--- src/CMakeLists.txt	2010-01-09 17:30:13.000000000 +0100
+++ src/CMakeLists.txt.old	2010-02-14 19:48:49.000000000 +0100
@@ -46,7 +46,7 @@
 
 kde4_add_executable(partitionmanager-bin ${partitionmanager-bin_SRCS})
 
-target_link_libraries(partitionmanager-bin partitionmanagerprivate)
+target_link_libraries(partitionmanager-bin partitionmanagerprivate ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS})
 
 if(UNIX)
 	add_custom_target(partitionmanager ALL
