diff -up at-spi2-core-0.1.8/registryd/Makefile.am.libs at-spi2-core-0.1.8/registryd/Makefile.am
--- at-spi2-core-0.1.8/registryd/Makefile.am.libs	2010-03-30 22:55:32.756091945 -0400
+++ at-spi2-core-0.1.8/registryd/Makefile.am	2010-03-30 22:56:05.214101095 -0400
@@ -12,7 +12,8 @@ at_spi2_registryd_LDADD = $(DBUS_GLIB_LI
 			  $(X_LIBS)		\
 			  $(SM_LIBS)		\
 			  $(XTST_LIBS)		\
-			  $(XEVIE_LIBS)
+			  $(XEVIE_LIBS)		\
+			  -ldl
 
 at_spi2_registryd_SOURCES =	\
 	de-marshaller.h         \
