diff -up x3270-3.3.10ga4/c3270-3.3/c3270.man.ibmhostspath x3270-3.3.10ga4/c3270-3.3/c3270.man
--- x3270-3.3.10ga4/c3270-3.3/c3270.man.ibmhostspath	2009-12-07 13:06:22.000000000 +0100
+++ x3270-3.3.10ga4/c3270-3.3/c3270.man	2009-12-07 13:09:17.000000000 +0100
@@ -3204,7 +3204,7 @@ backslash character.
 If more than one \fB\-xrm\fP option is given for the same resource,
 the last one on the command line is used.
 .SH "FILES"
-/usr/local/lib/x3270/ibm_hosts
+/etc/ibm_hosts
 .br
 $HOME/.c3270pro
 
diff -up x3270-3.3.10ga4/c3270-3.3/glue.c.ibmhostspath x3270-3.3.10ga4/c3270-3.3/glue.c
diff -up x3270-3.3.10ga4/c3270-3.3/host.c.ibmhostspath x3270-3.3.10ga4/c3270-3.3/host.c
--- x3270-3.3.10ga4/c3270-3.3/host.c.ibmhostspath	2009-12-07 13:13:03.000000000 +0100
+++ x3270-3.3.10ga4/c3270-3.3/host.c	2009-12-07 13:14:35.000000000 +0100
@@ -126,7 +126,7 @@ hostfile_init(void)
 	hostfile_initted = True;
 	hostfile_name = appres.hostsfile;
 	if (hostfile_name == CN)
-		hostfile_name = xs_buffer("%s/ibm_hosts", appres.conf_dir);
+		hostfile_name = xs_buffer("%s/ibm_hosts", "/etc");
 	else
 		hostfile_name = do_subst(appres.hostsfile, True, True);
 	hf = fopen(hostfile_name, "r");
diff -up x3270-3.3.10ga4/c3270-3.3/html/c3270-man.html.ibmhostspath x3270-3.3.10ga4/c3270-3.3/html/c3270-man.html
diff -up x3270-3.3.10ga4/c3270-3.3/html/Resources.html.ibmhostspath x3270-3.3.10ga4/c3270-3.3/html/Resources.html
diff -up x3270-3.3.10ga4/c3270-3.3/Makefile.ibmhostspath x3270-3.3.10ga4/c3270-3.3/Makefile
diff -up x3270-3.3.10ga4/c3270-3.3/Makefile.in.ibmhostspath x3270-3.3.10ga4/c3270-3.3/Makefile.in
--- x3270-3.3.10ga4/c3270-3.3/Makefile.in.ibmhostspath	2009-12-07 13:06:43.000000000 +0100
+++ x3270-3.3.10ga4/c3270-3.3/Makefile.in	2009-12-07 13:12:43.000000000 +0100
@@ -84,10 +84,10 @@ install:: c3270 x3270if
 	$(INSTALL_PROGRAM) c3270 $(DESTDIR)$(BINDIR)/c3270
 	$(INSTALL_PROGRAM) x3270if $(DESTDIR)$(BINDIR)/x3270if
 install::
-	[ -d $(DESTDIR)$(LIBX3270DIR) ] || \
-		mkdir -p $(DESTDIR)$(LIBX3270DIR)
-	[ -r $(DESTDIR)$(LIBX3270DIR)/ibm_hosts ] || \
-		$(INSTALL_DATA) ibm_hosts $(DESTDIR)$(LIBX3270DIR)/ibm_hosts
+	[ -d $(DESTDIR)/etc ] || \
+		mkdir -p $(DESTDIR)/etc
+	[ -r $(DESTDIR)/etc/ibm_hosts ] || \
+		$(INSTALL_DATA) ibm_hosts $(DESTDIR)/etc/ibm_hosts
 @PR@install::
 
 install.man:
diff -up x3270-3.3.10ga4/s3270-3.3/host.c.ibmhostspath x3270-3.3.10ga4/s3270-3.3/host.c
--- x3270-3.3.10ga4/s3270-3.3/host.c.ibmhostspath	2009-12-07 13:17:07.000000000 +0100
+++ x3270-3.3.10ga4/s3270-3.3/host.c	2009-12-07 13:18:19.000000000 +0100
@@ -126,7 +126,7 @@ hostfile_init(void)
 	hostfile_initted = True;
 	hostfile_name = appres.hostsfile;
 	if (hostfile_name == CN)
-		hostfile_name = xs_buffer("%s/ibm_hosts", appres.conf_dir);
+		hostfile_name = xs_buffer("%s/ibm_hosts", "/etc");
 	else
 		hostfile_name = do_subst(appres.hostsfile, True, True);
 	hf = fopen(hostfile_name, "r");
diff -up x3270-3.3.10ga4/s3270-3.3/html/Resources.html.ibmhostspath x3270-3.3.10ga4/s3270-3.3/html/Resources.html
diff -up x3270-3.3.10ga4/s3270-3.3/html/s3270-man.html.ibmhostspath x3270-3.3.10ga4/s3270-3.3/html/s3270-man.html
diff -up x3270-3.3.10ga4/s3270-3.3/s3270.man.ibmhostspath x3270-3.3.10ga4/s3270-3.3/s3270.man
--- x3270-3.3.10ga4/s3270-3.3/s3270.man.ibmhostspath	2009-12-07 13:16:49.000000000 +0100
+++ x3270-3.3.10ga4/s3270-3.3/s3270.man	2009-12-07 13:18:34.000000000 +0100
@@ -2015,7 +2015,7 @@ the \fBtraceDir\fP resource.
 If more than one \fB\-xrm\fP option is given for the same resource,
 the last one on the command line is used.
 .SH "FILES"
-/usr/local/lib/x3270/ibm_hosts
+/etc/ibm_hosts
 .br
 
 .SH "SEE ALSO"
diff -up x3270-3.3.10ga4/s3270-3.3/X3270.xad.ibmhostspath x3270-3.3.10ga4/s3270-3.3/X3270.xad
--- x3270-3.3.10ga4/s3270-3.3/X3270.xad.ibmhostspath	2009-12-07 13:17:00.000000000 +0100
+++ x3270-3.3.10ga4/s3270-3.3/X3270.xad	2009-12-07 13:18:45.000000000 +0100
@@ -221,7 +221,7 @@ x3270.charsetList:			U.S. English (CP 03
 ! x3270.doConfirms:			true
 ! x3270.debugTracing:			true
 ! x3270.disconnectClear:		false
-! x3270.hostsFile:			/usr/lib/X11/x3270/ibm_hosts
+! x3270.hostsFile:			/etc/ibm_hosts
 ! x3270.highlightSelect:		true
 ! x3270.idleCommand:
 ! x3270.idleTimeout:			~7m
diff -up x3270-3.3.10ga4/x3270-3.3/html/ibm_hosts.html.ibmhostspath x3270-3.3.10ga4/x3270-3.3/html/ibm_hosts.html
diff -up x3270-3.3.10ga4/x3270-3.3/html/Resources.html.ibmhostspath x3270-3.3.10ga4/x3270-3.3/html/Resources.html
diff -up x3270-3.3.10ga4/x3270-3.3/html/x3270-man.html.ibmhostspath x3270-3.3.10ga4/x3270-3.3/html/x3270-man.html
diff -up x3270-3.3.10ga4/x3270-3.3/ibm_hosts.man.ibmhostspath x3270-3.3.10ga4/x3270-3.3/ibm_hosts.man
--- x3270-3.3.10ga4/x3270-3.3/ibm_hosts.man.ibmhostspath	2009-12-07 12:59:42.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/ibm_hosts.man	2009-12-07 13:01:03.000000000 +0100
@@ -3,7 +3,7 @@
 .SH "NAME"
 ibm_hosts \- host database for x3270 and c3270
 .SH "SYNOPSIS"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .SH "DESCRIPTION"
 The \fBibm_hosts\fP
 file contains information regarding IBM hosts on the network.
@@ -82,7 +82,7 @@ The name \fBvm\fP will cause the 3270 em
 enter the string `3' on the second data-entry field on the screen, and
 send the Enter \s-1AID\s+1 sequence.
 .SH "FILES"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .SH "SEE ALSO"
 x3270(1),
 c3270(1)
diff -up x3270-3.3.10ga4/x3270-3.3/Imakefile.in.ibmhostspath x3270-3.3.10ga4/x3270-3.3/Imakefile.in
--- x3270-3.3.10ga4/x3270-3.3/Imakefile.in.ibmhostspath	2009-12-07 12:57:19.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/Imakefile.in	2009-12-07 13:03:46.000000000 +0100
@@ -190,8 +190,8 @@ $(DESTDIR)$(FONTINSTDIR)/fonts.dir::
 /* Extra install rule for the sample ibm_hosts file (runs only if there is
    no ibm_hosts file installed yet). */
 install:: $(HOSTSFILE)
-	@[ -r $(DESTDIR)$(LIBX3270DIR)/$(HOSTSFILE) ] || \
-		$(INSTALL) -c $(INSTDATFLAGS) $(HOSTSFILE) $(DESTDIR)$(LIBX3270DIR)/$(HOSTSFILE)
+	@[ -r $(DESTDIR)/etc/$(HOSTSFILE) ] || \
+		$(INSTALL) -c $(INSTDATFLAGS) $(HOSTSFILE) $(DESTDIR)/etc/$(HOSTSFILE)
 
 /*
  * Install rules for x3270if, x3270-script and ibm_hosts man pages.
@@ -220,8 +220,8 @@ install.byprefix::
 	$(MKFONTDIR) $(DESTDIR)$(autoconf_fontinstdir)
 	chmod u=rwx,go=rx $(DESTDIR)$(autoconf_fontinstdir)
 	chmod a=r $(DESTDIR)$(autoconf_fontinstdir)/fonts.dir
-	@[ -r $(DESTDIR)$(LIBX3270DIR)/$(HOSTSFILE) ] || \
-		(set -x; $(INSTALL) -c $(INSTDATFLAGS) $(HOSTSFILE) $(DESTDIR)$(LIBX3270DIR)/$(HOSTSFILE))
+	@[ -r $(DESTDIR)/etc/$(HOSTSFILE) ] || \
+		(set -x; $(INSTALL) -c $(INSTDATFLAGS) $(HOSTSFILE) $(DESTDIR)/etc/$(HOSTSFILE))
 
 /******************************************************************************
  * Clean rules.                                                               *
diff -up x3270-3.3.10ga4/x3270-3.3/Makefile.in.ibmhostspath x3270-3.3.10ga4/x3270-3.3/Makefile.in
--- x3270-3.3.10ga4/x3270-3.3/Makefile.in.ibmhostspath	2009-12-07 12:59:22.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/Makefile.in	2009-12-07 13:16:01.000000000 +0100
@@ -127,10 +127,10 @@ install:: x3270 x3270if
 	$(INSTALL_PROGRAM) x3270 $(DESTDIR)$(BINDIR)/x3270
 	$(INSTALL_PROGRAM) x3270if $(DESTDIR)$(BINDIR)/x3270if
 install::
-	[ -d $(DESTDIR)$(LIBX3270DIR) ] || \
-		mkdir -p $(DESTDIR)$(LIBX3270DIR)
-	[ -r $(DESTDIR)$(LIBX3270DIR)/ibm_hosts ] || \
-		$(INSTALL_DATA) ibm_hosts $(DESTDIR)$(LIBX3270DIR)/ibm_hosts
+	[ -d $(DESTDIR)/etc ] || \
+		mkdir -p $(DESTDIR)/etc
+	[ -r $(DESTDIR)/etc/ibm_hosts ] || \
+		$(INSTALL_DATA) ibm_hosts $(DESTDIR)/etc/ibm_hosts
 install::
 	[ -d $(DESTDIR)$(CIFONTDIR) ] || \
 		mkdir -p $(DESTDIR)$(CIFONTDIR)
diff -up x3270-3.3.10ga4/x3270-3.3/X3270.ad.ibmhostspath x3270-3.3.10ga4/x3270-3.3/X3270.ad
--- x3270-3.3.10ga4/x3270-3.3/X3270.xad.ibmhostspath	2009-12-07 12:59:53.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/X3270.xad	2009-12-07 13:05:28.000000000 +0100
@@ -214,7 +214,7 @@ x3270.charsetList:			U.S. English (CP 03
 ! x3270.doConfirms:			true
 ! x3270.debugTracing:			true
 ! x3270.disconnectClear:		false
-! x3270.hostsFile:			/usr/lib/X11/x3270/ibm_hosts
+! x3270.hostsFile:			/etc/ibm_hosts
 ! x3270.highlightSelect:		true
 ! x3270.idleCommand:
 ! x3270.idleTimeout:			~7m
diff -up x3270-3.3.10ga4/x3270-3.3/x3270.man.ibmhostspath x3270-3.3.10ga4/x3270-3.3/x3270.man
--- x3270-3.3.10ga4/x3270-3.3/x3270.man.ibmhostspath	2009-12-07 12:59:05.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/x3270.man	2009-12-07 13:05:42.000000000 +0100
@@ -5008,7 +5008,7 @@ T}
 The special types \fBsocks4a\fP and \fBsocks5d\fP can also be used to force
 the proxy server to do the hostname resolution for the SOCKS protocol.
 .SH "FILES"
-/usr/lib/X11/x3270/ibm_hosts
+/etc/ibm_hosts
 .br
 $HOME/.x3270pro
 
