diff -up x3270-3.3.10ga4/x3270-3.3/Examples/peer_script.expect.expectpath x3270-3.3.10ga4/x3270-3.3/Examples/peer_script.expect
--- x3270-3.3.10ga4/x3270-3.3/Examples/peer_script.expect.expectpath	2009-12-04 13:56:39.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/Examples/peer_script.expect	2009-12-04 13:56:44.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/local/bin/expect
+#! /usr/bin/expect
 # TSO login script, which runs as a peer of x3270.
 # expect version
 
diff -up x3270-3.3.10ga4/x3270-3.3/Examples/cms_logon.bash.bashpath x3270-3.3.10ga4/x3270-3.3/Examples/cms_logon.bash
--- x3270-3.3.10ga4/x3270-3.3/Examples/cms_logon.bash.bashpath	2009-12-04 13:58:21.000000000 +0100
+++ x3270-3.3.10ga4/x3270-3.3/Examples/cms_logon.bash	2009-12-04 13:59:25.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/local/bin/bash
+#! /bin/bash
 
 # Copyright (c) 1995-2009, Paul Mattes.
 # All rights reserved.
