--- mariadb-10.1.8/scripts/CMakeLists.txt.systemd	2015-10-15 17:43:45.000000000 +0200
+++ mariadb-10.1.8/scripts/CMakeLists.txt	2015-10-21 17:39:41.179737915 +0200
@@ -323,6 +323,33 @@ ELSE()
       COMPONENT ${${file}_COMPONENT}
      )
   ENDFOREACH()
+
+  # files for systemd
+  SET(SYSTEMD_SCRIPTS
+    mysql.tmpfiles.d
+    mysql.service
+    mysql-prepare-db-dir
+    mysql-wait-ready
+    mysql-wait-stop
+    mysql-check-socket
+    mysql-check-upgrade
+    mysql-scripts-common
+    mysql_config_multilib
+    mysql.init
+    my.cnf
+  )
+  FOREACH(file ${SYSTEMD_SCRIPTS})
+    IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSEIF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.in)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.in
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSE()
+      MESSAGE(FATAL_ERROR "Can not find ${file}.sh or ${file}.in in "
+        "${CMAKE_CURRENT_SOURCE_DIR}" )
+    ENDIF()
+  ENDFOREACH()
 ENDIF()
 
 # Install libgcc as  mylibgcc.a
