--- mariadb-10.0.10/include/my_sys.h.p1	2014-03-30 19:56:37.000000000 +0200
+++ mariadb-10.0.10/include/my_sys.h	2014-04-07 15:30:20.627060157 +0200
@@ -209,13 +209,7 @@ extern void my_large_free(uchar *ptr);
 #define my_safe_afree(ptr, size, max_alloca_sz) my_afree(ptr)
 #endif /* HAVE_ALLOCA */
 
-#ifndef errno				/* did we already get it? */
-#ifdef HAVE_ERRNO_AS_DEFINE
-#include <errno.h>			/* errno is a define */
-#else
-extern int errno;			/* declare errno */
-#endif
-#endif					/* #ifndef errno */
+#include <errno.h>				/* errno is a define */
 extern char *home_dir;			/* Home directory for user */
 extern MYSQL_PLUGIN_IMPORT char  *mysql_data_home;
 extern const char *my_progname;		/* program-name (printed in errors) */
