diff -up mariadb-10.0.16/scripts/mysql_install_db.pl.in.createtestdb mariadb-10.0.16/scripts/mysql_install_db.pl.in
--- mariadb-10.0.16/scripts/mysql_install_db.pl.in.createtestdb	2015-03-03 23:02:33.287620715 +0100
+++ mariadb-10.0.16/scripts/mysql_install_db.pl.in	2015-03-03 23:02:56.664641960 +0100
@@ -398,10 +398,10 @@ if ( $opt->{'skip-name-resolve'} and $re
 }
 
 # ----------------------------------------------------------------------
-# Create database directories mysql & test
+# Create database directories mysql
 # ----------------------------------------------------------------------
 
-foreach my $dir ( $opt->{ldata}, "$opt->{ldata}/mysql", "$opt->{ldata}/test" )
+foreach my $dir ( $opt->{ldata}, "$opt->{ldata}/mysql" )
 {
   # FIXME not really the same as original "mkdir -p", but ok?
   mkdir($dir, 0700) unless -d $dir;
