diff -rup mariadb-10.1.8.orig/include/my_sys.h mariadb-10.1.8/include/my_sys.h
--- mariadb-10.1.8.orig/include/my_sys.h	2015-10-15 17:43:37.000000000 +0200
+++ mariadb-10.1.8/include/my_sys.h	2015-10-21 15:48:00.948590204 +0200
@@ -218,13 +218,7 @@ extern void my_large_free(uchar *ptr);
 #define my_safe_afree(ptr, size) my_afree(ptr)
 #endif /* HAVE_ALLOCA */
 
-#ifndef errno				/* did we already get it? */
-#ifdef HAVE_ERRNO_AS_DEFINE
-#include <errno.h>			/* errno is a define */
-#else
-extern int errno;			/* declare errno */
-#endif
-#endif					/* #ifndef errno */
+#include <errno.h>				/* errno is a define */
 extern char *home_dir;			/* Home directory for user */
 extern MYSQL_PLUGIN_IMPORT char  *mysql_data_home;
 extern const char *my_progname;		/* program-name (printed in errors) */
