diff -up mariadb-10.0.16/scripts/CMakeLists.txt.systemd mariadb-10.0.16/scripts/CMakeLists.txt
--- mariadb-10.0.16/scripts/CMakeLists.txt.systemd	2015-01-25 16:21:37.000000000 +0100
+++ mariadb-10.0.16/scripts/CMakeLists.txt	2015-02-03 10:53:05.261790495 +0100
@@ -367,6 +367,33 @@ ELSE()
       COMPONENT ${${file}_COMPONENT}
      )
   ENDFOREACH()
+
+  # files for systemd
+  SET(SYSTEMD_SCRIPTS
+    mysql.tmpfiles.d
+    mysql.service
+    mysql-prepare-db-dir
+    mysql-wait-ready
+    mysql-wait-stop
+    mysql-check-socket
+    mysql-check-upgrade
+    mysql-scripts-common
+    mysql_config_multilib
+    mysql.init
+    my.cnf
+  )
+  FOREACH(file ${SYSTEMD_SCRIPTS})
+    IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSEIF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.in)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.in
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSE()
+      MESSAGE(FATAL_ERROR "Can not find ${file}.sh or ${file}.in in "
+        "${CMAKE_CURRENT_SOURCE_DIR}" )
+    ENDIF()
+  ENDFOREACH()
 ENDIF()
 
 # Install libgcc as  mylibgcc.a
diff -up mariadb-10.0.15/support-files/CMakeLists.txt.cmakescripts mariadb-10.0.15/support-files/CMakeLists.txt
--- mariadb-10.0.15/support-files/CMakeLists.txt.cmakescripts	2014-11-27 15:07:13.203821563 +0100
+++ mariadb-10.0.15/support-files/CMakeLists.txt	2014-11-27 15:04:26.252530666 +0100
@@ -112,6 +112,8 @@ IF(UNIX)
               COMPONENT SharedLibraries)
       INSTALL(FILES rpm/mysql-clients.cnf DESTINATION ${INSTALL_SYSCONF2DIR}
               COMPONENT Client)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/rpm/server.cnf 
+                     ${CMAKE_CURRENT_BINARY_DIR}/rpm/server.cnf @ONLY )
       INSTALL(FILES rpm/server.cnf DESTINATION ${INSTALL_SYSCONF2DIR}
               COMPONENT IniFiles)
     ENDIF()
