diff -up mariadb-10.0.12/scripts/CMakeLists.txt.systemd mariadb-10.0.12/scripts/CMakeLists.txt
--- mariadb-10.0.12/scripts/CMakeLists.txt.systemd	2014-07-21 10:49:58.491470586 +0200
+++ mariadb-10.0.12/scripts/CMakeLists.txt	2014-07-21 14:21:22.673329708 +0200
@@ -368,6 +368,33 @@ ELSE()
       COMPONENT ${${file}_COMPONENT}
      )
   ENDFOREACH()
+
+  # files for systemd
+  SET(SYSTEMD_SCRIPTS
+    mysql.tmpfiles.d
+    mysql.service
+    mysql-prepare-db-dir
+    mysql-wait-ready
+    mysql-check-socket
+    mysql-scripts-common
+    mysql_config_multilib
+    mysql.init
+    mysql-compat.service
+    mysql-compat.conf
+    my.cnf
+  )
+  FOREACH(file ${SYSTEMD_SCRIPTS})
+    IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSEIF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.in)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.in
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSE()
+      MESSAGE(FATAL_ERROR "Can not find ${file}.sh or ${file}.in in "
+        "${CMAKE_CURRENT_SOURCE_DIR}" )
+    ENDIF()
+  ENDFOREACH()
 ENDIF()
 
 # Install libgcc as  mylibgcc.a
diff -up mariadb-10.0.12/support-files/CMakeLists.txt.cmakescripts mariadb-10.0.12/support-files/CMakeLists.txt
--- mariadb-10.0.12/support-files/CMakeLists.txt.cmakescripts	2014-07-24 22:31:38.116002679 +0200
+++ mariadb-10.0.12/support-files/CMakeLists.txt	2014-07-24 22:35:00.532233645 +0200
@@ -110,6 +110,8 @@ IF(UNIX)
               COMPONENT SharedLibraries)
       INSTALL(FILES rpm/mysql-clients.cnf DESTINATION ${INSTALL_SYSCONF2DIR}
               COMPONENT Client)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/rpm/server.cnf 
+                     ${CMAKE_CURRENT_BINARY_DIR}/rpm/server.cnf @ONLY )
       INSTALL(FILES rpm/server.cnf DESTINATION ${INSTALL_SYSCONF2DIR}
               COMPONENT IniFiles)
     ENDIF()
