diff --git a/libs/service/secretstorage.cpp b/libs/service/secretstorage.cpp
index fabb1f9..52c2188 100644
--- a/libs/service/secretstorage.cpp
+++ b/libs/service/secretstorage.cpp
@@ -19,6 +19,8 @@ License along with this library.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include "secretstorage.h"
+#include "../internals/settings/802-11-wireless-security.h"
+#include "../internals/settings/802-1x.h"
 
 #include <KConfigGroup>
 #include <kwallet.h>
@@ -153,6 +155,18 @@ void SecretStorage::walletOpenedForRead(bool success)
                     if (i.next().key() != con->uuid())
                         continue;
 
+                    Knm::WirelessSecuritySetting * settingSecurity = static_cast<Knm::WirelessSecuritySetting *>(con->setting(Knm::Setting::WirelessSecurity));
+                    if (settingSecurity) {
+                        Knm::Security8021xSetting * setting8021x = static_cast<Knm::Security8021xSetting *>(con->setting(Knm::Setting::Security8021x));
+
+                        if (setting8021x) {
+                            if (settingSecurity->securityType() == Knm::WirelessSecuritySetting::EnumSecurityType::DynamicWep ||
+                                settingSecurity->securityType() == Knm::WirelessSecuritySetting::EnumSecurityType::WpaEap) {
+                                setting8021x->setEnabled(true); // needed for needSecrets() below, otherwise needSecrets() returns an empty list.
+                            }
+                        }
+                    }
+
                     QPair<QString,GetSecretsFlags> pair = i.value();
                     bool settingsFound = false;
                     foreach (Knm::Setting * setting, con->settings()) {
