--- configure.ac.orig	2014-02-21 19:24:32.803750944 -0300
+++ configure.ac	2014-02-21 19:25:04.342124560 -0300
@@ -533,7 +533,7 @@ if
 	AC_MSG_ERROR([No client interfaces configured!])
 fi
 
-MPD_AUTO_PKG(systemd_daemon, SYSTEMD_DAEMON, libsystemd-daemon,
+MPD_AUTO_PKG(systemd_daemon, SYSTEMD_DAEMON, libsystemd,
 	[systemd activation], [libsystemd-daemon not found])
 AM_CONDITIONAL(ENABLE_SYSTEMD_DAEMON, test x$enable_systemd_daemon = xyes)
 if test x$enable_systemd_daemon = xyes; then
