--- src/decoder/plugins/FfmpegDecoderPlugin.cxx.orig	2014-10-19 22:39:58.996638204 -0200
+++ src/decoder/plugins/FfmpegDecoderPlugin.cxx	2014-10-19 22:44:48.744993325 -0200
@@ -422,12 +422,14 @@ ffmpeg_probe(Decoder *decoder, InputStre
 	avpd.buf_size = nbytes;
 	avpd.filename = is.GetURI();
 
+#if LIBAVFORMAT_VERSION_INT <= AV_VERSION_INT(56, 9, 99)
 #ifdef AVPROBE_SCORE_MIME
 	/* this attribute was added in libav/ffmpeg version 11, but
 	   unfortunately it's "uint8_t" instead of "char", and it's
 	   not "const" - wtf? */
 	avpd.mime_type = (uint8_t *)const_cast<char *>(is.GetMimeType());
 #endif
+#endif
 
 	return av_probe_input_format(&avpd, true);
 }
