diff -up phonon-4.7.0/PhononConfig.cmake.in.rootDir phonon-4.7.0/PhononConfig.cmake.in
--- phonon-4.7.0/PhononConfig.cmake.in.rootDir	2013-11-11 10:15:33.835839769 -0600
+++ phonon-4.7.0/PhononConfig.cmake.in	2013-11-11 10:16:55.063365497 -0600
@@ -4,7 +4,7 @@
 set(PHONON_VERSION "@PHONON_LIB_MAJOR_VERSION@.@PHONON_LIB_MINOR_VERSION@.@PHONON_LIB_PATCH_VERSION@")
 
 get_filename_component(currentDir ${CMAKE_CURRENT_LIST_FILE} PATH) # get the directory where I myself am
-get_filename_component(rootDir ${currentDir}/@relInstallDir@ ABSOLUTE) # get the chosen install prefix
+get_filename_component(rootDir @CMAKE_INSTALL_PREFIX@ ABSOLUTE) # get the chosen install prefix
 
 set(PHONON_NO_GRAPHICSVIEW @PHONON_NO_GRAPHICSVIEW@)
 set(PHONON_PULSESUPPORT @PHONON_PULSESUPPORT@)
