From 6b596e197111aa47d9d4054b0a5b4e3c66e48b56 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Mon, 8 Aug 2011 16:54:35 +0200
Subject: [PATCH] pulse: Spell a11y propery (those are ones dammit)

---
 phonon/pulsesupport.cpp |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/phonon/pulsesupport.cpp b/phonon/pulsesupport.cpp
index e998c6f..4fdad55 100644
--- a/phonon/pulsesupport.cpp
+++ b/phonon/pulsesupport.cpp
@@ -194,7 +194,7 @@ static Phonon::Category pulseRoleToPhononCategory(const char *role, bool *succes
         return Phonon::NotificationCategory;
     if (r == "phone")
         return Phonon::CommunicationCategory;
-    if (r == "ally")
+    if (r == "a11y")
         return Phonon::AccessibilityCategory;
 
     // ^^ "animation" and "production" have no mapping
@@ -215,7 +215,7 @@ static Phonon::CaptureCategory pulseRoleToPhononCaptureCategory(const char *role
         return Phonon::CommunicationCaptureCategory;
     if (r == "production")
         return Phonon::RecordingCaptureCategory;
-    if (r == "ally")
+    if (r == "a11y")
         return Phonon::ControlCaptureCategory;
 
     *success = false;
@@ -238,7 +238,7 @@ static const QByteArray phononCategoryToPulseRole(Phonon::Category category)
     case Phonon::CommunicationCategory:
         return QByteArray("phone");
     case Phonon::AccessibilityCategory:
-        return QByteArray("ally");
+        return QByteArray("a11y");
     default:
         return QByteArray();
     }
@@ -254,7 +254,7 @@ static const QByteArray phononCaptureCategoryToPulseRole(Phonon::CaptureCategory
     case Phonon::RecordingCaptureCategory:
         return QByteArray("production");
     case Phonon::ControlCaptureCategory:
-        return QByteArray("ally");
+        return QByteArray("a11y");
     default:
         return QByteArray();
     }
-- 
1.7.6

