diff -up phonon-4.5.0/phonon/addoninterface.h.subtitles phonon-4.5.0/phonon/addoninterface.h
--- phonon-4.5.0/phonon/addoninterface.h.subtitles	2011-03-25 10:55:56.000000000 -0300
+++ phonon-4.5.0/phonon/addoninterface.h	2011-03-25 20:01:26.185363014 -0300
@@ -90,8 +90,11 @@ class AddonInterface
             availableSubtitles, /**< \returns \c int representing the amount of
                                       available subtitles on the media source */
             currentSubtitle,    /**< \returns \c int representing the current subtitle */
-            setCurrentSubtitle  /**< Sets the current subtitle to the first
+            setCurrentSubtitle,  /**< Sets the current subtitle to the first
                                      \c int in the QList */
+            setAutoDetectSubtitles,
+            setFileSubtitle,
+            setSubtitleEncoding
         };
 
         enum AudioChannelCommand {
diff -up phonon-4.5.0/phonon/mediacontroller.cpp.subtitles phonon-4.5.0/phonon/mediacontroller.cpp
--- phonon-4.5.0/phonon/mediacontroller.cpp.subtitles	2011-03-25 10:55:56.000000000 -0300
+++ phonon-4.5.0/phonon/mediacontroller.cpp	2011-03-25 20:00:38.751454271 -0300
@@ -276,6 +276,27 @@ void MediaController::setCurrentSubtitle
         AddonInterface::setCurrentSubtitle, QList<QVariant>() << qVariantFromValue(stream));
 }
 
+void MediaController::setAutoDetectSubtitles(bool state)
+{
+    IFACE;
+    iface->interfaceCall(AddonInterface::SubtitleInterface,
+        AddonInterface::setAutoDetectSubtitles, QList<QVariant>() << QVariant(state));
+}
+
+void MediaController::setFileSubtitle (const MediaSource &subtitle)
+{
+    IFACE;
+    iface->interfaceCall(AddonInterface::SubtitleInterface,
+        AddonInterface::setFileSubtitle, QList<QVariant>() << qVariantFromValue(subtitle));
+}
+
+void MediaController::setSubtitleEncoding(const QString &encoding)
+{
+    IFACE;
+    iface->interfaceCall(AddonInterface::SubtitleInterface,
+        AddonInterface::setSubtitleEncoding, QList<QVariant>() <<  QVariant(encoding));
+}
+
 #undef IFACE
 
 } // namespace Phonon
diff -up phonon-4.5.0/phonon/mediacontroller.h.subtitles phonon-4.5.0/phonon/mediacontroller.h
--- phonon-4.5.0/phonon/mediacontroller.h.subtitles	2011-03-25 10:55:56.000000000 -0300
+++ phonon-4.5.0/phonon/mediacontroller.h	2011-03-25 20:00:38.751454271 -0300
@@ -39,6 +39,7 @@ namespace Phonon
 {
 class MediaControllerPrivate;
 class MediaObject;
+class MediaSource;
 
 /** \class MediaController mediacontroller.h phonon/MediaController
  * \brief Controls optional features of a media file/device like title, chapter, angle.
@@ -189,6 +190,41 @@ class PHONON_EXPORT MediaController : pu
          */
         void setCurrentSubtitle(const Phonon::SubtitleDescription &stream);
 
+	/**
+	 * Enable auto-detection for files subtitles
+	 *
+	 * If one subtitle exists in the same directory,
+	 * matching the same filename, it will be selected as default.
+	 *
+	 * \see setFileSubtitle()
+	 */
+        void setAutoDetectSubtitles(bool autodetect);
+
+	/**
+	 * Selects a subtitle file for the media
+	 *
+	 * Choose the following subtitle to be read with the media.
+	 *
+	 * \param subtitle MediaSource for the subtitle to load
+	 *
+	 * \see setSubtitleEncoding()
+	 */
+        void setFileSubtitle(const MediaSource &subtitle);
+
+        /**
+         * Change the encoding for the current subtitle.
+         *
+         * Choose the following encoding to be used when a subtitle
+         * is set (auto-detected or manually).
+         * This is especially useful when a subtitle using a specific charset is loaded.
+         * As default the encoding is "iso-8859-1", because a lot of subs still use this charset.
+         *
+         * \param encoding name of the encoding to be used
+         *
+         * \see setSubtitleFont()
+         */
+        void setSubtitleEncoding(const QString &encoding);
+
     public Q_SLOTS:
         void setCurrentAngle(int angleNumber);
         void setCurrentChapter(int chapterNumber);
diff -up phonon-4.5.0/phonon/mediasource.h.subtitles phonon-4.5.0/phonon/mediasource.h
--- phonon-4.5.0/phonon/mediasource.h.subtitles	2011-03-25 10:55:56.000000000 -0300
+++ phonon-4.5.0/phonon/mediasource.h	2011-03-25 20:00:38.752454058 -0300
@@ -31,6 +31,7 @@
 
 #include <QtCore/QSharedData>
 #include <QtCore/QString>
+#include <QtCore/QMetaType>
 
 QT_BEGIN_HEADER
 QT_BEGIN_NAMESPACE
@@ -333,6 +334,7 @@ class PHONON_EXPORT MediaSource
 } // namespace Phonon
 
 QT_END_NAMESPACE
+Q_DECLARE_METATYPE(Phonon::MediaSource);
 QT_END_HEADER
 
 #endif // PHONON_MEDIASOURCE_H
