diff -ur phonon-4.7.0-fix-rpath-handling/cmake/FindPhononInternal.cmake phonon-4.7.0-really-fix-rpath-handling/cmake/FindPhononInternal.cmake
--- phonon-4.7.0-fix-rpath-handling/cmake/FindPhononInternal.cmake	2013-11-10 23:25:47.000000000 +0100
+++ phonon-4.7.0-really-fix-rpath-handling/cmake/FindPhononInternal.cmake	2013-11-10 23:28:10.000000000 +0100
@@ -216,7 +216,6 @@
 # These two options below make it set the RPATH of the installed targets to all
 # RPATH directories outside the current CMAKE_BINARY_DIR and also the library 
 # install directory. Alex
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
 set(_abs_LIB_INSTALL_DIR "${LIB_INSTALL_DIR}")
 if (NOT IS_ABSOLUTE "${_abs_LIB_INSTALL_DIR}")
    set(_abs_LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}")
