From 4ab9f6763bf3647d3238e75527dabb9a3433a4e9 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Mon, 8 Aug 2011 18:22:05 +0200
Subject: [PATCH] globalconfig: Properly show pulseaudio device orders for
 capture.

---
 phonon/globalconfig.cpp |   24 ++++++++++++++----------
 1 files changed, 14 insertions(+), 10 deletions(-)

diff --git a/phonon/globalconfig.cpp b/phonon/globalconfig.cpp
index d0d256d..f47c8f8 100644
--- a/phonon/globalconfig.cpp
+++ b/phonon/globalconfig.cpp
@@ -173,19 +173,23 @@ static QList<int> sortDevicesByCategoryPriority(const GlobalConfig *config, cons
     }
 
     QList<int> deviceList;
-
-    QString categoryKey = QLatin1String("Category_") + QString::number(static_cast<int>(category));
-    if (!backendConfig->hasKey(categoryKey)) {
-        // no list in config for the given category
-        categoryKey = QLatin1String("Category_") + QString::number(static_cast<int>(NoCategory));
+    PulseSupport *pulse = PulseSupport::getInstance();
+    if (pulse->isActive()) {
+        deviceList = pulse->objectIndexesByCategory(type, category);
+    } else {
+        QString categoryKey = QLatin1String("Category_") + QString::number(static_cast<int>(category));
         if (!backendConfig->hasKey(categoryKey)) {
-            // no list in config for NoCategory
-            return defaultList;
+            // no list in config for the given category
+            categoryKey = QLatin1String("Category_") + QString::number(static_cast<int>(NoCategory));
+            if (!backendConfig->hasKey(categoryKey)) {
+                // no list in config for NoCategory
+                return defaultList;
+            }
         }
-    }
 
-    //Now the list from d->config
-    deviceList = backendConfig->value(categoryKey, QList<int>());
+        //Now the list from d->config
+        deviceList = backendConfig->value(categoryKey, QList<int>());
+    }
 
     //if there are devices in d->config that the backend doesn't report, remove them from the list
     QMutableListIterator<int> i(deviceList);
-- 
1.7.6

