--- phonon-4.5.0/phonon/pulsestream.cpp~	2011-08-04 16:36:29.406729335 +0100
+++ phonon-4.5.0/phonon/pulsestream.cpp	2011-08-04 16:36:29.442728719 +0100
@@ -21,6 +21,7 @@
 */
 
 #include "pulsestream_p.h"
+#include <QtCore/qmath.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -81,7 +82,7 @@
         // AudioOutput expects the "backend" to supply values that have been
         // adjusted for Stephens' law, so we need to fudge them accordingly
         // so that the %ages match up in KMix/the application's own slider.
-        emit volumeChanged(pow(vol, VOLTAGE_TO_LOUDNESS_EXPONENT));
+        emit volumeChanged(qPow(vol, VOLTAGE_TO_LOUDNESS_EXPONENT));
     }
 }
 
