diff -Naur phonon-4.3.50.orig/phonon/mediaobject.cpp phonon-4.3.50/phonon/mediaobject.cpp
--- phonon-4.3.50.orig/phonon/mediaobject.cpp   2009-02-26 22:33:54.000000000 +0100
+++ phonon-4.3.50/phonon/mediaobject.cpp    2009-10-12 22:48:22.000000000 +0200
@@ -114,7 +114,7 @@
 void MediaObject::stop()
 {
     K_D(MediaObject);
-    if (d->backendObject() && isPlayable(d->mediaSource.type())) {
+    if (d->backendObject() && d->mediaSource.type() != MediaSource::Invalid) {
         INTERFACE_CALL(stop());
     }
 }

