diff -up phonon-4.4.3/phonon/addoninterface.h.subtitles phonon-4.4.3/phonon/addoninterface.h
--- phonon-4.4.3/phonon/addoninterface.h.subtitles	2010-11-24 16:26:19.000000000 -0200
+++ phonon-4.4.3/phonon/addoninterface.h	2010-11-24 21:37:50.771105989 -0200
@@ -77,7 +77,10 @@ class AddonInterface
         enum SubtitleCommand {
             availableSubtitles,
             currentSubtitle,
-            setCurrentSubtitle
+            setCurrentSubtitle,
+            setAutoDetectSubtitles,
+            setFileSubtitle,
+            setSubtitleEncoding
         };
         enum AudioChannelCommand {
             availableAudioChannels,
diff -up phonon-4.4.3/phonon/mediacontroller.cpp.subtitles phonon-4.4.3/phonon/mediacontroller.cpp
--- phonon-4.4.3/phonon/mediacontroller.cpp.subtitles	2010-11-24 16:26:19.000000000 -0200
+++ phonon-4.4.3/phonon/mediacontroller.cpp	2010-11-24 21:37:50.771105989 -0200
@@ -226,6 +226,27 @@ void MediaController::setCurrentSubtitle
         AddonInterface::setCurrentSubtitle, QList<QVariant>() << qVariantFromValue(stream));
 }
 
+void MediaController::setAutoDetectSubtitles(bool state)
+{
+    IFACE;
+    iface->interfaceCall(AddonInterface::SubtitleInterface,
+        AddonInterface::setAutoDetectSubtitles, QList<QVariant>() << QVariant(state));
+}
+
+void MediaController::setFileSubtitle (const MediaSource &subtitle)
+{
+    IFACE;
+    iface->interfaceCall(AddonInterface::SubtitleInterface,
+        AddonInterface::setFileSubtitle, QList<QVariant>() << qVariantFromValue(subtitle));
+}
+
+void MediaController::setSubtitleEncoding(const QString &encoding)
+{
+    IFACE;
+    iface->interfaceCall(AddonInterface::SubtitleInterface,
+        AddonInterface::setSubtitleEncoding, QList<QVariant>() <<  QVariant(encoding));
+}
+
 #undef IFACE
 
 } // namespace Phonon
diff -up phonon-4.4.3/phonon/mediacontroller.h.subtitles phonon-4.4.3/phonon/mediacontroller.h
--- phonon-4.4.3/phonon/mediacontroller.h.subtitles	2010-11-24 16:26:19.000000000 -0200
+++ phonon-4.4.3/phonon/mediacontroller.h	2010-11-24 21:37:50.771105989 -0200
@@ -38,6 +38,7 @@ namespace Phonon
 {
 class MediaControllerPrivate;
 class MediaObject;
+class MediaSource;
 
 /** \class MediaController mediacontroller.h phonon/MediaController
  * \brief Controls optional features of a media file/device like title, chapter, angle.
@@ -133,6 +134,41 @@ class PHONON_EXPORT MediaController : pu
          */
         void setCurrentSubtitle(const Phonon::SubtitleDescription &stream);
 
+	/**
+	 * Enable auto-detection for files subtitles
+	 *
+	 * If one subtitle exists in the same directory,
+	 * matching the same filename, it will be selected as default.
+	 *
+	 * \see setFileSubtitle()
+	 */
+        void setAutoDetectSubtitles(bool autodetect);
+
+	/**
+	 * Selects a subtitle file for the media
+	 *
+	 * Choose the following subtitle to be read with the media.
+	 *
+	 * \param subtitle MediaSource for the subtitle to load
+	 *
+	 * \see setSubtitleEncoding()
+	 */
+        void setFileSubtitle(const MediaSource &subtitle);
+
+        /**
+         * Change the encoding for the current subtitle.
+         *
+         * Choose the following encoding to be used when a subtitle
+         * is set (auto-detected or manually).
+         * This is especially useful when a subtitle using a specific charset is loaded.
+         * As default the encoding is "iso-8859-1", because a lot of subs still use this charset.
+         *
+         * \param encoding name of the encoding to be used
+         *
+         * \see setSubtitleFont()
+         */
+        void setSubtitleEncoding(const QString &encoding);
+
     public Q_SLOTS:
         void setCurrentAngle(int angleNumber);
         void setCurrentChapter(int chapterNumber);
diff -up phonon-4.4.3/phonon/mediasource.h.subtitles phonon-4.4.3/phonon/mediasource.h
--- phonon-4.4.3/phonon/mediasource.h.subtitles	2010-11-24 16:26:19.000000000 -0200
+++ phonon-4.4.3/phonon/mediasource.h	2010-11-24 21:37:50.772105783 -0200
@@ -28,6 +28,7 @@
 #include "objectdescription.h"
 #include <QtCore/QSharedData>
 #include <QtCore/QString>
+#include <QtCore/QMetaType>
 
 QT_BEGIN_HEADER
 QT_BEGIN_NAMESPACE
@@ -310,6 +311,7 @@ class PHONON_EXPORT MediaSource
 } // namespace Phonon
 
 QT_END_NAMESPACE
+Q_DECLARE_METATYPE(Phonon::MediaSource);
 QT_END_HEADER
 
 #endif // PHONON_MEDIASOURCE_H
