--- compizconfig-python-0.9.2.1/src/compizconfig.pyx	2010-11-07 04:55:00.000000000 +0000
+++ compizconfig-python-0.9.2.1/src/compizconfig.pyx.new	2011-01-15 23:29:20.819139836 +0000
@@ -570,7 +570,7 @@
     cdef object extendedStrRestrictions
     cdef object baseStrRestrictions
 
-    def __new__ (self, Plugin plugin, name):
+    def __init__ (self, Plugin plugin, name):
         cdef CCSSettingType t
         cdef CCSSettingInfo * i
 
@@ -674,7 +674,7 @@
 cdef class SSGroup:
     cdef object screen
 
-    def __new__ (self, screen):
+    def __init__ (self, screen):
         self.screen = screen
 
     property Screen:
@@ -692,7 +692,7 @@
     cdef object ranking
     cdef object hasExtendedString
     
-    def __new__ (self, Context context, name):
+    def __init__ (self, Context context, name):
         self.ccsPlugin = ccsFindPlugin (context.ccsContext, name)
         self.context = context
         self.screen = {}
@@ -1007,7 +1007,7 @@
     cdef Context context
     cdef char * name
 
-    def __new__ (self, Context context, name):
+    def __init__ (self, Context context, name):
         self.context = context
         self.name = strdup (name)
 
@@ -1029,7 +1029,7 @@
     cdef Bool profileSupport
     cdef Bool integrationSupport
 
-    def __new__ (self, Context context, info):
+    def __init__ (self, Context context, info):
         self.context = context
         self.name = strdup (info[0])
         self.shortDesc = strdup (info[1])
@@ -1072,7 +1072,7 @@
     cdef object currentBackend
     cdef Bool integration
 
-    def __new__ (self, screenNum = 0, plugins = [], basic_metadata = False):
+    def __init__ (self, screenNum = 0, plugins = [], basic_metadata = False):
         cdef CCSPlugin * pl
         cdef CCSList * pll
         if basic_metadata:
