diff -up gnome-games-2.31.92.1/configure.in.no-lsb gnome-games-2.31.92.1/configure.in
--- gnome-games-2.31.92.1/configure.in.no-lsb	2010-09-16 15:52:43.000000000 -0400
+++ gnome-games-2.31.92.1/configure.in	2010-09-24 08:15:17.509500718 -0400
@@ -229,14 +229,8 @@ GNOME_MAINTAINER_MODE_DEFINES
 # Distribution info
 # *****************
 
-LSB_DISTRIBUTOR=
-LSB_DISTRIBUTION=
-
-AC_PATH_PROG([LSB_RELEASE],[lsb_release],[false])
-if test "$LSB_RELEASE" != "false"; then
-  LSB_DISTRIBUTOR="$($LSB_RELEASE -s -i | sed 's,^\",,;s,\"$,,')"
-  LSB_DISTRIBUTION="$($LSB_RELEASE -s -d | sed 's,^\",,;s,\"$,,')"
-fi
+LSB_DISTRIBUTOR="_SLKDISTR_"
+LSB_DISTRIBUTION="_SLKDIST_"
 
 AC_MSG_CHECKING([for LSB distributor])
 AC_MSG_RESULT([$LSB_DISTRIBUTOR])
