diff --git a/version.h.in b/version.h.in
index cf429f2..da0ad15 100644
--- a/version.h.in
+++ b/version.h.in
@@ -45,12 +45,6 @@ extern "C" {
 
 #define NETTLE_USE_MINI_GMP @NETTLE_USE_MINI_GMP@
 
-/* We need a preprocessor constant for GMP_NUMB_BITS, simply using
-   sizeof(mp_limb_t) * CHAR_BIT is not good enough. */
-#if NETTLE_USE_MINI_GMP
-# define GMP_NUMB_BITS @GMP_NUMB_BITS@
-#endif
-
 int
 nettle_version_major (void);
 
