diff -up chrony-1.31/examples/chronyd.service.service-helper chrony-1.31/examples/chronyd.service
--- chrony-1.31/examples/chronyd.service.service-helper	2014-09-10 17:00:54.000000000 +0200
+++ chrony-1.31/examples/chronyd.service	2014-07-03 18:09:45.000000000 +0200
@@ -8,6 +8,7 @@ Type=forking
 PIDFile=/var/run/chronyd.pid
 EnvironmentFile=-/etc/sysconfig/chronyd
 ExecStart=/usr/sbin/chronyd $OPTIONS
+ExecStartPost=/usr/libexec/chrony-helper update-daemon
 
 [Install]
 WantedBy=multi-user.target
