From 122c6556917bf154c3d78613802046fa63a74523 Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Thu, 26 Jul 2012 15:13:50 +0200
Subject: [PATCH 4/4] warning--

---
 gstreamer/mediaobject.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gstreamer/mediaobject.cpp b/gstreamer/mediaobject.cpp
index 8ba8574..170c34b 100644
--- a/gstreamer/mediaobject.cpp
+++ b/gstreamer/mediaobject.cpp
@@ -77,8 +77,8 @@ MediaObject::MediaObject(Backend *backend, QObject *parent)
         , m_waitingForNextSource(false)
         , m_waitingForPreviousSource(false)
         , m_skippingEOS(false)
-        , m_skipGapless(false)
         , m_doingEOS(false)
+        , m_skipGapless(false)
         , m_handlingAboutToFinish(false)
 {
     qRegisterMetaType<GstCaps*>("GstCaps*");
-- 
1.7.11.2

