From 2db4c430740da89fb22319b2ded63e770f3d6fac Mon Sep 17 00:00:00 2001
From: Trever Fischer <tdfischer@fedoraproject.org>
Date: Sat, 22 Sep 2012 19:41:14 +0200
Subject: [PATCH 59/63] Use a sync connection to update the xvimagesink window
 handle

BUG: 305333
---
 gstreamer/videowidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gstreamer/videowidget.cpp b/gstreamer/videowidget.cpp
index 96eea1c..3b38ab0 100644
--- a/gstreamer/videowidget.cpp
+++ b/gstreamer/videowidget.cpp
@@ -86,7 +86,7 @@ void VideoWidget::updateWindowID()
 void VideoWidget::finalizeLink()
 {
     connect(root()->pipeline(), SIGNAL(mouseOverActive(bool)), this, SLOT(mouseOverActive(bool)));
-    connect(root()->pipeline(), SIGNAL(windowIDNeeded()), this, SLOT(updateWindowID()));
+    connect(root()->pipeline(), SIGNAL(windowIDNeeded()), this, SLOT(updateWindowID()), Qt::DirectConnection);
 }
 
 void VideoWidget::prepareToUnlink()
-- 
1.8.0

