From: Piotr FILICIAK <piotr@filiciak.pl>
Subject: Report proper BD-R capacity (for LG BH10 and maybe others)
Forwarded: yes
Bug: http://bugs.kde.org/show_bug.cgi?id=243549
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=587996
Origin: other, http://bugsfiles.kde.org/attachment.cgi?id=51896
Last-Update: 2010-12-29

--- a/libk3bdevice/k3bdevice_mmc.cpp
+++ b/libk3bdevice/k3bdevice_mmc.cpp
@@ -322,7 +322,7 @@ bool K3b::Device::Device::readTrackInfor
     // DVD-DL: 48 (MMC5)
     // CD:     36 (MMC2)
     //
-    if( dataLen <= 4 ) {
+    if( dataLen <= 6 ) {
         int m = mediaType();
         if( m & (MEDIA_DVD_R_DL|MEDIA_DVD_R_DL_SEQ|MEDIA_DVD_R_DL_JUMP) )
             dataLen = 48;
