--- k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.ffmpeg	2013-03-17 01:05:40.157149232 -0300
+++ k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2013-03-17 01:10:05.069108135 -0300
@@ -42,6 +42,9 @@ extern "C" {
 #define FFMPEG_BUILD_PRE_4629
 #endif
 
+#if LIBAVCODEC_VERSION_MAJOR >= 55
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
+#endif
 
 K3bFFMpegWrapper* K3bFFMpegWrapper::s_instance = 0;
 
@@ -88,14 +91,18 @@ bool K3bFFMpegFile::open()
     close();
 
     // open the file
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53,2,0)
+    int err = ::avformat_open_input( &d->formatContext, m_filename.toLocal8Bit(), 0, 0 );
+#else
     int err = ::av_open_input_file( &d->formatContext, m_filename.toLocal8Bit(), 0, 0, 0 );
+#endif
     if( err < 0 ) {
         kDebug() << "(K3bFFMpegFile) unable to open " << m_filename << " with error " << err;
         return false;
     }
 
     // analyze the streams
-    ::av_find_stream_info( d->formatContext );
+    ::avformat_find_stream_info( d->formatContext, NULL );
 
     // we only handle files containing one audio stream
     if( d->formatContext->nb_streams != 1 ) {
@@ -129,7 +136,7 @@ bool K3bFFMpegFile::open()
 
     // open the codec on our context
     kDebug() << "(K3bFFMpegFile) found codec for " << m_filename;
-    if( ::avcodec_open( codecContext, d->codec ) < 0 ) {
+    if( ::avcodec_open2( codecContext, d->codec, NULL ) < 0 ) {
         kDebug() << "(K3bFFMpegDecoderFactory) could not open codec.";
         return false;
     }
@@ -143,7 +150,11 @@ bool K3bFFMpegFile::open()
     }
 
     // dump some debugging info
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(53,2,0)
+    ::av_dump_format( d->formatContext, 0, m_filename.toLocal8Bit(), 0 );
+#else
     ::dump_format( d->formatContext, 0, m_filename.toLocal8Bit(), 0 );
+#endif
 
     return true;
 }
@@ -165,7 +176,7 @@ void K3bFFMpegFile::close()
     }
 
     if( d->formatContext ) {
-        ::av_close_input_file( d->formatContext );
+        ::avformat_close_input( &d->formatContext );
         d->formatContext = 0;
     }
 }
@@ -324,10 +335,14 @@ int K3bFFMpegFile::fillOutputBuffer()
 #if LIBAVCODEC_VERSION_MAJOR < 52
         int len = ::avcodec_decode_audio(
 #else
-   #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(52, 64, 0)
-        int len = ::avcodec_decode_audio3(
+   #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55, 0, 0)
+        int len = ::avcodec_decode_audio4(
    #else
-        int len = ::avcodec_decode_audio2(
+      #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(52, 64, 0)
+              int len = ::avcodec_decode_audio3(
+      #else
+              int len = ::avcodec_decode_audio2(
+      #endif
    #endif
 #endif
 
@@ -336,7 +351,11 @@ int K3bFFMpegFile::fillOutputBuffer()
 #else
             d->formatContext->streams[0]->codec,
 #endif
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55, 0, 0)
+            (AVFrame*)d->alignedOutputBuffer,
+#else
             (short*)d->alignedOutputBuffer,
+#endif
             &d->outputBufferSize,
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(52, 64, 0)
             &d->packet );
