--- plugins/decoder/libsndfile/CMakeLists.txt.orig	2010-10-08 23:37:01.406668863 -0300
+++ plugins/decoder/libsndfile/CMakeLists.txt	2010-10-08 23:37:50.572861068 -0300
@@ -4,6 +4,10 @@
 
 kde4_add_plugin(k3blibsndfiledecoder ${k3blibsndfiledecoder_PART_SRCS})
 
+include_directories(
+  ${SNDFILE_INCLUDE_DIR}
+)
+
 target_link_libraries(k3blibsndfiledecoder k3bdevice k3b ${KDE4_KDECORE_LIBS} ${KDE4_KDEUI_LIBS} ${SNDFILE_LIBRARIES} )
 
 install(TARGETS k3blibsndfiledecoder DESTINATION ${PLUGIN_INSTALL_DIR} )
