--- mp3mesa.desktop.orig	2008-07-27 19:48:10.000000000 -0300
+++ mp3mesa.desktop	2010-05-21 20:20:46.509853264 -0300
@@ -1,23 +1,11 @@
 [Desktop Entry]
 Name=MP3 Mesa
-Name[ca]=MP3 Mesa
-Name[es]=MP3 Mesa
-Name[de]=MP3 Mesa
-Name[fi]=MP3 Mesa
-Name[zh_CN]=MP3 Mesa å<88>¶ä½<9C>ç¨<8B>åº<8F>
-Name[zh_CN.GB2312]=MP3 Mesa å<88>¶ä½<9C>ç¨<8B>åº<8F>
-Name[zh_TW]=MP3 Mesa è£½ä½<9C>ç¨<8B>å¼<8F>
-Name[zh_TW.Big5]=MP3 Mesa è£½ä½<9C>ç¨<8B>å¼<8F>
 Comment=CD player/ripper
 Comment[ca]=Reproductor/extractor de CD d'audio
 Comment[es]=Reproductor/extractor de CDs de audio
 Comment[de]=Audio-CDs rippen und MP3s/OGGs kodieren
 Comment[fr]=Lecteur/extracteur de CD audio
 Comment[fi]=CD-audion soitto- ja kopiointiohjelma
-Comment[zh_CN]=MP3 Mesa è®©æ<82>¨è½»æ<9D>¾å<88>¶ä½<9C> MP3 æ<96><87>ä»¶
-Comment[zh_CN.GB2312]=MP3 Mesa è®©æ<82>¨è½»æ<9D>¾å<88>¶ä½<9C> MP3 æ<96><87>ä»¶
-Comment[zh_TW]=MP3 Mesa è®<93>æ<82>¨è¼<95>é¬<86>è£½ä½<9C> MP3 æª<94>
-Comment[zh_TW.Big5]=MP3 Mesa è®<93>æ<82>¨è¼<95>é¬<86>è£½ä½<9C> MP3 æª<94>
 Exec=mp3mesa
 Icon=mp3mesaicon.png
 Type=Application
