--- linphone-3.6.1/coreapi/linphonecore.c	2013-06-12 17:20:14.020039579 +0300
+++ linphone-3.6.1/coreapi/linphonecore.c	2013-07-26 11:36:51.014342101 +0300
@@ -638,7 +638,7 @@
 	} 
 
 #ifdef __linux
-	sal_set_root_ca(lc->sal, lp_config_get_string(lc->config,"sip","root_ca", "/etc/ssl/certs"));
+	sal_set_root_ca(lc->sal, lp_config_get_string(lc->config,"sip","root_ca", "/etc/ssl/certs/ca-bundle.crt"));
 #else
 	sal_set_root_ca(lc->sal, lp_config_get_string(lc->config,"sip","root_ca", ROOT_CA_FILE));
 #endif
