--- linphone-3.9.0/coreapi/linphonecore.c.orig	2015-11-05 00:02:26.388863101 -0200
+++ linphone-3.9.0/coreapi/linphonecore.c	2015-11-05 00:04:42.423244503 -0200
@@ -924,10 +924,10 @@ static void certificates_config_read(Lin
 	const char *rootca;
 #ifdef __linux
 	struct stat sb;
-	rootca=lp_config_get_string(lc->config,"sip","root_ca", "/etc/ssl/certs");
-	if (stat("/etc/ssl/certs", &sb) != 0 || !S_ISDIR(sb.st_mode))
+	rootca=lp_config_get_string(lc->config,"sip","root_ca", "/etc/ssl/certs/ca-bundle.crt");
+	if (stat("/etc/ssl/certs/ca-bundle.crt", &sb) != 0 || !S_ISDIR(sb.st_mode))
 	{
-		ms_warning("/etc/ssl/certs not found, using %s instead", ROOT_CA_FILE);
+		ms_warning("/etc/ssl/certs/ca-bundle.crt not found, using %s instead", ROOT_CA_FILE);
 		rootca=lp_config_get_string(lc->config,"sip","root_ca", ROOT_CA_FILE);
 	}
 #else
