--- linphone-3.5.0/mediastreamer2/tests/mediastream.c	2011-12-20 10:35:18.000000000 +0200
+++ linphone-3.5.0/mediastreamer2/tests/mediastream.c	2011-12-26 04:55:01.468278706 +0200
@@ -430,8 +430,8 @@
 
 
 void setup_media_streams(MediastreamDatas* args) {
-	/*create the rtp session */
-	MSWebCam *cam=NULL;
+	/*create the rtp session
+	MSWebCam *cam=NULL; */
 
 	ortp_init();
 	if (args->is_verbose) {
