--- linphone-3.5.1/mediastreamer2/tests/mediastream.c	2012-01-18 13:00:15.000000000 +0200
+++ linphone-3.5.1/mediastreamer2/tests/mediastream.c	2012-02-20 16:45:05.961678898 +0200
@@ -400,7 +400,9 @@
 
 void setup_media_streams(MediastreamDatas* args) {
 	/*create the rtp session */
+#ifdef VIDEO_ENABLED
 	MSWebCam *cam=NULL;
+#endif
 
 	ortp_init();
 	if (args->is_verbose) {
--- linphone-3.5.1/mediastreamer2/src/nowebcam.c	2012-01-18 13:00:15.000000000 +0200
+++ linphone-3.5.1/mediastreamer2/src/nowebcam.c	2012-02-19 19:24:12.643525630 +0200
@@ -1875,10 +1875,11 @@
 }
 
 int static_image_set_vsize(MSFilter *f, void* data){
-	SIData *d=(SIData*)f->data;
 #ifndef NO_FFMPEG
+	SIData *d=(SIData*)f->data;
 	d->vsize=*(MSVideoSize*)data;
 #else
+	__attribute__((__unused__)) SIData *d=(SIData*)f->data;
     // no rescaling without ffmpeg
 #endif
 	return 0;
