--- linphone-3.8.0/coreapi/linphonecore.c.orig	2015-03-15 21:07:39.315978383 -0300
+++ linphone-3.8.0/coreapi/linphonecore.c	2015-03-15 21:09:17.103166813 -0300
@@ -827,7 +827,7 @@ static void certificates_config_read(Lin
 {
 	const char *rootca;
 #ifdef __linux
-	rootca=lp_config_get_string(lc->config,"sip","root_ca", "/etc/ssl/certs");
+	rootca=lp_config_get_string(lc->config,"sip","root_ca", "/etc/ssl/certs/ca-bundle.crt");
 #else
 	rootca=lp_config_get_string(lc->config,"sip","root_ca", ROOT_CA_FILE);
 #endif
