From 0e1ca304dd2ebecbc3f91d16d9e71947afd85b3f Mon Sep 17 00:00:00 2001
From: Ralph Bean <rbean@redhat.com>
Date: Wed, 17 Sep 2014 14:34:12 -0400
Subject: [PATCH] Namespace.

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index dc82951..09b30eb 100644
--- a/setup.py
+++ b/setup.py
@@ -16,6 +16,7 @@ setup(
 
     package_dir = {'': 'src'},
     packages = find_packages('src'),
+    namespace_packages=['backports'],
     include_package_data = True,
     install_requires = [],
     entry_points = '',
-- 
1.9.3

