diff -up kphotoalbum-4.6.1/ImageManager/VideoThumbnails.cpp.gcc kphotoalbum-4.6.1/ImageManager/VideoThumbnails.cpp
--- kphotoalbum-4.6.1/ImageManager/VideoThumbnails.cpp.gcc	2015-03-17 23:28:37.000000000 -0500
+++ kphotoalbum-4.6.1/ImageManager/VideoThumbnails.cpp	2015-04-15 21:51:28.671045728 -0500
@@ -44,7 +44,7 @@ void ImageManager::VideoThumbnails::setV
         return;
 
     cancelPreviousJobs();
-    m_pendingRequest = nullptr;
+    m_pendingRequest = false;
     for ( int i= 0; i < 10; ++i )
         m_cache[i] = QImage();
 
