--- src/config_pepperflash.c.orig	2014-07-08 23:03:37.894653449 -0300
+++ src/config_pepperflash.c	2014-07-08 23:04:00.754785504 -0300
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 
 const char *pepperflash_path_list[] = {
+    "/usr/lib/pepperflash",                     // pepperflash-plugin (Chinforinfula)
     "/opt/google/chrome/PepperFlash",           // Chrome
     "/opt/google/chrome-beta/PepperFlash",      // Chrome beta
     "/opt/google/chrome-unstable/PepperFlash",  // Chrome unstable
