diff -up plplot-5.11.1/examples/c/Makefile.examples.in.multiarch plplot-5.11.1/examples/c/Makefile.examples.in
--- plplot-5.11.1/examples/c/Makefile.examples.in.multiarch	2015-08-12 15:07:59.505179310 -0600
+++ plplot-5.11.1/examples/c/Makefile.examples.in	2015-08-12 15:08:22.439080205 -0600
@@ -25,9 +25,6 @@ SHELL = @SH_EXECUTABLE@
 CC = @CC@
 EXEEXT = @EXEEXT@
 
-PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-RPATHCMD = @RPATHCMD@
-
 @extXdrawable_true@extXdrawable_EXECUTABLES_list = \
 @extXdrawable_true@	extXdrawable_demo$(EXEEXT)
 
diff -up plplot-5.11.1/examples/c++/Makefile.examples.in.multiarch plplot-5.11.1/examples/c++/Makefile.examples.in
--- plplot-5.11.1/examples/c++/Makefile.examples.in.multiarch	2015-08-12 15:07:59.505179310 -0600
+++ plplot-5.11.1/examples/c++/Makefile.examples.in	2015-08-12 15:08:37.205016396 -0600
@@ -25,10 +25,7 @@ SHELL = @SH_EXECUTABLE@
 CXX = @CXX@
 EXEEXT = @EXEEXT@
 
-PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-RPATHCMD = @RPATHCMD@
 @qt_gui_true@QT_MOC_EXECUTABLE = @QT_MOC_EXECUTABLE@
-@qt_gui_true@qt_RPATHCMD = @qt_RPATHCMD@
 
 @wxwidgets_true@PLPLOTWXWIDGETS_EXECUTABLES_list = \
 @wxwidgets_true@	@wxdemo_name@$(EXEEXT)
diff -up plplot-5.11.1/examples/f95/Makefile.examples.in.multiarch plplot-5.11.1/examples/f95/Makefile.examples.in
--- plplot-5.11.1/examples/f95/Makefile.examples.in.multiarch	2015-08-12 11:35:28.000000000 -0600
+++ plplot-5.11.1/examples/f95/Makefile.examples.in	2015-08-12 15:07:59.506179306 -0600
@@ -32,9 +32,6 @@ SHELL = @SH_EXECUTABLE@
 F95 = @FC@
 EXEEXT = @EXEEXT@
 
-PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-RPATHCMD = @RPATHCMD@
-
 EXECUTABLES_list = \
 		 x00f$(EXEEXT) \
 		 x01f$(EXEEXT) \
diff -up plplot-5.11.1/examples/tk/Makefile.examples.in.multiarch plplot-5.11.1/examples/tk/Makefile.examples.in
--- plplot-5.11.1/examples/tk/Makefile.examples.in.multiarch	2015-08-12 11:35:28.000000000 -0600
+++ plplot-5.11.1/examples/tk/Makefile.examples.in	2015-08-12 15:07:59.506179306 -0600
@@ -24,9 +24,6 @@ SHELL = @SH_EXECUTABLE@
 CC = @CC@
 EXEEXT = @EXEEXT@
 
-PKG_CONFIG_ENV = @PKG_CONFIG_ENV@
-plplottcltk_Main_RPATHCMD = @plplottcltk_Main_RPATHCMD@
-
 EXECUTABLES_list = xtk01$(EXEEXT)
 # Second and fourth examples depend on itk.
 @itk_true@itk_EXECUTABLES_list = xtk02$(EXEEXT) xtk04$(EXEEXT)
