--- src/modules/avformat/producer_avformat.c.ffmpeg	2013-03-15 18:33:00.025674597 -0300
+++ src/modules/avformat/producer_avformat.c	2013-03-15 18:33:34.067524976 -0300
@@ -77,6 +77,10 @@ const char *avcodec_get_sample_fmt_name(
 #include <libavcodec/opt.h>
 #endif
 
+#if LIBAVCODEC_VERSION_MAJOR >= 55
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
+#endif
+
 #define POSITION_INITIAL (-2)
 #define POSITION_INVALID (-1)
 
