From 5428142140f578a0a73f288fdf821ace9fea6a60 Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Mon, 8 Feb 2010 12:51:47 +1000
Subject: [PATCH 2/2] set canDoBGNoneRoot

---
 src/nv_driver.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/nv_driver.c b/src/nv_driver.c
index 248a151..b5bc0a7 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -673,6 +673,7 @@ NVPreInit(ScrnInfoPtr pScrn, int flags)
 
 	/* Set pScrn->monitor */
 	pScrn->monitor = pScrn->confScreen->monitor;
+	pScrn->canDoBGNoneRoot = 1;
 
 	/*
 	 * The first thing we should figure out is the depth, bpp, etc.
-- 
1.6.6.1

