diff -Nur webkit-1.3.10.orig/Source/JavaScriptCore/wtf/Platform.h webkit-1.3.10/Source/JavaScriptCore/wtf/Platform.h
--- webkit-1.3.10.orig/Source/JavaScriptCore/wtf/Platform.h	2011-01-10 18:07:11.000000000 -0700
+++ webkit-1.3.10/Source/JavaScriptCore/wtf/Platform.h	2011-01-10 20:53:01.422468389 -0700
@@ -1036,7 +1036,7 @@
 #endif
 /* Setting this flag prevents the assembler from using RWX memory; this may improve
    security but currectly comes at a significant performance cost. */
-#if PLATFORM(IOS)
+#if PLATFORM(IOS) || PLATFORM(X86) || PLATFORM(X86_64)
 #define ENABLE_ASSEMBLER_WX_EXCLUSIVE 1
 #else
 #define ENABLE_ASSEMBLER_WX_EXCLUSIVE 0
