--- Source/WebKit/gtk/tests/testatk.c.orig	2011-11-26 18:45:12.987514283 -0200
+++ Source/WebKit/gtk/tests/testatk.c	2011-11-26 18:45:15.290014607 -0200
@@ -1686,7 +1686,7 @@ static void initializeTestingFramework(i
     g_test_init(&argc, &argv, 0);
     gtk_disable_setlocale();
     setlocale(LC_ALL, "C");
-    gdk_disable_multidevice();
+    //gdk_disable_multidevice();
     gtk_init(&argc, &argv);
 }
 
