diff -ur krusader-2.3.0-beta1-xz/krusader/Konfigurator/kgprotocols.cpp krusader-2.3.0-beta1-default-mimetypes/krusader/Konfigurator/kgprotocols.cpp
--- krusader-2.3.0-beta1-xz/krusader/Konfigurator/kgprotocols.cpp	2011-01-30 20:11:26.000000000 +0100
+++ krusader-2.3.0-beta1-default-mimetypes/krusader/Konfigurator/kgprotocols.cpp	2011-01-30 20:14:15.000000000 +0100
@@ -45,11 +45,12 @@
 QString KgProtocols::defaultKrarcMimes = "application/x-7z,application/x-7z-compressed,"
         "application/x-ace,application/x-ace-compressed,"
         "application/x-arj,application/x-arj-compressed,"
-        "application/x-bzip2,"
+        "application/x-bzip,"
         "application/x-cpio,application/x-deb,"
         "application/x-debian-package,"
         "application/x-gzip,application/x-jar,"
         "application/x-lha,application/x-lha-compressed,"
+        "application/x-lzma,"
         "application/x-rar,application/x-rar-compressed,"
         "application/x-xz,"
         "application/x-rpm,application/zip,"
@@ -57,6 +58,7 @@
 QString KgProtocols::defaultTarMimes   = "application/x-tar,application/x-tarz,"
         "application/x-bzip-compressed-tar,"
         "application/x-compressed-tar,"
+        "application/x-lzma-compressed-tar,"
         "application/x-tbz,application/x-tgz,"
         "application/x-xz-compressed-tar";
 
