diff -ur krusader-2.2.0-beta1/krusader/Panel/krinterviewitemdelegate.cpp krusader-2.2.0-beta1-rename-selects-ext-regression/krusader/Panel/krinterviewitemdelegate.cpp
--- krusader-2.2.0-beta1/krusader/Panel/krinterviewitemdelegate.cpp	2010-04-30 18:41:02.000000000 +0200
+++ krusader-2.2.0-beta1-rename-selects-ext-regression/krusader/Panel/krinterviewitemdelegate.cpp	2010-06-04 00:43:32.000000000 +0200
@@ -24,6 +24,8 @@
 #include <QtGui/QPainter>
 #include <QtGui/QLineEdit>
 
+#include "../krglobal.h"
+
 KrInterViewItemDelegate::KrInterViewItemDelegate(QObject *parent) :
         QItemDelegate(parent), _currentlyEdited(-1) {}
 
@@ -53,9 +55,14 @@
     if (editor->inherits("QLineEdit")) {
         QLineEdit *lineEdit = qobject_cast<QLineEdit *> (editor);
         if (lineEdit) {
-            QString nameWithoutExt = index.data(Qt::UserRole).toString();
-            lineEdit->deselect();
-            lineEdit->setSelection(0, nameWithoutExt.length());
+            KConfigGroup gl(krConfig, "Look&Feel");
+            if (gl.readEntry("Rename Selects Extension", true))
+                lineEdit->selectAll();
+            else {
+                QString nameWithoutExt = index.data(Qt::UserRole).toString();
+                lineEdit->deselect();
+                lineEdit->setSelection(0, nameWithoutExt.length());
+            }
         }
     }
 }
