diff -ur krusader-2.2.0-beta1/krusader/KViewer/panelviewer.cpp krusader-2.2.0-beta1-readonly/krusader/KViewer/panelviewer.cpp
--- krusader-2.2.0-beta1/krusader/KViewer/panelviewer.cpp	2010-04-30 18:41:03.000000000 +0200
+++ krusader-2.2.0-beta1-readonly/krusader/KViewer/panelviewer.cpp	2010-06-03 23:57:13.000000000 +0200
@@ -172,6 +172,8 @@
 
     if (!cpart) cpart = getPart("text/plain");
     if (!cpart) cpart = getPart("all/allfiles");
+    if (cpart && cpart->inherits("KParts::ReadWritePart"))
+      static_cast<KParts::ReadWritePart *>(cpart.data())->setReadWrite(false);
 
     if (cpart) {
         addWidget(cpart->widget());
