diff -Nur krusader-2.2.0-beta1/krusader/CMakeLists.txt krusader-2.2.0-beta1-viewer-menu/krusader/CMakeLists.txt
--- krusader-2.2.0-beta1/krusader/CMakeLists.txt	2010-04-30 18:41:03.000000000 +0200
+++ krusader-2.2.0-beta1-viewer-menu/krusader/CMakeLists.txt	2010-06-03 23:23:06.000000000 +0200
@@ -94,6 +94,7 @@
         DESTINATION ${XDG_APPS_INSTALL_DIR} )
 install(FILES krusaderui.rc
               krusaderlisterui.rc
+              krviewer.rc
               midnight_commander.color
               total_commander.color
               total_commander.keymap
diff -Nur krusader-2.2.0-beta1/krusader/krviewer.rc krusader-2.2.0-beta1-viewer-menu/krusader/krviewer.rc
--- krusader-2.2.0-beta1/krusader/krviewer.rc	1970-01-01 01:00:00.000000000 +0100
+++ krusader-2.2.0-beta1-viewer-menu/krusader/krviewer.rc	2010-06-03 23:21:40.000000000 +0200
@@ -0,0 +1,5 @@
+<!DOCTYPE kpartgui >
+<kpartgui version="1" name="krviewer" >
+ <MenuBar>
+ </MenuBar>
+</kpartgui>
