diff -ur krusader-2.2.0-beta1/krusader/Panel/krmousehandler.cpp krusader-2.2.0-beta1-singleclick-regression/krusader/Panel/krmousehandler.cpp
--- krusader-2.2.0-beta1/krusader/Panel/krmousehandler.cpp	2010-04-30 18:41:02.000000000 +0200
+++ krusader-2.2.0-beta1-singleclick-regression/krusader/Panel/krmousehandler.cpp	2010-06-03 21:48:04.000000000 +0200
@@ -161,7 +161,7 @@
         _rightClickedItem = 0;
         _contextMenuTimer.stop();
     }
-    if (_singleClick && e->button() == Qt::LeftButton) {
+    if (_singleClick && e->button() == Qt::LeftButton && e->modifiers() == Qt::NoModifier) {
         CANCEL_TWO_CLICK_RENAME;
         e->accept();
         if (item == 0)
