diff -ur krusader-2.2.0-beta1/krusader/KViewer/panelviewer.cpp krusader-2.2.0-beta1-force-reload/krusader/KViewer/panelviewer.cpp
--- krusader-2.2.0-beta1/krusader/KViewer/panelviewer.cpp	2010-04-30 18:41:03.000000000 +0200
+++ krusader-2.2.0-beta1-force-reload/krusader/KViewer/panelviewer.cpp	2010-06-03 23:09:52.000000000 +0200
@@ -177,7 +177,9 @@
         addWidget(cpart->widget());
         setCurrentWidget(cpart->widget());
     }
-    if (cpart && cpart->openUrl(curl)) {
+    KParts::OpenUrlArguments args;
+    args.setReload(true);
+    if (cpart && (cpart->setArguments(args), cpart->openUrl(curl))) {
         curl = url;
         connect(cpart, SIGNAL(destroyed()), this, SLOT(slotCPartDestroyed()));
         return cpart;
@@ -294,6 +296,9 @@
             return cpart;
         }
     } else {
+        KParts::OpenUrlArguments args;
+        args.setReload(true);
+        cpart->setArguments(args);
         if (cpart->openUrl(curl)) {
             connect(cpart, SIGNAL(destroyed()), this, SLOT(slotCPartDestroyed()));
             return cpart;
