diff -up ModemManager-0.6.0.0/src/77-mm-usb-device-blacklist.rules.arduino ModemManager-0.6.0.0/src/77-mm-usb-device-blacklist.rules
--- ModemManager-0.6.0.0/src/77-mm-usb-device-blacklist.rules.arduino	2013-01-31 07:57:12.321964272 -0500
+++ ModemManager-0.6.0.0/src/77-mm-usb-device-blacklist.rules	2013-01-31 07:59:02.546965926 -0500
@@ -68,5 +68,14 @@ ATTRS{idVendor}=="050d", ATTRS{idProduct
 # ATEN Intl UC-232A (Prolific)
 ATTRS{idVendor}=="0557", ATTRS{idProduct}=="2008", ENV{ID_MM_DEVICE_IGNORE}="1"
 
+# FT232 USB-Serial (UART) IC (Arduino)
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6001", ENV{ID_MM_DEVICE_IGNORE}="1"
+
+# Atmel Corp. LUFA USB to Serial Adapter Project (Arduino)
+ATTRS{idVendor}=="03eb", ATTRS{idProduct}=="204b", ENV{ID_MM_DEVICE_IGNORE}="1"
+
+# Anything with the Arduino Vendor ID
+ATTRS{idVendor}=="2341", ENV{ID_MM_DEVICE_IGNORE}="1"
+
 LABEL="mm_usb_device_blacklist_end"
 
