diff -up flac-1.3.1/configure.ac.cflags flac-1.3.1/configure.ac
--- flac-1.3.1/configure.ac.cflags	2014-11-27 03:45:33.598155763 +0100
+++ flac-1.3.1/configure.ac	2014-11-27 13:45:18.092749862 +0100
@@ -352,8 +352,7 @@ if test "x$debug" = xtrue; then
 	CFLAGS="-g $CFLAGS"
 else
 	CPPFLAGS="-DNDEBUG $CPPFLAGS"
-	CFLAGS=$(echo "$CFLAGS" | sed 's/-O2//;s/-g//')
-	CFLAGS="-O3 -funroll-loops $CFLAGS"
+	CFLAGS="$user_cflags"
 fi
 
 XIPH_GCC_VERSION
@@ -363,7 +362,6 @@ if test x$ac_cv_c_compiler_gnu = xyes ;
 	CXXFLAGS="$CXXFLAGS -Wall -Wextra -Wcast-align -Wshadow -Wwrite-strings -Wctor-dtor-privacy -Wnon-virtual-dtor -Wreorder -Wsign-promo -Wundef -Wunreachable-code " # -Wcast-qual -Wbad-function-cast -Wwrite-strings -Woverloaded-virtual -Wmissing-declarations
 
 	XIPH_ADD_CFLAGS([-Wdeclaration-after-statement])
-	XIPH_ADD_CFLAGS([-D_FORTIFY_SOURCE=2])
 
 	AC_LANG_PUSH([C++])
 	XIPH_ADD_CXXFLAGS([-Weffc++])
@@ -385,10 +383,6 @@ if test x$ac_cv_c_compiler_gnu = xyes ;
 		XIPH_ADD_CFLAGS([-fgnu89-inline])
 		fi
 
-	if test "x$asm_optimisation$sse_os" = "xyesyes" ; then
-		XIPH_ADD_CFLAGS([-msse2])
-		fi
-
 	fi
 
 XIPH_ADD_CFLAGS([-Wextra])
