--- lxpanel-0.8.1/plugins/xkb/xkb-plugin.c.revert	2015-05-07 04:44:51.000000000 +0900
+++ lxpanel-0.8.1/plugins/xkb/xkb-plugin.c	2015-05-24 15:58:10.987343168 +0900
@@ -190,7 +190,7 @@
 
     /* Change to next or previous group. */
     xkb_change_group(xkb,
-        (((event->direction == GDK_SCROLL_UP) || (event->direction == GDK_SCROLL_RIGHT)) ? 1 : -1));
+        (((event->direction == GDK_SCROLL_DOWN) || (event->direction == GDK_SCROLL_RIGHT)) ? 1 : -1));
     return TRUE;
 }
 
