diff -up fltk-1.1.9/fltk-config.in.ldstaticflags fltk-1.1.9/fltk-config.in
--- fltk-1.1.9/fltk-config.in.ldstaticflags	2008-02-20 12:10:13.000000000 -0600
+++ fltk-1.1.9/fltk-config.in	2009-05-13 10:23:15.071926233 -0500
@@ -216,7 +216,7 @@ else
     includes=
 fi
 
-if test "$libdir" != /usr/lib -a "$libdir" != /usr/lib32; then
+if test "$libdir" != /usr/lib -a "$libdir" != /usr/lib32 -a "$libdir" != /usr/lib64; then
     libs=-L$libdir
 else
     libs=
