--- FL/x.H.orig	2014-11-09 23:08:46.883465646 -0200
+++ FL/x.H	2014-11-09 23:09:29.588418455 -0200
@@ -114,7 +114,7 @@ FL_EXPORT Fl_Region XRectangleRegion(int
 // With Xlib / X11 fonts, fl_xfont will return the current selected font.
 // With XFT / X11 fonts, fl_xfont will attempt to return the bitmap "core" font most
 // similar to (usually the same as) the current XFT font.
-class Fl_XFont_On_Demand
+class FL_EXPORT Fl_XFont_On_Demand
 {
 public:
   Fl_XFont_On_Demand(XFontStruct* p = NULL) : ptr(p) { }
