diff -up fltk-1.3.0rc5/src/filename_list.cxx.gcc44 fltk-1.3.0rc5/src/filename_list.cxx
--- fltk-1.3.0rc5/src/filename_list.cxx.gcc44	2011-01-05 17:50:10.000000000 +0100
+++ fltk-1.3.0rc5/src/filename_list.cxx	2011-05-25 16:26:16.842727901 +0200
@@ -119,7 +119,7 @@ int fl_filename_list(const char *d, dire
   // The vast majority of UNIX systems want the sort function to have this
   // prototype, most likely so that it can be passed to qsort without any
   // changes:
-  int n = scandir(dirloc, list, 0, (int(*)(const void*,const void*))sort);
+  int n = scandir(d, list, 0, (int(*)(const dirent **,const dirent **))sort);
 #endif
 
 #ifndef __APPLE__
