diff -up fltk-1.1.10/fluid/Makefile.fluid_target fltk-1.1.10/fluid/Makefile
--- fltk-1.1.10/fluid/Makefile.fluid_target	2008-12-29 14:21:29.000000000 -0600
+++ fltk-1.1.10/fluid/Makefile	2010-12-07 12:54:32.836737792 -0600
@@ -52,7 +52,7 @@ OBJECTS = $(CPPFILES:.cxx=.o)
 
 include ../makeinclude
 
-all:	$(FLUID) fluid$(EXEEXT)
+all:	$(FLUID)
 
 fluid$(EXEEXT):		$(OBJECTS) $(LIBNAME) $(FLLIBNAME) \
 			$(IMGLIBNAME)
