diff -up fltk-1.3.0rc5/src/Makefile.undefined fltk-1.3.0rc5/src/Makefile
--- fltk-1.3.0rc5/src/Makefile.undefined	2011-02-06 15:08:08.000000000 +0100
+++ fltk-1.3.0rc5/src/Makefile	2011-05-25 16:19:41.430727963 +0200
@@ -312,7 +312,7 @@ $(GLLIBNAME): $(GLOBJECTS)
 
 libfltk_gl.so.1.3: $(GLOBJECTS) libfltk.so.1.3
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(GLOBJECTS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(GLOBJECTS) -L. $(GLDLIBS) -lfltk
 	$(RM) libfltk_gl.so
 	$(LN) libfltk_gl.so.1.3 libfltk_gl.so
 
