--- compiz-plugins-extra-0.9.2.1/notification/src/notification.cpp	2010-11-07 04:58:15.000000000 +0000
+++ compiz-plugins-extra-0.9.2.1/notification/src/notification.cpp.new	2011-01-20 11:13:59.149668918 +0000
@@ -55,7 +55,11 @@
     logLevel = (char *) logLevelToString (level);
 
     n = notify_notification_new (logLevel, message,
-				 iconUri.c_str (), NULL);
+				 iconUri.c_str ()
+#if !defined(NOTIFY_VERSION_MINOR) || (NOTIFY_VERSION_MAJOR == 0 && NOTIFY_VERSION_MINOR < 7)
+                                            , NULL
+#endif
+                                            );
 
     timeout = optionGetTimeout ();
     if (timeout > 0)
