diff --git a/Makefile.in b/Makefile.in
index 93be806..70ada3b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -719,46 +719,46 @@ library_validators_include_HEADERS = $(h_validators_sources_public)
 
 @ENABLE_EXAMPLES_TRUE@examples_textreader_textreader_SOURCES = examples/textreader/main.cc
 @ENABLE_EXAMPLES_TRUE@examples_dom_build_dom_build_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_dom_parse_entities_dom_parse_entities_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_dom_parser_dom_parser_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_dom_parser_raw_dom_parser_raw_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_dom_read_write_dom_read_write_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_dom_xpath_dom_xpath_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_dtdvalidation_dtdvalidation_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_import_node_import_node_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_sax_exception_sax_exception_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_sax_parser_sax_parser_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_sax_parser_build_dom_sax_parser_build_dom_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_sax_parser_entities_sax_parser_entities_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_schemavalidation_schemavalidation_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_EXAMPLES_TRUE@examples_textreader_textreader_LDADD = \
-@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la
+@ENABLE_EXAMPLES_TRUE@	libxml++/libxml++-$(LIBXMLXX_API_VERSION).la $(LIBXMLXX_LIBS)
 
 @ENABLE_DOCUMENTATION_TRUE@book_name = $(LIBXMLXX_MODULE_NAME)
 @ENABLE_DOCUMENTATION_TRUE@doc_input = $(h_sources_public)
