diff --git a/src/fuser.c b/src/fuser.c
index 4688dfe..d19075a 100644
--- a/src/fuser.c
+++ b/src/fuser.c
@@ -1112,7 +1112,13 @@ print_matches(struct names *names_head, const opt_type opts,
 
 	for (nptr = names_head; nptr != NULL; nptr = nptr->next) {
 		if (opts & OPT_SILENT) {
-			have_match = nptr->matched_procs ? 1 : have_match;
+			for (pptr = nptr->matched_procs; pptr != NULL;
+			     pptr = pptr->next) {
+				if(pptr->proc_type != PTYPE_NORMAL)
+					continue;
+
+				have_match = 1;
+			}
 		} else {	/* We're not silent */
 			if ((opts & OPT_ALLFILES) == 0) {
 				name_has_procs = 0;
