diff -urNp coreutils-8.4-orig/src/Makefile.am coreutils-8.4/src/Makefile.am
--- coreutils-8.4-orig/src/Makefile.am	2010-09-03 17:34:43.399747649 +0200
+++ coreutils-8.4/src/Makefile.am	2010-09-03 17:36:13.005765125 +0200
@@ -367,6 +367,7 @@ factor_LDADD += $(LIB_GMP)
 
 # for crypt and pam
 su_LDADD += $(LIB_CRYPT) $(PAM_LIBS)
+su_LDFLAGS = -pie -Wl,-z,relro,-z,now
 
 # for various ACL functions
 copy_LDADD += $(LIB_ACL)
