diff -urNp coreutils-8.20-orig/src/longlong.h coreutils-8.20/src/longlong.h
--- coreutils-8.20-orig/src/longlong.h	2012-10-23 16:14:12.000000000 +0200
+++ coreutils-8.20/src/longlong.h	2012-12-06 17:09:12.865695463 +0100
@@ -1392,7 +1392,7 @@ extern UWtype __MPN(udiv_qrnnd) (UWtype 
 
 /* We should test _IBMR2 here when we add assembly support for the system
    vendor compilers.  */
-#if HAVE_HOST_CPU_FAMILY_powerpc && W_TYPE_SIZE == 64
+#if HAVE_HOST_CPU_FAMILY_powerpc && W_TYPE_SIZE == 64 && defined (_LP64)
 #if !defined (_LONG_LONG_LIMB)
 /* _LONG_LONG_LIMB is ABI=mode32 where adde operates on 32-bit values.  So
    use adde etc only when not _LONG_LONG_LIMB.  */
