diff --git a/src/md5sum.c b/src/md5sum.c
index 8e21609..a857d62 100644
--- a/src/md5sum.c
+++ b/src/md5sum.c
@@ -266,6 +266,9 @@ Print or check %s (%d-bit) checksums.\n\
         fputs (_("\
   -t, --text           read in text mode (default)\n\
 "), stdout);
+  fputs (_("\
+  Note: There is no difference between binary and text mode option on GNU system.\n\
+"), stdout);
       fputs (_("\
 \n\
 The following five options are useful only when verifying checksums:\n\
