diff -up mesa-20110620/configure.ac.classic mesa-20110620/configure.ac
--- mesa-20110620/configure.ac.classic	2011-06-20 13:52:52.000000000 +1000
+++ mesa-20110620/configure.ac	2011-06-20 13:55:09.000000000 +1000
@@ -1023,20 +1023,20 @@ if test "$mesa_driver" = dri -o "$mesa_d
             # x86-64 system where they could *ever* be used.
             if test "x$DRI_DIRS" = "xyes"; then
                 DRI_DIRS="i915 i965 mga nouveau r128 r200 r300 r600 \
-                    radeon savage tdfx unichrome swrast"
+                    radeon savage tdfx unichrome"
             fi
             ;;
         powerpc*)
             # Build only the drivers for cards that exist on PowerPC.
             # At some point MGA will be added, but not yet.
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="r128 r200 r300 r600 radeon tdfx swrast"
+                DRI_DIRS="r128 r200 r300 r600 radeon tdfx nouveau"
             fi
             ;;
         sparc*)
             # Build only the drivers for cards that exist on sparc`
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="r128 r200 r300 r600 radeon swrast"
+                DRI_DIRS="r128 r200 r300 r600 radeon nouveau"
             fi
             ;;
         esac
@@ -1080,7 +1080,7 @@ if test "$mesa_driver" = dri -o "$mesa_d
     # default drivers
     if test "x$DRI_DIRS" = "xyes"; then
         DRI_DIRS="i810 i915 i965 mga nouveau r128 r200 r300 r600 radeon \
-            savage sis tdfx unichrome swrast"
+            savage sis tdfx unichrome"
     fi
 
     DRI_DIRS=`echo "$DRI_DIRS" | $SED 's/  */ /g'`
