diff -upNr Mesa-7.8.1.orign/src/mesa/drivers/dri/i915/intel_tex_image.c Mesa-7.8.1/src/mesa/drivers/dri/i915/intel_tex_image.c
--- Mesa-7.8.1.orign/src/mesa/drivers/dri/i915/intel_tex_image.c	2010-04-05 18:05:15.000000000 +0200
+++ Mesa-7.8.1/src/mesa/drivers/dri/i915/intel_tex_image.c	2010-04-22 19:11:06.881488899 +0200
@@ -728,8 +728,7 @@ intelSetTexBuffer2(__DRIcontext *pDRICtx
    if (!intelObj)
       return;
 
-   if (dPriv->lastStamp != dPriv->dri2.stamp)
-      intel_update_renderbuffers(pDRICtx, dPriv);
+   intel_update_renderbuffers(pDRICtx, dPriv);
 
    rb = intel_get_renderbuffer(fb, BUFFER_FRONT_LEFT);
    /* If the region isn't set, then intel_update_renderbuffers was unable
diff -upNr Mesa-7.8.1.orign/src/mesa/drivers/dri/i965/intel_tex_image.c Mesa-7.8.1/src/mesa/drivers/dri/i965/intel_tex_image.c
--- Mesa-7.8.1.orign/src/mesa/drivers/dri/i965/intel_tex_image.c	2010-04-05 18:05:15.000000000 +0200
+++ Mesa-7.8.1/src/mesa/drivers/dri/i965/intel_tex_image.c	2010-04-22 19:11:06.881488899 +0200
@@ -728,8 +728,7 @@ intelSetTexBuffer2(__DRIcontext *pDRICtx
    if (!intelObj)
       return;
 
-   if (dPriv->lastStamp != dPriv->dri2.stamp)
-      intel_update_renderbuffers(pDRICtx, dPriv);
+   intel_update_renderbuffers(pDRICtx, dPriv);
 
    rb = intel_get_renderbuffer(fb, BUFFER_FRONT_LEFT);
    /* If the region isn't set, then intel_update_renderbuffers was unable
diff -upNr Mesa-7.8.1.orign/src/mesa/drivers/dri/intel/intel_tex_image.c Mesa-7.8.1/src/mesa/drivers/dri/intel/intel_tex_image.c
--- Mesa-7.8.1.orign/src/mesa/drivers/dri/intel/intel_tex_image.c	2010-04-05 18:05:15.000000000 +0200
+++ Mesa-7.8.1/src/mesa/drivers/dri/intel/intel_tex_image.c	2010-04-22 19:11:06.881488899 +0200
@@ -728,8 +728,7 @@ intelSetTexBuffer2(__DRIcontext *pDRICtx
    if (!intelObj)
       return;
 
-   if (dPriv->lastStamp != dPriv->dri2.stamp)
-      intel_update_renderbuffers(pDRICtx, dPriv);
+   intel_update_renderbuffers(pDRICtx, dPriv);
 
    rb = intel_get_renderbuffer(fb, BUFFER_FRONT_LEFT);
    /* If the region isn't set, then intel_update_renderbuffers was unable
