diff -up mesa-20130610/src/gallium/drivers/r600/evergreen_state.c.jx mesa-20130610/src/gallium/drivers/r600/evergreen_state.c
--- mesa-20130610/src/gallium/drivers/r600/evergreen_state.c.jx	2013-06-10 13:58:51.000000000 -0400
+++ mesa-20130610/src/gallium/drivers/r600/evergreen_state.c	2013-06-17 10:24:14.687160428 -0400
@@ -615,7 +615,7 @@ static uint32_t r600_translate_colorform
 
 static uint32_t r600_colorformat_endian_swap(uint32_t colorformat)
 {
-	if (R600_BIG_ENDIAN) {
+	if (0 && R600_BIG_ENDIAN) {
 		switch(colorformat) {
 
 		/* 8-bit buffers. */
