diff -up mesa-20110620/configure.ac.no-mach64 mesa-20110620/configure.ac
--- mesa-20110620/configure.ac.no-mach64	2011-06-20 12:19:16.000000000 +1000
+++ mesa-20110620/configure.ac	2011-06-20 13:52:27.000000000 +1000
@@ -1022,7 +1022,7 @@ if test "$mesa_driver" = dri -o "$mesa_d
             # the new interface.  i810 are missing because there is no
             # x86-64 system where they could *ever* be used.
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="i915 i965 mach64 mga nouveau r128 r200 r300 r600 \
+                DRI_DIRS="i915 i965 mga nouveau r128 r200 r300 r600 \
                     radeon savage tdfx unichrome swrast"
             fi
             ;;
@@ -1030,13 +1030,13 @@ if test "$mesa_driver" = dri -o "$mesa_d
             # Build only the drivers for cards that exist on PowerPC.
             # At some point MGA will be added, but not yet.
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="mach64 r128 r200 r300 r600 radeon tdfx swrast"
+                DRI_DIRS="r128 r200 r300 r600 radeon tdfx swrast"
             fi
             ;;
         sparc*)
             # Build only the drivers for cards that exist on sparc`
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="mach64 r128 r200 r300 r600 radeon swrast"
+                DRI_DIRS="r128 r200 r300 r600 radeon swrast"
             fi
             ;;
         esac
@@ -1050,7 +1050,7 @@ if test "$mesa_driver" = dri -o "$mesa_d
         fi
 
         if test "x$DRI_DIRS" = "xyes"; then
-            DRI_DIRS="i810 i915 i965 mach64 mga nouveau r128 r200 r300 r600 \
+            DRI_DIRS="i810 i915 i965 mga nouveau r128 r200 r300 r600 \
                 radeon tdfx unichrome savage sis swrast"
         fi
         ;;
@@ -1079,7 +1079,7 @@ if test "$mesa_driver" = dri -o "$mesa_d
 
     # default drivers
     if test "x$DRI_DIRS" = "xyes"; then
-        DRI_DIRS="i810 i915 i965 mach64 mga nouveau r128 r200 r300 r600 radeon \
+        DRI_DIRS="i810 i915 i965 mga nouveau r128 r200 r300 r600 radeon \
             savage sis tdfx unichrome swrast"
     fi
 
