diff -uNr mesa-20131201.orig/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp mesa-20131201/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp
--- mesa-20131201.orig/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp	2013-12-01 11:29:21.000000000 +0400
+++ mesa-20131201/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp	2013-12-01 16:05:06.488417551 +0400
@@ -739,7 +739,7 @@
    if (!i)
       i = new_Instruction(pol.context(), op, dType);
 #ifndef NDEBUG // non-conformant assert, so this is required
-   assert(typeid(*i) == typeid(*this));
+   //assert(typeid(*i) == typeid(*this));
 #endif
 
    pol.set<Instruction>(this, i);
