diff -up mesa-20110620/src/mesa/drivers/dri/intel/intel_fbo.c.da mesa-20110620/src/mesa/drivers/dri/intel/intel_fbo.c
--- mesa-20110620/src/mesa/drivers/dri/intel/intel_fbo.c.da	2011-06-20 13:51:25.000000000 +1000
+++ mesa-20110620/src/mesa/drivers/dri/intel/intel_fbo.c	2011-06-20 13:51:32.000000000 +1000
@@ -612,11 +612,6 @@ intel_render_texture(struct gl_context *
        return;
    }
 
-   DBG("Begin render texture tid %lx tex=%u w=%d h=%d refcount=%d\n",
-       _glthread_GetID(),
-       att->Texture->Name, image->Width, image->Height,
-       irb->Base.RefCount);
-
    intel_renderbuffer_set_draw_offset(irb, intel_image, att->Zoffset);
    intel_image->used_as_render_target = GL_TRUE;
 
