diff -up mesa-20110412/configure.ac.jx mesa-20110412/configure.ac
--- mesa-20110412/configure.ac.jx	2011-05-09 15:04:19.000000000 -0400
+++ mesa-20110412/configure.ac	2011-05-09 15:05:11.000000000 -0400
@@ -1679,7 +1679,7 @@ if test "x$enable_gallium_llvm" = xyes; 
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags`
-	LLVM_LIBS="`$LLVM_CONFIG --libs` -lstdc++"
+	LLVM_LIBS="-lLLVM-`llvm-config --version` -lstdc++"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS llvmpipe"
