diff -up src/mesa/drivers/dri/nouveau/nv20_context.c.nouveau src/mesa/drivers/dri/nouveau/nv20_context.c
--- src/mesa/drivers/dri/nouveau/nv20_context.c.nouveau	2010-04-06 16:46:06.235249028 -0300
+++ src/mesa/drivers/dri/nouveau/nv20_context.c	2010-04-06 16:46:21.877286306 -0300
@@ -298,8 +298,8 @@ nv20_hwctx_init(GLcontext *ctx)
 	OUT_RING  (chan, 0);
 
 	BEGIN_RING(chan, kelvin, NV20TCL_TX_GEN_S(0),
-		   4 * NV20TCL_TX_GEN_S__SIZE);
-	for (i=0; i < 4 * NV20TCL_TX_GEN_S__SIZE; i++)
+		   4 * NV20TCL_TX_GEN_MODE_S__SIZE);
+	for (i=0; i < 4 * NV20TCL_TX_GEN_MODE_S__SIZE; i++)
 		OUT_RING(chan, 0);
 
 	BEGIN_RING(chan, kelvin, NV20TCL_FOG_EQUATION_CONSTANT, 3);
