diff -up mesa-20100529/configure.ac.nouveau mesa-20100529/configure.ac
--- mesa-20100529/configure.ac.nouveau	2010-05-29 18:57:30.000000000 +1000
+++ mesa-20100529/configure.ac	2010-05-29 18:58:24.000000000 +1000
@@ -820,20 +820,20 @@ if test "$mesa_driver" = dri; then
             # x86-64 system where they could *ever* be used.
             if test "x$DRI_DIRS" = "xyes"; then
                 DRI_DIRS="i915 i965 mga r128 r200 r300 r600 radeon \
-                    savage tdfx unichrome swrast"
+                    savage tdfx unichrome swrast nouveau"
             fi
             ;;
         powerpc*)
             # Build only the drivers for cards that exist on PowerPC.
             # At some point MGA will be added, but not yet.
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="r128 r200 r300 r600 radeon tdfx swrast"
+                DRI_DIRS="r128 r200 r300 r600 radeon tdfx swrast nouveau"
             fi
             ;;
         sparc*)
             # Build only the drivers for cards that exist on sparc`
             if test "x$DRI_DIRS" = "xyes"; then
-                DRI_DIRS="r128 r200 r300 r600 radeon swrast"
+                DRI_DIRS="r128 r200 r300 r600 radeon swrast nouveau"
             fi
             ;;
         esac
@@ -870,7 +870,7 @@ if test "$mesa_driver" = dri; then
     # default drivers
     if test "x$DRI_DIRS" = "xyes"; then
         DRI_DIRS="i810 i915 i965 mga r128 r200 r300 r600 radeon \
-            savage sis tdfx unichrome swrast"
+            savage sis tdfx unichrome swrast nouveau"
     fi
 
     DRI_DIRS=`echo "$DRI_DIRS" | $SED 's/  */ /g'`
