diff -up mesa-20130213/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp.rtti mesa-20130213/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp
--- mesa-20130213/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp.rtti	2013-02-13 18:08:17.533677028 +1000
+++ mesa-20130213/src/gallium/drivers/nouveau/codegen/nv50_ir.cpp	2013-02-13 18:08:20.496752128 +1000
@@ -716,7 +716,7 @@ Instruction::clone(ClonePolicy<Function>
    if (!i)
       i = new_Instruction(pol.context(), op, dType);
 #ifndef NDEBUG // non-conformant assert, so this is required
-   assert(typeid(*i) == typeid(*this));
+  // assert(typeid(*i) == typeid(*this));
 #endif
 
    pol.set<Instruction>(this, i);
