diff -uNr mesa-20140521.orig/configure.ac mesa-20140521/configure.ac
--- mesa-20140521.orig/configure.ac	2014-05-21 20:11:53.662619590 +0400
+++ mesa-20140521/configure.ac	2014-05-21 20:14:19.361398680 +0400
@@ -1642,7 +1642,7 @@
                 LLVM_COMPONENTS="${LLVM_COMPONENTS} option"
             fi
         fi
-        DEFINES="${DEFINES} -DHAVE_LLVM=0x0$LLVM_VERSION_INT -DLLVM_VERSION_PATCH=$LLVM_VERSION_PATCH"
+        DEFINES="${DEFINES} -DHAVE_LLVM=0x0$LLVM_VERSION_INT -DLLVM_VERSION_PATCH=1"
         MESA_LLVM=1
 
         dnl Check for Clang internal headers
