diff -up slrn/src/Makefile.in.makefile slrn/src/Makefile.in
--- slrn/src/Makefile.in.makefile	2008-03-03 20:39:26.000000000 +0100
+++ slrn/src/Makefile.in	2008-06-09 12:32:47.000000000 +0200
@@ -27,7 +27,7 @@ ICONV_LIB       = @LIBICONV@ @LIBINTL@
 SYSLIBS = @X_EXTRA_LIBS@ @TERMCAP@
 
 #-----------------------------------------------------------------------------
-RPATH		= @RPATH@
+RPATH		= 
 
 #---------------------------------------------------------------------------
 # Installation Directories
@@ -204,8 +204,8 @@ installslang:
 installlocales:
 	cd ../po; $(MAKE) DESTDIR=$(DESTDIR) LOCALEDIR=$(LOCALEDIR) install 
 installbin:
-	$(INSTALL) -m 755 -s $(OBJDIR)/slrn $(DEST_BINDIR)
-	-$(INSTALL) -m 755 -s $(OBJDIR)/slrnpull $(DEST_BINDIR)
+	$(INSTALL) -m 755 $(OBJDIR)/slrn $(DEST_BINDIR)
+	-$(INSTALL) -m 755 $(OBJDIR)/slrnpull $(DEST_BINDIR)
 install: all installdirs installbin installdocs installslang installlocales
 
 # The symlinks target is for my own private use.  It simply creates the object
