--- a/mozilla/toolkit/library/symverscript.in	2012-11-07 11:08:34.000000000 -0500
+++ b/mozilla/toolkit/library/symverscript.in	2012-11-07 11:08:15.000000000 -0500
@@ -0,0 +1,5 @@
+#filter substitution
+@VERSION@ {
+	global: *;
+};
+
--- a/mozilla/toolkit/library/Makefile.in	2012-11-07 11:08:34.000000000 -0500
+++ b/mozilla/toolkit/library/Makefile.in	2012-11-07 11:08:15.000000000 -0500
@@ -120,6 +120,14 @@
 ifeq (Linux,$(OS_ARCH))
 ifneq (Android,$(OS_TARGET))
 OS_LIBS += -lrt
+EXTRA_DSO_LDOPTS += -Wl,-version-script,symverscript
+
+symverscript: symverscript.in
+	$(PYTHON) $(topsrcdir)/config/Preprocessor.py \
+		-DVERSION="$(LIBRARY_NAME)$(MOZILLA_VERSION)" $< > $@
+
+EXTRA_DEPS += symverscript
+
 endif
 endif
 
--- a/mozilla/js/src/symverscript.in	2012-11-07 09:36:16.000000000 -0500
+++ b/mozilla/js/src/symverscript.in	2012-11-07 09:37:05.000000000 -0500
@@ -0,0 +1,5 @@
+#filter substitution
+@VERSION@ {
+	global: *;
+};
+
--- a/mozilla/js/src/Makefile.in	2012-11-07 09:36:16.000000000 -0500
+++ b/mozilla/js/src/Makefile.in	2012-11-07 09:37:09.000000000 -0500
@@ -437,6 +437,12 @@
 DEFINES += -DIMPL_MFBT
 endif

+# ensure symbol versions of shared library on linux do not conflict with those in libxul
+ifeq (Linux,$(OS_TARGET))
+EXTRA_DSO_LDOPTS += -Wl,-version-script,symverscript
+EXTRA_DEPS += symverscript
+endif
+
 EXTRA_DSO_LDOPTS += $(NSPR_LIBS)
 
 ifndef BUILD_OPT
--- a/mozilla/js/src/Makefile.in	2012-11-07 09:36:16.000000000 -0500
+++ b/mozilla/js/src/Makefile.in	2012-11-07 09:37:09.000000000 -0500
@@ -914,4 +914,9 @@
 #
 # END kludges for the Nitro assembler
 ###############################################
+
+# build target for symverscript, needs to be here otherwise it overrides 'all' as default
+symverscript: symverscript.in
+	$(PYTHON) $(topsrcdir)/config/Preprocessor.py \
+		-DVERSION="$(subst -,_,$(LIBRARY_NAME))" $< > $@
 

