diff -up firefox-36.0/mozilla-release/js/src/jit/none/Lowering-none.h.old firefox-36.0/mozilla-release/js/src/jit/none/Lowering-none.h
--- firefox-36.0/mozilla-release/js/src/jit/none/Lowering-none.h.old	2015-02-20 03:01:48.000000000 +0100
+++ firefox-36.0/mozilla-release/js/src/jit/none/Lowering-none.h	2015-03-06 12:00:43.486675228 +0100
@@ -89,6 +89,7 @@ class LIRGeneratorNone : public LIRGener
     bool visitSimdTernaryBitwise(MSimdTernaryBitwise *ins) { MOZ_CRASH(); }
     bool visitSimdSplatX4(MSimdSplatX4 *ins) { MOZ_CRASH(); }
     bool visitSimdValueX4(MSimdValueX4 *lir) { MOZ_CRASH(); }
+    bool visitSubstr(MSubstr *) { MOZ_CRASH(); }
 };
 
 typedef LIRGeneratorNone LIRGeneratorSpecific;
