diff -up firefox-27.0/mozilla-release/security/build/Makefile.in.bld firefox-27.0/mozilla-release/security/build/Makefile.in
--- firefox-27.0/mozilla-release/security/build/Makefile.in.bld	2014-01-28 05:04:01.000000000 +0100
+++ firefox-27.0/mozilla-release/security/build/Makefile.in	2014-02-03 10:56:16.010749049 +0100
@@ -128,7 +128,6 @@ DEFAULT_GMAKE_FLAGS += NSPR_INCLUDE_DIR=
 DEFAULT_GMAKE_FLAGS += NSPR_LIB_DIR=$(NSPR_LIB_DIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
-DEFAULT_GMAKE_FLAGS += NSS_ENABLE_ECC=1
 ifeq ($(OS_ARCH)_$(GNU_CC),WINNT_1)
 DEFAULT_GMAKE_FLAGS += OS_DLLFLAGS="-static-libgcc"
 endif
diff -up firefox-27.0/mozilla-release/security/manager/ssl/src/Makefile.in.bld firefox-27.0/mozilla-release/security/manager/ssl/src/Makefile.in
--- firefox-27.0/mozilla-release/security/manager/ssl/src/Makefile.in.bld	2014-02-03 11:01:58.000000000 +0100
+++ firefox-27.0/mozilla-release/security/manager/ssl/src/Makefile.in	2014-02-03 11:02:05.941855059 +0100
@@ -5,7 +5,6 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 DEFINES += \
-  -DNSS_ENABLE_ECC \
   -DDLL_PREFIX=\"$(DLL_PREFIX)\" \
   -DDLL_SUFFIX=\"$(DLL_SUFFIX)\" \
   $(NULL)
diff -up firefox-27.0/mozilla-release/security/manager/ssl/src/moz.build.bld firefox-27.0/mozilla-release/security/manager/ssl/src/moz.build
