diff -up mozilla-release/toolkit/xre/nsAppRunner.cpp.old mozilla-release/toolkit/xre/nsAppRunner.cpp
--- mozilla-release/toolkit/xre/nsAppRunner.cpp.old	2014-11-26 03:17:40.000000000 +0100
+++ mozilla-release/toolkit/xre/nsAppRunner.cpp	2015-01-05 14:23:05.977933308 +0100
@@ -4168,10 +4168,8 @@ XREMain::XRE_main(int argc, char* argv[]
 
 #if defined(MOZ_WIDGET_GTK)
 #if defined(MOZ_MEMORY) || defined(__FreeBSD__) || defined(__NetBSD__)
-  // Disable the slice allocator, since jemalloc already uses similar layout
-  // algorithms, and using a sub-allocator tends to increase fragmentation.
-  // This must be done before g_thread_init() is called.
-  g_slice_set_config(G_SLICE_CONFIG_ALWAYS_MALLOC, 1);
+  // rhbz#1014858 - enable slice allocator for child processes
+  unsetenv("G_SLICE");
 #endif
   g_thread_init(nullptr);
 #endif
