diff -up gui/Makefile.am.dso gui/Makefile.am
--- gui/Makefile.am.dso	2010-06-03 19:43:02.073923763 -0300
+++ gui/Makefile.am	2010-06-03 19:52:55.656553304 -0300
@@ -124,6 +124,7 @@ gdmchooser_LDADD = \
 	$(X_LIBS)		\
 	$(XDMCP_LIBS)		\
 	-lX11			\
+	-lXau \
 	-lfontconfig		\
 	$(NULL)
 
@@ -144,6 +145,8 @@ gdmlogin_LDADD = \
 	$(XINERAMA_LIBS)	\
 	$(X_LIBS)		\
 	-lX11			\
+	-lXau \
+	-lm \
 	-lfontconfig		\
 	$(NULL)
 
@@ -184,6 +187,7 @@ gdmphotosetup_LDADD = \
 	$(EXTRA_SOCKET_LIB)	\
 	$(EXTRA_NSL_LIB)	\
 	-lX11			\
+	-lXau \
 	-lfontconfig		\
 	$(top_builddir)/common/libgdmcommon.a \
 	$(NULL)
diff -up gui/Makefile.in.dso gui/Makefile.in
--- gui/Makefile.in.dso	2010-06-03 19:42:55.623677952 -0300
+++ gui/Makefile.in	2010-06-03 19:52:54.988549673 -0300
@@ -581,6 +581,7 @@ gdmchooser_LDADD = \
 	$(X_LIBS)		\
 	$(XDMCP_LIBS)		\
 	-lX11			\
+	-lXau \
 	-lfontconfig		\
 	$(NULL)
 
@@ -601,6 +602,8 @@ gdmlogin_LDADD = \
 	$(XINERAMA_LIBS)	\
 	$(X_LIBS)		\
 	-lX11			\
+	-lXau \
+	-lm \
 	-lfontconfig		\
 	$(NULL)
 
@@ -642,6 +645,7 @@ gdmphotosetup_LDADD = \
 	$(EXTRA_SOCKET_LIB)	\
 	$(EXTRA_NSL_LIB)	\
 	-lX11			\
+	-lXau \
 	-lfontconfig		\
 	$(top_builddir)/common/libgdmcommon.a \
 	$(NULL)
diff -up gui/greeter/Makefile.am.dso gui/greeter/Makefile.am
--- gui/greeter/Makefile.am.dso	2010-06-03 19:50:46.822549249 -0300
+++ gui/greeter/Makefile.am	2010-06-03 19:52:54.330663359 -0300
@@ -92,6 +92,7 @@ gdmgreeter_LDADD = \
 	$(XINERAMA_LIBS)	\
 	$(X_LIBS)		\
 	-lX11			\
+	-lXau			\
 	-lfontconfig
 
 EXTRA_DIST = \
diff -up gui/greeter/Makefile.in.dso gui/greeter/Makefile.in
--- gui/greeter/Makefile.in.dso	2010-06-03 19:50:32.773557490 -0300
+++ gui/greeter/Makefile.in	2010-06-03 19:52:53.637928476 -0300
@@ -463,6 +463,7 @@ gdmgreeter_LDADD = \
 	$(XINERAMA_LIBS)	\
 	$(X_LIBS)		\
 	-lX11			\
+	-lXau			\
 	-lfontconfig
 
 EXTRA_DIST = \
