diff -rupN mscore-1.1.old/mscore/mscore/mscore.cpp mscore-1.1/mscore/mscore/mscore.cpp
--- mscore-1.1.old/mscore/mscore/mscore.cpp	2011-07-27 09:32:43.000000000 -0400
+++ mscore-1.1/mscore/mscore/mscore.cpp	2011-07-28 21:54:28.000000000 -0400
@@ -1883,7 +1883,7 @@ int main(int argc, char* av[])
       if (!useFactorySettings)
             preferences.read();
       else {
-            preferences.soundFont = mscoreGlobalShare+"sound/TimGM6mb.sf2";
+            preferences.soundFont = QString(INSTPREFIX "/share/soundfonts/default.sf2");
             }
 
       QSplashScreen* sc = 0;
@@ -3288,4 +3288,4 @@ void MuseScore::closeWebPanelPermanently
       showWebPanel(false);
       preferences.showWebPanel = false;
       preferences.dirty  = true;
-      }
\ No newline at end of file
+      }
diff -rupN mscore-1.1.old/mscore/mscore/preferences.cpp mscore-1.1/mscore/mscore/preferences.cpp
--- mscore-1.1.old/mscore/mscore/preferences.cpp	2011-07-27 09:32:44.000000000 -0400
+++ mscore-1.1/mscore/mscore/preferences.cpp	2011-07-28 21:56:37.000000000 -0400
@@ -361,12 +361,8 @@ void Preferences::read()
       playNotes       = s.value("playNotes", true).toBool();
       lPort           = s.value("lPort").toString();
       rPort           = s.value("rPort").toString();
-      defaultSoundfont = mscoreGlobalShare+"sound/TimGM6mb.sf2";
+      defaultSoundfont = QString(INSTPREFIX "/share/soundfonts/default.sf2");
       soundFont       = s.value("soundFont", defaultSoundfont).toString();
-      if (soundFont == ":/data/piano1.sf2") {
-            // silently change to new default sound font
-            soundFont = defaultSoundfont;
-            }
       showNavigator   = s.value("showNavigator", true).toBool();
       showStatusBar   = s.value("showStatusBar", true).toBool();
       showPlayPanel   = s.value("showPlayPanel", false).toBool();
diff -rupN mscore-1.1.old/mscore/share/sound/CMakeLists.txt mscore-1.1/mscore/share/sound/CMakeLists.txt
--- mscore-1.1.old/mscore/share/sound/CMakeLists.txt	2011-07-27 09:32:55.000000000 -0400
+++ mscore-1.1/mscore/share/sound/CMakeLists.txt	2011-07-28 21:57:16.000000000 -0400
@@ -20,6 +20,6 @@
 
 install(FILES
       TimGM6mb.sf2
-      DESTINATION ${Mscore_SHARE_NAME}${Mscore_INSTALL_NAME}sound
+      DESTINATION ${Mscore_SHARE_NAME}/soundfonts
       )
 
