diff -rupN mscore-1.0.old/mscore/mscore/exportxml.cpp mscore-1.0/mscore/mscore/exportxml.cpp
--- mscore-1.0.old/mscore/mscore/exportxml.cpp	2011-02-04 16:13:12.000000000 -0500
+++ mscore-1.0/mscore/mscore/exportxml.cpp	2011-02-08 23:25:47.000000000 -0500
@@ -2340,7 +2340,7 @@ static void directionTag(Xml& xml, Attri
                    el->userOff().y());
 */
             if (el->type() == HAIRPIN || el->type() == OTTAVA || el->type() == TEXTLINE) {
-                  SLine* sl = static_cast<const SLine*>(el);
+                  SLine* sl = static_cast<SLine*>(el);
 //                  printf("slin segsz=%d", sl->lineSegments().size());
                   if (sl->lineSegments().size() > 0) {
                         LineSegment* seg = sl->lineSegments().at(0);
