--- mscore-0.9.6.old/mscore/mscore/barline.cpp	2010-06-07 17:26:55.000000000 -0400
+++ mscore-0.9.6/mscore/mscore/barline.cpp	2010-06-12 00:20:33.000000000 -0400
@@ -90,7 +90,7 @@ void BarLine::draw(QPainter& p) const
       {
       double lw = point(score()->styleS(ST_barWidth));
       double y1, y2;
-      getY(&y1, &y2);
+      getY((double*)&y1, (double*)&y2);
 
       QPen pen(p.pen());
       pen.setWidthF(lw);
@@ -340,7 +340,7 @@ void BarLine::updateGrips(int* grips, QR
       *grips   = 1;
       double lw = point(score()->styleS(ST_barWidth));
       double y1, y2;
-      getY(&y1, &y2);
+      getY((double*)&y1, (double*)&y2);
       grip[0].translate(QPointF(lw * .5, y2) + canvasPos());
       }
 
@@ -387,7 +387,7 @@ void BarLine::editDrag(int, const QPoint
 void BarLine::endEditDrag()
       {
       double y1, h2;
-      getY(&y1, &h2);
+      getY((double*)&y1, (double*)&h2);
       yoff      = 0.0;
       double ay1 = canvasPos().y();
       double ay2 = ay1 + h2;
@@ -443,7 +443,7 @@ void BarLine::layout()
       _span = staff() ? staff()->barLineSpan() : 1;
 
       double y1, y2;
-      getY(&y1, &y2);
+      getY((double*)&y1, (double*)&y2);
       Spatium w = score()->styleS(ST_barWidth);
       double dw  = 0.0;
 
