diff -rupN mscore-1.0.old/mscore/CMakeLists.txt mscore-1.0/mscore/CMakeLists.txt
--- mscore-1.0.old/mscore/CMakeLists.txt	2011-02-04 16:13:45.000000000 -0500
+++ mscore-1.0/mscore/CMakeLists.txt	2011-02-08 23:18:48.000000000 -0500
@@ -55,6 +55,8 @@ if (APPLE)
       set(OsxFrameworks ${AudioToolboxFW} ${AudioUnitFW} ${CoreAudioFW} ${CoreMidiFW} ${SystemConfigurationFW} ${CoreServicesFW})
 endif (APPLE)
 
+option(USE_SYSTEM_QTSINGLEAPPLICATION "Use system QtSingleApplication" OFF)
+
 set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
 set(CMAKE_SKIP_RULE_DEPENDENCY TRUE)
@@ -236,7 +238,16 @@ if (NOT MINGW AND NOT APPLE)
 endif (NOT MINGW AND NOT APPLE)
 
 set (subdirs al awl bww2mxml osdabzip rtf2html mscore share manual demos
-      singleapp fluid)
+      fluid)
+
+if(USE_SYSTEM_QTSINGLEAPPLICATION)
+  find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES QtSolutions)
+  find_library(QTSINGLEAPPLICATION_LIBRARIES QtSolutions_SingleApplication-2.6)
+else(USE_SYSTEM_QTSINGLEAPPLICATION)
+  set (subdirs ${subdirs} singleapp)
+  set(QTSINGLEAPPLICATION_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/singleapp/src)
+  set(QTSINGLEAPPLICATION_LIBRARIES qtsingleapp)
+endif(USE_SYSTEM_QTSINGLEAPPLICATION)
 
 if (BUILD_SCRIPTGEN)
       set (subdirs ${subdirs} scriptgen)
diff -rupN mscore-1.0.old/mscore/mscore/CMakeLists.txt mscore-1.0/mscore/mscore/CMakeLists.txt
--- mscore-1.0.old/mscore/mscore/CMakeLists.txt	2011-02-08 23:16:27.000000000 -0500
+++ mscore-1.0/mscore/mscore/CMakeLists.txt	2011-02-08 23:19:38.000000000 -0500
@@ -67,6 +67,7 @@ include_directories(
       ${PROJECT_SOURCE_DIR}/scriptdebug
       ${PROJECT_SOURCE_DIR}/osdabzip
       ${PROJECT_SOURCE_DIR}/fluid
+      ${QTSINGLEAPPLICATION_INCLUDE_DIRS}
       ${PROJECT_SOURCE_DIR}/bww2mxml
       )
 
@@ -234,7 +235,7 @@ if (MINGW)
       bww
       osdabzip
       rtf2html
-      qtsingleapp
+      ${QTSINGLEAPPLICATION_LIBRARIES}
       ${QT_mingw_LIBRARIES}
       portaudio
       portmidi
@@ -286,7 +287,7 @@ else (MINGW)
       bww
       osdabzip
       rtf2html
-      qtsingleapp
+      ${QTSINGLEAPPLICATION_LIBRARIES}
       fluid
       z
       dl
diff -rupN mscore-1.0.old/mscore/mscore/mscore.h mscore-1.0/mscore/mscore/mscore.h
--- mscore-1.0.old/mscore/mscore/mscore.h	2011-02-04 16:13:12.000000000 -0500
+++ mscore-1.0/mscore/mscore/mscore.h	2011-02-08 23:19:54.000000000 -0500
@@ -25,7 +25,7 @@
 #include "ui_measuresdialog.h"
 #include "ui_insertmeasuresdialog.h"
 #include "ui_aboutbox.h"
-#include "singleapp/src/qtsingleapplication.h"
+#include <QtSingleApplication>
 #include "updatechecker.h"
 
 class ScoreView;
