diff -rupN mscore-1.2.old/mscore/CMakeLists.txt mscore-1.2/mscore/CMakeLists.txt
--- mscore-1.2.old/mscore/CMakeLists.txt	2012-03-13 09:37:09.000000000 -0400
+++ mscore-1.2/mscore/CMakeLists.txt	2012-03-13 21:03:47.093427409 -0400
@@ -64,6 +64,8 @@ if (APPLE)
       set(OsxFrameworks ${AudioToolboxFW} ${AudioUnitFW} ${CoreAudioFW} ${CoreMidiFW} ${SystemConfigurationFW} ${CoreServicesFW})
 endif (APPLE)
 
+option(USE_SYSTEM_QTSINGLEAPPLICATION "Use system QtSingleApplication" OFF)
+
 set(CMAKE_INCLUDE_CURRENT_DIR TRUE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH ON)
 set(CMAKE_SKIP_RULE_DEPENDENCY TRUE)
@@ -305,7 +307,16 @@ if (NOT MINGW AND NOT APPLE)
 endif (NOT MINGW AND NOT APPLE)
 
 subdirs (al awl bww2mxml osdabzip rtf2html mscore share manual demos
-      singleapp fluid)
+      fluid)
+
+if(USE_SYSTEM_QTSINGLEAPPLICATION)
+  find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES QtSolutions)
+  find_library(QTSINGLEAPPLICATION_LIBRARIES QtSolutions_SingleApplication-2.6)
+else(USE_SYSTEM_QTSINGLEAPPLICATION)
+  subdirs (singleapp)
+  set(QTSINGLEAPPLICATION_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}/singleapp/src)
+  set(QTSINGLEAPPLICATION_LIBRARIES qtsingleapp)
+endif(USE_SYSTEM_QTSINGLEAPPLICATION)
 
 if (BUILD_SCRIPTGEN)
       subdirs (scriptgen)
diff -rupN mscore-1.2.old/mscore/mscore/CMakeLists.txt mscore-1.2/mscore/mscore/CMakeLists.txt
--- mscore-1.2.old/mscore/mscore/CMakeLists.txt	2012-03-13 21:13:46.278302004 -0400
+++ mscore-1.2/mscore/mscore/CMakeLists.txt	2012-03-13 21:08:55.224819711 -0400
@@ -38,6 +38,7 @@ include_directories(
       ${PROJECT_SOURCE_DIR}/scriptdebug
       ${PROJECT_SOURCE_DIR}/osdabzip
       ${PROJECT_SOURCE_DIR}/fluid
+      ${QTSINGLEAPPLICATION_INCLUDE_DIRS}
       ${PROJECT_SOURCE_DIR}/bww2mxml
       )
 
@@ -217,7 +218,7 @@ if (MINGW)
       bww
       osdabzip
       rtf2html
-      qtsingleapp
+      ${QTSINGLEAPPLICATION_LIBRARIES}
       ${QT_mingw_LIBRARIES}
       portaudio
       portmidi
@@ -271,7 +272,7 @@ else (MINGW)
       bww
       osdabzip
       rtf2html
-      qtsingleapp
+      ${QTSINGLEAPPLICATION_LIBRARIES}
       fluid
       z
       dl
diff -rupN mscore-1.2.old/mscore/mscore/mscore.h mscore-1.2/mscore/mscore/mscore.h
--- mscore-1.2.old/mscore/mscore/mscore.h	2012-03-13 09:32:27.000000000 -0400
+++ mscore-1.2/mscore/mscore/mscore.h	2012-03-13 21:05:04.665022579 -0400
@@ -26,7 +26,7 @@
 #include "ui_measuresdialog.h"
 #include "ui_insertmeasuresdialog.h"
 #include "ui_aboutbox.h"
-#include "singleapp/src/qtsingleapplication.h"
+#include <QtSingleApplication>
 #include "updatechecker.h"
 
 class ScoreView;
