diff -rupN mscore-0.9.6.old/mscore/mscore/capella.cpp mscore-0.9.6/mscore/mscore/capella.cpp
--- mscore-0.9.6.old/mscore/mscore/capella.cpp	2010-06-07 17:26:55.000000000 -0400
+++ mscore-0.9.6/mscore/mscore/capella.cpp	2010-06-14 22:48:05.000000000 -0400
@@ -115,7 +115,7 @@ void SimpleTextObj::read()
       align  = cap->readByte();
       _font  = cap->readFont();
       _text  = cap->readString();
-printf("read SimpletextObj(%d,%d) len %d <%s> char0: %02x\n",
+printf("read SimpletextObj(%d,%d) len %zd <%s> char0: %02x\n",
       relPos.x(), relPos.y(), strlen(_text), _text, _text[0]);
       }
 
diff -rupN mscore-0.9.6.old/mscore/mscore/importxml.cpp mscore-0.9.6/mscore/mscore/importxml.cpp
--- mscore-0.9.6.old/mscore/mscore/importxml.cpp	2010-06-07 17:26:55.000000000 -0400
+++ mscore-0.9.6/mscore/mscore/importxml.cpp	2010-06-14 22:47:46.000000000 -0400
@@ -2616,7 +2616,7 @@ void MusicXml::xmlNote(Measure* measure,
             } // for (int s ...
       if (!found) {
             if (voicelist[relStaff].size() >= unsigned(VOICES))
-                  printf("ImportMusicXml: too many voices (staff %d, relStaff %d, %d >= %d)\n",
+                  printf("ImportMusicXml: too many voices (staff %d, relStaff %d, %zd >= %d)\n",
                          staff, relStaff, voicelist[relStaff].size(), VOICES);
             else {
                   voicelist[relStaff].push_back(voice);
