diff -rupN mscore-0.9.6.old/mscore/mscore/mscore.cpp mscore-0.9.6/mscore/mscore/mscore.cpp
--- mscore-0.9.6.old/mscore/mscore/mscore.cpp	2010-06-07 17:26:55.000000000 -0400
+++ mscore-0.9.6/mscore/mscore/mscore.cpp	2010-06-14 20:44:00.000000000 -0400
@@ -1845,7 +1845,7 @@ int main(int argc, char* av[])
       if (!useFactorySettings)
             preferences.read();
       else {
-            preferences.soundFont = mscoreGlobalShare + "/sound/TimGM6mb.sf2";
+	    preferences.soundFont = QString(INSTPREFIX "/share/soundfonts/default.sf2");
             }
 
       QSplashScreen* sc = 0;
diff -rupN mscore-0.9.6.old/mscore/mscore/preferences.cpp mscore-0.9.6/mscore/mscore/preferences.cpp
--- mscore-0.9.6.old/mscore/mscore/preferences.cpp	2010-06-07 17:26:55.000000000 -0400
+++ mscore-0.9.6/mscore/mscore/preferences.cpp	2010-06-14 20:45:20.000000000 -0400
@@ -360,10 +360,10 @@ void Preferences::read()
       lPort           = s.value("lPort").toString();
       rPort           = s.value("rPort").toString();
 
-      soundFont       = s.value("soundFont", mscoreGlobalShare+"/sound/TimGM6mb.sf2").toString();
+      soundFont       = s.value("soundFont", QString(INSTPREFIX "/share/soundfonts/default.sf2")).toString();
       if (soundFont == ":/data/piano1.sf2") {
             // silently change to new default sound font
-            soundFont = mscoreGlobalShare + "/sound/TimGM6mb.sf2";
+	    soundFont = QString(INSTPREFIX "/share/soundfonts/default.sf2");
             }
       showNavigator   = s.value("showNavigator", true).toBool();
       showStatusBar   = s.value("showStatusBar", true).toBool();
diff -rupN mscore-0.9.6.old/mscore/share/sound/CMakeLists.txt mscore-0.9.6/mscore/share/sound/CMakeLists.txt
--- mscore-0.9.6.old/mscore/share/sound/CMakeLists.txt	2010-06-07 17:27:27.000000000 -0400
+++ mscore-0.9.6/mscore/share/sound/CMakeLists.txt	2010-06-14 20:46:18.000000000 -0400
@@ -20,6 +20,6 @@
 
 install(FILES
       TimGM6mb.sf2
-      DESTINATION ${Mscore_SHARE_NAME}${Mscore_INSTALL_NAME}sound
+      DESTINATION ${Mscore_SHARE_NAME}/soundfonts
       )
 
