diff -rupN mscore-1.1.old/mscore/mscore/sym.cpp mscore-1.1/mscore/mscore/sym.cpp
--- mscore-1.1.old/mscore/mscore/sym.cpp	2011-07-27 09:32:43.000000000 -0400
+++ mscore-1.1/mscore/mscore/sym.cpp	2012-03-03 16:47:43.831015608 -0500
@@ -509,7 +509,14 @@ void Sym::draw(QPainter& painter, double
       {
       double imag = 1.0 / mag;
       painter.scale(mag, mag);
+#if 1 // backported from revision 3193
+      painter.setFont(_font);
+      QString s(_code);
+      painter.drawText(x * imag, y * imag, s);
+#else
+      // does not work with surrogates?
       tl->draw(&painter, QPointF(x * imag, y * imag));
+#endif
       painter.scale(imag, imag);
       }
 
