diff -rupN mscore-1.2.old/mscore/mscore/capella.cpp mscore-1.2/mscore/mscore/capella.cpp
--- mscore-1.2.old/mscore/mscore/capella.cpp	2012-03-13 09:32:27.000000000 -0400
+++ mscore-1.2/mscore/mscore/capella.cpp	2012-03-13 20:58:33.969062225 -0400
@@ -116,7 +116,7 @@ void SimpleTextObj::read()
       align  = cap->readByte();
       _font  = cap->readFont();
       _text  = cap->readString();
-printf("read SimpletextObj(%d,%d) len %d <%s> char0: %02x\n",
+printf("read SimpletextObj(%d,%d) len %zd <%s> char0: %02x\n",
       relPos.x(), relPos.y(), strlen(_text), _text, _text[0]);
       }
 
