diff -rupN mscore-0.9.6.old/mscore/mscore/mscore.cpp mscore-0.9.6/mscore/mscore/mscore.cpp
--- mscore-0.9.6.old/mscore/mscore/mscore.cpp	2010-06-07 17:26:55.000000000 -0400
+++ mscore-0.9.6/mscore/mscore/mscore.cpp	2010-06-14 22:19:19.000000000 -0400
@@ -907,7 +907,7 @@ void MuseScore::helpBrowser()
                 if (!mscoreHelp.isReadable()) {
                       QString info(tr("MuseScore handbook not found at: \n"));
                       info += mscoreHelp.filePath();
-                      info += tr("\n\nFrom the \"Help\" menu try choosing \"Online Handbook\" instead.");
+                      info += tr("\n\nEither install the \"mscore-doc\" package and try again, or\nfrom the \"Help\" menu try choosing \"Online Handbook\" instead.");
                       QMessageBox::critical(this, tr("MuseScore: Open Help"), info);
                       return;
                       }
