--- silly-0.1.0.orig/src/loaders/SILLYPNGImageLoader.cpp
+++ silly-0.1.0/src/loaders/SILLYPNGImageLoader.cpp
@@ -66,7 +66,11 @@ void PNG_error_function(png_structp png_
     //  printf("PNG Error: %s\n", error);
     // copied from libpng's pngerror.cpp
     jmp_buf buf;
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+    memcpy(buf, png_jmpbuf((png_ptr)), sizeof(jmp_buf));
+#else
     memcpy(buf, png_ptr->jmpbuf, sizeof(jmp_buf));
+#endif
     longjmp(buf, 1);
 }
 
