--- qt4/kqt4-wrapper.cmake.orig	2009-03-11 21:19:40.000000000 -0300
+++ qt4/kqt4-wrapper.cmake	2010-06-15 21:09:26.095753468 -0300
@@ -15,7 +15,7 @@
 app=`basename $0`
 
 if [ "$app" = "kqt4-wrapper" ] ; then
-    LD_PRELOAD=@CMAKE_INSTALL_PREFIX@/lib/kgtk/libkqt4.so:$LD_PRELOAD "$@"
+    LD_PRELOAD=@CMAKE_INSTALL_PREFIX@/lib@LIB_SUFFIX@/kgtk/libkqt4.so:$LD_PRELOAD "$@"
 else
     dir=`dirname $0`
     oldPath=$PATH
