diff -urp gnote-0.7.3/src/addinmanager.hpp gnote-0.7.3-pm/src/addinmanager.hpp
--- gnote-0.7.3/src/addinmanager.hpp	2011-02-05 00:01:56.000000000 +0100
+++ gnote-0.7.3-pm/src/addinmanager.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -26,6 +26,7 @@
 #include <map>
 #include <string>
 
+#include <cstddef>
 #include <sigc++/signal.h>
 
 #include "sharp/modulemanager.hpp"
diff -urp gnote-0.7.3/src/notebooks/notebookmanager.hpp gnote-0.7.3-pm/src/notebooks/notebookmanager.hpp
--- gnote-0.7.3/src/notebooks/notebookmanager.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/notebooks/notebookmanager.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -23,6 +23,7 @@
 #ifndef _NOTEBOOK_MANAGER_HPP__
 #define _NOTEBOOK_MANAGER_HPP__
 
+#include <cstddef>
 #include <sigc++/signal.h>
 #include <gtkmm/liststore.h>
 #include <gtkmm/treemodel.h>
diff -urp gnote-0.7.3/src/note.hpp gnote-0.7.3-pm/src/note.hpp
--- gnote-0.7.3/src/note.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/note.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -30,6 +30,7 @@
 
 #include <libxml/tree.h>
 
+#include <cstddef>
 #include <sigc++/signal.h>
 #include <gtkmm/textbuffer.h>
 
diff -urp gnote-0.7.3/src/notemanager.hpp gnote-0.7.3-pm/src/notemanager.hpp
--- gnote-0.7.3/src/notemanager.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/notemanager.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include <gconf/gconf.h>
 
+#include <cstddef>
 #include <sigc++/signal.h>
 
 #include "preferences.hpp"
diff -urp gnote-0.7.3/src/notetag.hpp gnote-0.7.3-pm/src/notetag.hpp
--- gnote-0.7.3/src/notetag.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/notetag.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -24,6 +24,7 @@
 
 #include <string>
 
+#include <cstddef>
 #include <sigc++/signal.h>
 #include <glibmm/refptr.h>
 #include <gtkmm/textbuffer.h>
diff -urp gnote-0.7.3/src/preferences.hpp gnote-0.7.3-pm/src/preferences.hpp
--- gnote-0.7.3/src/preferences.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/preferences.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -25,6 +25,7 @@
 
 #include <string>
 #include <gconf/gconf-client.h>
+#include <cstddef>
 #include <sigc++/signal.h>
 
 #include "base/singleton.hpp"
diff -urp gnote-0.7.3/src/prefskeybinder.hpp gnote-0.7.3-pm/src/prefskeybinder.hpp
--- gnote-0.7.3/src/prefskeybinder.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/prefskeybinder.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -23,6 +23,7 @@
 #define _PREFSKEYBINDER_HPP_
 
 #include <string>
+#include <cstddef>
 #include <sigc++/signal.h>
 #include <sigc++/slot.h>
 
diff -urp gnote-0.7.3/src/tagmanager.hpp gnote-0.7.3-pm/src/tagmanager.hpp
--- gnote-0.7.3/src/tagmanager.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/tagmanager.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -24,6 +24,7 @@
 #define __TAG_MANAGER_HPP_
 
 
+#include <cstddef>
 #include <sigc++/signal.h>
 
 #include <glibmm/thread.h>
diff -urp gnote-0.7.3/src/undo.hpp gnote-0.7.3-pm/src/undo.hpp
--- gnote-0.7.3/src/undo.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/undo.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include <boost/noncopyable.hpp>
 
+#include <cstddef>
 #include <sigc++/signal.h>
 #include <gtkmm/textbuffer.h>
 #include <gtkmm/texttag.h>
diff -urp gnote-0.7.3/src/utils.hpp gnote-0.7.3-pm/src/utils.hpp
--- gnote-0.7.3/src/utils.hpp	2010-09-26 10:44:19.000000000 +0200
+++ gnote-0.7.3-pm/src/utils.hpp	2011-02-05 00:11:50.000000000 +0100
@@ -22,6 +22,7 @@
 #ifndef _GNOTE_UTILS_HPP__
 #define _GNOTE_UTILS_HPP__
 
+#include <cstddef>
 #include <sigc++/signal.h>
 
 #include <gdkmm/pixbuf.h>
diff -urp gnote-0.7.3/src/noteaddin.hpp gnote-0.7.3-pm/src/noteaddin.hpp
--- gnote-0.7.3/src/noteaddin.hpp	2011-02-05 00:19:37.000000000 +0100
+++ gnote-0.7.3-pm/src/noteaddin.hpp	2011-02-05 00:18:37.000000000 +0100
@@ -26,6 +26,7 @@
 #include <list>
 #include <map>
 
+#include <cstddef>
 #include <sigc++/connection.h>
 
 #include <gtkmm/menuitem.h>
