diff -up gwenhywfar-4.3.1/gui/qt4/libtest.cpp.foo gwenhywfar-4.3.1/gui/qt4/libtest.cpp
--- gwenhywfar-4.3.1/gui/qt4/libtest.cpp.foo	2012-01-19 09:59:00.445183383 -0500
+++ gwenhywfar-4.3.1/gui/qt4/libtest.cpp	2012-01-19 09:59:11.796183699 -0500
@@ -4,6 +4,8 @@
 #include "qt4_gui.hpp"
 #include "../testdialogs/dlg_test.h"
 
+#include <unistd.h>
+
 #include <gwenhywfar/gwenhywfar.h>
 #include <gwenhywfar/gui.h>
 #include <gwenhywfar/dialog.h>
