diff -up cantata-0.8.3.1/CMakeLists.txt.qtsingleapplication cantata-0.8.3.1/CMakeLists.txt
--- cantata-0.8.3.1/CMakeLists.txt.qtsingleapplication	2012-09-04 06:23:24.000000000 -0500
+++ cantata-0.8.3.1/CMakeLists.txt	2012-09-25 15:03:17.951850931 -0500
@@ -467,9 +467,20 @@ ELSE( ENABLE_KDE_SUPPORT )
     ADD_DEFINITIONS( -DQT_NO_STL -DQT_NO_CAST_TO_ASCII -Wall -Wextra )
     SET( XDG_APPS_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/applications" )
 
-    add_subdirectory(qtsingleapplication)
-    TARGET_LINK_LIBRARIES(cantata qtsingleapplication)
-    include_directories(${CMAKE_BINARY_DIR}/qtsingleapplication)
+    option(USE_SYSTEM_QTSINGLEAPPLICATION "Use System QtSingleApplication" OFF)
+    if(USE_SYSTEM_QTSINGLEAPPLICATION)
+        find_path(QTSINGLEAPPLICATION_INCLUDE_DIRS qtsingleapplication.h PATH_SUFFIXES QtSolutions)
+        find_library(QTSINGLEAPPLICATION_LIBRARIES QtSolutions_SingleApplication-2.6)
+        find_library(QTSINGLECOREAPPLICATION_LIBRARIES QtSolutions_SingleCoreApplication-2.6)
+    else(USE_SYSTEM_QTSINGLEAPPLICATION)
+        add_subdirectory(qtsingleapplication)
+        set(QTSINGLEAPPLICATION_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/qtsingleapplication)
+        set(QTSINGLEAPPLICATION_LIBRARIES qtsingleapplication)
+    endif(USE_SYSTEM_QTSINGLEAPPLICATION)
+
+    TARGET_LINK_LIBRARIES(cantata ${QTSINGLEAPPLICATION_LIBRARIES})
+    include_directories($QTSINGLEAPPLICATION_INCLUDE_DIRS})
+
 ENDIF( ENABLE_KDE_SUPPORT )
 
 ADD_SUBDIRECTORY( icons )
