--- replaygain/ffmpeginput.cpp.ffmpeg	2013-05-02 20:20:55.677656685 -0300
+++ replaygain/ffmpeginput.cpp	2013-05-02 20:21:30.270319131 -0300
@@ -37,6 +37,10 @@ extern "C" {
 
 static QMutex mutex;
 
+#if LIBAVCODEC_VERSION_MAJOR >= 55
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
+#endif
+
 #define BUFFER_SIZE ((((AVCODEC_MAX_AUDIO_FRAME_SIZE * 3) / 2) * sizeof(int16_t)) + FF_INPUT_BUFFER_PADDING_SIZE)
 
 struct FfmpegInput::Handle {
