diff -up cantata-1.3.3/CMakeLists.txt.kde4_includes cantata-1.3.3/CMakeLists.txt
--- cantata-1.3.3/CMakeLists.txt.kde4_includes	2014-06-07 10:45:10.477192972 -0500
+++ cantata-1.3.3/CMakeLists.txt	2014-06-07 10:47:24.482770660 -0500
@@ -229,6 +229,7 @@ endif (ENABLE_HTTP_STREAM_PLAYBACK)
 if (ENABLE_KDE)
     find_package(KDE4 REQUIRED)
     add_definitions(-DENABLE_KDE_SUPPORT)
+    include_directories(${KDE4_INCLUDES})
     set(ENABLE_KDE_SUPPORT TRUE)
 endif (ENABLE_KDE)
 
