diff -up cantata-0.8.3.1/dynamic/cantata-dynamic.perl_env cantata-0.8.3.1/dynamic/cantata-dynamic
--- cantata-0.8.3.1/dynamic/cantata-dynamic.perl_env	2012-09-04 06:23:24.000000000 -0500
+++ cantata-0.8.3.1/dynamic/cantata-dynamic	2012-11-28 08:08:13.564436152 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 
 #  Cantata-Dynamic
 #
diff -up cantata-0.8.3.1/dynamic/cantata-dynamic-server.perl_env cantata-0.8.3.1/dynamic/cantata-dynamic-server
--- cantata-0.8.3.1/dynamic/cantata-dynamic-server.perl_env	2012-09-04 06:23:24.000000000 -0500
+++ cantata-0.8.3.1/dynamic/cantata-dynamic-server	2012-11-28 08:08:56.485906354 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 
 #  Cantata-Dynamic
 #
