diff -up cantata-1.3.4/CMakeLists.txt.system-qtiocompressor cantata-1.3.4/CMakeLists.txt
--- cantata-1.3.4/CMakeLists.txt.system-qtiocompressor	2014-04-08 14:44:37.000000000 -0500
+++ cantata-1.3.4/CMakeLists.txt	2014-06-07 11:12:22.816867598 -0500
@@ -637,10 +638,9 @@ else (QJSON_FOUND)
     target_link_libraries(cantata qjson)
 endif (QJSON_FOUND)
 
-add_subdirectory(3rdparty/qtiocompressor)
 add_subdirectory(icons)
 
-target_link_libraries(cantata support qtiocompressor ${QTLIBS} ${ZLIB_LIBRARIES})
+target_link_libraries(cantata support QtSolutions_IOCompressor-2.3 ${QTLIBS} ${ZLIB_LIBRARIES})
 include_directories(${QTINCLUDES} ${ZLIB_INCLUDE_DIRS})
 
 if (TAGLIB_FOUND)
diff -up cantata-1.3.4/context/albumview.cpp.system-qtiocompressor cantata-1.3.4/context/albumview.cpp
--- cantata-1.3.4/context/albumview.cpp.system-qtiocompressor	2014-03-25 15:27:18.000000000 -0500
+++ cantata-1.3.4/context/albumview.cpp	2014-06-07 11:12:22.810867662 -0500
@@ -27,7 +27,7 @@
 #include "covers.h"
 #include "networkaccessmanager.h"
 #include "utils.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "musiclibrarymodel.h"
 #include "contextengine.h"
 #include "textbrowser.h"
diff -up cantata-1.3.4/context/artistview.cpp.system-qtiocompressor cantata-1.3.4/context/artistview.cpp
--- cantata-1.3.4/context/artistview.cpp.system-qtiocompressor	2014-03-25 15:27:18.000000000 -0500
+++ cantata-1.3.4/context/artistview.cpp	2014-06-07 11:12:22.811867651 -0500
@@ -27,7 +27,7 @@
 #include "utils.h"
 #include "musiclibrarymodel.h"
 #include "networkaccessmanager.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "textbrowser.h"
 #include "contextengine.h"
 #include "actioncollection.h"
diff -up cantata-1.3.4/context/wikipediasettings.cpp.system-qtiocompressor cantata-1.3.4/context/wikipediasettings.cpp
--- cantata-1.3.4/context/wikipediasettings.cpp.system-qtiocompressor	2014-02-20 13:17:51.000000000 -0600
+++ cantata-1.3.4/context/wikipediasettings.cpp	2014-06-07 11:12:22.811867651 -0500
@@ -28,7 +28,7 @@
 #include "icon.h"
 #include "spinner.h"
 #include "settings.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "utils.h"
 #include "action.h"
 #include "thread.h"
diff -up cantata-1.3.4/models/dirviewmodel.cpp.system-qtiocompressor cantata-1.3.4/models/dirviewmodel.cpp
--- cantata-1.3.4/models/dirviewmodel.cpp.system-qtiocompressor	2014-03-09 01:57:44.000000000 -0600
+++ cantata-1.3.4/models/dirviewmodel.cpp	2014-06-07 11:20:23.030770690 -0500
@@ -45,7 +45,7 @@
 #include "icons.h"
 #include "config.h"
 #include "utils.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 
 #if defined ENABLE_MODEL_TEST
 #include "modeltest.h"
diff -up cantata-1.3.4/models/musiclibraryitempodcast.cpp.system-qtiocompressor cantata-1.3.4/models/musiclibraryitempodcast.cpp
--- cantata-1.3.4/models/musiclibraryitempodcast.cpp.system-qtiocompressor	2014-03-25 15:27:18.000000000 -0500
+++ cantata-1.3.4/models/musiclibraryitempodcast.cpp	2014-06-07 11:12:22.811867651 -0500
@@ -30,7 +30,7 @@
 #include "onlineservice.h"
 #include "song.h"
 #include "icons.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "utils.h"
 #include "covers.h"
 #include "rssparser.h"
diff -up cantata-1.3.4/models/musiclibraryitemroot.cpp.system-qtiocompressor cantata-1.3.4/models/musiclibraryitemroot.cpp
--- cantata-1.3.4/models/musiclibraryitemroot.cpp.system-qtiocompressor	2014-03-25 15:27:18.000000000 -0500
+++ cantata-1.3.4/models/musiclibraryitemroot.cpp	2014-06-07 11:12:22.812867641 -0500
@@ -32,7 +32,7 @@
 #include "mpdconnection.h"
 #include "song.h"
 #include "localize.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "musicmodel.h"
 #include <QXmlStreamReader>
 #include <QXmlStreamWriter>
diff -up cantata-1.3.4/models/musiclibrarymodel.cpp.system-qtiocompressor cantata-1.3.4/models/musiclibrarymodel.cpp
--- cantata-1.3.4/models/musiclibrarymodel.cpp.system-qtiocompressor	2014-02-20 13:17:51.000000000 -0600
+++ cantata-1.3.4/models/musiclibrarymodel.cpp	2014-06-07 11:12:22.812867641 -0500
@@ -41,7 +41,7 @@
 #include "utils.h"
 #include "icons.h"
 #include "stdactions.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include <QCommonStyle>
 #include <QFile>
 #include <QTimer>
diff -up cantata-1.3.4/models/streamsmodel.cpp.system-qtiocompressor cantata-1.3.4/models/streamsmodel.cpp
--- cantata-1.3.4/models/streamsmodel.cpp.system-qtiocompressor	2014-03-25 15:27:18.000000000 -0500
+++ cantata-1.3.4/models/streamsmodel.cpp	2014-06-07 11:12:22.813867630 -0500
@@ -40,7 +40,7 @@
 #include "actioncollection.h"
 #include "digitallyimported.h"
 #include "qjson/parser.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "utils.h"
 #include "config.h"
 #include <QModelIndex>
diff -up cantata-1.3.4/online/onlineservice.cpp.system-qtiocompressor cantata-1.3.4/online/onlineservice.cpp
--- cantata-1.3.4/online/onlineservice.cpp.system-qtiocompressor	2014-03-25 15:27:19.000000000 -0500
+++ cantata-1.3.4/online/onlineservice.cpp	2014-06-07 11:12:22.813867630 -0500
@@ -28,7 +28,7 @@
 #include "networkaccessmanager.h"
 #include "mpdparseutils.h"
 #include "covers.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "thread.h"
 #include "settings.h"
 #include <QFile>
diff -up cantata-1.3.4/streams/tar.cpp.system-qtiocompressor cantata-1.3.4/streams/tar.cpp
--- cantata-1.3.4/streams/tar.cpp.system-qtiocompressor	2014-02-20 13:17:54.000000000 -0600
+++ cantata-1.3.4/streams/tar.cpp	2014-06-07 11:12:22.813867630 -0500
@@ -22,7 +22,7 @@
  */
 
 #include "tar.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 
 Tar::Tar(const QString &fileName)
     : file(fileName)
