diff -up cantata-1.4.1/CMakeLists.txt.system-qtiocompressor cantata-1.4.1/CMakeLists.txt
--- cantata-1.4.1/CMakeLists.txt.system-qtiocompressor	2014-08-27 11:15:24.084393939 -0500
+++ cantata-1.4.1/CMakeLists.txt	2014-08-27 11:16:11.182889573 -0500
@@ -714,8 +714,8 @@ endif (WIN32)
 
 add_subdirectory(po)
 add_subdirectory(support)
-add_subdirectory(3rdparty/qtiocompressor)
-target_link_libraries(cantata support-core qtiocompressor ${CANTATA_LIBS} ${QTLIBS} ${ZLIB_LIBRARIES})
+#add_subdirectory(3rdparty/qtiocompressor)
+target_link_libraries(cantata support-core QtSolutions_IOCompressor-2.3 ${CANTATA_LIBS} ${QTLIBS} ${ZLIB_LIBRARIES})
 
 # enable warnings
 add_definitions(-DQT_NO_DEBUG_OUTPUT)
