diff -up cantata-1.4.1/CMakeLists.txt.system-qxt cantata-1.4.1/CMakeLists.txt
--- cantata-1.4.1/CMakeLists.txt.system-qxt	2014-08-27 11:35:04.684766778 -0500
+++ cantata-1.4.1/CMakeLists.txt	2014-08-27 11:38:39.057480514 -0500
@@ -671,8 +672,9 @@ else (ENABLE_KDE)
     endif (ENABLE_QT5)
 
     if (WIN32 OR (NOT APPLE AND NOT ENABLE_QT5))
-        add_definitions(-DQXT_STATIC)
-        add_subdirectory(3rdparty/qxt)
+        #add_definitions(-DQXT_STATIC)
+        #add_subdirectory(3rdparty/qxt)
+        include_directories(/usr/include/QxtCore /usr/include/QxtGui)
         set(CANTATA_SRCS ${CANTATA_SRCS} gui/qxtmediakeys.cpp)
     endif (WIN32 OR (NOT APPLE AND NOT ENABLE_QT5))
 
@@ -696,7 +698,7 @@ else (ENABLE_KDE)
         target_link_libraries(cantata qtsingleapplication)
     endif (WIN32 OR APPLE)
     if (WIN32 OR (NOT APPLE AND NOT ENABLE_QT5))
-        target_link_libraries(cantata qxt)
+        target_link_libraries(cantata QxtGui)
     endif (WIN32 OR (NOT APPLE AND NOT ENABLE_QT5))
 endif (ENABLE_KDE)
 
