diff -up cantata-1.3.4/3rdparty/solid-lite/CMakeLists.txt.libsolid_static cantata-1.3.4/3rdparty/solid-lite/CMakeLists.txt
--- cantata-1.3.4/3rdparty/solid-lite/CMakeLists.txt.libsolid_static	2014-02-20 13:17:53.000000000 -0600
+++ cantata-1.3.4/3rdparty/solid-lite/CMakeLists.txt	2014-06-09 12:52:02.823317653 -0500
@@ -289,7 +289,7 @@ if (ENABLE_QT5)
 else (ENABLE_QT5)
     QT4_WRAP_CPP(solidlite_LIB_MOC_SRCS ${solidlite_LIB_MOC_HDRS} )
 endif (ENABLE_QT5)
-add_library(solidlite ${solidlite_LIB_SRCS} ${solidlite_LIB_MOC_SRCS})
+add_library(solidlite STATIC ${solidlite_LIB_SRCS} ${solidlite_LIB_MOC_SRCS})
 # set_target_properties(solidlite PROPERTIES AUTOMOC TRUE)
 
 if ( UDEV_FOUND )
