diff -up cantata-1.1.0/support/CMakeLists.txt.kde_headers cantata-1.1.0/support/CMakeLists.txt
--- cantata-1.1.0/support/CMakeLists.txt.kde_headers	2013-08-14 13:40:45.000000000 -0500
+++ cantata-1.1.0/support/CMakeLists.txt	2013-08-14 21:33:49.843128020 -0500
@@ -15,6 +15,7 @@ if (NOT WIN32)
 endif (NOT WIN32)
 
 if (ENABLE_KDE_SUPPORT )
+    include_directories( ${KDE4_INCLUDES} )
     macro_ensure_version("4.7.0" ${KDE_VERSION} KDE_VERSION_47)
     if (NOT KDE_VERSION_47)
         set(SUPPORT_SRCS ${SUPPORT_SRCS} kmessagewidget.cpp )
