diff -up cantata-1.2.2/CMakeLists.txt~ cantata-1.2.2/CMakeLists.txt
--- cantata-1.2.2/CMakeLists.txt~	2014-03-18 22:58:11.867871608 +0200
+++ cantata-1.2.2/CMakeLists.txt	2014-03-18 23:22:12.290305025 +0200
@@ -543,10 +543,9 @@ else (QJSON_FOUND)
     target_link_libraries(cantata qjson)
 endif (QJSON_FOUND)
 
-add_subdirectory(3rdparty/qtiocompressor)
 add_subdirectory(icons)
 
-target_link_libraries(cantata support qtiocompressor ${QTLIBS} ${ZLIB_LIBRARIES})
+target_link_libraries(cantata support QtSolutions_IOCompressor-2.3 ${QTLIBS} ${ZLIB_LIBRARIES})
 include_directories(${QTINCLUDES} ${ZLIB_INCLUDE_DIRS})
 
 if (TAGLIB_FOUND)
diff -up cantata-1.2.2/context/albumview.cpp~ cantata-1.2.2/context/albumview.cpp
--- cantata-1.2.2/context/albumview.cpp~	2013-12-14 17:57:49.000000000 +0200
+++ cantata-1.2.2/context/albumview.cpp	2014-03-18 23:17:33.833465344 +0200
@@ -27,7 +27,7 @@
 #include "covers.h"
 #include "networkaccessmanager.h"
 #include "utils.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "musiclibrarymodel.h"
 #include "contextengine.h"
 #include "textbrowser.h"
diff -up cantata-1.2.2/context/artistview.cpp~ cantata-1.2.2/context/artistview.cpp
--- cantata-1.2.2/context/artistview.cpp~	2013-12-14 17:57:49.000000000 +0200
+++ cantata-1.2.2/context/artistview.cpp	2014-03-18 23:18:20.209441072 +0200
@@ -27,7 +27,7 @@
 #include "utils.h"
 #include "musiclibrarymodel.h"
 #include "networkaccessmanager.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "textbrowser.h"
 #include "contextengine.h"
 #include "actioncollection.h"
diff -up cantata-1.2.2/context/wikipediasettings.cpp~ cantata-1.2.2/context/wikipediasettings.cpp
--- cantata-1.2.2/context/wikipediasettings.cpp~	2013-12-14 17:57:49.000000000 +0200
+++ cantata-1.2.2/context/wikipediasettings.cpp	2014-03-18 23:17:24.025258756 +0200
@@ -28,7 +28,7 @@
 #include "icon.h"
 #include "spinner.h"
 #include "settings.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "utils.h"
 #include "action.h"
 #include "thread.h"
diff -up cantata-1.2.2/models/musiclibraryitempodcast.cpp~ cantata-1.2.2/models/musiclibraryitempodcast.cpp
--- cantata-1.2.2/models/musiclibraryitempodcast.cpp~	2013-12-23 11:58:18.000000000 +0200
+++ cantata-1.2.2/models/musiclibraryitempodcast.cpp	2014-03-18 23:18:10.498236891 +0200
@@ -30,7 +30,7 @@
 #include "onlineservice.h"
 #include "song.h"
 #include "icons.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "utils.h"
 #include "covers.h"
 #include "rssparser.h"
diff -up cantata-1.2.2/models/musiclibraryitemroot.cpp~ cantata-1.2.2/models/musiclibraryitemroot.cpp
--- cantata-1.2.2/models/musiclibraryitemroot.cpp~	2013-12-14 17:57:49.000000000 +0200
+++ cantata-1.2.2/models/musiclibraryitemroot.cpp	2014-03-18 23:17:00.536763662 +0200
@@ -32,7 +32,7 @@
 #include "mpdconnection.h"
 #include "song.h"
 #include "localize.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "musicmodel.h"
 #include <QXmlStreamReader>
 #include <QXmlStreamWriter>
diff -up cantata-1.2.2/models/musiclibrarymodel.cpp~ cantata-1.2.2/models/musiclibrarymodel.cpp
--- cantata-1.2.2/models/musiclibrarymodel.cpp~	2013-12-14 17:57:49.000000000 +0200
+++ cantata-1.2.2/models/musiclibrarymodel.cpp	2014-03-18 23:18:00.747031799 +0200
@@ -41,7 +41,7 @@
 #include "utils.h"
 #include "icons.h"
 #include "stdactions.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include <QCommonStyle>
 #include <QFile>
 #include <QTimer>
diff -up cantata-1.2.2/models/streamsmodel.cpp~ cantata-1.2.2/models/streamsmodel.cpp
--- cantata-1.2.2/models/streamsmodel.cpp~	2013-12-14 17:57:49.000000000 +0200
+++ cantata-1.2.2/models/streamsmodel.cpp	2014-03-18 23:17:12.507016038 +0200
@@ -37,7 +37,7 @@
 #include "actioncollection.h"
 #include "digitallyimported.h"
 #include "qjson/parser.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "utils.h"
 #include "config.h"
 #include <QModelIndex>
diff -up cantata-1.2.2/online/onlineservice.cpp~ cantata-1.2.2/online/onlineservice.cpp
--- cantata-1.2.2/online/onlineservice.cpp~	2013-12-14 17:57:50.000000000 +0200
+++ cantata-1.2.2/online/onlineservice.cpp	2014-03-18 23:17:46.033722203 +0200
@@ -28,7 +28,7 @@
 #include "networkaccessmanager.h"
 #include "mpdparseutils.h"
 #include "covers.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 #include "thread.h"
 #include "settings.h"
 #include <QFile>
diff -up cantata-1.2.2/streams/tar.cpp~ cantata-1.2.2/streams/tar.cpp
--- cantata-1.2.2/streams/tar.cpp~	2013-12-14 17:57:50.000000000 +0200
+++ cantata-1.2.2/streams/tar.cpp	2014-03-18 23:16:50.335548470 +0200
@@ -22,7 +22,7 @@
  */
 
 #include "tar.h"
-#include "qtiocompressor/qtiocompressor.h"
+#include <QtIOCompressor>
 
 Tar::Tar(const QString &fileName)
     : file(fileName)
