diff -up cantata-1.2.0/support/CMakeLists.txt.kde4_includes cantata-1.2.0/support/CMakeLists.txt
--- cantata-1.2.0/support/CMakeLists.txt.kde4_includes	2013-12-17 23:26:20.554067977 -0500
+++ cantata-1.2.0/support/CMakeLists.txt	2013-12-17 23:27:01.488396435 -0500
@@ -15,6 +15,7 @@ if (NOT WIN32)
 endif (NOT WIN32)
 
 if (ENABLE_KDE_SUPPORT )
+    include_directories( ${KDE4_INCLUDES} )
     # macro_ensure_version("4.7.0" ${KDE_VERSION} KDE_VERSION_47)
     # if (NOT KDE_VERSION_47)
         set(SUPPORT_SRCS ${SUPPORT_SRCS} kmessagewidget.cpp )
