diff -up cantata-1.4.1/support/icon.h.icons_crash cantata-1.4.1/support/icon.h
--- cantata-1.4.1/support/icon.h.icons_crash	2014-07-22 13:58:44.000000000 -0500
+++ cantata-1.4.1/support/icon.h	2014-08-27 10:39:03.457961536 -0500
@@ -54,7 +54,7 @@ public:
     static void init(QToolButton *btn, bool setFlat=true);
     #ifdef ENABLE_KDE_SUPPORT
     static Icon getMediaIcon(const QString &name) { return Icon(name); }
-    static QString currentTheme() { return KIconLoader::global()->theme()->name(); }
+    static QString currentTheme() { return (KIconLoader::global()->theme() ? KIconLoader::global()->theme()->name() : QString()); }
     #else
     static Icon getMediaIcon(const QString &name);
     static QString currentTheme() { return QIcon::themeName(); }
diff -up cantata-1.4.1/widgets/icons.cpp.icons_crash cantata-1.4.1/widgets/icons.cpp
--- cantata-1.4.1/widgets/icons.cpp.icons_crash	2014-07-22 13:58:44.000000000 -0500
+++ cantata-1.4.1/widgets/icons.cpp	2014-08-27 10:01:28.270480754 -0500
@@ -370,7 +370,7 @@ Icons::Icons()
     #endif
 
     #ifdef ENABLE_STREAMS
-    streamCategoryIcon=Icon(QLatin1String("oxygen")==Icon::currentTheme().toLower() ? "inode-directory" : "folder-music");
+    streamCategoryIcon=Icon(QString::compare(QLatin1String("oxygen"),Icon::currentTheme(), Qt::CaseInsensitive)==0 ? "inode-directory" : "folder-music");
     #endif
 
     QString iconFile=QString(CANTATA_SYS_ICONS_DIR+"stream.png");
@@ -568,7 +568,7 @@ void Icons::initToolbarIcons(const QColo
             toolbarMenuIcon=createMenuIcon(toolbarText);
         }
         #endif
-        if (QLatin1String("gnome")==Icon::currentTheme().toLower()) {
+        if (QString::compare(QLatin1String("gnome"),Icon::currentTheme(), Qt::CaseInsensitive)==0) {
             QColor col=QApplication::palette().color(QPalette::Active, QPalette::WindowText);
             infoIcon=loadSidebarIcon("info", col, col);
         }
@@ -579,7 +579,7 @@ void Icons::initToolbarIcons(const QColo
     }
 
     #if !defined ENABLE_KDE_SUPPORT && !defined Q_OS_WIN && !defined Q_OS_MAC
-    if (QLatin1String("gnome")==Icon::currentTheme().toLower()) {
+    if (QString::compare(QLatin1String("gnome"),Icon::currentTheme(), Qt::CaseInsensitive)==0) {
         QColor col=QApplication::palette().color(QPalette::Active, QPalette::WindowText);
         contextIcon=loadSidebarIcon("info", col, col);
     } else
