diff -up LVM2.2.02.95/lib/misc/lvm-wrappers.c.udevpath LVM2.2.02.95/lib/misc/lvm-wrappers.c
--- LVM2.2.02.95/lib/misc/lvm-wrappers.c.udevpath	2012-05-26 16:56:01.235290748 -0300
+++ LVM2.2.02.95/lib/misc/lvm-wrappers.c	2012-05-26 16:56:35.422020454 -0300
@@ -72,7 +72,7 @@ const char *udev_get_dev_dir(void)
 		return NULL;
 	}
 
-	return udev_get_dev_path(_udev);
+	return "/dev";
 }
 
 struct udev* udev_get_library_context(void)
diff -up LVM2.2.02.95/tools/dmsetup.c.udevpath LVM2.2.02.95/tools/dmsetup.c
--- LVM2.2.02.95/tools/dmsetup.c.udevpath	2012-03-05 23:42:31.000000000 -0300
+++ LVM2.2.02.95/tools/dmsetup.c	2012-05-26 16:54:27.138294189 -0300
@@ -1031,7 +1031,7 @@ static int _set_up_udev_support(const ch
 			  _udev_cookie);
 
 	if (!(udev = udev_new()) ||
-	    !(udev_dev_dir = udev_get_dev_path(udev)) ||
+	    !(udev_dev_dir = "/dev") ||
 	    !*udev_dev_dir) {
 		log_error("Could not get udev dev path.");
 		return 0;
