--- LVM2.2.02.108/conf/example.conf.in	2014-07-23 16:21:02.000000000 +0100
+++ LVM2.2.02.108-new/conf/example.conf.in	2014-07-23 23:05:06.000000000 +0100
@@ -90,7 +90,7 @@ devices {
 	# Rule 3 prefers a symlink.
 	# Rule 4 prefers the path with least value in lexicographical order.
 	# Example:
-	# preferred_names = [ "^/dev/mpath/", "^/dev/mapper/mpath", "^/dev/[hs]d" ]
+	preferred_names = [ "^/dev/mpath/", "^/dev/mapper/mpath", "^/dev/[hs]d" ]
 	# This configuration option does not have a default value defined.
 	# preferred_names=[]
