 libdm/libdevmapper.pc.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libdm/libdevmapper.pc.in b/libdm/libdevmapper.pc.in
index 63c0720..540d0db 100644
--- a/libdm/libdevmapper.pc.in
+++ b/libdm/libdevmapper.pc.in
@@ -8,4 +8,5 @@ Description: device-mapper library
 Version: @DM_LIB_PATCHLEVEL@
 Cflags: -I${includedir} 
 Libs: -L${libdir} -ldevmapper
-Requires.private: @SELINUX_PC@ @UDEV_PC@ @RT_PC@
+Requires.private: @SELINUX_PC@ @UDEV_PC@
+Libs.private: -lrt
