diff -up LVM2.2.02.89/configure.in.static LVM2.2.02.89/configure.in
--- LVM2.2.02.89/configure.in.static	2012-01-23 15:48:18.000000000 -0200
+++ LVM2.2.02.89/configure.in	2012-01-27 17:53:03.073280423 -0200
@@ -32,6 +32,7 @@ case "$host_os" in
 		COPTIMISE_FLAG="-O2"
 		CLDFLAGS="$CLDFLAGS -Wl,--version-script,.export.sym"
 		ELDFLAGS="-Wl,--export-dynamic"
+		STATIC_LDFLAGS="-Wl,--no-export-dynamic"
 		# FIXME Generate list and use --dynamic-list=.dlopen.sym
 		CLDWHOLEARCHIVE="-Wl,-whole-archive"
 		CLDNOWHOLEARCHIVE="-Wl,-no-whole-archive"
@@ -1381,6 +1382,7 @@ AC_SUBST(SELINUX_LIBS)
 AC_SUBST(SELINUX_PC)
 AC_SUBST(SNAPSHOTS)
 AC_SUBST(STATICDIR)
+AC_SUBST(STATIC_LDFLAGS)
 AC_SUBST(STATIC_LINK)
 AC_SUBST(TESTING)
 AC_SUBST(THIN)
diff -up LVM2.2.02.89/daemons/dmeventd/Makefile.in.static LVM2.2.02.89/daemons/dmeventd/Makefile.in
--- LVM2.2.02.89/daemons/dmeventd/Makefile.in.static	2011-11-14 19:30:36.000000000 -0200
+++ LVM2.2.02.89/daemons/dmeventd/Makefile.in	2012-01-27 17:52:46.324832694 -0200
@@ -64,7 +64,7 @@ dmeventd: $(LIB_SHARED) dmeventd.o
 	$(DL_LIBS) $(LVMLIBS) $(LIBS) -rdynamic
 
 dmeventd.static: $(LIB_STATIC) dmeventd.o $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) $(ELDFLAGS) -static -L. -L$(interfacebuilddir) -o $@ \
+	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L. -L$(interfacebuilddir) -o $@ \
 	dmeventd.o $(DL_LIBS) $(LVMLIBS) $(LIBS) $(STATIC_LIBS)
 
 ifeq ("@PKGCONFIG@", "yes")
diff -up LVM2.2.02.89/make.tmpl.in.static LVM2.2.02.89/make.tmpl.in
--- LVM2.2.02.89/make.tmpl.in.static	2012-01-27 17:52:46.321833330 -0200
+++ LVM2.2.02.89/make.tmpl.in	2012-01-27 17:52:46.325832482 -0200
@@ -38,6 +38,7 @@ CLDFLAGS += @CLDFLAGS@
 ELDFLAGS += @ELDFLAGS@
 LDDEPS += @LDDEPS@
 LDFLAGS += @LDFLAGS@
+STATIC_LDFLAGS += @STATIC_LDFLAGS@
 LIB_SUFFIX = @LIB_SUFFIX@
 LVMINTERNAL_LIBS = -llvm-internal $(UDEV_LIBS) $(DL_LIBS)
 DL_LIBS = @DL_LIBS@
diff -up LVM2.2.02.89/tools/Makefile.in.static LVM2.2.02.89/tools/Makefile.in
--- LVM2.2.02.89/tools/Makefile.in.static	2011-11-14 19:30:36.000000000 -0200
+++ LVM2.2.02.89/tools/Makefile.in	2012-01-27 17:52:46.325832482 -0200
@@ -126,7 +126,7 @@ dmsetup: dmsetup.o $(top_builddir)/libdm
 	      -o $@ dmsetup.o -ldevmapper $(LIBS)
 
 dmsetup.static: dmsetup.o $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) \
+	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L$(interfacebuilddir) \
 	      -o $@ dmsetup.o -ldevmapper $(STATIC_LIBS) $(LIBS)
 
 all: device-mapper
@@ -136,7 +136,7 @@ lvm: $(OBJECTS) lvm.o $(top_builddir)/li
 		$(LVMLIBS) $(READLINE_LIBS) $(LIBS) -rdynamic
 
 lvm.static: $(OBJECTS) lvm-static.o $(top_builddir)/lib/liblvm-internal.a  $(interfacebuilddir)/libdevmapper.a
-	$(CC) $(CFLAGS) $(LDFLAGS) -static -L$(interfacebuilddir) -o $@ \
+	$(CC) $(CFLAGS) $(LDFLAGS) $(STATIC_LDFLAGS) -static -L$(interfacebuilddir) -o $@ \
 	      $(OBJECTS) lvm-static.o $(LVMLIBS) $(STATIC_LIBS) $(LIBS)
 
 liblvm2cmd.a: $(top_builddir)/lib/liblvm-internal.a $(OBJECTS) lvmcmdlib.o lvm2cmd.o
