diff -up kdebindings-4.5.95/generator/type_compiler.cpp.valgrind kdebindings-4.5.95/generator/type_compiler.cpp
--- kdebindings-4.5.95/generator/type_compiler.cpp.valgrind	2010-10-12 08:06:55.000000000 -0500
+++ kdebindings-4.5.95/generator/type_compiler.cpp	2011-01-08 20:02:55.623757621 -0600
@@ -191,6 +191,8 @@ void TypeCompiler::visitParameterDeclara
 
 void TypeCompiler::visitPtrOperator(PtrOperatorAST* node)
 {
+    if ( ! m_session->token_stream ) return;
+    if ( ! token_text(m_session->token_stream->kind(node->op)) ) return;
     if (token_text(m_session->token_stream->kind(node->op))[0] == '*') {
         QPair<bool, bool> cv = m_visitor->parseCv(node->cv);
         pointerDepth.append(cv.first);
