From 867f150c4d7999e859a098c795594708b7cffbe8 Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Tue, 1 Apr 2014 11:52:59 +0200
Subject: [PATCH 03/14] kstartupinfo: timestamp from data is obsolete since
 2004

---
 kwin/activation.cpp | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/kwin/activation.cpp b/kwin/activation.cpp
index f924bbc..2a95d3f 100644
--- a/kwin/activation.cpp
+++ b/kwin/activation.cpp
@@ -721,13 +721,9 @@ xcb_timestamp_t Client::readUserTimeMapTimestamp(const KStartupInfoId *asn_id, c
     // newer ASN timestamp always replaces user timestamp, unless user timestamp is 0
     // helps e.g. with konqy reusing
     if (asn_data != NULL && time != 0) {
-        // prefer timestamp from ASN id (timestamp from data is obsolete way)
         if (asn_id->timestamp() != 0
                 && (time == -1U || timestampCompare(asn_id->timestamp(), time) > 0)) {
             time = asn_id->timestamp();
-        } else if (asn_data->timestamp() != -1U
-                  && (time == -1U || timestampCompare(asn_data->timestamp(), time) > 0)) {
-            time = asn_data->timestamp();
         }
     }
     kDebug(1212) << "User timestamp, ASN:" << time;
@@ -850,8 +846,6 @@ void Client::startupIdChanged()
     if (asn_data.xinerama() != -1)
         workspace()->sendClientToScreen(this, asn_data.xinerama());
     Time timestamp = asn_id.timestamp();
-    if (timestamp == 0 && asn_data.timestamp() != -1U)
-        timestamp = asn_data.timestamp();
     if (timestamp != 0) {
         bool activate = workspace()->allowClientActivation(this, timestamp);
         if (asn_data.desktop() != 0 && !isOnCurrentDesktop())
@@ -885,12 +879,9 @@ void Group::startupIdChanged()
     bool asn_valid = workspace()->checkStartupNotification(leader_wid, asn_id, asn_data);
     if (!asn_valid)
         return;
-    if (asn_id.timestamp() != 0 && user_time != -1U
-            && timestampCompare(asn_id.timestamp(), user_time) > 0) {
+    if (asn_id.timestamp() != -1U && user_time != -1U
+        && timestampCompare(asn_id.timestamp(), user_time) > 0) {
         user_time = asn_id.timestamp();
-    } else if (asn_data.timestamp() != -1U && user_time != -1U
-              && timestampCompare(asn_data.timestamp(), user_time) > 0) {
-        user_time = asn_data.timestamp();
     }
 }
 
-- 
1.9.0

