diff --git a/libs/plasmaclock/calendartable.cpp b/libs/plasmaclock/calendartable.cpp
index 47ffd9d..745e92f 100644
--- a/libs/plasmaclock/calendartable.cpp
+++ b/libs/plasmaclock/calendartable.cpp
@@ -101,6 +101,7 @@ class CalendarTablePrivate
             delayedPopulationTimer->setInterval(0);
             delayedPopulationTimer->setSingleShot(true);
             QObject::connect(delayedPopulationTimer, SIGNAL(timeout()), q, SLOT(populateCalendar()));
+            QObject::connect(KGlobalSettings::self(), SIGNAL(settingsChanged(int)), q, SLOT(settingsChanged(int)));

             setDate(initialDate);
         }
@@ -362,6 +363,7 @@ class CalendarTablePrivate
         void populateHolidays();
         void populateEvents();
         void populateCalendar();
+        void settingsChanged(int category);

         CalendarTable *q;
         QString calendarType;
@@ -819,6 +821,17 @@ void CalendarTablePrivate::populateEvents()
     delayedPopulationTimer->start();
 }

+void CalendarTablePrivate::settingsChanged(int category)
+{
+    if (category != KGlobalSettings::SETTINGS_LOCALE) {
+        return;
+    }
+
+    calendar = KGlobal::locale()->calendar();
+
+    q->update();
+}
+
 void CalendarTable::dataUpdated(const QString &source, const Plasma::DataEngine::Data &data)
 {
     Q_UNUSED(source)
diff --git a/libs/plasmaclock/calendartable.h b/libs/plasmaclock/calendartable.h
index 969e4fc..8678593 100644
--- a/libs/plasmaclock/calendartable.h
+++ b/libs/plasmaclock/calendartable.h
@@ -127,6 +127,7 @@ private:
     CalendarTablePrivate* const d;

     Q_PRIVATE_SLOT(d, void populateCalendar())
+    Q_PRIVATE_SLOT(d, void settingsChanged(int category))
 };

 }
