commit 32d7d19aee2aa9ea910903217712325d33f733ea
Author: Christoph Feck <christoph@maxiom.de>
Date:   Sun Sep 14 11:07:28 2014 +0200

    Set foreground color for GTK2 menus
    
    BUG: 127861
    FIXED-IN: 4.11.13

diff --git a/kcontrol/krdb/krdb.cpp b/kcontrol/krdb/krdb.cpp
index 31e8820..73c585f 100644
--- a/kcontrol/krdb/krdb.cpp
+++ b/kcontrol/krdb/krdb.cpp
@@ -372,6 +372,7 @@ static void createGtkrc( bool exportColors, const QPalette& cg, bool exportGtkTh
         t << "style \"MenuItem\"" << endl;
         t << "{" << endl;
         t << "  bg[PRELIGHT] = " << color( cg.color(QPalette::Highlight) ) << endl;
+        t << "  fg[PRELIGHT] = " << color( cg.color(QPalette::HighlightedText) ) << endl;
         t << "}" << endl;
         t << endl;
         t << "class \"*MenuItem\" style \"MenuItem\"" << endl;
