commit 39273c7f728d28618d6c2644db1f1d2fc99d2bb2
Author: Christoph Feck <christoph@maxiom.de>
Date:   Tue Jan 14 02:41:07 2014 +0100

    Allow global shortcut for "Clear Clipboard History"
    
    CCBUG: 329735

diff --git a/klipper/klipper.cpp b/klipper/klipper.cpp
index cf41bd2..2bb5aae 100644
--- a/klipper/klipper.cpp
+++ b/klipper/klipper.cpp
@@ -171,6 +171,7 @@ Klipper::Klipper(QObject* parent, const KSharedConfigPtr& config)
     m_clearHistoryAction = m_collection->addAction( "clear-history" );
     m_clearHistoryAction->setIcon( KIcon("edit-clear-history") );
     m_clearHistoryAction->setText( i18n("C&lear Clipboard History") );
+    m_clearHistoryAction->setGlobalShortcut(KShortcut());
     connect(m_clearHistoryAction, SIGNAL(triggered()), SLOT(slotAskClearHistory()));
 
     m_configureAction = m_collection->addAction( "configure" );
