commit 246badf2c8edd93f6352797b9ddde1a626f4167a
Author: Christoph Feck <christoph@maxiom.de>
Date:   Sun May 4 12:35:02 2014 +0200

    Fix wrong icons
    
    BUG: 334309
    FIXED-IN: 4.11.10

diff --git a/kinfocenter/infocenter.cpp b/kinfocenter/infocenter.cpp
index b0ed9d1..7cd5640 100644
--- a/kinfocenter/infocenter.cpp
+++ b/kinfocenter/infocenter.cpp
@@ -134,7 +134,7 @@ void KInfoCenter::createToolBar()
 
 	m_aboutKcm = actionCollection()->addAction("help_about_module");
 	m_aboutKcm->setText(i18nc("Information about current module located in about menu","About Current Information Module"));
-	m_aboutKcm->setIcon(KIcon("help"));
+	m_aboutKcm->setIcon(KIcon("help-about"));
 	m_aboutKcm->setEnabled(false);
 
 	m_exportAction = new KAction(this);
@@ -192,7 +192,7 @@ void KInfoCenter::createMenuFrame()
 	m_searchAction = new KAction(this);
 	m_searchAction->setShortcut(KShortcut(QKeySequence(Qt::CTRL + Qt::Key_F)));
 	m_searchAction->setText(i18nc("Kaction search label", "Search Modules"));
-	m_searchAction->setIcon(KIcon("search"));
+	m_searchAction->setIcon(KIcon("edit-find"));
 
 	actionCollection()->addAction("search",m_searchAction);
 
