commit aed179611d5d51264c79159311385cf5bfedd1c4
Author: Thomas Lübking <thomas.luebking@gmail.com>
Date:   Sun Oct 6 22:58:20 2013 +0200

    don't leak fpstext texture
    
    BUG: 325610
    FIXED-IN: 4.11.3
    REVIEW: 113136

diff --git a/kwin/effects/showfps/showfps.cpp b/kwin/effects/showfps/showfps.cpp
index 1ab7d57..01bce31 100644
--- a/kwin/effects/showfps/showfps.cpp
+++ b/kwin/effects/showfps/showfps.cpp
@@ -47,7 +47,6 @@ const int MAX_TIME = 100;
 ShowFpsEffect::ShowFpsEffect()
     : paints_pos(0)
     , frames_pos(0)
-    , fpsText(0)
     , m_noBenchmark(effects->effectFrame(EffectFrameUnstyled, false))
 {
     for (int i = 0;
@@ -233,8 +232,7 @@ void ShowFpsEffect::paintGL(int fps)
 
     // Paint FPS numerical value
     if (fpsTextRect.isValid()) {
-        delete fpsText;
-        fpsText = new GLTexture(fpsTextImage(fps));
+        fpsText.reset(new GLTexture(fpsTextImage(fps)));
         fpsText->bind();
         ShaderBinder binder(ShaderManager::SimpleShader);
         if (effects->compositingType() == OpenGL2Compositing) {
diff --git a/kwin/effects/showfps/showfps.h b/kwin/effects/showfps/showfps.h
index 3c30c39..27900b8 100644
--- a/kwin/effects/showfps/showfps.h
+++ b/kwin/effects/showfps/showfps.h
@@ -93,7 +93,7 @@ private:
     int x;
     int y;
     QRect fps_rect;
-    GLTexture *fpsText;
+    QScopedPointer<GLTexture> fpsText;
     int textPosition;
     QFont textFont;
     QColor textColor;
