commit 67014bc3961f78aeb3a16362c5a1b778fdb00c8e
Author: Jan Grulich <jgrulich@redhat.com>
Date:   Sat Mar 1 13:03:30 2014 +0100

    Place a new panel to bottom when possible
    
    REVIEW:116117

diff --git a/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js b/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
index 47f635c..356e689 100644
--- a/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
+++ b/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
@@ -1,5 +1,14 @@
 var panel = new Panel
-if (panelIds.length == 1) {
+var panelScreen = panel.screen
+var numberOfPanelsOnScreen = 0
+
+for (i = 0; i < panelIds.length; ++i) {
+    if (panelById(panelIds[i]).screen == panelScreen) {
+        numberOfPanelsOnScreen += 1
+    }
+}
+
+if (numberOfPanelsOnScreen == 1) {
     // we are the only panel, so set the location for the user
     panel.location = 'bottom'
 }
diff --git a/plasma/desktop/shell/desktopcorona.cpp b/plasma/desktop/shell/desktopcorona.cpp
index d3635d9..b910366 100644
--- a/plasma/desktop/shell/desktopcorona.cpp
+++ b/plasma/desktop/shell/desktopcorona.cpp
@@ -532,12 +532,13 @@ void DesktopCorona::addPanel(const QString &plugin)
     panel->setScreen(screen);
 
     QList<Plasma::Location> freeEdges = DesktopCorona::freeEdges(screen);
+
     //kDebug() << freeEdges;
     Plasma::Location destination;
-    if (freeEdges.contains(Plasma::TopEdge)) {
-        destination = Plasma::TopEdge;
-    } else if (freeEdges.contains(Plasma::BottomEdge)) {
+    if (freeEdges.contains(Plasma::BottomEdge)) {
         destination = Plasma::BottomEdge;
+    } else if (freeEdges.contains(Plasma::TopEdge)) {
+        destination = Plasma::TopEdge;
     } else if (freeEdges.contains(Plasma::LeftEdge)) {
         destination = Plasma::LeftEdge;
     } else if (freeEdges.contains(Plasma::RightEdge)) {
