commit b933e4d24937d79baf19033cf441de88fade35bd
Author: Eike Hein <hein@kde.org>
Date:   Sun Jan 12 16:44:01 2014 +0100

    Timer.start doesn't behave like QTimer::start() ...

diff --git a/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml b/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
index 03da0b2..f68649c 100644
--- a/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
+++ b/plasma/desktop/applets/tasks/package/contents/ui/MouseHandler.qml
@@ -51,7 +51,7 @@ Item {
                 }
             } else if (!tasks.dragSource && above && hoveredItem != above) {
                 hoveredItem = above;
-                activationTimer.start();
+                activationTimer.restart();
             } else if (!above) {
                 hoveredItem = null;
                 activationTimer.stop();
