commit 20239fd28c5929a5b2f211666a7b2b93ff9e1133
Author: Cedric Bellegarde <gnumdk@gmail.com>
Date:   Mon Mar 4 11:36:35 2013 +0100

    BUG: 313269 Cache importer sooner to get shortcuts working

diff --git a/appmenu/appmenu.cpp b/appmenu/appmenu.cpp
index cf8f6ca..4ae90c1 100644
--- a/appmenu/appmenu.cpp
+++ b/appmenu/appmenu.cpp
@@ -154,11 +154,6 @@ void AppMenuModule::slotAboutToHide()
 // New window registered
 void AppMenuModule::slotWindowRegistered(WId id, const QString& service, const QDBusObjectPath& path)
 {
-    if (m_menuStyle == "ButtonVertical") {
-        // Tell Kwin menu is available
-        emit menuAvailable(id);
-    }
-
     KDBusMenuImporter* importer = m_importers.take(id);
      if (importer) {
         delete importer;
@@ -167,7 +162,12 @@ void AppMenuModule::slotWindowRegistered(WId id, const QString& service, const Q
     // Application already active so check if we need create menubar
     if ( m_menuStyle == "TopMenuBar" && id == KWindowSystem::self()->activeWindow()) {
         slotActiveWindowChanged(id);
+    } else if (m_menuStyle == "ButtonVertical") {
+        // Tell Kwin menu is available
+        emit menuAvailable(id);
+        getImporter(id);
     }
+
     // Send a signal on bus for others dbus interface registrars
     emit WindowRegistered(id, service, path);
 }
