--- libs/kworkspace/kdisplaymanager.cpp.muj	2012-10-22 17:05:31.733455101 +0200
+++ libs/kworkspace/kdisplaymanager.cpp	2012-10-22 17:07:28.577100360 +0200
@@ -666,10 +666,15 @@ KDisplayManager::localSessions(SessList
                 if (lsess.isValid()) {
                     SessEnt se;
                     getSessionLocation(lsess, se);
-                    if (lsess.property("Class").toString() != QLatin1String("greeter")) {
+                    if ((lsess.property("Class").toString() != QLatin1String("greeter")) &&
+                            (lsess.property("State").toString() != QLatin1String("closing"))) {
                         NumberedDBusObjectPath numberedPath = lsess.getUser();
+                        se.display = lsess.property("Display").toString();
+                        se.vt = lsess.property("VTNr").toInt();
                         se.user = KUser(K_UID(numberedPath.num)).loginName();
                         se.session = "<unknown>";
+                        se.self = lsess.property("Display").toString() == ::getenv("DISPLAY"); /* Bleh once again */
+                        se.tty = !lsess.property("TTY").toString().isEmpty();
                     }
                     se.self = (sp == currentSession);
                     list.append(se);
