diff -up kde-workspace-4.11.0/plasma/desktop/shell/data/plasma-desktop.desktop.plasma_startup_delay kde-workspace-4.11.0/plasma/desktop/shell/data/plasma-desktop.desktop
--- kde-workspace-4.11.0/plasma/desktop/shell/data/plasma-desktop.desktop.plasma_startup_delay	2013-07-09 17:44:32.000000000 -0500
+++ kde-workspace-4.11.0/plasma/desktop/shell/data/plasma-desktop.desktop	2013-08-19 20:48:11.251312586 -0500
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Exec=plasma-desktop
-X-DBUS-StartupType=unique
+X-DBUS-StartupType=multi
 Name=Plasma Desktop Workspace
 Name[ar]=مساحة عمل بلازما
 Name[ast]=Espaciu de trabayu del escritoriu Plasma
