--- kcontrol/krdb/krdb.cpp.gtkrc	2013-10-21 17:43:58.096029302 -0200
+++ kcontrol/krdb/krdb.cpp	2013-10-21 18:00:15.881807159 -0200
@@ -76,6 +76,18 @@ inline const char * sysGtkrc(int version
     }
 }
 
+inline const char * sysdataGtkrc(int version)
+{
+    if(2==version)
+    {
+	    return "/usr/share/gtk-2.0/gtkrc";
+    }
+    else
+    {
+	    return "/usr/share/gtk/gtkrc";
+    }
+}
+
 inline const char * userGtkrc(int version)
 {
     return 2==version  ? "/.gtkrc-2.0" : "/.gtkrc";
@@ -90,6 +102,9 @@ static void applyGtkStyles(bool active,
    QString userHomeGtkrc = QDir::homePath()+userGtkrc(version);
    if (!list.contains(userHomeGtkrc))
       list.prepend(userHomeGtkrc);
+   QLatin1String systemdataGtkrc = QLatin1String(sysdataGtkrc(version));
+   if (!list.contains(systemdataGtkrc))
+      list.prepend(systemdataGtkrc);
    QLatin1String systemGtkrc = QLatin1String(sysGtkrc(version));
    if (!list.contains(systemGtkrc))
       list.prepend(systemGtkrc);
