diff -up kde-workspace-4.7.3/ksmserver/shutdown.cpp.fix-wetab-power-button-freeze kde-workspace-4.7.3/ksmserver/shutdown.cpp
--- kde-workspace-4.7.3/ksmserver/shutdown.cpp.fix-wetab-power-button-freeze	2011-05-20 15:32:07.000000000 -0500
+++ kde-workspace-4.7.3/ksmserver/shutdown.cpp	2011-11-17 11:08:39.785434855 -0600
@@ -58,15 +58,15 @@ CONNECTION WITH THE SOFTWARE OR THE USE
 #include <QTimer>
 #include <QtDBus/QtDBus>
 
-#include <klocale.h>
-#include <kglobal.h>
-#include <kconfig.h>
-#include <kstandarddirs.h>
-#include <kapplication.h>
-#include <ktemporaryfile.h>
-#include <kconfiggroup.h>
-#include <knotification.h>
-#include <kdisplaymanager.h>
+#include <KApplication>
+#include <KConfig>
+#include <KConfigGroup>
+#include <KGlobal>
+#include <KLocale>
+#include <KNotification>
+#include <KStandardDirs>
+#include <KTemporaryFile>
+#include <kworkspace/kdisplaymanager.h>
 #include "server.h"
 #include "global.h"
 #include "client.h"
@@ -152,9 +152,9 @@ void KSMServer::shutdown( KWorkSpace::Sh
     dialogActive = true;
     QString bopt;
     if ( !logoutConfirmed ) {
+        KApplication::kApplication()->updateUserTimestamp();
         KSMShutdownFeedback::start(); // make the screen gray
-        logoutConfirmed =
-            KSMShutdownDlg::confirmShutdown( maysd, choose, sdtype, bopt );
+        logoutConfirmed = KSMShutdownDlg::confirmShutdown( maysd, choose, sdtype, bopt );
         // ###### We can't make the screen remain gray while talking to the apps,
         // because this prevents interaction ("do you want to save", etc.)
         // TODO: turn the feedback widget into a list of apps to be closed,
diff -up kde-workspace-4.7.3/libs/kworkspace/kworkspace.cpp.fix-wetab-power-button-freeze kde-workspace-4.7.3/libs/kworkspace/kworkspace.cpp
--- kde-workspace-4.7.3/libs/kworkspace/kworkspace.cpp.fix-wetab-power-button-freeze	2011-08-22 08:22:52.000000000 -0500
+++ kde-workspace-4.7.3/libs/kworkspace/kworkspace.cpp	2011-11-17 11:08:39.783434881 -0600
@@ -174,29 +174,25 @@ static void cleanup_sm()
 }
 #endif
 
-bool requestShutDown(
-    ShutdownConfirm confirm, ShutdownType sdtype, ShutdownMode sdmode )
+void requestShutDown(ShutdownConfirm confirm, ShutdownType sdtype, ShutdownMode sdmode)
 {
 #ifdef Q_WS_X11
-    QApplication::syncX();
-    kapp->updateRemoteUserTimestamp( "org.kde.ksmserver" );
     /*  use ksmserver's dcop interface if necessary  */
     if ( confirm == ShutdownConfirmYes ||
          sdtype != ShutdownTypeDefault ||
          sdmode != ShutdownModeDefault )
     {
         org::kde::KSMServerInterface ksmserver("org.kde.ksmserver", "/KSMServer", QDBusConnection::sessionBus());
-        QDBusReply<void> reply = ksmserver.logout((int)confirm,  (int)sdtype,  (int)sdmode);
-        return (reply.isValid());
+        ksmserver.logout((int)confirm,  (int)sdtype,  (int)sdmode);
+        return;
     }
+
     if( helper == NULL )
     {
         helper = new KRequestShutdownHelper();
         qAddPostRoutine(cleanup_sm);
     }
-    return helper->requestShutdown( confirm );
-#else
-    return false;
+    helper->requestShutdown( confirm );
 #endif
 }
 
diff -up kde-workspace-4.7.3/libs/kworkspace/kworkspace.h.fix-wetab-power-button-freeze kde-workspace-4.7.3/libs/kworkspace/kworkspace.h
--- kde-workspace-4.7.3/libs/kworkspace/kworkspace.h.fix-wetab-power-button-freeze	2011-08-22 08:22:52.000000000 -0500
+++ kde-workspace-4.7.3/libs/kworkspace/kworkspace.h	2011-11-17 11:08:39.785434855 -0600
@@ -110,10 +110,8 @@ namespace KWorkSpace
    * ShutdownConfirm
    * @param sdtype The action to take after logging out. ShutdownType
    * @param sdmode If/When the action should be taken. ShutdownMode
-   * @return true on success, false if the session manager could not be
-   * contacted.
    */
-  KDE_EXPORT bool requestShutDown( ShutdownConfirm confirm = ShutdownConfirmDefault,
+  KDE_EXPORT void requestShutDown( ShutdownConfirm confirm = ShutdownConfirmDefault,
                                    ShutdownType    sdtype  = ShutdownTypeDefault,
                                    ShutdownMode    sdmode  = ShutdownModeDefault );
 
diff -up kde-workspace-4.7.3/plasma/desktop/applets/kickoff/core/itemhandlers.cpp.fix-wetab-power-button-freeze kde-workspace-4.7.3/plasma/desktop/applets/kickoff/core/itemhandlers.cpp
--- kde-workspace-4.7.3/plasma/desktop/applets/kickoff/core/itemhandlers.cpp.fix-wetab-power-button-freeze	2011-11-17 11:08:39.000000000 -0600
+++ kde-workspace-4.7.3/plasma/desktop/applets/kickoff/core/itemhandlers.cpp	2011-11-17 11:11:42.109155494 -0600
@@ -165,10 +165,7 @@ void LeaveItemHandler::logout()
         type = KWorkSpace::ShutdownTypeDefault;
     }
 
-//FIXME: the proper fix is to implement the KWorkSpace methods for Windows
-#ifndef Q_WS_WIN
     KWorkSpace::requestShutDown(confirm, type);
-#endif
 }
 
 void LeaveItemHandler::lock()
diff -up kde-workspace-4.7.3/plasma/generic/applets/lock_logout/lockout.cpp.fix-wetab-power-button-freeze kde-workspace-4.7.3/plasma/generic/applets/lock_logout/lockout.cpp
--- kde-workspace-4.7.3/plasma/generic/applets/lock_logout/lockout.cpp.fix-wetab-power-button-freeze	2011-05-20 15:32:08.000000000 -0500
+++ kde-workspace-4.7.3/plasma/generic/applets/lock_logout/lockout.cpp	2011-11-17 11:08:39.786434842 -0600
@@ -274,11 +274,7 @@ void LockOut::clickLogout()
     }
 
     kDebug()<<"LockOut:: logout clicked ";
-#ifndef Q_OS_WIN
-    KWorkSpace::requestShutDown( KWorkSpace::ShutdownConfirmDefault,
-                                 KWorkSpace::ShutdownTypeDefault,
-                                 KWorkSpace::ShutdownModeDefault);
-#endif
+    KWorkSpace::requestShutDown();
 }
 
 void LockOut::clickSwitchUser()
diff -up kde-workspace-4.7.3/plasma/generic/containmentactions/contextmenu/menu.cpp.fix-wetab-power-button-freeze kde-workspace-4.7.3/plasma/generic/containmentactions/contextmenu/menu.cpp
--- kde-workspace-4.7.3/plasma/generic/containmentactions/contextmenu/menu.cpp.fix-wetab-power-button-freeze	2011-11-17 11:08:39.706435844 -0600
+++ kde-workspace-4.7.3/plasma/generic/containmentactions/contextmenu/menu.cpp	2011-11-17 11:08:39.787434830 -0600
@@ -255,11 +255,8 @@ void ContextMenu::logout()
     if (!KAuthorized::authorizeKAction("logout")) {
         return;
     }
-#ifndef Q_WS_WIN
-    KWorkSpace::requestShutDown(KWorkSpace::ShutdownConfirmDefault,
-                                KWorkSpace::ShutdownTypeDefault,
-                                KWorkSpace::ShutdownModeDefault);
-#endif
+
+    KWorkSpace::requestShutDown();
 }
 
 QWidget* ContextMenu::createConfigurationInterface(QWidget* parent)
