diff --git a/powerdevil/daemon/actions/dpms/powerdevildpmsaction.cpp b/powerdevil/daemon/actions/dpms/powerdevildpmsaction.cpp
index a16bf7ee254b9a40c6f033c93e5ca63226469e6e..4b8667da23e6739e5dd00917c3424778fd280f15 100644
--- a/powerdevil/daemon/actions/dpms/powerdevildpmsaction.cpp
+++ b/powerdevil/daemon/actions/dpms/powerdevildpmsaction.cpp
@@ -121,6 +121,7 @@ void PowerDevilDPMSAction::onProfileUnload()
     } else {
         kDebug() << "Not performing DPMS action due to inhibition";
     }
+    DPMSSetTimeouts(dpy, 0, 0, 0);
 }
 
 void PowerDevilDPMSAction::onWakeupFromIdle()
@@ -185,6 +186,9 @@ void PowerDevilDPMSAction::triggerImpl(const QVariantMap& args)
             DPMSForceLevel(dpy, DPMSModeSuspend);
         }
     }
+
+    // this leaves DPMS enabled but if it's meant to be disabled
+    // then the timeouts will be zero and so effectively disabled
 }
 
 bool PowerDevilDPMSAction::loadAction(const KConfigGroup& config)
@@ -200,7 +204,8 @@ void PowerDevilDPMSAction::onUnavailablePoliciesChanged(PowerDevil::PolicyAgent:
         // Inhibition triggered: disable DPMS
         kDebug() << "Disabling DPMS due to inhibition";
         Display *dpy = QX11Info::display();
-        DPMSDisable(dpy);
+        DPMSSetTimeouts(dpy, 0, 0, 0);
+        DPMSDisable(dpy); // wakes the screen - do we want this?
     } else {
         // Inhibition removed: let's start again
         onProfileLoad();
