--- a/plasma/generic/dataengines/powermanagement/powermanagementengine.cpp
+++ b/plasma/generic/dataengines/powermanagement/powermanagementengine.cpp
@@ -147,7 +147,7 @@ bool PowermanagementEngine::sourceReques
                                                               "/org/kde/Solid/PowerManagement",
                                                               "org.kde.Solid.PowerManagement",
                                                               "batteryRemainingTime");
-            QDBusPendingReply< int > reply = QDBusConnection::sessionBus().asyncCall(msg);
+            QDBusPendingReply< qulonglong > reply = QDBusConnection::sessionBus().asyncCall(msg);
             reply.waitForFinished();
             if (reply.isValid()) {
                 //kDebug() << "Remaining time 1:" << reply.value();
@@ -335,7 +335,7 @@ void PowermanagementEngine::reloadPowerD
                                                           "/org/kde/Solid/PowerManagement",
                                                           "org.kde.Solid.PowerManagement",
                                                           "batteryRemainingTime");
-        QDBusPendingReply< int > reply = QDBusConnection::sessionBus().asyncCall(msg);
+        QDBusPendingReply< qulonglong > reply = QDBusConnection::sessionBus().asyncCall(msg);
         reply.waitForFinished();
         if (reply.isValid()) {
             batteryRemainingTimeChanged(reply.value());
