diff -up kde-workspace-4.10.2/plasma/generic/applets/systemtray/ui/applet.cpp.systray_ktp_presence kde-workspace-4.10.2/plasma/generic/applets/systemtray/ui/applet.cpp
--- kde-workspace-4.10.2/plasma/generic/applets/systemtray/ui/applet.cpp.systray_ktp_presence	2013-03-01 00:32:25.120846173 -0600
+++ kde-workspace-4.10.2/plasma/generic/applets/systemtray/ui/applet.cpp	2013-04-09 12:59:57.666691688 -0500
@@ -707,6 +707,10 @@ void Applet::checkDefaultApplets()
         engines->unloadEngine("powermanagement");
     }
 
+    if (!applets.contains("org.kde.ktp-presence")) {
+        s_manager->addApplet("org.kde.ktp-presence", this);
+    }
+
     config().writeEntry("DefaultAppletsAdded", true);
 }
 
