diff -ur kde-workspace-4.11.16/kcontrol/colors/CMakeLists.txt kde-workspace-4.11.16-colorschemes-kde4/kcontrol/colors/CMakeLists.txt
--- kde-workspace-4.11.16/kcontrol/colors/CMakeLists.txt	2015-01-29 19:49:17.000000000 +0100
+++ kde-workspace-4.11.16-colorschemes-kde4/kcontrol/colors/CMakeLists.txt	2015-03-07 02:36:25.000000000 +0100
@@ -9,7 +9,7 @@
 
 install(TARGETS kcm_colors  DESTINATION ${PLUGIN_INSTALL_DIR})
 install( FILES colors.desktop  DESTINATION  ${SERVICES_INSTALL_DIR} )
-install( FILES colorschemes.knsrc DESTINATION  ${CONFIG_INSTALL_DIR} )
+install( FILES colorschemes.knsrc DESTINATION  ${CONFIG_INSTALL_DIR} RENAME "colorschemes-kde4.knsrc" )
 
 # built-in color schemes
 FILE(GLOB schemefiles schemes/*.colors)
diff -ur kde-workspace-4.11.16/kcontrol/colors/colorscm.cpp kde-workspace-4.11.16-colorschemes-kde4/kcontrol/colors/colorscm.cpp
--- kde-workspace-4.11.16/kcontrol/colors/colorscm.cpp	2015-01-29 19:49:17.000000000 +0100
+++ kde-workspace-4.11.16-colorschemes-kde4/kcontrol/colors/colorscm.cpp	2015-03-07 02:36:54.000000000 +0100
@@ -397,7 +397,7 @@
 
 void KColorCm::on_schemeKnsButton_clicked()
 {
-    KNS3::DownloadDialog dialog("colorschemes.knsrc", this);
+    KNS3::DownloadDialog dialog("colorschemes-kde4.knsrc", this);
     dialog.exec();
     if ( ! dialog.changedEntries().isEmpty() )
     {
@@ -428,7 +428,7 @@
         }
 
         // upload
-        KNS3::UploadDialog dialog("colorschemes.knsrc", this);
+        KNS3::UploadDialog dialog("colorschemes-kde4.knsrc", this);
         dialog.setUploadFile(KUrl(path) );
         dialog.exec();
     }
