diff -up kde-workspace-4.7.97/ksplash/ksplashqml/main.cpp.gcc47 kde-workspace-4.7.97/ksplash/ksplashqml/main.cpp
--- kde-workspace-4.7.97/ksplash/ksplashqml/main.cpp.gcc47	2012-01-03 13:43:13.000000000 -0600
+++ kde-workspace-4.7.97/ksplash/ksplashqml/main.cpp	2012-01-04 06:45:43.494619293 -0600
@@ -20,6 +20,7 @@
 #include "SplashApp.h"
 
 #include <iostream>
+#include <unistd.h>
 #include <X11/Xlib.h>
 
 int main(int argc, char **argv)
diff -up kde-workspace-4.7.97/libs/ksysguard/processui/ProcessModel.cpp.gcc47 kde-workspace-4.7.97/libs/ksysguard/processui/ProcessModel.cpp
