diff -up kde-workspace-4.10.3/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js.bz#747982-launchers kde-workspace-4.10.3/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
--- kde-workspace-4.10.3/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js.bz#747982-launchers	2013-05-02 23:36:45.000000000 -0500
+++ kde-workspace-4.10.3/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js	2013-05-07 07:29:47.754712122 -0500
@@ -9,10 +9,6 @@ panel.addWidget("launcher")
 panel.addWidget("org.kde.showActivityManager")
 pager = panel.addWidget("pager")
 pager.writeConfig("hideWhenSingleDesktop", "true")
-tasks = panel.addWidget("tasks")
+panel.addWidget("tasks")
 panel.addWidget("systemtray")
 panel.addWidget("digital-clock")
-
-tasks.currentConfigGroup = new Array("Launchers")
-tasks.writeConfig("browser", "preferred://browser, , , ")
-tasks.writeConfig("filemanager", "preferred://filemanager, , , ")
