commit 06d67a3a0ca6c5e9ea545b40d04a300f316b9405
Author: Dan Vrátil <dvratil@redhat.com>
Date:   Thu Apr 11 13:27:34 2013 +0200

    Clear screenlocker password field on ESC key
    
    REVIEW: 109945

diff --git a/ksmserver/screenlocker/greeter/greeter.cpp b/ksmserver/screenlocker/greeter/greeter.cpp
index f5918f9..4ba67f8 100644
--- a/ksmserver/screenlocker/greeter/greeter.cpp
+++ b/ksmserver/screenlocker/greeter/greeter.cpp
@@ -114,6 +114,11 @@ void GreeterItem::verify()
     m_unlocker->verify();
 }
 
+void GreeterItem::clear()
+{
+    m_unlocker->clear();
+}
+
 void GreeterItem::focusInEvent(QFocusEvent *event)
 {
     QGraphicsItem::focusInEvent(event);
@@ -275,6 +280,11 @@ void Greeter::verify()
     m_greet->next();
 }
 
+void Greeter::clear()
+{
+    m_greet->clear();
+}
+
 void Greeter::failedTimer()
 {
     emit greeterReady();
diff --git a/ksmserver/screenlocker/greeter/greeter.h b/ksmserver/screenlocker/greeter/greeter.h
index a0a1ac3..e50a40c 100644
--- a/ksmserver/screenlocker/greeter/greeter.h
+++ b/ksmserver/screenlocker/greeter/greeter.h
@@ -49,6 +49,7 @@ public:
 
 public Q_SLOTS:
     void verify();
+    void clear();
 
 private Q_SLOTS:
     void themeChanged();
@@ -60,6 +61,7 @@ Q_SIGNALS:
     void greeterAccepted();
 protected:
     virtual void focusInEvent(QFocusEvent *event);
+
 private:
     void init();
     QGraphicsProxyWidget *m_proxy;
@@ -142,6 +144,11 @@ public Q_SLOTS:
      **/
     void verify();
 
+    /**
+     * Invoke to clear password field
+     */
+    void clear();
+
 private Q_SLOTS:
     void handleVerify();
     void failedTimer();
diff --git a/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/Greeter.qml b/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/Greeter.qml
index 3dd9fc0..66a3bb4 100644
--- a/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/Greeter.qml
+++ b/ksmserver/screenlocker/greeter/themes/org.kde.passworddialog/contents/ui/Greeter.qml
@@ -91,6 +91,7 @@ Item {
 
                 Keys.onEnterPressed: verify()
                 Keys.onReturnPressed: verify()
+                Keys.onEscapePressed: clear()
             }
             KeyboardItem {
                 anchors {
