diff -up kde-workspace-4.7.97/startkde.cmake.redhat_startkde kde-workspace-4.7.97/startkde.cmake
--- kde-workspace-4.7.97/startkde.cmake.redhat_startkde	2012-01-03 13:43:14.000000000 -0600
+++ kde-workspace-4.7.97/startkde.cmake	2012-01-04 06:43:55.711665134 -0600
@@ -28,22 +28,6 @@ fi
 # we have to unset this for Darwin since it will screw up KDE's dynamic-loading
 unset DYLD_FORCE_FLAT_NAMESPACE
 
-# in case we have been started with full pathname spec without being in PATH
-bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
-if [ -n "$bindir" ]; then
-  qbindir=`$bindir/kde4-config --qt-binaries`
-  if [ -n "$qbindir" ]; then
-    case $PATH in
-      $qbindir|$qbindir:*|*:$qbindir|*:$qbindir:*) ;;
-      *) PATH=$qbindir:$PATH; export PATH;;
-    esac
-  fi
-  case $PATH in
-    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
-    *) PATH=$bindir:$PATH; export PATH;;
-  esac
-fi
-
 # Boot sequence:
 #
 # kdeinit is used to fork off processes which improves memory usage
@@ -81,7 +96,15 @@ krandrrc [Screen2]
 krandrrc [Screen3]
 kcmfonts General forceFontDPI 0
 kdeglobals Locale Language '' # trigger requesting languages from KLocale
+kdeglobals Locale Country ''
 EOF
+# read the default KSplash theme to use out of kde-settings
+if [ -e /usr/share/kde-settings/kde-profile/default/share/config/ksplashrc ]
+  then eval `grep '^Theme=' /usr/share/kde-settings/kde-profile/default/share/config/ksplashrc`
+       if [ -n "$Theme" ]
+         then sed -i -e "s/Default/$Theme/g" $kdehome/share/config/startupconfigkeys
+       fi
+fi
 kstartupconfig4
 returncode=$?
 if test $returncode -ne 0; then
@@ -159,8 +182,9 @@ fi
 # better use the Autostart folder.
 
 libpath=`kde4-config --path lib | tr : '\n'`
+envpath=/etc/kde/env/
 
-for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/env/,p'`; do
+for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/env/,p'` $envpath ; do
   for file in "$prefix"*.sh; do
     test -r "$file" && . "$file"
   done
@@ -369,7 +393,8 @@ kde3 dcopserver_shutdown --wait 2>/dev/n
 echo 'startkde: Running shutdown scripts...'  1>&2
 
 # Run scripts found in $KDEDIRS/shutdown
-for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/shutdown/,p'`; do
+shutdownpath=/etc/kde/shutdown/
+for prefix in `echo "$libpath" | sed -n -e 's,/lib[^/]*/,/shutdown/,p'` $shutdownpath; do
   for file in `ls "$prefix" 2> /dev/null | egrep -v '(~|\.bak)$'`; do
     test -x "$prefix$file" && "$prefix$file"
   done
