diff -up kde-workspace-4.10.90/kwin/workspace.h.kde#171685 kde-workspace-4.10.90/kwin/workspace.h
--- kde-workspace-4.10.90/kwin/workspace.h.kde#171685	2013-06-27 16:31:25.455394124 -0500
+++ kde-workspace-4.10.90/kwin/workspace.h	2013-06-27 16:33:44.773681745 -0500
@@ -113,7 +113,7 @@ public:
     void setShouldGetFocus(Client*);
     bool activateNextClient(Client* c);
     bool focusChangeEnabled() {
-        return block_focus == 0;
+        return block_focus < 1;
     }
 
     /**
