diff -urN kde-workspace-4.10.3/kdm/kcm/main.cpp kde-workspace-4.10.3-patched/kdm/kcm/main.cpp
--- kde-workspace-4.10.3/kdm/kcm/main.cpp	2013-05-03 15:36:45.175122217 +1100
+++ kde-workspace-4.10.3-patched/kdm/kcm/main.cpp	2013-05-24 16:14:40.815684198 +1100
@@ -303,7 +303,7 @@
         config->group("X-*-Greeter").readEntry(
             "BackgroundCfg", KDE_CONFDIR "/kdm/backgroundrc");
 
-    KConfig systemBackgroundConfig(systemBackgroundConfigName);
+    KConfig systemBackgroundConfig(systemBackgroundConfigName, KConfig::SimpleConfig);
     KSharedConfigPtr pTempConfig = KSharedConfig::openConfig(tempBackgroundConfigName);
     systemBackgroundConfig.copyTo(tempBackgroundConfigName, pTempConfig.data());
     pTempConfig->sync();
