diff -up kde-workspace-4.9.1/plasma/generic/runners/solid/plasma-runner-solid.desktop.solid_krunner_disable kde-workspace-4.9.1/plasma/generic/runners/solid/plasma-runner-solid.desktop
--- kde-workspace-4.9.1/plasma/generic/runners/solid/plasma-runner-solid.desktop.solid_krunner_disable	2012-08-13 03:54:14.000000000 -0500
+++ kde-workspace-4.9.1/plasma/generic/runners/solid/plasma-runner-solid.desktop	2012-09-27 10:54:07.670651957 -0500
@@ -142,5 +142,5 @@ X-KDE-PluginInfo-Email=wilderkde@gmail.c
 X-KDE-PluginInfo-Name=solid
 X-KDE-PluginInfo-Version=1.0
 X-KDE-PluginInfo-License=GPL
-X-KDE-PluginInfo-EnabledByDefault=true
+X-KDE-PluginInfo-EnabledByDefault=false
 X-Plasma-AdvertiseSingleRunnerQueryMode=true
