commit 9c841a55667d4b6679a34e744586e097c820710d
Author: Oliver Henshaw <oliver.henshaw@gmail.com>
Date:   Mon Feb 11 15:13:34 2013 +0000

    Use qBound for succinctness
    
    REVIEW: 108425

diff --git a/ksmserver/screenlocker/ksldapp.cpp b/ksmserver/screenlocker/ksldapp.cpp
index c07ac0e..b3eebf5 100644
--- a/ksmserver/screenlocker/ksldapp.cpp
+++ b/ksmserver/screenlocker/ksldapp.cpp
@@ -150,12 +150,7 @@ void KSldApp::configure()
         // timeout stored in seconds
         m_idleId = KIdleTime::instance()->addIdleTimeout(timeout*1000);
     }
-    m_lockGrace = KScreenSaverSettings::lockGrace();
-    if (m_lockGrace < 0) {
-        m_lockGrace = 0;
-    } else if (m_lockGrace > 300000) {
-        m_lockGrace = 300000; // 5 minutes, keep the value sane
-    }
+    m_lockGrace = qBound(0, KScreenSaverSettings::lockGrace(), 300000);  // max 5 minutes, keep the value sane
     m_autoLogoutTimeout = KScreenSaverSettings::autoLogout() ? KScreenSaverSettings::autoLogoutTimeout() * 1000 : 0;
     m_plasmaEnabled = KScreenSaverSettings::plasmaEnabled();
 }
