diff -up kde-workspace-4.11.0/powerdevil/daemon/backends/upower/backlighthelper.cpp.actual_brightness kde-workspace-4.11.0/powerdevil/daemon/backends/upower/backlighthelper.cpp
--- kde-workspace-4.11.0/powerdevil/daemon/backends/upower/backlighthelper.cpp.actual_brightness	2013-07-09 17:44:32.000000000 -0500
+++ kde-workspace-4.11.0/powerdevil/daemon/backends/upower/backlighthelper.cpp	2013-08-21 08:07:27.207287869 -0500
@@ -252,7 +252,7 @@ ActionReply BacklightHelper::brightness(
         return reply;
     }
 #else
-    QFile file(m_dirname + "/brightness");
+    QFile file(m_dirname + "/actual_brightness");
     if (!file.open(QIODevice::ReadOnly)) {
         reply = ActionReply::HelperErrorReply;
         reply.setErrorCode(file.error());
