diff -up kde-workspace-4.7.95/kdm/backend/auth.c.selinux_wtf kde-workspace-4.7.95/kdm/backend/auth.c
--- kde-workspace-4.7.95/kdm/backend/auth.c.selinux_wtf	2011-12-20 18:40:36.000000000 -0600
+++ kde-workspace-4.7.95/kdm/backend/auth.c	2012-01-03 14:33:38.931306044 -0600
@@ -1292,6 +1292,12 @@ removeUserAuthorization(struct display *
 
     if (!(auths = d->authorizations))
         return;
+
+    /* given that the code below doesn't handle the forceUserAuthDir case, */
+    /* let's just bail out for now, see http://bugs.kde.org/242065         */
+    if ( d->forceUserAuthDir ) 
+        return;
+
     debug("removeUserAuthorization\n");
     startUserAuth(name, new_name, &old, &new);
     if (new) {
