diff -up kde-workspace-4.7.2/libs/plasmaclock/calendartable.cpp.displayEvents_default kde-workspace-4.7.2/libs/plasmaclock/calendartable.cpp
--- kde-workspace-4.7.2/libs/plasmaclock/calendartable.cpp.displayEvents_default	2011-06-22 09:29:19.000000000 -0500
+++ kde-workspace-4.7.2/libs/plasmaclock/calendartable.cpp	2011-10-12 17:58:04.818558317 -0500
@@ -867,7 +867,7 @@ void CalendarTable::applyConfiguration(K
     }
 
     setDisplayHolidays(holidays);
-    setDisplayEvents(cg.readEntry("displayEvents", true));
+    setDisplayEvents(cg.readEntry("displayEvents", false));
 }
 
 void CalendarTable::writeConfiguration(KConfigGroup cg)
