diff -up kde-workspace-4.6.80/kwin/workspace.h.kde#171685 kde-workspace-4.6.80/kwin/workspace.h
--- kde-workspace-4.6.80/kwin/workspace.h.kde#171685	2011-05-20 22:32:07.000000000 +0200
+++ kde-workspace-4.6.80/kwin/workspace.h	2011-05-27 17:12:15.661430903 +0200
@@ -144,7 +144,7 @@ public:
     void unfakeActivity(Client* c);
     bool activateNextClient(Client* c);
     bool focusChangeEnabled() {
-        return block_focus == 0;
+        return block_focus < 1;
     }
 
     void updateColormap();
