diff -Nur dhcp-4.1.1.orig//client/scripts/linux dhcp-4.1.1/client/scripts/linux
--- dhcp-4.1.1.orig//client/scripts/linux	2009-04-21 09:21:09.000000000 -0500
+++ dhcp-4.1.1/client/scripts/linux	2010-03-03 09:14:33.261713322 -0600
@@ -22,6 +22,9 @@
 # 4. TIMEOUT not tested. ping has a flag I don't know, and I'm suspicious
 # of the $1 in its args.
 
+PATH=/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin
+export PATH
+
 # 'ip' just looks too weird.  /sbin/ip looks less weird.
 ip=/sbin/ip
 
