diff -up dhcp-4.2.4-P2/common/parse.c.dupl-key dhcp-4.2.4-P2/common/parse.c
--- dhcp-4.2.4-P2/common/parse.c.dupl-key	2012-11-16 15:31:30.568561745 +0100
+++ dhcp-4.2.4-P2/common/parse.c	2012-11-16 15:31:30.577561619 +0100
@@ -2893,6 +2893,9 @@ int parse_zone (struct dns_zone *zone, s
 			    }
 			    val = key_name;
 		    }
+		    if (zone->key)
+			log_fatal("Multiple key definition for zone %s.",
+				  zone->name);
 		    if (omapi_auth_key_lookup_name (&zone -> key, val) !=
 			ISC_R_SUCCESS)
 			    parse_warn (cfile, "unknown key %s", val);
