diff -up dhcp-4.1.1/client/dhclient.conf.supersede dhcp-4.1.1/client/dhclient.conf
--- dhcp-4.1.1/client/dhclient.conf.supersede	1997-06-03 00:50:44.000000000 +0200
+++ dhcp-4.1.1/client/dhclient.conf	2010-01-20 17:24:00.000000000 +0100
@@ -1,10 +1,10 @@
 send host-name "andare.fugue.com";
 send dhcp-client-identifier 1:0:a0:24:ab:fb:9c;
 send dhcp-lease-time 3600;
-supersede domain-name "fugue.com home.vix.com";
+supersede domain-search "fugue.com home.vix.com";
 prepend domain-name-servers 127.0.0.1;
 request subnet-mask, broadcast-address, time-offset, routers,
-	domain-name, domain-name-servers, host-name;
+	domain-search, domain-name, domain-name-servers, host-name;
 require subnet-mask, domain-name-servers;
 timeout 60;
 retry 60;
