diff -up dhcp-4.2.4/server/ldap.c.failOverPeer dhcp-4.2.4/server/ldap.c
--- dhcp-4.2.4/server/ldap.c.failOverPeer	2012-07-23 12:53:26.815262322 +0200
+++ dhcp-4.2.4/server/ldap.c	2012-07-23 12:54:31.002119299 +0200
@@ -893,7 +893,7 @@ ldap_start (void)
 static void
 parse_external_dns (LDAPMessage * ent)
 {
-  char *search[] = {"dhcpOptionsDN", "dhcpSharedNetworkDN", "dhcpSubnetDN",
+  char *search[] = {"dhcpFailOverPeerDN", "dhcpOptionsDN", "dhcpSharedNetworkDN", "dhcpSubnetDN",
                     "dhcpGroupDN", "dhcpHostDN", "dhcpClassesDN",
                     "dhcpPoolDN", NULL};
   LDAPMessage * newres, * newent;
