diff -up dhcp-4.3.0a1/common/conflex.c.forward-updates dhcp-4.3.0a1/common/conflex.c
--- dhcp-4.3.0a1/common/conflex.c.forward-updates	2013-12-20 13:52:11.315466686 +0100
+++ dhcp-4.3.0a1/common/conflex.c	2013-12-20 13:52:11.492464218 +0100
@@ -896,7 +896,7 @@ intern(char *atom, enum dhcp_token dfv)
 			if (!strcasecmp(atom + 7, "list"))
 				return DOMAIN_LIST;
 		}
-		if (!strcasecmp (atom + 1, "o-forward-update"))
+		if (!strcasecmp (atom + 1, "o-forward-updates"))
 			return DO_FORWARD_UPDATE;
 		if (!strcasecmp (atom + 1, "ebug"))
 			return TOKEN_DEBUG;
