diff -up dhcp-4.2.4-P2/common/conflex.c.orig dhcp-4.2.4-P2/common/conflex.c
--- dhcp-4.2.4-P2/common/conflex.c.orig	2012-08-28 04:13:03.000000000 +0200
+++ dhcp-4.2.4-P2/common/conflex.c	2012-10-09 14:01:29.922104149 +0200
@@ -889,7 +889,7 @@ intern(char *atom, enum dhcp_token dfv)
 			if (!strcasecmp(atom + 7, "list"))
 				return DOMAIN_LIST;
 		}
-		if (!strcasecmp (atom + 1, "o-forward-update"))
+		if (!strcasecmp (atom + 1, "o-forward-updates"))
 			return DO_FORWARD_UPDATE;
 		if (!strcasecmp (atom + 1, "ebug"))
 			return TOKEN_DEBUG;
