diff -up dhcp-4.1.1/client/dhclient.c.logpid dhcp-4.1.1/client/dhclient.c
--- dhcp-4.1.1/client/dhclient.c.logpid	2010-01-20 17:39:50.000000000 +0100
+++ dhcp-4.1.1/client/dhclient.c	2010-01-20 17:39:50.000000000 +0100
@@ -150,7 +150,7 @@ main(int argc, char **argv) {
 	else if (fd != -1)
 		close(fd);
 
-	openlog("dhclient", LOG_NDELAY, LOG_DAEMON);
+	openlog("dhclient", LOG_NDELAY | LOG_PID, LOG_DAEMON);
 
 #if !(defined(DEBUG) || defined(__CYGWIN32__))
 	setlogmask(LOG_UPTO(LOG_INFO));
