diff -up dhcp-4.1.1/server/confpars.c.failover-ports dhcp-4.1.1/server/confpars.c
--- dhcp-4.1.1/server/confpars.c.failover-ports	2010-01-20 17:16:18.000000000 +0100
+++ dhcp-4.1.1/server/confpars.c	2010-01-20 17:16:18.000000000 +0100
@@ -1156,10 +1156,17 @@ void parse_failover_peer (cfile, group, 
 		parse_warn (cfile, "peer address may not be omitted");
 
 	/* XXX - when/if we get a port number assigned, just set as default */
+	/* See Red Hat Bugzilla 167292:
+	 * we do now:  dhcp-failover 647/tcp
+	 *             dhcp-failover 647/udp
+	 *             dhcp-failover 847/tcp
+	 *             dhcp-failover 847/udp
+	 * IANA registration by Bernard Volz <volz@cisco.com>
+	 */
 	if (!peer -> me.port)
-		parse_warn (cfile, "local port may not be omitted");
+		peer -> me.port = 647;
 	if (!peer -> partner.port)
-		parse_warn (cfile, "peer port may not be omitted");
+		peer -> partner.port = 847;
 
 	if (peer -> i_am == primary) {
 	    if (!peer -> hba) {
