diff -up dhcp-4.3.0a1/common/packet.c.ffff dhcp-4.3.0a1/common/packet.c
--- dhcp-4.3.0a1/common/packet.c.ffff	2013-12-20 13:59:42.947165977 +0100
+++ dhcp-4.3.0a1/common/packet.c	2013-12-20 13:59:43.137163326 +0100
@@ -326,6 +326,7 @@ decode_udp_ip_header(struct interface_in
   len = ulen - sizeof(udp);
 
   usum = udp.uh_sum;
+  if (usum == 0xffff) usum = 0;
   udp.uh_sum = 0;
 
   /* XXX: We have to pass &udp, because we have to zero the checksum
