diff -up dhcp-4.2.4-P1/configure.ac.remove-dst dhcp-4.2.4-P1/configure.ac
--- dhcp-4.2.4-P1/configure.ac.remove-dst	2012-08-17 15:24:29.066454140 +0200
+++ dhcp-4.2.4-P1/configure.ac	2012-08-17 15:24:29.071454073 +0200
@@ -608,7 +608,6 @@ AC_OUTPUT([
   common/Makefile
   common/tests/Makefile
   dhcpctl/Makefile
-  dst/Makefile
   includes/Makefile
   omapip/Makefile
   relay/Makefile
diff -up dhcp-4.2.4-P1/includes/Makefile.am.remove-dst dhcp-4.2.4-P1/includes/Makefile.am
--- dhcp-4.2.4-P1/includes/Makefile.am.remove-dst	2012-07-13 08:17:54.000000000 +0200
+++ dhcp-4.2.4-P1/includes/Makefile.am	2012-08-17 15:25:08.253922458 +0200
@@ -1,7 +1,6 @@
 nobase_include_HEADERS = omapip/alloc.h omapip/buffer.h omapip/convert.h \
 			 omapip/hash.h omapip/isclib.h omapip/omapip.h \
-			 omapip/omapip_p.h omapip/result.h omapip/trace.h \
-			 isc-dhcp/dst.h
+			 omapip/omapip_p.h omapip/result.h omapip/trace.h
 
 EXTRA_DIST = cdefs.h ctrace.h dhcp.h dhcp6.h dhcpd.h dhctoken.h failover.h \
 	     heap.h inet.h minires.h osdep.h site.h statement.h tree.h \
diff -up dhcp-4.2.4-P1/Makefile.am.remove-dst dhcp-4.2.4-P1/Makefile.am
--- dhcp-4.2.4-P1/Makefile.am.remove-dst	2012-08-17 15:24:29.067454126 +0200
+++ dhcp-4.2.4-P1/Makefile.am	2012-08-17 15:24:29.071454073 +0200
@@ -27,7 +27,7 @@ else
 SUBDIRS = 
 endif
 
-SUBDIRS += includes tests common dst omapip client dhcpctl relay server
+SUBDIRS += includes tests common omapip client dhcpctl relay server
 
 nobase_include_HEADERS = dhcpctl/dhcpctl.h
 
diff -up dhcp-4.2.4-P1/server/ddns.c.remove-dst dhcp-4.2.4-P1/server/ddns.c
--- dhcp-4.2.4-P1/server/ddns.c.remove-dst	2012-07-13 08:18:05.000000000 +0200
+++ dhcp-4.2.4-P1/server/ddns.c	2012-08-17 15:24:29.072454060 +0200
@@ -34,7 +34,6 @@
  */
 
 #include "dhcpd.h"
-#include "dst/md5.h"
 #include <dns/result.h>
 
 #ifdef NSUPDATE
