diff -up dhcp-4.3.0a1/client/dhclient.c.logpid dhcp-4.3.0a1/client/dhclient.c
--- dhcp-4.3.0a1/client/dhclient.c.logpid	2013-12-19 15:51:50.588388633 +0100
+++ dhcp-4.3.0a1/client/dhclient.c	2013-12-19 15:51:50.593388563 +0100
@@ -165,7 +165,7 @@ main(int argc, char **argv) {
 	else if (fd != -1)
 		close(fd);
 
-	openlog("dhclient", LOG_NDELAY, LOG_DAEMON);
+	openlog("dhclient", LOG_NDELAY | LOG_PID, LOG_DAEMON);
 
 #if !(defined(DEBUG) || defined(__CYGWIN32__))
 	setlogmask(LOG_UPTO(LOG_INFO));
