diff -up dhcp-4.3.0a1/common/parse.c.dupl-key dhcp-4.3.0a1/common/parse.c
--- dhcp-4.3.0a1/common/parse.c.dupl-key	2013-12-20 13:53:01.121772037 +0100
+++ dhcp-4.3.0a1/common/parse.c	2013-12-20 13:53:01.211770781 +0100
@@ -2930,6 +2930,9 @@ int parse_zone (struct dns_zone *zone, s
 			    }
 			    val = key_name;
 		    }
+		    if (zone->key)
+			log_fatal("Multiple key definition for zone %s.",
+				  zone->name);
 		    if (omapi_auth_key_lookup_name (&zone -> key, val) !=
 			ISC_R_SUCCESS)
 			    parse_warn (cfile, "unknown key %s", val);
