diff -up dhcp-4.1.1/client/dhclient.c.usage dhcp-4.1.1/client/dhclient.c
--- dhcp-4.1.1/client/dhclient.c.usage	2010-01-20 17:17:18.000000000 +0100
+++ dhcp-4.1.1/client/dhclient.c	2010-01-20 17:17:18.000000000 +0100
@@ -992,6 +992,10 @@ static void usage()
 		  "[-s server]");
 	log_error("                [-cf config-file] [-lf lease-file]%s",
 		  "[-pf pid-file] [-e VAR=val]");
+	log_error("                [-I <dhcp-client-identifier>] [-B]");
+	log_error("                [-H <host-name> | -F <fqdn.fqdn>] [-timeout <timeout>]");
+	log_error("                [-V <vendor-class-identifier>]");
+	log_error("                [-R <request option list>]");
 	log_fatal("                [-sf script-file] [interface]");
 }
 
