diff -up dhcp-4.2.4-P2/doc/examples/dhcpd-dhcpv6.conf.paths dhcp-4.2.4-P2/doc/examples/dhcpd-dhcpv6.conf
--- dhcp-4.2.4-P2/doc/examples/dhcpd-dhcpv6.conf.paths  2012-08-24 21:11:21.000000000 +0200
+++ dhcp-4.2.4-P2/doc/examples/dhcpd-dhcpv6.conf        2012-10-26 18:10:08.716560729 +0200
@@ -42,7 +42,7 @@ option dhcp6.domain-search "test.example
 option dhcp6.info-refresh-time 21600;
 
 # The path of the lease file
-dhcpv6-lease-file-name "/usr/local/var/db/dhcpd6.leases";
+dhcpv6-lease-file-name "/var/lib/dhcpd/dhcpd6.leases";
 
 # Static definition (must be global)
 host myclient {
diff -up dhcp-4.2.4b1/includes/dhcpd.h.paths dhcp-4.2.4b1/includes/dhcpd.h
--- dhcp-4.2.4b1/includes/dhcpd.h.paths	2012-04-18 11:12:34.000000000 +0200
+++ dhcp-4.2.4b1/includes/dhcpd.h	2012-04-18 11:14:38.541272405 +0200
@@ -1420,7 +1420,7 @@ typedef unsigned char option_mask [16];
 #else /* !DEBUG */
 
 #ifndef _PATH_DHCPD_CONF
-#define _PATH_DHCPD_CONF	"/etc/dhcpd.conf"
+#define _PATH_DHCPD_CONF	"/etc/dhcp/dhcpd.conf"
 #endif /* DEBUG */
 
 #ifndef _PATH_DHCPD_DB
@@ -1442,11 +1442,11 @@ typedef unsigned char option_mask [16];
 #endif /* DEBUG */
 
 #ifndef _PATH_DHCLIENT_CONF
-#define _PATH_DHCLIENT_CONF	"/etc/dhclient.conf"
+#define _PATH_DHCLIENT_CONF	"/etc/dhcp/dhclient.conf"
 #endif
 
 #ifndef _PATH_DHCLIENT_SCRIPT
-#define _PATH_DHCLIENT_SCRIPT	"/sbin/dhclient-script"
+#define _PATH_DHCLIENT_SCRIPT	"/usr/sbin/dhclient-script"
 #endif
 
 #ifndef _PATH_DHCLIENT_PID
