diff -up dhcp-4.2.3-P2/server/dhcpd.c.log_perror dhcp-4.2.3-P2/server/dhcpd.c
--- dhcp-4.2.3-P2/server/dhcpd.c.log_perror	2012-02-22 14:24:57.000000000 +0100
+++ dhcp-4.2.3-P2/server/dhcpd.c	2012-02-22 14:29:09.964576687 +0100
@@ -315,6 +315,7 @@ main(int argc, char **argv) {
 #ifndef DEBUG
 			daemon = 0;
 #endif
+			log_perror = 0;
 		} else if (!strcmp (argv [i], "-d")) {
 #ifndef DEBUG
 			daemon = 0;
