diff -up imlib-1.9.15/imlib-config.in.multilib imlib-1.9.15/imlib-config.in
--- imlib-1.9.15/imlib-config.in.multilib	2004-08-27 13:03:11.000000000 -0300
+++ imlib-1.9.15/imlib-config.in	2013-12-24 23:44:44.149154119 -0200
@@ -7,9 +7,13 @@ gx_libs="@SUPPORT_LIBS@"
 gdk_gx_libs=" @GDK_SUPPORT_LIBS@"
 gdk_libs="@LIBS@"
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
+version=`pkg-config imlib --modversion`
+prefix=`pkg-config imlib --variable prefix`
+exec_prefix=`pkg-config imlib --variable exec_prefix`
 exec_prefix_set=no
+cflags=`pkg-config imlib --cflags`
+libs=`pkg-config imlib --libs`
+libsgdk=`pkg-config imlibgdk --libs`
 
 usage="\
 Usage: imlib-config [--prefix[=DIR]] [--exec-prefix[=DIR]] [--version] [--libs] [--libs-gdk] [--cflags] [--cflags-gdk]"
@@ -43,13 +47,10 @@ while test $# -gt 0; do
       echo $exec_prefix
       ;;
     --version)
-      echo @VERSION@
+      echo $version
       ;;
     --cflags)
-      if test @includedir@ != /usr/include ; then
-        includes=-I@includedir@
-      fi
-      echo $includes @X_CFLAGS@ 
+      echo $cflags
       ;;
     --cflags-gdk)
       if test @includedir@ != /usr/include ; then
@@ -58,12 +59,10 @@ while test $# -gt 0; do
       echo `@GTK_CONFIG@ --cflags` $includes @X_CFLAGS@
       ;;
     --libs)
-      libdirs=-L@libdir@
-      echo $libdirs -lImlib @LDFLAGS@ $gx_libs @X_LIBS@ @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
+      echo $libs
       ;;
     --libs-gdk)
-      libdirs=-L@libdir@
-      echo $libdirs -lgdk_imlib$gdk_gx_libs `@GTK_CONFIG@ --libs` $gdk_libs
+      echo $libsgdk
       ;;
     *)
       echo "${usage}" 1>&2
