diff -up nepomuk-core-4.9.2/services/filewatch/nepomukfilewatch.cpp.isEmpty_crash nepomuk-core-4.9.2/services/filewatch/nepomukfilewatch.cpp
--- nepomuk-core-4.9.2/services/filewatch/nepomukfilewatch.cpp.isEmpty_crash	2012-09-27 19:43:41.000000000 -0500
+++ nepomuk-core-4.9.2/services/filewatch/nepomukfilewatch.cpp	2012-10-01 13:08:47.588285323 -0500
@@ -322,8 +322,8 @@ void Nepomuk2::FileWatch::connectToKDirW
 #ifdef BUILD_KINOTIFY
 void Nepomuk2::FileWatch::slotInotifyWatchUserLimitReached()
 {
-    // we do it the brutal way for now hoping with new kernels and defaults this will never happen
-    delete m_dirWatch;
+    // Delete the KInotify and switch to KDirNotify dbus signals
+    m_dirWatch->deleteLater();
     m_dirWatch = 0;
     connectToKDirWatch();
 }
