From 33d305fc63d88ce163fd3aba41a0dd0e1a74be38 Mon Sep 17 00:00:00 2001
From: Roland McGrath <roland@redhat.com>
Date: Tue, 30 Nov 2010 17:39:55 -0800
Subject: [PATCH] dwfl_module_relocations: Remove over-eager assert.

---
 libdwfl/ChangeLog    |    4 ++++
 libdwfl/derelocate.c |    3 +--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/libdwfl/ChangeLog b/libdwfl/ChangeLog
index dd4eb10..c5746c8 100644
--- a/libdwfl/ChangeLog
+++ b/libdwfl/ChangeLog
@@ -1,3 +1,7 @@
+2010-11-30  Roland McGrath  <roland@redhat.com>
+
+	* derelocate.c (dwfl_module_relocations): Remove over-eager assert.
+
 2010-11-12  Roland McGrath  <roland@redhat.com>
 
 	* libdwflP.h (struct Dwfl_Module): New member main_bias.
diff --git a/libdwfl/derelocate.c b/libdwfl/derelocate.c
index 483b75e..ba6d9db 100644
--- a/libdwfl/derelocate.c
+++ b/libdwfl/derelocate.c
@@ -1,5 +1,5 @@
 /* Recover relocatibility for addresses computed from debug information.
-   Copyright (C) 2005-2009 Red Hat, Inc.
+   Copyright (C) 2005-2010 Red Hat, Inc.
    This file is part of Red Hat elfutils.
 
    Red Hat elfutils is free software; you can redistribute it and/or modify
@@ -240,7 +240,6 @@ dwfl_module_relocations (Dwfl_Module *mod)
 
     case ET_EXEC:
       assert (mod->main.vaddr == mod->low_addr);
-      assert (mod->debug.address_sync == mod->main.address_sync);
       break;
     }
 
-- 
1.7.2.3

