diff --git a/src/tuning/bluetooth.cpp b/src/tuning/bluetooth.cpp
index 9be327e..52a9278 100644
--- a/src/tuning/bluetooth.cpp
+++ b/src/tuning/bluetooth.cpp
@@ -46,8 +46,8 @@
 bt_tunable::bt_tunable(void) : tunable("", 1.0, _("Good"), _("Bad"), _("Unknown"))
 {
 	sprintf(desc, _("Bluetooth device interface status"));
-	strcpy(toggle_bad, "/usr/sbin/hciconfig hci0 up &> /dev/null &");
-	strcpy(toggle_good, "/usr/sbin/hciconfig hci0 down &> /dev/null");
+	toggle_bad = "/usr/sbin/hciconfig hci0 up &> /dev/null &";
+	toggle_good = "/usr/sbin/hciconfig hci0 down &> /dev/null";
 }
 
 
@@ -190,9 +190,9 @@ const char *bt_tunable::toggle_script(void)
 	good = good_bad();
 
 	if (good == TUNE_GOOD) {
-		return toggle_bad;
+		return toggle_bad.c_str();
 	}
-	return toggle_good;
+	return toggle_good.c_str();
 }
 
 
diff --git a/src/tuning/ethernet.cpp b/src/tuning/ethernet.cpp
index 3d36112..f3f0224 100644
--- a/src/tuning/ethernet.cpp
+++ b/src/tuning/ethernet.cpp
@@ -49,11 +49,13 @@ extern void create_all_nics(callback fn);
 
 ethernet_tunable::ethernet_tunable(const char *iface) : tunable("", 0.3, _("Good"), _("Bad"), _("Unknown"))
 {
+	char tmp[4096];
+
 	memset(interf, 0, sizeof(interf));
 	strncpy(interf, iface, sizeof(interf));
 	sprintf(desc, _("Wake-on-lan status for device %s"), iface);
-	sprintf(toggle_good, "ethtool -s %s wol d;", iface);
-
+	snprintf(tmp, 4096, "ethtool -s %s wol d;", iface);
+	toggle_good = tmp;
 }
 
 
@@ -134,7 +136,7 @@ const char *ethernet_tunable::toggle_script(void)
 	good = good_bad();
 
 	if (good != TUNE_GOOD) {
-		return toggle_good;
+		return toggle_good.c_str();
 	}
 
 	return NULL;
diff --git a/src/tuning/runtime.cpp b/src/tuning/runtime.cpp
index 3201fdd..6d6bb98 100644
--- a/src/tuning/runtime.cpp
+++ b/src/tuning/runtime.cpp
@@ -42,6 +42,7 @@ runtime_tunable::runtime_tunable(const char *path, const char *bus, const char *
 {
 	ifstream file;
 	sprintf(runtime_path, "%s/power/control", path);
+	char tmp[4096];
 
 
 	sprintf(desc, _("Runtime PM for %s device %s"), bus, dev);
@@ -77,8 +78,10 @@ runtime_tunable::runtime_tunable(const char *path, const char *bus, const char *
 
 
 	}
-	sprintf(toggle_good, "echo 'auto' > '%s';", runtime_path);
-	sprintf(toggle_bad, "echo 'on' > '%s';", runtime_path);
+	snprintf(tmp, 4096, "echo 'auto' > '%s';", runtime_path);
+	toggle_good = tmp;
+	snprintf(tmp, 4096, "echo 'on' > '%s';", runtime_path);
+	toggle_bad = tmp;
 }
 
 int runtime_tunable::good_bad(void)
@@ -112,10 +115,10 @@ const char *runtime_tunable::toggle_script(void)
 	good = good_bad();
 
 	if (good == TUNE_GOOD) {
-		return toggle_bad;
+		return toggle_bad.c_str();
 	}
 
-	return toggle_good;
+	return toggle_good.c_str();
 }
 
 
diff --git a/src/tuning/tunable.h b/src/tuning/tunable.h
index 3372378..8f90ee2 100644
--- a/src/tuning/tunable.h
+++ b/src/tuning/tunable.h
@@ -26,6 +26,7 @@
 #define _INCLUDE_GUARD_TUNABLE_H
 
 #include <vector>
+#include <string>
 
 #include "../lib.h"
 
@@ -43,8 +44,8 @@ class tunable {
 	char bad_string[128];
 	char neutral_string[128];
 protected:
-	char toggle_good[4096];
-	char toggle_bad[4096];
+	string toggle_good;
+	string toggle_bad;
 public:
 	char desc[4096];
 	double score;
diff --git a/src/tuning/tuningsysfs.cpp b/src/tuning/tuningsysfs.cpp
index 0859a0c..3bfe0e3 100644
--- a/src/tuning/tuningsysfs.cpp
+++ b/src/tuning/tuningsysfs.cpp
@@ -43,11 +43,15 @@
 
 sysfs_tunable::sysfs_tunable(const char *str, const char *_sysfs_path, const char *_target_content) : tunable(str, 1.0, _("Good"), _("Bad"), _("Unknown"))
 {
+	char tmp[4096];
+
 	strcpy(sysfs_path, _sysfs_path);
 	strcpy(target_value, _target_content);
 	bad_value[0] = 0;
-	sprintf(toggle_good, "echo '%s' > '%s';", target_value, sysfs_path);
-	sprintf(toggle_bad, "echo '%s' > '%s';", bad_value, sysfs_path);
+	snprintf(tmp, 4096, "echo '%s' > '%s';", target_value, sysfs_path);
+	toggle_good = tmp;
+	snprintf(tmp, 4096, "echo '%s' > '%s';", bad_value, sysfs_path);
+	toggle_bad = tmp;
 }
 
 int sysfs_tunable::good_bad(void)
@@ -92,11 +96,11 @@ const char *sysfs_tunable::toggle_script(void) {
 
 	if (good == TUNE_GOOD) {
 		if (strlen(bad_value) > 0)
-			return toggle_bad;
+			return toggle_bad.c_str();
 		return NULL;
 	}
 
-	return toggle_good;
+	return toggle_good.c_str();
 }
 
 
diff --git a/src/tuning/tuningusb.cpp b/src/tuning/tuningusb.cpp
index d2a0c11..ec869ec 100644
--- a/src/tuning/tuningusb.cpp
+++ b/src/tuning/tuningusb.cpp
@@ -41,6 +41,7 @@ usb_tunable::usb_tunable(const char *path, const char *name) : tunable("", 0.9,
 	char filename[4096];
 	char vendor[2048];
 	char product[2048];
+	char tmp[4096];
 	string str1, str2;
 	sprintf(usb_path, "%s/power/control", path);
 
@@ -73,8 +74,10 @@ usb_tunable::usb_tunable(const char *path, const char *name) : tunable("", 0.9,
 	else if (strlen(vendor))
 		sprintf(desc, _("Autosuspend for USB device %s [%s]"), vendor, name);
 
-	sprintf(toggle_good, "echo 'auto' > '%s';", usb_path);
-	sprintf(toggle_bad, "echo 'on' > '%s';", usb_path);
+	snprintf(tmp, 4096, "echo 'auto' > '%s';", usb_path);
+	toggle_good = tmp;
+	snprintf(tmp, 4096, "echo 'on' > '%s';", usb_path);
+	toggle_bad = tmp;
 }
 
 int usb_tunable::good_bad(void)
@@ -108,10 +111,10 @@ const char *usb_tunable::toggle_script(void)
 	good = good_bad();
 
 	if (good == TUNE_GOOD) {
-		return toggle_bad;
+		return toggle_bad.c_str();
 	}
 
-	return toggle_good;
+	return toggle_good.c_str();
 }
 
 static void add_usb_callback(const char *d_name)
diff --git a/src/tuning/wifi.cpp b/src/tuning/wifi.cpp
index 77cdfcc..2cc01b8 100644
--- a/src/tuning/wifi.cpp
+++ b/src/tuning/wifi.cpp
@@ -44,11 +44,15 @@ extern "C" {
 
 wifi_tunable::wifi_tunable(const char *_iface) : tunable("", 1.5, _("Good"), _("Bad"), _("Unknown"))
 {
+	char tmp[4096];
+
 	strcpy(iface, _iface);
 	sprintf(desc, _("Wireless Power Saving for interface %s"), iface);
 
-	sprintf(toggle_good, "iw dev %s set power_save on", iface);
-	sprintf(toggle_bad, "iw dev %s set power_save off", iface);
+	snprintf(tmp, 4096, "iw dev %s set power_save on", iface);
+	toggle_good = tmp;
+	snprintf(tmp, 4096, "iw dev %s set power_save off", iface);
+	toggle_bad = tmp;
 }
 
 int wifi_tunable::good_bad(void)
@@ -78,10 +82,10 @@ const char *wifi_tunable::toggle_script(void)
 	good = good_bad();
 
 	if (good == TUNE_GOOD) {
-		return toggle_bad;
+		return toggle_bad.c_str();
 	}
 
-	return toggle_good;
+	return toggle_good.c_str();
 }
 
 void add_wifi_tunables(void)
