diff --git a/src/display.cpp b/src/display.cpp
index b6c9200..c051520 100644
--- a/src/display.cpp
+++ b/src/display.cpp
@@ -127,7 +127,9 @@ void show_tab(unsigned int tab)
 	if (c && strlen(c) > 0)
 		mvwprintw(bottom_line, 0,0, c);
 	else
-		mvwprintw(bottom_line, 0,0,"<ESC> %s | ",_("Exit"));
+		mvwprintw(bottom_line, 0, 0,
+			"<ESC> %s | <TAB> / <Shift + TAB> %s | ", _("Exit"),
+			_("Navigate"));
 
 
 	current_tab = tab;
