diff -up kgpg-4.8.80/kgpgoptions.cpp.gpg2 kgpg-4.8.80/kgpgoptions.cpp
--- kgpg-4.8.80/kgpgoptions.cpp.gpg2	2012-05-23 02:38:31.000000000 +0200
+++ kgpg-4.8.80/kgpgoptions.cpp	2012-06-03 11:58:35.151732768 +0200
@@ -132,6 +132,16 @@ kgpgOptions::~kgpgOptions()
 	delete m_page7;
 }
 
+static QString gpgExecutable()
+{
+  QString gpgExe = KStandardDirs::findExe( "gpg" );
+  if ( gpgExe.isEmpty() )
+    gpgExe = KStandardDirs::findExe( "gpg2" );
+  if ( gpgExe.isEmpty() )
+    return QLatin1String( "gpg" );
+  return gpgExe;
+}
+
 void kgpgOptions::slotChangeHome()
 {
 	QString gpgHome = KFileDialog::getExistingDirectory(m_page4->gpg_home_path->text(), this, i18n("New GnuPG Home Location"));
@@ -154,7 +164,7 @@ void kgpgOptions::slotChangeHome()
 				// start GnuPG so that it will create a config file
 				QString gpgbin = m_page4->kcfg_GpgBinaryPath->text();
 				if (!QFile::exists(gpgbin))
-					gpgbin = QLatin1String( "gpg" );
+					gpgbin = gpgExecutable(); 
 
 				KProcess p;
 				p << gpgbin << QLatin1String( "--homedir" ) << gpgHome << QLatin1String( "--no-tty" ) << QLatin1String( "--list-secret-keys" );
