diff -up amanith/examples/opengl/testshaders/testshaders.pro.DSO amanith/examples/opengl/testshaders/testshaders.pro
--- amanith/examples/opengl/testshaders/testshaders.pro.DSO	2010-02-10 11:39:49.917762524 -0500
+++ amanith/examples/opengl/testshaders/testshaders.pro	2010-02-10 11:40:23.487766249 -0500
@@ -9,9 +9,11 @@ CONFIG        += qt warn_on opengl
 SOURCES        = main.cpp \
                  testshaders.cpp
 
+LIBS          += -lGLEW
+
 win32: RC_FILE = example.rc
 
 # Qt4 support
 contains(DEFINES, USE_QT4) {
     QT += opengl
-}
\ No newline at end of file
+}
