--- fontforge-20100501.orig/fontforge/charview.c	2010-04-15 03:26:28.000000000 +0100
+++ fontforge-20100501/fontforge/charview.c	2011-03-01 15:09:22.821753002 +0000
@@ -8597,7 +8597,7 @@
 static int CVNumForePointsSelected(CharView *cv, BasePoint **bp) {
     SplineSet *spl;
     SplinePoint *test, *first;
-    BasePoint *bps[4];
+    BasePoint *bps[5];
     int i, cnt;
 
     if ( cv->b.drawmode!=dm_fore )
