--- fontforge-20120731-b/fontforge/parsepdf.c.orig	2013-04-15 17:47:51.113706334 +0100
+++ fontforge-20120731-b/fontforge/parsepdf.c	2013-04-15 17:55:55.531682304 +0100
@@ -960,6 +960,8 @@
 	    start = 0;
 	    num = pdf_getinteger(pt,pc);
 	}
+	if ( (pt=PSDictHasEntry(&pc->pdfdict,"DecodeParms"))!=NULL && strstr(pt, "/Predictor")!=NULL)
+return( NULL );
 	if ( (pt=PSDictHasEntry(&pc->pdfdict,"Index"))!=NULL ) {
 	    if ( sscanf(pt,"[%d %d]", &start, &num )!=2 )
 return( NULL );
