diff -up libLASi-1.1.1/examples/Makefile.examples.in.multilib libLASi-1.1.1/examples/Makefile.examples.in
--- libLASi-1.1.1/examples/Makefile.examples.in.multilib	2012-06-12 21:36:04.797674580 -0600
+++ libLASi-1.1.1/examples/Makefile.examples.in	2011-09-21 13:09:55.000000000 -0600
@@ -1,7 +1,7 @@
 CXX = @CMAKE_CXX_COMPILER@
 EXEEXT = @EXEEXT@
 RPATHCMD = @RPATHCMD@
-PKG_ENV = PKG_CONFIG_PATH=@PKG_CONFIG_PATH@
+PKG_ENV =
 
 all: example0.ps example1.ps example2.ps
 
