--- compizconfig-backend-kconfig4-0.9.2.1/src/CMakeLists.txt	2010-11-07 04:51:10.000000000 +0000
+++ compizconfig-backend-kconfig4-0.9.2.1/src/CMakeLists.txt.new	2011-01-19 21:31:14.516969722 +0000
@@ -1,11 +1,10 @@
 find_package(KDE4 REQUIRED)
-find_package (Compiz REQUIRED)
-
 add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
-
 include(KDE4Defaults)
 include(FindPkgConfig)
 include(MacroLibrary)
+
+find_package (Compiz REQUIRED)
 include(CompizCommon)
 include(CompizPackage)
 
