diff -up src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp.ffmpeg src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp
--- src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp.ffmpeg	2012-04-05 17:21:48.913436236 -0300
+++ src/plugins/Input/ffmpeg/decoder_ffmpeg.cpp	2012-04-05 17:26:07.003916129 -0300
@@ -133,9 +133,6 @@ bool DecoderFFmpeg::initialize()
     m_stream->seekable = !input()->isSequential();
     m_stream->max_packet_size = INPUT_BUFFER_SIZE;
 
-    AVFormatParameters ap;
-    memset(&ap, 0, sizeof(ap));
-
     if(avformat_open_input(&ic, m_path.toLocal8Bit().constData(), fmt, 0) != 0)
     {
         qDebug("DecoderFFmpeg: av_open_input_stream() failed");
