From 0be678837d3ab4b74a27b689e4df0f6e92f9aa7c Mon Sep 17 00:00:00 2001
From: Thomas Goyne <tgoyne@gmail.com>
Date: Sun, 6 Sep 2015 09:31:44 -0700
Subject: [PATCH] Fix configure checks

---
 configure.ac |  6 +-----
 2 files changed, 1 insertion(+), 17 deletions(-)

diff --git a/configure.ac b/configure.ac
index 91064e3..0bb736b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -95,7 +95,7 @@ AC_DEFUN([TEST_IS_FFMPEG],
                 /* is ffmpeg */
                 #elif LIBAVCODEC_VERSION_INT < AV_VERSION_INT(53, 47, 100)
                 /* hopefully libav doesn''t add G2M before 53.47.... */
-                (void)AV_CODEC_ID_G2M;
+                (void)CODEC_ID_G2M;
                 #else
                 #error is libav or old enough that the distinction doesn't matter
                 #endif
@@ -121,10 +121,6 @@ AC_DEFUN([TEST_LIBAV],
             ]],[[
                 avcodec_register_all();
                 swscale_version();
-                #ifdef FFMS_USE_FFMPEG_COMPAT
-                int bogus = AV_CODEC_ID_G2M;
-                (void) bogus;
-                #endif
             ]])], [eval $1=yes], [eval $1=no])
         ])
 
