diff -up net-tools-1.60/lib/interface.c.statalias net-tools-1.60/lib/interface.c
--- net-tools-1.60/lib/interface.c.statalias	2011-12-02 16:37:32.258671807 +0100
+++ net-tools-1.60/lib/interface.c	2011-12-02 16:37:33.193660117 +0100
@@ -403,9 +403,11 @@ static int if_readlist_rep(char *target,
 	char *s, name[IFNAMSIZ];
 	s = get_name(name, buf);    
 	get_dev_fields(s, ife);
-	ife->statistics_valid = 1;
 	if (target && !strcmp(target,name))
+	{
+		ife->statistics_valid = 1;
 		break;
+	}
     }
     if (ferror(fh)) {
 	perror(_PATH_PROCNET_DEV);
