diff -up net-tools-1.60/netstat.c.probe net-tools-1.60/netstat.c
--- net-tools-1.60/netstat.c.probe	2008-10-15 12:47:41.000000000 +0200
+++ net-tools-1.60/netstat.c	2008-10-15 12:49:48.000000000 +0200
@@ -1061,6 +1061,11 @@ static void tcp_do_one(int lnr, const ch
 			 (double) time_len / HZ, retr, timeout);
 		break;
 
+	    case 4:
+		snprintf(timers, sizeof(timers), _("probe (%2.2f/%ld/%d)"),
+			 (double) time_len / HZ, retr, timeout);
+		break;
+
 	    default:
 		snprintf(timers, sizeof(timers), _("unkn-%d (%2.2f/%ld/%d)"),
 			 timer_run, (double) time_len / HZ, retr, timeout);
