diff -up net-tools-1.60/man/en_US/mii-tool.8.arbitrary-device-names net-tools-1.60/man/en_US/mii-tool.8
--- net-tools-1.60/man/en_US/mii-tool.8.arbitrary-device-names	2011-03-07 16:41:22.000000000 +0100
+++ net-tools-1.60/man/en_US/mii-tool.8	2011-03-07 16:41:22.000000000 +0100
@@ -16,7 +16,7 @@ mii-tool \- view, manipulate media-indep
 [\fB\-l\fR, \fB\-\-log\fR]
 [\fB\-A\fR, \fB\-\-advertise=\fImedia,...\fR]
 [\fB\-F\fR, \fB\-\-force=\fImedia\fR]
-.RI [ "interface\ ..." ]
+.RI  "interface\ ..."
 
 .SH NOTE
 .P 
@@ -42,9 +42,7 @@ one mode, instead of autonegotiating.  T
 options are mutually exclusive.
 .PP
 The default short output reports the negotiated link speed and link
-status for each interface.  If an interface or interfaces are not
-specified on the command line, then \fBmii\-tool\fR will check any
-available interfaces from \fBeth0\fR through \fBeth7\fR.
+status for each interface.
 .SH OPTIONS
 .TP
 \fB\-v\fR, \fB\-\-verbose\fR
diff -up net-tools-1.60/mii-tool.c.arbitrary-device-names net-tools-1.60/mii-tool.c
--- net-tools-1.60/mii-tool.c.arbitrary-device-names	2011-03-07 16:41:22.000000000 +0100
+++ net-tools-1.60/mii-tool.c	2011-03-07 16:44:52.000000000 +0100
@@ -419,7 +419,7 @@ static void watch_one_xcvr(int skfd, cha
 /*--------------------------------------------------------------------*/
 
 const char *usage =
-"usage: %s [-VvRrwl] [-A media,... | -F media] [interface ...]\n"
+"usage: %s [-VvRrwl] [-A media,... | -F media] <interface> ...\n"
 "       -V, --version               display version information\n"
 "       -v, --verbose               more verbose output\n"
 "       -R, --reset                 reset MII to poweron state\n"
@@ -474,13 +474,10 @@ int main(int argc, char **argv)
 
     /* No remaining args means show all interfaces. */
     if (optind == argc) {
-	ret = 1;
-	for (i = 0; i < MAX_ETH; i++) {
-	    sprintf(s, "eth%d", i);
-	    ret &= do_one_xcvr(skfd, s, 1);
-	}
-	if (ret)
-	    fprintf(stderr, "no MII interfaces found\n");
+	fprintf(stderr, "No interface specified\n");
+	fprintf(stderr, usage, argv[0]);
+	close(skfd);
+	return 2;
     } else {
 	ret = 0;
 	for (i = optind; i < argc; i++) {
@@ -491,15 +488,8 @@ int main(int argc, char **argv)
     if (opt_watch && (ret == 0)) {
 	while (1) {
 	    sleep(1);
-	    if (optind == argc) {
-		for (i = 0; i < MAX_ETH; i++) {
-		    sprintf(s, "eth%d", i);
-		    watch_one_xcvr(skfd, s, i);
-		}
-	    } else {
-		for (i = optind; i < argc; i++)
+	    for (i = optind; i < argc; i++)
 		    watch_one_xcvr(skfd, argv[i], i-optind);
-	    }
 	}
     }
 
