diff -up net-tools-2.0/netstat.c.probe net-tools-2.0/netstat.c
--- net-tools-2.0/netstat.c.probe	2012-10-04 11:36:15.085940621 +0200
+++ net-tools-2.0/netstat.c	2012-10-04 11:36:15.092940511 +0200
@@ -1267,6 +1267,11 @@ static void tcp_do_one(int lnr, const ch
 			 (double) time_len / HZ, retr, timeout);
 		break;
 
+	    case 4:
+		snprintf(timers, sizeof(timers), _("probe (%2.2f/%ld/%d)"),
+			 (double) time_len / HZ, retr, timeout);
+		break;
+
 	    default:
 		snprintf(timers, sizeof(timers), _("unkn-%d (%2.2f/%ld/%d)"),
 			 timer_run, (double) time_len / HZ, retr, timeout);
