diff -up net-tools-1.60/lib/masq_info.c.masqinfo-raw-ip net-tools-1.60/lib/masq_info.c
--- net-tools-1.60/lib/masq_info.c.masqinfo-raw-ip	2000-10-28 12:59:42.000000000 +0200
+++ net-tools-1.60/lib/masq_info.c	2009-09-15 17:05:24.000000000 +0200
@@ -119,7 +119,9 @@ static int read_masqinfo(FILE * f, struc
 	ms->src.sin_family = AF_INET;
 	ms->dst.sin_family = AF_INET;
 
-	if (strcmp("TCP", buf) == 0)
+	if (strcmp("IP", buf) == 0)
+	    ms->proto = "ip";
+	else if (strcmp("TCP", buf) == 0)
 	    ms->proto = "tcp";
 	else if (strcmp("UDP", buf) == 0)
 	    ms->proto = "udp";
