diff -up net-tools-2.0/lib/interface.c.statalias net-tools-2.0/lib/interface.c
--- net-tools-2.0/lib/interface.c.statalias	2012-10-04 11:33:05.490889090 +0200
+++ net-tools-2.0/lib/interface.c	2012-10-04 11:33:05.513888785 +0200
@@ -405,9 +405,11 @@ static int if_readlist_rep(char *target,
 	char *s, name[IFNAMSIZ];
 	s = get_name(name, buf);    
 	get_dev_fields(s, ife);
-	ife->statistics_valid = 1;
 	if (target && !strcmp(target,name))
+	{
+		ife->statistics_valid = 1;
 		break;
+	}
     }
     if (ferror(fh)) {
 	perror(_PATH_PROCNET_DEV);
