diff -up net-tools-1.60/lib/ipx_gr.c.ipx net-tools-1.60/lib/ipx_gr.c
--- net-tools-1.60/lib/ipx_gr.c.ipx	2011-11-21 02:31:57.000000000 +0100
+++ net-tools-1.60/lib/ipx_gr.c	2011-12-01 17:02:53.261805292 +0100
@@ -70,7 +70,7 @@ int IPX_rprint(int options)
 	    continue;
 
 	/* Fetch and resolve the Destination */
-	(void) ap->input(5, net, &sa);
+	(void) ap->input(1, net, &sa);
 	strcpy(net, ap->sprint(&sa, numeric));
 
 	/* Fetch and resolve the Router Net */
diff -up net-tools-1.60/netstat.c.ipx net-tools-1.60/netstat.c
--- net-tools-1.60/netstat.c.ipx	2011-12-01 17:02:53.000000000 +0100
+++ net-tools-1.60/netstat.c	2011-12-01 17:06:26.071144809 +0100
@@ -1582,13 +1582,13 @@ static int ipx_info(void)
 	}
 
 	/* Fetch and resolve the Source */
-	(void) ap->input(4, sad, &sa);
+	(void) ap->input(0, sad, &sa);
 	safe_strncpy(buf, ap->sprint(&sa, flag_not & FLAG_NUM_HOST), sizeof(buf));
 	snprintf(sad, sizeof(sad), "%s:%04X", buf, sport);
 
 	if (!nc) {
 	    /* Fetch and resolve the Destination */
-	    (void) ap->input(4, dad, &sa);
+	    (void) ap->input(0, dad, &sa);
 	    safe_strncpy(buf, ap->sprint(&sa, flag_not & FLAG_NUM_HOST), sizeof(buf));
 	    snprintf(dad, sizeof(dad), "%s:%04X", buf, dport);
 	} else
