diff -up pulseaudio-2.0/src/modules/module-udev-detect.c.udevpath pulseaudio-2.0/src/modules/module-udev-detect.c
--- pulseaudio-2.0/src/modules/module-udev-detect.c.udevpath	2012-05-26 17:04:10.488163452 -0300
+++ pulseaudio-2.0/src/modules/module-udev-detect.c	2012-05-26 17:05:39.930299608 -0300
@@ -123,7 +123,7 @@ static char *card_get_sysattr(const char
         goto finish;
     }
 
-    t = pa_sprintf_malloc("%s/class/sound/card%s", udev_get_sys_path(udev), card_idx);
+    t = pa_sprintf_malloc("/sys/class/sound/card%s", card_idx);
     card = udev_device_new_from_syspath(udev, t);
     pa_xfree(t);
 
@@ -282,7 +282,7 @@ static void verify_access(struct userdat
     pa_assert(u);
     pa_assert(d);
 
-    cd = pa_sprintf_malloc("%s/snd/controlC%s", udev_get_dev_path(u->udev), path_get_card_id(d->path));
+    cd = pa_sprintf_malloc("/dev/snd/controlC%s", path_get_card_id(d->path)); 
     accessible = access(cd, R_OK|W_OK) >= 0;
     pa_log_debug("%s is accessible: %s", cd, pa_yes_no(accessible));
 
@@ -621,7 +621,6 @@ fail:
 }
 
 static int setup_inotify(struct userdata *u) {
-    char *dev_snd;
     int r;
 
     if (u->inotify_fd >= 0)
@@ -632,9 +631,7 @@ static int setup_inotify(struct userdata
         return -1;
     }
 
-    dev_snd = pa_sprintf_malloc("%s/snd", udev_get_dev_path(u->udev));
-    r = inotify_add_watch(u->inotify_fd, dev_snd, IN_ATTRIB|IN_CLOSE_WRITE|IN_DELETE_SELF|IN_MOVE_SELF);
-    pa_xfree(dev_snd);
+    r = inotify_add_watch(u->inotify_fd, "/dev/snd", IN_ATTRIB|IN_CLOSE_WRITE|IN_DELETE_SELF|IN_MOVE_SELF);
 
     if (r < 0) {
         int saved_errno = errno;
diff -up pulseaudio-2.0/src/modules/udev-util.c.udevpath pulseaudio-2.0/src/modules/udev-util.c
--- pulseaudio-2.0/src/modules/udev-util.c.udevpath	2012-05-26 17:05:49.535273535 -0300
+++ pulseaudio-2.0/src/modules/udev-util.c	2012-05-26 17:06:20.289785782 -0300
@@ -180,7 +180,7 @@ int pa_udev_get_info(int card_idx, pa_pr
         goto finish;
     }
 
-    t = pa_sprintf_malloc("%s/class/sound/card%i", udev_get_sys_path(udev), card_idx);
+    t = pa_sprintf_malloc("/sys/class/sound/card%i", card_idx);
     card = udev_device_new_from_syspath(udev, t);
     pa_xfree(t);
 
@@ -277,7 +277,7 @@ char* pa_udev_get_property(int card_idx,
         goto finish;
     }
 
-    t = pa_sprintf_malloc("%s/class/sound/card%i", udev_get_sys_path(udev), card_idx);
+    t = pa_sprintf_malloc("/sys/class/sound/card%i", card_idx);
     card = udev_device_new_from_syspath(udev, t);
     pa_xfree(t);
 
