diff -up pulseaudio-7.0/configure.ac.localedir pulseaudio-7.0/configure.ac
--- pulseaudio-7.0/configure.ac.localedir	2015-09-16 21:47:47.000000000 -0500
+++ pulseaudio-7.0/configure.ac	2015-09-27 18:11:47.719668163 -0500
@@ -110,7 +110,7 @@ AC_SUBST([GETTEXT_PACKAGE])
 AC_DEFINE_UNQUOTED([GETTEXT_PACKAGE],["$GETTEXT_PACKAGE"],[Gettext package])
 
 pulselocaledir='${prefix}/${DATADIRNAME}/locale'
-AX_DEFINE_DIR(PULSE_LOCALEDIR, pulselocaledir, [Gettext locale dir])
+AX_DEFINE_DIR(PULSE_LOCALEDIR, localedir, [Gettext locale dir])
 else
 # workaround till an intltool m4 bug is fixed upstream
 # (https://bugs.launchpad.net/intltool/+bug/904647)
diff -up pulseaudio-7.0/configure.localedir pulseaudio-7.0/configure
--- pulseaudio-7.0/configure.localedir	2015-09-27 18:04:46.606210648 -0500
+++ pulseaudio-7.0/configure	2015-09-27 18:10:17.603152941 -0500
@@ -8939,7 +8939,7 @@ pulselocaledir='${prefix}/${DATADIRNAME}
   exec_prefix_NONE=
   test "x$prefix" = xNONE && prefix_NONE=yes && prefix=$ac_default_prefix
   test "x$exec_prefix" = xNONE && exec_prefix_NONE=yes && exec_prefix=$prefix
-  eval ax_define_dir="\"$pulselocaledir\""
+  eval ax_define_dir="\"$localedir\""
   eval ax_define_dir="\"$ax_define_dir\""
   PULSE_LOCALEDIR="$ax_define_dir"
 
