diff -rupN libqxt.old/src/gui/qxtglobalshortcut_x11.cpp libqxt/src/gui/qxtglobalshortcut_x11.cpp
--- libqxt.old/src/gui/qxtglobalshortcut_x11.cpp	2010-04-18 10:35:25.000000000 -0400
+++ libqxt/src/gui/qxtglobalshortcut_x11.cpp	2010-07-18 15:32:53.000000000 -0400
@@ -72,6 +72,8 @@ quint32 QxtGlobalShortcutPrivate::native
         native |= ControlMask;
     if (modifiers & Qt::AltModifier)
         native |= Mod1Mask;
+    if (modifiers & Qt::MetaModifier)
+        native |= Mod4Mask;
     // TODO: resolve these?
     //if (modifiers & Qt::MetaModifier)
     //if (modifiers & Qt::KeypadModifier)
