--- config.tests/db/main.cpp.db51	2010-11-02 14:22:36.000000000 -0200
+++ config.tests/db/main.cpp	2010-11-02 14:25:09.928367892 -0200
@@ -1,14 +1,14 @@
 #include <db.h>
 
-#if DB_VERSION_MAJOR < 4
-#error db>=4.4 required
+#if DB_VERSION_MAJOR < 5
+#error db>=5.1 required
 #endif
 
-#if DB_VERSION_MINOR < 4
-#error db>=4.4 required
+#if DB_VERSION_MINOR < 1
+#error db>=4.1 required
 #endif
 
-#if DB_VERSION_MINOR < 6
+#if DB_VERSION_MINOR < 1
 #warning db < 4.6, some features will be disabled
 #endif
 
