diff -up kBuild-0.1.5-p1/Config.kmk.pthreads kBuild-0.1.5-p1/Config.kmk
--- kBuild-0.1.5-p1/Config.kmk.pthreads	2010-02-18 08:27:24.287688965 +0100
+++ kBuild-0.1.5-p1/Config.kmk	2010-02-18 08:28:07.505493345 +0100
@@ -282,6 +282,8 @@ ifndef TEMPLATE_BIN_TOOL
  endif
  ifeq ($(KBUILD_TARGET),linux)
   TEMPLATE_BIN_LIBS            += rt
+  TEMPLATE_BIN_LIBS            += pthread
+  TEMPLATE_BIN_LDFLAGS         += -Wl,--as-needed
  endif
   TEMPLATE_BIN_CFLAGS.x86      += -m32
   TEMPLATE_BIN_CFLAGS.sparc32  += -m32
