--- Config.kmk.slk	2010-05-28 23:44:56.333946943 -0300
+++ Config.kmk	2010-05-28 23:44:59.493938397 -0300
@@ -273,9 +273,9 @@
   TEMPLATE_BIN_CFLAGS          += -Wall -Wextra -pedantic -Wno-unused-parameter
   TEMPLATE_BIN_DEFS            += NO_ENUM_BITFIELDS
  endif
- TEMPLATE_BIN_LDFLAGS           = -g
+ TEMPLATE_BIN_LDFLAGS           = -g _SLKLDFLAGS_
  TEMPLATE_BIN_LDFLAGS.profile   = -pg -p
- TEMPLATE_BIN_CFLAGS.release    = -O3
+ TEMPLATE_BIN_CFLAGS.release    = _SLKCFLAGS_
  TEMPLATE_BIN_CFLAGS.profile    = -O3 -pg -p
  ifeq ($(KBUILD_TARGET),freebsd)
   TEMPLATE_BIN_INCS            += $(PATH_GNUMAKE_SRC)/glob /usr/local/include
