--- Config.kmk.orig	2009-01-10 00:24:02.000000000 -0200
+++ Config.kmk	2010-05-28 23:40:02.273173268 -0300
@@ -281,7 +281,7 @@
   TEMPLATE_BIN_INCS            += $(PATH_GNUMAKE_SRC)/glob /usr/local/include
  endif
  ifeq ($(KBUILD_TARGET),linux)
-  TEMPLATE_BIN_LIBS            += rt
+  TEMPLATE_BIN_LIBS            += rt pthread
  endif
   TEMPLATE_BIN_CFLAGS.x86      += -m32
   TEMPLATE_BIN_CFLAGS.sparc32  += -m32
