From 947f108f0860b5d55e4e506fa9521a797257d2b2 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Fri, 11 May 2012 13:51:03 +0200
Subject: [PATCH 10/12] libv4l2: Improve VIDIOC_*_FMT logging

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 lib/libv4l2/log.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/libv4l2/log.c b/lib/libv4l2/log.c
index f7ce06f..377f553 100644
--- a/lib/libv4l2/log.c
+++ b/lib/libv4l2/log.c
@@ -154,6 +154,8 @@ void v4l2_log_ioctl(unsigned long int request, void *arg, int result)
 					(int)fmt->fmt.pix.sizeimage);
 			fprintf(v4l2_log_file, "  colorspace: %d, priv: %x\n",
 					(int)fmt->fmt.pix.colorspace, (int)fmt->fmt.pix.priv);
+		} else {
+			 fprintf(v4l2_log_file, "  type: %d\n", (int)fmt->type);
 		}
 		break;
 	}
-- 
1.7.10

