diff -Nur libsidplay-1.36.57-orig/src/libcfg.h.in libsidplay-1.36.57/src/libcfg.h.in
--- libsidplay-1.36.57-orig/src/libcfg.h.in	2002-03-08 17:00:47.000000000 +0100
+++ libsidplay-1.36.57/src/libcfg.h.in	2007-10-20 12:04:31.000000000 +0200
@@ -39,11 +39,16 @@
  *
  * This option is highly used and tested. A failing little endian system
  * has not been reported so far.  */
-@DIRECT_FIXPOINT@
+#if defined(__i386__)
+#define DIRECT_FIXPOINT
 
 /* This will cause little(big) endian machines to directly access little(big)
  * endian values in memory structures or arrays, disregarding alignment.  */
+#define OPTIMIZE_ENDIAN_ACCESS
+#else
+@DIRECT_FIXPOINT@
 @OPTIMIZE_ENDIAN_ACCESS@
+#endif
 	
 /* --------------------------------------------------------------------------
  * Don't touch these!
