diff -up lensfun-0.2.6/libs/lensfun/CMakeLists.txt.cmake_pkgconfig lensfun-0.2.6/libs/lensfun/CMakeLists.txt
--- lensfun-0.2.6/libs/lensfun/CMakeLists.txt.cmake_pkgconfig	2012-02-12 05:19:39.000000000 -0600
+++ lensfun-0.2.6/libs/lensfun/CMakeLists.txt	2012-06-28 10:05:08.767440482 -0500
@@ -26,3 +26,8 @@ INSTALL(TARGETS lensfun
         RUNTIME DESTINATION ${BINDIR}
         LIBRARY DESTINATION ${LIBDIR}
         ARCHIVE DESTINATION ${LIBDIR})
+
+# pkgconfig support
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/lensfun.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/lensfun.pc @ONLY)
+INSTALL( FILES ${CMAKE_CURRENT_BINARY_DIR}/lensfun.pc DESTINATION ${LIBDIR}/pkgconfig )
+
diff -up lensfun-0.2.6/libs/lensfun/lensfun.pc.cmake.cmake_pkgconfig lensfun-0.2.6/libs/lensfun/lensfun.pc.cmake
--- lensfun-0.2.6/libs/lensfun/lensfun.pc.cmake.cmake_pkgconfig	2012-06-28 10:05:08.767440482 -0500
+++ lensfun-0.2.6/libs/lensfun/lensfun.pc.cmake	2012-06-28 10:37:50.337917472 -0500
@@ -0,0 +1,13 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+bindir=@CMAKE_INSTALL_PREFIX@/@BINDIR@
+libdir=@CMAKE_INSTALL_PREFIX@/@LIBDIR@
+includedir=@CMAKE_INSTALL_PREFIX@/@INCLUDEDIR@
+datadir=@CMAKE_INSTALL_PREFIX@/@DATADIR@
+docdir=@CMAKE_INSTALL_PREFIX@/@DOCDIR@
+
+Name: lensfun
+Description: A photographic lens database and access library
+Version: @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_MICRO@.@VERSION_BUGFIX@
+Requires.private: glib-2.0
+Libs: -L${libdir} -llensfun
+Cflags: -I${includedir}
