diff -up lensfun-0.2.6/CMakeLists.txt.LIB_SUFFIX lensfun-0.2.6/CMakeLists.txt
--- lensfun-0.2.6/CMakeLists.txt.LIB_SUFFIX	2012-06-10 09:19:35.000000000 -0500
+++ lensfun-0.2.6/CMakeLists.txt	2012-06-28 08:32:16.628591449 -0500
@@ -74,7 +74,7 @@ ENDIF()
  
 # define destination dirs (relative to prefix)
 SET(BINDIR bin)
-SET(LIBDIR lib)
+SET(LIBDIR lib${LIB_SUFFIX})
 SET(INCLUDEDIR include)
 SET(DATADIR share/lensfun)
 SET(DOCDIR share/doc/lensfun-${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_MICRO}.${VERSION_BUGFIX})
