diff -up hugin-2014.0.0/CMakeLists.txt.orig hugin-2014.0.0/CMakeLists.txt
--- hugin-2014.0.0/CMakeLists.txt.orig	2014-10-09 19:56:03.642854890 -0300
+++ hugin-2014.0.0/CMakeLists.txt	2014-10-09 19:56:20.792240447 -0300
@@ -271,7 +271,7 @@ IF(WIN32)
 ENDIF(WIN32)
 FIND_PACKAGE(Boost 1.54 COMPONENTS thread date_time regex filesystem iostreams system)
 IF(NOT Boost_FOUND)
-  FIND_PACKAGE(Boost 1.44 COMPONENTS thread date_time regex filesystem iostreams system signals)
+  FIND_PACKAGE(Boost 1.47 COMPONENTS thread date_time regex filesystem iostreams system signals)
   IF (NOT Boost_FOUND)
     IF(WIN32)
       MESSAGE(FATAL_ERROR "Boost not found. Bad BOOST_ROOT? On windows, try editing Boost_root_suffix")
diff -up hugin-2014.0.0/mac/BuildConfig.xcconfig.orig hugin-2014.0.0/mac/BuildConfig.xcconfig
diff -up hugin-2014.0.0/src/tools/ParseExp.cpp.orig hugin-2014.0.0/src/tools/ParseExp.cpp
--- hugin-2014.0.0/src/tools/ParseExp.cpp.orig	2014-10-09 19:52:44.781752471 -0300
+++ hugin-2014.0.0/src/tools/ParseExp.cpp	2014-10-09 19:55:23.300356845 -0300
@@ -34,6 +34,7 @@
 #include <limits>
 #include <iterator>
 
+#define BOOST_SPIRIT_USE_PHOENIX_V3 1
 #include <boost/spirit/version.hpp>
 #if !defined(SPIRIT_VERSION) || SPIRIT_VERSION < 0x2010
 #error "At least Spirit version 2.1 required"
@@ -50,11 +51,9 @@ namespace Parser
 //power function
 struct lazy_pow_
 {
-    template <typename X, typename Y>
-    struct result { typedef X type; };
+    typedef double result_type;
 
-    template <typename X, typename Y>
-    X operator()(X x, Y y) const
+    double operator()(double x, double y) const
     {
         return std::pow(x, y);
     }
@@ -63,11 +62,9 @@ struct lazy_pow_
 // modulus for double values
 struct lazy_mod_
 {
-    template <typename X, typename Y>
-    struct result { typedef X type; };
+    typedef double result_type;
 
-    template <typename X, typename Y>
-    X operator()(X x, Y y) const
+    double operator()(double x, double y) const
     {
         return std::fmod(x,y);
     }
@@ -76,31 +73,27 @@ struct lazy_mod_
 // if statement
 struct lazy_if_
 {
-    template <typename X, typename Y, typename Z>
-    struct result { typedef Y type; };
+    typedef double result_type;
 
-    template <typename X, typename Y, typename Z>
-    X operator()(X x, Y y, Z z) const
+    double operator()(double x, double y, double z) const
     {
-        return x ? y : z;
+        return (std::fabs(x)>1e-5) ? y : z;
     }
 };
 
 // wrapper for unary function
 struct lazy_ufunc_
 {
-    template <typename F, typename A1>
-    struct result { typedef A1 type; };
+    typedef double result_type;
 
-    template <typename F, typename A1>
-    A1 operator()(F f, A1 a1) const
+    double operator()(double (*f)(double), double a1) const
     {
         return f(a1);
     }
 };
 
 // convert rad into deg
-double deg(const double d)
+const double deg(const double d)
 {
     return d*180.0/boost::math::constants::pi<double>();
 };
