diff -up hugin-2013.0.0/src/hugin_base/lensdb/LensDB.cpp.lensfun hugin-2013.0.0/src/hugin_base/lensdb/LensDB.cpp
--- hugin-2013.0.0/src/hugin_base/lensdb/LensDB.cpp.lensfun	2013-10-27 04:19:02.000000000 -0500
+++ hugin-2013.0.0/src/hugin_base/lensdb/LensDB.cpp	2014-11-26 07:52:42.060305692 -0600
@@ -627,7 +627,9 @@ bool LensDB::GetDistortion(double focal,
                 break;
             case LF_DIST_MODEL_NONE:
             case LF_DIST_MODEL_POLY5:
+#ifdef LF_DIST_MODEL_FOV1
             case LF_DIST_MODEL_FOV1:
+#endif
             default:
                 return false;
                 break;
