diff -Nur imms-3.1.0-rc8-orig/clients/bmp/bmpinterface.c imms-3.1.0-rc8/clients/bmp/bmpinterface.c
--- imms-3.1.0-rc8-orig/clients/bmp/bmpinterface.c	2009-02-15 05:05:53.000000000 +0100
+++ imms-3.1.0-rc8/clients/bmp/bmpinterface.c	2010-01-17 22:41:19.000000000 +0100
@@ -125,6 +125,7 @@
     read_config();
 
     configure_win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    gtk_window_set_type_hint(GTK_WINDOW(configure_win), GDK_WINDOW_TYPE_HINT_DIALOG);
     gtk_signal_connect(GTK_OBJECT(configure_win), "destroy",
             GTK_SIGNAL_FUNC(gtk_widget_destroyed), &configure_win);
     gtk_window_set_title(GTK_WINDOW(configure_win), "IMMS Configuration");
