--- Makefile	2012-11-15 12:28:50.000000000 -0500
+++ Makefile.new	2012-11-17 07:33:52.334486249 -0500
@@ -5,7 +5,7 @@
 XDGCONFDIR 	?= /etc/xdg
 
 EXAILEBINDIR  = ${DESTDIR}${PREFIX}/bin
-EXAILELIBDIR 	= ${DESTDIR}${PREFIX}${LIBINSTALLDIR}/exaile
+EXAILELIBDIR 	= ${DESTDIR}${LIBINSTALLDIR}/exaile
 EXAILESHAREDIR 	= ${DESTDIR}${PREFIX}/share/exaile
 EXAILECONFDIR 	= ${DESTDIR}${XDGCONFDIR}/exaile
 
@@ -120,7 +120,7 @@
 	tools/generate-launcher "${DESTDIR}" "${PREFIX}" "${LIBINSTALLDIR}" \
 		"${PYTHON2_CMD}" && \
 	  chmod 755 ${DESTDIR}${PREFIX}/bin/exaile
-	sed "s|\@bindir\@|${EXAILEBINDIR}|" data/org.exaile.Exaile.service.in > \
+	sed "s|\@bindir\@|${PREFIX}/bin|" data/org.exaile.Exaile.service.in > \
 		${DESTDIR}${PREFIX}/share/dbus-1/services/org.exaile.Exaile.service
 	${MAKE} -C plugins install
 
--- tools/generate-launcher	2012-11-15 12:28:50.000000000 -0500
+++ tools/generate-launcher.new	2012-11-17 07:27:37.237377857 -0500
@@ -34,7 +34,7 @@
 fi
 XDG_DATA_DIRS="\${EXAILE_PREFIX}/share/:\${XDG_DATA_DIRS}"
 export XDG_DATA_DIRS
-exec ${PYTHON2_CMD} \${EXAILE_PREFIX}${LIBDIR}/exaile/exaile.py --datadir=\${EXAILE_PREFIX}/share/exaile/data --startgui ${DEFAULTARGS} "\$@"
+exec ${PYTHON2_CMD} ${LIBDIR}/exaile/exaile.py --datadir=\${EXAILE_PREFIX}/share/exaile/data --startgui ${DEFAULTARGS} "\$@"
 exit \$?
 
 EOF
