diff -up qtbase-opensource-src-5.1.1/src/3rdparty/sha3/KeccakF-1600-opt64.c.bigendian qtbase-opensource-src-5.1.1/src/3rdparty/sha3/KeccakF-1600-opt64.c
--- qtbase-opensource-src-5.1.1/src/3rdparty/sha3/KeccakF-1600-opt64.c.bigendian	2013-09-23 11:10:42.000000000 +0200
+++ qtbase-opensource-src-5.1.1/src/3rdparty/sha3/KeccakF-1600-opt64.c	2013-09-23 11:16:02.000000000 +0200
@@ -324,7 +324,7 @@ static void KeccakPermutation(unsigned c
     KeccakPermutationOnWords((UINT64*)state);
 }
 
-#if 0 // Unused in the Qt configuration
+#if (PLATFORM_BYTE_ORDER == IS_BIG_ENDIAN)
 static void fromBytesToWord(UINT64 *word, const UINT8 *bytes)
 {
     unsigned int i;
@@ -445,7 +445,7 @@ static void KeccakAbsorb(unsigned char *
 #endif
 }
 
-#if 0 // Unused in the Qt configuration
+#if (PLATFORM_BYTE_ORDER == IS_BIG_ENDIAN)
 static void fromWordToBytes(UINT8 *bytes, const UINT64 word)
 {
     unsigned int i;
