diff -up qtbase-opensource-src-5.0.2/src/gui/painting/qdrawhelper.cpp.lowmem qtbase-opensource-src-5.0.2/src/gui/painting/qdrawhelper.cpp
--- qtbase-opensource-src-5.0.2/src/gui/painting/qdrawhelper.cpp.lowmem	2013-04-08 19:11:09.000000000 -0500
+++ qtbase-opensource-src-5.0.2/src/gui/painting/qdrawhelper.cpp	2013-04-18 19:34:44.600940784 -0500
@@ -43,8 +43,8 @@
     && (__GNUC__ * 100 + __GNUC_MINOR__ * 10 + __GNUC_PATCHLEVEL__ >= 440)
 // GCC 4.4 supports #pragma GCC optimize and #pragma GCC target
 
-#    if (__GNUC__ * 100 + __GNUC_MINOR__ * 10 + __GNUC_PATCHLEVEL__ < 473)
-// From GCC 4.7.3 onwards, GCC optimize can result in gcc bailing out with OOM
+#    if (__GNUC__ * 100 + __GNUC_MINOR__ * 10 + __GNUC_PATCHLEVEL__ < 472)
+// From GCC 4.7.2 onwards, GCC optimize can result in gcc bailing out with OOM
 #        pragma GCC optimize "O3"
 #    endif
 
