diff -up qtbase-opensource-src-5.1.0/mkspecs/linux-g++-32/qmake.conf.multilib_optflags qtbase-opensource-src-5.1.0/mkspecs/linux-g++-32/qmake.conf
--- qtbase-opensource-src-5.1.0/mkspecs/linux-g++-32/qmake.conf.multilib_optflags	2013-08-27 10:28:26.599972122 -0500
+++ qtbase-opensource-src-5.1.0/mkspecs/linux-g++-32/qmake.conf	2013-08-27 10:30:03.780950078 -0500
@@ -9,6 +9,8 @@ QMAKE_INCREMENTAL_STYLE = sublib
 QMAKE_CFLAGS            = -m32
 QMAKE_LFLAGS            = -m32
 
+QMAKE_CFLAGS_RELEASE   += -O2
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
diff -up qtbase-opensource-src-5.1.0/mkspecs/linux-g++-64/qmake.conf.multilib_optflags qtbase-opensource-src-5.1.0/mkspecs/linux-g++-64/qmake.conf
--- qtbase-opensource-src-5.1.0/mkspecs/linux-g++-64/qmake.conf.multilib_optflags	2013-08-27 10:28:26.600972112 -0500
+++ qtbase-opensource-src-5.1.0/mkspecs/linux-g++-64/qmake.conf	2013-08-27 10:29:49.188103550 -0500
@@ -12,6 +12,8 @@ QMAKE_INCREMENTAL_STYLE = sublib
 QMAKE_CFLAGS            = -m64
 QMAKE_LFLAGS            = -m64
 
+QMAKE_CFLAGS_RELEASE   += -O2
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
diff -up qtbase-opensource-src-5.1.0/mkspecs/linux-g++/qmake.conf.multilib_optflags qtbase-opensource-src-5.1.0/mkspecs/linux-g++/qmake.conf
--- qtbase-opensource-src-5.1.0/mkspecs/linux-g++/qmake.conf.multilib_optflags	2013-08-27 10:28:26.601972101 -0500
+++ qtbase-opensource-src-5.1.0/mkspecs/linux-g++/qmake.conf	2013-08-27 10:29:30.196303285 -0500
@@ -6,6 +6,8 @@ MAKEFILE_GENERATOR      = UNIX
 CONFIG                 += incremental gdb_dwarf_index
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_CFLAGS_RELEASE   += -O2
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
