diff -up qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-32/qmake.conf.multilib_optflags qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-32/qmake.conf
--- qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-32/qmake.conf.multilib_optflags	2012-11-10 13:03:34.000000000 -0600
+++ qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-32/qmake.conf	2012-11-13 13:21:06.055794647 -0600
@@ -9,6 +9,8 @@ QMAKE_INCREMENTAL_STYLE = sublib
 QMAKE_CFLAGS            = -m32
 QMAKE_LFLAGS            = -m32
 
+QMAKE_CFLAGS_RELEASE   += -O2
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
diff -up qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-64/qmake.conf.multilib_optflags qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-64/qmake.conf
--- qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-64/qmake.conf.multilib_optflags	2012-11-10 13:03:34.000000000 -0600
+++ qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++-64/qmake.conf	2012-11-13 13:21:06.056794635 -0600
@@ -12,6 +12,8 @@ QMAKE_INCREMENTAL_STYLE = sublib
 QMAKE_CFLAGS            = -m64
 QMAKE_LFLAGS            = -m64
 
+QMAKE_CFLAGS_RELEASE   += -O2
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
diff -up qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++/qmake.conf.multilib_optflags qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++/qmake.conf
--- qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++/qmake.conf.multilib_optflags	2012-11-10 13:03:34.000000000 -0600
+++ qtbase-opensource-src-5.0.0-beta2/mkspecs/linux-g++/qmake.conf	2012-11-13 13:21:55.171189736 -0600
@@ -6,6 +6,8 @@ MAKEFILE_GENERATOR	= UNIX
 CONFIG += incremental gdb_dwarf_index
 QMAKE_INCREMENTAL_STYLE = sublib
 
+QMAKE_CFLAGS_RELEASE   += -O2
+
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
