diff -up qtbase-opensource-src-5.4.1/src/dbus/qdbusconnection.cpp.qdbusconnection_no_debug qtbase-opensource-src-5.4.1/src/dbus/qdbusconnection.cpp
--- qtbase-opensource-src-5.4.1/src/dbus/qdbusconnection.cpp.qdbusconnection_no_debug	2015-02-16 22:56:38.000000000 -0600
+++ qtbase-opensource-src-5.4.1/src/dbus/qdbusconnection.cpp	2015-04-25 10:48:52.099668703 -0500
@@ -1056,8 +1056,10 @@ public:
         // make sure this connection is running on the main thread
         QCoreApplication *instance = QCoreApplication::instance();
         if (!instance) {
+#ifndef QT_NO_DEBUG
             qWarning("QDBusConnection: %s D-Bus connection created before QCoreApplication. Application may misbehave.",
                      type == SessionBus ? "session" : type == SystemBus ? "system" : "generic");
+#endif
         } else if (QDBusConnectionPrivate::d(*this)) {
             QDBusConnectionPrivate::d(*this)->moveToThread(instance->thread());
         }
