diff -up libofx-0.9.8/lib/ofx_preproc.cpp.foo libofx-0.9.8/lib/ofx_preproc.cpp
--- libofx-0.9.8/lib/ofx_preproc.cpp.foo	2013-08-23 23:45:14.151148860 -0400
+++ libofx-0.9.8/lib/ofx_preproc.cpp	2013-08-23 23:45:53.419525113 -0400
@@ -327,7 +327,7 @@ int ofx_proc_file(LibofxContextPtr ctx,
             free (iconv_buffer);
 #endif
           }
-          cout << s_buffer << "\n";
+          //cout << s_buffer << "\n";
           tmp_file.write(s_buffer.c_str(), s_buffer.length());
         }
 
