diff -up kdebindings-4.6.0/ruby/korundum/modules/krossruby/rubyconfig.h.rbconfig kdebindings-4.6.0/ruby/korundum/modules/krossruby/rubyconfig.h
--- kdebindings-4.6.0/ruby/korundum/modules/krossruby/rubyconfig.h.rbconfig	2011-01-18 19:23:46.000000000 -0200
+++ kdebindings-4.6.0/ruby/korundum/modules/krossruby/rubyconfig.h	2011-01-27 19:18:48.516850877 -0200
@@ -21,12 +21,12 @@
 #define KROSS_RUBY_CONFIG_H
 
 #include <ruby.h>
-#include <env.h>
-#include <rubysig.h>
-#include <node.h>
-#include <version.h>
+//#include <env.h>
+#include <ruby/backward/rubysig.h>
+#include <ruby/node.h>
+#include <ruby/version.h>
 
-#include <st.h>
+#include <ruby/st.h>
 //#include <typeinfo>
 
 #include <kross/core/krossconfig.h>
