diff -urNp stunnel-5.07/tools/stunnel.service.in stunnel-5.07-patched/tools/stunnel.service.in
--- stunnel-5.07/tools/stunnel.service.in	2011-05-02 18:07:34.000000000 -0400
+++ stunnel-5.07-patched/tools/stunnel.service.in	2014-11-04 14:21:15.825057474 -0500
@@ -1,10 +1,11 @@
 [Unit]
 Description=SSL tunnel for network daemons
-After=syslog.target
+After=syslog.target network.target
 
 [Service]
 ExecStart=@prefix@/bin/stunnel
 Type=forking
+PrivateTmp=true
 
 [Install]
 WantedBy=multi-user.target
