--- Makefile.am.link	2010-04-25 01:06:32.000000000 -0300
+++ Makefile.am	2010-04-26 23:41:40.098179677 -0300
@@ -169,8 +169,8 @@
 # but it doesn't for whatever reason (evaluates to empty string).
 # For now, using explicit `ln -s`
 install-exec-hook:
-	rm -fv "$(prefix)/bin/tovid"
-	ln -s "$(DESTDIR)$(bindir)/tovid" "$(prefix)/bin/tovid"
+	rm -fv "$(DESTDIR)$(prefix)/bin/tovid"
+	ln -s "$(bindir)/tovid" "$(DESTDIR)$(prefix)/bin/tovid"
 
 # After uninstalling tovid, remove the python module directories
 uninstall-hook:
