--- Makefile.am.make	2009-11-09 05:14:28.000000000 -0200
+++ Makefile.am	2010-07-29 15:39:57.921121477 -0300
@@ -12,11 +12,11 @@
 
 if GCONF_SCHEMAS_INSTALL
 install-data-local:
-                if test -z "$(DESTDIR)" ; then \
-                        for p in $(schemas_DATA) ; do \
-                                GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/data/$$p ; \
-                        done \
-                fi
+	if test -z "$(DESTDIR)" ; then \
+		for p in $(schemas_DATA) ; do \
+			GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(top_builddir)/data/$$p ; \
+		done \
+	fi
 else
 install-data-local:
 endif
