diff -rupN kmplayer-0.11.3.old/src/kmplayer.desktop kmplayer-0.11.3/src/kmplayer.desktop
--- kmplayer-0.11.3.old/src/kmplayer.desktop	2011-10-11 13:36:18.000000000 -0400
+++ kmplayer-0.11.3/src/kmplayer.desktop	2011-10-19 00:09:47.000000000 -0400
@@ -46,7 +46,6 @@ Comment[sv]=KDE-gränssnitt för Mplayer
 Comment[th]=ระบบติดต่อผู้ใช้ของ MPlayer บน KDE 
 Comment[tr]=MPlayer için KDE arayüzü
 Comment[uk]=KDE-інтерфейс для MPlayer
-Comment[x-test]=xxKDE interface for MPlayerxx
 Comment[zh_CN]=MPlayer 的 KDE 界面
 Comment[zh_TW]=MPlayer 的 KDE 介面
 Terminal=false
@@ -97,7 +96,6 @@ Name[th]=KMPlayer
 Name[tr]=KMPlayer
 Name[ug]=KMPlayer
 Name[uk]=KMPlayer
-Name[x-test]=xxKMPlayerxx
 Name[zh_CN]=KMPlayer
 Name[zh_TW]=KMPlayer
 MimeType=application/ogg;application/smil;application/vnd.ms-asf;application/vnd.rn-realmedia;application/x-kmplayer;application/x-mplayer2;application/x-ogg;application/xspf+xml;audio/basic;audio/m3u;audio/mp4;audio/mpeg;audio/mpegurl;audio/vnd.rn-realaudio;audio/vorbis;audio/x-aiff;audio/x-flac;audio/x-m3u;audio/x-matroska;audio/x-mp2;audio/x-mp3;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/x-mod;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-wav;video/avi;video/flv;video/mediaplayer;video/mp4;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/webm;video/x-avi;video/x-flic;video/x-matroska;video/x-ms-asf;video/x-flv;video/x-msvideo;video/x-ms-wmp;video/x-ms-wmv;video/x-ogm;video/x-theora;x-scheme-handler/mms;x-scheme-handler/pnm;x-scheme-handler/rtspt;x-scheme-handler/rtspu;
@@ -150,6 +148,5 @@ GenericName[th]=โปรแกรมเ
 GenericName[tr]=Medya Oynatıcı
 GenericName[ug]=ۋاسىتە قويغۇچ
 GenericName[uk]=Програвач
-GenericName[x-test]=xxMedia Playerxx
 GenericName[zh_CN]=媒体播放器
 GenericName[zh_TW]=媒體播放器
diff -rupN kmplayer-0.11.3.old/src/kmplayer_part.desktop kmplayer-0.11.3/src/kmplayer_part.desktop
--- kmplayer-0.11.3.old/src/kmplayer_part.desktop	2011-10-11 13:36:18.000000000 -0400
+++ kmplayer-0.11.3/src/kmplayer_part.desktop	2011-10-19 00:11:56.000000000 -0400
@@ -43,7 +43,6 @@ Name[sv]=Inbäddad Mplayer för KDE
 Name[th]=ตัวเล่นสื่อ MPlayer แบบฝังตัวได้สำหรับ KDE
 Name[tr]=KDE için gömülü MPlayer
 Name[uk]=Вбудований MPlayer для KDE
-Name[x-test]=xxEmbedded MPlayer for KDExx
 Name[zh_CN]=KDE 嵌入式 MPlayer
 Name[zh_TW]=KDE 嵌入式 MPlayer
 Comment=Video player plugin (QuickTime, XMovie, Windows Media and RealPlayer compatible)
@@ -88,7 +87,6 @@ Comment[sv]=Insticksprogram för videosp
 Comment[th]=ส่วนเสริมตัวเล่นวิดีโอ (ใช้ด้วยกันได้กับ QuickTime, XMovie, Windows Media และ RealPlayer)
 Comment[tr]=Video oynatıcı eklentisi (QuickTime, XMovie, Windows Media ve RealPlayer uyumlu)
 Comment[uk]=Додаток відтворення відео (QuickTime, XMovie, Windows Media і сумісних з RealPlayer)
-Comment[x-test]=xxVideo player plugin (QuickTime, XMovie, Windows Media and RealPlayer compatible)xx
 Comment[zh_CN]=视频播放器插件(兼容 QuickTime、XMovie、Windows Media 和 RealPlayer)
 Comment[zh_TW]=影像播放器外掛程式（與 QuickTime，XMovie，Windows Media 與 RealPlayer 相容）
 X-KDE-Library=libkmplayerpart
