diff -rupN kmplayer-0.11.2.old/po/CMakeLists.txt kmplayer-0.11.2/po/CMakeLists.txt
--- kmplayer-0.11.2.old/po/CMakeLists.txt	2010-03-04 13:33:08.000000000 -0500
+++ kmplayer-0.11.2/po/CMakeLists.txt	2010-03-05 04:42:49.000000000 -0500
@@ -35,6 +35,5 @@ add_subdirectory(sk)
 add_subdirectory(sv)
 add_subdirectory(tr)
 add_subdirectory(uk)
-add_subdirectory(x-test)
 add_subdirectory(zh_CN)
 add_subdirectory(zh_TW)
diff -rupN kmplayer-0.11.2.old/src/kmplayer.desktop kmplayer-0.11.2/src/kmplayer.desktop
--- kmplayer-0.11.2.old/src/kmplayer.desktop	2010-03-04 13:33:08.000000000 -0500
+++ kmplayer-0.11.2/src/kmplayer.desktop	2010-03-05 04:43:38.000000000 -0500
@@ -38,7 +38,6 @@ Comment[sv]=KDE-gränssnitt för Mplayer
 Comment[th]=ระบบติดต่อผู้ใช้ของ MPlayer บน KDE 
 Comment[tr]=MPlayer için KDE arayüzü
 Comment[uk]=KDE-інтерфейс для MPlayer
-Comment[x-test]=xxKDE interface for MPlayerxx
 Comment[zh_CN]=MPlayer 的 KDE 界面
 Terminal=false
 Name=KMPlayer
@@ -78,7 +77,6 @@ Name[sv]=Kmplayer
 Name[th]=KMPlayer
 Name[tr]=KMPlayer
 Name[uk]=KMPlayer
-Name[x-test]=xxKMPlayerxx
 Name[zh_CN]=KMPlayer
 Name[zh_TW]=KMPlayer
 MimeType=application/ogg;application/smil;application/vnd.ms-asf;application/vnd.rn-realmedia;application/x-kmplayer;application/x-mplayer2;application/x-ogg;application/xspf+xml;audio/basic;audio/m3u;audio/mp4;audio/mpeg;audio/mpegurl;audio/vnd.rn-realaudio;audio/vorbis;audio/x-aiff;audio/x-flac;audio/x-m3u;audio/x-matroska;audio/x-mp2;audio/x-mp3;audio/x-mpegurl;audio/x-ms-asx;audio/x-ms-wma;audio/x-musepack;audio/x-mod;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-wav;video/avi;video/flv;video/mediaplayer;video/mp4;video/mpeg;video/quicktime;video/vnd.rn-realvideo;video/x-avi;video/x-flic;video/x-matroska;video/x-ms-asf;video/x-flv;video/x-msvideo;video/x-ms-wmp;video/x-ms-wmv;video/x-ogm;video/x-theora;uri/mms;uri/pnm;uri/rtspt;uri/rtspu;
@@ -121,6 +119,5 @@ GenericName[sv]=Mediaspelare
 GenericName[th]=โปรแกรมเล่นสื่อ
 GenericName[tr]=Medya Oynatıcı
 GenericName[uk]=Програвач
-GenericName[x-test]=xxMedia Playerxx
 GenericName[zh_CN]=媒体播放器
 GenericName[zh_TW]=媒體播放器
diff -rupN kmplayer-0.11.2.old/src/kmplayer_part.desktop kmplayer-0.11.2/src/kmplayer_part.desktop
--- kmplayer-0.11.2.old/src/kmplayer_part.desktop	2010-03-04 13:33:08.000000000 -0500
+++ kmplayer-0.11.2/src/kmplayer_part.desktop	2010-03-05 04:44:01.000000000 -0500
@@ -35,7 +35,6 @@ Name[sv]=Inbäddad Mplayer för KDE
 Name[th]=ตัวเล่นสื่อ MPlayer แบบฝังตัวได้สำหรับ KDE
 Name[tr]=KDE için gömülmüş MPlayer
 Name[uk]=Вбудований MPlayer для KDE
-Name[x-test]=xxEmbedded MPlayer for KDExx
 Name[zh_CN]=KDE 嵌入式 MPlayer
 Comment=Video player plugin (QuickTime, XMovie, Windows Media and RealPlayer compatible)
 Comment[be]=Утулка прайгравання відэа (сумяшчальны з QuickTime, XMovie, Windows Media і RealPlayer)
@@ -71,7 +70,6 @@ Comment[sv]=Insticksprogram för videosp
 Comment[th]=ส่วนเสริมตัวเล่นวิดีโอ (ใช้ด้วยกันได้กับ QuickTime, XMovie, Windows Media และ RealPlayer)
 Comment[tr]=Video oynatıcı eklentisi (QuickTime, XMovie, Windows Media ve RealPlayer ile uyumlu)
 Comment[uk]=Додаток відтворення відео (QuickTime, XMovie, Windows Media і сумісних з RealPlayer)
-Comment[x-test]=xxVideo player plugin (QuickTime, XMovie, Windows Media and RealPlayer compatible)xx
 Comment[zh_CN]=视频播放器插件(兼容 QuickTime、XMovie、Windows Media 和 RealPlayer)
 X-KDE-Library=libkmplayerpart
 # Keep in sync with the audio types in kmplayer.desktop
