diff -up vte-0.27.4/src/vte.c.grow-up vte-0.27.4/src/vte.c
--- vte-0.27.4/src/vte.c.grow-up	2011-01-11 13:37:37.000000000 -0500
+++ vte-0.27.4/src/vte.c	2011-01-14 11:31:27.159520002 -0500
@@ -8133,6 +8133,24 @@ vte_terminal_init(VteTerminal *terminal)
 
 	_vte_debug_print(VTE_DEBUG_LIFECYCLE, "vte_terminal_init()\n");
 
+#if GTK_CHECK_VERSION (2, 91, 2)
+       static GtkCssProvider *provider;
+       GtkStyleContext *context;
+
+       if (provider == NULL) {
+               provider = gtk_css_provider_new ();
+               gtk_css_provider_load_from_data (GTK_CSS_PROVIDER (provider),
+                                                "VteTerminal {\n"
+                                                "  -VteTerminal-inner-border: 1;\n"
+                                                "}\n", -1, NULL);
+       }
+
+       context = gtk_widget_get_style_context (GTK_WIDGET (terminal));
+       gtk_style_context_add_provider (context,
+                                       GTK_STYLE_PROVIDER (provider),
+                                       GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+#endif
+
 	/* Initialize private data. */
 	pvt = terminal->pvt = G_TYPE_INSTANCE_GET_PRIVATE (terminal, VTE_TYPE_TERMINAL, VteTerminalPrivate);
 
