--- pcmanfm-1.1.0/src/tab-page.c.dot	2012-10-31 03:47:28.000000000 +0900
+++ pcmanfm-1.1.0/src/tab-page.c	2013-04-01 22:30:56.000000000 +0900
@@ -493,7 +493,9 @@
     /* create tab label */
     tab_label = (FmTabLabel*)fm_tab_label_new("");
     gtk_label_set_max_width_chars(tab_label->label, app_config->max_tab_chars);
+#if ! GTK_CHECK_VERSION(3, 0, 0)
     gtk_label_set_ellipsize(tab_label->label, PANGO_ELLIPSIZE_END);
+#endif
     page->tab_label = tab_label;
 
     g_signal_connect(page->folder_view, "sel-changed",
