Index: src/filemanager/file.c
===================================================================
--- src/filemanager/file.c	(revision 0fe4734fd6efb169a4057b84a44c223108b957cb)
+++ src/filemanager/file.c	(revision 54a5139eec69438f8771ffddcff96a2e72a197ff)
@@ -1273,6 +1273,6 @@
 /** Initialize variables for progress bars */
 static FileProgressStatus
-panel_operate_init_totals (FileOperation operation, const WPanel * panel, const char *source,
-                           FileOpContext * ctx, filegui_dialog_type_t dialog_type)
+panel_operate_init_totals (const WPanel * panel, const char *source, FileOpContext * ctx,
+                           filegui_dialog_type_t dialog_type)
 {
     FileProgressStatus status;
@@ -1283,5 +1283,5 @@
 #endif
 
-    if (operation != OP_MOVE && verbose && file_op_compute_totals)
+    if (verbose && file_op_compute_totals)
     {
         ComputeDirSizeUI *ui;
@@ -2379,11 +2379,4 @@
     if (ctx->erase_at_end)
     {
-        /* remove files after move */
-        if (erase_list != NULL)
-        {
-            file_op_context_destroy_ui (ctx);
-            file_op_context_create_ui (ctx, FALSE, FILEGUI_DIALOG_DELETE_ITEM);
-        }
-
         while (erase_list != NULL && return_status != FILE_ABORT)
         {
@@ -2793,12 +2786,10 @@
         if (operation == OP_DELETE)
             dialog_type = FILEGUI_DIALOG_DELETE_ITEM;
+        else if (single_entry && S_ISDIR (selection (panel)->st.st_mode))
+            dialog_type = FILEGUI_DIALOG_MULTI_ITEM;
+        else if (single_entry || force_single)
+            dialog_type = FILEGUI_DIALOG_ONE_ITEM;
         else
-        {
-            dialog_type = ((operation != OP_COPY) || single_entry || force_single)
-                ? FILEGUI_DIALOG_ONE_ITEM : FILEGUI_DIALOG_MULTI_ITEM;
-
-            if (single_entry && (operation == OP_COPY) && S_ISDIR (selection (panel)->st.st_mode))
-                dialog_type = FILEGUI_DIALOG_MULTI_ITEM;
-        }
+            dialog_type = FILEGUI_DIALOG_MULTI_ITEM;
     }
 
@@ -2845,6 +2836,6 @@
             source_with_vpath = vfs_path_append_new (panel->cwd_vpath, source, (char *) NULL);
 #endif /* WITH_FULL_PATHS */
-        if (panel_operate_init_totals
-            (operation, panel, vfs_path_as_str (source_with_vpath), ctx, dialog_type) == FILE_CONT)
+        if (panel_operate_init_totals (panel, vfs_path_as_str (source_with_vpath), ctx, dialog_type)
+            == FILE_CONT)
         {
             if (operation == OP_DELETE)
@@ -2937,5 +2928,5 @@
         }
 
-        if (panel_operate_init_totals (operation, panel, NULL, ctx, dialog_type) == FILE_CONT)
+        if (panel_operate_init_totals (panel, NULL, ctx, dialog_type) == FILE_CONT)
         {
             /* Loop for every file, perform the actual copy operation */
