Index: doc/man/mc.1.in
===================================================================
--- doc/man/mc.1.in	(revision 6da546bcf3a09b7710bc3b69d38e736e3273c34b)
+++ doc/man/mc.1.in	(revision 446748619b5cb77ea63360a81bb16cd2dd2caff7)
@@ -173,9 +173,10 @@
 .PP
 If both paths are specified, the first path name is the directory to show
-in the left panel; the second path name is the directory to be shown in
-the right panel.
+in the active panel; the second path name is the directory to be shown in
+the other panel.
 .PP
 If one path is specified, the path name is the directory to show
-in the active panel; current directory is shown in the passive panel.
+in the active panel; value of "other_dir" from panels.ini is the directory
+to be shown in the passive panel.
 .PP
 If no paths are specified, current directory is shown in the active panel;
Index: doc/man/ru/mc.1.in
===================================================================
--- doc/man/ru/mc.1.in	(revision 6da546bcf3a09b7710bc3b69d38e736e3273c34b)
+++ doc/man/ru/mc.1.in	(revision 446748619b5cb77ea63360a81bb16cd2dd2caff7)
@@ -150,11 +150,12 @@
 .PP
 Если в командной строке указаны два каталога, первый будет отображаться
-в левой панели, второй \- в правой.
+в активной панели, второй \- в другой панели.
 .PP
 Если указан только один каталог, он будет отображён в активной панели,
-во второй панели будет отображён текущий каталог.
+во второй панели будет отображён каталог, указанный в параметре "other_dir"
+в файле panels.ini.
 .PP
 Если каталоги не указаны, в активной панели будет отображён текущий каталог,
-а во второй панели \- каталог, указанный в параметре "other_dir" в файла panels.ini.
+а во второй панели \- каталог, указанный в параметре "other_dir" в файле panels.ini.
 .\"NODE "Overview"
 .SH "Главное окно программы"
Index: src/filemanager/midnight.c
===================================================================
--- src/filemanager/midnight.c	(revision b451e8d13ddf89f1b1a00b1c2f240e4deaf8abe7)
+++ src/filemanager/midnight.c	(revision 446748619b5cb77ea63360a81bb16cd2dd2caff7)
@@ -585,13 +585,13 @@
      * 'mc' (no arguments):            mc_run_param0 == NULL, mc_run_param1 == NULL
      *                                 active panel uses current directory
-     *                                 passive panel uses "other_dir" from ini
+     *                                 passive panel uses "other_dir" from panels.ini
      *
      * 'mc dir1 dir2' (two arguments): mc_run_param0 != NULL, mc_run_param1 != NULL
-     *                                 left panel uses mc_run_param0
-     *                                 right panel uses mc_run_param1
+     *                                 active panel uses mc_run_param0
+     *                                 passive panel uses mc_run_param1
      *
      * 'mc dir1' (single argument):    mc_run_param0 != NULL, mc_run_param1 == NULL
-     *                                 active panel uses current directory
-     *                                 passive panel uses mc_run_param0
+     *                                 active panel uses mc_run_param0
+     *                                 passive panel uses "other_dir" from panels.ini
      */
 
@@ -620,6 +620,6 @@
         {
             /* one argument */
-            current_dir = NULL; /* assume current dir */
-            other_dir = (char *) mc_run_param0;
+            current_dir = (char *) mc_run_param0;
+            other_dir = saved_other_dir;        /* from ini */
         }
     }
@@ -641,12 +641,12 @@
         {
             /* two arguments */
-            current_dir = mc_run_param1;
-            other_dir = (char *) mc_run_param0;
+            current_dir = (char *) mc_run_param0;
+            other_dir = mc_run_param1;
         }
         else                    /* mc_run_param0 != NULL && mc_run_param1 == NULL */
         {
             /* one argument */
-            current_dir = NULL; /* assume current dir */ ;
-            other_dir = (char *) mc_run_param0;
+            current_dir = (char *) mc_run_param0;
+            other_dir = saved_other_dir;        /* from ini */
         }
     }
