diff -up mc-4.8.10/src/vfs/cpio/cpio.c.cpiosegfault mc-4.8.10/src/vfs/cpio/cpio.c
--- mc-4.8.10/src/vfs/cpio/cpio.c.cpiosegfault	2013-08-03 07:40:32.068958650 +0200
+++ mc-4.8.10/src/vfs/cpio/cpio.c	2013-08-03 07:42:11.926345838 +0200
@@ -784,7 +784,7 @@ cpio_super_same (const vfs_path_element_
         return 0;
 
     /* Has the cached archive been changed on the disk? */
-    if (((cpio_super_data_t *) parc->data)->st.st_mtime < archive_stat->st_mtime)
+    if (parc->data && ((cpio_super_data_t *) parc->data)->st.st_mtime < archive_stat->st_mtime)
     {
         /* Yes, reload! */
         (*vfs_cpiofs_ops.free) ((vfsid) parc);
