diff -up mc-4.7.5-pre1/src/editor/editwidget.c.mcviewsegfault mc-4.7.5-pre1/src/editor/editwidget.c
--- mc-4.7.5-pre1/src/editor/editwidget.c.mcviewsegfault	2010-12-03 09:53:38.000000000 +0100
+++ mc-4.7.5-pre1/src/editor/editwidget.c	2010-12-08 12:33:35.943724419 +0100
@@ -387,7 +387,7 @@ edit_file (const char *_file, int line)
 const char *
 edit_get_file_name (const WEdit * edit)
 {
-    return edit->filename;
+    return edit?edit->filename:NULL;
 }
 
 /* --------------------------------------------------------------------------------------------- */
