diff -up mc-4.8.10/lib/widget/widget-common.h.widgetsegfault mc-4.8.10/lib/widget/widget-common.h
--- mc-4.8.10/lib/widget/widget-common.h.widgetsegfault	2013-08-03 07:42:45.329299880 +0200
+++ mc-4.8.10/lib/widget/widget-common.h	2013-08-03 07:44:22.162755951 +0200
@@ -159,7 +159,7 @@ gboolean mouse_global_in_widget (const G
 static inline cb_ret_t
 send_message (void *w, void *sender, widget_msg_t msg, int parm, void *data)
 {
-    return WIDGET (w)->callback (WIDGET (w), WIDGET (sender), msg, parm, data);
+    return w ? WIDGET (w)->callback (WIDGET (w), WIDGET (sender), msg, parm, data) : 1;
 }
 
 /* --------------------------------------------------------------------------------------------- */
