library location in Fedora depends on architecture (lib/lib64)

diff -u -uNPrp fatrat-1.1.3/config.h.in fatrat-1.1.3.new/config.h.in
--- fatrat-1.1.3/config.h.in	2010-09-06 11:43:46.000000000 +0200
+++ fatrat-1.1.3.new/config.h.in	2010-09-27 12:14:55.596653999 +0200
@@ -19,7 +19,7 @@
 #cmakedefine LIBTORRENT_0_15
 
 #define DATA_LOCATION "${CMAKE_INSTALL_PREFIX}/share/fatrat"
-#define PLUGIN_LOCATION "${CMAKE_INSTALL_PREFIX}/lib/fatrat/plugins"
+#define PLUGIN_LOCATION "${CMAKE_INSTALL_LIBDIR}/fatrat/plugins"
 
 #ifndef _LARGEFILE64_SOURCE
 #	define _LARGEFILE64_SOURCE
