From 7f5cd68ce2643444cf977e6ad7d5465cbff21ea3 Mon Sep 17 00:00:00 2001
From: Lu Wang <WANG Lu coolwanglu@gmail.com>
Date: Wed, 4 Sep 2013 14:52:27 +0800
Subject: [PATCH] do not free variables allocated by alloca()

---
 handlers.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/handlers.c b/handlers.c
index 7cb7e42..dbee1ca 100644
--- a/handlers.c
+++ b/handlers.c
@@ -170,7 +170,8 @@ bool handler__set(globals_t * vars, char **argv, unsigned argc)
     if (INTERRUPTABLE()) {
         
         /* control returns here when interrupted */
-        free(settings);
+// settings is allocated with alloca, do not free it
+//        free(settings);
         detach(vars->target);
         ENDINTERRUPTABLE();
         return true;
@@ -1084,12 +1085,14 @@ bool handler__shell(globals_t * vars, char **argv, unsigned argc)
 
     /* finally execute command */
     if (system(command) == -1) {
-        free(command);
+// command is allocated with alloca, do not free it
+//        free(command);
         show_error("system() failed, command was not executed.\n");
         return false;
     }
 
-    free(command);
+// command is allocated with alloca, do not free it
+//    free(command);
     return true;
 }
 
-- 
1.8.3.1

