diff -up xf86-video-ati-20160518/src/drmmode_display.c.da xf86-video-ati-20160518/src/drmmode_display.c
--- xf86-video-ati-20160518/src/drmmode_display.c.da	2016-05-18 06:14:02.886748541 +1000
+++ xf86-video-ati-20160518/src/drmmode_display.c	2016-05-18 06:14:16.236094662 +1000
@@ -44,7 +44,7 @@
 #include "radeon_reg.h"
 
 #ifdef RADEON_PIXMAP_SHARING
-#include <dri.h>
+#include <dri2.h>
 #endif
 
 #include "drmmode_display.h"
diff -up xf86-video-ati-20160518/src/radeon_probe.c.da xf86-video-ati-20160518/src/radeon_probe.c
--- xf86-video-ati-20160518/src/radeon_probe.c.da	2016-05-18 06:13:57.751615418 +1000
+++ xf86-video-ati-20160518/src/radeon_probe.c	2016-05-18 06:14:10.043934112 +1000
@@ -48,7 +48,7 @@
 #include "xf86.h"
 
 #include "xf86drmMode.h"
-#include "dri.h"
+#include "dri2.h"
 
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <xf86_OSproc.h>
