--- src/gcm-session.c.orig	2010-07-20 05:36:46.000000000 -0300
+++ src/gcm-session.c	2011-02-12 01:50:31.170357606 -0200
@@ -145,7 +145,7 @@ gcm_session_notify_recalibrate (const gc
 	NotifyNotification *notification;
 
 	/* show a bubble */
-	notification = notify_notification_new (title, message, GCM_STOCK_ICON, NULL);
+	notification = notify_notification_new (title, message, GCM_STOCK_ICON);
 	notify_notification_set_timeout (notification, GCM_SESSION_NOTIFY_TIMEOUT);
 	notify_notification_set_urgency (notification, NOTIFY_URGENCY_LOW);
 
