--- videowrapper.cpp.orig	2013-07-30 08:46:19.000000000 -0300
+++ videowrapper.cpp	2013-07-30 18:42:29.158909551 -0300
@@ -38,14 +38,9 @@ void videowrapper::delete_videowrapper()
 int videowrapper::load_video(QString filename)
 {
     // *** OUVERTURE DU FICHIER VIDEO ***
-    #ifdef avformat_open_input
         if (avformat_open_input(&pFormatCtx,filename.toAscii().data(),NULL,NULL)!=0)
         {
-    #else
-        if (av_open_input_file(&pFormatCtx,filename.toAscii().data(),NULL,0,NULL)!=0)
-        {
-    #endif
-        qDebug() << "CAN'T OPEN VIDEO FILE : " << filename;
+         qDebug() << "CAN'T OPEN VIDEO FILE : " << filename;
         return 1;
     }
 
@@ -91,13 +86,8 @@ int videowrapper::load_video(QString fil
 
     // *** OUVERTURE DU CODEC ***
 
-    #ifdef av_find_stream_info
-    if(avcodec_open(pCodecCtx, pCodec)<0)
-    {
-    #else
     if (avcodec_open2(pCodecCtx, pCodec, NULL) < 0)
     {
-    #endif
         qDebug() << "CAN'T FIND VIDEO CODEC";
         return 1;
     }
