Index: 2ManDVD/mainfrm.cpp
===================================================================
--- 2ManDVD.orig/mainfrm.cpp
+++ 2ManDVD/mainfrm.cpp
@@ -1045,8 +1045,8 @@ mainfrm::mainfrm(QWidget *parent) : QMai
 	outext << "growisofs" << endl;
 	outext << "mpeg2enc" << endl;
         outext << "jpegtopnm /dev/null" << endl;
-        outext << "cat " + "/usr/share/2mandvd//fake.pl" << endl;
-        outext << "rm -f " + homeDir.path() + "/.mdvdep.sh" << endl;
+        outext << "cat " << "/usr/share/2mandvd//fake.pl" << endl;
+        outext << "rm -f " << homeDir.path()<<"/.mdvdep.sh" << endl;
 	outext << "exit" << endl;
 	wmyficext.close();
 	QString programf = "bash";
