--- videowrapper.cpp.ffmpeg	2012-05-31 18:45:52.846389443 -0300
+++ videowrapper.cpp	2012-05-31 18:46:54.929065673 -0300
@@ -38,14 +38,9 @@ void videowrapper::delete_videowrapper()
 int videowrapper::load_video(QString filename)
 {
     // *** OUVERTURE DU FICHIER VIDEO ***
-    #ifdef avformat_open_input
         if (avformat_open_input(&pFormatCtx,filename.toAscii().data(),NULL,NULL)!=0)
         {
-    #else
-        if (av_open_input_file(&pFormatCtx,filename.toAscii().data(),NULL,0,NULL)!=0)
-        {
-    #endif
-        qDebug() << "CAN'T OPEN VIDEO FILE : " << filename;
+         qDebug() << "CAN'T OPEN VIDEO FILE : " << filename;
         return 1;
     }
 
