From fb1b02802aa368f43c8d0a871e017c14cb39b180 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Fri, 4 Apr 2014 08:11:15 +0200
Subject: [PATCH 075/152] Add missing #endif

---
 include/grub/libgcc.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/grub/libgcc.h b/include/grub/libgcc.h
index 97233d7..8e93b67 100644
--- a/include/grub/libgcc.h
+++ b/include/grub/libgcc.h
@@ -42,6 +42,7 @@ void EXPORT_FUNC (__bswapsi2) (void);
 # ifdef HAVE___BSWAPDI2
 void EXPORT_FUNC (__bswapdi2) (void);
 # endif
+#endif
 
 #ifdef HAVE__RESTGPR_14_X
 void EXPORT_FUNC (_restgpr_14_x) (void);
-- 
1.9.3

