--- util/grub.d/10_linux.in.orig	2010-12-25 10:57:11.000000000 -0200
+++ util/grub.d/10_linux.in	2011-01-29 02:40:35.211863649 -0200
@@ -111,7 +111,7 @@ EOF
 EOF
 }
 
-list=`for i in /boot/vmlinuz-* /boot/vmlinux-* /vmlinuz-* /vmlinux-* ; do
+list=`for i in /boot/kernel-* /boot/vmlinuz-* /boot/vmlinux-* /vmlinuz-* /vmlinux-* ; do
         if grub_file_is_not_garbage "$i" ; then echo -n "$i " ; fi
       done`
 prepare_boot_cache=
