From 8b424dd42b64453e50a49a4c6fc455584f931b46 Mon Sep 17 00:00:00 2001
From: Fedora Ninjas <pjones@fedoraproject.org>
Date: Thu, 23 Jun 2011 00:08:03 +0400
Subject: [PATCH] Workaround for 'variable set but not used' issue

Signed-off-by: Fedora Ninjas <pjones@fedoraproject.org>
---
 grub-core/lib/relocator.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/grub-core/lib/relocator.c b/grub-core/lib/relocator.c
index dbd5fe4..606b545 100644
--- a/grub-core/lib/relocator.c
+++ b/grub-core/lib/relocator.c
@@ -1010,6 +1010,9 @@ malloc_in_range (struct grub_relocator *rel,
 		      = ALIGN_UP (alloc_end,
 				  GRUB_RELOCATOR_FIRMWARE_REQUESTS_QUANT);
 
+		      grub_dprintf ("relocator", "requesting %lx-%lx\n",
+				    (unsigned long) fstart,
+				    (unsigned long) fend);
 #if GRUB_RELOCATOR_HAVE_LEFTOVERS
 		    {
 		      struct grub_relocator_fw_leftover *lo1 = NULL;
-- 
1.7.5.1

