--- util/grub.d/10_linux.in.chinfo	2011-04-18 18:18:49.000000000 -0300
+++ util/grub.d/10_linux.in	2011-04-22 13:54:51.251426450 -0300
@@ -31,7 +31,7 @@ CLASS="--class gnu-linux --class gnu --c
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  OS="${GRUB_DISTRIBUTOR}"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr 'A-Z' 'a-z' | cut -d' ' -f1) ${CLASS}"
 fi
 
