--- util/grub.d/10_linux.in.chinfo	2011-01-29 02:42:20.000000000 -0200
+++ util/grub.d/10_linux.in	2011-01-29 02:43:17.719252340 -0200
@@ -31,7 +31,7 @@ CLASS="--class gnu-linux --class gnu --c
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
 else
-  OS="${GRUB_DISTRIBUTOR} GNU/Linux"
+  OS="${GRUB_DISTRIBUTOR}"
   CLASS="--class $(echo ${GRUB_DISTRIBUTOR} | tr '[A-Z]' '[a-z]' | cut -d' ' -f1) ${CLASS}"
 fi
 
