--- util/grub-mkconfig_lib.in.orig	2012-10-05 23:18:39.552518547 -0300
+++ util/grub-mkconfig_lib.in	2012-10-05 23:19:11.184839452 -0300
@@ -209,6 +209,7 @@ grub_file_is_not_garbage ()
   if test -f "$1" ; then
     case "$1" in
       *.dpkg-*) return 1 ;; # debian dpkg
+      *.new) return 1 ;; # slackware new
       *.rpmsave|*.rpmnew) return 1 ;;
       README*|*/README*)  return 1 ;; # documentation
     esac
