From d50a99c5ceeb7107f624c5d3238d37509b2217a8 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 29 Jan 2014 08:37:43 +0100
Subject: [PATCH] NEWS: update for version 036

---
 NEWS | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/NEWS b/NEWS
index 5b88600..ccf279f 100644
--- a/NEWS
+++ b/NEWS
@@ -1,3 +1,7 @@
+dracut-036
+==========
+- fixed skipcpio signature checking
+
 dracut-035
 ==========
 - changed dracut tarball compression to xz
