From dece5ee1f530335d2c2c10472733f838f6027434 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 1 Aug 2012 16:35:46 +0200
Subject: [PATCH] dracut.conf.5.asc: s#dracut/conf.d#dracut.conf.d#g

---
 dracut.conf.5.asc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dracut.conf.5.asc b/dracut.conf.5.asc
index 89bbf61..f3a7de6 100644
--- a/dracut.conf.5.asc
+++ b/dracut.conf.5.asc
@@ -127,8 +127,8 @@ _/etc/dracut.conf_::
     Old configuration file. You better use your own file in
     _/etc/dracut/conf.d/_.
 
-_/etc/dracut/conf.d/_::
-    Any _/etc/dracut/conf.d/*.conf_ file can overwrite the values in
+_/etc/dracut.conf.d/_::
+    Any _/etc/dracut.conf.d/*.conf_ file can overwrite the values in
     _/etc/dracut.conf_. The configuration files are read in alphanumerical
     order.
 
