From f57850d7e8800cc28c766c77ad0a881a9b6a1f36 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 30 Jan 2014 14:27:18 +0100
Subject: [PATCH] dracut-functions.sh: also search in the updates directory

---
 dracut-functions.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index 7cfa097..c766ca3 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -1551,7 +1551,7 @@ find_kernel_modules_by_path () {
     _OLDIFS=$IFS
     IFS=:
     while read a rest; do
-        [[ $a = */$1/* ]] || continue
+        [[ $a = */$1/* ]] || [[ $a = */updates/* ]] || continue
         printf "%s\n" "$srcmods/$a"
     done < "$srcmods/modules.dep"
     IFS=$_OLDIFS
