From c8608373039e66768d34bcf8d43da48d68c5be2a Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 31 Jan 2014 15:33:05 +0100
Subject: [PATCH] base/rdsosreport.sh: add dracut version to rdsosreport

---
 modules.d/99base/rdsosreport.sh | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/modules.d/99base/rdsosreport.sh b/modules.d/99base/rdsosreport.sh
index 5e11166..3e2ac87 100755
--- a/modules.d/99base/rdsosreport.sh
+++ b/modules.d/99base/rdsosreport.sh
@@ -8,13 +8,7 @@ exec >/run/initramfs/rdsosreport.txt 2>&1
 
 set -x
 
-cat /proc/self/mountinfo
-cat /proc/mounts
-
-blkid
-blkid -o udev
-
-ls -l /dev/disk/by*
+cat /lib/dracut/dracut-*
 
 cat /proc/cmdline
 
@@ -26,6 +20,14 @@ for _i in /etc/cmdline.d/*.conf; do
     cat $_i
 done
 
+cat /proc/self/mountinfo
+cat /proc/mounts
+
+blkid
+blkid -o udev
+
+ls -l /dev/disk/by*
+
 for _i in /etc/conf.d/*.conf; do
     [ -f "$_i" ] || break
     echo $_i
