From 8d21728942b52f1fe258ab63acfa5a0542d8b1fc Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 5 Feb 2014 13:58:33 +0100
Subject: [PATCH] systemd: add sys-kernel-config.mount

---
 modules.d/98systemd/module-setup.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules.d/98systemd/module-setup.sh b/modules.d/98systemd/module-setup.sh
index d625ea0..036f1c1 100755
--- a/modules.d/98systemd/module-setup.sh
+++ b/modules.d/98systemd/module-setup.sh
@@ -44,6 +44,7 @@ install() {
         $systemdutildir/systemd-modules-load \
         $systemdutildir/systemd-vconsole-setup \
         $systemdutildir/system-generators/systemd-fstab-generator \
+        \
         $systemdsystemunitdir/cryptsetup.target \
         $systemdsystemunitdir/emergency.target \
         $systemdsystemunitdir/sysinit.target \
@@ -72,6 +73,9 @@ install() {
         $systemdsystemunitdir/timers.target \
         $systemdsystemunitdir/paths.target \
         $systemdsystemunitdir/umount.target \
+        \
+        $systemdsystemunitdir/sys-kernel-config.mount \
+        \
         $systemdsystemunitdir/kmod-static-nodes.service \
         $systemdsystemunitdir/systemd-tmpfiles-setup-dev.service \
         $systemdsystemunitdir/systemd-ask-password-console.path \
