--- xfce4-genmon-plugin-3.2/panel-plugin/Makefile.am	2007-11-23 15:32:40.000000000 +0000
+++ xfce4-genmon-plugin-3.2/panel-plugin/Makefile.am.new	2011-01-14 18:13:18.662136456 +0000
@@ -3,10 +3,12 @@
 
 xfce4_genmon_plugin_CFLAGS =					\
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"			\
-	@LIBXFCE4PANEL_CFLAGS@
+	@LIBXFCE4PANEL_CFLAGS@					\
+	@LIBXFCEGUI4_CFLAGS@
 
 xfce4_genmon_plugin_LDADD =					\
-	@LIBXFCE4PANEL_LIBS@
+	@LIBXFCE4PANEL_LIBS@					\
+	@LIBXFCEGUI4_LIBS@
 
 xfce4_genmon_plugin_SOURCES =	\
 	main.c			\
--- xfce4-genmon-plugin-3.2/configure.ac	2011-01-14 18:18:53.410130665 +0000
+++ xfce4-genmon-plugin-3.2/configure.ac.new	2011-01-14 18:18:58.431354545 +0000
@@ -32,6 +32,7 @@
 
 dnl configure the panel plugin
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.22])
+XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.3.22])
 
 dnl Translations
 XDT_I18N([ca cs en_GB eu fr pl pt_PT sq])
