--- src/appearancegtk2.cpp.mine	2013-03-01 04:44:37.000000000 -0300
+++ src/appearancegtk2.cpp	2013-03-11 22:56:17.394059505 -0300
@@ -110,6 +110,7 @@ bool AppearanceGTK2::saveSettings(const
     flow << "gtk-toolbar-style=" << m_settings["toolbar_style"] << "\n";
     flow << "gtk-menu-images=" << m_settings["show_icons_buttons"] << "\n";
     flow << "gtk-button-images=" << m_settings["show_icons_menus"] << "\n";
+    flow << "\n" << "include \"" << QDir::homePath() << "/.gtkrc-2.0.mine\"\n";
     
     //we're done with the  ~/.gtk-2.0 file
     gtkrc.close();
