--- lxqt-config-0.9.0/lxqt-config-monitor/lxqt-config-monitor.desktop.in.orig	2015-02-08 12:20:02.101279529 -0200
+++ lxqt-config-0.9.0/lxqt-config-monitor/lxqt-config-monitor.desktop.in	2015-02-08 12:20:14.642280113 -0200
@@ -5,7 +5,7 @@
 Comment=Configure monitors
 Exec=lxqt-config-monitor
 Icon=preferences-desktop-display
-Categories=Settings;DesktopSettings;Qt;LXQt;;HardwareSettings;
+Categories=Settings;DesktopSettings;Qt;LXQt;HardwareSettings;
 OnlyShowIn=LXQt;
 
 #TRANSLATIONS_DIR=translations
