diff -ur clucene-core-0.9.21b/src/CLucene/debug/lucenebase.h clucene-core-0.9.21b-gcc48/src/CLucene/debug/lucenebase.h
--- clucene-core-0.9.21b/src/CLucene/debug/lucenebase.h	2008-10-23 19:44:35.000000000 +0200
+++ clucene-core-0.9.21b-gcc48/src/CLucene/debug/lucenebase.h	2014-10-25 20:54:09.000000000 +0200
@@ -58,7 +58,11 @@
 		__cl_refcount--;
 		return __cl_refcount;
 	}
-    virtual ~LuceneBase(){};
+    virtual ~LuceneBase()
+#if __cplusplus >= 201103L
+        noexcept(false)
+#endif
+        {};
 };
 
 class LuceneVoidBase{
