diff -ur clucene-core-0.9.21b/src/CLucene/queryParser/Lexer.cpp clucene-core-0.9.21b-fix-unescaping/src/CLucene/queryParser/Lexer.cpp
--- clucene-core-0.9.21b/src/CLucene/queryParser/Lexer.cpp	2008-10-23 19:44:35.000000000 +0200
+++ clucene-core-0.9.21b-fix-unescaping/src/CLucene/queryParser/Lexer.cpp	2014-10-25 23:43:21.000000000 +0200
@@ -354,6 +354,7 @@
    TCHAR ch = prev;
    int bp=0;
    buf[bp++] = ch;
+   buf[bp] = 0;
 
    ch = reader->GetNext();
    int32_t idx = _tcscspn( buf, _T("\\+-!():^[]{}\"~*") );
diff -ur clucene-core-0.9.21b/src/CLucene/queryParser/QueryParserBase.cpp clucene-core-0.9.21b-fix-unescaping/src/CLucene/queryParser/QueryParserBase.cpp
--- clucene-core-0.9.21b/src/CLucene/queryParser/QueryParserBase.cpp	2008-10-23 19:44:35.000000000 +0200
+++ clucene-core-0.9.21b-fix-unescaping/src/CLucene/queryParser/QueryParserBase.cpp	2014-10-25 23:57:30.000000000 +0200
@@ -14,6 +14,8 @@
 #include "CLucene/search/WildcardQuery.h"
 #include "CLucene/search/PrefixQuery.h"
 
+#include <cstring>
+
 
 CL_NS_USE(search)
 CL_NS_USE(util)
@@ -44,7 +46,9 @@
 	int j = 0;
 	for (int i = 0; i < len; i++) {
 		if (source[i] == '\\' && source[i+1] != '\0' ) {
-			_tcscpy(source+i,source+i+1);
+			// move from i+1 to len+1 (terminating 0)
+			// i.e. (len+1)-i = len-i+1 characters
+			std::memmove(source+i,source+i+1,(len-i+1)*sizeof(TCHAR));
 			len--;
 		}
 	}
diff -ur clucene-core-0.9.21b/src/CLucene/queryParser/QueryToken.cpp clucene-core-0.9.21b-fix-unescaping/src/CLucene/queryParser/QueryToken.cpp
--- clucene-core-0.9.21b/src/CLucene/queryParser/QueryToken.cpp	2008-10-23 19:44:35.000000000 +0200
+++ clucene-core-0.9.21b-fix-unescaping/src/CLucene/queryParser/QueryToken.cpp	2014-10-25 23:43:35.000000000 +0200
@@ -57,7 +57,7 @@
     Value = STRDUP_TtoT(value);
   #else
     _tcsncpy(Value,value,LUCENE_TOKEN_WORD_LENGTH);
-    Value[LUCENE_TOKEN_WORD_LENGTH];
+    Value[LUCENE_TOKEN_WORD_LENGTH] = 0;
   #endif
     this->Start = start;
     this->End = end;
