diff -ur clucene-core-0.9.21b/test/CuTest.cpp clucene-core-0.9.21b-verbose-tests/test/CuTest.cpp
--- clucene-core-0.9.21b/test/CuTest.cpp	2008-10-23 19:44:35.000000000 +0200
+++ clucene-core-0.9.21b-verbose-tests/test/CuTest.cpp	2014-10-25 22:34:15.000000000 +0200
@@ -8,7 +8,7 @@
 #include "CuTest.h"
 #include <assert.h>
 
-static int verbose = 0;
+static int verbose = 1;
 static int messyPrinting = 0;
 
 void CuInit(int argc, char *argv[])
