--- configure.ac.orig	2014-02-20 18:30:34.249893070 -0300
+++ configure.ac	2014-02-20 18:30:39.777721229 -0300
@@ -154,7 +154,7 @@ AC_ARG_WITH([systemd],
   [], [with_systemd=no]
 )
 if test "x$with_systemd" != xno; then
-  PKG_CHECK_MODULES([SYSTEMD], [libsystemd-login >= 206], [], [
+  PKG_CHECK_MODULES([SYSTEMD], [libsystemd >= 209], [], [
     AC_CHECK_LIB(systemd-login, sd_pid_get_slice, [have_systemd=yes], [have_systemd=no])
     if test "x$have_systemd" = xno; then
       AC_MSG_ERROR([systemd support missing/incomplete])
