diff -Naur procps-ng-3.3.7.orig/testsuite/config/unix.exp procps-ng-3.3.7/testsuite/config/unix.exp
--- procps-ng-3.3.7.orig/testsuite/config/unix.exp	2013-03-25 11:50:21.000000000 +0100
+++ procps-ng-3.3.7/testsuite/config/unix.exp	2013-03-26 17:50:54.860228114 +0100
@@ -80,7 +80,7 @@
     }
 
     set do_loop 1
-    set last_value &9
+    set last_value 99999999
     set found_item 0
     while { $do_loop ==1 } {
     expect {
