diff -up gnome-session-2.32.0/gnome-session/Makefile.am.cflags gnome-session-2.32.0/gnome-session/Makefile.am
--- gnome-session-2.32.0/gnome-session/Makefile.am.cflags	2010-09-30 08:19:16.757603002 -0400
+++ gnome-session-2.32.0/gnome-session/Makefile.am	2010-09-30 08:19:50.708603005 -0400
@@ -80,6 +80,10 @@ gnome_session_LDADD =				\
 	$(GCONF_LIBS)				\
 	$(EXECINFO_LIBS)
 
+libgsmutil_la_CFLAGS =				\
+	$(AM_CPPFLAGS)				\
+	$(GCONF_CFLAGS)
+
 libgsmutil_la_SOURCES =				\
 	gsm-util.c				\
 	gsm-util.h
