--- gnome-session/gsm-manager.c.libnotify	2011-02-12 01:43:26.000000000 -0200
+++ gnome-session/gsm-manager.c	2011-02-12 01:45:00.984858465 -0200
@@ -2567,7 +2567,7 @@ update_max_idle_notification (GsmManager
                                                     NULL);
                 } else {
                         manager->priv->max_idle_notification
-                                = notify_notification_new (summary, body, NULL, NULL);
+                                = notify_notification_new (summary, body, NULL);
                         notify_notification_set_timeout (manager->priv->max_idle_notification,
                                                          NOTIFY_EXPIRES_NEVER);
 
