diff -up enscript-1.6.5.1/afm/Makefile.am.rh477382 enscript-1.6.5.1/afm/Makefile.am
--- enscript-1.6.5.1/afm/Makefile.am.rh477382	2010-03-06 14:02:53.000000000 +0100
+++ enscript-1.6.5.1/afm/Makefile.am	2010-05-21 13:09:28.000346827 +0200
@@ -29,9 +29,12 @@ hvnbo.afm hvno.afm hvo.afm ncb.afm ncbi.
 pobi.afm poi.afm por.afm sy.afm tib.afm tibi.afm tii.afm tir.afm	\
 zcmi.afm zd.afm
 
-public_fonts = matrix.afm matrix.pfa
+public_fonts = matrix.afm matrix.eps
 
 afmdir = $(pkgdatadir)/afm
 dist_afm_DATA = font.map $(default_afm) $(public_fonts)
 
 EXTRA_DIST = ChangeLog.old
+
+matrix.eps: matrix.pfa
+	cp matrix.pfa matrix.eps
diff -up enscript-1.6.5.1/src/util.c.rh477382 enscript-1.6.5.1/src/util.c
--- enscript-1.6.5.1/src/util.c.rh477382	2009-12-28 00:57:12.000000000 +0100
+++ enscript-1.6.5.1/src/util.c	2010-05-21 13:08:10.920347803 +0200
@@ -1079,9 +1079,16 @@ download_font (char *name)
       buffer_append (&fname, ".pfb");
       if (stat (buffer_ptr (&fname), &stat_st) != 0)
 	{
-	  /* Couldn't find font description file, nothing to download. */
-	  buffer_uninit (&fname);
-	  return;
+	  /* .eps */
+	  buffer_clear (&fname);
+	  buffer_append (&fname, prefix);
+	  buffer_append (&fname, ".eps");
+	  if (stat (buffer_ptr (&fname), &stat_st) != 0)
+	    {
+	      /* Couldn't find font description file, nothing to download. */
+	      buffer_uninit (&fname);
+	      return;
+	    }
 	}
     }
 
