--- xfce4-power-manager-1.0.1.orig/src/xfpm-notify.c	2010-05-19 11:22:27.000000000 +0200
+++ xfce4-power-manager-1.0.1/src/xfpm-notify.c	2010-12-17 11:05:26.087627800 +0100
@@ -209,14 +209,11 @@
 {
     NotifyNotification *n;
     
-    n = notify_notification_new (title, message, NULL, NULL);
+    n = notify_notification_new (title, message, NULL);
     
     if ( icon_name )
     	xfpm_notify_set_notification_icon (n, icon_name);
 	
-    if ( icon )
-    	notify_notification_attach_to_status_icon (n, icon);
-	
     notify_notification_set_urgency (n, (NotifyUrgency)urgency);
     
     if ( timeout != 0)
