--- wordwarvi-1.00/Makefile.orig	2011-06-09 01:06:22.000000000 -0300
+++ wordwarvi-1.00/Makefile	2011-07-06 21:29:53.395989601 -0300
@@ -1,5 +1,5 @@
 PREFIX=/usr
-DATADIR=${PREFIX}/share/wordwarvi
+DATADIR?=${PREFIX}/share/wordwarvi
 MANDIR?=${PREFIX}/share/man
 MANPAGEDIR=${MANDIR}/man6
 
@@ -24,18 +24,6 @@ endif
 CC ?= gcc
 BUILD_CC ?= gcc
 
-# DEBUG=-g
-# DEBUG=
-# PROFILE_FLAG=-pg
-#PROFILE_FLAG=
-#OPTIMIZE_FLAG=
-# OPTIMIZE_FLAG=-O3
-#OPTIMIZE_FLAG=-O3 -pedantic -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security
-OPTIMIZE_FLAG=-O3 -pedantic
-
-
-LDFLAGS=${PROFILE_FLAG}
-
 DEFINES=${SNDFLAGS} -DDATADIR=\"${DATADIR}/\"
 
 all:	wordwarvi wordwarvi.6.gz
@@ -53,32 +41,32 @@ else
 endif
 
 joystick.o:	joystick.c joystick.h Makefile
-	$(CC) ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} -pthread -Wall -c joystick.c
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} -pthread -Wall -c joystick.c
 
 ogg_to_pcm.o:	ogg_to_pcm.c ogg_to_pcm.h Makefile
-	$(CC) ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} `pkg-config --cflags vorbisfile` \
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} `pkg-config --cflags vorbisfile` \
 		-pthread -Wall -c ogg_to_pcm.c
 
 wwviaudio.o:	wwviaudio.c wwviaudio.h ogg_to_pcm.h my_point.h Makefile
-	$(CC) -Wall ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} \
+	$(CC) -Wall ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} \
 		${DEFINES} \
 		-pthread `pkg-config --cflags vorbisfile` \
 		-c wwviaudio.c
 
 rumble.o:	rumble.c rumble.h Makefile
-	$(CC) ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} `pkg-config --cflags vorbisfile` \
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} `pkg-config --cflags vorbisfile` \
 		-pthread -Wall -c rumble.c
 
 wwvi_font.o:	wwvi_font.c wwvi_font.h my_point.h Makefile
-	$(CC) ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} -pthread -Wall -c wwvi_font.c
+	$(CC) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} -pthread -Wall -c wwvi_font.c
 
 stamp:	stamp.c
-	$(BUILD_CC) -o stamp stamp.c
+	$(BUILD_CC) $(LDFLAGS) -o stamp stamp.c
 
 wordwarvi:	wordwarvi.c joystick.o rumble.o ${OGGOBJ} wwviaudio.o wwvi_font.o \
 		Makefile version.h stamp levels.h rumble.h
 	./stamp > stamp.h
-	$(CC) ${DEBUG} ${PROFILE_FLAG} ${OPTIMIZE_FLAG} ${SCREENSAVERFLAG} -pthread -Wall  ${DEFINES} \
+	$(CC) $(LDFLAGS) ${DEBUG} ${PROFILE_FLAG} ${CFLAGS} ${SCREENSAVERFLAG} -pthread -Wall  ${DEFINES} \
 		joystick.o \
 		rumble.o \
 		wwvi_font.o \
