--- migemo-0.40/migemo-dict.rb.bz830559	2012-06-11 23:17:27.000000000 +0900
+++ migemo-0.40/migemo-dict.rb	2012-06-11 23:18:06.000000000 +0900
@@ -39,7 +39,7 @@
 
   private
   def decompose (line)
-    array = line.chomp.split("\t").delete_if do |x| x == nil end
+    array = line.chomp.force_encoding("EUC-JP").split("\t").delete_if do |x| x == nil end
     key = array.shift
     values = array
     raise if key == nil
