--- smpeg2-config.in.orig	2013-06-01 12:51:06.000000000 -0300
+++ smpeg2-config.in	2013-09-08 14:39:16.431729472 -0300
@@ -45,10 +45,9 @@ while test $# -gt 0; do
       echo $includes -I@includedir@/smpeg2 `@SDL_CONFIG@ --cflags`
       ;;
     --libs)
+      unset libdirs
       if [ "`uname`" = "SunOS" ]; then
         libdirs="-L@libdir@ -R@libdir@"
-      else
-        libdirs="-L@libdir@ @SMPEG_RLD_FLAGS@"
       fi
       echo $libdirs -lsmpeg2 `@SDL_CONFIG@ --libs`
       ;;
