From: =?UTF-8?q?Ond=C5=99ej=20Va=C5=A1=C3=ADk?= <ovasik@redhat.com>
Date: Mon, 14 Sep 2015 09:47:05 +0200
Subject: [PATCH 5/7] fix segfault with nonexisting file with patternnames
 (#567022)

diff --git a/src/copyin.c b/src/copyin.c
index 12bd27c..183b5b5 100644
--- a/src/copyin.c
+++ b/src/copyin.c
@@ -870,21 +870,24 @@ read_pattern_file ()
 
   pattern_fp = fopen (pattern_file_name, "r");
   if (pattern_fp == NULL)
-    open_fatal (pattern_file_name);
-  while (ds_fgetstr (pattern_fp, &pattern_name, '\n') != NULL)
-    {
-      if (new_num_patterns >= max_new_patterns)
-	{
-	  max_new_patterns += 1;
-	  new_save_patterns = (char **)
-	    xrealloc ((char *) new_save_patterns,
-		      max_new_patterns * sizeof (char *));
-	}
-      new_save_patterns[new_num_patterns] = xstrdup (pattern_name.ds_string);
-      ++new_num_patterns;
-    }
-  if (ferror (pattern_fp) || fclose (pattern_fp) == EOF)
-    close_error (pattern_file_name);
+    open_error (pattern_file_name);
+  else
+  {
+    while (ds_fgetstr (pattern_fp, &pattern_name, '\n') != NULL)
+      {
+        if (new_num_patterns >= max_new_patterns)
+        {
+          max_new_patterns += 1;
+          new_save_patterns = (char **)
+          xrealloc ((char *) new_save_patterns,
+            max_new_patterns * sizeof (char *));
+        }
+        new_save_patterns[new_num_patterns] = xstrdup (pattern_name.ds_string);
+        ++new_num_patterns;
+      }
+    if (ferror (pattern_fp) || fclose (pattern_fp) == EOF)
+      close_error (pattern_file_name);
+  }
 
   for (i = 0; i < num_patterns; ++i)
     new_save_patterns[i] = save_patterns[i];
