diff -up sudo-1.8.0/configure.in.getgrouplist sudo-1.8.0/configure.in
--- sudo-1.8.0/configure.in.getgrouplist	2011-03-01 18:32:10.000000000 -0300
+++ sudo-1.8.0/configure.in	2011-03-01 18:34:56.577526726 -0300
@@ -1967,7 +1967,7 @@ dnl
 AC_FUNC_GETGROUPS
 AC_CHECK_FUNCS(strrchr sysconf tzset strftime initgroups getgroups fstat \
 	       regcomp setlocale nl_langinfo getaddrinfo mbr_check_membership \
-	       setrlimit64 sysctl)
+	       setrlimit64 sysctl getgrouplist)
 AC_CHECK_FUNCS(getline, [], [
     AC_LIBOBJ(getline)
     AC_CHECK_FUNCS(fgetln)
diff -up sudo-1.8.0/plugins/sudoers/pwutil.c.getgrouplist sudo-1.8.0/plugins/sudoers/pwutil.c
--- sudo-1.8.0/plugins/sudoers/pwutil.c.getgrouplist	2011-01-24 17:54:57.000000000 -0200
+++ sudo-1.8.0/plugins/sudoers/pwutil.c	2011-03-01 18:35:53.709970354 -0300
@@ -709,6 +709,24 @@ user_in_group(struct passwd *pw, const c
     }
 #endif /* HAVE_MBR_CHECK_MEMBERSHIP */
 
+#ifdef HAVE_GETGROUPLIST
+    {
+	gid_t *grouplist, grouptmp;
+	int n_groups, i;
+	n_groups = 1;
+	if (getgrouplist(user_name, user_gid, &grouptmp, &n_groups) == -1) {
+	    grouplist = (gid_t *) emalloc(sizeof(gid_t) * (n_groups + 1));
+	    if (getgrouplist(user_name, user_gid, grouplist, &n_groups) > 0)
+		for (i = 0; i < n_groups; i++)
+		    if (grouplist[i] == grp->gr_gid) {
+			free(grouplist);
+			return(TRUE);
+		    }
+	    free(grouplist);
+	}
+    }
+#endif /* HAVE_GETGROUPLIST */
+
 done:
     if (grp != NULL)
 	gr_delref(grp);
