diff -up sudo-1.7.3/check.c.getgrouplist sudo-1.7.3/check.c
--- sudo-1.7.3/check.c.getgrouplist	2010-06-29 13:27:04.000000000 -0300
+++ sudo-1.7.3/check.c	2010-07-01 00:04:59.532964763 -0300
@@ -370,6 +370,25 @@ user_is_exempt()
 {
     if (!def_exempt_group)
 	return(FALSE);
+
+#ifdef HAVE_GETGROUPLIST
+    {
+	gid_t *grouplist, grouptmp;
+	int n_groups, i;
+	n_groups = 1;
+	if (getgrouplist(user_name, user_gid, &grouptmp, &n_groups) == -1) {
+	    grouplist = (gid_t *) emalloc(sizeof(gid_t) * (n_groups + 1));
+	    if (getgrouplist(user_name, user_gid, grouplist, &n_groups) > 0)
+		for (i = 0; i < n_groups; i++)
+		    if (grouplist[i] == grp->gr_gid) {
+			free(grouplist);
+			return(TRUE);
+		    }
+	    free(grouplist);
+	}
+    }
+#endif
+
     return(user_in_group(sudo_user.pw, def_exempt_group));
 }
 
diff -up sudo-1.7.3/configure.in.getgrouplist sudo-1.7.3/configure.in
