diff -up sudo-1.8.14b4/doc/sudoers.cat.docpassexpire sudo-1.8.14b4/doc/sudoers.cat
--- sudo-1.8.14b4/doc/sudoers.cat.docpassexpire	2015-06-09 00:47:07.000000000 +0200
+++ sudo-1.8.14b4/doc/sudoers.cat	2015-07-14 13:11:11.116000185 +0200
@@ -1328,8 +1328,8 @@ SSUUDDOOEERRSS OOPPTTIIOONN
                        fractional component if minute granularity is
                        insufficient, for example 2.5.  The default is 5.  Set
                        this to 0 to always prompt for a password.  If set to a
-                       value less than 0 the user's time stamp will never
-                       expire.  This can be used to allow users to create or
+                       value less than 0 the user's time stamp will not
+                       expire until reboot.  This can be used to allow users to create or
                        delete their own time stamps via ``sudo -v'' and ``sudo
                        -k'' respectively.
 
diff -up sudo-1.8.14b4/doc/sudoers.man.in.docpassexpire sudo-1.8.14b4/doc/sudoers.man.in
--- sudo-1.8.14b4/doc/sudoers.man.in.docpassexpire	2015-07-14 13:11:11.116000185 +0200
+++ sudo-1.8.14b4/doc/sudoers.man.in	2015-07-14 13:14:17.261222481 +0200
@@ -2822,7 +2822,7 @@ Set this to
 to always prompt for a password.
 If set to a value less than
 \fR0\fR
-the user's time stamp will never expire.
+the user's time stamp will not expire until reboot.
 This can be used to allow users to create or delete their own time stamps via
 \(Lq\fRsudo -v\fR\(Rq
 and
diff -up sudo-1.8.14b4/doc/sudoers.mdoc.in.docpassexpire sudo-1.8.14b4/doc/sudoers.mdoc.in
--- sudo-1.8.14b4/doc/sudoers.mdoc.in.docpassexpire	2015-04-07 18:15:50.000000000 +0200
+++ sudo-1.8.14b4/doc/sudoers.mdoc.in	2015-07-14 13:11:11.117000176 +0200
@@ -2647,7 +2647,7 @@ Set this to
 to always prompt for a password.
 If set to a value less than
 .Li 0
-the user's time stamp will never expire.
+the user's time stamp will not expire until reboot.
 This can be used to allow users to create or delete their own time stamps via
 .Dq Li sudo -v
 and
