--- src/preferences.py.prefs	2010-05-30 08:56:07.000000000 -0300
+++ src/preferences.py	2010-06-14 02:00:25.667895661 -0300
@@ -23,15 +23,15 @@
 PREFS = {
     "card_index": 0,
     "control": "Master",
-    "mixer": "",
+    "mixer": "alsamixer",
     "icon_theme": "Default",
-    "run_in_terminal": 0,
+    "run_in_terminal": 1,
     "scale_increment": 1.0,
-    "scale_show_value": 0,
+    "scale_show_value": 1,
     "show_tooltip": 1,
     "toggle": "mute",
-    "keys": 0,
-    "keys_backend": "hal",
+    "keys": 1,
+    "keys_backend": "xlib",
     "show_notify": 0,
     "notify_timeout": 2.0,
     "notify_position": 0,
