--- src/preferences.py.prefs	2015-09-29 20:55:40.251732177 -0300
+++ src/preferences.py	2015-09-29 20:56:53.359268640 -0300
@@ -25,8 +25,8 @@ from debug import log
 PREFS = {
     "card_index": 0,
     "control": "Master",
-    "mixer": "alsamixer",
-    "run_in_terminal": 1,
+    "mixer": "_DEF_MIX_",
+    "run_in_terminal": _TERM_MODE_,
     "mixer_internal": 1,
     "mixer_show_values": 1,
     "icon_theme": "Default",
@@ -34,7 +34,7 @@ PREFS = {
     "scale_show_value": 0,
     "show_tooltip": 1,
     "toggle": "mute",
-    "keys": 0,
+    "keys": 1,
     "keys_backend": "xlib",
     "show_notify": 0,
     "notify_timeout": 2.0,
