--- src/m64py/platform.py.libs	2013-11-15 21:55:25.301546089 -0200
+++ src/m64py/platform.py	2013-11-15 22:07:42.370485250 -0200
@@ -23,7 +23,9 @@ if sys.platform.startswith("linux"):
     DLL_FILTER = ".so.2"
     DEFAULT_DYNLIB = "libmupen64plus.so.2"
     SEARCH_DIRS = [
+            "/usr/local/lib64/mupen64plus",
             "/usr/local/lib/mupen64plus",
+            "/usr/lib64/mupen64plus",
             "/usr/lib/mupen64plus",
             "/usr/games/lib64/mupen64plus",
             "/usr/games/lib/mupen64plus",
