--- src/m64py/platform.py.orig	2014-06-01 12:47:10.085462827 -0300
+++ src/m64py/platform.py	2014-06-01 12:48:11.278417595 -0300
@@ -23,7 +23,9 @@ if sys.platform.startswith("linux"):
     DLL_FILTER = ".so.2"
     DEFAULT_DYNLIB = "libmupen64plus.so.2"
     SEARCH_DIRS = [
+        "/usr/local/lib64/mupen64plus",
         "/usr/local/lib/mupen64plus",
+        "/usr/lib64/mupen64plus",
         "/usr/lib/mupen64plus",
         "/usr/games/lib64/mupen64plus",
         "/usr/games/lib/mupen64plus",
