--- src/m64py/frontend/input.py.orig	2014-01-11 14:17:53.445212922 -0200
+++ src/m64py/frontend/input.py	2014-01-11 14:18:27.622020937 -0200
@@ -94,7 +94,7 @@ class Input(QDialog, Ui_InputDialog):
 
         devices = [(self.tr("Keyboard/Mouse"), -1)]
         for num, joy in enumerate(self.joystick.joystick_names):
-            devices.append((self.tr("Joystick %s (%s)" % (num, joy), num)))
+            devices.append((self.tr("Joystick %s (%s)" % (num, joy)), num))
 
         for device, dtype in devices:
             self.comboDevice.addItem(device, dtype)
