From 4d038ae3163aba04218b05f3983473b25c943b8b Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@linux-m68k.org>
Date: Wed, 5 Sep 2012 17:35:14 +0200
Subject: [PATCH] Increase timeout for stdlib/tst-strtod-overflow

---
 stdlib/tst-strtod-overflow.c | 1 +
 2 files changed, 5 insertions(+)

diff --git a/stdlib/tst-strtod-overflow.c b/stdlib/tst-strtod-overflow.c
index 668d55b..6c5b282 100644
--- a/stdlib/tst-strtod-overflow.c
+++ b/stdlib/tst-strtod-overflow.c
@@ -45,4 +45,5 @@ do_test (void)
 }
 
 #define TEST_FUNCTION do_test ()
+#define TIMEOUT 5
 #include "../test-skeleton.c"
-- 
1.7.12.2

