From 6b18bea6256e1eefebd27811873831d0e3f171de Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Tue, 19 Mar 2013 22:38:25 +0000
Subject: [PATCH] Fix types of constants in k_casinh*.c.

---
 ChangeLog        | 7 +++++++
 math/k_casinhf.c | 4 ++--
 math/k_casinhl.c | 4 ++--
 3 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/math/k_casinhf.c b/math/k_casinhf.c
index 3152ea2..7ff4b03 100644
--- a/math/k_casinhf.c
+++ b/math/k_casinhf.c
@@ -79,8 +79,8 @@ __kernel_casinhf (__complex__ float x, int adj)
     }
   else
     {
-      __real__ y = (rx - ix) * (rx + ix) + 1.0;
-      __imag__ y = 2.0 * rx * ix;
+      __real__ y = (rx - ix) * (rx + ix) + 1.0f;
+      __imag__ y = 2.0f * rx * ix;
 
       y = __csqrtf (y);
 
diff --git a/math/k_casinhl.c b/math/k_casinhl.c
index 110ae33..aec501b 100644
--- a/math/k_casinhl.c
+++ b/math/k_casinhl.c
@@ -86,8 +86,8 @@ __kernel_casinhl (__complex__ long double x, int adj)
     }
   else
     {
-      __real__ y = (rx - ix) * (rx + ix) + 1.0;
-      __imag__ y = 2.0 * rx * ix;
+      __real__ y = (rx - ix) * (rx + ix) + 1.0L;
+      __imag__ y = 2.0L * rx * ix;
 
       y = __csqrtl (y);
 
-- 
1.8.2.3

