From fd80f0b7a144a17222795a68691db3d97da5f730 Mon Sep 17 00:00:00 2001
From: Allan McRae <allan@archlinux.org>
Date: Sun, 2 Dec 2012 19:56:13 +1000
Subject: [PATCH] Fix "localedef --posix" description

---
 locale/programs/localedef.c | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/locale/programs/localedef.c b/locale/programs/localedef.c
index cc94828..16129bb 100644
--- a/locale/programs/localedef.c
+++ b/locale/programs/localedef.c
@@ -128,7 +128,7 @@ static const struct argp_option options[] =
     N_("Create output even if warning messages were issued") },
   { "old-style", OPT_OLDSTYLE, NULL, 0, N_("Create old-style tables") },
   { "prefix", OPT_PREFIX, "PATH", 0, N_("Optional output file prefix") },
-  { "posix", OPT_POSIX, NULL, 0, N_("Be strictly POSIX conform") },
+  { "posix", OPT_POSIX, NULL, 0, N_("Strictly conform to POSIX") },
   { "quiet", OPT_QUIET, NULL, 0,
     N_("Suppress warnings and information messages") },
   { "verbose", 'v', NULL, 0, N_("Print more messages") },
-- 
1.8.2

