From e11f5155b28e0a48e347ef0a68de1cdd229aa18c Mon Sep 17 00:00:00 2001
From: Andreas Jaeger <aj@suse.de>
Date: Tue, 14 Aug 2012 08:02:08 +0200
Subject: [PATCH 1/2] Update i386 ULPs

---
 sysdeps/i386/fpu/libm-test-ulps | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/sysdeps/i386/fpu/libm-test-ulps b/sysdeps/i386/fpu/libm-test-ulps
index f70311b..b683532 100644
--- a/sysdeps/i386/fpu/libm-test-ulps
+++ b/sysdeps/i386/fpu/libm-test-ulps
@@ -1077,6 +1077,8 @@ idouble: 1
 Test "Imaginary part of: clog10 (0x1.000566p0 + 0x1.234p-100 i) == 3.577293486783822178310971763308187385546e-5 + 3.897399639875661463735636919790792140598e-31 i":
 ildouble: 1
 ldouble: 1
+Test "Real part of: clog10 (0x1.234566p-30 + 1.0 i) == 2.438200411482400072282924063740535840474e-19 + 6.821881764607257184291586401763604544928e-1 i":
+ldouble: 1
 Test "Imaginary part of: clog10 (0x1.234566p-30 + 1.0 i) == 2.438200411482400072282924063740535840474e-19 + 6.821881764607257184291586401763604544928e-1 i":
 float: 1
 ifloat: 1
-- 
1.7.12

