From ddc7e412ab252e7360a4357664beb3b5d9c4f42b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20B=C3=ADlka?= <neleai@seznam.cz>
Date: Fri, 25 Oct 2013 19:04:47 +0200
Subject: [PATCH] Make strptime %Z consistent between doc and code. Fixes bug
 14876

---
 time/strptime_l.c   |  6 +++++-
 time/tst-strptime.c |  1 +
 3 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/time/strptime_l.c b/time/strptime_l.c
index 00fc1ef..c2c2ee8 100644
--- a/time/strptime_l.c
+++ b/time/strptime_l.c
@@ -744,7 +744,11 @@ __strptime_internal (rp, fmt, tmp, statep LOCALE_PARAM)
 	  s.want_xday = 1;
 	  break;
 	case 'Z':
-	  /* XXX How to handle this?  */
+	  /* Read timezone but perform no conversion.  */
+	  while (ISSPACE (*rp))
+	    rp++;
+	  while (!ISSPACE (*rp) && *rp != '\0')
+	    rp++;
 	  break;
 	case 'z':
 	  /* We recognize two formats: if two digits are given, these
diff --git a/time/tst-strptime.c b/time/tst-strptime.c
index 21fbb63..193f7db 100644
--- a/time/tst-strptime.c
+++ b/time/tst-strptime.c
@@ -41,6 +41,7 @@ static const struct
   { "C", "19990502123412", "%Y%m%d%H%M%S", 0, 121, 4, 2 },
   { "C", "2001 20 Mon", "%Y %U %a", 1, 140, 4, 21 },
   { "C", "2001 21 Mon", "%Y %W %a", 1, 140, 4, 21 },
+  { "C", "2001 EST 21 Mon", "%Y %Z %W %a", 1, 140, 4, 21 },
   { "C", "2012 00 Sun", "%Y %W %a", 0, 0, 0, 1 },
   { "ja_JP.EUC-JP", "2000-01-01 08:12:21 AM", "%Y-%m-%d %I:%M:%S %p",
     6, 0, 0, 1 },
-- 
1.8.4.2

