From d84f25c7d871dac31ed3ef136ebfb80d3ac73601 Mon Sep 17 00:00:00 2001
From: Liubov Dmitrieva <ldmitrie@sourceware.org>
Date: Thu, 19 Sep 2013 16:51:13 +0400
Subject: [PATCH] Fix buffer overrun in strtod_l

---
 stdlib/strtod_l.c | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/stdlib/strtod_l.c b/stdlib/strtod_l.c
index 8f60653..90541cd 100644
--- a/stdlib/strtod_l.c
+++ b/stdlib/strtod_l.c
@@ -1752,7 +1752,7 @@ ____STRTOF_INTERNAL (nptr, endptr, group, loc)
 	      got_limb;
 	    }
 
-	  for (i = densize; num[i] == 0 && i >= 0; --i)
+	  for (i = densize; i >= 0 && num[i] == 0; --i)
 	    ;
 	  return round_and_return (retval, exponent - 1, negative,
 				   quot, BITS_PER_MP_LIMB - 1 - used,
-- 
1.8.4

