From b0037103771cc02e092884c39aa9e4982073099a Mon Sep 17 00:00:00 2001
From: "Maciej W. Rozycki" <macro@codesourcery.com>
Date: Wed, 26 Jun 2013 19:14:29 +0100
Subject: [PATCH] [BZ #15022] Avoid repeated calls to DL_STATIC_INIT for the
 same module.

---
 elf/dl-open.c | 8 ++++----
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/elf/dl-open.c b/elf/dl-open.c
index 0f054bf..15221c8 100644
--- a/elf/dl-open.c
+++ b/elf/dl-open.c
@@ -568,6 +568,10 @@ cannot load any more object with static TLS"));
   if (relocation_in_progress)
     LIBC_PROBE (reloc_complete, 3, args->nsid, r, new);
 
+#ifndef SHARED
+  DL_STATIC_INIT (new);
+#endif
+
   /* Run the initializer functions of new objects.  */
   _dl_init (new, args->argc, args->argv, args->env);
 
@@ -721,10 +725,6 @@ no more namespaces available for dlmopen()"));
   /* Release the lock.  */
   __rtld_lock_unlock_recursive (GL(dl_load_lock));
 
-#ifndef SHARED
-  DL_STATIC_INIT (args.map);
-#endif
-
   return args.map;
 }
 
-- 
1.8.3.2

