From b2e25af00c4b6268624a06472477112fc015e7d8 Mon Sep 17 00:00:00 2001
From: Roland McGrath <roland@hack.frob.com>
Date: Tue, 5 Feb 2013 11:38:14 -0800
Subject: [PATCH] Fix aliases in stub lseek.

---
 io/lseek.c | 7 ++++---
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/io/lseek.c b/io/lseek.c
index b65c50d..63cd75f 100644
--- a/io/lseek.c
+++ b/io/lseek.c
@@ -21,7 +21,7 @@
 
 /* Seek to OFFSET on FD, starting from WHENCE.  */
 off_t
-__lseek (fd, offset, whence)
+__libc_lseek (fd, offset, whence)
      int fd;
      off_t offset;
      int whence;
@@ -45,7 +45,8 @@ __lseek (fd, offset, whence)
   __set_errno (ENOSYS);
   return -1;
 }
+weak_alias (__libc_lseek, __lseek)
+weak_alias (__libc_lseek, lseek)
 stub_warning (lseek)
-libc_hidden_def (__lseek)
 
-weak_alias (__lseek, lseek)
+libc_hidden_def (__lseek)
-- 
1.8.1.3

