From c13e07830800dd95ca7684b3c473394c77059c97 Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Wed, 20 May 2015 13:50:00 +0200
Subject: [PATCH] Set errno to ENOMEM on overflow in sbrk (bug 18592)

---
 misc/sbrk.c | 13 +++++++++----
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/misc/sbrk.c b/misc/sbrk.c
index 87b5472..89ab9c4 100644
--- a/misc/sbrk.c
+++ b/misc/sbrk.c
@@ -47,10 +47,15 @@ __sbrk (intptr_t increment)
     return __curbrk;
 
   oldbrk = __curbrk;
-  if ((increment > 0
-       ? ((uintptr_t) oldbrk + (uintptr_t) increment < (uintptr_t) oldbrk)
-       : ((uintptr_t) oldbrk < (uintptr_t) -increment))
-      || __brk (oldbrk + increment) < 0)
+  if (increment > 0
+      ? ((uintptr_t) oldbrk + (uintptr_t) increment < (uintptr_t) oldbrk)
+      : ((uintptr_t) oldbrk < (uintptr_t) -increment))
+    {
+      __set_errno (ENOMEM);
+      return (void *) -1;
+    }
+
+  if (__brk (oldbrk + increment) < 0)
     return (void *) -1;
 
   return oldbrk;
-- 
2.4.6

