From 11b8a0e1d7b6175470ffc9077e94104c896093b7 Mon Sep 17 00:00:00 2001
From: Liubov Dmitrieva <liubov.dmitrieva@intel.com>
Date: Wed, 26 Jun 2013 12:28:43 -0700
Subject: [PATCH 2/2] Fix buffers overrun in x86_64 memcmp-ssse3.S

---
 sysdeps/x86_64/multiarch/memcmp-ssse3.S | 6 ++----
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/sysdeps/x86_64/multiarch/memcmp-ssse3.S b/sysdeps/x86_64/multiarch/memcmp-ssse3.S
index bdd2ed2..e319df9 100644
--- a/sysdeps/x86_64/multiarch/memcmp-ssse3.S
+++ b/sysdeps/x86_64/multiarch/memcmp-ssse3.S
@@ -1463,10 +1463,8 @@ L(next_24_bytes):
 	test	$0x40, %dh
 	jnz	L(Byte22)
 
-	mov	-9(%rdi), %eax
-	and	$0xff, %eax
-	mov	-9(%rsi), %edx
-	and	$0xff, %edx
+	movzbl	-9(%rdi), %eax
+	movzbl	-9(%rsi), %edx
 	sub	%edx, %eax
 	ret
 # else
-- 
1.8.3.2

