From fab967007b1ebfff374f40d6f321f14d45674525 Mon Sep 17 00:00:00 2001
From: Andreas Jaeger <aj@suse.de>
Date: Tue, 14 Aug 2012 08:04:51 +0200
Subject: [PATCH 2/2] Another ULPs update.

---
 sysdeps/i386/fpu/libm-test-ulps | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysdeps/i386/fpu/libm-test-ulps b/sysdeps/i386/fpu/libm-test-ulps
index b683532..abd98a2 100644
--- a/sysdeps/i386/fpu/libm-test-ulps
+++ b/sysdeps/i386/fpu/libm-test-ulps
@@ -1078,6 +1078,7 @@ Test "Imaginary part of: clog10 (0x1.000566p0 + 0x1.234p-100 i) == 3.57729348678
 ildouble: 1
 ldouble: 1
 Test "Real part of: clog10 (0x1.234566p-30 + 1.0 i) == 2.438200411482400072282924063740535840474e-19 + 6.821881764607257184291586401763604544928e-1 i":
+ildouble: 1
 ldouble: 1
 Test "Imaginary part of: clog10 (0x1.234566p-30 + 1.0 i) == 2.438200411482400072282924063740535840474e-19 + 6.821881764607257184291586401763604544928e-1 i":
 float: 1
-- 
1.7.12

