From 0bfcf2c73cf6a68be978e66248cec5c4c03f07fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20B=C3=ADlka?= <neleai@seznam.cz>
Date: Fri, 18 Oct 2013 09:32:35 +0200
Subject: [PATCH 2/2] Remove assert in malloc statistic. Fixes bug 12486.

---
 malloc/malloc.c | 9 ---------
 2 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/malloc/malloc.c b/malloc/malloc.c
index 2938234..1a18c3f 100644
--- a/malloc/malloc.c
+++ b/malloc/malloc.c
@@ -2216,15 +2216,6 @@ static void do_check_malloc_state(mstate av)
   /* top chunk is OK */
   check_chunk(av, av->top);
 
-  /* sanity checks for statistics */
-
-  assert(mp_.n_mmaps <= mp_.max_n_mmaps);
-
-  assert((unsigned long)(av->system_mem) <=
-	 (unsigned long)(av->max_system_mem));
-
-  assert((unsigned long)(mp_.mmapped_mem) <=
-	 (unsigned long)(mp_.max_mmapped_mem));
 }
 #endif
 
-- 
1.8.4.2

