--- glibc-2.15/nss/nsswitch.c.orig	2012-01-01 10:16:32.000000000 -0200
+++ glibc-2.15/nss/nsswitch.c	2012-04-07 02:57:48.751819125 -0300
@@ -331,9 +331,20 @@ nss_load_library (service_user *ni)
       ni->library->lib_handle = __libc_dlopen (shlib_name);
       if (ni->library->lib_handle == NULL)
 	{
-	  /* Failed to load the library.  */
-	  ni->library->lib_handle = (void *) -1l;
-	  __set_errno (saved_errno);
+	  /* Failed to load the library. Try a fallback.  */
+	  int n = __snprintf(shlib_name, shlen, "libnss_%s.so.%d.%d",
+		   ni->library->name, __GLIBC__, __GLIBC_MINOR__);
+	  if (n >= shlen)
+	    ni->library->lib_handle = NULL;
+	  else
+	    ni->library->lib_handle = __libc_dlopen (shlib_name);
+
+	  if (ni->library->lib_handle == NULL)
+	    {
+	    /* Ok, really fail now.  */
+	    ni->library->lib_handle = (void *) -1l;
+	    __set_errno (saved_errno);
+	    }
 	}
       else if (is_nscd)
 	{
