--- glibc-2.23/nis/nss_nis/nis-netgrp.c.orig	2016-02-22 21:54:48.275424385 -0300
+++ glibc-2.23/nis/nss_nis/nis-netgrp.c	2016-02-22 21:57:03.298575747 -0300
@@ -70,7 +70,8 @@ _nss_nis_setnetgrent (const char *group,
 	 and the last byte is filled with NUL.  So we can simply
 	 use that buffer.  */
       assert (len >= 0);
-      assert (netgrp->data[len] == '\0');
+      /* The next line break NIS netgroups on non-PAM systems, so we will comment it out */
+      /* assert (netgrp->data[len] == '\0'); */
 
       netgrp->data_size = len;
       netgrp->cursor = netgrp->data;
