--- glibc-2.22/sysdeps/posix/nice.c.orig
+++ glibc-2.22/sysdeps/posix/nice.c
@@ -41,7 +41,12 @@ nice (int incr)
 	return -1;
     }
 
-  result = __setpriority (PRIO_PROCESS, 0, prio + incr);
+  prio += incr;
+  if (prio < PRIO_MIN)
+      prio = PRIO_MIN;
+  else if (prio >= PRIO_MAX)
+      prio = PRIO_MAX - 1;
+  result = __setpriority (PRIO_PROCESS, 0, prio);
   if (result == -1)
     {
       if (errno == EACCES)
