--- glibc-2.22/localedata/Makefile.orig	2015-08-05 19:27:39.591176778 -0300
+++ glibc-2.22/localedata/Makefile	2015-08-05 19:28:23.401869408 -0300
@@ -211,6 +211,7 @@ $(INSTALL-SUPPORTED-LOCALES): install-lo
 	echo -n '...'; \
 	input=`echo $$locale | sed 's/\([^.]*\)[^@]*\(.*\)/\1\2/'`; \
 	$(LOCALEDEF) --alias-file=../intl/locale.alias \
+		     --no-archive \
 		     -i locales/$$input -c -f charmaps/$$charset \
 		     $(addprefix --prefix=,$(install_root)) $$locale \
 	&& echo ' done'; \
