--- glibc-2.15/scripts/check-local-headers.sh.orig	2012-01-01 10:16:32.000000000 -0200
+++ glibc-2.15/scripts/check-local-headers.sh	2012-04-07 02:59:57.587501948 -0300
@@ -29,7 +29,7 @@ exec ${AWK} -v includedir="$includedir"
 BEGIN {
   status = 0
   exclude = "^" includedir \
-    "/(asm[-/]|linux/|selinux/|gd|nss3/|sys/capability\\.h|libaudit\\.h)"
+    "/(asm[-/]|linux/|selinux/|multiarch-|gd|nss3/|sys/capability\\.h|libaudit\\.h)"
 }
 /^[^ ]/ && $1 ~ /.*:/ { obj = $1 }
 {
