--- glibc-2.16.0/scripts/check-local-headers.sh.orig	2012-07-03 22:45:04.747042236 -0300
+++ glibc-2.16.0/scripts/check-local-headers.sh	2012-07-03 22:45:10.465809654 -0300
@@ -28,7 +28,7 @@ exec ${AWK} -v includedir="$includedir"
 BEGIN {
   status = 0
   exclude = "^" includedir \
-    "/(.*-.*-.*/|)(asm[-/]|linux/|selinux/|gd|nss3/|sys/capability\\.h|libaudit\\.h)"
+    "/(.*-.*-.*/|)(asm[-/]|linux/|selinux/|multiarch-|gd|nss3/|sys/capability\\.h|libaudit\\.h)"
 }
 /^[^ ]/ && $1 ~ /.*:/ { obj = $1 }
 {
