*** /tmp/ediff2864wKI	2010-10-06 08:30:42.639138469 -0400
--- mercurial/i18n.py	2010-10-06 08:29:26.720138409 -0400
***************
*** 15,21 ****
      module = __file__
  
  base = os.path.dirname(module)
! for dir in ('.', '..'):
      localedir = os.path.join(base, dir, 'locale')
      if os.path.isdir(localedir):
          break
--- 15,21 ----
      module = __file__
  
  base = os.path.dirname(module)
! for dir in ('.', '..', '/usr/share'):
      localedir = os.path.join(base, dir, 'locale')
      if os.path.isdir(localedir):
          break
