diff -ur clucene-core-0.9.21b/src/CLucene/CLConfig.h clucene-core-0.9.21b-enable-refcount/src/CLucene/CLConfig.h
--- clucene-core-0.9.21b/src/CLucene/CLConfig.h	2008-10-23 19:44:37.000000000 +0200
+++ clucene-core-0.9.21b-enable-refcount/src/CLucene/CLConfig.h	2014-10-25 19:08:04.000000000 +0200
@@ -147,7 +147,8 @@
 //If reference counting is enabled, the wrapper can add a reference to any class
 //and when _CLDECDELETE is called, the reference is decremented and only deleted
 //if the refcount is zero.
-//#define LUCENE_ENABLE_REFCOUNT
+//enabled because Qt Assistant requires this option
+#define LUCENE_ENABLE_REFCOUNT
 
 
 ////////////////////////////////////////////////////////////////////
