diff -ur clucene-core-0.9.21b/src/Makefile.am clucene-core-0.9.21b-fix-soversion/src/Makefile.am
--- clucene-core-0.9.21b/src/Makefile.am	2008-10-23 19:44:37.000000000 +0200
+++ clucene-core-0.9.21b-fix-soversion/src/Makefile.am	2014-10-25 21:16:25.000000000 +0200
@@ -3,7 +3,7 @@
 
 include_HEADERS = CLucene.h
 lib_LTLIBRARIES        = libclucene.la
-libclucene_la_LDFLAGS  = --version-info $(CLLIB_VERSION) --release $(PACKAGE_VERSION)
+libclucene_la_LDFLAGS  = -version-info $(CLLIB_VERSION)
 
 lsrcdir 	= $(top_srcdir)/src/CLucene
 EXTRA_DIST 	= $(lsrcdir)/CLMonolithic.cpp \
diff -ur clucene-core-0.9.21b/src/Makefile.in clucene-core-0.9.21b-fix-soversion/src/Makefile.in
--- clucene-core-0.9.21b/src/Makefile.in	2008-10-23 20:18:25.000000000 +0200
+++ clucene-core-0.9.21b-fix-soversion/src/Makefile.in	2014-10-25 21:16:33.000000000 +0200
@@ -281,7 +281,7 @@
 AUTOMAKE_OPTIONS = 1.6
 include_HEADERS = CLucene.h
 lib_LTLIBRARIES = libclucene.la
-libclucene_la_LDFLAGS = --version-info $(CLLIB_VERSION) --release $(PACKAGE_VERSION)
+libclucene_la_LDFLAGS = -version-info $(CLLIB_VERSION)
 lsrcdir = $(top_srcdir)/src/CLucene
 EXTRA_DIST = $(lsrcdir)/CLMonolithic.cpp \
               $(lsrcdir)/CMakeLists.txt \
