diff -up texmf-dist/scripts/context/ruby/base/exa.rb.rb19 texmf-dist/scripts/context/ruby/base/exa.rb
--- texmf-dist/scripts/context/ruby/base/exa.rb.rb19	2006-11-15 23:02:47.000000000 -0200
+++ texmf-dist/scripts/context/ruby/base/exa.rb	2010-08-06 19:57:26.128036431 -0300
@@ -3,7 +3,7 @@
 # tex.setup.setuplayout.width.[integer|real|dimension|string|key]
 # tex.[mp]var.whatever.width.[integer|real|dimension|string|key]
 
-require 'ftools'
+require 'fileutils'
 require 'md5'
 
 # this can become a lua thing
diff -up texmf-dist/scripts/context/ruby/base/file.rb.rb19 texmf-dist/scripts/context/ruby/base/file.rb
--- texmf-dist/scripts/context/ruby/base/file.rb.rb19	2006-11-15 23:02:47.000000000 -0200
+++ texmf-dist/scripts/context/ruby/base/file.rb	2010-08-06 19:57:26.128036431 -0300
@@ -8,7 +8,7 @@
 # info      : j.hagen@xs4all.nl
 # www       : www.pragma-ade.com
 
-require 'ftools'
+require 'fileutils'
 
 class File
 
diff -up texmf-dist/scripts/context/ruby/base/tool.rb.rb19 texmf-dist/scripts/context/ruby/base/tool.rb
--- texmf-dist/scripts/context/ruby/base/tool.rb.rb19	2010-08-06 19:58:03.098961612 -0300
+++ texmf-dist/scripts/context/ruby/base/tool.rb	2010-08-06 19:58:29.247250525 -0300
@@ -24,9 +24,9 @@ module Tool
                     t = Time.now
                     u = t.usec.to_s % [1..2] [0..3]
                     pth = t.strftime("#{mainpath}%Y%m%d-%H%M%S-#{u}-#{Process.pid}")
-                    if pth == $constructedtempdir
+                    loop do
+                        pth == $constructedtempdir
                         # sleep(0.01)
-                        retry
                     end
                     Dir.mkdir(pth) if create
                     $constructedtempdir = pth
diff -up texmf-dist/scripts/context/ruby/ctxtools.rb.rb19 texmf-dist/scripts/context/ruby/ctxtools.rb
--- texmf-dist/scripts/context/ruby/ctxtools.rb.rb19	2007-01-12 09:44:15.000000000 -0200
+++ texmf-dist/scripts/context/ruby/ctxtools.rb	2010-08-06 19:54:21.147437232 -0300
@@ -56,7 +56,7 @@ require 'base/file'
 
 require 'rexml/document'
 require 'net/http'
-require 'ftools'
+require 'fileutils'
 require 'kconv'
 
 exit if defined?(REQUIRE2LIB)
diff -up texmf-dist/scripts/context/ruby/graphics/gs.rb.rb19 texmf-dist/scripts/context/ruby/graphics/gs.rb
--- texmf-dist/scripts/context/ruby/graphics/gs.rb.rb19	2007-01-12 09:44:15.000000000 -0200
+++ texmf-dist/scripts/context/ruby/graphics/gs.rb	2010-08-06 19:57:26.130035994 -0300
@@ -13,7 +13,7 @@
 
 require 'base/variables'
 require 'base/system'
-require 'ftools'
+require 'fileutils'
 
 class GhostScript
 
diff -up texmf-dist/scripts/context/ruby/pdftools.rb.rb19 texmf-dist/scripts/context/ruby/pdftools.rb
--- texmf-dist/scripts/context/ruby/pdftools.rb.rb19	2007-01-12 09:44:15.000000000 -0200
+++ texmf-dist/scripts/context/ruby/pdftools.rb	2010-08-06 19:54:21.150436577 -0300
@@ -20,7 +20,7 @@ $: << File.expand_path(File.dirname($0))
 require 'base/switch'
 require 'base/logger'
 
-require 'ftools'
+require 'fileutils'
 
 class File
 
diff -up texmf-dist/scripts/context/ruby/rlxtools.rb.rb19 texmf-dist/scripts/context/ruby/rlxtools.rb
--- texmf-dist/scripts/context/ruby/rlxtools.rb.rb19	2007-01-12 09:44:15.000000000 -0200
+++ texmf-dist/scripts/context/ruby/rlxtools.rb	2010-08-06 19:54:21.151436359 -0300
@@ -19,7 +19,7 @@ require 'base/logger'
 require 'base/system'
 require 'base/kpse'
 
-require 'ftools'
+require 'fileutils'
 require 'rexml/document'
 
 class Commands
diff -up texmf-dist/scripts/context/ruby/rsfiltool.rb.rb19 texmf-dist/scripts/context/ruby/rsfiltool.rb
--- texmf-dist/scripts/context/ruby/rsfiltool.rb.rb19	2006-11-15 23:02:47.000000000 -0200
+++ texmf-dist/scripts/context/ruby/rsfiltool.rb	2010-08-06 19:54:21.151436359 -0300
@@ -18,7 +18,7 @@ end
 
 # todo : split session stuff from xmpl/base into an xmpl/session module and "include xmpl/session" into base and here and ...
 
-require 'ftools'
+require 'fileutils'
 require 'xmpl/base'
 require 'xmpl/switch'
 require 'xmpl/request'
diff -up texmf-dist/scripts/context/ruby/runtools.rb.rb19 texmf-dist/scripts/context/ruby/runtools.rb
--- texmf-dist/scripts/context/ruby/runtools.rb.rb19	2006-11-15 23:02:47.000000000 -0200
+++ texmf-dist/scripts/context/ruby/runtools.rb	2010-08-06 19:54:21.152436141 -0300
@@ -1,5 +1,5 @@
 require 'timeout'
-require 'ftools'
+require 'fileutils'
 require 'rbconfig'
 
 class File
diff -up texmf-dist/scripts/context/ruby/texexec.rb.rb19 texmf-dist/scripts/context/ruby/texexec.rb
--- texmf-dist/scripts/context/ruby/texexec.rb.rb19	2007-01-12 09:44:15.000000000 -0200
+++ texmf-dist/scripts/context/ruby/texexec.rb	2010-08-06 19:54:21.153435922 -0300
@@ -2,7 +2,7 @@ banner = ['TeXExec', 'version 6.2.0', '1
 
 $: << File.expand_path(File.dirname($0)) ; $: << File.join($:.last,'lib') ; $:.uniq!
 
-require 'ftools'     # needed ?
+require 'fileutils'     # needed ?
 
 require 'base/switch'
 require 'base/logger'
diff -up texmf-dist/scripts/context/ruby/textools.rb.rb19 texmf-dist/scripts/context/ruby/textools.rb
--- texmf-dist/scripts/context/ruby/textools.rb.rb19	2006-12-01 08:42:41.000000000 -0200
+++ texmf-dist/scripts/context/ruby/textools.rb	2010-08-06 19:54:21.157435048 -0300
@@ -20,7 +20,7 @@ $: << File.expand_path(File.dirname($0))
 require 'base/switch'
 require 'base/logger'
 
-require 'ftools'
+require 'fileutils'
 
 # Remark
 #
diff -up texmf-dist/scripts/context/ruby/www/lib.rb.rb19 texmf-dist/scripts/context/ruby/www/lib.rb
--- texmf-dist/scripts/context/ruby/www/lib.rb.rb19	2006-11-17 23:37:52.000000000 -0200
+++ texmf-dist/scripts/context/ruby/www/lib.rb	2010-08-06 19:57:26.133035339 -0300
@@ -18,7 +18,7 @@
 
 require 'base/variables'
 
-require 'ftools'
+require 'fileutils'
 require 'fileutils'
 require 'tempfile'
 require 'timeout'
diff -up texmf-dist/scripts/context/ruby/wwwserver.rb.rb19 texmf-dist/scripts/context/ruby/wwwserver.rb
--- texmf-dist/scripts/context/ruby/wwwserver.rb.rb19	2006-11-17 23:37:52.000000000 -0200
+++ texmf-dist/scripts/context/ruby/wwwserver.rb	2010-08-06 19:54:21.159434611 -0300
@@ -20,7 +20,7 @@ require 'www/login'
 require 'www/exa'
 
 require 'tempfile'
-require 'ftools'
+require 'fileutils'
 require 'webrick'
 
 class Server
diff -up texmf-dist/scripts/context/ruby/wwwwatch.rb.rb19 texmf-dist/scripts/context/ruby/wwwwatch.rb
--- texmf-dist/scripts/context/ruby/wwwwatch.rb.rb19	2006-11-17 23:37:52.000000000 -0200
+++ texmf-dist/scripts/context/ruby/wwwwatch.rb	2010-08-06 19:54:21.159434611 -0300
@@ -11,7 +11,7 @@ require 'www/common'
 
 require 'monitor'
 require 'fileutils'
-require 'ftools'
+require 'fileutils'
 require 'tempfile'
 require 'timeout'
 require 'thread'
