--- texmf/web2c/fmtutil.cnf.orig	2007-01-23 17:46:02.000000000 -0200
+++ texmf/web2c/fmtutil.cnf	2011-02-07 23:17:04.758262125 -0200
@@ -74,7 +74,7 @@ amstex		pdftex	-		-translate-file=cp227.
 #! cont-de	pdftex	cont-usr.tex	-8bit *cont-de.ini
 #! cont-de	xetex	cont-usr.tex	-8bit *cont-de.ini
 cont-en		pdftex	cont-usr.tex	-8bit *cont-en.ini
-cont-en		xetex	cont-usr.tex	-8bit *cont-en.ini
+#! cont-en		xetex	cont-usr.tex	-8bit *cont-en.ini
 #! cont-nl	pdftex	cont-usr.tex	-8bit *cont-nl.ini
 #! cont-nl	xetex	cont-usr.tex	-8bit *cont-nl.ini
 #! cont-uk	pdftex	cont-usr.tex	-8bit *cont-uk.ini
