diff -up ./configure.ac.orig ./configure.ac
--- ./configure.ac.orig	2015-07-08 22:07:23.117117124 +0300
+++ ./configure.ac	2015-07-08 22:07:43.673324210 +0300
@@ -41,7 +41,7 @@ m4_define([gimp_full_name], [GNU Image M
 
 # required versions of other packages
 m4_define([babl_required_version], [0.1.10])
-m4_define([gegl_required_version], [0.2.0])
+m4_define([gegl_required_version], [0.3.0])
 m4_define([glib_required_version], [2.30.2])
 m4_define([atk_required_version], [2.2.0])
 m4_define([gtk_required_version], [2.24.10])
@@ -519,7 +519,7 @@ AC_SUBST(ISO_CODES_LOCALEDIR)
 ###############################
 
 PKG_CHECK_MODULES(BABL, babl >= babl_required_version)
-PKG_CHECK_MODULES(GEGL, gegl-0.2 >= gegl_required_version)
+PKG_CHECK_MODULES(GEGL, gegl-0.3 >= gegl_required_version)
 PKG_CHECK_MODULES(ATK, atk >= atk_required_version)
 
 AM_PATH_GLIB_2_0(glib_required_version, :,
