diff -up kdevelop-4.2.81/CMakeLists.txt.no-rpath kdevelop-4.2.81/CMakeLists.txt
--- kdevelop-4.2.81/CMakeLists.txt.no-rpath	2012-01-23 11:57:24.890826586 +0100
+++ kdevelop-4.2.81/CMakeLists.txt	2012-01-23 11:57:40.104206627 +0100
@@ -23,7 +23,6 @@ list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_D
 if("${_isSystemLibDir}" STREQUAL "-1")
  set(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}")
 endif("${_isSystemLibDir}" STREQUAL "-1")
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 find_package(KDevPlatform 1.2.80 REQUIRED)
 
