diff -up kdevelop-4.2.81/debuggers/gdb/unittests/debugeeslow.cpp.include-unistd-h kdevelop-4.2.81/debuggers/gdb/unittests/debugeeslow.cpp
--- kdevelop-4.2.81/debuggers/gdb/unittests/debugeeslow.cpp.include-unistd-h	2012-01-15 01:51:45.000000000 +0100
+++ kdevelop-4.2.81/debuggers/gdb/unittests/debugeeslow.cpp	2012-01-25 08:37:38.341179180 +0100
@@ -17,6 +17,7 @@
 */
 
 #include <iostream>
+#include <unistd.h>
 
 void foo() {
     static int i=0;
