diff -up QtZeitgeist-0.7.0-Source/QtZeitgeist.pc.in.pkgconfig_version QtZeitgeist-0.7.0-Source/QtZeitgeist.pc.in
--- QtZeitgeist-0.7.0-Source/QtZeitgeist.pc.in.pkgconfig_version	2011-05-20 07:47:17.000000000 -0500
+++ QtZeitgeist-0.7.0-Source/QtZeitgeist.pc.in	2011-05-20 07:55:53.973084817 -0500
@@ -5,7 +5,7 @@ includedir=${CMAKE_INSTALL_PREFIX}/inclu
 
 Name: QtZeitgeist
 Description: Qt Zeitgeist library
-Version: 0.0.1
+Version: ${CPACK_PACKAGE_VERSION}
 Requires: QtCore QtDBus
 Libs: -L${libdir} -lQtZeitgeist
 Cflags: -I${CMAKE_INSTALL_PREFIX}/include/QtZeitgeist -I${CMAKE_INSTALL_PREFIX}/include/
