diff --git gsoap-2.7.orig/gsoap/samples/autotest/Makefile.am gsoap-2.7/gsoap/samples/autotest/Makefile.am
index 20af1a5..cddc41b 100644
--- gsoap-2.7.orig/gsoap/samples/autotest/Makefile.am
+++ gsoap-2.7/gsoap/samples/autotest/Makefile.am
@@ -22,7 +22,28 @@ $(SOAP_CPP_SRC) : $(WSDLINPUT)
 
 #LIBS=
 
-noinst_PROGRAMS=autotest
+autotestdir = $(pkgdatadir)/examples/autotest
+
+nobase_autotest_PROGRAMS=autotest
+
+nobase_autotest_DATA = 	LICENSE.txt 			\
+						README.txt				\
+						ChameleonIncluded.xsd	\
+						test-patterns11.sh		\
+						redefineschema.xsd		\
+						test-patterns12.sh		\
+						examples.wsdl			\
+						Makefile.am  			\
+						RelativeIncluded.xsd	\
+						test-self.sh			\
+						Included.xsd          	\
+						Makefile.in  			\
+						strict.xsd				\
+						typemap.dat
+
+install-data-local:
+	$(MKDIR_P) $(DESTDIR)$(autotestdir)
+	cp -r databinding $(DESTDIR)$(autotestdir)
 
 autotest_CFLAGS=$(C_DEBUG_FLAGS) -D$(platform)
 autotest_CPPFLAGS=$(C_DEBUG_FLAGS) -D$(platform)
diff --git gsoap-2.7.orig/gsoap/samples/calc++/Makefile.am gsoap-2.7/gsoap/samples/calc++/Makefile.am
index 490a4f0..6c6155e 100644
--- gsoap-2.7.orig/gsoap/samples/calc++/Makefile.am
+++ gsoap-2.7/gsoap/samples/calc++/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = calcclient calcserver
+calcdir = $(pkgdatadir)/examples/calc++
+nobase_calc_PROGRAMS = calcclient calcserver
+nobase_calc_DATA =	calcclient.cpp	\
+					calc.h			\
+					calcserver.cpp	\
+					Makefile.am 	\
+					Makefile.in  	\
+					README.txt
+
 SOAPHEADER = calc.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/calc/Makefile.am gsoap-2.7/gsoap/samples/calc/Makefile.am
index 0224270..22c0181 100644
--- gsoap-2.7.orig/gsoap/samples/calc/Makefile.am
+++ gsoap-2.7/gsoap/samples/calc/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = calcclient calcserver
+calcdir = $(pkgdatadir)/examples/calc
+nobase_calc_PROGRAMS = calcclient calcserver
+nobase_calc_DATA = 	calcclient.c	\
+					calc.h  		\
+					calcserver.c  	\
+					Makefile.am  	\
+					Makefile.in  	\
+					README.txt
+
 SOAPHEADER = calc.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/databinding/Makefile.am gsoap-2.7/gsoap/samples/databinding/Makefile.am
index abe5404..bc77843 100644
--- gsoap-2.7.orig/gsoap/samples/databinding/Makefile.am
+++ gsoap-2.7/gsoap/samples/databinding/Makefile.am
@@ -22,7 +22,20 @@ $(SOAP_CPP_SRC) : $(WSDLINPUT)
 
 #LIBS=
 
-noinst_PROGRAMS=address
+addresdir = $(pkgdatadir)/examples/databinding
+nobase_addres_PROGRAMS=address
+nobase_addres_DATA = 	address.cpp 		\
+						addresstypemap.dat  \
+						Doxyfile  			\
+						Makefile.am			\
+						README.txt			\
+						address.h 			\
+						address.xsd			\
+						Makefile.in
+
+install-data-local:
+	$(MKDIR_P) $(DESTDIR)$(addresdir)
+	cp -r html $(DESTDIR)$(addresdir)
 
 address_CFLAGS=$(C_DEBUG_FLAGS) -D$(platform)
 address_CPPFLAGS=$(C_DEBUG_FLAGS) -D$(platform)
diff --git gsoap-2.7.orig/gsoap/samples/dime/Makefile.am gsoap-2.7/gsoap/samples/dime/Makefile.am
index d382ae8..bad1ef6 100644
--- gsoap-2.7.orig/gsoap/samples/dime/Makefile.am
+++ gsoap-2.7/gsoap/samples/dime/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = dimeclient dimeserver
+dimedir = $(pkgdatadir)/examples/dime
+nobase_dime_PROGRAMS = dimeclient dimeserver
+nobase_dime_DATA = 	dimeclient.cpp 	\
+					dime.h			\
+					dimeserver.cpp	\
+					Makefile.am		\
+					Makefile.in		\
+					README.txt		
+
 SOAPHEADER = dime.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/dom/Makefile.am gsoap-2.7/gsoap/samples/dom/Makefile.am
index 22fa6c7..603bc87 100644
--- gsoap-2.7.orig/gsoap/samples/dom/Makefile.am
+++ gsoap-2.7/gsoap/samples/dom/Makefile.am
@@ -3,7 +3,14 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = dom2calc
+domdir = $(pkgdatadir)/examples/dom
+nobase_dom_PROGRAMS = dom2calc
+nobase_dom_DATA =	dom2calc.cpp	\
+					dom2calc.h		\
+					Makefile.am		\
+					Makefile.in		\
+					README.txt
+
 SOAPHEADER = dom2calc.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/factory/Makefile.am gsoap-2.7/gsoap/samples/factory/Makefile.am
index f625c08..e918b00 100644
--- gsoap-2.7.orig/gsoap/samples/factory/Makefile.am
+++ gsoap-2.7/gsoap/samples/factory/Makefile.am
@@ -3,7 +3,13 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = factory
+factorydir = $(pkgdatadir)/examples/factory
+nobase_factory_PROGRAMS = factory
+nobase_factory_DATA =	factory.cpp	\
+						factory.h	\
+						Makefile.am \
+						Makefile.in \
+						README.txt
 SOAPHEADER = factory.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/factorytest/Makefile.am gsoap-2.7/gsoap/samples/factorytest/Makefile.am
index 95516fa..c1f01f0 100644
--- gsoap-2.7.orig/gsoap/samples/factorytest/Makefile.am
+++ gsoap-2.7/gsoap/samples/factorytest/Makefile.am
@@ -3,7 +3,14 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = factorytest
+factorytestdir = $(pkgdatadir)/examples/factorytest
+nobase_factorytest_PROGRAMS = factorytest
+nobase_factorytest_DATA =	factorytest.cpp	\
+							factorytest.h  	\
+							Makefile.am  	\
+							Makefile.in  	\
+							README.txt
+
 SOAPHEADER = factorytest.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/gmt/Makefile.am gsoap-2.7/gsoap/samples/gmt/Makefile.am
index 9dfdf5e..3771356 100644
--- gsoap-2.7.orig/gsoap/samples/gmt/Makefile.am
+++ gsoap-2.7/gsoap/samples/gmt/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = gmtclient gmtserver
+gmtdir = $(pkgdatadir)/examples/gmt
+nobase_gmt_PROGRAMS = gmtclient gmtserver
+nobase_gmt_DATA =	gmtclient.cpp	\
+					gmt.h  			\
+					gmtserver.cpp  	\
+					Makefile.am  	\
+					Makefile.in  	\
+					README.txt
+
 SOAPHEADER = gmt.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/googleapi/Makefile.am gsoap-2.7/gsoap/samples/googleapi/Makefile.am
index b64ee33..f7ffc55 100644
--- gsoap-2.7.orig/gsoap/samples/googleapi/Makefile.am
+++ gsoap-2.7/gsoap/samples/googleapi/Makefile.am
@@ -3,7 +3,14 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = googleapi
+googleapidir = $(pkgdatadir)/examples/googleapi
+googleapi_PROGRAMS = googleapi
+googleapi_DATA =	googleapi.c \
+					googleapi.h \
+					Makefile.am \
+					Makefile.in \
+					README.txt	
+
 SOAPHEADER = googleapi.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/hello/Makefile.am gsoap-2.7/gsoap/samples/hello/Makefile.am
index 84d38b1..b7a1728 100644
--- gsoap-2.7.orig/gsoap/samples/hello/Makefile.am
+++ gsoap-2.7/gsoap/samples/hello/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = helloclient helloserver
+hellodir = $(pkgdatadir)/examples/hello
+nobase_hello_PROGRAMS = helloclient helloserver
+nobase_hello_DATA = helloclient.cpp  \
+					hello.h          \
+					helloserver.cpp  \
+					Makefile.am      \
+					Makefile.in  	 \
+					README.txt
+
 SOAPHEADER = hello.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/httpcookies/Makefile.am gsoap-2.7/gsoap/samples/httpcookies/Makefile.am
index 5bc4cce..eb4d01b 100644
--- gsoap-2.7.orig/gsoap/samples/httpcookies/Makefile.am
+++ gsoap-2.7/gsoap/samples/httpcookies/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = ck.cgi ckclient
+httpcookiesdir = $(pkgdatadir)/examples/httpcookies
+nobase_httpcookies_PROGRAMS = ck.cgi ckclient
+nobase_httpcookies_DATA = 	ckclient.cpp	\
+							ck.h  			\
+							ckserver.cpp  	\
+							Makefile.am  	\
+							Makefile.in  	\
+							README.txt
+
 SOAPHEADER = ck.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/oneway++/Makefile.am gsoap-2.7/gsoap/samples/oneway++/Makefile.am
index b7ecb0c..95170bc 100644
--- gsoap-2.7.orig/gsoap/samples/oneway++/Makefile.am
+++ gsoap-2.7/gsoap/samples/oneway++/Makefile.am
@@ -3,7 +3,14 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = event handler
+onewaydir = $(pkgdatadir)/examples/oneway++
+nobase_oneway_PROGRAMS = event handler
+nobase_oneway_DATA = 	event.cpp  	\
+						event.h  	\
+						handler.cpp \
+						Makefile.am \
+						Makefile.in \
+						README.txt
 SOAPHEADER = event.h
 
 include ../Makefile.defines
diff --git gsoap-2.7.orig/gsoap/samples/oneway/Makefile.am gsoap-2.7/gsoap/samples/oneway/Makefile.am
index cb527b6..def5e16 100644
--- gsoap-2.7.orig/gsoap/samples/oneway/Makefile.am
+++ gsoap-2.7/gsoap/samples/oneway/Makefile.am
@@ -3,7 +3,15 @@
 ## you have all needed files, that a GNU package needs
 AUTOMAKE_OPTIONS = foreign 1.4
 
-noinst_PROGRAMS = event handler
+onewaydir = $(pkgdatadir)/examples/oneway
+nobase_oneway_PROGRAMS = event handler
+nobase_oneway_DATA =	event.c		\
+						event.h  	\
+						handler.c  	\
+						Makefile.am \
+						Makefile.in \
+						README.txt
+
 SOAPHEADER = event.h
 
 include ../Makefile.defines
