diff -ur gsoap-2.8.orig/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.cpp gsoap-2.8/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.cpp
--- gsoap-2.8.orig/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.cpp	2014-08-24 18:38:54.000000000 +0200
+++ gsoap-2.8/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.cpp	2014-09-11 15:37:46.494978203 +0200
@@ -4122,6 +4122,11 @@
             soap_set_sender_error(soap, "Timeout", "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4132,6 +4137,11 @@
             soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4146,6 +4156,11 @@
         soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
         soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+        if (res->ai_next)
+        {
+          res = res->ai_next;
+          goto again;
+        }
         freeaddrinfo(ressave);
 #endif
         return SOAP_INVALID_SOCKET;
diff -ur gsoap-2.8.orig/gsoap/stdsoap2.c gsoap-2.8/gsoap/stdsoap2.c
--- gsoap-2.8.orig/gsoap/stdsoap2.c	2014-08-24 18:38:55.000000000 +0200
+++ gsoap-2.8/gsoap/stdsoap2.c	2014-09-11 15:37:46.497978168 +0200
@@ -4122,6 +4122,11 @@
             soap_set_sender_error(soap, "Timeout", "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4132,6 +4137,11 @@
             soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4146,6 +4156,11 @@
         soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
         soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+        if (res->ai_next)
+        {
+          res = res->ai_next;
+          goto again;
+        }
         freeaddrinfo(ressave);
 #endif
         return SOAP_INVALID_SOCKET;
diff -ur gsoap-2.8.orig/gsoap/stdsoap2.cpp gsoap-2.8/gsoap/stdsoap2.cpp
--- gsoap-2.8.orig/gsoap/stdsoap2.cpp	2014-08-24 18:38:55.000000000 +0200
+++ gsoap-2.8/gsoap/stdsoap2.cpp	2014-09-11 15:37:46.499978145 +0200
@@ -4122,6 +4122,11 @@
             soap_set_sender_error(soap, "Timeout", "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4132,6 +4137,11 @@
             soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4146,6 +4156,11 @@
         soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
         soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+        if (res->ai_next)
+        {
+          res = res->ai_next;
+          goto again;
+        }
         freeaddrinfo(ressave);
 #endif
         return SOAP_INVALID_SOCKET;
diff -ur gsoap-2.8.orig/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.cpp gsoap-2.8/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.cpp
--- gsoap-2.8.orig/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.cpp	2014-08-24 18:38:55.000000000 +0200
+++ gsoap-2.8/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.cpp	2014-09-11 15:37:46.502978110 +0200
@@ -4122,6 +4122,11 @@
             soap_set_sender_error(soap, "Timeout", "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4132,6 +4137,11 @@
             soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
             soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+            if (res->ai_next)
+            {
+              res = res->ai_next;
+              goto again;
+            }
             freeaddrinfo(ressave);
 #endif
             return SOAP_INVALID_SOCKET;
@@ -4146,6 +4156,11 @@
         soap_set_sender_error(soap, tcp_error(soap), "connect failed in tcp_connect()", SOAP_TCP_ERROR);
         soap->fclosesocket(soap, sk);
 #ifdef WITH_IPV6
+        if (res->ai_next)
+        {
+          res = res->ai_next;
+          goto again;
+        }
         freeaddrinfo(ressave);
 #endif
         return SOAP_INVALID_SOCKET;
