diff -up gsoap-2.7/gsoap/samples/factorytest/factorytest.cpp.iostream gsoap-2.7/gsoap/samples/factorytest/factorytest.cpp
--- gsoap-2.7/gsoap/samples/factorytest/factorytest.cpp.iostream	2010-07-26 18:31:00.000000000 +0600
+++ gsoap-2.7/gsoap/samples/factorytest/factorytest.cpp	2010-07-26 18:31:22.000000000 +0600
@@ -49,7 +49,9 @@ A commercial use license is available fr
 #include "soapH.h"
 #include "factorytest.nsmap"
 
-#include <iostream.h>
+#include <iostream>
+
+using namespace std;
 
 // default factory service endpoint:
 const char *factory = "http://localhost:18085";
diff -up gsoap-2.7/gsoap/samples/lu/luclient.cpp.iostream gsoap-2.7/gsoap/samples/lu/luclient.cpp
--- gsoap-2.7/gsoap/samples/lu/luclient.cpp.iostream	2010-07-26 18:29:34.000000000 +0600
+++ gsoap-2.7/gsoap/samples/lu/luclient.cpp	2010-07-26 18:29:58.000000000 +0600
@@ -32,7 +32,9 @@ A commercial use license is available fr
 */
 
 #include "soapH.h"
-#include <iostream.h>
+#include <iostream>
+
+using namespace std;
 
 const char luserver[] = "http://websrv.cs.fsu.edu/~engelen/luserver.cgi";
 
