--- gsoap/stdsoap2.c.orig	2013-01-09 22:49:08.983666641 +0100
+++ gsoap/stdsoap2.c	2013-01-09 22:50:19.587915594 +0100
@@ -4212,7 +4212,7 @@ again:
           { X509_EXTENSION *ext = X509_get_ext(peer, i);
             const char *ext_str = OBJ_nid2sn(OBJ_obj2nid(X509_EXTENSION_get_object(ext)));
             if (ext_str && !strcmp(ext_str, "subjectAltName"))
-            { X509V3_EXT_METHOD *meth = (X509V3_EXT_METHOD*)X509V3_EXT_get(ext);
+            { const X509V3_EXT_METHOD *meth = X509V3_EXT_get(ext);
               unsigned char *data;
               if (!meth)
                 break;
--- gsoap/stdsoap2.cpp.orig	2013-01-09 22:51:09.381121892 +0100
+++ gsoap/stdsoap2.cpp	2013-01-09 22:51:39.993307100 +0100
@@ -4212,7 +4212,7 @@ again:
           { X509_EXTENSION *ext = X509_get_ext(peer, i);
             const char *ext_str = OBJ_nid2sn(OBJ_obj2nid(X509_EXTENSION_get_object(ext)));
             if (ext_str && !strcmp(ext_str, "subjectAltName"))
-            { X509V3_EXT_METHOD *meth = (X509V3_EXT_METHOD*)X509V3_EXT_get(ext);
+            { const X509V3_EXT_METHOD *meth = X509V3_EXT_get(ext);
               unsigned char *data;
               if (!meth)
                 break;
