diff -ur gsoap-2.8.12.orig/gsoap/samples/chaining/Makefile.am gsoap-2.8.12/gsoap/samples/chaining/Makefile.am
--- gsoap-2.8.12.orig/gsoap/samples/chaining/Makefile.am	2012-12-11 01:36:45.000000000 +0100
+++ gsoap-2.8.12/gsoap/samples/chaining/Makefile.am	2013-01-10 10:32:10.265892484 +0100
@@ -22,4 +22,4 @@
 SOAP_NS = c.nsmap
 
 chaining_SOURCES = chaining.c quote.h calc.h env.h qServerLib.c cServerLib.c envC.c
-chaining_LDADD = $(SOAP_C_LIB) $(SAMPLE_EXTRA_LIBS)
+chaining_LDADD = $(SOAP_C_LIB) $(SAMPLE_EXTRA_LIBS) -lm
