--- src/config.c.orig	2015-02-18 02:45:59.000000000 -0200
+++ src/config.c	2015-02-18 03:29:12.894735530 -0200
@@ -458,11 +458,11 @@ void JE_initProcessorType( void )
 	/* SYN: Originally this proc looked at your hardware specs and chose appropriate options. We don't care, so I'll just set
 	   decent defaults here. */
 
-	wild = false;
-	superWild = false;
+	wild = true;
+	superWild = true;
 	smoothScroll = true;
 	explosionTransparent = true;
-	filtrationAvail = false;
+	filtrationAvail = true;
 	background2 = true;
 	displayScore = true;
 
