=== modified file 'filter/pdftoopvp/OPVPOutputDev.cxx'
--- filter/pdftoopvp/OPVPOutputDev.cxx	2015-02-18 03:17:28 +0000
+++ filter/pdftoopvp/OPVPOutputDev.cxx	2015-07-14 19:22:49 +0000
@@ -1811,10 +1811,14 @@
 #if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
   maskSplash->drawImage(&imageSrc, &imgMaskData,
 			splashModeMono8, gFalse, maskWidth, maskHeight, mat);
-#else
+#elif POPPLER_VERSION_MAJOR <= 0 && POPPLER_VERSION_MINOR <= 33
   maskSplash->drawImage(&imageSrc, &imgMaskData,
 			splashModeMono8, gFalse, maskWidth, maskHeight,
                         mat,gFalse);
+#else
+  maskSplash->drawImage(&imageSrc, 0, &imgMaskData,
+                          splashModeMono8, gFalse, maskWidth, maskHeight,
+			                          mat,gFalse);
 #endif
   delete imgMaskData.imgStr;
   maskStr->close();

=== modified file 'filter/pdftoopvp/oprs/OPRS.cxx'
--- filter/pdftoopvp/oprs/OPRS.cxx	2013-01-09 10:42:32 +0000
+++ filter/pdftoopvp/oprs/OPRS.cxx	2015-07-14 19:22:49 +0000
@@ -240,8 +240,10 @@
     if (rasterMode) {
 #if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
 	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
-#else
+#elif POPPLER_VERSION_MAJOR <= 0 && POPPLER_VERSION_MINOR <= 33
 	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
+#else
+	return splash->drawImage(src,0,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
 #endif
     } else {
 	return opvpSplash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);

