--- jed-B0.99-12/src/xkeys.c.xkeys	Sat Oct 21 23:15:58 2000
+++ jed-B0.99-12/src/xkeys.c	Thu Dec 28 12:24:21 2000
@@ -943,17 +943,17 @@
    "\003\033OQ",	/* 0xFF92 	 XK_KP_F2        PF2 */
    "\003\033OR",	/* 0xFF93 	 XK_KP_F3        PF3 */
    "\003\033OS",	/* 0xFF94 	 XK_KP_F4        PF4 */
-   "",			/* 0xFF95 */
-   "",			/* 0xFF96 */
-   "",			/* 0xFF97 */
-   "",			/* 0xFF98 */
-   "",			/* 0xFF99 */
-   "",			/* 0xFF9A */
-   "",			/* 0xFF9B */
-   "",			/* 0xFF9C */
-   "",			/* 0xFF9D */
-   "",			/* 0xFF9E */
-   "",			/* 0xFF9F */
+   "\003\033Ow",	/* 0xFF95 	 XK_KP_Home */
+   "\003\033Ot",	/* 0xFF96 	 XK_KP_Left */
+   "\003\033Ox",	/* 0xFF97 	 XK_KP_Up */
+   "\003\033Ov",	/* 0xFF98 	 XK_KP_Right */
+   "\003\033Or",	/* 0xFF99 	 XK_KP_Down */
+   "\003\033Oy",	/* 0xFF9A 	 XK_KP_Prior */
+   "\003\033Os",	/* 0xFF9B 	 XK_KP_Next */
+   "\003\033Oq",	/* 0xFF9C 	 XK_KP_End */
+   "\003\033Ou",	/* 0xFF9D 	 XK_KP_Begin */
+   "\003\033Op",	/* 0xFF9E 	 XK_KP_Insert */
+   "\003\033On",	/* 0xFF9F 	 XK_KP_Delete */
    "",			/* 0xFFA0 */
    "",			/* 0xFFA1 */
    "",			/* 0xFFA2 */
