--- jed-0.99-19.org/lib/site.sl	2010-01-16 12:48:54.802785974 +0530
+++ jed-0.99-19/lib/site.sl	2010-01-16 12:42:49.525739941 +0530
@@ -3309,13 +3309,7 @@
      $1 = getenv ("JED_CONF_DIR");
      if ($1 == NULL)
        {
-	  $1 = guess_jed_install_prefix ();
-	  if ($1 != NULL)
-	    {
-	       $1 = path_concat ($1, "etc");
-	       if (($1 == "/usr/etc") and (0 == file_status ($1)))
-		 $1 = "/etc";
-	    }
+          $1 = "/etc";
        }
      if ($1 != NULL)
        {
