--- jed-0.99-19.org/autoconf/aclocal.m4	2010-01-16 12:38:55.849789880 +0530
+++ jed-0.99-19/autoconf/aclocal.m4	2010-01-16 12:57:11.071786401 +0530
@@ -218,7 +218,7 @@
 
 JD_Search_Dirs="$JD_Search_Dirs \
                 $jd_prefix_incdir,$jd_prefix_libdir \
-		$HOME/include,$HOME/lib"
+		$HOME/include,$HOME/lib64"
 
 if test -n "$ARCH"
 then
@@ -231,10 +231,10 @@
 # version.
 
 JD_Search_Dirs="$JD_Search_Dirs \
-                /usr/local/include,/usr/local/lib \
-		/usr/include,/usr/lib \
-		/usr/include/$1,/usr/lib \
-		/usr/include/$1,/usr/lib/$1"
+    /usr/local/include,/usr/local/lib64 \
+    /usr/include,/usr/lib64 \
+    /usr/include/$1,/usr/lib64 \
+    /usr/include/$1,/usr/lib64/$1"
 
 echo looking for the JD_UP_NAME library
 
@@ -290,7 +290,7 @@
     JD_UP_NAME[]_INC=""
 fi
 
-if test "[$]JD_UP_NAME[]_LIB" = "-L/usr/lib"
+if test "[$]JD_UP_NAME[]_LIB" = "-L/usr/lib64"
 then
     JD_UP_NAME[]_LIB=""
     RPATH_[]JD_UP_NAME=""
@@ -510,7 +510,6 @@
 AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_GCC_TRADITIONAL
-AC_ISC_POSIX
 
 dnl #This stuff came from Yorick config script
 dnl
@@ -775,7 +774,7 @@
  fi
 
  AC_ARG_WITH($1,
-  [  --with-$1=DIR      Use DIR/lib and DIR/include for $1],
+  [  --with-$1=DIR      Use DIR/lib64 and DIR/include for $1],
   [jd_with_$1_arg=$withval], [jd_with_$1_arg=unspecified])
   
  case "x$jd_with_$1_arg" in
@@ -794,7 +793,7 @@
    *)
     jd_with_$1_library="yes"
     jd_$1_include_dir="$jd_with_$1_arg"/include
-    jd_$1_library_dir="$jd_with_$1_arg"/lib
+    jd_$1_library_dir="$jd_with_$1_arg"/lib64
     ;;
  esac
 
@@ -862,15 +861,15 @@
     then
       inc_and_lib_dirs="\
          $jd_prefix_incdir,$jd_prefix_libdir \
-	 /usr/local/$1/include,/usr/local/$1/lib \
-	 /usr/local/include/$1,/usr/local/lib \
-	 /usr/local/include,/usr/local/lib \
-	 /usr/include/$1,/usr/lib \
-	 /usr/$1/include,/usr/$1/lib \
-	 /usr/include,/usr/lib \
-	 /opt/include/$1,/opt/lib \
-	 /opt/$1/include,/opt/$1/lib \
-	 /opt/include,/opt/lib"
+	 /usr/local/$1/include,/usr/local/$1/lib64 \
+	 /usr/local/include/$1,/usr/local/lib64 \
+	 /usr/local/include,/usr/local/lib64 \
+	 /usr/include/$1,/usr/lib64 \
+	 /usr/$1/include,/usr/$1/lib64 \
+	 /usr/include,/usr/lib64 \
+	 /opt/include/$1,/opt/lib64 \
+	 /opt/$1/include,/opt/$1/lib64 \
+	 /opt/include,/opt/lib64"
 	 
       if test X$3 != X
       then
@@ -924,10 +923,10 @@
   then
     AC_MSG_RESULT(yes: $jd_$1_library_dir and $jd_$1_include_dir)
     jd_with_$1_library="yes"
-    dnl#  Avoid using /usr/lib and /usr/include because of problems with
+    dnl#  Avoid using /usr/lib64 and /usr/include because of problems with
     dnl#  gcc on some solaris systems.
     JD_ARG1[]_LIB=-L$jd_$1_library_dir
-    if test "X$jd_$1_library_dir" = "X/usr/lib"
+    if test "X$jd_$1_library_dir" = "X/usr/lib64"
     then
       JD_ARG1[]_LIB=""
     else
