From: Ansgar Burchardt <ansgar@43-1.org>

See http://bugs.debian.org/500415

diff --git a/src/CAudio.cpp b/src/CAudio.cpp
index 6adc59b..4d38015 100644
--- a/src/CAudio.cpp
+++ b/src/CAudio.cpp
@@ -72,7 +72,7 @@ bool Audio::loadSound(int i, char *filename)
 	}
 
 	#if USEPAK
-		engine->unpack(filename, PAK_SOUND);
+	if (engine->unpack(filename, PAK_SOUND))
 		sound[i] = Mix_LoadWAV_RW(engine->sdlrw, 1);
 	#else
 		sound[i] = Mix_LoadWAV(filename);
@@ -111,7 +111,7 @@ bool Audio::loadMusic(char *filename)
 	}
 
 	#if USEPAK
-		engine->unpack(filename, PAK_MUSIC);
+	if (engine->unpack(filename, PAK_MUSIC))
 		music = Mix_LoadMUS(tempPath);
 	#else
 		music = Mix_LoadMUS(filename);
