--- branches/KDE/4.5/kdebase/apps/plasma/applets/folderview/folderview.cpp	2010/10/12 20:59:24	1185295
+++ branches/KDE/4.5/kdebase/apps/plasma/applets/folderview/folderview.cpp	2010/10/12 21:01:29	1185296
@@ -360,7 +360,7 @@
     m_model->setMimeTypeFilterList(m_filterFilesMimeList);
     m_model->setFileNameFilter(m_filterFiles);
     m_model->setSortDirectoriesFirst(m_sortDirsFirst);
-    m_model->setDynamicSortFilter(true);
+    m_model->setDynamicSortFilter(m_sortColumn != -1);
     m_model->sort(m_sortColumn != -1 ? m_sortColumn : KDirModel::Name, Qt::AscendingOrder);
 
     DirLister *lister = new DirLister(this);
@@ -712,8 +712,10 @@
         if (m_sortColumn != -1) {
             m_model->invalidate();
             m_model->sort(m_sortColumn, Qt::AscendingOrder);
+            m_model->setDynamicSortFilter(true);
         } else if (m_iconView) {
             m_iconView->setCustomLayout(true);
+            m_model->setDynamicSortFilter(false);
         }
         updateSortActionsState();
         cg.writeEntry("sortColumn", m_sortColumn);
@@ -1583,6 +1585,7 @@
     if (column != m_sortColumn) {
         m_model->invalidate();
         m_model->sort(column, Qt::AscendingOrder);
+        m_model->setDynamicSortFilter(true);
         m_sortColumn = column;
         config().writeEntry("sortColumn", m_sortColumn);
         emit configNeedsSaving();
@@ -1683,6 +1686,7 @@
     // If the user has rearranged the icons, the view is no longer sorted
     if (m_sortColumn != -1) {
         m_sortColumn = -1;
+        m_model->setDynamicSortFilter(false);
         updateSortActionsState();
         config().writeEntry("sortColumn", m_sortColumn);
         emit configNeedsSaving();
