diff -p -up apps/dolphin/src/dolphinfileitemdelegate.cpp.orig apps/dolphin/src/dolphinfileitemdelegate.cpp
--- apps/dolphin/src/dolphinfileitemdelegate.cpp.orig	2010-04-01 11:19:18.000000000 -0300
+++ apps/dolphin/src/dolphinfileitemdelegate.cpp	2010-04-01 13:38:31.000000000 -0300
@@ -18,6 +18,8 @@
  ***************************************************************************/
 
 #include "dolphinfileitemdelegate.h"
+#include "settings/dolphinsettings.h"
+#include "dolphin_iconsmodesettings.h"
 
 #include <dolphinmodel.h>
 #include <kfileitem.h>
@@ -37,6 +39,11 @@ DolphinFileItemDelegate::DolphinFileItem
     m_cachedSize(),
     m_cachedEmblems()
 {
+    const IconsModeSettings* settings = DolphinSettings::instance().iconsModeSettings();
+    Q_ASSERT(settings != 0);
+
+    if (settings->firstLineTextEliding())
+	    setWrapMode(QTextOption::WordWrap);
 }
 
 DolphinFileItemDelegate::~DolphinFileItemDelegate()
diff -p -up apps/dolphin/src/dolphiniconsview.cpp.orig apps/dolphin/src/dolphiniconsview.cpp
diff -p -up apps/dolphin/src/dolphiniconsview.h.orig apps/dolphin/src/dolphiniconsview.h
diff -p -up apps/dolphin/src/settings/dolphin_iconsmodesettings.kcfg.orig apps/dolphin/src/settings/dolphin_iconsmodesettings.kcfg
--- apps/dolphin/src/settings/dolphin_iconsmodesettings.kcfg.orig	2010-03-30 11:07:43.000000000 -0300
+++ apps/dolphin/src/settings/dolphin_iconsmodesettings.kcfg	2010-04-01 10:37:48.000000000 -0300
@@ -63,5 +63,9 @@
             <label>Preview size</label>
             <default code="true">KIconLoader::SizeHuge</default>
         </entry>
+        <entry name="FirstLineTextEliding" type="Bool">
+            <label>First line eliding</label>
+            <default>false</default>
+        </entry>
     </group>
 </kcfg>
diff -p -up apps/dolphin/src/settings/iconsviewsettingspage.cpp.orig apps/dolphin/src/settings/iconsviewsettingspage.cpp
--- apps/dolphin/src/settings/iconsviewsettingspage.cpp.orig	2010-03-29 14:30:34.000000000 -0300
+++ apps/dolphin/src/settings/iconsviewsettingspage.cpp	2010-04-01 10:37:45.000000000 -0300
@@ -33,7 +33,6 @@
 #include <kcombobox.h>
 #include <KNumInput>
 
-#include <QCheckBox>
 #include <QGroupBox>
 #include <QLabel>
 #include <QListView>
@@ -48,7 +47,8 @@ IconsViewSettingsPage::IconsViewSettings
     m_fontRequester(0),
     m_textlinesCountBox(0),
     m_arrangementBox(0),
-    m_gridSpacingBox(0)
+    m_gridSpacingBox(0),
+    m_firstLineTextEliding(0)
 {
     const int spacing = KDialog::spacingHint();
     const int margin = KDialog::marginHint();
@@ -94,6 +94,9 @@ IconsViewSettingsPage::IconsViewSettings
     m_textWidthBox->addItem(i18nc("@item:inlistbox Text width", "Huge"));
     connect(m_textWidthBox, SIGNAL(currentIndexChanged(int)), this, SIGNAL(changed()));
 
+    m_firstLineTextEliding = new QCheckBox(i18nc("@option:check", "First line eliding"));
+    connect(m_firstLineTextEliding, SIGNAL(toggled(bool)), this, SIGNAL(changed()));
+
     QGridLayout* textGroupLayout = new QGridLayout(textGroup);
     textGroupLayout->addWidget(fontLabel, 0, 0, Qt::AlignRight);
     textGroupLayout->addWidget(m_fontRequester, 0, 1);
@@ -101,6 +104,7 @@ IconsViewSettingsPage::IconsViewSettings
     textGroupLayout->addWidget(m_textlinesCountBox, 1, 1);
     textGroupLayout->addWidget(textWidthLabel, 2, 0, Qt::AlignRight);
     textGroupLayout->addWidget(m_textWidthBox, 2, 1);
+    textGroupLayout->addWidget(m_firstLineTextEliding, 3, 1);
 
     // create the 'Grid' group for selecting the arrangement and the grid spacing
     QGroupBox* gridGroup = new QGroupBox(i18nc("@title:group", "Grid"), this);
@@ -187,6 +191,8 @@ void IconsViewSettingsPage::applySetting
         settings->setGridSpacing(GridSpacingBase + (index - 1) * GridSpacingInc);
     }
 
+    settings->setFirstLineTextEliding(m_firstLineTextEliding->isChecked());
+
     settings->writeConfig();
 }
 
@@ -241,6 +247,8 @@ void IconsViewSettingsPage::loadSettings
     const int spacing = settings->gridSpacing();
     const int index = (spacing <= 0) ? 0 : 1 + (spacing - GridSpacingBase) / GridSpacingInc;
     m_gridSpacingBox->setCurrentIndex(index);
+
+    m_firstLineTextEliding->setChecked(settings->firstLineTextEliding());
 }
 
 #include "iconsviewsettingspage.moc"
diff -p -up apps/dolphin/src/settings/iconsviewsettingspage.h.orig apps/dolphin/src/settings/iconsviewsettingspage.h
--- apps/dolphin/src/settings/iconsviewsettingspage.h.orig	2010-03-29 14:39:17.000000000 -0300
+++ apps/dolphin/src/settings/iconsviewsettingspage.h	2010-03-31 11:43:01.000000000 -0300
@@ -23,6 +23,8 @@
 #include <dolphiniconsview.h>
 #include <settings/viewsettingspagebase.h>
 
+#include <QCheckBox>
+
 class DolphinFontRequester;
 class IconSizeGroupBox;
 class KComboBox;
@@ -77,6 +79,7 @@ private:
 
     IconSizeGroupBox* m_iconSizeGroupBox;
     KComboBox* m_textWidthBox;
+    QCheckBox* m_firstLineTextEliding;
     DolphinFontRequester* m_fontRequester;
     KIntSpinBox* m_textlinesCountBox;
 
