Index: apps/konqueror/src/konqsessiondlg.cpp
===================================================================
--- apps/konqueror/src/konqsessiondlg.cpp
+++ apps/konqueror/src/konqsessiondlg.cpp	2010-05-09 09:44:42.000000000 +0200
@@ -1,4 +1,4 @@
-/*  This file is part of the KDE project
+/*  This file is part of the KDE project
     Copyright (C) 2008 Eduardo Robles Elvira <edulix@gmail.com>
 
     This program is free software; you can redistribute it and/or modify
@@ -64,7 +64,6 @@
 };
 
 #define BTN_OPEN KDialog::User1
-#define BTN_CLOSE KDialog::User2
 
 KonqSessionDlg::KonqSessionDlg( KonqViewManager *manager, QWidget *parent )
     : KDialog( parent )
@@ -77,7 +76,7 @@
     setModal( true );
     setCaption( i18n( "Manage Sessions" ) );
     setButtons( BTN_OPEN | Close );
-    setDefaultButton( BTN_CLOSE );
+    setDefaultButton( Close );
     showButtonSeparator( true );
     
     setButtonGuiItem( BTN_OPEN, KGuiItem( i18n( "&Open" ), "document-open" ) );
@@ -202,7 +201,6 @@
 }
 
 #undef BTN_OPEN
-#undef BTN_CLOSE
 
 class KonqNewSessionDlg::KonqNewSessionDlgPrivate : public QWidget,
     public Ui::KonqNewSessionDlgBase
