Index: apps/lib/konq/konq_operations.cpp
===================================================================
--- apps/lib/konq/konq_operations.cpp	(révision 1107032)
+++ apps/lib/konq/konq_operations.cpp	(copie de travail)
@@ -113,7 +113,8 @@
 void KonqOperations::emptyTrash( QWidget* parent )
 {
     KonqOperations *op = new KonqOperations( parent );
-    op->_del( EMPTYTRASH, KUrl("trash:/"), SKIP_CONFIRMATION );
+    ConfirmationType confirmation = DEFAULT_CONFIRMATION;
+    op->_del( EMPTYTRASH, KUrl("trash:/"), confirmation );
 }
 
 void KonqOperations::restoreTrashedItems( const KUrl::List& urls, QWidget* parent )
@@ -187,7 +188,7 @@
         delete this; // this one is ok, _del is always called directly
         return;
     }
-
+    
     if ( confirmation == SKIP_CONFIRMATION || askDeleteConfirmation( selectedUrls, method, confirmation, parentWidget() ) )
     {
         //m_srcUrls = selectedUrls;
@@ -239,7 +240,21 @@
 
 bool KonqOperations::askDeleteConfirmation( const KUrl::List & selectedUrls, int method, ConfirmationType confirmation, QWidget* widget )
 {
-    KIO::JobUiDelegate::DeletionType deletionType = method == DEL ? KIO::JobUiDelegate::Delete : KIO::JobUiDelegate::Trash;
+    KIO::JobUiDelegate::DeletionType deletionType ;
+    
+    switch (method)
+    {
+	    case EMPTYTRASH:
+		    deletionType = KIO::JobUiDelegate::EmptyTrash ;
+		    break;
+	    case DEL:
+		    deletionType = KIO::JobUiDelegate::Delete;
+		    break;
+	    default:
+		    deletionType = KIO::JobUiDelegate::Trash;
+		    break;
+    }
+
     KIO::JobUiDelegate::ConfirmationType confirmationType = confirmation == FORCE_CONFIRMATION ? KIO::JobUiDelegate::ForceConfirmation : KIO::JobUiDelegate::DefaultConfirmation;
     KIO::JobUiDelegate uiDelegate;
     uiDelegate.setWindow(widget);
