diff -ur kdebase-4.4.3/apps/konsole/src/ViewManager.cpp kdebase-4.4.3-konsole-activetabindex/apps/konsole/src/ViewManager.cpp
--- kdebase-4.4.3/apps/konsole/src/ViewManager.cpp	2010-04-29 21:54:14.000000000 +0200
+++ kdebase-4.4.3-konsole-activetabindex/apps/konsole/src/ViewManager.cpp	2010-05-01 20:02:13.000000000 +0200
@@ -936,7 +936,7 @@
 void ViewManager::restoreSessions(const KConfigGroup& group)
 {
     QList<int> ids = group.readEntry("Sessions", QList<int>());
-    int activeTab  = group.readEntry("Active", 0);
+    int activeTab  = group.readEntry("Active", group.readEntry("ActiveTabIndex", 0));
     TerminalDisplay *display = 0;
 
     int tab = 1;
