Index: kdebase/apps/lib/konq/konq_popupmenu.cpp
===================================================================
--- kdebase/apps/lib/konq/konq_popupmenu.cpp	(révision 1107032)
+++ kdebase/apps/lib/konq/konq_popupmenu.cpp	(copie de travail)
@@ -48,6 +48,7 @@
 #include <kacceleratormanager.h>
 
 #include <QFileInfo>
+#include <QtCore/QProcess>
 
 /*
  Test cases:
@@ -97,6 +98,7 @@
     void slotPopupNewDir();
     void slotPopupNewView();
     void slotPopupEmptyTrashBin();
+    void slotConfigTrashBin();
     void slotPopupRestoreTrashedItems();
     void slotPopupAddToBookmark();
     void slotPopupMimeType();
@@ -340,7 +342,17 @@
         QObject::connect(act, SIGNAL(triggered()), q, SLOT(slotPopupEmptyTrashBin()));
         q->addAction(act);
     }
+    if ( isCurrentTrash )
+    {
+	act = new KAction(m_parentWidget);
+	m_ownActions.append(act);
+	act->setIcon( KIcon("trash-empty") );
+	act->setText( i18n( "&Configure Trash Bin" ) );
+	QObject::connect(act, SIGNAL(triggered()), q, SLOT(slotConfigTrashBin()));
+	q->addAction(act);
+    }
 
+
     // This is used by KHTML, see khtml_popupmenu.rc (copy, selectAll, searchProvider etc.)
     // and by DolphinPart (rename, trash, delete)
     addGroup( "editactions" );
@@ -494,6 +506,12 @@
   KonqOperations::emptyTrash(m_parentWidget);
 }
 
+void KonqPopupMenuPrivate::slotConfigTrashBin()
+{
+  QProcess *proc = new QProcess();
+  proc->start("kcmshell4 kcmtrash");
+}
+
 void KonqPopupMenuPrivate::slotPopupRestoreTrashedItems()
 {
   KonqOperations::restoreTrashedItems(m_popupItemProperties.urlList(), m_parentWidget);
Index: kdebase/apps/lib/konq/konq_popupmenu.h
===================================================================
--- kdebase/apps/lib/konq/konq_popupmenu.h	(révision 1107032)
+++ kdebase/apps/lib/konq/konq_popupmenu.h	(copie de travail)
@@ -106,6 +106,7 @@
   Q_PRIVATE_SLOT(d, void slotPopupNewDir())
   Q_PRIVATE_SLOT(d, void slotPopupNewView())
   Q_PRIVATE_SLOT(d, void slotPopupEmptyTrashBin())
+  Q_PRIVATE_SLOT(d, void slotConfigTrashBin())
   Q_PRIVATE_SLOT(d, void slotPopupRestoreTrashedItems())
   Q_PRIVATE_SLOT(d, void slotPopupAddToBookmark())
   Q_PRIVATE_SLOT(d, void slotPopupMimeType())
