Index: apps/konsole/src/CharacterColor.h
===================================================================
--- apps/konsole/src/CharacterColor.h	(révision 1104288)
+++ apps/konsole/src/CharacterColor.h	(copie de travail)
@@ -27,6 +27,7 @@
 #include <QtGui/QColor>
 
 #include <kdemacros.h>
+#include <KDebug>
 
 namespace Konsole
 {
@@ -271,6 +272,8 @@
 {
   switch (_colorSpace)
   {
+	  kDebug() << "DEBUG: _colorSpace" << _colorSpace;
+
     case COLOR_SPACE_DEFAULT: return base[_u+0+(_v?BASE_COLORS:0)].color;
     case COLOR_SPACE_SYSTEM: return base[_u+2+(_v?BASE_COLORS:0)].color;
     case COLOR_SPACE_256: return color256(_u,base);
