<https://rt.cpan.org/Public/Bug/Display.html?id=11659>

--- a/lib/Parse/Yapp.pm     2005-02-25 13:24:39.000000000 +0300
+++ b/lib/Parse/Yapp.pm     2005-02-25 13:25:13.000000000 +0300
@@ -375,7 +375,7 @@

 So you will need an Error reporting sub.

-item C<Error reporting routine>
+=item C<Error reporting routine>

 If you want one, write it knowing that it is passed as parameter
 the parser object. So you can share information whith the lexer
