From: Ansgar Burchardt <ansgar@43-1.org>
Subject: Correct spelling errors

<https://rt.cpan.org/Public/Bug/Display.html?id=54410>

--- libparse-yapp-perl.orig/lib/Parse/Yapp.pm
+++ libparse-yapp-perl/lib/Parse/Yapp.pm
@@ -173,7 +173,7 @@
 
 A rhs may be followed by an optional C<%prec> directive, followed
 by a token, giving the rule an explicit precedence (see yacc manuals
-for its precise meaning) and optionnal semantic action code block (see
+for its precise meaning) and optional semantic action code block (see
 below).
 
     exp:   '-' exp %prec NEG { -$_[1] }
@@ -351,7 +351,7 @@
 It is its duty to return the next token and value to the parser.
 They C<must> be returned as a list of two variables, the first one
 is the token known by the parser (symbolic or literal), the second
-one beeing anything you want (usualy the content of the token, or the
+one beeing anything you want (usually the content of the token, or the
 literal value) from a simple scalar value to any complex reference,
 as the parsing driver never use it but to call semantic actions:
 
