--- cmake_unofficial/CMakeLists.txt.orig	2015-05-15 20:04:36.326860531 -0300
+++ cmake_unofficial/CMakeLists.txt	2015-05-15 20:06:13.667343026 -0300
@@ -42,19 +42,20 @@ if(BUILD_LIBS)
     
     set_target_properties(liblz4 PROPERTIES
     OUTPUT_NAME lz4
-    SOVERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
+    VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
+    SOVERSION "${CPACK_PACKAGE_VERSION_MAJOR}"
     )
         
     install(TARGETS liblz4
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     )
 
     install(FILES
     ${LZ4_DIR}/lz4.h
     ${LZ4_DIR}/lz4hc.h
     ${LZ4_DIR}/lz4frame.h
-    DESTINATION include
+    DESTINATION include/lz4
     )
     
     if(BUILD_TOOLS)
