--- cmake_unofficial/CMakeLists.txt.orig	2014-01-07 22:06:24.859006354 -0200
+++ cmake_unofficial/CMakeLists.txt	2014-01-07 22:07:28.163653819 -0200
@@ -46,18 +46,19 @@ if(BUILD_LIBS)
     
     set_target_properties(liblz4 PROPERTIES
     OUTPUT_NAME lz4
-    SOVERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
+    VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
+    SOVERSION "${CPACK_PACKAGE_VERSION_MAJOR}"
     )
         
     install(TARGETS liblz4
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     )
 
     install(FILES
     ${LZ4_DIR}/lz4.h
     ${LZ4_DIR}/lz4hc.h
-    DESTINATION include
+    DESTINATION include/lz4
     )
     
     if(BUILD_TOOLS)
