--- cmake/CMakeLists.txt.orig	2014-01-01 19:42:28.464083160 -0200
+++ cmake/CMakeLists.txt	2014-01-01 19:44:42.305669811 -0200
@@ -45,18 +45,19 @@ if(BUILD_LIBS)
     
     set_target_properties(liblz4 PROPERTIES
     OUTPUT_NAME lz4
-    SOVERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
+    VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}"
+    SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR}
     )
         
     install(TARGETS liblz4
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
     )
 
     install(FILES
     ${SRC_DIR}/lz4.h
     ${SRC_DIR}/lz4hc.h
-    DESTINATION include
+    DESTINATION include/lz4
     )
     
     if(BUILD_TOOLS)
