diff -up wrk/src/mcompand.c.wrk wrk/src/mcompand.c
--- wrk/src/mcompand.c.wrk	2013-02-15 10:15:03.426639189 +0100
+++ wrk/src/mcompand.c	2013-02-15 10:18:42.408008144 +0100
@@ -395,7 +395,7 @@ static int flow(sox_effect_t * effp, con
     (void)sox_mcompand_flow_1(effp, c,l,bbuf,abuf,len, (size_t)effp->out_signal.channels);
     for (i=0;i<len;++i)
     {
-      out = obuf[i] + abuf[i];
+      out = (double)obuf[i] + (double)abuf[i];
       SOX_SAMPLE_CLIP_COUNT(out, effp->clips);
       obuf[i] = out;
     }
