diff -dur xfce4-clipman-plugin-1.1.3.orig/daemon/gsd-clipboard-manager.c xfce4-clipman-plugin-1.1.3/daemon/gsd-clipboard-manager.c
--- xfce4-clipman-plugin-1.1.3.orig/daemon/gsd-clipboard-manager.c	2009-11-15 21:24:52.000000000 +0100
+++ xfce4-clipman-plugin-1.1.3/daemon/gsd-clipboard-manager.c	2010-01-07 01:02:10.000000000 +0100
@@ -121,7 +121,7 @@
         GtkSelectionData *selection_data_cache = NULL;
 
         list = manager->priv->default_cache;
-        for (; list->next != NULL; list = list->next) {
+        for (; list != NULL && list->next != NULL; list = list->next) {
                 selection_data_cache = list->data;
                 if (selection_data->target == selection_data_cache->target) {
                         break;
@@ -154,8 +154,11 @@
         GtkSelectionData *sdata;
         GSList           *list;
 
-        target_list = gtk_target_list_new (NULL, 0);
         list = manager->priv->default_cache;
+        if (list == NULL) {
+                return;
+        }
+        target_list = gtk_target_list_new (NULL, 0);
         for (; list->next != NULL; list = list->next) {
                 sdata = list->data;
                 gtk_target_list_add (target_list, sdata->target, 0, 0);
Nur in xfce4-clipman-plugin-1.1.3/daemon: gsd-clipboard-manager.c.orig.
