diff -up gnome-netstatus-2.28.1/configure.in.libs gnome-netstatus-2.28.1/configure.in
--- gnome-netstatus-2.28.1/configure.in.libs	2010-03-30 21:14:59.730092643 -0400
+++ gnome-netstatus-2.28.1/configure.in	2010-03-30 21:15:58.486091876 -0400
@@ -49,6 +49,8 @@ PKG_CHECK_MODULES(NETSTATUS,
 
 AC_CHECK_HEADERS(sys/sockio.h)
 
+NETSTATUS_LIBS="$NETSTATUS_LIBS -lm"
+
 # Blatantly stolen from configure.in in the sample code from
 # "UNIX Network Programming, Volume 1" by W. Richard Stevens
 AC_CACHE_CHECK(if sockaddr has sa_len member,
