--- lib/pluginregistry.c.libdir	2010-11-14 12:17:44.134179617 -0200
+++ lib/pluginregistry.c	2010-11-14 12:21:39.897087715 -0200
@@ -854,9 +854,9 @@
   
   env = getenv("LADSPA_PATH");
   if(env)
-    path = bg_sprintf("%s:/usr/lib64/ladspa:/usr/local/lib64/ladspa:/usr/lib/ladspa:/usr/local/lib/ladspa", env);
+    path = bg_sprintf("%s:_LADSPA_LIBDIR_/ladspa", env);
   else
-    path = bg_sprintf("/usr/lib64/ladspa:/usr/local/lib64/ladspa:/usr/lib/ladspa:/usr/local/lib/ladspa");
+    path = bg_sprintf("_LADSPA_LIBDIR_/ladspa");
 
   tmp_info = scan_multi(path, &file_info, section, BG_PLUGIN_API_LADSPA, opt);
   if(tmp_info)
@@ -865,7 +865,7 @@
   free(path);
   
   /* Frei0r */
-  tmp_info = scan_multi("/usr/lib64/frei0r-1:/usr/local/lib64/frei0r-1:/usr/lib/frei0r-1:/usr/local/lib/frei0r-1", &file_info, 
+  tmp_info = scan_multi("_FREI0R_LIBDIR_/frei0r-1", &file_info, 
                         section, BG_PLUGIN_API_FREI0R, opt);
   if(tmp_info)
     ret->entries = append_to_list(ret->entries, tmp_info);
