--- fusion-icon-5e2dc9/FusionIcon/data.py	2008-03-16 22:58:07.000000000 +0000
+++ fusion-icon-5e2dc9/FusionIcon/data.py.new	2011-01-19 01:14:49.907802877 +0000
@@ -33,7 +33,7 @@
 	'/usr/share/nvidia-glx/diversions/libGL.so.1.2',
 )
 
-compiz_args = ['--replace', '--sm-disable', '--ignore-desktop-hints', 'ccp']
+compiz_args = ['--replace', '--sm-disable', 'ccp']
 
 config_home = os.environ.get('XDG_CONFIG_HOME',
 			os.path.join(os.environ['HOME'], '.config'))
