--- kactivities-4.10.2/src/service/Application.cpp.orig	2013-04-18 18:11:25.988256844 -0300
+++ kactivities-4.10.2/src/service/Application.cpp	2013-04-18 18:11:47.320775915 -0300
@@ -236,8 +236,8 @@ int main(int argc, char ** argv)
 
     KCmdLineArgs::init(argc, argv, &about);
 
-    if (!KCmdLineArgs::allArguments().contains("--nofork"))
-        initSignalCatching();
+//    if (!KCmdLineArgs::allArguments().contains("--nofork"))
+//        initSignalCatching();
 
     return Application::self()->exec();
 }
