From 64849958c830b6e3f8025230b37aa0cc1481cf6e Mon Sep 17 00:00:00 2001
From: Maarten Lankhorst <m.b.lankhorst@gmail.com>
Date: Sun, 8 May 2011 10:47:25 +0200
Subject: [PATCH 03/31] winemenubuilder: silence an err

---
 programs/winemenubuilder/winemenubuilder.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/programs/winemenubuilder/winemenubuilder.c b/programs/winemenubuilder/winemenubuilder.c
index 3746bd0..17f3e99 100644
--- a/programs/winemenubuilder/winemenubuilder.c
+++ b/programs/winemenubuilder/winemenubuilder.c
@@ -2489,7 +2489,7 @@ static BOOL write_freedesktop_association_entry(const char *desktopPath, const c
         fclose(desktop);
     }
     else
-        WINE_ERR("error writing association file %s\n", wine_dbgstr_a(desktopPath));
+        WINE_WARN("error writing association file %s\n", wine_dbgstr_a(desktopPath));
     return ret;
 }
 
-- 
1.7.9.5

