--- wine-1.7.6/dlls/appwiz.cpl/addons.c.orig	2013-11-08 20:01:17.309837677 -0200
+++ wine-1.7.6/dlls/appwiz.cpl/addons.c	2013-11-08 20:02:04.093017452 -0200
@@ -65,8 +65,8 @@ WINE_DEFAULT_DEBUG_CHANNEL(appwizcpl);
 #define GECKO_SHA "???"
 #endif
 
-#define MONO_VERSION "0.0.8"
-#define MONO_SHA "dd349e72249ce5ff981be0e9dae33ac4a46a9f60"
+#define MONO_VERSION "4.5.0"
+#define MONO_SHA "4c01b2a64104e5fa4b73740355aa57c4cbeb710e"
 
 typedef struct {
     const char *version;
