--- configure.orig	2010-05-25 19:44:56.000000000 +0200
+++ configure	2010-05-25 19:46:11.000000000 +0200
@@ -5132,10 +5132,10 @@
     then
       convert_version_major=`expr "$convert_version" : '.* \([0-9]*\)\.[0-9]*'`
       convert_version_minor=`expr "$convert_version" : '.* [0-9]*\.\([0-9]*\)'`
-      if test "$convert_version_major" -eq 6 -a "$convert_version_minor" -lt 6
+      if test "$convert_version_major" -eq 6 -a "$convert_version_minor" -lt 5
       then
         CONVERT=false
-        as_fn_append wine_warnings "|imagemagick version 6.6 or newer is needed to rebuild icons."
+        as_fn_append wine_warnings "|imagemagick version 6.5 or newer is needed to rebuild icons."
       fi
     fi
     if test "$CONVERT" = false
--- configure.ac.orig	2010-05-25 19:45:16.000000000 +0200
+++ configure.ac	2010-05-25 19:46:44.000000000 +0200
@@ -303,10 +303,10 @@
     then
       convert_version_major=`expr "$convert_version" : '.* \([[0-9]]*\)\.[[0-9]]*'`
       convert_version_minor=`expr "$convert_version" : '.* [[0-9]]*\.\([[0-9]]*\)'`
-      if test "$convert_version_major" -eq 6 -a "$convert_version_minor" -lt 6
+      if test "$convert_version_major" -eq 6 -a "$convert_version_minor" -lt 5
       then
         CONVERT=false
-        WINE_WARNING([imagemagick version 6.6 or newer is needed to rebuild icons.])
+        WINE_WARNING([imagemagick version 6.5 or newer is needed to rebuild icons.])
       fi
     fi
     if test "$CONVERT" = false
