diff --git a/dlls/gdi32/freetype.c b/dlls/gdi32/freetype.c
index 424fb3d..b25aa2e 100644
--- a/dlls/gdi32/freetype.c
+++ b/dlls/gdi32/freetype.c
@@ -6605,6 +6605,21 @@ static DWORD get_glyph_outline(GdiFont *
 	unsigned int max_level, row, col;
 	BYTE *start, *ptr;
 
+if(!strcmp(ft_face->family_name, "csd"))
+{
+    int i;
+    if(lpgm->gmBlackBoxX)
+        lpgm->gmBlackBoxX--;
+    for(i = 0; i < 2; i++)
+    {
+        if(lpgm->gmBlackBoxY)
+        {
+            lpgm->gmBlackBoxY--;
+            lpgm->gmptGlyphOrigin.y--;
+        }
+    }
+}
+
 	pitch = (width + 3) / 4 * 4;
 	needed = pitch * height;
 
