diff -up wine-1.1.41/dlls/wined3d/directx.c.intz wine-1.1.41/dlls/wined3d/directx.c
--- wine-1.1.41/dlls/wined3d/directx.c.intz	2010-03-19 16:02:16.000000000 -0300
+++ wine-1.1.41/dlls/wined3d/directx.c	2010-03-19 19:33:49.741164678 -0300
@@ -3417,6 +3417,7 @@ static BOOL CheckTextureCapability(struc
         case WINED3DFMT_S8_UINT_D24_FLOAT:
         case WINED3DFMT_D32_UNORM:
         case WINED3DFMT_D32_FLOAT:
+	 case WINED3DFMT_INTZ:
             return TRUE;
 
         /*****
diff -up wine-1.1.41/dlls/wined3d/utils.c.intz wine-1.1.41/dlls/wined3d/utils.c
--- wine-1.1.41/dlls/wined3d/utils.c.intz	2010-03-19 16:02:16.000000000 -0300
+++ wine-1.1.41/dlls/wined3d/utils.c	2010-03-19 19:33:49.742164059 -0300
@@ -127,6 +127,7 @@ static const struct StaticPixelFormatDes
     {WINED3DFMT_R16_UINT,                   0x0,        0x0,        0x0,        0x0,        2,      0,      0},
     {WINED3DFMT_R32_UINT,                   0x0,        0x0,        0x0,        0x0,        4,      0,      0},
     {WINED3DFMT_R16G16B16A16_SNORM,         0x0,        0x0,        0x0,        0x0,        8,      0,      0},
+    {WINED3DFMT_INTZ,                       0x0,        0x0,        0x0,        0x0,        4,      24,     8},
     /* Vendor-specific formats */
     {WINED3DFMT_ATI2N,                      0x0,        0x0,        0x0,        0x0,        1,      0,      0},
     {WINED3DFMT_NVHU,                       0x0,        0x0,        0x0,        0x0,        2,      0,      0},
@@ -518,6 +519,10 @@ static const struct wined3d_format_textu
             GL_DEPTH_STENCIL,           GL_FLOAT_32_UNSIGNED_INT_24_8_REV,
             WINED3DFMT_FLAG_DEPTH | WINED3DFMT_FLAG_STENCIL,
             ARB_DEPTH_BUFFER_FLOAT},
+    {WINED3DFMT_INTZ,                   GL_DEPTH24_STENCIL8_EXT,         GL_DEPTH24_STENCIL8_EXT,               0,
+            GL_DEPTH_STENCIL,         GL_UNSIGNED_INT_24_8,
+            WINED3DFMT_FLAG_POSTPIXELSHADER_BLENDING | WINED3DFMT_FLAG_FILTERING | WINED3DFMT_FLAG_DEPTH | WINED3DFMT_FLAG_STENCIL,
+            ARB_DEPTH_TEXTURE},
     /* Vendor-specific formats */
     {WINED3DFMT_ATI2N,                  GL_COMPRESSED_LUMINANCE_ALPHA_3DC_ATI, GL_COMPRESSED_LUMINANCE_ALPHA_3DC_ATI, 0,
             GL_LUMINANCE_ALPHA,         GL_UNSIGNED_BYTE,
@@ -1334,6 +1339,7 @@ const char* debug_d3dformat(WINED3DFORMA
     FMT_TO_STR(WINED3DFMT_B5G5R5A1_UNORM);
     FMT_TO_STR(WINED3DFMT_B8G8R8A8_UNORM);
     FMT_TO_STR(WINED3DFMT_B8G8R8X8_UNORM);
+    FMT_TO_STR(WINED3DFMT_INTZ);
 #undef FMT_TO_STR
   default:
     {
@@ -2179,6 +2185,7 @@ BOOL getDepthStencilBits(const struct wi
         case WINED3DFMT_S8_UINT_D24_FLOAT:
         case WINED3DFMT_D32_UNORM:
         case WINED3DFMT_D32_FLOAT:
+	case WINED3DFMT_INTZ:
             break;
         default:
             FIXME("Unsupported stencil format: %s\n", debug_d3dformat(format_desc->format));
diff -up wine-1.1.41/include/wine/wined3d.idl.intz wine-1.1.41/include/wine/wined3d.idl
--- wine-1.1.41/include/wine/wined3d.idl.intz	2010-03-19 16:02:16.000000000 -0300
+++ wine-1.1.41/include/wine/wined3d.idl	2010-03-19 19:33:49.742164059 -0300
@@ -269,6 +269,7 @@ typedef enum _WINED3DFORMAT
     WINED3DFMT_MULTI2_ARGB8                 = WINEMAKEFOURCC('M','E','T','1'),
     WINED3DFMT_G8R8_G8B8                    = WINEMAKEFOURCC('G','R','G','B'),
     WINED3DFMT_R8G8_B8G8                    = WINEMAKEFOURCC('R','G','B','G'),
+    WINED3DFMT_INTZ                         = WINEMAKEFOURCC('I','N','T','Z'),
     WINED3DFMT_ATI2N                        = WINEMAKEFOURCC('A','T','I','2'),
     WINED3DFMT_INST                         = WINEMAKEFOURCC('I','N','S','T'),
     WINED3DFMT_NVHU                         = WINEMAKEFOURCC('N','V','H','U'),
