diff --git a/dlls/gdi32/freetype.c b/dlls/gdi32/freetype.c
index 424fb3d..b25aa2e 100644
--- a/dlls/gdi32/freetype.c
+++ b/dlls/gdi32/freetype.c
@@ -3580,6 +3580,21 @@ DWORD WineEngGetGlyphOutline(GdiFont *incoming_font, UINT glyph, UINT format,
        unsigned int mult, row, col;
        BYTE *start, *ptr;
 
+if(!strcmp(ft_face->family_name, "csd"))
+{
+    int i;
+    if(lpgm->gmBlackBoxX)
+        lpgm->gmBlackBoxX--;
+    for(i = 0; i < 2; i++)
+    {
+        if(lpgm->gmBlackBoxY)
+        {
+            lpgm->gmBlackBoxY--;
+            lpgm->gmptGlyphOrigin.y--;
+        }
+    }
+}
+
         width = lpgm->gmBlackBoxX;
        height = lpgm->gmBlackBoxY;
        pitch = (width + 3) / 4 * 4;

