--- wine-1.7.38/dlls/ntdll/sec.c.orig	2015-03-06 07:27:36.000000000 -0600
+++ wine-1.7.38/dlls/ntdll/sec.c	2015-03-19 10:42:37.117665983 -0500
@@ -1714,7 +1714,7 @@
  * the native function returns something like
  * "S-1-5-21-0000000000-000000000-0000000000-500";
  */
-NTSTATUS WINAPI RtlConvertSidToUnicodeString(
+NTSTATUS WINAPI __attribute__((optimize("O0"))) RtlConvertSidToUnicodeString(
        PUNICODE_STRING String,
        PSID pSid,
        BOOLEAN AllocateString)
--- wine-1.7.38/dlls/ntdll/directory.c.orig	2015-03-19 13:53:47.952588902 -0500
+++ wine-1.7.38/dlls/ntdll/directory.c	2015-03-19 13:52:51.571954998 -0500
@@ -3008,7 +3008,7 @@
  * element doesn't have to exist; in that case STATUS_NO_SUCH_FILE is
  * returned, but the unix name is still filled in properly.
  */
-NTSTATUS CDECL wine_nt_to_unix_file_name( const UNICODE_STRING *nameW, ANSI_STRING *unix_name_ret,
+NTSTATUS CDECL __attribute__((optimize("O0"))) wine_nt_to_unix_file_name( const UNICODE_STRING *nameW, ANSI_STRING *unix_name_ret,
                                           UINT disposition, BOOLEAN check_case )
 {
     static const WCHAR unixW[] = {'u','n','i','x'};
--- wine-1.7.38/dlls/kernel32/volume.c.orig	2015-03-19 15:01:23.647111174 -0500
+++ wine-1.7.38/dlls/kernel32/volume.c	2015-03-19 15:00:59.460854627 -0500
@@ -714,7 +714,7 @@
 /***********************************************************************
  *           GetVolumeInformationW   (KERNEL32.@)
  */
-BOOL WINAPI GetVolumeInformationW( LPCWSTR root, LPWSTR label, DWORD label_len,
+BOOL WINAPI __attribute__((optimize("O0"))) GetVolumeInformationW( LPCWSTR root, LPWSTR label, DWORD label_len,
                                    DWORD *serial, DWORD *filename_len, DWORD *flags,
                                    LPWSTR fsname, DWORD fsname_len )
 {
--- wine-1.7.39/dlls/ntdll/virtual.c.orig	2015-03-23 09:22:18.512141978 -0500
+++ wine-1.7.39/dlls/ntdll/virtual.c	2015-03-23 09:22:47.951415202 -0500
@@ -2346,7 +2346,7 @@
  *             NtQueryVirtualMemory   (NTDLL.@)
  *             ZwQueryVirtualMemory   (NTDLL.@)
  */
-NTSTATUS WINAPI NtQueryVirtualMemory( HANDLE process, LPCVOID addr,
+NTSTATUS WINAPI __attribute__((optimize("O0"))) NtQueryVirtualMemory( HANDLE process, LPCVOID addr,
                                       MEMORY_INFORMATION_CLASS info_class, PVOID buffer,
                                       SIZE_T len, SIZE_T *res_len )
 {
--- wine-1.7.39/dlls/kernel32/process.c.orig	2015-03-23 11:35:21.490412754 -0500
+++ wine-1.7.39/dlls/kernel32/process.c	2015-03-23 12:12:36.928665968 -0500
@@ -1088,7 +1088,7 @@
  *
  * Startup routine of a new process. Runs on the new process stack.
  */
-static DWORD WINAPI start_process( PEB *peb )
+static DWORD WINAPI __attribute__((optimize("O0"))) start_process( PEB *peb )
 {
     IMAGE_NT_HEADERS *nt;
     LPTHREAD_START_ROUTINE entry;
--- wine-1.7.39/dlls/winex11.drv/window.c.orig	2015-03-23 12:14:36.789785691 -0500
+++ wine-1.7.39/dlls/winex11.drv/window.c	2015-03-23 12:35:56.656863213 -0500
@@ -1270,7 +1270,7 @@
  *
  * Synchronize the X client window position with the Windows one
  */
-static void sync_client_position( struct x11drv_win_data *data,
+static void __attribute__((optimize("O0"))) sync_client_position( struct x11drv_win_data *data,
                                   const RECT *old_client_rect, const RECT *old_whole_rect )
 {
     int mask = 0;
--- wine-1.7.39/dlls/gdi32/freetype.c.orig	2015-03-20 08:41:44.000000000 -0500
+++ wine-1.7.39/dlls/gdi32/freetype.c	2015-03-23 14:15:25.884759565 -0500
@@ -6389,7 +6389,7 @@
 
 static const BYTE masks[8] = {0x80, 0x40, 0x20, 0x10, 0x08, 0x04, 0x02, 0x01};
 
-static DWORD get_glyph_outline(GdiFont *incoming_font, UINT glyph, UINT format,
+static DWORD __attribute__((optimize("O0"))) get_glyph_outline(GdiFont *incoming_font, UINT glyph, UINT format,
                                LPGLYPHMETRICS lpgm, ABC *abc, DWORD buflen, LPVOID buf,
                                const MAT2* lpmat)
 {
--- wine-1.7.39/dlls/rpcrt4/rpc_binding.c.orig	2015-03-20 08:41:44.000000000 -0500
+++ wine-1.7.39/dlls/rpcrt4/rpc_binding.c	2015-03-23 17:00:45.215424859 -0500
@@ -984,7 +984,7 @@
 /***********************************************************************
  *             RpcBindingCopy (RPCRT4.@)
  */
-RPC_STATUS RPC_ENTRY RpcBindingCopy(
+RPC_STATUS RPC_ENTRY __attribute__((optimize("O0"))) RpcBindingCopy(
   RPC_BINDING_HANDLE SourceBinding,
   RPC_BINDING_HANDLE* DestinationBinding)
 {
--- wine-1.7.39/dlls/gdi32/font.c.orig	2015-03-23 20:33:48.751656009 -0500
+++ wine-1.7.39/dlls/gdi32/font.c	2015-03-23 20:33:52.295693767 -0500
@@ -1297,7 +1297,7 @@
  * Return the size of the string as it would be if it was output properly by
  * e.g. TextOut.
  */
-BOOL WINAPI GetTextExtentExPointW( HDC hdc, LPCWSTR str, INT count, INT max_ext,
+BOOL WINAPI __attribute__((optimize("O0"))) GetTextExtentExPointW( HDC hdc, LPCWSTR str, INT count, INT max_ext,
                                    LPINT nfit, LPINT dxs, LPSIZE size )
 {
     DC *dc;
--- wine-1.7.39/dlls/gdi32/mapping.c.orig	2015-03-23 20:21:39.025790911 -0500
+++ wine-1.7.39/dlls/gdi32/mapping.c	2015-03-23 20:27:31.380589949 -0500
@@ -324,7 +324,7 @@
 /***********************************************************************
  *           DPtoLP    (GDI32.@)
  */
-BOOL WINAPI DPtoLP( HDC hdc, LPPOINT points, INT count )
+BOOL WINAPI __attribute__((optimize("O0"))) DPtoLP( HDC hdc, LPPOINT points, INT count )
 {
     DC * dc = get_dc_ptr( hdc );
     if (!dc) return FALSE;
@@ -352,7 +352,7 @@
 /***********************************************************************
  *           LPtoDP    (GDI32.@)
  */
-BOOL WINAPI LPtoDP( HDC hdc, LPPOINT points, INT count )
+BOOL WINAPI __attribute__((optimize("O0"))) LPtoDP( HDC hdc, LPPOINT points, INT count )
 {
     DC * dc = get_dc_ptr( hdc );
     if (!dc) return FALSE;
--- wine-1.7.39/dlls/user32/win.c.orig	2015-03-23 21:42:20.861715642 -0500
+++ wine-1.7.39/dlls/user32/win.c	2015-03-23 21:45:49.223517788 -0500
@@ -2955,7 +2955,7 @@
 /*****************************************************************
  *		SetParent (USER32.@)
  */
-HWND WINAPI SetParent( HWND hwnd, HWND parent )
+HWND WINAPI __attribute__((optimize("O0"))) SetParent( HWND hwnd, HWND parent )
 {
     HWND full_handle;
     HWND old_parent = 0;
--- wine-1.7.39/programs/winecfg/appdefaults.c.orig	2015-03-23 21:15:50.402096990 -0500
+++ wine-1.7.39/programs/winecfg/appdefaults.c	2015-03-23 21:16:22.345452810 -0500
@@ -78,7 +78,7 @@
 static const char szKeyNT[] = "Software\\Microsoft\\Windows NT\\CurrentVersion";
 static const char szKeyProdNT[] = "System\\CurrentControlSet\\Control\\ProductOptions";
 
-static int get_registry_version(void)
+static int __attribute__((optimize("O0"))) get_registry_version(void)
 {
     int i, best = -1, platform, major, minor = 0, build = 0;
     char *p, *ver, *type = NULL;
--- wine-1.7.39/programs/winemenubuilder/winemenubuilder.c.orig	2015-03-23 21:18:24.651815189 -0500
+++ wine-1.7.39/programs/winemenubuilder/winemenubuilder.c	2015-03-23 21:24:20.565779739 -0500
@@ -2323,7 +2323,7 @@
     return ret;
 }
 
-static void update_association(LPCWSTR extension, LPCSTR mimeType, LPCWSTR progId,
+static void __attribute__((optimize("O0"))) update_association(LPCWSTR extension, LPCSTR mimeType, LPCWSTR progId,
     LPCSTR appName, LPCSTR desktopFile, LPCSTR openWithIcon)
 {
     static const WCHAR ProgIDW[] = {'P','r','o','g','I','D',0};
@@ -2400,7 +2400,7 @@
     HeapFree(GetProcessHeap(), 0, openWithIconW);
 }
 
-static BOOL cleanup_associations(void)
+static BOOL __attribute__((optimize("O0"))) cleanup_associations(void)
 {
     static const WCHAR openW[] = {'o','p','e','n',0};
     static const WCHAR DesktopFileW[] = {'D','e','s','k','t','o','p','F','i','l','e',0};
