diff -up kdebindings-4.6.4/python/pykde4/tools/pykdeuic4/pykdeuic4.py.python-bz#684419 kdebindings-4.6.4/python/pykde4/tools/pykdeuic4/pykdeuic4.py
--- kdebindings-4.6.4/python/pykde4/tools/pykdeuic4/pykdeuic4.py.python-bz#684419	2011-06-02 05:03:32.000000000 -0500
+++ kdebindings-4.6.4/python/pykde4/tools/pykdeuic4/pykdeuic4.py	2011-06-16 15:22:04.791151529 -0500
@@ -45,7 +45,7 @@ class kde_i18n_string(qtproxies.i18n_str
     def __init__(self,string, disambig=None):
         original_i18n_string.__init__(self,string, disambig)
     def __str__(self):
-        return "kdecore.i18n(%s)" % (qtproxies.escape(self.string),)
+        return "kdecore.i18n(%s)" % (qtproxies.as_string(self.string, encode=False),)
 
 qtproxies.i18n_string = kde_i18n_string
 
@@ -78,7 +78,7 @@ def processUI(uifile, output_filename=No
     indenter.indentwidth = indent
     comp = compiler.UICompiler()
     comp.factory._cwFilters.append(kdeFilter())
-    winfo = comp.compileUi(uifile, output)
+    winfo = comp.compileUi(uifile, output, False)
 
     if exe:
         output.write("""
