--- ffmpeg.c.orig	2015-11-03 23:25:13.890492686 -0200
+++ ffmpeg.c	2015-11-03 23:29:59.663056603 -0200
@@ -39,7 +39,6 @@
 #include <libavformat/avio.h>
 #include <libswresample/swresample.h>
 #include <libavutil/opt.h>
-#include <libavutil/audioconvert.h>
 #ifndef AVUTIL_MATHEMATICS_H
 #include <libavutil/mathematics.h>
 #endif
@@ -235,7 +234,7 @@ static int ffmpeg_open(struct input_plug
 
 		codec = avcodec_find_decoder(cc->codec_id);
 		if (!codec) {
-			d_print("codec not found: %d, %s\n", cc->codec_id, cc->codec_name);
+			d_print("codec not found: %d, %s\n", cc->codec_id, cc->codec_type);
 			err = -IP_ERROR_UNSUPPORTED_FILE_TYPE;
 			break;
 		}
@@ -248,7 +247,7 @@ static int ffmpeg_open(struct input_plug
 #else
 		if (avcodec_open2(cc, codec, NULL) < 0) {
 #endif
-			d_print("could not open codec: %d, %s\n", cc->codec_id, cc->codec_name);
+			d_print("could not open codec: %d, %s\n", cc->codec_id, cc->codec_type);
 			err = -IP_ERROR_UNSUPPORTED_FILE_TYPE;
 			break;
 		}
@@ -347,7 +346,7 @@ static int ffmpeg_fill_buffer(AVFormatCo
 			      struct ffmpeg_output *output, SwrContext *swr)
 {
 #if (LIBAVCODEC_VERSION_INT >= ((53<<16) + (25<<8) + 0))
-	AVFrame *frame = avcodec_alloc_frame();
+	AVFrame *frame = av_frame_alloc();
 	int got_frame;
 #endif
 	while (1) {
@@ -363,7 +362,7 @@ static int ffmpeg_fill_buffer(AVFormatCo
 			if (av_read_frame(ic, &input->pkt) < 0) {
 				/* Force EOF once we can read no longer. */
 #if (LIBAVCODEC_VERSION_INT >= ((53<<16) + (25<<8) + 0))
-				avcodec_free_frame(&frame);
+				av_frame_free(&frame);
 #endif
 				return 0;
 			}
@@ -426,7 +425,7 @@ static int ffmpeg_fill_buffer(AVFormatCo
 				res = 0;
 			output->buffer_pos = output->buffer;
 			output->buffer_used_len = res * cc->channels * sizeof(int16_t);
-			avcodec_free_frame(&frame);
+			av_frame_free(&frame);
 			return output->buffer_used_len;
 		}
 #endif
@@ -566,7 +565,7 @@ static long ffmpeg_current_bitrate(struc
 	long bitrate = -1;
 #if (LIBAVFORMAT_VERSION_INT > ((51<<16)+(43<<8)+0))
 	/* ape codec returns silly numbers */
-	if (priv->codec->id == CODEC_ID_APE)
+	if (priv->codec->id == AV_CODEC_ID_APE)
 		return -1;
 #endif
 	if (priv->input->curr_duration > 0) {
