diff -up iris-1.0.0/src/irisnet/corelib/corelib.pri.jdns_install iris-1.0.0/src/irisnet/corelib/corelib.pri
--- iris-1.0.0/src/irisnet/corelib/corelib.pri.jdns_install	2010-03-05 19:48:58.000000000 -0600
+++ iris-1.0.0/src/irisnet/corelib/corelib.pri	2011-11-08 13:59:36.546525284 -0600
@@ -3,9 +3,16 @@ QT *= network
 # libidn
 #LIBS += -lidn
 
-include(../../jdns/jdns.pri)
 INCLUDEPATH += $$PWD/../../jdns
 
+iris_bundle:{
+	include(../../jdns/jdns.pri)
+}
+else {
+        QMAKE_LFLAGS += -L$$IRIS_BASE/lib
+        LIBS += -lqjdns
+}
+
 HEADERS += \
 	$$PWD/jdnsshared.h \
 	$$PWD/objectsession.h \
diff -up iris-1.0.0/src/jdns/jdns.pro.jdns_install iris-1.0.0/src/jdns/jdns.pro
--- iris-1.0.0/src/jdns/jdns.pro.jdns_install	2007-09-17 19:50:18.000000000 -0500
+++ iris-1.0.0/src/jdns/jdns.pro	2011-11-08 14:01:50.093855711 -0600
@@ -1,9 +1,39 @@
+
+include(jdns.pri)
+
+iris_bundle:{
+
 CONFIG += console
 CONFIG -= app_bundle
 QT -= gui
 QT += network
 
-include(jdns.pri)
-
 SOURCES += \
 	main.cpp
+}
+else {
+
+IRIS_BASE = ../..
+
+TEMPLATE = lib
+TARGET   = qjdns
+DESTDIR  = $$IRIS_BASE/lib
+CONFIG  += dll create_prl
+
+headers.path = $$[QT_INSTALL_HEADERS]/jdns
+headers.files = *.h
+
+INSTALLS += headers
+
+target.path = $$[QT_INSTALL_LIBS]
+
+CONFIG     += create_pc no_install_prl
+QMAKE_PKGCONFIG_NAME = qjdns
+QMAKE_PKGCONFIG_DESCRIPTION = Qt bindings for JDNS
+QMAKE_PKGCONFIG_LIBDIR = $$[QT_INSTALL_LIBS]
+QMAKE_PKGCONFIG_INCDIR = $$[QT_INSTALL_HEADERS]/jdns
+QMAKE_PKGCONFIG_CFLAGS = -I$$[QT_INSTALL_HEADERS]
+QMAKE_PKGCONFIG_DESTDIR = pkgconfig
+
+INSTALLS += target
+}
diff -up iris-1.0.0/src/src.pro.jdns_install iris-1.0.0/src/src.pro
--- iris-1.0.0/src/src.pro.jdns_install	2009-07-03 13:45:50.000000000 -0500
+++ iris-1.0.0/src/src.pro	2011-11-08 13:59:36.547525271 -0600
@@ -5,6 +5,7 @@ include(libbase.pri)
 sub_irisnet.subdir = irisnet
 sub_xmpp.subdir = xmpp
 sub_xmpp.depends = sub_irisnet
+sub_jdns.subdir = jdns
 
 SUBDIRS += sub_irisnet
-!iris_bundle:SUBDIRS += sub_xmpp
+!iris_bundle:SUBDIRS += sub_xmpp sub_jdns
