diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/bsocket.cpp iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/bsocket.cpp
--- iris-1.0.0/src/irisnet/noncore/cutestuff/bsocket.cpp	2010-03-06 02:49:47.000000000 +0100
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/bsocket.cpp	2013-02-13 23:08:51.000000000 +0100
@@ -208,6 +208,11 @@
 	d->srv.resolve(srv, type, "tcp");
 }
 
+QAbstractSocket* BSocket::abstractSocket() const
+{
+	return d->qsock;
+}
+
 int BSocket::socket() const
 {
 	if(d->qsock)
diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/bsocket.h iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/bsocket.h
--- iris-1.0.0/src/irisnet/noncore/cutestuff/bsocket.h	2010-03-06 02:49:47.000000000 +0100
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/bsocket.h	2013-02-13 23:08:11.000000000 +0100
@@ -43,6 +43,7 @@
 	void connectToHost(const QString &host, quint16 port);
 	void connectToHost(const QHostAddress &addr, quint16 port);
 	void connectToServer(const QString &srv, const QString &type);
+	virtual QAbstractSocket* abstractSocket() const;
 	int socket() const;
 	void setSocket(int);
 	int state() const;
diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/bytestream.h iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/bytestream.h
--- iris-1.0.0/src/irisnet/noncore/cutestuff/bytestream.h	2008-06-02 20:37:53.000000000 +0200
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/bytestream.h	2013-02-13 23:08:28.000000000 +0100
@@ -24,6 +24,7 @@
 #include <QObject>
 #include <QByteArray>
 
+class QAbstractSocket;
 // CS_NAMESPACE_BEGIN
 
 // CS_EXPORT_BEGIN
@@ -45,6 +46,8 @@
 	static void appendArray(QByteArray *a, const QByteArray &b);
 	static QByteArray takeArray(QByteArray *from, int size=0, bool del=true);
 
+	virtual QAbstractSocket* abstractSocket() const { return 0; }
+
 signals:
 	void connectionClosed();
 	void delayedCloseFinished();
diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/httppoll.cpp iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/httppoll.cpp
--- iris-1.0.0/src/irisnet/noncore/cutestuff/httppoll.cpp	2010-01-27 00:25:55.000000000 +0100
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/httppoll.cpp	2013-02-13 23:08:37.000000000 +0100
@@ -109,6 +109,11 @@
 	delete d;
 }
 
+QAbstractSocket* HttpPoll::abstractSocket() const
+{
+	return d->http.abstractSocket();
+}
+
 void HttpPoll::reset(bool clear)
 {
 	if(d->http.isActive())
@@ -465,6 +470,11 @@
 	delete d;
 }
 
+QAbstractSocket* HttpProxyPost::abstractSocket() const
+{
+	return d->sock.abstractSocket();
+}
+
 void HttpProxyPost::reset(bool clear)
 {
 	if(d->sock.state() != BSocket::Idle)
diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/httppoll.h iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/httppoll.h
--- iris-1.0.0/src/irisnet/noncore/cutestuff/httppoll.h	2007-09-20 20:39:14.000000000 +0200
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/httppoll.h	2013-02-13 23:08:59.000000000 +0100
@@ -33,6 +33,8 @@
 	HttpPoll(QObject *parent=0);
 	~HttpPoll();
 
+	virtual QAbstractSocket* abstractSocket() const;
+
 	void setAuth(const QString &user, const QString &pass="");
 	void connectToUrl(const QString &url);
 	void connectToHost(const QString &proxyHost, int proxyPort, const QString &url);
@@ -75,6 +77,8 @@
 	HttpProxyPost(QObject *parent=0);
 	~HttpProxyPost();
 
+	QAbstractSocket* abstractSocket() const;
+
 	void setAuth(const QString &user, const QString &pass="");
 	bool isActive() const;
 	void post(const QString &proxyHost, int proxyPort, const QString &url, const QByteArray &data, bool asProxy=true);
diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/socks.cpp iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/socks.cpp
--- iris-1.0.0/src/irisnet/noncore/cutestuff/socks.cpp	2010-01-27 00:25:55.000000000 +0100
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/socks.cpp	2013-02-13 23:08:19.000000000 +0100
@@ -477,6 +477,11 @@
 	delete d;
 }
 
+QAbstractSocket* SocksClient::abstractSocket() const
+{
+	return d->sock.abstractSocket();
+}
+
 void SocksClient::reset(bool clear)
 {
 	if(d->sock.state() != BSocket::Idle)
diff -Nur iris-1.0.0/src/irisnet/noncore/cutestuff/socks.h iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/socks.h
--- iris-1.0.0/src/irisnet/noncore/cutestuff/socks.h	2008-06-02 21:09:45.000000000 +0200
+++ iris-1.0.0-027_add_socket_access_function/src/irisnet/noncore/cutestuff/socks.h	2013-02-13 23:08:43.000000000 +0100
@@ -63,6 +63,8 @@
 	SocksClient(int, QObject *parent=0);
 	~SocksClient();
 
+	virtual QAbstractSocket* abstractSocket() const;
+
 	bool isIncoming() const;
 
 	// outgoing
