diff -up iris-1.0.0/src/xmpp/jid/jid.cpp.system_libidn iris-1.0.0/src/xmpp/jid/jid.cpp
--- iris-1.0.0/src/xmpp/jid/jid.cpp.system_libidn	2011-03-02 03:37:32.000000000 -0600
+++ iris-1.0.0/src/xmpp/jid/jid.cpp	2013-02-11 07:50:45.855157286 -0600
@@ -24,7 +24,7 @@
 #include <QCoreApplication>
 #include <QByteArray>
 #include <QHash>
-#include <libidn/stringprep.h>
+#include <stringprep.h>
 
 #ifndef NO_IRISNET
 #include "irisnetglobal_p.h"
diff -up iris-1.0.0/src/xmpp/jid/jid.pri.system_libidn iris-1.0.0/src/xmpp/jid/jid.pri
--- iris-1.0.0/src/xmpp/jid/jid.pri.system_libidn	2008-09-26 04:46:03.000000000 -0500
+++ iris-1.0.0/src/xmpp/jid/jid.pri	2013-02-11 07:52:25.379913056 -0600
@@ -7,7 +7,4 @@ HEADERS += \
 SOURCES += \
 	$$PWD/jid.cpp
 
-# Normally, we shouldn't include other modules, but since Jid is the only
-# user of libidn, and many modules include this module, we include this
-# module here
-include($$PWD/../../libidn/libidn.pri)
+LIBS += -lidn
diff -up iris-1.0.0/src/xmpp/xmpp-core/connector.cpp.system_libidn iris-1.0.0/src/xmpp/xmpp-core/connector.cpp
--- iris-1.0.0/src/xmpp/xmpp-core/connector.cpp.system_libidn	2010-03-05 19:49:47.000000000 -0600
+++ iris-1.0.0/src/xmpp/xmpp-core/connector.cpp	2013-02-11 07:50:45.855157286 -0600
@@ -36,7 +36,7 @@
 #include <QList>
 #include <QUrl>
 #include <QTimer>
-#include <libidn/idna.h>
+#include <idna.h>
 #include "safedelete.h"
 #include "bsocket.h"
 #include "httpconnect.h"
