diff -Nur iris-1.0.0/src/xmpp/xmpp-im/filetransfer.cpp iris-1.0.0-009_filetransferpreview/src/xmpp/xmpp-im/filetransfer.cpp
--- iris-1.0.0/src/xmpp/xmpp-im/filetransfer.cpp	2011-02-24 19:05:06.000000000 +0100
+++ iris-1.0.0-009_filetransferpreview/src/xmpp/xmpp-im/filetransfer.cpp	2013-02-13 19:54:58.000000000 +0100
@@ -58,6 +58,7 @@
 	qlonglong size;
 	qlonglong sent;
 	QString desc;
+	QString preview;
 	bool rangeSupported;
 	qlonglong rangeOffset, rangeLength, length;
 	QString streamType;
@@ -129,19 +130,20 @@
 	d->proxy = proxy;
 }
 
-void FileTransfer::sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc)
+void FileTransfer::sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc, const QString& preview)
 {
 	d->state = Requesting;
 	d->peer = to;
 	d->fname = fname;
 	d->size = size;
 	d->desc = desc;
+	d->preview = preview;
 	d->sender = true;
 	d->id = d->m->link(this);
 
 	d->ft = new JT_FT(d->m->client()->rootTask());
 	connect(d->ft, SIGNAL(finished()), SLOT(ft_finished()));
-	d->ft->request(to, d->id, fname, size, desc, d->m->streamPriority() );
+	d->ft->request(to, d->id, fname, size, desc, d->m->streamPriority(), preview);
 	d->ft->go(true);
 }
 
@@ -194,6 +196,12 @@
 	return d->desc;
 }
 
+
+QString XMPP::FileTransfer::preview() const
+{
+	return d->preview;
+}
+
 bool FileTransfer::rangeSupported() const
 {
 	return d->rangeSupported;
@@ -347,6 +355,7 @@
 	d->fname = req.fname;
 	d->size = req.size;
 	d->desc = req.desc;
+	d->preview = req.preview;
 	d->rangeSupported = req.rangeSupported;
 	d->streamType = streamType;
 }
@@ -548,7 +557,7 @@
 }
 
 void JT_FT::request(const Jid &to, const QString &_id, const QString &fname,
-					qlonglong size, const QString &desc, const QStringList &streamTypes)
+					qlonglong size, const QString &desc, const QStringList &streamTypes, const QString& preview)
 {
 	QDomElement iq;
 	d->to = to;
@@ -567,6 +576,12 @@
 		de.appendChild(doc()->createTextNode(desc));
 		file.appendChild(de);
 	}
+	if(!preview.isEmpty()) {
+		QDomElement pr = doc()->createElement("preview");
+		pr.setAttribute("xmlns", "http://kopete.kde.org/protocol/file-preview");
+		pr.appendChild(doc()->createTextNode(preview));
+		file.appendChild(pr);
+	}
 	QDomElement range = doc()->createElement("range");
 	file.appendChild(range);
 	si.appendChild(file);
@@ -800,6 +815,11 @@
 	QDomElement de = file.elementsByTagName("desc").item(0).toElement();
 	if(!de.isNull())
 		desc = de.text();
+	
+	QString preview;
+	QDomElement pr = file.elementsByTagName("preview").item(0).toElement();
+	if(!pr.isNull())
+		preview= pr.text();
 
 	bool rangeSupported = false;
 	QDomElement range = file.elementsByTagName("range").item(0).toElement();
@@ -831,6 +851,7 @@
 	r.fname = fname;
 	r.size = size;
 	r.desc = desc;
+	r.preview = preview;
 	r.rangeSupported = rangeSupported;
 	r.streamTypes = streamTypes;
 
diff -Nur iris-1.0.0/src/xmpp/xmpp-im/filetransfer.h iris-1.0.0-009_filetransferpreview/src/xmpp/xmpp-im/filetransfer.h
--- iris-1.0.0/src/xmpp/xmpp-im/filetransfer.h	2011-02-24 19:05:06.000000000 +0100
+++ iris-1.0.0-009_filetransferpreview/src/xmpp/xmpp-im/filetransfer.h	2013-02-13 19:48:35.000000000 +0100
@@ -55,7 +55,7 @@
 		void setProxy(const Jid &proxy);
 
 		// send
-		void sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc);
+		void sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc, const QString& preview=QString());
 		qlonglong offset() const;
 		qlonglong length() const;
 		int dataSizeNeeded() const;
@@ -66,6 +66,7 @@
 		QString fileName() const;
 		qlonglong fileSize() const;
 		QString description() const;
+		QString preview() const;
 		bool rangeSupported() const;
 		void accept(qlonglong offset=0, qlonglong length=0);
 
@@ -144,7 +145,7 @@
 		JT_FT(Task *parent);
 		~JT_FT();
 
-		void request(const Jid &to, const QString &id, const QString &fname, qlonglong size, const QString &desc, const QStringList &streamTypes);
+		void request(const Jid &to, const QString &id, const QString &fname, qlonglong size, const QString &desc, const QStringList &streamTypes, const QString &preview=QString());
 		qlonglong rangeOffset() const;
 		qlonglong rangeLength() const;
 		QString streamType() const;
@@ -164,6 +165,7 @@
 		QString fname;
 		qlonglong size;
 		QString desc;
+		QString preview;
 		bool rangeSupported;
 		QStringList streamTypes;
 	};
