diff -Nur iris-1.0.0/src/xmpp/jid/jid.cpp iris-1.0.0-003_case_insensitive_jid/src/xmpp/jid/jid.cpp
--- iris-1.0.0/src/xmpp/jid/jid.cpp	2011-03-02 10:37:32.000000000 +0100
+++ iris-1.0.0-003_case_insensitive_jid/src/xmpp/jid/jid.cpp	2013-02-13 19:48:34.000000000 +0100
@@ -274,6 +274,9 @@
 		b = d;
 	else
 		b = n + '@' + d;
+	
+	b=b.toLower(); // JID are not case sensitive
+	
 	if(r.isEmpty())
 		f = b;
 	else
