diff -Nur iris-1.0.0/src/xmpp/xmpp-core/parser.cpp iris-1.0.0-014_fix_semicolons/src/xmpp/xmpp-core/parser.cpp
--- iris-1.0.0/src/xmpp/xmpp-core/parser.cpp	2008-08-19 01:03:07.000000000 +0200
+++ iris-1.0.0-014_fix_semicolons/src/xmpp/xmpp-core/parser.cpp	2013-02-13 19:48:35.000000000 +0100
@@ -547,7 +547,7 @@
 		QList<Parser::Event*> eventList;
 		bool needMore;
 	};
-};
+}
 
 
 //----------------------------------------------------------------------------
diff -Nur iris-1.0.0/src/xmpp/xmpp-im/xmpp_xmlcommon.cpp iris-1.0.0-014_fix_semicolons/src/xmpp/xmpp-im/xmpp_xmlcommon.cpp
--- iris-1.0.0/src/xmpp/xmpp-im/xmpp_xmlcommon.cpp	2008-06-02 18:58:17.000000000 +0200
+++ iris-1.0.0-014_fix_semicolons/src/xmpp/xmpp-im/xmpp_xmlcommon.cpp	2013-02-13 19:48:35.000000000 +0100
@@ -561,5 +561,5 @@
 	}
 }
 
-};
+}
 
