diff -up iris-1.0.0/iris.pro.install_jingle iris-1.0.0/iris.pro
--- iris-1.0.0/iris.pro.install_jingle	2013-02-14 07:00:02.426980206 -0600
+++ iris-1.0.0/iris.pro	2013-02-14 07:14:10.891372941 -0600
@@ -34,6 +34,10 @@ src/irisnet/corelib/irisnetexport.h \
 src/irisnet/corelib/irisnetglobal.h \
 src/irisnet/corelib/irisnetplugin.h \
 src/irisnet/corelib/jdnsshared.h \
+src/xmpp/jingle/jinglesession.h \
+src/xmpp/jingle/jinglesessionmanager.h \
+src/xmpp/jingle/jinglecontent.h \
+src/xmpp/jingle/jingletasks.h \
 src/irisnet/noncore/legacy/ndns.h \
 src/irisnet/corelib/netavailability.h \
 src/irisnet/corelib/netinterface.h \
