diff -up libjingle-0.5.1/talk/xmllite/xmlbuilder.cc.system-expat libjingle-0.5.1/talk/xmllite/xmlbuilder.cc
--- libjingle-0.5.1/talk/xmllite/xmlbuilder.cc.system-expat	2010-11-19 11:05:16.419967001 -0500
+++ libjingle-0.5.1/talk/xmllite/xmlbuilder.cc	2010-11-19 11:05:28.161967001 -0500
@@ -33,7 +33,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "lib/expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 namespace buzz {
diff -up libjingle-0.5.1/talk/xmllite/xmlbuilder.h.system-expat libjingle-0.5.1/talk/xmllite/xmlbuilder.h
--- libjingle-0.5.1/talk/xmllite/xmlbuilder.h.system-expat	2010-11-19 11:04:49.214967000 -0500
+++ libjingle-0.5.1/talk/xmllite/xmlbuilder.h	2010-11-19 11:05:05.753967002 -0500
@@ -36,7 +36,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "lib/expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 namespace buzz {
diff -up libjingle-0.5.1/talk/xmllite/xmlparser.cc.system-expat libjingle-0.5.1/talk/xmllite/xmlparser.cc
--- libjingle-0.5.1/talk/xmllite/xmlparser.cc.system-expat	2010-11-19 11:05:55.471967002 -0500
+++ libjingle-0.5.1/talk/xmllite/xmlparser.cc	2010-11-19 11:06:07.578967000 -0500
@@ -38,7 +38,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "lib/expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 namespace buzz {
diff -up libjingle-0.5.1/talk/xmllite/xmlparser.h.system-expat libjingle-0.5.1/talk/xmllite/xmlparser.h
--- libjingle-0.5.1/talk/xmllite/xmlparser.h.system-expat	2010-11-19 11:05:38.778967001 -0500
+++ libjingle-0.5.1/talk/xmllite/xmlparser.h	2010-11-19 11:05:50.330967001 -0500
@@ -34,7 +34,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "lib/expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif  // EXPAT_RELATIVE_PATH
 
 struct XML_ParserStruct;
diff -up libjingle-0.5.1/talk/xmpp/xmppstanzaparser.cc.system-expat libjingle-0.5.1/talk/xmpp/xmppstanzaparser.cc
--- libjingle-0.5.1/talk/xmpp/xmppstanzaparser.cc.system-expat	2010-11-19 11:06:25.677966995 -0500
+++ libjingle-0.5.1/talk/xmpp/xmppstanzaparser.cc	2010-11-19 11:06:37.048967002 -0500
@@ -32,7 +32,7 @@
 #ifdef EXPAT_RELATIVE_PATH
 #include "lib/expat.h"
 #else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
+#include <expat.h>
 #endif
 
 namespace buzz {
