diff -up libjingle-0.5.1/talk/base/physicalsocketserver.cc.statfix libjingle-0.5.1/talk/base/physicalsocketserver.cc
--- libjingle-0.5.1/talk/base/physicalsocketserver.cc.statfix	2010-11-18 16:25:20.475761998 -0500
+++ libjingle-0.5.1/talk/base/physicalsocketserver.cc	2010-11-18 16:26:37.294762015 -0500
@@ -34,6 +34,7 @@
 #ifdef POSIX
 #include <string.h>
 #include <errno.h>
+#include <time.h>
 #include <fcntl.h>
 #include <sys/time.h>
 #include <unistd.h>
diff -up libjingle-0.5.1/talk/base/stream.cc.statfix libjingle-0.5.1/talk/base/stream.cc
--- libjingle-0.5.1/talk/base/stream.cc.statfix	2010-11-18 16:27:09.870761999 -0500
+++ libjingle-0.5.1/talk/base/stream.cc	2010-11-18 16:27:43.842762002 -0500
@@ -26,6 +26,7 @@
  */
 
 #if defined(POSIX)
+#include <time.h>
 #include <sys/file.h>
 #endif  // POSIX
 #include <sys/types.h>
