diff -up libjingle-0.5.1/talk/session/phone/v4llookup.cc.v4lfix libjingle-0.5.1/talk/session/phone/v4llookup.cc
--- libjingle-0.5.1/talk/session/phone/v4llookup.cc.v4lfix	2010-11-19 12:37:12.153967002 -0500
+++ libjingle-0.5.1/talk/session/phone/v4llookup.cc	2010-11-19 12:38:47.910967002 -0500
@@ -6,7 +6,8 @@
  * The functionality is implemented as a class with virtual methods for
  * the purpose of unit testing.
  */
-#include "talk/session/phone/v4llookup.h"
+
+#include "config.h"
 
 #include <errno.h>
 #include <fcntl.h>
@@ -18,6 +19,7 @@
 
 #include <cstring>
 
+#include "talk/session/phone/v4llookup.h"
 #include "talk/base/logging.h"
 
 namespace cricket {
