diff -up libjingle-0.5.1/talk/session/phone/devicemanager.cc.alsa libjingle-0.5.1/talk/session/phone/devicemanager.cc
--- libjingle-0.5.1/talk/session/phone/devicemanager.cc.alsa	2010-11-19 12:26:06.864967001 -0500
+++ libjingle-0.5.1/talk/session/phone/devicemanager.cc	2010-11-19 12:30:59.090967002 -0500
@@ -25,7 +25,7 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "talk/session/phone/devicemanager.h"
+#include "config.h"
 
 #if WIN32
 #include <atlbase.h>
@@ -44,9 +44,7 @@
 #include <QuickTime/QuickTime.h>
 #elif LINUX
 #include <unistd.h>
-#ifndef USE_TALK_SOUND
 #include <alsa/asoundlib.h>
-#endif
 #include "talk/base/linux.h"
 #include "talk/base/fileutils.h"
 #include "talk/base/pathutils.h"
@@ -54,6 +52,7 @@
 #include "talk/session/phone/v4llookup.h"
 #endif
 
+#include "talk/session/phone/devicemanager.h"
 #include "talk/base/logging.h"
 #include "talk/base/stringutils.h"
 #include "talk/session/phone/mediaengine.h"
