diff -up libjingle-0.6.10/talk/base/messagequeue.h.nextfix libjingle-0.6.10/talk/base/messagequeue.h
--- libjingle-0.6.10/talk/base/messagequeue.h.nextfix	2012-01-12 17:59:48.000000000 -0500
+++ libjingle-0.6.10/talk/base/messagequeue.h	2012-02-13 20:59:48.528641310 -0500
@@ -42,7 +42,7 @@
 #include "talk/base/scoped_ref_ptr.h"
 #include "talk/base/sigslot.h"
 #include "talk/base/socketserver.h"
-#include "talk/base/timeutils.h"
+#include "talk/base/time.h"
 
 namespace talk_base {
 
diff -up libjingle-0.6.10/talk/base/timeutils.h.nextfix libjingle-0.6.10/talk/base/timeutils.h
--- libjingle-0.6.10/talk/base/timeutils.h.nextfix	2012-01-12 17:59:48.000000000 -0500
+++ libjingle-0.6.10/talk/base/timeutils.h	2012-02-13 20:59:48.529641300 -0500
@@ -30,8 +30,7 @@
 
 #ifndef WIN32
 #include <time.h>
-#endif
-
+#else
 #include "talk/base/basictypes.h"
 
 namespace talk_base {
@@ -86,5 +85,6 @@ inline int32 TimeUntil(uint32 later) {
 }
 
 } // namespace talk_base
+#endif // WIN32
 
 #endif // TALK_BASE_TIMEUTILS_H_
diff -up libjingle-0.6.10/talk/p2p/base/pseudotcp.cc.nextfix libjingle-0.6.10/talk/p2p/base/pseudotcp.cc
--- libjingle-0.6.10/talk/p2p/base/pseudotcp.cc.nextfix	2012-01-12 17:59:48.000000000 -0500
+++ libjingle-0.6.10/talk/p2p/base/pseudotcp.cc	2012-02-13 20:59:48.531641280 -0500
@@ -38,7 +38,7 @@
 #include "talk/base/logging.h"
 #include "talk/base/socket.h"
 #include "talk/base/stringutils.h"
-#include "talk/base/timeutils.h"
+#include "talk/base/time.h"
 
 // The following logging is for detailed (packet-level) analysis only.
 #define _DBG_NONE     0
diff -up libjingle-0.6.10/talk/session/phone/rtpdump.cc.nextfix libjingle-0.6.10/talk/session/phone/rtpdump.cc
--- libjingle-0.6.10/talk/session/phone/rtpdump.cc.nextfix	2012-01-12 17:59:48.000000000 -0500
+++ libjingle-0.6.10/talk/session/phone/rtpdump.cc	2012-02-13 20:59:48.532641270 -0500
@@ -33,7 +33,7 @@
 
 #include "talk/base/byteorder.h"
 #include "talk/base/logging.h"
-#include "talk/base/timeutils.h"
+#include "talk/base/time.h"
 #include "talk/session/phone/rtputils.h"
 
 namespace cricket {
diff -up libjingle-0.6.10/talk/session/phone/videocommon.h.nextfix libjingle-0.6.10/talk/session/phone/videocommon.h
--- libjingle-0.6.10/talk/session/phone/videocommon.h.nextfix	2012-01-12 17:59:48.000000000 -0500
+++ libjingle-0.6.10/talk/session/phone/videocommon.h	2012-02-13 20:59:48.533641259 -0500
@@ -31,7 +31,7 @@
 #include <string>
 
 #include "talk/base/basictypes.h"
-#include "talk/base/timeutils.h"
+#include "talk/base/time.h"
 
 namespace cricket {
 
