diff -up graphviz-2.38.0/rtest/rtest.sh.origy graphviz-2.38.0/rtest/rtest.sh
--- graphviz-2.38.0/rtest/rtest.sh.origy	2014-04-13 22:40:25.000000000 +0200
+++ graphviz-2.38.0/rtest/rtest.sh	2014-04-14 15:07:49.113994881 +0200
@@ -311,7 +311,7 @@ function doTest
 
 trap 'rm -f $TMPFILE1 $TMPFILE2 $TMPINFILE errout; exit' 0 1 2 3 15
 
-Usage=rrtest [-gvn] [TESTFILE]\n
+Usage='rrtest [-gvn] [TESTFILE]\n
  -g : generate test data\n
  -v : verbose\n
  -n : print test'
