--- which-2.21/bash.c.orig	2015-03-19 15:06:11.000000000 -0300
+++ which-2.21/bash.c	2015-09-21 23:20:54.311871063 -0300
@@ -34,7 +34,6 @@
  * The following macro's are used in bash, and below:
  */
 #undef SHELL
-#undef AFS
 #undef NOGROUP
 
 /*
@@ -266,7 +265,7 @@ file_status (char const* name)
     r |= FS_READABLE;
 
   return r;
-#elif defined (AFS)
+#elif (getenv("AFS")) {
   /* We have to use access(2) to determine access because AFS does not
      support Unix file system semantics.  This may produce wrong
      answers for non-AFS files when ruid != euid.  I hate AFS. */
@@ -276,7 +275,9 @@ file_status (char const* name)
     r |= FS_READABLE;
 
   return r;
-#else /* !AFS */
+  }
+  else /* !AFS */
+  {
 
   /* Find out if the file is actually executable.  By definition, the
      only other criteria is that the file has an execute bit set that
@@ -320,7 +321,7 @@ file_status (char const* name)
     }
 
   return r;
-#endif /* !AFS */
+  } /* !AFS */
 }
 
 /* From bash-4.3 / general.c / line 604 ; Changes: Using 'strchr' instead of 'mbschr'. */
