--- src/sys_stuff.c.orig	2012-01-04 21:44:27.052373902 -0200
+++ src/sys_stuff.c	2012-01-04 21:45:12.458365550 -0200
@@ -786,7 +786,7 @@ void enter_data_dir() {
         if (!slash_pos) break;
 
         // Add "/data"
-        strncpy(slash_pos, "/data", sizeof(data_dir) - (slash_pos - data_dir));
+        strncpy(data_dir, "data", sizeof(data_dir));
 #else
         /* ### TODO ### Get the working directory of the program
          * Mac OS X: _NSGetExecutablePath() (man 3 dyld)
