--- ffmpegthumbs/ffmpegthumbnailer/moviedecoder.cpp.orig	2011-05-18 23:28:37.955299375 -0300
+++ ffmpegthumbs/ffmpegthumbnailer/moviedecoder.cpp	2011-05-18 23:28:58.638881814 -0300
@@ -122,7 +122,7 @@ QString MovieDecoder::getCodec()
 void MovieDecoder::initializeVideo()
 {
     for (unsigned int i = 0; i < m_pFormatContext->nb_streams; i++) {
-        if (m_pFormatContext->streams[i]->codec->codec_type == CODEC_TYPE_VIDEO) {
+        if (m_pFormatContext->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO) {
             m_pVideoStream = m_pFormatContext->streams[i];
             m_VideoStream = i;
             break;
