Index: src/app/videoWindow.cpp
===================================================================
--- src/app/videoWindow.cpp	(Revision 1251584)
+++ src/app/videoWindow.cpp	(Arbeitskopie)
@@ -29,10 +29,6 @@
 #include "actions.h"        //::seek() FIXME unfortunate
 #include "theStream.h"
 
-#ifdef HAVE_XINE
-#include <xine.h>
-#endif
-
 #include <QActionGroup>
 #include <QContextMenuEvent>
 #include <QLabel>
@@ -88,7 +84,6 @@
         , m_cursorTimer( new QTimer( this ) )
         , m_justLoaded( false )
         , m_adjustedSize( false)
-        , m_xineStream( 0 )
         , m_subLanguages( new QActionGroup( this ) )
         , m_audioLanguages( new QActionGroup( this ) )
         , m_logo( new QLabel( this ) )
@@ -508,22 +503,6 @@
 }
 
 void
-VideoWindow::refreshXineStream()
-{
-   if( m_media->property( "xine_stream_t" ).canConvert<void*>() )
-  //  if( m_media->property( "xine_stream_t" ).isValid() )
-    {
-        kDebug() << "value property " <<  m_media->property( "xine_stream_t" ).type();
-        m_xineStream = (xine_stream_t*) m_media->property( "xine_stream_t" ).value<void*>();
-    }
-    else
-    {
-        kDebug() << "mrrrrrr, QVariant property xine_stream_t isn't a void*.";
-        m_xineStream = 0;
-    }
-}
-
-void
 VideoWindow::stateChanged(Phonon::State currentState, Phonon::State oldstate) // slot
 {
 kDebug() << "chapters: " << m_controller->availableChapters() << " titles: " << m_controller->availableTitles();
@@ -531,14 +510,10 @@
     states << QLatin1String( "Loading" ) << QLatin1String( "Stopped" ) << QLatin1String( "Playing" ) << QLatin1String( "Buffering" ) << QLatin1String( "Paused" ) << QLatin1String( "Error" );
     kDebug() << "going from " << states.at(oldstate) << " to " << states.at(currentState);
 
-    if( currentState == Phonon::LoadingState )
-      m_xineStream = 0;
-
     if( currentState == Phonon::PlayingState  && m_media->hasVideo() )
     {
         m_logo->hide();
         m_vWidget->show();
-        refreshXineStream();
         updateChannels();
 
         if(m_adjustedSize==false)
@@ -670,15 +645,8 @@
 void
 VideoWindow::toggleDVDMenu()
 {
-#ifdef HAVE_XINE
-    if( m_xineStream )
-    {
-        xine_event_t e;
-        e.type = XINE_EVENT_INPUT_MENU1;
-        e.data = NULL;
-        e.data_length = 0;
-        xine_event_send( m_xineStream, &e );
-    }
+#if PHONON_VERSION >= PHONON_VERSION_CHECK(4, 5, 0)
+    m_controller->setCurrentMenu(MediaController::RootMenu);
 #endif
 }
 
Index: src/app/extern.h
===================================================================
--- src/app/extern.h	(Revision 1251584)
+++ src/app/extern.h	(Arbeitskopie)
@@ -24,11 +24,6 @@
 
 #include <QMenu>
 //I think this whole file is deprecated - Dave
-extern "C"
-{
-   typedef struct xine_s xine_t;
-}
-
 class QWidget;
 
 namespace Dragon
@@ -37,7 +32,6 @@
    
 
    void showVideoSettingsDialog( QWidget* );
-   void showXineConfigurationDialog( QWidget*, xine_t* );
    void insertAspectRatioMenuItems( QMenu* );
 }
 
Index: src/app/mainWindow.cpp
===================================================================
--- src/app/mainWindow.cpp	(Revision 1251584)
+++ src/app/mainWindow.cpp	(Arbeitskopie)
@@ -218,8 +218,7 @@
 
     if( !engine()->init() ) {
         KMessageBox::error( this, i18n(
-            "<qt>xine could not be successfully initialised. Dragon Player will now exit. "
-            "You can try to identify what is wrong with your xine installation using the <b>xine-check</b> command at a command-prompt.</qt>") );
+            "<qt>Phonon could not be successfully initialized. Dragon Player will now exit.</qt>") );
         QApplication::exit( 2 );
     }
 
Index: src/app/CMakeLists.txt
===================================================================
--- src/app/CMakeLists.txt	(Revision 1251584)
+++ src/app/CMakeLists.txt	(Arbeitskopie)
@@ -1,7 +1,3 @@
-if(XINE_XCB_FOUND)
-    add_definitions(-DHAVE_XINE)
-endif(XINE_XCB_FOUND)
-
 set(videoplayer_common_SRCS
     theStream.cpp
     actions.cpp
@@ -45,11 +41,6 @@
     ${KDE4_SOLID_LIBS}
     ${KDE4_PLASMA_LIBS} )
 
-if(XINE_XCB_FOUND)
-target_link_libraries(dragon
-    ${XINE_LIBRARY})
-endif(XINE_XCB_FOUND)
-
 install(TARGETS dragon ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 set(videoplayer_part_SRCS
@@ -66,9 +57,4 @@
     ${KDE4_PHONON_LIBS}
     ${KDE4_SOLID_LIBS})
 
-if(XINE_XCB_FOUND)
-target_link_libraries(dragonpart
-    ${XINE_LIBRARY})
-endif(XINE_XCB_FOUND)
-
 install(TARGETS dragonpart DESTINATION ${PLUGIN_INSTALL_DIR} )
Index: src/app/theStream.cpp
===================================================================
--- src/app/theStream.cpp	(Revision 1251584)
+++ src/app/theStream.cpp	(Arbeitskopie)
@@ -34,10 +34,6 @@
 #include <Solid/Device>
 #include <Solid/StorageVolume>
 
-#ifdef HAVE_XINE
-#include <xine.h>
-#endif
-
 #include "videoWindow.h"
 
 namespace Dragon
Index: src/app/videoWindow.h
===================================================================
--- src/app/videoWindow.h	(Revision 1251584)
+++ src/app/videoWindow.h	(Arbeitskopie)
@@ -44,8 +44,6 @@
      class AudioDataOutput;
 }
 
-typedef struct xine_stream_s xine_stream_t;
-
 #include "phonon/phononnamespace.h" //Phonon::State
 
 namespace Dragon
@@ -65,7 +63,6 @@
         QTimer* m_cursorTimer;
         bool m_justLoaded;
 		bool m_adjustedSize;
-        xine_stream_t* m_xineStream;
         QActionGroup* m_subLanguages;
         QActionGroup* m_audioLanguages;
         QWidget* m_logo;
@@ -130,8 +127,6 @@
         qint64 currentTime() const;
         int videoSetting( const QString& );
 
-        const xine_stream_t* xineStream() const { return m_xineStream; }
-
     public slots:
         void pause();
         void playPause();
@@ -159,7 +154,6 @@
         virtual void contextMenuEvent( QContextMenuEvent * event );
         virtual void mouseDoubleClickEvent( QMouseEvent* );
         virtual QSize sizeHint() const;
-        void refreshXineStream();
         Phonon::State state( Phonon::State state ) const;
         void setSubtitle( int channel );
         void setAudioChannel( int channel );
Index: src/codeine.h
===================================================================
--- src/codeine.h	(Revision 1251584)
+++ src/codeine.h	(Arbeitskopie)
@@ -33,7 +33,7 @@
         QWidget *mainWindow(); //defined in mainWindow.cpp
 }
 
-/// used by mainWindow.h and xineEngine.h
+/// used by mainWindow.h
 int main( int, char** );
 
 #define APP_VERSION "2.0"
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(Revision 1251584)
+++ CMakeLists.txt	(Arbeitskopie)
@@ -6,11 +6,6 @@
     ${CMAKE_CURRENT_SOURCE_DIR}/src
     ${CMAKE_CURRENT_SOURCE_DIR}/src/app
     )
-if(XINE_XCB_FOUND)
-include_directories(
-    ${XINE_INCLUDE_DIR}
-    )
-endif(XINE_XCB_FOUND)    
 
 add_subdirectory( src/app )
 add_subdirectory( misc )
Index: README
===================================================================
--- README	(Revision 1251584)
+++ README	(Arbeitskopie)
@@ -24,14 +24,9 @@
 REQUIREMENTS
   You will need at least:
 
-    * kdelibs 4.0
-    * The Xine backend for Phonon, located in kdebase
-    * xine-lib 1.1
-CMake is also required to build Video Player.
+    * kdelibs 4.x
+    * Phonon
+    * any Phonon backend
+CMake is also required to build Dragon Player.
 
-Note that currently Xine is required since Phonon in KDE 4.0 lacks a few
-features (subtitle and multiple audio tracks). For KDE 4.1, Video Player
-should be a pure-Phonon player, with no explicit Xine requirement.
-
-You could hack the CMake build system and use Video Player without Xine,
-it would probably work fine just without the aforementioned features.
+Note that not all Phonon backends support DVD menus.
\ No newline at end of file
