From c3fb6c90ab8b4243b752919d0e491df2a2d4f047 Mon Sep 17 00:00:00 2001
From: Chris Spiel <cspiel@freenet.de>
Date: Mon, 13 Oct 2014 16:08:50 +0200
Subject: [PATCH] Fix an off-by-one error in function exportImageAlpha.

---
 include/vigra/impexalpha.hxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/vigra/impexalpha.hxx b/include/vigra/impexalpha.hxx
index 0426d67..41299fc 100644
--- a/include/vigra/impexalpha.hxx
+++ b/include/vigra/impexalpha.hxx
@@ -789,7 +789,7 @@ namespace vigra
 
             encoder->setPixelType(pixel_type);
 
-            vigra_precondition(isBandNumberSupported(encoder->getFileType(), image_accessor.size(image_upper_left)),
+            vigra_precondition(isBandNumberSupported(encoder->getFileType(), image_accessor.size(image_upper_left) + 1U),
                                "exportImageAlpha(): file format does not support requested number of bands (color channels)");
 
             const range_t image_source_range(find_source_value_range(export_info,
