diff -Naupr epydoc-3.0.1.orig/epydoc/docwriter/dotgraph.py epydoc-3.0.1/epydoc/docwriter/dotgraph.py
--- epydoc-3.0.1.orig/epydoc/docwriter/dotgraph.py	2008-01-28 19:15:33.000000000 +0100
+++ epydoc-3.0.1/epydoc/docwriter/dotgraph.py	2008-12-22 20:19:42.037246526 +0100
@@ -140,7 +140,7 @@ class DotGraph:
         # the cmapx with a single call to dot.  Otherwise, we need to
         # run dot twice.
         if get_dot_version() > [1,8,10]:
-            cmapx = self._run_dot('-Tgif', '-o%s' % image_file, '-Tcmapx')
+            cmapx = self._run_dot('-Tpng', '-o%s' % image_file, '-Tcmapx')
             if cmapx is None: return '' # failed to render
         else:
             if not self.write(image_file):
@@ -220,7 +220,7 @@ class DotGraph:
                 if url: attribs['href'] = url
                 else: del attribs['href']
                 
-    def write(self, filename, language='gif'):
+    def write(self, filename, language='png'):
         """
         Render the graph using the output format `language`, and write
         the result to `filename`.
@@ -234,7 +234,7 @@ class DotGraph:
             result = result.decode('utf-8')
         return (result is not None)
 
-    def render(self, language='gif'):
+    def render(self, language='png'):
         """
         Use the ``dot`` command to render this graph, using the output
         format `language`.  Return the result as a string, or ``None``
diff -Naupr epydoc-3.0.1.orig/epydoc/docwriter/html.py epydoc-3.0.1/epydoc/docwriter/html.py
--- epydoc-3.0.1.orig/epydoc/docwriter/html.py	2008-01-29 13:43:04.000000000 +0100
+++ epydoc-3.0.1/epydoc/docwriter/html.py	2008-12-22 20:19:56.195371915 +0100
@@ -1627,7 +1627,7 @@ class HTMLWriter:
     def render_graph(self, graph):
         if graph is None: return ''
         graph.caption = graph.title = None
-        image_url = '%s.gif' % graph.uid
+        image_url = '%s.png' % graph.uid
         image_file = os.path.join(self._directory, image_url)
         return graph.to_html(image_file, image_url)
     
diff -Naupr epydoc-3.0.1.orig/epydoc/markup/epytext.py epydoc-3.0.1/epydoc/markup/epytext.py
--- epydoc-3.0.1.orig/epydoc/markup/epytext.py	2007-09-26 06:45:35.000000000 +0200
+++ epydoc-3.0.1/epydoc/markup/epytext.py	2008-12-22 20:20:14.630371505 +0100
@@ -1878,7 +1878,7 @@ class ParsedEpytextDocstring(ParsedDocst
                                       docindex, context)
             if not graph: return ''
             # Write the graph.
-            image_url = '%s.gif' % graph.uid
+            image_url = '%s.png' % graph.uid
             image_file = os.path.join(directory, image_url)
             return graph.to_html(image_file, image_url)
         else:
diff -Naupr epydoc-3.0.1.orig/epydoc/markup/restructuredtext.py epydoc-3.0.1/epydoc/markup/restructuredtext.py
--- epydoc-3.0.1.orig/epydoc/markup/restructuredtext.py	2008-01-28 19:15:33.000000000 +0100
+++ epydoc-3.0.1/epydoc/markup/restructuredtext.py	2008-12-22 20:20:23.679309245 +0100
@@ -663,7 +663,7 @@ class _EpydocHTMLTranslator(HTMLTranslat
         if graph is None: return
         
         # Write the graph.
-        image_url = '%s.gif' % graph.uid
+        image_url = '%s.png' % graph.uid
         image_file = os.path.join(self._directory, image_url)
         self.body.append(graph.to_html(image_file, image_url))
         raise SkipNode()
