From b293e11299566005b5d918c735bdf9c0ab5ded6f Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Thu, 1 Apr 2010 20:23:12 +0200
Subject: [PATCH] Fix restructuredtext formatting for python-docutils-0.6

---
 epydoc/markup/restructuredtext.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/epydoc/markup/restructuredtext.py b/epydoc/markup/restructuredtext.py
index 8b6ac04..4726cb5 100644
--- a/epydoc/markup/restructuredtext.py
+++ b/epydoc/markup/restructuredtext.py
@@ -304,10 +304,10 @@ class _SummaryExtractor(NodeVisitor):
         # Extract the first sentence.
         for child in node:
             if isinstance(child, docutils.nodes.Text):
-                m = self._SUMMARY_RE.match(child.data)
+                m = self._SUMMARY_RE.match(child)
                 if m:
                     summary_pieces.append(docutils.nodes.Text(m.group(1)))
-                    other = child.data[m.end():]
+                    other = child[m.end():]
                     if other and not other.isspace():
                         self.other_docs = True
                     break
-- 
1.7.0.1

