diff -up ./cmd/ossdevlinks/ossdevlinks.c.orig ./cmd/ossdevlinks/ossdevlinks.c
--- ./cmd/ossdevlinks/ossdevlinks.c.orig	2010-11-28 18:09:57.790812151 -0200
+++ ./cmd/ossdevlinks/ossdevlinks.c	2010-11-28 18:26:12.436178611 -0200
@@ -70,55 +70,15 @@ static int find_midilink (oss_midi_info
 #define PATH_MAX 1024 
 #endif
 
-#if defined(sun)
 #define DEFAULT_OSSLIBDIR "/etc/oss"
-#else
-#define DEFAULT_OSSLIBDIR "/usr/lib/oss"
-#endif
 
 static char *
 get_mapname (void)
 {
   FILE *f;
-#define OSSLIBDIRLEN PATH_MAX
-  char tmp[OSSLIBDIRLEN+11];
+  #define OSSLIBDIRLEN PATH_MAX
   static char name[OSSLIBDIRLEN];
-  struct stat st;
-
-  if ((f = fopen ("/etc/oss.conf", "r")) == NULL)
-    {
-      // perror ("/etc/oss.conf");
-      goto oexit;
-    }
-
-  while (fgets (tmp, sizeof (tmp), f) != NULL)
-    {
-      int l = strlen (tmp);
-      if (l > 0 && tmp[l - 1] == '\n')
-	tmp[l - 1] = '\0';
-
-      if (strncmp (tmp, "OSSLIBDIR=", 10) == 0)
-	{
-	  l = snprintf (name, sizeof (name), "%s", &tmp[10]);
-	  if ((l >= OSSLIBDIRLEN) || (l < 0))
-	    {
-	      fprintf (stderr, "String in /etc/oss.conf is too long!\n");
-	      goto oexit;
-	    }
-	  if ((stat (name, &st) == -1) || !S_ISDIR(st.st_mode))
-	    {
-	      fprintf (stderr, "Directory %s from /etc/oss.conf cannot "
-			       "be used!\n", name);
-	      goto oexit;
-	    }
-	  fclose (f);
-	  return name;
-	}
-    }
-
-oexit:
-  fclose (f);
-  snprintf (name, sizeof (name), DEFAULT_OSSLIBDIR);
+  snprintf (name, sizeof (name), "/etc/oss");
   return name;
 }
 
@@ -989,7 +949,7 @@ save_links (void)
       char lfile[PATH_MAX+20], * osslibdir;
 
       osslibdir = get_mapname ();
-      snprintf (lfile, sizeof (lfile), "%s/%s", osslibdir, "etc/legacy_devices");
+      snprintf (lfile, sizeof (lfile), "%s/%s", osslibdir, "legacy_devices");
       if ((f = fopen (lfile, "w")) == NULL)
 	{
 	  perror (lfile);
diff -up ./cmd/ossdevlinks/ossdevlinks.man.orig ./cmd/ossdevlinks/ossdevlinks.man
--- ./cmd/ossdevlinks/ossdevlinks.man.orig	2010-11-28 18:24:11.982861247 -0200
+++ ./cmd/ossdevlinks/ossdevlinks.man	2010-11-28 18:24:29.098927149 -0200
@@ -32,8 +32,7 @@ SEE ALSO
 ossinfo(1)
 
 FILES
-/usr/sbin/ossdevlinks
-/usr/lib/oss/etc/legacy_devices
+/etc/oss/legacy_devices
 /dev/dspN
 /dev/midiNN
 /dev/mixerN
diff -up ./cmd/savemixer/savemixer.c.orig ./cmd/savemixer/savemixer.c
--- ./cmd/savemixer/savemixer.c.orig	2010-11-28 18:16:34.494578746 -0200
+++ ./cmd/savemixer/savemixer.c	2010-11-28 18:29:03.683835687 -0200
@@ -35,7 +35,7 @@ static void reorder_dspdevs (void);
 #define ETCDIRLEN 512
 #define SLINELEN 256
 
-static char ossetcdir[ETCDIRLEN] = "/usr/lib/oss/etc";
+static char ossetcdir[ETCDIRLEN] = "/etc/oss";
 	/* This is the usual place */
 static oss_mixerinfo mixerinfo;
 static oss_mixext *mixerdefs = NULL;
@@ -268,8 +268,6 @@ load_applist (void)
 static char *
 get_mapname (void)
 {
-  FILE *f;
-  char tmp[ETCDIRLEN+11]; /* Adding 'OSSLIBDIR=' */
   static char name[ETCDIRLEN+15]; /* Adding '/etc/mixer.save' */
   struct stat st;
 
@@ -280,47 +278,8 @@ get_mapname (void)
       return name;
     }
 
-  if ((f = fopen ("/etc/oss.conf", "r")) == NULL)
-    {
-      // perror ("/etc/oss.conf");
-      goto dexit;
-    }
-
-  while (fgets (tmp, sizeof (tmp), f) != NULL)
-    {
-      size_t l = strlen (tmp);
-      if (l > 0 && tmp[l - 1] == '\n')
-	tmp[l - 1] = '\0';
-
-      if (strncmp (tmp, "OSSLIBDIR=", 10) == 0)
-	{
-	  l = snprintf (name, sizeof (name), "%s/etc/mixer.save", &tmp[10]);
-	  if ((l >= sizeof (name)) || (l < 0))
-	    {
-	      fprintf (stderr, "String in /etc/oss.conf is too long!\n");
-	      goto oexit;
-	    }
-	  snprintf (ossetcdir, sizeof (ossetcdir), "%s/etc", &tmp[10]);
-	  if ((l >= sizeof (ossetcdir)) || (l < 0))
-	    {
-	      fprintf (stderr, "String in /etc/oss.conf is too long!\n");
-	      goto oexit;
-	    }
-	  fclose (f);
-	  return name;
-	}
-    }
-
-  fclose (f);
-  fprintf (stderr, "Error: OSSLIBDIR not set in /etc/oss.conf\n");
-
-dexit:
   snprintf (name, sizeof (name), "%s/mixer.save", ossetcdir);
   return name;
-
-oexit:
-  fclose (f);
-  exit (-1);
 }
 
 static int
diff -up ./cmd/savemixer/savemixer.man.orig ./cmd/savemixer/savemixer.man
--- ./cmd/savemixer/savemixer.man.orig	2010-11-28 18:31:39.349085735 -0200
+++ ./cmd/savemixer/savemixer.man	2010-11-28 18:31:50.653490046 -0200
@@ -9,8 +9,7 @@ The savemixer program saves mixer settin
 settings back into the mixer.
 
 Running this program without any parameters will save the current mixer
-settings into /etc/oss/mixer.save or $OSSLIBDIR/etc/mixer.save file.
-OSSLIBDIR is decided by reading /etc/oss.conf, and defaults to /usr/lib/oss.
+settings into /etc/oss/mixer.save.
 
 AUTOMATIC SAVE
 By default the soundoff command will automatically run savemixer to save
@@ -27,11 +26,10 @@ SEE ALSO
 soundoff(1), soundon(1), ossdetect(1), ossdevlinks(1), ossmix(1), ossxmix(1)
 
 FILES
-/etc/oss.conf
 /usr/sbin/savemixer
-/usr/lib/oss/etc/mixer.save
-/usr/lib/oss/etc/dspdevs.map
-/usr/lib/oss/etc/applist.conf
+/etc/oss/mixer.save
+/etc/oss/dspdevs.map
+/etc/oss/applist.conf
 
 AUTHOR
 4Front Technologies
diff -up ./os_cmd/Linux/ossdetect/ossdetect.c.orig ./os_cmd/Linux/ossdetect/ossdetect.c
--- ./os_cmd/Linux/ossdetect/ossdetect.c.orig	2010-11-28 18:19:55.101924671 -0200
+++ ./os_cmd/Linux/ossdetect/ossdetect.c	2010-11-28 18:21:11.157433672 -0200
@@ -74,46 +74,8 @@ static void usb_detect (void);
 static char *
 get_mapname (void)
 {
-  FILE *f;
-  static char name[OSSLIBDIRLEN], tmp[OSSLIBDIRLEN+11];
-  struct stat st;
-
-  if ((f = fopen ("/etc/oss.conf", "r")) == NULL)
-    {
-      perror ("/etc/oss.conf");
-      goto oexit;
-    }
-
-  while (fgets (tmp, sizeof (tmp), f) != NULL)
-    {
-      int l = strlen (tmp);
-      if (l > 0 && tmp[l - 1] == '\n')
-	tmp[l - 1] = '\0';
-
-      if (strncmp (tmp, "OSSLIBDIR=", 10) == 0)
-	{
-	  l = snprintf (name, sizeof (name), "%s", &tmp[10]);
-	  if ((l >= OSSLIBDIRLEN) || (l < 0))
-	    {
-	      fprintf (stderr, "String in /etc/oss.conf is too long!\n");
-	      goto oexit;
-	    }
-	  if ((stat (name, &st) == -1) || !S_ISDIR(st.st_mode))
-	    {
-	      fprintf (stderr, "Directory %s from /etc/oss.conf cannot "
-			       "be used!\n", name);
-	      goto oexit;
-	    }
-	  fclose (f);
-	  return name;
-	}
-    }
-
-  fprintf (stderr, "OSSLIBDIR not set in /etc/oss.conf, using default "
-		   "/usr/lib/oss\n");
-oexit:
-  fclose (f);
-  snprintf (name, sizeof (name), "/usr/lib/oss");
+  static char name[OSSLIBDIRLEN];
+  snprintf (name, sizeof (name), "/etc/oss");
   return name;
 }
 
@@ -642,14 +604,14 @@ main (int argc, char *argv[])
 
   if (do_license == 1)
     {
-      load_license ("etc/license.asc");
+      load_license ("license.asc");
       exit (0);
     }
 
-  load_devlist ("etc/devices.list", 0);
+  load_devlist ("devices.list", 0);
 
-  if (stat ("/etc/oss_3rdparty", &st) != -1)
-    load_devlist ("/etc/oss_3rdparty", 1);
+  if (stat ("oss_3rdparty", &st) != -1)
+    load_devlist ("oss_3rdparty", 1);
 
   pci_detect (NULL);
   usb_detect ();
@@ -672,7 +634,7 @@ main (int argc, char *argv[])
     }
 
   snprintf (instfname, sizeof (instfname), "%s/%s", osslibdir,
-	    "etc/installed_drivers");
+	    "installed_drivers");
  
   if ((f = fopen (instfname, "w")) == NULL)
     {
diff -up ./os_cmd/Linux/ossdetect/ossdetect.man.orig ./os_cmd/Linux/ossdetect/ossdetect.man
--- ./os_cmd/Linux/ossdetect/ossdetect.man.orig	2010-11-28 18:21:23.891505454 -0200
+++ ./os_cmd/Linux/ossdetect/ossdetect.man	2010-11-28 18:22:02.955523214 -0200
@@ -6,9 +6,9 @@ ossdetect [-diuv] [-m<mode>] [-L<path>]
 
 DESCRIPTION
 The ossdetect application performs automatic detection of soundcards.
-The application looks at the /usr/lib/oss/etc/device.list and performs
+The application looks at the /etc/oss/device.list and performs
 automatic detection. It will create a file with the list of the configured
-drivers in /usr/lib/oss/etc/installed_drivers.
+drivers in /etc/oss/installed_drivers.
 
 The Input Multiplexer driver (IMUX) is not configured by default and
 can be added to the OSS configuration using the -i option.
@@ -19,8 +19,6 @@ OPTIONS
 -m<mode>	Create nodes with permissions set to octal <mode>.
 -u		Adds the oss_userdev driver.
 -v		Verbose output.
--L<path>	Use <path> as root directory for OSS files.
-		Default: Path from /etc/oss.conf, followed by /usr/lib/oss.
 
 SEE ALSO
 ossdevlinks(1), ossinfo(1)
diff -up ./setup/Linux/build.sh.orig ./setup/Linux/build.sh
--- ./setup/Linux/build.sh.orig	2010-11-28 18:22:11.464566795 -0200
+++ ./setup/Linux/build.sh	2010-11-28 18:23:53.388135411 -0200
@@ -129,18 +129,18 @@ do
 
 	if test -f $SRCDIR/kernel/drv/$N/$N.man
 	then
-	  sed "s:CONFIGFILEPATH:$OSSLIBDIR/conf:g" < $SRCDIR/kernel/drv/$N/$N.man > /tmp/ossman.txt
+	  sed "s:CONFIGFILEPATH:/etc/oss/conf:g" < $SRCDIR/kernel/drv/$N/$N.man > /tmp/ossman.txt
 	  $TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 /tmp/ossman.txt | gzip -9 > prototype/usr/share/man/man7/$N.7.gz
 	else
 		if test -f $SRCDIR/kernel/nonfree/drv/$N/$N.man
 		then
-	  		sed "s:CONFIGFILEPATH:$OSSLIBDIR/conf:g" < $SRCDIR/kernel/nonfree/drv/$N/$N.man > /tmp/ossman.txt
+	  		sed "s:CONFIGFILEPATH:/etc/oss/conf:g" < $SRCDIR/kernel/nonfree/drv/$N/$N.man > /tmp/ossman.txt
 	  		$TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 $SRCDIR/kernel/nonfree/drv/$N/$N.man | gzip -9 > prototype/usr/share/man/man7/$N.7.gz
 		fi
 	fi
 done
 
-sed "s:CONFIGFILEPATH:$OSSLIBDIR/conf:g" < $SRCDIR/kernel/drv/osscore/osscore.man > /tmp/ossman.txt
+sed "s:CONFIGFILEPATH:/etc/oss/conf:g" < $SRCDIR/kernel/drv/osscore/osscore.man > /tmp/ossman.txt
 $TXT2MAN -t "osscore" -v "OSS Devices" -s 7 /tmp/ossman.txt | gzip -9 > prototype/usr/share/man/man7/osscore.7.gz
 rm -f /tmp/ossman.txt
 
