diff -Nrbu seamonkey-2.22/comm-release/mozilla/build/unix/elfhack/Makefile.in seamonkey-2.22-OK/comm-release/mozilla/build/unix/elfhack/Makefile.in
--- seamonkey-2.22/comm-release/mozilla/build/unix/elfhack/Makefile.in	2013-10-29 05:20:42.000000000 +0400
+++ seamonkey-2.22-OK/comm-release/mozilla/build/unix/elfhack/Makefile.in	2013-11-01 18:54:56.575000337 +0400
@@ -22,7 +22,7 @@
   elfhack.cpp \
   $(NULL)
 
-OS_CXXFLAGS := $(filter-out -fno-exceptions,$(OS_CXXFLAGS)) -fexceptions
+HOST_CXXFLAGS := $(filter-out -fno-exceptions,$(HOST_CXXFLAGS)) -fexceptions
 
 CSRCS := \
   test-ctors.c \
