diff -Nrbu seamonkey-2.29/comm-release/mail/app/profile/all-thunderbird.js seamonkey-2.29-OK/comm-release/mail/app/profile/all-thunderbird.js
--- seamonkey-2.29/comm-release/mail/app/profile/all-thunderbird.js	2014-09-05 09:30:37.000000000 +0400
+++ seamonkey-2.29-OK/comm-release/mail/app/profile/all-thunderbird.js	2014-09-20 21:58:30.849935519 +0400
@@ -174,7 +174,8 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+//pref("extensions.showMismatchUI", false);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
diff -Nrbu seamonkey-2.29/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm seamonkey-2.29-OK/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm
--- seamonkey-2.29/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2014-09-05 09:37:10.000000000 +0400
+++ seamonkey-2.29-OK/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2014-09-20 22:01:03.526931450 +0400
@@ -1888,11 +1888,7 @@
       // Changes to installed extensions may have changed which theme is selected
       this.applyThemeChange();
 
-      if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
-        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
-      }
-      else if (aAppChanged && !this.allAppGlobal &&
+      if (aAppChanged && !this.allAppGlobal &&
                Prefs.getBoolPref(PREF_EM_SHOW_MISMATCH_UI, true)) {
         if (!Prefs.getBoolPref(PREF_SHOWN_SELECTION_UI, false)) {
           // Flip a flag to indicate that we interrupted startup with an interactive prompt
@@ -1913,6 +1909,10 @@
           }
         }
       }
+      else {
+        // For all profiles we will never need to show the add-on selection UI
+        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
+      }
 
       if (flushCaches) {
         flushStartupCache();
