diff -Nrbu seamonkey-2.35/comm-release/mail/app/profile/all-thunderbird.js seamonkey-2.35-OK/comm-release/mail/app/profile/all-thunderbird.js
--- seamonkey-2.35/comm-release/mail/app/profile/all-thunderbird.js	2015-09-02 12:16:10.000000000 +0300
+++ seamonkey-2.35-OK/comm-release/mail/app/profile/all-thunderbird.js	2015-09-13 18:25:27.567623985 +0300
@@ -173,7 +173,8 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+//pref("extensions.showMismatchUI", false);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
diff -Nrbu seamonkey-2.35/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm seamonkey-2.35-OK/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm
--- seamonkey-2.35/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2015-09-02 12:29:18.000000000 +0300
+++ seamonkey-2.35-OK/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2015-09-13 18:41:26.711622868 +0300
@@ -2102,11 +2102,7 @@
 
       AddonManagerPrivate.markProviderSafe(this);
 
-      if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
-        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
-      }
-      else if (aAppChanged && !this.allAppGlobal &&
+      if (aAppChanged && !this.allAppGlobal &&
                Preferences.get(PREF_EM_SHOW_MISMATCH_UI, true)) {
         if (!Preferences.get(PREF_SHOWN_SELECTION_UI, false)) {
           // Flip a flag to indicate that we interrupted startup with an interactive prompt
@@ -2126,6 +2122,9 @@
             flushCaches = true;
           }
         }
+      } else {
+        // For new profiles we will never need to show the add-on selection UI
+        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
       }
 
       if (flushCaches) {
