--- mozilla/xpcom/idl-parser/Makefile.in.orig	2012-06-07 15:08:55.192504035 -0400
+++ mozilla/xpcom/idl-parser/Makefile.in	2012-06-07 15:09:03.105850394 -0400
@@ -61,7 +61,7 @@
 	$(PYTHON_PATH) \
 	  -I$(topsrcdir)/other-licenses/ply \
 	  -I$(topsrcdir)/xpcom/idl-parser \
-	  $(topsrcdir)/xpcom/idl-parser/header.py --cachedir=. --regen
+	  $(topsrcdir)/xpcom/idl-parser/header.py --cachedir=cache --regen
 
 check::
 	$(PYTHON_PATH) \
--- mozilla/xpcom/typelib/xpidl/Makefile.in.orig	2012-06-07 17:05:20.594538497 -0400
+++ mozilla/xpcom/typelib/xpidl/Makefile.in	2012-06-07 17:05:35.930599502 -0400
@@ -48,8 +48,8 @@
   $(topsrcdir)/xpcom/idl-parser/xpidl.py \
   $(topsrcdir)/xpcom/idl-parser/header.py \
   $(topsrcdir)/xpcom/idl-parser/typelib.py \
-  $(DEPTH)/xpcom/idl-parser/xpidllex.py \
-  $(DEPTH)/xpcom/idl-parser/xpidlyacc.py \
+  $(DEPTH)/xpcom/idl-parser/cache/xpidllex.py \
+  $(DEPTH)/xpcom/idl-parser/cache/xpidlyacc.py \
   $(NULL)
 
 ifndef MOZ_SYSTEM_PLY

