diff -Nrbu seamonkey-2.26/comm-release/mail/app/profile/all-thunderbird.js seamonkey-2.26-OK/comm-release/mail/app/profile/all-thunderbird.js
--- seamonkey-2.26/comm-release/mail/app/profile/all-thunderbird.js	2014-04-29 08:55:07.000000000 +0400
+++ seamonkey-2.26-OK/comm-release/mail/app/profile/all-thunderbird.js	2014-05-06 21:38:43.918496556 +0400
@@ -174,7 +174,7 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
diff -Nrbu seamonkey-2.26/comm-release/mozilla/toolkit/mozapps/extensions/XPIProvider.jsm seamonkey-2.26-OK/comm-release/mozilla/toolkit/mozapps/extensions/XPIProvider.jsm
--- seamonkey-2.26/comm-release/mozilla/toolkit/mozapps/extensions/XPIProvider.jsm	2014-04-29 09:03:18.000000000 +0400
+++ seamonkey-2.26-OK/comm-release/mozilla/toolkit/mozapps/extensions/XPIProvider.jsm	2014-05-06 21:38:43.920496556 +0400
@@ -2051,8 +2051,8 @@
         this.showUpgradeUI();
         flushCaches = true;
       }
-      else if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
+    else {
+      // For all profiles we will never need to show the add-on selection UI
         Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
       }
 
