diff -Nrbu seamonkey-2.32/comm-release/mail/app/profile/all-thunderbird.js seamonkey-2.32-OK/comm-release/mail/app/profile/all-thunderbird.js
--- seamonkey-2.32/comm-release/mail/app/profile/all-thunderbird.js	2015-01-13 07:25:42.000000000 +0300
+++ seamonkey-2.32-OK/comm-release/mail/app/profile/all-thunderbird.js	2015-01-21 00:11:56.545827322 +0300
@@ -174,7 +174,8 @@
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
+//pref("extensions.showMismatchUI", false);
 
 // Preferences for AMO integration
 pref("extensions.getAddons.cache.enabled", true);
diff -Nrbu seamonkey-2.32/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm seamonkey-2.32-OK/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm
--- seamonkey-2.32/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2015-01-13 07:35:13.000000000 +0300
+++ seamonkey-2.32-OK/comm-release/mozilla/toolkit/mozapps/extensions/internal/XPIProvider.jsm	2015-01-21 00:16:10.854827053 +0300
@@ -2093,11 +2093,7 @@
       // Changes to installed extensions may have changed which theme is selected
       this.applyThemeChange();
 
-      if (aAppChanged === undefined) {
-        // For new profiles we will never need to show the add-on selection UI
-        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
-      }
-      else if (aAppChanged && !this.allAppGlobal &&
+      if (aAppChanged && !this.allAppGlobal &&
                Preferences.get(PREF_EM_SHOW_MISMATCH_UI, true)) {
         if (!Preferences.get(PREF_SHOWN_SELECTION_UI, false)) {
           // Flip a flag to indicate that we interrupted startup with an interactive prompt
@@ -2118,6 +2114,10 @@
           }
         }
       }
+      else {
+        // For all profiles we will never need to show the add-on selection UI
+        Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
+      }
 
       if (flushCaches) {
         flushStartupCache();
