diff -Nrbu seamonkey-2.17/comm-release/mozilla/media/webrtc/trunk/tools/gyp/pylib/gyp/generator/mozmake.py seamonkey-2.17-OK/comm-release/mozilla/media/webrtc/trunk/tools/gyp/pylib/gyp/generator/mozmake.py
--- seamonkey-2.17/comm-release/mozilla/media/webrtc/trunk/tools/gyp/pylib/gyp/generator/mozmake.py	2013-04-01 07:31:07.000000000 +0400
+++ seamonkey-2.17-OK/comm-release/mozilla/media/webrtc/trunk/tools/gyp/pylib/gyp/generator/mozmake.py	2013-04-03 16:23:51.027633271 +0400
@@ -377,7 +377,7 @@
     WriteMakefile(output_file, data, build_file, depth, top,
                   # we set srcdir up one directory, since the subdir
                   # doesn't actually exist in the source directory
-                  swapslashes(os.path.join(top, self.relative_srcdir, os.path.split(rel_path)[0])),
+                  swapslashes(os.path.normpath(os.path.join(top, self.relative_srcdir, os.path.split(rel_path)[0]))),
                   self.relative_srcdir,
                   self.common_mk_path)
     return True
