diff -Nru src-old/mess/mess.mak src/mess/mess.mak
--- src-old/mess/mess.mak	2012-09-17 08:37:37.000000000 +0200
+++ src/mess/mess.mak	2012-10-03 16:54:09.000000000 +0200
@@ -2154,6 +2156,7 @@
 $(MESS_DRIVERS)/lynx.o:		$(MESS_LAYOUT)/lynx.lh
 $(MESS_DRIVERS)/lx800.o:	$(MESS_LAYOUT)/lx800.lh
 $(MESS_DRIVERS)/mac.o:		$(MESS_LAYOUT)/mac.lh
+$(MAME_MACHINE)/megacd.o:	$(MAME_LAYOUT)/megacd.lh
 $(MAME_MACHINE)/megadriv.o:	$(MAME_LAYOUT)/megacd.lh
 $(MESS_DRIVERS)/megadriv.o:	$(MAME_LAYOUT)/megacd.lh
 $(MESS_DRIVERS)/mekd2.o:	$(MESS_LAYOUT)/mekd2.lh
