--- src/emu/mame.c.gcc45	2010-04-30 17:35:13.247626567 -0300
+++ src/emu/mame.c	2010-04-30 17:38:14.197375560 -0300
@@ -811,7 +811,7 @@
 UINT32 memory_region_length(running_machine *machine, const char *name)
 {
 	const region_info *region = machine->region(name);
-	return (region != NULL) ? region->length : NULL;
+	return (region != 0) ? region->length : 0;
 }
 
 
@@ -823,7 +823,7 @@
 UINT32 memory_region_flags(running_machine *machine, const char *name)
 {
 	const region_info *region = machine->region(name);
-	return (region != NULL) ? region->flags : NULL;
+	return (region != 0) ? region->flags : 0;
 }
 
 
