diff -up src/osd/sdl/dview.c.nogconf src/osd/sdl/dview.c
--- src/osd/sdl/dview.c.nogconf	2013-01-14 18:05:45.537932138 -0200
+++ src/osd/sdl/dview.c	2013-01-14 18:08:29.240989942 -0200
@@ -1,8 +1,5 @@
 
 #include "dview.h"
-#ifndef SDLMAME_WIN32
-#include <gconf/gconf-client.h>
-#endif
 
 G_DEFINE_TYPE(DView, dview, GTK_TYPE_CONTAINER);
 
@@ -347,13 +344,9 @@ static void dview_forall(GtkContainer *d
 static void dview_class_init(DViewClass *dvc)
 {
 #ifndef SDLMAME_WIN32
-	GConfClient *conf = gconf_client_get_default();
 	char *name = 0;
 	dvc->fixedfont = 0;
 
-	if(conf)
-		name = gconf_client_get_string(conf, "/desktop/gnome/interface/monospace_font_name", 0);
-
 	if(name) {
 		dvc->fixedfont = pango_font_description_from_string(name);
 		g_free(name);
diff -up src/osd/sdl/sdl.mak.nogconf src/osd/sdl/sdl.mak
--- src/osd/sdl/sdl.mak.nogconf	2013-01-14 18:05:45.522935339 -0200
+++ src/osd/sdl/sdl.mak	2013-01-14 18:05:52.255498283 -0200
@@ -685,9 +685,9 @@ LIBS += -lX11 -lXinerama
 
 # the new debugger relies on GTK+ in addition to the base SDLMAME needs
 # Non-X11 builds can not use the debugger
-INCPATH += `pkg-config --cflags-only-I gtk+-2.0` `pkg-config --cflags-only-I gconf-2.0`
-CCOMFLAGS += `pkg-config --cflags-only-other gtk+-2.0` `pkg-config --cflags-only-other gconf-2.0`
-LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
+INCPATH += `pkg-config --cflags-only-I gtk+-2.0`
+CCOMFLAGS += `pkg-config --cflags-only-other gtk+-2.0`
+LIBS += `pkg-config --libs gtk+-2.0`
 #CCOMFLAGS += -DGTK_DISABLE_DEPRECATED
 
 # The newer debugger uses QT
