diff -up playgsf-0.7.1/VBA/GBA.cpp.gcc46 playgsf-0.7.1/VBA/GBA.cpp
--- playgsf-0.7.1/VBA/GBA.cpp.gcc46	2011-10-09 21:37:24.000000000 -0300
+++ playgsf-0.7.1/VBA/GBA.cpp	2011-10-09 21:41:49.993984244 -0300
@@ -1054,12 +1054,12 @@ int CPULoadRom(const char *szFile)
   {
 	  rom = (u8 *)malloc(0x200);
 	  loadedsize=0;
-	  i = (int)utilLoad(szFile,utilIsGBAImage,whereToLoad,size);
+	  i = (long)utilLoad(szFile,utilIsGBAImage,whereToLoad,size);
   }
   else
   {
 	  rom = utilLoad(szFile,utilIsGBAImage,whereToLoad,size);
-	  i = (int) rom;
+	  i = (long) rom;
   }
 
   //loadedsize = sizeof(*rom);
diff -up playgsf-0.7.1/VBA/Util.cpp.gcc46 playgsf-0.7.1/VBA/Util.cpp
--- playgsf-0.7.1/VBA/Util.cpp.gcc46	2006-02-03 15:57:16.000000000 -0200
+++ playgsf-0.7.1/VBA/Util.cpp	2011-10-09 21:35:53.432791542 -0300
@@ -917,7 +917,7 @@ bool utilIsGSF(const char * file)
   
 
   if(strlen(file) > 4) {
-    char *p = strrchr(file,'.');
+    const char *p = strrchr(file,'.');
 
 	if(p != NULL) {
 	  if(_stricmp(p, ".gsf") == 0)
@@ -934,7 +934,7 @@ bool utilIsGBAImage(const char * file)
 {
   cpuIsMultiBoot = false;
   if(strlen(file) > 4) {
-    char * p = strrchr(file,'.');
+    const char * p = strrchr(file,'.');
 
     if(p != NULL) {
       //if(_stricmp(p, ".gba") == 0)
