diff -up playgsf-0.7.1/Makefile.in.bsys playgsf-0.7.1/Makefile.in
--- playgsf-0.7.1/Makefile.in.bsys	2006-02-03 15:57:17.000000000 -0200
+++ playgsf-0.7.1/Makefile.in	2011-10-09 21:46:28.796738110 -0300
@@ -8,15 +8,9 @@ LDFLAGS=@LDFLAGS@
 
 OBJS=gsf.o VBA/GBA.o VBA/Globals.o VBA/Sound.o VBA/Util.o VBA/bios.o VBA/memgzio.o VBA/snd_interp.o VBA/unzip.o linuxmain.o VBA/psftag.o
 
-all: libresample-0.1.3/libresample.a $(OBJS) 
+all: $(OBJS) 
 	$(LD) $(LDFLAGS) $(OBJS) -lresample -o playgsf
 
-libresample-0.1.3/libresample.a: libresample-0.1.3/Makefile
-	$(MAKE) -C libresample-0.1.3
-
-libresample-0.1.3/Makefile:
-	cd libresample-0.1.3 ; ./configure ; cd ..
-
 %.o: %.c %.h
 	$(CC) $(CFLAGS) -c $< -o $@
 
diff -up playgsf-0.7.1/configure.in.bsys playgsf-0.7.1/configure.in
--- playgsf-0.7.1/configure.in.bsys	2006-02-03 15:57:17.000000000 -0200
+++ playgsf-0.7.1/configure.in	2011-10-09 21:46:29.374615310 -0300
@@ -13,8 +13,8 @@ AC_CHECK_LIB(ao, ao_initialize, has_liba
 	exit
 ])
 
-CFLAGS="-DLINUX -I./VBA -DVERSION_STR=\\\"0.07\\\" -DHA_VERSION_STR=\\\"0.11\\\" -I./libresample-0.1.3/include"
-LDFLAGS="-lz -lresample -L./libresample-0.1.3 -lao"
+CFLAGS="$CFLAGS -DLINUX -I./VBA -DVERSION_STR=\\\"0.07\\\" -DHA_VERSION_STR=\\\"0.11\\\""
+LDFLAGS="$LDFLAGS -lz -lresample -lao"
 
 use_c_core=yes
 auto_c_core=yes
