From 00593fa55c16deb22304bbe7316911570e7c33b9 Mon Sep 17 00:00:00 2001
From: Kalev Lember <kalevlember@gmail.com>
Date: Sun, 27 Jan 2013 16:15:01 +0100
Subject: [PATCH] [GTK] Pass ICU cppflags to libWebCoreSVG.la build

---
 Source/WebCore/GNUmakefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/WebCore/GNUmakefile.am b/Source/WebCore/GNUmakefile.am
index 95a23ed..d3463ca 100644
--- a/Source/WebCore/GNUmakefile.am
+++ b/Source/WebCore/GNUmakefile.am
@@ -253,7 +253,8 @@ libWebCoreSVG_la_CPPFLAGS = \
 	-fno-strict-aliasing \
 	$(CAIRO_CFLAGS) \
 	$(FREETYPE_CFLAGS) \
-	$(LIBSOUP_CFLAGS)
+	$(LIBSOUP_CFLAGS) \
+	$(UNICODE_CFLAGS)
 
 feature_defines_overrides += ENABLE_SVG=1
 WEBCORE_CSS_PROPERTY_NAMES += $(WebCore)/css/SVGCSSPropertyNames.in
-- 
1.8.1

