Index: Source/JavaScriptCore/llint/LowLevelInterpreter.cpp
===================================================================
--- webkitgtk-1.11.2/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp	(Revision 137796)
+++ webkitgtk-1.11.2/Source/JavaScriptCore/llint/LowLevelInterpreter.cpp	(Arbeitskopie)
@@ -116,6 +116,17 @@
     u.ival64 = (static_cast<uint64_t>(hi) << 32) | lo;
     return u.dval;
 }
+
+static void Double2Ints(double input, intptr_t lo, intptr_t hi)
+{
+    union {
+        double dval;
+        int64_t ival64;
+    } u;
+    u.dval = input;
+    *(reinterpret_cast<int32_t*>(hi)) = static_cast<int32_t>(u.ival64 >> 32);
+    *(reinterpret_cast<int32_t*>(lo)) = static_cast<int32_t>(u.ival64);
+}
 #endif // USE(JSVALUE32_64)
 
 } // namespace LLint
