diff -up streamdev-0.6.0/Makefile.mkfix streamdev-0.6.0/Makefile
--- streamdev-0.6.0/Makefile.mkfix	2011-05-02 17:16:31.000000000 -0300
+++ streamdev-0.6.0/Makefile	2013-03-01 19:38:31.984269414 -0300
@@ -27,15 +27,12 @@ TMPDIR = /tmp
 
 ### The version number of VDR (taken from VDR's "config.h"):
 
-APIVERSION = $(shell grep 'define APIVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
-APIVERSNUM = $(shell grep 'define APIVERSNUM ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
-TSPLAYVERSNUM = $(shell grep 'define TSPLAY_PATCH_VERSION ' $(VDRDIR)/device.h | awk '{ print $$3 }')
+APIVERSION = $(shell grep 'define APIVERSION ' _INCDIR_/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+APIVERSNUM = $(shell grep 'define APIVERSNUM ' _INCDIR_/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+TSPLAYVERSNUM = $(shell grep 'define TSPLAY_PATCH_VERSION ' _INCDIR_/device.h | awk '{ print $$3 }')
 
 ### Allow user defined options to overwrite defaults:
 
-ifeq ($(shell test $(APIVERSNUM) -ge 10713; echo $$?),0)
-include $(VDRDIR)/Make.global
-else
 ifeq ($(shell test $(APIVERSNUM) -ge 10704 -o -n "$(TSPLAYVERSNUM)" ; echo $$?),0)
 DEFINES  += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
 CFLAGS   += -fPIC
@@ -44,7 +41,6 @@ else
 CFLAGS   += -fPIC
 CXXFLAGS += -fPIC
 endif
-endif
 
 -include $(VDRDIR)/Make.config
 
@@ -62,7 +58,7 @@ PACKAGE = vdr-$(ARCHIVE)
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I..
+INCLUDES += -I_INCDIR_ -I..
 
 DEFINES += -D_GNU_SOURCE
 
