--- emesene-2.12.3/emesene/data/share/applications/emesene.desktop.orig	2012-05-21 18:00:04.112377445 -0300
+++ emesene-2.12.3/emesene/data/share/applications/emesene.desktop	2012-05-21 18:00:11.106887251 -0300
@@ -7,9 +7,3 @@ Icon=emesene
 Terminal=false
 Type=Application
 Categories=Network;InstantMessaging;
-X-Ayatana-Desktop-Shortcuts=NewInstance;
-
-[NewInstance Shortcut Group]
-Name=Open a new instance
-Exec=emesene
-TargetEnvironment=Unity
