--- configure.in.ppl10	2012-07-06 19:00:48.860177319 -0300
+++ configure.in	2012-07-06 19:01:08.384035483 -0300
@@ -338,9 +338,11 @@ if test "x$with_ppl" != "x" -a "x$with_p
 
 	AC_MSG_CHECKING([for version 0.10 (or later revision) of PPL])
 	AC_TRY_COMPILE([#include "ppl_c.h"],[
+	#if PPL_VERSION_MAJOR != 1
 	#if PPL_VERSION_MAJOR != 0 || PPL_VERSION_MINOR < 10
 	choke me
 	#endif
+	#endif
 	], [AC_MSG_RESULT([yes])], [AC_MSG_ERROR(Can't find correct version of PPL.) ])
 
 	LIBS="$LIBS -lppl_c -lppl -lgmpxx"
