--- data/xfce/xfce-applications.menu.orig	2015-06-29 18:57:31.161269933 -0300
+++ data/xfce/xfce-applications.menu	2015-06-29 18:58:27.025376476 -0300
@@ -41,13 +41,6 @@
             <Merge type="all"/>
         </Layout>
 
-        <Menu>
-            <Name>Screensavers</Name>
-            <Directory>xfce-screensavers.directory</Directory>
-            <Include>
-                <Category>Screensaver</Category>
-            </Include>
-        </Menu>
     </Menu>
 
     <Menu>
@@ -156,7 +149,23 @@
         <Directory>xfce-other.directory</Directory>
         <OnlyUnallocated/>
         <Include>
-            <All/>
+            <And>
+                <Not><Category>X-KDE-settings-accessibility</Category></Not>
+                <Not><Category>X-KDE-settings-components</Category></Not>
+                <Not><Category>X-KDE-settings-desktop</Category></Not>
+                <Not><Category>X-KDE-settings-looknfeel</Category></Not>
+                <Not><Category>X-KDE-settings-network</Category></Not>
+                <Not><Category>X-KDE-settings-webbrowsing</Category></Not>
+                <Not><Category>X-KDE-settings-peripherals</Category></Not>
+                <Not><Category>X-KDE-settings-hardware</Category></Not>
+                <Not><Category>X-KDE-settings-power</Category></Not>
+                <Not><Category>X-KDE-settings-security</Category></Not>
+                <Not><Category>X-KDE-settings-sound</Category></Not>
+                <Not><Category>X-KDE-settings-system</Category></Not>
+                <Not><Category>X-KDE-information</Category></Not>
+                <Not><Category>X-KDE-ScreenSaver</Category></Not>
+                <Not><Category>Screensaver</Category></Not>
+            </And>
         </Include>
     </Menu>
 
