--- bluez-4.75/src/bluetooth.conf.orig	2010-10-04 08:41:01.000000000 -0300
+++ bluez-4.75/src/bluetooth.conf	2010-10-07 22:55:54.248119682 -0300
@@ -11,13 +11,18 @@
     <allow own="org.bluez"/>
     <allow send_destination="org.bluez"/>
     <allow send_interface="org.bluez.Agent"/>
+    <allow send_interface="org.bluez.Adapter"/>
     <allow send_interface="org.bluez.HandsfreeAgent"/>
     <allow send_interface="org.bluez.MediaEndpoint"/>
     <allow send_interface="org.bluez.Watcher"/>
   </policy>
 
-  <policy at_console="true">
-    <allow send_destination="org.bluez"/>
+  <policy group="plugdev">
+     <allow send_destination="org.bluez"/>
+     <allow send_interface="org.bluez.Agent"/>
+     <allow send_interface="org.bluez.Adapter"/>
+     <allow send_destination="org.bluez"
+            send_interface="org.freedesktop.DBus.Introspectable"/>
   </policy>
 
   <!-- allow users of lp group (printing subsystem) to 
