--- bluez-4.97/src/bluetooth.conf.orig	2011-12-21 20:53:54.000000000 -0200
+++ bluez-4.97/src/bluetooth.conf	2012-01-01 20:00:22.414857505 -0200
@@ -11,6 +11,7 @@
     <allow own="org.bluez"/>
     <allow send_destination="org.bluez"/>
     <allow send_interface="org.bluez.Agent"/>
+    <allow send_interface="org.bluez.Adapter"/>
     <allow send_interface="org.bluez.HandsfreeAgent"/>
     <allow send_interface="org.bluez.MediaEndpoint"/>
     <allow send_interface="org.bluez.MediaPlayer"/>
@@ -18,8 +19,12 @@
     <allow send_interface="org.bluez.ThermometerWatcher"/>
   </policy>
 
-  <policy at_console="true">
-    <allow send_destination="org.bluez"/>
+  <policy group="plugdev">
+     <allow send_destination="org.bluez"/>
+     <allow send_interface="org.bluez.Agent"/>
+     <allow send_interface="org.bluez.Adapter"/>
+     <allow send_destination="org.bluez"
+            send_interface="org.freedesktop.DBus.Introspectable"/>
   </policy>
 
   <!-- allow users of lp group (printing subsystem) to 
