Index: bluez/sbc/sbc_primitives_mmx.c
===================================================================
--- bluez.orig/sbc/sbc_primitives_mmx.c	2012-01-04 12:06:47.231873000 -0500
+++ bluez/sbc/sbc_primitives_mmx.c	2012-01-04 12:07:34.748103759 -0500
@@ -318,7 +318,7 @@
 				"movl          %k0, 4(%3)\n"
 			: "+r" (blk)
 			: "r" (&sb_sample_f[0][ch][sb]),
-				"i" ((char *) &sb_sample_f[1][0][0] -
+				"r" ((char *) &sb_sample_f[1][0][0] -
 					(char *) &sb_sample_f[0][0][0]),
 				"r" (&scale_factor[ch][sb]),
 				"r" (&consts),
