--- nspluginwrapper-1.4.4/configure.orig	2011-06-30 23:18:57.000000000 -0400
+++ nspluginwrapper-1.4.4/configure	2012-05-14 15:01:01.000000000 -0400
@@ -985,6 +985,10 @@
     echo "TARGET_ARCH=ppc" >> $config_mak
     echo "#define TARGET_ARCH \"ppc\"" >> $config_h
     echo "#define TARGET_PPC 1" >> $config_h
+elif test "$target_cpu" = "arm" ; then
+    echo "TARGET_ARCH=arm" >> $config_mak
+    echo "#define TARGET_ARCH \"arm\"" >> $config_h
+    echo "#define TARGET_ARM 1" >> $config_h
 else
     echo "Unsupported target CPU"
     exit 1
