diff -up nspluginwrapper-1.4.4/Makefile.pthread nspluginwrapper-1.4.4/Makefile
--- nspluginwrapper-1.4.4/Makefile.pthread	2011-10-27 17:11:08.048040662 +0200
+++ nspluginwrapper-1.4.4/Makefile	2011-10-27 17:11:08.057040729 +0200
@@ -37,7 +37,7 @@ ifneq (,$(findstring $(OS),linux))
 libdl_LIBS = -ldl
 endif
 
-libpthread_LIBS = -lpthread
+libpthread_LIBS = __PTHREAD_LIBS__
 ifeq ($(OS),dragonfly)
 libpthread_LDFLAGS = -pthread
 endif
--- nspluginwrapper-1.4.4/src/npw-player.c.pthread	2011-12-29 05:44:50.000000000 +0100
+++ nspluginwrapper-1.4.4/src/npw-player.c	2012-09-14 14:31:47.669931757 +0200
@@ -29,4 +29,5 @@
 #include <gdk/gdkx.h>
 #include <gdk/gdkkeysyms.h>
+#include <pthread.h>
 
 #include "rpc.h"
