diff -up nspluginwrapper-1.4.4/Makefile.make nspluginwrapper-1.4.4/Makefile
--- nspluginwrapper-1.4.4/Makefile.make	2011-12-29 05:44:50.000000000 +0100
+++ nspluginwrapper-1.4.4/Makefile	2012-09-14 14:21:50.207020448 +0200
@@ -104,6 +104,7 @@ npwrapper_CFLAGS  = $(CFLAGS) $(X_CFLAGS
 npwrapper_LDFLAGS = $(LDFLAGS) $(libpthread_LDFLAGS)
 npwrapper_LIBS    = $(X_LIBS) $(libpthread_LIBS) $(libsocket_LIBS)
 npwrapper_LIBS   += $(GLIB_LIBS)
+npwrapper_LIBS   += $(LDFLAGS)
 
 npviewer_PROGRAM  = npviewer.bin
 npviewer_RAWSRCS  = npw-viewer.c npw-common.c npw-malloc.c npw-rpc.c rpc.c debug.c utils.c npruntime.c
@@ -118,6 +119,7 @@ npviewer_LDFLAGS  = $(LDFLAGS_32)
 npviewer_LDFLAGS += $(libpthread_LDFLAGS)
 npviewer_LIBS     = $(GTK_LIBS_32) $(GLIB_LIBS_32) $(X_LIBS_32)
 npviewer_LIBS    += $(libdl_LIBS) $(libpthread_LIBS)
+npviewer_LIBS    += $(LDFLAGS)
 ifeq ($(TARGET_OS):$(TARGET_ARCH),linux:i386)
 npviewer_MAPFILE  = $(SRC_PATH)/src/npw-viewer.map
 endif
@@ -302,13 +304,13 @@ npplayer-%.o: $(SRC_PATH)/src/tidy/%.c
 	$(CC) -o $@ -c $< $(CPPFLAGS) $(npplayer_CFLAGS) -DBUILD_PLAYER
 
 $(libnoxshm_LIBRARY): $(libnoxshm_OBJECTS) $(LSB_OBJ_DIR) $(LSB_LIBS)
-	$(CC) $(DSO_LDFLAGS) $(libnoxshm_LDFLAGS) -o $@ $(libnoxshm_OBJECTS) -Wl,$(LD_soname),libnoxshm.so
+	$(CC) $(LDFLAGS) $(DSO_LDFLAGS) $(libnoxshm_LDFLAGS) -o $@ $(libnoxshm_OBJECTS) -Wl,$(LD_soname),libnoxshm.so
 
 libnoxshm-%.o: $(SRC_PATH)/src/%.c
 	$(CC) -o $@ -c $< $(CPPFLAGS) $(libnoxshm_CFLAGS)
 
 $(npconfig_PROGRAM): $(npconfig_OBJECTS)
-	$(CC) $(npconfig_LDFLAGS) -o $@ $(npconfig_OBJECTS) $(npconfig_LIBS)
+	$(CC) $(npconfig_LDFLAGS) -o $@ $(npconfig_OBJECTS) $(npconfig_LIBS) $(LDFLAGS)
 
 npconfig-%.o: $(SRC_PATH)/src/%.c
 	$(CC) $(npconfig_CFLAGS) -o $@ -c $< $(CPPFLAGS)
@@ -334,7 +336,7 @@ $(LSB_OBJ_DIR)/libgcc_s_32.so: $(LSB_OBJ
 	$(LN_S) libgcc_s.so $@
 
 $(LSB_OBJ_DIR)/%.so: $(LSB_OBJ_DIR)/%.o | $(LSB_OBJ_DIR)
-	$(CC) $(LDFLAGS_32) -nostdlib $(DSO_LDFLAGS) $< -o $@ \
+	$(CC) $(LDFLAGS) $(LDFLAGS_32) -nostdlib $(DSO_LDFLAGS) $< -o $@ \
 		-Wl,--version-script,$(patsubst $(LSB_OBJ_DIR)/%.o,$(LSB_SRC_DIR)/%.Version,$<) \
 		-Wl,-soname,`grep "$(patsubst $(LSB_OBJ_DIR)/%.o,%,$<) " $(LSB_SRC_DIR)/LibNameMap.txt | cut -f2 -d' '`
 
