diff -up nspluginwrapper-1.1.2/plugin-config-1.9/src/plugin-config.c.old nspluginwrapper-1.1.2/plugin-config-1.9/src/plugin-config.c
--- nspluginwrapper-1.1.2/plugin-config-1.9/src/plugin-config.c.old	2008-10-22 13:35:55.000000000 +0200
+++ nspluginwrapper-1.1.2/plugin-config-1.9/src/plugin-config.c	2008-10-22 13:37:23.000000000 +0200
@@ -453,7 +453,7 @@ int wrap_plugin_2(WRAP_PLUGIN *p_plugin,
         NULL
       };
     
-      info("Running: %s");
+      info("Running: %s", p_wrapper);
       for(i = 0; i < sizeof(params)/sizeof(params[0]); i++) {
         if(params[i]) {
           info("         %s",params[i]);
