--- nspluginwrapper-1.1.0/plugin-config-1.9/src/plugin-config.c.old	2008-02-26 12:52:02.000000000 +0100
+++ nspluginwrapper-1.1.0/plugin-config-1.9/src/plugin-config.c	2008-10-17 12:18:39.000000000 +0200
@@ -785,6 +785,8 @@ void configure(void)
   cfg.action = A_CHECK;
 
   create_exclude_lists();  
+
+  umask(S_IWGRP|S_IWOTH); 
 }
 
 /**********************************************************************************
