diff -up nspluginwrapper-1.1.0/plugin-config-1.9/src/Makefile.in.set nspluginwrapper-1.1.0/plugin-config-1.9/src/Makefile.in
--- nspluginwrapper-1.1.0/plugin-config-1.9/src/Makefile.in.set	2008-02-26 12:52:28.000000000 +0100
+++ nspluginwrapper-1.1.0/plugin-config-1.9/src/Makefile.in	2008-07-21 10:40:57.000000000 +0200
@@ -44,7 +44,7 @@ mkinstalldirs = $(install_sh) -d
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 am__installdirs = "$(DESTDIR)$(bindir)"
-binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
+binPROGRAMS_INSTALL = $(INSTALL_PROGRAM) -m 4755
 PROGRAMS = $(bin_PROGRAMS)
 am_plugin_config_OBJECTS = plugin-config.$(OBJEXT) \
 	plugin-detection.$(OBJEXT) plugin-dir.$(OBJEXT)
