--- nspluginwrapper-1.4.4/plugin-config-1.9/src/plugin-config.c	2012-12-13 09:24:27.988306638 +0100
+++ nspluginwrapper-1.4.4/plugin-config-1.9/src/plugin-config.c.help	2012-12-13 09:25:23.015335592 +0100
@@ -889,11 +889,11 @@ to \n\
                                                                           \n\
 Usage:                                                                    \n\
                                                                           \n\
-  plugin-setup  [options]                                                 \n\
+  plugin-config  [options]                                                \n\
                                                                           \n\
 Options:                                                                  \n\
                                                                           \n\
-  -i  --install Install/wrap all availabile plug-ins                      \n\
+  -i  --install Install/wrap all available plug-ins                       \n\
   -r  --remove  Remove all installed plug-ins                             \n\
                                                                           \n\
   -l  --list    List all installed plug-ins                               \n\
@@ -908,7 +908,7 @@ Options:
   -v  --verbose Verbose mode                                              \n\
   -h  --help    This help                                                 \n\
                                                                           \n\
-  If it is run without any command line argument, --check is used.        \n\\n",
+  If it is run without any command line argument, --check is used.        \n\n",
   NPW_PLUGIN_DIR_32,NPW_PLUGIN_DIR_64,
   NPW_PLUGIN_DIR_32_WRAPPED,NPW_PLUGIN_DIR_64_WRAPPED );
   
