diff -up xaos-3.6/src/engine/zoom.c.proto xaos-3.6/src/engine/zoom.c
--- xaos-3.6/src/engine/zoom.c.proto	2013-11-01 08:54:21.000000000 +0100
+++ xaos-3.6/src/engine/zoom.c	2015-07-09 18:56:49.884124804 +0200
@@ -138,6 +138,7 @@ int nperi = 0;
 #endif
 
 #ifndef USE_i386ASM
+REGISTERS(0)
 static void
 moveoldpoints(void *data1, struct taskinfo *task, int r1, int r2)
 REGISTERS(0);
@@ -933,6 +934,7 @@ static /*INLINE */ void preparemoveoldpo
 }
 
 #ifndef USE_i386ASM
+REGISTERS(0)
 static /*INLINE */ void
 moveoldpoints(void /*@unused@ */ *data1,
 	      struct taskinfo /*@unused@ */ *task,
diff -up xaos-3.6/src/engine/zoomd.c.proto xaos-3.6/src/engine/zoomd.c
--- xaos-3.6/src/engine/zoomd.c.proto	2013-11-01 08:54:21.000000000 +0100
+++ xaos-3.6/src/engine/zoomd.c	2015-07-09 18:53:38.665002524 +0200
@@ -280,7 +280,8 @@ dosymetry2(void /*@unused@ */ *data, str
 #endif
 #undef bpp1
 
-static INLINE void fillline(int line)
+REGISTERS(0)
+static void fillline(int line)
 {
     register unsigned char *RESTRICT vbuff = cimage.currlines[line];
     const struct filltable *RESTRICT table = (struct filltable *) tmpdata;
