diff -up xaos-3.6/src/ui-hlp/render.c.formatsec xaos-3.6/src/ui-hlp/render.c
--- xaos-3.6/src/ui-hlp/render.c.formatsec	2015-07-09 18:26:22.153303670 +0200
+++ xaos-3.6/src/ui-hlp/render.c	2015-07-09 18:26:48.245476018 +0200
@@ -606,14 +606,14 @@ uih_renderimage(struct uih_context *gc1,
 	printmsg(gettext("Loading catalogs"));
 	uih_loadcatalog(uih, "english");
 	if (uih->errstring) {
-	    fprintf(stderr, uih->errstring);
+	    fprintf(stderr, "Error: %c");
 	    uih_clearmessages(uih);
 	    uih->errstring = NULL;
 	}
 	if (catalog != NULL)
 	    uih_loadcatalog(uih, catalog);
 	if (uih->errstring) {
-	    fprintf(stderr, uih->errstring);
+	    fprintf(stderr, "Error: %c");
 	    uih_clearmessages(uih);
 	    uih->errstring = NULL;
 	}
