diff -up kmix-4.11.97/kmix_autostart.desktop.autostart kmix-4.11.97/kmix_autostart.desktop
--- kmix-4.11.97/kmix_autostart.desktop.autostart	2013-09-03 04:57:22.000000000 -0500
+++ kmix-4.11.97/kmix_autostart.desktop	2013-12-08 22:05:18.359384464 -0600
@@ -6,6 +6,7 @@ Type=Application
 MimeType=
 Terminal=false
 X-KDE-autostart-after=panel
+X-KDE-autostart-after=pulseaudio
 X-KDE-StartupNotify=false
 X-DBUS-StartupType=Unique
 X-KDE-autostart-condition=kmixrc:Global:AutoStart:true
diff -up kmix-4.11.97/restore_kmix_volumes.desktop.autostart kmix-4.11.97/restore_kmix_volumes.desktop
--- kmix-4.11.97/restore_kmix_volumes.desktop.autostart	2013-09-03 04:57:22.000000000 -0500
+++ kmix-4.11.97/restore_kmix_volumes.desktop	2013-12-08 22:10:12.757238970 -0600
@@ -1,5 +1,6 @@
 [Desktop Entry]
 X-KDE-autostart-phase=1
+X-KDE-autostart-after=pulseaudio
 X-KDE-autostart-condition=kmixrc:Global:startkdeRestore:true
 Type=Application
 Exec=kmixctrl --restore
