From 3271ba0553cd8805ea0669edd67b16f4ecee332e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Sat, 24 Oct 2015 18:18:04 -0400
Subject: [PATCH] getcwd

---
 makeint.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git makeint.h makeint.h
index fdcae75bd2..98c022deac 100644
--- makeint.h
+++ makeint.h
@@ -548,7 +548,7 @@ long int lseek ();
 #endif  /* Not GNU C library or POSIX.  */
 
 #ifdef  HAVE_GETCWD
-# if !defined(VMS) && !defined(__DECC)
+# if !defined(VMS) && !defined(__DECC) && !defined(getcwd)
 char *getcwd ();
 # endif
 #else
-- 
2.5.0

