--- make-3.81/tests/scripts/features/recursion~	2005-02-10 01:10:58.000000000 +0100
+++ make-3.81/tests/scripts/features/recursion	2010-05-26 15:30:30.000000000 +0200
@@ -16,7 +16,7 @@ last:
 	@echo MAKELEVEL = $(MAKELEVEL)
 	@echo THE END
 ',
-              ('CFLAGS=-O -w' . ($parallel_jobs ? '-j 2' : '')),
+              ('CFLAGS=-O -w' . ($parallel_jobs ? ' -j 2' : '')),
               ($vos
                ? "#MAKE#: Entering directory `#PWD#'
 make 'CFLAGS=-O' -f #MAKEFILE# foo

Diff finished.  Wed May 26 15:30:43 2010
