diff -Nrup a/main.c b/main.c
--- a/main.c	2013-10-09 00:22:40.000000000 -0400
+++ b/main.c	2014-02-03 17:18:04.238609236 -0500
@@ -1987,6 +1987,20 @@ main (int argc, char **argv, char **envp
     }
 #endif
 
+#ifdef PIPE_BUF
+  if (job_slots > PIPE_BUF)
+#elif defined _POSIX_PIPE_BUF
+  if (job_slots > _POSIX_PIPE_BUF)
+#else
+  if (job_slots > 512)
+#endif
+    {
+      O (error, NILF,
+            _("More parallel jobs (-jN) than this platform can handle requested."));
+      O (error, NILF, _("Resetting to single job (-j1) mode."));
+      job_slots = 1;
+    }
+
 #ifdef MAKE_JOBSERVER
   /* If we have >1 slot but no jobserver-fds, then we're a top-level make.
      Set up the pipe and install the fds option for our children.  */
