From 5467b24cddaa4c5d108f98b8266fe787d778c9d0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Sat, 24 Oct 2015 18:19:30 -0400
Subject: [PATCH] newlines

---
 job.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git job.c job.c
index 2989249063..7e5bae2a56 100644
--- job.c
+++ job.c
@@ -3320,13 +3320,14 @@ construct_command_argv_internal (char *line, char **restp, const char *shell,
 #endif
             if (PRESERVE_BSNL)
               {
-                *(ap++) = '\\';
+                *(ap++) = '\'';
                 /* Only non-batch execution needs another backslash,
                    because it will be passed through a recursive
                    invocation of this function.  */
                 if (!batch_mode_shell)
                   *(ap++) = '\\';
                 *(ap++) = '\n';
+                *(ap++) = '\'';
               }
             ++p;
             continue;
-- 
2.5.0

