diff -Nrup a/job.c b/job.c
--- a/job.c	2014-02-03 18:23:45.936436714 -0500
+++ b/job.c	2014-02-04 00:17:53.232074893 -0500
@@ -3269,13 +3269,14 @@ construct_command_argv_internal (char *l
 #endif
             if (PRESERVE_BSNL)
               {
-                *(ap++) = '\\';
+                *(ap++) = '\'';
                 /* Only non-batch execution needs another backslash,
                    because it will be passed through a recursive
                    invocation of this function.  */
                 if (!batch_mode_shell)
                   *(ap++) = '\\';
                 *(ap++) = '\n';
+                *(ap++) = '\'';
               }
             ++p;
             continue;
