commit 8c5332984e32d27d28f9a440947b070af0d14c45
Author: Panu Matilainen <pmatilai@redhat.com>
Date:   Mon Mar 15 09:45:49 2010 +0200

    Tell python about RPMTRANS_FLAG_NOCONTEXTS

diff --git a/python/rpmmodule.c b/python/rpmmodule.c
index 4ace4bc..6159aee 100644
--- a/python/rpmmodule.c
+++ b/python/rpmmodule.c
@@ -386,6 +386,7 @@ static int initModule(PyObject *m)
     REGISTER_ENUM(RPMTRANS_FLAG_NODOCS);
     REGISTER_ENUM(RPMTRANS_FLAG_ALLFILES);
     REGISTER_ENUM(RPMTRANS_FLAG_KEEPOBSOLETE);
+    REGISTER_ENUM(RPMTRANS_FLAG_NOCONTEXTS);
     REGISTER_ENUM(RPMTRANS_FLAG_REPACKAGE);
     REGISTER_ENUM(RPMTRANS_FLAG_REVERSE);
     REGISTER_ENUM(RPMTRANS_FLAG_NOPRE);
