diff -up rpm-4.8.0/scripts/pkgconfigdeps.sh.pkgconfig-private rpm-4.8.0/scripts/pkgconfigdeps.sh
--- rpm-4.8.0/scripts/pkgconfigdeps.sh.pkgconfig-private	2010-05-27 15:24:48.000000000 +0300
+++ rpm-4.8.0/scripts/pkgconfigdeps.sh	2010-05-27 15:25:01.000000000 +0300
@@ -39,7 +39,7 @@ case $1 in
 	[ $i -eq 1 ] && echo "$pkgconfig"
 	DIR="`dirname ${filename}`"
 	export PKG_CONFIG_PATH="$DIR:$DIR/../../share/pkgconfig"
-	$pkgconfig --print-requires "$filename" 2> /dev/null | while read n r v ; do
+	$pkgconfig --print-requires --print-requires-private "$filename" 2> /dev/null | while read n r v ; do
 	    [ -n "$n" ] || continue
 	    echo -n "pkgconfig($n) "
 	    [ -n "$r" ] && [ -n "$v" ] && echo -n "$r" "$v"
