diff -up rpm-4.9.0/lib/manifest.c.manifest-fix rpm-4.9.0/lib/manifest.c
--- rpm-4.9.0/lib/manifest.c.manifest-fix	2010-12-03 13:11:57.000000000 +0100
+++ rpm-4.9.0/lib/manifest.c	2011-03-16 14:54:49.502557299 +0100
@@ -96,6 +96,7 @@ rpmRC rpmReadPackageManifest(FD_t fd, in
 
 	/* Sanity checks: skip obviously binary lines and dash (for stdin) */
 	if (*s < 32 || rstreq(s, "-")) {
+	    s = NULL;
 	    rpmrc = RPMRC_NOTFOUND;
 	    goto exit;
 	}
