diff -up rpm-4.11.2-rc1/fileattrs/kmod.attr.kmod-provides rpm-4.11.2-rc1/fileattrs/kmod.attr
--- rpm-4.11.2-rc1/fileattrs/kmod.attr.kmod-provides	2014-01-20 11:47:48.325409810 +0200
+++ rpm-4.11.2-rc1/fileattrs/kmod.attr	2014-01-20 11:47:48.325409810 +0200
@@ -0,0 +1,2 @@
+%__kmod_provides       %{_rpmconfigdir}/kmod.prov
+%__kmod_path           ^/lib/modules/.*$
diff -up rpm-4.11.2-rc1/fileattrs/Makefile.am.kmod-provides rpm-4.11.2-rc1/fileattrs/Makefile.am
--- rpm-4.11.2-rc1/fileattrs/Makefile.am.kmod-provides	2014-01-20 11:47:48.325409810 +0200
+++ rpm-4.11.2-rc1/fileattrs/Makefile.am	2014-01-20 11:49:31.764544198 +0200
@@ -7,6 +7,6 @@ fattrsdir = $(rpmconfigdir)/fileattrs
 fattrs_DATA = \
 	appdata.attr desktop.attr elf.attr font.attr libtool.attr perl.attr \
 	perllib.attr pkgconfig.attr python.attr ocaml.attr script.attr \
-	mono.attr
+	mono.attr kmod.attr
 
 EXTRA_DIST = $(fattrs_DATA)
diff -up rpm-4.11.2-rc1/scripts/kmod.prov.kmod-provides rpm-4.11.2-rc1/scripts/kmod.prov
--- rpm-4.11.2-rc1/scripts/kmod.prov.kmod-provides	2014-01-20 11:47:48.325409810 +0200
+++ rpm-4.11.2-rc1/scripts/kmod.prov	2014-01-20 11:47:48.325409810 +0200
@@ -0,0 +1,17 @@
+#!/bin/sh +x
+
+IFS=$'\n'
+
+for i in $(grep -E '(/lib/modules/.*\.ko|/lib/modules/.*/modules.builtin)');
+do
+	kmod=$(basename $i);
+
+	if [ $kmod == "modules.builtin" ]; then
+		for j in $(cat $i); do
+			j=$(basename $j);
+			echo "kmod($j)"
+		done
+	else
+		echo "kmod($kmod)"
+	fi
+done
diff -up rpm-4.11.2-rc1/scripts/Makefile.am.kmod-provides rpm-4.11.2-rc1/scripts/Makefile.am
--- rpm-4.11.2-rc1/scripts/Makefile.am.kmod-provides	2014-01-17 13:17:29.000000000 +0200
+++ rpm-4.11.2-rc1/scripts/Makefile.am	2014-01-20 11:50:24.065601168 +0200
@@ -20,7 +20,7 @@ EXTRA_DIST = \
 	ocaml-find-requires.sh ocaml-find-provides.sh \
 	pkgconfigdeps.sh libtooldeps.sh appdata.prov \
 	fontconfig.prov desktop-file.prov script.req \
-	macros.perl macros.php macros.python
+	macros.perl macros.php macros.python kmod.prov
 
 rpmconfig_SCRIPTS = \
 	brp-compress brp-python-bytecompile brp-java-gcjcompile \
@@ -34,7 +34,7 @@ rpmconfig_SCRIPTS = \
 	pkgconfigdeps.sh libtooldeps.sh \
 	ocaml-find-requires.sh ocaml-find-provides.sh \
 	fontconfig.prov desktop-file.prov script.req \
-	rpmdb_loadcvt rpm2cpio.sh tcl.req tgpg 
+	rpmdb_loadcvt rpm2cpio.sh tcl.req tgpg kmod.prov
 
 rpmconfig_DATA = \
 	rpm.daily rpm.log rpm.supp \
