--- lib/backend/db3.c.db51	2010-11-02 17:52:07.714234930 -0200
+++ lib/backend/db3.c	2010-11-02 17:52:13.187925237 -0200
@@ -67,7 +67,7 @@
     return 0;
 }
 
-#if (DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 5)
+#if (DB_VERSION_MAJOR >= 5 && DB_VERSION_MINOR >= 0)
 /*
  * dbenv->failchk() callback method for determining is the given pid/tid 
  * is alive. We only care about pid's though. 
@@ -130,7 +130,7 @@
     dbenv->set_errfile(dbenv, rpmdb->db_errfile);
     dbenv->set_errpfx(dbenv, rpmdb->db_errpfx);
 
-#if (DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 5)
+#if (DB_VERSION_MAJOR >= 5 && DB_VERSION_MINOR >= 0)
     /* 
      * These enable automatic stale lock removal. 
      * thread_count 8 is some kind of "magic minimum" value...
@@ -139,7 +139,7 @@
     dbenv->set_isalive(dbenv, db3isalive);
 #endif
 
-#if !(DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 3)
+#if !(DB_VERSION_MAJOR == 5 && DB_VERSION_MINOR >= 0)
     dbenv->set_verbose(dbenv, DB_VERB_CHKPOINT,
 	    (dbi->dbi_verbose & DB_VERB_CHKPOINT));
 #endif
@@ -183,7 +183,7 @@
     if (rc)
 	goto errxit;
 
-#if (DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 5)
+#if (DB_VERSION_MAJOR >= 5 && DB_VERSION_MINOR >= 0)
     /* stale lock removal */
     rc = dbenv->failchk(dbenv, 0);
     rc = cvtdberr(dbi, "dbenv->failchk", rc, _debug);
@@ -362,7 +362,7 @@
 static int db3stat(dbiIndex dbi, unsigned int flags)
 {
     DB * db = dbi->dbi_db;
-#if (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 3)
+#if (DB_VERSION_MAJOR == 5 && DB_VERSION_MINOR >= 0)
     DB_TXN * txnid = NULL;
 #endif
     int rc = 0;
@@ -375,7 +375,7 @@
 #endif
 	flags = 0;
     dbi->dbi_stats = _free(dbi->dbi_stats);
-#if (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 3)
+#if (DB_VERSION_MAJOR == 5 && DB_VERSION_MINOR >= 0)
     rc = db->stat(db, txnid, &dbi->dbi_stats, flags);
 #else
     rc = db->stat(db, &dbi->dbi_stats, flags);
@@ -424,7 +424,7 @@
 	dbenv->set_errcall(dbenv, (void *) rpmdb->db_errcall);
 	dbenv->set_errfile(dbenv, rpmdb->db_errfile);
 	dbenv->set_errpfx(dbenv, rpmdb->db_errpfx);
-#if !(DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 3)
+#if !(DB_VERSION_MAJOR == 5 && DB_VERSION_MINOR >= 0)
 	xx = dbenv->set_verbose(dbenv, DB_VERB_CHKPOINT,
 		(dbi->dbi_verbose & DB_VERB_CHKPOINT));
 #endif
