From f6771b6722f0df097f9c61fc1b487f6f0ee402e8 Mon Sep 17 00:00:00 2001
From: Florian Festi <ffesti@redhat.com>
Date: Tue, 30 Jul 2013 16:35:21 +0200
Subject: [PATCH] Do not filter ld64.* and ld64-* provides and requires Fixes
 #988373

---
 tools/elfdeps.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/elfdeps.c b/tools/elfdeps.c
index 906de10..8679f89 100644
--- a/tools/elfdeps.c
+++ b/tools/elfdeps.c
@@ -52,7 +52,8 @@ static int skipSoname(const char *soname)
 	if (!strstr(soname, ".so"))
 	    return 1;
 
-	if (rstreqn(soname, "ld.", 3) || rstreqn(soname, "ld-", 3))
+	if (rstreqn(soname, "ld.", 3) || rstreqn(soname, "ld-", 3) ||
+	    rstreqn(soname, "ld64.", 3) || rstreqn(soname, "ld64-", 3))
 	    return 0;
 
 	if (rstreqn(soname, "lib", 3))
-- 
1.7.11.7

