diff -up cinepaint-0.22-1/plug-ins/pdf/pdf.cpp.enoy cinepaint-0.22-1/plug-ins/pdf/pdf.cpp
--- cinepaint-0.22-1/plug-ins/pdf/pdf.cpp.enoy	2010-08-27 17:12:34.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/pdf/pdf.cpp	2010-08-27 17:13:38.000000000 +0200
@@ -377,7 +377,9 @@ load_image(const char* filename)
     // set a default profile for cmyk
 #ifdef OYRANOS_H
 # if OYRANOS_API > 12
+#  if OYRANOS_API < 16
     using namespace oyranos;
+#endif
 
     if( vals.colourspace == PDF_CMYK )
       profile_name = oyGetDefaultProfileName (oyEDITING_CMYK, myAllocFunc);
