diff -ur cinepaint-0.25.0/plug-ins/pdf/pdf.cpp ./plug-ins/pdf/pdf.cpp
--- cinepaint-0.25.0/plug-ins/pdf/pdf.cpp	2008-10-10 07:33:07.000000000 +0200
+++ ./plug-ins/pdf/pdf.cpp	2008-12-30 13:12:27.000000000 +0100
@@ -47,8 +47,16 @@
 #include "lib/wire/libtile.h"
 #include "plugin_pdb.h"
 #include "libgimp/stdplugins-intl.h"
-#ifdef HAVE_OY
+
+#if HAVE_OY
+# if OYRANOS_NVERSION > 107
+#include <oyranos.h>
+# else
 #include <oyranos/oyranos.h>
+# endif
+# ifndef OYRANOS_VERSION
+# define OYRANOS_VERSION 0
+# endif
 #endif
 }
 
diff -ur cinepaint-0.25.0/plug-ins/tiff/tiff.c ./plug-ins/tiff/tiff.c
--- cinepaint-0.25.0/plug-ins/tiff/tiff.c	2007-05-11 14:58:30.000000000 +0200
+++ ./plug-ins/tiff/tiff.c	2008-12-30 13:12:16.000000000 +0100
@@ -168,8 +168,15 @@
 #include "info.h"
 #include "gui.h"
 
-#ifdef HAVE_OY
+#if HAVE_OY
+# if OYRANOS_NVERSION > 107
+#include <oyranos.h>
+# else
 #include <oyranos/oyranos.h>
+# endif
+# ifndef OYRANOS_VERSION
+# define OYRANOS_VERSION 0
+# endif
 #endif
 
 /*** struct definitions ***/
