diff -up cinepaint-0.22-1/app/Makefile.am.dso cinepaint-0.22-1/app/Makefile.am
--- cinepaint-0.22-1/app/Makefile.am.dso	2010-06-23 11:58:30.000000000 +0200
+++ cinepaint-0.22-1/app/Makefile.am	2010-06-23 12:00:54.000000000 +0200
@@ -418,12 +418,14 @@ cinepaint_LDADD = \
 	$(GTK_LIBS) \
 	$(X_LIBS) \
 	$(OYRANOS_LIBS) \
-	$(LCMS_LIB)
+	$(LCMS_LIB) \
+	-lX11 -lm
 
 cinepaint_remote_LDADD = \
 	$(GTK_LIBS) \
 	$(X_LIBS) \
-	$(XMU_LIB)
+	$(XMU_LIB) \
+	-lX11
 
 ## DEPS = $(top_builddir)/lib/libcinepainti.a
 
diff -up cinepaint-0.22-1/app/Makefile.in.dso cinepaint-0.22-1/app/Makefile.in
--- cinepaint-0.22-1/app/Makefile.in.dso	2010-06-23 11:58:31.000000000 +0200
+++ cinepaint-0.22-1/app/Makefile.in	2010-06-23 12:00:47.000000000 +0200
@@ -742,12 +742,14 @@ cinepaint_LDADD = \
 	$(GTK_LIBS) \
 	$(X_LIBS) \
 	$(OYRANOS_LIBS) \
-	$(LCMS_LIB)
+	$(LCMS_LIB) \
+	-lX11 -lm
 
 cinepaint_remote_LDADD = \
 	$(GTK_LIBS) \
 	$(X_LIBS) \
-	$(XMU_LIB)
+	$(XMU_LIB) \
+	-lX11
 
 all: all-recursive
 
diff -up cinepaint-0.22-1/plug-ins/cineon/Makefile.am.dso cinepaint-0.22-1/plug-ins/cineon/Makefile.am
--- cinepaint-0.22-1/plug-ins/cineon/Makefile.am.dso	2010-06-23 11:47:47.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/cineon/Makefile.am	2010-06-23 11:47:52.000000000 +0200
@@ -27,7 +27,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/cineon/Makefile.in.dso cinepaint-0.22-1/plug-ins/cineon/Makefile.in
--- cinepaint-0.22-1/plug-ins/cineon/Makefile.in.dso	2007-06-12 07:04:36.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/cineon/Makefile.in	2010-06-23 11:48:24.000000000 +0200
@@ -292,7 +292,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 cineon_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.am.dso cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.am
--- cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.am.dso	2010-06-23 11:49:30.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.am	2010-06-23 11:49:39.000000000 +0200
@@ -16,7 +16,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.in.dso cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.in
--- cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.in.dso	2010-06-23 11:49:34.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/gauss_rle/Makefile.in	2010-06-23 11:49:47.000000000 +0200
@@ -281,7 +281,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 gauss_rle_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/mblur/Makefile.am.dso cinepaint-0.22-1/plug-ins/mblur/Makefile.am
--- cinepaint-0.22-1/plug-ins/mblur/Makefile.am.dso	2010-06-23 11:51:06.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/mblur/Makefile.am	2010-06-23 11:51:16.000000000 +0200
@@ -16,7 +16,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/mblur/Makefile.in.dso cinepaint-0.22-1/plug-ins/mblur/Makefile.in
--- cinepaint-0.22-1/plug-ins/mblur/Makefile.in.dso	2010-06-23 11:51:11.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/mblur/Makefile.in	2010-06-23 11:51:24.000000000 +0200
@@ -281,7 +281,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 mblur_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/psd/Makefile.am.dso cinepaint-0.22-1/plug-ins/psd/Makefile.am
--- cinepaint-0.22-1/plug-ins/psd/Makefile.am.dso	2010-06-23 11:52:05.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/psd/Makefile.am	2010-06-23 11:52:16.000000000 +0200
@@ -15,7 +15,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
-	-lc
+	-lc -lm
 
 DEPS = \
 	$(top_builddir)/lib/libcinepaint.la
diff -up cinepaint-0.22-1/plug-ins/psd/Makefile.in.dso cinepaint-0.22-1/plug-ins/psd/Makefile.in
--- cinepaint-0.22-1/plug-ins/psd/Makefile.in.dso	2010-06-23 11:52:11.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/psd/Makefile.in	2010-06-23 11:52:21.000000000 +0200
@@ -280,7 +280,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
-	-lc
+	-lc -lm
 
 DEPS = \
 	$(top_builddir)/lib/libcinepaint.la
diff -up cinepaint-0.22-1/plug-ins/retinex/Makefile.am.dso cinepaint-0.22-1/plug-ins/retinex/Makefile.am
--- cinepaint-0.22-1/plug-ins/retinex/Makefile.am.dso	2010-06-23 11:53:45.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/retinex/Makefile.am	2010-06-23 11:53:54.000000000 +0200
@@ -18,7 +18,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la \
 	$(X_LIBS) \
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/retinex/Makefile.in.dso cinepaint-0.22-1/plug-ins/retinex/Makefile.in
--- cinepaint-0.22-1/plug-ins/retinex/Makefile.in.dso	2010-06-23 11:53:49.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/retinex/Makefile.in	2010-06-23 11:53:59.000000000 +0200
@@ -283,7 +283,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la \
 	$(X_LIBS) \
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 retinex_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/script-fu/Makefile.am.dso cinepaint-0.22-1/plug-ins/script-fu/Makefile.am
--- cinepaint-0.22-1/plug-ins/script-fu/Makefile.am.dso	2010-06-23 11:53:05.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/script-fu/Makefile.am	2010-06-23 11:53:16.000000000 +0200
@@ -61,7 +61,7 @@ LDADD = \
 	$(top_builddir)/plug-ins/dbbrowser/dbbrowser_utils.o	\
 	$(top_builddir)/lib/libcinepaint.la			\
 	$(X_LIBS)						\
-	-lc
+	-lc -lm
 
 DEPS = \
 	$(top_builddir)/plug-ins/dbbrowser/dbbrowser_utils.o	\
diff -up cinepaint-0.22-1/plug-ins/script-fu/Makefile.in.dso cinepaint-0.22-1/plug-ins/script-fu/Makefile.in
--- cinepaint-0.22-1/plug-ins/script-fu/Makefile.in.dso	2010-06-23 11:53:09.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/script-fu/Makefile.in	2010-06-23 11:53:22.000000000 +0200
@@ -337,7 +337,7 @@ LDADD = \
 	$(top_builddir)/plug-ins/dbbrowser/dbbrowser_utils.o	\
 	$(top_builddir)/lib/libcinepaint.la			\
 	$(X_LIBS)						\
-	-lc
+	-lc -lm
 
 DEPS = \
 	$(top_builddir)/plug-ins/dbbrowser/dbbrowser_utils.o	\
diff -up cinepaint-0.22-1/plug-ins/snoise/Makefile.am.dso cinepaint-0.22-1/plug-ins/snoise/Makefile.am
--- cinepaint-0.22-1/plug-ins/snoise/Makefile.am.dso	2010-06-23 11:54:18.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/snoise/Makefile.am	2010-06-23 11:54:26.000000000 +0200
@@ -15,7 +15,7 @@ INCLUDES = \
 LDADD = $(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = \
 	$(top_builddir)/lib/libcinepaint.la
diff -up cinepaint-0.22-1/plug-ins/snoise/Makefile.in.dso cinepaint-0.22-1/plug-ins/snoise/Makefile.in
--- cinepaint-0.22-1/plug-ins/snoise/Makefile.in.dso	2010-06-23 11:54:21.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/snoise/Makefile.in	2010-06-23 11:54:32.000000000 +0200
@@ -280,7 +280,7 @@ INCLUDES = \
 LDADD = $(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = \
 	$(top_builddir)/lib/libcinepaint.la
diff -up cinepaint-0.22-1/plug-ins/sobel/Makefile.am.dso cinepaint-0.22-1/plug-ins/sobel/Makefile.am
--- cinepaint-0.22-1/plug-ins/sobel/Makefile.am.dso	2010-06-23 11:54:47.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/sobel/Makefile.am	2010-06-23 11:54:57.000000000 +0200
@@ -16,7 +16,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/sobel/Makefile.in.dso cinepaint-0.22-1/plug-ins/sobel/Makefile.in
--- cinepaint-0.22-1/plug-ins/sobel/Makefile.in.dso	2010-06-23 11:54:51.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/sobel/Makefile.in	2010-06-23 11:55:02.000000000 +0200
@@ -281,7 +281,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 sobel_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/spread/Makefile.am.dso cinepaint-0.22-1/plug-ins/spread/Makefile.am
--- cinepaint-0.22-1/plug-ins/spread/Makefile.am.dso	2010-06-23 11:55:17.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/spread/Makefile.am	2010-06-23 11:55:30.000000000 +0200
@@ -16,7 +16,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/spread/Makefile.in.dso cinepaint-0.22-1/plug-ins/spread/Makefile.in
--- cinepaint-0.22-1/plug-ins/spread/Makefile.in.dso	2010-06-23 11:55:24.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/spread/Makefile.in	2010-06-23 11:55:36.000000000 +0200
@@ -281,7 +281,7 @@ LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
 					\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 spread_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/tiff/Makefile.am.dso cinepaint-0.22-1/plug-ins/tiff/Makefile.am
--- cinepaint-0.22-1/plug-ins/tiff/Makefile.am.dso	2010-06-23 11:55:50.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/tiff/Makefile.am	2010-06-23 11:55:57.000000000 +0200
@@ -26,7 +26,7 @@ LDADD = \
 	$(OYRANOS_LIBS) \
 	@LCMS_LIB@                           \
 	@LIBTIFF_LIB@				\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/tiff/Makefile.in.dso cinepaint-0.22-1/plug-ins/tiff/Makefile.in
--- cinepaint-0.22-1/plug-ins/tiff/Makefile.in.dso	2010-06-23 11:55:51.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/tiff/Makefile.in	2010-06-23 11:56:03.000000000 +0200
@@ -292,7 +292,7 @@ LDADD = \
 	$(OYRANOS_LIBS) \
 	@LCMS_LIB@                           \
 	@LIBTIFF_LIB@				\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 tiff_DEPENDENCIES = $(DEPS)
diff -up cinepaint-0.22-1/plug-ins/unsharp/Makefile.am.dso cinepaint-0.22-1/plug-ins/unsharp/Makefile.am
--- cinepaint-0.22-1/plug-ins/unsharp/Makefile.am.dso	2010-06-23 11:56:14.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/unsharp/Makefile.am	2010-06-23 11:56:19.000000000 +0200
@@ -15,7 +15,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 
diff -up cinepaint-0.22-1/plug-ins/unsharp/Makefile.in.dso cinepaint-0.22-1/plug-ins/unsharp/Makefile.in
--- cinepaint-0.22-1/plug-ins/unsharp/Makefile.in.dso	2010-06-23 11:56:15.000000000 +0200
+++ cinepaint-0.22-1/plug-ins/unsharp/Makefile.in	2010-06-23 11:56:25.000000000 +0200
@@ -280,7 +280,7 @@ INCLUDES = \
 LDADD = \
 	$(top_builddir)/lib/libcinepaint.la	\
 	$(X_LIBS)				\
-	-lc
+	-lc -lm
 
 DEPS = $(top_builddir)/lib/libcinepaint.la
 unsharp_DEPENDENCIES = $(DEPS)
