diff -up icc_examin-0.46/icc_examin.cpp.overflow icc_examin-0.46/icc_examin.cpp
--- icc_examin-0.46/icc_examin.cpp.overflow	2010-06-23 12:37:26.000000000 +0200
+++ icc_examin-0.46/icc_examin.cpp	2010-06-23 12:36:58.000000000 +0200
@@ -802,7 +802,7 @@ void
 ICCexamin::setzeFensterTitel()
 {
   DBG_PROG_START
-  char* t = (char*) malloc(256);
+  char* t = (char*) malloc(256 + 4); // 256 chars + "..."
   const char* title = NULL;
   unsigned int t_len = 256;
 
