--- Makefile.in.make	2007-09-02 01:50:47.000000000 -0300
+++ Makefile.in	2010-08-04 02:39:37.698630729 -0300
@@ -1,13 +1,13 @@
 PIDGIN_CFLAGS = @PIDGIN_CFLAGS@
 AUD_CFLAGS = @AUD_CFLAGS@
 DBUS_GLIB_CFLAGS = @DBUS_GLIB_CFLAGS@
-CFLAGS = -fPIC -shared -Wall $(PIDGIN_CFLAGS) $(AUD_CFLAGS) $(DBUS_GLIB_CFLAGS)
+CFLAGS += -fPIC -shared -Wall $(PIDGIN_CFLAGS) $(AUD_CFLAGS) $(DBUS_GLIB_CFLAGS)
 
 AUD_LIBS = @AUD_LIBS@
 AUD_LIB_DIR = @AUD_LIB_DIR@
 DBUS_GLIB_LIBS = @DBUS_GLIB_LIBS@
 DBUS_GLIB_LIB_DIR = @DBUS_GLIB_LIB_DIR@
-LDFLAGS = $(AUD_LIBS) $(DBUS_GLIB_LIBS)
+LIBS = $(AUD_LIBS) $(DBUS_GLIB_LIBS)
 
 PIDGIN_LIB_DIR = @PIDGIN_LIB_DIR@
 
@@ -15,11 +15,12 @@
 default: pidgin-audacious.so
 
 pidgin-audacious.so: pidgin-audacious.c
-	gcc -o $@ $< $(CFLAGS) $(LDFLAGS) -g -Wl,-rpath=$(AUD_LIB_DIR)
+	gcc -o $@ $< $(CFLAGS) $(LDFLAGS) $(LIBS)
 
 
 install: pidgin-audacious.so
-	install -m 755 pidgin-audacious.so $(PIDGIN_LIB_DIR)/pidgin
+	mkdir -p $(DESTDIR)/$(PIDGIN_LIB_DIR)/pidgin
+	install -m 755 pidgin-audacious.so $(DESTDIR)/$(PIDGIN_LIB_DIR)/pidgin
 
 
 clean:
