diff -ur kdelibs-3.5.10/kdeprint/cups/cupsdconf2/cups-util.c kdelibs-3.5.10-cups-util-missing-header/kdeprint/cups/cupsdconf2/cups-util.c
--- kdelibs-3.5.10/kdeprint/cups/cupsdconf2/cups-util.c	2007-10-08 11:52:10.000000000 +0200
+++ kdelibs-3.5.10-cups-util-missing-header/kdeprint/cups/cupsdconf2/cups-util.c	2011-06-17 18:18:56.000000000 +0200
@@ -2,6 +2,7 @@
 #include <cups/http.h>
 #include <cups/cups.h>
 #include <stdlib.h>
+#include <stdio.h>
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <unistd.h>
