diff -ur kdelibs-3.5.10/admin/cvs.sh kdelibs-3.5.10-automake-version/admin/cvs.sh
--- kdelibs-3.5.10/admin/cvs.sh	2008-08-19 22:28:39.000000000 +0200
+++ kdelibs-3.5.10-automake-version/admin/cvs.sh	2012-06-12 01:24:13.000000000 +0200
@@ -63,12 +63,7 @@
 AUTOMAKE_STRING=`$AUTOMAKE --version | head -n 1`
 required_automake_version="1.6.1 or newer"
 case $AUTOMAKE_STRING in
-  automake*1.5d* | automake*1.5* | automake*1.5-* )
-    echo "*** YOU'RE USING $AUTOMAKE_STRING."
-    echo "*** KDE requires automake $required_automake_version"
-    exit 1
-    ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
