--- branches/KDE/3.5/kdelibs/kdeui/kmenubar.cpp	2005/09/29 09:47:40	465272
+++ branches/KDE/3.5/kdelibs/kdeui/kmenubar.cpp	2009/08/30 08:02:08	1017229
@@ -308,7 +308,7 @@
         if( d->fallback_mode )
         {
             d->fallback_mode = false;
-//            KWin::setStrut( winId(), 0, 0, 0, 0 ); KWin will set strut as it will see fit
+            KWin::setStrut( winId(), 0, 0, 0, 0 );
             setMinimumSize( 0, 0 );
             setMaximumSize( QWIDGETSIZE_MAX, QWIDGETSIZE_MAX );
             updateMenuBarSize();
