diff -up kdelibs-3.5.10/kate/part/katehighlight.cpp.gcc_ftbfs kdelibs-3.5.10/kate/part/katehighlight.cpp
--- kdelibs-3.5.10/kate/part/katehighlight.cpp.gcc_ftbfs	2007-05-14 02:52:27.000000000 -0500
+++ kdelibs-3.5.10/kate/part/katehighlight.cpp	2010-11-18 11:12:45.453403959 -0600
@@ -3225,7 +3225,7 @@ void KateHlManager::getDefaults(uint sch
   list.append(charAttribute);
 
   KateAttribute* string = new KateAttribute();
-  string->setTextColor(QColor::QColor("#D00"));
+  string->setTextColor(QColor("#D00"));
   string->setSelectedTextColor(Qt::red);
   list.append(string);
 
@@ -3242,9 +3242,9 @@ void KateHlManager::getDefaults(uint sch
 
   KateAttribute* alert = new KateAttribute();
   alert->setTextColor(Qt::black);
-  alert->setSelectedTextColor( QColor::QColor("#FCC") );
+  alert->setSelectedTextColor( QColor("#FCC") );
   alert->setBold(true);
-  alert->setBGColor( QColor::QColor("#FCC") );
+  alert->setBGColor( QColor("#FCC") );
   list.append(alert);
 
   KateAttribute* functionAttribute = new KateAttribute();
