diff -up kdelibs-3.5.10/kdecore/kde-config.cpp.in.kde-config_kde-version kdelibs-3.5.10/kdecore/kde-config.cpp.in
--- kdelibs-3.5.10/kdecore/kde-config.cpp.in.kde-config_kde-version	2005-09-10 03:27:12.000000000 -0500
+++ kdelibs-3.5.10/kdecore/kde-config.cpp.in	2010-01-29 10:30:16.846715263 -0600
@@ -20,7 +20,7 @@ static KCmdLineOptions options[] =
     { "exec-prefix", I18N_NOOP("Compiled in exec_prefix for KDE libraries"), 0 },
     { "libsuffix", I18N_NOOP("Compiled in library path suffix"), 0 },
     { "localprefix", I18N_NOOP("Prefix in $HOME used to write files"), 0},
-    { "version",   I18N_NOOP("Compiled in version string for KDE libraries"), 0 },
+    { "kde-version",   I18N_NOOP("Compiled in version string for KDE libraries"), 0 },
     { "types",     I18N_NOOP("Available KDE resource types"), 0 },
     { "path type", I18N_NOOP("Search path for resource type"), 0 },
     { "userpath type", I18N_NOOP("User path: desktop|autostart|trash|document"), 0 },
@@ -154,7 +154,7 @@ int main(int argc, char **argv)
         return 0;
     }
 
-    if (args->isSet("version"))
+    if (args->isSet("kde-version"))
     {
         printf("%s\n", KDE_VERSION_STRING);
         return 0;
