--- slip/util/files.py.orig	2010-10-11 07:17:26.000000000 -0300
+++ slip/util/files.py	2011-05-28 00:14:25.268875041 -0300
@@ -24,7 +24,6 @@ __all__ = ["issamefile", "linkfile", "co
            "overwrite_safely"]
 
 import os
-import selinux
 import shutil
 import tempfile
 import errno
@@ -135,9 +134,6 @@ def copyfile(srcpath, dstpath, copy_mode
 
     os.rename(dsttmpfile.name, dstpath)
 
-    if run_restorecon and selinux.is_selinux_enabled() > 0:
-        selinux.restorecon(dstpath)
-
 
 def linkorcopyfile(srcpath, dstpath, copy_mode_from_dst=True,
     run_restorecon=True):
@@ -164,7 +160,7 @@ def linkorcopyfile(srcpath, dstpath, cop
 def overwrite_safely(path, content, preserve_mode=True, preserve_context=True):
     """Safely overwrite a file by creating a temporary file in the same
     directory, writing it, moving it over the original file, eventually
-    preserving file mode and SELinux context."""
+    preserving file mode."""
 
     path = os.path.realpath(path)
     dir_ = os.path.dirname(path)
@@ -176,9 +172,6 @@ def overwrite_safely(path, content, pres
 
     exists = os.path.exists(path)
 
-    if preserve_context and selinux.is_selinux_enabled() <= 0:
-        preserve_context = False
-
     try:
         fd, tmpname = tempfile.mkstemp(prefix=base + os.path.extsep,
                                        dir=dir_)
@@ -186,11 +179,6 @@ def overwrite_safely(path, content, pres
         if exists and preserve_mode:
             shutil.copymode(path, tmpname)
 
-        if exists and preserve_context:
-            ret, ctx = selinux.getfilecon(path)
-            if ret < 0:
-                raise RuntimeError("getfilecon(%r) failed" % path)
-
         f = os.fdopen(fd, "w")
         fd = None
 
@@ -201,12 +189,6 @@ def overwrite_safely(path, content, pres
 
         os.rename(tmpname, path)
 
-        if preserve_context:
-            if exists:
-                selinux.setfilecon(path, ctx)
-            else:
-                selinux.restorecon(path)
-
     finally:
         if f:
             f.close()
