diff -up man-1.6g/src/man.conf.in.upd man-1.6g/src/man.conf.in
--- man-1.6g/src/man.conf.in.upd	2010-10-25 22:22:24.000000000 -0200
+++ man-1.6g/src/man.conf.in	2011-01-02 03:21:12.397310114 -0200
@@ -138,3 +138,13 @@ MANSECT		@sections@
 .F		@fcat@
 .Y		@unyabba@
 .xz		@unxz@
+#
+# Enable/disable makewhatis database cron updates.
+# If MAKEWHATISDBUPDATES variable is uncommented
+# and set to n or N, cron scripts 
+# /etc/cron.daily/makewhatis.cron
+# /etc/cron.weekly/makewhatis.cron
+# will not update makewhatis database.
+# Otherwise the database will be updated.
+# 
+#MAKEWHATISDBUPDATES	n
diff -up man-1.6g/src/paths.h.in.upd man-1.6g/src/paths.h.in
--- man-1.6g/src/paths.h.in.upd	2005-08-20 20:26:06.000000000 -0300
+++ man-1.6g/src/paths.h.in	2011-01-02 03:21:12.397310114 -0200
@@ -35,5 +35,6 @@ static struct paths {
     { "COMPRESS_EXT", "@compress_ext@" }, /* not a path, just a string variable */
     { "DECOMPRESS",	"@decompress@" },
     { "MANSECT",  "@sections@"},          /* idem */
-    { "MANDEFOPTIONS", ""}                /* idem */
+    { "MANDEFOPTIONS", ""},               /* idem */
+    { "MAKEWHATISDBUPDATES", ""}	  /* option is used in makewhatis.cron{weekly|daily} scripts */
 };
