diff -up libkcddb-4.9.98/cmake/FindMusicBrainz5.cmake.cflags libkcddb-4.9.98/cmake/FindMusicBrainz5.cmake
--- libkcddb-4.9.98/cmake/FindMusicBrainz5.cmake.cflags	2012-12-18 02:40:19.000000000 -0600
+++ libkcddb-4.9.98/cmake/FindMusicBrainz5.cmake	2013-01-20 00:14:44.860102865 -0600
@@ -1,7 +1,7 @@
 # Module to find the musicbrainz-4 library
 #
 # It defines
-#  MUSICBRAINZ5_INCLUDE_DIR - the include dir
+#  MUSICBRAINZ5_INCLUDE_DIRS - the include dirs
 #  MUSICBRAINZ5_LIBRARIES - the required libraries
 #  MUSICBRAINZ5_FOUND - true if both of the above have been found
 
@@ -10,17 +10,23 @@
 # Redistribution and use is allowed according to the terms of the BSD license.
 # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 
-if(MUSICBRAINZ5_INCLUDE_DIR AND MUSICBRAINZ5_LIBRARIES)
+if(MUSICBRAINZ5_INCLUDE_DIRS AND MUSICBRAINZ5_LIBRARIES)
    set(MUSICBRAINZ5_FIND_QUIETLY TRUE)
-endif(MUSICBRAINZ5_INCLUDE_DIR AND MUSICBRAINZ5_LIBRARIES)
+endif(MUSICBRAINZ5_INCLUDE_DIRS AND MUSICBRAINZ5_LIBRARIES)
 
-FIND_PATH(MUSICBRAINZ5_INCLUDE_DIR musicbrainz5/Disc.h)
-
-FIND_LIBRARY( MUSICBRAINZ5_LIBRARIES NAMES musicbrainz5)
+IF (NOT WIN32)
+   # use pkg-config to get the directories and then use these values
+   # in the FIND_PATH() and FIND_LIBRARY() calls
+   find_package(PkgConfig)
+   PKG_SEARCH_MODULE( MUSICBRAINZ5 libmusicbrainz5 )
+ELSE (NOT WIN32)
+  FIND_PATH( MUSICBRAINZ5_INCLUDE_DIRS musicbrainz5/Disc.h )
+  FIND_LIBRARY( MUSICBRAINZ5_LIBRARIES NAMES musicbrainz5 )
+ENDIF (NOT WIN32)
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args( MusicBrainz5 DEFAULT_MSG
-                                   MUSICBRAINZ5_INCLUDE_DIR MUSICBRAINZ5_LIBRARIES)
+                                   MUSICBRAINZ5_INCLUDE_DIRS MUSICBRAINZ5_LIBRARIES)
 
-MARK_AS_ADVANCED(MUSICBRAINZ5_INCLUDE_DIR MUSICBRAINZ5_LIBRARIES)
+MARK_AS_ADVANCED(MUSICBRAINZ5_INCLUDE_DIRS MUSICBRAINZ5_LIBRARIES)
 
diff -up libkcddb-4.9.98/CMakeLists.txt.cflags libkcddb-4.9.98/CMakeLists.txt
--- libkcddb-4.9.98/CMakeLists.txt.cflags	2012-12-18 02:40:19.000000000 -0600
+++ libkcddb-4.9.98/CMakeLists.txt	2013-01-20 00:15:42.811916101 -0600
@@ -13,7 +13,7 @@ macro_log_feature(MUSICBRAINZ5_FOUND "Mu
 
 if(MUSICBRAINZ5_FOUND)
     set(HAVE_MUSICBRAINZ5 1)
-    include_directories(${MUSICBRAINZ5_INCLUDE_DIR})
+    include_directories(${MUSICBRAINZ5_INCLUDE_DIRS})
 
     set(libmusicbrainz_SRCS ${libmusicbrainz_SRCS}
        musicbrainz/musicbrainzlookup.cpp
