diff -up kdeutils-4.6.80/kgpg/kgpg.kcfg.gpg2 kdeutils-4.6.80/kgpg/kgpg.kcfg
--- kdeutils-4.6.80/kgpg/kgpg.kcfg.gpg2	2010-05-16 22:05:42.000000000 +0200
+++ kdeutils-4.6.80/kgpg/kgpg.kcfg	2011-06-14 10:24:46.878450069 +0200
@@ -60,7 +60,7 @@
         </entry>
         <entry name="GpgBinaryPath" type="Path" key="gpg_bin_path">
             <label>The path of the gpg binary used by KGpg.</label>
-            <default>gpg</default>
+            <default>gpg2</default>
         </entry>
         <entry name="Groups" type="String">
             <label>GPG groups</label>
diff -up kdeutils-4.6.80/kgpg/kgpgoptions.cpp.gpg2 kdeutils-4.6.80/kgpg/kgpgoptions.cpp
--- kdeutils-4.6.80/kgpg/kgpgoptions.cpp.gpg2	2011-05-20 22:16:54.000000000 +0200
+++ kdeutils-4.6.80/kgpg/kgpgoptions.cpp	2011-06-14 10:25:45.515218869 +0200
@@ -156,7 +156,7 @@ void kgpgOptions::slotChangeHome()
 				// start GnuPG so that it will create a config file
 				QString gpgbin = m_page4->kcfg_GpgBinaryPath->text();
 				if (!QFile::exists(gpgbin))
-					gpgbin = QLatin1String( "gpg" );
+					gpgbin = QLatin1String( "gpg2" );
 
 				KProcess p;
 				p << gpgbin << QLatin1String( "--homedir" ) << gpgHome << QLatin1String( "--no-tty" ) << QLatin1String( "--list-secret-keys" );
