diff -up kdeutils-4.6.1/kgpg/kgpg.kcfg.gnupg2 kdeutils-4.6.1/kgpg/kgpg.kcfg
--- kdeutils-4.6.1/kgpg/kgpg.kcfg.gnupg2	2011-01-18 15:24:55.000000000 -0600
+++ kdeutils-4.6.1/kgpg/kgpg.kcfg	2011-03-24 11:27:15.186864517 -0500
@@ -60,7 +60,7 @@
         </entry>
         <entry name="GpgBinaryPath" type="Path" key="gpg_bin_path">
             <label>The path of the gpg binary used by KGpg.</label>
-            <default>gpg</default>
+            <default>gpg2</default>
         </entry>
         <entry name="Groups" type="String">
             <label>GPG groups</label>
diff -up kdeutils-4.6.1/kgpg/kgpgoptions.cpp.gnupg2 kdeutils-4.6.1/kgpg/kgpgoptions.cpp
--- kdeutils-4.6.1/kgpg/kgpgoptions.cpp.gnupg2	2011-03-24 11:36:02.627895937 -0500
+++ kdeutils-4.6.1/kgpg/kgpgoptions.cpp	2011-03-24 11:36:08.912361046 -0500
@@ -154,7 +154,7 @@ void kgpgOptions::slotChangeHome()
 				// start GnuPG so that it will create a config file
 				QString gpgbin = m_page4->gpg_bin_path->text();
 				if (!QFile::exists(gpgbin))
-					gpgbin = QLatin1String( "gpg" );
+					gpgbin = QLatin1String( "gpg2" );
 
 				KProcess p;
 				p << gpgbin << QLatin1String( "--homedir" ) << gpgHome << QLatin1String( "--no-tty" ) << QLatin1String( "--list-secret-keys" );
