--- monsterz-0.7.1/Makefile.orig	2007-12-17 22:05:00.000000000 +0000
+++ monsterz-0.7.1/Makefile	2008-01-06 17:28:03.000000000 +0000
@@ -6,6 +6,7 @@
 scoredir = /var/games
 scorefile = $(scoredir)/monsterz
 
+CFLAGS=
 VERSION = 0.7.1
 DIRECTORY = monsterz-$(VERSION)
 
@@ -22,7 +23,7 @@
 all: monsterz
 
 monsterz: monsterz.c
-	$(CC) -Wall monsterz.c -DDATADIR=\"$(pkgdatadir)\" -DSCOREFILE=\"$(scorefile)\" -o monsterz
+	$(CC) $(CFLAGS) -Wall monsterz.c -DDATADIR=\"$(pkgdatadir)\" -DSCOREFILE=\"$(scorefile)\" -o monsterz
 
 bitmap: $(BITMAP)
 
