--- monsterz-0.7.1/monsterz.py.orig	2007-12-17 22:05:00.000000000 +0000
+++ monsterz-0.7.1/monsterz.py	2008-02-26 14:23:34.000000000 +0000
@@ -108,7 +108,7 @@
                 M = int(max(r, g, b))
                 m = int(min(r, g, b))
                 val = (2 * M + r + g + b) / 5
-                p[:] = (val + r) / 2, (val + g) / 2, (val + b) / 2
+                #p[:] = (val + r) / 2, (val + g) / 2, (val + b) / 2
                 if alpha[y][x] >= 250:
                     alpha[y][x] = 255 - (M - m) * 3 / 4
         del pixels
@@ -129,7 +129,7 @@
                 r, g, b = p
                 M = int(max(r, g, b))
                 m = int(min(r, g, b))
-                p[:] = (m + r) / 2, (m + g) / 2, (m + b) / 2
+                #p[:] = (m + r) / 2, (m + g) / 2, (m + b) / 2
                 if alpha[y][x] >= 250:
                     alpha[y][x] = 255 - M * 2 / 3
         del pixels
