diff -up src/cli/report_bdtree.cpp.ldpaths src/cli/report_bdtree.cpp
--- src/cli/report_bdtree.cpp.ldpaths	2015-11-06 23:53:10.523352456 -0200
+++ src/cli/report_bdtree.cpp	2015-11-06 23:56:14.966399690 -0200
@@ -155,8 +155,8 @@ findInLdSoCache (ElfFile::Arch arch, con
 ElfFile
 findInLdDefaultPathes (ElfFile::Arch arch, const std::string& soname)
 {
-  StringVec defdirs32{ "/lib","/usr/lib","/usr/local/lib" } ;
-  StringVec defdirs64{ "/lib64","/usr/lib64","/usr/local/lib64" } ;
+  StringVec defdirs32{ "/lib","/usr/lib","/usr/local/lib","/usr/lib/mysql","/usr/lib/wine","/usr/lib/mesa","/usr/lib/dcmtk","/usr/lib/llvm","/usr/lib/samba","/usr/lib/deprecated" } ;
+  StringVec defdirs64{ "/lib64","/usr/lib64","/usr/local/lib64","/usr/lib64/mysql","/usr/lib64/wine","/usr/lib64/mesa","/usr/lib64/dcmtk","/usr/lib64/hugin","/usr/lib64/llvm","/usr/lib64/samba","/usr/lib64/deprecated" } ;
 
   if (arch == ElfFile::Arch32)
     return findInPaths(arch, soname, defdirs32) ;
diff -up src/sbbdep/ldconf.cpp.ldpaths src/sbbdep/ldconf.cpp
--- src/sbbdep/ldconf.cpp.ldpaths	2015-11-06 23:53:18.049762944 -0200
+++ src/sbbdep/ldconf.cpp	2015-11-06 23:55:20.304943714 -0200
@@ -245,7 +245,13 @@ Ldconf::Ldconf ()
 
   { // this should not be required, but to have a base ..
     const auto lds = {"/lib", "/usr/lib", "/usr/local/lib",
-        "/lib64", "/usr/lib64", "/usr/local/lib64"};
+        "/usr/lib/mysql", "/usr/lib/wine", "/usr/lib/mesa",
+        "/usr/lib/dcmtk", "/usr/lib/hugin", "/usr/lib/llvm",
+        "/usr/lib/samba", "/usr/lib/deprecated",
+        "/lib64", "/usr/lib64", "/usr/local/lib64",
+        "/usr/lib64/mysql", "/usr/lib64/wine", "/usr/lib64/mesa",
+        "/usr/lib64/dcmtk","/usr/lib64/hugin", "/usr/lib64/llvm",
+        "/usr/lib64/samba", "/usr/lib64/deprecated"};
 
     for (auto&& p : lds)
       {
