--- gtest-1.6.0.orig/CMakeLists.txt	2011-04-16 04:49:11.000000000 +0900
+++ gtest-1.6.0/CMakeLists.txt	2012-09-28 14:20:19.000000000 +0900
@@ -68,7 +68,9 @@ link_directories(${gtest_BINARY_DIR}/src
 # are used for other targets, to ensure that gtest can be compiled by a user
 # aggressive about warnings.
 cxx_library(gtest "${cxx_strict}" src/gtest-all.cc)
+set_target_properties(gtest PROPERTIES VERSION 0.0.0 SOVERSION 0)
 cxx_library(gtest_main "${cxx_strict}" src/gtest_main.cc)
+set_target_properties(gtest_main PROPERTIES VERSION 0.0.0 SOVERSION 0)
 target_link_libraries(gtest_main gtest)
 
 ########################################################################
