diff -up SFML-1.6/include/SFML/System/Resource.hpp.gcc SFML-1.6/include/SFML/System/Resource.hpp
--- SFML-1.6/include/SFML/System/Resource.hpp.gcc	2010-01-27 14:53:54.000000000 +0100
+++ SFML-1.6/include/SFML/System/Resource.hpp	2012-01-12 15:20:59.693521927 +0100
@@ -29,6 +29,7 @@
 // Headers
 ////////////////////////////////////////////////////////////
 #include <set>
+#include <cstddef>
 
 
 namespace sf
diff -up SFML-1.6/src/SFML/Window/Linux/Joystick.cpp.gcc SFML-1.6/src/SFML/Window/Linux/Joystick.cpp
--- SFML-1.6/src/SFML/Window/Linux/Joystick.cpp.gcc	2012-01-12 15:22:54.550833303 +0100
+++ SFML-1.6/src/SFML/Window/Linux/Joystick.cpp	2012-01-12 15:22:57.362842023 +0100
@@ -26,6 +26,7 @@
 // Headers
 ////////////////////////////////////////////////////////////
 #include <SFML/Window/Joystick.hpp>
+#include <unistd.h>
 #include <sstream>
 
 
