diff -up src/SFML/Audio/Makefile.sys src/SFML/Audio/Makefile
--- src/SFML/Audio/Makefile.sys	2010-05-20 02:40:32.151612273 -0300
+++ src/SFML/Audio/Makefile	2010-05-20 02:40:32.342364998 -0300
@@ -19,7 +19,7 @@ libsfml-audio-s.a: $(OBJ) $(OBJVORBIS)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJVORBIS)
 
 libsfml-audio.so: $(OBJ) $(OBJVORBIS)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJVORBIS) -lsndfile -lopenal
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB)._SFMLSOVER_ -o $(LIBNAME) $(OBJ) $(OBJVORBIS) -lsndfile -lopenal
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
diff -up src/SFML/Graphics/GraphicsContext.hpp.sys src/SFML/Graphics/GraphicsContext.hpp
--- src/SFML/Graphics/GraphicsContext.hpp.sys	2010-05-20 02:40:32.123616648 -0300
+++ src/SFML/Graphics/GraphicsContext.hpp	2010-05-20 02:40:32.342364998 -0300
@@ -30,7 +30,7 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/Config.hpp>
 #include <SFML/System/NonCopyable.hpp>
-#include <SFML/Graphics/GLEW/glew.h>
+#include <GL/glew.h>
 #include <iostream>
 #include <string>
 
diff -up src/SFML/Graphics/ImageLoader.cpp.sys src/SFML/Graphics/ImageLoader.cpp
--- src/SFML/Graphics/ImageLoader.cpp.sys	2010-05-20 02:40:32.122616708 -0300
+++ src/SFML/Graphics/ImageLoader.cpp	2010-05-20 02:46:58.997767129 -0300
@@ -28,11 +28,11 @@
 #include <SFML/Graphics/ImageLoader.hpp>
 extern "C"
 {
-    #include <SFML/Graphics/libjpeg/jpeglib.h>
-    #include <SFML/Graphics/libjpeg/jerror.h>
+    #include <jpeglib.h>
+    #include <jerror.h>
 }
-#include <SFML/Graphics/libpng/png.h>
-#include <SFML/Graphics/SOIL/SOIL.h>
+#include <png.h>
+#include <SOIL/SOIL.h>
 #include <iostream>
 
 
diff -up src/SFML/Graphics/Makefile.sys src/SFML/Graphics/Makefile
--- src/SFML/Graphics/Makefile.sys	2010-05-20 02:40:32.122616708 -0300
+++ src/SFML/Graphics/Makefile	2010-05-20 02:47:49.905413352 -0300
@@ -5,11 +5,16 @@ SRCPNG  = $(wildcard ./libpng/*.c)
 SRCSOIL = $(wildcard ./SOIL/*.c)
 SRCZLIB = $(wildcard ./zlib/*.c)
 OBJ     = $(SRC:.cpp=.o)
-OBJGLEW = $(SRCGLEW:.c=.o)
-OBJJPEG = $(SRCJPEG:.c=.o)
-OBJPNG  = $(SRCPNG:.c=.o)
-OBJSOIL = $(SRCSOIL:.c=.o)
-OBJZLIB = $(SRCZLIB:.c=.o)
+GLEWINC = `pkg-config --cflags glew`
+GLEWLIBS = `pkg-config --libs glew`
+JPEGINC = 
+JPEGLIBS = -ljpeg
+PNGINC  = `pkg-config --cflags libpng`
+PNGLIBS  = `pkg-config --libs libpng`
+SOILINC = 
+SOILLIBS = -lSOIL
+ZLIBINC = 
+ZLIBLIBS = -lz
 
 ifeq ($(STATIC), yes)
     LIB     = libsfml-graphics-s.a
@@ -23,22 +28,19 @@ endif
 
 all: $(LIB)
 
-libsfml-graphics-s.a: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+libsfml-graphics-s.a: $(OBJ)
+	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(GLEWLIBS) $(JPEGLIBS) $(PNGLIBS) $(SOILLIBS) $(ZLIBLIBS)
 
-libsfml-graphics.so: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB) -lfreetype -lX11 -lGL
+libsfml-graphics.so: $(OBJ)
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB)._SFMLSOVER_ -o $(LIBNAME) $(OBJ) $(GLEWLIBS) $(JPEGLIBS) $(PNGLIBS) $(SOILLIBS) $(ZLIBLIBS) -lfreetype -lX11 -lGL
 
 $(OBJ): %.o: %.cpp
-	$(CPP) -o $@ -c $< $(CFLAGS) -I/usr/include/freetype2
-
-$(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB): %.o: %.c
-	$(CC) -o $@ -c $< $(CFLAGSEXT) -DSTBI_FAILURE_USERMSG
+	$(CPP) -o $@ -c $< $(CFLAGS) $(GLEWINC) $(JPEGINC) $(PNGINC) $(SOILINC) $(ZLIBINC) -I/usr/include/freetype2
 
 .PHONY: clean mrproper
 
 clean:
-	@rm -rf $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
+	@rm -rf $(OBJ)
 
 mrproper: clean
 	@rm -rf $(LIBNAME)
diff -up src/SFML/Network/Makefile.sys src/SFML/Network/Makefile
--- src/SFML/Network/Makefile.sys	2010-05-20 02:40:32.117616783 -0300
+++ src/SFML/Network/Makefile	2010-05-20 02:40:32.343363397 -0300
@@ -17,7 +17,7 @@ libsfml-network-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
 libsfml-network.so: $(OBJ)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ)
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB)._SFMLSOVER_ -o $(LIBNAME) $(OBJ)
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
diff -up src/SFML/System/Makefile.sys src/SFML/System/Makefile
--- src/SFML/System/Makefile.sys	2010-05-20 02:40:32.115371635 -0300
+++ src/SFML/System/Makefile	2010-05-20 02:40:32.343363397 -0300
@@ -17,7 +17,7 @@ libsfml-system-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
 libsfml-system.so: $(OBJ)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) -lpthread
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB)._SFMLSOVER_ -o $(LIBNAME) $(OBJ) -lpthread
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
diff -up src/SFML/Window/Linux/WindowImplX11.cpp.sys src/SFML/Window/Linux/WindowImplX11.cpp
--- src/SFML/Window/Linux/WindowImplX11.cpp.sys	2010-05-20 02:40:32.109371491 -0300
+++ src/SFML/Window/Linux/WindowImplX11.cpp	2010-05-20 02:40:32.344363337 -0300
@@ -27,8 +27,8 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/Window/WindowStyle.hpp> // important to be included first (conflict with None)
 #include <SFML/Window/Linux/WindowImplX11.hpp>
-#include <SFML/Window/glext/glxext.h>
-#include <SFML/Window/glext/glext.h>
+#include <GL/glxext.h>
+#include <GL/glext.h>
 #include <SFML/System/Unicode.hpp>
 #include <X11/keysym.h>
 #include <X11/extensions/Xrandr.h>
diff -up src/SFML/Window/Makefile.sys src/SFML/Window/Makefile
--- src/SFML/Window/Makefile.sys	2010-05-20 02:40:32.106366674 -0300
+++ src/SFML/Window/Makefile	2010-05-20 02:40:32.344363337 -0300
@@ -1,6 +1,7 @@
 SRC     = $(wildcard *.cpp ./Linux/*.cpp)
 OBJ     = $(SRC:.cpp=.o)
-OBJGLEW = $(SRCGLEW:.c=.o)
+GLEWINC = `pkg-config --cflags glew`
+GLEWLIBS = `pkg-config --libs glew`
 
 ifeq ($(STATIC), yes)
     LIB     = libsfml-window-s.a
@@ -18,10 +19,10 @@ libsfml-window-s.a: $(OBJ)
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
 libsfml-window.so: $(OBJ)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) -lX11 -lXrandr -lGL
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB)._SFMLSOVER_ -o $(LIBNAME) $(OBJ) $(GLEWLIBS) -lX11 -lXrandr -lGL
 
 $(OBJ): %.o: %.cpp
-	$(CPP) -o $@ -c $< $(CFLAGS)
+	$(CPP) -o $@ -c $< $(CFLAGS) $(GLEWINC)
 
 .PHONY: clean mrproper
 
