diff -up libdbusmenu-qt-0.3.2/dbusmenu-qt.pc.in.pkgconfig libdbusmenu-qt-0.3.2/dbusmenu-qt.pc.in
--- libdbusmenu-qt-0.3.2/dbusmenu-qt.pc.in.pkgconfig	2010-04-02 09:38:25.000000000 -0500
+++ libdbusmenu-qt-0.3.2/dbusmenu-qt.pc.in	2010-04-25 16:50:29.729374725 -0500
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=@CMAKE_INSTALL_PREFIX@
-libdir=@CMAKE_INSTALL_PREFIX@/lib
+libdir=@CMAKE_INSTALL_PREFIX@/lib@LIB_SUFFIX@
 includedir=@CMAKE_INSTALL_PREFIX@/include/dbusmenu-qt
 
 Name: libdbusmenu-qt
