--- src/joystick.cpp.orig	2010-09-04 20:49:12.506574504 -0300
+++ src/joystick.cpp	2010-09-04 20:49:19.810055576 -0300
@@ -45,6 +45,12 @@
 	}
 
 	foreach(SDL_Joystick* j, joysticks) {
+		if(SDL_JoystickGetAxis(j, 1) < -threshold ||
+		   SDL_JoystickGetAxis(j, 3) < -threshold ||
+		   SDL_JoystickGetAxis(j, 5) < -threshold) {
+			return true;
+		}
+
 		const int nhats = SDL_JoystickNumHats(j);
 		for(int n = 0; n != nhats; ++n) {
 			const Uint8 state = SDL_JoystickGetHat(j, n);
@@ -67,6 +73,12 @@
 	}
 
 	foreach(SDL_Joystick* j, joysticks) {
+		if(SDL_JoystickGetAxis(j, 1) > threshold ||
+		   SDL_JoystickGetAxis(j, 3) > threshold ||
+		   SDL_JoystickGetAxis(j, 5) > threshold) {
+			return true;
+		}
+
 		const int nhats = SDL_JoystickNumHats(j);
 		for(int n = 0; n != nhats; ++n) {
 			const Uint8 state = SDL_JoystickGetHat(j, n);
@@ -78,9 +90,7 @@
 			}
 		}
 
-	}
-
-	return false;
+	}	return false;
 }
 
 bool left() {
@@ -89,6 +99,12 @@
 	}
 
 	foreach(SDL_Joystick* j, joysticks) {
+		if(SDL_JoystickGetAxis(j, 0) < -threshold ||
+		   SDL_JoystickGetAxis(j, 2) < -threshold ||
+		   SDL_JoystickGetAxis(j, 4) < -threshold) {
+			return true;
+		}
+
 		const int nhats = SDL_JoystickNumHats(j);
 		for(int n = 0; n != nhats; ++n) {
 			const Uint8 state = SDL_JoystickGetHat(j, n);
@@ -111,6 +127,12 @@
 	}
 
 	foreach(SDL_Joystick* j, joysticks) {
+		if(SDL_JoystickGetAxis(j, 0) > threshold ||
+		   SDL_JoystickGetAxis(j, 2) > threshold ||
+		   SDL_JoystickGetAxis(j, 4) > threshold) {
+			return true;
+		}
+
 		const int nhats = SDL_JoystickNumHats(j);
 		for(int n = 0; n != nhats; ++n) {
 			const Uint8 state = SDL_JoystickGetHat(j, n);
