--- iproute2-2.6.35/ip/iproute.c	2010-08-31 13:01:39.151185561 +0200
+++ iproute2-print-route-u32.c	    2010-08-31 13:41:04.072928779 +0200
@@ -368,7 +368,7 @@
 				    abuf, sizeof(abuf)));
 	}
 	if (tb[RTA_PRIORITY])
-		fprintf(fp, " metric %d ", *(__u32*)RTA_DATA(tb[RTA_PRIORITY]));
+		fprintf(fp, " metric %u ", *(__u32*)RTA_DATA(tb[RTA_PRIORITY]));
 	if (r->rtm_flags & RTNH_F_DEAD)
 		fprintf(fp, "dead ");
 	if (r->rtm_flags & RTNH_F_ONLINK)
@@ -487,7 +487,7 @@
 			if (i < sizeof(mx_names)/sizeof(char*) && mx_names[i])
 				fprintf(fp, " %s", mx_names[i]);
 			else
-				fprintf(fp, " metric %d", i);
+				fprintf(fp, " metric %u", i);
 			if (mxlock & (1<<i))
 				fprintf(fp, " lock");
 
