--- src/main.cpp.orig	2010-11-04 08:01:46.000000000 -0200
+++ src/main.cpp	2011-02-11 23:47:49.739929193 -0200
@@ -341,24 +341,10 @@ void write_error(const char* message, bo
 #ifdef HAVE_LIBNOTIFY
     else if (prog_config.error_mode == Error::libnotify) { 
       GobjHandle<NotifyNotification> notification_h;
-      if (prog_config.tray_icon && main_win_p) {
-	notification_h = GobjHandle<NotifyNotification>(notify_notification_new_with_status_icon(
-                                                          gettext("Mount-gtk: error"),
-							  message,
-							  0,
-							  main_win_p->get_status_icon()));
-      }
-      else {
-	GtkWidget* attach_p;
-	if (main_win_p) attach_p = GTK_WIDGET(main_win_p->get_win());
-	else attach_p = 0;
-			  
 	notification_h = GobjHandle<NotifyNotification>(notify_notification_new(
 							  gettext("Mount-gtk: error"),
 							  message,
-							  0,
-							  attach_p));
-      }
+							  NULL));
       notify_notification_set_icon_from_pixbuf(notification_h, prog_config.window_icon_h);
       GError* error_p = 0;
       if (!notify_notification_show(notification_h, &error_p)) {
