--- src/Makefile_Emile.am.liblz4	2015-07-24 07:55:02.000000000 -0300
+++ src/Makefile_Emile.am	2015-08-04 20:15:28.321826594 -0300
@@ -45,18 +45,13 @@ lib_emile_libemile_la_CPPFLAGS = -I$(top
 @EMILE_CFLAGS@ \
 @NEON_CFLAGS@
 
-if ! ENABLE_LIBLZ4
-lib_emile_libemile_la_CPPFLAGS += \
--I$(top_srcdir)/src/static_libs/lz4
-endif
+lib_emile_libemile_la_CPPFLAGS += @LIBLZ4_CFLAGS@
 
 lib_emile_libemile_la_LIBADD = @EMILE_LIBS@
 lib_emile_libemile_la_DEPENDENCIES = @EMILE_INTERNAL_LIBS@
 lib_emile_libemile_la_LDFLAGS = @EFL_LTLIBRARY_FLAGS@
 
-if ! ENABLE_LIBLZ4
 lib_emile_libemile_la_LIBADD += @LIBLZ4_LIBS@
-endif
 
 EXTRA_DIST += static_libs/lz4/README.md \
 static_libs/lz4/LICENSE \
