diff -up irrlicht18-svn3629/source/Irrlicht/CZipReader.cpp.libaesgm irrlicht18-svn3629/source/Irrlicht/CZipReader.cpp
--- irrlicht18-svn3629/source/Irrlicht/CZipReader.cpp.libaesgm	2012-08-23 13:56:12.824576588 -0500
+++ irrlicht18-svn3629/source/Irrlicht/CZipReader.cpp	2012-08-23 13:56:31.497576866 -0500
@@ -27,7 +27,7 @@ extern "C" void bz_internal_error(int er
 	#endif
 
 	#ifdef _IRR_COMPILE_WITH_ZIP_ENCRYPTION_
-	#include "aesGladman/fileenc.h"
+	#include <aes/fileenc.h>
 	#endif
 	#ifdef _IRR_COMPILE_WITH_BZIP2_
 	#ifndef _IRR_USE_NON_SYSTEM_BZLIB_
diff -up irrlicht18-svn3629/source/Irrlicht/Makefile.libaesgm irrlicht18-svn3629/source/Irrlicht/Makefile
--- irrlicht18-svn3629/source/Irrlicht/Makefile.libaesgm	2012-08-23 13:56:41.010577010 -0500
+++ irrlicht18-svn3629/source/Irrlicht/Makefile	2012-08-23 13:57:08.195577417 -0500
@@ -55,7 +55,7 @@ LINKOBJ = $(IRRMESHOBJ) $(IRROBJ) $(IRRP
 ###############
 #Compiler flags
 CXXINCS = -I../../include `libpng-config --cflags`
-CPPFLAGS += $(CXXINCS) -DIRRLICHT_EXPORTS=1 $(RPM_OPT_FLAGS)
+CPPFLAGS += $(CXXINCS) -DIRRLICHT_EXPORTS=1 -DUSE_SHA1 $(RPM_OPT_FLAGS)
 CXXFLAGS += $(RPM_OPT_FLAGS) -Wall -pipe -fno-exceptions -fno-rtti -fstrict-aliasing
 ifndef NDEBUG
 CXXFLAGS += -g -D_DEBUG
