--- irrlicht-1.7.2/source/Irrlicht/CImageLoaderPNG.cpp.orig	2010-12-21 22:01:03.000000000 -0200
+++ irrlicht-1.7.2/source/Irrlicht/CImageLoaderPNG.cpp	2010-12-21 22:01:35.417928277 -0200
@@ -162,7 +162,11 @@ IImage* CImageLoaderPng::loadImage(io::I
 	if (BitDepth < 8)
 	{
 		if (ColorType==PNG_COLOR_TYPE_GRAY || ColorType==PNG_COLOR_TYPE_GRAY_ALPHA)
-			png_set_gray_1_2_4_to_8(png_ptr);
+			#if PNG_LIBPNG_VER < 10400
+				png_set_gray_1_2_4_to_8(png_ptr);
+			#else
+				png_set_expand_gray_1_2_4_to_8(png_ptr);
+			#endif
 		else
 			png_set_packing(png_ptr);
 	}
