diff -up screen-4.0.3/config.h.in.configh screen-4.0.3/config.h.in
--- screen-4.0.3/config.h.in.configh	2006-10-23 15:06:32.000000000 +0200
+++ screen-4.0.3/config.h.in	2008-05-19 15:47:19.000000000 +0200
@@ -291,7 +291,7 @@
  * Define USE_LOCALE if you want screen to use the locale names
  * for the name of the month and day of the week.
  */
-#define USE_LOCALE
+#undef USE_LOCALE
 
 /*
  * Define USE_PAM if your system supports PAM (Pluggable Authentication
@@ -563,7 +563,6 @@
  * If you are on a SYS V machine that restricts filename length to 14 
  * characters, you may need to enforce that by setting NAME_MAX to 14
  */
-#undef NAME_MAX		/* KEEP_UNDEF_HERE override system value */
 #undef NAME_MAX
 
 /*
