diff -up screen-4.0.3/configure.in.libs screen-4.0.3/configure.in
--- screen-4.0.3/configure.in.libs	2003-06-03 13:58:24.000000000 +0200
+++ screen-4.0.3/configure.in	2008-02-26 13:58:29.000000000 +0100
@@ -196,7 +196,9 @@ AC_EGREP_CPP(yes,
 ], LIBS="$LIBS -lsocket -linet";seqptx=1)
 
 oldlibs="$LIBS"
-LIBS="$LIBS -lelf"
+# Humm we don't actually need to link against libelf for linux
+#LIBS="$LIBS -lelf"
+LIBS="$LIBS"
 AC_CHECKING(SVR4)
 AC_TRY_LINK([#include <utmpx.h>
 ],,
@@ -643,14 +645,10 @@ dnl
 AC_CHECKING(for tgetent)
 AC_TRY_LINK(,tgetent((char *)0, (char *)0);,,
 olibs="$LIBS"
-LIBS="-lcurses $olibs"
-AC_CHECKING(libcurses)
+LIBS="-ltinfo $olibs"
+AC_CHECKING(libtinfo)
 AC_TRY_LINK(,[
-#ifdef __hpux
-__sorry_hpux_libcurses_is_totally_broken_in_10_10();
-#else
 tgetent((char *)0, (char *)0);
-#endif
 ],,
 LIBS="-ltermcap $olibs"
 AC_CHECKING(libtermcap)
@@ -666,7 +664,7 @@ AC_MSG_ERROR(!!! no tgetent - no screen)
 AC_TRY_RUN([
 main()
 {
- exit(strcmp(tgoto("%p1%d", 0, 1), "1") ? 0 : 1);
+	return 1;
 }], AC_NOTE(- you use the termcap database),
 AC_NOTE(- you use the terminfo database) AC_DEFINE(TERMINFO))
 AC_CHECKING(ospeed)
@@ -864,9 +862,6 @@ fi
 dnl
 dnl    ****  loadav  ****
 dnl
-AC_CHECKING(for libutil(s))
-test -f /usr/lib/libutils.a && LIBS="$LIBS -lutils"
-test -f /usr/lib/libutil.a && LIBS="$LIBS -lutil"
 
 AC_CHECKING(getloadavg)
 AC_TRY_LINK(,[getloadavg((double *)0, 0);],
