diff -up patch-2.7.1/src/patch.c.args-segfault patch-2.7.1/src/patch.c
--- patch-2.7.1/src/patch.c.args-segfault	2013-06-12 16:48:50.617108731 +0100
+++ patch-2.7.1/src/patch.c	2013-06-12 16:49:06.790181916 +0100
@@ -150,6 +150,8 @@ main (int argc, char **argv)
     else if ((version_control = getenv ("VERSION_CONTROL")))
       version_control_context = "$VERSION_CONTROL";
 
+    init_files_to_output ();
+
     /* parse switches */
     Argc = argc;
     Argv = argv;
@@ -164,7 +166,6 @@ main (int argc, char **argv)
 
     init_backup_hash_table ();
     init_files_to_delete ();
-    init_files_to_output ();
 
     init_output (&outstate);
     if (outfile)
