diff -ur smc-1.9/configure.ac smc-1.9-fix-implicit-linking/configure.ac
--- smc-1.9/configure.ac	2009-08-17 15:59:00.000000000 +0200
+++ smc-1.9-fix-implicit-linking/configure.ac	2010-05-19 07:16:42.000000000 +0200
@@ -57,6 +57,11 @@
 CPPFLAGS="$CPPFLAGS $CEGUI_CFLAGS"
 LIBS="$LIBS $CEGUI_LIBS"
 
+# Check for the X11 library
+PKG_CHECK_MODULES(X11, x11)
+CPPFLAGS="$CPPFLAGS $X11_CFLAGS"
+LIBS="$LIBS $X11_LIBS"
+
 # Check for the gettext library
 AM_GNU_GETTEXT([external])
 
