--- imlib2-config.in.multilib	2009-07-28 15:35:58.000000000 -0300
+++ imlib2-config.in	2010-06-22 20:51:34.011045371 -0300
@@ -1,8 +1,11 @@
 #!/bin/sh
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
+version=`pkg-config imlib2 --modversion`
+prefix=`pkg-config imlib2 --variable prefix`
+exec_prefix=`pkg-config imlib2 --variable exec_prefix`
 exec_prefix_set=no
+cflags=`pkg-config imlib2 --cflags`
+libs=`pkg-config imlib2 --libs`
 
 usage="\
 Usage: imlib2-config [--prefix[=DIR]] [--exec-prefix[=DIR]] [--version] [--libs] [--cflags]"
@@ -36,17 +39,13 @@
       echo $exec_prefix
       ;;
     --version)
-      echo @VERSION@
+      echo $version
       ;;
     --cflags)
-      if test @prefix@/include != /usr/include ; then
-        includes="-I@prefix@/include"
-      fi
-      echo $includes
+      echo $cflags
       ;;
     --libs)
-      libdirs=-L@libdir@
-      echo $libdirs -lImlib2 @my_libs@
+      echo $libs
       ;;
     *)
       echo "${usage}" 1>&2
