--- zenmap/zenmapGUI/App.py.py27	2010-08-11 13:35:28.420537292 -0300
+++ zenmap/zenmapGUI/App.py	2010-08-11 13:35:57.656742493 -0300
@@ -96,7 +96,7 @@
 # produces a warning, but the lack of a display eventually causes a
 # segmentation fault. See http://live.gnome.org/PyGTK/WhatsNew210.
 import warnings
-warnings.filterwarnings("error", module = "gtk")
+warnings.filterwarnings("error", module = "gtk", append=True)
 try:
     import gtk
 except Exception:
@@ -176,7 +176,7 @@
         # produces a warning, but the lack of a display eventually causes a
         # segmentation fault. See http://live.gnome.org/PyGTK/WhatsNew210.
         import warnings
-        warnings.filterwarnings("error", module = "gtk")
+        warnings.filterwarnings("error", module = "gtk", append=True)
         import gtk
         warnings.resetwarnings()
 
