diff -up nmap-5.50/zenmap/zenmapGUI/App.py.bz637403 nmap-5.50/zenmap/zenmapGUI/App.py
--- nmap-5.50/zenmap/zenmapGUI/App.py.bz637403	2011-01-21 01:04:51.000000000 +0100
+++ nmap-5.50/zenmap/zenmapGUI/App.py	2011-02-07 13:01:36.275314680 +0100
@@ -179,7 +179,7 @@ def install_excepthook():
         # produces a warning, but the lack of a display eventually causes a
         # segmentation fault. See http://live.gnome.org/PyGTK/WhatsNew210.
         import warnings
-        warnings.filterwarnings("error", module = "gtk")
+        warnings.filterwarnings("error", module = "gtk", append=True)
         import gtk
         warnings.resetwarnings()
 
