From 06ddd9b0cca5a340fd65cd80a08544e71ee95ee8 Mon Sep 17 00:00:00 2001
From: Uoti Urpala <uau@mplayer2.org>
Date: Fri, 21 Sep 2012 02:27:03 +0300
Subject: [PATCH 5/9] stream_ftp: fix double free in one error case

---
 stream/stream_ftp.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/stream/stream_ftp.c b/stream/stream_ftp.c
index 00027d5..53b9e94 100644
--- a/stream/stream_ftp.c
+++ b/stream/stream_ftp.c
@@ -417,7 +417,6 @@ static int open_f(stream_t *stream,int mode, void* opts, int* file_format) {
 
   if (readresp(p, NULL) == 0) {
     close_f(stream);
-    m_struct_free(&stream_opts,opts);
     return STREAM_ERROR;
   }
 
-- 
1.7.12.3

