--- etc/example.conf~	2006-06-11 21:35:46.000000000 +0300
+++ etc/example.conf	2006-08-10 01:18:25.000000000 +0300
@@ -3,3 +3,3 @@
 #
-# Configuration files are read system-wide from /usr/local/etc/mplayer.conf
+# Configuration files are read system-wide from /etc/mplayer/mplayer.conf
 # and per user from ~/.mplayer/config, where per-user settings override
@@ -120,4 +120,9 @@
 
+# A plugged in joystick can interfere with keyboard shortcuts
+joystick = no
+
+# FriBidi can do "funny" things to some ISO-8859-* subtitles
+flip-hebrew = no
 
 # You can also include other configuration files.
-#include = /path/to/the/file/you/want/to/include
+#include = /home/me/.mplayer/something
