diff -rupN qhexedit2-0.7.2/example/qhexedit.pro qhexedit2-0.7.2-new/example/qhexedit.pro
--- qhexedit2-0.7.2/example/qhexedit.pro	2015-05-19 18:13:44.000000000 +0200
+++ qhexedit2-0.7.2-new/example/qhexedit.pro	2015-05-20 18:50:14.385852832 +0200
@@ -3,9 +3,6 @@ greaterThan(QT_MAJOR_VERSION, 4): QT +=
 HEADERS = \
     mainwindow.h \
     optionsdialog.h \
-    ../src/qhexedit.h \
-    ../src/chunks.h \
-    ../src/commands.h \
     searchdialog.h
 
 
@@ -13,11 +10,10 @@ SOURCES = \
     main.cpp \
     mainwindow.cpp \
     optionsdialog.cpp \
-    ../src/qhexedit.cpp \
-    ../src/chunks.cpp \
-    ../src/commands.cpp \
     searchdialog.cpp
 
+LIBS = -L../build-lib-qt5 -lqhexedit-qt5
+
 RESOURCES = \
     qhexedit.qrc
 
diff -rupN qhexedit2-0.7.2/sip/configure.py qhexedit2-0.7.2-new/sip/configure.py
--- qhexedit2-0.7.2/sip/configure.py	2015-05-19 18:13:44.000000000 +0200
+++ qhexedit2-0.7.2-new/sip/configure.py	2015-05-20 18:50:14.386852851 +0200
@@ -38,7 +38,7 @@ makefile = pyqtconfig.QtGuiModuleMakefil
 # Add the library we are wrapping.  The name doesn't include any platform
 # specific prefixes or extensions (e.g. the "lib" prefix on UNIX, or the
 # ".dll" extension on Windows).
-makefile.LFLAGS.append("-L../c++")
+makefile.LFLAGS.append("-L../build-lib-qt4")
 makefile.extra_libs = ["qhexedit"]
 
 # Generate the Makefile itself.
diff -rupN qhexedit2-0.7.2/sip/qhexedit.pro qhexedit2-0.7.2-new/sip/qhexedit.pro
--- qhexedit2-0.7.2/sip/qhexedit.pro	2015-05-19 18:13:44.000000000 +0200
+++ qhexedit2-0.7.2-new/sip/qhexedit.pro	2015-05-20 18:50:14.386852851 +0200
@@ -1,4 +1,5 @@
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
+equals(QT_MAJOR_VERSION, 5): QTSUFFIX = -qt5
 
 TEMPLATE    = lib
 
@@ -17,10 +18,4 @@ SOURCES     = \
     ../src/chunks.cpp \
     ../src/commands.cpp
 
-TARGET      = qhexedit
-
-DESTDIR     = /usr/lib
-
-win32 {
-DESTDIR     = D:\\Programme\\Python27\\Lib\\site-packages\\PyQt4
-}
+TARGET      = qhexedit$$QTSUFFIX
