diff -r -u lxqt-common-0.10.0.orig/autostart/lxqt-policykit-agent.desktop.in lxqt-common-0.10.0/autostart/lxqt-policykit-agent.desktop.in
--- lxqt-common-0.10.0.orig/autostart/lxqt-policykit-agent.desktop.in	2015-10-08 13:38:12.000000000 -0300
+++ lxqt-common-0.10.0/autostart/lxqt-policykit-agent.desktop.in	2015-11-02 15:56:15.525422740 -0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=PolicyKit Handler
+Comment=PolicyKit Handler
 TryExec=lxqt-policykit-agent
 Exec=lxqt-policykit-agent
 OnlyShowIn=LXQt;
diff -r -u lxqt-common-0.10.0.orig/autostart/lxqt-qlipper-autostart.desktop.in lxqt-common-0.10.0/autostart/lxqt-qlipper-autostart.desktop.in
--- lxqt-common-0.10.0.orig/autostart/lxqt-qlipper-autostart.desktop.in	2015-10-08 13:38:12.000000000 -0300
+++ lxqt-common-0.10.0/autostart/lxqt-qlipper-autostart.desktop.in	2015-11-02 15:58:12.368021720 -0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=Qlipper
+Comment=Qlipper
 TryExec=qlipper
 Exec=qlipper
 OnlyShowIn=LXQt;
