From f45dc4a373acf9ed5fed905df6deb709506d27e2 Mon Sep 17 00:00:00 2001
From: Rex Dieter <rdieter@math.unl.edu>
Date: Mon, 25 Nov 2013 07:54:55 -0600
Subject: [PATCH] fix pkgconfig Version:

---
 src/kscreen.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kscreen.pc.in b/src/kscreen.pc.in
index 6f4bb39..2687d2e 100644
--- a/src/kscreen.pc.in
+++ b/src/kscreen.pc.in
@@ -4,7 +4,7 @@ includedir=@CMAKE_INSTALL_PREFIX@/include
 
 Name: kscreen
 Description: Qt Based library to manage screens with backends (xrandr, whatevercomesnext)
-Version: @VERSION@
+Version: @libkscreen_VERSION@
 
 Libs: -L${libdir} -lkscreen
 Cflags: -I${includedir}
-- 
1.8.4.2

