--- cutter-1.03.orig/cutter.c
+++ cutter-1.03/cutter.c
@@ -57,6 +57,7 @@
 #include <arpa/inet.h>
 #include <net/if.h>
 #include <errno.h>
+#include <time.h>
 
 #define ETHHDR	 sizeof(struct ethhdr)
 #define TCPHDR	 sizeof(struct tcphdr)
@@ -149,7 +150,6 @@
 int getmac(in_addr_t ip, uchar *mac)
 {
 	FILE *id = fopen( "/proc/net/arp", "r" );
-	union { uchar c[4]; in_addr_t n; } ipu;
 	in_addr_t ipn;
 	int mac0, mac1, mac2, mac3, mac4, mac5;
 	int hwtype, flags;
@@ -240,7 +240,7 @@
 	u_short toport
 )
 {
-	int i_result, raw_sock, rtn;
+	int i_result, raw_sock;
 	in_addr_t gateway_ip;
 	struct sockaddr_ll myaddr, hisaddr;
 	struct tpack tpack;
@@ -377,7 +377,7 @@
 
 	for ( ; time(0) < tstart + 15; ) {	// give the peer 15 seconds to respond
 		struct sockaddr_ll gotaddr;
-		int addrlen = sizeof(gotaddr);
+		unsigned int addrlen = sizeof(gotaddr);
 		fd_set readfds;
 		struct timeval tv;
 
@@ -478,8 +478,7 @@
 {
 	FILE *id = fopen( "/proc/net/ip_conntrack", "r" );
 	char src1[32], dst1[32], src2[32], dst2[32];
-	int sport1, dport1, sport2, dport2, i;
-	int packets1, packets2, bytes1, bytes2;
+	int sport1, dport1, sport2, dport2;
 	in_addr_t src1n, src2n, dst1n, dst2n;
 	char buff[1024], *p;
 	int found = 0;
