diff -Nur audacious-plugins-fedora-2.4.4-orig/src/adplug/core/binio_virtual.h audacious-plugins-fedora-2.4.4/src/adplug/core/binio_virtual.h
--- audacious-plugins-fedora-2.4.4-orig/src/adplug/core/binio_virtual.h	2011-02-22 18:29:46.000000000 +0100
+++ audacious-plugins-fedora-2.4.4/src/adplug/core/binio_virtual.h	2011-02-23 12:43:21.523837441 +0100
@@ -22,10 +22,6 @@
 	};
 
 	~vfsistream() {
-		if (this->fd) {
-			vfs_fclose(this->fd);
-			this->fd = 0;
-		}
 	};
 
 	vfsistream(VFSFile *fd) {
