diff -Nur audacious-plugins-fedora-2.5.1-orig/src/vorbis/vcedit.c audacious-plugins-fedora-2.5.1/src/vorbis/vcedit.c
--- audacious-plugins-fedora-2.5.1-orig/src/vorbis/vcedit.c	2011-05-19 20:10:38.000000000 +0200
+++ audacious-plugins-fedora-2.5.1/src/vorbis/vcedit.c	2011-06-08 17:24:16.918706254 +0200
@@ -145,7 +145,7 @@
 {
     int result;
     char *buffer;
-    int bytes;
+    gint64 bytes;
 
     result = ogg_stream_packetout(s->os, p);
 
@@ -189,7 +189,7 @@
                       vcedit_write_func write_func)
 {
     char *buffer;
-    int bytes, i;
+    gint64 bytes, i;
     ogg_packet *header;
     ogg_packet header_main;
     ogg_packet header_comments;
@@ -317,7 +317,7 @@
     ogg_int64_t granpos = 0;
     int result;
     char *buffer;
-    int bytes;
+    gint64 bytes;
     int needflush = 0, needout = 0;
 
     state->eosin = 0;
diff -Nur audacious-plugins-fedora-2.5.1-orig/src/vorbis/vcedit.h audacious-plugins-fedora-2.5.1/src/vorbis/vcedit.h
--- audacious-plugins-fedora-2.5.1-orig/src/vorbis/vcedit.h	2011-05-19 20:10:38.000000000 +0200
+++ audacious-plugins-fedora-2.5.1/src/vorbis/vcedit.h	2011-06-08 17:23:20.655472870 +0200
@@ -19,8 +19,8 @@
 #include <vorbis/codec.h>
 #include <audacious/plugin.h>
 
-typedef size_t (*vcedit_read_func)(void *, size_t, size_t, void *);
-typedef size_t (*vcedit_write_func)(const void *, size_t, size_t, void *);
+typedef gint64 (*vcedit_read_func)(void *, gint64, gint64, void *);
+typedef gint64 (*vcedit_write_func)(const void *, gint64, gint64, void *);
 
 typedef struct {
 	ogg_sync_state	 *oy;
