diff -Nurb --strip-trailing-cr audacious-plugins-3.2-beta1-orig/src/amidi-plug/Makefile audacious-plugins-3.2-beta1/src/amidi-plug/Makefile
--- audacious-plugins-3.2-beta1-orig/src/amidi-plug/Makefile	2011-12-30 00:21:49.000000000 +0100
+++ audacious-plugins-3.2-beta1/src/amidi-plug/Makefile	2012-01-02 02:54:53.646129222 +0100
@@ -21,4 +21,4 @@
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS} ${PANGO_CFLAGS} ${GMODULE_CFLAGS} \
 	-DAMIDIPLUGBACKENDDIR=\"${plugindir}/${AMIDIPLUG_BACKEND_DIR}\" ${PTHREAD_CFLAGS} -I../..
-LIBS += ${GTK_LIBS} ${PANGO_LIBS} ${GMODULE_LIBS} ${PTHREAD_LIBS} ./pcfg/libpcfg.a ${GLIB_LIBS}
+LIBS += ${GTK_LIBS} ${PANGO_LIBS} ${GMODULE_LIBS} ${PTHREAD_LIBS} ./pcfg/libpcfg.a ${GLIB_LIBS} `pkg-config --libs gmodule-2.0`
diff -Nurb --strip-trailing-cr audacious-plugins-3.2-beta1-orig/src/ladspa/Makefile audacious-plugins-3.2-beta1/src/ladspa/Makefile
--- audacious-plugins-3.2-beta1-orig/src/ladspa/Makefile	2011-12-30 00:21:49.000000000 +0100
+++ audacious-plugins-3.2-beta1/src/ladspa/Makefile	2012-01-02 02:56:17.904557465 +0100
@@ -12,4 +12,4 @@
 
 CPPFLAGS += -I../.. ${PTHREAD_CFLAGS} ${GTK_CFLAGS}
 CFLAGS += ${PLUGIN_CFLAGS}
-LIBS += -lm ${PTHREAD_LIBS} ${GTK_LIBS}
+LIBS += -lm ${PTHREAD_LIBS} ${GTK_LIBS} `pkg-config --libs gmodule-2.0`
