diff -Nur audacious-plugins-2.4-alpha1-orig/src/ladspa/ladspa.c audacious-plugins-2.4-alpha1/src/ladspa/ladspa.c
--- audacious-plugins-2.4-alpha1-orig/src/ladspa/ladspa.c	2010-06-08 16:00:01.000000000 +0200
+++ audacious-plugins-2.4-alpha1/src/ladspa/ladspa.c	2010-06-11 19:48:21.000000000 +0200
@@ -224,8 +224,8 @@
     if (ladspa_path == NULL)
     {
 	/* Fallback, look in obvious places */
-	find_plugins("/usr/lib/ladspa");
-	find_plugins("/usr/local/lib/ladspa");
+	find_plugins("__RPM_LIBDIR__/ladspa");
+	find_plugins("/usr/local/__RPM_LIB__/ladspa");
     }
     else
     {
