diff -Nur audacious-plugins-2.5-alpha1-orig/src/skins/ui_skinselector.c audacious-plugins-2.5-alpha1/src/skins/ui_skinselector.c
--- audacious-plugins-2.5-alpha1-orig/src/skins/ui_skinselector.c	2011-01-30 20:40:47.000000000 +0100
+++ audacious-plugins-2.5-alpha1/src/skins/ui_skinselector.c	2011-01-31 21:22:34.805577066 +0100
@@ -253,9 +253,13 @@
 skinlist_update(void)
 {
     gchar *skinsdir;
+    gchar xmmsskinsdir[] = "/usr/share/xmms/Skins";
 
     skinlist_clear();
 
+    if (g_file_test(xmmsskinsdir, G_FILE_TEST_IS_DIR)) {
+        scan_skindir( xmmsskinsdir );
+    }
     scan_skindir(skins_paths[SKINS_PATH_USER_SKIN_DIR]);
 
     gchar * path = g_strdup_printf ("%s/Skins", aud_get_path (AUD_PATH_DATA_DIR));
