diff -Nur audacious-plugins-fedora-3.0-alpha1-orig/src/skins/ui_skinselector.c audacious-plugins-fedora-3.0-alpha1/src/skins/ui_skinselector.c
--- audacious-plugins-fedora-3.0-alpha1-orig/src/skins/ui_skinselector.c	2011-06-12 03:42:34.000000000 +0200
+++ audacious-plugins-fedora-3.0-alpha1/src/skins/ui_skinselector.c	2011-06-14 14:08:31.582444777 +0200
@@ -235,9 +235,13 @@
 skinlist_update(void)
 {
     gchar *skinsdir;
+    gchar xmmsskinsdir[] = "/usr/share/xmms/Skins";
 
     skinlist_clear();
 
+    if (g_file_test(xmmsskinsdir, G_FILE_TEST_IS_DIR)) {
+        scan_skindir( xmmsskinsdir );
+    }
     if (g_file_test (skins_paths[SKINS_PATH_USER_SKIN_DIR], G_FILE_TEST_EXISTS))
         scan_skindir (skins_paths[SKINS_PATH_USER_SKIN_DIR]);
 
