--- configure.ac.newline	2010-08-03 01:27:33.897856598 -0300
+++ configure.ac	2010-08-03 01:27:38.289843708 -0300
@@ -166,8 +166,7 @@
  enable_gio=$enableval, enable_gio=yes)
 
 if test "x$enable_gio" = "xyes"; then
-    PKG_CHECK_MODULES(GIO, [gio-2.0 >= 2.0], [ADD_PC_REQUIRES([gio-2.0 >=
-     2.0])], [enable_gio=no])
+    PKG_CHECK_MODULES(GIO, [gio-2.0 >= 2.0], [ADD_PC_REQUIRES([gio-2.0 >= 2.0])], [enable_gio=no])
 fi
 
 if test "x$enable_gio" = "xyes"; then
