diff -Nur audacious-plugins-3.7-alpha1-orig/src/skins/skinselector.cc audacious-plugins-3.7-alpha1/src/skins/skinselector.cc
--- audacious-plugins-3.7-alpha1-orig/src/skins/skinselector.cc	2015-06-30 02:58:08.000000000 +0200
+++ audacious-plugins-3.7-alpha1/src/skins/skinselector.cc	2015-07-06 12:47:34.785280539 +0200
@@ -124,6 +124,11 @@
 {
     skinlist.clear ();
 
+    char xmmsskinsdir[] = "/usr/share/xmms/Skins";
+    if (g_file_test(xmmsskinsdir, G_FILE_TEST_IS_DIR)) {
+        dir_foreach (xmmsskinsdir, scan_skindir_func);
+    }
+
     const char * user_skin_dir = skins_get_user_skin_dir ();
     if (g_file_test (user_skin_dir, G_FILE_TEST_EXISTS))
         dir_foreach (user_skin_dir, scan_skindir_func);
