diff -Nur audacious-plugins-fedora-2.4.0-orig/src/cue/cue.c audacious-plugins-fedora-2.4.0/src/cue/cue.c
--- audacious-plugins-fedora-2.4.0-orig/src/cue/cue.c	2010-08-26 01:07:50.000000000 +0200
+++ audacious-plugins-fedora-2.4.0/src/cue/cue.c	2010-11-04 16:34:59.528946835 +0100
@@ -72,7 +72,7 @@
 
 static void playlist_load_cue (const gchar * cue_filename, gint at)
 {
-    void * buffer;
+    void * buffer = NULL;
     gint64 size;
     vfs_file_get_contents (cue_filename, & buffer, & size);
     if (buffer == NULL)
