diff -Nur audacious-plugins-3.6-orig/src/ladspa/plugin.cc audacious-plugins-3.6/src/ladspa/plugin.cc
--- audacious-plugins-3.6-orig/src/ladspa/plugin.cc	2015-02-28 22:57:53.000000000 +0100
+++ audacious-plugins-3.6/src/ladspa/plugin.cc	2015-03-02 15:57:15.499371104 +0100
@@ -35,6 +35,7 @@
 
 const char * const LADSPAHost::defaults[] = {
  "plugin_count", "0",
+ "module_path", "__RPM_LIBDIR__/ladspa",
  nullptr};
 
 pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
