diff -Nur audacious-plugins-2.2-orig/src/skins/plugin.c audacious-plugins-2.2/src/skins/plugin.c
--- audacious-plugins-2.2-orig/src/skins/plugin.c	2009-11-22 23:49:53.000000000 +0100
+++ audacious-plugins-2.2/src/skins/plugin.c	2010-04-10 11:08:35.000000000 +0200
@@ -255,7 +255,7 @@
                                            GTK_DIALOG_DESTROY_WITH_PARENT,
                                            GTK_MESSAGE_ERROR,
                                            GTK_BUTTONS_OK,
-                                           _(markup));
+                                           "%s",_(markup));
 
     gtk_window_set_modal(GTK_WINDOW(dialog), TRUE);
     gtk_widget_show(GTK_WIDGET(dialog));
