diff -Nur audacious-plugins-fedora-2.4.2-orig/src/notify/libnotify-aosd_osd.c audacious-plugins-fedora-2.4.2/src/notify/libnotify-aosd_osd.c
--- audacious-plugins-fedora-2.4.2-orig/src/notify/libnotify-aosd_osd.c	2010-12-08 00:14:55.000000000 +0100
+++ audacious-plugins-fedora-2.4.2/src/notify/libnotify-aosd_osd.c	2010-12-09 19:37:35.037726989 +0100
@@ -55,7 +55,7 @@
 	GError *error = NULL;
 
 	if(notification == NULL) {
-		notification = notify_notification_new(title, message, pb == NULL ? icon : NULL, NULL);
+		notification = notify_notification_new(title, message, pb == NULL ? icon : NULL);
 		g_signal_connect(notification, "closed", G_CALLBACK(osd_closed_handler), NULL);
 		AUDDBG("new osd created! (notification=%p)\n", notification);
 	} else {
