diff -Nur audacious-plugins-2.4.3-orig/src/psf/plugin.c audacious-plugins-2.4.3/src/psf/plugin.c
--- audacious-plugins-2.4.3-orig/src/psf/plugin.c	2011-01-12 12:13:26.000000000 +0100
+++ audacious-plugins-2.4.3/src/psf/plugin.c	2011-01-28 23:47:46.829315597 +0100
@@ -96,6 +96,7 @@
 	return AO_SUCCESS;
 }
 
+static GStaticMutex mutex = G_STATIC_MUTEX_INIT;
 static gint seek = 0;
 gboolean stop_flag = FALSE;
 
@@ -182,7 +183,6 @@
 			}
 			else
 			{
-				data->output->close_audio();
 				break;
 			}
 		}
@@ -192,15 +192,17 @@
 		while (!stop_flag && data->output->buffer_playing())
 			g_usleep(10000);
 
-		data->output->close_audio();
-
 		break;
 	}
 
+    g_static_mutex_lock (& mutex);
+    stop_flag = TRUE;
+    data->output->close_audio ();
+    g_static_mutex_unlock (& mutex);
+
 	g_free(buffer);
 	g_free(path);
 
-	stop_flag = TRUE;
 	return ! error;
 }
 
@@ -232,8 +234,13 @@
 
 void psf2_Stop(InputPlayback *playback)
 {
-	stop_flag = TRUE;
-	playback->output->abort_write();
+    g_static_mutex_lock (& mutex);
+    if (! stop_flag)
+    {
+        stop_flag = TRUE;
+        playback->output->abort_write ();
+    }
+    g_static_mutex_unlock (& mutex);
 }
 
 void psf2_pause(InputPlayback *playback, gshort pause)
