diff -up ClanLib-0.6.5/Sources/Core/DatafileCompiler/datafile_writer.cpp~ ClanLib-0.6.5/Sources/Core/DatafileCompiler/datafile_writer.cpp
--- ClanLib-0.6.5/Sources/Core/DatafileCompiler/datafile_writer.cpp~	2011-04-03 19:19:42.000000000 +0200
+++ ClanLib-0.6.5/Sources/Core/DatafileCompiler/datafile_writer.cpp	2011-04-03 19:20:03.625836085 +0200
@@ -147,7 +147,7 @@ OutputSource_Datafile::OutputSource_Data
 	output_fd = provider->create_index(handle);
 	char compression = provider->get_compression() ? 1 : 0;
 	::write(output_fd, &compression,1);
-	if (provider->get_compression()) output_zfd = gzdopen(output_fd, "w+b");
+	if (provider->get_compression()) output_zfd = gzdopen(output_fd, "wb");
 	pos = 1;
 }
 
@@ -347,7 +347,7 @@ void CL_Datafile_Write::create_index(cha
 	write(index, &data_pos, sizeof(long));
 
 	index_open=true;
-	data_indexed=gzdopen(dup(data), "w+b");
+	data_indexed=gzdopen(dup(data), "wb");
 
 	num_indexes++;
 
