--- src/gtk/Makefile.am.orig	2013-12-01 19:13:43.888580986 -0200
+++ src/gtk/Makefile.am	2013-12-01 19:13:56.374926633 -0200
@@ -26,6 +26,7 @@ desmume_SOURCES = \
 	../filter/scanline.cpp \
 	../filter/videofilter.cpp ../filter/videofilter.h \
 	main.cpp main.h
+main.o: CXXFLAGS += -O1
 desmume_LDADD = ../libdesmume.a \
 	$(SDL_LIBS) $(GTK_LIBS) $(GTHREAD_LIBS) $(ALSA_LIBS) $(LIBAGG_LIBS) $(LIBSOUNDTOUCH_LIBS)
 if HAVE_GDB_STUB
