--- Hurrican/src/DX8Graphics.cpp.orig	2012-08-08 18:47:21.688576855 -0300
+++ Hurrican/src/DX8Graphics.cpp	2012-08-08 18:51:22.427029061 -0300
@@ -228,7 +228,7 @@ _ModeFound:
 bool DirectGraphicsClass::Init(HWND hwnd, DWORD dwBreite, DWORD dwHoehe,
                            DWORD dwZ_Bits, bool VSync)
 {
-    bool isFullscreen       = !CommandLineParams.RunWindowMode;
+    bool isFullscreen       = CommandLineParams.RunWindowMode;
     uint16_t ScreenWidth    = SCREENWIDTH;
     uint16_t ScreenHeight   = SCREENHEIGHT;
     uint16_t ScreenDepth    = SCREENBPP;
