--- proftpd.conf.orig	2013-10-02 22:56:59.565069285 -0300
+++ proftpd.conf	2013-10-02 22:57:54.953367208 -0300
@@ -22,6 +22,20 @@ Umask				022
 # (such as xinetd)
 MaxInstances			30
 
+# Use pam to authenticate (default) and be authoritative
+AuthPAMConfig			proftpd
+AuthOrder			mod_auth_pam.c* mod_auth_unix.c
+
+# Use PAM to provide a 'keyboard-interactive' SSH2 authentication method for
+# mod_sftp (http://www.castaglia.org/proftpd/modules/mod_sftp_pam.html)
+#   LoadModule mod_sftp_pam.c
+
+# Implement a virtual chroot capability that does not require root privileges
+# (http://www.castaglia.org/proftpd/modules/mod_vroot.html)
+# Using this module rather than the kernel's chroot() system call works
+# around issues with PAM and chroot (http://bugzilla.redhat.com/506735)
+LoadModule mod_vroot.c
+
 # Set the user and group that the server normally runs at.
 User				nobody
 Group				nogroup
