Index: contrib/mod_deflate.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_deflate.c,v
retrieving revision 1.8
diff -u -r1.8 mod_deflate.c
--- contrib/mod_deflate.c	10 Jun 2011 02:57:35 -0000	1.8
+++ contrib/mod_deflate.c	23 Nov 2011 17:58:21 -0000
@@ -898,7 +898,7 @@
 
     pr_signals_block();
     PRIVS_ROOT
-    res = pr_log_openfile(c->argv[0], &deflate_logfd, 0640);
+    res = pr_log_openfile(c->argv[0], &deflate_logfd, PR_LOG_SYSTEM_MODE);
     PRIVS_RELINQUISH
     pr_signals_unblock();
 
Index: contrib/mod_exec.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_exec.c,v
retrieving revision 1.20
diff -u -r1.20 mod_exec.c
--- contrib/mod_exec.c	24 Sep 2011 06:44:36 -0000	1.20
+++ contrib/mod_exec.c	23 Nov 2011 17:58:21 -0000
@@ -192,7 +192,7 @@
 
   pr_signals_block();
   PRIVS_ROOT
-  res = pr_log_openfile(exec_logname, &exec_logfd, 0640);
+  res = pr_log_openfile(exec_logname, &exec_logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
   pr_signals_unblock();
 
Index: contrib/mod_quotatab.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_quotatab.c,v
retrieving revision 1.76
diff -u -r1.76 mod_quotatab.c
--- contrib/mod_quotatab.c	26 May 2011 23:14:01 -0000	1.76
+++ contrib/mod_quotatab.c	23 Nov 2011 17:58:21 -0000
@@ -429,7 +429,7 @@
 
   pr_signals_block();
   PRIVS_ROOT
-  res = pr_log_openfile(quota_logname, &quota_logfd, 0640);
+  res = pr_log_openfile(quota_logname, &quota_logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
   pr_signals_unblock();
 
Index: contrib/mod_radius.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_radius.c,v
retrieving revision 1.66
diff -u -r1.66 mod_radius.c
--- contrib/mod_radius.c	16 Aug 2011 16:13:02 -0000	1.66
+++ contrib/mod_radius.c	23 Nov 2011 17:58:21 -0000
@@ -1825,7 +1825,7 @@
 
   pr_signals_block();
   PRIVS_ROOT
-  res = pr_log_openfile(radius_logname, &radius_logfd, 0640);
+  res = pr_log_openfile(radius_logname, &radius_logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
   pr_signals_unblock();
 
Index: contrib/mod_sql.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_sql.c,v
retrieving revision 1.218
diff -u -r1.218 mod_sql.c
--- contrib/mod_sql.c	4 Oct 2011 05:27:18 -0000	1.218
+++ contrib/mod_sql.c	23 Nov 2011 17:58:22 -0000
@@ -5383,7 +5383,7 @@
 
   pr_signals_block();
   PRIVS_ROOT
-  res = pr_log_openfile(sql_logfile, &sql_logfd, 0640);
+  res = pr_log_openfile(sql_logfile, &sql_logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
   pr_signals_unblock();
 
Index: contrib/mod_tls.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_tls.c,v
retrieving revision 1.264
diff -u -r1.264 mod_tls.c
--- contrib/mod_tls.c	19 Nov 2011 02:54:13 -0000	1.264
+++ contrib/mod_tls.c	23 Nov 2011 17:58:22 -0000
@@ -6152,7 +6152,7 @@
 
   pr_signals_block();
   PRIVS_ROOT
-  res = pr_log_openfile(tls_logname, &tls_logfd, 0600);
+  res = pr_log_openfile(tls_logname, &tls_logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
   pr_signals_unblock();
 
Index: contrib/mod_sftp/mod_sftp.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_sftp/mod_sftp.c,v
retrieving revision 1.61
diff -u -r1.61 mod_sftp.c
--- contrib/mod_sftp/mod_sftp.c	12 Oct 2011 17:15:56 -0000	1.61
+++ contrib/mod_sftp/mod_sftp.c	23 Nov 2011 17:58:22 -0000
@@ -1675,7 +1675,7 @@
 
     pr_signals_block();
     PRIVS_ROOT
-    res = pr_log_openfile(sftp_logname, &sftp_logfd, 0600);
+    res = pr_log_openfile(sftp_logname, &sftp_logfd, PR_LOG_SYSTEM_MODE);
     PRIVS_RELINQUISH
     pr_signals_unblock();
 
Index: contrib/mod_wrap2/mod_wrap2.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/contrib/mod_wrap2/mod_wrap2.c,v
retrieving revision 1.39
diff -u -r1.39 mod_wrap2.c
--- contrib/mod_wrap2/mod_wrap2.c	6 Nov 2011 21:56:12 -0000	1.39
+++ contrib/mod_wrap2/mod_wrap2.c	23 Nov 2011 17:58:22 -0000
@@ -150,7 +150,7 @@
 
   pr_signals_block();
   PRIVS_ROOT
-  res = pr_log_openfile(wrap2_logname, &wrap2_logfd, 0640);
+  res = pr_log_openfile(wrap2_logname, &wrap2_logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
   pr_signals_unblock();
 
Index: modules/mod_ctrls.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/modules/mod_ctrls.c,v
retrieving revision 1.51
diff -u -r1.51 mod_ctrls.c
--- modules/mod_ctrls.c	17 Nov 2011 23:40:28 -0000	1.51
+++ modules/mod_ctrls.c	23 Nov 2011 17:58:22 -0000
@@ -110,7 +110,7 @@
     return 0;
 
   PRIVS_ROOT
-  res = pr_log_openfile(ctrls_logname, &logfd, 0640);
+  res = pr_log_openfile(ctrls_logname, &logfd, PR_LOG_SYSTEM_MODE);
   PRIVS_RELINQUISH
 
   if (res == 0) {
Index: modules/mod_memcache.c
===================================================================
RCS file: /cvsroot/proftp/proftpd/modules/mod_memcache.c,v
retrieving revision 1.15
diff -u -r1.15 mod_memcache.c
--- modules/mod_memcache.c	23 May 2011 21:11:56 -0000	1.15
+++ modules/mod_memcache.c	23 Nov 2011 17:58:22 -0000
@@ -347,7 +347,7 @@
 
       pr_signals_block();
       PRIVS_ROOT
-      res = pr_log_openfile(path, &memcache_logfd, 0600);
+      res = pr_log_openfile(path, &memcache_logfd, PR_LOG_SYSTEM_MODE);
       PRIVS_RELINQUISH
       pr_signals_unblock();
 
