--- src/apprentice.c.orig	2013-02-24 02:18:27.659103027 -0300
+++ src/apprentice.c	2013-02-24 02:19:23.012366736 -0300
@@ -419,8 +419,8 @@ apprentice_1(struct magic_set *ms, const
 #ifndef COMPILE_ONLY
 	map = apprentice_map(ms, fn);
 	if (map == NULL) {
-		if (ms->flags & MAGIC_CHECK)
-			file_magwarn(ms, "using regular magic file `%s'", fn);
+		/* if (ms->flags & MAGIC_CHECK)
+			file_magwarn(ms, "using regular magic file `%s'", fn); */
 		map = apprentice_load(ms, fn, action);
 		if (map == NULL)
 			return -1;
