diff --git a/magic/Magdir/filesystems b/magic/Magdir/filesystems
index 3323593..0f28d7b 100644
--- a/magic/Magdir/filesystems
+++ b/magic/Magdir/filesystems
@@ -62,7 +62,7 @@
 30		search/481	\x55\xAA	
 # to display x86 boot sector (40) before old one (strength=50+21),Syslinux bootloader (71),SYSLINUX MBR (37+36),NetBSD mbr (110),AdvanceMAME mbr (111)
 # DOS BPB information (70) and after DOS floppy (120) like in previous file version
-!:strength +72
+!:strength +0
 # for sector sizes < 512 Bytes
 >11		uleshort	<512		
 >>(11.s-2)	uleshort	0xAA55		x86 boot sector
@@ -1176,7 +1176,7 @@
 >11		uleshort&0xf001f	0	
 >>11		uleshort	<32769		
 >>>11		uleshort	>31		
->>>>21		ubyte&0xf0	0xF0		
+>>>>21		ubyte&0xf0	0xF0		x86 boot sector
 >>>>>0		ubyte		0xEB		
 >>>>>>1		ubyte		x		\b, code offset 0x%x+2
 >>>>>0		ubyte		0xE9		
