diff -up file-5.04/src/file.c.separ file-5.04/src/file.c
--- file-5.04/src/file.c.separ	2010-03-24 10:24:19.000000000 +0100
+++ file-5.04/src/file.c	2010-03-24 10:24:52.000000000 +0100
@@ -406,9 +406,8 @@ process(struct magic_set *ms, const char
 	if (wid > 0 && !bflag) {
 		(void)printf("%s", std_in ? "/dev/stdin" : inname);
 		if (nulsep)
-			(void)putc('\0', stdout);
-		else
-			(void)printf("%s", separator);
+			(void)putc('\0', stdout);		
+		(void)printf("%s", separator);
 		(void)printf("%*s ",
 		    (int) (nopad ? 0 : (wid - file_mbswidth(inname))), "");
 	}
