diff -up file-5.04/src/fsmagic.c.retval file-5.04/src/fsmagic.c
--- file-5.04/src/fsmagic.c.retval	2010-04-14 12:37:12.000000000 +0200
+++ file-5.04/src/fsmagic.c	2010-04-14 12:55:50.000000000 +0200
@@ -127,14 +127,8 @@ file_fsmagic(struct magic_set *ms, const
 	ret = stat(fn, sb);	/* don't merge into if; see "ret =" above */
 
 	if (ret) {
-		if (ms->flags & MAGIC_ERROR) {
-			file_error(ms, errno, "cannot stat `%s'", fn);
-			return -1;
-		}
-		if (file_printf(ms, "cannot open `%s' (%s)",
-		    fn, strerror(errno)) == -1)
-			return -1;
-		return 1;
+	    file_error(ms, errno, "cannot open `%s'", fn);
+	    return -1;
 	}
 
 	if (!mime) {
