diff --git a/magic/Magdir/perl b/magic/Magdir/perl
index 1d7bd31..94536e2 100644
--- a/magic/Magdir/perl
+++ b/magic/Magdir/perl
@@ -33,7 +33,8 @@
 0	search/1	!p
 >0	regex		\^package[\ \t]+[0-9A-Za-z_:]+\ *;
 >>0	regex		\^1\ *;|\^(use|sub|my)\ .*[(;{=]	Perl5 module source text
-!:strength + 10
+# Originally 12, we want at least 50
+!:strength + 50
 
 # Perl POD documents
 # From: Tom Hukins <tom@eborcom.com>
