diff --git a/magic/Magdir/elf b/magic/Magdir/elf
index 83bdbbf..aa9f539 100644
--- a/magic/Magdir/elf
+++ b/magic/Magdir/elf
@@ -189,9 +189,9 @@
 >4	byte		1		32-bit
 >4	byte		2		64-bit
 >5	byte		0		invalid byte order
->5	byte		1		LSB 
+>5	byte		1		LSB
 >>0	use		elf-le
->5	byte		2		MSB 
+>5	byte		2		MSB
 >>0	use		\^elf-le
 # Up to now only 0, 1 and 2 are defined; I've seen a file with 0x83, it seemed
 # like proper ELF, but extracting the string had bad results.
