--- magic/Magdir/images	2008-08-27 12:23:45.000000000 +0200
+++ magic/Magdir/images.oden	2008-12-26 00:20:58.000000000 +0100
@@ -567,6 +567,12 @@
 0	string	\211HDF\r\n\032	Hierarchical Data Format (version 5) data
 !:mime	application/x-hdf
 
+# The boot loaders syslinux and isolinux use a RLE based image format
+# called SLL16 to store splash screens.
+0	lelong		0x1413f33d	Syslinux SLL16 image data,
+>4	leshort		>0		%d x
+>6	leshort		>0		%d
+
 # From: Tobias Burnus <burnus@net-b.de>
 # Xara (for a while: Corel Xara) is a graphic package, see
 # http://www.xara.com/ for Windows and as GPL application for Linux
