From d9e9aa9f06757947a1e51f80925445b2f18a6fc3 Mon Sep 17 00:00:00 2001
From: Reuben Thomas <rrt@sc3d.org>
Date: Mon, 14 Jul 2014 14:21:33 +0000
Subject: [PATCH] Comment out too-sensitive Pascal magic

---
 magic/Magdir/pascal | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/magic/Magdir/pascal b/magic/Magdir/pascal
index a95939c..d8e61c6 100644
--- a/magic/Magdir/pascal
+++ b/magic/Magdir/pascal
@@ -1,10 +1,10 @@
 #------------------------------------------------------------------------------
-# $File: pascal,v 1.1 2011/12/08 12:12:46 rrt Exp $
+# $File: pascal,v 1.1 2011/12/08 12:12:46 rrt Exp $
 # pascal:  file(1) magic for Pascal source
 #
 0	search/8192	(input,		Pascal source text
 !:mime	text/x-pascal
-0	regex		\^program	Pascal source text
-!:mime	text/x-pascal
-0	regex           	\^record		Pascal source text
-!:mime	text/x-pascal
+#0	regex		\^program	Pascal source text
+#!:mime	text/x-pascal
+#0	regex           	\^record		Pascal source text
+#!:mime	text/x-pascal
