diff --git a/magic/Magdir/perl b/magic/Magdir/perl
index b5b54fb..490dbeb 100644
--- a/magic/Magdir/perl
+++ b/magic/Magdir/perl
@@ -7,40 +7,54 @@
 # Send additions to <perl5-porters@perl.org>
 0	search/1024	eval\ "exec\ perl		Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ "exec\ /bin/perl		Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ "exec\ /usr/bin/perl	Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ "exec\ /usr/local/bin/perl	Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ 'exec\ perl		Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ 'exec\ /bin/perl		Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ 'exec\ /usr/bin/perl	Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ 'exec\ /usr/local/bin/perl	Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	eval\ '(exit\ $?0)'\ &&\ eval\ 'exec	Perl script text
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	#!/usr/bin/env\ perl	Perl script text executable
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	#!\ /usr/bin/env\ perl	Perl script text executable
 !:mime	text/x-perl
+!:strength + 30
 0	search/1024	#!
 >0	regex	\^#!.*/bin/perl([[:space:]].*)*$	Perl script text executable
 !:mime	text/x-perl
+!:strength + 30
 
 # by Dmitry V. Levin and Alexey Tourbin
 # check the first line
 0	search/1024	package
 >0	regex		\^package[\ \t]+[0-9A-Za-z_:]+\ *;	Perl5 module source text
-!:strength + 10
+!:mime  text/x-perl
+!:strength + 45
 # not 'p', check other lines
 0	search/1024	!p
 >0	regex		\^package[\ \t]+[0-9A-Za-z_:]+\ *;
 >>0	regex		\^1\ *;|\^(use|sub|my)\ .*[(;{=]	Perl5 module source text
-!:strength + 10
+!:mime  text/x-perl
+!:strength + 45
 
 # Perl POD documents
 # From: Tom Hukins <tom@eborcom.com>
