diff --git a/magic/Magdir/msdos b/magic/Magdir/msdos
index f8852cb..36bd51c 100644
--- a/magic/Magdir/msdos
+++ b/magic/Magdir/msdos
@@ -841,7 +841,12 @@
 # Dell's .hdr
 0	string $RBU
 >23	string Dell			%s system BIOS
->48	string x			version %.3s
+>5	byte   2
+>>48	byte   x			version %d.
+>>49	byte   x			\b%d.
+>>50	byte   x			\b%d
+>5	byte   <2
+>>48	string x			version %.3s
 
 # Type: Microsoft DirectDraw Surface
 # URL:	http://msdn.microsoft.com/library/default.asp?url=/library/en-us/directx9_c/directx/graphics/reference/DDSFileReference/ddsfileformat.asp
