--- file-5.05/magic/Magdir/audio.audio	2010-10-06 22:13:42.000000000 -0300
+++ file-5.05/magic/Magdir/audio	2011-01-18 18:49:42.860695868 -0200
@@ -129,7 +129,7 @@
 # Oct 31, 1995
 # fixed by <doj@cubic.org> 2003-06-24
 # Too short...
-#0	string		MTM		MultiTracker Module sound file
+#0	string		MTM		MultiTracker Module sound data
 #0	string		if		Composer 669 Module sound data
 #0	string		JN		Composer 669 Module sound data (extended format)
 0	string		MAS_U		ULT(imate) Module sound data
