diff --git a/magic/Magdir/compress b/magic/Magdir/compress
index beb8ebe..ae1c90f 100644
--- a/magic/Magdir/compress
+++ b/magic/Magdir/compress
@@ -257,7 +257,7 @@
 !:mime	application/x-qpress
 
 # Zlib https://www.ietf.org/rfc/rfc6713.txt
-0	beshort%31	=0	
->0	byte&0xf	=8
->>0	byte&0x80 	=0	zlib compressed data
-!:mime	application/zlib
+#0	beshort%31	=0	
+#>0	byte&0xf	=8
+#>>0	byte&0x80 	=0	zlib compressed data
+#!:mime	application/zlib
