--- file-5.05/magic/Magdir/compress.xz_container	2011-01-18 18:55:44.832968263 -0200
+++ file-5.05/magic/Magdir/compress	2011-01-18 18:55:57.577196415 -0200
@@ -200,7 +200,9 @@
 >>5	lequad		!0xffffffffffffffff	non-streamed, size %lld
 
 # http://tukaani.org/xz/xz-file-format.txt
-0	ustring		\xFD7zXZ\x00		XZ compressed data
+0	ustring		\xFD7zXZ\x00		XZ compressed data,
+>6	ubyte&0x10		0x00		single-block stream
+>6	ubyte&0x10		0x10		multi-block stream
 !:strength * 2
 !:mime	application/x-xz
 
