From bf3fea9e6c0b18e9a645d2a796a1e3ae29be5ee5 Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Tue, 5 Aug 2014 07:38:45 +0000
Subject: [PATCH] PR/360: Andreas Koehler: Escape regex properly.

---
 magic/Magdir/msooxml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/magic/Magdir/msooxml b/magic/Magdir/msooxml
index c908b71..80bdf13 100644
--- a/magic/Magdir/msooxml
+++ b/magic/Magdir/msooxml
@@ -16,7 +16,7 @@
 0		string		PK\003\004
 !:strength +10
 # make sure the first file is correct
->0x1E		regex		\[Content_Types\]\.xml|_rels/\.rels
+>0x1E		regex		\\[Content_Types\\]\\.xml|_rels/\\.rels
 # skip to the second local file header
 # since some documents include a 520-byte extra field following the file
 # header, we need to scan for the next header
