diff --git a/python/magic.py b/python/magic.py
index 35a7182..1cfc035 100644
--- a/python/magic.py
+++ b/python/magic.py
@@ -19,17 +19,17 @@ _libraries = {}
 _libraries['magic'] = _init()
 
 # Flag constants for open and setflags
-NONE = 0
-DEBUG = 1
-SYMLINK = 2
-COMPRESS = 4
-DEVICES = 8
-MIME_TYPE = 16
-CONTINUE = 32
-CHECK = 64
-PRESERVE_ATIME = 128
+NONE = MAGIC_NONE = 0
+DEBUG = MAGIC_DEBUG = 1
+SYMLINK = MAGIC_SYMLINK = 2
+COMPRESS = MAGIC_COMPRESS = 4
+DEVICES = MAGIC_DEVICES = 8
+MIME_TYPE = MAGIC_MIME = 16
+CONTINUE = MAGIC_CONTINUE = 32
+CHECK = MAGIC_CHECK = 64
+PRESERVE_ATIME = MAGIC_PRESERVE_ATIME = 128
 RAW = 256
-ERROR = 512
+ERROR = MAGIC_ERROR = 512
 MIME_ENCODING = 1024
 MIME = 1040
 APPLE = 2048
