diff --git a/doc/file.man b/doc/file.man
index fff7730..3a524f5 100644
--- a/doc/file.man
+++ b/doc/file.man
@@ -542,6 +542,11 @@ program, and are not covered by the above license.
 .Sh RETURN CODE
 .Nm
 returns 0 on success, and non-zero on error.
+.Pp
+If the file named by the file
+operand does not exist, cannot be read, or the type of the file named by
+the file operand cannot be determined, this is not be considered an error
+that affects the exit status.
 .Sh BUGS
 .Pp
 Please report bugs and send patches to the bug tracker at
