diff --git a/magic/Magdir/sgml b/magic/Magdir/sgml
index 0a57375..1dceba5 100644
--- a/magic/Magdir/sgml
+++ b/magic/Magdir/sgml
@@ -34,21 +34,13 @@
 # HyperText Markup Language (HTML) is an SGML document type,
 # from Daniel Quinlan (quinlan@yggdrasil.com)
 # adapted to string extenstions by Anthon van der Neut <anthon@mnt.org)
-0	search/4096/cWbt	\<!doctype\ html	HTML document text
+0	search/1/cb	\<!doctype\ html	HTML document text
 !:mime	text/html
-0	search/4096/cwbt	\<head			HTML document text
+0	search/1/cb	\<head			HTML document text
 !:mime	text/html
-0	search/4096/cwbt	\<title			HTML document text
+0	search/1/cb	\<title			HTML document text
 !:mime	text/html
-0	search/4096/cwbt	\<html			HTML document text
-!:mime	text/html
-0	search/4096/cwbt	\<script 		HTML document text
-!:mime	text/html
-0	search/4096/cwbt	\<style 		HTML document text
-!:mime	text/html
-0	search/4096/cwbt	\<table			HTML document text
-!:mime	text/html
-0	search/4096/cwbt	\<a\ href=		HTML document text
+0	search/1/cb	\<html			HTML document text
 !:mime	text/html
 
 # Extensible markup language (XML), a subset of SGML
@@ -76,10 +68,9 @@
 
 
 # SGML, mostly from rph@sq
-0	search/4096/cwbt	\<!doctype		exported SGML document text
-0	search/4096/cwbt	\<!subdoc		exported SGML subdocument text
-0	search/4096/cwbt	\<!--			exported SGML document text
-!:strength - 10
+0	search/1/cb	\<!doctype		exported SGML document text
+0	search/1/cb	\<!subdoc		exported SGML subdocument text
+0	search/1/cb	\<!--			exported SGML document text
 
 # Web browser cookie files
 # (Mozilla, Galeon, Netscape 4, Konqueror..)
