diff --git a/magic/Magdir/msdos b/magic/Magdir/msdos
index d1a19a6..9e89f11 100644
--- a/magic/Magdir/msdos
+++ b/magic/Magdir/msdos
@@ -319,9 +319,9 @@
 # but it isn't feasible to match all COM files since there must be at least
 # two dozen different one-byte "magics".
 # test too generic ?
-0	byte		0xe9		DOS executable (COM)
->0x1FE leshort		0xAA55		\b, boot code
->6	string		SFX\ of\ LHarc	(%s)
+#0	byte		0xe9		DOS executable (COM)
+#>0x1FE leshort		0xAA55		\b, boot code
+#>6	string		SFX\ of\ LHarc	(%s)
 
 # DOS device driver updated by Joerg Jenderek at May 2011
 # http://maben.homeip.net/static/S100/IBM/software/DOS/DOS%20techref/CHAPTER.009
@@ -396,28 +396,28 @@
 #>4	uleshort&0x4000			0x4000			\b,control strings-support)
 
 # test too generic ?
-0	byte		0x8c		DOS executable (COM)
+#0	byte		0x8c		DOS executable (COM)
 # updated by Joerg Jenderek at Oct 2008
 0	ulelong		0xffff10eb	DR-DOS executable (COM)
 # byte 0xeb conflicts with "sequent" magic leshort 0xn2eb
-0	ubeshort&0xeb8d	>0xeb00		
+#0	ubeshort&0xeb8d	>0xeb00		
 # DR-DOS STACKER.COM SCREATE.SYS missed
->0	byte		0xeb
->>0x1FE leshort		0xAA55		DOS executable (COM), boot code
->>85	string		UPX		DOS executable (COM), UPX compressed
->>4	string		\ $ARX		DOS executable (COM), ARX self-extracting archive
->>4	string		\ $LHarc	DOS executable (COM), LHarc self-extracting archive
->>0x20e string		SFX\ by\ LARC	DOS executable (COM), LARC self-extracting archive
+#>0	byte		0xeb
+#>>0x1FE leshort		0xAA55		DOS executable (COM), boot code
+#>>85	string		UPX		DOS executable (COM), UPX compressed
+#>>4	string		\ $ARX		DOS executable (COM), ARX self-extracting archive
+#>>4	string		\ $LHarc	DOS executable (COM), LHarc self-extracting archive
+#>>0x20e string		SFX\ by\ LARC	DOS executable (COM), LARC self-extracting archive
 # updated by Joerg Jenderek at Oct 2008
 #0	byte		0xb8		COM executable
-0	uleshort&0x80ff	0x00b8		
+#0	uleshort&0x80ff	0x00b8		
 # modified by Joerg Jenderek
->1	lelong		!0x21cd4cff	COM executable for DOS
+#>1	lelong		!0x21cd4cff	COM executable for DOS
 # http://syslinux.zytor.com/comboot.php
 # (32-bit COMBOOT) programs *.C32 contain 32-bit code and run in flat-memory 32-bit protected mode
 # start with assembler instructions mov eax,21cd4cffh
-0	uleshort&0xc0ff	0xc0b8		
->1	lelong		0x21cd4cff	COM executable (32-bit COMBOOT)
+#0	uleshort&0xc0ff	0xc0b8		
+#>1	lelong		0x21cd4cff	COM executable (32-bit COMBOOT)
 # syslinux:doc/comboot.txt
 # A COM32R program must start with the byte sequence B8 FE 4C CD 21 (mov
 # eax,21cd4cfeh) as a magic number.
@@ -437,27 +437,27 @@
 # GRR search is not working
 #2	search/28	\xcd\x21	COM executable for MS-DOS
 #WHICHFAT.cOM
-2	string	\xcd\x21		COM executable for DOS
+#2	string	\xcd\x21		COM executable for DOS
 #DELTREE.cOM DELTREE2.cOM
-4	string	\xcd\x21		COM executable for DOS
+#4	string	\xcd\x21		COM executable for DOS
 #IFMEMDSK.cOM ASSIGN.cOM COMP.cOM
-5	string	\xcd\x21		COM executable for DOS
+#5	string	\xcd\x21		COM executable for DOS
 #DELTMP.COm HASFAT32.cOM
-7	string	\xcd\x21		
->0	byte	!0xb8			COM executable for DOS
+#7	string	\xcd\x21		
+#>0	byte	!0xb8			COM executable for DOS
 #COMP.cOM MORE.COm
-10	string	\xcd\x21		
->5	string	!\xcd\x21		COM executable for DOS
+#10	string	\xcd\x21		
+#>5	string	!\xcd\x21		COM executable for DOS
 #comecho.com
-13	string	\xcd\x21		COM executable for DOS
+#13	string	\xcd\x21		COM executable for DOS
 #HELP.COm EDIT.coM
-18	string	\xcd\x21		COM executable for MS-DOS
+#18	string	\xcd\x21		COM executable for MS-DOS
 #NWRPLTRM.COm
-23	string	\xcd\x21		COM executable for MS-DOS
+#23	string	\xcd\x21		COM executable for MS-DOS
 #LOADFIX.cOm LOADFIX.cOm
-30	string	\xcd\x21		COM executable for MS-DOS
+#30	string	\xcd\x21		COM executable for MS-DOS
 #syslinux.com 3.11
-70	string	\xcd\x21		COM executable for DOS
+#70	string	\xcd\x21		COM executable for DOS
 # many compressed/converted COMs start with a copy loop instead of a jump
 0x6	search/0xa	\xfc\x57\xf3\xa5\xc3	COM executable for MS-DOS
 0x6	search/0xa	\xfc\x57\xf3\xa4\xc3	COM executable for DOS
