diff -up samba/filepropertiesplugin/sambausershareplugin.cpp.orig samba/filepropertiesplugin/sambausershareplugin.cpp
--- samba/filepropertiesplugin/sambausershareplugin.cpp.orig	2013-08-15 18:13:18.351112085 -0300
+++ samba/filepropertiesplugin/sambausershareplugin.cpp	2013-08-15 18:14:43.517114408 -0300
@@ -115,6 +115,7 @@ SambaUserSharePlugin::~SambaUserSharePlu
 {
 }
 
+#ifdef SAMBA_INSTALL
 void SambaUserSharePlugin::installSamba()
 {
     unsigned int xid = 0;
@@ -131,6 +132,7 @@ void SambaUserSharePlugin::installSamba(
     }
     QDBusReply<int> reply = device.call("InstallPackageNames", xid, packages, interaction);
 }
+#endif
 
 void SambaUserSharePlugin::setupModel()
 {
diff -up samba/filepropertiesplugin/sambausershareplugin.h.orig samba/filepropertiesplugin/sambausershareplugin.h
--- samba/filepropertiesplugin/sambausershareplugin.h.orig	2013-08-15 18:16:49.970389422 -0300
+++ samba/filepropertiesplugin/sambausershareplugin.h	2013-08-15 18:17:17.269619669 -0300
@@ -42,7 +42,9 @@ public:
 private Q_SLOTS:
     void load();
     void toggleShareStatus(bool checked);
+#ifdef SAMBA_INSTALL
     void installSamba();
+#endif
     void checkShareName(const QString &name);
 
 private:
