From 6fedeb66cc3c59bce24146f079295a1d404c749c Mon Sep 17 00:00:00 2001
From: Jos van den Oever <jos@vandenoever.info>
Date: Sat, 1 Jun 2013 17:59:36 +0200
Subject: [PATCH 3/5] Build fix for gcc 4.8.

---
 include/strigi/bufferedstream.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/strigi/bufferedstream.h b/include/strigi/bufferedstream.h
index 563da8a..05f5500 100644
--- a/include/strigi/bufferedstream.h
+++ b/include/strigi/bufferedstream.h
@@ -34,7 +34,7 @@ namespace Strigi {
  * BufferedStream will do the rest.
  */
 template <class T>
-class BufferedStream : public StreamBase<T> {
+class STREAMS_EXPORT BufferedStream : public StreamBase<T> {
 private:
     StreamBuffer<T> buffer;
     bool finishedWritingToBuffer;
-- 
1.8.4.2

