From 22d1b9923a61e0f890a39c80f661aeffaaeb2cb5 Mon Sep 17 00:00:00 2001
From: Aaron Seigo <aseigo@kde.org>
Date: Wed, 13 Feb 2013 17:29:52 +0100
Subject: [PATCH 2/5] order matters for systems that have things already
 installed

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7819ddf..6dee36b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,13 +119,13 @@ configure_file(${libstreamanalyzer_SOURCE_DIR}/strigi_thread.h.cmake
                ${libstreamanalyzer_BINARY_DIR}/include/strigi/strigi_thread.h)
 
 # every directory needs the headers that will be installed
-include_directories(${LIBSTREAMS_INCLUDE_DIRS}
-                    ${libstreamanalyzer_SOURCE_DIR}/include
+include_directories(${libstreamanalyzer_SOURCE_DIR}/include
                     ${libstreamanalyzer_SOURCE_DIR}/include/strigi
                     ${libstreamanalyzer_BINARY_DIR}/include
                     ${libstreamanalyzer_BINARY_DIR}/include/strigi
                     ${libstreamanalyzer_SOURCE_DIR}/lib
                     ${libstreamanalyzer_BINARY_DIR}/lib
+                    ${LIBSTREAMS_INCLUDE_DIRS}
                     ${ICONV_INCLUDE_DIR}
                     ${LIBSTREAMS_LIBRARY_DIRS}
                     ${EXIV2_INCLUDEDIR}
-- 
1.8.4.2

