diff -up strigi-0.7.5/libstreamanalyzer/lib/CMakeLists.txt.no_rpm strigi-0.7.5/libstreamanalyzer/lib/CMakeLists.txt
--- strigi-0.7.5/libstreamanalyzer/lib/CMakeLists.txt.no_rpm	2011-06-02 22:12:19.575180654 -0300
+++ strigi-0.7.5/libstreamanalyzer/lib/CMakeLists.txt	2011-06-02 22:12:32.658387695 -0300
@@ -39,7 +39,7 @@ set(streamanalyzer_SRCS
 	endanalyzers/oleendanalyzer.cpp
 	endanalyzers/pdfendanalyzer.cpp
 	endanalyzers/pngendanalyzer.cpp
-	endanalyzers/rpmendanalyzer.cpp
+#	endanalyzers/rpmendanalyzer.cpp
 	endanalyzers/sdfendanalyzer.cpp
 	endanalyzers/tarendanalyzer.cpp
 	endanalyzers/textendanalyzer.cpp
diff -up strigi-0.7.5/libstreamanalyzer/lib/streamanalyzer.cpp.no_rpm strigi-0.7.5/libstreamanalyzer/lib/streamanalyzer.cpp
--- strigi-0.7.5/libstreamanalyzer/lib/streamanalyzer.cpp.no_rpm	2011-06-02 22:13:03.468810501 -0300
+++ strigi-0.7.5/libstreamanalyzer/lib/streamanalyzer.cpp	2011-06-02 22:13:41.260743113 -0300
@@ -34,7 +34,7 @@
 #include "endanalyzers/zipexeendanalyzer.h"
 #include "endanalyzers/odfendanalyzer.h"
 #include "endanalyzers/oleendanalyzer.h"
-#include "endanalyzers/rpmendanalyzer.h"
+// #include "endanalyzers/rpmendanalyzer.h"
 #include "endanalyzers/cpioendanalyzer.h"
 #include "endanalyzers/pdfendanalyzer.h"
 #include "endanalyzers/sdfendanalyzer.h"
@@ -322,7 +322,7 @@ StreamAnalyzerPrivate::initializeEndFact
     addFactory(new OdfEndAnalyzerFactory());
     addFactory(new ZipEndAnalyzerFactory());
     addFactory(new ZipExeEndAnalyzerFactory());
-    addFactory(new RpmEndAnalyzerFactory());
+//    addFactory(new RpmEndAnalyzerFactory());
     addFactory(new CpioEndAnalyzerFactory());
     addFactory(new PngEndAnalyzerFactory());
     addFactory(new BmpEndAnalyzerFactory());
diff -up strigi-0.7.5/libstreams/lib/CMakeLists.txt.no_rpm strigi-0.7.5/libstreams/lib/CMakeLists.txt
--- strigi-0.7.5/libstreams/lib/CMakeLists.txt.no_rpm	2011-06-02 22:15:18.726938216 -0300
+++ strigi-0.7.5/libstreams/lib/CMakeLists.txt	2011-06-02 22:15:30.973324220 -0300
@@ -22,7 +22,7 @@ set(streams_SRCS
         lzmainputstream.cpp
 	mailinputstream.cpp
 	oleinputstream.cpp
-	rpminputstream.cpp
+#	rpminputstream.cpp
 	sdfinputstream.cpp
 #	signatureinputstream.cpp # not done yet
 	skippingfileinputstream.cpp
diff -up strigi-0.7.5/libstreams/lib/archivereader.cpp.no_rpm strigi-0.7.5/libstreams/lib/archivereader.cpp
--- strigi-0.7.5/libstreams/lib/archivereader.cpp.no_rpm	2011-06-02 22:14:05.469575448 -0300
+++ strigi-0.7.5/libstreams/lib/archivereader.cpp	2011-06-02 22:14:55.795832921 -0300
@@ -21,7 +21,7 @@
 #include <strigi/sdfinputstream.h>
 #include <strigi/tarinputstream.h>
 #include <strigi/mailinputstream.h>
-#include <strigi/rpminputstream.h>
+// #include <strigi/rpminputstream.h>
 #include <strigi/arinputstream.h>
 #include <strigi/zipinputstream.h>
 #include "archiveentrycache.h"
@@ -177,8 +177,8 @@ ArchiveReader::ArchiveReaderPrivate::Arc
 
     subs.insert(SubsPair(MailInputStream::checkHeader,
         MailInputStream::factory));
-    subs.insert(SubsPair(RpmInputStream::checkHeader,
-        RpmInputStream::factory));
+//    subs.insert(SubsPair(RpmInputStream::checkHeader,
+//        RpmInputStream::factory));
     subs.insert(SubsPair(ArInputStream::checkHeader,
         ArInputStream::factory));
     subs.insert(SubsPair(ZipInputStream::checkHeader,
diff -up strigi-0.7.5/libstreams/tests/api/CMakeLists.txt.no_rpm strigi-0.7.5/libstreams/tests/api/CMakeLists.txt
--- strigi-0.7.5/libstreams/tests/api/CMakeLists.txt.no_rpm	2011-06-02 22:16:41.409290030 -0300
+++ strigi-0.7.5/libstreams/tests/api/CMakeLists.txt	2011-06-02 22:16:53.096795455 -0300
@@ -13,7 +13,7 @@ set(streamtests
 	LZMAInputStreamTest.cpp
 	MailInputStreamTest.cpp
 	OleInputStreamTest.cpp
-	RpmInputStreamTest.cpp
+#	RpmInputStreamTest.cpp
 	SdfInputStreamTest.cpp
 #	SignatureInputStreamTest.cpp # disabled because that stream is not done
 	StringTerminatedSubStreamTest.cpp
