diff -up strigi-0.7.5/strigiutils/bin/xmlindexer/CMakeLists.txt.static strigi-0.7.5/strigiutils/bin/xmlindexer/CMakeLists.txt
--- strigi-0.7.5/strigiutils/bin/xmlindexer/CMakeLists.txt.static	2011-04-23 06:00:17.000000000 -0500
+++ strigi-0.7.5/strigiutils/bin/xmlindexer/CMakeLists.txt	2011-08-07 12:27:46.112100236 -0500
@@ -1,11 +1,11 @@
-add_library(librdfindexer
+add_library(librdfindexer STATIC
         rdfindexer.cpp
         tagmapping.cpp
 	rdfindexwriter.cpp
 )
 target_link_libraries(librdfindexer ${LIBSTREAMANALYZER_LIBRARIES} ${LIBSTREAMS_LIBRARIES})
 
-add_library(libxmlindexer
+add_library(libxmlindexer STATIC
         xmlindexer.cpp
         tagmapping.cpp
 	xmlindexwriter.cpp
