diff -up libnetfilter_cthelper-1.0.0/configure.ac.headers libnetfilter_cthelper-1.0.0/configure.ac
--- libnetfilter_cthelper-1.0.0/configure.ac.headers	2012-10-08 20:01:02.000000000 -0300
+++ libnetfilter_cthelper-1.0.0/configure.ac	2012-10-08 20:01:10.762920525 -0300
@@ -31,5 +31,5 @@ regular_CFLAGS="-Wall -Waggregate-return
 	-Wformat=2 -pipe"
 AC_SUBST([regular_CPPFLAGS])
 AC_SUBST([regular_CFLAGS])
-AC_CONFIG_FILES([Makefile src/Makefile include/Makefile include/libnetfilter_cthelper/Makefile include/linux/Makefile include/linux/netfilter/Makefile examples/Makefile libnetfilter_cthelper.pc doxygen.cfg])
+AC_CONFIG_FILES([Makefile src/Makefile include/Makefile include/libnetfilter_cthelper/Makefile examples/Makefile libnetfilter_cthelper.pc doxygen.cfg])
 AC_OUTPUT
diff -up libnetfilter_cthelper-1.0.0/include/Makefile.am.headers libnetfilter_cthelper-1.0.0/include/Makefile.am
--- libnetfilter_cthelper-1.0.0/include/Makefile.am.headers	2012-06-05 13:59:22.000000000 -0300
+++ libnetfilter_cthelper-1.0.0/include/Makefile.am	2012-10-08 19:59:23.978440727 -0300
@@ -1 +1 @@
-SUBDIRS = libnetfilter_cthelper linux
+SUBDIRS = libnetfilter_cthelper
