From a6a4b6fdb966ca408411d9002082f3a112b1c365 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Mon, 20 Oct 2014 15:19:44 +0200
Subject: [PATCH 6/7] tcpslice: don't test the pointer but pointee for NULL

---
 tcpslice-1.2a3/tcpslice.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tcpslice-1.2a3/tcpslice.c b/tcpslice-1.2a3/tcpslice.c
index 895e54f..a91439b 100644
--- a/tcpslice-1.2a3/tcpslice.c
+++ b/tcpslice-1.2a3/tcpslice.c
@@ -402,7 +402,9 @@ fill_tm(char *time_string, int is_delta, struct tm *t, time_t *usecs_addr)
 
 		while (isdigit(*t_stop))
 			++t_stop;
-		if (! t_stop)
+
+                if (!(*t_stop))
+                        /* we've reached end of string -> bad date format */
 			error("bad date format %s, problem starting at %s",
 			      time_string, t_start);
 
-- 
1.8.3.1

