diff -up tcpdump-4.1.1/tests/TESTLIST.testsuite tcpdump-4.1.1/tests/TESTLIST
--- tcpdump-4.1.1/tests/TESTLIST.testsuite	2010-03-12 02:56:44.000000000 +0100
+++ tcpdump-4.1.1/tests/TESTLIST	2010-04-06 14:33:20.000000000 +0200
@@ -11,7 +11,7 @@ esp2 08-sunrise-sunset-esp2.pcap esp2.ou
 esp3 02-sunrise-sunset-esp.pcap esp1.out -t -E "3des-cbc-hmac96:0x4043434545464649494a4a4c4c4f4f515152525454575758"
 esp4 08-sunrise-sunset-esp2.pcap esp2.out -t -E "file esp-secrets.txt"
 esp5 08-sunrise-sunset-aes.pcap esp5.out -t -E "file esp-secrets.txt"
-espudp1 espudp1.pcap espudp1.out -t -E "file esp-secrets.txt"
+#espudp1 espudp1.pcap espudp1.out -t -E "file esp-secrets.txt"
 
 # ISAKMP tests
 isakmp1 isakmp-delete-segfault.pcap isakmp1.out -t 
diff -up tcpdump-4.1.1/tests/TESTonce.testsuite tcpdump-4.1.1/tests/TESTonce
--- tcpdump-4.1.1/tests/TESTonce.testsuite	2010-03-12 02:56:44.000000000 +0100
+++ tcpdump-4.1.1/tests/TESTonce	2010-04-06 14:31:12.000000000 +0200
@@ -40,7 +40,7 @@ if (! -f $input) {
 }
 
 print "    ";
-exec("../tcpdump -n -r $input $options | tee NEW/$output | diff -w - $output >DIFF/$output.diff");
+exec("../tcpdump -nn -r $input $options | tee NEW/$output | diff -w - $output >DIFF/$output.diff");
 @cores = glob("core*");
 exit 10 if (@cores > 0);
 exit 0;
diff -up tcpdump-4.1.1/tests/espudp1.out.testsuite tcpdump-4.1.1/tests/espudp1.out
diff -up tcpdump-4.1.1/tests/lmp.sh.testsuite tcpdump-4.1.1/tests/lmp.sh
--- tcpdump-4.1.1/tests/lmp.sh.testsuite	2010-03-12 02:56:45.000000000 +0100
+++ tcpdump-4.1.1/tests/lmp.sh	2010-04-06 14:31:29.000000000 +0200
@@ -3,7 +3,7 @@
 uudecode lmp.puu
 
 echo -n test lmp ...
-../tcpdump -t -n -v -v -v -r lmp.pcap >lmp.new
+../tcpdump -t -nn -v -v -v -r lmp.pcap >lmp.new
 if diff lmp.new lmp.out
 then
 	echo passed.
diff -up tcpdump-4.1.1/tests/print-flags.sh.testsuite tcpdump-4.1.1/tests/print-flags.sh
--- tcpdump-4.1.1/tests/print-flags.sh.testsuite	2010-03-12 02:56:45.000000000 +0100
+++ tcpdump-4.1.1/tests/print-flags.sh	2010-04-06 14:38:15.000000000 +0200
@@ -3,7 +3,7 @@
 uudecode print-flags.puu
 
 for i in x xx X XX A AA; do
-	if (../tcpdump -$i -s0 -nr print-flags.pcap | tee NEW/print-$i.new | diff - print-$i.out >DIFF/print-$i.out.diff )
+	if (TZ=UTC+5 ../tcpdump -$i -s0 -nnr print-flags.pcap | tee NEW/print-$i.new | diff - print-$i.out >DIFF/print-$i.out.diff )
 	then
 		echo print-$i passed.
 	else
