diff -up tcpdump-4.2.0/tcpdump.1.in.droproot tcpdump-4.2.0/tcpdump.1.in
--- tcpdump-4.2.0/tcpdump.1.in.droproot	2011-07-24 17:58:55.000000000 -0300
+++ tcpdump-4.2.0/tcpdump.1.in	2011-11-27 21:17:34.298813596 -0200
@@ -214,6 +214,9 @@ have the name specified with the
 flag, with a number after it, starting at 1 and continuing upward.
 The units of \fIfile_size\fP are millions of bytes (1,000,000 bytes,
 not 1,048,576 bytes).
+
+Note that when used with \fB\-Z\fR option (enabled by default), privileges
+are dropped before opening first savefile.
 .TP
 .B \-d
 Dump the compiled packet-matching code in a human readable form to
@@ -646,7 +649,9 @@ but before opening any savefiles for out
 and the group ID to the primary group of
 .IR user .
 .IP
-This behavior can also be enabled by default at compile time.
+This behavior is enabled by default (\fB\-Z tcpdump\fR), and can
+be disabled by \fB\-Z root\fR.
+
 .IP "\fI expression\fP"
 .RS
 selects which packets will be dumped.
diff -up tcpdump-4.2.0/tcpdump.c.droproot tcpdump-4.2.0/tcpdump.c
--- tcpdump-4.2.0/tcpdump.c.droproot	2011-07-24 17:58:55.000000000 -0300
+++ tcpdump-4.2.0/tcpdump.c	2011-11-27 21:19:44.426171476 -0200
@@ -1032,7 +1032,7 @@ main(int argc, char **argv)
 
 #ifdef WITH_CHROOT
 	/* if run as root, prepare for chrooting */
-	if (getuid() == 0 || geteuid() == 0) {
+	if (Cflag == 0 && (getuid() == 0 || geteuid() == 0)) {
 		/* future extensibility for cmd-line arguments */
 		if (!chroot_dir)
 			chroot_dir = WITH_CHROOT;
@@ -1041,7 +1041,7 @@ main(int argc, char **argv)
 
 #ifdef WITH_USER
 	/* if run as root, prepare for dropping root privileges */
-	if (getuid() == 0 || geteuid() == 0) {
+	if (Cflag == 0 && (getuid() == 0 || geteuid() == 0)) {
 		/* Run with '-Z root' to restore old behaviour */ 
 		if (!username)
 			username = WITH_USER;
@@ -1281,12 +1281,17 @@ main(int argc, char **argv)
 	 * Switching to the -Z user ID only after opening the first
 	 * savefile doesn't handle the general case.
 	 */
-	if (getuid() == 0 || geteuid() == 0) {
+	if (Cflag == 0 && (getuid() == 0 || geteuid() == 0)) {
 		if (username || chroot_dir)
 			droproot(username, chroot_dir);
 	}
 #endif /* WIN32 */
 
+	if (Cflag != 0 && (getuid() == 0 || geteuid() == 0)) {
+		if (username || chroot_dir)
+			droproot(username, chroot_dir);
+	}
+
 	if (pcap_setfilter(pd, &fcode) < 0)
 		error("%s", pcap_geterr(pd));
 	if (WFileName) {
