diff -up tcpdump/print-icmp6.c.icmp6msec tcpdump/print-icmp6.c
--- tcpdump/print-icmp6.c.icmp6msec	2009-08-19 00:02:37.000000000 +0200
+++ tcpdump/print-icmp6.c	2009-08-20 16:33:59.000000000 +0200
@@ -397,7 +397,7 @@ icmp6_print(const u_char *bp, u_int leng
 			p = (struct nd_router_advert *)dp;
 			TCHECK(p->nd_ra_retransmit);
 			printf("\n\thop limit %u, Flags [%s]" \
-                               ", pref %s, router lifetime %us, reachable time %us, retrans time %us",
+                               ", pref %s, router lifetime %us, reachable time %ums, retrans time %ums",
                                (u_int)p->nd_ra_curhoplimit,
                                bittok2str(icmp6_opt_ra_flag_values,"none",(p->nd_ra_flags_reserved)),
                                get_rtpref(p->nd_ra_flags_reserved),
