diff -up xsel-1.2.0/xsel.c~ xsel-1.2.0/xsel.c
--- xsel-1.2.0/xsel.c~	2012-04-05 16:18:33.000000000 +0200
+++ xsel-1.2.0/xsel.c	2012-04-05 16:30:15.833900611 +0200
@@ -505,7 +505,8 @@ get_append_property (XSelectionEvent * x
 
   debug_property (D_TRACE, xsl->requestor, xsl->property, target, length);
 
-  if (target != XA_STRING) {
+  if (target != utf8_atom && target != XA_STRING &&
+      target != compound_text_atom) {
     print_debug (D_OBSC, "target %s not XA_STRING in get_append_property()",
                  get_atom_name (target));
     free (*buffer);
