diff -up ./src/plugin.cpp.static-init-crash ./src/plugin.cpp
--- ./src/plugin.cpp.static-init-crash	2011-03-16 16:12:23.000000000 +0100
+++ ./src/plugin.cpp	2011-03-16 16:12:45.000000000 +0100
@@ -19,7 +19,7 @@
 
 #include "tokenhelpdialog.h"
 
-const QString Plugin::S_TOKEN_SEPARATOR = TokenHelpDialog::S_TOKEN_SEPARATOR;
+const QString Plugin::S_TOKEN_SEPARATOR = ";;";
 
 Plugin::Plugin( PluginLoader* loader )
     : m_pluginLoader( loader ), m_enabled( false )
