Index: clamav-0.96.2/clamd/dazukoio_compat12.c
===================================================================
--- clamav-0.96.2.orig/clamd/dazukoio_compat12.c
+++ clamav-0.96.2/clamd/dazukoio_compat12.c
@@ -89,7 +89,7 @@ int dazukoRegister_TS_compat12(struct da
 	if (dazuko->device < 0)
 	{
 
-		dazuko->device = open("/dev/dazuko", 0);
+		dazuko->device = open("/dev/dazuko", O_RDONLY);
 		if (dazuko->device < 0)
 			return -1;
 
