From 900b2fb63dac15fe1013d9e326a7a5c8d4da1cca Mon Sep 17 00:00:00 2001
From: Tim-Philipp Müller <tim.muller@collabora.co.uk>
Date: Mon, 23 May 2011 12:25:44 +0000
Subject: flacdec: also try upstream first for duration query in DEFAULT format

https://bugzilla.gnome.org/show_bug.cgi?id=650785
---
diff --git a/ext/flac/gstflacdec.c b/ext/flac/gstflacdec.c
index 4957129..2e4085d 100644
--- a/ext/flac/gstflacdec.c
+++ b/ext/flac/gstflacdec.c
@@ -1724,8 +1724,9 @@ gst_flac_dec_src_query (GstPad * pad, GstQuery * query)
 
       gst_query_parse_duration (query, &fmt, NULL);
 
-      /* try any demuxers before us first */
-      if (fmt == GST_FORMAT_TIME && peer && gst_pad_query (peer, query)) {
+      /* try any demuxers or parsers before us first */
+      if ((fmt == GST_FORMAT_TIME || fmt == GST_FORMAT_DEFAULT) &&
+          peer != NULL && gst_pad_query (peer, query)) {
         gst_query_parse_duration (query, NULL, &len);
         GST_DEBUG_OBJECT (flacdec, "peer returned duration %" GST_TIME_FORMAT,
             GST_TIME_ARGS (len));
--
cgit v0.9.0.2-2-gbebe

