From 15017b22ac829fb179e6bbb1a11ffa2a055c4ba5 Mon Sep 17 00:00:00 2001
From: Eike Hein <hein@kde.org>
Date: Fri, 19 Oct 2012 00:52:46 +0200
Subject: [PATCH 01/56] Adjust for next dev cycle.

---
 app/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/main.cpp b/app/main.cpp
index e384682..ecfe94d 100644
--- a/app/main.cpp
+++ b/app/main.cpp
@@ -34,7 +34,7 @@ int main (int argc, char *argv[])
     KAboutData aboutData("yakuake",
         0,
         ki18nc("@title", "<application>Yakuake</application>"),
-        "2.9.9",
+        "2.9.9+",
         ki18nc("@title", "A drop-down terminal emulator based on KDE Konsole technology."),
         KAboutData::License_GPL,
         ki18nc("@info:credit", "(c) 2008-2012 The Yakuake Team"),
-- 
1.9.3

