From 5a0e8f8bcf549459d431ab36244eb2223a7cc732 Mon Sep 17 00:00:00 2001
From: Jekyll Wu <adaptee@gmail.com>
Date: Tue, 6 Nov 2012 20:46:08 +0800
Subject: [PATCH 05/56] Fix the small typo and thus the warning

---
 app/config/behaviorsettings.ui | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/config/behaviorsettings.ui b/app/config/behaviorsettings.ui
index 3759f7b..ca7176a 100644
--- a/app/config/behaviorsettings.ui
+++ b/app/config/behaviorsettings.ui
@@ -165,7 +165,7 @@
   <tabstop>kcfg_KeepOpen</tabstop>
   <tabstop>kcfg_KeepAbove</tabstop>
   <tabstop>kcfg_ToggleToFocus</tabstop>
-  <tabstop>kcg_KeepOpenAfterLastSessionCloses</tabstop>
+  <tabstop>kcfg_KeepOpenAfterLastSessionCloses</tabstop>
   <tabstop>kcfg_FocusFollowsMouse</tabstop>
   <tabstop>kcfg_ConfirmQuit</tabstop>
  </tabstops>
-- 
1.9.3

