From 9a3cc71209191aba129b15c19d8b617c6af028f7 Mon Sep 17 00:00:00 2001
From: Eike Hein <hein@kde.org>
Date: Wed, 21 Aug 2013 19:37:17 +0200
Subject: [PATCH 21/56] Change default to appear on the screen the pointer is
 on.

---
 app/config/yakuake.kcfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/config/yakuake.kcfg b/app/config/yakuake.kcfg
index a71e9b3..a6d34c0 100644
--- a/app/config/yakuake.kcfg
+++ b/app/config/yakuake.kcfg
@@ -8,7 +8,7 @@
     <entry name="Screen" type="Int">
       <label context="@label">Screen to use</label>
       <whatsthis context="@info:whatsthis">The screen on which the application window will appear. 0 is understood to be the screen the mouse pointer is on.</whatsthis>
-      <default>1</default>
+      <default>0</default>
     </entry>
     <entry name="Width" type="Int">
       <label context="@label">Window width</label>
-- 
1.9.3

