From 5e31ccfdbc3a46e6bffd79fc6bcafc3a2bde0ed3 Mon Sep 17 00:00:00 2001
From: Vangelis Tasoulas <vangelis@tasoulas.net>
Date: Sun, 23 Nov 2014 00:16:11 +0100
Subject: [PATCH 41/56] Fixes incorrect Yakuake height handling on dualhead
 setups.

BUG:310155
---
 app/mainwindow.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/app/mainwindow.cpp b/app/mainwindow.cpp
index ca5df80..c11de41 100644
--- a/app/mainwindow.cpp
+++ b/app/mainwindow.cpp
@@ -1269,8 +1269,6 @@ QRect MainWindow::getDesktopGeometry()
     if (action->isChecked())
         return screenGeometry;
 
-    int currentDesktop = KWindowSystem::windowInfo(winId(), NET::WMDesktop).desktop();
-
     if (KApplication::desktop()->numScreens() > 1)
     {
         const QList<WId> allWindows = KWindowSystem::windows();
@@ -1286,7 +1284,9 @@ QRect MainWindow::getDesktopGeometry()
             {
                 KWindowInfo windowInfo = KWindowSystem::windowInfo(windowId, NET::WMDesktop, NET::WM2ExtendedStrut);
 
-                if (windowInfo.valid() && (windowInfo.desktop() == currentDesktop || windowInfo.desktop() == -1))
+                // If windowInfo is valid and the window is located at the same (current)
+                // desktop with the yakuake window...
+                if (windowInfo.valid() && windowInfo.isOnCurrentDesktop())
                 {
                     NETExtendedStrut strut = windowInfo.extendedStrut();
 
@@ -1316,10 +1316,10 @@ QRect MainWindow::getDesktopGeometry()
             }
         }
 
-        return KWindowSystem::workArea(offScreenWindows, currentDesktop).intersect(screenGeometry);
+        return KWindowSystem::workArea(offScreenWindows).intersect(screenGeometry);
     }
 
-    return KWindowSystem::workArea(currentDesktop);
+    return KWindowSystem::workArea();
 }
 
 void MainWindow::whatsThis()
-- 
1.9.3

