diff -ur libindi-0.6.2/CMakeLists.txt libindi-0.6.2.new/CMakeLists.txt
--- libindi-0.6.2/CMakeLists.txt	2010-07-20 06:40:26.000000000 +0200
+++ libindi-0.6.2.new/CMakeLists.txt	2010-07-29 12:23:34.079146062 +0200
@@ -115,7 +115,7 @@
 ######## INDI Static Library #########
 ######################################
 add_library(indidriver STATIC ${indimain_SRCS})
-install(TARGETS indidriver ARCHIVE DESTINATION lib${LIB_POSTFIX})
+install(TARGETS indidriver ARCHIVE DESTINATION lib${LIB_SUFFIX})
 
 ######################################
 ######## INDI Shared Library #########
@@ -133,7 +133,7 @@
 target_link_libraries(indi ${FLI_LIBRARIES})
 endif(FLI_FOUND)
 
-install(TARGETS indi LIBRARY DESTINATION lib${LIB_POSTFIX})
+install(TARGETS indi LIBRARY DESTINATION lib${LIB_SUFFIX})
 set_target_properties(indi PROPERTIES VERSION ${CMAKE_INDI_VERSION_STRING} SOVERSION ${INDI_SOVERSION})
 
 #####################################
