diff -ur libindi-0.9.8/libs/indibase/alignment/CMakeLists.txt libindi-0.9.8.fix/libs/indibase/alignment/CMakeLists.txt
--- libindi-0.9.8/libs/indibase/alignment/CMakeLists.txt	2014-03-30 17:20:23.000000000 +0200
+++ libindi-0.9.8.fix/libs/indibase/alignment/CMakeLists.txt	2014-04-23 23:58:32.619150626 +0200
@@ -62,7 +62,7 @@
 
 add_library(AlignmentClient ${AlignmentClient_SRCS})
 set_target_properties(AlignmentClient PROPERTIES OUTPUT_NAME indiAlignmentClient)
-install(TARGETS AlignmentClient ARCHIVE DESTINATION ${LIB_DESTINATION})
+install(TARGETS AlignmentClient ARCHIVE DESTINATION ${LIB_DESTINATION} LIBRARY DESTINATION ${LIB_DESTINATION})
 
 ##################################################
 ############ LoaderCLient test program ###########
