--- CMakeLists.txt.cfits	2010-05-23 23:25:46.655368179 -0300
+++ CMakeLists.txt	2010-05-23 23:26:11.734558545 -0300
@@ -31,13 +31,14 @@
 include (MacroLogFeature)
 include (MacroBoolTo01)
 include (CheckIncludeFiles)
+include (FindPkgConfig)
 
 find_package(ZLIB REQUIRED)
 
 macro_optional_find_package(USB)
 macro_log_feature(LIBUSB_FOUND "libusb" "User level access to USB devices" "http://indi.sf.net" FALSE "" "Provides support for USB based drivers in INDI.")
 
-macro_optional_find_package(CFITSIO)
+pkg_check_modules(CFITSIO REQUIRED cfitsio)
 
 if (NOT CFITSIO_FOUND OR CFITSIO_VERSION_MAJOR LESS 3)
   message(FATAL_ERROR "CFITSIO version too old, Please install cfitsio 3.x and try again. http://indi.sf.net")
@@ -66,7 +67,7 @@
 include_directories( ${CMAKE_SOURCE_DIR}/libs/webcam)
 
 if (CFITSIO_FOUND)
-  include_directories(${CFITSIO_INCLUDE_DIR})
+  include_directories(${CFITSIO_INCLUDE_DIRS})
 endif (CFITSIO_FOUND)
 
 if (NOVA_FOUND)
