--- kdepimlibs-4.7.3/kimap/loginjob.cpp.ark	2011-11-08 00:15:20.339317872 +0100
+++ kdepimlibs-4.7.3/kimap/loginjob.cpp	2011-11-08 00:19:51.685974706 +0100
@@ -323,11 +323,7 @@ void LoginJob::handleResponse( const Mes
       }
     }
   } else if ( response.content.size() >= 2 ) {
-    if ( d->authState == LoginJobPrivate::Authenticate ) {
-      if (!d->answerChallenge(QByteArray::fromBase64(response.content[1].toString()))) {
-        emitResult(); //error, we're done
-      }
-    } else if ( response.content[1].toString()=="CAPABILITY" ) {
+    if ( response.content[1].toString()=="CAPABILITY" ) {
       bool authModeSupported = d->authMode.isEmpty();
       for (int i = 2; i < response.content.size(); ++i) {
         QString capability = response.content[i].toString();
@@ -346,6 +342,10 @@ void LoginJob::handleResponse( const Mes
         setErrorText( i18n("Login failed, authentication mode %1 is not supported by the server.", d->authMode) );
         d->authState = LoginJobPrivate::Login; //just to treat the upcoming OK correctly
       }
+    } else if ( d->authState == LoginJobPrivate::Authenticate ) {
+      if (!d->answerChallenge(QByteArray::fromBase64(response.content[1].toString()))) {
+        emitResult(); //error, we're done
+      }
     }
   }
 }
