Index: kdepimlibs/akonadi/CMakeLists.txt
===================================================================
--- kdepimlibs/akonadi/CMakeLists.txt	(révision 1088150)
+++ kdepimlibs/akonadi/CMakeLists.txt	(révision 1088151)
@@ -171,6 +171,7 @@
   subscriptiondialog.ui
   controlprogressindicator.ui
   selftestdialog.ui
+  erroroverlay.ui
 )
 
 kde4_add_library( akonadi-kde SHARED ${akonadikde_LIB_SRC} )
