Index: kdepimlibs/akonadi/agentmanager_p.h
===================================================================
--- kdepimlibs/akonadi/agentmanager_p.h	(révision 1088320)
+++ kdepimlibs/akonadi/agentmanager_p.h	(révision 1088321)
@@ -76,7 +76,7 @@
     void synchronize( const AgentInstance& );
     void synchronizeCollectionTree( const AgentInstance& );
 
-    void serviceOwnderChanged( const QString &name, const QString &oldOwner, const QString &newOwner );
+    void serviceOwnerChanged( const QString &name, const QString &oldOwner, const QString &newOwner );
 
     AgentType fillAgentType( const QString &identifier ) const;
     AgentInstance fillAgentInstance( const QString &identifier ) const;
Index: kdepimlibs/akonadi/agentmanager.cpp
===================================================================
--- kdepimlibs/akonadi/agentmanager.cpp	(révision 1088320)
+++ kdepimlibs/akonadi/agentmanager.cpp	(révision 1088321)
@@ -264,7 +264,7 @@
   return instance;
 }
 
-void AgentManagerPrivate::serviceOwnderChanged( const QString &service, const QString &oldOwner, const QString &newOwner )
+void AgentManagerPrivate::serviceOwnerChanged( const QString &service, const QString &oldOwner, const QString &newOwner )
 {
   Q_UNUSED( newOwner );
   if ( service == QLatin1String( AKONADI_DBUS_CONTROL_SERVICE ) && oldOwner.isEmpty() )
Index: kdepimlibs/akonadi/agentmanager.h
===================================================================
--- kdepimlibs/akonadi/agentmanager.h	(révision 1088320)
+++ kdepimlibs/akonadi/agentmanager.h	(révision 1088321)
@@ -198,6 +198,7 @@
     Q_PRIVATE_SLOT( d, void agentInstanceWarning( const QString&, const QString& ) )
     Q_PRIVATE_SLOT( d, void agentInstanceError( const QString&, const QString& ) )
     Q_PRIVATE_SLOT( d, void agentInstanceOnlineChanged( const QString&, bool ) )
+    Q_PRIVATE_SLOT( d, void serviceOwnerChanged(const QString&, const QString&, const QString&) )
     //@endcond
 };
 
