Index: kdepimlibs/akonadi/selftestdialog.cpp
===================================================================
--- kdepimlibs/akonadi/selftestdialog.cpp	(révision 1088256)
+++ kdepimlibs/akonadi/selftestdialog.cpp	(révision 1088257)
@@ -86,6 +86,7 @@
   connect( this, SIGNAL(user1Clicked()), SLOT(saveReport()) );
   connect( this, SIGNAL(user2Clicked()), SLOT(copyReport()) );
 
+  connect( ServerManager::self(), SIGNAL(stateChanged(ServerManager::State)), SLOT(runTests()) );
   runTests();
 }
 
@@ -133,6 +134,8 @@
 
 void SelfTestDialog::runTests()
 {
+  mTestModel->clear();
+
   const QString driver = serverSetting( QLatin1String("General"), "Driver", QLatin1String("QMYSQL") ).toString();
   testSQLDriver();
   if (driver == QLatin1String( "QPSQL" )) {
Index: kdepimlibs/akonadi/selftestdialog_p.h
===================================================================
--- kdepimlibs/akonadi/selftestdialog_p.h	(révision 1088256)
+++ kdepimlibs/akonadi/selftestdialog_p.h	(révision 1088257)
@@ -62,6 +62,7 @@
     void saveReport();
     void copyReport();
     void linkActivated( const QString &link );
+    void runTests();
 
   private:
     enum ResultType {
@@ -71,7 +72,6 @@
       Error
     };
     QStandardItem* report( ResultType type, const KLocalizedString &summary, const KLocalizedString &details );
-    void runTests();
     QVariant serverSetting( const QString &group, const char *key, const QVariant &def ) const;
     bool useStandaloneMysqlServer() const;
     bool runProcess( const QString &app, const QStringList &args, QString &result ) const;
