Index: kdepimlibs/kcal/icalformat_p.cpp
===================================================================
--- kdepimlibs/kcal/icalformat_p.cpp	(révision 1122721)
+++ kdepimlibs/kcal/icalformat_p.cpp	(révision 1122722)
@@ -784,11 +784,7 @@
   if ( att->isUri() ) {
     attach = icalattach_new_from_url( att->uri().toUtf8().data() );
   } else {
-#if ( LIBICAL_MAJOR_VERSION == 0 ) && ( LIBICAL_MINOR_VERSION < 45 )
     attach = icalattach_new_from_data ( ( unsigned char * )att->data(), 0, 0 );
-#else
-    attach = icalattach_new_from_data ( att->data(), 0, 0 );
-#endif
   }
   icalproperty *p = icalproperty_new_attach( attach );
 
Index: kdepimlibs/kcal/CMakeLists.txt
===================================================================
--- kdepimlibs/kcal/CMakeLists.txt	(révision 1122721)
+++ kdepimlibs/kcal/CMakeLists.txt	(révision 1122722)
@@ -3,8 +3,6 @@
 set(LIBICAL_MIN_VERSION "0.42")
 find_package(Libical)
 macro_log_feature(LIBICAL_FOUND "libical" "Reference implementation of the iCalendar data type and serialization format" "http://sourceforge.net/projects/freeassociation" TRUE "${LIBICAL_MIN_VERSION}" "Required for the critical PIM kcal library.")
-add_definitions(-DLIBICAL_MAJOR_VERSION=${LIBICAL_MAJOR_VERSION}
-                -DLIBICAL_MINOR_VERSION=${LIBICAL_MINOR_VERSION})
 
 add_definitions(-DKDE_DEFAULT_DEBUG_AREA=5800)
 
