diff -up src/Makefile.jansson src/Makefile
--- src/Makefile.jansson	2014-04-08 18:33:57.140800824 -0300
+++ src/Makefile	2014-04-08 18:36:25.095248747 -0300
@@ -60,15 +60,14 @@ ifndef VDPAU_CFLAGS
 endif
 
 ifndef JANSSON_CFLAGS
-  JANSSON_CFLAGS  = -Wno-cast-qual
-  JANSSON_CFLAGS += -Wno-strict-prototypes
-  JANSSON_CFLAGS += -Wno-unused-function
-  JANSSON_CFLAGS += -DHAVE_CONFIG_H
-  ifeq ($(TARGET_ARCH),armv7l)
-    JANSSON_CFLAGS += -Wno-unused-but-set-variable
+  JANSSON_CFLAGS        := $(shell $(PKG_CONFIG) --cflags jansson 2> /dev/null)
+  ifeq ($(VDPAU_CFLAGS),)
+    JANSSON_CFLAGS       = -I /usr/include
   endif
 endif
 
+JANSSON_LIBS            := $(shell $(PKG_CONFIG) --libs jansson 2> /dev/null)
+
 ##############################################################################
 # The XF86Config-parser, libXNVCtrl, and common-utils directories may
 # be in one of two places: either elsewhere in the driver source tree
@@ -157,7 +156,6 @@ CFLAGS     += -I $(XCONFIG_PARSER_DIR)/.
 CFLAGS     += -I libXNVCtrlAttributes
 CFLAGS     += -I xpm_data
 CFLAGS     += -I gtk+-2.x
-CFLAGS     += -I jansson
 CFLAGS     += -I $(COMMON_UTILS_DIR)
 CFLAGS     += -I $(VIRTUAL_RESOLUTIONS_DIR)
 CFLAGS     += -I $(OUTPUTDIR)
@@ -166,8 +164,6 @@ CFLAGS     += -DPROGRAM_NAME=\"nvidia-se
 
 $(call BUILD_OBJECT_LIST,$(GTK_SRC)): CFLAGS += $(GTK_CFLAGS)
 
-$(call BUILD_OBJECT_LIST,$(JANSSON_SRC)): CFLAGS += $(JANSSON_CFLAGS)
-
 
 ##############################################################################
 # build rules
@@ -185,7 +181,7 @@ NVIDIA_SETTINGS_install: $(NVIDIA_SETTIN
 
 $(NVIDIA_SETTINGS): $(OBJS)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) -o $@ $(OBJS) \
-	    $(XNVCTRL_LIB) $(LIBS)
+	    $(XNVCTRL_LIB) $(JANSSON_LIBS) $(LIBS)
 	$(call quiet_cmd,STRIP_CMD) $@
 
 # define the rule to build each object file
diff -up src/src.mk.jansson src/src.mk
--- src/src.mk.jansson	2014-04-08 18:36:45.993908300 -0300
+++ src/src.mk	2014-04-08 18:36:54.177207388 -0300
@@ -269,33 +269,5 @@ GTK_EXTRA_DIST += gtk+-2.x/ctkvdpau.h
 
 NVIDIA_SETTINGS_EXTRA_DIST += $(GTK_EXTRA_DIST)
 
-#
-# files in the src/jansson directory of nvidia-settings
-#
-JANSSON_SRC += jansson/dump.c
-JANSSON_SRC += jansson/error.c
-JANSSON_SRC += jansson/hashtable.c
-JANSSON_SRC += jansson/hashtable_seed.c
-JANSSON_SRC += jansson/load.c
-JANSSON_SRC += jansson/memory.c
-JANSSON_SRC += jansson/pack_unpack.c
-JANSSON_SRC += jansson/strbuffer.c
-JANSSON_SRC += jansson/strconv.c
-JANSSON_SRC += jansson/utf.c
-JANSSON_SRC += jansson/value.c
-
-NVIDIA_SETTINGS_SRC += $(JANSSON_SRC)
-
-JANSSON_EXTRA_DIST += jansson/hashtable.h
-JANSSON_EXTRA_DIST += jansson/jansson_config.h
-JANSSON_EXTRA_DIST += jansson/jansson.h
-JANSSON_EXTRA_DIST += jansson/jansson_private_config.h
-JANSSON_EXTRA_DIST += jansson/jansson_private.h
-JANSSON_EXTRA_DIST += jansson/lookup3.h
-JANSSON_EXTRA_DIST += jansson/strbuffer.h
-JANSSON_EXTRA_DIST += jansson/utf.h
-
-NVIDIA_SETTINGS_EXTRA_DIST += $(JANSSON_EXTRA_DIST)
-
 NVIDIA_SETTINGS_DIST_FILES += $(NVIDIA_SETTINGS_SRC)
 NVIDIA_SETTINGS_DIST_FILES += $(NVIDIA_SETTINGS_EXTRA_DIST)
