--- src/Makefile.orig	2012-04-27 16:03:52.000000000 -0300
+++ src/Makefile	2012-05-07 17:48:18.482794255 -0300
@@ -34,9 +34,9 @@ include $(UTILS_MK_DIR)/utils.mk
 
 ifndef X_LDFLAGS
   ifeq ($(TARGET_OS)-$(TARGET_ARCH),Linux-x86_64)
-    X_LDFLAGS          = -L/usr/X11R6/lib64
+    X_LDFLAGS          = -L/usr/lib64
   else
-    X_LDFLAGS          = -L/usr/X11R6/lib
+    X_LDFLAGS          = -L/usr/lib
   endif
 endif
 
@@ -67,7 +67,7 @@ endif
 ##############################################################################
 
 XNVCTRL_DIR           ?= libXNVCtrl
-XNVCTRL_ARCHIVE       ?= $(XNVCTRL_DIR)/libXNVCtrl.a
+XNVCTRL_LIB           ?= -lXNVCtrl
 XCONFIG_PARSER_DIR    ?= XF86Config-parser
 COMMON_UTILS_DIR      ?= common-utils
 
@@ -113,7 +113,7 @@ endif
 # $(OBJECTS) on the link commandline, causing libraries for linking to
 # be named after the objects that depend on those libraries (needed
 # for "--as-needed" linker behavior).
-LIBS += -lX11 -lXext -lm $(LIBDL_LIBS) $(GTK_LDFLAGS)
+LIBS += -lX11 -lm $(LIBDL_LIBS) $(GTK_LDFLAGS)
 
 # Include all the source lists; dist-files.mk will define NVIDIA_SETTINGS_SRC
 include src.mk
@@ -157,9 +157,9 @@ NVIDIA_SETTINGS_install: $(NVIDIA_SETTIN
 	$(MKDIR) $(BINDIR)
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
-$(NVIDIA_SETTINGS): $(OBJS) $(XNVCTRL_ARCHIVE)
+$(NVIDIA_SETTINGS): $(OBJS)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) -o $@ $(OBJS) \
-	    $(XNVCTRL_ARCHIVE) $(LIBS)
+	    $(XNVCTRL_LIB) $(LIBS)
 	$(call quiet_cmd,STRIP_CMD) $@
 
 # define the rule to build each object file
