--- src/Makefile.orig	2014-04-04 19:36:17.000000000 -0300
+++ src/Makefile	2014-04-08 18:26:15.580215961 -0300
@@ -34,9 +34,9 @@ include $(UTILS_MK_DIR)/utils.mk
 
 ifndef X_LDFLAGS
   ifeq ($(TARGET_OS)-$(TARGET_ARCH),Linux-x86_64)
-    X_LDFLAGS          = -L/usr/X11R6/lib64
+    X_LDFLAGS          = -L/usr/lib64
   else
-    X_LDFLAGS          = -L/usr/X11R6/lib
+    X_LDFLAGS          = -L/usr/lib
   endif
 endif
 
@@ -82,7 +82,7 @@ endif
 
 XNVCTRL_DIR             ?= libXNVCtrl
 XNVCTRL_MAKEFILE        ?= Makefile
-XNVCTRL_ARCHIVE         ?= $(XNVCTRL_DIR)/libXNVCtrl.a
+XNVCTRL_LIB             ?= -lXNVCtrl
 XCONFIG_PARSER_DIR      ?= XF86Config-parser
 COMMON_UTILS_DIR        ?= common-utils
 COMMON_UNIX_DIR         ?= common-unix
@@ -131,7 +131,7 @@ endif
 # $(OBJECTS) on the link commandline, causing libraries for linking to
 # be named after the objects that depend on those libraries (needed
 # for "--as-needed" linker behavior).
-LIBS += -lX11 -lXext -lm $(LIBDL_LIBS) $(GTK_LDFLAGS)
+LIBS += -lX11 -lm $(LIBDL_LIBS) $(GTK_LDFLAGS)
 
 # Include all the source lists; dist-files.mk will define NVIDIA_SETTINGS_SRC
 include src.mk
@@ -183,9 +183,9 @@ NVIDIA_SETTINGS_install: $(NVIDIA_SETTIN
 	$(MKDIR) $(BINDIR)
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(BINDIR)/$(notdir $<)
 
-$(NVIDIA_SETTINGS): $(OBJS) $(XNVCTRL_ARCHIVE)
+$(NVIDIA_SETTINGS): $(OBJS)
 	$(call quiet_cmd,LINK) $(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS) -o $@ $(OBJS) \
-	    $(XNVCTRL_ARCHIVE) $(LIBS)
+	    $(XNVCTRL_LIB) $(LIBS)
 	$(call quiet_cmd,STRIP_CMD) $@
 
 # define the rule to build each object file
