Fix bug #460734.


diff -Naur libdbi-drivers-0.8.3/drivers/sqlite3/dbd_sqlite3.c libdbi-drivers-0.8.3-1/drivers/sqlite3/dbd_sqlite3.c
--- libdbi-drivers-0.8.3/drivers/sqlite3/dbd_sqlite3.c	2008-01-15 09:07:24.000000000 -0500
+++ libdbi-drivers-0.8.3-1/drivers/sqlite3/dbd_sqlite3.c	2008-03-03 12:32:26.000000000 -0500
@@ -731,7 +731,7 @@
 				      &errmsg);
 
   if (query_res || !table_numrows) {
-    _dbi_internal_error_handler(conn, NULL, DBI_ERROR_BADNAME);
+    _dbd_internal_error_handler(conn, NULL, DBI_ERROR_BADNAME);
     /*       printf("field not found\n"); */
     return 0;
   }
