--- ./scripts/init.d/cgred.in.orig	2011-03-08 05:29:42.000000000 -0600
+++ ./scripts/init.d/cgred.in	2013-02-23 16:15:54.054745738 -0600
@@ -39,9 +39,8 @@
 # Sanity checks
 [ -x $CGRED_BIN ] || exit 1
 
-# Source function library & LSB routines
+# Source function library
 . /etc/rc.d/init.d/functions
-. /lib/lsb/init-functions
 
 # Read in configuration options.
 if [ -f "/etc/sysconfig/cgred.conf" ] ; then
@@ -70,16 +69,16 @@
 {
 	echo -n $"Starting CGroup Rules Engine Daemon: "
 	if [ -f "$lockfile" ]; then
-		log_failure_msg "$servicename is already running with PID `cat ${pidfile}`"
+		echo "$servicename is already running with PID `cat ${pidfile}`"
 		return 0
 	fi
 	if [ ! -s $CGRED_CONF ]; then
-		log_failure_msg "not configured"
+		echo "not configured"
 		return 6
 	fi
 	if ! grep "^cgroup" /proc/mounts &>/dev/null; then
 		echo
-		log_failure_msg $"Cannot find cgroups, is cgconfig service running?"
+		echo $"Cannot find cgroups, is cgconfig service running?"
 		return 1
 	fi
 	daemon --check $servicename --pidfile $pidfile $CGRED_BIN $OPTIONS
@@ -100,7 +99,7 @@
 {
 	echo -n $"Stopping CGroup Rules Engine Daemon..."
 	if [ ! -f $pidfile ]; then
-		log_success_msg
+		#log_success_msg
 		return 0
 	fi
 	killproc -p $pidfile -TERM "$processname"
@@ -146,13 +145,13 @@
 			echo $"Reloading rules configuration..."
 			kill -s 12 `cat ${pidfile}`
 			RETVAL=$?
-			if [ $RETVAL -eq 0 ] ; then
-				log_success_msg
-			else
-				log_failure_msg
-			fi
+			#if [ $RETVAL -eq 0 ] ; then
+			#	log_success_msg
+			#else
+			#	log_failure_msg
+			#fi
 		else
-			log_failure_msg "$servicename is not running."
+			echo "$servicename is not running."
 		fi
 		;;
 	*)
--- ./scripts/init.d/cgconfig.in.orig	2011-12-06 03:42:15.000000000 -0600
+++ ./scripts/init.d/cgconfig.in	2013-02-23 16:23:51.530775095 -0600
@@ -36,11 +36,6 @@
 servicename=cgconfig
 lockfile=/var/lock/subsys/$servicename
 
-#
-# Source LSB routines
-#
-. /lib/lsb/init-functions
-
 # read the config
 CREATE_DEFAULT=yes
 if [ -e /etc/sysconfig/cgconfig ]; then
@@ -54,8 +49,8 @@
 	    grep -m1 '^\*[[:space:]]\+' /etc/cgrules.conf | \
 		    read user ctrl defaultcgroup
             if [ -n "$defaultcgroup" -a "$defaultcgroup" = "*" ]; then
-                log_warning_msg "/etc/cgrules.conf incorrect"
-                log_warning_msg "Overriding it"
+                echo "/etc/cgrules.conf incorrect"
+                echo "Overriding it"
                 defaultcgroup=
             fi
         fi
@@ -98,20 +93,20 @@
 start() {
         echo -n "Starting cgconfig service: "
 	if [ -f "$lockfile" ]; then
-            log_warning_msg "lock file already exists"
+            echo "lock file already exists"
             return 0
         fi
 
         if [ $? -eq 0 ]; then
                 if [ ! -s $CONFIG_FILE ]; then
-                    log_failure_msg $CONFIG_FILE "is not configured"
+                    echo $CONFIG_FILE "is not configured"
                     return 6
                 fi
 
                 $CGCONFIGPARSER_BIN -l $CONFIG_FILE
                 retval=$?
                 if [ $retval -ne 0 ]; then
-                    log_failure_msg "Failed to parse " $CONFIG_FILE
+                    echo "Failed to parse " $CONFIG_FILE
                     return 1
                 fi
         fi
@@ -123,18 +118,18 @@
         touch "$lockfile"
         retval=$?
         if [ $retval -ne 0 ]; then
-            log_failure_msg "Failed to touch $lockfile"
+            echo "Failed to touch $lockfile"
             return 1
         fi
-        log_success_msg
+        #log_success_msg
         return 0
 }
 
 stop() {
     echo -n "Stopping cgconfig service: "
-    cgclear
+    /usr/sbin/cgclear
     rm -f "$lockfile"
-    log_success_msg
+    #log_success_msg
     return 0
 }
 
