diff -up shared-desktop-ontologies-0.4/shared-desktop-ontologies.pc.cmake.pkgconfig shared-desktop-ontologies-0.4/shared-desktop-ontologies.pc.cmake
--- shared-desktop-ontologies-0.4/shared-desktop-ontologies.pc.cmake.pkgconfig	2009-11-25 08:17:13.000000000 -0600
+++ shared-desktop-ontologies-0.4/shared-desktop-ontologies.pc.cmake	2010-05-06 07:56:38.396156575 -0500
@@ -2,7 +2,7 @@ prefix=@CMAKE_INSTALL_PREFIX@
 
 Name: shared-desktop-ontologies
 Description: Common OSCAF desktop ontologies
-Version: @CMAKE_ONTOLOGIES_VERSION_STRING@
+Version: @SHAREDDESKTOPONTOLOGIES_VERSION@ 
 Requires:
 Libs:
 Cflags:
