diff -up nss-3.13.1/mozilla/security/nss/lib/util/pkcs11n.h.fo nss-3.13.1/mozilla/security/nss/lib/util/pkcs11n.h
--- nss-3.13.1/mozilla/security/nss/lib/util/pkcs11n.h.fo	2011-11-10 12:44:17.683967574 -0600
+++ nss-3.13.1/mozilla/security/nss/lib/util/pkcs11n.h	2011-11-10 12:44:24.146886778 -0600
@@ -362,7 +362,7 @@ typedef CK_ULONG          CK_TRUST;
  *  cast the resulting value to the deprecated type in the #define, thus
  *  producting the warning when the #define is used.
  */
-#if (__GNUC__  == 4) && (__GNUC_MINOR < 5)
+#if (__GNUC__  == 4) && (__GNUC_MINOR__ < 5)
 /* The mac doesn't like the friendlier deprecate messages. I'm assuming this
  * is a gcc version issue rather than mac or ppc specific */
 typedef CK_TRUST __CKT_NSS_UNTRUSTED __attribute__((deprecated));
