diff -up nss/cmd/bltest/Makefile.iquote nss/cmd/bltest/Makefile
--- nss/cmd/bltest/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/bltest/Makefile	2014-05-06 07:15:41.173387799 -0700
@@ -45,6 +45,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
+INCLUDES += -iquote $(DIST)/../private/nss
 
 
 #######################################################################
diff -up nss/cmd/certcgi/Makefile.iquote nss/cmd/certcgi/Makefile
--- nss/cmd/certcgi/Makefile.iquote	2014-08-19 10:18:35.713017904 -0700
+++ nss/cmd/certcgi/Makefile	2014-08-19 10:19:36.106528087 -0700
@@ -36,7 +36,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/certutil/Makefile.iquote nss/cmd/certutil/Makefile
--- nss/cmd/certutil/Makefile.iquote	2014-08-19 10:23:39.697585905 -0700
+++ nss/cmd/certutil/Makefile	2014-08-19 10:24:31.060019803 -0700
@@ -37,7 +37,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/lib/Makefile.iquote nss/cmd/lib/Makefile
--- nss/cmd/lib/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/lib/Makefile	2014-05-06 07:15:41.174387806 -0700
@@ -38,7 +38,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../private/nss
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/modutil/Makefile.iquote nss/cmd/modutil/Makefile
--- nss/cmd/modutil/Makefile.iquote	2014-05-06 07:34:30.055124213 -0700
+++ nss/cmd/modutil/Makefile	2014-05-06 07:35:36.016602770 -0700
@@ -41,6 +41,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
+INCLUDES += -iquote $(DIST)/../public/nss
 
 
 #######################################################################
diff -up nss/cmd/selfserv/Makefile.iquote nss/cmd/selfserv/Makefile
--- nss/cmd/selfserv/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/selfserv/Makefile	2014-05-06 07:15:41.175387813 -0700
@@ -35,7 +35,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/ssltap/Makefile.iquote nss/cmd/ssltap/Makefile
--- nss/cmd/ssltap/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/ssltap/Makefile	2014-05-06 07:15:41.176387820 -0700
@@ -39,7 +39,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../private/nss
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/strsclnt/Makefile.iquote nss/cmd/strsclnt/Makefile
--- nss/cmd/strsclnt/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/strsclnt/Makefile	2014-05-06 07:15:41.177387827 -0700
@@ -36,7 +36,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/tstclnt/Makefile.iquote nss/cmd/tstclnt/Makefile
--- nss/cmd/tstclnt/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/tstclnt/Makefile	2014-05-06 07:15:41.178387834 -0700
@@ -37,6 +37,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 #######################################################################
 
 #include ../platlibs.mk
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/cmd/vfyserv/Makefile.iquote nss/cmd/vfyserv/Makefile
--- nss/cmd/vfyserv/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/cmd/vfyserv/Makefile	2014-05-06 07:15:41.179387841 -0700
@@ -37,6 +37,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 #######################################################################
 
 #include ../platlibs.mk
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/coreconf/location.mk.iquote nss/coreconf/location.mk
--- nss/coreconf/location.mk.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/coreconf/location.mk	2014-05-06 07:15:41.180387848 -0700
@@ -45,6 +45,10 @@ endif
 
 ifdef NSS_INCLUDE_DIR
     INCLUDES += -I$(NSS_INCLUDE_DIR)
+    ifdef IN_TREE_FREEBL_HEADERS_FIRST
+        INCLUDES += -iquote $(DIST)/../public/nss
+        INCLUDES += -iquote $(DIST)/../private/nss
+    endif
 endif
 
 ifndef NSS_LIB_DIR
diff -up nss/lib/certhigh/Makefile.iquote nss/lib/certhigh/Makefile
--- nss/lib/certhigh/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/lib/certhigh/Makefile	2014-05-06 07:15:41.181387855 -0700
@@ -38,7 +38,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/lib/cryptohi/Makefile.iquote nss/lib/cryptohi/Makefile
--- nss/lib/cryptohi/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/lib/cryptohi/Makefile	2014-05-06 07:15:41.182387862 -0700
@@ -38,7 +38,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/lib/nss/Makefile.iquote nss/lib/nss/Makefile
--- nss/lib/nss/Makefile.iquote	2014-05-01 20:27:18.000000000 -0700
+++ nss/lib/nss/Makefile	2014-05-06 07:15:41.183387869 -0700
@@ -37,7 +37,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
