diff -up nss/cmd/bltest/Makefile.iquote nss/cmd/bltest/Makefile
--- nss/cmd/bltest/Makefile.iquote	2014-01-03 11:59:10.000000000 -0800
+++ nss/cmd/bltest/Makefile	2014-01-07 13:30:04.465429623 -0800
@@ -45,6 +45,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
+INCLUDES += -iquote $(DIST)/../private/nss
 
 
 #######################################################################
diff -up nss/cmd/lib/Makefile.iquote nss/cmd/lib/Makefile
--- nss/cmd/lib/Makefile.iquote	2014-01-03 11:59:10.000000000 -0800
+++ nss/cmd/lib/Makefile	2014-01-07 13:30:04.465429623 -0800
@@ -38,7 +38,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../private/nss
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/coreconf/location.mk.iquote nss/coreconf/location.mk
--- nss/coreconf/location.mk.iquote	2014-01-03 11:59:10.000000000 -0800
+++ nss/coreconf/location.mk	2014-01-07 13:30:04.465429623 -0800
@@ -45,6 +45,10 @@ endif
 
 ifdef NSS_INCLUDE_DIR
     INCLUDES += -I$(NSS_INCLUDE_DIR)
+    ifdef IN_TREE_FREEBL_HEADERS_FIRST
+        INCLUDES += -iquote $(DIST)/../public/nss
+        INCLUDES += -iquote $(DIST)/../private/nss
+    endif
 endif
 
 ifndef NSS_LIB_DIR
diff -up nss/lib/certhigh/Makefile.iquote nss/lib/certhigh/Makefile
--- nss/lib/certhigh/Makefile.iquote	2014-01-03 11:59:10.000000000 -0800
+++ nss/lib/certhigh/Makefile	2014-01-07 13:30:04.466429634 -0800
@@ -38,7 +38,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
diff -up nss/lib/cryptohi/Makefile.iquote nss/lib/cryptohi/Makefile
--- nss/lib/cryptohi/Makefile.iquote	2014-01-03 11:59:10.000000000 -0800
+++ nss/lib/cryptohi/Makefile	2014-01-07 13:30:04.466429634 -0800
@@ -38,7 +38,7 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
--- nss/lib/nss/Makefile.iquote	2014-01-03 11:59:10.000000000 -0800
+++ nss/lib/nss/Makefile	2014-01-07 13:30:04.466429634 -0800
@@ -37,7 +37,8 @@ include $(CORE_DEPTH)/coreconf/rules.mk
 # (6) Execute "component" rules. (OPTIONAL)                           #
 #######################################################################
 
-
+INCLUDES += -iquote $(DIST)/../public/nss
+INCLUDES += -iquote $(DIST)/../private/nss
 
 #######################################################################
 # (7) Execute "local" rules. (OPTIONAL).                              #
