diff -up ./mozilla/security/nss/cmd/selfserv/selfserv.c.539183 ./mozilla/security/nss/cmd/selfserv/selfserv.c
--- ./mozilla/security/nss/cmd/selfserv/selfserv.c.539183	2011-04-27 15:24:07.922128850 -0700
+++ ./mozilla/security/nss/cmd/selfserv/selfserv.c	2011-04-27 15:27:11.053271675 -0700
@@ -1493,14 +1493,14 @@ getBoundListenSocket(unsigned short port
     PRSocketOptionData opt;
     PRUint16           socketDomain = PR_AF_INET;
 
-    addr.inet.family = PR_AF_INET;
-    addr.inet.ip     = PR_INADDR_ANY;
-    addr.inet.port   = PR_htons(port);
+    if (PR_SetNetAddr(PR_IpAddrAny, PR_AF_INET6, port, &addr) != PR_SUCCESS) {
+	errExit("PR_SetNetAddr");
+    }
 
     if (PR_GetEnv("NSS_USE_SDP")) {
         socketDomain = PR_AF_INET_SDP;
     }
-    listen_sock = PR_OpenTCPSocket(socketDomain);
+    listen_sock = PR_OpenTCPSocket(PR_AF_INET6);
     if (listen_sock == NULL) {
         errExit("PR_OpenTCPSocket error");
     }
