diff -up mozilla/security/coreconf/Linux.mk.zlib mozilla/security/coreconf/Linux.mk
--- nss/coreconf/Linux.mk.zlib	2010-01-15 20:19:00.000000000 -0200
+++ nss/coreconf/Linux.mk	2010-03-06 20:46:14.451781165 -0300
@@ -159,7 +159,6 @@ G++INCLUDES		= -I/usr/include/g++
 CPU_TAG = _$(CPU_ARCH)
 
 USE_SYSTEM_ZLIB = 1
-ZLIB_LIBS = -lz
 
 # The -rpath '$$ORIGIN' linker option instructs this library to search for its
 # dependencies in the same directory where it resides.
diff -up mozilla/security/nss/cmd/modutil/manifest.mn.zlib mozilla/security/nss/cmd/modutil/manifest.mn
--- nss/cmd/modutil/manifest.mn.zlib	2004-04-25 12:02:47.000000000 -0300
+++ nss/cmd/modutil/manifest.mn	2010-03-06 20:46:14.451781165 -0300
@@ -64,3 +64,4 @@ DEFINES = -DNSPR20
 # USE_STATIC_LIBS = 1 
 
 EXTRA_LIBS = $(JAR_LIBS)
+EXTRA_SHARED_LIBS += -lz
diff -up mozilla/security/nss/cmd/signtool/manifest.mn.zlib mozilla/security/nss/cmd/signtool/manifest.mn
--- nss/cmd/signtool/manifest.mn.zlib	2008-06-30 18:26:58.000000000 -0300
+++ nss/cmd/signtool/manifest.mn	2010-03-06 20:46:14.452780971 -0300
@@ -55,3 +55,4 @@ PROGRAM =  signtool
 REQUIRES = seccmd
 
 EXTRA_LIBS = $(JAR_LIBS)
+EXTRA_SHARED_LIBS += -lz
diff -up mozilla/security/nss/lib/ssl/manifest.mn.zlib mozilla/security/nss/lib/ssl/manifest.mn
--- nss/lib/ssl/manifest.mn.zlib	2008-09-22 20:47:00.000000000 -0300
+++ nss/lib/ssl/manifest.mn	2010-03-06 20:46:14.453780826 -0300
@@ -81,3 +81,5 @@ LIBRARY_VERSION = 3
 
 # This part of the code, including all sub-dirs, can be optimized for size
 export ALLOW_OPT_CODE_SIZE = 1
+
+EXTRA_SHARED_LIBS += -lz
