diff -up mozilla/security/nss/lib/freebl/stubs.c.prelink mozilla/security/nss/lib/freebl/stubs.c
--- mozilla/security/nss/lib/freebl/stubs.c.prelink	2011-03-29 18:20:12.000000000 -0700
+++ mozilla/security/nss/lib/freebl/stubs.c	2011-10-07 10:02:54.851452105 -0700
@@ -156,6 +156,7 @@ STUB_DECLARE(void,PR_DestroyCondVar,(PRC
 STUB_DECLARE(void,PR_Free,(void *ptr));
 STUB_DECLARE(char * ,PR_GetLibraryFilePathname,(const char *name,
 			PRFuncPtr addr));
+STUB_DECLARE(PRFileDesc *,PR_ImportFile,(PROsfd osfd));
 STUB_DECLARE(PRFileDesc *,PR_ImportPipe,(PROsfd osfd));
 STUB_DECLARE(void,PR_Lock,(PRLock *lock));
 STUB_DECLARE(PRCondVar *,PR_NewCondVar,(PRLock *lock));
@@ -307,6 +308,20 @@ PR_Open_stub(const char *name, PRIntn fl
 }
 
 extern PRFileDesc *
+PR_ImportFile_stub(PROsfd fd)
+{
+    int *lfd = NULL;
+
+    STUB_SAFE_CALL1(PR_ImportFile, fd);
+
+    lfd = PORT_New_stub(int);
+    if (lfd != NULL) {
+	*lfd = fd;
+    }
+    return (PRFileDesc *)lfd;
+}
+
+extern PRFileDesc *
 PR_ImportPipe_stub(PROsfd fd)
 {
     int *lfd = NULL;
@@ -566,6 +581,7 @@ freebl_InitNSPR(void *lib)
 {
     STUB_FETCH_FUNCTION(PR_Free);
     STUB_FETCH_FUNCTION(PR_Open);
+    STUB_FETCH_FUNCTION(PR_ImportFile);
     STUB_FETCH_FUNCTION(PR_ImportPipe);
     STUB_FETCH_FUNCTION(PR_Close);
     STUB_FETCH_FUNCTION(PR_Read);
