diff -up ./mozilla/security/nss/lib/ckfw/pem/pobject.c.717338 ./mozilla/security/nss/lib/ckfw/pem/pobject.c
--- ./mozilla/security/nss/lib/ckfw/pem/pobject.c.717338	2010-11-25 10:49:27.000000000 -0800
+++ ./mozilla/security/nss/lib/ckfw/pem/pobject.c	2011-09-10 10:16:58.752726964 -0700
@@ -1179,6 +1179,7 @@ pem_CreateObject
             return (NSSCKMDObject *) NULL;
 
         certDER.len = 0; /* in case there is no equivalent cert */
+        certDER.data = NULL;
 
         objid = -1;
         for (i = 0; i < pem_nobjs; i++) {
