diff -up ./mozilla/security/nss/lib/ckfw/pem/pinst.c.754771 ./mozilla/security/nss/lib/ckfw/pem/pinst.c
--- ./mozilla/security/nss/lib/ckfw/pem/pinst.c.754771	2011-12-12 09:38:51.839104295 -0800
+++ ./mozilla/security/nss/lib/ckfw/pem/pinst.c	2011-12-12 09:44:40.437096761 -0800
@@ -350,6 +350,9 @@ AddObjectIfNeeded(CK_OBJECT_CLASS objCla
     if (io == NULL)
         return NULL;
 
+    /* initialize pointers to functions */
+    pem_CreateMDObject(NULL, io, NULL);
+
     io->gobjIndex = count;
 
     /* add object to global array */
