From 72457bee3adc228121d0be57398d6dd5497bca4c Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Sat, 8 Nov 2014 19:14:00 +0500
Subject: [PATCH 03/16] qca-ossl: fixed compilation warnings

---
 plugins/qca-ossl/qca-ossl.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/qca-ossl/qca-ossl.cpp b/plugins/qca-ossl/qca-ossl.cpp
index 783a20d..95b2bc4 100644
--- a/plugins/qca-ossl/qca-ossl.cpp
+++ b/plugins/qca-ossl/qca-ossl.cpp
@@ -1539,10 +1539,12 @@ static BigInteger decode(const QString &prime)
 	return BigInteger(SecureArray(a));
 }
 
+#ifndef OPENSSL_FIPS
 static QByteArray decode_seed(const QString &hex_seed)
 {
 	return dehex(hex_seed);
 }
+#endif
 
 class DLParams
 {
@@ -1550,6 +1552,7 @@ public:
 	BigInteger p, q, g;
 };
 
+#ifndef OPENSSL_FIPS
 static bool make_dlgroup(const QByteArray &seed, int bits, int counter, DLParams *params)
 {
 	int ret_counter;
@@ -1564,6 +1567,7 @@ static bool make_dlgroup(const QByteArray &seed, int bits, int counter, DLParams
 	DSA_free(dsa);
 	return true;
 }
+#endif
 
 static bool get_dlgroup(const BigInteger &p, const BigInteger &g, DLParams *params)
 {
-- 
1.9.3

