From f8fe237c3c117c01e35a767e93d5e355a07d9a5c Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Wed, 12 Nov 2014 15:37:09 +0500
Subject: [PATCH 04/16] cmake: dropped dead variable

---
 CMakeLists.txt     | 1 +
 src/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d15a3a7..0f5a42c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,7 @@ cmake_policy(SET CMP0020 OLD)
 option(BUILD_TESTS "Create test" ON)
 option(BUILD_TOOLS "Compile mozcerts and qcatool" ON)
 set(BUILD_PLUGINS "auto" CACHE STRING "Plugins for building (also possible values: none, all and auto)")
+# BUILD_SHARED_LIBS is cmake variable. Need to change default value.
 option(BUILD_SHARED_LIBS "Build shared library" ON)
 option(DEVELOPER_MODE "Special developer mode" OFF)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f2a2fbf..290f403 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -140,7 +140,7 @@ qt4_wrap_cpp( SOURCES "${qca_INCLUDEDIR}/QtCrypto/qpipe.h")
 qt4_wrap_cpp( SOURCES "qca_safeobj.h")
 qt4_wrap_cpp( SOURCES "${qca_INCLUDEDIR}/QtCrypto/qca_safetimer.h")
 
-ADD_LIBRARY(${QCA_LIB_NAME} ${LIBRARY_TYPE} ${SOURCES}  ${public_HEADERS})
+add_library(${QCA_LIB_NAME} ${SOURCES}  ${public_HEADERS})
 TARGET_LINK_LIBRARIES(${QCA_LIB_NAME} ${QT_QTCORE_LIBRARY})
 
 if(WIN32)
-- 
1.9.3

