From 20686e955cc86c2b0ac451de79f713d5fcf24f91 Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Thu, 6 Nov 2014 22:11:28 +0500
Subject: [PATCH 10/16] fixed compilation on android

---
 src/support/console.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/support/console.cpp b/src/support/console.cpp
index 32eb7cd..c3c5570 100644
--- a/src/support/console.cpp
+++ b/src/support/console.cpp
@@ -30,7 +30,11 @@
 #ifdef Q_OS_WIN
 # include <windows.h>
 #else
-# include <sys/termios.h>
+# ifdef Q_OS_ANDROID
+#  include <termios.h>
+# else
+#  include <sys/termios.h>
+# endif
 # include <unistd.h>
 # include <fcntl.h>
 #endif
-- 
1.9.3

