From 98be0a1234538cb0eda66609f8707eccc6f96724 Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Fri, 28 Nov 2014 23:50:13 +0500
Subject: [PATCH 13/16] docs: fixed no images in docs when build out of source

---
 Doxyfile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Doxyfile.in b/Doxyfile.in
index 7e2bd30..59d9afe 100644
--- a/Doxyfile.in
+++ b/Doxyfile.in
@@ -567,7 +567,7 @@ EXAMPLE_RECURSIVE      = NO
 # directories that contain image that are included in the documentation (see 
 # the \image command).
 
-IMAGE_PATH             = images
+IMAGE_PATH             = @CMAKE_SOURCE_DIR@/images
 
 # The INPUT_FILTER tag can be used to specify a program that doxygen should 
 # invoke to filter for each input file. Doxygen will invoke the filter program 
-- 
1.9.3

