From 9a9c16dfd1affc92962acd92e4a5246830ccbc04 Mon Sep 17 00:00:00 2001
From: Ivan Romanov <drizt@land.ru>
Date: Sat, 29 Nov 2014 00:57:31 +0500
Subject: [PATCH 14/16] cmake: fixed cmake config module when used QCA_SUFFIX

---
 QcaConfig.cmake.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QcaConfig.cmake.in b/QcaConfig.cmake.in
index cb053cf..0eea618 100644
--- a/QcaConfig.cmake.in
+++ b/QcaConfig.cmake.in
@@ -1,7 +1,7 @@
 @PACKAGE_INIT@
 
 if(NOT TARGET qca)
-  include("${CMAKE_CURRENT_LIST_DIR}/QcaTargets.cmake")
+  include("${CMAKE_CURRENT_LIST_DIR}/@QCA_CONFIG_NAME_BASE@Targets.cmake")
 endif()
 
 set(Qca_LIBRARY qca)
-- 
1.9.3

