--- kfilebox.pro.orig	2010-09-24 09:11:46.000000000 -0300
+++ kfilebox.pro	2010-12-20 20:01:10.666188908 -0200
@@ -127,18 +127,6 @@ transinstallde.path = $$DATADIR/locale/d
 transinstallde.files = locale/de/kfilebox.mo
 INSTALLS += transinstallde
 
-transinstallen.extra = msgmerge locale/en/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/en/kfilebox.po; \
-    msgfmt -o locale/en/kfilebox.mo locale/en/kfilebox.po; \
-    cp -f locale/en/kfilebox.mo $$DATADIR/locale/en/LC_MESSAGES
-transinstallen.path = $$DATADIR/locale/en/LC_MESSAGES
-transinstallen.files = locale/en/kfilebox.mo
-INSTALLS += transinstallen
-
-transinstallen_gb.extra = cp -f locale/en/kfilebox.mo $$DATADIR/locale/en_GB/LC_MESSAGES
-transinstallen_gb.path = $$DATADIR/locale/en_GB/LC_MESSAGES
-transinstallen_gb.files = locale/en/kfilebox.mo
-INSTALLS += transinstallen_gb
-
 transinstalles.extra = msgmerge locale/es/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/es/kfilebox.po; \
     msgfmt -o locale/es/kfilebox.mo locale/es/kfilebox.po; \
     cp -f locale/es/kfilebox.mo $$DATADIR/locale/es/LC_MESSAGES
@@ -233,3 +221,6 @@ transinstallzh_cn.extra =  cp -f locale/
 transinstallzh_cn.path = $$DATADIR/locale/zh_CN/LC_MESSAGES
 transinstallzh_cn.files = locale/zh/kfilebox.mo
 INSTALLS += transinstallzh_cn
+QMAKE_CFLAGS_RELEASE = -O2 -march=native -mtune=native  -pipe -fPIC
+QMAKE_CXXFLAGS_RELEASE = -O2 -march=native -mtune=native  -pipe -fPIC
+QMAKE_LFLAGS_RELEASE = -Wl,--hash-style=both
