diff -up kfilebox-0.4.5/core/DropboxClient.cpp.paths kfilebox-0.4.5/core/DropboxClient.cpp
--- kfilebox-0.4.5/core/DropboxClient.cpp.paths	2010-06-24 07:12:13.000000000 -0300
+++ kfilebox-0.4.5/core/DropboxClient.cpp	2010-12-20 20:27:33.314198971 -0200
@@ -14,9 +14,7 @@ bool DropboxClient::start()
 {
 
     bool returnValue=false;
-    QString path(QDir::home().path());
-    path.append(QDir::separator()).append(".dropbox-dist/dropboxd");
-    path = QDir::toNativeSeparators(path);
+    QString path("_LIBDIR_/dropbox/dropboxd");
 
     SystemCall *sc = new SystemCall();
     sc->executeDetached(path);
diff -up kfilebox-0.4.5/mainwindow.cpp.paths kfilebox-0.4.5/mainwindow.cpp
--- kfilebox-0.4.5/mainwindow.cpp.paths	2010-12-20 20:27:55.443533014 -0200
+++ kfilebox-0.4.5/mainwindow.cpp	2010-12-20 20:29:11.566488050 -0200
@@ -19,7 +19,7 @@ MainWindow::MainWindow(QWidget *parent)
     ui->setupUi(this);    
     connect(trayIcon,SIGNAL(prefsWindowActionTrigered()),this,SLOT(openPrefsWindow()));
 
-    QDir *dropbox_dist= new QDir(QDir::homePath().append(QDir::separator()).append(".dropbox-dist"));
+    QDir *dropbox_dist= new QDir("_LIBDIR_/dropbox");
     if (!dropbox_dist->exists(dropbox_dist->path())){
         using namespace installer;
         Daemoninstaller *di=new Daemoninstaller();
