Description: Fix the fd leak in mpc_mpd_connect()
Author: Andrey Rahmatullin <wrar@wrar.name>
Origin: vendor, http://git.altlinux.org/people/wrar/packages/gkrellm-gkrellmpc.git?f=gkrellmpc-0.1_beta10-alt-fd-leak-fix.patch;h=0706caae
Last-Update: 2010-12-18

--- a/mpd.c
+++ b/mpd.c
@@ -37,7 +37,10 @@
 	}
 
 	if ((sockfd = socket(PF_INET, SOCK_STREAM, 0)) == -1) return(FALSE);
-	if (!(server = gethostbyname(mpc_conf_hostname))) return(FALSE);
+	if (!(server = gethostbyname(mpc_conf_hostname))) {
+		close(sockfd);
+		return(FALSE);
+	}
 
 	bzero((char *) &serv_addr, sizeof(serv_addr));
 	serv_addr.sin_family = AF_INET;
@@ -46,7 +49,10 @@
 			server->h_length);
 	serv_addr.sin_port = htons(mpc_conf_port);
 
-	if (connect(sockfd, (struct sockaddr*)&serv_addr, sizeof(serv_addr)) < 0) return(FALSE);
+	if (connect(sockfd, (struct sockaddr*)&serv_addr, sizeof(serv_addr)) < 0) {
+		close(sockfd);
+		return(FALSE);
+	}
 	
 	/* Getup the mpd object */
 	mpc_mpd = g_io_channel_unix_new(sockfd);
