--- gkrellmpc.c.orig	2010-09-19 15:44:25.000000000 -0300
+++ gkrellmpc.c	2010-09-19 15:50:34.835752031 -0300
@@ -444,6 +444,8 @@
 	gchar * time;
 	gchar * artist;
 	gchar * title;
+	gchar * album;
+	gchar * track;
 	gchar * name;
 	gint    oldid;
 	gchar * file;
@@ -505,14 +507,18 @@
 		file = g_hash_table_lookup(currentsong, "file");
 		artist = g_hash_table_lookup(currentsong, "artist");
 		title = g_hash_table_lookup(currentsong, "title");
+		album = g_hash_table_lookup(currentsong, "album");
+		track = g_hash_table_lookup(currentsong, "track");
 		name = g_hash_table_lookup(currentsong, "name");
 		state = g_hash_table_lookup(status, "state");
 
 		if (file) {
 			tiptext = g_strdup_printf(
-				_("Artist: %s\nTitle: %s\nFile: %s")
+				_("Artist: %s\nTitle: %s\nTrack: %s\nAlbum: %s\nFile: %s")
 				, artist ? artist : _("N/A")
 				, title ? title : _("N/A")
+				, track ? track : _("N/A")
+				, album ? album : _("N/A")
 				,  file
 			);
 		}
