--- gkrellmpc.c.orig	2005-03-05 23:40:57.000000000 -0300
+++ gkrellmpc.c	2011-06-20 21:38:20.604575548 -0300
@@ -444,6 +444,11 @@ void mpc_sync_with_mpd() {
 	gchar * time;
 	gchar * artist;
 	gchar * title;
+	gchar * album;
+	gchar * track;
+	gchar * date;
+	gchar * disc;
+	gchar * plpos;
 	gchar * name;
 	gint    oldid;
 	gchar * file;
@@ -505,14 +510,24 @@ void mpc_sync_with_mpd() {
 		file = g_hash_table_lookup(currentsong, "file");
 		artist = g_hash_table_lookup(currentsong, "artist");
 		title = g_hash_table_lookup(currentsong, "title");
+		album = g_hash_table_lookup(currentsong, "album");
+		track = g_hash_table_lookup(currentsong, "track");
+		date = g_hash_table_lookup(currentsong, "date");
+		disc = g_hash_table_lookup(currentsong, "disc");
+		plpos = g_hash_table_lookup(currentsong, "pos");
 		name = g_hash_table_lookup(currentsong, "name");
 		state = g_hash_table_lookup(status, "state");
 
 		if (file) {
 			tiptext = g_strdup_printf(
-				_("Artist: %s\nTitle: %s\nFile: %s")
+				_("Artist: %s\nTitle: %s\nAlbum: %s\nDisc: %s\nTrack: %s\nDate: %s\nPlaylist Position: %s\nFile: %s")
 				, artist ? artist : _("N/A")
 				, title ? title : _("N/A")
+				, album ? album : _("N/A")
+				, disc ? disc : _("N/A")
+				, track ? track : _("N/A")
+				, date ? date : _("N/A")
+				, plpos ? plpos : _("N/A")
 				,  file
 			);
 		}
