diff -up gnome-panel-2.29.92.1/applets/clock/Makefile.am.libs gnome-panel-2.29.92.1/applets/clock/Makefile.am
--- gnome-panel-2.29.92.1/applets/clock/Makefile.am.libs	2010-03-11 22:23:51.692914036 -0500
+++ gnome-panel-2.29.92.1/applets/clock/Makefile.am	2010-03-11 22:38:20.988697805 -0500
@@ -134,7 +135,8 @@ gnome_clock_applet_mechanism_INCLUDES =	
 
 gnome_clock_applet_mechanism_LDADD =	\
 	$(CLOCK_MECHANISM_LIBS)		\
-	$(POLKIT_LIBS)
+	$(POLKIT_LIBS)			\
+	-lm
 
 if CLOCK_INPROCESS
 APPLET_TYPE     = shlib
diff -up gnome-panel-2.29.92.1/configure.in.libs gnome-panel-2.29.92.1/configure.in
--- gnome-panel-2.29.92.1/configure.in.libs	2010-03-11 21:31:25.628657996 -0500
+++ gnome-panel-2.29.92.1/configure.in	2010-03-11 22:11:11.098665958 -0500
@@ -72,6 +72,9 @@ dnl pkg-config dependency checks
 PKG_CHECK_MODULES(EGG_SMCLIENT, ice sm gtk+-2.0)
 
 PKG_CHECK_MODULES(PANEL, ORBit-2.0 >= $ORBIT_REQUIRED gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED pango >= $PANGO_REQUIRED gtk+-2.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED gio-2.0 >= $GLIB_REQUIRED gio-unix-2.0 >= $GLIB_REQUIRED libbonoboui-2.0 >= $LIBBONOBOUI_REQUIRED gnome-desktop-2.0 >= $LIBGNOME_DESKTOP_REQUIRED gconf-2.0 >= $GCONF_REQUIRED libgnome-menu >= $LIBGNOME_MENU_REQUIRED dbus-glib-1 >= $DBUS_GLIB_REQUIRED libcanberra-gtk)
+
+PANEL_LIBS="$PANEL_LIBS -lm"
+
 AC_SUBST(PANEL_CFLAGS)
 AC_SUBST(PANEL_LIBS)
 
@@ -256,16 +259,17 @@ PKG_CHECK_MODULES(X, x11 xau, :, [
   X_LIBS="$X_PRE_LIBS $X_LIBS -lX11 -lXau $X_EXTRA_LIBS"
 ])
 
-AC_SUBST(X_LIBS)
-
 dnl X RANDR extension
 
-if $PKG_CONFIG --exists "xrandr >= 1.2.0" ; then
-	AC_DEFINE(HAVE_RANDR, 1, [Have the Xrandr extension library])
-	have_randr=yes
-else
-	have_randr=no
-fi
+PKG_CHECK_MODULES(XRANDR, xrandr >= 1.2.0, [
+  AC_DEFINE(HAVE_RANDR, 1, [Have the Xrandr extension library])
+  have_randr=yes
+  X_LIBS="$X_LIBS $XRANDR_LIBS"
+], [
+  have_randr=no
+])
+
+AC_SUBST(X_LIBS)
 
 dnl IDL directories
 
diff -up gnome-panel-2.29.92.1/gnome-panel/Makefile.am.libs gnome-panel-2.29.92.1/gnome-panel/Makefile.am
