diff -up gnome-panel-2.32.0.2/applets/notification_area/Makefile.am.dso gnome-panel-2.32.0.2/applets/notification_area/Makefile.am
--- gnome-panel-2.32.0.2/applets/notification_area/Makefile.am.dso	2010-10-09 02:24:55.603746117 -0300
+++ gnome-panel-2.32.0.2/applets/notification_area/Makefile.am	2010-10-09 02:27:25.712492354 -0300
@@ -39,7 +39,7 @@ testtray_SOURCES = testtray.c
 testtray_LDADD =			\
 	$(X_LIBS)			\
 	$(NOTIFICATION_AREA_LIBS)	\
-	libtray.la
+	libtray.la -lX11 -lcairo
 
 if NOTIFICATION_AREA_INPROCESS
 APPLET_IN_PROCESS = true
@@ -57,7 +57,7 @@ APPLET_LOCATION   = $(libexecdir)/notifi
 
 libexec_PROGRAMS = notification-area-applet
 notification_area_applet_SOURCES = $(NOTIFICATION_AREA_SOURCES)
-notification_area_applet_LDADD = $(NOTIFICATION_AREA_LDADD)
+notification_area_applet_LDADD = $(NOTIFICATION_AREA_LDADD) -lgdk-x11-2.0 -lcairo -lX11
 notification_area_applet_CFLAGS = $(AM_CFLAGS)
 endif
 
diff -up gnome-panel-2.32.0.2/configure.ac.dso gnome-panel-2.32.0.2/configure.ac
--- gnome-panel-2.32.0.2/configure.ac.dso	2010-09-27 17:25:45.000000000 -0300
+++ gnome-panel-2.32.0.2/configure.ac	2010-10-09 02:23:33.807496581 -0300
@@ -239,10 +239,15 @@ AC_SUBST(X_LIBS)
 
 dnl X RANDR extension
 
-PKG_CHECK_MODULES(XRANDR, xrandr >= 1.2.0, have_randr=yes, have_randr=no)
-if test "x$have_randr" = "xyes"; then
+PKG_CHECK_MODULES(XRANDR, xrandr >= 1.2.0, [
   AC_DEFINE(HAVE_RANDR, 1, [Have the Xrandr extension library])
-fi
+  have_randr=yes
+  X_LIBS="$X_LIBS $XRANDR_LIBS"
+], [
+  have_randr=no
+])
+
+AC_SUBST(X_LIBS)
 
 dnl Modules dir
 AC_SUBST([modulesdir],"\$(libdir)/gnome-panel/modules")
diff -up gnome-panel-2.32.0.2/gnome-panel/Makefile.am.dso gnome-panel-2.32.0.2/gnome-panel/Makefile.am
--- gnome-panel-2.32.0.2/gnome-panel/Makefile.am.dso	2010-09-27 15:04:32.000000000 -0300
+++ gnome-panel-2.32.0.2/gnome-panel/Makefile.am	2010-10-09 02:23:33.807496581 -0300
@@ -134,6 +134,7 @@ gnome_panel_LDADD =		\
 	$(top_builddir)/gnome-panel/libpanel-util/libpanel-util.la			\
 	$(PANEL_LIBS)									\
 	$(XRANDR_LIBS)									\
+	-lm \
 	$(X_LIBS)
 
 gnome_panel_LDFLAGS = -export-dynamic
@@ -147,7 +148,8 @@ gnome_desktop_item_edit_SOURCES =	\
 
 gnome_desktop_item_edit_LDADD = 	\
 	$(top_builddir)/gnome-panel/libpanel-util/libpanel-util.la \
-	$(PANEL_LIBS)
+	$(PANEL_LIBS) \
+	-lm -lX11
 
 panel_test_applets_SOURCES =		\
 	panel-modules.c			\
