diff -up gnome-panel-2.32.0.2/applets/notification_area/Makefile.am.libs gnome-panel-2.32.0.2/applets/notification_area/Makefile.am
--- gnome-panel-2.32.0.2/applets/notification_area/Makefile.am.libs	2010-09-30 12:29:17.884603002 -0400
+++ gnome-panel-2.32.0.2/applets/notification_area/Makefile.am	2010-09-30 13:11:01.996603010 -0400
@@ -1,5 +1,4 @@
 noinst_LTLIBRARIES = libtray.la
-noinst_PROGRAMS = testtray
 
 AM_CPPFLAGS =							\
 	$(NOTIFICATION_AREA_CFLAGS)				\
@@ -30,15 +29,15 @@ NOTIFICATION_AREA_SOURCES = main.c
 
 NOTIFICATION_AREA_LDADD =				\
 	../../libpanel-applet/libpanel-applet-3.la	\
-	$(X_LIBS)					\
+	libtray.la					\
 	$(NOTIFICATION_AREA_LIBS)			\
 	$(LIBPANEL_APPLET_LIBS)				\
-	libtray.la
+	$(X_LIBS)
 
 testtray_SOURCES = testtray.c
 testtray_LDADD =			\
-	$(X_LIBS)			\
 	$(NOTIFICATION_AREA_LIBS)	\
+	$(X_LIBS)			\
 	libtray.la
 
 if NOTIFICATION_AREA_INPROCESS
diff -up gnome-panel-2.32.0.2/gnome-panel/Makefile.am.libs gnome-panel-2.32.0.2/gnome-panel/Makefile.am
--- gnome-panel-2.32.0.2/gnome-panel/Makefile.am.libs	2010-09-27 14:04:32.000000000 -0400
+++ gnome-panel-2.32.0.2/gnome-panel/Makefile.am	2010-09-30 12:27:42.928603001 -0400
@@ -147,7 +147,8 @@ gnome_desktop_item_edit_SOURCES =	\
 
 gnome_desktop_item_edit_LDADD = 	\
 	$(top_builddir)/gnome-panel/libpanel-util/libpanel-util.la \
-	$(PANEL_LIBS)
+	$(PANEL_LIBS) \
+	$(X_LIBS)
 
 panel_test_applets_SOURCES =		\
 	panel-modules.c			\
