diff -up src/include/syntaxreader.h.orig src/include/syntaxreader.h
--- src/include/syntaxreader.h.orig	2014-12-04 19:19:08.634450001 -0200
+++ src/include/syntaxreader.h	2014-12-04 19:19:37.573465775 -0200
@@ -36,7 +36,9 @@ along with Highlight.  If not, see <http
 #include <iterator>
 #include <sstream>
 
+#ifndef Q_MOC_RUN
 #include <boost/xpressive/xpressive_dynamic.hpp>
+#endif
 
 #include <Diluculum/LuaState.hpp>
 #include <Diluculum/LuaVariable.hpp>
