From 7f1656b2cd73c7a721bc04b087a66d92765cdc0c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ha=C3=AFkel=20Gu=C3=A9mar?= <hguemar@fedoraproject.org>
Date: Wed, 20 Mar 2013 09:53:45 +0100
Subject: [PATCH] workaround RHBZ #904790

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 8d0841a..b70723b 100755
--- a/configure
+++ b/configure
@@ -19451,7 +19451,7 @@ if test -n "$GRAPHVIZ_LIBS"; then
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GRAPHVIZ_LIBS=`$PKG_CONFIG --libs ""libgvc"" 2>/dev/null`
+  pkg_cv_GRAPHVIZ_LIBS="-lcdt -lcgraph -lgvc"
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
-- 
1.8.1.4

