diff -up kdegraphics-4.5.2/okular/generators/xps/libokularGenerator_xps.desktop.hack kdegraphics-4.5.2/okular/generators/xps/libokularGenerator_xps.desktop
--- kdegraphics-4.5.2/okular/generators/xps/libokularGenerator_xps.desktop.hack	2010-07-23 16:14:58.000000000 -0500
+++ kdegraphics-4.5.2/okular/generators/xps/libokularGenerator_xps.desktop	2010-10-19 11:49:20.698706403 -0500
@@ -106,7 +106,7 @@ Comment[x-test]=xxXPS backend for Okular
 Comment[zh_CN]=Okular 的 XPS 格式后端
 Comment[zh_TW]=Okular 的 XPS 後端
 X-KDE-ServiceTypes=okular/Generator
-MimeType=application/vnd.ms-xpsdocument;
+MimeType=application/oxps;application/vnd.ms-xpsdocument;
 X-KDE-Library=okularGenerator_xps
 X-KDE-Priority=4
 X-KDE-okularAPIVersion=1
