diff -ruN okular-4.6.4.orig/okular/CMakeLists.txt okular-4.6.4/okular/CMakeLists.txt
--- okular-4.6.4.orig/okular/CMakeLists.txt	2011-01-06 00:36:07.000000000 +0100
+++ okular-4.6.4/okular/CMakeLists.txt	2011-06-14 23:54:25.681769727 +0200
@@ -1,8 +1,27 @@
 project(okular)
+
+find_package(KDE4 4.6.0 REQUIRED)
+include(KDE4Defaults)
+include(MacroLibrary)
+include(MacroOptionalAddSubdirectory)
+
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake/modules)
+
+macro_optional_find_package(QImageBlitz)
+macro_log_feature(QIMAGEBLITZ_FOUND "QImageBlitz" "An image effects library" "http://sourceforge.net/projects/qimageblitz" TRUE "kdesupport" "Required to build Okular.")
+
+add_definitions(${QT_DEFINITIONS} ${KDE4_DEFINITIONS})
+include_directories(
+   ${CMAKE_CURRENT_SOURCE_DIR}
+   ${KDE4_INCLUDES}
+   ${QIMAGEBLITZ_INCLUDES}
+)
+
 add_subdirectory( ui )
 add_subdirectory( shell )
 add_subdirectory( generators )
 add_subdirectory( tests )
+macro_optional_add_subdirectory(doc)
 
 include(OkularConfigureChecks.cmake)
 include(MacroWriteBasicCMakeVersionFile)
@@ -12,11 +31,6 @@
 else(NOT WIN32)
  set(MATH_LIB)
 endif(NOT WIN32)
-include_directories(
-   ${CMAKE_CURRENT_SOURCE_DIR}
-   ${QIMAGEBLITZ_INCLUDES}
-)
-
 
 # okularcore
 
@@ -189,3 +203,6 @@
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/OkularConfig.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/OkularConfigVersion.cmake
         DESTINATION ${LIB_INSTALL_DIR}/cmake/Okular )
+
+macro_display_feature_log()
+
diff -ruN okular-4.6.4.orig/okular/VERSION okular-4.6.4/okular/VERSION
--- okular-4.6.4.orig/okular/VERSION	2011-02-24 15:38:41.000000000 +0100
+++ okular-4.6.4/okular/VERSION	2011-06-14 23:54:25.682769725 +0200
@@ -1 +1 @@
-okular v0.12.1
+okular v0.12.4
diff -ruN okular-4.6.4.orig/okular/core/action.h okular-4.6.4/okular/core/action.h
--- okular-4.6.4.orig/okular/core/action.h	2009-05-13 13:13:04.000000000 +0200
+++ okular-4.6.4/okular/core/action.h	2011-06-14 23:54:25.682769725 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_ACTION_H_
 #define _OKULAR_ACTION_H_
 
-#include <okular/core/global.h>
-#include <okular/core/okular_export.h>
+#include "global.h"
+#include "okular_export.h"
 
 #include <QtCore/QString>
 
diff -ruN okular-4.6.4.orig/okular/core/annotations.h okular-4.6.4/okular/core/annotations.h
--- okular-4.6.4.orig/okular/core/annotations.h	2009-12-31 10:56:31.000000000 +0100
+++ okular-4.6.4/okular/core/annotations.h	2011-06-14 23:54:25.682769725 +0200
@@ -18,8 +18,8 @@
 #include <QtXml/QDomDocument>
 #include <QtXml/QDomElement>
 
-#include <okular/core/okular_export.h>
-#include <okular/core/area.h>
+#include "okular_export.h"
+#include "area.h"
 
 namespace Okular {
 
diff -ruN okular-4.6.4.orig/okular/core/area.h okular-4.6.4/okular/core/area.h
--- okular-4.6.4.orig/okular/core/area.h	2009-11-12 22:45:35.000000000 +0100
+++ okular-4.6.4/okular/core/area.h	2011-06-14 23:54:25.682769725 +0200
@@ -15,8 +15,8 @@
 #include <QtGui/QPainterPath>
 #include <kdebug.h>
 
-#include <okular/core/global.h>
-#include <okular/core/okular_export.h>
+#include "global.h"
+#include "okular_export.h"
 
 class QPolygonF;
 class QRect;
diff -ruN okular-4.6.4.orig/okular/core/audioplayer.h okular-4.6.4/okular/core/audioplayer.h
--- okular-4.6.4.orig/okular/core/audioplayer.h	2007-07-10 20:24:18.000000000 +0200
+++ okular-4.6.4/okular/core/audioplayer.h	2011-06-14 23:54:25.682769725 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_AUDIOPLAYER_H_
 #define _OKULAR_AUDIOPLAYER_H_
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 #include <QtCore/QObject>
 
diff -ruN okular-4.6.4.orig/okular/core/document.h okular-4.6.4/okular/core/document.h
--- okular-4.6.4.orig/okular/core/document.h	2010-08-11 23:01:33.000000000 +0200
+++ okular-4.6.4/okular/core/document.h	2011-06-14 23:54:25.682769725 +0200
@@ -11,10 +11,10 @@
 #ifndef _OKULAR_DOCUMENT_H_
 #define _OKULAR_DOCUMENT_H_
 
-#include <okular/core/okular_export.h>
-#include <okular/core/area.h>
-#include <okular/core/global.h>
-#include <okular/core/pagesize.h>
+#include "okular_export.h"
+#include "area.h"
+#include "global.h"
+#include "pagesize.h"
 
 #include <QtCore/QObject>
 #include <QtCore/QStringList>
diff -ruN okular-4.6.4.orig/okular/core/fileprinter.h okular-4.6.4/okular/core/fileprinter.h
--- okular-4.6.4.orig/okular/core/fileprinter.h	2010-06-04 21:04:05.000000000 +0200
+++ okular-4.6.4/okular/core/fileprinter.h	2011-06-14 23:54:25.682769725 +0200
@@ -17,8 +17,8 @@
 #include <QtCore/QList>
 #include <QtCore/QString>
 
-#include <okular/core/okular_export.h>
-#include <okular/core/generator.h>
+#include "okular_export.h"
+#include "generator.h"
 
 class QPrinter;
 class QSize;
diff -ruN okular-4.6.4.orig/okular/core/fontinfo.h okular-4.6.4/okular/core/fontinfo.h
--- okular-4.6.4.orig/okular/core/fontinfo.h	2009-11-21 01:50:13.000000000 +0100
+++ okular-4.6.4/okular/core/fontinfo.h	2011-06-14 23:54:25.683769724 +0200
@@ -15,7 +15,7 @@
 #include <QtCore/QSharedDataPointer>
 #include <QtCore/QString>
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 namespace Okular {
 
diff -ruN okular-4.6.4.orig/okular/core/form.h okular-4.6.4/okular/core/form.h
--- okular-4.6.4.orig/okular/core/form.h	2008-03-20 18:42:17.000000000 +0100
+++ okular-4.6.4/okular/core/form.h	2011-06-14 23:54:25.683769724 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_FORM_H_
 #define _OKULAR_FORM_H_
 
-#include <okular/core/okular_export.h>
-#include <okular/core/area.h>
+#include "okular_export.h"
+#include "area.h"
 
 #include <QtCore/QStringList>
 
diff -ruN okular-4.6.4.orig/okular/core/generator.h okular-4.6.4/okular/core/generator.h
--- okular-4.6.4.orig/okular/core/generator.h	2010-04-15 01:33:14.000000000 +0200
+++ okular-4.6.4/okular/core/generator.h	2011-06-14 23:54:25.683769724 +0200
@@ -12,10 +12,10 @@
 #ifndef _OKULAR_GENERATOR_H_
 #define _OKULAR_GENERATOR_H_
 
-#include <okular/core/okular_export.h>
-#include <okular/core/fontinfo.h>
-#include <okular/core/global.h>
-#include <okular/core/pagesize.h>
+#include "okular_export.h"
+#include "fontinfo.h"
+#include "global.h"
+#include "pagesize.h"
 
 #include <QtCore/QList>
 #include <QtCore/QObject>
diff -ruN okular-4.6.4.orig/okular/core/misc.h okular-4.6.4/okular/core/misc.h
--- okular-4.6.4.orig/okular/core/misc.h	2007-01-03 15:30:48.000000000 +0100
+++ okular-4.6.4/okular/core/misc.h	2011-06-14 23:54:25.683769724 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_MISC_H_
 #define _OKULAR_MISC_H_
 
-#include <okular/core/okular_export.h>
-#include <okular/core/area.h>
+#include "okular_export.h"
+#include "area.h"
 
 namespace Okular {
 
diff -ruN okular-4.6.4.orig/okular/core/movie.h okular-4.6.4/okular/core/movie.h
--- okular-4.6.4.orig/okular/core/movie.h	2008-10-24 09:10:18.000000000 +0200
+++ okular-4.6.4/okular/core/movie.h	2011-06-14 23:54:25.683769724 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_MOVIE_H_
 #define _OKULAR_MOVIE_H_
 
-#include <okular/core/global.h>
-#include <okular/core/okular_export.h>
+#include "global.h"
+#include "okular_export.h"
 
 #include <QtCore/QSize>
 
diff -ruN okular-4.6.4.orig/okular/core/observer.h okular-4.6.4/okular/core/observer.h
--- okular-4.6.4.orig/okular/core/observer.h	2008-05-19 01:06:21.000000000 +0200
+++ okular-4.6.4/okular/core/observer.h	2011-06-14 23:54:25.683769724 +0200
@@ -13,7 +13,7 @@
 
 #include <QtCore/QVector>
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 namespace Okular {
 
diff -ruN okular-4.6.4.orig/okular/core/page.h okular-4.6.4/okular/core/page.h
--- okular-4.6.4.orig/okular/core/page.h	2009-11-12 22:48:43.000000000 +0100
+++ okular-4.6.4/okular/core/page.h	2011-06-14 23:54:25.683769724 +0200
@@ -12,10 +12,10 @@
 
 #include <QtCore/QLinkedList>
 
-#include <okular/core/okular_export.h>
-#include <okular/core/area.h>
-#include <okular/core/global.h>
-#include <okular/core/textpage.h>
+#include "okular_export.h"
+#include "area.h"
+#include "global.h"
+#include "textpage.h"
 
 class QPixmap;
 
diff -ruN okular-4.6.4.orig/okular/core/pagesize.h okular-4.6.4/okular/core/pagesize.h
--- okular-4.6.4.orig/okular/core/pagesize.h	2007-09-02 00:48:28.000000000 +0200
+++ okular-4.6.4/okular/core/pagesize.h	2011-06-14 23:54:25.683769724 +0200
@@ -14,7 +14,7 @@
 #include <QtCore/QSharedDataPointer>
 #include <QtCore/QString>
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 namespace Okular {
 
diff -ruN okular-4.6.4.orig/okular/core/pagetransition.h okular-4.6.4/okular/core/pagetransition.h
--- okular-4.6.4.orig/okular/core/pagetransition.h	2007-01-13 23:55:00.000000000 +0100
+++ okular-4.6.4/okular/core/pagetransition.h	2011-06-14 23:54:25.683769724 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_PAGETRANSITION_H_
 #define _OKULAR_PAGETRANSITION_H_
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 namespace Okular {
 
diff -ruN okular-4.6.4.orig/okular/core/sound.h okular-4.6.4/okular/core/sound.h
--- okular-4.6.4.orig/okular/core/sound.h	2007-05-27 11:59:52.000000000 +0200
+++ okular-4.6.4/okular/core/sound.h	2011-06-14 23:54:25.683769724 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_SOUND_H_
 #define _OKULAR_SOUND_H_
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 #include <QtCore/QByteArray>
 #include <QtCore/QString>
diff -ruN okular-4.6.4.orig/okular/core/sourcereference.h okular-4.6.4/okular/core/sourcereference.h
--- okular-4.6.4.orig/okular/core/sourcereference.h	2007-01-05 17:40:07.000000000 +0100
+++ okular-4.6.4/okular/core/sourcereference.h	2011-06-14 23:54:25.683769724 +0200
@@ -10,7 +10,7 @@
 #ifndef OKULAR_SOURCEREFERENCE_H
 #define OKULAR_SOURCEREFERENCE_H
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
 class QString;
 
diff -ruN okular-4.6.4.orig/okular/core/textdocumentgenerator.cpp okular-4.6.4/okular/core/textdocumentgenerator.cpp
--- okular-4.6.4.orig/okular/core/textdocumentgenerator.cpp	2010-04-11 19:54:54.000000000 +0200
+++ okular-4.6.4/okular/core/textdocumentgenerator.cpp	2011-06-14 23:54:25.684769722 +0200
@@ -23,10 +23,10 @@
 #include <QtGui/QTextDocumentWriter>
 #endif
 
-#include <okular/core/action.h>
-#include <okular/core/annotations.h>
-#include <okular/core/page.h>
-#include <okular/core/textpage.h>
+#include "action.h"
+#include "annotations.h"
+#include "page.h"
+#include "textpage.h"
 
 #include "document.h"
 
diff -ruN okular-4.6.4.orig/okular/core/textdocumentgenerator.h okular-4.6.4/okular/core/textdocumentgenerator.h
--- okular-4.6.4.orig/okular/core/textdocumentgenerator.h	2008-02-15 00:23:14.000000000 +0100
+++ okular-4.6.4/okular/core/textdocumentgenerator.h	2011-06-14 23:54:25.684769722 +0200
@@ -10,10 +10,10 @@
 #ifndef _OKULAR_TEXTDOCUMENTGENERATOR_H_
 #define _OKULAR_TEXTDOCUMENTGENERATOR_H_
 
-#include <okular/core/okular_export.h>
+#include "okular_export.h"
 
-#include <okular/core/document.h>
-#include <okular/core/generator.h>
+#include "document.h"
+#include "generator.h"
 
 class QTextBlock;
 class QTextDocument;
diff -ruN okular-4.6.4.orig/okular/core/textpage.cpp okular-4.6.4/okular/core/textpage.cpp
--- okular-4.6.4.orig/okular/core/textpage.cpp	2010-09-01 20:53:49.000000000 +0200
+++ okular-4.6.4/okular/core/textpage.cpp	2011-06-14 23:54:25.684769722 +0200
@@ -510,7 +510,7 @@
             }
             SearchPoint* sp = *sIt;
             sp->it_begin = it_begin;
-            sp->it_end = it;
+            sp->it_end = it - 1;
             sp->offset_begin = j;
             sp->offset_end = j + qMin( queryLeft, len );
             ret->simplify();
diff -ruN okular-4.6.4.orig/okular/core/textpage.h okular-4.6.4/okular/core/textpage.h
--- okular-4.6.4.orig/okular/core/textpage.h	2009-11-12 22:48:43.000000000 +0100
+++ okular-4.6.4/okular/core/textpage.h	2011-06-14 23:54:25.684769722 +0200
@@ -13,8 +13,8 @@
 #include <QtCore/QList>
 #include <QtCore/QString>
 
-#include <okular/core/okular_export.h>
-#include <okular/core/global.h>
+#include "okular_export.h"
+#include "global.h"
 
 class QMatrix;
 
diff -ruN okular-4.6.4.orig/okular/core/utils.h okular-4.6.4/okular/core/utils.h
--- okular-4.6.4.orig/okular/core/utils.h	2009-06-09 03:03:50.000000000 +0200
+++ okular-4.6.4/okular/core/utils.h	2011-06-14 23:54:25.684769722 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_UTILS_H_
 #define _OKULAR_UTILS_H_
 
-#include <okular/core/okular_export.h>
-#include <okular/core/area.h>
+#include "okular_export.h"
+#include "area.h"
 
 class QRect;
 class QImage;
diff -ruN okular-4.6.4.orig/okular/core/version.h okular-4.6.4/okular/core/version.h
--- okular-4.6.4.orig/okular/core/version.h	2011-02-24 15:38:41.000000000 +0100
+++ okular-4.6.4/okular/core/version.h	2011-06-14 23:54:25.684769722 +0200
@@ -10,10 +10,10 @@
 #ifndef _OKULAR_VERSION_H_
 #define _OKULAR_VERSION_H_
 
-#define OKULAR_VERSION_STRING "0.12.1"
+#define OKULAR_VERSION_STRING "0.12.4"
 #define OKULAR_VERSION_MAJOR 0
 #define OKULAR_VERSION_MINOR 12
-#define OKULAR_VERSION_RELEASE 1
+#define OKULAR_VERSION_RELEASE 4
 #define OKULAR_MAKE_VERSION( a,b,c ) (((a) << 16) | ((b) << 8) | (c))
 
 #define OKULAR_VERSION \
diff -ruN okular-4.6.4.orig/okular/generators/CMakeLists.txt okular-4.6.4/okular/generators/CMakeLists.txt
--- okular-4.6.4.orig/okular/generators/CMakeLists.txt	2009-12-12 14:57:32.000000000 +0100
+++ okular-4.6.4/okular/generators/CMakeLists.txt	2011-06-14 23:54:49.137769847 +0200
@@ -73,4 +73,3 @@
   add_subdirectory(epub)
 endif(EPUB_FOUND)
 
-add_subdirectory(mobipocket)
diff -ruN okular-4.6.4.orig/okular/generators/chm/generator_chm.cpp okular-4.6.4/okular/generators/chm/generator_chm.cpp
--- okular-4.6.4.orig/okular/generators/chm/generator_chm.cpp	2010-01-29 15:56:51.000000000 +0100
+++ okular-4.6.4/okular/generators/chm/generator_chm.cpp	2011-06-14 23:54:49.137769847 +0200
@@ -24,11 +24,11 @@
 #include <dom/dom_node.h>
 #include <dom/dom_html.h>
 
-#include <okular/core/action.h>
-#include <okular/core/observer.h> //for PAGEVIEW_ID
-#include <okular/core/page.h>
-#include <okular/core/textpage.h>
-#include <okular/core/utils.h>
+#include <core/action.h>
+#include <core/observer.h> //for PAGEVIEW_ID
+#include <core/page.h>
+#include <core/textpage.h>
+#include <core/utils.h>
 
 static KAboutData createAboutData()
 {
diff -ruN okular-4.6.4.orig/okular/generators/chm/generator_chm.h okular-4.6.4/okular/generators/chm/generator_chm.h
--- okular-4.6.4.orig/okular/generators/chm/generator_chm.h	2009-05-11 18:02:35.000000000 +0200
+++ okular-4.6.4/okular/generators/chm/generator_chm.h	2011-06-14 23:54:49.137769847 +0200
@@ -11,8 +11,8 @@
 #ifndef _OKULAR_CHMGENERATOR_H_
 #define _OKULAR_CHMGENERATOR_H_ 
 
-#include <okular/core/document.h>
-#include <okular/core/generator.h>
+#include <core/document.h>
+#include <core/generator.h>
 
 #include "lib/libchmfile.h"
 
diff -ruN okular-4.6.4.orig/okular/generators/chm/okularApplication_chm.desktop okular-4.6.4/okular/generators/chm/okularApplication_chm.desktop
--- okular-4.6.4.orig/okular/generators/chm/okularApplication_chm.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/chm/okularApplication_chm.desktop	2011-06-14 23:54:49.137769847 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/chm/okularChm.desktop okular-4.6.4/okular/generators/chm/okularChm.desktop
--- okular-4.6.4.orig/okular/generators/chm/okularChm.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/chm/okularChm.desktop	2011-06-14 23:54:49.137769847 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/comicbook/generator_comicbook.cpp okular-4.6.4/okular/generators/comicbook/generator_comicbook.cpp
--- okular-4.6.4.orig/okular/generators/comicbook/generator_comicbook.cpp	2011-01-06 00:46:08.000000000 +0100
+++ okular-4.6.4/okular/generators/comicbook/generator_comicbook.cpp	2011-06-14 23:54:49.137769847 +0200
@@ -15,9 +15,9 @@
 #include <kaboutdata.h>
 #include <klocale.h>
 
-#include <okular/core/document.h>
-#include <okular/core/page.h>
-#include <okular/core/fileprinter.h>
+#include <core/document.h>
+#include <core/page.h>
+#include <core/fileprinter.h>
 
 static KAboutData createAboutData()
 {
diff -ruN okular-4.6.4.orig/okular/generators/comicbook/okularApplication_comicbook.desktop okular-4.6.4/okular/generators/comicbook/okularApplication_comicbook.desktop
--- okular-4.6.4.orig/okular/generators/comicbook/okularApplication_comicbook.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/comicbook/okularApplication_comicbook.desktop	2011-06-14 23:54:49.137769847 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/comicbook/okularComicbook.desktop okular-4.6.4/okular/generators/comicbook/okularComicbook.desktop
--- okular-4.6.4.orig/okular/generators/comicbook/okularComicbook.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/comicbook/okularComicbook.desktop	2011-06-14 23:54:49.137769847 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/djvu/generator_djvu.cpp okular-4.6.4/okular/generators/djvu/generator_djvu.cpp
--- okular-4.6.4.orig/okular/generators/djvu/generator_djvu.cpp	2011-01-06 00:46:08.000000000 +0100
+++ okular-4.6.4/okular/generators/djvu/generator_djvu.cpp	2011-06-14 23:54:49.137769847 +0200
@@ -9,14 +9,14 @@
 
 #include "generator_djvu.h"
 
-#include <okular/core/action.h>
-#include <okular/core/annotations.h>
-#include <okular/core/area.h>
-#include <okular/core/document.h>
-#include <okular/core/page.h>
-#include <okular/core/textpage.h>
-#include <okular/core/utils.h>
-#include <okular/core/fileprinter.h>
+#include <core/action.h>
+#include <core/annotations.h>
+#include <core/area.h>
+#include <core/document.h>
+#include <core/page.h>
+#include <core/textpage.h>
+#include <core/utils.h>
+#include <core/fileprinter.h>
 
 #include <qdom.h>
 #include <qmutex.h>
diff -ruN okular-4.6.4.orig/okular/generators/djvu/generator_djvu.h okular-4.6.4/okular/generators/djvu/generator_djvu.h
--- okular-4.6.4.orig/okular/generators/djvu/generator_djvu.h	2007-12-02 21:57:24.000000000 +0100
+++ okular-4.6.4/okular/generators/djvu/generator_djvu.h	2011-06-14 23:54:49.138769833 +0200
@@ -10,7 +10,7 @@
 #ifndef _GENERATOR_DJVU_H_
 #define _GENERATOR_DJVU_H_
 
-#include <okular/core/generator.h>
+#include <core/generator.h>
 
 #include <qvector.h>
 
diff -ruN okular-4.6.4.orig/okular/generators/djvu/okularApplication_djvu.desktop okular-4.6.4/okular/generators/djvu/okularApplication_djvu.desktop
--- okular-4.6.4.orig/okular/generators/djvu/okularApplication_djvu.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/djvu/okularApplication_djvu.desktop	2011-06-14 23:54:49.138769833 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/djvu/okularDjvu.desktop okular-4.6.4/okular/generators/djvu/okularDjvu.desktop
--- okular-4.6.4.orig/okular/generators/djvu/okularDjvu.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/djvu/okularDjvu.desktop	2011-06-14 23:54:49.138769833 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/dvi/dviexport.cpp okular-4.6.4/okular/generators/dvi/dviexport.cpp
--- okular-4.6.4.orig/okular/generators/dvi/dviexport.cpp	2009-10-08 23:55:51.000000000 +0200
+++ okular-4.6.4/okular/generators/dvi/dviexport.cpp	2011-06-14 23:54:49.138769833 +0200
@@ -17,7 +17,7 @@
  */
 
 #include <config.h>
-#include <okular/core/fileprinter.h>
+#include <core/fileprinter.h>
 
 #include "dviexport.h"
 
diff -ruN okular-4.6.4.orig/okular/generators/dvi/generator_dvi.cpp okular-4.6.4/okular/generators/dvi/generator_dvi.cpp
--- okular-4.6.4.orig/okular/generators/dvi/generator_dvi.cpp	2010-07-28 17:00:52.000000000 +0200
+++ okular-4.6.4/okular/generators/dvi/generator_dvi.cpp	2011-06-14 23:54:49.138769833 +0200
@@ -7,13 +7,13 @@
  *   (at your option) any later version.                                   *
  ***************************************************************************/
 
-#include <okular/core/action.h>
-#include <okular/core/document.h>
-#include <okular/core/fileprinter.h>
-#include <okular/core/page.h>
-#include <okular/core/sourcereference.h>
-#include <okular/core/textpage.h>
-#include <okular/core/utils.h>
+#include <core/action.h>
+#include <core/document.h>
+#include <core/fileprinter.h>
+#include <core/page.h>
+#include <core/sourcereference.h>
+#include <core/textpage.h>
+#include <core/utils.h>
 
 #include "generator_dvi.h"
 #include "dviFile.h"
diff -ruN okular-4.6.4.orig/okular/generators/dvi/generator_dvi.h okular-4.6.4/okular/generators/dvi/generator_dvi.h
--- okular-4.6.4.orig/okular/generators/dvi/generator_dvi.h	2010-01-08 01:01:28.000000000 +0100
+++ okular-4.6.4/okular/generators/dvi/generator_dvi.h	2011-06-14 23:54:49.138769833 +0200
@@ -10,7 +10,7 @@
 #ifndef _DVI_GENERATOR_H_
 #define _DVI_GENERATOR_H_
 
-#include <okular/core/generator.h>
+#include <core/generator.h>
 
 #include <qbitarray.h>
 
diff -ruN okular-4.6.4.orig/okular/generators/dvi/okularApplication_dvi.desktop okular-4.6.4/okular/generators/dvi/okularApplication_dvi.desktop
--- okular-4.6.4.orig/okular/generators/dvi/okularApplication_dvi.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/dvi/okularApplication_dvi.desktop	2011-06-14 23:54:49.138769833 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/dvi/okularDvi.desktop okular-4.6.4/okular/generators/dvi/okularDvi.desktop
--- okular-4.6.4.orig/okular/generators/dvi/okularDvi.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/dvi/okularDvi.desktop	2011-06-14 23:54:49.138769833 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/epub/converter.cpp okular-4.6.4/okular/generators/epub/converter.cpp
--- okular-4.6.4.orig/okular/generators/epub/converter.cpp	2010-10-06 00:41:06.000000000 +0200
+++ okular-4.6.4/okular/generators/epub/converter.cpp	2011-06-14 23:54:49.138769833 +0200
@@ -16,7 +16,7 @@
 
 #include <kdebug.h>
 #include <klocale.h>
-#include <okular/core/action.h>
+#include <core/action.h>
 
 using namespace Epub;
 
diff -ruN okular-4.6.4.orig/okular/generators/epub/converter.h okular-4.6.4/okular/generators/epub/converter.h
--- okular-4.6.4.orig/okular/generators/epub/converter.h	2010-10-05 22:48:29.000000000 +0200
+++ okular-4.6.4/okular/generators/epub/converter.h	2011-06-14 23:54:49.138769833 +0200
@@ -10,8 +10,8 @@
 #ifndef EPUB_CONVERTER_H
 #define EPUB_CONVERTER_H
 
-#include <okular/core/textdocumentgenerator.h>
-#include <okular/core/document.h>
+#include <core/textdocumentgenerator.h>
+#include <core/document.h>
 
 #include "epubdocument.h"
 
diff -ruN okular-4.6.4.orig/okular/generators/epub/generator_epub.h okular-4.6.4/okular/generators/epub/generator_epub.h
--- okular-4.6.4.orig/okular/generators/epub/generator_epub.h	2010-10-05 22:48:29.000000000 +0200
+++ okular-4.6.4/okular/generators/epub/generator_epub.h	2011-06-14 23:54:49.138769833 +0200
@@ -9,7 +9,7 @@
 
 #ifndef _OKULAR_GENERATOR_EPUB_H_
 #define _OKULAR_GENERATOR_EPUB_H_
-#include <okular/core/textdocumentgenerator.h>
+#include <core/textdocumentgenerator.h>
 
 class EPubGenerator : public Okular::TextDocumentGenerator
 {
diff -ruN okular-4.6.4.orig/okular/generators/epub/okularApplication_epub.desktop okular-4.6.4/okular/generators/epub/okularApplication_epub.desktop
--- okular-4.6.4.orig/okular/generators/epub/okularApplication_epub.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/epub/okularApplication_epub.desktop	2011-06-14 23:54:49.138769833 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/epub/okularEPub.desktop okular-4.6.4/okular/generators/epub/okularEPub.desktop
--- okular-4.6.4.orig/okular/generators/epub/okularEPub.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/epub/okularEPub.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/fax/generator_fax.cpp okular-4.6.4/okular/generators/fax/generator_fax.cpp
--- okular-4.6.4.orig/okular/generators/fax/generator_fax.cpp	2011-01-06 00:46:08.000000000 +0100
+++ okular-4.6.4/okular/generators/fax/generator_fax.cpp	2011-06-14 23:54:49.139769820 +0200
@@ -17,8 +17,8 @@
 #include <kaboutdata.h>
 #include <klocale.h>
 
-#include <okular/core/document.h>
-#include <okular/core/page.h>
+#include <core/document.h>
+#include <core/page.h>
 
 static KAboutData createAboutData()
 {
diff -ruN okular-4.6.4.orig/okular/generators/fax/okularApplication_fax.desktop okular-4.6.4/okular/generators/fax/okularApplication_fax.desktop
--- okular-4.6.4.orig/okular/generators/fax/okularApplication_fax.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/fax/okularApplication_fax.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/fax/okularFax.desktop okular-4.6.4/okular/generators/fax/okularFax.desktop
--- okular-4.6.4.orig/okular/generators/fax/okularFax.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/fax/okularFax.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/fictionbook/converter.cpp okular-4.6.4/okular/generators/fictionbook/converter.cpp
--- okular-4.6.4.orig/okular/generators/fictionbook/converter.cpp	2010-04-02 20:49:13.000000000 +0200
+++ okular-4.6.4/okular/generators/fictionbook/converter.cpp	2011-06-14 23:54:49.139769820 +0200
@@ -21,8 +21,8 @@
 #include <kglobal.h>
 #include <klocale.h>
 
-#include <okular/core/action.h>
-#include <okular/core/document.h>
+#include <core/action.h>
+#include <core/document.h>
 
 #include "document.h"
 
diff -ruN okular-4.6.4.orig/okular/generators/fictionbook/converter.h okular-4.6.4/okular/generators/fictionbook/converter.h
--- okular-4.6.4.orig/okular/generators/fictionbook/converter.h	2009-05-23 00:56:52.000000000 +0200
+++ okular-4.6.4/okular/generators/fictionbook/converter.h	2011-06-14 23:54:49.139769820 +0200
@@ -10,7 +10,7 @@
 #ifndef FICTIONBOOK_CONVERTER_H
 #define FICTIONBOOK_CONVERTER_H
 
-#include <okular/core/textdocumentgenerator.h>
+#include <core/textdocumentgenerator.h>
 
 class QDomElement;
 class QTextCursor;
diff -ruN okular-4.6.4.orig/okular/generators/fictionbook/generator_fb.h okular-4.6.4/okular/generators/fictionbook/generator_fb.h
--- okular-4.6.4.orig/okular/generators/fictionbook/generator_fb.h	2007-12-02 21:57:24.000000000 +0100
+++ okular-4.6.4/okular/generators/fictionbook/generator_fb.h	2011-06-14 23:54:49.139769820 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_GENERATOR_FB_H_
 #define _OKULAR_GENERATOR_FB_H_
 
-#include <okular/core/textdocumentgenerator.h>
+#include <core/textdocumentgenerator.h>
 
 class FictionBookGenerator : public Okular::TextDocumentGenerator
 {
diff -ruN okular-4.6.4.orig/okular/generators/fictionbook/okularApplication_fb.desktop okular-4.6.4/okular/generators/fictionbook/okularApplication_fb.desktop
--- okular-4.6.4.orig/okular/generators/fictionbook/okularApplication_fb.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/fictionbook/okularApplication_fb.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/fictionbook/okularFb.desktop okular-4.6.4/okular/generators/fictionbook/okularFb.desktop
--- okular-4.6.4.orig/okular/generators/fictionbook/okularFb.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/fictionbook/okularFb.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/kimgio/generator_kimgio.cpp okular-4.6.4/okular/generators/kimgio/generator_kimgio.cpp
--- okular-4.6.4.orig/okular/generators/kimgio/generator_kimgio.cpp	2010-12-28 00:03:13.000000000 +0100
+++ okular-4.6.4/okular/generators/kimgio/generator_kimgio.cpp	2011-06-14 23:54:49.139769820 +0200
@@ -23,7 +23,7 @@
 #include <kimageio.h>
 #include <klocale.h>
 
-#include <okular/core/page.h>
+#include <core/page.h>
 
 static KAboutData createAboutData()
 {
diff -ruN okular-4.6.4.orig/okular/generators/kimgio/okularApplication_kimgio.desktop okular-4.6.4/okular/generators/kimgio/okularApplication_kimgio.desktop
--- okular-4.6.4.orig/okular/generators/kimgio/okularApplication_kimgio.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/kimgio/okularApplication_kimgio.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/kimgio/okularKimgio.desktop okular-4.6.4/okular/generators/kimgio/okularKimgio.desktop
--- okular-4.6.4.orig/okular/generators/kimgio/okularKimgio.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/kimgio/okularKimgio.desktop	2011-06-14 23:54:49.139769820 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/CMakeLists.txt okular-4.6.4/okular/generators/mobipocket/CMakeLists.txt
--- okular-4.6.4.orig/okular/generators/mobipocket/CMakeLists.txt	2010-10-19 19:00:35.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
@@ -1,24 +0,0 @@
-include_directories(
-                    ${PROJECT_SOURCE_DIR}/../libs/mobipocket
-                   )
-
-########### next target ###############
-
-set(okularGenerator_mobi_PART_SRCS
-  ${PROJECT_SOURCE_DIR}/../libs/mobipocket/mobipocket.cpp
-  ${PROJECT_SOURCE_DIR}/../libs/mobipocket/decompressor.cpp
-  mobidocument.cpp
-  generator_mobi.cpp
-  converter.cpp
-)
-
-kde4_add_plugin(okularGenerator_mobi ${okularGenerator_mobi_PART_SRCS})
-
-target_link_libraries(okularGenerator_mobi okularcore ${KDE4_KDECORE_LIBS} ${QT_QTGUI_LIBRARY})
-
-install(TARGETS okularGenerator_mobi DESTINATION ${PLUGIN_INSTALL_DIR})
-
-########### install files ###############
-
-install( FILES libokularGenerator_mobi.desktop okularMobi.desktop  DESTINATION  ${SERVICES_INSTALL_DIR} )
-install( PROGRAMS okularApplication_mobi.desktop  DESTINATION  ${XDG_APPS_INSTALL_DIR} )
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/Messages.sh okular-4.6.4/okular/generators/mobipocket/Messages.sh
--- okular-4.6.4.orig/okular/generators/mobipocket/Messages.sh	2009-02-20 11:28:04.000000000 +0100
+++ okular-4.6.4/okular/generators/mobipocket/Messages.sh	1970-01-01 01:00:00.000000000 +0100
@@ -1,2 +0,0 @@
-#!/bin/sh
-$XGETTEXT *.cpp -o $podir/okular_mobi.pot
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/converter.cpp okular-4.6.4/okular/generators/mobipocket/converter.cpp
--- okular-4.6.4.orig/okular/generators/mobipocket/converter.cpp	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/converter.cpp	1970-01-01 01:00:00.000000000 +0100
@@ -1,107 +0,0 @@
-/***************************************************************************
- *   Copyright (C) 2008 by Jakub Stachowski <qbast@go2.pl>                 *
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- ***************************************************************************/
-
-#include "converter.h"
-
-#include <QtGui/QAbstractTextDocumentLayout>
-#include <QtGui/QTextDocument>
-#include <QtGui/QTextBlock>
-#include <QtGui/QTextFrame>
-#include <QtGui/QTextDocumentFragment>
-#include <QtCore/QDebug>
-#include <QtCore/QFile>
-#include "mobipocket.h"
-
-#include <klocale.h>
-#include <okular/core/action.h>
-
-using namespace Mobi;
-
-Converter::Converter() 
-{
-  
-}
-
-Converter::~Converter()
-{    
-}
-
-void Converter::handleMetadata(const QMap<Mobipocket::Document::MetaKey,QString> metadata)
-{
-  QMapIterator<Mobipocket::Document::MetaKey,QString> it(metadata);
-  while (it.hasNext()) {
-    it.next();
-    switch (it.key()) {
-        case Mobipocket::Document::Title: addMetaData(Okular::DocumentInfo::Title, it.value()); break;
-        case Mobipocket::Document::Author: addMetaData(Okular::DocumentInfo::Author, it.value()); break;
-        case Mobipocket::Document::Description: addMetaData(Okular::DocumentInfo::Description, it.value()); break;
-        case Mobipocket::Document::Subject: addMetaData(Okular::DocumentInfo::Subject, it.value()); break;
-        case Mobipocket::Document::Copyright: addMetaData(Okular::DocumentInfo::Copyright, it.value()); break;
-    }
-  }
-}
-
-QTextDocument* Converter::convert( const QString &fileName )
-{
-  MobiDocument* newDocument=new MobiDocument(fileName);
-  if (!newDocument->mobi()->isValid()) {
-    emit error(i18n("Error while opening the Mobipocket document."), -1);
-    delete newDocument;
-    return NULL;
-  }
-  if (newDocument->mobi()->hasDRM()) {
-    emit error(i18n("This book is protected by DRM and can be displayed only on designated device"), -1);
-    delete newDocument;
-    return NULL;
-  }
-  
-  handleMetadata(newDocument->mobi()->metadata());
-  newDocument->setPageSize(QSizeF(600, 800));
-
-  QTextFrameFormat frameFormat;
-  frameFormat.setMargin( 20 );
-  QTextFrame *rootFrame = newDocument->rootFrame();
-  rootFrame->setFrameFormat( frameFormat ); 
-  QMap<QString,QPair<int,int> > links;
-  QMap<QString,QTextBlock> targets;
-
-  // go over whole document and add all <a> tags to links or targets map
-  for (QTextBlock it = newDocument->begin(); it != newDocument->end(); it = it.next()) 
-   for (QTextBlock::iterator fit=it.begin(); !fit.atEnd(); ++fit) {
-    QTextFragment frag=fit.fragment();
-    QTextCharFormat format=frag.charFormat();
-    if (!format.isAnchor()) continue;
-    //link
-    if (!format.anchorHref().isEmpty()) links[format.anchorHref()]=
-      QPair<int,int>(frag.position(), frag.position()+frag.length());
-    if (!format.anchorNames().isEmpty()) {
-      // link targets
-      Q_FOREACH(const QString& name, format.anchorNames()) 
-	targets['#'+name]=it;
-    }
-  }
-
-  // create link actions
-  QMapIterator<QString,QPair<int,int> > it(links);
-  while (it.hasNext()) {
-    it.next();
-    QUrl u(it.key());
-    // external or internal link
-    if (!u.isRelative()) emit addAction(new Okular::BrowseAction(it.key()), it.value().first, it.value().second);
-    else {
-      // is there valid target?
-      if (!targets.contains( it.key() ) || !targets[it.key()].isValid()) continue;
-      emit addAction(new Okular::GotoAction(QString(), calculateViewport( newDocument, targets[it.key()] )),
-	    it.value().first, it.value().second);
-    }
-    
-  }
-
-  return newDocument;
-}
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/converter.h okular-4.6.4/okular/generators/mobipocket/converter.h
--- okular-4.6.4.orig/okular/generators/mobipocket/converter.h	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/converter.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,32 +0,0 @@
-/***************************************************************************
- *   Copyright (C) 2008 by Jakub Stachowski <qbast@go2.pl>                 *
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- ***************************************************************************/
-#ifndef MOBI_CONVERTER_H
-#define MOBI_CONVERTER_H
-
-#include <okular/core/textdocumentgenerator.h>
-#include <okular/core/document.h>
-
-#include "mobidocument.h"
-#include "mobipocket.h"
-
-
-namespace Mobi {
-  class Converter : public Okular::TextDocumentConverter
-    {
-    public:
-      Converter();
-      ~Converter();
-      
-      virtual QTextDocument *convert( const QString &fileName );
-    private:
-      void handleMetadata(const QMap<Mobipocket::Document::MetaKey, QString> metadata);
-    };
-}
-
-#endif
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/generator_mobi.cpp okular-4.6.4/okular/generators/mobipocket/generator_mobi.cpp
--- okular-4.6.4.orig/okular/generators/mobipocket/generator_mobi.cpp	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/generator_mobi.cpp	1970-01-01 01:00:00.000000000 +0100
@@ -1,37 +0,0 @@
-/***************************************************************************
- *   Copyright (C) 2008 by Ely Levy <elylevy@cs.huji.ac.il>                *
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- ***************************************************************************/
-#include "generator_mobi.h"
-
-#include "converter.h"
-
-#include <kaboutdata.h>
-
-static KAboutData createAboutData()
-{
-  KAboutData aboutData(
-                       "okular_mobi",
-                       "okular_mobi",
-                       ki18n("Mobipocket Backend"),
-                       "0.1",
-                       ki18n("A mobipocket backend"),
-                       KAboutData::License_GPL,
-                       ki18n("© 2008-2009 Jakub Stachowski")
-                       );
-  aboutData.addAuthor(ki18n("Jakub Stachowski"), KLocalizedString(),
-                      "qbast@go2.pl");
-
-  return aboutData;
-}
-
-OKULAR_EXPORT_PLUGIN( MobiGenerator, createAboutData() )
-
-MobiGenerator::MobiGenerator( QObject *parent, const QVariantList &args )
-: Okular::TextDocumentGenerator( new Mobi::Converter, parent, args )
-{
-}
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/generator_mobi.h okular-4.6.4/okular/generators/mobipocket/generator_mobi.h
--- okular-4.6.4.orig/okular/generators/mobipocket/generator_mobi.h	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/generator_mobi.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,20 +0,0 @@
-/***************************************************************************
- *   Copyright (C) 2008 by Ely Levy <elylevy@cs.huji.ac.il>                *
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- ***************************************************************************/
-#ifndef OKULAR_GENERATOR_MOBI_H
-#define OKULAR_GENERATOR_MOBI_H
-#include <okular/core/textdocumentgenerator.h>
-
-class MobiGenerator : public Okular::TextDocumentGenerator
-{
- public:
-  MobiGenerator( QObject *parent, const QVariantList &args );
-  ~MobiGenerator() {}
-};
-
-#endif
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/libokularGenerator_mobi.desktop okular-4.6.4/okular/generators/mobipocket/libokularGenerator_mobi.desktop
--- okular-4.6.4.orig/okular/generators/mobipocket/libokularGenerator_mobi.desktop	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/libokularGenerator_mobi.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,108 +0,0 @@
-[Desktop Entry]
-Type=Service
-Name=Mobipocket document
-Name[ar]=مستند Mobipocket
-Name[ast]=Documentu Mobipocket
-Name[bg]=Документ Mobipocket
-Name[ca]=Document Mobipocket
-Name[ca@valencia]=Document Mobipocket
-Name[cs]=Mobipocket dokument
-Name[da]=Mobipocket-dokument
-Name[de]=Mobipocket-Dokument
-Name[el]=Έγγραφο Mobipocket
-Name[en_GB]=Mobipocket document
-Name[es]=Documento Mobipocket
-Name[et]=Mobipocketi dokument
-Name[eu]=Mobipocket dokumentua
-Name[fi]=Mobipocket-asiakirja
-Name[fr]=Document Mobipocket
-Name[ga]=Cáipéis Mobipocket
-Name[gl]=Documento Mobipocket
-Name[hr]=Dokument oblika Mobipocket
-Name[hu]=Mobipocket-dokumentum
-Name[is]=Mobipocket skjal
-Name[it]=Documento Mobipocket
-Name[ja]=Mobipocket 文書
-Name[kk]=Mobipocket құжаты
-Name[km]=ឯកសារ Mobipocket
-Name[ko]=Mobipocket 문서
-Name[lt]=Mobipocket dokumentas
-Name[lv]=Mobipocket dokuments
-Name[nb]=Mobipocket-dokument
-Name[nds]=Mobipocket-Dokment
-Name[nl]=Mobipocket-document
-Name[nn]=Mobipocket-dokument
-Name[pa]=ਮੋਬੀਪਾਕਟ ਡੌਕੂਮੈਂਟ
-Name[pl]=Dokument Mobipocket
-Name[pt]=Documento do Mobipocket
-Name[pt_BR]=Documento Mobipocket
-Name[ru]=Документ Mobipocket
-Name[sk]=Mobipocket dokument
-Name[sl]=Dokument Mobipocket
-Name[sr]=Мобипокет
-Name[sr@ijekavian]=Мобипокет
-Name[sr@ijekavianlatin]=Mobipocket
-Name[sr@latin]=Mobipocket
-Name[sv]=Mobipocket-dokument
-Name[th]=เอกสาร Mobipocket
-Name[tr]=Mobipocket belgesi
-Name[uk]=Документ Mobipocket
-Name[x-test]=xxMobipocket documentxx
-Name[zh_CN]=Mobipocket 文档
-Name[zh_TW]=Mobipocket 文件
-Comment=Mobipocket backend for Okular
-Comment[ar]=خلفية Mobipocket لأوكلار
-Comment[ast]=Motor Mobipocket pa Okular
-Comment[bg]=Ядро на Okular за Mobipocket
-Comment[ca]=Dorsal de Mobipocket per a l'Okular
-Comment[ca@valencia]=Dorsal de Mobipocket per a l'Okular
-Comment[cs]= Implementace Mobipocket pro Okular
-Comment[da]=Mobipocket-motor til Okular
-Comment[de]=Ein Anzeigemodul für Mobipocket-Dateien in Okular
-Comment[el]=Σύστημα υποστήριξης Mobipocket για το Okular
-Comment[en_GB]=Mobipocket backend for Okular
-Comment[es]=Motor Mobipocket para Okular
-Comment[et]=Okulari Mobipocketi taustaprogramm
-Comment[eu]=Okular-en Mobipocket motorra
-Comment[fi]=Mobipocket-taustaosa Okular-lukijalle
-Comment[fr]=Interface d'Okular pour le format Mobipocket
-Comment[ga]=Inneall Mobipocket le haghaidh Okular
-Comment[gl]=Infraestrutura de Mobipocket para Okular
-Comment[hr]=Podrška za Mobipocket za Okular
-Comment[hu]=Mobipocket-bővítmény az Okularhoz
-Comment[is]=Mobipocket stuðningur fyrir Okular
-Comment[it]=Backend Mobipocket per Okular
-Comment[ja]=Okular の Mobipocket 用バックエンド
-Comment[kk]=Okular-дың Mobipocket тетігі
-Comment[km]=កម្មវិធី​ខាងក្រោយ​សម្រាប់ Mobipocket សម្រាប់ Okular
-Comment[ko]=Okular의 Mobipocket 백엔드
-Comment[lt]=Mobipocket programinė sąsaja, skirta Okular
-Comment[lv]=Mobipocket Okular aizmugure
-Comment[nb]=Mobipocket-motor for okular
-Comment[nds]=Mobipocket-Hülppropgramm för Okular
-Comment[nl]=Mobipocket-backend voor Okular
-Comment[nn]=Mobipocket-motor for Okular
-Comment[pa]=ਓਕੁਲਾਰ ਲਈ ਮੋਬੀਪਾਕਟ ਬੈਕਐਂਡ
-Comment[pl]=Obsługa formatu Mobipocket dla Okulara
-Comment[pt]=Infra-estrutura do Mobipocket para o Okular
-Comment[pt_BR]=Infraestrutura Mobipocket para o Okular
-Comment[ru]=Модуль поддержки формата Mobipocket для Okular
-Comment[sk]=Backend Mobipocket pre Okular
-Comment[sl]=Hrbtenica za Mobipocket Okular
-Comment[sr]=Позадина Мобипокета за Окулар
-Comment[sr@ijekavian]=Позадина Мобипокета за Окулар
-Comment[sr@ijekavianlatin]=Pozadina Mobipocketa za Okular
-Comment[sr@latin]=Pozadina Mobipocketa za Okular
-Comment[sv]=Mobipocket-gränssnitt för Okular
-Comment[th]=แบ็กเอนด์แฟ้มแบบ Mobipocket สำหรับโอกูลาร์
-Comment[tr]=Okular için Mobipocket arka ucu
-Comment[uk]=Сервер Mobipocket для Okular
-Comment[x-test]=xxMobipocket backend for Okularxx
-Comment[zh_CN]=Okular 的 Mobipocket 格式后端
-Comment[zh_TW]=Okular 的 Mobipocket 後端介面
-X-KDE-ServiceTypes=okular/Generator
-MimeType=application/x-mobipocket-ebook;
-X-KDE-Library=okularGenerator_mobi
-X-KDE-Priority=1
-X-KDE-okularAPIVersion=1
-X-KDE-okularHasInternalSettings=false
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/mobidocument.cpp okular-4.6.4/okular/generators/mobipocket/mobidocument.cpp
--- okular-4.6.4.orig/okular/generators/mobipocket/mobidocument.cpp	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/mobidocument.cpp	1970-01-01 01:00:00.000000000 +0100
@@ -1,97 +0,0 @@
-/***************************************************************************
- *   Copyright (C) 2008 by Jakub Stachowski <qbast@go2.pl>                 *
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- ***************************************************************************/
-#include "mobidocument.h"
-#include "mobipocket.h"
-#include "qfilestream.h"
-#include <QtCore/QFile>
-#include <QtCore/QRegExp>
-#include <kdebug.h>
-
-using namespace Mobi;
-
-MobiDocument::MobiDocument(const QString &fileName) : QTextDocument() 
-{
-  file = new Mobipocket::QFileStream(fileName);
-  doc = new Mobipocket::Document(file);
-  if (doc->isValid()) {
-      QString text=doc->text();
-      QString header=text.left(1024);
-      if (header.contains("<html>") || header.contains("<HTML>")) setHtml(fixMobiMarkup(text));
-      else setPlainText(text);
-  }
-}
-
-MobiDocument::~MobiDocument() 
-{
-    delete doc;
-    delete file;
-}
-  
-QVariant MobiDocument::loadResource(int type, const QUrl &name) 
-{
-  if (type!=QTextDocument::ImageResource || name.scheme()!=QString("pdbrec")) return QVariant();
-  bool ok;
-  quint16 recnum=name.path().mid(1).toUShort(&ok);
-  if (!ok || recnum>=doc->imageCount()) return QVariant();
-   
-  QVariant resource;
-  resource.setValue(doc->getImage(recnum-1));
-  addResource(type, name, resource); 
-    
-  return resource;
-}
-
-// starting from 'pos', find position in the string that is not inside a tag
-int outsideTag(const QString& data, int pos)
-{
-  for (int i=pos-1;i>=0;i--) {
-    if (data[i]=='>') return pos;
-    if (data[i]=='<') return i;
-  }
-  return pos;
-}
-
-QString MobiDocument::fixMobiMarkup(const QString& data) 
-{
-    QString ret=data;
-    QMap<int,QString> anchorPositions;
-    static QRegExp anchors("<a(?: href=\"[^\"]*\"){0,1}[\\s]+filepos=['\"]{0,1}([\\d]+)[\"']{0,1}", Qt::CaseInsensitive);
-    int pos=0;
-
-    // find all link destinations
-    while ( (pos=anchors.indexIn( data, pos ))!=-1) {
-	int filepos=anchors.cap( 1 ).toUInt(  );
-	if (filepos) anchorPositions[filepos]=anchors.cap(1);
-	pos+=anchors.matchedLength();
-    }
-
-    // put HTML anchors in all link destinations
-    int offset=0;
-    QMapIterator<int,QString> it(anchorPositions);
-    while (it.hasNext()) {
-      it.next();
-      // link pointing outside the document, ignore
-      if ( (it.key()+offset) >= ret.size()) continue;
-      int fixedpos=outsideTag(ret, it.key()+offset);
-      ret.insert(fixedpos,QString("<a name=\"")+it.value()+QString("\">&nbsp;</a>"));
-      // inserting anchor shifts all offsets after the anchor
-      offset+=21+it.value().size();
-    }
-
-    // replace links referencing filepos with normal internal links
-    ret.replace(anchors,"<a href=\"#\\1\"");
-    // Mobipocket uses strange variang of IMG tags: <img recindex="3232"> where recindex is number of 
-    // record containing image
-    static QRegExp imgs("<img.*recindex=\"([\\d]*)\".*>", Qt::CaseInsensitive);
-    
-    imgs.setMinimal(true);
-    ret.replace(imgs,"<img src=\"pdbrec:/\\1\">");
-    ret.replace("<mbp:pagebreak/>","<p style=\"page-break-after:always\"></p>");
-    return ret;
-}
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/mobidocument.h okular-4.6.4/okular/generators/mobipocket/mobidocument.h
--- okular-4.6.4.orig/okular/generators/mobipocket/mobidocument.h	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/mobidocument.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,42 +0,0 @@
-/***************************************************************************
- *   Copyright (C) 2008 by Jakub Stachowski <qbast@go2.pl>                 *
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- ***************************************************************************/
-#ifndef MOBI_DOCUMENT_H
-#define MOBI_DOCUMENT_H
-
-#include <QTextDocument>
-#include <QUrl>
-#include <QVariant>
-
-class QFile;
-namespace Mobipocket {
-class Document;
-class QFileStream;
-}
-
-namespace Mobi {
-
-  class MobiDocument : public QTextDocument {
-      
-  public:
-    MobiDocument(const QString &fileName);  
-    ~MobiDocument();   
-    
-    Mobipocket::Document* mobi() const { return doc; }
-    
-  protected:
-    virtual QVariant loadResource(int type, const QUrl &name);
-    
-  private:
-    QString fixMobiMarkup(const QString& data);
-    Mobipocket::Document *doc;
-    Mobipocket::QFileStream* file;
-  };
-
-}
-#endif
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/okularApplication_mobi.desktop okular-4.6.4/okular/generators/mobipocket/okularApplication_mobi.desktop
--- okular-4.6.4.orig/okular/generators/mobipocket/okularApplication_mobi.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/okularApplication_mobi.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,122 +0,0 @@
-[Desktop Entry]
-MimeType=application/x-mobipocket-ebook;
-Terminal=false
-Name=Okular
-Name[ar]=اوكلار
-Name[ast]=Okular
-Name[bg]=Okular
-Name[ca]=Okular
-Name[ca@valencia]=Okular
-Name[cs]=Okular
-Name[da]=Okular
-Name[de]=Okular
-Name[el]=Okular
-Name[en_GB]=Okular
-Name[eo]=Okular
-Name[es]=Okular
-Name[et]=Okular
-Name[eu]=Okular
-Name[fi]=Okular
-Name[fr]=Okular
-Name[ga]=Okular
-Name[gl]=Okular
-Name[hne]=ओकुलर
-Name[hr]=Okular
-Name[hu]=Okular
-Name[is]=Okular
-Name[it]=Okular
-Name[ja]=Okular
-Name[kk]=Okular
-Name[km]=Okular
-Name[ko]=Okular
-Name[ku]=Okular
-Name[lt]=Okular
-Name[lv]=Okular
-Name[nb]=Okular
-Name[nds]=Okular
-Name[nl]=Okular
-Name[nn]=Okular
-Name[pa]=ਓਕੁਲਾਰ
-Name[pl]=Okular
-Name[pt]=Okular
-Name[pt_BR]=Okular
-Name[ro]=Okular
-Name[ru]=Okular
-Name[sk]=Okular
-Name[sl]=Okular
-Name[sr]=Окулар
-Name[sr@ijekavian]=Окулар
-Name[sr@ijekavianlatin]=Okular
-Name[sr@latin]=Okular
-Name[sv]=Okular
-Name[th]=โอกูลาร์
-Name[tr]=Okular
-Name[uk]=Okular
-Name[x-test]=xxOkularxx
-Name[zh_CN]=Okular
-Name[zh_TW]=Okular
-GenericName=Document Viewer
-GenericName[ar]=عارض المستندات
-GenericName[ast]=Visor de documentos
-GenericName[bg]=Преглед на доументи
-GenericName[ca]=Visualitzador de documents
-GenericName[ca@valencia]=Visualitzador de documents
-GenericName[cs]=Prohlížeč dokumentů
-GenericName[da]=Dokumentfremviser
-GenericName[de]=Dokumentenbetrachter
-GenericName[el]=Προβολέας εγγράφων
-GenericName[en_GB]=Document Viewer
-GenericName[eo]=Dokumenta rigardilo
-GenericName[es]=Visor de documentos
-GenericName[et]=Dokumendinäitaja
-GenericName[eu]=Dokumentu ikustailea
-GenericName[fa]=مشاهده‌گر سند
-GenericName[fi]=Asiakirjan näyttöohjelma
-GenericName[fr]=Visionneuse de document
-GenericName[ga]=Amharcán Cáipéisí
-GenericName[gl]=Visor de documentos
-GenericName[hi]=दस्तावेज़ प्रदर्शक
-GenericName[hne]=कागद प्रदर्सक
-GenericName[hr]=Preglednik dokumenata
-GenericName[hu]=Dokumentumnézegető
-GenericName[is]=Skjalaskoðari
-GenericName[it]=Visore di documenti
-GenericName[ja]=文書ビューア
-GenericName[kk]=Құжатты қарау құралы
-GenericName[km]=កម្មវិធី​មើល​ឯកសារ
-GenericName[ko]=문서 뷰어
-GenericName[ku]=Nîşanderê Belgeyan
-GenericName[lt]=Dokumentų žiūryklė
-GenericName[lv]=Dokumentu skatītājs
-GenericName[nb]=Dokumentviser
-GenericName[nds]=Dokmentkieker
-GenericName[ne]=कागजात दर्शक
-GenericName[nl]=Documentenviewer
-GenericName[nn]=Dokumentvisar
-GenericName[oc]=Visualizaire de documents
-GenericName[pa]=ਡੌਕੂਮੈਂਟ ਦਰਸ਼ਕ
-GenericName[pl]=Przeglądarka dokumentów
-GenericName[pt]=Visualizador de Documentos
-GenericName[pt_BR]=Visualizador de documentos
-GenericName[ro]=Vizualizor de documente
-GenericName[ru]=Просмотр документов
-GenericName[sk]=Prehliadač dokumentov
-GenericName[sl]=Pregledovalnik dokumentov
-GenericName[sr]=Приказивач докумената
-GenericName[sr@ijekavian]=Приказивач докумената
-GenericName[sr@ijekavianlatin]=Prikazivač dokumenata
-GenericName[sr@latin]=Prikazivač dokumenata
-GenericName[sv]=Dokumentvisare
-GenericName[th]=เครื่องมือแสดงเอกสาร
-GenericName[tr]=Belge Gösterici
-GenericName[uk]=Переглядач документів
-GenericName[x-test]=xxDocument Viewerxx
-GenericName[zh_CN]=文档查看器
-GenericName[zh_TW]=文件檢視器
-Exec=okular %U %i -caption "%c"
-Icon=okular
-Type=Application
-InitialPreference=1
-Categories=Qt;KDE;Graphics;Viewer;
-NoDisplay=true
-X-KDE-Keywords=mobipocket
diff -ruN okular-4.6.4.orig/okular/generators/mobipocket/okularMobi.desktop okular-4.6.4/okular/generators/mobipocket/okularMobi.desktop
--- okular-4.6.4.orig/okular/generators/mobipocket/okularMobi.desktop	2010-07-28 17:00:31.000000000 +0200
+++ okular-4.6.4/okular/generators/mobipocket/okularMobi.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,60 +0,0 @@
-[Desktop Entry]
-Icon=okular
-Name=Okular
-Name[ar]=اوكلار
-Name[ast]=Okular
-Name[bg]=Okular
-Name[ca]=Okular
-Name[ca@valencia]=Okular
-Name[cs]=Okular
-Name[da]=Okular
-Name[de]=Okular
-Name[el]=Okular
-Name[en_GB]=Okular
-Name[eo]=Okular
-Name[es]=Okular
-Name[et]=Okular
-Name[eu]=Okular
-Name[fi]=Okular
-Name[fr]=Okular
-Name[ga]=Okular
-Name[gl]=Okular
-Name[hne]=ओकुलर
-Name[hr]=Okular
-Name[hu]=Okular
-Name[is]=Okular
-Name[it]=Okular
-Name[ja]=Okular
-Name[kk]=Okular
-Name[km]=Okular
-Name[ko]=Okular
-Name[ku]=Okular
-Name[lt]=Okular
-Name[lv]=Okular
-Name[nb]=Okular
-Name[nds]=Okular
-Name[nl]=Okular
-Name[nn]=Okular
-Name[pa]=ਓਕੁਲਾਰ
-Name[pl]=Okular
-Name[pt]=Okular
-Name[pt_BR]=Okular
-Name[ro]=Okular
-Name[ru]=Okular
-Name[sk]=Okular
-Name[sl]=Okular
-Name[sr]=Окулар
-Name[sr@ijekavian]=Окулар
-Name[sr@ijekavianlatin]=Okular
-Name[sr@latin]=Okular
-Name[sv]=Okular
-Name[th]=โอกูลาร์
-Name[tr]=Okular
-Name[uk]=Okular
-Name[x-test]=xxOkularxx
-Name[zh_CN]=Okular
-Name[zh_TW]=Okular
-X-KDE-ServiceTypes=KParts/ReadOnlyPart
-X-KDE-Library=okularpart
-Type=Service
-MimeType=application/x-mobipocket-ebook;
diff -ruN okular-4.6.4.orig/okular/generators/ooo/converter.cpp okular-4.6.4/okular/generators/ooo/converter.cpp
--- okular-4.6.4.orig/okular/generators/ooo/converter.cpp	2009-06-05 19:45:59.000000000 +0200
+++ okular-4.6.4/okular/generators/ooo/converter.cpp	2011-06-14 23:54:49.140769807 +0200
@@ -20,10 +20,10 @@
 #include <QtXml/QDomText>
 #include <QtXml/QXmlSimpleReader>
 
-#include <okular/core/action.h>
-#include <okular/core/annotations.h>
-#include <okular/core/document.h>
-#include <okular/core/utils.h>
+#include <core/action.h>
+#include <core/annotations.h>
+#include <core/document.h>
+#include <core/utils.h>
 
 #include <klocale.h>
 
diff -ruN okular-4.6.4.orig/okular/generators/ooo/converter.h okular-4.6.4/okular/generators/ooo/converter.h
--- okular-4.6.4.orig/okular/generators/ooo/converter.h	2009-01-06 07:48:28.000000000 +0100
+++ okular-4.6.4/okular/generators/ooo/converter.h	2011-06-14 23:54:49.140769807 +0200
@@ -13,7 +13,7 @@
 #include <QtGui/QTextCharFormat>
 #include <QtXml/QDomDocument>
 
-#include <okular/core/textdocumentgenerator.h>
+#include <core/textdocumentgenerator.h>
 
 #include "styleinformation.h"
 
diff -ruN okular-4.6.4.orig/okular/generators/ooo/generator_ooo.h okular-4.6.4/okular/generators/ooo/generator_ooo.h
--- okular-4.6.4.orig/okular/generators/ooo/generator_ooo.h	2007-12-02 21:57:24.000000000 +0100
+++ okular-4.6.4/okular/generators/ooo/generator_ooo.h	2011-06-14 23:54:49.140769807 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_GENERATOR_OOO_H_
 #define _OKULAR_GENERATOR_OOO_H_
 
-#include <okular/core/textdocumentgenerator.h>
+#include <core/textdocumentgenerator.h>
 
 class KOOOGenerator : public Okular::TextDocumentGenerator
 {
diff -ruN okular-4.6.4.orig/okular/generators/ooo/okularApplication_ooo.desktop okular-4.6.4/okular/generators/ooo/okularApplication_ooo.desktop
--- okular-4.6.4.orig/okular/generators/ooo/okularApplication_ooo.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/ooo/okularApplication_ooo.desktop	2011-06-14 23:54:49.140769807 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/ooo/okularOoo.desktop okular-4.6.4/okular/generators/ooo/okularOoo.desktop
--- okular-4.6.4.orig/okular/generators/ooo/okularOoo.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/ooo/okularOoo.desktop	2011-06-14 23:54:49.141769801 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/plucker/generator_plucker.cpp okular-4.6.4/okular/generators/plucker/generator_plucker.cpp
--- okular-4.6.4.orig/okular/generators/plucker/generator_plucker.cpp	2011-01-06 00:46:08.000000000 +0100
+++ okular-4.6.4/okular/generators/plucker/generator_plucker.cpp	2011-06-14 23:54:49.141769801 +0200
@@ -18,7 +18,7 @@
 #include <kaboutdata.h>
 #include <klocale.h>
 
-#include <okular/core/page.h>
+#include <core/page.h>
 
 static KAboutData createAboutData()
 {
diff -ruN okular-4.6.4.orig/okular/generators/plucker/generator_plucker.h okular-4.6.4/okular/generators/plucker/generator_plucker.h
--- okular-4.6.4.orig/okular/generators/plucker/generator_plucker.h	2007-12-02 21:57:24.000000000 +0100
+++ okular-4.6.4/okular/generators/plucker/generator_plucker.h	2011-06-14 23:54:49.141769801 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_GENERATOR_PLUCKER_H_
 #define _OKULAR_GENERATOR_PLUCKER_H_
 
-#include <okular/core/document.h>
-#include <okular/core/generator.h>
+#include <core/document.h>
+#include <core/generator.h>
 
 #include <QtGui/QTextBlock>
 
diff -ruN okular-4.6.4.orig/okular/generators/plucker/okularApplication_plucker.desktop okular-4.6.4/okular/generators/plucker/okularApplication_plucker.desktop
--- okular-4.6.4.orig/okular/generators/plucker/okularApplication_plucker.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/plucker/okularApplication_plucker.desktop	2011-06-14 23:54:49.141769801 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/plucker/okularPlucker.desktop okular-4.6.4/okular/generators/plucker/okularPlucker.desktop
--- okular-4.6.4.orig/okular/generators/plucker/okularPlucker.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/plucker/okularPlucker.desktop	2011-06-14 23:54:49.141769801 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/plucker/unpluck/qunpluck.cpp okular-4.6.4/okular/generators/plucker/unpluck/qunpluck.cpp
--- okular-4.6.4.orig/okular/generators/plucker/unpluck/qunpluck.cpp	2007-10-06 05:08:55.000000000 +0200
+++ okular-4.6.4/okular/generators/plucker/unpluck/qunpluck.cpp	2011-06-14 23:54:49.141769801 +0200
@@ -26,8 +26,8 @@
 
 #include <QtGui/QLabel>
 
-#include <okular/core/action.h>
-#include <okular/core/document.h>
+#include <core/action.h>
+#include <core/document.h>
 
 #include "qunpluck.h"
 #include "image.h"
diff -ruN okular-4.6.4.orig/okular/generators/poppler/annots.cpp okular-4.6.4/okular/generators/poppler/annots.cpp
--- okular-4.6.4.orig/okular/generators/poppler/annots.cpp	2008-08-23 02:12:55.000000000 +0200
+++ okular-4.6.4/okular/generators/poppler/annots.cpp	2011-06-14 23:54:49.141769801 +0200
@@ -12,7 +12,7 @@
 // qt/kde includes
 #include <qvariant.h>
 
-#include <okular/core/annotations.h>
+#include <core/annotations.h>
 
 #include "popplerembeddedfile.h"
 #include "config-okular-poppler.h"
diff -ruN okular-4.6.4.orig/okular/generators/poppler/generator_pdf.cpp okular-4.6.4/okular/generators/poppler/generator_pdf.cpp
--- okular-4.6.4.orig/okular/generators/poppler/generator_pdf.cpp	2010-07-28 17:00:52.000000000 +0200
+++ okular-4.6.4/okular/generators/poppler/generator_pdf.cpp	2011-06-14 23:54:49.142769787 +0200
@@ -31,16 +31,16 @@
 #include <kdebug.h>
 #include <kglobal.h>
 
-#include <okular/core/action.h>
-#include <okular/core/page.h>
-#include <okular/core/annotations.h>
-#include <okular/core/movie.h>
-#include <okular/core/pagetransition.h>
-#include <okular/core/sound.h>
-#include <okular/core/sourcereference.h>
-#include <okular/core/textpage.h>
-#include <okular/core/fileprinter.h>
-#include <okular/core/utils.h>
+#include <core/action.h>
+#include <core/page.h>
+#include <core/annotations.h>
+#include <core/movie.h>
+#include <core/pagetransition.h>
+#include <core/sound.h>
+#include <core/sourcereference.h>
+#include <core/textpage.h>
+#include <core/fileprinter.h>
+#include <core/utils.h>
 
 #include <config-okular-poppler.h>
 
@@ -52,6 +52,8 @@
 #endif
 
 static const int PDFDebug = 4710;
+static const int defaultPageWidth = 595;
+static const int defaultPageHeight = 842;
 
 class PDFOptionsPage : public QWidget
 {
@@ -275,7 +277,7 @@
          "okular_poppler",
          "okular_poppler",
          ki18n( "PDF Backend" ),
-         "0.4",
+         "0.4.1",
          ki18n( "A PDF file renderer" ),
          KAboutData::License_GPL,
          ki18n( "© 2005-2008 Albert Astals Cid" )
@@ -474,49 +476,57 @@
     {
         // get xpdf page
         Poppler::Page * p = pdfdoc->page( i );
-        const QSizeF pSize = p->pageSizeF();
-        w = pSize.width() / 72.0 * dpiX;
-        h = pSize.height() / 72.0 * dpiY;
-        Okular::Rotation orientation = Okular::Rotation0;
-        switch (p->orientation())
-        {
-          case Poppler::Page::Landscape: orientation = Okular::Rotation90; break;
-          case Poppler::Page::UpsideDown: orientation = Okular::Rotation180; break;
-          case Poppler::Page::Seascape: orientation = Okular::Rotation270; break;
-          case Poppler::Page::Portrait: orientation = Okular::Rotation0; break;
-        }
-        if (rotation % 2 == 1)
-          qSwap(w,h);
-        // init a Okular::page, add transition and annotation information
-        Okular::Page * page = new Okular::Page( i, w, h, orientation );
-        addTransition( p, page );
-        if ( true ) //TODO real check
-          addAnnotations( p, page );
-        Poppler::Link * tmplink = p->action( Poppler::Page::Opening );
-        if ( tmplink )
-        {
-            page->setPageAction( Okular::Page::Opening, createLinkFromPopplerLink( tmplink ) );
-            delete tmplink;
-        }
-        tmplink = p->action( Poppler::Page::Closing );
-        if ( tmplink )
+        Okular::Page * page;
+        if (p)
         {
-            page->setPageAction( Okular::Page::Closing, createLinkFromPopplerLink( tmplink ) );
-            delete tmplink;
-        }
-        page->setDuration( p->duration() );
-        page->setLabel( p->label() );
+            const QSizeF pSize = p->pageSizeF();
+            w = pSize.width() / 72.0 * dpiX;
+            h = pSize.height() / 72.0 * dpiY;
+            Okular::Rotation orientation = Okular::Rotation0;
+            switch (p->orientation())
+            {
+            case Poppler::Page::Landscape: orientation = Okular::Rotation90; break;
+            case Poppler::Page::UpsideDown: orientation = Okular::Rotation180; break;
+            case Poppler::Page::Seascape: orientation = Okular::Rotation270; break;
+            case Poppler::Page::Portrait: orientation = Okular::Rotation0; break;
+            }
+            if (rotation % 2 == 1)
+            qSwap(w,h);
+            // init a Okular::page, add transition and annotation information
+            page = new Okular::Page( i, w, h, orientation );
+            addTransition( p, page );
+            if ( true ) //TODO real check
+            addAnnotations( p, page );
+            Poppler::Link * tmplink = p->action( Poppler::Page::Opening );
+            if ( tmplink )
+            {
+                page->setPageAction( Okular::Page::Opening, createLinkFromPopplerLink( tmplink ) );
+                delete tmplink;
+            }
+            tmplink = p->action( Poppler::Page::Closing );
+            if ( tmplink )
+            {
+                page->setPageAction( Okular::Page::Closing, createLinkFromPopplerLink( tmplink ) );
+                delete tmplink;
+            }
+            page->setDuration( p->duration() );
+            page->setLabel( p->label() );
 
-        addFormFields( p, page );
+            addFormFields( p, page );
 //        kWarning(PDFDebug).nospace() << page->width() << "x" << page->height();
 
 #ifdef PDFGENERATOR_DEBUG
-        kDebug(PDFDebug) << "load page" << i << "with rotation" << rotation << "and orientation" << orientation;
+            kDebug(PDFDebug) << "load page" << i << "with rotation" << rotation << "and orientation" << orientation;
 #endif
-	delete p;
+            delete p;
 
-        if (clear && pagesVector[i])
-            delete pagesVector[i];
+            if (clear && pagesVector[i])
+                delete pagesVector[i];
+        }
+        else
+        {
+            page = new Okular::Page( i, defaultPageWidth, defaultPageHeight, Okular::Rotation0 );
+        }
         // set the Okular::page at the right position in document's pages vector
         pagesVector[i] = page;
     }
@@ -793,6 +803,7 @@
     Poppler::Page *p = pdfdoc->page(page->number());
 
     // 2. Take data from outputdev and attach it to the Page
+    if (p)
     {
         QImage img( p->renderToImage(fakeDpiX, fakeDpiY, -1, -1, -1, -1, Poppler::Page::Rotate0 ) );
         if ( !page->isBoundingBoxKnown() )
@@ -800,8 +811,14 @@
 
         page->setPixmap( request->id(), new QPixmap( QPixmap::fromImage( img ) ) );
     }
+    else
+    {
+        QPixmap *dummyPixmap = new QPixmap( request->width(), request->height() );
+        dummyPixmap->fill( Qt::white );
+        page->setPixmap( request->id(), dummyPixmap );
+    }
 
-    if ( genObjectRects )
+    if ( p && genObjectRects )
     {
     	// TODO previously we extracted Image type rects too, but that needed porting to poppler
         // and as we are not doing anything with Image type rects i did not port it, have a look at
@@ -812,7 +829,7 @@
 
     // 3. UNLOCK [re-enables shared access]
     userMutex()->unlock();
-    if ( genTextPage )
+    if ( p && genTextPage )
     {
         QList<Poppler::TextBox*> textList = p->textList();
         const QSizeF s = p->pageSizeF();
@@ -838,16 +855,26 @@
     kDebug(PDFDebug) << "page" << page->number();
 #endif
     // build a TextList...
+    QList<Poppler::TextBox*> textList;
+    double pageWidth, pageHeight;
     Poppler::Page *pp = pdfdoc->page( page->number() );
-    userMutex()->lock();
-    QList<Poppler::TextBox*> textList = pp->textList();
-    userMutex()->unlock();
+    if (pp)
+    {
+        userMutex()->lock();
+        textList = pp->textList();
+        userMutex()->unlock();
 
-    QSizeF s = pp->pageSizeF();
-    const double pageWidth = s.width();
-    const double pageHeight = s.height();
+        QSizeF s = pp->pageSizeF();
+        pageWidth = s.width();
+        pageHeight = s.height();
 
-    delete pp;
+        delete pp;
+    }
+    else   
+    {
+        pageWidth = defaultPageWidth;
+        pageHeight = defaultPageHeight;
+    }
 
     Okular::TextPage *tp = abstractTextPage(textList, pageHeight, pageWidth, (Poppler::Page::Rotation)page->orientation());
     qDeleteAll(textList);
@@ -1088,9 +1115,13 @@
         int num = document()->pages();
         for ( int i = 0; i < num; ++i )
         {
+            QString text;
             userMutex()->lock();
             Poppler::Page *pp = pdfdoc->page(i);
-            QString text = pp->text(QRect());
+            if (pp)
+            {
+                text = pp->text(QRect());
+            }
             userMutex()->unlock();
             ts << text;
             delete pp;
@@ -1838,29 +1869,37 @@
     // 1. set OutputDev parameters and Generate contents
     Poppler::Page *pp = d->generator->pdfdoc->page( page->number() );
     
-    double fakeDpiX = width * d->generator->dpiX / pageWidth,
-           fakeDpiY = height * d->generator->dpiY / pageHeight;
+    if (pp)
+    {
+        double fakeDpiX = width * d->generator->dpiX / pageWidth,
+            fakeDpiY = height * d->generator->dpiY / pageHeight;
 
-    // 2. grab data from the OutputDev and store it locally (note takeIMAGE)
+        // 2. grab data from the OutputDev and store it locally (note takeIMAGE)
 #ifndef NDEBUG
-    if ( d->m_image )
-        kDebug(PDFDebug) << "PDFPixmapGeneratorThread: previous image not taken";
-    if ( !d->m_textList.isEmpty() )
-        kDebug(PDFDebug) << "PDFPixmapGeneratorThread: previous text not taken";
+        if ( d->m_image )
+            kDebug(PDFDebug) << "PDFPixmapGeneratorThread: previous image not taken";
+        if ( !d->m_textList.isEmpty() )
+            kDebug(PDFDebug) << "PDFPixmapGeneratorThread: previous text not taken";
 #endif
-    d->m_image = new QImage( pp->renderToImage( fakeDpiX, fakeDpiY, -1, -1, -1, -1, Poppler::Page::Rotate0 ) );
-    
-    if ( genObjectRects )
-    {
-    	d->m_rects = generateLinks(pp->links());
-    }
-    else d->m_rectsTaken = false;
+        d->m_image = new QImage( pp->renderToImage( fakeDpiX, fakeDpiY, -1, -1, -1, -1, Poppler::Page::Rotate0 ) );
+        
+        if ( genObjectRects )
+        {
+            d->m_rects = generateLinks(pp->links());
+        }
+        else d->m_rectsTaken = false;
 
-    if ( genTextPage )
+        if ( genTextPage )
+        {
+            d->m_textList = pp->textList();
+        }
+        delete pp;
+    }
+    else
     {
-        d->m_textList = pp->textList();
+        d->m_image = new QImage( width, height, QImage::Format_ARGB32 );
+        d->m_image->fill( Qt::white );
     }
-    delete pp;
     
     // 3. [UNLOCK] mutex
     d->generator->userMutex()->unlock();
diff -ruN okular-4.6.4.orig/okular/generators/poppler/generator_pdf.h okular-4.6.4/okular/generators/poppler/generator_pdf.h
--- okular-4.6.4.orig/okular/generators/poppler/generator_pdf.h	2010-04-15 01:33:14.000000000 +0200
+++ okular-4.6.4/okular/generators/poppler/generator_pdf.h	2011-06-14 23:54:49.142769787 +0200
@@ -21,11 +21,11 @@
 #include <qpointer.h>
 #include <qthread.h>
 
-#include <okular/core/document.h>
-#include <okular/core/generator.h>
-#include <okular/interfaces/configinterface.h>
-#include <okular/interfaces/printinterface.h>
-#include <okular/interfaces/saveinterface.h>
+#include <core/document.h>
+#include <core/generator.h>
+#include <interfaces/configinterface.h>
+#include <interfaces/printinterface.h>
+#include <interfaces/saveinterface.h>
 
 namespace Okular {
 class ObjectRect;
diff -ruN okular-4.6.4.orig/okular/generators/poppler/libokularGenerator_poppler.desktop okular-4.6.4/okular/generators/poppler/libokularGenerator_poppler.desktop
--- okular-4.6.4.orig/okular/generators/poppler/libokularGenerator_poppler.desktop	2011-01-04 08:23:35.000000000 +0100
+++ okular-4.6.4/okular/generators/poppler/libokularGenerator_poppler.desktop	2011-06-14 23:54:49.142769787 +0200
@@ -111,7 +111,7 @@
 Comment[zh_CN]=使用 poppler 的 Okular PDF 格式后端
 Comment[zh_TW]=Okular 使用 poppler 的 PDF 後端
 X-KDE-ServiceTypes=okular/Generator
-MimeType=application/x-pdf;application/pdf;
+MimeType=application/x-pdf;application/pdf;application/x-wwf;
 X-KDE-Library=okularGenerator_poppler
 X-KDE-Priority=1
 X-KDE-okularAPIVersion=1
diff -ruN okular-4.6.4.orig/okular/generators/poppler/okularApplication_pdf.desktop okular-4.6.4/okular/generators/poppler/okularApplication_pdf.desktop
--- okular-4.6.4.orig/okular/generators/poppler/okularApplication_pdf.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/poppler/okularApplication_pdf.desktop	2011-06-14 23:54:49.142769787 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-MimeType=application/pdf;application/x-gzpdf;application/x-bzpdf;
+MimeType=application/pdf;application/x-gzpdf;application/x-bzpdf;application/x-wwf;
 Terminal=false
 Name=Okular
 Name[ar]=اوكلار
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/poppler/okularPoppler.desktop okular-4.6.4/okular/generators/poppler/okularPoppler.desktop
--- okular-4.6.4.orig/okular/generators/poppler/okularPoppler.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/poppler/okularPoppler.desktop	2011-06-14 23:54:49.142769787 +0200
@@ -53,8 +53,8 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
-MimeType=application/x-pdf;application/pdf;application/x-gzpdf;application/x-bzpdf;
+MimeType=application/x-pdf;application/pdf;application/x-gzpdf;application/x-bzpdf;application/x-wwf;
diff -ruN okular-4.6.4.orig/okular/generators/poppler/popplerembeddedfile.h okular-4.6.4/okular/generators/poppler/popplerembeddedfile.h
--- okular-4.6.4.orig/okular/generators/poppler/popplerembeddedfile.h	2008-04-13 18:14:40.000000000 +0200
+++ okular-4.6.4/okular/generators/poppler/popplerembeddedfile.h	2011-06-14 23:54:49.142769787 +0200
@@ -12,7 +12,7 @@
 
 #include <poppler-qt4.h>
 
-#include <okular/core/document.h>
+#include <core/document.h>
 
 class PDFEmbeddedFile : public Okular::EmbeddedFile
 {
diff -ruN okular-4.6.4.orig/okular/generators/spectre/generator_ghostview.cpp okular-4.6.4/okular/generators/spectre/generator_ghostview.cpp
--- okular-4.6.4.orig/okular/generators/spectre/generator_ghostview.cpp	2010-06-04 10:25:25.000000000 +0200
+++ okular-4.6.4/okular/generators/spectre/generator_ghostview.cpp	2011-06-14 23:54:49.142769787 +0200
@@ -23,10 +23,10 @@
 #include <kmimetype.h>
 #include <ktemporaryfile.h>
 
-#include <okular/core/document.h>
-#include <okular/core/page.h>
-#include <okular/core/fileprinter.h>
-#include <okular/core/utils.h>
+#include <core/document.h>
+#include <core/page.h>
+#include <core/fileprinter.h>
+#include <core/utils.h>
 
 #include "ui_gssettingswidget.h"
 #include "gssettings.h"
diff -ruN okular-4.6.4.orig/okular/generators/spectre/generator_ghostview.h okular-4.6.4/okular/generators/spectre/generator_ghostview.h
--- okular-4.6.4.orig/okular/generators/spectre/generator_ghostview.h	2008-12-01 01:35:21.000000000 +0100
+++ okular-4.6.4/okular/generators/spectre/generator_ghostview.h	2011-06-14 23:54:49.142769787 +0200
@@ -10,8 +10,8 @@
 #ifndef _OKULAR_GENERATOR_GHOSTVIEW_H_
 #define _OKULAR_GENERATOR_GHOSTVIEW_H_
 
-#include <okular/core/generator.h>
-#include <okular/interfaces/configinterface.h>
+#include <core/generator.h>
+#include <interfaces/configinterface.h>
 
 #include <libspectre/spectre.h>
 
diff -ruN okular-4.6.4.orig/okular/generators/spectre/okularApplication_ghostview.desktop okular-4.6.4/okular/generators/spectre/okularApplication_ghostview.desktop
--- okular-4.6.4.orig/okular/generators/spectre/okularApplication_ghostview.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/spectre/okularApplication_ghostview.desktop	2011-06-14 23:54:49.142769787 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/spectre/okularGhostview.desktop okular-4.6.4/okular/generators/spectre/okularGhostview.desktop
--- okular-4.6.4.orig/okular/generators/spectre/okularGhostview.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/spectre/okularGhostview.desktop	2011-06-14 23:54:49.142769787 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 InitialPreference=6
diff -ruN okular-4.6.4.orig/okular/generators/tiff/generator_tiff.cpp okular-4.6.4/okular/generators/tiff/generator_tiff.cpp
--- okular-4.6.4.orig/okular/generators/tiff/generator_tiff.cpp	2010-06-04 10:25:25.000000000 +0200
+++ okular-4.6.4/okular/generators/tiff/generator_tiff.cpp	2011-06-14 23:54:49.142769787 +0200
@@ -23,10 +23,10 @@
 #include <kglobal.h>
 #include <klocale.h>
 
-#include <okular/core/document.h>
-#include <okular/core/page.h>
-#include <okular/core/fileprinter.h>
-#include <okular/core/utils.h>
+#include <core/document.h>
+#include <core/page.h>
+#include <core/fileprinter.h>
+#include <core/utils.h>
 
 #include <tiff.h>
 #include <tiffio.h>
diff -ruN okular-4.6.4.orig/okular/generators/tiff/okularApplication_tiff.desktop okular-4.6.4/okular/generators/tiff/okularApplication_tiff.desktop
--- okular-4.6.4.orig/okular/generators/tiff/okularApplication_tiff.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/generators/tiff/okularApplication_tiff.desktop	2011-06-14 23:54:49.142769787 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/tiff/okularTiff.desktop okular-4.6.4/okular/generators/tiff/okularTiff.desktop
--- okular-4.6.4.orig/okular/generators/tiff/okularTiff.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/generators/tiff/okularTiff.desktop	2011-06-14 23:54:49.143769773 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/generators/xps/generator_xps.cpp okular-4.6.4/okular/generators/xps/generator_xps.cpp
--- okular-4.6.4.orig/okular/generators/xps/generator_xps.cpp	2011-01-06 00:46:08.000000000 +0100
+++ okular-4.6.4/okular/generators/xps/generator_xps.cpp	2011-06-14 23:54:49.143769773 +0200
@@ -32,10 +32,10 @@
 #include <QImageReader>
 #include <QMutex>
 
-#include <okular/core/document.h>
-#include <okular/core/page.h>
-#include <okular/core/area.h>
-#include <okular/core/fileprinter.h>
+#include <core/document.h>
+#include <core/page.h>
+#include <core/area.h>
+#include <core/fileprinter.h>
 
 const int XpsDebug = 4712;
 
@@ -345,7 +345,7 @@
                 {
                     QPointF point1 = getPointFromString(&token, isRelative, currPos);
                     QPointF point2 = getPointFromString(&token, isRelative, currPos);
-                    path.quadTo(point2, point2);
+                    path.quadTo(point1, point2);
                 }
                 break;
             case 's': // Smooth cubic bezier curve
@@ -1980,7 +1980,7 @@
 
     m_docInfo = new Okular::DocumentInfo();
 
-    m_docInfo->set( Okular::DocumentInfo::MimeType, "application/vnd.ms-xpsdocument" );
+    m_docInfo->set( Okular::DocumentInfo::MimeType, "application/oxps" );
 
     if ( ! m_corePropertiesFileName.isEmpty() ) {
         const KZipFileEntry* corepropsFile = static_cast<const KZipFileEntry *>(m_xpsArchive->directory()->entry(m_corePropertiesFileName));
diff -ruN okular-4.6.4.orig/okular/generators/xps/generator_xps.h okular-4.6.4/okular/generators/xps/generator_xps.h
--- okular-4.6.4.orig/okular/generators/xps/generator_xps.h	2008-09-04 19:30:56.000000000 +0200
+++ okular-4.6.4/okular/generators/xps/generator_xps.h	2011-06-14 23:54:49.143769773 +0200
@@ -20,8 +20,8 @@
 #ifndef _OKULAR_GENERATOR_XPS_H_
 #define _OKULAR_GENERATOR_XPS_H_
 
-#include <okular/core/generator.h>
-#include <okular/core/textpage.h>
+#include <core/generator.h>
+#include <core/textpage.h>
 
 #include <QColor>
 #include <QDomDocument>
diff -ruN okular-4.6.4.orig/okular/generators/xps/okularApplication_xps.desktop okular-4.6.4/okular/generators/xps/okularApplication_xps.desktop
--- okular-4.6.4.orig/okular/generators/xps/okularApplication_xps.desktop	2010-10-19 22:14:21.000000000 +0200
+++ okular-4.6.4/okular/generators/xps/okularApplication_xps.desktop	2011-06-14 23:54:49.143769773 +0200
@@ -54,7 +54,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/generators/xps/okularXps.desktop okular-4.6.4/okular/generators/xps/okularXps.desktop
--- okular-4.6.4.orig/okular/generators/xps/okularXps.desktop	2010-10-19 22:14:21.000000000 +0200
+++ okular-4.6.4/okular/generators/xps/okularXps.desktop	2011-06-14 23:54:49.143769773 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/interfaces/configinterface.h okular-4.6.4/okular/interfaces/configinterface.h
--- okular-4.6.4.orig/okular/interfaces/configinterface.h	2007-07-12 20:14:01.000000000 +0200
+++ okular-4.6.4/okular/interfaces/configinterface.h	2011-06-14 23:54:49.143769773 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_CONFIGINTERFACE_H_
 #define _OKULAR_CONFIGINTERFACE_H_
 
-#include <okular/core/okular_export.h>
+#include "../core/okular_export.h"
 
 #include <QtCore/QObject>
 
diff -ruN okular-4.6.4.orig/okular/interfaces/guiinterface.h okular-4.6.4/okular/interfaces/guiinterface.h
--- okular-4.6.4.orig/okular/interfaces/guiinterface.h	2007-08-29 01:17:00.000000000 +0200
+++ okular-4.6.4/okular/interfaces/guiinterface.h	2011-06-14 23:54:49.143769773 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_GUIINTERFACE_H_
 #define _OKULAR_GUIINTERFACE_H_
 
-#include <okular/core/okular_export.h>
+#include "../core/okular_export.h"
 
 #include <QtCore/QObject>
 
diff -ruN okular-4.6.4.orig/okular/interfaces/printinterface.h okular-4.6.4/okular/interfaces/printinterface.h
--- okular-4.6.4.orig/okular/interfaces/printinterface.h	2007-10-16 01:01:27.000000000 +0200
+++ okular-4.6.4/okular/interfaces/printinterface.h	2011-06-14 23:54:49.143769773 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_PRINTINTERFACE_H_
 #define _OKULAR_PRINTINTERFACE_H_
 
-#include <okular/core/okular_export.h>
+#include "../core/okular_export.h"
 
 #include <QtCore/QObject>
 
diff -ruN okular-4.6.4.orig/okular/interfaces/saveinterface.h okular-4.6.4/okular/interfaces/saveinterface.h
--- okular-4.6.4.orig/okular/interfaces/saveinterface.h	2009-10-20 01:00:52.000000000 +0200
+++ okular-4.6.4/okular/interfaces/saveinterface.h	2011-06-14 23:54:49.143769773 +0200
@@ -10,7 +10,7 @@
 #ifndef _OKULAR_SAVEINTERFACE_H_
 #define _OKULAR_SAVEINTERFACE_H_
 
-#include <okular/core/okular_export.h>
+#include "../core/okular_export.h"
 
 #include <QtCore/QObject>
 
diff -ruN okular-4.6.4.orig/okular/okular_part.desktop okular-4.6.4/okular/okular_part.desktop
--- okular-4.6.4.orig/okular/okular_part.desktop	2010-07-15 09:11:30.000000000 +0200
+++ okular-4.6.4/okular/okular_part.desktop	2011-06-14 23:54:49.143769773 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 X-KDE-ServiceTypes=KParts/ReadOnlyPart
 X-KDE-Library=okularpart
 Type=Service
diff -ruN okular-4.6.4.orig/okular/part.cpp okular-4.6.4/okular/part.cpp
--- okular-4.6.4.orig/okular/part.cpp	2010-10-29 23:47:17.000000000 +0200
+++ okular-4.6.4/okular/part.cpp	2011-06-14 23:54:49.144769761 +0200
@@ -615,8 +615,8 @@
     slotNewConfig();
 
     // [SPEECH] check for KTTSD presence and usability
-    KService::List offers = KServiceTypeTrader::self()->query("DBUS/Text-to-Speech", "Name == 'KTTSD'");
-    Okular::Settings::setUseKTTSD( !offers.isEmpty() );
+    const KService::Ptr kttsd = KService::serviceByDesktopName("kttsd");
+    Okular::Settings::setUseKTTSD( kttsd );
     Okular::Settings::self()->writeConfig();
 
     rebuildBookmarkMenu( false );
@@ -2264,7 +2264,7 @@
     if ( m_bookmarkActions.isEmpty() )
     {
         havebookmarks = false;
-        QAction * a = new QAction( 0 );
+        QAction * a = new KAction( 0 );
         a->setText( i18n( "No Bookmarks" ) );
         a->setEnabled( false );
         m_bookmarkActions.append( a );
diff -ruN okular-4.6.4.orig/okular/shell/okular.desktop okular-4.6.4/okular/shell/okular.desktop
--- okular-4.6.4.orig/okular/shell/okular.desktop	2010-10-05 01:51:07.000000000 +0200
+++ okular-4.6.4/okular/shell/okular.desktop	2011-06-14 23:54:49.144769761 +0200
@@ -53,7 +53,7 @@
 Name[uk]=Okular
 Name[x-test]=xxOkularxx
 Name[zh_CN]=Okular
-Name[zh_TW]=Okular
+Name[zh_TW]=文件檢視_Okular
 GenericName=Document Viewer
 GenericName[ar]=عارض المستندات
 GenericName[ast]=Visor de documentos
diff -ruN okular-4.6.4.orig/okular/shell/shell.cpp okular-4.6.4/okular/shell/shell.cpp
--- okular-4.6.4.orig/okular/shell/shell.cpp	2010-11-13 21:12:35.000000000 +0100
+++ okular-4.6.4/okular/shell/shell.cpp	2011-06-14 23:54:49.144769761 +0200
@@ -69,6 +69,7 @@
   setXMLFile("shell.rc");
   m_doc=0L;
   m_fileformatsscanned = false;
+  m_showMenuBarAction = 0;
   // this routine will find and load our Part.  it finds the Part by
   // name which is a bad idea usually.. but it's alright in this
   // case since our Part is made for this Shell
@@ -280,7 +281,8 @@
 
 void Shell::showEvent(QShowEvent *e)
 {
-    m_showMenuBarAction->setChecked( menuBar()->isVisible() );
+    if (m_showMenuBarAction)
+        m_showMenuBarAction->setChecked( menuBar()->isVisible() );
 
     KParts::MainWindow::showEvent(e);
 }
diff -ruN okular-4.6.4.orig/okular/ui/pagepainter.cpp okular-4.6.4/okular/ui/pagepainter.cpp
--- okular-4.6.4.orig/okular/ui/pagepainter.cpp	2010-03-23 22:50:13.000000000 +0100
+++ okular-4.6.4/okular/ui/pagepainter.cpp	2011-06-14 23:54:49.144769761 +0200
@@ -738,7 +738,7 @@
     unsigned int * destData = (unsigned int *)dest.bits();
 
     // source image (1:1 conversion from pixmap)
-    QImage srcImage = src->toImage();
+    QImage srcImage = src->toImage().convertToFormat(format);
     unsigned int * srcData = (unsigned int *)srcImage.bits();
 
     // precalc the x correspondancy conversion in a lookup table
diff -ruN okular-4.6.4.orig/okular/ui/pageview.cpp okular-4.6.4/okular/ui/pageview.cpp
--- okular-4.6.4.orig/okular/ui/pageview.cpp	2011-02-10 01:55:01.000000000 +0100
+++ okular-4.6.4/okular/ui/pageview.cpp	2011-06-14 23:54:49.145769753 +0200
@@ -132,7 +132,7 @@
     //text annotation dialogs list
     QHash< Okular::Annotation *, AnnotWindow * > m_annowindows;
     // other stuff
-    QTimer * delayRelayoutTimer;
+    QTimer * delayResizeEventTimer;
     bool dirtyLayout;
     bool blockViewport;                 // prevents changes to viewport
     bool blockPixmapsRequest;           // prevent pixmap requests
@@ -284,9 +284,9 @@
     d->aPageSizes=0;
     d->setting_viewCols = Okular::Settings::viewColumns();
 
-    d->delayRelayoutTimer = new QTimer( this );
-    d->delayRelayoutTimer->setSingleShot( true );
-    connect( d->delayRelayoutTimer, SIGNAL( timeout() ), this, SLOT( slotRelayoutPages() ) );
+    d->delayResizeEventTimer = new QTimer( this );
+    d->delayResizeEventTimer->setSingleShot( true );
+    connect( d->delayResizeEventTimer, SIGNAL( timeout() ), this, SLOT( delayedResizeEvent() ) );
 
     setFrameStyle(QFrame::NoFrame);
 
@@ -779,7 +779,7 @@
         // because we might end up in notifyViewportChanged while slotRelayoutPages
         // has not been done and we don't want that to happen
         d->dirtyLayout = true;
-        d->delayRelayoutTimer->start(0);
+        QMetaObject::invokeMethod(this, "slotRelayoutPages", Qt::QueuedConnection);
     }
     else
     {
@@ -1264,7 +1264,7 @@
     }
 
     // start a timer that will refresh the pixmap after 0.2s
-    d->delayRelayoutTimer->start( 200 );
+    d->delayResizeEventTimer->start( 200 );
     d->verticalScrollBarVisible = verticalScrollBar()->isVisible();
 }
 
@@ -2985,9 +2985,6 @@
 void PageView::slotRelayoutPages()
 // called by: notifySetup, viewportResizeEvent, slotViewMode, slotContinuousToggled, updateZoom
 {
-    // If we already got here we don't need to execute the timer slot again
-    d->delayRelayoutTimer->stop();
-
     // set an empty container if we have no pages
     int pageCount = d->items.count();
     if ( pageCount < 1 )
@@ -3166,6 +3163,14 @@
         viewport()->update();
 }
 
+void PageView::delayedResizeEvent()
+{
+    // If we already got here we don't need to execute the timer slot again
+    d->delayResizeEventTimer->stop();
+    slotRelayoutPages();
+    slotRequestVisiblePixmaps();
+}
+
 void PageView::slotRequestVisiblePixmaps( int newValue )
 {
     // if requests are blocked (because raised by an unwanted event), exit
diff -ruN okular-4.6.4.orig/okular/ui/pageview.h okular-4.6.4/okular/ui/pageview.h
--- okular-4.6.4.orig/okular/ui/pageview.h	2010-09-11 15:34:22.000000000 +0200
+++ okular-4.6.4/okular/ui/pageview.h	2011-06-14 23:54:49.145769753 +0200
@@ -181,8 +181,10 @@
         class PageViewPrivate * d;
 
     private slots:
-        // activated either directly or via QTimer on the viewportResizeEvent
+        // activated either directly or via queued connection on notifySetup
         void slotRelayoutPages();
+        // activated by the resize event delay timer
+        void delayedResizeEvent();
         // activated either directly or via the contentsMoving(int,int) signal
         void slotRequestVisiblePixmaps( int newValue = -1 );
         // activated by the viewport move timer
