--- config.layout.orig	2012-02-02 10:05:28.000000000 -0200
+++ config.layout	2012-03-27 22:29:05.353876994 -0300
@@ -347,3 +347,26 @@
     proxycachedir: ${localstatedir}/proxy
 </Layout>
 
+# FHS layout
+<Layout Slackware-FHS>
+    prefix:          /etc/httpd
+    exec_prefix:     /usr
+    bindir:          ${exec_prefix}/bin
+    sbindir:         ${exec_prefix}/sbin
+    libdir:          ${exec_prefix}/lib/httpd
+    libexecdir:      ${exec_prefix}/lib/httpd/modules
+    installbuilddir: ${exec_prefix}/lib/httpd/build
+    mandir:          ${exec_prefix}/man
+    sysconfdir:      /etc/httpd
+    datadir:         /srv/httpd
+    iconsdir:        ${datadir}/icons
+    htdocsdir:       ${datadir}/htdocs
+    manualdir:       ${htdocsdir}/manual
+    cgidir:          ${datadir}/cgi-bin
+    errordir:        ${datadir}/error
+    includedir:      ${exec_prefix}/include/httpd
+    localstatedir:   /var
+    runtimedir:      ${localstatedir}/run/httpd
+    logfiledir:      ${localstatedir}/log/httpd
+    proxycachedir:   ${localstatedir}/cache/httpd
+</Layout>
