diff --git a/server/protocol.c b/server/protocol.c
index e1ef204..d6d9165 100644
--- a/server/protocol.c
+++ b/server/protocol.c
@@ -1049,6 +1049,7 @@ request_rec *ap_read_request(conn_rec *conn)
      * now read. may update status.
      */
     ap_update_vhost_from_headers(r);
+    access_status = r->status;
 
     /* Toggle to the Host:-based vhost's timeout mode to fetch the
      * request body and send the response body, if needed.
