--- ./X11/neXtaw/Scrollbar.c.orig	2003-09-02 02:56:35.000000000 -0600
+++ ./X11/neXtaw/Scrollbar.c	2013-03-26 08:15:27.738940661 -0600
@@ -1362,7 +1362,12 @@ static void NotifyThumb (w, event, param
     Cardinal *num_params;	/* unused */
 {
     register ScrollbarWidget sbw = (ScrollbarWidget) w;
-    float top = sbw->scrollbar.top;
+    union {
+      float top;
+      XtPointer ptr;
+    } tp;
+
+    tp.top = sbw->scrollbar.top;
 
     if (LookAhead (w, event)) return;
 
@@ -1384,10 +1389,10 @@ static void NotifyThumb (w, event, param
        there is no problem since in this case there is always a constant
        integer number of pixels the thumb must be moved in order to scroll
        to the next line/column. */
-    top += 0.0001;
+    tp.top += 0.0001;
 
-    XtCallCallbacks (w, XtNthumbProc, *(XtPointer*)&top);
-    XtCallCallbacks (w, XtNjumpProc, (XtPointer)&top);
+    XtCallCallbacks (w, XtNthumbProc, tp.ptr);
+    XtCallCallbacks (w, XtNjumpProc, (XtPointer)&tp.top);
 }
 
 
