--- ./X11/neXtaw/XawIm.c.orig	2003-08-24 00:24:13.000000000 -0600
+++ ./X11/neXtaw/XawIm.c	2013-03-26 08:38:45.927486081 -0600
@@ -62,6 +62,7 @@ in this Software without prior written a
 #include <X11/Xos.h>
 #include <X11/Xfuncs.h>
 #include <X11/ShellP.h>
+#include <X11/ResourceI.h>
 #include "TextP.h"
 #include "MultiSrc.h"
 #include "MultiSinkP.h"
--- ./X11/neXtaw/laygram.c.orig	2003-01-15 08:50:49.000000000 -0700
+++ ./X11/neXtaw/laygram.c	2013-03-26 08:41:32.790915480 -0600
@@ -57,6 +57,7 @@ typedef union {
     Operator	    oval;
 } YYSTYPE;
 #include <stdio.h>
+#include <stdlib.h>
 
 #ifndef __cplusplus
 #ifndef __STDC__
