diff -up cmake-2.8.6-rc1/Modules/FindDCMTK.cmake.dcmtk cmake-2.8.6-rc1/Modules/FindDCMTK.cmake
--- cmake-2.8.6-rc1/Modules/FindDCMTK.cmake.dcmtk	2011-08-26 10:51:46.000000000 -0600
+++ cmake-2.8.6-rc1/Modules/FindDCMTK.cmake	2011-08-29 16:58:07.299550799 -0600
@@ -30,6 +30,10 @@
 # Written for VXL by Amitha Perera.
 # Upgraded for GDCM by Mathieu Malaterre.
 # Modified for EasyViz by Thomas Sondergaard.
+# 
+# Wed Jul 27 2011 Ankur Sinha <ankursinha AT fedoraproject DOT org> 
+# - Add all dcmtk libs
+# - Add usr/lib to paths
 #
 
 if(NOT DCMTK_FOUND AND NOT DCMTK_DIR)
@@ -44,20 +48,26 @@ endif()
 
 foreach(lib
     dcmdata
+    dcmdsig
     dcmimage
     dcmimgle
     dcmjpeg
+    dcmjpls
     dcmnet
     dcmpstat
     dcmqrdb
-    dcmsign
     dcmsr
     dcmtls
+    dcmwlm
     ijg12
     ijg16
     ijg8
+    libi2d
+    oflog
     ofstd)
 
+
+
   find_library(DCMTK_${lib}_LIBRARY
     ${lib}
     PATHS
@@ -66,7 +76,8 @@ foreach(lib
     ${DCMTK_DIR}/${lib}/libsrc/Debug
     ${DCMTK_DIR}/${lib}/Release
     ${DCMTK_DIR}/${lib}/Debug
-    ${DCMTK_DIR}/lib)
+    ${DCMTK_DIR}/lib
+    /usr/lib/dcmtk)
 
   mark_as_advanced(DCMTK_${lib}_LIBRARY)
 
@@ -82,13 +93,16 @@ set(DCMTK_dcmdata_TEST_HEADER dctypes.h)
 set(DCMTK_dcmimage_TEST_HEADER dicoimg.h)
 set(DCMTK_dcmimgle_TEST_HEADER dcmimage.h)
 set(DCMTK_dcmjpeg_TEST_HEADER djdecode.h)
+set(DCMTK_dcmjpls_TEST_HEADER djcodecd.h)
 set(DCMTK_dcmnet_TEST_HEADER assoc.h)
 set(DCMTK_dcmpstat_TEST_HEADER dcmpstat.h)
 set(DCMTK_dcmqrdb_TEST_HEADER dcmqrdba.h)
 set(DCMTK_dcmsign_TEST_HEADER sicert.h)
 set(DCMTK_dcmsr_TEST_HEADER dsrtree.h)
 set(DCMTK_dcmtls_TEST_HEADER tlslayer.h)
+set(DCMTK_dcmwlm_TEST_HEADER wldsfs.h)
 set(DCMTK_ofstd_TEST_HEADER ofstdinc.h)
+set(DCMTK_oflog_TEST_HEADER oflog.h)
 
 foreach(dir
     config
@@ -96,16 +110,20 @@ foreach(dir
     dcmimage
     dcmimgle
     dcmjpeg
+    dcmjpls
     dcmnet
     dcmpstat
     dcmqrdb
     dcmsign
     dcmsr
     dcmtls
+    dcmwlen
+    oflog
     ofstd)
   find_path(DCMTK_${dir}_INCLUDE_DIR
     ${DCMTK_${dir}_TEST_HEADER}
     PATHS
+    /usr/include/dcmtk
     ${DCMTK_DIR}/${dir}/include
     ${DCMTK_DIR}/${dir}
     ${DCMTK_DIR}/include/${dir}
