diff -up cmake-2.8.11/Source/CMakeLists.txt\~ cmake-2.8.11/Source/CMakeLists.txt
--- cmake-2.8.11/Source/CMakeLists.txt~	2013-05-15 19:38:13.000000000 +0200
+++ cmake-2.8.11/Source/CMakeLists.txt	2013-07-25 16:35:01.200389140 +0200
@@ -308,6 +308,13 @@ if(APPLE)
     cmLocalXCodeGenerator.h)
 endif()
 
+# GCC shows strict aliasing warnings with cm_sha2.c.  Turn off the
+# corresponding optimizations.
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_GNUCC)
+  set_source_files_properties(cm_sha2.c PROPERTIES
+			      COMPILE_FLAGS "-fno-strict-aliasing")
+endif()
+
 
 if (WIN32)
   set(SRCS ${SRCS}

Diff finished.  Thu Jul 25 16:35:18 2013
