--- Modules/FindPythonLibs.cmake.pylibs	2010-11-03 17:58:27.000000000 -0200
+++ Modules/FindPythonLibs.cmake	2010-11-05 18:37:08.620679581 -0200
@@ -27,7 +27,9 @@
 # Search for the python framework on Apple.
 CMAKE_FIND_FRAMEWORKS(Python)
 
-FOREACH(_CURRENT_VERSION 2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0 1.6 1.5)
+EXECUTE_PROCESS(COMMAND python -c "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
+  OUTPUT_VARIABLE _PYTHON_VERSION)
+FOREACH(_CURRENT_VERSION ${_PYTHON_VERSION})
   STRING(REPLACE "." "" _CURRENT_VERSION_NO_DOTS ${_CURRENT_VERSION})
   IF(WIN32)
     FIND_LIBRARY(PYTHON_DEBUG_LIBRARY
