--- cmake-3.0.0/Modules/FindLAPACK.cmake.orig	2014-06-09 12:46:52.000000000 -0300
+++ cmake-3.0.0/Modules/FindLAPACK.cmake	2014-06-10 23:58:09.996419645 -0300
@@ -2,6 +2,11 @@
 # FindLAPACK
 # ----------
 #
+# Version modified for Gentoo Linux
+# If a valid PkgConfig configuration is found, this overrides and cancels
+# all further checks.
+#
+
 # Find LAPACK library
 #
 # This module finds an installed fortran library that implements the
@@ -46,6 +51,24 @@
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
 
+#
+# first, try PkgConfig
+#
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(PC_LAPACK lapack)
+if(PC_LAPACK_FOUND)
+  foreach(PC_LIB ${PC_LAPACK_LIBRARIES})
+    find_library(${PC_LIB}_LIBRARY NAMES ${PC_LIB} HINTS ${PC_LAPACK_LIBRARY_DIRS} )
+    if (NOT ${PC_LIB}_LIBRARY)
+      message(FATAL_ERROR "Something is wrong in your pkg-config file - lib ${PC_LIB} not found in ${PC_LAPACK_LIBRARY_DIRS}")
+    endif (NOT ${PC_LIB}_LIBRARY)
+    list(APPEND LAPACK_LIBRARIES ${${PC_LIB}_LIBRARY}) 
+  endforeach(PC_LIB)
+  find_package_handle_standard_args(LAPACK DEFAULT_MSG LAPACK_LIBRARIES)
+  mark_as_advanced(LAPACK_LIBRARIES)
+else(PC_LAPACK_FOUND)
+message(STATUS "No PkgConfig configuration for LAPACK found; starting more extensive search.")
+
 set(_lapack_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
 
 get_property(_LANGUAGES_ GLOBAL PROPERTY ENABLED_LANGUAGES)
@@ -348,3 +371,5 @@ else()
 endif()
 
 set(CMAKE_FIND_LIBRARY_SUFFIXES ${_lapack_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES})
+
+endif(PC_LAPACK_FOUND)
