--- cmake-3.0.0-rc3/Modules/FindwxWidgets.cmake.orig	2014-03-20 10:50:27.000000000 -0600
+++ cmake-3.0.0-rc3/Modules/FindwxWidgets.cmake	2014-04-04 09:34:41.194072417 -0600
@@ -230,10 +230,10 @@
 
 #=====================================================================
 #=====================================================================
-if(WIN32 AND NOT CYGWIN AND NOT MSYS)
+if(WIN32 AND NOT CYGWIN AND NOT MSYS AND NOT CMAKE_CROSSCOMPILING)
   set(wxWidgets_FIND_STYLE "win32")
 else()
-  if(UNIX OR MSYS)
+  if(UNIX OR MSYS OR WIN32 AND CMAKE_CROSSCOMPILING)
     set(wxWidgets_FIND_STYLE "unix")
   endif()
 endif()
