diff -urN cmake-2.8.3.orig//Source/kwsys/SystemInformation.cxx cmake-2.8.3//Source/kwsys/SystemInformation.cxx
--- cmake-2.8.3.orig//Source/kwsys/SystemInformation.cxx	2010-12-25 16:56:42.609163498 +0100
+++ cmake-2.8.3//Source/kwsys/SystemInformation.cxx	2010-12-25 17:25:18.002163499 +0100
@@ -152,7 +152,7 @@
 
 public:
 #define VENDOR_STRING_LENGTH    (12 + 1)
-#define CHIPNAME_STRING_LENGTH    (48 + 1)
+#define CHIPNAME_STRING_LENGTH    (68 + 1)
 #define SERIALNUMBER_STRING_LENGTH  (29 + 1)
 
   typedef struct tagID
