--- Modules/FindPythonLibs.cmake.pylibs	2011-02-15 15:47:27.000000000 -0200
+++ Modules/FindPythonLibs.cmake	2011-02-16 19:53:40.806967110 -0200
@@ -29,9 +29,11 @@ CMAKE_FIND_FRAMEWORKS(Python)
 
 # Set up the versions we know about, in the order we will search. Always add
 # the user supplied additional versions to the front.
+EXECUTE_PROCESS(COMMAND python -c "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
+  OUTPUT_VARIABLE _PYTHON_VERSION)
 set(_Python_VERSIONS
   ${Python_ADDITIONAL_VERSIONS}
-  2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0 1.6 1.5)
+  ${_PYTHON_VERSION})
 
 FOREACH(_CURRENT_VERSION ${_Python_VERSIONS})
   STRING(REPLACE "." "" _CURRENT_VERSION_NO_DOTS ${_CURRENT_VERSION})
