diff -up cmake-2.8.10/CMakeLists.txt.libform cmake-2.8.10/CMakeLists.txt
--- cmake-2.8.10/CMakeLists.txt.libform	2012-11-02 18:52:43.442466458 -0200
+++ cmake-2.8.10/CMakeLists.txt	2012-11-02 18:55:28.740504412 -0200
@@ -395,9 +395,6 @@ macro (CMAKE_BUILD_UTILITIES)
   else ()
     set(BUILD_CursesDialog 0)
   endif ()
-  if(BUILD_CursesDialog)
-    add_subdirectory(Source/CursesDialog/form)
-  endif()
 endmacro ()
 
 #-----------------------------------------------------------------------
diff -up cmake-2.8.10/Source/CursesDialog/CMakeLists.txt.libform cmake-2.8.10/Source/CursesDialog/CMakeLists.txt
--- cmake-2.8.10/Source/CursesDialog/CMakeLists.txt.libform	2012-11-02 18:52:43.443466247 -0200
+++ cmake-2.8.10/Source/CursesDialog/CMakeLists.txt	2012-11-02 18:55:47.783477668 -0200
@@ -25,13 +25,15 @@ set( CURSES_SRCS
               CursesDialog/ccmake
    )
 
-include_directories(${CMake_SOURCE_DIR}/Source/CursesDialog/form
-                    ${CMake_BINARY_DIR}/Source/CursesDialog/form)
 include_directories(${CURSES_INCLUDE_PATH})
 
 
 add_executable(ccmake ${CURSES_SRCS} )
 target_link_libraries(ccmake CMakeLib)
-target_link_libraries(ccmake cmForm)
+target_link_libraries(ccmake form)
+target_link_libraries(ccmake ${CURSES_LIBRARY})
+if(CURSES_EXTRA_LIBRARY)
+  target_link_libraries(ccmake ${CURSES_EXTRA_LIBRARY})
+endif()
 
 install_targets(/bin ccmake)
diff -up cmake-2.8.10/Source/CursesDialog/cmCursesStandardIncludes.h.libform cmake-2.8.10/Source/CursesDialog/cmCursesStandardIncludes.h
--- cmake-2.8.10/Source/CursesDialog/cmCursesStandardIncludes.h.libform	2012-10-31 13:32:05.000000000 -0200
+++ cmake-2.8.10/Source/CursesDialog/cmCursesStandardIncludes.h	2012-11-02 18:52:43.443466247 -0200
@@ -15,8 +15,6 @@
  #define _MSE_INT_H
 #endif
 
-#include <cmFormConfigure.h>
-
 #if defined(__hpux)
 # define _BOOL_DEFINED
 # include <sys/time.h>
