diff -Naru tilda-0.9.6-bak/src/key_grabber.c tilda-0.9.6/src/key_grabber.c
--- tilda-0.9.6-bak/src/key_grabber.c	2009-02-23 15:52:39.000000000 +0100
+++ tilda-0.9.6/src/key_grabber.c	2009-02-23 16:13:35.000000000 +0100
@@ -123,10 +123,10 @@
     DEBUG_FUNCTION ("tilda_window_set_active");
     DEBUG_ASSERT (tw != NULL);
 
-    Display *x11_display = GDK_WINDOW_XDISPLAY( tw->window->window );
-    Window *x11_window = GDK_WINDOW_XWINDOW( tw->window->window );
-    Window *x11_root_window = GDK_WINDOW_XWINDOW ( gtk_widget_get_root_window (tw->window) );
-    GdkScreen *screen = gtk_widget_get_screen (tw->window);
+    Display *x11_display = GDK_WINDOW_XDISPLAY( (GdkWindow*)(tw->window->window) );
+    Window *x11_window = (Window*)GDK_WINDOW_XWINDOW( (GdkWindow*)(tw->window->window) );
+    Window *x11_root_window = (Window*)GDK_WINDOW_XWINDOW ( gtk_widget_get_root_window ((GtkWidget*)(tw->window)) );
+    GdkScreen *screen = gtk_widget_get_screen ((GtkWidget*)(tw->window));
 
     XEvent event;
     long mask = SubstructureRedirectMask | SubstructureNotifyMask;
