diff -up iotop-0.6/iotop/ui.py.noendcurses iotop-0.6/iotop/ui.py
--- iotop-0.6/iotop/ui.py.noendcurses	2014-12-03 17:50:38.941430261 +0100
+++ iotop-0.6/iotop/ui.py	2014-12-03 17:51:40.108064465 +0100
@@ -520,6 +525,17 @@ Please do not file bugs on iotop about t
             sys.exit(1)
         else:
             raise
+    except curses.error as e:
+        stre = str(e)
+        if stre.find('ERR')>=0 and (
+            stre.find('nocbreak()')>=0 or stre.find('endwin()')>=0
+            ):
+            pass
+            # endwin and nocbreak can cause error (and raise hard to catch 
+            # exception) if iotop was running in the terminal and that 
+            # terminal got closed while iotop was still running
+        else:
+            raise
 
 #
 # Profiling
