diff -ur po4a-0.44.old/lib/Locale/Po4a/Po.pm po4a-0.44/lib/Locale/Po4a/Po.pm
--- po4a-0.44.old/lib/Locale/Po4a/Po.pm	2012-10-21 00:03:24.000000000 +0100
+++ po4a-0.44/lib/Locale/Po4a/Po.pm	2013-04-17 11:26:54.749652129 +0100
@@ -572,7 +572,8 @@
 
     if (-e $filename) {
         my ($tmp_filename);
-        (undef,$tmp_filename)=File::Temp->tempfile($filename."XXXX",
+        my $basename = basename($filename);
+        (undef,$tmp_filename)=File::Temp::tempfile($basename."XXXX",
                                                    DIR    => "/tmp",
                                                    OPEN   => 0,
                                                    UNLINK => 0);
diff -ur po4a-0.44.old/lib/Locale/Po4a/Wml.pm po4a-0.44/lib/Locale/Po4a/Wml.pm
--- po4a-0.44.old/lib/Locale/Po4a/Wml.pm	2012-10-21 00:03:24.000000000 +0100
+++ po4a-0.44/lib/Locale/Po4a/Wml.pm	2013-04-17 11:27:43.904492845 +0100
@@ -81,7 +81,7 @@
 sub read {
     my ($self,$filename)=@_;
     my $tmp_filename;
-    (undef,$tmp_filename)=File::Temp->tempfile("po4aXXXX",
+    (undef,$tmp_filename)=File::Temp::tempfile("po4aXXXX",
                                                 DIR    => "/tmp",
                                                 SUFFIX => ".xml",
                                                 OPEN   => 0,
diff -ur po4a-0.44.old/po4a po4a-0.44/po4a
--- po4a-0.44.old/po4a	2012-10-21 00:03:24.000000000 +0100
+++ po4a-0.44/po4a	2013-04-17 11:26:05.938810267 +0100
@@ -1209,7 +1209,7 @@
     chdir $po4a_opts{"srcdir"}
 	if (defined $po4a_opts{"srcdir"});
     if ($po4a_opts{"split"}) {
-        (undef,$pot_filename)=File::Temp->tempfile("po4aXXXX",
+        (undef,$pot_filename)=File::Temp::tempfile("po4aXXXX",
                                                    DIR    => "/tmp",
                                                    SUFFIX => ".pot",
                                                    OPEN   => 0,
@@ -1239,7 +1239,7 @@
         # Create a temporary POT, and check if the old one needs to be
         # updated (unless --force was specified).
         unless ($po4a_opts{"force"}) {
-            (undef,$tmp_file)=File::Temp->tempfile("po4aXXXX",
+            (undef,$tmp_file)=File::Temp::tempfile("po4aXXXX",
                                                    DIR    => "/tmp",
                                                    SUFFIX => ".pot",
                                                    OPEN   => 0,
@@ -1270,7 +1270,7 @@
     # Generate a complete .po
     foreach my $lang (sort keys %po_filename) {
         my $tmp_bigpo;
-        (undef,$tmp_bigpo)=File::Temp->tempfile("po4aXXXX",
+        (undef,$tmp_bigpo)=File::Temp::tempfile("po4aXXXX",
                                                 DIR    => "/tmp",
                                                 SUFFIX => "-$lang.po",
                                                 OPEN   => 0,
@@ -1336,7 +1336,7 @@
             my $tmp_file;
             # Create a temporary PO, and check if the old one needs to be
             # updated (unless --force was specified).
-            (undef,$tmp_file)=File::Temp->tempfile("po4aXXXX",
+            (undef,$tmp_file)=File::Temp::tempfile("po4aXXXX",
                                                    DIR    => "/tmp",
                                                    SUFFIX => ".po",
                                                    OPEN   => 0,
diff -ur po4a-0.44.old/po4a-updatepo po4a-0.44/po4a-updatepo
--- po4a-0.44.old/po4a-updatepo	2012-10-21 00:03:24.000000000 +0100
+++ po4a-0.44/po4a-updatepo	2013-04-17 11:26:05.938810267 +0100
@@ -248,7 +248,7 @@
 	if $_ eq '-'  && !-e '-'} @pofiles;
 
 my ($pot_filename);
-(undef,$pot_filename)=File::Temp->tempfile("po4a-updatepoXXXX",
+(undef,$pot_filename)=File::Temp::tempfile("po4a-updatepoXXXX",
 					   DIR    => "/tmp",
 					   SUFFIX => ".pot",
 					   OPEN   => 0,
diff -ur po4a-0.44.old/scripts/msguntypot po4a-0.44/scripts/msguntypot
--- po4a-0.44.old/scripts/msguntypot	2012-10-21 00:03:24.000000000 +0100
+++ po4a-0.44/scripts/msguntypot	2013-04-17 11:26:05.939810264 +0100
@@ -195,7 +195,7 @@
 
 # Get all po files and report differences in them
 my ($pofile);
-(undef,$pofile)=File::Temp->tempfile("po4aXXXX",
+(undef,$pofile)=File::Temp::tempfile("po4aXXXX",
     DIR    => "/tmp",
     SUFFIX => ".po",
     OPEN   => 0,
