From 9a36e0227c4dd8c3ebd1fbb313737c9b95fbd7e1 Mon Sep 17 00:00:00 2001
From: Steve Dickson <steved@redhat.com>
Date: Thu, 1 May 2014 10:35:12 -0400
Subject: [PATCH] start-statd: rpc.statd's systemd unit changed names.

Signed-off-by: Steve Dickson <steved@redhat.com>
---
 utils/statd/start-statd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/statd/start-statd b/utils/statd/start-statd
index 8ac3798..dcdaf77 100644
--- a/utils/statd/start-statd
+++ b/utils/statd/start-statd
@@ -5,7 +5,7 @@
 # It should run statd with whatever flags are apropriate for this
 # site.
 PATH="/sbin:/usr/sbin:/bin:/usr/bin"
-if systemctl start statd.service
+if systemctl start rpc-statd.service
 then :
 else
     exec rpc.statd --no-notify
-- 
1.9.0

