diff -up nfs-utils-1.3.2/systemd/rpc-gssd.service.orig nfs-utils-1.3.2/systemd/rpc-gssd.service
--- nfs-utils-1.3.2/systemd/rpc-gssd.service.orig	2015-01-30 15:42:15.000000000 -0500
+++ nfs-utils-1.3.2/systemd/rpc-gssd.service	2015-06-10 09:24:57.224547972 -0400
@@ -16,4 +16,4 @@ After=nfs-config.service
 EnvironmentFile=-/run/sysconfig/nfs-utils
 
 Type=forking
-ExecStart=/usr/sbin/rpc.gssd $GSSDARGS
+ExecStart=/usr/sbin/rpc.gssd $RPCGSSDARGS
diff -up nfs-utils-1.3.2/systemd/rpc-svcgssd.service.orig nfs-utils-1.3.2/systemd/rpc-svcgssd.service
--- nfs-utils-1.3.2/systemd/rpc-svcgssd.service.orig	2015-06-10 08:28:05.471097916 -0400
+++ nfs-utils-1.3.2/systemd/rpc-svcgssd.service	2015-06-10 09:25:44.691158360 -0400
@@ -17,4 +17,4 @@ After=nfs-config.service
 [Service]
 EnvironmentFile=-/run/sysconfig/nfs-utils
 Type=forking
-ExecStart=/usr/sbin/rpc.svcgssd $SVCGSSDARGS
+ExecStart=/usr/sbin/rpc.svcgssd $RPCSVCGSSDARGS
