diff -up nfs-utils-1.2.8/utils/statd/sm-notify.man.orig nfs-utils-1.2.8/utils/statd/sm-notify.man
--- nfs-utils-1.2.8/utils/statd/sm-notify.man.orig	2013-04-22 12:47:20.000000000 -0400
+++ nfs-utils-1.2.8/utils/statd/sm-notify.man	2013-09-18 15:32:01.511494000 -0400
@@ -184,7 +184,7 @@ where NSM state information resides.
 If this option is not specified,
 .B sm-notify
 uses
-.I /var/lib/nfs
+.I /var/lib/nfs/statd
 by default.
 .IP
 After starting,
@@ -285,13 +285,13 @@ Currently, the
 command supports sending notification only via datagram transport protocols.
 .SH FILES
 .TP 2.5i
-.I /var/lib/nfs/sm
+.I /var/lib/nfs/statd/sm
 directory containing monitor list
 .TP 2.5i
-.I /var/lib/nfs/sm.bak
+.I /var/lib/nfs/statd/sm.bak
 directory containing notify list
 .TP 2.5i
-.I /var/lib/nfs/state
+.I /var/lib/nfs/statd/state
 NSM state number for this host
 .TP 2.5i
 .I /proc/sys/fs/nfs/nsm_local_state
diff -up nfs-utils-1.2.8/utils/statd/statd.man.orig nfs-utils-1.2.8/utils/statd/statd.man
--- nfs-utils-1.2.8/utils/statd/statd.man.orig	2013-04-22 12:47:20.000000000 -0400
+++ nfs-utils-1.2.8/utils/statd/statd.man	2013-09-18 15:32:01.520491000 -0400
@@ -233,7 +233,7 @@ where NSM state information resides.
 If this option is not specified,
 .B rpc.statd
 uses
-.I /var/lib/nfs
+.I /var/lib/nfs/statd
 by default.
 .IP
 After starting,
@@ -357,13 +357,13 @@ As long as at least one network transpor
 will operate.
 .SH FILES
 .TP 2.5i
-.I /var/lib/nfs/sm
+.I /var/lib/nfs/statd/sm
 directory containing monitor list
 .TP 2.5i
-.I /var/lib/nfs/sm.bak
+.I /var/lib/nfs/statd/sm.bak
 directory containing notify list
 .TP 2.5i
-.I /var/lib/nfs/state
+.I /var/lib/nfs/statd/state
 NSM state number for this host
 .TP 2.5i
 .I /var/run/run.statd.pid
