diff -up nfs-utils-1.2.8/support/nfs/exports.c.orig nfs-utils-1.2.8/support/nfs/exports.c
--- nfs-utils-1.2.8/support/nfs/exports.c.orig	2013-09-24 15:35:26.022548905 -0400
+++ nfs-utils-1.2.8/support/nfs/exports.c	2013-09-24 15:35:51.274547991 -0400
@@ -508,7 +508,7 @@ void fix_pseudoflavor_flags(struct expor
 static int
 parseopts(char *cp, struct exportent *ep, int warn, int *had_subtree_opt_ptr)
 {
-	int	had_subtree_opt = 0;
+	int	had_subtree_opt = 1;
 	char 	*flname = efname?efname:"command line";
 	int	flline = efp?efp->x_line:0;
 	unsigned int active = 0;
