diff -up evolution-2.24.1/configure.ldaphack evolution-2.24.1/configure
--- evolution-2.24.1/configure.ldaphack	2008-10-19 23:37:55.000000000 -0400
+++ evolution-2.24.1/configure	2008-10-21 13:16:16.000000000 -0400
@@ -28471,7 +28471,7 @@ if test "${ac_cv_lib_ldap_ldap_open+set}
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lldap $LDAP_LDFLAGS $LDAP_LIBS $LIBS"
+LIBS="-lldap -lresolv $LDAP_LDFLAGS $LDAP_LIBS $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
