diff -up texlive-2007/libs/icu-xetex/tools/pkgdata/make.c.execshield texlive-2007/libs/icu-xetex/tools/pkgdata/make.c
--- texlive-2007/libs/icu-xetex/tools/pkgdata/make.c.execshield	2006-07-25 14:37:43.000000000 +0200
+++ texlive-2007/libs/icu-xetex/tools/pkgdata/make.c	2010-05-24 15:07:38.721047876 +0200
@@ -429,7 +429,7 @@ pkg_mak_writeAssemblyHeader(FileStream *
     T_FileStream_writeLine(f, "\n");
     T_FileStream_writeLine(f, "$(TEMP_DIR)/$(NAME)_dat.o : $(TEMP_DIR)/$(NAME).dat\n");
     T_FileStream_writeLine(f, "\t$(INVOKE) $(GENCCODE) $(GENCCODE_ASSEMBLY) -n $(NAME) -e $(ENTRYPOINT) -d $(TEMP_DIR) $<\n");
-    T_FileStream_writeLine(f, "\t$(COMPILE.c) $(DYNAMICCPPFLAGS) $(DYNAMICCXXFLAGS) -o $@ $(TEMP_DIR)/$(NAME)_dat"ASM_SUFFIX"\n");
+    T_FileStream_writeLine(f, "\t$(COMPILE.c) $(DYNAMICCPPFLAGS) $(DYNAMICCXXFLAGS) -Wa,--noexecstack -o $@ $(TEMP_DIR)/$(NAME)_dat"ASM_SUFFIX"\n");
     T_FileStream_writeLine(f, "\t$(RMV) $(TEMP_DIR)/$(NAME)_dat"ASM_SUFFIX"\n");
     T_FileStream_writeLine(f, "\n");
     T_FileStream_writeLine(f, "else\n");
