diff -up libs/icu-xetex/common/Makefile.in.newmake libs/icu-xetex/common/Makefile.in
--- libs/icu-xetex/common/Makefile.in.newmake	2010-08-06 17:02:36.889621273 -0300
+++ libs/icu-xetex/common/Makefile.in	2010-08-06 18:13:03.931578011 -0300
@@ -158,7 +158,8 @@ unicode/platform.h: $(srcdir)/unicode/pl
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
diff -up libs/icu-xetex/i18n/Makefile.in.newmake libs/icu-xetex/i18n/Makefile.in
--- libs/icu-xetex/i18n/Makefile.in.newmake	2010-08-06 18:08:34.388695983 -0300
+++ libs/icu-xetex/i18n/Makefile.in	2010-08-06 18:13:05.284281341 -0300
@@ -149,7 +149,8 @@ Makefile: $(srcdir)/Makefile.in  $(top_b
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
diff -up libs/icu-xetex/io/Makefile.in.newmake libs/icu-xetex/io/Makefile.in
--- libs/icu-xetex/io/Makefile.in.newmake	2010-08-06 18:17:15.446462952 -0300
+++ libs/icu-xetex/io/Makefile.in	2010-08-06 18:17:23.537692887 -0300
@@ -133,7 +133,8 @@ Makefile: $(srcdir)/Makefile.in  $(top_b
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
diff -up libs/icu-xetex/layout/Makefile.in.newmake libs/icu-xetex/layout/Makefile.in
--- libs/icu-xetex/layout/Makefile.in.newmake	2010-08-06 18:10:06.679453515 -0300
+++ libs/icu-xetex/layout/Makefile.in	2010-08-06 18:13:06.482018658 -0300
@@ -194,7 +194,8 @@ Makefile: $(srcdir)/Makefile.in  $(top_b
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
diff -up libs/icu-xetex/layoutex/Makefile.in.newmake libs/icu-xetex/layoutex/Makefile.in
--- libs/icu-xetex/layoutex/Makefile.in.newmake	2010-08-06 18:10:55.807678314 -0300
+++ libs/icu-xetex/layoutex/Makefile.in	2010-08-06 18:13:08.175647218 -0300
@@ -129,7 +129,8 @@ Makefile: $(srcdir)/Makefile.in  $(top_b
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
diff -up libs/icu-xetex/tools/ctestfw/Makefile.in.newmake libs/icu-xetex/tools/ctestfw/Makefile.in
--- libs/icu-xetex/tools/ctestfw/Makefile.in.newmake	2010-08-06 18:16:22.693003399 -0300
+++ libs/icu-xetex/tools/ctestfw/Makefile.in	2010-08-06 18:16:35.984095982 -0300
@@ -96,7 +96,8 @@ Makefile: $(srcdir)/Makefile.in  $(top_b
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
diff -up libs/icu-xetex/tools/toolutil/Makefile.in.newmake libs/icu-xetex/tools/toolutil/Makefile.in
--- libs/icu-xetex/tools/toolutil/Makefile.in.newmake	2010-08-06 18:12:02.566036642 -0300
+++ libs/icu-xetex/tools/toolutil/Makefile.in	2010-08-06 18:13:50.642333652 -0300
@@ -98,7 +98,8 @@ Makefile: $(srcdir)/Makefile.in  $(top_b
 	 && CONFIG_FILES=$(subdir)/$@ CONFIG_HEADERS= $(SHELL) ./config.status
 
 ifneq ($(ENABLE_STATIC),)
-$(TARGET): $(TARGET)($(STATIC_OBJECTS))
+$(TARGET): $(STATIC_OBJECTS)
+	$(AR) rv $@ $^
 	$(RANLIB) $@
 endif
 
