--- ptex-src-3.1.11/ptex-base.ch.fixptex	2011-11-12 00:41:52.705478048 -0200
+++ ptex-src-3.1.11/ptex-base.ch	2011-11-12 00:43:43.402053138 -0200
@@ -92,7 +92,7 @@ message_printing: boolean; { true if mes
 {Initialize enc\TeX\ data.}
 for i:=0 to 255 do mubyte_read[i]:=null;
 for i:=0 to 255 do mubyte_write[i]:=0;
-for i:=0 to 128 do mubyte_cswrite[i]:=null;
+for i:=0 to 127 do mubyte_cswrite[i]:=null;
 mubyte_keep := 0; mubyte_start := false;
 write_noexpanding := false; cs_converting := false;
 special_printing := false; message_printing := false;
