--- texk/tetex/context/texmfstart.rb19	2010-08-06 16:51:30.119725569 -0300
+++ texk/tetex/context/texmfstart	2010-08-06 16:52:43.498644583 -0300
@@ -34,7 +34,7 @@
 $: << File.expand_path(File.dirname($0)) ; $: << File.join($:.last,'lib') ; $:.uniq!
 
 require "rbconfig"
-require "md5"
+require "digest/md5"
 
 # kpse_merge_done: require 'base/kpseremote'
 # kpse_merge_done: require 'base/kpsedirect'
