--- texlive.tlpdb.orig	2014-05-09 06:00:14.065003389 +0100
+++ texlive.tlpdb	2014-05-14 08:06:39.583267360 +0100
@@ -76,7 +76,7 @@
 catalogue-ctan /macros/latex/contrib/abc
 catalogue-date 2008-03-08 20:47:21 +0100
 catalogue-license lppl
-catalogue-version .0
+catalogue-version 0
 
 name abntex2
 category Package
@@ -9852,7 +9852,7 @@
 catalogue-license pd
 catalogue-version 1.0a
 
-name Asana-Math
+name asana-math
 category Package
 revision 31750
 shortdesc A font to typeset maths in Xe(La)TeX and Lua(La)TeX.
@@ -15095,38 +15095,6 @@
 catalogue-date 2006-12-29 13:11:21 +0100
 catalogue-license gpl
 
-name biber
-category Package
-revision 32012
-shortdesc A BibTeX replacement for users of biblatex.
-longdesc Biber is a BibTeX replacement for users of biblatex. Biber
-longdesc supports full UTF-8, can (re)-encode input and output, supports
-longdesc highly configurable sorting, dynamic bibliography sets and many
-longdesc other features. The CTAN distribution offers a compressed tar
-longdesc archive of the sources, etc., together with "binary"
-longdesc distributions for a variety of platforms. Note: on SourceForge
-longdesc biber is formally named "biblatex-biber", to distinguish it
-longdesc from an earlier (now apparently moribund) project called
-longdesc "biber".
-depend biber.ARCH
-docfiles size=87
- texmf-dist/doc/bibtex/biber/biber.pdf
-srcfiles size=542
- texmf-dist/source/bibtex/biber/Changes
- texmf-dist/source/bibtex/biber/biblatex-biber.tar.gz
- texmf-dist/source/bibtex/biber/utf8-macro-map.html
-catalogue-ctan /biblio/biber
-catalogue-date 2013-10-27 19:01:04 +0100
-catalogue-license artistic
-catalogue-version 1.8
-
-name biber.i386-linux
-category Package
-revision 32012
-shortdesc i386-linux files of biber
-binfiles arch=i386-linux size=4315
- bin/i386-linux/biber
-
 name bibexport
 category Package
 revision 29349
@@ -29596,7 +29564,6 @@
 depend tex
 depend texconfig
 depend texlive-common
-depend texlive-docindex
 depend texlive-en
 depend texlive-msg-translations
 depend texlive-scripts
@@ -29820,7 +29787,7 @@
 shortdesc Additional fonts
 relocated 1
 depend collection-basic
-depend Asana-Math
+depend asana-math
 depend accanthis
 depend adforn
 depend adfsymbols
@@ -29953,7 +29920,7 @@
 depend librebaskerville
 depend librecaslon
 depend libris
-depend linearA
+depend lineara
 depend lobster2
 depend lxfonts
 depend ly1
@@ -32040,7 +32007,7 @@
 depend m-tx
 depend musixguit
 depend musixtex
-depend musixtex-fonts
+depend musixtex-fnts
 depend pmx
 depend pmxchords
 depend songbook
@@ -32518,25 +32485,6 @@
 depend xymtex
 depend youngtab
 
-name collection-texworks
-category Collection
-revision 25704
-shortdesc TeXworks editor
-relocated 1
-longdesc See http://tug.org/texworks.
-depend collection-basic
-depend texworks
-
-name collection-wintools
-category Collection
-revision 30307
-shortdesc Windows-only support programs
-relocated 1
-longdesc Utilities for Windows, since they are not readily available
-longdesc there: chktex, unzip, wget, xpdf, and the dviout previewer.
-depend wintools.win32
-depend dviout.win32
-
 name collection-xetex
 category Collection
 revision 30396
@@ -41736,7 +41684,7 @@
  texmf-dist/source/support/dosepsbin/install-sh
 catalogue-ctan /support/dosepsbin
 catalogue-date 2012-03-22 21:51:23 +0100
-catalogue-license artistic
+catalogue-license gpl
 catalogue-version 1.2
 
 name dosepsbin.i386-linux
@@ -69599,13 +69547,6 @@
 catalogue-license lppl
 catalogue-version v1.7
 
-name installfont.i386-linux
-category Package
-revision 19743
-shortdesc i386-linux files of installfont
-binfiles arch=i386-linux size=1
- bin/i386-linux/installfont-tl
-
 name interactiveworkbook
 category Package
 revision 15878
@@ -80497,7 +80438,7 @@
 catalogue-ctan /macros/latex/contrib/labels
 catalogue-date 2009-09-29 00:01:21 +0200
 catalogue-license lppl1.2
-catalogue-version .13
+catalogue-version 13
 
 name labyrinth
 category Package
@@ -81837,73 +81778,6 @@
 binfiles arch=i386-linux size=1
  bin/i386-linux/latexindent
 
-name latexmk
-category Package
-revision 32312
-shortdesc Fully automated LaTeX document generation.
-longdesc Latexmk completely automates the process of generating a LaTeX
-longdesc document. Given the source files for a document, latexmk issues
-longdesc the appropriate sequence of commands to generate a .dvi, .ps,
-longdesc .pdf or hardcopy version of the document. An important feature
-longdesc is the "preview continuous mode", where the script watches all
-longdesc of the source files (primary file and included TeX and graphics
-longdesc files), and reruns LaTeX, etc., whenever a source file has
-longdesc changed. Thus a previewer can offer a display of the document's
-longdesc latest state.
-depend latexmk.ARCH
-runfiles size=79
- texmf-dist/scripts/latexmk/latexmk.pl
-docfiles size=182
- texmf-dist/doc/man/man1/latexmk.1
- texmf-dist/doc/man/man1/latexmk.man1.pdf
- texmf-dist/doc/support/latexmk/CHANGES
- texmf-dist/doc/support/latexmk/COPYING
- texmf-dist/doc/support/latexmk/INSTALL
- texmf-dist/doc/support/latexmk/README
- texmf-dist/doc/support/latexmk/example_rcfiles/README
- texmf-dist/doc/support/latexmk/example_rcfiles/Sweave_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/TeX4ht-latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/asymptote_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/clean_pattern_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/exceltex_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/glossary_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/lualatex_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/mpost_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/pdflatexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/pythontex-latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/sagetex_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/sweave_latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/tex4ht-latexmkrc
- texmf-dist/doc/support/latexmk/example_rcfiles/xelatex_latexmkrc
- texmf-dist/doc/support/latexmk/extra-scripts/README1
- texmf-dist/doc/support/latexmk/extra-scripts/dvipdfm_call
- texmf-dist/doc/support/latexmk/extra-scripts/dvipdfm_call.bat
- texmf-dist/doc/support/latexmk/extra-scripts/dvipdfmx_call
- texmf-dist/doc/support/latexmk/extra-scripts/dvipdfmx_call.bat
- texmf-dist/doc/support/latexmk/extra-scripts/htlatexonly
- texmf-dist/doc/support/latexmk/extra-scripts/kickxdvi
- texmf-dist/doc/support/latexmk/extra-scripts/l1
- texmf-dist/doc/support/latexmk/extra-scripts/l2
- texmf-dist/doc/support/latexmk/extra-scripts/myhtlatex2
- texmf-dist/doc/support/latexmk/extra-scripts/pst2pdf_for_latexmk
- texmf-dist/doc/support/latexmk/extra-scripts/pst2pdf_for_latexmk_README.txt
- texmf-dist/doc/support/latexmk/extra-scripts/startacroread
- texmf-dist/doc/support/latexmk/latexmk.pdf
- texmf-dist/doc/support/latexmk/latexmk.txt
-srcfiles size=1
- texmf-dist/source/support/latexmk/latexmk.bat
-catalogue-ctan /support/latexmk
-catalogue-date 2013-12-03 18:34:20 +0100
-catalogue-license gpl
-catalogue-version 4.39
-
-name latexmk.i386-linux
-category Package
-revision 10937
-shortdesc i386-linux files of latexmk
-binfiles arch=i386-linux size=1
- bin/i386-linux/latexmk
-
 name latexmp
 category Package
 revision 15878
@@ -87297,7 +87171,7 @@
 catalogue-date 2007-01-08 23:14:30 +0100
 catalogue-license lppl
 
-name linearA
+name lineara
 category Package
 revision 15878
 shortdesc Linear A script fonts.
@@ -98850,7 +98724,7 @@
  texmf-dist/source/generic/mkjobtexmf/install-sh
 catalogue-ctan /support/mkjobtexmf
 catalogue-date 2011-11-16 11:07:17 +0100
-catalogue-license artistic
+catalogue-license gpl
 catalogue-version 0.8
 
 name mkjobtexmf.i386-linux
@@ -100911,7 +100785,7 @@
 catalogue-license lppl1.3
 catalogue-version 1.2.2
 
-name musixtex-fonts
+name musixtex-fnts
 category Package
 revision 28934
 shortdesc Fonts used by MusixTeX.
@@ -111584,6 +111458,7 @@
 longdesc LuaTeX).
 depend kpathsea
 depend pdftex.ARCH
+depend pdftex-def
 execute AddFormat name=pdftex  engine=pdftex patterns=language.def                     options="-translate-file=cp227.tcx *pdfetex.ini"
 execute AddFormat name=etex    engine=pdftex patterns=language.def                     options="-translate-file=cp227.tcx *etex.ini"
 execute AddFormat name=pdfetex engine=pdftex patterns=language.def                     options="-translate-file=cp227.tcx *pdfetex.ini"
@@ -119632,7 +119507,7 @@
 catalogue-ctan /graphics/pstricks/contrib/pst-layout
 catalogue-date 2012-04-27 12:45:13 +0200
 catalogue-license lppl
-catalogue-version .95
+catalogue-version 95
 
 name pst-lens
 category Package
@@ -132434,7 +132309,7 @@
 depend kurier
 depend poltawski
 depend xits
-depend Asana-Math
+depend asana-math
 depend gentium-tug
 depend txfonts
 depend pxfonts
@@ -139799,7 +139674,7 @@
  texmf-dist/doc/man/man1/texdirflatten.man1.pdf
 catalogue-ctan /support/texdirflatten
 catalogue-date 2012-01-12 19:29:56 +0100
-catalogue-license artistic
+catalogue-license gpl
 catalogue-version 1.1
 
 name texdirflatten.i386-linux
@@ -141373,7 +141248,6 @@
 revision 29741
 shortdesc i386-linux files of texlive-scripts
 binfiles arch=i386-linux size=2
- bin/i386-linux/man
  bin/i386-linux/rungs
 
 name texlive-sr
@@ -150645,7 +150519,7 @@
 catalogue-ctan /macros/latex209/contrib/trees/tree-dvips
 catalogue-date 2011-03-14 14:09:23 +0100
 catalogue-license lppl1
-catalogue-version .91
+catalogue-version 91
 
 name treetex
 category Package
@@ -162874,7 +162748,6 @@
 revision 30845
 shortdesc i386-linux files of xetex
 binfiles arch=i386-linux size=5465
- bin/i386-linux/teckit_compile
  bin/i386-linux/xdvipdfmx
  bin/i386-linux/xelatex
  bin/i386-linux/xetex
