diff -Nur parole-0.2.0.2.orig/browser-plugin/media-plugin/Makefile.in parole-0.2.0.2/browser-plugin/media-plugin/Makefile.in
--- parole-0.2.0.2.orig/browser-plugin/media-plugin/Makefile.in	2010-01-25 04:47:24.000000000 -0700
+++ parole-0.2.0.2/browser-plugin/media-plugin/Makefile.in	2010-02-18 15:16:52.000000000 -0700
@@ -198,7 +198,7 @@
 INTLTOOL_PERL = @INTLTOOL_PERL@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lX11
 LIBNOTIFY_CFLAGS = @LIBNOTIFY_CFLAGS@
 LIBNOTIFY_LIBS = @LIBNOTIFY_LIBS@
 LIBNOTIFY_REQUIRED_VERSION = @LIBNOTIFY_REQUIRED_VERSION@
diff -Nur parole-0.2.0.2.orig/src/Makefile.in parole-0.2.0.2/src/Makefile.in
--- parole-0.2.0.2.orig/src/Makefile.in	2010-01-25 04:47:27.000000000 -0700
+++ parole-0.2.0.2/src/Makefile.in	2010-02-18 13:55:57.000000000 -0700
@@ -205,7 +205,7 @@
 INTLTOOL_PERL = @INTLTOOL_PERL@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lX11
 LIBNOTIFY_CFLAGS = @LIBNOTIFY_CFLAGS@
 LIBNOTIFY_LIBS = @LIBNOTIFY_LIBS@
 LIBNOTIFY_REQUIRED_VERSION = @LIBNOTIFY_REQUIRED_VERSION@
