diff -Nur parole-0.2.0.2.orig/plugins/tray/tray-provider.c parole-0.2.0.2/plugins/tray/tray-provider.c
--- parole-0.2.0.2.orig/plugins/tray/tray-provider.c	2010-01-25 04:22:07.000000000 -0700
+++ parole-0.2.0.2/plugins/tray/tray-provider.c	2010-11-05 16:51:49.170270398 -0600
@@ -261,7 +261,7 @@
     
     message = g_strdup_printf ("%s %s %s %s", _("<b>Playing:</b>"), title, _("<b>Duration:</b>"), timestring);
     
-    tray->n = notify_notification_new (title, message, NULL, NULL);
+    tray->n = notify_notification_new (title, message, NULL);
     g_free (title);
     g_free (message);
     
