diff -up ode-0.13.1/ode-double-config~ ode-0.13.1/ode-double-config
--- ode-0.13.1/ode-double-config~	2014-10-29 13:22:58.000000000 +0100
+++ ode-0.13.1/ode-double-config	2014-10-29 13:24:17.436154237 +0100
@@ -39,10 +39,10 @@ while test $# -gt 0; do
       echo 0.13.1
       ;;
     --cflags)
-      echo
+      echo -DdIDEDOUBLE
       ;;
     --libs)
-      echo -lode
+      echo -lode-double
       ;;
     *)
       echo "${usage}" 1>&2
diff -up ode-0.13.1/ode-double.pc~ ode-0.13.1/ode-double.pc
--- ode-0.13.1/ode-double.pc~	2014-10-29 13:22:58.000000000 +0100
+++ ode-0.13.1/ode-double.pc	2014-10-29 13:24:19.693095427 +0100
@@ -7,6 +7,6 @@ precision=dDOUBLE
 Name: ode
 Description: Open Dynamics Engine
 Version: 0.13.1
-Libs: -L${libdir} -lode
+Libs: -L${libdir} -lode-double
 Libs.private: -lstdc++ -lm
-Cflags: -I${includedir}
+Cflags: -I${includedir} -DdIDEDOUBLE
