diff -up ode-0.10.0/configure.ac~ ode-0.10.0/configure.in
--- ode-0.10.0/configure.ac~	2008-07-02 11:41:50.000000000 +0200
+++ ode-0.10.0/configure.ac	2008-07-02 11:41:50.000000000 +0200
@@ -60,7 +60,7 @@ dnl this may NOT be the machine on which
 dnl so allow users to compile programs for their target machine.
 pentium=no
 cpu64=no
-case "$host_cpu" in
+case "$target_cpu" in
   i586 | i686 | i786 )
         pentium=yes
         AC_DEFINE(PENTIUM,1,[compiling for a pentium on a gcc-based platform?])
diff -up ode-0.10.0/configure~ ode-0.10.0/configure
--- ode-0.10.0/configure~	2008-07-02 11:41:48.000000000 +0200
+++ ode-0.10.0/configure	2008-07-02 11:41:48.000000000 +0200
@@ -20857,7 +20857,7 @@ echo "${ECHO_T}$enable_demos" >&6; }
 
 pentium=no
 cpu64=no
-case "$host_cpu" in
+case "$target_cpu" in
   i586 | i686 | i786 )
         pentium=yes
 
