diff -up ode-0.11.1/ode-config.in~ ode-0.11.1/ode-config.in
--- ode-0.11.1/ode-config.in~	2008-08-23 02:31:01.000000000 +0200
+++ ode-0.11.1/ode-config.in	2009-11-12 08:02:24.000000000 +0100
@@ -39,10 +39,10 @@ while test $# -gt 0; do
       echo @ODE_RELEASE@
       ;;
     --cflags)
-      echo  -I@includedir@ @ODE_PRECISION@
+      echo @ODE_PRECISION@
       ;;
     --libs)
-      echo  -L@libdir@ -lode
+      echo -lode
       ;;
     *)
       echo "${usage}" 1>&2
