diff -up ortp-0.23.0/include/ortp/ortp_srtp.h.srtp-15 ortp-0.23.0/include/ortp/ortp_srtp.h
--- ortp-0.23.0/include/ortp/ortp_srtp.h.srtp-15	2014-01-22 12:02:48.000000000 -0500
+++ ortp-0.23.0/include/ortp/ortp_srtp.h	2014-11-14 12:07:11.982503995 -0500
@@ -25,6 +25,8 @@
 // Android and Windows phone don't use make install
 #include <srtp.h>
 #else
+#include <srtp/config.h>
+#include <srtp/crypto_kernel.h>
 #include <srtp/srtp.h>
 #endif
 #else
