Index: numpy/core/src/multiarray/number.c
===================================================================
--- numpy/core/src/multiarray/number.c (revision 7888)
+++ numpy/core/src/multiarray/number.c (revision 8364)
@@ -31,4 +31,5 @@
             return -1;                                    \
         }                                                 \
+        Py_INCREF(temp);                                  \
         Py_XDECREF(n_ops.op);                             \
         n_ops.op = temp;                                  \
