diff -up numpy-1.4.1/numpy/core/src/multiarray/numpyos.c.remove-PyOS_ascii_strtod numpy-1.4.1/numpy/core/src/multiarray/numpyos.c
--- numpy-1.4.1/numpy/core/src/multiarray/numpyos.c.remove-PyOS_ascii_strtod	2010-07-22 19:15:18.846696632 -0400
+++ numpy-1.4.1/numpy/core/src/multiarray/numpyos.c	2010-07-22 19:16:24.454744969 -0400
@@ -506,7 +506,11 @@ NumPyOS_ascii_strtod(const char *s, char
             }
             memcpy(buffer, s, n);
             buffer[n] = '\0';
+#if PY_VERSION_HEX >= 0x02070000
+            result = PyOS_string_to_double(buffer, &q, NULL);
+#else
             result = PyOS_ascii_strtod(buffer, &q);
+#endif
             if (endptr != NULL) {
                 *endptr = (char*)(s + (q - buffer));
             }
@@ -515,7 +519,11 @@ NumPyOS_ascii_strtod(const char *s, char
     }
     /* End of ##2 */
 
+#if PY_VERSION_HEX >= 0x02070000
+    return PyOS_string_to_double(s, endptr, NULL);
+#else
     return PyOS_ascii_strtod(s, endptr);
+#endif
 }
 
 
