diff -up epdfview-0.1.7/configure.ac.poppler epdfview-0.1.7/configure.ac
--- epdfview-0.1.7/configure.ac.poppler	2009-02-28 18:00:49.000000000 -0300
+++ epdfview-0.1.7/configure.ac	2010-12-29 00:48:23.428213053 -0200
@@ -40,18 +40,22 @@ PKG_CHECK_MODULES([POPPLER], [poppler-gl
 AC_SUBST([POPPLER_CFLAGS])
 AC_SUBST([POPPLER_LIBS])
 dnl Check if we have poppler version 0.5.2 or higher.
-PKG_CHECK_EXISTS([poppler-glib >= 0.5.2], [have_poppler_052=yes])
-if test "x$have_poppler_052" = "xyes"; then
+PKG_CHECK_EXISTS([poppler-glib >= 0.5.2], [have_poppler_0_5_2=yes])
+if test "x$have_poppler_0_5_2" = "xyes"; then
     AC_DEFINE([HAVE_POPPLER_0_5_2], [1], [Define to 1 if you have Poppler version 0.5.2 or higher.])
 fi
-PKG_CHECK_EXISTS([poppler-glib >= 0.6], [have_poppler_060=yes])
-if test "x$have_poppler_060" = "xyes"; then
+PKG_CHECK_EXISTS([poppler-glib >= 0.6], [have_poppler_0_6_0=yes])
+if test "x$have_poppler_0_6_0" = "xyes"; then
     AC_DEFINE([HAVE_POPPLER_0_6_0], [1], [Define to 1 if you have Poppler version 0.6.0 or higher.])
 fi
-PKG_CHECK_EXISTS([poppler-glib >= 0.8], [have_poppler_080=yes])
-if test "x$have_poppler_080" = "xyes"; then
+PKG_CHECK_EXISTS([poppler-glib >= 0.8], [have_poppler_0_8_0=yes])
+if test "x$have_poppler_0_8_0" = "xyes"; then
     AC_DEFINE([HAVE_POPPLER_0_8_0], [1], [Define to 1 if you have Poppler version 0.8.0 or higher.])
 fi
+PKG_CHECK_EXISTS([poppler-glib >= 0.15], [have_poppler_0_15_0=yes])
+if test "x$have_poppler_0_15_0" = "xyes"; then
+    AC_DEFINE([HAVE_POPPLER_0_15_0], [1], [Define to 1 if you have Poppler version 0.15.0 or higher.])
+fi
 
 EPDFVIEW_PATH_CUPS([CUPS], [have_cups=yes])
 AM_CONDITIONAL(cups_printing, test "x$have_cups" = "xyes")
diff -up epdfview-0.1.7/src/PDFDocument.cxx.poppler epdfview-0.1.7/src/PDFDocument.cxx
--- epdfview-0.1.7/src/PDFDocument.cxx.poppler	2009-02-28 18:00:35.000000000 -0300
+++ epdfview-0.1.7/src/PDFDocument.cxx	2010-12-29 00:51:12.994899390 -0200
@@ -678,14 +678,24 @@ PDFDocument::setTextSelection (DocumentR
     gdouble pageWidth, pageHeight;
     poppler_page_get_size(page, &pageWidth, &pageHeight);
 
+#if defined (HAVE_POPPLER_0_15_0)
+    PopplerRectangle textRect = { rect->getX1() / getZoom(),
+                                  rect->getY1() / getZoom(),
+                                  rect->getX2() / getZoom(),
+                                  rect->getY2() / getZoom()};
+#else // !HAVE_POPPLER_0_15_0
     //for get text we must exchange y coordinate, don't ask me where logic here.
     PopplerRectangle textRect = { rect->getX1() / getZoom(),
                                   (pageHeight - rect->getY2() / getZoom()),
                                   rect->getX2() * getZoom() / getZoom(),
                                   (pageHeight - rect->getY1() / getZoom())};
+#endif // HAVE_POPPLER_0_15_0
     repairEmpty(textRect);
 
-#if defined (HAVE_POPPLER_0_6_0)
+#if defined (HAVE_POPPLER_0_15_0)
+    gchar *text = poppler_page_get_selected_text(page, POPPLER_SELECTION_GLYPH,
+            &textRect);
+#elif defined (HAVE_POPPLER_0_6_0)
     gchar *text = poppler_page_get_text(page, POPPLER_SELECTION_GLYPH,
             &textRect);
 #else // !HAVE_POPPLER_0_6_0
