diff -up openssh-5.8p1/sshd.c.exit openssh-5.8p1/sshd.c
--- openssh-5.8p1/sshd.c.exit	2011-01-11 07:20:31.000000000 +0100
+++ openssh-5.8p1/sshd.c	2011-03-15 12:06:20.623854928 +0100
@@ -1115,7 +1115,7 @@ server_accept_loop(int *sock_in, int *so
 			    (int) received_sigterm);
 			close_listen_socks();
 			unlink(options.pid_file);
-			exit(255);
+			exit(0);
 		}
 		if (key_used && key_do_regen) {
 			generate_ephemeral_server_key();
