diff -up openssh/configure.ac.seccomp openssh/configure.ac
diff -up openssh/sandbox-seccomp-filter.c.seccomp openssh/sandbox-seccomp-filter.c
--- openssh/sandbox-seccomp-filter.c.seccomp	2015-06-24 11:45:44.001581471 +0200
+++ openssh/sandbox-seccomp-filter.c	2015-06-24 11:51:54.032635297 +0200
@@ -165,6 +165,9 @@ static const struct sock_filter preauth_
 #ifdef __NR__newselect
 	SC_ALLOW(_newselect),
 #endif
+#ifdef __NR_pselect6 /* AArch64 */
+	SC_ALLOW(pselect6),
+#endif
 #ifdef __NR_poll
 	SC_ALLOW(poll),
 #endif
