diff -up openssh-6.8p1/packet.c.packet openssh-6.8p1/packet.c
--- openssh-6.8p1/packet.c.packet	2015-03-18 10:56:32.286930601 +0100
+++ openssh-6.8p1/packet.c	2015-03-18 10:58:38.535629739 +0100
@@ -371,6 +371,8 @@ ssh_packet_connection_is_on_socket(struc
 	struct sockaddr_storage from, to;
 	socklen_t fromlen, tolen;
 
+	if (!state)
+		return 0;
 	/* filedescriptors in and out are the same, so it's a socket */
 	if (state->connection_in == state->connection_out)
 		return 1;
