diff -up openssh-6.1p1/servconf.c.required-authentication openssh-6.1p1/servconf.c
--- openssh-6.1p1/servconf.c.required-authentication	2012-11-30 21:13:14.375382453 +0100
+++ openssh-6.1p1/servconf.c	2012-11-30 21:33:56.972017545 +0100
@@ -468,6 +468,8 @@ static struct {
 	{ "authorizedkeyscommanduser", sAuthorizedKeysCommandUser, SSHCFG_ALL },
 	{ "versionaddendum", sVersionAddendum, SSHCFG_GLOBAL },
 	{ "authenticationmethods", sAuthenticationMethods, SSHCFG_ALL },
+	{ "requiredauthentications1", sAuthenticationMethods, SSHCFG_ALL },
+	{ "requiredauthentications2", sAuthenticationMethods, SSHCFG_ALL },
 	{ NULL, sBadOption, 0 }
 };
 
@@ -1560,6 +1562,9 @@ process_server_config_line(ServerOptions
 		return 0;
 
 	case sAuthenticationMethods:
+		if (strncasecmp(arg, "requiredauthentications", 23) == 0)
+			logit("%s line %d: Option %s is obsolete. Please use AuthenticationMethods",
+			    filename, linenum, arg);
 		if (*activep && options->num_auth_methods == 0) {
 			while ((arg = strdelim(&cp)) && *arg != '\0') {
 				if (options->num_auth_methods >=
