diff -up pam_ssh_agent_auth-0.9.2/pam_ssh_agent_auth.c.visibility pam_ssh_agent_auth-0.9.2/pam_ssh_agent_auth.c
--- pam_ssh_agent_auth-0.9.2/pam_ssh_agent_auth.c.visibility	2009-12-21 20:57:34.000000000 +0100
+++ pam_ssh_agent_auth-0.9.2/pam_ssh_agent_auth.c	2012-06-21 20:01:31.356259429 +0200
@@ -68,7 +68,7 @@ char           *__progname;
 extern char    *__progname;
 #endif
 
-PAM_EXTERN int
+PAM_EXTERN int __attribute__ ((visibility ("default")))
 pam_sm_authenticate(pam_handle_t * pamh, int flags, int argc, const char **argv)
 {
     char          **argv_ptr;
@@ -184,7 +184,7 @@ pam_sm_authenticate(pam_handle_t * pamh,
 }
 
 
-PAM_EXTERN int
+PAM_EXTERN int __attribute__ ((visibility ("default")))
 pam_sm_setcred(pam_handle_t * pamh, int flags, int argc, const char **argv)
 {
     return PAM_SUCCESS;
