diff -Nru /tmp/bMVxNqare1/libvisual-0.4.0/libvisual/lv_cpu.c /tmp/AxCYZ8Ol5p/libvisual-0.4.0/libvisual/lv_cpu.c
--- libvisual-0.4.0/libvisual/lv_cpu.c	2008-02-22 01:03:41.000000000 +0000
+++ libvisual-0.4.0/libvisual/lv_cpu.c	2008-02-22 01:03:41.000000000 +0000
@@ -227,7 +227,7 @@
 		__asm __volatile ("xorps %xmm0, %xmm0");
 		SetUnhandledExceptionFilter(exc_fil);
 	}
-#elif defined(VISUAL_OS_LINUX)
+#elif defined(VISUAL_OS_LINUX) && defined(_POSIX_SOURCE) && defined(X86_FXSR_MAGIC)
 //	printf ("omg1\n");
 //	printf ("omg2\n");
 	struct sigaction saved_sigill;
