From 9cdd6c00920f3ff90e99d0805ce4512982cfea06 Mon Sep 17 00:00:00 2001
From: Curtis Gedak <gedakc@gmail.com>
Date: Tue, 07 Jun 2011 17:01:13 +0000
Subject: Minor format changes to final configuration summary

---
diff --git a/configure.in b/configure.in
index 06fa2c5..427de2a 100644
--- a/configure.in
+++ b/configure.in
@@ -210,14 +210,14 @@ dnl Summary
 dnl======================
 
 echo ""
-echo "================ Final configuration ==================="
-echo "                 Installing into prefix  :  $prefix"
+echo "===================== Final configuration ========================"
+echo "                      Installing into prefix  :  $prefix"
 echo ""
-echo "                   Build documentation?  :  $enable_doc"
+echo "                        Build documentation?  :  $enable_doc"
 echo ""
-echo "   Need part table re-read work around?  :  $need_pt_reread_work_around"
-echo "     Supports sector sizes > 512 bytes?  :  $support_sector_size_gt_512"
-echo "   Use native libparted dmraid support?  :  $enable_libparted_dmraid"
+echo "   Need partition table re-read work around?  :  $need_pt_reread_work_around"
+echo "          Supports sector sizes > 512 bytes?  :  $support_sector_size_gt_512"
+echo "        Use native libparted dmraid support?  :  $enable_libparted_dmraid"
 echo ""
 echo " If all settings are OK, type make and make install "
-echo "========================================================"
+echo "=================================================================="
--
cgit v0.9
