--- configure.in.gtk3	2015-08-16 12:39:12.038361045 -0300
+++ configure.in	2015-08-16 13:01:12.030491539 -0300
@@ -390,7 +390,17 @@ CONFIG_MATH_LIB="${USE_LIBM}"
 
 dnl configure use of features
 
-AM_PATH_GTK(1.2.0, HAVE_GTK="yes", HAVE_GTK="no")
+PKG_CHECK_MODULES(GTK, gtk+-3.0 >= 3.0, HAVE_GTK="yes", HAVE_GTK="no")
+
+if test "${HAVE_GTK}" = "yes" -o "x${HAVE_GTK}" != "x" \
+  -o "x${SNDFILE_CFLAGS}" != "x"; then
+      ac_save_CFLAGS="$CFLAGS"
+      ac_save_LIBS="$LIBS"
+      CFLAGS="$CFLAGS $GTK_CFLAGS"
+      LIBS="$GTK_LIBS $LIBS"
+      AC_SUBST(GTK_CFLAGS)
+      AC_SUBST(GTK_LIBS)
+fi
 
 dnl ElectricFence malloc debugging
 AC_MSG_CHECKING(use of ElectricFence malloc debugging)
