diff -Naur util-linux-2.21.2/text-utils/column.c util-linux-2.21.2.patched//text-utils/column.c
--- util-linux-2.21.2/text-utils/column.c	2012-05-25 05:44:59.000000000 -0400
+++ util-linux-2.21.2.patched//text-utils/column.c	2012-08-14 05:22:17.000000000 -0400
@@ -121,9 +121,9 @@
 	{
 		{ "help",	0, 0, 'h' },
 		{ "version",    0, 0, 'V' },
-		{ "columns",	0, 0, 'c' },
+		{ "columns",	1, 0, 'c' },
 		{ "table",	0, 0, 't' },
-		{ "separator",	0, 0, 's' },
+		{ "separator",	1, 0, 's' },
 		{ "fillrows",	0, 0, 'x' },
 		{ NULL,		0, 0, 0 },
 	};
