--- util-linux-2.24-rc1/include/pt-mbr-partnames.h.orig	2013-10-01 16:48:08.930215049 -0300
+++ util-linux-2.24-rc1/include/pt-mbr-partnames.h	2013-10-01 16:48:24.590914746 -0300
@@ -49,7 +49,7 @@
 	{0x75, N_("PC/IX")},
 	{0x80, N_("Old Minix")},	/* Minix 1.4a and earlier */
 	{0x81, N_("Minix / old Linux")},/* Minix 1.4b and later */
-	{0x82, N_("Linux swap / Solaris")},
+	{0x82, N_("Linux swap")},
 	{0x83, N_("Linux")},
 	{0x84, N_("OS/2 hidden C: drive")},
 	{0x85, N_("Linux extended")},
