diff -up util-linux-2.22.2/login-utils/login.c.kzak util-linux-2.22.2/login-utils/login.c
--- util-linux-2.22.2/login-utils/login.c.kzak	2012-12-12 21:04:47.906355128 +0100
+++ util-linux-2.22.2/login-utils/login.c	2013-02-19 16:14:33.224339106 +0100
@@ -514,7 +514,7 @@ static void log_lastlog(struct login_con
 	if (!cxt->pwd)
 		return;
 
-	fd = open(_PATH_LASTLOG, O_RDWR, 0);
+	fd = open(_PATH_LASTLOG, O_RDWR | O_CREAT, 0);
 	if (fd < 0)
 		return;
 
