diff -up lxdvdrip/Makefile.orig lxdvdrip/Makefile
--- lxdvdrip/Makefile.orig	2011-10-21 16:48:46.000000000 -0200
+++ lxdvdrip/Makefile	2011-11-02 02:20:07.200962839 -0200
@@ -1,29 +1,33 @@
-INSTALLDIR = /usr/local
-HINWEIS0 = 'Bitte Konfigurationsdatei in /etc mit Changelog abgleichen'
-HINWEIS1 = 'Check the configuration file in /etc with the Changelog'
-HINWEIS2 = 'Comparez le fichier de config. en /etc avec le Changelog'
+INSTALLDIR =
+CFLAGS ?= -O2
+LDFLAGS ?=
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+MANDIR ?= $(PREFIX)/man
+DATADIR ?= $(PREFIX)/share
+SYSCONFDIR ?= /etc
 
 all:
 	if test -f badsect.o; then rm *.o; fi
-	gcc -g -c badsect.c
-	gcc -g -c cputest.c
-	gcc -g -c dvdbackup.c
-	gcc -g -c dvdcell.c
-	gcc -g -c dvdcopy.c
-	gcc -g -c dvdformat.c
-	gcc -g -c dvdinfo.c
-	gcc -g -c dvdtools.c
-	gcc -g -c ifo.c
-	gcc -g -c lxdvdrip.c
-	gcc -g -c mpeg2dec.c
-	gcc -g -c requant.c
-	gcc -g -c streamanalyze.c
-	gcc -g -c systools.c
-	gcc -g -c tcmemcpy.c
-	gcc -g -c vaporize.c
-	gcc -pthread  -g -o lxdvdrip *o -ldvdread -lm
+	gcc -g $(CFLAGS) -c badsect.c
+	gcc -g $(CFLAGS) -c cputest.c
+	gcc -g $(CFLAGS) -c dvdbackup.c
+	gcc -g $(CFLAGS) -c dvdcell.c
+	gcc -g $(CFLAGS) -c dvdcopy.c
+	gcc -g $(CFLAGS) -c dvdformat.c
+	gcc -g $(CFLAGS) -c dvdinfo.c
+	gcc -g $(CFLAGS) -c dvdtools.c
+	gcc -g $(CFLAGS) -c ifo.c
+	gcc -g $(CFLAGS) -c lxdvdrip.c
+	gcc -g $(CFLAGS) -c mpeg2dec.c
+	gcc -g $(CFLAGS) -c requant.c
+	gcc -g $(CFLAGS) -c streamanalyze.c
+	gcc -g $(CFLAGS) -c systools.c
+	gcc -g $(CFLAGS) -c tcmemcpy.c
+	gcc -g $(CFLAGS) -c vaporize.c
+	gcc -pthread  -g $(CFLAGS) -o lxdvdrip *o $(LDFLAGS) -ldvdread -lm
 	if test -f badsect.o; then rm *.o; fi
-	gcc -g -lm -o lxac3scan lxac3scan.c
+	gcc -g $(CFLAGS) $(LDFLAGS) -lm -o lxac3scan lxac3scan.c
 	cd vamps && make && cd ..
 	cd dvdbackup && make && cd ..
 	cd requant && make && cd ..
@@ -37,25 +41,17 @@ clean:
 	cd mbuffer && make clean && cd ..
 
 install:
-	cp lxdvdrip $(INSTALLDIR)/bin
-	cp lxac3scan $(INSTALLDIR)/bin
-	cp lxdvdrip.1 $(INSTALLDIR)/man/man1
-	cp lxdvdrip.wav $(INSTALLDIR)/share
+	mkdir -p $(INSTALLDIR)/$(BINDIR)
+	mkdir -p $(INSTALLDIR)/$(MANDIR)/man1
+	mkdir -p $(INSTALLDIR)/$(DATADIR)/lxdvdrip
+	mkdir -p $(INSTALLDIR)/$(SYSCONFDIR)
+	cp lxdvdrip $(INSTALLDIR)/$(BINDIR)
+	cp lxac3scan $(INSTALLDIR)/$(BINDIR)
+	cp lxdvdrip.1 $(INSTALLDIR)/$(MANDIR)/man1
+	cp lxdvdrip.wav $(INSTALLDIR)/$(DATADIR)/lxdvdrip
+	cp lxdvdrip.conf $(INSTALLDIR)/$(SYSCONFDIR)
 	cd vamps && make install && cd ..
 	cd dvdbackup && make install && cd ..
 	cd requant && make install && cd ..
 	cd mbuffer && make install && cd ..
-	if test -f /etc/lxdvdrip.conf; then echo $(HINWEIS0); fi;
-	if test -f /etc/lxdvdrip.conf; then echo $(HINWEIS1); fi;
-	if test -f /etc/lxdvdrip.conf; then echo $(HINWEIS2); else cp lxdvdrip.conf /etc; fi;
 
-uninstall:
-	rm $(INSTALLDIR)/bin/lxdvdrip
-	rm $(INSTALLDIR)/bin/lxac3scan
-	rm $(INSTALLDIR)/man/man1/lxdvdrip.1
-	rm $(INSTALLDIR)/share/lxdvdrip.wav
-	rm $(INSTALLDIR)/bin/vamps_lxdvdrip
-	rm $(INSTALLDIR)/bin/play_cell_lxdvdrip
-	rm $(INSTALLDIR)/bin/dvdbackup_lxdvdrip
-	rm $(INSTALLDIR)/bin/requant_lxdvdrip
-	rm $(INSTALLDIR)/bin/mbuffer_lxdvdrip
diff -up lxdvdrip/dvdbackup/Makefile.orig lxdvdrip/dvdbackup/Makefile
--- lxdvdrip/dvdbackup/Makefile.orig	2011-10-05 18:24:37.000000000 -0300
+++ lxdvdrip/dvdbackup/Makefile	2011-11-02 02:18:02.024616065 -0200
@@ -1,14 +1,15 @@
 # This file is part of DVDBACKUP.
 #
 
-
-PREFIX    ?= /usr/local
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INSTALLDIR = 
 
 CC         = gcc
-CFLAGS    ?= -O2  -g -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 \
+CFLAGS    ?= -O2
+CFLAGS    += -Wall -g -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 \
 	       -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_REENTRANT
-CFLAGS    += -Wall
-LDFLAGS   += -s
+LDFLAGS   ?=
 LOADLIBES += -lm -ldvdread 
 
 INSTALL   ?= install
@@ -19,11 +20,8 @@ all:			dvdbackup_lxdvdrip
 dvdbackup_lxdvdrip:	dvdbackup_lxdvdrip.o
 
 install:	all
-		$(INSTALL) -d $(PREFIX)/bin
-		$(INSTALL) -m 755 dvdbackup_lxdvdrip $(PREFIX)/bin
-
-uninstall:
-		rm -f $(PREFIX)/bin/dvdbackup_lxdvdrip
+		$(INSTALL) -d $(INSTALLDIR)/$(BINDIR)
+		$(INSTALL) -m 755 dvdbackup_lxdvdrip $(INSTALLDIR)/$(BINDIR)
 
 clean:
 		rm -f *_lxdvdrip  *.o *~ core
diff -up lxdvdrip/mbuffer/Makefile.orig lxdvdrip/mbuffer/Makefile
--- lxdvdrip/mbuffer/Makefile.orig	2011-10-24 19:46:37.000000000 -0200
+++ lxdvdrip/mbuffer/Makefile	2011-11-02 02:18:33.591894554 -0200
@@ -1,6 +1,10 @@
 
-# Where to install buffer 
-INSTBIN=/usr/local/bin
+CFLAGS ?= -O2
+LDFLAGS ?=
+# Where to install buffer
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INSTALLDIR = 
 
 RM=/bin/rm
 ALL=Makefile log.c mbuffer.c network.c
@@ -9,18 +13,15 @@ all: buffer
 
 buffer: log.c mbuffer.c network.c
 	if test -f log.o; then rm *.o; fi
-	gcc -g -O2 -c log.c -Wno-unused-result
-	gcc -g -O2 -c mbuffer.c -w
-	gcc -g -O2 -c network.c -Wno-unused-result
-	gcc -pthread -O2 -g -o mbuffer_lxdvdrip *o -lrt -lm
+	gcc -g $(CFLAGS) -c log.c -Wno-unused-result
+	gcc -g $(CFLAGS) -c mbuffer.c -w
+	gcc -g $(CFLAGS) -c network.c -Wno-unused-result
+	gcc -pthread $(CFLAGS) -g -o mbuffer_lxdvdrip *o $(LDFLAGS) -lrt -lm
 	if test -f log.o; then rm *.o; fi
 
 clean:
 	$(RM) -f *.o core mbuffer_lxdvdrip .merrs
 
 install: buffer
-	cp mbuffer_lxdvdrip $(INSTBIN)/mbuffer_lxdvdrip
-	chmod 755 $(INSTBIN)/mbuffer_lxdvdrip
-
-uninstall: buffer
-	rm -f $(INSTBIN)/mbuffer_lxdvdrip
+	cp mbuffer_lxdvdrip $(INSTALLDIR)/$(BINDIR)/mbuffer_lxdvdrip
+	chmod 755 $(INSTALLDIR)/$(BINDIR)/mbuffer_lxdvdrip
diff -up lxdvdrip/requant/Makefile.orig lxdvdrip/requant/Makefile
--- lxdvdrip/requant/Makefile.orig	2011-10-17 19:22:14.000000000 -0200
+++ lxdvdrip/requant/Makefile	2011-11-02 02:18:05.974774969 -0200
@@ -1,14 +1,15 @@
 # This file is part of REQUANT
 #
 
-
-PREFIX    ?= /usr/local
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INSTALLDIR = 
 
 CC         = gcc
-CFLAGS    ?= -O2  -g -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 \
+CFLAGS    ?= -O2
+CFLAGS    += -Wall -Wno-unused-result -g -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 \
 	       -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_REENTRANT
-CFLAGS    += -Wall -Wno-unused-result
-LDFLAGS   += -s
+LDFLAGS   ?=
 LOADLIBES += -lm
 
 INSTALL   ?= install
@@ -19,11 +20,8 @@ all:			requant_lxdvdrip
 requant_lxdvdrip:	requant_lxdvdrip.o
 
 install:	all
-		$(INSTALL) -d $(PREFIX)/bin
-		$(INSTALL) -m 755 requant_lxdvdrip $(PREFIX)/bin
-
-uninstall:
-		rm -f $(PREFIX)/bin/requant_lxdvdrip
+		$(INSTALL) -d $(INSTALLDIR)/$(BINDIR)
+		$(INSTALL) -m 755 requant_lxdvdrip $(INSTALLDIR)/$(BINDIR)
 
 clean:
 		rm -f *_lxdvdrip  *.o *~ core
diff -up lxdvdrip/vamps/Makefile.orig lxdvdrip/vamps/Makefile
--- lxdvdrip/vamps/Makefile.orig	2011-10-17 19:22:14.000000000 -0200
+++ lxdvdrip/vamps/Makefile	2011-11-02 02:12:48.544365625 -0200
@@ -15,13 +15,15 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 
-PREFIX    ?= /usr/local
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INSTALLDIR = 
 
 CC         = gcc
-CFLAGS    ?= -O2  -g -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 \
+CFLAGS    ?= -O2
+CFLAGS    += -Wall -g -D_LARGEFILE_SOURCE -DLARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 \
 	       -D_XOPEN_SOURCE=500 -D_BSD_SOURCE -D_REENTRANT
-CFLAGS    += -Wall
-LDFLAGS   += -s
+LDFLAGS   ?= -s
 LOADLIBES += -lm -pthread -ldvdread
 
 INSTALL   ?= install
@@ -38,13 +40,9 @@ requant.o:		requant.h
 play_cell_lxdvdrip:	play_cell_lxdvdrip.o
 
 install:	all
-		$(INSTALL) -d $(PREFIX)/bin
-		$(INSTALL) -m 755 vamps_lxdvdrip $(PREFIX)/bin
-		$(INSTALL) -m 755 play_cell_lxdvdrip $(PREFIX)/bin
-
-uninstall:
-		rm -f $(PREFIX)/bin/vamps_lxdvdrip
-		rm -f $(PREFIX)/bin/play_cell_lxdvdrip
+		$(INSTALL) -d $(INSTALLDIR)/$(BINDIR)
+		$(INSTALL) -m 755 vamps_lxdvdrip $(INSTALLDIR)/$(BINDIR)
+		$(INSTALL) -m 755 play_cell_lxdvdrip $(INSTALLDIR)/$(BINDIR)
 
 clean:
 		rm -f *_lxdvdrip  *.o *~ core
