diff -up rtmpdump-2.2e/Makefile.shared rtmpdump-2.2e/Makefile
--- rtmpdump-2.2e/Makefile.shared	2010-06-14 18:00:04.000000000 -0300
+++ rtmpdump-2.2e/Makefile	2010-06-14 18:00:29.248792454 -0300
@@ -40,7 +40,7 @@ THREADLIB_mingw=
 THREADLIB=$(THREADLIB_$(SYS))
 SLIBS=$(THREADLIB) $(LIBS)
 
-LIBRTMP=librtmp/librtmp.a
+LIBRTMP=librtmp/librtmp.so
 INCRTMP=librtmp/rtmp_sys.h librtmp/rtmp.h librtmp/log.h librtmp/amf.h
 
 EXT_posix=
diff -up rtmpdump-2.2e/librtmp/Makefile.shared rtmpdump-2.2e/librtmp/Makefile
--- rtmpdump-2.2e/librtmp/Makefile.shared	2010-05-29 04:42:20.000000000 -0300
+++ rtmpdump-2.2e/librtmp/Makefile	2010-06-14 17:57:56.267792525 -0300
@@ -1,4 +1,5 @@
 VERSION=v2.2e
+SONAME=$(shell echo $(VERSION) | cut -d. -f1 | sed -e 's|^v||g')
 
 prefix=/usr/local
 
@@ -19,7 +20,7 @@ CRYPTO_DEF=$(DEF_$(CRYPTO))
 
 DEF=-DRTMPDUMP_VERSION=\"$(VERSION)\" $(CRYPTO_DEF) $(XDEF)
 OPT=-O2
-CFLAGS=-Wall $(XCFLAGS) $(INC) $(DEF) $(OPT)
+CFLAGS=-Wall -fPIC $(XCFLAGS) $(INC) $(DEF) $(OPT)
 
 incdir=$(prefix)/include/librtmp
 libdir=$(prefix)/lib
@@ -28,14 +29,18 @@ INCDIR=$(DESTDIR)$(incdir)
 LIBDIR=$(DESTDIR)$(libdir)
 MANDIR=$(DESTDIR)$(mandir)
 
-all:	librtmp.a
+all:	librtmp.a librtmp.so
 
 clean:
-	rm -f *.o *.a
+	rm -f *.o *.a *.so
 
 librtmp.a: rtmp.o log.o amf.o hashswf.o parseurl.o
 	$(AR) rs $@ $?
 
+librtmp.so: rtmp.o log.o amf.o hashswf.o parseurl.o
+	$(CC) $(CFLAGS) -fPIC $(LDFLAGS) -shared -Wl,-soname,$@.$(SONAME) -o $@.$(SONAME).0.0 $?
+	ln -sf $@.$(SONAME).0.0 $@
+
 log.o: log.c log.h Makefile
 rtmp.o: rtmp.c rtmp.h rtmp_sys.h handshake.h dh.h log.h amf.h Makefile
 amf.o: amf.c amf.h bytes.h log.h Makefile
@@ -46,9 +51,12 @@ librtmp.pc: librtmp.pc.in Makefile
 	sed -e "s;@prefix@;$(prefix);" -e "s;@VERSION@;$(VERSION);" \
 		-e "s;@CRYPTO_REQ@;$(CRYPTO_REQ);" librtmp.pc.in > $@
 
-install:	librtmp.a librtmp.pc
+install:	librtmp.a librtmp.so librtmp.pc
 	-mkdir -p $(INCDIR) $(LIBDIR)/pkgconfig $(MANDIR)/man3
 	cp amf.h http.h log.h rtmp.h $(INCDIR)
 	cp librtmp.a $(LIBDIR)
+	cp librtmp.so.$(SONAME).0.0 $(LIBDIR)
+	ln -s librtmp.so.$(SONAME).0.0 $(LIBDIR)/librtmp.so.$(SONAME) 
+	ln -s librtmp.so.$(SONAME) $(LIBDIR)/librtmp.so
 	cp librtmp.pc $(LIBDIR)/pkgconfig
 	cp librtmp.3 $(MANDIR)/man3
