diff -up soprano-2.4.3/client/localsocketclient.cpp.missing-qdebug-header soprano-2.4.3/client/localsocketclient.cpp
--- soprano-2.4.3/client/localsocketclient.cpp.missing-qdebug-header	2010-04-22 04:35:08.000000000 -0500
+++ soprano-2.4.3/client/localsocketclient.cpp	2010-04-22 08:15:27.604825134 -0500
@@ -23,6 +23,7 @@
 #include "clientconnection.h"
 #include "clientmodel.h"
 
+#include <QtCore/QDebug>
 #include <QtCore/QDir>
 #include <QtCore/QThread>
 #include <QtNetwork/QLocalSocket>
diff -up soprano-2.4.3/server/localserver.cpp.missing-qdebug-header soprano-2.4.3/server/localserver.cpp
--- soprano-2.4.3/server/localserver.cpp.missing-qdebug-header	2010-04-12 07:14:57.000000000 -0500
+++ soprano-2.4.3/server/localserver.cpp	2010-04-22 08:23:15.025563785 -0500
@@ -23,6 +23,7 @@
 #include "serverconnection.h"
 #include "servercore_p.h"
 
+#include <QtCore/QDebug>
 #include <QtNetwork/QLocalSocket>
 
 namespace {
diff -up soprano-2.4.3/server/tcpserver.cpp.missing-qdebug-header soprano-2.4.3/server/tcpserver.cpp
--- soprano-2.4.3/server/tcpserver.cpp.missing-qdebug-header	2010-04-12 07:14:57.000000000 -0500
+++ soprano-2.4.3/server/tcpserver.cpp	2010-04-22 08:25:26.484562301 -0500
@@ -23,6 +23,7 @@
 #include "serverconnection.h"
 #include "servercore_p.h"
 
+#include <QtCore/QDebug>
 #include <QtNetwork/QTcpSocket>
 
 namespace {
