--- kioslave/nfs/CMakeLists.txt.orig	2011-02-25 18:54:12.000000000 -0300
+++ kioslave/nfs/CMakeLists.txt	2011-06-11 00:30:06.146863488 -0300
@@ -3,8 +3,13 @@ set(kio_nfs_PART_SRCS kio_nfs.cpp mount_
 
 kde4_add_plugin(kio_nfs ${kio_nfs_PART_SRCS})
 
+if(NOT DEFINED TIRPC_INCLUDE_DIR)
+  set(TIRPC_INCLUDE_DIR "${INCLUDE_INSTALL_DIR}/tirpc" CACHE PATH "Where TIRPC include files are located")
+endif()
 
-target_link_libraries(kio_nfs   ${KDE4_KIO_LIBS})
+include_directories ( ${TIRPC_INCLUDE_DIR} )
+
+target_link_libraries(kio_nfs tirpc  ${KDE4_KIO_LIBS})
 
 install(TARGETS kio_nfs  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
