Index: nepomuk/server/nepomukserver.cpp
===================================================================
--- nepomuk/server/nepomukserver.cpp
+++ nepomuk/server/nepomukserver.cpp	2010-05-09 01:21:28.000000000 +0200
@@ -94,6 +94,8 @@
 
             // now nepomuk is enabled
             m_enabled = true;
+            KConfig cnf( "kdeglobals" );
+            cnf.group( "KFileDialog Settings" ).writeEntry( "file module", "smartfilemodule" );
         }
         else {
             // stop all running services
@@ -107,7 +109,10 @@
             m_legacyStorageBridge = 0;
 
             // nepomuk is disabled
-            m_enabled = false;
+            m_enabled = false;           
+            
+            KConfig cnf( "kdeglobals" );
+            cnf.group( "KFileDialog Settings" ).writeEntry( "file module", "kfilemodule" );
         }
     }
 }
Index: nepomuk/services/strigi/eventmonitor.cpp
===================================================================
--- nepomuk/services/strigi/eventmonitor.cpp
+++ nepomuk/services/strigi/eventmonitor.cpp	2010-05-09 01:22:06.000000000 +0200
@@ -52,8 +52,8 @@
     m_periodicUpdateTimer.setInterval( 2*60*60*1000 );
 
     // monitor the powermanagement to not drain the battery
-    connect( Solid::PowerManagement::notifier(), SIGNAL( appShouldConserveResourcesChanged( bool ) ),
-             this, SLOT( slotPowerManagementStatusChanged( bool ) ) );
+    //connect( Solid::PowerManagement::notifier(), SIGNAL( appShouldConserveResourcesChanged( bool ) ),
+    //         this, SLOT( slotPowerManagementStatusChanged( bool ) ) );
 
     // setup the avail disk usage monitor
     connect( &m_availSpaceTimer, SIGNAL( timeout() ),
@@ -81,7 +81,7 @@
         m_periodicUpdateTimer.start();
     }
 
-    slotPowerManagementStatusChanged( Solid::PowerManagement::appShouldConserveResources() );
+//    slotPowerManagementStatusChanged( Solid::PowerManagement::appShouldConserveResources() );
 }
 
 
