diff -up kdebase-runtime-4.5.3/phonon/kcm/speakersetup.cpp.kde255736 kdebase-runtime-4.5.3/phonon/kcm/speakersetup.cpp
--- kdebase-runtime-4.5.3/phonon/kcm/speakersetup.cpp.kde255736	2010-11-11 22:46:07.998955398 -0600
+++ kdebase-runtime-4.5.3/phonon/kcm/speakersetup.cpp	2010-11-11 22:42:15.713947645 -0600
@@ -312,11 +312,16 @@ SpeakerSetup::SpeakerSetup(QWidget *pare
 
 SpeakerSetup::~SpeakerSetup()
 {
-    ca_context_destroy(m_Canberra);
-    pa_context_unref(s_context);
-    s_context = NULL;
-    pa_glib_mainloop_free(s_mainloop);
-    s_mainloop = NULL;
+    if (m_Canberra)
+        ca_context_destroy(m_Canberra);
+    if (s_context) {
+        pa_context_unref(s_context);
+        s_context = NULL;
+    } 
+    if (s_mainloop) {
+        pa_glib_mainloop_free(s_mainloop);
+        s_mainloop = NULL;
+    } 
 }
 
 void SpeakerSetup::load()
