diff -up libmikmod-3.3.2/libmikmod-config.in~ libmikmod-3.3.2/libmikmod-config.in
--- libmikmod-3.3.2/libmikmod-config.cmake~	2013-09-28 16:10:55.000000000 +0200
+++ libmikmod-3.3.2/libmikmod-config.cmake	2013-10-14 12:21:12.242028265 +0200
@@ -49,8 +49,7 @@ while test $# -gt 0 ; do
 		echo @LIB_LDADD@
 	;;
 	--libs)
-		libdir=-L"$prefix/lib"
-		echo $libdir -lmikmod @LIBRARY_LIB@
+		echo -lmikmod @LIBRARY_LIB@
 	;;
 	*)
 		echo "${usage}" 1>&2
