diff --git a/configure.ac b/configure.ac
index 51471b1..6b59976 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1227,11 +1227,6 @@ AC_ARG_ENABLE(fortify, [AC_HELP_STRING([--disable-fortify], [compile without FOR
 
 DEBUG_CFLAGS="$DEBUG_CFLAGS -DPURPLE_DISABLE_DEPRECATED -DPIDGIN_DISABLE_DEPRECATED -DFINCH_DISABLE_DEPRECATED -DGNT_DISABLE_DEPRECATED"
 if test "x$GCC" = "xyes"; then
-	dnl We enable -Wall later.
-	dnl If it's set after the warning CFLAGS in the compiler invocation, it counteracts the -Wno... flags.
-	dnl This leads to warnings we don't want.
-	CFLAGS=`echo $CFLAGS |$sedpath 's/-Wall//'`
-
 	dnl ENABLE WARNINGS SUPPORTED BY THE VERSION OF GCC IN USE
 	dnl
 	dnl Future Possibilities
