diff -up waf-1.6.8/waflib/extras/qnxnto.py.orig waf-1.6.8/waflib/extras/qnxnto.py
--- waf-1.6.8/waflib/extras/qnxnto.py.orig	2011-10-01 19:06:38.000000000 +0200
+++ waf-1.6.8/waflib/extras/qnxnto.py	2011-10-03 12:29:27.970114038 +0200
@@ -34,7 +34,7 @@ class Popen(object):
 			else:
 				if Popen.verbose:
 					print("Emulating")
-		except Exception, e:
+		except Exception as e:
 			if Popen.verbose:
 				print("Exception: %s" % e)
 			raise
