diff -up waf-1.8.11/waf-light.orig waf-1.8.11/waf-light
--- waf-1.8.11/waf-light.orig	2015-06-20 22:24:00.000000000 +0200
+++ waf-1.8.11/waf-light	2015-06-22 20:23:07.171667528 +0200
@@ -149,6 +149,9 @@ def find_lib():
 		w = test(i + '/lib/' + dirname)
 		if w: return w
 
+	w = test('/usr/share/%s' % WAF)
+	if w: return w
+
 	#waf-local
 	dir = join(base, (sys.platform != 'win32' and '.' or '') + dirname)
 	w = test(dir)
