Index: waflib/extras/review.py
===================================================================
--- waflib/extras/review.py	(revision 11401)
+++ waflib/extras/review.py	(revision 11402)
@@ -47,7 +47,7 @@
 
 	def __delitem__(self, key):
 		if not key in self._keys:
-			raise KeyError, key
+			raise KeyError(key)
 		dict.__delitem__(self, key)
 		self._keys.remove(key)
 
Index: waflib/extras/md5_tstamp.py
===================================================================
--- waflib/extras/md5_tstamp.py	(revision 11401)
+++ waflib/extras/md5_tstamp.py	(revision 11402)
@@ -33,7 +33,7 @@
 	self.restore_real()
 	try:
 		Build.hash_cache = self.hash_cache or {}
-	except Exception, e:
+	except Exception as e:
 		Build.hash_cache = {}
 Build.BuildContext.restore_real = Build.BuildContext.restore
 Build.BuildContext.restore      = restore
