diff -up waf-1.6.9/docs/sphinx/conf.py.orig waf-1.6.9/docs/sphinx/conf.py
--- waf-1.6.9/docs/sphinx/conf.py.orig	2011-11-20 15:51:43.000000000 +0100
+++ waf-1.6.9/docs/sphinx/conf.py	2011-11-26 14:27:33.996165116 +0100
@@ -415,7 +415,7 @@ html_theme = 'default'
 
 # The name of an image file (relative to this directory) to place at the top
 # of the sidebar.
-html_logo = '_images/waf-64x64.png'
+#html_logo = '_images/waf-64x64.png'
 
 # The name of an image file (within the static path) to use as favicon of the
 # docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
