diff --git a/waf-light b/waf-light
index 2068d44..c4ed3e6 100755
--- a/waf-light
+++ b/waf-light
@@ -127,7 +127,7 @@ def find_lib():
 		if w: return w
 		err("waf-light requires wafadmin -> export WAFDIR=/folder")
 
-	dir = "/lib/%s-%s-%s/" % (WAF, VERSION, REVISION)
+	dir = "/share/waf/"
 	for i in [INSTALL,'/usr','/usr/local','/opt']:
 		w = test(i+dir)
 		if w: return w
diff --git a/wscript b/wscript
index 8313ad3..02a7b6a 100644
--- a/wscript
+++ b/wscript
@@ -313,7 +313,7 @@ def build(bld):
 		if val != True and val != "y": sys.exit(1)
 		create_waf()
 
-	dir = os.path.join('lib', 'waf-%s-%s' % (VERSION, REVISION), 'wafadmin')
+	dir = os.path.join('share', 'waf', 'wafadmin')
 
 	wafadmin = bld(features = 'py')
 	wafadmin.find_sources_in_dirs('wafadmin', exts=['.py'])
