diff --git a/converter/other/Makefile b/converter/other/Makefile
index f51a780..9df16bc 100644
--- a/converter/other/Makefile
+++ b/converter/other/Makefile
@@ -121,7 +121,7 @@ PORTBINARIES =  avstopam bmptopnm fitstopnm \
 		pamtowinicon pamtoxvmini \
 		pbmtopgm pdbimgtopam pfmtopam \
 	        pgmtopbm pgmtoppm ppmtopgm pnmtoddif \
-		pnmtopclxl pnmtorast \
+                pnmtopclxl pnmtops pnmtorast \
 		pnmtosgi pnmtosir pamtotga pnmtoxwd \
 		rasttopnm rlatopam sgitopnm sirtopnm srftopam sunicontopnm \
 		winicontopam xwdtopnm yuy2topam zeisstopnm
