diff -up netpbm-10.58.01/converter/other/pnmtopclxl.c.pnmtopclxl netpbm-10.58.01/converter/other/pnmtopclxl.c
--- netpbm-10.58.01/converter/other/pnmtopclxl.c.pnmtopclxl	2012-04-09 15:31:42.000000000 +0200
+++ netpbm-10.58.01/converter/other/pnmtopclxl.c	2012-06-13 15:39:17.566141565 +0200
@@ -269,10 +269,10 @@ XY_RLEnew(size_t const size) {
 
     MALLOCVAR(rleP);
     if (rleP) {
-        rleP->fbuf = malloc(size);
+        rleP->fbufsize = MAX(1024, size);
+        rleP->fbuf = malloc(rleP->fbufsize);
 
         if (rleP->fbuf) {
-            rleP->fbufsize = MAX(1024, size);
             retval = rleP;
         } else
             retval = NULL;
