diff --git a/config.mk.in b/config.mk.in
index 5910146..bd1c70b 100644
--- a/config.mk.in
+++ b/config.mk.in
@@ -476,12 +476,12 @@ JBIGLIB = $(INTERNAL_JBIGLIB)
 JBIGHDR_DIR = $(INTERNAL_JBIGHDR_DIR)
 
 # The Jasper JPEG-2000 image compression library (aka JasPer):
-JASPERLIB = $(INTERNAL_JASPERLIB)
-JASPERHDR_DIR = $(INTERNAL_JASPERHDR_DIR)
+JASPERLIB = ""
+JASPERHDR_DIR = "/usr/include/jasper"
 # JASPERDEPLIBS is the libraries (-l options or file names) on which
 # The Jasper library depends -- i.e. what you have to link into any
 # executable that links in the Jasper library.
-JASPERDEPLIBS =
+JASPERDEPLIBS = -ljasper
 #JASPERDEPLIBS = -ljpeg
 
 # And the Utah Raster Toolkit (aka URT aka RLE) library:
