diff -up netpbm-10.47.12/buildtools/manpage.mk.noppmtompeg netpbm-10.47.12/buildtools/manpage.mk
--- netpbm-10.47.12/buildtools/manpage.mk.noppmtompeg	2010-04-27 09:54:52.121522491 -0400
+++ netpbm-10.47.12/buildtools/manpage.mk	2010-04-27 09:55:10.337642487 -0400
@@ -251,7 +251,6 @@ MAN1 = \
 	ppmtolj.1 \
 	ppmtomap.1 \
 	ppmtomitsu.1 \
-	ppmtompeg.1 \
 	ppmtoneo.1 \
 	ppmtopcx.1 \
 	ppmtopgm.1 \
diff -up netpbm-10.47.12/converter/ppm/Makefile.noppmtompeg netpbm-10.47.12/converter/ppm/Makefile
--- netpbm-10.47.12/converter/ppm/Makefile.noppmtompeg	2010-04-27 09:55:20.293517254 -0400
+++ netpbm-10.47.12/converter/ppm/Makefile	2010-04-27 09:55:41.534642567 -0400
@@ -7,7 +7,7 @@ VPATH=.:$(SRCDIR)/$(SUBDIR)
 
 include $(BUILDDIR)/config.mk
 
-SUBDIRS = hpcdtoppm ppmtompeg
+SUBDIRS = hpcdtoppm
 
 PORTBINARIES =	411toppm eyuvtoppm gouldtoppm ilbmtoppm imgtoppm \
 		leaftoppm mtvtoppm neotoppm \
