diff --git a/converter/other/xwdtopnm.c b/converter/other/xwdtopnm.c
index d49a2b0..45d66b4 100644
--- a/converter/other/xwdtopnm.c
+++ b/converter/other/xwdtopnm.c
@@ -1025,7 +1025,7 @@ pixelReader_getbits(pixelReader * const rdrP,
             unsigned long const bitsToTakeMask = lsbmask[nBitsToTake];
                 /* E.g. if nbitsToTake is 4, this is 0x0000000F */
 
-            unsigned long bitsToTake;
+            unsigned int bitsToTake;
                 /* The actual bits we take, in the 'nBitsToTake' low bits */
 
             assert(nBitsToTake <= 32);
