diff --git a/lib/pm.h b/lib/pm.h
index 47cbfe8..e9f1405 100644
--- a/lib/pm.h
+++ b/lib/pm.h
@@ -18,6 +18,7 @@
 #include <sys/types.h>
 #include <ctype.h>
 #include <stdio.h>
+#include <string.h>
 #include <errno.h>
 #include <setjmp.h>
 #include <sys/stat.h>
diff --git a/urt/rle.h b/urt/rle.h
index 0766d22..8d72cb9 100644
--- a/urt/rle.h
+++ b/urt/rle.h
@@ -31,6 +31,7 @@
 #define RLE_H
 
 #include <stdio.h>      /* Declare FILE. */
+#include <string.h>
 
 enum rle_dispatch {
     NO_DISPATCH = -1,
