diff --git a/converter/ppm/ppmtopict.c b/converter/ppm/ppmtopict.c
index 36464b6..034e705 100644
--- a/converter/ppm/ppmtopict.c
+++ b/converter/ppm/ppmtopict.c
@@ -208,7 +208,7 @@ putRow(FILE *         const ofP,
 
    Use buffer 'outBuf'.
 -----------------------------------------------------------------------------*/
-    unsigned int i;
+    int i;
     unsigned int count;
     unsigned int run;
     unsigned int rep;
