diff --git a/converter/pbm/cmuwmtopbm.c b/converter/pbm/cmuwmtopbm.c
index ccf8cfc..1f9bc6c 100644
--- a/converter/pbm/cmuwmtopbm.c
+++ b/converter/pbm/cmuwmtopbm.c
@@ -18,7 +18,7 @@
    This program does not check the pad bits at the end of each row.
 */
 
-
+#include <stdint.h>
 #include "pbm.h"
 
 /*--------------------------
@@ -42,7 +42,7 @@ readCmuwmHeader(FILE *         const ifP,
         "CMU window manager header EOF / read error";
     uint32_t const cmuwmMagic = 0xf10040bb;
 
-    long l;
+    uint32_t l;
     short s;
     int rc;
 
