diff -up netpbm-10.56.03/converter/other/pstopnm.c.CAN-2005-2471 netpbm-10.56.03/converter/other/pstopnm.c
--- netpbm-10.56.03/converter/other/pstopnm.c.CAN-2005-2471	2011-11-09 09:18:00.000000000 +0100
+++ netpbm-10.56.03/converter/other/pstopnm.c	2011-11-11 11:43:51.827409226 +0100
@@ -747,11 +747,11 @@ execGhostscript(int          const input
                    ghostscriptProg, arg0,
                    deviceopt, outfileopt, gopt, ropt, textalphabitsopt,
                    "-q", "-dNOPAUSE", 
-                   "-dSAFER", "-");
+                   "-dPARANOIDSAFER", "-");
     }
 
     execl(ghostscriptProg, arg0, deviceopt, outfileopt, gopt, ropt, "-q",
-          "-dNOPAUSE", "-dSAFER", "-", NULL);
+          "-dNOPAUSE", "-dPARANOIDSAFER", "-", NULL);
     
     pm_error("execl() of Ghostscript ('%s') failed, errno=%d (%s)",
              ghostscriptProg, errno, strerror(errno));
