diff --git a/GNUmakefile b/GNUmakefile
index caf8e37..4418076 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -394,7 +394,7 @@ install.lib:
 endif
 
 .PHONY: install.manwebmain
-install.manwebmain: $(PKGDIR)/$(PKGMANDIR)/web/netpbm.url $(PKGDIR)/bin/doc.url
+install.manwebmain: $(PKGDIR)/$(PKGMANDIR)/web/netpbm.url
 
 $(PKGDIR)/$(PKGMANDIR)/web/netpbm.url: $(PKGDIR)/$(PKGMANDIR)/web
 	echo "$(NETPBM_DOCURL)" > $@
