diff --git a/converter/other/bmptopnm.c b/converter/other/bmptopnm.c
index a069092..20a0331 100644
--- a/converter/other/bmptopnm.c
+++ b/converter/other/bmptopnm.c
@@ -1328,7 +1328,7 @@ readBmp(FILE *               const ifP,
         xel **               const colormapP,
         bool                 const verbose) {
 
-    xel * colormap;  /* malloc'ed */
+    xel * colormap = NULL;  /* malloc'ed */
     unsigned int pos;
         /* Current byte position in the BMP file */
 
