diff -up netpbm-10.58.01/converter/ppm/ppmtopict.c.ppmtopict netpbm-10.58.01/converter/ppm/ppmtopict.c
--- netpbm-10.58.01/converter/ppm/ppmtopict.c.ppmtopict	2012-06-13 09:52:42.000000000 +0200
+++ netpbm-10.58.01/converter/ppm/ppmtopict.c	2012-06-13 11:33:11.134747363 +0200
@@ -200,7 +200,7 @@ putRow(FILE * const ifP,
        pixel *      const rowpixels,
        char *       const packed) {
 
-    unsigned int i;
+    int i;
     unsigned int count;
     unsigned int run;
     unsigned int rep;
