diff -up libproxy-0.4.4/libproxy/libproxy-1.0.pc.in.fix libproxy-0.4.4/libproxy/libproxy-1.0.pc.in
--- libproxy-0.4.4/libproxy/libproxy-1.0.pc.in.fix	2010-07-13 09:58:27.000000000 +0200
+++ libproxy-0.4.4/libproxy/libproxy-1.0.pc.in	2010-07-13 10:00:26.000000000 +0200
@@ -1,6 +1,6 @@
-prefix=@prefix@
-libdir=@libdir@
-includedir=@includedir@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_LIBDIR@
+includedir=@CMAKE_INSTALL_PREFIX@/include
 
 Name: libproxy-1.0
 Description: Proxy Configuration Library
