Index: bindings/perl/src/CMakeLists.txt
===================================================================
--- bindings/perl/src/CMakeLists.txt	(revisão 872)
+++ bindings/perl/src/CMakeLists.txt	(revisão 873)
@@ -12,7 +12,7 @@
 set(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/perl/blib/arch/auto/Net)
 add_library(PLlibproxy SHARED ${Libproxy_LIB_SRCS})
 
-target_link_libraries(PLlibproxy ${PERL_LIBRARY} libproxy)
+target_link_libraries(PLlibproxy ${PERL_LIBRARY} libproxy pthread)
 set_target_properties(PLlibproxy PROPERTIES OUTPUT_NAME "Libproxy")
 set_target_properties(PLlibproxy PROPERTIES PREFIX "")
 
