diff -up libproxy-0.4.0/bindings/CMakeLists.txt.pyarch libproxy-0.4.0/bindings/CMakeLists.txt
--- libproxy-0.4.0/bindings/CMakeLists.txt.pyarch	2010-02-25 21:29:54.000000000 +0100
+++ libproxy-0.4.0/bindings/CMakeLists.txt	2010-03-14 21:03:16.059831098 +0100
@@ -3,7 +3,7 @@ find_package(PythonInterp)
 if(PYTHONINTERP_FOUND)
   execute_process(COMMAND
 		    ${PYTHON_EXECUTABLE}
-		    -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib(plat_specific=0)"
+		    -c "import distutils.sysconfig ; print distutils.sysconfig.get_python_lib()"
 		    OUTPUT_VARIABLE pysitepkgdir
 		    OUTPUT_STRIP_TRAILING_WHITESPACE)
   install(FILES libproxy.py DESTINATION ${pysitepkgdir})
