From a6ddd108cf8e70c4037065143f968e994f5ef581 Mon Sep 17 00:00:00 2001
From: "Dwayne C. Litzenberger" <dlitz@dlitz.net>
Date: Sat, 18 Feb 2012 12:42:23 -0500
Subject: [PATCH 2/3] Fix typo

---
 src/_fastmath.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/_fastmath.c b/src/_fastmath.c
index c0f8c4e..8c1a517 100644
--- a/src/_fastmath.c
+++ b/src/_fastmath.c
@@ -1139,7 +1139,7 @@ getRNG (void)
 	if (!PyCallable_Check (new_func))
 	{
 		PyErr_SetString (PyExc_RuntimeError,
-						 "Cryptor.Random.new is not callable.");
+						 "Crypto.Random.new is not callable.");
 		return NULL;
 	}
 	rng = PyObject_CallObject (new_func, NULL);
-- 
1.7.7.6

