Index: oprofile/libop/op_cpu_type.c
diff -u oprofile/libop/op_cpu_type.c:1.54 oprofile/libop/op_cpu_type.c:1.55
--- oprofile/libop/op_cpu_type.c:1.54	Wed Nov 25 20:35:19 2009
+++ oprofile/libop/op_cpu_type.c	Mon Jun 21 23:12:11 2010
@@ -83,5 +83,6 @@
    	{ "Intel Core/i7", "i386/core_i7", CPU_CORE_I7, 4 },
    	{ "Intel Atom", "i386/atom", CPU_ATOM, 2 },
+	{ "Intel Nehalem microarchitecture", "i386/nehalem", CPU_NEHALEM, 4 },
 };
  
 static size_t const nr_cpu_descrs = sizeof(cpu_descrs) / sizeof(struct cpu_descr);
Index: oprofile/libop/op_cpu_type.h
diff -u oprofile/libop/op_cpu_type.h:1.47 oprofile/libop/op_cpu_type.h:1.48
--- oprofile/libop/op_cpu_type.h:1.47	Wed Nov 25 20:35:19 2009
+++ oprofile/libop/op_cpu_type.h	Mon Jun 21 23:12:11 2010
@@ -80,5 +80,6 @@
    	CPU_CORE_I7, /* Intel Core i7, Nehalem */
    	CPU_ATOM, /* First generation Intel Atom */
+	CPU_NEHALEM, /* Intel Nehalem microarchitecture */
 	MAX_CPU_TYPE
 } op_cpu;
 
Index: oprofile/libop/op_events.c
diff -u oprofile/libop/op_events.c:1.98 oprofile/libop/op_events.c:1.99
--- oprofile/libop/op_events.c:1.98	Wed Nov 25 20:35:19 2009
+++ oprofile/libop/op_events.c	Mon Jun 21 23:12:11 2010
@@ -970,6 +970,7 @@
 		case CPU_FAMILY11H:
  		case CPU_ATOM:
  		case CPU_CORE_I7:
+		case CPU_NEHALEM:
 			descr->name = "CPU_CLK_UNHALTED";
 			break;
 
Index: oprofile/utils/ophelp.c
diff -u oprofile/utils/ophelp.c:1.35 oprofile/utils/ophelp.c:1.36
--- oprofile/utils/ophelp.c:1.35	Wed Nov 25 20:35:20 2009
+++ oprofile/utils/ophelp.c	Mon Jun 21 23:12:11 2010
@@ -483,6 +483,7 @@
 	case CPU_CORE:
 	case CPU_CORE_2:
 	case CPU_CORE_I7:
+	case CPU_NEHALEM:
 	case CPU_ATOM:
 		event_doc =
 			"See Intel Architecture Developer's Manual Volume 3B, Appendix A and\n"
