diff -rupN hydrogen-0.9.5.old/data/hydrogen.default.conf hydrogen-0.9.5/data/hydrogen.default.conf
--- hydrogen-0.9.5.old/data/hydrogen.default.conf	2010-01-13 17:20:49.000000000 -0500
+++ hydrogen-0.9.5/data/hydrogen.default.conf	2011-03-26 08:36:07.000000000 -0400
@@ -14,7 +14,7 @@
         <useLash>false</useLash>
 	<useTimeLine>false</useTimeLine>
 	<maxBars>400</maxBars>
-	<showDevelWarning>true</showDevelWarning>
+	<showDevelWarning>false</showDevelWarning>
 	<hearNewNotes>true</hearNewNotes>
 	<recordEvents>false</recordEvents>
 	<quantizeEvents>true</quantizeEvents>
