diff -rupN 0.9.4.1.old/libs/hydrogen/src/preferences.cpp 0.9.4.1/libs/hydrogen/src/preferences.cpp
--- 0.9.4.1.old/libs/hydrogen/src/preferences.cpp	2009-08-26 08:12:27.000000000 -0400
+++ 0.9.4.1/libs/hydrogen/src/preferences.cpp	2010-04-10 07:14:14.000000000 -0400
@@ -99,6 +99,8 @@ Preferences::Preferences()
 #else
 		m_ladspaPathVect.push_back( "/usr/lib/ladspa" );
 		m_ladspaPathVect.push_back( "/usr/local/lib/ladspa" );
+		m_ladspaPathVect.push_back( "/usr/lib64/ladspa" );
+		m_ladspaPathVect.push_back( "/usr/local/lib64/ladspa" );
 #endif
 
 	}
