diff -up 0.9.4.1/Sconstruct.tinyxml 0.9.4.1/Sconstruct
--- 0.9.4.1/Sconstruct.tinyxml	2009-09-07 18:31:13.000000000 -0300
+++ 0.9.4.1/Sconstruct	2010-07-04 18:13:52.998678600 -0300
@@ -185,6 +185,8 @@ def get_hydrogen_lib( opts ):
         if str(env['lash']) == "1":
             env.ParseConfig('pkg-config --cflags --libs lash-1.0')
 
+        env.ParseConfig('pkg-config --cflags --libs tinyxml')
+
 
 	#env.Decider is not known in older scons version
 	try:
@@ -269,6 +271,8 @@ def get_hydrogen_gui( lib_hydrogen , opt
 	if str(env['lash']) == "1":
 	        env.ParseConfig('pkg-config --cflags --libs lash-1.0')
 
+	env.ParseConfig('pkg-config --cflags --libs tinyxml')
+
 	if sys.platform == "darwin":
 		env.Append( LIBS = ["z"] )
 
diff -up 0.9.4.1/libs/hydrogen/hydrogen.pro.tinyxml 0.9.4.1/libs/hydrogen/hydrogen.pro
--- 0.9.4.1/libs/hydrogen/hydrogen.pro.tinyxml	2009-08-07 11:14:20.000000000 -0300
+++ 0.9.4.1/libs/hydrogen/hydrogen.pro	2010-07-04 18:11:25.761803172 -0300
@@ -75,8 +75,6 @@ HEADERS += src/flac_file.h \
            src/IO/PortMidiDriver.h \
            src/table/getTableValue.h \
            src/table/TableExponential.h \
-           src/xml/tinystr.h \
-           src/xml/tinyxml.h \
            include/hydrogen/fx/Effects.h \
            include/hydrogen/fx/ladspa.h \
            include/hydrogen/fx/LadspaFX.h \
@@ -129,8 +127,4 @@ SOURCES += src/action.cpp \
            src/sampler/sampler.cpp \
            src/smf/smf.cpp \
            src/smf/smf_event.cpp \
-           src/synth/synth.cpp \
-           src/xml/tinystr.cpp \
-           src/xml/tinyxml.cpp \
-           src/xml/tinyxmlerror.cpp \
-           src/xml/tinyxmlparser.cpp 
+           src/synth/synth.cpp
diff -up 0.9.4.1/libs/hydrogen/src/local_file_mgr.cpp.tinyxml 0.9.4.1/libs/hydrogen/src/local_file_mgr.cpp
--- 0.9.4.1/libs/hydrogen/src/local_file_mgr.cpp.tinyxml	2009-09-06 08:00:03.000000000 -0300
+++ 0.9.4.1/libs/hydrogen/src/local_file_mgr.cpp	2010-07-04 18:11:25.761803172 -0300
@@ -48,7 +48,7 @@
 #include <QVector>
 #include <QLocale>
 
-#include "xml/tinyxml.h"
+#include <tinyxml.h>
 
 #include <algorithm>
 //#include <cstdio>
diff -up 0.9.4.1/libs/hydrogen/src/preferences.cpp.tinyxml 0.9.4.1/libs/hydrogen/src/preferences.cpp
--- 0.9.4.1/libs/hydrogen/src/preferences.cpp.tinyxml	2009-08-26 09:12:27.000000000 -0300
+++ 0.9.4.1/libs/hydrogen/src/preferences.cpp	2010-07-04 18:11:25.761803172 -0300
@@ -42,7 +42,7 @@
 #include "config.h"
 #include "version.h"
 
-#include "xml/tinyxml.h"
+#include <tinyxml.h>
 #include <QDir>
 #include <QApplication>
 
diff -up 0.9.4.1/libs/hydrogen/src/song.cpp.tinyxml 0.9.4.1/libs/hydrogen/src/song.cpp
--- 0.9.4.1/libs/hydrogen/src/song.cpp.tinyxml	2009-07-14 22:10:17.000000000 -0300
+++ 0.9.4.1/libs/hydrogen/src/song.cpp	2010-07-04 18:11:25.761803172 -0300
@@ -25,7 +25,7 @@
 
 #include <cassert>
 
-#include "xml/tinyxml.h"
+#include <tinyxml.h>
 
 #include <hydrogen/adsr.h>
 #include <hydrogen/data_path.h>
