--- hydrogen-0.9.6.1/data/hydrogen.default.conf.orig	2015-01-19 21:49:15.442765573 -0200
+++ hydrogen-0.9.6.1/data/hydrogen.default.conf	2015-01-19 21:49:56.283142339 -0200
@@ -15,7 +15,7 @@
 	<useTimeLine>false</useTimeLine>
         <maxBars>400</maxBars>
         <useTheRubberbandBpmChangeEvent>false</useTheRubberbandBpmChangeEvent>
-	<showDevelWarning>true</showDevelWarning>
+	<showDevelWarning>false</showDevelWarning>
 	<hearNewNotes>true</hearNewNotes>
 	<recordEvents>false</recordEvents>
 	<quantizeEvents>true</quantizeEvents>
