--- src/core/src/preferences.cpp.orig	2015-01-19 22:16:00.593423246 -0200
+++ src/core/src/preferences.cpp	2015-01-19 22:16:48.855262594 -0200
@@ -121,10 +121,8 @@ Preferences::Preferences()
 		m_ladspaPathVect.push_back( "/Library/Audio/Plug-Ins/LADSPA/" );
 		m_ladspaPathVect.push_back( QDir::homePath().append( "/Library/Audio/Plug-Ins/LADSPA" ));
 #else
-		m_ladspaPathVect.push_back( "/usr/lib/ladspa" );
-		m_ladspaPathVect.push_back( "/usr/local/lib/ladspa" );
-		m_ladspaPathVect.push_back( "/usr/lib64/ladspa" );
-		m_ladspaPathVect.push_back( "/usr/local/lib64/ladspa" );
+		m_ladspaPathVect.push_back( "/usr/__SB_LIBDIR__/ladspa" );
+		m_ladspaPathVect.push_back( "/usr/local/__SB_LIBDIR__/ladspa" );
 #endif
 
 	}
