diff -rupN 0.9.4.1.old/plugins/wasp/wasp_booster/wasp_booster.pro 0.9.4.1/plugins/wasp/wasp_booster/wasp_booster.pro
--- 0.9.4.1.old/plugins/wasp/wasp_booster/wasp_booster.pro	2007-06-18 14:30:19.000000000 -0400
+++ 0.9.4.1/plugins/wasp/wasp_booster/wasp_booster.pro	2010-04-10 06:43:30.000000000 -0400
@@ -5,14 +5,15 @@ QT -= core
 DESTDIR = ../../
 INCLUDEPATH += ../include
 SOURCES += booster.c
+QMAKE_LFLAGS_PLUGIN += -nostartfiles
 
 linux-g++* {
 	CONFIG = plugin
 
 	# there's something weird in linux qmake...this is a simple and ugly workaround
-	QMAKE_LINK_SHLIB_CMD = ld -shared -o libwasp_booster.so booster.o
+        #QMAKE_LINK_SHLIB_CMD = gcc -nostartfiles -lm -shared booster.o -o libwasp_booster.so
 }
 
 win32 {
 	CONFIG = dll
-}
\ No newline at end of file
+}
diff -rupN 0.9.4.1.old/plugins/wasp/wasp_noisifier/wasp_noisifier.pro 0.9.4.1/plugins/wasp/wasp_noisifier/wasp_noisifier.pro
--- 0.9.4.1.old/plugins/wasp/wasp_noisifier/wasp_noisifier.pro	2007-06-18 14:30:19.000000000 -0400
+++ 0.9.4.1/plugins/wasp/wasp_noisifier/wasp_noisifier.pro	2010-04-10 06:44:36.000000000 -0400
@@ -5,14 +5,15 @@ QT -= core
 DESTDIR = ../../
 INCLUDEPATH += ../include
 SOURCES += noisifier.c
+QMAKE_LFLAGS_PLUGIN += -nostartfiles
 
 linux-g++* {
 	CONFIG = plugin
 	
 	# there's something weird in linux qmake...this is a simple and ugly workaround
-	QMAKE_LINK_SHLIB_CMD = ld -shared -o libwasp_noisifier.so noisifier.o
+        #QMAKE_LINK_SHLIB_CMD = gcc -lm -shared -o wasp_noisifier.so noisifier.o
 }
 
 win32 {
 	CONFIG = dll
-}
\ No newline at end of file
+}
diff -rupN 0.9.4.1.old/plugins/wasp/wasp_xshaper/wasp_xshaper.pro 0.9.4.1/plugins/wasp/wasp_xshaper/wasp_xshaper.pro
--- 0.9.4.1.old/plugins/wasp/wasp_xshaper/wasp_xshaper.pro	2007-06-18 14:30:19.000000000 -0400
+++ 0.9.4.1/plugins/wasp/wasp_xshaper/wasp_xshaper.pro	2010-04-10 06:44:55.000000000 -0400
@@ -5,13 +5,14 @@ QT -= core
 DESTDIR = ../../
 INCLUDEPATH += ../include
 SOURCES += xshaper.c
+QMAKE_LFLAGS_PLUGIN += -nostartfiles
 
 linux-g++* {
 	CONFIG = plugin
 	# there's something weird in linux qmake...this is a simple and ugly workaround
-	QMAKE_LINK_SHLIB_CMD = ld -shared -o libwasp_xshaper.so xshaper.o
+        #QMAKE_LINK_SHLIB_CMD = gcc -lm -shared -o wasp_xshaper.so xshaper.o
 }
 
 win32 {
 	CONFIG = dll
-}
\ No newline at end of file
+}
