diff -up crda-1.1.3_2013.01.11/crda-1.1.3/udev/regulatory.rules.setregdomain crda-1.1.3_2013.01.11/crda-1.1.3/udev/regulatory.rules
--- crda-1.1.3_2013.01.11/crda-1.1.3/udev/regulatory.rules.setregdomain	2012-01-18 21:19:09.000000000 -0500
+++ crda-1.1.3_2013.01.11/crda-1.1.3/udev/regulatory.rules	2013-01-25 14:10:57.818931320 -0500
@@ -2,4 +2,6 @@
 # For more information see:
 # http://wireless.kernel.org/en/developers/Regulatory/CRDA
 
+SUBSYSTEM=="ieee80211", ACTION=="add", RUN+="/sbin/setregdomain"
+
 KERNEL=="regulatory*", ACTION=="change", SUBSYSTEM=="platform", RUN+="$(SBINDIR)crda"
