From 3bd8afbff29e50b38f889b2f688785a669b9aafc Mon Sep 17 00:00:00 2001
From: Cory Benfield <lukasaoz@gmail.com>
Date: Sat, 14 Mar 2015 08:49:55 +0000
Subject: [PATCH] Don't ascribe cookies to the target domain.

---
 requests/sessions.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requests/sessions.py b/requests/sessions.py
index 4f30696..9d5498c 100644
--- a/requests/sessions.py
+++ b/requests/sessions.py
@@ -171,7 +171,7 @@ def resolve_redirects(self, resp, req, stream=False, timeout=None,
             except KeyError:
                 pass
 
-            extract_cookies_to_jar(prepared_request._cookies, prepared_request, resp.raw)
+            extract_cookies_to_jar(prepared_request._cookies, req, resp.raw)
             prepared_request._cookies.update(self.cookies)
             prepared_request.prepare_cookies(prepared_request._cookies)
 
