diff -up glew-1.11.0/Makefile.jx glew-1.11.0/Makefile
--- glew-1.11.0/Makefile.jx	2014-08-21 18:46:29.948417527 -0300
+++ glew-1.11.0/Makefile	2014-08-21 18:51:24.252884697 -0300
@@ -42,9 +42,9 @@ endif
 
 GLEW_PREFIX ?= /usr
 GLEW_DEST ?= /usr
-BINDIR    ?= $(GLEW_DEST)/bin
-LIBDIR    ?= $(GLEW_DEST)/lib
-INCDIR    ?= $(GLEW_DEST)/include/GL
+BINDIR    ?= $(GLEW_DEST)$(bindir)
+LIBDIR    ?= $(GLEW_DEST)$(libdir)
+INCDIR    ?= $(GLEW_DEST)$(includedir)/GL
 
 ifneq ($(GLEW_NO_GLU), -DGLEW_NO_GLU)
 LIBGLU = glu
@@ -66,7 +66,7 @@ DIST_DIR := $(shell mktemp -d /tmp/glew.
 
 AR      ?= ar
 INSTALL ?= install
-STRIP   ?= strip
+STRIP   ?= true
 RM      ?= rm -f
 LN      ?= ln -sf
 
@@ -133,10 +133,10 @@ tmp/$(SYSTEM)/default/shared/glew.o: src
 
 glew.pc: glew.pc.in
 	sed \
-		-e "s|@prefix@|$(GLEW_PREFIX)|g" \
-		-e "s|@libdir@|$(LIBDIR)|g" \
-		-e "s|@exec_prefix@|$(BINDIR)|g" \
-		-e "s|@includedir@|$(INCDIR)|g" \
+		-e "s|@prefix@|/usr|g" \
+		-e "s|@libdir@|$(libdir)|g" \
+		-e "s|@exec_prefix@|$(bindir)|g" \
+		-e "s|@includedir@|$(includedir)|g" \
 		-e "s|@version@|$(GLEW_VERSION)|g" \
 		-e "s|@cflags@||g" \
 		-e "s|@libname@|$(NAME)|g" \
@@ -244,7 +244,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -d -m 0755 "$(DESTDIR)$(BINDIR)"
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED) "$(DESTDIR)$(BINDIR)/"
 else
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED) "$(DESTDIR)$(LIBDIR)/"
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED) "$(DESTDIR)$(LIBDIR)/"
 endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED) "$(DESTDIR)$(LIBDIR)/$(LIB.SONAME)"
@@ -266,7 +266,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -d -m 0755 "$(DESTDIR)$(BINDIR)"
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) "$(DESTDIR)$(BINDIR)/"
 else
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED.MX) "$(DESTDIR)$(LIBDIR)/"
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) "$(DESTDIR)$(LIBDIR)/"
 endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED.MX) "$(DESTDIR)$(LIBDIR)/$(LIB.SONAME.MX)"
@@ -278,7 +278,7 @@ endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED.MX) "$(DESTDIR)$(LIBDIR)/$(LIB.DEVLNK.MX)"
 endif
-	$(INSTALL) -m 0644 lib/$(LIB.STATIC.MX) "$(DESTDIR)$(LIBDIR)/"
+	$(INSTALL) -m 0755 lib/$(LIB.STATIC.MX) "$(DESTDIR)$(LIBDIR)/"
 
 install.bin: glew.bin
 	$(INSTALL) -d -m 0755 "$(DESTDIR)$(BINDIR)"
diff -up glew-1.11.0/config/Makefile.linux.jx glew-1.11.0/config/Makefile.linux
--- glew-1.11.0/config/Makefile.linux.jx	2014-08-21 18:46:30.080389486 -0300
+++ glew-1.11.0/config/Makefile.linux	2014-08-21 18:54:27.741650391 -0300
@@ -9,14 +9,8 @@ endif
 ifeq (ppc64,${M_ARCH})
   ARCH64 = true
 endif
-ifeq (${ARCH64},true)
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
-  LIBDIR = $(GLEW_DEST)/lib64
-else
-  LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
-  LIBDIR = $(GLEW_DEST)/lib
-endif
-LDFLAGS.GL = -lGL -lX11
+LIBDIR = $(GLEW_DEST)$(libdir)
+LDFLAGS.GL = -lGL -lX11 -lGLU
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
