diff -up glew-1.9.0/Makefile.jx glew-1.9.0/Makefile
--- glew-1.9.0/Makefile.jx	2012-12-02 12:08:32.902710195 -0200
+++ glew-1.9.0/Makefile	2012-12-02 12:08:33.027683812 -0200
@@ -41,9 +41,9 @@ $(error "Platform '$(SYSTEM)' not suppor
 endif
 
 GLEW_DEST ?= /usr
-BINDIR    ?= $(GLEW_DEST)/bin
-LIBDIR    ?= $(GLEW_DEST)/lib
-INCDIR    ?= $(GLEW_DEST)/include/GL
+BINDIR    ?= $(GLEW_DEST)$(bindir)
+LIBDIR    ?= $(GLEW_DEST)$(libdir)
+INCDIR    ?= $(GLEW_DEST)$(includedir)/GL
 
 TARDIR = ../glew-$(GLEW_VERSION)
 TARBALL = ../glew-$(GLEW_VERSION).tar.gz
@@ -62,7 +62,7 @@ DIST_SRC_TGZ = glew-$(GLEW_VERSION).tgz
 
 AR      ?= ar
 INSTALL ?= install
-STRIP   ?= strip
+STRIP   ?= true
 RM      ?= rm -f
 LN      ?= ln -sf
 
@@ -224,7 +224,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -d -m 0755 $(BINDIR)
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(BINDIR)/
 else
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(LIBDIR)/
 endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
@@ -246,7 +246,7 @@ ifeq ($(filter-out mingw% cygwin,$(SYSTE
 	$(INSTALL) -d -m 0755 $(BINDIR)
 	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) $(BINDIR)/
 else
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED.MX) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED.MX) $(LIBDIR)/
 endif
 ifneq ($(LN),)
 	$(LN) $(LIB.SHARED.MX) $(LIBDIR)/$(LIB.SONAME.MX)
@@ -262,7 +262,7 @@ endif
 
 install.bin: glew.bin
 	$(INSTALL) -d -m 0755 $(BINDIR)
-	$(INSTALL) -s -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
+	$(INSTALL) -m 0755 bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN) $(BINDIR)/
 
 install.include:
 	$(INSTALL) -d -m 0755 $(INCDIR)
diff -up glew-1.9.0/config/Makefile.linux.jx glew-1.9.0/config/Makefile.linux
--- glew-1.9.0/config/Makefile.linux.jx	2012-12-02 12:08:33.028683603 -0200
+++ glew-1.9.0/config/Makefile.linux	2012-12-02 12:12:30.729851863 -0200
@@ -3,14 +3,8 @@ CC = cc
 LD = cc
 PICFLAG = -fPIC
 M_ARCH ?= $(shell uname -m)
-ifeq (x86_64,${M_ARCH})
-LDFLAGS.EXTRA = -L/usr/X11R6/lib64 -L/usr/lib64
-LIBDIR = $(GLEW_DEST)/lib64
-else
-LDFLAGS.EXTRA = -L/usr/X11R6/lib -L/usr/lib
-LIBDIR = $(GLEW_DEST)/lib
-endif
-LDFLAGS.GL = -lXmu -lXi -lGL -lXext -lX11
+LIBDIR = $(GLEW_DEST)$(libdir)
+LDFLAGS.GL = -lGL -lX11 -lGLU
 LDFLAGS.STATIC = -Wl,-Bstatic
 LDFLAGS.DYNAMIC = -Wl,-Bdynamic
 NAME = GLEW
