--- src/keyboard_gui.py.orig	2009-07-26 21:31:30.000000000 -0300
+++ src/keyboard_gui.py	2012-03-02 23:34:01.122540056 -0300
@@ -35,10 +35,10 @@ import gettext
 
 import system_config_keyboard.keyboard as keyboard
 
-from firstboot.config import *
-from firstboot.constants import *
-from firstboot.functions import *
-from firstboot.module import *
+from firstboot_config import *
+from firstboot_constants import *
+from firstboot_functions import *
+from firstboot_module import *
 
 sys.path.append('/usr/share/system-config-keyboard')
 import keyboard_backend
