diff -up firstboot/firstboot/config.py.orig firstboot/firstboot/config.py
--- firstboot/firstboot/config.py.orig	2011-07-26 05:19:19.000000000 -0300
+++ firstboot/firstboot/config.py	2012-03-02 23:50:34.687871002 -0300
@@ -18,7 +18,7 @@
 # subject to the GNU General Public License and may only be used or replicated
 # with the express permission of Red Hat, Inc. 
 #
-from constants import *
+from firstboot_constants import *
 
 class Config:
     def __init__(self):
diff -up firstboot/firstboot/module.py.orig firstboot/firstboot/module.py
--- firstboot/firstboot/module.py.orig	2011-07-26 05:19:19.000000000 -0300
+++ firstboot/firstboot/module.py	2012-03-02 23:49:37.682956703 -0300
@@ -17,8 +17,8 @@
 # subject to the GNU General Public License and may only be used or replicated
 # with the express permission of Red Hat, Inc. 
 #
-from firstboot.config import *
-from firstboot.constants import *
+from firstboot_config import *
+from firstboot_constants import *
 import logging
 
 import gettext
@@ -161,7 +161,7 @@ class Module:
             raise SystemError, "Module %s has not initializes its UI" % self.title
 
         import gtk
-        from firstboot.functions import loadToImage
+        from firstboot_functions import loadToImage
 
         # Create the large label that goes at the top of the right side.
         label = gtk.Label("")
