diff -r 018b43db7c47 -r 211cabb6719b CHANGELOG
--- a/CHANGELOG	Fri Mar 18 23:36:53 2011 -0500
+++ b/CHANGELOG	Fri Mar 18 23:38:55 2011 -0500
@@ -1,4 +1,8 @@
-1.0
+1.0.1
+
+- Fixed problems with SkipTest in Python 3.2 (#389)
+
+1.0.0
 
 - Made nose compatible with python 3. **Huge** thanks to Alex "foogod"
   Stewart!
diff -r 018b43db7c47 -r 211cabb6719b nose/proxy.py
--- a/nose/proxy.py	Fri Mar 18 23:36:53 2011 -0500
+++ b/nose/proxy.py	Fri Mar 18 23:38:55 2011 -0500
@@ -148,6 +148,9 @@
         from nose.plugins.skip import SkipTest
         self.assertMyTest(test)
         plugins = self.plugins
+        if not isinstance(reason, Exception):
+            # for Python 3.2+
+            reason = Exception(reason)
         plugins.addError(self.test, (SkipTest, reason, None))
         self.result.addSkip(self.test, reason)
 
