--- Makefile.orig	2011-01-13 15:59:05.000000000 -0200
+++ Makefile	2011-01-14 19:12:34.234340889 -0200
@@ -20,26 +20,34 @@ SHELL = /bin/sh
 
 # The C compiler and linker to use
 CC	= gcc
-CFLAGS	= -g -O2 -Wall
+CFLAGS	?= -g -O2 -Wall
 LD	= gcc
-LDFLAGS =
+LDFLAGS ?=
 
 # Where the executables should be put
-BINDIR	= /usr/local/bin
+ifndef BINDIR
+  BINDIR = /usr/bin
+endif
 
 # Where the man page should be put
-MANDIR	= /usr/local/man
+ifndef MANDIR
+  MANDIR = /usr/man/man6
+endif
 
 # Where figlet will search first for fonts (the ".flf" files).
-DEFAULTFONTDIR = /usr/local/share/figlet
+ifndef DEFAULTFONTDIR
+  DEFAULTFONTDIR = /usr/share/figlet
+endif
 # Use this definition if you can't put things in /usr/local/share/figlet
-DEFAULTFONTDIR = fonts
+#DEFAULTFONTDIR = fonts
 
 # The filename of the font to be used if no other is specified
 #   (standard.flf is recommended, but any other can be used).
 #   This font file should reside in the directory specified by
 #   DEFAULTFONTDIR.
-DEFAULTFONTFILE = standard.flf
+ifndef DEFAULTFONTFILE
+  DEFAULTFONTFILE = standard.flf
+endif
 
 ##
 ##  END OF CONFIGURATION SECTION
