diff -up system-config-printer-1.2.0/check-device-ids.py.cdi-no-drivers system-config-printer-1.2.0/check-device-ids.py
--- system-config-printer-1.2.0/check-device-ids.py.cdi-no-drivers	2010-03-17 14:20:35.000000000 +0000
+++ system-config-printer-1.2.0/check-device-ids.py	2010-03-18 16:29:56.065865306 +0000
@@ -167,7 +167,7 @@ for device, attrs in devices.iteritems (
     try:
         drivers = ppds.ids[id_fields['MFG'].lower ()][id_fields['MDL'].lower ()]
     except KeyError:
-        drivers = None
+        drivers = []
 
     if i < n:
         more = unichr (0x2502)
