diff -up system-config-printer-1.1.91/installdriver.py.lowercase-mfg-mdl system-config-printer-1.1.91/installdriver.py
--- system-config-printer-1.1.91/installdriver.py.lowercase-mfg-mdl	2010-02-08 17:25:56.000000000 +0000
+++ system-config-printer-1.1.91/installdriver.py	2010-02-17 15:49:19.064410563 +0000
@@ -43,6 +43,8 @@ class PrinterDriversInstaller(dbus.servi
         obj = bus.get_object ("org.freedesktop.PackageKit",
                               "/org/freedesktop/PackageKit")
         proxy = dbus.Interface (obj, "org.freedesktop.PackageKit.Modify")
+        mfg = mfg.lower ()
+        mdl = mdl.lower ()
         proxy.InstallPrinterDrivers (0, ["MFG:%s;MDL:%s;" % (mfg, mdl)],
                                      "hide-finished",
                                      reply_handler=reply_handler,
