diff -up system-config-printer-1.2.0/check-device-ids.py.cdi-cmd system-config-printer-1.2.0/check-device-ids.py
--- system-config-printer-1.2.0/check-device-ids.py.cdi-cmd	2010-03-19 10:39:18.000000000 +0000
+++ system-config-printer-1.2.0/check-device-ids.py	2010-03-19 10:39:55.011621602 +0000
@@ -160,9 +160,16 @@ for device, attrs in devices.iteritems (
     else:
         line = last
 
-    print "%s %s: MFG:%s;MDL:%s;" % (line, make_and_model,
-                                     id_fields['MFG'],
-                                     id_fields['MDL'])
+    cmd = id_fields['CMD']
+    if cmd:
+        cmd = "CMD:%s;" % reduce (lambda x, y: x + ',' + y, cmd)
+    else:
+        cmd = ""
+
+    print "%s %s: MFG:%s;MDL:%s;%s" % (line, make_and_model,
+                                       id_fields['MFG'],
+                                       id_fields['MDL'],
+                                       cmd)
 
     try:
         drivers = ppds.ids[id_fields['MFG'].lower ()][id_fields['MDL'].lower ()]
