Author: Alastair McKinstry <mckinstry@debian.org>
Description: Correct minor version number
Last-Updated: 2016-06-10
Forwarded: no

Index: mxml-2.9/mxml.h
===================================================================
--- mxml-2.9.orig/mxml.h
+++ mxml-2.9/mxml.h
@@ -37,7 +37,7 @@
  */
 
 #  define MXML_MAJOR_VERSION	2	/* Major version number */
-#  define MXML_MINOR_VERSION	8	/* Minor version number */
+#  define MXML_MINOR_VERSION	9	/* Minor version number */
 
 #  define MXML_TAB		8	/* Tabs every N columns */
 
