diff -Nurp mxml-2.7/configure mxml-2.7.mod/configure
--- mxml-2.7/configure	2011-03-24 06:02:54.000000000 +0100
+++ mxml-2.7.mod/configure	2013-12-04 14:31:56.704610051 +0100
@@ -4715,8 +4715,8 @@ echo "${ECHO_T}yes" >&6; }
 echo "${ECHO_T}yes" >&6; }
 			LIBMXML="libmxml.so.1.5"
 			DSO="\$(CC)"
-			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1,-rpath,\$(libdir) -shared \$(OPTIM)"
-                        LDFLAGS="$LDFLAGS -Wl,-rpath,\$(libdir)"
+			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1 -shared \$(OPTIM)"
+                        LDFLAGS="$LDFLAGS -Wl,-soname,libmxml.so.1"
 			;;
 
 		*BSD*)
diff -Nurp mxml-2.7/configure.in mxml-2.7.mod/configure.in
--- mxml-2.7/configure.in	2011-01-03 03:03:29.000000000 +0100
+++ mxml-2.7.mod/configure.in	2013-12-04 14:33:45.288611088 +0100
@@ -196,8 +196,8 @@ if test x$enable_shared != xno; then
 			AC_MSG_RESULT(yes)
 			LIBMXML="libmxml.so.1.5"
 			DSO="\$(CC)"
-			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1,-rpath,\$(libdir) -shared \$(OPTIM)"
-                        LDFLAGS="$LDFLAGS -Wl,-rpath,\$(libdir)"
+			DSOFLAGS="$DSOFLAGS -Wl,-soname,libmxml.so.1 -shared \$(OPTIM)"
+                        LDFLAGS="$LDFLAGS -Wl,-soname,libmxml.so.1"
 			;;
 
 		*BSD*)
