From 3c20d6ec49a97c24cfb1fe9fd61ac80944819088 Mon Sep 17 00:00:00 2001
From: Dag Wieers <dag@wieers.com>
Date: Fri, 28 Sep 2012 19:33:01 +0200
Subject: [PATCH] Fix a broken --export option and add -V as alternative to
 --version

This closes #93.
---
 unoconv | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/unoconv b/unoconv
index 972e962..30e6706 100755
--- a/unoconv
+++ b/unoconv
@@ -524,8 +524,8 @@ class Options:
 
         ### Get options from the commandline
         try:
-            opts, args = getopt.getopt (args, 'c:Dd:e:f:hi:Llo:np:s:T:t:v',
-                ['connection=', 'debug', 'doctype=', 'export', 'format=',
+            opts, args = getopt.getopt (args, 'c:Dd:e:f:hi:Llo:np:s:T:t:vV',
+                ['connection=', 'debug', 'doctype=', 'export=', 'format=',
                  'help', 'import', 'listener', 'no-launch', 'output=',
                  'outputpath', 'password=', 'pipe=', 'port=', 'server=',
                  'timeout=', 'show', 'stdout', 'template', 'verbose',
@@ -609,7 +609,7 @@ class Options:
                 self.timeout = int(arg)
             elif opt in ['-v', '--verbose']:
                 self.verbose = self.verbose + 1
-            elif opt in ['--version']:
+            elif opt in ['-V', '--version']:
                 self.version()
                 sys.exit(255)
 
-- 
1.7.11.7

