diff -up ./Ambiance/index.theme.orig ./Ambiance/index.theme
--- ./Ambiance/index.theme.orig	2013-11-02 13:36:19.293911000 +0100
+++ ./Ambiance/index.theme	2013-11-02 13:36:41.483462971 +0100
@@ -8,5 +8,5 @@ Encoding=UTF-8
 GtkTheme=Ambiance
 MetacityTheme=Ambiance
 IconTheme=ubuntu-mono-dark
-CursorTheme=DMZ-White
+CursorTheme=dmz
 ButtonLayout=close,minimize,maximize:
diff -up ./Radiance/index.theme.orig ./Radiance/index.theme
--- ./Radiance/index.theme.orig	2013-11-02 13:36:19.293911000 +0100
+++ ./Radiance/index.theme	2013-11-02 13:36:41.483462971 +0100
@@ -8,5 +8,5 @@ Encoding=UTF-8
 GtkTheme=Radiance
 MetacityTheme=Radiance
 IconTheme=ubuntu-mono-light
-CursorTheme=DMZ-White
+CursorTheme=dmz
 ButtonLayout=close,minimize,maximize:
