From 85a39afe95f1c3ec50d198b01b103801418fc204 Mon Sep 17 00:00:00 2001
From: Viktor Gedzenko <foxinushka@gmail.com>
Date: Wed, 14 Jan 2015 11:43:26 +0200
Subject: [PATCH] Added timeout handling while SMTP auth

---
 src/low-level/smtp/mailsmtp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/low-level/smtp/mailsmtp.c b/src/low-level/smtp/mailsmtp.c
index 304b608..7893095 100644
--- a/src/low-level/smtp/mailsmtp.c
+++ b/src/low-level/smtp/mailsmtp.c
@@ -1300,6 +1300,10 @@ int mailesmtp_auth_sasl(mailsmtp * session, const char * auth_type,
   while (1) {
     r = read_response(session);
     switch (r) {
+    case 0:
+      res = MAILSMTP_ERROR_STREAM;
+      goto free_sasl_conn;
+        
     case 220:
     case 235:
       res = MAILSMTP_NO_ERROR;
