diff -up libetpan-1.6/src/data-types/mailstream_ssl.c.orig libetpan-1.6/src/data-types/mailstream_ssl.c
--- libetpan-1.6/src/data-types/mailstream_ssl.c.orig	2015-02-03 13:58:14.819035381 +0100
+++ libetpan-1.6/src/data-types/mailstream_ssl.c	2015-02-03 13:59:57.805757755 +0100
@@ -566,7 +566,6 @@ static struct mailstream_ssl_data * ssl_
   gnutls_certificate_set_retrieve_function(xcred, mailstream_gnutls_client_cert_cb);
 #endif
   gnutls_set_default_priority(session);
-  gnutls_priority_set_direct(session, "NORMAL", NULL);
 
   gnutls_record_disable_padding(session);
   gnutls_dh_set_prime_bits(session, 512);
