--- libetpan-config.in.orig	2013-06-04 03:48:38.000000000 -0300
+++ libetpan-config.in	2014-02-06 17:43:41.420328982 -0200
@@ -3,6 +3,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 exec_prefix_set=no
+libs="-letpan@LIBSUFFIX@"
 
 usage="\
 Usage: libetpan-config [--prefix[=DIR]] [--exec-prefix[=DIR]] [--version] [--libs] [--cflags]"
@@ -47,8 +48,10 @@ while test $# -gt 0; do
       echo $includedir
       ;;
     --libs)
-      libdir=-L@libdir@
-      echo $libdir -letpan@LIBSUFFIX@ @LDFLAGS@ @SSLLIBS@ @GNUTLSLIB@ @LIBICONV@ @DBLIB@ @LIBS@ @SASLLIBS@
+      echo $libs
+      ;;
+    --static)
+      echo $libs @SSLLIBS@ @GNUTLSLIB@ @LIBICONV@ @DBLIB@ @LIBS@ @SASLLIBS@
       ;;
     *)
       echo "${usage}" 1>&2
