diff -up live-0.2.0/Makefile.mkfix live-0.2.0/Makefile
--- live-0.2.0/Makefile.mkfix	2013-03-01 20:13:27.830658843 -0300
+++ live-0.2.0/Makefile	2013-03-01 20:13:58.009227315 -0300
@@ -41,7 +41,7 @@ TMPDIR	 ?= /tmp
 
 ### The version number of VDR's plugin API (taken from VDR's "config.h"):
 
-APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
+APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' _INCDIR_/config.h)
 I18NTARG   = $(shell if [ `echo $(APIVERSION) | tr [.] [0]` -ge "10507" ]; then echo "i18n"; fi)
 TNTVERS7   = $(shell ver=`tntnet-config --version | sed -e's/\.//g' | awk '/^..$$/ { print $$1."00"} /^...$$/ { print $$1."0"} /^....$$/ { print $$1 }'`; if [ $$ver -ge "1606" ]; then echo "yes"; fi)
 
@@ -52,7 +52,7 @@ PACKAGE = vdr-$(ARCHIVE)
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include
+INCLUDES += -I_INCDIR_
 ifneq ($(TNTVERS7),yes)
 	INCLUDES += -Ihttpd
 	LIBS	 += httpd/libhttpd.a
