diff -up ghostscript-8.71/cups/gdevcups.c.gdevcups-ripcache ghostscript-8.71/cups/gdevcups.c
--- ghostscript-8.71/cups/gdevcups.c.gdevcups-ripcache	2010-09-02 14:38:53.886818591 +0100
+++ ghostscript-8.71/cups/gdevcups.c	2010-09-02 14:40:49.925943426 +0100
@@ -915,8 +915,7 @@ cups_get_space_params(const gx_device_pr
     switch (sscanf(cache_env, "%f%254s", &cache_size, cache_units))
     {
       case 0 :
-          cache_size = 8 * 1024 * 1024;
-	  break;
+          return;
       case 1 :
           cache_size *= 4 * CUPS_TILE_SIZE * CUPS_TILE_SIZE;
 	  break;
@@ -933,12 +932,15 @@ cups_get_space_params(const gx_device_pr
     }
   }
   else
-    cache_size = 8 * 1024 * 1024;
+    return;
+
+  if (cache_size == 0)
+    return;
 
   dprintf1("DEBUG2: cache_size = %.0f\n", cache_size);
 
   space_params->MaxBitmap   = (int)cache_size;
-  space_params->BufferSpace = (int)cache_size / 10;
+  space_params->BufferSpace = (int)cache_size;
 }
 
 
