diff -up ghostscript-9.00/cups/gdevcups.c.gdevcups-debug-uninit ghostscript-9.00/cups/gdevcups.c
--- ghostscript-9.00/cups/gdevcups.c.gdevcups-debug-uninit	2010-08-12 19:10:47.000000000 +0100
+++ ghostscript-9.00/cups/gdevcups.c	2010-10-14 17:39:39.834994296 +0100
@@ -807,9 +807,10 @@ cups_get_matrix(gx_device *pdev,	/* I - 
   dprintf4("DEBUG2: PageSize = [ %d %d ], HWResolution = [ %d %d ]\n",
 	   cups->header.PageSize[0], cups->header.PageSize[1],
 	   cups->header.HWResolution[0], cups->header.HWResolution[1]);
-  dprintf4("DEBUG2: HWMargins = [ %.3f %.3f %.3f %.3f ]\n",
-	   pdev->HWMargins[0], pdev->HWMargins[1], pdev->HWMargins[2],
-	   pdev->HWMargins[3]);
+  if (size_set)
+    dprintf4("DEBUG2: HWMargins = [ %.3f %.3f %.3f %.3f ]\n",
+	     pdev->HWMargins[0], pdev->HWMargins[1], pdev->HWMargins[2],
+	     pdev->HWMargins[3]);
   dprintf6("DEBUG2: matrix = [ %.3f %.3f %.3f %.3f %.3f %.3f ]\n",
 	   pmat->xx, pmat->xy, pmat->yx, pmat->yy, pmat->tx, pmat->ty);
 #endif /* DEBUG */
