diff -up ghostscript-9.16/psi/int.mak.noopt ghostscript-9.16/psi/int.mak
--- ghostscript-9.16/psi/int.mak.noopt	2015-03-30 17:03:47.120475345 +0100
+++ ghostscript-9.16/psi/int.mak	2015-03-30 17:04:53.004247278 +0100
@@ -116,7 +116,7 @@ $(PSOBJ)igcref.$(OBJ) : $(PSSRC)igcref.c
  $(gsexit_h) $(gsstruct_h)\
  $(iastate_h) $(idebug_h) $(igc_h) $(iname_h) $(ipacked_h) $(store_h)\
  $(MAKEDIRS)
-	$(PSCC) $(PSO_)igcref.$(OBJ) $(C_) $(PSSRC)igcref.c
+	$(PSCC) -O0 $(PSO_)igcref.$(OBJ) $(C_) $(PSSRC)igcref.c
 
 $(PSOBJ)igcstr.$(OBJ) : $(PSSRC)igcstr.c $(GH) $(memory__h)\
  $(gsmdebug_h) $(gsstruct_h) $(iastate_h) $(igcstr_h) $(MAKEDIRS)
