diff -up ghostscript-8.71/base/gdevbbox.c.bbox-close ghostscript-8.71/base/gdevbbox.c
--- ghostscript-8.71/base/gdevbbox.c.bbox-close	2010-01-19 15:48:57.000000000 +0000
+++ ghostscript-8.71/base/gdevbbox.c	2010-09-02 13:03:37.742943978 +0100
@@ -249,7 +249,7 @@ bbox_close_device(gx_device * dev)
 	 * This device was created as a wrapper for a compositor.
 	 * Just free the devices.
 	 */
-	int code = (bdev->forward_open_close ? gs_closedevice(tdev) : 0);
+	int code = (tdev && bdev->forward_open_close ? gs_closedevice(tdev) : 0);
 
 	gs_free_object(dev->memory, dev, "bbox_close_device(composite)");
 	return code;
