diff -up ghostscript-9.10/base/gsicc.c.gs694154 ghostscript-9.10/base/gsicc.c
--- ghostscript-9.10/base/gsicc.c.gs694154	2013-08-30 11:37:28.000000000 +0100
+++ ghostscript-9.10/base/gsicc.c	2013-12-10 16:06:58.364001185 +0000
@@ -354,6 +354,8 @@ gx_remap_ICC(const gs_client_color * pcc
     cmm_dev_profile_t *dev_profile;
 
     code = dev_proc(dev, get_profile)(dev, &dev_profile);
+    if (dev_profile == NULL)
+        return gs_throw(gs_error_Fatal, "Attempting to do ICC remap with no profile");
     num_des_comps = gsicc_get_device_profile_comps(dev_profile);
     rendering_params.black_point_comp = pis->blackptcomp;
     rendering_params.graphics_type_tag = dev->graphics_type_tag;
diff -up ghostscript-9.10/base/gsstate.c.gs694154 ghostscript-9.10/base/gsstate.c
--- ghostscript-9.10/base/gsstate.c.gs694154	2013-08-30 11:37:28.000000000 +0100
+++ ghostscript-9.10/base/gsstate.c	2013-12-10 16:06:58.364001185 +0000
@@ -708,7 +708,7 @@ gs_setoverprint(gs_state * pgs, bool ovp
         } else {
             dev_proc(dev, get_profile)(dev,  &profile_struct);
         }
-        if (profile_struct->sim_overprint == false) return;
+        if (profile_struct != NULL && profile_struct->sim_overprint == false) return;
     }
     pgs->overprint = ovp;
     if (prior_ovp != ovp)
