diff -up ghostscript-9.00/cups/gdevcups.c.width-height ghostscript-9.00/cups/gdevcups.c
--- ghostscript-9.00/cups/gdevcups.c.width-height	2011-01-07 11:59:02.093383442 +0000
+++ ghostscript-9.00/cups/gdevcups.c	2011-01-07 11:59:48.520453332 +0000
@@ -3460,7 +3460,7 @@ cups_put_params(gx_device     *pdev,	/* 
       sp = ((gx_device_printer *)pdev)->space_params;
 
       if ((code = gdev_prn_maybe_realloc_memory((gx_device_printer *)pdev, &sp, 
-						width_old, height_old,
+						width, height,
 						cups->page_uses_transparency))
 	  < 0)
 	return (code);
