diff -up ghostscript-8.70/cups/cups.mak.cups-filters ghostscript-8.70/cups/cups.mak
--- ghostscript-8.70/cups/cups.mak.cups-filters	2009-10-15 12:42:27.531402610 +0100
+++ ghostscript-8.70/cups/cups.mak	2009-10-15 12:44:14.835402533 +0100
@@ -63,9 +63,10 @@ install-cups: cups
 	    $(INSTALL_PROGRAM) $(GSTORASTER_XE) $(DESTDIR)$(CUPSSERVERBIN)/filter; \
 	fi
 	$(INSTALL_PROGRAM) cups/pstopxl $(DESTDIR)$(CUPSSERVERBIN)/filter
-	-mkdir -p $(DESTDIR)$(CUPSSERVERROOT)
+	-mkdir -p $(DESTDIR)$(CUPSDATA)/mime
+	$(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSDATA)/mime
 	if [ "$(CUPSPDFTORASTER)" = "1" ]; then \
-	    $(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSSERVERROOT); \
+	    $(INSTALL_DATA) cups/gstoraster.convs $(DESTDIR)$(CUPSDATA)/mime; \
 	fi
 	-mkdir -p $(DESTDIR)$(CUPSDATA)/model
 	$(INSTALL_DATA) cups/pxlcolor.ppd $(DESTDIR)$(CUPSDATA)/model
