diff -up ghostscript-9.08rc1/base/gsicc_manage.c.icc-missing-check ghostscript-9.08rc1/base/gsicc_manage.c
--- ghostscript-9.08rc1/base/gsicc_manage.c.icc-missing-check	2013-08-09 14:12:18.000000000 +0100
+++ ghostscript-9.08rc1/base/gsicc_manage.c	2013-08-09 16:54:11.167695694 +0100
@@ -599,6 +599,8 @@ gsicc_set_srcgtag_struct(gsicc_manager_t
                             icc_profile =
                                 gsicc_profile_new(str, mem, curr_ptr, strlen(curr_ptr));
                             code = sfclose(str);
+                        }
+                        if (str != NULL && icc_profile != NULL) {
                             gsicc_init_profile_info(icc_profile);
                             cmm = gsCMM_DEFAULT;
                             /* Check if this object is a devicelink profile. 
