diff -up ghostscript-9.04/base/gxtype1.c.SEAC ghostscript-9.04/base/gxtype1.c
--- ghostscript-9.04/base/gxtype1.c.SEAC	2011-08-05 12:12:20.000000000 +0100
+++ ghostscript-9.04/base/gxtype1.c	2011-11-08 09:30:19.000262809 +0000
@@ -409,7 +409,12 @@ gs_type1_piece_codes(/*const*/ gs_font_t
             cnext;
             goto out;
         case c2_shortint:
-            cip += 2;
+            {
+                short sint = *cip++;
+                sint = (sint << 8) + *cip++;
+                CS_CHECK_PUSH(csp, cstack);
+                *++csp = int2fixed(sint);
+            }
             break;
         case c2_hstemhm:
             hhints += ((csp - cstack) + 1) / 2;
