--- sun/util/calendar/LocalGregorianCalendar.java.keiths	2007-09-07 14:48:19.000000000 -0700
+++ sun/util/calendar/LocalGregorianCalendar.java	2007-09-07 14:52:58.000000000 -0700
@@ -120,8 +120,7 @@ public class LocalGregorianCalendar exte
     static LocalGregorianCalendar getLocalGregorianCalendar(String name) {
 	Properties calendarProps = null;
 	try {
-	    String homeDir = (String) AccessController.doPrivileged(
-				new sun.security.action.GetPropertyAction("java.home"));
+  	    String homeDir = (String) System.getProperty("java.home");
 	    final String fname = homeDir + File.separator + "lib" + File.separator
 				 + "calendars.properties";
 	    calendarProps = (Properties) AccessController.doPrivileged(new PrivilegedExceptionAction() {
--- sun/util/calendar/ZoneInfoFile.java.keiths	2007-09-07 14:54:58.000000000 -0700
+++ sun/util/calendar/ZoneInfoFile.java	2007-09-07 14:55:36.000000000 -0700
@@ -1021,8 +1021,7 @@ public class ZoneInfoFile {
 	byte[] buffer = null;
 
 	try {
-	    String zi_dir = (String) AccessController.doPrivileged(
-			        new sun.security.action.GetPropertyAction("user.zoneinfo.dir"));
+	    String zi_dir = (String) System.getProperty("user.zoneinfo.dir");
 	    File dir = null;
 	    if (zi_dir != null)
 	      dir = new File(zi_dir);
@@ -1035,8 +1034,7 @@ public class ZoneInfoFile {
 	    }
 
 	    if (dir == null) {
-	      String homeDir = (String) AccessController.doPrivileged(
-				new sun.security.action.GetPropertyAction("java.home"));
+	      String homeDir = (String) System.getProperty("java.home");
 	      zi_dir = homeDir + File.separator + "lib" + File.separator
 		+ "zi";
 	    }
