--- src/xdgmenumaker.orig	2012-06-24 08:23:25.000000000 -0300
+++ src/xdgmenumaker	2012-06-24 16:36:20.725310444 -0300
@@ -45,6 +45,9 @@ settings_icon = de.getIcon()
 de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/xdgmenumaker-system.directory')
 system = de.getName().encode('utf-8')
 system_icon = de.getIcon()
+de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/wine.directory')
+wine = de.getName().encode('utf-8')
+wine_icon = de.getIcon()
 de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/xdgmenumaker-other.directory')
 other = de.getName().encode('utf-8')
 other_icon = de.getIcon()
@@ -170,6 +173,8 @@ def get_entry_info(desktopfile):
 		category = settings
 	elif 'Utility' in categories:
 		category = accessories
+	elif 'X-Wine' in categories:
+		category = wine
 	else:
 		category = other
 
@@ -251,6 +256,8 @@ def category_icon(category):
 		icon = settings_icon
 	elif category == system:
 		icon = system_icon
+	elif category == wine:
+		icon = wine_icon
 	elif category == other:
 		icon = other_icon
 	else:
