--- src/xdgmenumaker.orig	2012-09-22 15:17:31.000000000 -0300
+++ src/xdgmenumaker	2012-10-02 00:20:22.876733933 -0300
@@ -46,6 +46,9 @@ settings_icon = de.getIcon()
 de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/xdgmenumaker-system.directory')
 system = de.getName().encode('utf-8')
 system_icon = de.getIcon()
+de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/wine.directory')
+wine = de.getName().encode('utf-8')
+wine_icon = de.getIcon()
 de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/xdgmenumaker-other.directory')
 other = de.getName().encode('utf-8')
 other_icon = de.getIcon()
@@ -175,6 +178,8 @@ def get_entry_info(desktopfile):
 		category = settings
 	elif 'Utility' in categories:
 		category = accessories
+	elif 'X-Wine' in categories:
+		category = wine
 	else:
 		category = other
 
@@ -212,6 +217,16 @@ def desktopfilelist():
 			for i in os.listdir(xdgdir):
 				if i.endswith('.desktop'):
 					filelist.append(xdgdir+'/'+i)
+		xdgdir = d+'/applications/kde4'
+		if os.path.isdir(xdgdir):
+			for i in os.listdir(xdgdir):
+				if i.endswith('.desktop'):
+					filelist.append(xdgdir+'/'+i)
+		xdgdir = d+'/applications/wine'
+		if os.path.isdir(xdgdir):
+			for i in os.listdir(xdgdir):
+				if i.endswith('.desktop'):
+					filelist.append(xdgdir+'/'+i)
 	return filelist
 
 def menu():
@@ -256,6 +271,8 @@ def category_icon(category):
 		icon = settings_icon
 	elif category == system:
 		icon = system_icon
+	elif category == wine:
+		icon = wine_icon
 	elif category == other:
 		icon = other_icon
 	else:
