--- src/xdgmenumaker.orig	2014-01-18 12:42:40.000000000 -0200
+++ src/xdgmenumaker	2014-01-20 17:35:37.841887760 -0200
@@ -48,6 +48,9 @@ settings_icon = de.getIcon()
 de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/xdgmenumaker-system.directory')
 system = de.getName().encode('utf-8')
 system_icon = de.getIcon()
+de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/wine.directory')
+wine = de.getName().encode('utf-8')
+wine_icon = de.getIcon()
 de = dentry.DesktopEntry(filename = '/usr/share/desktop-directories/xdgmenumaker-other.directory')
 other = de.getName().encode('utf-8')
 other_icon = de.getIcon()
@@ -211,6 +214,8 @@ def get_entry_info(desktopfile):
 		category = settings
 	elif 'Utility' in categories:
 		category = accessories
+	elif 'X-Wine' in categories:
+		category = wine
 	else:
 		category = other
 
@@ -257,6 +262,21 @@ def desktopfilelist():
 					if i not in df_temp:
 						df_temp.append(i)
 						filelist.append(xdgdir+'/'+i)
+		xdgdir = d+'/applications/kde'
+		if os.path.isdir(xdgdir):
+			for i in os.listdir(xdgdir):
+				if i.endswith('.desktop'):
+					filelist.append(xdgdir+'/'+i)
+		xdgdir = d+'/applications/kde4'
+		if os.path.isdir(xdgdir):
+			for i in os.listdir(xdgdir):
+				if i.endswith('.desktop'):
+					filelist.append(xdgdir+'/'+i)
+		xdgdir = d+'/applications/wine'
+		if os.path.isdir(xdgdir):
+			for i in os.listdir(xdgdir):
+				if i.endswith('.desktop'):
+					filelist.append(xdgdir+'/'+i)
 	return filelist
 
 def menu():
@@ -302,6 +322,8 @@ def category_icon(category):
 		icon = settings_icon
 	elif category == system:
 		icon = system_icon
+	elif category == wine:
+		icon = wine_icon
 	elif category == other:
 		icon = other_icon
 	else:
