--- src/xdgmenumaker.orig	2015-09-27 23:24:55.013611940 -0300
+++ src/xdgmenumaker	2015-09-27 23:43:23.955062886 -0300
@@ -39,7 +39,7 @@ pekwmdynamic = False
 # 'not_set' it looks in the currect directory tree for the .directory
 # files. If it is actually set to something else, it looks under there
 # for them, where they should be if this was installed properly
-prefix = 'not_set'
+prefix = '/usr'
 if prefix == 'not_set':
     desktop_dir = '../desktop-directories/'
 else:
@@ -135,6 +135,10 @@ de = dentry.DesktopEntry(filename=deskto
                          'xdgmenumaker-system.directory')
 system = de.getName().encode('utf-8')
 system_icon = de.getIcon()
+de = dentry.DesktopEntry(filename=desktop_dir +
+                         'wine.directory')
+wine = de.getName().encode('utf-8')
+wine_icon = de.getIcon()
 de = dentry.DesktopEntry(filename=desktop_dir + 'xdgmenumaker-other.directory')
 other = de.getName().encode('utf-8')
 other_icon = de.getIcon()
@@ -330,6 +334,8 @@ def get_entry_info(desktopfile, ico_path
         category = settings
     elif 'Utility' in categories:
         category = accessories
+    elif 'X-Wine' in categories:
+        category = wine
     else:
         category = other
 
@@ -376,6 +382,27 @@ def desktopfilelist():
                     if i not in df_temp:
                         df_temp.append(i)
                         filelist.append('{}/{}'.format(xdgdir, i))
+        xdgdir = '{}/applications/kde'.format(d)
+        if os.path.isdir(xdgdir):
+            for i in os.listdir(xdgdir):
+                if i.endswith('.desktop'):
+                    if i not in df_temp:
+                        df_temp.append(i)
+                        filelist.append('{}/{}'.format(xdgdir, i))
+        xdgdir = '{}/applications/kde4'.format(d)
+        if os.path.isdir(xdgdir):
+            for i in os.listdir(xdgdir):
+                if i.endswith('.desktop'):
+                    if i not in df_temp:
+                        df_temp.append(i)
+                        filelist.append('{}/{}'.format(xdgdir, i))
+        xdgdir = '{}/applications/wine'.format(d)
+        if os.path.isdir(xdgdir):
+            for i in os.listdir(xdgdir):
+                if i.endswith('.desktop'):
+                    if i not in df_temp:
+                        df_temp.append(i)
+                        filelist.append('{}/{}'.format(xdgdir, i))
     return filelist
 
 
@@ -423,6 +450,8 @@ def category_icon(category):
         icon = settings_icon
     elif category == system:
         icon = system_icon
+    elif category == wine:
+        icon = wine_icon
     elif category == other:
         icon = other_icon
     else:
