--- wxPython-src-2.8.9.2/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp.orig	2009-04-10 23:03:18.000000000 +0200
+++ wxPython-src-2.8.9.2/wxPython/contrib/gizmos/gtk/gizmos_wrap.cpp	2009-04-10 23:03:41.000000000 +0200
@@ -10321,7 +10321,7 @@ SWIGINTERN PyObject *_wrap_TreeListCtrl_
   PyObject *resultobj = 0;
   wxPyTreeListCtrl *arg1 = (wxPyTreeListCtrl *) 0 ;
   wxTreeItemId *arg2 = 0 ;
-  wxTreeItemId const &arg3_defvalue = NULL ;
+  wxTreeItemId const &arg3_defvalue = (wxTreeItemId *) NULL ;
   wxTreeItemId *arg3 = (wxTreeItemId *) &arg3_defvalue ;
   bool arg4 = (bool) true ;
   void *argp1 = 0 ;
