From a26837d20ea626943398f2390bfd00c22be3468b Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@dhcp-40-90.bne.redhat.com>
Date: Mon, 10 Sep 2012 12:02:55 +1000
Subject: [PATCH] scan pci after probing devices

---
 hw/xfree86/common/xf86platformBus.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hw/xfree86/common/xf86platformBus.c b/hw/xfree86/common/xf86platformBus.c
index 502d3c4..5e05791 100644
--- a/hw/xfree86/common/xf86platformBus.c
+++ b/hw/xfree86/common/xf86platformBus.c
@@ -213,11 +213,12 @@ xf86platformProbe(void)
     int i;
     Bool pci = TRUE;
 
+    config_odev_probe(&xf86PlatformDeviceProbe);
+
     if (!xf86scanpci()) {
         pci = FALSE;
     }
 
-    config_odev_probe(&xf86PlatformDeviceProbe);
     for (i = 0; i < xf86_num_platform_devices; i++) {
         char *busid = xf86_get_platform_attrib(i, ODEV_ATTRIB_BUSID);
 
-- 
1.7.12

