--- configure.ac.orig	2010-08-24 19:43:11.087631059 -0300
+++ configure.ac	2010-08-24 19:44:08.918568181 -0300
@@ -812,7 +812,7 @@
 LIBGLIB="glib-2.0 >= 2.16"
 LIBUDEV="libudev >= 143"
 LIBSELINUX="libselinux >= 2.0.86"
-LIBDBUS="dbus-1 >= 1.0"
+LIBDBUS="dbus-1-disabled >= 1.0"
 LIBPIXMAN="pixman-1 >= 0.15.20"
 
 dnl Pixman is always required, but we separate it out so we can link
@@ -863,7 +863,7 @@
 fi
 AM_CONDITIONAL(CONFIG_DBUS_API, [test "x$CONFIG_DBUS_API" = xyes])
 
-PKG_CHECK_MODULES(HAL, hal, [HAVE_HAL=yes], [HAVE_HAL=no])
+PKG_CHECK_MODULES(HAL, hal-disabled, [HAVE_HAL=yes], [HAVE_HAL=no])
 if test "x$CONFIG_HAL" = xauto; then
 	CONFIG_HAL="$HAVE_HAL"
 fi
