From d8df4c83537cd2041141478b0fdc512f4b46219b Mon Sep 17 00:00:00 2001
From: Linus Arver <linusarver@gmail.com>
Date: Tue, 9 Nov 2010 15:01:06 -0500
Subject: [PATCH] http://lists.x.org/archives/xorg-devel/2010-October/014150.html

---
 Xext/panoramiXprocs.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Xext/panoramiXprocs.c b/Xext/panoramiXprocs.c
index 67b4030..d843168 100644
--- a/Xext/panoramiXprocs.c
+++ b/Xext/panoramiXprocs.c
@@ -634,7 +634,7 @@ int PanoramiXTranslateCoords(ClientPtr client)
     rep.dstX = x - pDst->drawable.x;
     rep.dstY = y - pDst->drawable.y;
     if((pDst == screenInfo.screens[0]->root) ||
-       (pWin->drawable.id == screenInfo.screens[0]->screensaver.wid))
+       (pDst->drawable.id == screenInfo.screens[0]->screensaver.wid))
     {
 	rep.dstX += screenInfo.screens[0]->x;
 	rep.dstY += screenInfo.screens[0]->y;
-- 
1.7.2.3

