From 92a02d58bf4e1d0b4779d37401d40c28e3a4ba2a Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Thu, 4 Mar 2010 15:18:38 +1000
Subject: [PATCH 11/17] tweak default mode list yet again

---
 hw/xfree86/modes/xf86Crtc.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/hw/xfree86/modes/xf86Crtc.c b/hw/xfree86/modes/xf86Crtc.c
index d907eb2..dc159a1 100644
--- a/hw/xfree86/modes/xf86Crtc.c
+++ b/hw/xfree86/modes/xf86Crtc.c
@@ -1630,6 +1630,9 @@ xf86ProbeOutputModes (ScrnInfoPtr scrn, int maxX, int maxY)
 	}
 	
 	output_modes = (*output->funcs->get_modes) (output);
+
+	/* if the driver found modes, don't second-guess it */
+	add_default_modes = (output_modes == NULL);
 	
 	edid_monitor = output->MonInfo;
 	
-- 
1.7.1

