From dc9f6e5021699fef97318ebc43f0e6c58a12d696 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 10 Dec 2007 11:26:57 -0500
Subject: [PATCH] Build libxf86config with -fPIC.


diff --git a/hw/xfree86/parser/Makefile.am b/hw/xfree86/parser/Makefile.am
index 49c191f..fea3b24 100644
--- a/hw/xfree86/parser/Makefile.am
+++ b/hw/xfree86/parser/Makefile.am
@@ -32,7 +32,7 @@ libxf86config_internal_la_SOURCES = \
 
 libxf86config_a_SOURCES = \
 	$(INTERNAL_SOURCES)
-libxf86config_a_CFLAGS = $(AM_CFLAGS)
+libxf86config_a_CFLAGS = $(AM_CFLAGS) -fPIC -fvisibility=hidden
 
 AM_CFLAGS = $(DIX_CFLAGS) $(XORG_CFLAGS)
 
-- 
1.6.6.1

