--- configure.ac.orig	2011-09-24 04:05:38.000000000 -0300
+++ configure.ac	2011-10-07 12:52:20.596671257 -0300
@@ -802,7 +802,7 @@ LIBXTST="xtst >= 1.0.99.2"
 LIBPCIACCESS="pciaccess >= 0.8.0"
 LIBUDEV="libudev >= 143"
 LIBSELINUX="libselinux >= 2.0.86"
-LIBDBUS="dbus-1 >= 1.0"
+LIBDBUS="dbus-1-disabled >= 1.0"
 LIBPIXMAN="pixman-1 >= 0.21.8"
 
 dnl Pixman is always required, but we separate it out so we can link
@@ -853,7 +853,7 @@ if test "x$CONFIG_DBUS_API" = xyes; then
 fi
 AM_CONDITIONAL(CONFIG_DBUS_API, [test "x$CONFIG_DBUS_API" = xyes])
 
-PKG_CHECK_MODULES(HAL, hal, [HAVE_HAL=yes], [HAVE_HAL=no])
+PKG_CHECK_MODULES(HAL, hal-disabled, [HAVE_HAL=yes], [HAVE_HAL=no])
 if test "x$CONFIG_HAL" = xauto; then
 	CONFIG_HAL="$HAVE_HAL"
 fi
