diff -up xorg-server-1.11.1/hw/xfree86/common/xf86Config.c.zap-warning xorg-server-1.11.1/hw/xfree86/common/xf86Config.c
--- xorg-server-1.11.1/hw/xfree86/common/xf86Config.c.zap-warning	2011-07-31 20:44:54.000000000 -0300
+++ xorg-server-1.11.1/hw/xfree86/common/xf86Config.c	2011-10-07 16:17:52.361519799 -0300
@@ -666,6 +666,7 @@ typedef enum {
     FLAG_NOTRAPSIGNALS,
     FLAG_DONTVTSWITCH,
     FLAG_DONTZAP,
+    FLAG_ZAPWARNING,
     FLAG_DONTZOOM,
     FLAG_DISABLEVIDMODE,
     FLAG_ALLOWNONLOCAL,
@@ -704,6 +705,8 @@ static OptionInfoRec FlagOptions[] = {
 	{0}, FALSE },
   { FLAG_DONTZAP,		"DontZap",			OPTV_BOOLEAN,
 	{0}, FALSE },
+  { FLAG_ZAPWARNING,		"ZapWarning",			OPTV_BOOLEAN,
+	{0}, FALSE },
   { FLAG_DONTZOOM,		"DontZoom",			OPTV_BOOLEAN,
 	{0}, FALSE },
   { FLAG_DISABLEVIDMODE,	"DisableVidModeExtension",	OPTV_BOOLEAN,
@@ -808,6 +811,7 @@ configServerFlags(XF86ConfFlagsPtr flags
     xf86GetOptValBool(FlagOptions, FLAG_NOTRAPSIGNALS, &xf86Info.notrapSignals);
     xf86GetOptValBool(FlagOptions, FLAG_DONTVTSWITCH, &xf86Info.dontVTSwitch);
     xf86GetOptValBool(FlagOptions, FLAG_DONTZAP, &xf86Info.dontZap);
+    xf86GetOptValBool(FlagOptions, FLAG_ZAPWARNING, &xf86Info.ZapWarning);
     xf86GetOptValBool(FlagOptions, FLAG_DONTZOOM, &xf86Info.dontZoom);
 
     xf86GetOptValBool(FlagOptions, FLAG_IGNORE_ABI, &xf86Info.ignoreABI);
diff -up xorg-server-1.11.1/hw/xfree86/common/xf86Events.c.zap-warning xorg-server-1.11.1/hw/xfree86/common/xf86Events.c
--- xorg-server-1.11.1/hw/xfree86/common/xf86Events.c.zap-warning	2011-05-13 18:06:03.000000000 -0300
+++ xorg-server-1.11.1/hw/xfree86/common/xf86Events.c	2011-10-07 16:17:52.361519799 -0300
@@ -179,12 +179,26 @@ xf86ProcessActionEvent(ActionEvent actio
     DebugF("ProcessActionEvent(%d,%x)\n", (int) action, arg);
     switch (action) {
     case ACTION_TERMINATE:
-	if (!xf86Info.dontZap) {
+	if (xf86Info.dontZap)
+	    break;
+
+ 	if (xf86Info.ZapWarning) {
+	    static struct timeval LastZap = { 0, 0};
+	    struct timeval NewZap;
+
+	    gettimeofday(&NewZap, NULL);
+
+	    if ((NewZap.tv_sec - LastZap.tv_sec) >= 2) {
+		xf86OSRingBell(30, 1000, 50);
+		LastZap = NewZap;
+		break;
+	    }
+	}
+
 #ifdef XFreeXDGA
-	    DGAShutdown();
+	DGAShutdown();
 #endif
-	    GiveUp(0);
-	}
+	GiveUp(0);
 	break;
     case ACTION_NEXT_MODE:
 	if (!xf86Info.dontZoom)
diff -up xorg-server-1.11.1/hw/xfree86/common/xf86Globals.c.zap-warning xorg-server-1.11.1/hw/xfree86/common/xf86Globals.c
--- xorg-server-1.11.1/hw/xfree86/common/xf86Globals.c.zap-warning	2011-03-02 01:09:32.000000000 -0300
+++ xorg-server-1.11.1/hw/xfree86/common/xf86Globals.c	2011-10-07 16:17:52.361519799 -0300
@@ -104,6 +104,7 @@ xf86InfoRec xf86Info = {
 #endif
     .dontVTSwitch               = FALSE,
     .dontZap                    = FALSE,
+    .ZapWarning                 = TRUE,
     .dontZoom                   = FALSE,
     .notrapSignals              = FALSE,
     .caughtSignal               = FALSE,
diff -up xorg-server-1.11.1/hw/xfree86/man/Xorg.man.zap-warning xorg-server-1.11.1/hw/xfree86/man/Xorg.man
--- xorg-server-1.11.1/hw/xfree86/man/Xorg.man.zap-warning	2011-06-25 15:31:50.000000000 -0300
+++ xorg-server-1.11.1/hw/xfree86/man/Xorg.man	2011-10-07 16:17:52.362519588 -0300
@@ -440,7 +440,7 @@ keymap.
 .B Ctrl+Alt+Backspace
 Immediately kills the server -- no questions asked. It can be disabled by
 setting the
-.B DontZap
+.B DontZap/ZapWarning
 __xconfigfile__(__filemansuffix__) file option to a TRUE value.
 .PP
 .RS 8
diff -up xorg-server-1.11.1/hw/xfree86/man/xorg.conf.man.zap-warning xorg-server-1.11.1/hw/xfree86/man/xorg.conf.man
--- xorg-server-1.11.1/hw/xfree86/man/xorg.conf.man.zap-warning	2011-05-30 05:18:18.000000000 -0300
+++ xorg-server-1.11.1/hw/xfree86/man/xorg.conf.man	2011-10-07 16:17:52.362519588 -0300
@@ -533,6 +533,13 @@ This action is normally used to terminat
 When this option is enabled, the action has no effect.
 Default: off.
 .TP 7
+.BI "Option \*qZapWarning\*q  \*q" boolean \*q
+This warns the user loudly when the
+.B Ctrl+Alt+Backspace
+sequence is pressed for the first time but still terminates the __xservername__
+server when this key-sequence is pressed again shortly after.
+Default: on.
+.TP 7
 .BI "Option \*qDontZoom\*q  \*q" boolean \*q
 This disallows the use of the
 .B Ctrl+Alt+Keypad\-Plus
