From a7949c46adda1d4bc223e7832361373577455c1b Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 19 Apr 2011 13:50:12 -0400
Subject: [PATCH] load swrastg

---
 glx/glxdriswrast.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/glx/glxdriswrast.c b/glx/glxdriswrast.c
index 08ea338..14e7469 100644
--- a/glx/glxdriswrast.c
+++ b/glx/glxdriswrast.c
@@ -453,6 +453,14 @@ __glXDRIscreenProbe(ScreenPtr pScreen)
 	     "%s/%s_dri.so", dri_driver_path, driverName);
 
     screen->driver = dlopen(filename, RTLD_LAZY | RTLD_LOCAL);
+
+    if (screen->driver == NULL) {
+	driverName = "swrastg";
+	snprintf(filename, sizeof filename,
+		 "%s/%s_dri.so", dri_driver_path, driverName);
+	screen->driver = dlopen(filename, RTLD_LAZY | RTLD_LOCAL);
+    }
+
     if (screen->driver == NULL) {
 	LogMessage(X_ERROR, "AIGLX error: dlopen of %s failed (%s)\n",
 		   filename, dlerror());
-- 
1.7.4.4

