From 677e7a9f0e58c3b9e6598104934da68ac12bc32c Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 13 Oct 2010 11:16:31 -0400
Subject: [PATCH] qxl v1 fallback

---
 hw/xfree86/common/xf86pciBus.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/hw/xfree86/common/xf86pciBus.c b/hw/xfree86/common/xf86pciBus.c
index 001625c..0d23f6a 100644
--- a/hw/xfree86/common/xf86pciBus.c
+++ b/hw/xfree86/common/xf86pciBus.c
@@ -1136,7 +1136,12 @@ videoPtrToDriverList(struct pci_device *dev,
 	    }
 	    break;
 	case 0x1106:		    driverList[0] = "openchrome"; break;
-        case 0x1b36:		    driverList[0] = "qxl"; break;
+	case 0x1b36:
+	    /* The new qxl driver does not work with the old spice-0.4 qxl
+	       device for now (falling back to the vesa driver does work). */
+	    if (dev->revision != 1)
+		driverList[0] = "qxl";
+	    break;
 	case 0x1163:		    driverList[0] = "rendition"; break;
 	case 0x5333:
 	    switch (dev->device_id)
-- 
1.7.2.3

