--- xorg-server-1.17.0/configure.ac.systemd209	2015-02-04 22:50:52.831850296 -0200
+++ xorg-server-1.17.0/configure.ac	2015-02-04 22:51:38.679093237 -0200
@@ -833,7 +833,7 @@ AC_ARG_WITH([systemd-daemon],
 	AS_HELP_STRING([--with-systemd-daemon],
 		[support systemd socket activation (default: auto)]),
 	[WITH_SYSTEMD_DAEMON=$withval], [WITH_SYSTEMD_DAEMON=auto])
-PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd-daemon],
+PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd],
                   [HAVE_SYSTEMD_DAEMON=yes], [HAVE_SYSTEMD_DAEMON=no])
 if test "x$WITH_SYSTEMD_DAEMON" = xauto; then
 	WITH_SYSTEMD_DAEMON="$HAVE_SYSTEMD_DAEMON"
@@ -842,8 +842,8 @@ if test "x$WITH_SYSTEMD_DAEMON" = xyes;
 	if test "x$HAVE_SYSTEMD_DAEMON" = xno; then
 		AC_MSG_ERROR([systemd support requested but no library has been found])
 	fi
-	AC_DEFINE(HAVE_SYSTEMD_DAEMON, 1, [Define to 1 if libsystemd-daemon is available])
-	REQUIRED_LIBS="$REQUIRED_LIBS libsystemd-daemon"
+	AC_DEFINE(HAVE_SYSTEMD_DAEMON, 1, [Define to 1 if libsystemd is available])
+	REQUIRED_LIBS="$REQUIRED_LIBS libsystemd"
 fi
 AM_CONDITIONAL([HAVE_SYSTEMD_DAEMON], [test "x$HAVE_SYSTEMD_DAEMON" = "xyes"])
 
