From ed61aafc4e2b2856955528ba8142075ce53d6dd9 Mon Sep 17 00:00:00 2001
From: Dan Horák <dan@danny.cz>
Date: Fri, 15 Jun 2012 06:25:58 -0400
Subject: [PATCH] build the touch test only when building Xorg

---
 test/Makefile.am |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/Makefile.am b/test/Makefile.am
index 32be00d..93a102a 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -1,11 +1,11 @@
 if ENABLE_UNIT_TESTS
 SUBDIRS= .
-noinst_PROGRAMS = list string touch
+noinst_PROGRAMS = list string
 if XORG
 # Tests that require at least some DDX functions in order to fully link
 # For now, requires xf86 ddx, could be adjusted to use another
 SUBDIRS += xi2
-noinst_PROGRAMS += xkb input xtest misc fixes xfree86
+noinst_PROGRAMS += xkb input xtest misc fixes xfree86 touch
 endif
 check_LTLIBRARIES = libxservertest.la
 
-- 
1.7.10.2

