--- xorg-server-1.16.0/configure.ac.orig	2014-07-18 19:44:42.841822268 -0300
+++ xorg-server-1.16.0/configure.ac	2014-07-18 19:47:46.062933249 -0300
@@ -894,7 +894,7 @@ if test "x$HAVE_DBUS" = xyes; then
 fi
 AM_CONDITIONAL(HAVE_DBUS, [test "x$HAVE_DBUS" = xyes])
 
-PKG_CHECK_MODULES(HAL, hal, [HAVE_HAL=yes], [HAVE_HAL=no])
+PKG_CHECK_MODULES(HAL, hal-disabled, [HAVE_HAL=yes], [HAVE_HAL=no])
 if test "x$CONFIG_HAL" = xauto; then
 	CONFIG_HAL="$HAVE_HAL"
 fi
