--- xorg-server-1.14.0/configure.ac.orig	2013-03-06 18:24:54.676064989 -0300
+++ xorg-server-1.14.0/configure.ac	2013-03-06 18:27:03.730795620 -0300
@@ -810,7 +810,7 @@ LIBXTST="xtst >= 1.0.99.2"
 LIBPCIACCESS="pciaccess >= 0.12.901"
 LIBUDEV="libudev >= 143"
 LIBSELINUX="libselinux >= 2.0.86"
-LIBDBUS="dbus-1 >= 1.0"
+LIBDBUS="dbus-1-disabled >= 1.0"
 LIBPIXMAN="pixman-1 >= 0.27.2"
 
 dnl Pixman is always required, but we separate it out so we can link
@@ -881,7 +881,7 @@ if test "x$CONFIG_DBUS_API" = xyes; then
 fi
 AM_CONDITIONAL(CONFIG_DBUS_API, [test "x$CONFIG_DBUS_API" = xyes])
 
-PKG_CHECK_MODULES(HAL, hal, [HAVE_HAL=yes], [HAVE_HAL=no])
+PKG_CHECK_MODULES(HAL, hal-disabled, [HAVE_HAL=yes], [HAVE_HAL=no])
 if test "x$CONFIG_HAL" = xauto; then
 	CONFIG_HAL="$HAVE_HAL"
 fi
