commit d9f9b8813fdd8bb09c46c1be605c76794782e33e
Author: Miroslav Lichvar <mlichvar@redhat.com>
Date:   Fri Aug 5 15:36:04 2011 +0200

    fix another spinner warning

diff --git a/src/obconf.glade b/src/obconf.glade
index 0e024f8..25feb72 100644
--- a/src/obconf.glade
+++ b/src/obconf.glade
@@ -1882,7 +1882,7 @@ Omnipresent (On all desktops)</property>
 				  <property name="update_policy">GTK_UPDATE_ALWAYS</property>
 				  <property name="snap_to_ticks">False</property>
 				  <property name="wrap">False</property>
-				  <property name="adjustment">1 1 100 1 10 10</property>
+				  <property name="adjustment">1 1 100 1 10 0</property>
 				  <signal name="value_changed" handler="on_fixed_monitor_value_changed" last_modification_time="Mon, 21 Dec 2009 18:32:45 GMT"/>
 				</widget>
 				<packing>
