diff -up ddrescue-1.13/testsuite/check.sh.test ddrescue-1.13/testsuite/check.sh
--- ddrescue-1.13/testsuite/check.sh.test	2010-08-27 19:50:11.000000000 +0300
+++ ddrescue-1.13/testsuite/check.sh	2010-11-18 14:39:05.076190104 +0200
@@ -7,7 +7,8 @@
 
 LC_ALL=C
 export LC_ALL
-objdir=`pwd`
+pwd=`pwd`
+objdir=`dirname $pwd`
 testdir=`cd "$1" ; pwd`
 DDRESCUE="${objdir}"/ddrescue
 framework_failure() { echo "failure in testing framework" ; exit 1 ; }
@@ -20,7 +21,7 @@ fi
 if [ -d tmp ] ; then rm -rf tmp ; fi
 mkdir tmp
 printf "testing ddrescue-%s..." "$2"
-cd "${objdir}"/tmp
+cd "${pwd}"/tmp
 
 cat "${testdir}"/test.txt > in || framework_failure
 fail=0
@@ -47,7 +48,7 @@ printf .
 echo
 if [ ${fail} = 0 ] ; then
 	echo "tests completed successfully."
-	cd "${objdir}" && rm -r tmp
+	cd "${pwd}" && rm -r tmp
 else
 	echo "tests failed."
 fi
