--- CMakeLists.txt.orig	2014-05-07 21:53:44.000000000 -0300
+++ CMakeLists.txt	2014-05-07 21:54:14.353769718 -0300
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
+project(lxqt-policykit-agent)
+
 # additional cmake files
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
@@ -13,8 +15,6 @@ if (NOT POLKITQT-1_LIB_DIR AND NOT POLKI
     message(WARNING "Qt bindings for policykit are not found. LxQt policykit integration won't be built")
     message(STATUS "Hint: package libpolkit-qt-1-devel on openSUSE")
 else ()
-    project(lxqt-policykit-agent)
-
     message(STATUS "PolkitQt-1: ${POLKITQT-1_VERSION_STRING}")
     message(STATUS "      libs: ${POLKITQT-1_CORE_LIBRARY} ${POLKITQT-1_AGENT_LIBRARY}")
     message(STATUS "   include: ${POLKITQT-1_INCLUDE_DIR}")
