diff -up openjdk/jdk/make/common/shared/Sanity.gmk.sav openjdk/jdk/make/common/shared/Sanity.gmk
--- openjdk/jdk/make/common/shared/Sanity.gmk.sav	2012-02-14 16:12:48.000000000 -0500
+++ openjdk/jdk/make/common/shared/Sanity.gmk	2012-03-07 17:31:26.153840755 -0500
@@ -814,12 +814,12 @@ ifdef OPENJDK
 	@(($(CD) $(BUILDDIR)/tools/freetypecheck && $(MAKE)) || \
 	    $(ECHO) "Failed to build freetypecheck." ) > $@
 
-    sane-freetype: $(TEMPDIR)/freetypeinfo
-	@if [ "`$(CAT) $< | $(GREP) Fail`" != "" ]; then \
-	  $(ECHO) "ERROR: FreeType version " $(REQUIRED_FREETYPE_VERSION) \
-	          " or higher is required. \n" \
-		  "`$(CAT) $<`  \n" >> $(ERROR_FILE) ; \
-	fi
+#    sane-freetype: $(TEMPDIR)/freetypeinfo
+#	@if [ "`$(CAT) $< | $(GREP) Fail`" != "" ]; then \
+#	  $(ECHO) "ERROR: FreeType version " $(REQUIRED_FREETYPE_VERSION) \
+#	          " or higher is required. \n" \
+#		  "`$(CAT) $<`  \n" >> $(ERROR_FILE) ; \
+#	fi
   else
     #do nothing  (cross-compiling)
     sane-freetype: 
