--- bonnie++-1.96.orig/Makefile.in	2009-12-09 10:01:09.000000000 -0500
+++ bonnie++-1.96/Makefile.in	2009-12-09 10:02:33.000000000 -0500
@@ -17,6 +17,8 @@ THREAD_LFLAGS=@thread_ldflags@
 INSTALL=@INSTALL@
 INSTALL_PROGRAM=@INSTALL_PROGRAM@
 
+DESTDIR=
+
 BONSRC=bonnie++.cpp bon_io.cpp bon_file.cpp bon_time.cpp semaphore.cpp \
  sync.cpp thread.cpp bon_suid.cpp duration.cpp rand.o util.o
 BONOBJS=$(BONSRC:.cpp=.o)
@@ -52,15 +54,15 @@ generate_randfile: generate_randfile.o
 	$(LINK) generate_randfile.o -o generate_randfile
 
 install-bin: $(EXE) $(EXES)
-	mkdir -p $(eprefix)/bin $(eprefix)/sbin
-	@INSTALL_PROGRAM@ @stripping@ $(EXES) $(eprefix)/sbin
-	@INSTALL_PROGRAM@ @stripping@ $(EXE) $(eprefix)/bin
-	@INSTALL_SCRIPT@ $(SCRIPTS) $(eprefix)/bin
+	mkdir -p $(DESTDIR)$(eprefix)/bin $(DESTDIR)$(eprefix)/sbin
+	@INSTALL_PROGRAM@ @stripping@ $(EXES) $(DESTDIR)$(eprefix)/sbin
+	@INSTALL_PROGRAM@ @stripping@ $(EXE) $(DESTDIR)$(eprefix)/bin
+	@INSTALL_SCRIPT@ $(SCRIPTS) $(DESTDIR)$(eprefix)/bin
 
 install: install-bin
-	mkdir -p @mandir@/man1 @mandir@/man8
-	@INSTALL_DATA@ $(MAN1) @mandir@/man1
-	@INSTALL_DATA@ $(MAN8) @mandir@/man8
+	mkdir -p $(DESTDIR)@mandir@/man1 $(DESTDIR)@mandir@/man8
+	@INSTALL_DATA@ $(MAN1) $(DESTDIR)@mandir@/man1
+	@INSTALL_DATA@ $(MAN8) $(DESTDIR)@mandir@/man8
 
 %.o: %.cpp
 	$(CXX) -c $<
