diff -Nur xfce4-settings-4.9.4.orig/dialogs/accessibility-settings/xfce4-accessibility-settings.desktop.in xfce4-settings-4.9.4/dialogs/accessibility-settings/xfce4-accessibility-settings.desktop.in
--- xfce4-settings-4.9.4.orig/dialogs/accessibility-settings/xfce4-accessibility-settings.desktop.in2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/dialogs/accessibility-settings/xfce4-accessibility-settings.desktop.in	2012-04-02 15:54:06.939388473 -0600
@@ -6,7 +6,7 @@
 Icon=preferences-desktop-accessibility
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-SystemSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-SystemSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfce4-settings-4.9.4.orig/dialogs/appearance-settings/xfce-ui-settings.desktop.in xfce4-settings-4.9.4/dialogs/appearance-settings/xfce-ui-settings.desktop.in
--- xfce4-settings-4.9.4.orig/dialogs/appearance-settings/xfce-ui-settings.desktop.in	2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/dialogs/appearance-settings/xfce-ui-settings.desktop.in	2012-04-02 15:54:12.520420061 -0600
@@ -6,7 +6,7 @@
 Icon=preferences-desktop-theme
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfce4-settings-4.9.4.orig/dialogs/display-settings/xfce-display-settings.desktop.in xfce4-settings-4.9.4/dialogs/display-settings/xfce-display-settings.desktop.in
--- xfce4-settings-4.9.4.orig/dialogs/display-settings/xfce-display-settings.desktop.in	2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/dialogs/display-settings/xfce-display-settings.desktop.in	2012-04-02 15:51:46.743627381 -0600
@@ -6,7 +6,7 @@
 Icon=video-display
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfce4-settings-4.9.4.orig/dialogs/keyboard-settings/xfce-keyboard-settings.desktop.in xfce4-settings-4.9.4/dialogs/keyboard-settings/xfce-keyboard-settings.desktop.in
--- xfce4-settings-4.9.4.orig/dialogs/keyboard-settings/xfce-keyboard-settings.desktop.in	2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/dialogs/keyboard-settings/xfce-keyboard-settings.desktop.in	2012-04-02 15:54:23.379481500 -0600
@@ -6,7 +6,7 @@
 Icon=preferences-desktop-keyboard
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfce4-settings-4.9.4.orig/dialogs/mime-settings/xfce4-mime-settings.desktop.in xfce4-settings-4.9.4/dialogs/mime-settings/xfce4-mime-settings.desktop.in
--- xfce4-settings-4.9.4.orig/dialogs/mime-settings/xfce4-mime-settings.desktop.in	2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/dialogs/mime-settings/xfce4-mime-settings.desktop.in	2012-04-02 15:54:18.154451947 -0600
@@ -6,6 +6,6 @@
 Icon=application-default-icon
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-SystemSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-SystemSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
diff -Nur xfce4-settings-4.9.4.orig/dialogs/mouse-settings/xfce-mouse-settings.desktop.in xfce4-settings-4.9.4/dialogs/mouse-settings/xfce-mouse-settings.desktop.in
--- xfce4-settings-4.9.4.orig/dialogs/mouse-settings/xfce-mouse-settings.desktop.in	2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/dialogs/mouse-settings/xfce-mouse-settings.desktop.in	2012-04-02 15:54:01.019354961 -0600
@@ -6,7 +6,7 @@
 Icon=preferences-desktop-peripherals
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfce4-settings-4.9.4.orig/xfce4-settings-editor/xfce4-settings-editor.desktop.in xfce4-settings-4.9.4/xfce4-settings-editor/xfce4-settings-editor.desktop.in
--- xfce4-settings-4.9.4.orig/xfce4-settings-editor/xfce4-settings-editor.desktop.in	2012-04-01 03:58:17.000000000 -0600
+++ xfce4-settings-4.9.4/xfce4-settings-editor/xfce4-settings-editor.desktop.in	2012-04-02 15:50:59.565376633 -0600
@@ -5,6 +5,6 @@
 Icon=preferences-system
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;
 OnlyShowIn=XFCE;
 X-XfceSettingsManagerHidden=true
