--- grep-2.21/build-aux/update-copyright.orig	2015-08-02 18:10:29.174256966 +0100
+++ grep-2.21/build-aux/update-copyright	2015-08-02 18:11:19.926287054 +0100
@@ -124,7 +124,7 @@
 use warnings;
 
 my $copyright_re = 'Copyright';
-my $circle_c_re = '(?:\([cC]\)|@copyright{}|&copy;)';
+my $circle_c_re = '(?:\([cC]\)|@copyright\{}|\\\\\(co|&copy;)';
 my $holder = $ENV{UPDATE_COPYRIGHT_HOLDER};
 $holder ||= 'Free Software Foundation, Inc.';
 my $prefix_max = 5;
