From 624fb2824cc4f318caf99c3912cfb0ad6c74b418 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Fri, 27 Jul 2012 23:54:26 +0000
Subject: maint: remove stat-size

* src/main.c: Don't include stat-size.h; no longer needed.
---
diff --git a/src/main.c b/src/main.c
index 6954760..84066d8 100644
--- a/src/main.c
+++ b/src/main.c
@@ -44,7 +44,6 @@
 #include "progname.h"
 #include "propername.h"
 #include "quote.h"
-#include "stat-size.h"
 #include "version-etc.h"
 #include "xalloc.h"
 #include "xstrtol.h"
--
cgit v0.9.0.2
