From 9e760c18472cd8ac04d228e3b25dee2bbc8975fc Mon Sep 17 00:00:00 2001
From: Tristan Matthews <tmatth@videolan.org>
Date: Sat, 6 Dec 2014 18:39:55 -0500
Subject: [PATCH 1/2] fix configure typo

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b555194..8bc54c7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -200,7 +200,7 @@ fi])
 
 AC_ARG_ENABLE(examples, [  --disable-examples      Do not build example programs, only the library])
 if test "$enableval" != no; then
-  AM_CONDITIONAL([BUILD_EXAMPLES], true),
+  AM_CONDITIONAL([BUILD_EXAMPLES], true)
 else
   AM_CONDITIONAL([BUILD_EXAMPLES], false)
 fi
-- 
2.2.0

