From ac91846ffae99867bbf076a9e3f0d01276d47646 Mon Sep 17 00:00:00 2001
From: Tristan Matthews <tmatth@videolan.org>
Date: Sat, 6 Dec 2014 18:49:24 -0500
Subject: [PATCH 2/2] ensure speex_buffer.h is included in dist but not
 installed

---
 include/speex/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/speex/Makefile.am b/include/speex/Makefile.am
index ebddaa8..68789ac 100644
--- a/include/speex/Makefile.am
+++ b/include/speex/Makefile.am
@@ -14,3 +14,5 @@ nodist_speexinclude_HEADERS = speexdsp_config_types.h
 speexinclude_HEADERS = speex_echo.h speex_jitter.h speex_preprocess.h speex_resampler.h \
 	speexdsp_types.h
 
+noinst_HEADERS=speex_buffer.h
+
-- 
2.2.0

