diff -Naur guichan-0.8.1.orig/include/guichan/key.hpp guichan-0.8.1/include/guichan/key.hpp > guichan-0.8.1-extended-utf8-support.patch
--- guichan-0.8.1.orig/include/guichan/key.hpp	2008-04-12 14:20:09.000000000 +0200
+++ guichan-0.8.1/include/guichan/key.hpp	2009-12-22 21:36:49.280614542 +0100
@@ -125,7 +125,7 @@
             SPACE              = ' ',
             TAB                = '\t',
             ENTER              = '\n',
-            LEFT_ALT           = 1000,
+            LEFT_ALT           = -1000,
             RIGHT_ALT,
             LEFT_SHIFT,
             RIGHT_SHIFT,
