diff -up ilmbase-1.0.3/IexMath/IexMathFpu.cpp.ucontext ilmbase-1.0.3/IexMath/IexMathFpu.cpp
--- ilmbase-1.0.3/IexMath/IexMathFpu.cpp.ucontext	2012-07-26 20:51:55.000000000 +0200
+++ ilmbase-1.0.3/IexMath/IexMathFpu.cpp	2012-08-02 23:33:14.815036891 +0200
@@ -228,6 +228,7 @@ clearExceptions ()
 const uint16_t cwRestoreMask = ~((3 << 10) | (3 << 8));
 const uint16_t cwRestoreVal = (0 << 10) | (3 << 8);
 
+#if 0
 inline void
 restoreControlRegs (const ucontext_t & ucon, bool clearExceptions)
 {
@@ -235,7 +236,7 @@ restoreControlRegs (const ucontext_t & u
     setMxcsr (ucon.uc_mcontext.fpregs->mxcsr, clearExceptions);
 }
 
-#if 0
+#endif
 
 //
 // Ugly, the mxcsr isn't defined in GNU libc ucontext_t, but
@@ -255,7 +256,6 @@ restoreControlRegs (const ucontext_t & u
     setMxcsr (kfp->magic == 0 ? kfp->mxcsr : 0, clearExceptions);
 }
 
-#endif
 
 } // namespace FpuControl
 
