diff -up at-3.1.13/at.c.add at-3.1.13/at.c
diff -up at-3.1.13/panic.c.add at-3.1.13/panic.c
--- at-3.1.13/panic.c.add	2012-01-27 13:54:46.216466452 +0100
+++ at-3.1.13/panic.c	2012-01-27 13:57:35.123747498 +0100
@@ -92,8 +92,8 @@ usage(void)
 {
 /* Print usage and exit.
  */
-    fprintf(stderr, "Usage: at [-V] [-q x] [-f file] [-mlbv] timespec ...\n"
-            "       at [-V] [-q x] [-f file] [-mlbv] -t time\n"
+    fprintf(stderr, "Usage: at [-V] [-q x] [-f file] [-mMlbv] timespec ...\n"
+            "       at [-V] [-q x] [-f file] [-mMlbv] -t time\n"
     	    "       at -c job ...\n"
 	    "       atq [-V] [-q x]\n"
 	    "       at [ -rd ] job ...\n"
