diff -up at-3.1.12/at.c.opt_V at-3.1.12/at.c
--- at-3.1.12/at.c.opt_V	2009-11-23 16:11:52.000000000 +0100
+++ at-3.1.12/at.c	2009-12-02 13:20:29.770215516 +0100
@@ -853,10 +853,9 @@ main(int argc, char **argv)
      */
 
     if (disp_version) {
-	fprintf(stderr, "at version " VERSION "\n"
-	   "Please report bugs to the Debian bug tracking system (http://bugs.debian.org/)\n"
-	   "or contact the maintainers (at@packages.debian.org).\n");
-	exit(EXIT_SUCCESS);
+	fprintf(stderr, "at version " VERSION "\n");
+	if (argc == 2)
+		exit(EXIT_SUCCESS);
     }
 
     /* select our program
