diff -up at-3.1.14/at.c.shell at-3.1.14/at.c
--- at-3.1.14/at.c.shell	2013-09-26 16:57:18.126861201 +0200
+++ at-3.1.14/at.c	2013-09-26 16:59:24.069127357 +0200
@@ -62,11 +62,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-#ifdef TM_IN_SYS_TIME
 #include <sys/time.h>
-#else
 #include <time.h>
-#endif
 
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
@@ -240,6 +237,12 @@ writefile(time_t runtimer, char queue)
     int kill_errno;
     int rc;
     int mailsize = 128;
+    struct timeval tv;
+    struct timezone tz;
+    long int i;
+
+    gettimeofday(&tv, &tz);
+    srandom(getpid()+tv.tv_usec);
 
 /* Install the signal handler for SIGINT; terminate after removing the
  * spool file if necessary
@@ -450,6 +453,9 @@ writefile(time_t runtimer, char queue)
     fprintf(fp, " || {\n\t echo 'Execution directory "
 	    "inaccessible' >&2\n\t exit 1\n}\n");
 
+    i = random();
+    fprintf(fp, "${SHELL:-/bin/sh} << \'marcinDELIMITER%08lx\'\n", i);
+
     istty = isatty(fileno(stdin));
     if (istty) {
 	fprintf(stderr, "at> ");
@@ -465,7 +471,7 @@ writefile(time_t runtimer, char queue)
     if (istty) {
 	fprintf(stderr, "<EOT>\n");
     }
-    fprintf(fp, "\n");
+    fprintf(fp, "marcinDELIMITER%08lx\n", i);
     if (ferror(fp))
 	panic("Output error");
 
@@ -915,7 +921,7 @@ main(int argc, char **argv)
 	   It also alows a warning diagnostic to be printed.  Because of the
 	   possible variance, we always output the diagnostic. */
 
-	fprintf(stderr, "warning: commands will be executed using /bin/sh\n");
+       /* fprintf(stderr, "warning: commands will be executed using /bin/sh\n"); */
 
 	writefile(timer, queue);
 	break;
