diff -up at-3.1.14/panic.c.help at-3.1.14/panic.c
--- at-3.1.14/panic.c.help	2013-12-02 12:38:46.208231065 +0100
+++ at-3.1.14/panic.c	2013-12-02 12:39:34.118317528 +0100
@@ -94,6 +94,7 @@ usage(void)
  */
     fprintf(stderr, "Usage: at [-V] [-q x] [-f file] [-mMlbv] timespec ...\n"
             "       at [-V] [-q x] [-f file] [-mMlbv] -t time\n"
+            "       at [-V] [-q x] [-f file] [-mMlbv] -t time\n"
     	    "       at -c job ...\n"
 	    "       atq [-V] [-q x]\n"
 	    "       at [ -rd ] job ...\n"
