diff -up NetworkManager-0.9.2.0/libnm-util/crypto_nss.c.foo NetworkManager-0.9.2.0/libnm-util/crypto_nss.c
--- NetworkManager-0.9.2.0/libnm-util/crypto_nss.c.foo	2011-11-10 12:40:14.365009479 -0600
+++ NetworkManager-0.9.2.0/libnm-util/crypto_nss.c	2011-11-10 12:40:48.787579140 -0600
@@ -26,6 +26,11 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 
+/* Work around NSS error:
+ * /usr/include/nss3/pkcs11n.h:365:26: error: "__GNUC_MINOR" is not defined [-Werror=undef]
+ */
+#define __GNUC_MINOR __GNUC_MINOR__
+
 #include <prinit.h>
 #include <nss.h>
 #include <pk11pub.h>
