diff -dur xfce4-panel-4.6.1.orig/panel/Makefile.in xfce4-panel-4.6.1/panel/Makefile.in
--- xfce4-panel-4.6.1.orig/panel/Makefile.in	2009-04-18 17:40:09.000000000 +0200
+++ xfce4-panel-4.6.1/panel/Makefile.in	2009-06-12 00:08:17.000000000 +0200
@@ -289,7 +289,7 @@
 	-I$(top_srcdir)								\
 	-I$(top_builddir)							\
 	-DG_LOG_DOMAIN=\"xfce4-panel\"						\
-	-DDATADIR=\"$(datadir)\"						\
+	-DDATADIR=\"$(libdir)\"						\
 	-DLIBDIR=\"$(libdir)/xfce4\"						\
 	-DLIBEXECDIR=\"$(libexecdir)/xfce4\"					\
 	-DSYSCONFDIR=\"$(sysconfdir)\"						\
diff -dur xfce4-panel-4.6.1.orig/plugins/actions/Makefile.in xfce4-panel-4.6.1/plugins/actions/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/actions/Makefile.in	2009-04-18 17:40:09.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/actions/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -325,7 +325,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	actions.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/clock/Makefile.in xfce4-panel-4.6.1/plugins/clock/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/clock/Makefile.in	2009-04-18 17:40:09.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/clock/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -341,7 +341,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	clock.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/iconbox/Makefile.in xfce4-panel-4.6.1/plugins/iconbox/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/iconbox/Makefile.in	2009-04-18 17:40:10.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/iconbox/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -328,7 +328,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	iconbox.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/launcher/Makefile.in xfce4-panel-4.6.1/plugins/launcher/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/launcher/Makefile.in	2009-04-18 17:40:12.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/launcher/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -339,7 +339,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	launcher.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/pager/Makefile.in xfce4-panel-4.6.1/plugins/pager/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/pager/Makefile.in	2009-04-18 17:40:12.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/pager/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -331,7 +331,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	pager.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/separator/Makefile.in xfce4-panel-4.6.1/plugins/separator/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/separator/Makefile.in	2009-04-18 17:40:12.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/separator/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -325,7 +325,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	separator.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/showdesktop/Makefile.in xfce4-panel-4.6.1/plugins/showdesktop/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/showdesktop/Makefile.in	2009-04-18 17:40:13.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/showdesktop/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -329,7 +329,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	showdesktop.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/systray/Makefile.in xfce4-panel-4.6.1/plugins/systray/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/systray/Makefile.in	2009-04-18 17:40:13.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/systray/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -348,7 +348,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	systray.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/tasklist/Makefile.in xfce4-panel-4.6.1/plugins/tasklist/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/tasklist/Makefile.in	2009-04-18 17:40:13.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/tasklist/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -332,7 +332,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	tasklist.desktop.in.in
diff -dur xfce4-panel-4.6.1.orig/plugins/windowlist/Makefile.in xfce4-panel-4.6.1/plugins/windowlist/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/windowlist/Makefile.in	2009-04-18 17:40:14.000000000 +0200
+++ xfce4-panel-4.6.1/plugins/windowlist/Makefile.in	2009-06-12 00:17:15.000000000 +0200
@@ -363,7 +363,7 @@
 # .desktop file
 #
 desktopdir = \
-	$(datadir)/xfce4/panel-plugins
+	$(libdir)/xfce4/panel-plugins
 
 desktop_in_in_files = \
 	windowlist.desktop.in.in
