diff -Nur xfce4-settings-4.6.4.orig/dialogs/mouse-settings/Makefile.in xfce4-settings-4.6.4/dialogs/mouse-settings/Makefile.in
--- xfce4-settings-4.6.4.orig/dialogs/mouse-settings/Makefile.in	2010-01-02 06:31:10.000000000 -0700
+++ xfce4-settings-4.6.4/dialogs/mouse-settings/Makefile.in	2010-02-14 11:52:59.000000000 -0700
@@ -177,7 +177,7 @@
 INTLTOOL_PERL = @INTLTOOL_PERL@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lm
 LIBNOTIFY_CFLAGS = @LIBNOTIFY_CFLAGS@
 LIBNOTIFY_LIBS = @LIBNOTIFY_LIBS@
 LIBNOTIFY_REQUIRED_VERSION = @LIBNOTIFY_REQUIRED_VERSION@
diff -Nur xfce4-settings-4.6.4.orig/xfsettingsd/Makefile.in xfce4-settings-4.6.4/xfsettingsd/Makefile.in
--- xfce4-settings-4.6.4.orig/xfsettingsd/Makefile.in	2010-01-02 06:31:11.000000000 -0700
+++ xfce4-settings-4.6.4/xfsettingsd/Makefile.in	2010-02-14 11:54:03.000000000 -0700
@@ -173,7 +173,7 @@
 INTLTOOL_PERL = @INTLTOOL_PERL@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lX11
 LIBNOTIFY_CFLAGS = @LIBNOTIFY_CFLAGS@
 LIBNOTIFY_LIBS = @LIBNOTIFY_LIBS@
 LIBNOTIFY_REQUIRED_VERSION = @LIBNOTIFY_REQUIRED_VERSION@
