diff -up p7zip_9.13/makefile.linux_amd64_asm.strip p7zip_9.13/makefile.linux_amd64_asm
--- p7zip_9.13/makefile.linux_amd64_asm.strip	2010-06-07 18:25:38.985371722 -0300
+++ p7zip_9.13/makefile.linux_amd64_asm	2010-06-07 18:26:00.515122430 -0300
@@ -2,7 +2,7 @@
 OPTFLAGS=-O
 
 # use "-m32" to have a 32bits executable
-ALLFLAGS=-m64 ${OPTFLAGS} -pipe -s \
+ALLFLAGS=-m64 ${OPTFLAGS} -pipe \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	-D_7ZIP_LARGE_PAGES \
diff -up p7zip_9.13/makefile.linux_x86_asm_gcc_4.X.strip p7zip_9.13/makefile.linux_x86_asm_gcc_4.X
--- p7zip_9.13/makefile.linux_x86_asm_gcc_4.X.strip	2010-06-07 18:25:24.086122178 -0300
+++ p7zip_9.13/makefile.linux_x86_asm_gcc_4.X	2010-06-07 18:25:57.250122430 -0300
@@ -4,7 +4,7 @@
 
 OPTFLAGS=-O
 
-ALLFLAGS=${OPTFLAGS} -pipe -m32 -s \
+ALLFLAGS=${OPTFLAGS} -pipe -m32 \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	-D_7ZIP_LARGE_PAGES \
