diff -up ksirk-4.12.90/ksirk/iris/src/CMakeLists.txt.iris_static ksirk-4.12.90/ksirk/iris/src/CMakeLists.txt
--- ksirk-4.12.90/ksirk/iris/src/CMakeLists.txt.iris_static	2014-03-03 23:56:11.000000000 -0600
+++ ksirk-4.12.90/ksirk/iris/src/CMakeLists.txt	2014-03-20 07:16:23.751934755 -0500
@@ -127,7 +127,7 @@ xmpp/zlib/zlibdecompressor.h
 )
 
 
-kde4_add_library(iris_ksirk SHARED ${libiris_SRCS})
+kde4_add_library(iris_ksirk STATIC ${libiris_SRCS})
 
 set(iris_ksirk_libs ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY}
 ${IDN_LIBRARY} ${QCA2_LIBRARIES} ${ZLIB_LIBRARIES})
@@ -147,7 +147,7 @@ set_target_properties(iris_ksirk PROPERT
                       SOVERSION 2
                       COMPILE_FLAGS -DIRISNET_STATIC
                       )
-install(TARGETS iris_ksirk ${INSTALL_TARGETS_DEFAULT_ARGS})
+#install(TARGETS iris_ksirk ${INSTALL_TARGETS_DEFAULT_ARGS})
     
     ########### install files ###############
     
