diff -up pykde4-4.9.90/CMakeLists.txt.respect_sip_flags pykde4-4.9.90/CMakeLists.txt
--- pykde4-4.9.90/CMakeLists.txt.respect_sip_flags	2012-11-30 14:39:11.000000000 -0600
+++ pykde4-4.9.90/CMakeLists.txt	2012-12-03 18:15:25.269843812 -0600
@@ -72,6 +72,9 @@ include_directories(
     ${KDEPIMLIBS_INCLUDE_DIRS}
 )
 
+# Pull in ARM configuration options if needed
+STRING(REGEX MATCH "PyQt_qreal_double" SIP_ARM_HACK ${PYQT4_SIP_FLAGS} "")
+
 set(SIP_INCLUDES ${PYQT4_SIP_DIR} sip)
 set(SIP_CONCAT_PARTS 8)
 if (WIN32)
@@ -79,7 +82,7 @@ if (WIN32)
 else ()
     set(SIP_TAGS ALL WS_X11 ${PYQT4_VERSION_TAG})
 endif ()
-set(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug)
+SET(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug ${SIP_ARM_HACK})
 
 # Use an extra option when compiling on Python 3.
 if (PYTHON_VERSION_MAJOR GREATER 2)
