diff -up pykde4-4.13.97/CMakeLists.txt.use_system_sip_dir pykde4-4.13.97/CMakeLists.txt
--- pykde4-4.13.97/CMakeLists.txt.use_system_sip_dir	2014-07-29 12:05:46.000000000 -0500
+++ pykde4-4.13.97/CMakeLists.txt	2014-08-07 10:02:34.770363996 -0500
@@ -108,7 +108,7 @@ else ()
 endif ()
 set(SIP_DISABLE_FEATURES VendorID PyQt_NoPrintRangeBug)
 
-set(SIP_FILES_INSTALL_DIR ${SHARE_INSTALL_PREFIX}/sip)
+set(SIP_FILES_INSTALL_DIR ${SIP_DEFAULT_SIP_DIR})
 
 # Use an extra option when compiling on Python 3.
 if (PYTHON_VERSION_MAJOR GREATER 2)
