diff -up pykde4-4.12.3/kpythonpluginfactory/CMakeLists.txt.python_library_realpath pykde4-4.12.3/kpythonpluginfactory/CMakeLists.txt
--- pykde4-4.12.3/kpythonpluginfactory/CMakeLists.txt.python_library_realpath	2014-01-18 13:18:35.000000000 -0600
+++ pykde4-4.12.3/kpythonpluginfactory/CMakeLists.txt	2014-03-07 22:38:06.228544932 -0600
@@ -3,7 +3,8 @@
 set(kpythonpluginfactory_SRCS
     kpythonpluginfactory.cpp)
 
-GET_FILENAME_COMPONENT(LIB_PYTHON ${PYTHON_LIBRARIES} NAME)
+GET_FILENAME_COMPONENT(PYTHON_LIBRARY_REALPATH "${PYTHON_LIBRARY}" REALPATH)
+GET_FILENAME_COMPONENT(LIB_PYTHON ${PYTHON_LIBRARY_REALPATH} NAME)
 ADD_DEFINITIONS(-DLIB_PYTHON="${LIB_PYTHON}")
 ADD_DEFINITIONS(-DKDE_DEFAULT_DEBUG_AREA=15000)
 
