--- a/src/gui/mainwindow.h
+++ b/src/gui/mainwindow.h
@@ -215,7 +215,6 @@
     void on_actionSpeed_in_title_bar_triggered();
     void on_actionTop_tool_bar_triggered();
     void on_action_Import_Torrent_triggered();
-    void on_actionDonate_money_triggered();
     void on_actionExecution_Logs_triggered(bool checked);
     void on_actionAutoExit_qBittorrent_toggled(bool );
     void on_actionAutoSuspend_system_toggled(bool );
--- a/src/gui/mainwindow.cpp
+++ b/src/gui/mainwindow.cpp
@@ -142,7 +142,6 @@
     actionBottomPriority->setIcon(IconProvider::instance()->getIcon("go-bottom"));
     actionDelete->setIcon(IconProvider::instance()->getIcon("list-remove"));
     actionDocumentation->setIcon(IconProvider::instance()->getIcon("help-contents"));
-    actionDonate_money->setIcon(IconProvider::instance()->getIcon("wallet-open"));
     actionExit->setIcon(IconProvider::instance()->getIcon("application-exit"));
     actionIncreasePriority->setIcon(IconProvider::instance()->getIcon("go-up"));
     actionTopPriority->setIcon(IconProvider::instance()->getIcon("go-top"));
@@ -1526,11 +1525,6 @@
 }
 #endif
 
-void MainWindow::on_actionDonate_money_triggered()
-{
-    QDesktopServices::openUrl(QUrl("http://sourceforge.net/donate/index.php?group_id=163414"));
-}
-
 void MainWindow::showConnectionSettings()
 {
     on_actionOptions_triggered();
--- a/src/gui/mainwindow.ui
+++ b/src/gui/mainwindow.ui
@@ -52,7 +52,6 @@
     <addaction name="actionDocumentation"/>
     <addaction name="actionCheck_for_updates"/>
     <addaction name="separator"/>
-    <addaction name="actionDonate_money"/>
     <addaction name="actionAbout"/>
    </widget>
    <widget class="QMenu" name="menu_Options">
@@ -334,14 +333,6 @@
     <string>Import Torrent...</string>
    </property>
   </action>
-  <action name="actionDonate_money">
-   <property name="text">
-    <string>Do&amp;nate!</string>
-   </property>
-   <property name="toolTip">
-    <string>If you like qBittorrent, please donate!</string>
-   </property>
-  </action>
   <action name="actionStart_All">
    <property name="text">
