--- xdg/DesktopEntry.py.orig	2012-12-09 16:30:06.788529302 -0200
+++ xdg/DesktopEntry.py	2012-12-09 16:30:57.371868153 -0200
@@ -396,7 +396,7 @@ class DesktopEntry(IniFile):
     def checkCategories(self, value):
         values = self.getList(value)
 
-        main = ["AudioVideo", "Audio", "Video", "Development", "Education", "Game", "Graphics", "Network", "Office", "Science", "Settings", "System", "Utility"]
+        main = ["AudioVideo", "Audio", "Video", "Development", "Education", "Game", "Graphics", "Network", "Office", "Science", "Settings", "System", "Utility", "X-Wine"]
         if not any(item in main for item in values):
             self.errors.append("Missing main category")
 
