--- xdg/DesktopEntry.py.orig	2012-07-25 18:15:25.000000000 -0300
+++ xdg/DesktopEntry.py	2012-07-31 13:29:13.938788927 -0300
@@ -383,7 +383,7 @@ class DesktopEntry(IniFile):
     def checkCategories(self, value):
         values = self.getList(value)
 
-        main = ["AudioVideo", "Audio", "Video", "Development", "Education", "Game", "Graphics", "Network", "Office", "Settings", "System", "Utility"]
+        main = ["AudioVideo", "Audio", "Video", "Development", "Education", "Game", "Graphics", "Network", "Office", "Settings", "System", "Utility", "X-Wine"]
         if not any(item in main for item in values):
             self.errors.append("Missing main category")
 
