diff -Naur hddtemp-0.3-beta15.orig/src/sata.c hddtemp-0.3-beta15/src/sata.c
--- hddtemp-0.3-beta15.orig/src/sata.c	2014-03-07 19:09:31.000000000 +0100
+++ hddtemp-0.3-beta15/src/sata.c	2014-03-07 19:29:07.738284373 +0100
@@ -88,7 +88,7 @@
     return strdup(_("unknown"));
   else
   {
-    sata_fixstring(identify + 54, 24);
+    sata_fixstring(identify + 54, 40);
     return strdup(identify + 54);
   }
 }
