From 1ad51d1470069cf69eb932dc40095c2978407839 Mon Sep 17 00:00:00 2001
From: Andrea Iacovitti <aiacovitti@libero.it>
Date: Tue, 7 May 2013 23:19:32 +0200
Subject: [PATCH 5/7] Already trimmed string.

---
 khtml/ecma/xmlhttprequest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/khtml/ecma/xmlhttprequest.cpp b/khtml/ecma/xmlhttprequest.cpp
index cd72fe8..8649954 100644
--- a/khtml/ecma/xmlhttprequest.cpp
+++ b/khtml/ecma/xmlhttprequest.cpp
@@ -631,7 +631,7 @@ void XMLHttpRequest::setRequestHeader(const QString& _name, const QString& _valu
   // TODO: Do something about "put" which kio_http sort of supports and
   // the webDAV headers such as PROPFIND etc...
   if (name == "get"  || name == "post") {
-    KUrl reqURL (doc->URL(), value.trimmed());
+    KUrl reqURL(doc->URL(), value);
     open(name, reqURL, async, ec);
     return;
   }
-- 
1.8.1.4

