commit e1923b25f26b4ee3a7508ae44e49d00a2ffeeb10
Author: Albert Astals Cid <aacid@kde.org>
Date:   Sun Nov 3 16:39:43 2013 +0100

    signals -> Q_SIGNALS

diff --git a/kdecore/network/ktcpsocket.h b/kdecore/network/ktcpsocket.h
index 67cf591..c57c5e9 100644
--- a/kdecore/network/ktcpsocket.h
+++ b/kdecore/network/ktcpsocket.h
@@ -240,7 +240,7 @@ QSslError::NoSslSupport                             Never happens :)
 protected:
     virtual qint64 readData (char *data, qint64 maxSize);
     virtual qint64 writeData (const char *data, qint64 maxSize);
-signals:
+Q_SIGNALS:
     /// @since 4.8.1
     /// Forwarded from QSslSocket
     void encryptedBytesWritten( qint64 written );
