commit c892034c064da3460f0730bcd0fccbcdbace08dd
Author: Andrea Iacovitti <aiacovitti@libero.it>
Date:   Sun Nov 3 13:00:17 2013 +0100

    Use QLatin1String.

diff --git a/kio/kio/global.cpp b/kio/kio/global.cpp
index 37efe46..71fc784 100644
--- a/kio/kio/global.cpp
+++ b/kio/kio/global.cpp
@@ -1205,15 +1205,15 @@ KIO::CacheControl KIO::parseCacheControl(const QString &cacheControl)
 {
   QString tmp = cacheControl.toLower();
 
-  if (tmp == "cacheonly")
+  if (tmp == QLatin1String("cacheonly"))
      return KIO::CC_CacheOnly;
-  if (tmp == "cache")
+  if (tmp == QLatin1String("cache"))
      return KIO::CC_Cache;
-  if (tmp == "verify")
+  if (tmp == QLatin1String("verify"))
      return KIO::CC_Verify;
-  if (tmp == "refresh")
+  if (tmp == QLatin1String("refresh"))
      return KIO::CC_Refresh;
-  if (tmp == "reload")
+  if (tmp == QLatin1String("reload"))
      return KIO::CC_Reload;
 
   kDebug() << "unrecognized Cache control option:"<<cacheControl;
