commit c8449628514d4bf6013d257520c859f4393f498a
Author: Lukáš Tinkl <lukas@kde.org>
Date:   Mon Aug 26 16:57:36 2013 +0200

    emit deviceRemoved for optical disks
    
    BUG: 324031

diff --git a/solid/solid/backends/udisks2/udisksmanager.cpp b/solid/solid/backends/udisks2/udisksmanager.cpp
index 373db81..d2fda69 100644
--- a/solid/solid/backends/udisks2/udisksmanager.cpp
+++ b/solid/solid/backends/udisks2/udisksmanager.cpp
@@ -220,7 +220,7 @@ void Manager::slotInterfacesRemoved(const QDBusObjectPath &object_path, const QS
 
     Device device(udi);
 
-    if (!udi.isEmpty() && (interfaces.isEmpty() || device.interfaces().isEmpty())) {
+    if (!udi.isEmpty() && (interfaces.isEmpty() || device.interfaces().isEmpty() || device.mightBeOpticalDisc())) {
         Q_EMIT deviceRemoved(udi);
         m_deviceCache.removeAll(udi);
         DeviceBackend::destroyBackend(udi);
