From 2173580f070e806d4715e13048c697c49ec262e2 Mon Sep 17 00:00:00 2001
From: Aaron Seigo <aseigo@kde.org>
Date: Thu, 21 Feb 2013 17:59:58 +0100
Subject: [PATCH 047/111] coding style fixes

---
 kdeui/icons/kiconloader.cpp | 27 ++++++++++++---------------
 1 file changed, 12 insertions(+), 15 deletions(-)

diff --git a/kdeui/icons/kiconloader.cpp b/kdeui/icons/kiconloader.cpp
index 6fed667..dba474d 100644
--- a/kdeui/icons/kiconloader.cpp
+++ b/kdeui/icons/kiconloader.cpp
@@ -938,32 +938,29 @@ K3Icon KIconLoaderPrivate::findMatchingIcon(const QString& name, int size) const
         }
     }
 
-    foreach (KIconThemeNode *themeNode, links)
-    {
+    foreach (KIconThemeNode *themeNode, links) {
         QString currentName = name;
 
-        while (!currentName.isEmpty())
-        {
-
+        while (!currentName.isEmpty()) {
             //kDebug(264) << "Looking up" << currentName;
 
-// The following code has been commented out because the Qt SVG renderer needs
-// to be improved. If you are going to change/remove some code from this part,
-// please contact me before (ereslibre@kde.org), or kde-core-devel@kde.org. (ereslibre)
-            for (int i = 0 ; i < 4 ; i++)
-            {
+            for (int i = 0 ; i < 4 ; i++) {
                 icon = themeNode->theme->iconPath(currentName + ext[i], size, KIconLoader::MatchExact);
-                if (icon.isValid())
-                    return icon;
+                if (icon.isValid()) {
+                    break;
+                }
 
                 icon = themeNode->theme->iconPath(currentName + ext[i], size, KIconLoader::MatchBest);
-                if (icon.isValid())
-                    return icon;
+                if (icon.isValid()) {
+                    break;
+                }
             }
+            //kDebug(264) << "Looking up" << currentName;
 
-            if (genericFallback)
+            if (genericFallback) {
                 // we already tested the base name
                 break;
+            }
 
             int rindex = currentName.lastIndexOf('-');
             if (rindex > 1) { // > 1 so that we don't split x-content or x-epoc
-- 
1.8.1.4

