Index: khtml/misc/loader.cpp
===================================================================
--- khtml/misc/loader.cpp
+++ khtml/misc/loader.cpp	2010-05-05 00:20:49.000000000 +0200
@@ -1215,6 +1215,13 @@
          ( doRedirectCheck && (m_doc && !KAuthorized::authorizeUrlAction("redirect", m_doc->URL(), fullURL)))) \
          return 0L;
 
+bool DocLoader::willLoadMediaElement( const DOM::DOMString &url)
+{
+    DOCLOADER_SECCHECK(true);
+
+    return true;
+}
+
 CachedImage *DocLoader::requestImage( const DOM::DOMString &url)
 {
     DOCLOADER_SECCHECK(true);
Index: khtml/misc/loader.h
===================================================================
--- khtml/misc/loader.h
+++ khtml/misc/loader.h	2010-05-05 00:20:01.000000000 +0200
@@ -429,6 +429,7 @@
         CachedScript *requestScript( const DOM::DOMString &url, const QString& charset);
         CachedSound *requestSound( const DOM::DOMString &url );
         CachedFont *requestFont( const DOM::DOMString &url );
+        bool willLoadMediaElement( const DOM::DOMString &url);
 
 	bool autoloadImages() const { return m_bautoloadImages; }
         KIO::CacheControl cachePolicy() const { return m_cachePolicy; }
Index: khtml/rendering/render_media.cpp
===================================================================
--- khtml/rendering/render_media.cpp
+++ khtml/rendering/render_media.cpp	2010-05-05 00:21:58.000000000 +0200
@@ -97,9 +97,12 @@
 
     QUrl url = mediaElement()->src().string();
     if (player()->mediaObject()->currentSource().url() != url) {
-        player()->stop();
-	player()->mediaObject()->setCurrentSource(url);
-        if (mediaElement()->autoplay()) player()->play(url);
+        if (!document()->docLoader()->willLoadMediaElement(mediaElement()->src()))
+            return;
+        if (mediaElement()->autoplay()) 
+            player()->play(url);
+        else
+            player()->load(url); 
     }
 }
 
