diff -up kdelibs-4.6.80/kio/kio/kprotocolmanager.cpp.branding kdelibs-4.6.80/kio/kio/kprotocolmanager.cpp
--- kdelibs-4.6.80/kio/kio/kprotocolmanager.cpp.branding	2011-05-20 22:24:54.000000000 +0200
+++ kdelibs-4.6.80/kio/kio/kprotocolmanager.cpp	2011-05-27 14:18:43.999356195 +0200
@@ -645,6 +645,7 @@ QString KProtocolManager::defaultUserAge
     d->useragent += QString::number(KDE::versionMajor());
     d->useragent += QL1C('.');
     d->useragent += QString::number(KDE::versionMinor());
+    d->useragent += QL1S(" _KDELIBS_SLK_DIST");
   }
   else
   {
