Index: kdelibs/khtml/html/dtd.cpp
===================================================================
--- kdelibs/khtml/html/dtd.cpp	(révision 1122691)
+++ kdelibs/khtml/html/dtd.cpp	(révision 1122692)
@@ -614,6 +614,8 @@
         if(childID == ID_OPTION) return true;
         return false;
     case ID_OPTION:
+        if(childID == ID_SCRIPT) return true;
+        // fallthrough intentional
     case ID_TEXTAREA:
     case ID_TITLE:
     case ID_STYLE:
