diff -up kdelibs-4.6.4/kdeui/notifications/kstatusnotifieritemdbus_p.cpp.kstatusnotifieritemdbus_leak kdelibs-4.6.4/kdeui/notifications/kstatusnotifieritemdbus_p.cpp
--- kdelibs-4.6.4/kdeui/notifications/kstatusnotifieritemdbus_p.cpp.kstatusnotifieritemdbus_leak	2011-04-01 08:55:50.000000000 -0500
+++ kdelibs-4.6.4/kdeui/notifications/kstatusnotifieritemdbus_p.cpp	2011-06-30 14:45:28.256526029 -0500
@@ -174,7 +174,9 @@ KStatusNotifierItemDBus::KStatusNotifier
 
 KStatusNotifierItemDBus::~KStatusNotifierItemDBus()
 {
+    m_dbus.unregisterObject("/StatusNotifierItem");
     m_dbus.unregisterService(m_service);
+    m_dbus.disconnectFromBus(m_service);
 }
 
 QDBusConnection KStatusNotifierItemDBus::dbusConnection() const
