--- branches/KDE/4.5/kdelibs/plasma/tooltipmanager.cpp	2010/09/28 01:19:58	1180436
+++ branches/KDE/4.5/kdelibs/plasma/tooltipmanager.cpp	2010/10/08 11:35:52	1183776
@@ -236,10 +236,12 @@
             }
         }
 
-        d->tipWidget->setContent(widget, data);
-        d->tipWidget->prepareShowing();
-        if (m_corona) {
-            d->tipWidget->moveTo(m_corona->popupPosition(widget, d->tipWidget->size(), Qt::AlignCenter));
+        if (d->tipWidget) {
+            d->tipWidget->setContent(widget, data);
+            d->tipWidget->prepareShowing();
+            if (m_corona) {
+                d->tipWidget->moveTo(m_corona->popupPosition(widget, d->tipWidget->size(), Qt::AlignCenter));
+            }
         }
     }
 }
