diff -up kdelibs-4.12.90/doc/common/Doxyfile.global.dot kdelibs-4.12.90/doc/common/Doxyfile.global
--- kdelibs-4.12.90/doc/common/Doxyfile.global.dot	2014-03-17 13:15:23.252517997 -0500
+++ kdelibs-4.12.90/doc/common/Doxyfile.global	2014-03-17 13:16:02.472100942 -0500
@@ -1360,7 +1360,7 @@ HIDE_UNDOC_RELATIONS   = NO
 # toolkit from AT&T and Lucent Bell Labs. The other options in this section
 # have no effect if this option is set to NO (the default)
 
-HAVE_DOT               = YES
+HAVE_DOT               = NO
 
 # The DOT_FONTSIZE tag can be used to set the size of the font of dot graphs.
 # The default size is 10pt.
