diff -up kdelibs-4.14.17/ConfigureChecks.cmake.gcc6 kdelibs-4.14.17/ConfigureChecks.cmake
--- kdelibs-4.14.17/ConfigureChecks.cmake.gcc6	2016-02-11 23:06:08.000000000 -0600
+++ kdelibs-4.14.17/ConfigureChecks.cmake	2016-02-12 19:14:07.707788177 -0600
@@ -244,7 +244,7 @@ check_prototype_exists(unsetenv stdlib.h
 check_prototype_exists(usleep unistd.h              HAVE_USLEEP_PROTO)
 check_prototype_exists(initgroups "unistd.h;sys/types.h;unistd.h;grp.h" HAVE_INITGROUPS_PROTO)
 check_prototype_exists(setreuid unistd.h            HAVE_SETREUID_PROTO)
-check_prototype_exists(trunc math.h                 HAVE_TRUNC)
+check_prototype_exists(truncf math.h                HAVE_TRUNC)
 
 # check for existing datatypes
 
