diff -up kdelibs-4.8.3/experimental/libkdeclarative/CMakeLists.txt.kdeclarative-install-location kdelibs-4.8.3/experimental/libkdeclarative/CMakeLists.txt
--- kdelibs-4.8.3/experimental/libkdeclarative/CMakeLists.txt.kdeclarative-install-location	2012-04-29 22:37:35.777207066 +0200
+++ kdelibs-4.8.3/experimental/libkdeclarative/CMakeLists.txt	2012-04-30 12:20:40.384393056 +0200
@@ -6,6 +6,9 @@ endif(KDE_PLATFORM_FEATURE_BINARY_COMPAT
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config-kdeclarative.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-kdeclarative.h)
 
+set (LIB_SUFFIX "" CACHE STRING "Define suffix of library directory name (eg. '64')")
+set( LIB_INSTALL_DIR lib${LIB_SUFFIX} )
+
 include_directories(
    ${CMAKE_SOURCE_DIR}
    ${CMAKE_BINARY_DIR}
