Index: solid/solid/CMakeLists.txt
===================================================================
--- solid/solid/CMakeLists.txt	(revision 1211253)
+++ solid/solid/CMakeLists.txt	(working copy)
@@ -39,7 +39,6 @@
 
 file(MAKE_DIRECTORY
    ${CMAKE_CURRENT_BINARY_DIR}/backends/fakehw
-   ${CMAKE_CURRENT_BINARY_DIR}/backends/hal
    ${CMAKE_CURRENT_BINARY_DIR}/backends/udev
    ${CMAKE_CURRENT_BINARY_DIR}/backends/wmi
 )
@@ -195,33 +194,6 @@
    endif( UDEV_FOUND )
 
 
-   message(STATUS "Building Solid HAL backend." )
-   set(solid_LIB_SRCS ${solid_LIB_SRCS}
-   backends/hal/halacadapter.cpp
-   backends/hal/halaudiointerface.cpp
-   backends/hal/halbattery.cpp
-   backends/hal/halblock.cpp
-   backends/hal/halbutton.cpp
-   backends/hal/halcamera.cpp
-   backends/hal/halcdrom.cpp
-   backends/hal/haldeviceinterface.cpp
-   backends/hal/haldvbinterface.cpp
-   backends/hal/halfstabhandling.cpp
-   backends/hal/halgenericinterface.cpp
-   backends/hal/haldevice.cpp
-   backends/hal/halmanager.cpp
-   backends/hal/halnetworkinterface.cpp
-   backends/hal/halserialinterface.cpp
-   backends/hal/halopticaldisc.cpp
-   backends/hal/halportablemediaplayer.cpp
-   backends/hal/halprocessor.cpp
-   backends/hal/halstorageaccess.cpp
-   backends/hal/halstorage.cpp
-   backends/hal/halvideo.cpp
-   backends/hal/halvolume.cpp
-   backends/hal/halsmartcardreader.cpp
-   )
-
    message(STATUS "Building Solid UPower backend." )
    set(solid_LIB_SRCS ${solid_LIB_SRCS}
    backends/upower/upowermanager.cpp
@@ -249,15 +221,6 @@
       )
    endif (CMAKE_SYSTEM_NAME MATCHES Linux)
 
-   message(STATUS "Building Solid fstab backend." )
-   set(solid_LIB_SRCS ${solid_LIB_SRCS}
-   backends/fstab/fstabmanager.cpp
-   backends/fstab/fstabdevice.cpp
-   backends/fstab/fstabstorageaccess.cpp
-   backends/fstab/fstabhandling.cpp
-   backends/fstab/fstabwatcher.cpp
-   )
-
 endif(NOT WIN32 AND NOT APPLE)
 
 if(APPLE)
Index: solid/solid/managerbase.cpp
===================================================================
--- solid/solid/managerbase.cpp	(revision 1211253)
+++ solid/solid/managerbase.cpp	(working copy)
@@ -30,7 +30,6 @@
 #if defined (Q_OS_MAC)
 #include "backends/iokit/iokitmanager.h"
 #elif defined (Q_OS_UNIX)
-#include "backends/hal/halmanager.h"
 #include "backends/udisks/udisksmanager.h"
 #include "backends/upower/upowermanager.h"
 
@@ -42,8 +41,6 @@
 #include "backends/udev/udevmanager.h"
 #endif
 
-#include "backends/fstab/fstabmanager.h"
-
 #elif defined (Q_WS_WIN) && defined(HAVE_WBEM) && !defined(_WIN32_WCE)
 #include "backends/wmi/wmimanager.h"
 #endif
@@ -71,22 +68,12 @@
 #        elif defined(Q_WS_WIN) && defined(HAVE_WBEM) && !defined(_WIN32_WCE)
             m_backends << new Solid::Backends::Wmi::WmiManager(0);
 
-#        elif defined(Q_OS_UNIX) && !defined(Q_OS_LINUX)
-            m_backends << new Solid::Backends::Hal::HalManager(0);
-
 #        elif defined(Q_OS_LINUX)
-            bool solidHalLegacyEnabled
-                = QString::fromLocal8Bit(qgetenv("SOLID_HAL_LEGACY")).toInt()==1;
-            if (solidHalLegacyEnabled) {
-                m_backends << new Solid::Backends::Hal::HalManager(0);
-            } else {
 #               if defined(UDEV_FOUND)
                     m_backends << new Solid::Backends::UDev::UDevManager(0);
 #               endif
                 m_backends << new Solid::Backends::UDisks::UDisksManager(0)
-                           << new Solid::Backends::UPower::UPowerManager(0)
-                           << new Solid::Backends::Fstab::FstabManager(0);
-            }
+                           << new Solid::Backends::UPower::UPowerManager(0);
 #        endif
 
 #        if defined (HUPNP_FOUND)
