diff -up kdelibs-4.6.80/solid/solid/CMakeLists.txt.halectomy kdelibs-4.6.80/solid/solid/CMakeLists.txt
--- kdelibs-4.6.80/solid/solid/CMakeLists.txt.halectomy	2011-05-20 22:24:54.000000000 +0200
+++ kdelibs-4.6.80/solid/solid/CMakeLists.txt	2011-05-30 14:55:43.380878553 +0200
@@ -39,7 +39,6 @@ configure_file(config-processor.h.cmake 
 
 file(MAKE_DIRECTORY
    ${CMAKE_CURRENT_BINARY_DIR}/backends/fakehw
-   ${CMAKE_CURRENT_BINARY_DIR}/backends/hal
    ${CMAKE_CURRENT_BINARY_DIR}/backends/udev
    ${CMAKE_CURRENT_BINARY_DIR}/backends/wmi
 )
@@ -199,33 +198,6 @@ if(NOT WIN32 AND NOT APPLE)
    endif( UDEV_FOUND )
 
 
-   message(STATUS "Building Solid HAL backend." )
-   set(solid_LIB_SRCS ${solid_LIB_SRCS}
-   backends/hal/halacadapter.cpp
-   backends/hal/halaudiointerface.cpp
-   backends/hal/halbattery.cpp
-   backends/hal/halblock.cpp
-   backends/hal/halbutton.cpp
-   backends/hal/halcamera.cpp
-   backends/hal/halcdrom.cpp
-   backends/hal/haldeviceinterface.cpp
-   backends/hal/haldvbinterface.cpp
-   backends/hal/halfstabhandling.cpp
-   backends/hal/halgenericinterface.cpp
-   backends/hal/haldevice.cpp
-   backends/hal/halmanager.cpp
-   backends/hal/halnetworkinterface.cpp
-   backends/hal/halserialinterface.cpp
-   backends/hal/halopticaldisc.cpp
-   backends/hal/halportablemediaplayer.cpp
-   backends/hal/halprocessor.cpp
-   backends/hal/halstorageaccess.cpp
-   backends/hal/halstorage.cpp
-   backends/hal/halvideo.cpp
-   backends/hal/halvolume.cpp
-   backends/hal/halsmartcardreader.cpp
-   )
-
    message(STATUS "Building Solid UPower backend." )
    set(solid_LIB_SRCS ${solid_LIB_SRCS}
    backends/upower/upowermanager.cpp
@@ -253,16 +225,6 @@ if(NOT WIN32 AND NOT APPLE)
       )
    endif (CMAKE_SYSTEM_NAME MATCHES Linux)
 
-   message(STATUS "Building Solid fstab backend." )
-   set(solid_LIB_SRCS ${solid_LIB_SRCS}
-   backends/fstab/fstabmanager.cpp
-   backends/fstab/fstabdevice.cpp
-   backends/fstab/fstabnetworkshare.cpp
-   backends/fstab/fstabstorageaccess.cpp
-   backends/fstab/fstabhandling.cpp
-   backends/fstab/fstabwatcher.cpp
-   )
-
 endif(NOT WIN32 AND NOT APPLE)
 
 if(APPLE)
diff -up kdelibs-4.6.80/solid/solid/managerbase.cpp.halectomy kdelibs-4.6.80/solid/solid/managerbase.cpp
--- kdelibs-4.6.80/solid/solid/managerbase.cpp.halectomy	2011-05-20 22:24:54.000000000 +0200
+++ kdelibs-4.6.80/solid/solid/managerbase.cpp	2011-05-30 14:55:43.381878481 +0200
@@ -30,7 +30,6 @@
 #if defined (Q_OS_MAC)
 #include "backends/iokit/iokitmanager.h"
 #elif defined (Q_OS_UNIX)
-#include "backends/hal/halmanager.h"
 #include "backends/udisks/udisksmanager.h"
 #include "backends/upower/upowermanager.h"
 
@@ -42,8 +41,6 @@
 #include "backends/udev/udevmanager.h"
 #endif
 
-#include "backends/fstab/fstabmanager.h"
-
 #elif defined (Q_WS_WIN) && defined(HAVE_WBEM) && !defined(_WIN32_WCE)
 #include "backends/wmi/wmimanager.h"
 #endif
@@ -71,22 +68,12 @@ void Solid::ManagerBasePrivate::loadBack
 #        elif defined(Q_WS_WIN) && defined(HAVE_WBEM) && !defined(_WIN32_WCE)
             m_backends << new Solid::Backends::Wmi::WmiManager(0);
 
-#        elif defined(Q_OS_UNIX) && !defined(Q_OS_LINUX)
-            m_backends << new Solid::Backends::Hal::HalManager(0);
-
 #        elif defined(Q_OS_LINUX)
-            bool solidHalLegacyEnabled
-                = QString::fromLocal8Bit(qgetenv("SOLID_HAL_LEGACY")).toInt()==1;
-            if (solidHalLegacyEnabled) {
-                m_backends << new Solid::Backends::Hal::HalManager(0);
-            } else {
 #               if defined(UDEV_FOUND)
                     m_backends << new Solid::Backends::UDev::UDevManager(0);
 #               endif
                 m_backends << new Solid::Backends::UDisks::UDisksManager(0)
-                           << new Solid::Backends::UPower::UPowerManager(0)
-                           << new Solid::Backends::Fstab::FstabManager(0);
-            }
+                           << new Solid::Backends::UPower::UPowerManager(0);
 #        endif
 
 #        if defined (HUPNP_FOUND)
diff -up kdelibs-4.6.80/solid/tests/CMakeLists.txt.halectomy kdelibs-4.6.80/solid/tests/CMakeLists.txt
--- kdelibs-4.6.80/solid/tests/CMakeLists.txt.halectomy	2011-05-20 22:24:54.000000000 +0200
+++ kdelibs-4.6.80/solid/tests/CMakeLists.txt	2011-05-30 15:13:39.179352524 +0200
@@ -15,21 +15,6 @@ endif(WIN32)
 target_link_libraries(fakehardwaretest solid_static ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTTEST_LIBRARY} )
 add_definitions(-DTEST_DATA="\\"${CMAKE_CURRENT_SOURCE_DIR}/../solid/backends/fakehw/fakecomputer.xml\\"")
 
-
-########### halbasictest ###############
-
-if(NOT WIN32 AND NOT APPLE)
-set(halbasictest_SRCS halbasictest.cpp )
-
-kde4_add_executable(halbasictest ${halbasictest_SRCS})
-
-if(WIN32)
-  set_target_properties(halbasictest PROPERTIES COMPILE_FLAGS -DSOLID_EXPORT=)
-endif(WIN32)
-
-target_link_libraries(halbasictest solid_static ${KDEWIN_LIBRARIES} ${QT_QTCORE_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTTEST_LIBRARY} )
-endif(NOT WIN32 AND NOT APPLE)
-
 ########### solidhwtest ###############
 
 set(solidhwtest_SRCS 
