diff -up kdelibs-4.10.3/plasma/private/dialogshadows.cpp.kde319137 kdelibs-4.10.3/plasma/private/dialogshadows.cpp
--- kdelibs-4.10.3/plasma/private/dialogshadows.cpp.kde319137	2013-05-02 23:41:11.209607748 -0500
+++ kdelibs-4.10.3/plasma/private/dialogshadows.cpp	2013-05-09 07:54:48.329139462 -0500
@@ -335,6 +335,8 @@ void DialogShadows::Private::setupData(P
 void DialogShadows::Private::freeX11Pixmaps()
 {
 #ifdef Q_WS_X11
+    if (!QX11Info::display())
+        return;
     foreach (const QPixmap &pixmap, m_shadowPixmaps) {
         if (!pixmap.isNull()) {
             XFreePixmap(QX11Info::display(), pixmap.handle());
