diff -up kdelibs-4.8.0/kdecore/kernel/kstandarddirs.cpp.libexecdir kdelibs-4.8.0/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.8.0/kdecore/kernel/kstandarddirs.cpp.libexecdir	2012-01-10 15:17:51.000000000 +0100
+++ kdelibs-4.8.0/kdecore/kernel/kstandarddirs.cpp	2012-01-19 13:38:07.424309462 +0100
@@ -1871,7 +1871,7 @@ void KStandardDirs::addKDEDefaults()
         addResourceType(types_string + types_indices[index], 0, types_string + types_indices[index+1], true);
         index+=2;
     }
-    addResourceType("exe", "lib", "kde4/libexec", true );
+    addResourceType("exe", 0, "libexec/kde4", true );
 
     addResourceDir("home", QDir::homePath(), false);
 
diff -up kdelibs-4.8.0/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir kdelibs-4.8.0/kdecore/kernel/kstandarddirs_unix.cpp
--- kdelibs-4.8.0/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir	2011-07-27 20:34:33.000000000 +0200
+++ kdelibs-4.8.0/kdecore/kernel/kstandarddirs_unix.cpp	2012-01-19 13:38:07.424309462 +0100
@@ -63,7 +63,7 @@ QString KStandardDirs::installPath(const
             if (strcmp("lib", type) == 0)
                 return QFile::decodeName(LIB_INSTALL_DIR "/");
             if (strcmp("libexec", type) == 0)
-                return QFile::decodeName(KDEDIR "/lib" KDELIBSUFF "/kde4/libexec/");
+                return QFile::decodeName(LIBEXEC_INSTALL_DIR "/");
             if (strcmp("locale", type) == 0)
                 return QFile::decodeName(LOCALE_INSTALL_DIR "/");
             break;
diff -up kdelibs-4.8.0/kdecore/tests/kstandarddirstest.cpp.libexecdir kdelibs-4.8.0/kdecore/tests/kstandarddirstest.cpp
--- kdelibs-4.8.0/kdecore/tests/kstandarddirstest.cpp.libexecdir	2012-01-18 19:06:22.000000000 +0100
+++ kdelibs-4.8.0/kdecore/tests/kstandarddirstest.cpp	2012-01-19 13:42:17.955103619 +0100
@@ -96,8 +96,9 @@ void KStandarddirsTest::testFindResource
 #define KIOSLAVE "bin/kioslave.exe"
 #else
 #define EXT ""
-#define KIOSLAVE "kde4/libexec/kioslave"
+#define KIOSLAVE "libexec/kde4/kioslave"
 #endif
+
     const QString bin = KGlobal::dirs()->findResource( "exe", "kioslave" EXT );
     QVERIFY( !bin.isEmpty() );
     QVERIFY( bin.endsWith( KIOSLAVE ) );
@@ -231,7 +232,8 @@ void KStandarddirsTest::testFindExe()
     // findExe with a result in libexec
     const QString lnusertemp = KGlobal::dirs()->findExe( "lnusertemp" );
     QVERIFY( !lnusertemp.isEmpty() );
-    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) );
+    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) 
+        || lnusertemp.endsWith( "libexec/kde4/lnusertemp" EXT, PATH_SENSITIVITY ) );
 #endif
 
 #ifndef Q_OS_MAC // kdeinit4 is a bundle on Mac, so the below doesn't work
diff -up kdelibs-4.8.0/kio/tests/krununittest.cpp.libexecdir kdelibs-4.8.0/kio/tests/krununittest.cpp
--- kdelibs-4.8.0/kio/tests/krununittest.cpp.libexecdir	2011-07-27 20:34:37.000000000 +0200
+++ kdelibs-4.8.0/kio/tests/krununittest.cpp	2012-01-19 13:38:07.427309472 +0100
@@ -160,7 +160,7 @@ void KRunUnitTest::testProcessDesktopExe
     if (kmailservice.isEmpty()) kmailservice = "kmailservice";
     if (!kdeinit.isEmpty()) {
         QVERIFY(!kmailservice.isEmpty());
-        QVERIFY(kmailservice.contains("kde4/libexec"));
+        QVERIFY(kmailservice.contains("kde4/libexec") || kmailservice.contains("libexec/kde4"));
     }
 
     QTest::newRow("%U l0") << "kdeinit4 %U" << l0 << false << kdeinit;
