--- trunk/KDE/kdelibs/plasma/containment.cpp	2010/10/14 15:42:21	1185911
+++ trunk/KDE/kdelibs/plasma/containment.cpp	2010/10/14 15:46:05	1185912
@@ -2079,6 +2079,17 @@
     Applet::showConfigurationInterface();
 }
 
+void Containment::configChanged()
+{
+    if (d->drawWallpaper) {
+        KConfigGroup group = config();
+        setWallpaper(group.readEntry("wallpaperplugin", defaultWallpaper),
+                     group.readEntry("wallpaperpluginmode", defaultWallpaperMode));
+    }
+
+    Applet::configChanged();
+}
+
 void ContainmentPrivate::requestConfiguration()
 {
     emit q->configureRequested(q);
--- trunk/KDE/kdelibs/plasma/containment.h	2010/10/14 15:42:21	1185911
+++ trunk/KDE/kdelibs/plasma/containment.h	2010/10/14 15:46:05	1185912
@@ -527,6 +527,13 @@
          */
         void showConfigurationInterface();
 
+        /**
+         * Called when applet configuration values have changed.
+         * @reimp
+         * @sa Applet::configChanged()
+         */
+        void configChanged();
+
     protected:
         /**
          * Sets the type of this containment.
