diff -up kdelibs-4.13.2/kdecore/kernel/ktoolinvocation_x11.cpp.orig kdelibs-4.13.2/kdecore/kernel/ktoolinvocation_x11.cpp
--- kdelibs-4.13.2/kdecore/kernel/ktoolinvocation_x11.cpp.orig	2014-07-11 16:54:51.000000000 +0200
+++ kdelibs-4.13.2/kdecore/kernel/ktoolinvocation_x11.cpp	2014-07-11 16:57:31.000000000 +0200
@@ -412,11 +412,9 @@ void KToolInvocation::invokeTerminal(con
         // directory before launching them, see below.
     }
 
-    QString error;
-    if (self()->startServiceInternal("kdeinit_exec_with_workdir",
-                                    cmd, cmdTokens, &error, 0, NULL, startup_id, false, workdir)) {
+    if (!QProcess::startDetached(cmd, cmdTokens)) {
       KMessage::message(KMessage::Error,
-                      i18n("Could not launch the terminal client:\n\n%1", error),
+                      i18n("Could not launch the terminal client"),
                       i18n("Could not launch Terminal Client"));
     }
 }
