diff -up kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.cpp.kde264487 kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.cpp
--- kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.cpp.kde264487	2011-02-25 16:00:34.000000000 -0600
+++ kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.cpp	2011-03-23 14:46:15.767586207 -0500
@@ -162,9 +162,9 @@ out:
 using namespace Solid::Backends::UDisks;
 
 OpticalDisc::OpticalDisc(UDisksDevice *device)
-    : UDisksStorageVolume(device)
+    : UDisksStorageVolume(device), m_needsReprobe(true), m_cachedContent(Solid::OpticalDisc::NoContent)
 {
-
+    connect(device, SIGNAL(changed()), this, SLOT(slotChanged()));
 }
 
 OpticalDisc::~OpticalDisc()
@@ -228,21 +228,35 @@ Solid::OpticalDisc::DiscType OpticalDisc
 
 Solid::OpticalDisc::ContentTypes OpticalDisc::availableContent() const
 {
-    Solid::OpticalDisc::ContentTypes content = Solid::OpticalDisc::NoContent;
-
-    if (!isBlank())
+    if (isBlank()) {
+        m_needsReprobe = false;
+        return Solid::OpticalDisc::NoContent;
+    }
+    
+    if (m_needsReprobe)
     {
+        m_cachedContent = Solid::OpticalDisc::NoContent;
+
         bool hasData = m_device->prop("OpticalDiscNumTracks").toInt() > 0;
         bool hasAudio = m_device->prop("OpticalDiscNumAudioTracks").toInt() > 0;
 
         if ( hasData )
-            content |= Solid::OpticalDisc::Data;
+            m_cachedContent |= Solid::OpticalDisc::Data;
         if ( hasAudio )
-            content |= Solid::OpticalDisc::Audio;
+            m_cachedContent |= Solid::OpticalDisc::Audio;
+
+        m_cachedContent |= advancedDiscDetect(m_device->property("DeviceFile").toString());
 
-        content |= advancedDiscDetect(m_device->prop("DeviceFile").toString());
+        m_needsReprobe = false;
     }
 
-    return content;
+    return m_cachedContent;
 }
 
+void Solid::Backends::UDisks::OpticalDisc::slotChanged()
+{
+    m_needsReprobe = true;
+    m_cachedContent = Solid::OpticalDisc::NoContent;
+}
+
+
diff -up kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.h.kde264487 kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.h
--- kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.h.kde264487	2011-02-25 15:53:32.000000000 -0600
+++ kdelibs-4.6.1/solid/solid/backends/udisks/udisksopticaldisc.h	2011-03-23 14:41:27.523063466 -0500
@@ -48,6 +48,13 @@ public:
     virtual bool isAppendable() const;
     virtual Solid::OpticalDisc::DiscType discType() const;
     virtual Solid::OpticalDisc::ContentTypes availableContent() const;
+
+private slots:
+    void slotChanged();
+
+private:
+    mutable bool m_needsReprobe;
+    mutable Solid::OpticalDisc::ContentTypes m_cachedContent;
 };
 
 }
