diff -up kdelibs-4.4.80/cmake/modules/FindKDE4Internal.cmake.no_rpath kdelibs-4.4.80/cmake/modules/FindKDE4Internal.cmake
--- kdelibs-4.4.80/cmake/modules/FindKDE4Internal.cmake.no_rpath	2010-05-21 10:00:15.000000000 +0200
+++ kdelibs-4.4.80/cmake/modules/FindKDE4Internal.cmake	2010-05-21 10:09:42.195154891 +0200
@@ -1014,12 +1014,20 @@ if (UNIX)
 
       list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${LIB_INSTALL_DIR}" _isSystemLibDir)
       if("${_isSystemLibDir}" STREQUAL "-1")
-         set(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}")
+         set(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}")
+         #message(STATUS "CMAKE_SYSTEM_LIBRARY_PATH: ${CMAKE_SYSTEM_LIBRARY_PATH}")
+         #message(STATUS "LIB_INSTALL_DIR: ${LIB_INSTALL_DIR}")
+         #list(FIND CMAKE_SYSTEM_LIBRARY_PATH ${LIB_INSTALL_DIR} _LIB_INSTALL_DIR_INDEX_IN_CMAKE_SYSTEM_LIBRARY_PATH)
+         #message(STATUS "_LIB_INSTALL_DIR_INDEX_IN_CMAKE_SYSTEM_LIBRARY_PATH: ${_LIB_INSTALL_DIR_INDEX_IN_CMAKE_SYSTEM_LIBRARY_PATH}")
+         #if(${_LIB_INSTALL_DIR_INDEX_IN_CMAKE_SYSTEM_LIBRARY_PATH} EQUAL -1)
+         #   message(STATUS "appending to CMAKE_INSTALL_RPATH: ${LIB_INSTALL_DIR}")
+         #   list(APPEND CMAKE_INSTALL_RPATH ${LIB_INSTALL_DIR})
+         #endif(${_LIB_INSTALL_DIR_INDEX_IN_CMAKE_SYSTEM_LIBRARY_PATH} EQUAL -1)
       endif("${_isSystemLibDir}" STREQUAL "-1")
 
       set(CMAKE_SKIP_BUILD_RPATH FALSE)
       set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
-      set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+      set(CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
    endif (APPLE)
 endif (UNIX)
 
