diff -up kdelibs-4.14.4/kdecore/services/kmimetyperepository.cpp.mimespam kdelibs-4.14.4/kdecore/services/kmimetyperepository.cpp
--- kdelibs-4.14.4/kdecore/services/kmimetyperepository.cpp.mimespam	2015-01-06 17:55:23.000000000 -0600
+++ kdelibs-4.14.4/kdecore/services/kmimetyperepository.cpp	2015-01-27 08:39:12.461004022 -0600
@@ -317,7 +317,7 @@ QStringList KMimeTypeRepository::parents
                     const QString derivedTypeName = line.left(pos);
                     KMimeType::Ptr derivedType = findMimeTypeByName(derivedTypeName, KMimeType::ResolveAliases);
                     if (!derivedType)
-                        kWarning(7012) << fileName << " refers to unknown mimetype " << derivedTypeName;
+                        kDebug(7012) << fileName << " refers to unknown mimetype " << derivedTypeName;
                     else {
                         const QString parentTypeName = line.mid(pos+1);
                         Q_ASSERT(!parentTypeName.isEmpty());
