diff -up kdelibs-4.8.4/nepomuk/core/nepomukmainmodel.cpp.me kdelibs-4.8.4/nepomuk/core/nepomukmainmodel.cpp
--- kdelibs-4.8.4/nepomuk/core/nepomukmainmodel.cpp.me	2012-06-14 13:10:17.239843879 +0200
+++ kdelibs-4.8.4/nepomuk/core/nepomukmainmodel.cpp	2012-06-14 13:11:00.724779658 +0200
@@ -94,10 +94,9 @@ public:
 
         // we may get disconnected from the server but we don't want to try
         // to connect every time the model is requested
-        if ( forced || (!m_socketConnectFailed && !localSocketClient.isConnected()) ) {
+        if ( !m_socketConnectFailed && !localSocketClient.isConnected() ) {
             delete localSocketModel;
             localSocketModel = 0;
-            localSocketClient.disconnect();
             QString socketName = KGlobal::dirs()->locateLocal( "socket", "nepomuk-socket" );
             kDebug() << "Connecting to local socket" << socketName;
             if ( localSocketClient.connect( socketName ) ) {
