diff -up kdelibs-4.8.0/CMakeLists.txt.version kdelibs-4.8.0/CMakeLists.txt
--- kdelibs-4.8.0/CMakeLists.txt.version	2012-01-18 12:06:22.000000000 -0600
+++ kdelibs-4.8.0/CMakeLists.txt	2012-02-04 10:16:47.076941064 -0600
@@ -12,7 +12,7 @@ set (KDE_VERSION_MAJOR 4)
 set (KDE_VERSION_MINOR 8)
 set (KDE_VERSION_RELEASE 00)
 set (KDE_VERSION "${KDE_VERSION_MAJOR}.${KDE_VERSION_MINOR}.${KDE_VERSION_RELEASE}" )
-set (KDE_VERSION_STRING "${KDE_VERSION} (4.8.0")
+set (KDE_VERSION_STRING "${KDE_VERSION} (4.8.0)")
 
 set (KDE_DISTRIBUTION_TEXT "compiled sources" CACHE STRING "Indicate the distribution in bug reports" )
 
