--- branches/KDE/4.5/kdelibs/kio/kio/krun.cpp	2010/10/15 15:53:49	1186277
+++ branches/KDE/4.5/kdelibs/kio/kio/krun.cpp	2010/10/15 16:00:27	1186278
@@ -368,9 +368,9 @@
     }
     else {
         if (supportedProtocols.isEmpty()) {
-            // compat mode: assume KIO if not set and it's a KDE app
+            // compat mode: assume KIO if not set and it's a KDE app (or a KDE service)
             const QStringList categories = _service.property("Categories").toStringList();
-            if (categories.contains("KDE")) {
+            if (categories.contains("KDE") || !_service.isApplication()) {
                 supportedProtocols.append("KIO");
             }
             else { // if no KDE app, be a bit over-generic
