diff -up kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspellclient.cpp.myspell_paths kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspellclient.cpp
--- kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspellclient.cpp.myspell_paths	2015-05-28 20:27:39.000000000 -0500
+++ kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspellclient.cpp	2015-06-04 17:59:10.824869047 -0500
@@ -52,7 +52,7 @@ SpellerPlugin *HunspellClient::createSpe
 QStringList HunspellClient::languages() const
 {
     QStringList lst;
-    QDir dir("/usr/share/myspell/dicts/");
+    QDir dir("/usr/share/myspell/");
     if(dir.exists())
     {
         QStringList lstDic = dir.entryList(QStringList("*.dic"), QDir::Files );
diff -up kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspelldict.cpp.myspell_paths kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspelldict.cpp
--- kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspelldict.cpp.myspell_paths	2015-05-28 20:27:39.000000000 -0500
+++ kdelibs-4.14.9/sonnet/plugins/hunspell/kspell_hunspelldict.cpp	2015-06-04 17:58:49.055774079 -0500
@@ -31,9 +31,9 @@ HunspellDict::HunspellDict( const QStrin
     : SpellerPlugin(lang), m_speller(0)
 {
     kDebug()<<" HunspellDict::HunspellDict( const QString& lang ):"<<lang;
-        QString dic=QString("/usr/share/myspell/dicts/%1.dic").arg(lang);
+        QString dic=QString("/usr/share/myspell/%1.dic").arg(lang);
     if (QFileInfo(dic).exists())
-        m_speller = new Hunspell(QString("/usr/share/myspell/dicts/%1.aff").arg(lang).toUtf8().constData(),dic.toUtf8().constData());
+        m_speller = new Hunspell(QString("/usr/share/myspell/%1.aff").arg(lang).toUtf8().constData(),dic.toUtf8().constData());
     else
         m_speller = 0;
     kDebug()<<" dddddd "<<m_speller;
