diff -up kdelibs-4.9.0/cmake/modules/FindKDE4Internal.cmake.htmldir kdelibs-4.9.0/cmake/modules/FindKDE4Internal.cmake
--- kdelibs-4.9.0/cmake/modules/FindKDE4Internal.cmake.htmldir	2012-09-02 14:55:03.462616900 -0300
+++ kdelibs-4.9.0/cmake/modules/FindKDE4Internal.cmake	2012-09-02 14:56:01.397306914 -0300
@@ -802,7 +802,7 @@ if (WIN32)
    set(IMPORTS_INSTALL_DIR       "${PLUGIN_INSTALL_DIR}/imports"   ) # "The subdirectory relative to the install prefix where imports will be installed
    set(CONFIG_INSTALL_DIR       "share/config"            ) # The config file install dir
    set(DATA_INSTALL_DIR         "share/apps"              ) # The parent directory where applications can install their data
-   set(HTML_INSTALL_DIR         "share/doc/HTML"          ) # The HTML install dir for documentation
+   set(HTML_INSTALL_DIR         "doc/HTML"                ) # The HTML install dir for documentation
    set(ICON_INSTALL_DIR         "share/icons"             ) # The icon install dir (default ${SHARE_INSTALL_PREFIX}/share/icons/)
    set(KCFG_INSTALL_DIR         "share/config.kcfg"       ) # The install dir for kconfig files
    set(LOCALE_INSTALL_DIR       "share/locale"            ) # The install dir for translations
@@ -875,7 +875,7 @@ else (WIN32)
    _set_fancy(IMPORTS_INSTALL_DIR       "${PLUGIN_INSTALL_DIR}/imports"                "The subdirectory relative to the install prefix where imports will be installed")
    _set_fancy(CONFIG_INSTALL_DIR       "${SHARE_INSTALL_PREFIX}/config"         "The config file install dir")
    _set_fancy(DATA_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/apps"           "The parent directory where applications can install their data")
-   _set_fancy(HTML_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/doc/HTML"       "The HTML install dir for documentation")
+   _set_fancy(HTML_INSTALL_DIR         "${CMAKE_INSTALL_PREFIX}/doc/HTML"       "The HTML install dir for documentation")
    _set_fancy(ICON_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/icons"          "The icon install dir (default ${SHARE_INSTALL_PREFIX}/share/icons/)")
    _set_fancy(KCFG_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/config.kcfg"    "The install dir for kconfig files")
    _set_fancy(LOCALE_INSTALL_DIR       "${SHARE_INSTALL_PREFIX}/locale"         "The install dir for translations")
diff -up kdelibs-4.9.0/cmake/modules/FindPyKDE4.cmake.htmldir kdelibs-4.9.0/cmake/modules/FindPyKDE4.cmake
--- kdelibs-4.9.0/cmake/modules/FindPyKDE4.cmake.htmldir	2012-09-02 14:54:33.150057747 -0300
+++ kdelibs-4.9.0/cmake/modules/FindPyKDE4.cmake	2012-09-02 14:54:48.379821708 -0300
@@ -70,7 +70,7 @@ IF(PYTHONINTERP_FOUND)
     _set_fancy(PLUGIN_INSTALL_DIR       "${LIB_INSTALL_DIR}/kde4"                "The subdirectory relative to the install prefix where plugins will be installed (default is ${LIB_INSTALL_DIR}/kde4)")
     _set_fancy(CONFIG_INSTALL_DIR       "${SHARE_INSTALL_PREFIX}/config"         "The config file install dir")
     _set_fancy(DATA_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/apps"           "The parent directory where applications can install their data")
-    _set_fancy(HTML_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/doc/HTML"       "The HTML install dir for documentation")
+    _set_fancy(HTML_INSTALL_DIR         "${CMAKE_INSTALL_PREFIX}/doc/HTML"       "The HTML install dir for documentation")
     _set_fancy(ICON_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/icons"          "The icon install dir (default ${SHARE_INSTALL_PREFIX}/share/icons/)")
     _set_fancy(KCFG_INSTALL_DIR         "${SHARE_INSTALL_PREFIX}/config.kcfg"    "The install dir for kconfig files")
     _set_fancy(LOCALE_INSTALL_DIR       "${SHARE_INSTALL_PREFIX}/locale"         "The install dir for translations")
