diff --git a/kio/misc/mms.protocol b/kio/misc/mms.protocol
index 286ccd8..e8afbdb 100644
--- a/kio/misc/mms.protocol
+++ b/kio/misc/mms.protocol
@@ -1,7 +1,7 @@
 [Protocol]
 helper=true
 exec=
-defaultMimetype=uri/mms
+defaultMimetype=x-scheme-handler/mms
 protocol=mms
 input=none
 output=none
diff --git a/kio/misc/mmst.protocol b/kio/misc/mmst.protocol
index 1f0f763..455694a 100644
--- a/kio/misc/mmst.protocol
+++ b/kio/misc/mmst.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-defaultMimetype=uri/mmst
+defaultMimetype=x-scheme-handler/mmst
 exec=
 protocol=mmst
 input=none
diff --git a/kio/misc/mmsu.protocol b/kio/misc/mmsu.protocol
index deda935..1c1a7f6 100644
--- a/kio/misc/mmsu.protocol
+++ b/kio/misc/mmsu.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-defaultMimetype=uri/mmsu
+defaultMimetype=x-scheme-handler/mmsu
 exec=
 protocol=mmsu
 input=none
diff --git a/kio/misc/pnm.protocol b/kio/misc/pnm.protocol
index 3ca3a7d..fe44c37 100644
--- a/kio/misc/pnm.protocol
+++ b/kio/misc/pnm.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-defaultMimetype=uri/pnm
+defaultMimetype=x-scheme-handler/pnm
 exec=
 protocol=pnm
 input=none
diff --git a/kio/misc/rtspt.protocol b/kio/misc/rtspt.protocol
index 9e4fe96..4d86282 100644
--- a/kio/misc/rtspt.protocol
+++ b/kio/misc/rtspt.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-defaultMimetype=uri/rtspt
+defaultMimetype=x-scheme-handler/rtspt
 exec=
 protocol=rtspt
 input=none
diff --git a/kio/misc/rtspu.protocol b/kio/misc/rtspu.protocol
index 9cc9299..5fc9d6b 100644
--- a/kio/misc/rtspu.protocol
+++ b/kio/misc/rtspu.protocol
@@ -1,5 +1,5 @@
 [Protocol]
-defaultMimetype=uri/rtspu
+defaultMimetype=x-scheme-handler/rtspu
 exec=
 protocol=rtspu
 input=none
diff --git a/mimetypes/kde.xml b/mimetypes/kde.xml
index 2810b5b..8de16f6 100644
--- a/mimetypes/kde.xml
+++ b/mimetypes/kde.xml
@@ -187,24 +187,30 @@ Notes:
     <comment>all files</comment>
   </mime-type>
 
-  <!-- uri/ fake mime types -->
-  <mime-type type="uri/mms">
+  <!-- x-scheme-handler/ fake(ish) mime types -->
+  <mime-type type="x-scheme-handler/mms">
     <comment>mms: URIs</comment>
+    <alias type="uri/mms"/>
   </mime-type>
-  <mime-type type="uri/mmst">
+  <mime-type type="x-scheme-handler/mmst">
     <comment>mmst: URIs</comment>
+    <alias type="uri/mmst"/>
   </mime-type>
-  <mime-type type="uri/mmsu">
+  <mime-type type="x-scheme-handler/mmsu">
     <comment>mmsu: URIs</comment>
+    <alias type="uri/mmsu"/>
   </mime-type>
-  <mime-type type="uri/pnm">
+  <mime-type type="x-scheme-handler/pnm">
     <comment>pnm: URIs</comment>
+    <alias type="uri/pnm"/>
   </mime-type>
-  <mime-type type="uri/rtspt">
+  <mime-type type="x-scheme-handler/rtspt">
     <comment>rtspt: URIs</comment>
+    <alias type="uri/rtspt"/>
   </mime-type>
-  <mime-type type="uri/rtspu">
+  <mime-type type="x-scheme-handler/rtspu">
     <comment>rtspu: URIs</comment>
+    <alias type="uri/rtspu"/>
   </mime-type>
 
   <mime-type type="application/vnd.kde.fontspackage">
