diff -up kdelibs-4.5.80/kio/kio/kprotocolmanager.cpp.branding kdelibs-4.5.80/kio/kio/kprotocolmanager.cpp
--- kdelibs-4.5.80/kio/kio/kprotocolmanager.cpp.branding	2010-11-05 05:48:26.000000000 -0500
+++ kdelibs-4.5.80/kio/kio/kprotocolmanager.cpp	2010-11-20 10:42:54.223363491 -0600
@@ -87,7 +87,7 @@ QL1S("Mozilla/5.0 (compatible; Konqueror
 QString::number(KDE::versionMajor()) % QL1C('.') % QString::number(KDE::versionMinor()) % \
 X % QL1S(") KHTML/") % \
 QString::number(KDE::versionMajor()) % QL1C('.') % QString::number(KDE::versionMinor()) % \
-QL1C('.') % QString::number(KDE::versionRelease()) % QL1S(" (like Gecko)")
+QL1C('.') % QString::number(KDE::versionRelease()) % QL1S(" (like Gecko) _KDELIBS_SLK_DIST")
 
 #define PRIVATE_DATA \
 KProtocolManagerPrivate *d = kProtocolManagerPrivate
