diff -up kdelibs-4.4.80/kdecore/kernel/kstandarddirs.cpp.libexecdir kdelibs-4.4.80/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.4.80/kdecore/kernel/kstandarddirs.cpp.libexecdir	2010-04-28 10:01:43.000000000 +0200
+++ kdelibs-4.4.80/kdecore/kernel/kstandarddirs.cpp	2010-05-21 09:51:22.456216633 +0200
@@ -1819,7 +1819,7 @@ void KStandardDirs::addKDEDefaults()
         addResourceType(types_string + types_indices[index], 0, types_string + types_indices[index+1], true);
         index+=2;
     }
-    addResourceType("exe", "lib", "kde4/libexec", true );
+    addResourceType("exe", 0, "libexec/kde4", true );
 
     addResourceDir("home", QDir::homePath(), false);
 
diff -up kdelibs-4.4.80/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir kdelibs-4.4.80/kdecore/kernel/kstandarddirs_unix.cpp
--- kdelibs-4.4.80/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir	2008-05-21 13:08:57.000000000 +0200
+++ kdelibs-4.4.80/kdecore/kernel/kstandarddirs_unix.cpp	2010-05-21 09:51:22.456216633 +0200
@@ -62,7 +62,7 @@ QString KStandardDirs::installPath(const
             if (strcmp("lib", type) == 0)
                 return QString::fromLatin1(LIB_INSTALL_DIR "/");
             if (strcmp("libexec", type) == 0)
-                return QString::fromLatin1(KDEDIR "/lib" KDELIBSUFF "/kde4/libexec/");
+                return QString::fromLatin1(LIBEXEC_INSTALL_DIR "/");
             if (strcmp("locale", type) == 0)
                 return QString::fromLatin1(LOCALE_INSTALL_DIR "/");
             break;
diff -up kdelibs-4.4.80/kdecore/tests/kstandarddirstest.cpp.libexecdir kdelibs-4.4.80/kdecore/tests/kstandarddirstest.cpp
--- kdelibs-4.4.80/kdecore/tests/kstandarddirstest.cpp.libexecdir	2010-04-28 10:01:43.000000000 +0200
+++ kdelibs-4.4.80/kdecore/tests/kstandarddirstest.cpp	2010-05-21 09:59:12.502278424 +0200
@@ -83,8 +83,9 @@ void KStandarddirsTest::testFindResource
 #define KIOSLAVE "bin/kioslave.exe"
 #else
 #define EXT ""
-#define KIOSLAVE "kde4/libexec/kioslave"
+#define KIOSLAVE "libexec/kde4/kioslave"
 #endif
+
     const QString bin = KGlobal::dirs()->findResource( "exe", "kioslave" EXT );
     QVERIFY( !bin.isEmpty() );
     QVERIFY( bin.endsWith( KIOSLAVE ) );
@@ -210,7 +211,8 @@ void KStandarddirsTest::testFindExe()
     // findExe with a result in libexec
     const QString lnusertemp = KGlobal::dirs()->findExe( "lnusertemp" );
     QVERIFY( !lnusertemp.isEmpty() );
-    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) );
+    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) 
+        || lnusertemp.endsWith( "libexec/kde4/lnusertemp" EXT, PATH_SENSITIVITY ) );
 #endif
 
     // Check the "exe" resource too
diff -up kdelibs-4.4.80/kio/tests/krununittest.cpp.libexecdir kdelibs-4.4.80/kio/tests/krununittest.cpp
--- kdelibs-4.4.80/kio/tests/krununittest.cpp.libexecdir	2010-02-24 13:52:53.000000000 +0100
+++ kdelibs-4.4.80/kio/tests/krununittest.cpp	2010-05-21 09:51:22.457216466 +0200
@@ -147,7 +147,7 @@ void KRunUnitTest::testProcessDesktopExe
     if (kmailservice.isEmpty()) kmailservice = "kmailservice";
     if (!kdeinit.isEmpty()) {
         QVERIFY(!kmailservice.isEmpty());
-        QVERIFY(kmailservice.contains("kde4/libexec"));
+        QVERIFY(kmailservice.contains("kde4/libexec") || kmailservice.contains("libexec/kde4"));
     }
 
     QTest::newRow("%U l0") << "kdeinit4 %U" << l0 << false << kdeinit;
