diff -up kdelibs-4.5.80/kdecore/kernel/kstandarddirs.cpp.libexecdir kdelibs-4.5.80/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.5.80/kdecore/kernel/kstandarddirs.cpp.libexecdir	2010-11-19 04:52:23.000000000 -0600
+++ kdelibs-4.5.80/kdecore/kernel/kstandarddirs.cpp	2010-11-20 10:44:54.913612398 -0600
@@ -1842,7 +1842,7 @@ void KStandardDirs::addKDEDefaults()
         addResourceType(types_string + types_indices[index], 0, types_string + types_indices[index+1], true);
         index+=2;
     }
-    addResourceType("exe", "lib", "kde4/libexec", true );
+    addResourceType("exe", 0, "libexec/kde4", true );
 
     addResourceDir("home", QDir::homePath(), false);
 
diff -up kdelibs-4.5.80/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir kdelibs-4.5.80/kdecore/kernel/kstandarddirs_unix.cpp
--- kdelibs-4.5.80/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir	2010-11-05 05:48:28.000000000 -0500
+++ kdelibs-4.5.80/kdecore/kernel/kstandarddirs_unix.cpp	2010-11-20 10:48:44.671378287 -0600
@@ -63,7 +63,7 @@ QString KStandardDirs::installPath(const
             if (strcmp("lib", type) == 0)
                 return QFile::decodeName(LIB_INSTALL_DIR "/");
             if (strcmp("libexec", type) == 0)
-                return QFile::decodeName(KDEDIR "/lib" KDELIBSUFF "/kde4/libexec/");
+                return QFile::decodeName(LIBEXEC_INSTALL_DIR "/");
             if (strcmp("locale", type) == 0)
                 return QFile::decodeName(LOCALE_INSTALL_DIR "/");
             break;
diff -up kdelibs-4.5.80/kdecore/tests/kstandarddirstest.cpp.libexecdir kdelibs-4.5.80/kdecore/tests/kstandarddirstest.cpp
--- kdelibs-4.5.80/kdecore/tests/kstandarddirstest.cpp.libexecdir	2010-11-11 08:41:42.000000000 -0600
+++ kdelibs-4.5.80/kdecore/tests/kstandarddirstest.cpp	2010-11-20 10:44:54.914612189 -0600
@@ -94,8 +94,9 @@ void KStandarddirsTest::testFindResource
 #define KIOSLAVE "bin/kioslave.exe"
 #else
 #define EXT ""
-#define KIOSLAVE "kde4/libexec/kioslave"
+#define KIOSLAVE "libexec/kde4/kioslave"
 #endif
+
     const QString bin = KGlobal::dirs()->findResource( "exe", "kioslave" EXT );
     QVERIFY( !bin.isEmpty() );
     QVERIFY( bin.endsWith( KIOSLAVE ) );
@@ -221,7 +222,8 @@ void KStandarddirsTest::testFindExe()
     // findExe with a result in libexec
     const QString lnusertemp = KGlobal::dirs()->findExe( "lnusertemp" );
     QVERIFY( !lnusertemp.isEmpty() );
-    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) );
+    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) 
+        || lnusertemp.endsWith( "libexec/kde4/lnusertemp" EXT, PATH_SENSITIVITY ) );
 #endif
 
     // Check the "exe" resource too
diff -up kdelibs-4.5.80/kio/tests/krununittest.cpp.libexecdir kdelibs-4.5.80/kio/tests/krununittest.cpp
--- kdelibs-4.5.80/kio/tests/krununittest.cpp.libexecdir	2010-11-05 05:48:27.000000000 -0500
+++ kdelibs-4.5.80/kio/tests/krununittest.cpp	2010-11-20 10:44:54.915612398 -0600
@@ -148,7 +148,7 @@ void KRunUnitTest::testProcessDesktopExe
     if (kmailservice.isEmpty()) kmailservice = "kmailservice";
     if (!kdeinit.isEmpty()) {
         QVERIFY(!kmailservice.isEmpty());
-        QVERIFY(kmailservice.contains("kde4/libexec"));
+        QVERIFY(kmailservice.contains("kde4/libexec") || kmailservice.contains("libexec/kde4"));
     }
 
     QTest::newRow("%U l0") << "kdeinit4 %U" << l0 << false << kdeinit;
