diff -up kdelibs-4.5.1/knewstuff/knewstuff2/core/security.cpp.gpg2 kdelibs-4.5.1/knewstuff/knewstuff2/core/security.cpp
--- kdelibs-4.5.1/knewstuff/knewstuff2/core/security.cpp.gpg2	2009-11-30 18:27:20.000000000 -0600
+++ kdelibs-4.5.1/knewstuff/knewstuff2/core/security.cpp	2010-08-27 10:49:10.619703933 -0500
@@ -61,7 +61,7 @@ void Security::readKeys()
     m_runMode = List;
     m_keys.clear();
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--no-tty"
     << "--with-colon"
@@ -87,7 +87,7 @@ void Security::readSecretKeys()
     }
     m_runMode = ListSecret;
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--no-tty"
     << "--with-colon"
@@ -258,7 +258,7 @@ void Security::slotCheckValidity()
 
     //verify the signature
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--status-fd=2"
     << "--command-fd=0"
@@ -340,7 +340,7 @@ void Security::slotSignFile()
 
     //verify the signature
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--status-fd=2"
     << "--command-fd=0"
diff -up kdelibs-4.5.1/knewstuff/knewstuff3/core/security.cpp.gpg2 kdelibs-4.5.1/knewstuff/knewstuff3/core/security.cpp
--- kdelibs-4.5.1/knewstuff/knewstuff3/core/security.cpp.gpg2	2009-11-30 18:27:19.000000000 -0600
+++ kdelibs-4.5.1/knewstuff/knewstuff3/core/security.cpp	2010-08-27 10:49:50.493454042 -0500
@@ -61,7 +61,7 @@ void Security::readKeys()
     m_runMode = List;
     m_keys.clear();
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--no-tty"
     << "--with-colon"
@@ -87,7 +87,7 @@ void Security::readSecretKeys()
     }
     m_runMode = ListSecret;
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--no-tty"
     << "--with-colon"
@@ -258,7 +258,7 @@ void Security::slotCheckValidity()
 
     //verify the signature
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--status-fd=2"
     << "--command-fd=0"
@@ -340,7 +340,7 @@ void Security::slotSignFile()
 
     //verify the signature
     m_process = new KProcess();
-    *m_process << "gpg"
+    *m_process << "gpg2"
     << "--no-secmem-warning"
     << "--status-fd=2"
     << "--command-fd=0"
