diff -up kdelibs-4.7.4/solid/solid/managerbase.cpp.UPNP kdelibs-4.7.4/solid/solid/managerbase.cpp
--- kdelibs-4.7.4/solid/solid/managerbase.cpp.UPNP	2011-12-01 19:22:43.644706401 -0600
+++ kdelibs-4.7.4/solid/solid/managerbase.cpp	2011-12-01 19:29:11.825853422 -0600
@@ -80,7 +80,11 @@ void Solid::ManagerBasePrivate::loadBack
 #        endif
 
 #        if defined (HUPNP_FOUND)
-            m_backends << new Solid::Backends::UPnP::UPnPDeviceManager(0);
+            bool solidUpnpEnabled
+                = QString::fromLocal8Bit(qgetenv("SOLID_UPNP")).toInt()==1;
+            if (solidUpnpEnabled) {
+                m_backends << new Solid::Backends::UPnP::UPnPDeviceManager(0);
+            }
 #        endif
     }
 }
