--- branches/KDE/4.4/kdelibs/kdecore/io/kdebug.cpp	2010/01/06 22:46:33	1070858
+++ branches/KDE/4.4/kdelibs/kdecore/io/kdebug.cpp	2010/03/20 01:43:08	1105396
@@ -317,7 +317,7 @@
         return groupName;
     }
 
-    OutputMode areaOutputMode(QtMsgType type, unsigned int area)
+    OutputMode areaOutputMode(QtMsgType type, unsigned int area, bool enableByDefault)
     {
         if (!config)
             return QtOutput;
@@ -343,7 +343,7 @@
         }
 
         const KConfigGroup cg(config, groupNameForArea(area));
-        const int mode = cg.readEntry(key, int(DefaultOutput));
+        const int mode = cg.readEntry(key, int(enableByDefault ? DefaultOutput : NoOutput));
         return OutputMode(mode);
     }
 
@@ -418,7 +418,7 @@
         const int lev = level(type);
         //qDebug() << "in cache for" << num << ":" << it->mode[lev];
         if (it->mode[lev] == Unknown)
-            it->mode[lev] = areaOutputMode(type, num);
+            it->mode[lev] = areaOutputMode(type, num, enableByDefault);
         if (it->mode[lev] == FileOutput && it->logFileName[lev].isEmpty())
             it->logFileName[lev] = logFileName(type, num);
 
