diff -up kdelibs-4.11.3/kdecore/kernel/kstandarddirs.cpp.libexecdir kdelibs-4.11.3/kdecore/kernel/kstandarddirs.cpp
--- kdelibs-4.11.3/kdecore/kernel/kstandarddirs.cpp.libexecdir	2013-06-28 12:03:40.883340083 -0500
+++ kdelibs-4.11.3/kdecore/kernel/kstandarddirs.cpp	2013-11-01 15:44:00.780783690 -0500
@@ -1871,7 +1871,7 @@ void KStandardDirs::addKDEDefaults()
         addResourceType(types_string + types_indices[index], 0, types_string + types_indices[index+1], true);
         index+=2;
     }
-    addResourceType("exe", "lib", "kde4/libexec", true );
+    addResourceType("exe", 0, "libexec/kde4", true );
 
     addResourceDir("home", QDir::homePath(), false);
 
diff -up kdelibs-4.11.3/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir kdelibs-4.11.3/kdecore/kernel/kstandarddirs_unix.cpp
--- kdelibs-4.11.3/kdecore/kernel/kstandarddirs_unix.cpp.libexecdir	2013-06-28 12:03:40.884340190 -0500
+++ kdelibs-4.11.3/kdecore/kernel/kstandarddirs_unix.cpp	2013-11-01 15:44:00.782783770 -0500
@@ -63,7 +63,7 @@ QString KStandardDirs::installPath(const
             if (strcmp("lib", type) == 0)
                 return QFile::decodeName(LIB_INSTALL_DIR "/");
             if (strcmp("libexec", type) == 0)
-                return QFile::decodeName(KDEDIR "/lib" KDELIBSUFF "/kde4/libexec/");
+                return QFile::decodeName(LIBEXEC_INSTALL_DIR "/");
             if (strcmp("locale", type) == 0)
                 return QFile::decodeName(LOCALE_INSTALL_DIR "/");
             break;
diff -up kdelibs-4.11.3/kdecore/tests/kstandarddirstest.cpp.libexecdir kdelibs-4.11.3/kdecore/tests/kstandarddirstest.cpp
--- kdelibs-4.11.3/kdecore/tests/kstandarddirstest.cpp.libexecdir	2013-11-01 10:45:56.409145508 -0500
+++ kdelibs-4.11.3/kdecore/tests/kstandarddirstest.cpp	2013-11-01 15:50:20.473658147 -0500
@@ -96,8 +96,9 @@ void KStandarddirsTest::testFindResource
 #define KIOSLAVE "bin/kioslave.exe"
 #else
 #define EXT ""
-#define KIOSLAVE "kde4/libexec/kioslave"
+#define KIOSLAVE "libexec/kde4/kioslave"
 #endif
+
     const QString bin = KGlobal::dirs()->findResource( "exe", "kioslave" EXT );
     QVERIFY( !bin.isEmpty() );
     QVERIFY( bin.endsWith( KIOSLAVE ) );
@@ -248,11 +249,13 @@ void KStandarddirsTest::testFindExe()
     // findExe with a result in libexec
     const QString lnusertemp = KGlobal::dirs()->findExe( "lnusertemp" );
     QVERIFY( !lnusertemp.isEmpty() );
-    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY ) );
+    QVERIFY( lnusertemp.endsWith( "lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY )
+	|| lnusertemp.endsWith( "libexec/kde4/lnusertemp" EXT, PATH_SENSITIVITY ) );
 
     // locate("exe") with a result in libexec
     const QString locateExeResult = KGlobal::dirs()->locate("exe", "lnusertemp");
-    QVERIFY(locateExeResult.endsWith("lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY));
+    QVERIFY(locateExeResult.endsWith("lib" KDELIBSUFF "/kde4/libexec/lnusertemp" EXT, PATH_SENSITIVITY) 
+	|| locateExeResult.endsWith("libexec/kde4/lnusertemp" EXT, PATH_SENSITIVITY) );
 
     // findExe with relative path
     const QString pwd = QDir::currentPath();
