From 0edfd42151ad57322a10a24ab4971b638e220e6e Mon Sep 17 00:00:00 2001
From: Aaron Seigo <aseigo@kde.org>
Date: Thu, 21 Feb 2013 18:14:54 +0100
Subject: [PATCH 049/111] return, not break. copy/paste error

---
 kdeui/icons/kiconloader.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kdeui/icons/kiconloader.cpp b/kdeui/icons/kiconloader.cpp
index dba474d..ce6aeea 100644
--- a/kdeui/icons/kiconloader.cpp
+++ b/kdeui/icons/kiconloader.cpp
@@ -947,12 +947,12 @@ K3Icon KIconLoaderPrivate::findMatchingIcon(const QString& name, int size) const
             for (int i = 0 ; i < 4 ; i++) {
                 icon = themeNode->theme->iconPath(currentName + ext[i], size, KIconLoader::MatchExact);
                 if (icon.isValid()) {
-                    break;
+                    return icon;
                 }
 
                 icon = themeNode->theme->iconPath(currentName + ext[i], size, KIconLoader::MatchBest);
                 if (icon.isValid()) {
-                    break;
+                    return icon;
                 }
             }
             //kDebug(264) << "Looking up" << currentName;
-- 
1.8.1.4

