diff -up openjpeg_v1_4_sources_r697/CMakeLists.txt.soversion openjpeg_v1_4_sources_r697/CMakeLists.txt
--- openjpeg_v1_4_sources_r697/CMakeLists.txt.soversion	2011-01-02 12:14:55.000000000 -0600
+++ openjpeg_v1_4_sources_r697/CMakeLists.txt	2011-01-03 15:41:02.739274204 -0600
@@ -27,6 +27,7 @@ INCLUDE_REGULAR_EXPRESSION("^.*$")
 
 #-----------------------------------------------------------------------------
 # OPENJPEG version number, useful for packaging and doxygen doc:
+SET(OPENJPEG_SOVERSION 3)
 SET(OPENJPEG_VERSION_MAJOR 1)
 SET(OPENJPEG_VERSION_MINOR 4)
 SET(OPENJPEG_VERSION_BUILD 0)
@@ -37,8 +38,8 @@ SET(PACKAGE_VERSION
 # This setting of SOVERSION assumes that any API change
 # will increment either the minor or major version number of openjpeg
 SET(OPENJPEG_LIBRARY_PROPERTIES
-  VERSION   "${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}.${OPENJPEG_VERSION_BUILD}"
-  SOVERSION "${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}"
+  VERSION   "${OPENJPEG_SOVERSION}.${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}.${OPENJPEG_VERSION_BUILD}"
+  SOVERSION "${OPENJPEG_SOVERSION}"
 )
 # You will also need to define a value for the following variables:
 # OPENJPEG_INSTALL_BIN_DIR          - binary dir (executables)
