diff -up openjpeg-1.5.1/CMakeLists.txt.cmake_libdir openjpeg-1.5.1/CMakeLists.txt
--- openjpeg-1.5.1/CMakeLists.txt.cmake_libdir	2012-09-13 02:58:39.000000000 -0500
+++ openjpeg-1.5.1/CMakeLists.txt	2012-09-13 08:55:59.493257595 -0500
@@ -183,7 +183,7 @@ IF(UNIX)
     ${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig )
   INSTALL( CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink
   \"libopenjpeg1.pc\"
-  \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${OPENJPEG_INSTALL_SHARE_DIR}/pkgconfig/libopenjpeg.pc\")")
+  \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig/libopenjpeg.pc\")")
 ENDIF(UNIX)
 
 #-----------------------------------------------------------------------------
