diff -up openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt.no_undefined openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt
--- openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt.no_undefined	2011-01-02 12:14:45.000000000 -0600
+++ openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt	2011-01-12 12:15:45.364398086 -0600
@@ -32,6 +32,10 @@ ENDIF(WIN32)
 ADD_LIBRARY(${OPENJPEG_LIBRARY_NAME} ${OPENJPEG_SRCS})
 SET_TARGET_PROPERTIES(${OPENJPEG_LIBRARY_NAME} PROPERTIES ${OPENJPEG_LIBRARY_PROPERTIES})
 
+IF(UNIX)
+  TARGET_LINK_LIBRARIES(${OPENJPEG_LIBRARY_NAME} m)
+ENDIF(UNIX)
+
 # Install library
 INSTALL(TARGETS ${OPENJPEG_LIBRARY_NAME}
   EXPORT OpenJPEGTargets
diff -up openjpeg_v1_4_sources_r697/libopenjpeg/Makefile.am.no_undefined openjpeg_v1_4_sources_r697/libopenjpeg/Makefile.am
--- openjpeg_v1_4_sources_r697/libopenjpeg/Makefile.am.no_undefined	2011-01-02 12:14:45.000000000 -0600
+++ openjpeg_v1_4_sources_r697/libopenjpeg/Makefile.am	2011-01-12 12:18:40.953012288 -0600
@@ -9,6 +9,7 @@ INCLUDES = -I.. -I.
 
 libopenjpeg_la_LDFLAGS = -no-undefined \
   -version-info $(CURRENT):$(BUILD):$(MINOR)
+libopenjpeg_la_LIBADD = -lm
 
 SRCS = bio.c cio.c dwt.c event.c image.c j2k.c j2k_lib.c jp2.c jpt.c \
  mct.c mqc.c openjpeg.c pi.c raw.c t1.c t2.c tcd.c tgt.c
