diff -up openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt.cmake_symlink_fix openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt
--- openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt.cmake_symlink_fix	2011-01-13 09:13:08.077608144 -0600
+++ openjpeg_v1_4_sources_r697/libopenjpeg/CMakeLists.txt	2011-01-13 09:14:36.417405082 -0600
@@ -47,7 +47,7 @@ INSTALL(FILES openjpeg.h
   DESTINATION ${OPENJPEG_INSTALL_INCLUDE_DIR}/${subdir} COMPONENT Headers
 )
 INSTALL(CODE
-  "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${OPENJPEG_INSTALL_INCLUDE_DIR}/${subdir}/openjpeg.h \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/${OPENJPEG_INSTALL_INCLUDE_DIR}/openjpeg.h)")
+  "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink \"${subdir}/openjpeg.h\" \"\$ENV{DESTDIR}${OPENJPEG_INSTALL_INCLUDE_DIR}/openjpeg.h\")")
 
 # install man page of the library
 INSTALL(
