--- build-5.2/src/CMakeLists.txt	2009-12-15 14:27:57.000000000 +0800
+++ build-5.2.new/src/CMakeLists.txt	2010-04-28 14:37:16.004826978 +0800
@@ -19,6 +19,8 @@
 	tab_index.cpp
 	tab_search.cpp
 	treeviewitem.cpp
+	toolbarmanager.cpp
+	toolbareditor.cpp
 	viewwindow.cpp
 	viewwindowmgr.cpp
 	viewwindow_qtextbrowser.cpp
@@ -36,6 +38,7 @@
 	tab_contents.ui
 	tab_index.ui
 	tab_search.ui
+	toolbareditor.ui
 	window_browser.ui
    )
 
