diff -up kchmviewer-5.1/CMakeLists.txt.no_msits kchmviewer-5.1/CMakeLists.txt
--- kchmviewer-5.1/CMakeLists.txt.no_msits      2010-03-19 12:53:57.000000000 +0100
+++ kchmviewer-5.1/CMakeLists.txt       2010-03-19 12:54:07.000000000 +0100
@@ -18,7 +18,7 @@ ADD_DEFINITIONS( -DUSE_KDE )
 
 # Run in those subdirectories
 ADD_SUBDIRECTORY( lib/libchmfile )
-ADD_SUBDIRECTORY( lib/kio-msits )
+#ADD_SUBDIRECTORY( lib/kio-msits )
 ADD_SUBDIRECTORY( src )
 ADD_SUBDIRECTORY( po )
 ADD_SUBDIRECTORY( packages )
