diff -Nur thunar-volman-0.7.0.orig/thunar-volman-settings/thunar-volman-settings.desktop.in.in thunar-volman-0.7.0/thunar-volman-settings/thunar-volman-settings.desktop.in.in
--- thunar-volman-0.7.0.orig/thunar-volman-settings/thunar-volman-settings.desktop.in.in	2012-03-28 10:42:29.000000000 -0600
+++ thunar-volman-0.7.0/thunar-volman-settings/thunar-volman-settings.desktop.in.in	2012-04-03 18:47:50.123807289 -0600
@@ -6,6 +6,6 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
 
 # vi:set encoding=UTF-8:
