--- configure
+++ configure
@@ -1061,6 +1061,7 @@
 with_jp2
 with_lcms
 with_lcms2
+with_lzma
 with_png
 with_tiff
 with_trio
@@ -20007,9 +20008,9 @@
 
 # Disable LZMA (lzma library)
 
-# Check whether --with-bzlib was given.
-if test "${with_bzlib+set}" = set; then :
-  withval=$with_bzlib; with_lzma=$withval
+# Check whether --with-lzma was given.
+if test "${with_lzma+set}" = set; then :
+  withval=$with_lzma; with_lzma=$withval
 else
   with_lzma='yes'
 fi
--- configure.ac
+++ configure.ac
@@ -694,7 +694,7 @@
 fi
 
 # Disable LZMA (lzma library)
-AC_ARG_WITH(bzlib,
+AC_ARG_WITH(lzma,
 	      [  --without-lzma          disable LZMA support],
 	      [with_lzma=$withval],
 	      [with_lzma='yes'])
