diff -up libtranslate-0.99/src/modules/Makefile.am.fix-modules libtranslate-0.99/src/modules/Makefile.am
diff -up libtranslate-0.99/src/Makefile.am.fix-modules libtranslate-0.99/src/Makefile.am
--- libtranslate-0.99/src/Makefile.am.fix-modules	2008-12-21 11:12:05.000000000 +0200
+++ libtranslate-0.99/src/Makefile.am	2008-12-21 11:19:42.000000000 +0200
@@ -1,6 +1,6 @@
 include $(top_srcdir)/common.make
 
-SUBDIRS = modules
+SUBDIRS = . modules
 
 public_headers = \
 	translate-common.h			\
diff -up libtranslate-0.99/src/modules/Makefile.am.fix-modules libtranslate-0.99/src/modules/Makefile.am
--- libtranslate-0.99/src/modules/Makefile.am.fix-modules	2005-01-05 14:14:17.000000000 +0200
+++ libtranslate-0.99/src/modules/Makefile.am	2008-12-21 11:10:51.000000000 +0200
@@ -18,12 +18,12 @@ generic_la_SOURCES = \
 	translate-generic-service.h		\
 	translate-generic-soup-cookie-jar.c	\
 	translate-generic-soup-cookie-jar.h
-generic_la_LDFLAGS = -avoid-version -module $(SOUP_LIBS) $(LIBXML_LIBS)
+generic_la_LDFLAGS = -avoid-version -module $(SOUP_LIBS) $(LIBXML_LIBS) $(top_builddir)/src/libtranslate.la
 generic_la_CPPFLAGS = $(WARN_CFLAGS) $(TRANSLATE_CFLAGS) $(SOUP_CFLAGS) $(LIBXML_CFLAGS) \
 	-DG_LOG_DOMAIN="\"$(PACKAGE)(generic)\"" \
 	-DSERVICESFILE="\"$(pkgdatadir)/services.xml\""
 
 talkfilters_la_SOURCES = translate-talkfilters.c
-talkfilters_la_LDFLAGS = -avoid-version -module $(TALKFILTERS_LIBS)
+talkfilters_la_LDFLAGS = -avoid-version -module $(TALKFILTERS_LIBS) $(top_builddir)/src/libtranslate.la
 talkfilters_la_CPPFLAGS = $(WARN_CFLAGS) $(TRANSLATE_CFLAGS) $(TALKFILTERS_CFLAGS) \
 	-DG_LOG_DOMAIN="\"$(PACKAGE)(talkfilters)\""
