--- lives-plugins/plugins/playback/video/Makefile.am.orig	2011-03-27 16:14:43.000000000 -0300
+++ lives-plugins/plugins/playback/video/Makefile.am	2011-03-27 16:14:51.728382068 -0300
@@ -48,12 +48,6 @@ yuv4mpeg_stream_la_LDFLAGS = $(shared_ld
 SDL_la_CFLAGS = $(shared_cflags) $(CFLAGS_SDL) $(weed_cflags)
 SDL_la_LDFLAGS = $(shared_ldflags) $(LDFLAGS_SDL)
 
-if IS_LINUX_GNU
-vloopback_libs = vloopback.la
-vloopback_la_CFLAGS = $(shared_cflags) $(weed_cflags)
-vloopback_la_LDFLAGS = $(shared_ldflags)
-endif
-
 videopluginslib_LTLIBRARIES = $(SDL_libs) $(YUV4MPEG_libs) $(vjack_output_libs) lives2lives_stream.la $(vloopback_libs)
 
 EXTRA_DIST = videoplugin.h
