--- lives-plugins/plugins/decoders/asf_decoder.c.ffmpeg	2011-10-15 15:51:33.458172472 -0300
+++ lives-plugins/plugins/decoders/asf_decoder.c	2011-10-15 15:51:50.565543187 -0300
@@ -1227,7 +1227,7 @@ static boolean attach_stream(lives_clip_
 	snprintf(cdata->audio_name,16,"unknown");
 
       } else if (type == LIVES_MEDIA_TYPE_VIDEO) {
-	priv->st->codec->codec_type = CODEC_TYPE_VIDEO;
+	priv->st->codec->codec_type = AVMEDIA_TYPE_VIDEO;
 	if (vidindex!=-1&&vidindex!=priv->st->id) {
 	  fprintf(stderr, "asf_decoder: unhandled multiple vidstreams %d and %d in %s\n",vidindex,priv->st->id,cdata->URI);
 	  got_vidst=TRUE;
