diff -p -up kdenetwork-4.2.2/kopete/kopete/kopeteui.rc.fixsearchbar kdenetwork-4.2.2/kopete/kopete/kopeteui.rc
--- kdenetwork-4.2.2/kopete/kopete/kopeteui.rc.fixsearchbar	2009-04-07 14:05:18.000000000 -0300
+++ kdenetwork-4.2.2/kopete/kopete/kopeteui.rc	2009-04-07 14:05:34.000000000 -0300
@@ -51,7 +51,7 @@
 		<Action name="settings_show_all_offline_empty"/>
 	</ToolBar>
 
-        <ToolBar fullWidth="true" name="quickSearchBar" noMerge="1" newline="false"><text>Quick Search Bar</text>
+        <ToolBar fullWidth="true" name="quickSearchBar" noMerge="1" newline="true"><text>Quick Search Bar</text>
 		<Action name="quicksearch_reset"/>
 		<Action name="quicksearch_label"/>
 		<Action name="quicksearch_bar"/>
