diff -up kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc.ssl kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc
--- kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc.ssl	2010-04-01 04:02:19.558192158 -0300
+++ kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc	2010-04-01 04:03:06.308071121 -0300
@@ -648,7 +648,11 @@ OpenSSLAdapter::SSLPostConnectionCheck(S
     int extension_nid = OBJ_obj2nid(X509_EXTENSION_get_object(extension));
 
     if (extension_nid == NID_subject_alt_name) {
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+      const X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#else
       X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#endif
       if (!meth)
         break;
 
