diff -up kdenetwork-4.10.2/kopete/libkopete/avdevice/videodevice.cpp.v4l_deprecations kdenetwork-4.10.2/kopete/libkopete/avdevice/videodevice.cpp
--- kdenetwork-4.10.2/kopete/libkopete/avdevice/videodevice.cpp.v4l_deprecations	2013-02-28 23:22:39.148721688 -0600
+++ kdenetwork-4.10.2/kopete/libkopete/avdevice/videodevice.cpp	2013-04-01 06:43:55.506441943 -0500
@@ -3061,8 +3061,12 @@ const char *VideoDevice::getUnifiedV4L2S
 		case V4L2_CID_GAIN:			return I18N_NOOP("Gain");
 		case V4L2_CID_HFLIP:			return I18N_NOOP("Horizontal Flip");
 		case V4L2_CID_VFLIP:			return I18N_NOOP("Vertical Flip");
+#ifdef V4L2_CID_HCENTER			// since kernel-3.9.0
 		case V4L2_CID_HCENTER:			return I18N_NOOP("Horizontal Center");
+#endif
+#ifdef V4L2_CID_VCENTER			// since kernel-3.9.0
 		case V4L2_CID_VCENTER:			return I18N_NOOP("Vertical Center");
+#endif
 #ifdef V4L2_CID_POWER_LINE_FREQUENCY	// since kernel 2.6.25
 		case V4L2_CID_POWER_LINE_FREQUENCY:	return I18N_NOOP("Power Line Frequency");
 		case V4L2_CID_HUE_AUTO:			return I18N_NOOP("Automatic Hue");
