diff -up kget/CMakeLists.txt.dso kget/CMakeLists.txt
--- kget/CMakeLists.txt.dso	2010-06-30 23:08:48.445214933 -0300
+++ kget/CMakeLists.txt	2010-06-30 23:09:08.529839662 -0300
@@ -280,7 +280,7 @@ endif(QCA2_FOUND)
 
 
 if (HAVE_NEPOMUK)
-  target_link_libraries(kget ${NEPOMUK_LIBRARIES})
+  target_link_libraries(kget ${SOPRANO_LIBRARIES} ${NEPOMUK_LIBRARIES})
 endif (HAVE_NEPOMUK)
 
 if (KONQUEROR_FOUND)
