diff -up kdenetwork-4.7.97/kopete/protocols/qq/evautil.cpp.fix-for-g++47 kdenetwork-4.7.97/kopete/protocols/qq/evautil.cpp
--- kdenetwork-4.7.97/kopete/protocols/qq/evautil.cpp.fix-for-g++47	2011-07-27 20:25:46.000000000 +0200
+++ kdenetwork-4.7.97/kopete/protocols/qq/evautil.cpp	2012-01-05 14:15:01.660149348 +0100
@@ -78,7 +78,9 @@ namespace Eva {
 		plain[0] = ( rand() & 0xf8 ) | pos;
 		memset( plain_pre, 0, 8 );
 		memset( crypted_pre, 0, 8 );
-		memset( plain+1, rand()& 0xff, pos++ );
+		if( pos )
+			memset( plain+1, rand()& 0xff, pos );
+		pos++;
 
 		// pad 2 bytes
 		for( i = 0; i< 2; i++ )
