diff -up kdenetwork-4.10.4/krdc/CMakeLists.txt.icon kdenetwork-4.10.4/krdc/CMakeLists.txt
--- kdenetwork-4.10.4/krdc/CMakeLists.txt.icon	2013-05-30 16:49:47.000000000 -0500
+++ kdenetwork-4.10.4/krdc/CMakeLists.txt	2013-06-01 13:38:17.506707887 -0500
@@ -87,7 +87,7 @@ kde4_add_ui_files(krdc_SRCS
     config/general.ui
 )
 
-kde4_add_app_icon(krdc_SRCS "${KDE4_INSTALL_DIR}/share/icons/oxygen/*/apps/krdc.png")
+kde4_add_app_icon(krdc_SRCS "hi*-apps-krdc.png")
 
 kde4_add_executable(krdc ${krdc_SRCS})
 
@@ -124,3 +124,5 @@ endif()
 if(NOT INSIDE_KDENETWORK)
     macro_display_feature_log()
 endif(NOT INSIDE_KDENETWORK)
+
+kde4_install_icons( ${ICON_INSTALL_DIR}   )
