--- kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc	2009-12-01 02:34:11.000000000 +0200
+++ kopete/protocols/jabber/googletalk/libjingle/talk/base/openssladapter.cc.new	2010-04-07 03:22:30.369164011 +0200
@@ -648,7 +648,16 @@
     int extension_nid = OBJ_obj2nid(X509_EXTENSION_get_object(extension));
 
     if (extension_nid == NID_subject_alt_name) {
+#if (OPENSSL_VERSION_NUMBER >= 0x10000000L)
+      const X509V3_EXT_METHOD *method;
+#else
+      X509V3_EXT_METHOD *method
+#endif
+#if (OPENSSL_VERSION_NUMBER >= 0x10000000L)
+      const X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#else
       X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#endif
       if (!meth)
         break;
 
