--- xfce-utils-4.7.4/scripts/xinitrc.in.orig	2011-01-02 11:21:26.000000000 -0200
+++ xfce-utils-4.7.4/scripts/xinitrc.in	2011-01-02 18:02:36.638017360 -0200
@@ -16,6 +16,13 @@ if test "x$XDG_MENU_PREFIX" = "x"; then
   export XDG_MENU_PREFIX
 fi
 
+#
+# Start pulseaudio for sound, if installed.
+#
+if test x"`which pulseaudio 2>/dev/null`" != x""; then
+  pulseaudio -D &
+fi
+
 # $XDG_CONFIG_HOME defines the base directory relative to which user specific
 # configuration files should be stored. If $XDG_CONFIG_HOME is either not set
 # or empty, a default equal to $HOME/.config should be used.
