--- qtgzmanager.pro.orig	2012-11-03 22:00:45.221781234 -0200
+++ qtgzmanager.pro	2012-11-03 22:00:54.445810645 -0200
@@ -4,7 +4,8 @@ QT += gui \
     network
 CONFIG += qt \
     console \
-    warn_on
+    warn_on \
+    qtsingleapplication
 DESTDIR += bin
 OBJECTS_DIR += build
 MOC_DIR += build
@@ -26,8 +27,6 @@ HEADERS += src/mainwindowimpl.h \
     src/tabwidget.h \
     src/searchlineedit.h \
     src/searchbar.h \
-    src/QtSolutions/qtsingleapplication.h \
-    src/QtSolutions/qtlocalpeer.h \
     src/packagecontroller.h \
     src/uihelper.h \
     src/wmhelper.h
@@ -52,8 +51,6 @@ SOURCES += src/main.cpp \
     src/tabwidget.cpp \
     src/searchlineedit.cpp \
     src/searchbar.cpp \
-    src/QtSolutions/qtsingleapplication.cpp \
-    src/QtSolutions/qtlocalpeer.cpp \
     src/packagecontroller.cpp \
     src/wmhelper.cpp
 FORMS += ui/mainwindow.ui \
