--- qtgzmanager.pro.orig	2014-07-11 22:14:50.000000000 -0300
+++ qtgzmanager.pro	2014-11-01 17:37:07.107090682 -0200
@@ -3,9 +3,9 @@ QT += gui \
     core \
     network
 CONFIG += qt \
-   debug \
    console \
-   warn_on
+   warn_on \
+   qtsingleapplication
 INCLUDEPATH += src
 DESTDIR += bin
 OBJECTS_DIR += build
@@ -28,8 +28,6 @@ HEADERS += src/mainwindowimpl.h \
     src/tabwidget.h \
     src/searchlineedit.h \
     src/searchbar.h \
-    src/QtSolutions/qtsingleapplication.h \
-    src/QtSolutions/qtlocalpeer.h \
     src/packagecontroller.h \
     src/uihelper.h \
     src/wmhelper.h
@@ -55,8 +53,6 @@ SOURCES += src/main.cpp \
     src/tabwidget.cpp \
     src/searchlineedit.cpp \
     src/searchbar.cpp \
-    src/QtSolutions/qtsingleapplication.cpp \
-    src/QtSolutions/qtlocalpeer.cpp \
     src/packagecontroller.cpp \
     src/wmhelper.cpp
 FORMS += ui/mainwindow.ui \
