diff -up src/firewall/core/ebtables.py.path src/firewall/core/ebtables.py
--- src/firewall/core/ebtables.py.path	2014-01-08 18:54:52.872999948 -0200
+++ src/firewall/core/ebtables.py	2014-01-08 18:55:01.882104881 -0200
@@ -37,7 +37,7 @@ class ebtables:
     ipv = "ipv4"
 
     def __init__(self):
-        self._command = "/sbin/ebtables"
+        self._command = "/usr/sbin/ebtables"
 
     def __run(self, args):
         # convert to string list
diff -up src/firewall/core/ipXtables.py.path src/firewall/core/ipXtables.py
--- src/firewall/core/ipXtables.py.path	2014-01-08 18:53:14.758635034 -0200
+++ src/firewall/core/ipXtables.py	2014-01-08 18:53:21.919129280 -0200
@@ -25,8 +25,8 @@ from firewall.core.prog import runProg
 from firewall.core.logger import log
 
 COMMAND = {
-    "ipv4": "/sbin/iptables",
-    "ipv6": "/sbin/ip6tables",
+    "ipv4": "/usr/sbin/iptables",
+    "ipv6": "/usr/sbin/ip6tables",
 }
 
 PROC_IPxTABLE_NAMES = {
