Index: gtk+-3.0.10/gtk/gtksettings.c
===================================================================
--- gtk+-3.0.10.orig/gtk/gtksettings.c	2011-05-23 04:12:33.000000000 +0200
+++ gtk+-3.0.10/gtk/gtksettings.c	2011-06-03 22:50:35.032963148 +0200
@@ -404,7 +404,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
                                                                   P_("Fallback Icon Theme Name"),
                                                                   P_("Name of a icon theme to fall back to"),
-                                                                  NULL,
+                                                                  "gnome",
                                                                   GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
