Index: gtk+-3.0.10/gtk/gtktreeview.c
===================================================================
--- gtk+-3.0.10.orig/gtk/gtktreeview.c	2011-05-23 04:12:26.000000000 +0200
+++ gtk+-3.0.10/gtk/gtktreeview.c	2011-06-03 22:50:40.652892494 +0200
@@ -10138,7 +10138,8 @@
 
       if (cursor_path)
 	{
-	  if (gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_MULTIPLE)
+	  if (gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_MULTIPLE ||
+	      gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
