Index: gtk+3.0-2.90.5/gtk/updateiconcache.c
===================================================================
--- gtk+3.0-2.90.5.orig/gtk/updateiconcache.c	2010-05-08 06:39:07.000000000 +0200
+++ gtk+3.0-2.90.5/gtk/updateiconcache.c	2010-07-30 19:12:50.495114783 +0200
@@ -660,7 +660,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 		
 	      image = g_new0 (Image, 1);
