Index: gtk+-3.3.20/gtk/updateiconcache.c
===================================================================
--- gtk+-3.3.20.orig/gtk/updateiconcache.c	2012-03-21 00:38:00.442485584 +0100
+++ gtk+-3.3.20/gtk/updateiconcache.c	2012-03-21 00:38:26.866484650 +0100
@@ -658,7 +658,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 		
 	      image = g_new0 (Image, 1);
