--- gtk+-3.10.0/gtk/gtktoolbar.c.orig	2013-09-24 21:22:15.918592820 -0300
+++ gtk+-3.10.0/gtk/gtktoolbar.c	2013-09-24 21:22:25.278612875 -0300
@@ -89,8 +89,8 @@ typedef struct _ToolbarContent ToolbarCo
 #define SPACE_LINE_START    2.0
 #define SPACE_LINE_END      8.0
 
-#define DEFAULT_ICON_SIZE GTK_ICON_SIZE_LARGE_TOOLBAR
-#define DEFAULT_TOOLBAR_STYLE GTK_TOOLBAR_BOTH_HORIZ
+#define DEFAULT_ICON_SIZE GTK_ICON_SIZE_SMALL_TOOLBAR
+#define DEFAULT_TOOLBAR_STYLE GTK_TOOLBAR_ICONS
 #define DEFAULT_ANIMATION_STATE TRUE
 
 #define MAX_HOMOGENEOUS_N_CHARS 13 /* Items that are wider than this do not participate
