--- gtk+-3.16.4/gtk/gtkscrolledwindow.c.orig	2015-06-30 18:53:05.210781889 -0300
+++ gtk+-3.16.4/gtk/gtkscrolledwindow.c	2015-06-30 18:59:37.370253819 -0300
@@ -596,7 +596,7 @@ gtk_scrolled_window_class_init (GtkScrol
                                    g_param_spec_boolean ("overlay-scrolling",
                                                          P_("Overlay Scrolling"),
                                                          P_("Overlay scrolling mode"),
-                                                         TRUE,
+                                                         FALSE,
                                                          GTK_PARAM_READWRITE|G_PARAM_EXPLICIT_NOTIFY));
 
   /**
@@ -1201,7 +1201,7 @@ gtk_scrolled_window_init (GtkScrolledWin
   priv->min_content_width = -1;
   priv->min_content_height = -1;
 
-  priv->overlay_scrolling = TRUE;
+  priv->overlay_scrolling = FALSE;
 
   priv->drag_gesture = gtk_gesture_drag_new (widget);
   gtk_gesture_single_set_touch_only (GTK_GESTURE_SINGLE (priv->drag_gesture), TRUE);
@@ -4052,8 +4052,8 @@ gtk_scrolled_window_update_use_indicator
 
   use_indicators = priv->overlay_scrolling;
 
-  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "0") == 0)
-    use_indicators = FALSE;
+  if (g_strcmp0 (g_getenv ("GTK_OVERLAY_SCROLLING"), "1") == 0)
+    use_indicators = TRUE;
 
   if (priv->use_indicators != use_indicators)
     {
