diff -up tbb30_20110419oss/build/linux.gcc.inc\~ tbb30_20110419oss/build/linux.gcc.inc
--- tbb30_20110419oss/build/linux.gcc.inc~	2011-04-19 13:48:58.000000000 +0200
+++ tbb30_20110419oss/build/linux.gcc.inc	2011-07-26 14:02:48.482006056 +0200
@@ -51,10 +51,10 @@ C_FLAGS = $(CPLUS_FLAGS)
 
 ITT_NOTIFY = -DDO_ITT_NOTIFY
 ifeq ($(cfg), release)
-        CPLUS_FLAGS = $(ITT_NOTIFY) -g -O2 -DUSE_PTHREAD
+        CPLUS_FLAGS = $(CXXFLAGS) $(ITT_NOTIFY) -O2 -DUSE_PTHREAD
 endif
 ifeq ($(cfg), debug)
-        CPLUS_FLAGS = -DTBB_USE_DEBUG $(ITT_NOTIFY) -g -O0 -DUSE_PTHREAD
+        CPLUS_FLAGS = $(CXXFLAGS) -DTBB_USE_DEBUG $(ITT_NOTIFY) -g -O0 -DUSE_PTHREAD
 endif
 
 ifneq (00,$(lambdas)$(cpp0x))

Diff finished.  Tue Jul 26 14:02:54 2011
