diff -up boost-1.54.0/libs/context/src/asm/jump_arm_aapcs_elf_gas.S~ boost-1.54.0/libs/context/src/asm/jump_arm_aapcs_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/jump_arm_aapcs_elf_gas.S~	2012-12-21 02:22:09.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/jump_arm_aapcs_elf_gas.S	2013-07-15 11:11:46.772543152 -0700
@@ -71,3 +71,6 @@ jump_fcontext:
                                     @ and as first arg in context function
     ldmia   a2, {v1-v8,sp-pc}       @ restore v1-V8,SP-PC
 .size jump_fcontext,.-jump_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/jump_i386_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/jump_i386_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/jump_i386_sysv_elf_gas.S~	2012-09-18 02:06:47.000000000 -0700
+++ boost-1.54.0/libs/context/src/asm/jump_i386_sysv_elf_gas.S	2013-07-15 11:10:32.829144835 -0700
@@ -70,3 +70,6 @@ jump_fcontext:
 
     jmp     *%edx                   /* indirect jump to context */
 .size jump_fcontext,.-jump_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/jump_ppc32_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/jump_ppc32_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/jump_ppc32_sysv_elf_gas.S~	2012-11-27 23:52:08.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/jump_ppc32_sysv_elf_gas.S	2013-07-15 11:10:57.848279687 -0700
@@ -178,3 +178,6 @@ jump_fcontext:
 
     bctr                        # jump to context
 .size jump_fcontext, .-jump_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/jump_ppc64_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/jump_ppc64_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/jump_ppc64_sysv_elf_gas.S~	2012-11-27 23:52:08.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/jump_ppc64_sysv_elf_gas.S	2013-07-15 11:11:07.189330597 -0700
@@ -199,3 +199,6 @@ jump_fcontext:
 
     bctr                        # jump to context
 .size .jump_fcontext, .-.jump_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/jump_x86_64_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/jump_x86_64_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/jump_x86_64_sysv_elf_gas.S~	2012-09-18 02:06:47.000000000 -0700
+++ boost-1.54.0/libs/context/src/asm/jump_x86_64_sysv_elf_gas.S	2013-07-15 11:09:49.724912924 -0700
@@ -80,3 +80,6 @@ jump_fcontext:
 
     jmp      *%rcx                  /* indirect jump to context */
 .size jump_fcontext,.-jump_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_arm_aapcs_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_arm_aapcs_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_arm_aapcs_elf_gas.S~	2013-01-15 01:51:39.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/make_arm_aapcs_elf_gas.S	2013-07-15 11:11:38.626499289 -0700
@@ -74,3 +74,6 @@ finish:
     mov     a1, #0          @ exit code is zero
     bl      _exit@PLT       @ exit application
 .size make_fcontext,.-make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_i386_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_i386_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_i386_sysv_elf_gas.S~	2013-06-07 08:38:10.000000000 -0700
+++ boost-1.54.0/libs/context/src/asm/make_i386_sysv_elf_gas.S	2013-07-15 11:10:40.333185226 -0700
@@ -75,3 +75,6 @@ finish:
     call   _exit@PLT                    /* exit application */
     hlt
 .size make_fcontext,.-make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_mips32_o32_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_mips32_o32_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_mips32_o32_elf_gas.S~	2012-11-27 23:52:08.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/make_mips32_o32_elf_gas.S	2013-07-15 11:12:34.392799546 -0700
@@ -83,3 +83,6 @@ finish:
     jalr    $t9                       # exit application
 .end make_fcontext
 .size make_fcontext, .-make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_ppc32_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_ppc32_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_ppc32_sysv_elf_gas.S~	2013-05-08 22:40:43.000000000 -0700
+++ boost-1.54.0/libs/context/src/asm/make_ppc32_sysv_elf_gas.S	2013-07-15 11:11:20.247400344 -0700
@@ -109,3 +109,6 @@ finish:
     li      %r3,  0             # exit code is zero
     bl      _exit@plt           # exit application
 .size make_fcontext, .-make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_ppc64_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_ppc64_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_ppc64_sysv_elf_gas.S~	2013-05-08 22:40:43.000000000 -0700
+++ boost-1.54.0/libs/context/src/asm/make_ppc64_sysv_elf_gas.S	2013-07-15 11:11:13.400363259 -0700
@@ -131,3 +131,6 @@ finish:
     bl      _exit               # exit application
     nop
 .size .make_fcontext, .-.make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_sparc64_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_sparc64_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_sparc64_sysv_elf_gas.S~	2013-02-12 04:55:22.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/make_sparc64_sysv_elf_gas.S	2013-07-15 11:12:41.643838416 -0700
@@ -84,3 +84,6 @@ finish:
      nop
 
 .size make_fcontext,.-make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_sparc_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_sparc_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_sparc_sysv_elf_gas.S~	2013-02-12 04:55:22.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/make_sparc_sysv_elf_gas.S	2013-07-15 11:12:48.733876794 -0700
@@ -80,3 +80,6 @@ finish:
      nop
 
 .size make_fcontext,.-make_fcontext
+
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
diff -up boost-1.54.0/libs/context/src/asm/make_x86_64_sysv_elf_gas.S~ boost-1.54.0/libs/context/src/asm/make_x86_64_sysv_elf_gas.S
--- boost-1.54.0/libs/context/src/asm/make_x86_64_sysv_elf_gas.S~	2012-11-29 23:38:52.000000000 -0800
+++ boost-1.54.0/libs/context/src/asm/make_x86_64_sysv_elf_gas.S	2013-07-15 11:10:06.151001366 -0700
@@ -72,3 +72,5 @@ finish:
     hlt
 .size make_fcontext,.-make_fcontext
 
+/* Mark that we don't need executable stack.  */
+.section .note.GNU-stack,"",@progbits
