--- tools/build/src/tools/gcc.jam.orig	2010-04-20 09:05:14.000000000 -0300
+++ tools/build/src/tools/gcc.jam	2010-08-17 12:31:12.583457030 -0300
@@ -383,7 +383,8 @@
     # OSF does have an option called -soname but it does not seem to work as
     # expected, therefore it has been disabled.
     HAVE_SONAME   = "" ;
-    SONAME_OPTION = -h ;
+    SONAME_OPTION = -soname ;
+    SONAME_VERSION = __SONAME__ ;
 }
 
 # HPUX, for some reason, seem to use '+h', not '-h'.
@@ -888,7 +889,7 @@
 # Differs from 'link' above only by -shared.
 actions link.dll bind LIBRARIES
 {
-    "$(CONFIG_COMMAND)" -L"$(LINKPATH)" -Wl,$(RPATH_OPTION:E=-R)$(SPACE)-Wl,$(RPATH) "$(.IMPLIB-COMMAND)$(<[1])" -o "$(<[-1])" $(HAVE_SONAME)-Wl,$(SONAME_OPTION)$(SPACE)-Wl,$(<[-1]:D=) -shared $(START-GROUP) "$(>)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS)
+    "$(CONFIG_COMMAND)" -L"$(LINKPATH)" -Wl,$(RPATH_OPTION:E=-R)$(SPACE)-Wl,"$(RPATH)" "$(.IMPLIB-COMMAND)$(<[1])" -o "$(<[-1])" $(HAVE_SONAME)-Wl,$(SONAME_OPTION)$(SPACE)-Wl,$(<[-1]:D=).$(SONAME_VERSION) -shared $(START-GROUP) "$(>)" "$(LIBRARIES)" $(FINDLIBS-ST-PFX) -l$(FINDLIBS-ST) $(FINDLIBS-SA-PFX) -l$(FINDLIBS-SA) $(END-GROUP) $(OPTIONS) $(USER_OPTIONS) $(SLKLDFLAGS)
 }
 
 rule setup-threading ( targets * : sources * : properties * )
