--- boostcpp.jam.orig	2012-10-27 18:28:54.975946733 -0200
+++ boostcpp.jam	2012-10-27 18:30:24.339144437 -0200
@@ -152,7 +152,7 @@ rule tag ( name : type ? : property-set
         # libFoo.1.2.3.dylib format. AIX linkers do not accept version suffixes
         # either. Pgi compilers can not accept a library with version suffix.
         if $(type) = SHARED_LIB &&
-          ! [ $(property-set).get <target-os> ] in windows cygwin darwin aix &&
+          ! [ $(property-set).get <target-os> ] in windows cygwin darwin aix linux &&
           ! [ $(property-set).get <toolset> ] in pgi
         {
             result = $(result).$(BOOST_VERSION)  ;
