--- Jamroot.orig	2009-09-18 00:02:06.000000000 -0300
+++ Jamroot	2009-09-18 00:12:44.033721195 -0300
@@ -384,7 +384,7 @@
         # suffixes either. Pgi compilers can not accept library with version
         # suffix.
         if $(type) = SHARED_LIB &&
-          ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix ) &&
+          ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix linux ) &&
             ! ( [ $(property-set).get <toolset> ] in pgi ) )
         {
             result = $(result).$(BOOST_VERSION)  ;
