diff -up boost_1_54_0/boost/math/distributions/inverse_gaussian.hpp~ boost_1_54_0/boost/math/distributions/inverse_gaussian.hpp
--- boost_1_54_0/boost/math/distributions/inverse_gaussian.hpp~	2012-09-07 01:43:19.000000000 -0700
+++ boost_1_54_0/boost/math/distributions/inverse_gaussian.hpp	2013-07-22 03:42:15.216142075 -0700
@@ -285,10 +285,8 @@ namespace detail
 
       // Define the distribution, using gamma_nooverflow:
       typedef gamma_distribution<RealType, no_overthrow_policy> gamma_nooverflow;
+      gamma_nooverflow g(static_cast<RealType>(0.5), static_cast<RealType>(1.));
 
-      gamma_distribution<RealType, no_overthrow_policy> g(static_cast<RealType>(0.5), static_cast<RealType>(1.));
-
-      // gamma_nooverflow g(static_cast<RealType>(0.5), static_cast<RealType>(1.));
       // R qgamma(0.2, 0.5, 1)  0.0320923
       RealType qg = quantile(complement(g, p));
       //RealType qg1 = qgamma(1.- p, 0.5, 1.0, true, false);
