--- boostcpp.jam.orig	2011-07-19 18:22:06.979491864 -0300
+++ boostcpp.jam	2011-07-19 18:22:15.855154557 -0300
@@ -154,7 +154,7 @@ rule tag ( name : type ? : property-set
         # suffixes either. Pgi compilers can not accept library with version
         # suffix.
         if $(type) = SHARED_LIB &&
-          ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix ) &&
+          ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix linux ) &&
             ! ( [ $(property-set).get <toolset> ] in pgi ) )
         {
             result = $(result).$(BOOST_VERSION)  ;
