diff -up boost_1_48_0/boost/lexical_cast.hpp\~ boost_1_48_0/boost/lexical_cast.hpp
--- boost_1_48_0/boost/lexical_cast.hpp~	2011-11-08 19:12:23.000000000 +0100
+++ boost_1_48_0/boost/lexical_cast.hpp	2011-11-29 01:27:29.368064527 +0100
@@ -1599,6 +1599,18 @@ namespace boost
         };
 
         template<typename T>
+        struct is_char
+        {
+            BOOST_STATIC_CONSTANT(bool, value = false );
+        };
+
+        template<>
+        struct is_char<char>
+        {
+            BOOST_STATIC_CONSTANT(bool, value = true );
+        };
+
+        template<typename T>
         struct is_char_or_wchar
         {
         private:
@@ -1732,11 +1744,7 @@ namespace boost
                                  is_arithmetic<src >::value,
                                  ::boost::type_traits::ice_and<
                                      is_pointer<src >::value,
-                                     is_char_or_wchar<removed_ptr_t >::value,
-                                     ::boost::type_traits::ice_eq<
-                                        sizeof(char_type),
-                                        sizeof(removed_ptr_t)
-                                     >::value
+                                     is_char<removed_ptr_t >::value
                                  >::value
                              >::value
                         );

Diff finished.  Tue Nov 29 01:27:49 2011
