diff -up kdeadmin-4.5.0/knetworkconf/backends/network-conf.cmake.slk kdeadmin-4.5.0/knetworkconf/backends/network-conf.cmake
--- kdeadmin-4.5.0/knetworkconf/backends/network-conf.cmake.slk	2010-08-15 22:58:20.228445081 -0300
+++ kdeadmin-4.5.0/knetworkconf/backends/network-conf.cmake	2010-08-15 23:10:03.979733338 -0300
@@ -77,7 +77,7 @@ $version = "@VERSION@";
               "mandriva-2006.1", "mandriva-2007.0", "mandriva-2007.1", "mandriva-2008.0", "mandriva-2008.1",
 	      "mandriva-2009.0", "mandriva-2009.1", "mandriva-2010.0",
               "vine-3.0", "vine-3.1",
-              "slackware-9.1.0", "slackware-10.0.0", "slackware-10.1.0",
+              "slackware",
 	      "freebsd-5", "freebsd-6", "freebsd-7", "freebsd-8", "freebsd-9",
               "gentoo", "vlos-1.2", "ark");
 
diff -up kdeadmin-4.5.0/knetworkconf/backends/network.pl.cmake.slk kdeadmin-4.5.0/knetworkconf/backends/network.pl.cmake
--- kdeadmin-4.5.0/knetworkconf/backends/network.pl.cmake.slk	2010-08-15 22:58:31.309054789 -0300
+++ kdeadmin-4.5.0/knetworkconf/backends/network.pl.cmake	2010-08-15 23:05:22.499387988 -0300
@@ -525,9 +525,7 @@ sub gst_network_get_broadcast_ping_cmd
        "specifix"     => "redhat-6.2",
        "vine-3.0"     => "redhat-6.2",
        "vine-3.1"     => "redhat-6.2",
-       "slackware-9.1.0" => "redhat-6.2",
-       "slackware-10.0.0" => "redhat-6.2",
-       "slackware-10.1.0" => "redhat-6.2",
+       "slackware"    => "redhat-6.2",
        "gentoo"       => "debian-2.2",
        "vlos-1.2"     => "debian-2.2",
        "freebsd-5"    => "debian-2.2",
@@ -1830,9 +1828,7 @@ sub gst_network_get_file
 	  "specifix"     => \&gst_network_rh72_get_file,
 	  "vine-3.0"     => \&gst_network_rh62_get_file,
 	  "vine-3.1"     => \&gst_network_rh62_get_file,
-	  "slackware-9.1.0" => \&gst_network_deb22_get_file,
-	  "slackware-10.0.0" => \&gst_network_deb22_get_file,
-	  "slackware-10.1.0" => \&gst_network_deb22_get_file,
+	  "slackware"    => \&gst_network_deb22_get_file,
 	  "gentoo"       => \&gst_network_deb22_get_file,
 	  "freebsd-5"    => \&gst_network_deb22_get_file,
 	  "freebsd-6"    => \&gst_network_deb22_get_file,
@@ -3955,9 +3951,7 @@ sub gst_network_ensure_loopback_interfac
 	  "specifix"     => "",
 	  "vine-3.0"     => "lo",
 	  "vine-3.1"     => "lo",
-	  "slackware-9.1.0" => "",
-	  "slackware-10.0.0" => "",
-	  "slackware-10.1.0" => "",
+	  "slackware"    => "",
 	  "gentoo"       => "",
 	  "vlos-1.2"     => "",
 	  "freebsd-5"    => "",
@@ -4106,9 +4100,7 @@ sub gst_network_get_parse_table
 	  "specifix"     => "redhat-7.2",
 	  "vine-3.0"     => "redhat-7.0",
 	  "vine-3.1"     => "redhat-7.0",
-	  "slackware-9.1.0" => "slackware-9.1.0",
-	  "slackware-10.0.0" => "slackware-9.1.0",
-	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware"    => "slackware",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -4378,7 +4370,7 @@ sub gst_network_get_parse_table
 		    ]
 			 },
 
-    "slackware-9.1.0" =>
+    "slackware" =>
     {
       fn =>
       {
@@ -4395,6 +4387,7 @@ sub gst_network_get_parse_table
        [ "hostname",      \&gst_parse_fq_hostname, HOSTNAME ],
        [ "nameserver",    \&gst_parse_split_all_unique_hash_comment, RESOLV_CONF,  "nameserver", "[ \t]+" ],
        [ "searchdomain",  \&gst_parse_split_first_array_unique, RESOLV_CONF,  "search", "[ \t]+", "[ \t]+" ],
+       [ "domain",        \&gst_parse_fq_domain,   HOSTNAME ],
        [ "domain",        \&gst_parse_split_first_str,   RESOLV_CONF,  "domain", "[ \t]+" ],
        [ "order",         \&gst_parse_split_first_array, HOST_CONF,    "order", "[ \t]+", ",[ \t]*" ],
        [ "hostmatch",     \&gst_parse_split_first_bool,  HOST_CONF,    "multi", "[ \t]+" ],
@@ -4608,9 +4601,7 @@ sub gst_network_get_interface_parse_tabl
 	  "specifix"     => "redhat-7.2",
 	  "vine-3.0"     => "vine-3.0",
 	  "vine-3.1"     => "vine-3.0",
-	  "slackware-9.1.0" => "slackware-9.1.0",
-	  "slackware-10.0.0" => "slackware-9.1.0",
-	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware"    => "slackware",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -5249,7 +5240,7 @@ sub gst_network_get_interface_parse_tabl
 #           [ "enabled",            \&gst_parse_trivial, 0 ]
           ]
     },
-    "slackware-9.1.0" =>
+    "slackware" =>
 	  {
       ifaces_get => \&gst_network_slackware91_ifaces_get_existing,
       fn =>
@@ -5505,9 +5496,7 @@ sub gst_network_get_replace_table
 	  "specifix"     => "redhat-7.2",
 	  "vine-3.0"     => "redhat-7.0",
 	  "vine-3.1"     => "redhat-7.0",
-	  "slackware-9.1.0" => "slackware-9.1.0",
-	  "slackware-10.0.0" => "slackware-9.1.0",
-	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware"    => "slackware",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -5764,7 +5753,7 @@ sub gst_network_get_replace_table
            [ "gateway",       \&gst_network_route_set_default_gw, "%gatewaydev%" ]
            ]
        },
-    "slackware-9.1.0" =>
+    "slackware" =>
     {
       fn =>
       {
@@ -5791,7 +5780,7 @@ sub gst_network_get_replace_table
        [ "smbdesc",       \&gst_replace_ini,              SMB_CONF,    "global", "server string" ],
        [ "winsserver",    \&gst_replace_ini,              SMB_CONF,    "global", "wins server" ],
        [ "winsuse",       \&gst_replace_ini_bool,         SMB_CONF,    "global", "wins support" ],
-       [ "smbuse",        \&gst_service_bsd_set_status,   "/etc/rc.d/rc.samba" ],
+       [ "smbuse",        \&gst_service_bsd_set_status,   "/etc/rc.d/rc.samba", "%smbuse%" ],
        [ "interface",     \&gst_network_interfaces_set,   OLD_HASH ],
        [ "gateway",       \&gst_network_route_set_default_gw, "%gatewaydev%" ]
       ]
@@ -5957,9 +5946,7 @@ sub gst_network_get_interface_replace_ta
 	  "specifix"     => "redhat-7.2",
 	  "vine-3.0"     => "vine-3.0",
 	  "vine-3.1"     => "vine-3.0",
-	  "slackware-9.1.0" => "slackware-9.1.0",
-	  "slackware-10.0.0" => "slackware-9.1.0",
-	  "slackware-10.1.0" => "slackware-9.1.0",
+	  "slackware"    => "slackware",
 	  "gentoo"       => "gentoo",
 	  "vlos-1.2"     => "vlos-1.2",
 	  "freebsd-5"    => "freebsd-5",
@@ -6562,7 +6549,7 @@ sub gst_network_get_interface_replace_ta
       ]
     },
 
-   "slackware-9.1.0" =>
+   "slackware" =>
    {
      iface_set    => \&gst_network_slackware91_interface_activate,
      iface_delete => \&gst_network_slackware91_interface_delete,
diff -up kdeadmin-4.5.0/knetworkconf/backends/platform.pl.cmake.slk kdeadmin-4.5.0/knetworkconf/backends/platform.pl.cmake
--- kdeadmin-4.5.0/knetworkconf/backends/platform.pl.cmake.slk	2010-08-15 22:58:42.874559896 -0300
+++ kdeadmin-4.5.0/knetworkconf/backends/platform.pl.cmake	2010-08-15 23:07:37.952200221 -0300
@@ -87,12 +87,7 @@ $PLATFORM_INFO = {
   "suse-9.0"        => "SuSE Linux 9.0",
   "suse-9.1"        => "SuSE Linux 9.1",
   "turbolinux-7.0"  => "Turbolinux 7.0",
-  "slackware-8.0.0" => "Slackware 8.0.0",
-  "slackware-8.1"   => "Slackware 8.1",
-  "slackware-9.0.0" => "Slackware 9.0.0",
-  "slackware-9.1.0" => "Slackware 9.1.0",
-  "slackware-10.0.0" => "Slackware 10.0.0",
-  "slackware-10.1.0" => "Slackware 10.1.0",
+  "slackware"       => "Slackware",
   "freebsd-4"       => "FreeBSD 4",
   "freebsd-5"       => "FreeBSD 5",
   "freebsd-6"       => "FreeBSD 6",
@@ -396,17 +391,7 @@ sub check_turbolinux
 
 sub check_slackware
 {
-  open RELEASE, "$gst_prefix/etc/slackware-version" or return -1;
-  while (<RELEASE>)
-  {
-    chomp;
-    if (/^Slackware ([0-9.]+)/)
-    {
-        close RELEASE;
-        return "slackware-$1";
-    }
-  }
-  close RELEASE;
+  return "slackware" if stat ("$gst_prefix/etc/slackware-version");
   return -1;
 }
 
diff -up kdeadmin-4.5.0/knetworkconf/backends/service.pl.cmake.slk kdeadmin-4.5.0/knetworkconf/backends/service.pl.cmake
--- kdeadmin-4.5.0/knetworkconf/backends/service.pl.cmake.slk	2010-08-15 22:59:02.969225120 -0300
+++ kdeadmin-4.5.0/knetworkconf/backends/service.pl.cmake	2010-08-15 23:11:25.730117081 -0300
@@ -278,9 +278,7 @@ sub gst_service_get_runlevel_roles
      "vine-3.0"       => "redhat-5.2",
      "vine-3.1"       => "redhat-5.2",
 
-     "slackware-9.1.0" => "slackware-9.1.0",
-     "slackware-10.0.0" => "slackware-9.1.0",
-     "slackware-10.1.0" => "slackware-9.1.0",
+     "slackware"      => "slackware",
 
      "gentoo"         => "gentoo",
      "vlos-1.2"       => "gentoo",
@@ -317,7 +315,7 @@ sub gst_service_get_runlevel_roles
                       "nonetwork" => "RECOVER"
                      },
      "freebsd-5"  => {"rc" => "GRAPHICAL" },
-     "slackware-9.1.0" => {"4" => "GRAPHICAL" }
+     "slackware"  => {"4" => "GRAPHICAL" }
     );
 
   $distro = $dist_map{$gst_dist};
