diff -up kdeadmin-4.8.4/ksystemlog/CMakeLists.txt.fedora kdeadmin-4.8.4/ksystemlog/CMakeLists.txt
--- kdeadmin-4.8.4/ksystemlog/CMakeLists.txt.fedora	2011-07-27 20:22:59.376235845 +0200
+++ kdeadmin-4.8.4/ksystemlog/CMakeLists.txt	2012-06-25 18:52:19.340210917 +0200
@@ -37,7 +37,6 @@ add_subdirectory(src/modes/xorg)
 add_subdirectory(src/modes/cron)
 add_subdirectory(src/modes/authentication) 
 add_subdirectory(src/modes/daemon) 
-add_subdirectory(src/modes/acpid)
 add_subdirectory(src/modes/xsession)
 add_subdirectory(src/modes/apache) 
 add_subdirectory(src/modes/postfix) 
diff -up kdeadmin-4.8.4/ksystemlog/src/CMakeLists.txt.fedora kdeadmin-4.8.4/ksystemlog/src/CMakeLists.txt
--- kdeadmin-4.8.4/ksystemlog/src/CMakeLists.txt.fedora	2011-07-27 20:22:59.296236648 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/CMakeLists.txt	2012-06-25 18:52:19.340210917 +0200
@@ -15,7 +15,6 @@ include_directories(
 	${CMAKE_CURRENT_SOURCE_DIR}/modes/daemon
 	${CMAKE_CURRENT_SOURCE_DIR}/modes/xorg
 	${CMAKE_CURRENT_SOURCE_DIR}/modes/cron
-	${CMAKE_CURRENT_SOURCE_DIR}/modes/acpid
 	${CMAKE_CURRENT_SOURCE_DIR}/modes/xsession
 	${CMAKE_CURRENT_SOURCE_DIR}/modes/apache
 	${CMAKE_CURRENT_SOURCE_DIR}/modes/cups
@@ -59,7 +58,6 @@ target_link_libraries(ksystemlog
 	ksystemlog_apache
 	ksystemlog_authentication 
 	ksystemlog_daemon 
-	ksystemlog_acpid 
 	ksystemlog_xsession
 	ksystemlog_postfix 
 	ksystemlog_cups 
diff -up kdeadmin-4.8.4/ksystemlog/src/logModePluginsLoader.cpp.fedora kdeadmin-4.8.4/ksystemlog/src/logModePluginsLoader.cpp
--- kdeadmin-4.8.4/ksystemlog/src/logModePluginsLoader.cpp.fedora	2011-07-27 20:22:59.295236659 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/logModePluginsLoader.cpp	2012-06-25 18:52:19.341210917 +0200
@@ -39,7 +39,6 @@
 #include "apacheFactory.h"
 #include "cupsFactory.h"
 #include "postfixFactory.h"
-#include "acpidFactory.h"
 #include "xsessionFactory.h"
 #include "sambaFactory.h"
 
@@ -87,9 +86,6 @@ void LogModePluginsLoader::loadPlugins()
 	//Samba
 	Globals::instance()->registerLogModeFactory(new SambaLogModeFactory());
 	
-	//Acpid
-	Globals::instance()->registerLogModeFactory(new AcpidLogModeFactory());
-	
 	//XSession
 	Globals::instance()->registerLogModeFactory(new XSessionLogModeFactory());
 
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/apache/apacheConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/apache/apacheConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/apache/apacheConfiguration.h.fedora	2011-07-27 20:22:58.794241689 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/apache/apacheConfiguration.h	2012-06-25 18:52:19.341210917 +0200
@@ -51,11 +51,11 @@ class ApacheConfiguration : public LogMo
 			configuration->setCurrentGroup(QLatin1String( "ApacheLogMode" ));
 
 			QStringList defaultApachePaths;
-			defaultApachePaths << QLatin1String( "/var/log/apache2/error.log" );
+			defaultApachePaths << QLatin1String( "/var/log/httpd/error_log" );
 			configuration->addItemStringList(QLatin1String( "ApacheLogFilesPaths" ), d->apachePaths, defaultApachePaths, QLatin1String( "ApacheLogFilesPaths" ));
 
 			QStringList defaultApacheAccessPaths;
-			defaultApacheAccessPaths << QLatin1String( "/var/log/apache2/access.log" );
+			defaultApacheAccessPaths << QLatin1String( "/var/log/httpd/access_log" );
 			configuration->addItemStringList(QLatin1String( "ApacheAccessLogFilesPaths" ), d->apacheAccessPaths, defaultApacheAccessPaths, QLatin1String( "ApacheAccessLogFilesPaths" ));
 		}
 
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/authentication/authenticationConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/authentication/authenticationConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/authentication/authenticationConfiguration.h.fedora	2011-07-27 20:22:58.534244296 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/authentication/authenticationConfiguration.h	2012-06-25 18:52:19.341210917 +0200
@@ -51,7 +51,7 @@ class AuthenticationConfiguration : publ
 
 			configuration->setCurrentGroup(QLatin1String( "AuthenticationLogMode" ));
 
-			QString defaultAuthenticationPath(QLatin1String( "/var/log/auth.log" ));
+			QString defaultAuthenticationPath(QLatin1String( "/var/log/secure" ));
 			configuration->addItemString(QLatin1String( "LogFilePath" ), d->authenticationPath, defaultAuthenticationPath, QLatin1String( "LogFilePath" ));
 
 			QStringList defaultWarningKeywords;
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/cron/cronConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/cron/cronConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/cron/cronConfiguration.h.fedora	2011-07-27 20:22:58.850241126 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/cron/cronConfiguration.h	2012-06-25 18:52:19.341210917 +0200
@@ -51,7 +51,7 @@ class CronConfiguration : public LogMode
 			configuration->setCurrentGroup(QLatin1String( "CronLogMode" ));
 
 			QStringList defaultCronPaths;
-			defaultCronPaths << QLatin1String( "/var/log/syslog" );
+			defaultCronPaths << QLatin1String( "/var/log/cron" );
 			configuration->addItemStringList(QLatin1String( "LogFilesPaths" ), d->cronPaths, defaultCronPaths, QLatin1String( "LogFilesPaths" ));
 
 			QString defaultProcessFilter(QLatin1String( "/usr/sbin/cron" ));
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/cups/CMakeLists.txt.fedora kdeadmin-4.8.4/ksystemlog/src/modes/cups/CMakeLists.txt
--- kdeadmin-4.8.4/ksystemlog/src/modes/cups/CMakeLists.txt.fedora	2012-06-25 19:00:17.116205526 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/cups/CMakeLists.txt	2012-06-25 19:00:24.882204993 +0200
@@ -24,10 +24,6 @@ set(ksystemlog_cups_sources
 	cupsPageItemBuilder.cpp
 	cupsPageLogMode.cpp
 	
-	cupsPdfAnalyzer.cpp
-	cupsPdfItemBuilder.cpp
-	cupsPdfLogMode.cpp
-	
 	cupsFactory.cpp
 	
 )
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsConfiguration.h.fedora	2011-07-27 20:22:59.198237633 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsConfiguration.h	2012-06-25 18:52:19.342210917 +0200
@@ -65,10 +65,6 @@ class CupsConfiguration : public LogMode
 			QStringList defaultCupsPagePaths;
 			defaultCupsPagePaths << QLatin1String( "/var/log/cups/page_log" );
 			configuration->addItemStringList(QLatin1String( "CupsPageLogFilesPaths" ), d->cupsPagePaths, defaultCupsPagePaths, QLatin1String( "CupsPageLogFilesPaths" ));
-
-			QStringList defaultCupsPdfPaths;
-			defaultCupsPdfPaths << QLatin1String( "/var/log/cups/cups-pdf_log" );
-			configuration->addItemStringList(QLatin1String( "CupsPdfLogFilesPaths" ), d->cupsPdfPaths, defaultCupsPdfPaths, QLatin1String( "CupsPdfLogFilesPaths" ));
 		}
 
 		virtual ~CupsConfiguration() {
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsFactory.cpp.fedora kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsFactory.cpp
--- kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsFactory.cpp.fedora	2012-06-25 19:02:20.105201541 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/cups/cupsFactory.cpp	2012-06-25 19:03:16.313199734 +0200
@@ -31,7 +31,6 @@
 #include "cupsLogMode.h"
 #include "cupsAccessLogMode.h"
 #include "cupsPageLogMode.h"
-#include "cupsPdfLogMode.h"
 
 #include "cupsConfigurationWidget.h"
 #include "cupsConfiguration.h"
@@ -47,7 +46,6 @@ QList<LogMode*> CupsLogModeFactory::crea
 	logModes.append(new CupsLogMode(logModeConfiguration, logModeConfigurationWidget));
 	logModes.append(new CupsAccessLogMode(logModeConfiguration, logModeConfigurationWidget));
 	logModes.append(new CupsPageLogMode(logModeConfiguration, logModeConfigurationWidget));
-	logModes.append(new CupsPdfLogMode(logModeConfiguration, logModeConfigurationWidget));
 
 	return logModes;
 }
@@ -59,7 +57,6 @@ LogModeAction* CupsLogModeFactory::creat
 	multipleActions->addInnerAction(cupsLogMode->action());
 	multipleActions->addInnerAction(Globals::instance()->findLogMode(QLatin1String( CUPS_ACCESS_LOG_MODE_ID ))->action());
 	multipleActions->addInnerAction(Globals::instance()->findLogMode(QLatin1String( CUPS_PAGE_LOG_MODE_ID ))->action());
-	multipleActions->addInnerAction(Globals::instance()->findLogMode(QLatin1String( CUPS_PDF_LOG_MODE_ID ))->action());
 
 	multipleActions->setInToolBar(false);
 	multipleActions->setCategory(LogModeAction::ServicesCategory);
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/daemon/daemonConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/daemon/daemonConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/daemon/daemonConfiguration.h.fedora	2011-07-27 20:22:58.950240123 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/daemon/daemonConfiguration.h	2012-06-25 18:52:19.342210917 +0200
@@ -47,7 +47,7 @@ class DaemonConfiguration : public LogMo
 			configuration->setCurrentGroup(QLatin1String( "DaemonLogMode" ));
 
 			QStringList defaultDaemonPaths;
-			defaultDaemonPaths << QLatin1String( "/var/log/daemon.log" );
+			defaultDaemonPaths << QLatin1String( "/var/log/boot.log" );
 			configuration->addItemStringList(QLatin1String( "LogFilesPaths" ), d->daemonPaths, defaultDaemonPaths, QLatin1String( "LogFilesPaths" ));
 
 		}
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/postfix/postfixConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/postfix/postfixConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/postfix/postfixConfiguration.h.fedora	2011-07-27 20:22:59.288236729 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/postfix/postfixConfiguration.h	2012-06-25 18:52:19.342210917 +0200
@@ -39,7 +39,7 @@ class PostfixConfiguration : public Gene
 		PostfixConfiguration() :
 			GenericLogModeConfiguration(
 					QLatin1String( POSTFIX_LOG_MODE_ID ),
-					QStringList() << QLatin1String( "/var/log/mail.log" ) << QLatin1String( "/var/log/mail.info" ) << QLatin1String( "/var/log/mail.warn" ) << QLatin1String( "/var/log/mail.err" ),
+					QStringList() << QLatin1String( "/var/log/maillog" ),
 					QList<int>() << NOTICE_LOG_LEVEL_ID << INFORMATION_LOG_LEVEL_ID << WARNING_LOG_LEVEL_ID << ERROR_LOG_LEVEL_ID
 			) {
 
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/samba/sambaConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/samba/sambaConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/samba/sambaConfiguration.h.fedora	2012-06-25 18:57:50.213208790 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/samba/sambaConfiguration.h	2012-06-25 18:58:08.394208697 +0200
@@ -55,14 +55,6 @@ class SambaConfiguration : public LogMod
 			QStringList defaultSambaPaths;
 			defaultSambaPaths << QLatin1String( "/var/log/samba/log.smbd" );
 			configuration->addItemStringList(QLatin1String( "SambaLogFilesPaths" ), d->sambaPaths, defaultSambaPaths, QLatin1String( "SambaLogFilesPaths" ));
-
-			QStringList defaultSambaAccessPaths;
-			defaultSambaAccessPaths << QLatin1String( "/var/log/samba/log.localhost" )  << QLatin1String( "/var/log/samba/log.127.0.0.1" );
-			configuration->addItemStringList(QLatin1String( "SambaAccessLogFilesPaths" ), d->sambaAccessPaths, defaultSambaAccessPaths, QLatin1String( "SambaAccessLogFilesPaths" ));
-
-			QStringList defaultNetbiosPaths;
-			defaultNetbiosPaths << QLatin1String( "/var/log/samba/log.nmbd" );
-			configuration->addItemStringList(QLatin1String( "NetbiosLogFilesPaths" ), d->netbiosPaths, defaultNetbiosPaths, QLatin1String( "NetbiosLogFilesPaths" ));
 		}
 
 		virtual ~SambaConfiguration() {
diff -up kdeadmin-4.8.4/ksystemlog/src/modes/system/systemConfiguration.h.fedora kdeadmin-4.8.4/ksystemlog/src/modes/system/systemConfiguration.h
--- kdeadmin-4.8.4/ksystemlog/src/modes/system/systemConfiguration.h.fedora	2011-07-27 20:22:59.067238948 +0200
+++ kdeadmin-4.8.4/ksystemlog/src/modes/system/systemConfiguration.h	2012-06-25 18:52:19.342210917 +0200
@@ -39,7 +39,7 @@ class SystemConfiguration : public Gener
 		SystemConfiguration() :
 			GenericLogModeConfiguration(
 					QLatin1String( SYSTEM_LOG_MODE_ID ),
-					QStringList() << QLatin1String( "/var/log/syslog" ),
+					QStringList() << QLatin1String( "/var/log/messages" ),
 					QList<int>() << INFORMATION_LOG_LEVEL_ID
 			) {
 
