diff --git a/conf/postfix-files b/conf/postfix-files
index 9e1804a..b4567e1 100644
--- a/conf/postfix-files
+++ b/conf/postfix-files
@@ -139,18 +139,13 @@ $command_directory/postqueue:f:root:$set
 $sendmail_path:f:root:-:755
 $newaliases_path:l:$sendmail_path
 $mailq_path:l:$sendmail_path
-$config_directory/LICENSE:f:root:-:644:1
-$config_directory/TLS_LICENSE:f:root:-:644:1
 $config_directory/access:f:root:-:644:p1
-$config_directory/aliases:f:root:-:644:p1
-$config_directory/bounce.cf.default:f:root:-:644:1
 $config_directory/canonical:f:root:-:644:p1
 $config_directory/cidr_table:f:root:-:644:o
 $config_directory/generic:f:root:-:644:p1
 $config_directory/generics:f:root:-:644:o
 $config_directory/header_checks:f:root:-:644:p1
 $config_directory/install.cf:f:root:-:644:o
-$config_directory/main.cf.default:f:root:-:644:1
 $config_directory/main.cf:f:root:-:644:p
 $config_directory/master.cf:f:root:-:644:p
 $config_directory/pcre_table:f:root:-:644:o
@@ -163,8 +158,8 @@ $config_directory/postfix-script:f:root:
 $config_directory/postfix-script-sgid:f:root:-:755:o
 $config_directory/postfix-script-nosgid:f:root:-:755:o
 $config_directory/post-install:f:root:-:755:o
-$manpage_directory/man1/mailq.1:f:root:-:644
-$manpage_directory/man1/newaliases.1:f:root:-:644
+$manpage_directory/man1/mailq.postfix.1:f:root:-:644
+$manpage_directory/man1/newaliases.postfix.1:f:root:-:644
 $manpage_directory/man1/postalias.1:f:root:-:644
 $manpage_directory/man1/postcat.1:f:root:-:644
 $manpage_directory/man1/postconf.1:f:root:-:644
@@ -177,9 +172,9 @@ $manpage_directory/man1/postmap.1:f:root
 $manpage_directory/man1/postmulti.1:f:root:-:644
 $manpage_directory/man1/postqueue.1:f:root:-:644
 $manpage_directory/man1/postsuper.1:f:root:-:644
-$manpage_directory/man1/sendmail.1:f:root:-:644
+$manpage_directory/man1/sendmail.postfix.1:f:root:-:644
 $manpage_directory/man5/access.5:f:root:-:644
-$manpage_directory/man5/aliases.5:f:root:-:644
+$manpage_directory/man5/aliases.postfix.5:f:root:-:644
 $manpage_directory/man5/body_checks.5:f:root:-:644
 $manpage_directory/man5/bounce.5:f:root:-:644
 $manpage_directory/man5/canonical.5:f:root:-:644
@@ -226,7 +221,7 @@ $manpage_directory/man8/qmqpd.8:f:root:-
 $manpage_directory/man8/scache.8:f:root:-:644
 $manpage_directory/man8/showq.8:f:root:-:644
 $manpage_directory/man8/smtp.8:f:root:-:644
-$manpage_directory/man8/smtpd.8:f:root:-:644
+$manpage_directory/man8/smtpd.postfix.8:f:root:-:644
 $manpage_directory/man8/spawn.8:f:root:-:644
 $manpage_directory/man8/tlsproxy.8:f:root:-:644
 $manpage_directory/man8/tlsmgr.8:f:root:-:644
