---
 common/xfpm-brightness.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/common/xfpm-brightness.c
+++ b/common/xfpm-brightness.c
@@ -65,10 +65,13 @@
 {
     XRRPropertyInfo *info;
     gboolean ret = TRUE;
+    gint rc;
 
+    gdk_error_trap_push ();
     info = XRRQueryOutputProperty (GDK_DISPLAY (), output, brightness->priv->backlight);
+    rc = gdk_error_trap_pop ();
     
-    if (info == NULL) 
+    if (info == NULL || rc != Success)
     {
 	g_warning ("could not get output property");
 	return FALSE;
