diff -Nur xfce4-power-manager-1.1.0.orig/settings/xfce4-power-manager-settings.desktop.in xfce4-power-manager-1.1.0/settings/xfce4-power-manager-settings.desktop.in
--- xfce4-power-manager-1.1.0.orig/settings/xfce4-power-manager-settings.desktop.in	2012-04-13 16:52:28.000000000 -0600
+++ xfce4-power-manager-1.1.0/settings/xfce4-power-manager-settings.desktop.in	2012-04-14 16:09:16.041310651 -0600
@@ -7,7 +7,7 @@
 Icon=xfpm-ac-adapter
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-HardwareSettings;
 OnlyShowIn=XFCE;
 StartupNotify=true
 X-XfcePluggable=true
