--- elinks-0.10.1/src/config/options.c.xterm	2005-01-04 01:50:29.000000000 +0100
+++ elinks-0.10.1/src/config/options.c	2005-01-28 18:32:34.523579120 +0100
@@ -689,8 +689,9 @@ register_autocreated_options(void)
 	get_opt_bool("terminal.linux.m11_hack") = 1;
 	get_opt_int("terminal.vt100.type") = TERM_VT100;
 	get_opt_int("terminal.vt110.type") = TERM_VT100;
-	get_opt_int("terminal.xterm.type") = TERM_VT100;
+	get_opt_int("terminal.xterm.type") = 2;
 	get_opt_bool("terminal.xterm.underline") = 1;
+	get_opt_int("terminal.xterm.colors") = 1;
 	get_opt_int("terminal.xterm-color.type") = TERM_VT100;
 	get_opt_int("terminal.xterm-color.colors") = COLOR_MODE_16;
 	get_opt_bool("terminal.xterm-color.underline") = 1;
