--- src/xml/repr-io.cpp~	2011-07-08 13:25:09.000000000 -0500
+++ src/xml/repr-io.cpp	2012-11-01 08:59:23.620360123 -0500
@@ -294,7 +294,7 @@
                              &src,
                              localFilename,
                              src.getEncoding(),
-                             XML_PARSE_NOENT );
+                             XML_PARSE_NOENT | XML_PARSE_HUGE);
         }
     }
 
