diff -up libatomic_ops-7.2/configure.ac.shared libatomic_ops-7.2/configure.ac
--- libatomic_ops-7.2/configure.ac.shared	2014-05-07 19:01:08.801942059 -0300
+++ libatomic_ops-7.2/configure.ac	2014-05-07 19:02:19.810982900 -0300
@@ -4,7 +4,8 @@ AC_INIT([libatomic_ops],[7.2])
 AC_CANONICAL_TARGET([])
 AC_CONFIG_SRCDIR(src/atomic_ops.c)
 AM_INIT_AUTOMAKE
-AC_PROG_RANLIB
+AC_LIBTOOL_DLOPEN
+AC_PROG_LIBTOOL
 
 AC_CONFIG_HEADERS([src/config.h])
 
diff -up libatomic_ops-7.2/src/Makefile.am.shared libatomic_ops-7.2/src/Makefile.am
--- libatomic_ops-7.2/src/Makefile.am.shared	2014-05-07 19:01:17.105192887 -0300
+++ libatomic_ops-7.2/src/Makefile.am	2014-05-07 19:02:19.819981004 -0300
@@ -4,14 +4,14 @@ AM_CFLAGS=@PICFLAG@
 AM_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
 
 include_HEADERS=atomic_ops.h atomic_ops_stack.h atomic_ops_malloc.h
-lib_LIBRARIES = libatomic_ops.a libatomic_ops_gpl.a
+lib_LTLIBRARIES = libatomic_ops.la libatomic_ops_gpl.la
 if NEED_ASM
-libatomic_ops_a_SOURCES = atomic_ops.c atomic_ops_sysdeps.S
+libatomic_ops_la_SOURCES = atomic_ops.c atomic_ops_sysdeps.S
 else
-libatomic_ops_a_SOURCES = atomic_ops.c
+libatomic_ops_la_SOURCES = atomic_ops.c
 endif
 
-libatomic_ops_gpl_a_SOURCES = atomic_ops_stack.c atomic_ops_malloc.c
+libatomic_ops_gpl_la_SOURCES = atomic_ops_stack.c atomic_ops_malloc.c
 
 EXTRA_DIST=Makefile.msft
 
diff -up libatomic_ops-7.2/tests/Makefile.am.shared libatomic_ops-7.2/tests/Makefile.am
--- libatomic_ops-7.2/tests/Makefile.am.shared	2014-05-07 19:01:32.645919034 -0300
+++ libatomic_ops-7.2/tests/Makefile.am	2014-05-07 19:02:20.324874638 -0300
@@ -20,15 +20,15 @@ endif
 check_PROGRAMS=$(TESTS)
 
 test_atomic_SOURCES=test_atomic.c
-test_atomic_LDADD=$(THREADDLLIBS) ../src/libatomic_ops.a
+test_atomic_LDADD=$(THREADDLLIBS) ../src/.libs/libatomic_ops.so
 
 test_stack_SOURCES=test_stack.c
-test_stack_LDADD=$(THREADDLLIBS) ../src/libatomic_ops_gpl.a \
-	../src/libatomic_ops.a
+test_stack_LDADD=$(THREADDLLIBS) ../src/.libs/libatomic_ops_gpl.so \
+	../src/.libs/libatomic_ops.so
 
 test_malloc_SOURCES=test_malloc.c
-test_malloc_LDADD=$(THREADDLLIBS) ../src/libatomic_ops_gpl.a \
-	../src/libatomic_ops.a
+test_malloc_LDADD=$(THREADDLLIBS) ../src/.libs/libatomic_ops_gpl.so \
+	../src/.libs/libatomic_ops.so
 
 test_atomic_include.h: test_atomic.template
 	sed -e s/XX// $? > $@
