diff -up epplets-0.12/epplets/Makefile.in~ epplets-0.12/epplets/Makefile.in
--- epplets-0.12/epplets/Makefile.in~	2009-01-03 10:10:34.000000000 +0100
+++ epplets-0.12/epplets/Makefile.in	2010-02-14 20:34:25.122238562 +0100
@@ -85,7 +85,7 @@ E_Biff_epplet_LINK = $(LIBTOOL) --tag=CC
 	$(E_Biff_epplet_LDFLAGS) $(LDFLAGS) -o $@
 am_E_Clock_epplet_OBJECTS = E-Clock.$(OBJEXT)
 E_Clock_epplet_OBJECTS = $(am_E_Clock_epplet_OBJECTS)
-E_Clock_epplet_LDADD = $(LDADD)
+E_Clock_epplet_LDADD = $(LDADD) -lm
 E_Clock_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_Clock_epplet_LDFLAGS) $(LDFLAGS) -o $@
@@ -137,13 +137,13 @@ E_Mixer_epplet_LINK = $(LIBTOOL) --tag=C
 am_E_MoonClock_epplet_OBJECTS = E-MoonClock.$(OBJEXT) \
 	CalcEphem.$(OBJEXT) Moon.$(OBJEXT)
 E_MoonClock_epplet_OBJECTS = $(am_E_MoonClock_epplet_OBJECTS)
-E_MoonClock_epplet_LDADD = $(LDADD)
+E_MoonClock_epplet_LDADD = $(LDADD) -lm
 E_MoonClock_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_MoonClock_epplet_LDFLAGS) $(LDFLAGS) -o $@
 am_E_Mountbox_epplet_OBJECTS = E-Mountbox.$(OBJEXT)
 E_Mountbox_epplet_OBJECTS = $(am_E_Mountbox_epplet_OBJECTS)
-E_Mountbox_epplet_LDADD = $(LDADD)
+E_Mountbox_epplet_LDADD = $(LDADD) -lm
 E_Mountbox_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_Mountbox_epplet_LDFLAGS) $(LDFLAGS) -o $@
@@ -161,7 +161,7 @@ E_NetFlame_epplet_LINK = $(LIBTOOL) --ta
 	$(E_NetFlame_epplet_LDFLAGS) $(LDFLAGS) -o $@
 am_E_NetGraph_epplet_OBJECTS = E-NetGraph.$(OBJEXT) net.$(OBJEXT)
 E_NetGraph_epplet_OBJECTS = $(am_E_NetGraph_epplet_OBJECTS)
-E_NetGraph_epplet_LDADD = $(LDADD)
+E_NetGraph_epplet_LDADD = $(LDADD) -lm
 E_NetGraph_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_NetGraph_epplet_LDFLAGS) $(LDFLAGS) -o $@
@@ -196,21 +196,21 @@ E_Power_epplet_LINK = $(LIBTOOL) --tag=C
 am_E_SD_epplet_OBJECTS = E-SD.$(OBJEXT) esdcloak.$(OBJEXT) \
 	utils.$(OBJEXT)
 E_SD_epplet_OBJECTS = $(am_E_SD_epplet_OBJECTS)
-E_SD_epplet_LDADD = $(LDADD)
+E_SD_epplet_LDADD = $(LDADD) -lm
 E_SD_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_SD_epplet_LDFLAGS) $(LDFLAGS) -o $@
 am_E_ScreenSave_epplet_OBJECTS = E-ScreenSave.$(OBJEXT) \
 	cloak.$(OBJEXT) utils.$(OBJEXT)
 E_ScreenSave_epplet_OBJECTS = $(am_E_ScreenSave_epplet_OBJECTS)
-E_ScreenSave_epplet_LDADD = $(LDADD)
+E_ScreenSave_epplet_LDADD = $(LDADD) -lm
 E_ScreenSave_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_ScreenSave_epplet_LDFLAGS) $(LDFLAGS) -o $@
 am_E_ScreenShoot_epplet_OBJECTS = E-ScreenShoot.$(OBJEXT) \
 	cloak.$(OBJEXT) utils.$(OBJEXT) E-ScreenShoot_ftp.$(OBJEXT)
 E_ScreenShoot_epplet_OBJECTS = $(am_E_ScreenShoot_epplet_OBJECTS)
-E_ScreenShoot_epplet_LDADD = $(LDADD)
+E_ScreenShoot_epplet_LDADD = $(LDADD) -lm
 E_ScreenShoot_epplet_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(E_ScreenShoot_epplet_LDFLAGS) $(LDFLAGS) -o $@
