From 95ae1809a18f37d50e7746324daccc46a317246d Mon Sep 17 00:00:00 2001
From: Vincent Penquerc'h <ogg.k.ogg.k@googlemail.com>
Date: Wed, 5 Oct 2011 22:07:50 +0100
Subject: [PATCH] kate: do not access the renderer if it fails to initialize

---
 modules/codec/kate.c |   17 +++++++++--------
 1 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/modules/codec/kate.c b/modules/codec/kate.c
index 49177da..769e0d9 100644
--- a/modules/codec/kate.c
+++ b/modules/codec/kate.c
@@ -404,14 +404,15 @@ static int OpenDecoder( vlc_object_t *p_this )
             p_sys->p_tr = NULL;
             p_sys->b_use_tiger = false;
         }
-
-        CHECK_TIGER_RET( tiger_renderer_set_surface_clear_color( p_sys->p_tr, 1, 0, 0, 0, 0 ) );
-
-        UpdateTigerFontEffect( p_dec );
-        UpdateTigerFontColor( p_dec );
-        UpdateTigerBackgroundColor( p_dec );
-        UpdateTigerQuality( p_dec );
-        UpdateTigerFontDesc( p_dec );
+        else {
+            CHECK_TIGER_RET( tiger_renderer_set_surface_clear_color( p_sys->p_tr, 1, 0, 0, 0, 0 ) );
+
+            UpdateTigerFontEffect( p_dec );
+            UpdateTigerFontColor( p_dec );
+            UpdateTigerBackgroundColor( p_dec );
+            UpdateTigerQuality( p_dec );
+            UpdateTigerFontDesc( p_dec );
+        }
     }
 
 #else
-- 
1.7.1

