--- vlc-2.0.0/modules/access/v4l2/video.c.orig	2012-02-12 14:28:47.000000000 -0200
+++ vlc-2.0.0/modules/access/v4l2/video.c	2012-02-19 03:39:44.403592760 -0200
@@ -337,7 +337,7 @@ vlc_module_begin ()
     add_float( CFG_PREFIX "fps", 0, FPS_TEXT, FPS_LONGTEXT, true )
         change_safe()
 #ifdef HAVE_LIBV4L2
-    add_bool( CFG_PREFIX "use-libv4l2", false, LIBV4L2_TEXT, LIBV4L2_LONGTEXT, true );
+    add_bool( CFG_PREFIX "use-libv4l2", true, LIBV4L2_TEXT, LIBV4L2_LONGTEXT, true );
 #endif
 
     set_section( N_( "Tuner" ), NULL )
