From 5694c98039d550984e3e8d3a13341cc9543d3950 Mon Sep 17 00:00:00 2001
From: Nicolas Chauvet <kwizart@gmail.com>
Date: Fri, 7 Oct 2011 21:46:37 +0200
Subject: [PATCH] Fix typo in vlc.desktop

---
 share/vlc.desktop |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/share/vlc.desktop b/share/vlc.desktop
index abc39d9..71bbb43 100644
--- a/share/vlc.desktop
+++ b/share/vlc.desktop
@@ -38,7 +38,7 @@ Name[nl]=VLC Mediaspeler
 GenericName[nl]=Mediaspeler
 Comment[nl]=Uw multimediastreams afspelen, opnemen en uitzenden
 Name[nn]=VLC mediespelar
-Name[nn]=Mediespelar
+GenericName[nn]=Mediespelar
 Comment[nn]=Spel av, ta opp og send ut multimedia
 Name[pa]=VLC ਮੀਡਿਆ ਪਲੇਅਰ
 Comment[pa]=ਆਪਣੀ ਮਲਟੀਮੀਡਿਆ ਸਟਰੀਮ ਪੜ੍ਹੋ, ਕੈਪਚਰ ਤੇ ਬਰਾਡਕਾਸਟ ਕਰੋ 
-- 
1.7.4.4

