diff -up vlc-2.1.0-pre1/Makefile.am.nocache vlc-2.1.0-pre1/Makefile.am
--- vlc-2.1.0-pre1/Makefile.am.nocache	2013-07-17 22:42:16.000252990 -0300
+++ vlc-2.1.0-pre1/Makefile.am	2013-07-17 22:42:50.505889442 -0300
@@ -662,15 +662,6 @@ dist_noinst_SCRIPTS += test/run_vlc.sh
 ###############################################################################
 # Installing plugins cache
 ###############################################################################
-install-exec-hook:
-	if test "$(build)" = "$(host)"; then \
-		PATH="$(DESTDIR)$(bindir):$$PATH" \
-		LD_LIBRARY_PATH="$(DESTDIR)$(libdir):$$LD_LIBRARY_PATH" \
-		"$(DESTDIR)$(vlclibdir)/vlc-cache-gen$(EXEEXT)" \
-			 "$(DESTDIR)$(vlclibdir)/plugins" ; \
-	else \
-		echo "Cross-compilation: cache generation skipped!" ; \
-	fi
 
 uninstall-hook:
 	rm -f -- "$(DESTDIR)$(vlclibdir)/plugins/plugins.dat"
diff -up vlc-2.1.0-pre1/bin/Makefile.am.nocache vlc-2.1.0-pre1/bin/Makefile.am
--- vlc-2.1.0-pre1/bin/Makefile.am.nocache	2013-07-17 22:42:44.274219257 -0300
+++ vlc-2.1.0-pre1/bin/Makefile.am	2013-07-17 22:43:07.838190840 -0300
@@ -76,8 +76,3 @@ MOSTLYCLEANFILES = $(noinst_DATA)
 
 ../modules/plugins.dat: vlc-cache-gen$(EXEEXT)
 	$(AM_V_at)rm -f ../modules/plugins.dat
-	$(AM_V_GEN)if test "$(build)" = "$(host)"; then \
-		./vlc-cache-gen$(EXEEXT) ../modules ; \
-	else \
-		echo "Cross-compilation: cache generation skipped!" ; \
-	fi
