diff -up vlc-2.0.0/Makefile.am.nocache vlc-2.0.0/Makefile.am
--- vlc-2.0.0/Makefile.am.nocache	2012-02-20 21:19:17.344828266 -0200
+++ vlc-2.0.0/Makefile.am	2012-02-20 21:19:50.162929435 -0200
@@ -452,14 +452,6 @@ vlc$(EXEEXT):
 ###############################################################################
 # Installing plugins cache
 ###############################################################################
-install-exec-hook:
-	if test "$(build)" = "$(host)"; then \
-		LD_LIBRARY_PATH="$(DESTDIR)$(libdir):$$LD_LIBRARY_PATH" \
-		"$(DESTDIR)$(vlclibdir)/vlc-cache-gen$(EXEEXT)" \
-			 "$(DESTDIR)$(vlclibdir)/plugins" ; \
-	else \
-		echo "Cross-compilation: cache generation skipped!" ; \
-	fi
 
 uninstall-hook:
 	rm -f -- "$(DESTDIR)$(vlclibdir)/plugins/plugins.dat"
diff -up vlc-2.0.0/bin/Makefile.am.nocache vlc-2.0.0/bin/Makefile.am
--- vlc-2.0.0/bin/Makefile.am.nocache	2012-02-20 21:18:41.437381158 -0200
+++ vlc-2.0.0/bin/Makefile.am	2012-02-20 21:19:03.480743893 -0200
@@ -79,8 +79,3 @@ MOSTLYCLEANFILES = $(noinst_DATA)
 
 ../modules/plugins.dat: vlc-cache-gen$(EXEEXT)
 	$(AM_V_at)rm -f ../modules/plugins.dat
-	$(AM_V_GEN)if test "$(build)" = "$(host)"; then \
-		./vlc-cache-gen$(EXEEXT) ../modules ; \
-	else \
-		echo "Cross-compilation: cache generation skipped!" ; \
-	fi
