--- vlc-1.1.10/modules/misc/freetype.c.orig	2011-06-05 20:31:20.000000000 -0300
+++ vlc-1.1.10/modules/misc/freetype.c	2011-06-06 19:05:48.261606768 -0300
@@ -55,7 +55,7 @@
 # define DEFAULT_FONT_FILE "/usr/share/fonts/nokia/nosnb.ttf"
 # define DEFAULT_FAMILY "Nokia Sans Bold"
 #else
-# define DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+# define DEFAULT_FONT_FILE "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
 # define DEFAULT_FAMILY "Serif Bold"
 #endif
 
