--- vlc-1.1.9/src/modules/cache.c.orig	2011-04-23 16:02:07.726604946 -0300
+++ vlc-1.1.9/src/modules/cache.c	2011-04-23 16:02:37.927175993 -0300
@@ -63,7 +63,7 @@ static int    CacheLoadConfig  ( module_
 
 /* Format string for the cache filename */
 #define CACHENAME_FORMAT \
-    "plugins-%.2zx%.2zx%.2"PRIx8"-%x.dat"
+    "plugins-cache.dat"
 /* Magic for the cache filename */
 #define CACHENAME_VALUES \
     sizeof(int), sizeof(void *), *(uint8_t *)&(uint16_t){ 0xbe1e }, vlc_CPU()
