diff -up vlc-1.1.12/projects/mozilla/support/npunix.cpp.xul8 vlc-1.1.12/projects/mozilla/support/npunix.cpp
--- vlc-1.1.12/projects/mozilla/support/npunix.cpp.xul8	2011-07-12 15:11:26.000000000 -0300
+++ vlc-1.1.12/projects/mozilla/support/npunix.cpp	2011-11-27 15:34:13.506461531 -0200
@@ -767,7 +767,7 @@ Private_GetJavaClass(void)
  *  - Netscape uses the return value to identify when an object instance
  *    of this plugin should be created.
  */
-char *
+const char *
 NP_GetMIMEDescription(void)
 {
     return NPP_GetMIMEDescription();
diff -up vlc-1.1.12/projects/mozilla/vlcshell.cpp.xul8 vlc-1.1.12/projects/mozilla/vlcshell.cpp
--- vlc-1.1.12/projects/mozilla/vlcshell.cpp.xul8	2011-11-27 15:31:48.878117130 -0200
+++ vlc-1.1.12/projects/mozilla/vlcshell.cpp	2011-11-27 15:33:10.910728747 -0200
@@ -71,7 +71,7 @@ static LRESULT CALLBACK Manage( HWND p_h
 /******************************************************************************
  * UNIX-only API calls
  *****************************************************************************/
-char * NPP_GetMIMEDescription( void )
+const char * NPP_GetMIMEDescription( void )
 {
     static char mimetype[] = PLUGIN_MIMETYPES;
     return mimetype;
diff -up vlc-1.1.12/projects/mozilla/vlcshell.h.xul8 vlc-1.1.12/projects/mozilla/vlcshell.h
--- vlc-1.1.12/projects/mozilla/vlcshell.h.xul8	2011-11-27 15:32:14.225744032 -0200
+++ vlc-1.1.12/projects/mozilla/vlcshell.h	2011-11-27 15:32:53.147493857 -0200
@@ -24,7 +24,7 @@
 #ifndef __VLCSHELL_H__
 #define __VLCSHELL_H__
 
-char * NPP_GetMIMEDescription( void );
+const char * NPP_GetMIMEDescription( void );
 
 NPError NPP_Initialize( void );
 
