diff -ur vlc-1.1.7.orig/modules/misc/notify/notify.c vlc-1.1.7/modules/misc/notify/notify.c
--- vlc-1.1.7.orig/modules/misc/notify/notify.c	2010-08-26 21:45:23.000000000 +0300
+++ vlc-1.1.7/modules/misc/notify/notify.c	2011-02-06 16:00:28.000000000 +0200
@@ -38,6 +38,10 @@
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 /*****************************************************************************
  * Module descriptor
  ****************************************************************************/
@@ -312,7 +316,12 @@
     }
 
     notification = notify_notification_new( _("Now Playing"),
-                                            psz_temp, NULL, NULL );
+                                            psz_temp, NULL
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+					    );
+#else
+					    , NULL );
+#endif
     notify_notification_set_timeout( notification,
                                 var_InheritInteger(p_this, "notify-timeout") );
     notify_notification_set_urgency( notification, NOTIFY_URGENCY_LOW );
