--- modules/demux/mkv/matroska_segment.cpp.orig	2010-09-27 11:38:23.000000000 -0300
+++ modules/demux/mkv/matroska_segment.cpp	2011-02-01 19:13:12.698674001 -0200
@@ -220,6 +220,7 @@ void matroska_segment_c::LoadTags( KaxTa
                     }
                     ep->Up();
                 }
+#if 0 // not valid anymore
                 else if( MKV_IS_ID( el, KaxTagGeneral ) )
                 {
                     msg_Dbg( &sys.demuxer, "|   + General" );
@@ -288,6 +289,7 @@ void matroska_segment_c::LoadTags( KaxTa
                 {
                     msg_Dbg( &sys.demuxer, "|   + Multi Title" );
                 }
+#endif
                 else
                 {
                     msg_Dbg( &sys.demuxer, "|   + LoadTag Unknown (%s)", typeid( *el ).name() );
--- modules/demux/mkv/mkv.hpp.orig	2011-01-30 12:45:14.000000000 -0200
+++ modules/demux/mkv/mkv.hpp	2011-02-01 19:09:16.360511543 -0200
@@ -92,7 +92,7 @@
 #include "matroska/KaxSegment.h"
 #include "matroska/KaxTag.h"
 #include "matroska/KaxTags.h"
-#include "matroska/KaxTagMulti.h"
+//#include "matroska/KaxTagMulti.h"
 #include "matroska/KaxTracks.h"
 #include "matroska/KaxTrackAudio.h"
 #include "matroska/KaxTrackVideo.h"
