--- vlc-2.1.0-pre1/modules/text_renderer/freetype.c.orig	2013-07-17 22:30:37.592066531 -0300
+++ vlc-2.1.0-pre1/modules/text_renderer/freetype.c	2013-07-17 22:32:18.121666073 -0300
@@ -67,9 +67,9 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
 # define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/TTF/DejaVuSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
 #endif
 
