diff -up vlc-2.2.0/modules/text_renderer/platform_fonts.h.deffont vlc-2.2.0/modules/text_renderer/platform_fonts.h
--- vlc-2.2.0/modules/text_renderer/platform_fonts.h.deffont	2015-02-27 19:56:47.044721878 -0300
+++ vlc-2.2.0/modules/text_renderer/platform_fonts.h	2015-02-27 19:57:41.090286023 -0300
@@ -55,9 +55,9 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/TTF/DejaVuSerif-Bold.ttf"
 # define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/TTF/DejaVuSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
 #endif
 
diff -up vlc-2.2.0/modules/visualization/projectm.cpp.deffont vlc-2.2.0/modules/visualization/projectm.cpp
--- vlc-2.2.0/modules/visualization/projectm.cpp.deffont	2015-02-27 20:05:24.327263264 -0300
+++ vlc-2.2.0/modules/visualization/projectm.cpp	2015-02-27 20:06:02.520193915 -0300
@@ -81,8 +81,8 @@ static void Close        ( vlc_object_t
 # define FONT_PATH_MENU "C:\\WINDOWS\\Fonts\\arial.ttf"
 # define PRESET_PATH    NULL
 #else
-# define FONT_PATH      "/usr/share/fonts/truetype/ttf-dejavu/DejaVuSans.ttf"
-# define FONT_PATH_MENU "/usr/share/fonts/truetype/ttf-dejavu/DejaVuSansMono.ttf"
+# define FONT_PATH      "/usr/share/fonts/TTF/DejaVuSans.ttf"
+# define FONT_PATH_MENU "/usr/share/fonts/TTF/DejaVuSansMono.ttf"
 # define PRESET_PATH    "/usr/share/projectM/presets"
 #endif
 
