From f848c1de94ef2f5ae7d53358d30e26400a69d3d4 Mon Sep 17 00:00:00 2001
From: Ivailo Monev <xakepa10@gmail.com>
Date: Mon, 24 Aug 2015 20:18:50 +0300
Subject: [PATCH 5/5] Filter out the kdm user

---
 src/ck-manager.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/ck-manager.c b/src/ck-manager.c
index 66da8b2..4a6e511 100644
--- a/src/ck-manager.c
+++ b/src/ck-manager.c
@@ -1111,9 +1111,10 @@ session_is_real_user (CkSession *session,
 
         username = get_user_name (uid);
 
-        /* filter out GDM/SDDM user */
+        /* filter out GDM/SDDM/KDM user */
         if (g_strcmp0 (username, "gdm")  == 0 ||
-            g_strcmp0 (username, "sddm") == 0) {
+            g_strcmp0 (username, "sddm") == 0 ||
+            g_strcmp0 (username, "kdm") == 0) {
                 ret = FALSE;
                 goto out;
         }
-- 
2.5.1

