From 22958330535c9e26ed537d1568925f64e3bafe4d Mon Sep 17 00:00:00 2001
From: Eric Hameleers <alien@slackware.com>
Date: Thu, 10 Sep 2015 14:19:43 +0200
Subject: [PATCH] Apply overlooked PREFIX/lib -> LIBDIR substitution

---
 src/ck-session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ck-session.c b/src/ck-session.c
index ca0f8ca..1b58889 100644
--- a/src/ck-session.c
+++ b/src/ck-session.c
@@ -1273,7 +1273,7 @@ ck_session_run_programs (CkSession  *session,
         g_assert(n <= G_N_ELEMENTS(extra_env));
 
         ck_run_programs (SYSCONFDIR "/ConsoleKit/run-session.d", action, extra_env);
-        ck_run_programs (PREFIX "/lib/ConsoleKit/run-session.d", action, extra_env);
+        ck_run_programs (LIBDIR "/ConsoleKit/run-session.d", action, extra_env);
 
         for (n = 0; extra_env[n] != NULL; n++) {
                 g_free (extra_env[n]);
