--- docs/tools/Makefile.am	2010-02-17 12:04:07.000000000 -0500
+++ docs/tools/Makefile.am.new	2010-02-20 01:49:57.034603685 -0500
@@ -20,7 +20,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 ttl2sgml_SOURCES =							\
 	ttl2sgml.c							\
@@ -38,7 +40,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GLIB2_LIBS)							\
-	$(GIO_LIBS)
+	$(GIO_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 gen-doc.stamp: ttl2sgml
 	$(top_srcdir)/docs/tools/gen-doc.sh
--- docs/tools/Makefile.in	2010-02-19 12:28:04.000000000 -0500
+++ docs/tools/Makefile.in.new	2010-02-20 01:49:17.190437481 -0500
@@ -377,7 +377,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 ttl2sgml_SOURCES = \
 	ttl2sgml.c							\
@@ -395,8 +397,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GLIB2_LIBS)							\
-	$(GIO_LIBS)
-
+	$(GIO_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 #
 # The origianal diagram but with .png extension
--- docs/reference/libtracker-client/Makefile.am	2010-02-19 12:52:38.000000000 -0500
+++ docs/reference/libtracker-client/Makefile.am.new	2010-03-02 13:16:20.942711481 -0500
@@ -46,7 +46,9 @@
 	$(UNAC_LIBS)							\
 	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-client/Makefile.in	2010-02-26 07:00:16.000000000 -0500
+++ docs/reference/libtracker-client/Makefile.in.new	2010-03-02 13:20:25.225089828 -0500
@@ -367,7 +367,9 @@
 	$(UNAC_LIBS)							\
 	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-common/Makefile.am	2010-02-19 12:52:38.000000000 -0500
+++ docs/reference/libtracker-common/Makefile.am.new	2010-03-02 13:17:17.902711556 -0500
@@ -43,7 +43,8 @@
 	$(UNAC_LIBS)							\
 	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-common/Makefile.in	2010-02-26 07:00:16.000000000 -0500
+++ docs/reference/libtracker-common/Makefile.in.new	2010-03-02 13:21:05.764818185 -0500
@@ -363,7 +363,8 @@
 	$(UNAC_LIBS)							\
 	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-extract/Makefile.am	2010-02-19 12:52:38.000000000 -0500
+++ docs/reference/libtracker-extract/Makefile.am.new	2010-03-02 13:18:16.565064140 -0500
@@ -34,7 +34,9 @@
 GTKDOC_LIBS =								\
 	$(top_builddir)/src/libtracker-extract/libtracker-extract-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-extract/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-extract/Makefile.in	2010-02-26 07:00:16.000000000 -0500
+++ docs/reference/libtracker-extract/Makefile.in.new	2010-03-02 13:21:47.998712414 -0500
@@ -354,7 +354,9 @@
 GTKDOC_LIBS = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-extract/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-miner/Makefile.am	2010-03-18 10:15:42.000000000 -0400
+++ docs/reference/libtracker-miner/Makefile.am.new	2010-03-18 14:31:43.572676975 -0400
@@ -52,7 +52,10 @@
 	$(UNAC_LIBS)							\
 	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-miner/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-miner/Makefile.in	2010-03-18 14:08:27.000000000 -0400
+++ docs/reference/libtracker-miner/Makefile.in.new	2010-03-18 14:32:33.372731452 -0400
@@ -377,7 +377,10 @@
 	$(UNAC_LIBS)							\
 	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-miner/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
