diff -up src/tracker-extract/tracker-extract-pdf.c.poppler src/tracker-extract/tracker-extract-pdf.c
--- src/tracker-extract/tracker-extract-pdf.c.poppler	2010-09-02 09:51:00.000000000 -0300
+++ src/tracker-extract/tracker-extract-pdf.c	2010-12-29 01:09:21.892313713 -0200
@@ -243,7 +243,7 @@ extract_content (PopplerDocument *docume
 		rect.x1 = rect.y1 = 0;
 		poppler_page_get_size (page, &rect.x2, &rect.y2);
 
-		text = poppler_page_get_text (page, POPPLER_SELECTION_WORD, &rect);
+		text = poppler_page_get_selected_text (page, POPPLER_SELECTION_WORD, &rect);
 		t = tracker_text_normalize (text, n_words - words, &normalized_words);
 
 		words += normalized_words;
