--- docs/tools/Makefile.am	2010-05-20 13:14:53.000000000 -0400
+++ docs/tools/Makefile.am.new	2010-05-27 13:47:46.082189121 -0400
@@ -23,7 +23,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 ttl2sgml_SOURCES =							\
 	ttl2sgml.c							\
@@ -41,6 +43,8 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GLIB2_LIBS)							\
-	$(GIO_LIBS)
+	$(GIO_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 EXTRA_DIST = gen-doc.sh
--- docs/tools/Makefile.in	2010-05-27 09:48:15.000000000 -0400
+++ docs/tools/Makefile.in.new	2010-05-27 13:49:52.845946280 -0400
@@ -385,7 +385,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 ttl2sgml_SOURCES = \
 	ttl2sgml.c							\
@@ -403,7 +405,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GLIB2_LIBS)							\
-	$(GIO_LIBS)
+	$(GIO_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 EXTRA_DIST = gen-doc.sh
 all: all-am
--- docs/reference/libtracker-client/Makefile.am	2010-04-15 06:14:08.000000000 -0400
+++ docs/reference/libtracker-client/Makefile.am.new	2010-04-19 09:56:26.248325370 -0400
@@ -40,7 +40,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-client/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-client/Makefile.in.new	2010-04-19 09:57:14.958188757 -0400
@@ -364,7 +364,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-common/Makefile.am	2010-04-15 06:14:08.000000000 -0400
+++ docs/reference/libtracker-common/Makefile.am.new	2010-04-19 09:59:04.634098797 -0400
@@ -37,7 +37,8 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-common/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-common/Makefile.in.new	2010-04-19 10:01:08.364095170 -0400
@@ -360,7 +360,8 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-extract/Makefile.am	2010-03-29 10:54:50.000000000 -0400
+++ docs/reference/libtracker-extract/Makefile.am.new	2010-04-19 10:02:46.533447803 -0400
@@ -34,7 +34,9 @@
 GTKDOC_LIBS =								\
 	$(top_builddir)/src/libtracker-extract/libtracker-extract-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-extract/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-extract/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-extract/Makefile.in.new	2010-04-19 10:03:29.565082879 -0400
@@ -357,7 +357,9 @@
 GTKDOC_LIBS = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-extract/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-miner/Makefile.am	2010-04-15 06:14:08.000000000 -0400
+++ docs/reference/libtracker-miner/Makefile.am.new	2010-04-19 10:04:20.494435581 -0400
@@ -50,7 +50,10 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-miner/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-miner/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-miner/Makefile.in.new	2010-04-19 10:05:07.469082598 -0400
@@ -374,7 +374,10 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-miner/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
