diff -up gpgme-1.2.0/src/gpgme-config.in.largefile gpgme-1.2.0/src/gpgme-config.in
--- gpgme-1.2.0/src/gpgme-config.in.largefile	2010-08-11 20:11:49.000000000 +0200
+++ gpgme-1.2.0/src/gpgme-config.in	2010-08-11 20:23:45.000000000 +0200
@@ -42,6 +42,10 @@ cflags_pthread=""
 cflags_glib="@GLIB_CFLAGS@"
 with_glib=
 
+if test "0@NEED__FILE_OFFSET_BITS@" -gt "0" ; then
+    cflags_lfs="-D_FILE_OFFSET_BITS=@NEED__FILE_OFFSET_BITS@"
+fi
+
 output=""
 
 usage()
@@ -101,6 +105,9 @@ while test $# -gt 0; do
 	    exit 0
 	    ;;
         --cflags)
+	    if test "x$cflags_lfs" != "x"; then
+		output="$output $cflags_lfs"
+	    fi
 	    if test "x$includedir" != "x/usr/include" -a "x$includedir" != "x/include"; then
 		output="$output -I$includedir"
 	    fi
