diff -up gpgme-1.3.2/src/Makefile.am.ImplicitDSOLinking gpgme-1.3.2/src/Makefile.am
--- gpgme-1.3.2/src/Makefile.am.ImplicitDSOLinking	2012-04-13 10:01:09.000000000 +0200
+++ gpgme-1.3.2/src/Makefile.am	2012-09-26 10:06:42.822675632 +0200
@@ -223,7 +223,7 @@ libgpgme_qt_la_LIBADD = $(gpgme_res) @LI
 endif
 
 noinst_PROGRAMS = gpgme-tool
-gpgme_tool_LDADD = libgpgme.la @LIBASSUAN_LIBS@
+gpgme_tool_LDADD = libgpgme.la @LIBASSUAN_LIBS@ @GPG_ERROR_LIBS@
 
 install-data-local: install-def-file
 
diff -up gpgme-1.3.2/src/Makefile.in.ImplicitDSOLinking gpgme-1.3.2/src/Makefile.in
--- gpgme-1.3.2/src/Makefile.in.ImplicitDSOLinking	2012-05-02 11:19:10.000000000 +0200
+++ gpgme-1.3.2/src/Makefile.in	2012-09-26 10:07:14.361355694 +0200
@@ -542,7 +542,7 @@ libgpgme_pthread_la_LIBADD = $(gpgme_res
 @BUILD_W32_QT_TRUE@libgpgme_qt_la_LIBADD = $(gpgme_res) @LIBASSUAN_LIBS@ @LTLIBOBJS@ \
 @BUILD_W32_QT_TRUE@	@GPG_ERROR_LIBS@ @QT4_CORE_LIBS@
 
-gpgme_tool_LDADD = libgpgme.la @LIBASSUAN_LIBS@
+gpgme_tool_LDADD = libgpgme.la @LIBASSUAN_LIBS@ @GPG_ERROR_LIBS@
 all: all-am
 
 .SUFFIXES:
diff -up gpgme-1.3.2/tests/gpg/Makefile.am.ImplicitDSOLinking gpgme-1.3.2/tests/gpg/Makefile.am
--- gpgme-1.3.2/tests/gpg/Makefile.am.ImplicitDSOLinking	2012-04-13 10:01:10.000000000 +0200
+++ gpgme-1.3.2/tests/gpg/Makefile.am	2012-09-26 10:08:30.059987972 +0200
@@ -47,8 +47,8 @@ EXTRA_DIST = pubdemo.asc secdemo.asc cip
 INCLUDES = -I$(top_builddir)/src
 
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
-t_thread1_LDADD = ../../src/libgpgme-pthread.la -lpthread
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
+t_thread1_LDADD = ../../src/libgpgme-pthread.la @GPG_ERROR_LIBS@ -pthread
 
 # We don't run t-genkey in the test suite, because it takes too long
 noinst_PROGRAMS = $(TESTS) t-genkey
diff -up gpgme-1.3.2/tests/gpg/Makefile.in.ImplicitDSOLinking gpgme-1.3.2/tests/gpg/Makefile.in
--- gpgme-1.3.2/tests/gpg/Makefile.in.ImplicitDSOLinking	2012-05-02 11:19:11.000000000 +0200
+++ gpgme-1.3.2/tests/gpg/Makefile.in	2012-09-26 10:09:09.254833125 +0200
@@ -380,8 +380,8 @@ EXTRA_DIST = pubdemo.asc secdemo.asc cip
 
 INCLUDES = -I$(top_builddir)/src
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
-t_thread1_LDADD = ../../src/libgpgme-pthread.la -lpthread
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
+t_thread1_LDADD = ../../src/libgpgme-pthread.la @GPG_ERROR_LIBS@ -pthread
 all: all-am
 
 .SUFFIXES:
diff -up gpgme-1.3.2/tests/gpgsm/Makefile.am.ImplicitDSOLinking gpgme-1.3.2/tests/gpgsm/Makefile.am
--- gpgme-1.3.2/tests/gpgsm/Makefile.am.ImplicitDSOLinking	2011-12-02 10:49:38.000000000 +0100
+++ gpgme-1.3.2/tests/gpgsm/Makefile.am	2012-09-26 10:04:36.600953929 +0200
@@ -32,7 +32,7 @@ EXTRA_DIST = cert_dfn_pca01.der cert_dfn
 INCLUDES = -I$(top_builddir)/src
 
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 
 # We don't run t-genkey in the test suite, because it takes too long
 # and needs a working pinentry.
diff -up gpgme-1.3.2/tests/gpgsm/Makefile.in.ImplicitDSOLinking gpgme-1.3.2/tests/gpgsm/Makefile.in
--- gpgme-1.3.2/tests/gpgsm/Makefile.in.ImplicitDSOLinking	2012-05-02 11:19:11.000000000 +0200
+++ gpgme-1.3.2/tests/gpgsm/Makefile.in	2012-09-26 10:09:38.607466053 +0200
@@ -308,7 +308,7 @@ EXTRA_DIST = cert_dfn_pca01.der cert_dfn
 
 INCLUDES = -I$(top_builddir)/src
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 key_id = 32100C27173EF6E9C4E9A25D3D69F86D37A4F939
 CLEANFILES = pubring.kbx pubring.kbx~ gpgsm.conf trustlist.txt \
 	random_seed S.gpg-agent  
diff -up gpgme-1.3.2/tests/opassuan/Makefile.am.ImplicitDSOLinking gpgme-1.3.2/tests/opassuan/Makefile.am
--- gpgme-1.3.2/tests/opassuan/Makefile.am.ImplicitDSOLinking	2010-12-30 16:19:33.000000000 +0100
+++ gpgme-1.3.2/tests/opassuan/Makefile.am	2012-09-26 10:04:36.601953950 +0200
@@ -27,7 +27,7 @@ EXTRA_DIST =
 INCLUDES = -I$(top_builddir)/src
 
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 
 noinst_PROGRAMS = $(TESTS) t-command
 
diff -up gpgme-1.3.2/tests/opassuan/Makefile.in.ImplicitDSOLinking gpgme-1.3.2/tests/opassuan/Makefile.in
--- gpgme-1.3.2/tests/opassuan/Makefile.in.ImplicitDSOLinking	2012-05-02 11:19:11.000000000 +0200
+++ gpgme-1.3.2/tests/opassuan/Makefile.in	2012-09-26 10:04:36.601953950 +0200
@@ -259,7 +259,7 @@ noinst_HEADERS =
 EXTRA_DIST = 
 INCLUDES = -I$(top_builddir)/src
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 DISTCLEANFILES = 
 all: all-am
 
