diff -up gpgme-1.3.2/src/gpgme-config.in.config_extras gpgme-1.3.2/src/gpgme-config.in
--- gpgme-1.3.2/src/gpgme-config.in.config_extras	2012-05-02 10:42:23.000000000 +0200
+++ gpgme-1.3.2/src/gpgme-config.in	2012-09-26 10:03:41.904413540 +0200
@@ -22,22 +22,22 @@ cflags="-I@includedir@"
 libs="-L@libdir@"
 
 # Network libraries.
-assuan_cflags="@LIBASSUAN_CFLAGS@"
-assuan_libs="@LIBASSUAN_LIBS@"
+#assuan_cflags="@LIBASSUAN_CFLAGS@"
+#assuan_libs="@LIBASSUAN_LIBS@"
 
 # Configure libgpg-error.
 gpg_error_cflags="@GPG_ERROR_CFLAGS@"
-gpg_error_libs="@GPG_ERROR_LIBS@"
+#gpg_error_libs="@GPG_ERROR_LIBS@"
 
 # Configure thread packages.
 thread_modules=""
 
 @HAVE_PTHREAD_TRUE@thread_modules="$thread_modules pthread"
-libs_pthread="-lpthread"
+#libs_pthread="-lpthread"
 cflags_pthread=""
 
 # Configure glib.
-libs_glib="@GLIB_LIBS@"
+#libs_glib="@GLIB_LIBS@"
 cflags_glib="@GLIB_CFLAGS@"
 with_glib=
 
