diff -uNr gpgme-1.7.0.orig/src/gpgme-config.in gpgme-1.7.0/src/gpgme-config.in
--- gpgme-1.7.0.orig/src/gpgme-config.in	2016-09-21 14:58:05.420362491 +0200
+++ gpgme-1.7.0/src/gpgme-config.in	2016-09-21 15:02:04.156591953 +0200
@@ -22,12 +22,12 @@
 libs="-L@libdir@"
 
 # Network libraries.
-assuan_cflags="@LIBASSUAN_CFLAGS@"
-assuan_libs="@LIBASSUAN_LIBS@"
+#assuan_cflags="@LIBASSUAN_CFLAGS@"
+#assuan_libs="@LIBASSUAN_LIBS@"
 
 # Configure libgpg-error.
 gpg_error_cflags="@GPG_ERROR_CFLAGS@"
-gpg_error_libs="@GPG_ERROR_LIBS@"
+#gpg_error_libs="@GPG_ERROR_LIBS@"
 
 # Configure thread packages.
 thread_modules=""
