diff -up gpgme-1.2.0/tests/gpgsm/Makefile.am.ImplicitDSOLinking gpgme-1.2.0/tests/gpgsm/Makefile.am
--- gpgme-1.2.0/tests/gpgsm/Makefile.am.ImplicitDSOLinking	2008-12-03 07:39:48.000000000 -0600
+++ gpgme-1.2.0/tests/gpgsm/Makefile.am	2010-07-02 18:51:30.336705172 -0500
@@ -32,7 +32,7 @@ EXTRA_DIST = cert_dfn_pca01.der cert_dfn
 INCLUDES = -I$(top_builddir)/src
 
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 
 # We don't run t-genkey in the test suite, because it takes too long
 # and needs a working pinentry.
diff -up gpgme-1.2.0/tests/gpgsm/Makefile.in.ImplicitDSOLinking gpgme-1.2.0/tests/gpgsm/Makefile.in
--- gpgme-1.2.0/tests/gpgsm/Makefile.in.ImplicitDSOLinking	2009-06-18 11:13:41.000000000 -0500
+++ gpgme-1.2.0/tests/gpgsm/Makefile.in	2010-07-02 18:51:30.336705172 -0500
@@ -285,7 +285,7 @@ EXTRA_DIST = cert_dfn_pca01.der cert_dfn
 
 INCLUDES = -I$(top_builddir)/src
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 key_id = 32100C27173EF6E9C4E9A25D3D69F86D37A4F939
 DISTCLEANFILES = pubring.kbx pubring.kbx~ gpgsm.conf trustlist.txt \
 	private-keys-v1.d/$(key_id).key random_seed
diff -up gpgme-1.2.0/tests/opassuan/Makefile.am.ImplicitDSOLinking gpgme-1.2.0/tests/opassuan/Makefile.am
--- gpgme-1.2.0/tests/opassuan/Makefile.am.ImplicitDSOLinking	2009-01-26 08:15:26.000000000 -0600
+++ gpgme-1.2.0/tests/opassuan/Makefile.am	2010-07-02 18:53:28.451456311 -0500
@@ -27,7 +27,7 @@ EXTRA_DIST = 
 INCLUDES = -I$(top_builddir)/src
 
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 
 noinst_PROGRAMS = $(TESTS) t-command
 
diff -up gpgme-1.2.0/tests/opassuan/Makefile.in.ImplicitDSOLinking gpgme-1.2.0/tests/opassuan/Makefile.in
--- gpgme-1.2.0/tests/opassuan/Makefile.in.ImplicitDSOLinking	2009-06-18 11:13:41.000000000 -0500
+++ gpgme-1.2.0/tests/opassuan/Makefile.in	2010-07-02 18:53:41.590456782 -0500
@@ -236,7 +236,7 @@ noinst_HEADERS = 
 EXTRA_DIST = 
 INCLUDES = -I$(top_builddir)/src
 AM_CPPFLAGS = @GPG_ERROR_CFLAGS@
-LDADD = ../../src/libgpgme.la
+LDADD = ../../src/libgpgme.la @GPG_ERROR_LIBS@
 DISTCLEANFILES = 
 all: all-am
 
