diff -up polkit-gnome-0.96/polkitgtk/polkitlockbutton.c.new-icons polkit-gnome-0.96/polkitgtk/polkitlockbutton.c
--- polkit-gnome-0.96/polkitgtk/polkitlockbutton.c.new-icons	2010-06-14 01:16:56.049851488 -0400
+++ polkit-gnome-0.96/polkitgtk/polkitlockbutton.c	2010-06-14 01:17:17.209606831 -0400
@@ -749,7 +749,7 @@ update_state (PolkitLockButton *button)
         }
     }
 
-  image = gtk_image_new_from_icon_name (button->priv->authorized ? "stock_lock-open" : "stock_lock",
+  image = gtk_image_new_from_icon_name (button->priv->authorized ? "changes-allow" : "changes-prevent",
                                         GTK_ICON_SIZE_SMALL_TOOLBAR);
   gtk_button_set_image (GTK_BUTTON (button->priv->button), image);
   gtk_label_set_text (GTK_LABEL (button->priv->label), text);
