diff -crB liferea-1.6.3/src/Makefile.in liferea-1.6.3-fedora/src/Makefile.in
*** liferea-1.6.3/src/Makefile.in	2010-02-22 16:30:22.000000000 -0500
--- liferea-1.6.3-fedora/src/Makefile.in	2010-03-27 14:59:37.502022524 -0400
***************
*** 368,373 ****
--- 368,374 ----
  		notification/libnotification.a \
  		ui/libliui.a \
  		webkit/libwebkit.a \
+                 -lX11 \
  		$(SYNC_LIB) \
  		$(PACKAGE_LIBS) $(SM_LIBS) \
  		$(DBUS_LIBS) $(NM_LIBS) $(INTLLIBS) $(GNUTLS_LIBS) $(AVAHI_LIBS) \
