--- fusion-icon-5e2dc9/FusionIcon/util.py	2008-03-16 22:58:07.000000000 +0000
+++ fusion-icon-5e2dc9/FusionIcon/util.py.new	2011-01-19 00:47:38.791230764 +0000
@@ -220,12 +220,12 @@
 
 		try:
 			context = compizconfig.Context( \
-				plugins=['decoration'], basic_metadata=True)
+				plugins=['decor'], basic_metadata=True)
 
 		except:
 			context = compizconfig.Context()
 
-		self.command = context.Plugins['decoration'].Display['command']
+		self.command = context.Plugins['decor'].Display['command']
 
 		for decorator in installed.decorators:
 			self[decorator] = CompizDecorator(decorator, self, installed)
