--- fusion-icon-5e2dc9/FusionIcon/util.py	2011-01-19 00:47:38.791230764 +0000
+++ fusion-icon-5e2dc9/FusionIcon/util.py.new2	2011-01-19 00:59:02.386674996 +0000
@@ -225,7 +225,7 @@
 		except:
 			context = compizconfig.Context()
 
-		self.command = context.Plugins['decor'].Display['command']
+		self.command = context.Plugins['decor'].Screen['command']
 
 		for decorator in installed.decorators:
 			self[decorator] = CompizDecorator(decorator, self, installed)
