diff -Nup xa-2.3.5.orig/Makefile xa-2.3.5/Makefile
--- xa-2.3.5.orig/Makefile	2009-02-08 07:49:30.000000000 +0100
+++ xa-2.3.5/Makefile	2009-04-02 11:18:47.000000000 +0200
@@ -15,11 +15,12 @@ LDFLAGS = -lc
 #CFLAGS =
 #LD = ld
 
-DESTDIR = /usr/local
+DESTDIR =
+PREFIX = /usr/local
 
-BINDIR = $(DESTDIR)/bin
-MANDIR = $(DESTDIR)/share/man/man1
-DOCDIR = $(DESTDIR)/share/doc
+BINDIR = $(DESTDIR)$(PREFIX)/bin
+MANDIR = $(DESTDIR)$(PREFIX)/share/man/man1
+DOCDIR = $(DESTDIR)$(PREFIX)/share/doc
 
 MKDIR = mkdir -p
 INSTALL = install
@@ -27,13 +28,13 @@ INSTALL = install
 all: xa uncpk
 
 xa:
-	(cd src && LD=${LD} CC="${CC} ${CFLAGS}" ${MAKE})
+	(cd src && LD=${LD} CC="${CC}" CFLAGS="${CFLAGS}" ${MAKE})
 
 load:	
-	(cd loader && CC="${CC} ${CFLAGS}" ${MAKE})
+	(cd loader && CC="${CC}" CFLAGS="${CFLAGS}" ${MAKE})
 
 uncpk:
-	(cd misc && CC="${CC} ${CFLAGS}" ${MAKE})
+	(cd misc && CC="${CC}" CFLAGS="${CFLAGS}" ${MAKE})
 
 dos: clean
 	(cd src && LD=gcc-go32 CC=gcc-go32 CFLAGS="-W -Wall -pedantic" ${MAKE})
@@ -53,8 +54,8 @@ clean:
 install: xa uncpk
 	$(MKDIR) $(BINDIR)
 	$(MKDIR) $(MANDIR)
-	$(INSTALL) xa reloc65 ldo65 file65 printcbm uncpk $(BINDIR)
-	$(INSTALL) man/file65.1 man/ldo65.1 man/printcbm.1 man/reloc65.1 man/uncpk.1 man/xa.1 $(MANDIR)
+	$(INSTALL) -m 0755 xa reloc65 ldo65 file65 printcbm uncpk $(BINDIR)
+	$(INSTALL) -m 0644 man/file65.1 man/ldo65.1 man/printcbm.1 man/reloc65.1 man/uncpk.1 man/xa.1 $(MANDIR)
 	#$(MKDIR) $(DOCDIR)/xa65
 
 dist: clean
