From d97f8b09fbc1db16ad257ad0f0c682611a95a94c Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward.hervey@collabora.co.uk>
Date: Mon, 17 Oct 2011 14:26:20 +0000
Subject: [PATCH 01/12] configure.ac: Fix for new libav

---
 configure |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 87e825e..83020f7 100755
--- a/configure
+++ b/configure
@@ -16696,8 +16696,8 @@ fi
   embffmpeg_configure_args="--prefix=$prefix"
 
   # Enable pic and static so that we get .a files, but with PIC code.
-  embffmpeg_configure_args="$embffmpeg_configure_args --disable-ffserver --disable-ffplay\
-        --disable-ffmpeg --disable-ffprobe --enable-static --enable-pic \
+  embffmpeg_configure_args="$embffmpeg_configure_args --disable-avserver --disable-avplay\
+        --disable-ffmpeg --disable-avprobe --enable-postproc --enable-gpl --enable-static --enable-pic \
 	--disable-encoder=flac --disable-decoder=cavs --disable-protocols --disable-devices\
 	--disable-network --disable-hwaccels --disable-filters --disable-doc\
 	--enable-optimizations"
-- 
1.7.10.4

