From d4e381c1cf8fa825d060b3c2797272e675d15937 Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward.hervey@collabora.co.uk>
Date: Mon, 17 Oct 2011 16:26:52 +0200
Subject: [PATCH 02/12] gstffmpegutils: Fix include

---
 ext/ffmpeg/gstffmpegutils.h |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ext/ffmpeg/gstffmpegutils.h b/ext/ffmpeg/gstffmpegutils.h
index 4b713de..dbc3fae 100644
--- a/ext/ffmpeg/gstffmpegutils.h
+++ b/ext/ffmpeg/gstffmpegutils.h
@@ -20,11 +20,17 @@
 #ifndef __GST_FFMPEG_UTILS_H__
 #define __GST_FFMPEG_UTILS_H__
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #ifdef HAVE_FFMPEG_UNINSTALLED
+#include <mathematics.h>
 #include <avcodec.h>
 #else
 #include <libavcodec/avcodec.h>
 #endif
+
 #include <gst/gst.h>
 
 /*
-- 
1.7.10.4

