From 68c379469f807bc8d34bd8bb0ea28fd25a9be1db Mon Sep 17 00:00:00 2001
From: Edward Hervey <edward.hervey@collabora.co.uk>
Date: Mon, 17 Oct 2011 16:27:36 +0200
Subject: [PATCH 03/12] gstffmpegdec: Re-enable MT-decoding by default

---
 ext/ffmpeg/gstffmpegdec.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/ffmpeg/gstffmpegdec.c b/ext/ffmpeg/gstffmpegdec.c
index cc07889..035307e 100644
--- a/ext/ffmpeg/gstffmpegdec.c
+++ b/ext/ffmpeg/gstffmpegdec.c
@@ -197,7 +197,7 @@ gst_ts_info_get (GstFFMpegDec * dec, gint idx)
 #define DEFAULT_DO_PADDING		TRUE
 #define DEFAULT_DEBUG_MV		FALSE
 #define DEFAULT_CROP			TRUE
-#define DEFAULT_MAX_THREADS		1
+#define DEFAULT_MAX_THREADS		0
 
 enum
 {
-- 
1.7.10.4

