From e0212a0b36fea9a3ad836bb0014e136675ad7ac8 Mon Sep 17 00:00:00 2001
From: Vincent Penquerc'h <vincent.penquerch@collabora.co.uk>
Date: Thu, 12 Jan 2012 14:57:48 +0000
Subject: [PATCH 07/12] ffmpegdemux: fix caps leak

---
 ext/ffmpeg/gstffmpegdemux.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/ffmpeg/gstffmpegdemux.c b/ext/ffmpeg/gstffmpegdemux.c
index 3eb5328..6122a40 100644
--- a/ext/ffmpeg/gstffmpegdemux.c
+++ b/ext/ffmpeg/gstffmpegdemux.c
@@ -1330,6 +1330,7 @@ gst_ffmpegdemux_type_find (GstTypeFind * tf, gpointer priv)
           in_plugin->name, sinkcaps, res);
 
       gst_type_find_suggest (tf, res, sinkcaps);
+      gst_caps_unref (sinkcaps);
     }
   }
 }
-- 
1.7.10.4

