From 283e0a54d04b2bf84fc365752dd4004c1e501f38 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Tue, 6 Mar 2012 15:27:22 +0100
Subject: [PATCH 11/12] ffmux: Use correct enum type for return value

---
 ext/ffmpeg/gstffmpegmux.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/ffmpeg/gstffmpegmux.c b/ext/ffmpeg/gstffmpegmux.c
index a041761..5486bda 100644
--- a/ext/ffmpeg/gstffmpegmux.c
+++ b/ext/ffmpeg/gstffmpegmux.c
@@ -785,7 +785,7 @@ gst_ffmpegmux_collected (GstCollectPads * pads, gpointer user_data)
 static GstStateChangeReturn
 gst_ffmpegmux_change_state (GstElement * element, GstStateChange transition)
 {
-  GstFlowReturn ret;
+  GstStateChangeReturn ret;
   GstFFMpegMux *ffmpegmux = (GstFFMpegMux *) (element);
 
   switch (transition) {
-- 
1.7.10.4

