diff -up mednafen/src/snes/src/lib/nall/array.hpp.buildfix mednafen/src/snes/src/lib/nall/array.hpp
--- mednafen/src/snes/src/lib/nall/array.hpp.buildfix	2010-01-06 09:57:25.000000000 +0100
+++ mednafen/src/snes/src/lib/nall/array.hpp	2012-02-10 22:40:27.697184404 +0100
@@ -2,6 +2,7 @@
 #define NALL_ARRAY_HPP
 
 #include <stdlib.h>
+#include <string.h>
 #include <nall/algorithm.hpp>
 #include <nall/bit.hpp>
 
