diff -up krfb-4.11.5/libvncserver/CMakeLists.txt.system_minilzo krfb-4.11.5/libvncserver/CMakeLists.txt
--- krfb-4.11.5/libvncserver/CMakeLists.txt.system_minilzo	2014-01-02 13:38:14.000000000 -0600
+++ krfb-4.11.5/libvncserver/CMakeLists.txt	2014-08-04 12:15:30.148919143 -0500
@@ -92,7 +92,6 @@ set(LIBVNCSERVER_SOURCES
     d3des.c
     vncauth.c
     cargs.c
-    minilzo.c
     ultra.c
     scale.c
 )
@@ -118,6 +117,10 @@ if(JPEG_FOUND)
   )
 endif(JPEG_FOUND)
 
+#if(MINOLZO_FOUND)
+  include_directories(/usr/include/lzo)
+  set(MINILZO_LIBRARIES -lminilzo)
+#endif
 
 add_library(vncserver STATIC ${LIBVNCSERVER_SOURCES})
 
@@ -129,4 +132,5 @@ target_link_libraries(vncserver
     ${ADDITIONAL_LIBS}
     ${ZLIB_LIBRARIES}
     ${JPEG_LIBRARIES}
+    ${MINILZO_LIBRARIES}
 )
