From 4713c10585de119b1bb8e35d4f0299aa91daefec Mon Sep 17 00:00:00 2001
From: Heinemann Jürgen (Undefined) <nospam __AT. hjcms __DOT. de>
Date: Sun, 03 Apr 2011 05:17:00 +0000
Subject: So 3. Apr 16:11:38 CEST 2011

---
diff --git a/src/ffprocess.cpp b/src/ffprocess.cpp
index 3b4b14e..efb91d1 100644
--- a/src/ffprocess.cpp
+++ b/src/ffprocess.cpp
@@ -122,8 +122,10 @@ bool FFProcess::stop()
 
   emit message ( trUtf8 ( "shutdown please wait ..." ) );
 
+  // TODO Keine Ahnung was der Aufeinmal hat ???
   char quit = 'q';
   m_QProcess->write ( &quit );
+  m_QProcess->closeWriteChannel();
 
   return isRunning();
 }
diff --git a/src/logviewer/highlighter.cpp b/src/logviewer/highlighter.cpp
index 4b65169..d226868 100644
--- a/src/logviewer/highlighter.cpp
+++ b/src/logviewer/highlighter.cpp
@@ -39,28 +39,24 @@ Highlighter::Highlighter ( QTextDocument *parent )
   RuleHighlight step1; // Strings
   step1.pattern = QRegExp ( "(?!=[\\s\\>\\/])([\\\"\\\'](.*)[\\\"\\\'])" );
   stringFormat.setForeground ( Qt::red );
-  stringFormat.setBackground ( bg );
   step1.format = stringFormat;
   highlightRules.append ( step1 );
 
   RuleHighlight step2; // Predicates
   step2.pattern = QRegExp ( "([^|\\s][\\w]+(?=[=]))" );
   attributeFormat.setForeground ( Qt::blue );
-  attributeFormat.setBackground ( bg );
   step2.format = attributeFormat;
   highlightRules.append ( step2 );
 
   RuleHighlight step3; // KeyWords
   step3.pattern = QRegExp ( "((^|\\s+)[a-zA-Z][\\w]+:)" );
   keywordFormat.setForeground ( Qt::blue );
-  keywordFormat.setBackground ( bg );
   step3.format = keywordFormat;
   highlightRules.append ( step3 );
 
   RuleHighlight step4; // Error
   step4.pattern = QRegExp ( "\\b(error|fatal|warning|invalid)", Qt::CaseInsensitive );
-  errorFormat.setForeground ( Qt::yellow );
-  errorFormat.setBackground ( Qt::darkGray );
+  errorFormat.setForeground ( Qt::darkGray );
   step4.format = errorFormat;
   highlightRules.append ( step4 );
 }
--
cgit v0.8.3.1
