--- src/mainwindow.cpp.orig	2012-02-14 17:18:14.000000000 -0200
+++ src/mainwindow.cpp	2012-02-14 21:42:35.672281451 -0200
@@ -608,9 +608,6 @@ void MainWindow::preparePreview ( bool b
       commandLine << expert;
   }
 
-  // Video Options
-  commandLine << m_videoEditor->getCmd ();
-
   // Meta Daten
   if ( m_metaData->isChecked() )
     commandLine << m_metaData->getCmd ( videoCodec() );
@@ -619,6 +616,9 @@ void MainWindow::preparePreview ( bool b
   if ( m_audioGroupBox->isChecked() )
     commandLine << m_audioEditor->getCmd ();
 
+  // Video Options
+  commandLine << m_videoEditor->getCmd ();
+
   // Output Options
   QString outFile = generateOutputFile();
   commandLine << "-y" << outFile;
