From 9cac90510a9c25cf070eb3dc90884c1554e07305 Mon Sep 17 00:00:00 2001
From: Jürgen Heinemann (Undefined) <nospam@hjcms.de>
Date: Wed, 14 Nov 2012 13:25:00 +0000
Subject: FFmpeg 1.0

---
diff --git a/src/avoptions/avoptions.cpp b/src/avoptions/avoptions.cpp
index af60851..94b813e 100644
--- a/src/avoptions/avoptions.cpp
+++ b/src/avoptions/avoptions.cpp
@@ -52,8 +52,12 @@ namespace QX11Grab
   static bool qx11grab_is_encoder ( AVCodec* _c )
   {
 #if ( LIBAVCODEC_VERSION_INT >= 3489124 )
-    // qDebug() << "NEW_ENCODER_API" << LIBAVCODEC_VERSION_INT;
-    return ( ( _c && ( _c->encode || _c->encode2 ) ) == 1 );
+// qDebug() << "NEW_ENCODER_API" << LIBAVCODEC_VERSION_INT;
+# if ( LIBAVCODEC_VERSION_INT >= 3554148 )
+  return ( ( _c && ( _c->encode2 ) ) == 1 );
+# else
+  return ( ( _c && ( _c->encode || _c->encode2 ) ) == 1 );
+# endif
 #else
     return ( ( _c && ( _c->encode ) ) == 1 );
 #endif
