--- src/qx11grab.cpp.orig	2011-07-09 01:03:22.000000000 -0300
+++ src/qx11grab.cpp	2011-07-09 01:03:33.904326171 -0300
@@ -648,9 +648,6 @@ void QX11Grab::perparePreview()
       QString::number ( r.y() )
   );
 
-  // Video Options
-  commandLine << m_videoEditor->getCmd ();
-
   // Meta Daten
   if ( m_grabberInfo->metaEnabled() )
     commandLine << m_metaData->getCmd ();
@@ -662,6 +659,9 @@ void QX11Grab::perparePreview()
     commandLine << m_audioEditor->getCmd ();
   }
 
+  // Video Options
+  commandLine << m_videoEditor->getCmd ();
+
   // Output Options
   commandLine << "-y" << m_defaults->output();
 
