--- src/mainwindow.cpp.orig	2015-04-23 19:15:44.289749327 -0300
+++ src/mainwindow.cpp	2015-04-23 19:15:52.753310486 -0300
@@ -621,9 +621,6 @@ void MainWindow::preparePreview ()
       commandLine << expert;
   }
 
-  // Video Options
-  commandLine << m_videoEditor->getCmd();
-
   // Audio Encoder Settings
   if ( m_audioGroupBox->isChecked() )
     commandLine << m_audioEditor->getCmd();
@@ -632,6 +629,9 @@ void MainWindow::preparePreview ()
   if ( m_metaData->isChecked() )
     commandLine << m_metaData->getCmd ( videoCodec() );
 
+  // Video Options
+  commandLine << m_videoEditor->getCmd();
+
   m_commandPreview->setCommandLine ( commandLine );
 
   cfg->setValue ( QLatin1String ( "SoundRecording" ), m_audioGroupBox->isChecked() );
