From 2185bdeea38efd2cbda00b6f1bcaa4abe30943ad Mon Sep 17 00:00:00 2001
From: Heinemann Jürgen (Undefined) <nospam __AT. hjcms __DOT. de>
Date: Wed, 13 Oct 2010 09:12:00 +0000
Subject: Mi 13. Okt 20:58:49 CEST 2010

---
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f7438a..1abb2b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,6 +111,9 @@ PKG_CHECK_MODULES (FFMPEG REQUIRED
   libavcodec${FFMPEG_SUFFIX}>=${FFMPEG_AVCODEC_VERSION}
 )
 
+## required by FFmpeg
+ADD_DEFINITIONS (-D__STDC_CONSTANT_MACROS)
+
 ##############################################################
 ## INCLUDES
 ##############################################################
--
cgit v0.8.3.1
