--- src/mainwindow.cpp.orig	2012-04-01 11:03:25.612334673 -0300
+++ src/mainwindow.cpp	2012-04-01 11:03:43.264582713 -0300
@@ -612,9 +612,6 @@ void MainWindow::preparePreview ()
       commandLine << expert;
   }
 
-  // Video Options
-  commandLine << m_videoEditor->getCmd();
-
   // Meta Daten
   if ( m_metaData->isChecked() )
     commandLine << m_metaData->getCmd ( videoCodec() );
@@ -623,6 +620,9 @@ void MainWindow::preparePreview ()
   if ( m_audioGroupBox->isChecked() )
     commandLine << m_audioEditor->getCmd();
 
+  // Video Options
+  commandLine << m_videoEditor->getCmd();
+
 // #ifdef MAINTAINER_REPOSITORY
 //   qDebug() << Q_FUNC_INFO << commandLine;
 // #endif
