--- CMakeLists.txt.prefix	2010-04-24 19:35:44.000000000 -0300
+++ CMakeLists.txt	2010-04-24 19:35:57.238789682 -0300
@@ -41,11 +41,7 @@
 SET (QX11GRAB_REQUIRES "" CACHE STRING "Defined in pkg-config file")
 SET (QX11GRAB_RESOURCE_FILE "${CMAKE_CURRENT_SOURCE_DIR}/src/qx11grab.qrc")
 SET (CMAKE_INSTALL_RPATH_USE_LINK_PATH  TRUE)
-
-EXEC_PROGRAM (${QT_QMAKE_EXECUTABLE}
-  ARGS "-query QT_INSTALL_PREFIX"
-  OUTPUT_VARIABLE CMAKE_INSTALL_PREFIX
-)
+SET (CMAKE_INSTALL_PREFIX "/usr" CACHE PATH "Define installation prefix")
 
 ##############################################################
 ## Set Header Definitions
