From bcebfa03ffb68159d8a73440738b31d07e55805e Mon Sep 17 00:00:00 2001
From: Heinemann Jürgen (Undefined) <nospam __AT. hjcms __DOT. de>
Date: Wed, 18 May 2011 11:54:00 +0000
Subject: Mi 18. Mai 16:14:26 CEST 2011

---
diff --git a/cmake.sh b/cmake.sh
index 57268a3..24f6d5c 100755
--- a/cmake.sh
+++ b/cmake.sh
@@ -1,22 +1,28 @@
 #!/bin/sh
 #################################################################
-# Author: Juergen Heinemann http://www.hjcms.de, (C) 2007-2011
+# Author: Juergen Heinemann http://www.hjcms.de, (C) 2007-2010
 #
 # Copyright: See COPYING file that comes with this distribution
 #################################################################
 
-if test -f $HOME/hjcms/projectconfigurations.txt ; then
-. $HOME/hjcms/projectconfigurations.txt
+if test -f $HOME/hjcms/projectconfigurations.sh ; then
+source $HOME/hjcms/projectconfigurations.sh
 fi
 
 cmake -Wdev \
 	-DCMAKE_CXX_FLAGS:STRING="${CFLAGS:-"-O2"}" \
 	-DCMAKE_INSTALL_PREFIX:PATH=/usr \
 	-DCMAKE_BUILD_TYPE:STRING=Debug \
-	-DXHTMLDBG_BUILD_VERBOSE:BOOL=ON \
-	-DCMAKE_SKIP_RPATH:BOOL=ON \
-	-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
-	-DXHTMLDBG_EXPERIMENTAL:BOOL=ON \
-	-DPHONON_STL_INCLUDE_DIR:PATH=/usr/include/KDE \
-	-DAUTOMOC4_EXECUTABLE:FILEPATH=$(which automoc4) \
-	-DCPACK_PACKAGE_INSTALL_DIRECTORY:PATH="$INSTALL_DESTDIR" $@ ../
+	$CMAKE_EXTRAS ../
+
+# cmake -Wdev \
+# 	-DCMAKE_CXX_FLAGS:STRING="${CFLAGS:-"-O2"}" \
+# 	-DCMAKE_INSTALL_PREFIX:PATH=/usr \
+# 	-DCMAKE_BUILD_TYPE:STRING=Debug \
+# 	-DXHTMLDBG_BUILD_VERBOSE:BOOL=ON \
+# 	-DCMAKE_SKIP_RPATH:BOOL=ON \
+# 	-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+# 	-DXHTMLDBG_EXPERIMENTAL:BOOL=ON \
+# 	-DPHONON_STL_INCLUDE_DIR:PATH=/usr/include/KDE \
+# 	-DAUTOMOC4_EXECUTABLE:FILEPATH=$(which automoc4) \
+# 	-DCPACK_PACKAGE_INSTALL_DIRECTORY:PATH="$INSTALL_DESTDIR" $@ ../
diff --git a/src/tableeditor/tableeditor.cpp b/src/tableeditor/tableeditor.cpp
index cdfda76..a52e86c 100644
--- a/src/tableeditor/tableeditor.cpp
+++ b/src/tableeditor/tableeditor.cpp
@@ -35,6 +35,7 @@
 extern "C"
 {
 #include <libavutil/common.h>
+#include <libavutil/avutil.h>
 #include <libavformat/avformat.h>
 #include <libavcodec/avcodec.h>
 }
@@ -72,7 +73,7 @@ void TableEditor::findVideoCodecs()
   int index = 0;
   for ( codec = av_codec_next ( 0 ); codec != NULL; codec = av_codec_next ( codec ) )
   {
-    if ( ( codec->type == CODEC_TYPE_VIDEO ) && ( codec->encode ) )
+    if ( ( codec->type == AVMEDIA_TYPE_VIDEO ) && ( codec->encode ) )
     {
       codecSelection->insertItem ( index, codec->name, QVariant ( codec->name ) );
       codecSelection->setItemData ( index, codec->name, Qt::EditRole );
@@ -104,7 +105,7 @@ void TableEditor::findAudioCodecs()
   int index = 0;
   for ( codec = av_codec_next ( 0 ); codec != NULL; codec = av_codec_next ( codec ) )
   {
-    if ( ( codec->type == CODEC_TYPE_AUDIO ) && ( codec->encode ) )
+    if ( ( codec->type == AVMEDIA_TYPE_AUDIO ) && ( codec->encode ) )
     {
       codecSelection->insertItem ( index, codec->name, QVariant ( codec->name ) );
       codecSelection->setItemData ( index, codec->name, Qt::EditRole );
--
cgit v0.8.3.1
