--- src/mainwindow.cpp.orig	2013-09-22 03:11:42.000000000 -0300
+++ src/mainwindow.cpp	2014-01-26 21:20:55.895495128 -0200
@@ -613,9 +613,6 @@ void MainWindow::preparePreview ()
       commandLine << expert;
   }
 
-  // Video Options
-  commandLine << m_videoEditor->getCmd();
-
   // Meta Daten
   if ( m_metaData->isChecked() )
     commandLine << m_metaData->getCmd ( videoCodec() );
@@ -624,6 +621,9 @@ void MainWindow::preparePreview ()
   if ( m_audioGroupBox->isChecked() )
     commandLine << m_audioEditor->getCmd();
 
+  // Video Options
+  commandLine << m_videoEditor->getCmd();
+
   m_commandPreview->setCommandLine ( commandLine );
 
   cfg->setValue ( QLatin1String ( "SoundRecording" ), m_audioGroupBox->isChecked() );
