diff -up src/settings/extensionselecter/extensionlineedit.cpp.ffmpeg src/settings/extensionselecter/extensionlineedit.cpp
--- src/settings/extensionselecter/extensionlineedit.cpp.ffmpeg	2013-07-05 20:55:32.400255165 -0300
+++ src/settings/extensionselecter/extensionlineedit.cpp	2013-07-05 20:55:42.542099390 -0300
@@ -39,7 +39,7 @@ ExtensionLineEdit::ExtensionLineEdit ( Q
   setValidator ( mask );
 }
 
-void ExtensionLineEdit::insertCompleter ( CodecID id )
+void ExtensionLineEdit::insertCompleter ( AVCodecID id )
 {
   if ( id == CODEC_ID_NONE )
     return;
diff -up src/settings/extensionselecter/extensionlineedit.h.ffmpeg src/settings/extensionselecter/extensionlineedit.h
--- src/settings/extensionselecter/extensionlineedit.h.ffmpeg	2013-07-05 20:54:52.224794759 -0300
+++ src/settings/extensionselecter/extensionlineedit.h	2013-07-05 20:55:17.243476889 -0300
@@ -40,7 +40,7 @@ class ExtensionLineEdit : public QLineEd
     Q_CLASSINFO ( "URL", "http://qx11grab.hjcms.de" )
 
   private:
-    void insertCompleter ( CodecID );
+    void insertCompleter ( AVCodecID );
 
   public Q_SLOTS:
     void setCompleterCodec ( const QString & );
diff -up src/settings/extensionselecter/extensionscombobox.cpp.ffmpeg src/settings/extensionselecter/extensionscombobox.cpp
--- src/settings/extensionselecter/extensionscombobox.cpp.ffmpeg	2013-07-05 20:52:29.023230616 -0300
+++ src/settings/extensionselecter/extensionscombobox.cpp	2013-07-05 20:52:40.844718258 -0300
@@ -48,7 +48,7 @@ ExtensionsComboBox::ExtensionsComboBox (
 * erstelle das Auswahl Menü!
 * TODO Alternativen aus Settings lesen!
 */
-void ExtensionsComboBox::insertExtensions ( CodecID id )
+void ExtensionsComboBox::insertExtensions ( AVCodecID id )
 {
   if ( id == CODEC_ID_NONE )
     return;
diff -up src/settings/extensionselecter/extensionscombobox.h.ffmpeg src/settings/extensionselecter/extensionscombobox.h
--- src/settings/extensionselecter/extensionscombobox.h.ffmpeg	2013-07-05 20:51:49.606607401 -0300
+++ src/settings/extensionselecter/extensionscombobox.h	2013-07-05 20:51:59.437518177 -0300
@@ -41,7 +41,7 @@ class ExtensionsComboBox : public QCombo
     Q_CLASSINFO ( "URL", "http://qx11grab.hjcms.de" )
 
   private:
-    void insertExtensions ( CodecID );
+    void insertExtensions ( AVCodecID );
 
   private Q_SLOTS:
     void updateItemList ( const QString & );
