--- src/avoptions/avoptions.cpp.ffmpeg	2015-11-07 02:46:31.543408927 -0200
+++ src/avoptions/avoptions.cpp	2015-11-07 02:48:39.085433692 -0200
@@ -178,9 +178,9 @@ namespace QX11Grab
   {
     QList<FFOption> list;
     avcodec_register_all();
-    for ( int i = 0; i < PIX_FMT_NB; i++ )
+    for ( int i = 0; i < AV_PIX_FMT_NB; i++ )
     {
-      const AVPixFmtDescriptor* pix_desc = &av_pix_fmt_descriptors[ static_cast<PixelFormat> ( i ) ];
+      const AVPixFmtDescriptor* pix_desc = av_pix_fmt_desc_get( static_cast<AVPixelFormat> ( i ) );
       if ( ! pix_desc->name )
         continue;
 
