diff -up Unicode-EastAsianWidth-1.33/MANIFEST.noinc Unicode-EastAsianWidth-1.33/MANIFEST
--- Unicode-EastAsianWidth-1.33/MANIFEST.noinc	2014-07-08 15:01:40.252642471 -0400
+++ Unicode-EastAsianWidth-1.33/MANIFEST	2014-07-08 15:01:47.592600413 -0400
@@ -1,16 +1,4 @@
 Changes
-inc/Module/Install.pm
-inc/Module/Install/Base.pm
-inc/Module/Install/Can.pm
-inc/Module/Install/Fetch.pm
-inc/Module/Install/GithubMeta.pm
-inc/Module/Install/Makefile.pm
-inc/Module/Install/Metadata.pm
-inc/Module/Install/Package.pm
-inc/Module/Install/Win32.pm
-inc/Module/Install/WriteAll.pm
-inc/Module/Package.pm
-inc/Pod/Markdown.pm
 lib/Unicode/EastAsianWidth.pm
 Makefile.PL
 MANIFEST			This list of files
