diff -dur QupZilla-1.6.3.orig/src/plugins/KWalletPasswords/KWalletPasswords.pro QupZilla-1.6.3/src/plugins/KWalletPasswords/KWalletPasswords.pro
--- QupZilla-1.6.3.orig/src/plugins/KWalletPasswords/KWalletPasswords.pro	2014-02-14 11:42:43.000000000 +0100
+++ QupZilla-1.6.3/src/plugins/KWalletPasswords/KWalletPasswords.pro	2014-02-18 15:35:46.924332871 +0100
@@ -45,4 +45,29 @@
 
 LIBS += -lkdeui
 
+KDEPREFIX = $$system(kde4-config --prefix)
+isEmpty(KDEPREFIX):error(KDE 4 kdelibs required.)
+exists($$KDEPREFIX/include/kde4/KDE) {
+  KDEINCDIR = $$KDEPREFIX/include/kde4
+} else:exists($$KDEPREFIX/include/kde/KDE) {
+  KDEINCDIR = $$KDEPREFIX/include/kde
+} else {
+  KDEINCDIR = $$KDEPREFIX/include
+}
+INCLUDEPATH += $$KDEINCDIR
+KDELIBDIR = $$KDEPREFIX/lib$$system(kde4-config --libsuffix)
+KDEDEVELLIBDIR = $$KDELIBDIR
+exists($$KDEDEVELLIBDIR/kde4/devel) {
+  KDEDEVELLIBDIR = $$KDEDEVELLIBDIR/kde4/devel
+}
+
+!equals(KDEDEVELLIBDIR,/usr/lib):!equals(KDEDEVELLIBDIR,/usr/lib64) {
+  QMAKE_LIBDIR = $$KDEDEVELLIBDIR $$QMAKE_LIBDIR
+}
+!equals(KDELIBDIR,/usr/lib):!equals(KDELIBDIR,/usr/lib64) {
+  !darwin-*:!macx-* {
+    LIBS += -Wl,--rpath,"$$KDELIBDIR"
+  }
+}
+
 include(../../plugins.pri)
