--- src/lib/other/qzsettings.cpp.orig	2014-03-09 00:16:32.599261069 -0300
+++ src/lib/other/qzsettings.cpp	2014-03-09 00:16:49.676600525 -0300
@@ -57,7 +57,7 @@ void QzSettings::loadSettings()
 
     settings.beginGroup("Browser-Tabs-Settings");
     newTabPosition = settings.value("OpenNewTabsSelected", false).toBool() ? Qz::NT_CleanSelectedTab : Qz::NT_CleanNotSelectedTab;
-    tabsOnTop = settings.value("TabsOnTop", true).toBool();
+    tabsOnTop = settings.value("TabsOnTop", false).toBool();
     alwaysSwitchTabsWithWheel = settings.value("AlwaysSwitchTabsWithWheel", false).toBool();
     settings.endGroup();
 }
