diff -up kdegraphics-mobipocket-4.12.0/QMobipocketConfig.cmake.kde_includes_hint kdegraphics-mobipocket-4.12.0/QMobipocketConfig.cmake
--- kdegraphics-mobipocket-4.12.0/QMobipocketConfig.cmake.kde_includes_hint	2013-11-04 14:42:18.000000000 -0600
+++ kdegraphics-mobipocket-4.12.0/QMobipocketConfig.cmake	2013-12-30 18:35:15.532983870 -0600
@@ -5,7 +5,7 @@ get_filename_component( _currentDir  ${_
 
 # find the full paths to the library and the includes:
 find_path(QMOBIPOCKET_INCLUDE_DIR qmobipocket/mobipocket.h
-          HINTS ${_currentDir}/include
+          HINTS ${_currentDir}/include ${KDE4_INCLUDE_DIR}
           NO_DEFAULT_PATH)
 
 find_library(QMOBIPOCKET_LIBRARY qmobipocket 
