--- scripts/CMakeLists.txt.orig	2012-11-21 20:21:21.671671570 -0200
+++ scripts/CMakeLists.txt	2012-11-21 20:21:31.172664113 -0200
@@ -222,15 +222,6 @@ if (DO_PERL_BINDINGS)
       endif (NOT WIN32)
     endif (RUN_SWIG)
 
-    add_library(bindings_perl MODULE ${openbabel_SOURCE_DIR}/scripts/perl/openbabel-perl.cpp)
-    target_link_libraries(bindings_perl ${PERL_LIBRARY} ${BABEL_LIBRARY} ${PERL_EXTRA_LINK_FLAGS})
-    set_target_properties(bindings_perl PROPERTIES
-        PREFIX ""
-        OUTPUT_NAME OpenBabel)
-    add_dependencies(bindings_perl openbabel)
-    install(TARGETS bindings_perl LIBRARY DESTINATION ${LIB_INSTALL_DIR})
-    install(FILES ${openbabel_SOURCE_DIR}/scripts/perl/OpenBabel.pm DESTINATION ${LIB_INSTALL_DIR})
-
     set(PERL_INSTALLDIRS_TEXT "")
     if (OBPERL_INSTALLDIRS)
       set(PERL_INSTALLDIRS_TEXT "INSTALLDIRS=${OBPERL_INSTALLDIRS}")
