diff -up midori-0.2.9/midori/midori-browser.c.orig midori-0.2.9/midori/midori-browser.c
--- midori-0.2.9/midori/midori-browser.c.orig	2010-10-31 20:20:48.000000000 -0200
+++ midori-0.2.9/midori/midori-browser.c	2010-11-01 23:37:06.108371833 -0200
@@ -4526,10 +4526,10 @@ _action_help_link_activate (GtkAction*
         }
         #else
         #ifdef DOCDIR
-        uri = "file://" DOCDIR "/midori/user/midori.html";
-        if (g_access (DOCDIR "/midori/user/midori.html", F_OK) != 0)
+        uri = "file://" DOCDIR "/user/midori.html";
+        if (g_access (DOCDIR "/user/midori.html", F_OK) != 0)
         #endif
-            uri = "error:nodocs " DOCDIR "/midori/user/midori.html";
+            uri = "error:nodocs " DOCDIR "/user/midori.html";
         #endif
     }
     else if  (!strncmp ("HelpFAQ", action_name, 7))
diff -up midori-0.2.9/wscript.orig midori-0.2.9/wscript
--- midori-0.2.9/wscript.orig	2010-10-31 20:20:48.000000000 -0200
+++ midori-0.2.9/wscript	2010-11-01 23:38:19.434267050 -0200
@@ -417,7 +417,7 @@ def build (bld):
     bld.add_group ()
 
     if bld.env['docs']:
-        bld.install_files ('${DOCDIR}/' + '/', \
+        bld.install_files ('${DOCDIR}/', \
             'AUTHORS COPYING ChangeLog EXPAT README')
 
     # Install default configuration
