diff -dur midori-0.2.4.orig/midori/midori-browser.c midori-0.2.4/midori/midori-browser.c
--- midori-0.2.4.orig/midori/midori-browser.c	2010-03-07 22:23:04.000000000 +0100
+++ midori-0.2.4/midori/midori-browser.c	2010-03-21 01:53:22.000000000 +0100
@@ -5050,10 +5050,10 @@
         }
         #else
         #ifdef DOCDIR
-        uri = "file://" DOCDIR "/midori/user/midori.html";
-        if (g_access (DOCDIR "/midori/user/midori.html", F_OK) != 0)
+        uri = "file://" DOCDIR "/user/midori.html";
+        if (g_access (DOCDIR "/user/midori.html", F_OK) != 0)
         #endif
-            uri = "error:nodocs " DOCDIR "/midori/user/midori.html";
+            uri = "error:nodocs " DOCDIR "/user/midori.html";
         #endif
     }
     else if  (!strncmp ("HelpFAQ", action_name, 7))
diff -dur midori-0.2.4.orig/wscript midori-0.2.4/wscript
--- midori-0.2.4.orig/wscript	2010-03-07 22:23:04.000000000 +0100
+++ midori-0.2.4/wscript	2010-03-21 01:51:52.000000000 +0100
@@ -426,7 +426,7 @@
     bld.add_group ()
 
     if bld.env['docs']:
-        bld.install_files ('${DOCDIR}/' + APPNAME + '/', \
+        bld.install_files ('${DOCDIR}/', \
             'AUTHORS COPYING ChangeLog EXPAT README')
 
     # Install default configuration
@@ -443,7 +443,7 @@
             '../../../docs/user/midori.txt ' + 'midori.html'
         Utils.exec_command (command)
         os.chdir ('../../..')
-        bld.install_files ('${DOCDIR}/midori/user/', blddir + '/docs/user/midori.html')
+        bld.install_files ('${DOCDIR}/user/', blddir + '/docs/user/midori.html')
 
     if bld.env['INTLTOOL']:
         obj = bld.new_task_gen ('intltool_po')
@@ -452,7 +452,7 @@
 
     if bld.env['GTKDOC_SCAN'] and Options.commands['build']:
         bld.add_subdirs ('docs/api')
-        bld.install_files ('${DOCDIR}/midori/api/', blddir + '/docs/api/*')
+        bld.install_files ('${DOCDIR}/api/', blddir + '/docs/api/*')
 
     if not is_mingw (bld.env) and Options.platform != 'win32':
         if bld.env['HAVE_HILDON']:
