diff -up src/ruby/audio/ao.cpp.ao src/ruby/audio/ao.cpp
--- bsnes/ruby/audio/ao.cpp.ao	2010-04-15 17:23:34.608679453 -0300
+++ bsnes/ruby/audio/ao.cpp	2010-04-15 17:23:49.756677876 -0300
@@ -51,6 +51,8 @@ public:
     driver_id = ao_default_driver_id(); //ao_driver_id((const char*)driver)
     if(driver_id < 0) return false;
 
+    memset(&driver_format, 0, sizeof driver_format);
+
     driver_format.bits = 16;
     driver_format.channels = 2;
     driver_format.rate = settings.frequency;
