diff -up conky-1.9.0/src/conky.c.nobool conky-1.9.0/src/conky.c
--- conky-1.9.0/src/conky.c.nobool	2012-05-03 23:22:21.000000000 +0200
+++ conky-1.9.0/src/conky.c	2012-05-07 15:56:40.000000000 +0200
@@ -882,12 +882,12 @@ void generate_text_internal(char *p, int
 			OBJ(cpu) {
 				if (cur->cpu_usage) {
 					if (obj->data.i > info.cpu_count) {
-						static bool warned = false;
+						static int warned = 0;
 						if(!warned) {
 							NORM_ERR("obj->data.i %i info.cpu_count %i",
 									obj->data.i, info.cpu_count);
 							NORM_ERR("attempting to use more CPUs than you have!");
-							warned = true;
+							warned = 1;
 						}
 					} else  {
 						percent_print(p, p_max_size,
