diff -U 3 -dHrNp -- bus/systemd-user.orig/dbus-setenv.service bus/systemd-user/dbus-setenv.service
--- bus/systemd-user.orig/dbus-setenv.service	1969-12-31 21:00:00.000000000 -0300
+++ bus/systemd-user/dbus-setenv.service	2015-10-19 18:32:48.000000000 -0200
@@ -0,0 +1,9 @@
+[Unit]
+Description=Set Environment DBUS_SESSION_BUS_ADDRESS
+DefaultDependencies=no
+Requisite=dbus.socket
+
+[Service]
+Type=oneshot
+Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=%t/bus
+ExecStart=/bin/systemctl --user import-environment DBUS_SESSION_BUS_ADDRESS
diff -U 3 -dHrNp -- bus/systemd-user.orig/dbus.socket.in bus/systemd-user/dbus.socket.in
--- bus/systemd-user.orig/dbus.socket.in	2015-10-19 18:30:44.000000000 -0200
+++ bus/systemd-user/dbus.socket.in	2015-10-19 18:30:58.000000000 -0200
@@ -1,5 +1,7 @@
 [Unit]
 Description=D-Bus User Message Bus Socket
+Requires=dbus-setenv.service
+After=dbus-setenv.service
 
 [Socket]
 ListenStream=%t/bus
