diff -Nur dbus-1.10.0.orig/bus/system.conf.in dbus-1.10.0/bus/system.conf.in
--- dbus-1.10.0.orig/bus/system.conf.in	2015-07-21 11:46:00.000000000 -0500
+++ dbus-1.10.0/bus/system.conf.in	2015-10-03 17:47:18.646635691 -0500
@@ -100,6 +100,14 @@
   <!-- Include legacy configuration location -->
   <include ignore_missing="yes">@DBUS_SYSCONFDIR@/dbus-1/system.conf</include>
 
+  <!-- Allow root to do anything over the messagebus.
+       Don't whine about "security" - anyone with root privileges
+       can edit this file anyway, so -ENOHOLE here. -->
+  <policy user="root">
+    <allow send_destination="*"/>
+    <allow send_interface="*"/>
+  </policy>
+
   <!-- Config files are placed here that among other things, punch 
        holes in the above policy for specific services. -->
   <includedir>system.d</includedir>
