diff -Nur dbus-1.4.0.orig//bus/system.conf.in dbus-1.4.0//bus/system.conf.in
--- dbus-1.4.0.orig//bus/system.conf.in	2010-08-31 16:44:19.000000000 -0500
+++ dbus-1.4.0//bus/system.conf.in	2010-09-14 09:17:56.080373096 -0500
@@ -70,6 +70,14 @@
           send_member="UpdateActivationEnvironment"/>
   </policy>
 
+  <!-- Allow root to do anything over the messagebus.
+       Don't whine about "security" - anyone with root privileges
+       can edit this file anyway, so -ENOHOLE here. -->
+  <policy user="root">
+    <allow send_destination="*"/>
+    <allow send_interface="*"/>
+  </policy>
+
   <!-- Config files are placed here that among other things, punch 
        holes in the above policy for specific services. -->
   <includedir>system.d</includedir>
