From d8afa0a10bc421b3fd32fe9320bfd277649b3f63 Mon Sep 17 00:00:00 2001
From: Christian Dywan <christian.dywan@lanedo.com>
Date: Thu, 15 Jul 2010 11:18:33 +0000
Subject: Assert in _dbus_decompose_path if path is NULL or len is wrong

---
diff --git a/dbus/dbus-object-tree.c b/dbus/dbus-object-tree.c
index 5aab9e6..28cfc8b 100644
--- a/dbus/dbus-object-tree.c
+++ b/dbus/dbus-object-tree.c
@@ -1080,6 +1080,7 @@ _dbus_decompose_path (const char*     data,
   int i, j, comp;
 
   _dbus_assert (data != NULL);
+  _dbus_assert (path != NULL);
   
 #if VERBOSE_DECOMPOSE
   _dbus_verbose ("Decomposing path \"%s\"\n",
@@ -1092,6 +1093,7 @@ _dbus_decompose_path (const char*     data,
       i = 0;
       while (i < len)
         {
+          _dbus_assert (data[i] != '\0');
           if (data[i] == '/')
             n_components += 1;
           ++i;
--
cgit v0.8.3-6-g21f6
