--- configure.ac.orig	2014-02-20 18:15:16.028982532 -0300
+++ configure.ac	2014-02-20 18:15:19.748195688 -0300
@@ -1141,7 +1141,7 @@ if test x$enable_systemd = xno ; then
     have_systemd=no;
 else
     PKG_CHECK_MODULES(SYSTEMD,
-        [libsystemd-login >= 32, libsystemd-daemon >= 32, libsystemd-journal >= 32],
+        [libsystemd >= 209],
         have_systemd=yes,
         have_systemd=no)
 fi
