diff -up jack-1.9.6/dbus/sigsegv.c.uc_regs jack-1.9.6/dbus/sigsegv.c
--- jack-1.9.6/dbus/sigsegv.c.uc_regs	2010-08-30 10:24:16.000000000 +0000
+++ jack-1.9.6/dbus/sigsegv.c	2011-03-11 20:40:46.000000000 +0000
@@ -101,8 +101,10 @@ static void signal_segv(int signum, sigi
 #if !defined(__alpha__) && !defined(__ia64__) && !defined(__FreeBSD_kernel__) && !defined(__arm__) && !defined(__hppa__) && !defined(__sh__)
     for(i = 0; i < NGREG; i++)
         jack_error("reg[%02d]       = 0x" REGFORMAT, i, 
-#if defined(__powerpc__)
+#if defined(__powerpc__) && !defined(__powerpc64__)
                 ucontext->uc_mcontext.uc_regs[i]
+#elif defined(__powerpc__) && defined(__powerpc64__)
+                ucontext->uc_mcontext.gp_regs[i]
 #elif defined(__sparc__) && defined(__arch64__)
                 ucontext->uc_mcontext.mc_gregs[i]
 #else
