diff -rupN jack-1.9.8.old/doxyfile jack-1.9.8/doxyfile
--- jack-1.9.8.old/doxyfile	2011-12-19 06:53:48.000000000 -0500
+++ jack-1.9.8/doxyfile	2011-12-24 14:04:23.569998760 -0500
@@ -574,22 +574,7 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories
 # with spaces.
 
-INPUT                  = common \
-                         posix \
-                         macosx \
-                         macosx/coreaudio/ \
-			 macosx/coremidi/ \
-                         linux \
-                         linux/alsa \
-			 linux/alsarawmidi \
-			 linux/firewire \
-			 linux/freebob \
-                         windows \
-                         windows/portaudio \
-                         windows/winmme \
-                         solaris \
-			 solaris/oss \
-                         common/jack/
+INPUT                  = common/jack/
 
 # This tag can be used to specify the character encoding of the source files
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is
