--- src/parallel.pod.orig	2013-06-13 00:24:40.301414641 +0200
+++ src/parallel.pod		2013-06-13 00:48:20.270905575 +0200
@@ -528,18 +528,18 @@
 
 =over 3
 
-=item 0
+=item .0
 
 Do not halt if a job fails. Exit status will be the number of jobs
 failed. This is the default.
 
-=item 1
+=item .1
 
 Do not start new jobs if a job fails, but complete the running jobs
 including cleanup. The exit status will be the exit status from the
 last failing job.
 
-=item 2
+=item .2
 
 Kill off all jobs immediately and exit without cleanup. The exit
 status will be the exit status from the failing job.
@@ -2756,7 +2756,7 @@
 
 =over 6
 
-=item 0
+=item .0
 
 All jobs ran without error.
 
@@ -2764,11 +2764,11 @@
 
 Some of the jobs failed. The exit status gives the number of failed jobs
 
-=item 254
+=item .254
 
 More than 253 jobs failed.
 
-=item 255
+=item .255
 
 Other error.
 
