diff -up unixodbc-gui-qt-20131023/ODBCTestQ4/DlgEnvAllocEnv.cpp.qstring unixodbc-gui-qt-20131023/ODBCTestQ4/DlgEnvAllocEnv.cpp
--- unixodbc-gui-qt-20131023/ODBCTestQ4/DlgEnvAllocEnv.cpp.qstring	2010-02-27 05:18:46.000000000 -0300
+++ unixodbc-gui-qt-20131023/ODBCTestQ4/DlgEnvAllocEnv.cpp	2013-10-23 18:24:00.275676387 -0200
@@ -63,7 +63,7 @@ void DlgEnvAllocEnv::slotDone()
         txt.sprintf( "    *phenv: %p", henv );
         pOdbcTest->out_win->append( txt );
 
-        pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_ENV, henv ) );
+        pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_ENV, henv, "" ) );
     }
 
     pOdbcTest->out_win->append( "" );
diff -up unixodbc-gui-qt-20131023/ODBCTestQ4/conn.cpp.qstring unixodbc-gui-qt-20131023/ODBCTestQ4/conn.cpp
--- unixodbc-gui-qt-20131023/ODBCTestQ4/conn.cpp.qstring	2012-07-25 12:02:10.000000000 -0300
+++ unixodbc-gui-qt-20131023/ODBCTestQ4/conn.cpp	2013-10-23 18:24:49.162333097 -0200
@@ -3113,9 +3113,9 @@ void dFullConnect::Ok()
      */
     if ( !hand )
     {
-        pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_ENV, henv ) );
+        pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_ENV, henv, "" ) );
     }
-    pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_DBC, hdbc ) );
+    pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_DBC, hdbc, "" ) );
     pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_STMT, hstmt, pOdbcTest->listHandle  ) );
 
     pOdbcTest->out_win->append( "Full Connect Succeeded\n" );
@@ -4058,7 +4058,7 @@ void dAllocConnect::Ok()
         txt.sprintf( "    *phdbc: %p", hdbc );
         pOdbcTest->out_win->append( txt );
 
-        pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_DBC, hdbc ) );
+        pOdbcTest->listHandle.append( new OdbcHandle( SQL_HANDLE_DBC, hdbc, "" ) );
     }
 
     pOdbcTest->out_win->append( "" );
