--- cpupower.c.orig	2016-06-05 07:19:23.000000000 -0300
+++ cpupower.c	2016-12-02 17:21:46.584414398 -0200
@@ -56,6 +56,7 @@
 
 #include <gkrellm2/gkrellm.h>
 #include <cpufreq.h>
+#include <cpuidle.h>
 
 /* version number */
 #define  VERSION        "0.1.6"
@@ -563,7 +564,7 @@ GkrellmMonitor* gkrellm_init_plugin(void
   monitor = &plugin_mon;
 
   /* determine number of cpus */
-  for( ncpu = 0; cpufreq_cpu_exists(ncpu)==0; ++ncpu )
+  for( ncpu = 0; cpupower_is_cpu_online(ncpu)!=0; ++ncpu )
     ;
   ncpu = ncpu > NCPU_MAX ? NCPU_MAX : ncpu;
 
