From 47322e554cc5388a3e6325f36b7d07a13f124594 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Thu, 31 Mar 2011 16:28:44 +0000
Subject: Add Unity to list of registered environments

See http://lists.freedesktop.org/archives/xdg/2011-March/011856.html
---
diff --git a/src/validate.c b/src/validate.c
index 04bc8d0..9c399a3 100644
--- a/src/validate.c
+++ b/src/validate.c
@@ -338,7 +338,7 @@ static struct {
 };
 
 static const char *show_in_registered[] = {
-  "GNOME", "KDE", "LXDE", "ROX", "XFCE", "Old"
+  "GNOME", "KDE", "LXDE", "ROX", "Unity", "XFCE", "Old"
 };
 
 static struct {
--
cgit v0.8.3-6-g21f6

