--- Makefile.orig	2010-07-15 15:46:35.000000000 -0300
+++ Makefile	2010-07-15 15:50:21.555151333 -0300
@@ -123,18 +123,15 @@
 
 default: $(DEP) x264$(EXE)
 
-libx264.a: .depend $(OBJS) $(OBJASM)
-	$(AR) rc libx264.a $(OBJS) $(OBJASM)
-	$(RANLIB) libx264.a
-
 $(SONAME): .depend $(OBJS) $(OBJASM) $(OBJSO)
 	$(CC) -shared -o $@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
+	ln -s $(SONAME) libx264.so
 
-x264$(EXE): $(OBJCLI) libx264.a
-	$(CC) -o $@ $+ $(LDFLAGSCLI) $(LDFLAGS)
+x264$(EXE): $(OBJCLI) $(SONAME)
+	$(CC) -o $@ $+ $(LDFLAGS) $(LDFLAGSCLI) -lx264 -L.
 
-checkasm: tools/checkasm.o libx264.a
-	$(CC) -o $@ $+ $(LDFLAGS)
+checkasm: tools/checkasm.o $(SONAME)
+	$(CC) -o $@ $+ $(LDFLAGS) -lx264 -L.
 
 %.o: %.asm
 	$(AS) $(ASFLAGS) -o $@ $<
@@ -202,10 +199,8 @@
 	install -d $(DESTDIR)$(libdir)
 	install -d $(DESTDIR)$(libdir)/pkgconfig
 	install -m 644 x264.h $(DESTDIR)$(includedir)
-	install -m 644 libx264.a $(DESTDIR)$(libdir)
 	install -m 644 x264.pc $(DESTDIR)$(libdir)/pkgconfig
 	install x264$(EXE) $(DESTDIR)$(bindir)
-	$(RANLIB) $(DESTDIR)$(libdir)/libx264.a
 ifeq ($(SYS),MINGW)
 	$(if $(SONAME), install -m 755 $(SONAME) $(DESTDIR)$(bindir))
 else
