Index: exmap-0.10/kernel/Makefile
===================================================================
--- exmap-0.10.orig/kernel/Makefile
+++ exmap-0.10/kernel/Makefile
@@ -7,7 +7,7 @@ obj-m += exmap.o
 build: kernel_modules
 
 kernel_modules:
-	make -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) modules
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) modules
 
 clean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) clean
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) clean
