diff -up nautilus-2.29.1/data/nautilus-autorun-software.desktop.in.desktop-key nautilus-2.29.1/data/nautilus-autorun-software.desktop.in
--- nautilus-2.29.1/data/nautilus-autorun-software.desktop.in.desktop-key	2009-12-17 17:18:10.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-autorun-software.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -2,6 +2,7 @@
 _Name=Autorun Prompt
 TryExec=nautilus-autorun-software
 Exec=nautilus-autorun-software %u
+X-Gnome-Vfs-System=gio
 Icon=application-x-executable
 NoDisplay=true
 Terminal=false
diff -up nautilus-2.29.1/data/nautilus-autorun-software.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus-autorun-software.desktop.in.in
--- nautilus-2.29.1/data/nautilus-autorun-software.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-autorun-software.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -2,6 +2,7 @@
 _Name=Autorun Prompt
 TryExec=nautilus-autorun-software
 Exec=nautilus-autorun-software %u
+X-Gnome-Vfs-System=gio
 Icon=application-x-executable
 NoDisplay=true
 Terminal=false
diff -up nautilus-2.29.1/data/nautilus-browser.desktop.in.desktop-key nautilus-2.29.1/data/nautilus-browser.desktop.in
--- nautilus-2.29.1/data/nautilus-browser.desktop.in.desktop-key	2009-12-17 17:18:10.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-browser.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=File Browser
 _Comment=Browse the file system with the file manager
 TryExec=nautilus
 Exec=nautilus --no-desktop --browser %U
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 StartupNotify=true
diff -up nautilus-2.29.1/data/nautilus-browser.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus-browser.desktop.in.in
--- nautilus-2.29.1/data/nautilus-browser.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-browser.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=File Browser
 _Comment=Browse the file system with the file manager
 TryExec=nautilus
 Exec=nautilus --no-desktop --browser %U
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 StartupNotify=true
diff -up nautilus-2.29.1/data/nautilus-computer.desktop.in.desktop-key nautilus-2.29.1/data/nautilus-computer.desktop.in
--- nautilus-2.29.1/data/nautilus-computer.desktop.in.desktop-key	2009-12-17 17:18:09.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-computer.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=Computer
 _Comment=Browse all local and remote disks and folders accessible from this computer
 TryExec=nautilus
 Exec=nautilus --no-desktop computer:
+X-Gnome-Vfs-System=gio
 Icon=computer
 Terminal=false
 StartupNotify=true
diff -up nautilus-2.29.1/data/nautilus-computer.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus-computer.desktop.in.in
--- nautilus-2.29.1/data/nautilus-computer.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-computer.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=Computer
 _Comment=Browse all local and remote disks and folders accessible from this computer
 TryExec=nautilus
 Exec=nautilus --no-desktop computer:
+X-Gnome-Vfs-System=gio
 Icon=computer
 Terminal=false
 StartupNotify=true
diff -up nautilus-2.29.1/data/nautilus.desktop.desktop-key nautilus-2.29.1/data/nautilus.desktop
--- nautilus-2.29.1/data/nautilus.desktop.desktop-key	2009-12-17 17:18:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus.desktop	2009-12-18 11:08:58.000000000 +0100
@@ -100,6 +100,7 @@ Name[zh_HK]=檔案管理程式
 Name[zh_TW]=檔案管理程式
 Name[zu]=Mphathi Hele
 Exec=nautilus
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -up nautilus-2.29.1/data/nautilus.desktop.in.desktop-key nautilus-2.29.1/data/nautilus.desktop.in
--- nautilus-2.29.1/data/nautilus.desktop.in.desktop-key	2009-12-17 17:18:09.000000000 +0100
+++ nautilus-2.29.1/data/nautilus.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=File Manager
 Exec=nautilus
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -up nautilus-2.29.1/data/nautilus.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus.desktop.in.in
--- nautilus-2.29.1/data/nautilus.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 _Name=File Manager
 Exec=nautilus
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -up nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in.desktop-key nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in
--- nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in.desktop-key	2009-12-17 17:18:09.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -2,6 +2,7 @@
 _Name=File Management
 _Comment=Change the behaviour and appearance of file manager windows
 Exec=nautilus-file-management-properties
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -up nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in.in
--- nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-file-management-properties.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -2,6 +2,7 @@
 _Name=File Management
 _Comment=Change the behaviour and appearance of file manager windows
 Exec=nautilus-file-management-properties
+X-Gnome-Vfs-System=gio
 Icon=system-file-manager
 Terminal=false
 Type=Application
diff -up nautilus-2.29.1/data/nautilus-folder-handler.desktop.in.desktop-key nautilus-2.29.1/data/nautilus-folder-handler.desktop.in
--- nautilus-2.29.1/data/nautilus-folder-handler.desktop.in.desktop-key	2009-12-17 17:18:10.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-folder-handler.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -2,6 +2,7 @@
 _Name=Open Folder
 TryExec=nautilus
 Exec=nautilus --no-desktop %U
+X-Gnome-Vfs-System=gio
 NoDisplay=true
 Terminal=false
 Icon=folder-open
diff -up nautilus-2.29.1/data/nautilus-folder-handler.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus-folder-handler.desktop.in.in
--- nautilus-2.29.1/data/nautilus-folder-handler.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-folder-handler.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -2,6 +2,7 @@
 _Name=Open Folder
 TryExec=nautilus
 Exec=nautilus --no-desktop %U
+X-Gnome-Vfs-System=gio
 NoDisplay=true
 Terminal=false
 Icon=folder-open
diff -up nautilus-2.29.1/data/nautilus-home.desktop.in.desktop-key nautilus-2.29.1/data/nautilus-home.desktop.in
--- nautilus-2.29.1/data/nautilus-home.desktop.in.desktop-key	2009-12-17 17:18:09.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-home.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=Home Folder
 _Comment=Open your personal folder
 TryExec=nautilus
 Exec=nautilus --no-desktop
+X-Gnome-Vfs-System=gio
 Icon=user-home
 Terminal=false
 StartupNotify=true
diff -up nautilus-2.29.1/data/nautilus-home.desktop.in.in.desktop-key nautilus-2.29.1/data/nautilus-home.desktop.in.in
--- nautilus-2.29.1/data/nautilus-home.desktop.in.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/data/nautilus-home.desktop.in.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=Home Folder
 _Comment=Open your personal folder
 TryExec=nautilus
 Exec=nautilus --no-desktop
+X-Gnome-Vfs-System=gio
 Icon=user-home
 Terminal=false
 StartupNotify=true
diff -up nautilus-2.29.1/src/network-scheme.desktop.in.desktop-key nautilus-2.29.1/src/network-scheme.desktop.in
--- nautilus-2.29.1/src/network-scheme.desktop.in.desktop-key	2009-12-15 11:22:12.000000000 +0100
+++ nautilus-2.29.1/src/network-scheme.desktop.in	2009-12-18 11:08:58.000000000 +0100
@@ -3,6 +3,7 @@ _Name=Network
 _Comment=Browse bookmarked and local network locations
 TryExec=nautilus
 Exec=nautilus --no-desktop network:
+X-Gnome-Vfs-System=gio
 Terminal=false
 StartupNotify=true
 Type=Application
