--- mozilla-firetray-0.3.1/components/SConscript.use-arch-dep-mozconfig	2011-07-05 22:02:01.000000000 +0100
+++ mozilla-firetray-0.3.1/components/SConscript	2011-09-11 00:23:04.000000000 +0000
@@ -1,4 +1,5 @@
 import sys, os, string, re
+import platform as plat
 
 platform = sys.platform
 
@@ -55,9 +56,10 @@ except:
         sys.exit(1)
 
 # Hack to detect Mozilla version
+arch = plat.architecture()[0].split("bit")[0]
 version_re = re.compile('#define MOZILLA_VERSION "(.*?)"')
 xpcom_libs = ['xpcomglue_s']
-with open(gecko_include + os.sep + 'mozilla-config.h', 'r') as f:
+with open(gecko_include + os.sep + 'mozilla-config' + arch + '.h', 'r') as f:
     for line in f:
         version_match = version_re.match(line)
         if version_match:
