--- mozilla-firetray-0.3.1/components/nsTray.cpp.use-our-icons	2011-07-05 22:02:01.000000000 +0100
+++ mozilla-firetray-0.3.1/components/nsTray.cpp	2011-09-10 09:54:44.000000000 +0000
@@ -749,7 +749,15 @@ NS_IMETHODIMP nsTray::SetDefaultXpmIcon(
  this->default_icon = gdk_pixbuf_new_from_xpm_data((const char**)df_icon);
  this->special_icon = gdk_pixbuf_new_from_xpm_data((const char**)sp_icon);
 
- gtk_status_icon_set_from_pixbuf(GTK_STATUS_ICON(this->systray_icon), GDK_PIXBUF(this->default_icon));
+ GList*  list=gtk_window_list_toplevels();
+
+ if(list) {
+   GList* pos=g_list_first(list);
+   GtkWindow *w=(GtkWindow *)pos->data;
+   gtk_status_icon_set_from_pixbuf(GTK_STATUS_ICON(this->systray_icon), GDK_PIXBUF(gtk_window_get_icon(w)));
+ }
+ else
+   gtk_status_icon_set_from_pixbuf(GTK_STATUS_ICON(this->systray_icon), GDK_PIXBUF(this->default_icon));
 /*
  gtk_status_icon_set_tooltip(this->systray_icon, text);
  gtk_status_icon_set_visible(this->systray_icon, TRUE);*/
