--- firetray-0.2.8-src/components/SConscript.old	2010-07-16 15:28:40.000000000 +0100
+++ firetray-0.2.8-src/components/SConscript	2010-07-16 15:29:19.000000000 +0100
@@ -43,7 +43,7 @@ bhdr = Builder(action = 'xpidl -w -m hea
 
 # Create environment object for build
 env = Environment(
-    CPPPATH = [gecko_include, '/usr/include/nspr' ],
+    CPPPATH = [gecko_include],
     LIBPATH = [gecko_lib],
     LIBS = ['xpcomglue', 'xpcomglue_s.a' ],
     ENV = os.environ)
@@ -53,6 +53,7 @@ env.Append(BUILDERS = {'MozXPT' : bxpt, 
 # Add additional param for different platforms
 if platform == 'linux2':
     env.ParseConfig('pkg-config --static --cflags gtk+-2.0') # libnotify
+    env.ParseConfig('pkg-config --cflags nspr')
     env.AppendUnique(
         CXXFLAGS = ['-include', 'xpcom-config.h', '-Wall', '-O2',
                     '-fno-rtti', '-fno-exceptions',
