--- mozilla-firetray-0.2.8/components/nsTrayModue.cpp.fix-xpcom-registration	2010-07-13 15:23:21.000000000 +0100
+++ mozilla-firetray-0.2.8/components/nsTrayModue.cpp	2010-09-14 15:45:59.000000000 +0000
@@ -1,16 +1,27 @@
-#include "nsIGenericFactory.h"
+#include "mozilla/ModuleUtils.h"
 #include "nsTray.h"
 
 NS_GENERIC_FACTORY_CONSTRUCTOR(nsTray)
 
-static nsModuleComponentInfo components[] =
-{
-    {
-       NS_ITRAY_CLASSNAME, 
-       NS_ITRAY_CID,
-       NS_ITRAY_CONTRACTID,
-       nsTrayConstructor,
-    }
+NS_DEFINE_NAMED_CID(NS_ITRAY_CID);
+
+static const mozilla::Module::CIDEntry kTrayCIDs[] = {
+     { &kNS_ITRAY_CID, false, NULL, nsTrayConstructor },
+     { NULL }
+};
+ 
+static const mozilla::Module::ContractIDEntry kTrayContracts[] = {
+    { NS_ITRAY_CONTRACTID, &kNS_ITRAY_CID },
+    { NULL }
+};
+
+static const mozilla::Module kTrayModule =
+{       
+       mozilla::Module::kVersion,
+       kTrayCIDs,
+       kTrayContracts,
+       NULL
 };
 
-NS_IMPL_NSGETMODULE("nsTrayMoudle", components)
+NSMODULE_DEFN(nsTrayModule) = &kTrayModule;
+NS_IMPL_MOZILLA192_NSGETMODULE(&kTrayModule)
--- mozilla-firetray-0.2.8/chrome.manifest.old	2010-09-14 15:51:16.000000000 +0000
+++ mozilla-firetray-0.2.8/chrome.manifest	2010-09-14 15:52:12.000000000 +0000
@@ -7,6 +7,14 @@ overlay chrome://sunbird/content/calenda
 overlay chrome://navigator/content/navigator.xul chrome://firetray/content/navigatorOverlay.xul	
 overlay	chrome://chatzilla/content/chatzilla.xul chrome://firetray/content/ircOverlay.xul
 
+binary-component components/libnptray.so
+ 
+component {77284574-9091-4b63-a5cf-533edb2897a1} components/nsMinimize.js
+contract @mozilla.org/Minimize;1 {77284574-9091-4b63-a5cf-533edb2897a1}
+
+interfaces components/nsITray.xpt
+interfaces components/nsIMinimize.xpt
+
 locale  firetray    en-US   jar:chrome/firetray.jar!/locale/en-US/
 locale  firetray    it-IT   jar:chrome/firetray.jar!/locale/it-IT/
 locale  firetray    bg-BG   jar:chrome/firetray.jar!/locale/bg-BG/
--- mozilla-firetray-0.2.8/components/nsMinimize.js.old	2010-09-14 15:54:25.000000000 +0000
+++ mozilla-firetray-0.2.8/components/nsMinimize.js	2010-09-14 15:59:39.000000000 +0000
@@ -1,25 +1,7 @@
 /***********************************************************
-constants
-***********************************************************/
-
-// reference to the interface defined in nsIMinimize.idl
-const nsIMinimize = Components.interfaces.nsIMinimize;
-
-// reference to the required base interface that all components must support
-const nsISupports = Components.interfaces.nsISupports;
-
-// UUID uniquely identifying our component
-const CLASS_ID = Components.ID("{77284574-9091-4b63-a5cf-533edb2897a1}");
-
-// description
-const CLASS_NAME = "Minimize Javascript XPCOM Component for FireTray";
-
-// textual unique identifier
-const CONTRACT_ID = "@mozilla.org/Minimize;1";
-
-/***********************************************************
 class definition
 ***********************************************************/
+Components.utils.import("resource://gre/modules/XPCOMUtils.jsm");
 
 //class constructor
 function Minimize() {
@@ -27,6 +9,13 @@ function Minimize() {
 
 // class definition
 Minimize.prototype = {
+
+    classDescription: "Minimize Javascript XPCOM Component for FireTray",
+    classID:          Components.ID("{77284574-9091-4b63-a5cf-533edb2897a1}"),
+    contractID:       "@mozilla.org/Minimize;1",
+    QueryInterface:   XPCOMUtils.generateQI([Components.interfaces.nsIMinimize,
+                                             Components.interfaces.nsISupports]),
+
     _all_hidden: false,
     _menu_window_list: 0,
 
@@ -45,69 +34,18 @@ Minimize.prototype = {
 
     set menu_window_list(menu) {
         this._menu_window_list = menu;
-    },
-
-    QueryInterface: function(aIID) {
-        if (!aIID.equals(nsIMinimize) && !aIID.equals(nsISupports)) {
-            throw Components.results.NS_ERROR_NO_INTERFACE;
-        }
-        return this;
-    }
-};
-
-/***********************************************************
-class factory
-
-This object is a member of the global-scope Components.classes.
-It is keyed off of the contract ID. Eg:
-
-myMinimize = Components.classes["@dietrich.ganx4.com/helloworld;1"].
-                                                    createInstance(Components.interfaces.nsIMinimize);
-
-***********************************************************/
-var MinimizeFactory = {
-    createInstance: function (aOuter, aIID) {
-        if (aOuter != null) {
-            throw Components.results.NS_ERROR_NO_AGGREGATION;
-        }
-        return (new Minimize()).QueryInterface(aIID);
     }
 };
 
 /***********************************************************
-module definition (xpcom registration)
-***********************************************************/
-var MinimizeModule = {
-    _firstTime: true,
-    registerSelf: function(aCompMgr, aFileSpec, aLocation, aType) {
-        aCompMgr = aCompMgr.QueryInterface(Components.interfaces.nsIComponentRegistrar);
-        aCompMgr.registerFactoryLocation(CLASS_ID, CLASS_NAME, CONTRACT_ID, aFileSpec, aLocation, aType);
-    },
-
-    unregisterSelf: function(aCompMgr, aLocation, aType) {
-        aCompMgr = aCompMgr.QueryInterface(Components.interfaces.nsIComponentRegistrar);
-        aCompMgr.unregisterFactoryLocation(CLASS_ID, aLocation);                
-    },
-    
-    getClassObject: function(aCompMgr, aCID, aIID) {
-        if (!aIID.equals(Components.interfaces.nsIFactory)) {
-            throw Components.results.NS_ERROR_NOT_IMPLEMENTED;
-        }
-
-        if (aCID.equals(CLASS_ID)) {
-            return MinimizeFactory;
-        }
-
-        throw Components.results.NS_ERROR_NO_INTERFACE;
-    },
-
-    canUnload: function(aCompMgr) { return true; }
-};
-
-/***********************************************************
 module initialization
 
 When the application registers the component, this function
 is called.
+XPCOMUtils.generateNSGetFactory was introduced in Mozilla 2 (Firefox 4).
+XPCOMUtils.generateNSGetModule is for Mozilla 1.9.2 (Firefox 3.6).
 ***********************************************************/
-function NSGetModule(aCompMgr, aFileSpec) { return MinimizeModule; }
+if (XPCOMUtils.generateNSGetFactory)
+    var NSGetFactory = XPCOMUtils.generateNSGetFactory([Minimize]);
+else
+    var NSGetModule = XPCOMUtils.generateNSGetModule([Minimize]);
--- mozilla-firetray-0.2.8/install.rdf.old	2010-09-14 16:05:29.000000000 +0000
+++ mozilla-firetray-0.2.8/install.rdf	2010-09-14 16:05:46.000000000 +0000
@@ -15,7 +15,7 @@
       <Description>
         <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
         <em:minVersion>3.0</em:minVersion>
-	    <em:maxVersion>3.6.*</em:maxVersion>
+	    <em:maxVersion>4.0.*</em:maxVersion>
       </Description>
     </em:targetApplication>
 
--- mozilla-firetray-0.2.8/components/SConscript.old	2010-09-14 16:37:32.000000000 +0000
+++ mozilla-firetray-0.2.8/components/SConscript	2010-09-14 16:37:58.000000000 +0000
@@ -18,7 +18,7 @@ print "linking: " + linking_options
 
 
 
-FLAGS += ' -fshort-wchar '
+FLAGS += ' -fshort-wchar -include mozilla-config.h '
 
 
 
@@ -63,7 +63,7 @@ bhdr = Builder(action = 'xpidl -w -m hea
 env = Environment(
     CPPPATH = [gecko_include],
     LIBPATH = [gecko_lib],
-    LIBS = ['xpcomglue', 'xpcomglue_s.a' ],
+    LIBS = ['xpcomglue', 'xpcomglue_s.a','mozalloc' ],
     ENV = os.environ)
 env.AppendENVPath('PATH', gecko_bin)
 env.Append(BUILDERS = {'MozXPT' : bxpt, 'MozHeader' : bhdr })
