--- mozilla-firetray-0.2.8/chrome/content/core.js.old	2010-07-16 19:33:54.000000000 +0100
+++ mozilla-firetray-0.2.8/chrome/content/core.js	2010-07-16 19:34:33.000000000 +0100
@@ -433,6 +433,20 @@ FireTray.getMailCount = function() {
     var msgs = folders[0].getNumUnread(true);
     var spam_msgs = 0;
 
+    var accounts = FireTray.accountManager.accounts;
+    for (var i = 0; i < accounts.Count(); i++) {  
+        var account = accounts.QueryElementAt(i, Components.interfaces.nsIMsgAccount);  
+        var rootfolder = account.incomingServer.rootFolder.QueryInterface(Components.interfaces.nsIMsgFolder);
+        if (rootfolder.hasSubFolders) {  
+            var subFolders=rootfolder.subFolders;
+            while(subFolders.hasMoreElements()) {  
+               var folder = subFolders.getNext().QueryInterface(Components.interfaces.nsIMsgFolder);  
+               if(folder.name == "Inbox" )
+               msgs+=folder.getNumUnread(true);
+            }
+        }
+    }  
+
     for(var i=0; i< allServers.Count(); i++)     // Gets all folders we need to check and all spam folders
     {
         var server = allServers.GetElementAt(i).QueryInterface(Components.interfaces.nsIMsgIncomingServer);
@@ -443,12 +457,6 @@ FireTray.getMailCount = function() {
 
         if(spamFolderURI!=null && spamFolderURIs.indexOf(spamFolderURI)<0 )
         { spamFolderURIs.push(spamFolderURI); }
-
-        if(folders.indexOf(folder)<0) //avoid considering folders multiple times
-        {
-          folders.push(folder);
-          msgs += folder.getNumUnread(true);
-        }
     }
 
     for(var i=0; i<spamFolderURIs.length; i++) //get spam mail count
