--- mozilla-firetray-0.2.8/install.rdf.add-chatzilla-support	2010-07-13 15:23:22.000000000 +0100
+++ mozilla-firetray-0.2.8/install.rdf	2010-07-16 20:41:49.000000000 +0100
@@ -15,7 +15,7 @@
       <Description>
         <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
         <em:minVersion>3.0</em:minVersion>
-	<em:maxVersion>3.6.*</em:maxVersion>
+	    <em:maxVersion>3.6.*</em:maxVersion>
       </Description>
     </em:targetApplication>
 
@@ -29,12 +29,12 @@
     </em:targetApplication>
     
     <em:targetApplication>
-	    <!-- Songbird -->
-	    <Description>
-		    <em:id>songbird@songbirdnest.com</em:id>
-		    <em:minVersion>0.8</em:minVersion>
-		    <em:maxVersion>1.2.0</em:maxVersion>
-	    </Description>
+	  <!-- Songbird -->
+	  <Description>
+		<em:id>songbird@songbirdnest.com</em:id>
+		<em:minVersion>0.8</em:minVersion>
+		<em:maxVersion>1.2.0</em:maxVersion>
+	  </Description>
     </em:targetApplication>   
     
     <em:targetApplication>  
@@ -49,11 +49,20 @@
     <em:targetApplication>
       <!-- Sunbird -->
       <Description>
-	      <em:id>{718e30fb-e89b-41dd-9da7-e25a45638b28}</em:id> 
-	<em:minVersion>0.5</em:minVersion>
-	<em:maxVersion>1.0pre</em:maxVersion>
+	    <em:id>{718e30fb-e89b-41dd-9da7-e25a45638b28}</em:id> 
+	    <em:minVersion>0.5</em:minVersion>
+	    <em:maxVersion>1.0pre</em:maxVersion>
       </Description>
     </em:targetApplication>
+
+    <em:targetApplication>
+      <!-- ChatZilla -->
+      <Description>
+	    <em:id>{59c81df5-4b7a-477b-912d-4e0fdf64e5f2}</em:id> 
+	    <em:minVersion>0.8</em:minVersion>
+	    <em:maxVersion>0.9.*</em:maxVersion>
+      </Description>
+    </em:targetApplication> 
     
     <!-- Front End MetaData -->
     <em:name>FireTray</em:name>
diff -Naur firetray-0.2.6/chrome/content/core.js firetray/chrome/content/core.js
--- firetray-0.2.6/chrome/content/core.js	2010-06-26 20:15:36.000000000 +0100
+++ firetray/chrome/content/core.js	2010-06-26 20:40:43.000000000 +0100
@@ -290,7 +290,11 @@ FireTray.getDefaultAppString = function(
    var text="";
    switch(appcode)
    {
-        case 10: //seamonkey
+    case 11: //chatzilla
+        text="Firetray (ChatZilla)";
+            break;
+
+    case 10: //seamonkey
   	   	text="Firetray (Seamonkey)";
    	        break;
 
@@ -356,6 +360,7 @@ FireTray.getMozillaAppCode = function() 
   const SONGBIRD_ID = "songbird@songbirdnest.com";
   const SUNBIRD_ID = "{718e30fb-e89b-41dd-9da7-e25a45638b28}";
   const SEAMONKEY_ID = "{92650c4d-4b8e-4d2a-b7eb-24ecf4f6b63a}";
+  const CHATZILLA_ID = "{59c81df5-4b7a-477b-912d-4e0fdf64e5f2}";
   
   var appname=appInfo.name.toLowerCase()
 
@@ -391,6 +396,11 @@ FireTray.getMozillaAppCode = function() 
         return 10;  //Seamonkey
         break;
 
+     case CHATZILLA_ID:
+		FireTray.isBrowser=true;
+        return 11;
+        break;
+
      default:
         return 0;
         break;
diff -Naur firetray-0.2.6/chrome/content/ircOverlay.xul firetray/chrome/content/ircOverlay.xul
--- firetray-0.2.6/chrome/content/ircOverlay.xul	1970-01-01 00:00:00.000000000 +0000
+++ firetray/chrome/content/ircOverlay.xul	2010-06-26 20:35:15.000000000 +0100
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!DOCTYPE overlay SYSTEM "chrome://firetray/locale/browserOverlay.dtd" >
+<overlay id="firetrayBrowserOverlay" xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul">
+    <script type="application/x-javascript" src="core.js" />
+    <menupopup id="menu_ChatZillaPopup">
+        <menuitem label="&firetray.close_to_tray;" id="menu_hideTray" oncommand="FireTray.hideToTray()" />
+        <menuseparator />
+    </menupopup>
+    <menupopup id="menu_ViewPopup">
+        <menuitem label="&firetray.hide;" id="menu_hide" accesskey="&firetray.hide.accesskey;" oncommand="FireTray.hideWindow()" />
+        <menuseparator />
+    </menupopup>
+</overlay>
diff -Naur firetray-0.2.6/chrome.manifest firetray/chrome.manifest
--- firetray-0.2.6/chrome.manifest	2010-06-26 20:15:36.000000000 +0100
+++ firetray/chrome.manifest	2010-06-26 20:43:36.000000000 +0100
@@ -5,6 +5,7 @@
 overlay chrome://songbird/content/xul/mainScriptsOverlay.xul  chrome://firetray/content/songOverlay.xul
 overlay chrome://sunbird/content/calendar.xul			chrome://firetray/content/mailOverlay.xul	
 overlay chrome://navigator/content/navigator.xul chrome://firetray/content/navigatorOverlay.xul	
+overlay	chrome://chatzilla/content/chatzilla.xul chrome://firetray/content/ircOverlay.xul
 
 locale  firetray    en-US   jar:chrome/firetray.jar!/locale/en-US/
 locale  firetray    it-IT   jar:chrome/firetray.jar!/locale/it-IT/
diff -Naur firetray-0.2.6/components/nsTray.cpp firetray/components/nsTray.cpp
--- firetray-0.2.6/components/nsTray.cpp	2010-06-26 20:15:36.000000000 +0100
+++ firetray/components/nsTray.cpp	2010-06-26 20:42:12.000000000 +0100
@@ -578,6 +578,10 @@ NS_IMETHODIMP nsTray::SetDefaultXpmIcon(
 
  switch(app)
  {
+   case 11: //chatzilla
+           df_icon=(char*)"/usr/share/icons/hicolor/32x32/apps/chatzilla.xpm";
+           sp_icon=(char*)"/usr/share/icons/hicolor/32x32/apps/chatzilla.xpm";
+           break;
    case 10: //seamonkey  
            df_icon=(char*)"/usr/share/pixmaps/seamonkey-icon.png";
            sp_icon=(char*)"/usr/share/mozilla-firetray/newmail.xpm";
diff -Naur firetray-0.2.6/chrome/SConscript firetray/chrome/SConscript
--- firetray-0.2.6/chrome/SConscript	2010-06-26 20:15:36.000000000 +0100
+++ firetray/chrome/SConscript	2010-06-26 20:45:02.000000000 +0100
@@ -3,7 +3,7 @@ import os
 Import('zipbld')
 
 contents = []
-content = 'core.js browserOverlay.xul mailOverlay.xul songOverlay.xul navigatorOverlay.xul options.xul options.js icon.png'
+content = 'core.js browserOverlay.xul mailOverlay.xul songOverlay.xul navigatorOverlay.xul ircOverlay.xul options.xul options.js icon.png'
 for c in content.split(' '):
     contents.append('content' + os.sep + c)
 
