--- firetray-0.2.5/components/nsTray.cpp.orig	2010-05-13 15:33:18.000000000 +0100
+++ firetray-0.2.5/components/nsTray.cpp	2010-05-13 18:30:05.000000000 +0100
@@ -3,18 +3,6 @@
 
 #include "nsTray.h"
 
-//BUILT IN PIXMAPS 
-#include "pixmaps/firefox.xpm"
-#include "pixmaps/thunderbird.xpm"
-#include "pixmaps/dove.xpm"
-#include "pixmaps/weasel.xpm"
-#include "pixmaps/icecat.xpm"
-#include "pixmaps/newmail.xpm"
-#include "pixmaps/seamonkey.xpm"
-#include "pixmaps/songbird.xpm"
-#include "pixmaps/songbirdegg.xpm"
-#include "pixmaps/sunbird.xpm"
-
 #include "nsMemory.h"
 #include "nsIBaseWindow.h"
 #include <pango/pangoft2.h>
@@ -177,7 +165,7 @@ nsTray::nsTray() {
 
     systray_icon = gtk_status_icon_new();
 
-    icon = gdk_pixbuf_new_from_xpm_data((const char**)firefox_xpm);
+    icon = gdk_pixbuf_new_from_file("/usr/share/icons/hicolor/48x48/apps/firefox.png",NULL);
 
     /* Connect signals */
     g_signal_connect(G_OBJECT(this->systray_icon), "activate", G_CALLBACK(nsTray::activate), this);
@@ -569,8 +557,8 @@ NS_IMETHODIMP nsTray::SetDefaultXpmIcon(
  if(this->default_icon) { g_object_unref(this->default_icon); this->default_icon=NULL;}
  if(this->special_icon) { g_object_unref(this->special_icon); this->special_icon=NULL;}
 
- char **df_icon;
- char **sp_icon;
+ char *df_icon;
+ char *sp_icon;
 
   /* APPS
 
@@ -591,50 +579,26 @@ NS_IMETHODIMP nsTray::SetDefaultXpmIcon(
  switch(app)
  {
    case 10: //seamonkey  
-           df_icon=(char**)seamonkey_xpm;
-           sp_icon=(char**)newmail_xpm;
+           df_icon=(char*)"/usr/share/pixmaps/seamonkey-icon.png";
+           sp_icon=(char*)"/usr/share/mozilla-firetray/newmail.xpm";
            break;
    case 9: //sunbird
-           df_icon=(char**)sunbird_xpm;
-           sp_icon=(char**)sunbird_xpm;
-           break;
-   case 8: //songbird
-           df_icon=(char**)songbirdegg_xpm;
-           sp_icon=(char**)songbird_xpm;
-           break;
-   case 7: //icecat
-           df_icon=(char**)icecat_xpm;
-           sp_icon=(char**)newmail_xpm;
-           break;
-   case 6: //iceweasel
-           df_icon=(char**)weasel_xpm;
-           sp_icon=(char**)newmail_xpm;
-           break;
-   case 5: //swiftdove
-           df_icon=(char**)dove_xpm;
-           sp_icon=(char**)newmail_xpm;
-           break;
-   case 4: //swiftweasel
-           df_icon=(char**)weasel_xpm;
-           sp_icon=(char**)newmail_xpm;
-           break;
-   case 3: //swiftdove
-           df_icon=(char**)dove_xpm;
-           sp_icon=(char**)newmail_xpm;
+           df_icon=(char*)"/usr/share/icons/hicolor/128x128/apps/sunbird.png";
+           sp_icon=(char*)"/usr/share/icons/hicolor/128x128/apps/sunbird.png";
            break;
    case 2: //thunderbird
-           df_icon=(char**)thunderbird_xpm;
-           sp_icon=(char**)newmail_xpm;
+           df_icon=(char*)"/usr/share/icons/hicolor/48x48/apps/thunderbird.png";
+           sp_icon=(char*)"/usr/share/mozilla-firetray/newmail.xpm";
            break;
    case 1: //firefox
    default:
-           df_icon=(char**)firefox_xpm;
-           sp_icon=(char**)firefox_xpm;
+           df_icon=(char*)"/usr/share/icons/hicolor/48x48/apps/firefox.png";
+           sp_icon=(char*)"/usr/share/icons/hicolor/48x48/apps/firefox.png";
            break;
  }
   
- this->default_icon = gdk_pixbuf_new_from_xpm_data((const char**)df_icon);
- this->special_icon = gdk_pixbuf_new_from_xpm_data((const char**)sp_icon);
+ this->default_icon = gdk_pixbuf_new_from_file(df_icon,NULL);
+ this->special_icon = gdk_pixbuf_new_from_file(sp_icon,NULL);
 
 /* gtk_status_icon_set_from_pixbuf(GTK_STATUS_ICON(this->systray_icon), GDK_PIXBUF(this->default_icon));
 
