--- a/drivers/acpi/acpica/Makefile.orig	2011-03-15 03:20:32.000000000 +0200
+++ a/drivers/acpi/acpica/Makefile	2011-03-20 02:38:15.340314871 +0200
@@ -14,7 +14,7 @@ acpi-y := dsfield.o   dsmthdat.o  dsopco
 
 acpi-y += evevent.o  evregion.o  evsci.o    evxfevnt.o \
 	 evmisc.o   evrgnini.o  evxface.o  evxfregn.o \
-	 evgpe.o    evgpeblk.o evgpeinit.o  evgpeutil.o evxfgpe.o evglock.o
+	 evgpe.o    evgpeblk.o evgpeinit.o  evgpeutil.o evxfgpe.o evglock.o proc.o
 
 acpi-y += exconfig.o  exfield.o  exnames.o   exoparg6.o  exresolv.o  exstorob.o\
 	 exconvrt.o  exfldio.o  exoparg1.o  exprep.o    exresop.o   exsystem.o\
