diff -up linux-2.6.32.noarch/drivers/gpu/drm/i915/intel_tv.c.jx linux-2.6.32.noarch/drivers/gpu/drm/i915/intel_tv.c
--- linux-2.6.32.noarch/drivers/gpu/drm/i915/intel_tv.c.jx	2010-01-14 16:20:20.000000000 -0500
+++ linux-2.6.32.noarch/drivers/gpu/drm/i915/intel_tv.c	2010-01-14 16:25:46.064428657 -0500
@@ -1841,7 +1841,6 @@ intel_tv_init(struct drm_device *dev)
 				   dev->mode_config.tv_bottom_margin_property,
 				   tv_priv->margin[TV_MARGIN_BOTTOM]);
 
-	dev_priv->hotplug_supported_mask |= TV_HOTPLUG_INT_STATUS;
 out:
 	drm_sysfs_connector_add(connector);
 }
