diff -up linux-2.6.33.noarch/drivers/gpu/drm/i915/intel_dp.c.dave linux-2.6.33.noarch/drivers/gpu/drm/i915/intel_dp.c
--- linux-2.6.33.noarch/drivers/gpu/drm/i915/intel_dp.c.dave	2010-06-25 16:30:13.000000000 +1000
+++ linux-2.6.33.noarch/drivers/gpu/drm/i915/intel_dp.c	2010-06-25 16:30:23.000000000 +1000
@@ -129,7 +129,7 @@ intel_dp_link_required(struct drm_device
 	struct drm_i915_private *dev_priv = dev->dev_private;
 
 	if (IS_eDP(intel_encoder))
-		return (pixel_clock * dev_priv->edp_bpp) / 8;
+		return (pixel_clock * ALIGN(dev_priv->edp_bpp, 8)) / 8;
 	else
 		return pixel_clock * 3;
 }
