diff --git a/drivers/block/floppy.c b/drivers/block/floppy.c
index 9955a53..aef66d1 100644
--- a/drivers/block/floppy.c
+++ b/drivers/block/floppy.c
@@ -1038,7 +1038,7 @@ static void floppy_disable_hlt(void)
 {
 	unsigned long flags;
 
-	WARN_ONCE(1, "floppy_disable_hlt() scheduled for removal in 2012");
+	printk(KERN_INFO "floppy_disable_hlt() scheduled for removal in 2012");
 	spin_lock_irqsave(&floppy_hlt_lock, flags);
 	if (!hlt_disabled) {
 		hlt_disabled = 1;
