From 81a06eed2491273b7d6d274ae4be1d333c100ab0 Mon Sep 17 00:00:00 2001
From: Mark Langsdorf <mark.langsdorf@calxeda.com>
Date: Mon, 12 Mar 2012 06:28:19 -0400
Subject: [PATCH] highbank: export clock functions

Signed-off-by: Mark Langsdorf <mark.langsdorf@calxeda.com>
---
 arch/arm/mach-highbank/clock.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git diff -up linux-3.2-rc4.orig/arch/arm/mach-highbank/clock.c diff -up linux-3.2-rc4/arch/arm/mach-highbank/clock.c
index c25a2ae..cdbc575 100644
--- a/arch/arm/mach-highbank/clock.c
+++ b/arch/arm/mach-highbank/clock.c
@@ -27,14 +27,17 @@ int clk_enable(struct clk *clk)
 {
 	return 0;
 }
+EXPORT_SYMBOL_GPL(clk_enable);
 
 void clk_disable(struct clk *clk)
 {}
+EXPORT_SYMBOL_GPL(clk_disable);
 
 unsigned long clk_get_rate(struct clk *clk)
 {
 	return clk->rate;
 }
+EXPORT_SYMBOL_GPL(clk_get_rate);
 
 long clk_round_rate(struct clk *clk, unsigned long rate)
 {
@@ -45,6 +48,7 @@ int clk_set_rate(struct clk *clk, unsigned long rate)
 {
 	return 0;
 }
+EXPORT_SYMBOL_GPL(clk_set_rate);
 
 static struct clk eclk = { .rate = 200000000 };
 static struct clk pclk = { .rate = 150000000 };
-- 
1.7.9.1

