From fe97564a6c40c1dd1e760ea9543d63709907d577 Mon Sep 17 00:00:00 2001
From: Kyle McMartin <kyle@treachery.i.jkkm.org>
Date: Mon, 11 Jan 2010 08:31:34 -0500
Subject: linux-2.6.30-hush-rom-warning.patch

---
 drivers/pci/setup-res.c |   11 ++++++++---
 1 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/pci/setup-res.c b/drivers/pci/setup-res.c
index 7d678bb..5acac72 100644
--- a/drivers/pci/setup-res.c
+++ b/drivers/pci/setup-res.c
@@ -95,6 +95,7 @@ int pci_claim_resource(struct pci_dev *dev, int resource)
 	struct resource *res = &dev->resource[resource];
 	struct resource *root;
 	int err;
+	const char *dtype = resource < PCI_BRIDGE_RESOURCES ? "device" : "bridge";
 
 	root = pci_find_parent_resource(dev, res);
 	if (!root) {
@@ -104,9 +105,13 @@ int pci_claim_resource(struct pci_dev *dev, int resource)
 	}
 
 	err = request_resource(root, res);
-	if (err)
-		dev_err(&dev->dev,
-			"address space collision: %pR already in use\n", res);
+	if (err) 
+		if (resource == 6) /* KERN_INFO on ROM. */
+			dev_info(&dev->dev,
+				"address space collision: %pR already in use\n", res);
+		else
+			dev_err(&dev->dev,
+				"address space collision: %pR already in use\n", res);
 
 	return err;
 }
-- 
1.6.5.2

