diff -up linux-2.6.33.noarch/drivers/pci/pci.c.orig linux-2.6.33.noarch/drivers/pci/pci.c
--- linux-2.6.33.noarch/drivers/pci/pci.c.orig	2010-04-22 13:53:51.649358165 -0400
+++ linux-2.6.33.noarch/drivers/pci/pci.c	2010-04-22 13:56:20.843232822 -0400
@@ -624,7 +624,7 @@ static void __pci_start_power_transition
  */
 int __pci_complete_power_transition(struct pci_dev *dev, pci_power_t state)
 {
-	return state > PCI_D0 ?
+	return state >= PCI_D0 ?
 			pci_platform_power_transition(dev, state) : -EINVAL;
 }
 EXPORT_SYMBOL_GPL(__pci_complete_power_transition);
@@ -661,10 +661,6 @@ int pci_set_power_state(struct pci_dev *
 		 */
 		return 0;
 
-	/* Check if we're already there */
-	if (dev->current_state == state)
-		return 0;
-
 	__pci_start_power_transition(dev, state);
 
 	/* This device is quirked not to be put into D3, so
