diff --git a/arch/x86/kernel/microcode_intel.c b/arch/x86/kernel/microcode_intel.c
index ebd193e..c57024a 100644
--- a/arch/x86/kernel/microcode_intel.c
+++ b/arch/x86/kernel/microcode_intel.c
@@ -453,9 +453,18 @@ static void microcode_fini_cpu(int cpu)
 	uci->mc = NULL;
 }
 
+/* we don't ship the broken out files...
+ *  instead, we'll just fail here, and load it with microcode_ctl
+ */
+static enum ucode_state noop_request_microcode_fw(int cpu,
+	struct device *device)
+{
+	return UCODE_NFOUND;
+}
+
 static struct microcode_ops microcode_intel_ops = {
 	.request_microcode_user		  = request_microcode_user,
-	.request_microcode_fw             = request_microcode_fw,
+	.request_microcode_fw             = noop_request_microcode_fw,
 	.collect_cpu_info                 = collect_cpu_info,
 	.apply_microcode                  = apply_microcode,
 	.microcode_fini_cpu               = microcode_fini_cpu,
