From 9b7ec0e07ce85c377b76626cef552d27d59cc405 Mon Sep 17 00:00:00 2001
From: Kyle McMartin <kyle@dreadnought.i.jkkm.org>
Date: Tue, 22 Jun 2010 13:57:47 +0100
Subject: utrace-ptrace: removed defunct arch_ptrace_untrace call

commit faa4602e removed the unused BTS code which had added
this hook.

Signed-off-by: Kyle McMartin <kyle@redhat.com>
---
 kernel/ptrace-utrace.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/kernel/ptrace-utrace.c b/kernel/ptrace-utrace.c
index 86234ee..1a8ba5e 100644
--- a/kernel/ptrace-utrace.c
+++ b/kernel/ptrace-utrace.c
@@ -50,8 +50,6 @@ void __ptrace_unlink(struct task_struct *child)
 	child->ptrace = 0;
 	child->parent = child->real_parent;
 	list_del_init(&child->ptrace_entry);
-
-	arch_ptrace_untrace(child);
 }
 
 struct ptrace_context {
-- 
1.7.0.1

