Index: linux-2.6.32.noarch/drivers/video/via/viafbdev.c
===================================================================
--- linux-2.6.32.noarch.orig/drivers/video/via/viafbdev.c
+++ linux-2.6.32.noarch/drivers/video/via/viafbdev.c
@@ -2161,6 +2161,8 @@ static int __init viafb_setup(char *opti
 #endif
 
 static struct pci_device_id viafb_pci_table[] __devinitdata = {
+/* We don't want this driver to autoload in F11/F-12 */
+/*
 	{ PCI_DEVICE(PCI_VENDOR_ID_VIA, UNICHROME_CLE266_DID),
 	  .driver_data = UNICHROME_CLE266 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_VIA, UNICHROME_PM800_DID),
@@ -2183,6 +2185,7 @@ static struct pci_device_id viafb_pci_ta
 	  .driver_data = UNICHROME_VX800 },
 	{ PCI_DEVICE(PCI_VENDOR_ID_VIA, UNICHROME_VX855_DID),
 	  .driver_data = UNICHROME_VX855 },
+*/
 	{ }
 };
 MODULE_DEVICE_TABLE(pci, viafb_pci_table);
