diff -up xterm-256/minstall.sh.man-page_paths xterm-256/minstall.sh
--- xterm-256/minstall.sh.man-page_paths	2010-03-04 00:44:49.000000000 +0100
+++ xterm-256/minstall.sh	2010-03-09 12:11:26.000000000 +0100
@@ -109,6 +109,10 @@ WTMP_PATH=$WTMP_PATH/$WTMP_NAME
 X_MANSECT=`man X 2>&1 | tr '\012' '\020' | sed -e 's/^[^0123456789]*\([^) ][^) ]*\).*/\1/'`
 test -z "$X_MANSECT" && X_MANSECT=$suffix
 
+fontpath=/usr/share/fonts
+xorgcfgdir=/etc/X11
+xorgcfgfil=xorg.conf
+X_MANSECT=7
 sed	-e 's%__vendorversion__%"X Window System"%' \
 	-e s%__apploaddir__%$APPS_DIR% \
 	-e s%__mansuffix__%$MY_MANSECT%g \
@@ -117,6 +121,9 @@ sed	-e 's%__vendorversion__%"X Window Sy
 	-e s%/etc/utmp%$UTMP_PATH%g \
 	-e s%fIwtmp'\\%fI'$WTMP_NAME'\\%g' \
 	-e s%/etc/wtmp%$WTMP_PATH%g \
+ 	-e s%__fontpath__%$fontpath%g \
+ 	-e s%__xorgcfgdir__%$xorgcfgdir%g \
+ 	-e s%__xorgcfgfil__%$xorgcfgfil%g \
 	$OLD_FILE >$NEW_FILE
 
 echo "$MINSTALL $OLD_FILE $END_FILE"
diff -up xterm-256/xterm.man.man-page_paths xterm-256/xterm.man
--- xterm-256/xterm.man.man-page_paths	2010-03-03 23:30:35.000000000 +0100
+++ xterm-256/xterm.man	2010-03-09 12:11:26.000000000 +0100
@@ -1699,17 +1699,17 @@ Since X11R6, bitmap fonts have been scal
 The font server claims to provide the bold font that \fIxterm\fP requests,
 but the result is not always readable.
 XFree86 provides a feature which can be used to suppress the scaling.
-In the X server's configuration file (e.g., "/etc/X11/XFree86"), you
+In the X server's configuration file (e.g., __xorgcfgdir__/__xorgcfgfil__), you
 can add ":unscaled" to the end of the directory specification for the
 "misc" fonts, which comprise the fixed-pitch fonts that are used by \fIxterm\fP.
 For example
 .RS
-	FontPath	"/usr/lib/X11/fonts/misc/"
+	FontPath	"__fontpath__"
 .RE
 .IP
 would become
 .RS
-	FontPath	"/usr/lib/X11/fonts/misc/:unscaled"
+	FontPath	"__fontpath__/misc/:unscaled"
 .RE
 .IP
 Depending on your configuration, the font server may have its own configuration
