diff -up libgit2-0.18.0/src/bswap.h.non-x86 libgit2-0.18.0/src/bswap.h
--- libgit2-0.18.0/src/bswap.h.non-x86	2013-06-19 10:01:31.000000000 +0200
+++ libgit2-0.18.0/src/bswap.h	2013-06-19 10:01:50.000000000 +0200
@@ -76,6 +76,10 @@ GIT_INLINE(uint16_t) default_swab16(uint
 #define bswap32(x) _byteswap_ulong(x)
 #define bswap16(x) _byteswap_ushort(x)
 
+#else
+
+#include <arpa/inet.h>
+
 #endif
 
 #ifdef bswap32
