--- xine-ui-0.99.5/m4/_xine.m4~	2006-07-01 05:26:26.000000000 +0300
+++ xine-ui-0.99.5/m4/_xine.m4	2007-07-13 17:29:54.000000000 +0300
@@ -35,15 +35,7 @@ AC_DEFUN([AC_CHECK_LIRC],
            [AC_CHECK_HEADER(lirc/lirc_client.h, true, have_lirc=no)], have_lirc=no)
      if test "$have_lirc" = "yes"; then
 
-        if test x"$LIRC_PREFIX" != "x"; then
-           lirc_libprefix="$LIRC_PREFIX/lib"
-  	   LIRC_INCLUDE="-I$LIRC_PREFIX/include"
-        fi
-        for llirc in $lirc_libprefix /lib /usr/lib /usr/local/lib; do
-          AC_CHECK_FILE(["$llirc/liblirc_client.a"],
-             [LIRC_LIBS="$llirc/liblirc_client.a"]
-             AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblir_client) installed]),,)
-        done
+         AC_DEFINE([HAVE_LIRC],,[Define this if you have LIRC (liblirc_client) installed])
      else
          AC_MSG_RESULT([*** LIRC client support not available, LIRC support will be disabled ***]);
      fi
--- xine-ui-0.99.5/configure~	2007-05-01 01:51:18.000000000 +0300
+++ xine-ui-0.99.5/configure	2007-07-13 17:28:19.000000000 +0300
@@ -14414,39 +14414,11 @@ fi
 
      if test "$have_lirc" = "yes"; then
 
-        if test x"$LIRC_PREFIX" != "x"; then
-           lirc_libprefix="$LIRC_PREFIX/lib"
-  	   LIRC_INCLUDE="-I$LIRC_PREFIX/include"
-        fi
-        for llirc in $lirc_libprefix /lib /usr/lib /usr/local/lib; do
-          as_ac_File=`echo "ac_cv_file_"$llirc/liblirc_client.a"" | $as_tr_sh`
-echo "$as_me:$LINENO: checking for \"$llirc/liblirc_client.a\"" >&5
-echo $ECHO_N "checking for \"$llirc/liblirc_client.a\"... $ECHO_C" >&6
-if eval "test \"\${$as_ac_File+set}\" = set"; then
-  echo $ECHO_N "(cached) $ECHO_C" >&6
-else
-  test "$cross_compiling" = yes &&
-  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
-echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
-   { (exit 1); exit 1; }; }
-if test -r ""$llirc/liblirc_client.a""; then
-  eval "$as_ac_File=yes"
-else
-  eval "$as_ac_File=no"
-fi
-fi
-echo "$as_me:$LINENO: result: `eval echo '${'$as_ac_File'}'`" >&5
-echo "${ECHO_T}`eval echo '${'$as_ac_File'}'`" >&6
-if test `eval echo '${'$as_ac_File'}'` = yes; then
-  LIRC_LIBS="$llirc/liblirc_client.a"
-
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_LIRC
 _ACEOF
 
-fi
-
-        done
+        LIRC_LIBS=-llirc_client
      else
          echo "$as_me:$LINENO: result: *** LIRC client support not available, LIRC support will be disabled ***" >&5
 echo "${ECHO_T}*** LIRC client support not available, LIRC support will be disabled ***" >&6;
