From e98eafec87cd3e506dbf0af294c6aec52214162d Mon Sep 17 00:00:00 2001
From: Fabrice Bellet <fabrice@bellet.info>
Date: Sun, 22 Sep 2013 12:21:23 +0200
Subject: [PATCH 2/2] make ShivaVG and FGAdminUI static libraries

---
 utils/fgadmin/src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/fgadmin/src/CMakeLists.txt b/utils/fgadmin/src/CMakeLists.txt
index 42b95d7..793452e 100644
--- a/utils/fgadmin/src/CMakeLists.txt
+++ b/utils/fgadmin/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 fltk_wrap_ui(FGAdminUI fgadmin.fl)
-add_library(FGAdminUI ${FGAdminUI_FLTK_UI_SRCS})
+add_library(FGAdminUI STATIC ${FGAdminUI_FLTK_UI_SRCS})
 
 add_executable(fgadmin WIN32 fgadmin_funcs.cxx main.cxx untarka.c)
 add_dependencies(fgadmin FGAdminUI)
 
-- 
1.8.3.1

