Index: trunk/lib/test/lookup_test.c
===================================================================
--- trunk/lib/test/lookup_test.c	(revision 737)
+++ trunk/lib/test/lookup_test.c	(revision 738)
@@ -385,5 +385,5 @@
 	// This is only for preventing collisions in s2i tests.
 	// If collisions are found in future, change the number. 
-	srand(2);
+	srand(3);
 #ifdef WITH_ALPHA
 	test_alpha_table();
