--- libffmpegthumbnailer/moviedecoder.cpp.orig	2011-05-18 23:19:55.392968832 -0300
+++ libffmpegthumbnailer/moviedecoder.cpp	2011-05-18 23:20:11.996418842 -0300
@@ -127,7 +127,7 @@ void MovieDecoder::initializeVideo()
 {
     for (unsigned int i = 0; i < m_pFormatContext->nb_streams; ++i)
     {
-        if (m_pFormatContext->streams[i]->codec->codec_type == CODEC_TYPE_VIDEO)
+        if (m_pFormatContext->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO)
         {
             m_pVideoStream = m_pFormatContext->streams[i];
             m_VideoStream = i;
