diff -up gutenprint-5.2.9/src/gutenprintui2/Makefile.am.yyin gutenprint-5.2.9/src/gutenprintui2/Makefile.am
--- gutenprint-5.2.9/src/gutenprintui2/Makefile.am.yyin	2010-09-12 14:52:53.000000000 +0100
+++ gutenprint-5.2.9/src/gutenprintui2/Makefile.am	2012-12-18 16:27:11.705127153 +0000
@@ -55,7 +55,8 @@ libgutenprintui2_la_CFLAGS = $(AM_CFLAGS
 # Uncomment to build a versioned library
 libgutenprintui2_la_LDFLAGS = \
  -version-info  $(GUTENPRINTUI2_CURRENT_INTERFACE):$(GUTENPRINTUI2_INTERFACE_AGE):$(GUTENPRINTUI2_BINARY_AGE) \
- -rpath $(libdir) -no-undefined
+ -rpath $(libdir) -no-undefined \
+ -export-symbols-regex '^([^y]|y[^y])'
 
 AM_LFLAGS = -i
 AM_YFLAGS = -d
diff -up gutenprint-5.2.9/src/gutenprintui2/Makefile.in.yyin gutenprint-5.2.9/src/gutenprintui2/Makefile.in
--- gutenprint-5.2.9/src/gutenprintui2/Makefile.in.yyin	2012-07-07 16:36:19.000000000 +0100
+++ gutenprint-5.2.9/src/gutenprintui2/Makefile.in	2012-12-18 16:27:11.705127153 +0000
@@ -367,7 +367,8 @@ libgutenprintui2_la_CFLAGS = $(AM_CFLAGS
 # Uncomment to build a versioned library
 libgutenprintui2_la_LDFLAGS = \
  -version-info  $(GUTENPRINTUI2_CURRENT_INTERFACE):$(GUTENPRINTUI2_INTERFACE_AGE):$(GUTENPRINTUI2_BINARY_AGE) \
- -rpath $(libdir) -no-undefined
+ -rpath $(libdir) -no-undefined \
+ -export-symbols-regex '^([^y]|y[^y])'
 
 AM_LFLAGS = -i
 AM_YFLAGS = -d
