diff -up hplip-3.12.9/Makefile.in.chgrp hplip-3.12.9/Makefile.in
--- hplip-3.12.9/Makefile.in.chgrp	2012-09-07 14:35:14.334691556 +0200
+++ hplip-3.12.9/Makefile.in	2012-09-07 14:36:02.974988844 +0200
@@ -8841,18 +8841,18 @@ uninstall-am: uninstall-binPROGRAMS unin
 
 install-dist_hplip_LogDATA:
 @FULL_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_Logdir)" || mkdir -p $(DESTDIR)$(hplip_Logdir)
-@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
+#@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
 @FULL_BUILD_TRUE@	chmod 774 $(DESTDIR)$(hplip_Logdir)
 @HPLIP_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_Logdir)" || mkdir -p $(DESTDIR)$(hplip_Logdir)
-@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
+#@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
 @HPLIP_BUILD_TRUE@	chmod 774 $(DESTDIR)$(hplip_Logdir)
 
 install-dist_hplip_tmpDATA:
 @FULL_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_tmpdir)" || mkdir -p $(DESTDIR)$(hplip_tmpdir)
-@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
+#@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
 @FULL_BUILD_TRUE@	chmod 1774 $(DESTDIR)$(hplip_tmpdir)
 @HPLIP_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_tmpdir)" || mkdir -p $(DESTDIR)$(hplip_tmpdir)
-@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
+#@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
 @HPLIP_BUILD_TRUE@	chmod 1774 $(DESTDIR)$(hplip_tmpdir)
 
 install-dist_cronDATA:$(dist_cron_DATA)
