diff -up hplip-3.10.2/prnt/ps/hp-laserjet_4250-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_4250-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_4250-ps.ppd.deviceIDs-ppd	2010-05-12 11:56:17.000000000 +0200
+++ hplip-3.10.2/prnt/ps/hp-laserjet_4250-ps.ppd	2010-05-12 11:58:30.000000000 +0200
@@ -53,6 +53,7 @@
 *HPPDEPanel: "HPImagingOptions"
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4250;DES:Hewlett-Packard LaserJet 4250;"
 *Product: "(hp LaserJet 4250)"
 *Product: "(Hewlett-Packard hp LaserJet 4250)"
 *Product: "(HP LaserJet 4250)"
diff -up hplip-3.10.2/prnt/ps/hp-color_laserjet_2605dn-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-color_laserjet_2605dn-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-color_laserjet_2605dn-ps.ppd.deviceIDs-ppd	2010-05-12 11:36:35.000000000 +0200
+++ hplip-3.10.2/prnt/ps/hp-color_laserjet_2605dn-ps.ppd	2010-05-12 11:39:06.000000000 +0200
@@ -50,6 +50,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 2605dn;DES:Hewlett-Packard Color LaserJet 2605dn;"
 *Product: "(HP Color LaserJet 2605)"
 *Product: "(Hewlett-Packard HP Color LaserJet 2605)"
 *Product: "(HP Color LaserJet 2605dn)"
diff -up hplip-3.10.2/prnt/ps/hp-color_laserjet_2840-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-color_laserjet_2840-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-color_laserjet_2840-ps.ppd.deviceIDs-ppd	2010-04-16 22:24:03.049147700 +0100
+++ hplip-3.10.2/prnt/ps/hp-color_laserjet_2840-ps.ppd	2010-04-16 22:25:34.077147648 +0100
@@ -46,6 +46,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID:  "MFG:Hewlett-Packard;CMD:PJL,PML,MLC,PCL,POSTSCRIPT,PCLXL;MDL:HP Color LaserJet 2840;DES:Hewlett-Packard Color LaserJet 2840;"
 *Product: "(HP Color LaserJet 2820)"
 *Product: "(Hewlett-Packard HP Color LaserJet 2820)"
 *Product: "(HP Color LaserJet 2830)"
diff -up hplip-3.10.2/prnt/ps/hp-color_laserjet_3800-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-color_laserjet_3800-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-color_laserjet_3800-ps.ppd.deviceIDs-ppd	2010-04-16 22:22:28.705273898 +0100
+++ hplip-3.10.2/prnt/ps/hp-color_laserjet_3800-ps.ppd	2010-04-16 22:23:28.914273477 +0100
@@ -51,6 +51,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID:  "MFG:Hewlett-Packard;MDL:HP Color LaserJet 3800;CMD:PJL,MLC,BIDI-ECP,PCLXL,PCL,PDF,PJL,POSTSCRIPT;"
 *Product: 	"(HP Color LaserJet 3800)"
 *Product: "(Hewlett-Packard HP Color LaserJet 3800)"
 *ModelName:	"HP Color LaserJet 3800"
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1200n-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1200n-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1200n-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:53.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1200n-ps.ppd	2010-04-16 22:21:49.286273004 +0100
@@ -56,6 +56,7 @@
 *Product: "(HP LaserJet 1200)"
 *Product: "(Hewlett-Packard HP LaserJet 1200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 1200;DES:Hewlett-Packard LaserJet 1200;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1200-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1200-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1200-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:53.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1200-ps.ppd	2010-04-16 22:21:49.292272590 +0100
@@ -56,6 +56,7 @@
 *Product: "(HP LaserJet 1200)"
 *Product: "(Hewlett-Packard HP LaserJet 1200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 1200;DES:Hewlett-Packard LaserJet 1200;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1320n-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1320n-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1320n-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1320n-ps.ppd	2010-04-16 22:21:49.300272653 +0100
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1320nw-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1320nw-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1320nw-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1320nw-ps.ppd	2010-04-16 22:21:49.304148148 +0100
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1320-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1320-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1320-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1320-ps.ppd	2010-04-16 22:21:49.317273026 +0100
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1320_series-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1320_series-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1320_series-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1320_series-ps.ppd	2010-04-16 22:21:49.321272466 +0100
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_1320tn-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_1320tn-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_1320tn-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_1320tn-ps.ppd	2010-04-16 22:21:49.334278100 +0100
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_2300-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_2300-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_2300-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_2300-ps.ppd	2010-04-16 22:21:49.342272544 +0100
@@ -61,6 +61,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2300L)"
 *Product: "(HP LaserJet 2300L)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2300 series;DES:Hewlett-Packard LaserJet 2300 series"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_2300_series-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_2300_series-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_2300_series-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:54.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_2300_series-ps.ppd	2010-04-16 22:21:49.346272524 +0100
@@ -61,6 +61,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2300L)"
 *Product: "(HP LaserJet 2300L)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2300 series;DES:Hewlett-Packard LaserJet 2300 series"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_p2015dn_series-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_p2015dn_series-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_p2015dn_series-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:55.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_p2015dn_series-ps.ppd	2010-04-16 22:21:49.356149235 +0100
@@ -64,6 +64,7 @@
 *Product: "(HP LaserJet P2015 Series)"
 *Product: "(Hewlett-Packard HP LaserJet P2015 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P2015 Series;DES:Hewlett-Packard LaserJet P2015 series;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_p2015n_series-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_p2015n_series-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_p2015n_series-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:55.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_p2015n_series-ps.ppd	2010-04-16 22:21:49.367148274 +0100
@@ -64,6 +64,7 @@
 *Product: "(HP LaserJet P2015 Series)"
 *Product: "(Hewlett-Packard HP LaserJet P2015 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P2015 Series;DES:Hewlett-Packard LaserJet P2015 series;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_p2015_series-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_p2015_series-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_p2015_series-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:55.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_p2015_series-ps.ppd	2010-04-16 22:21:49.376148194 +0100
@@ -64,6 +64,7 @@
 *Product: "(HP LaserJet P2015 Series)"
 *Product: "(Hewlett-Packard HP LaserJet P2015 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P2015 Series;DES:Hewlett-Packard LaserJet P2015 series;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.10.2/prnt/ps/hp-laserjet_p2015x_series-ps.ppd.deviceIDs-ppd hplip-3.10.2/prnt/ps/hp-laserjet_p2015x_series-ps.ppd
--- hplip-3.10.2/prnt/ps/hp-laserjet_p2015x_series-ps.ppd.deviceIDs-ppd	2010-02-24 23:09:55.000000000 +0000
+++ hplip-3.10.2/prnt/ps/hp-laserjet_p2015x_series-ps.ppd	2010-04-16 22:21:49.386147840 +0100
@@ -64,6 +64,7 @@
 *Product: "(HP LaserJet P2015 Series)"
 *Product: "(Hewlett-Packard HP LaserJet P2015 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P2015 Series;DES:Hewlett-Packard LaserJet P2015 series;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
