diff -up hplip-3.10.6/installer/core_install.py.network-timeout hplip-3.10.6/installer/core_install.py
--- hplip-3.10.6/installer/core_install.py.network-timeout	2010-07-19 04:17:37.000000000 +0200
+++ hplip-3.10.6/installer/core_install.py	2010-09-20 11:35:54.000000000 +0200
@@ -1546,7 +1546,7 @@ class CoreInstall(object):
         wget = utils.which("wget")
         if wget:
             wget = os.path.join(wget, "wget")
-            cmd = "%s --timeout=10 --output-document=- %s" % (wget, HTTP_GET_TARGET)
+            cmd = "%s --timeout=30 --output-document=- %s" % (wget, HTTP_GET_TARGET)
             log.debug(cmd)
             status, output = self.run(cmd)
             log.debug("wget returned: %d" % status)
@@ -1556,7 +1556,7 @@ class CoreInstall(object):
             curl = utils.which("curl")
             if curl:
                 curl = os.path.join(curl, "curl")
-                cmd = "%s --output - --connect-timeout 5 --max-time 10 %s" % (curl, HTTP_GET_TARGET)
+                cmd = "%s --output - --connect-timeout 15 --max-time 30 %s" % (curl, HTTP_GET_TARGET)
                 log.debug(cmd)
                 status, output = self.run(cmd)
                 log.debug("curl returned: %d" % status)
@@ -1567,7 +1567,7 @@ class CoreInstall(object):
 
                 if ping:
                     ping = os.path.join(ping, "ping")
-                    cmd = "%s -c1 -W1 -w10 %s" % (ping, PING_TARGET)
+                    cmd = "%s -c1 -W3 -w30 %s" % (ping, PING_TARGET)
                     log.debug(cmd)
                     status, output = self.run(cmd)
                     log.debug("ping returned: %d" % status)
