diff -up hplip-3.15.7/data/rules/56-hp_conf_usb.rules.udev-rules hplip-3.15.7/data/rules/56-hp_conf_usb.rules
--- hplip-3.15.7/data/rules/56-hp_conf_usb.rules.udev-rules	2015-07-29 16:55:43.945687431 +0200
+++ hplip-3.15.7/data/rules/56-hp_conf_usb.rules	2015-07-29 16:55:43.945687431 +0200
@@ -0,0 +1,4 @@
+# HPLIP udev rules file.
+
+# This rule will check the smart install feature, plugin status and firmware download for the required printers.
+ENV{hp_test}=="yes", PROGRAM="/bin/sh -c 'logger -p user.info loading HP Device $env{BUSNUM} $env{DEVNUM}'", RUN+="/bin/sh -c 'systemctl --no-block start hplip-printer@$env{BUSNUM}:$env{DEVNUM}.service'"
diff -up hplip-3.15.7/data/rules/56-hpmud.rules.udev-rules hplip-3.15.7/data/rules/56-hpmud.rules
--- hplip-3.15.7/data/rules/56-hpmud.rules.udev-rules	2015-07-13 09:48:49.000000000 +0200
+++ hplip-3.15.7/data/rules/56-hpmud.rules	2015-07-29 16:55:43.945687431 +0200
@@ -12,7 +12,4 @@ ENV{ID_USB_INTERFACES}=="", IMPORT{built
 # ENV{ID_HPLIP}="1" is for Ubuntu udev-acl
 ATTR{idVendor}=="03f0", ENV{ID_USB_INTERFACES}=="*:0701??:*|*:08????:", OWNER="root", GROUP="lp", MODE="0664", ENV{libsane_matched}="yes", ENV{hp_test}="yes", ENV{ID_HPLIP}="1"
 
-# This rule will check the smart install feature, plugin status and firmware download for the required printers.
-ENV{hp_test}=="yes", PROGRAM="/bin/sh -c 'logger -p user.info loading HP Device $env{BUSNUM} $env{DEVNUM}'", RUN+="/bin/sh -c '/usr/bin/nohup /usr/bin/hp-config_usb_printer $env{BUSNUM}:$env{DEVNUM}'"
-
 LABEL="hpmud_rules_end"
diff -up hplip-3.15.7/Makefile.in.udev-rules hplip-3.15.7/Makefile.in
--- hplip-3.15.7/Makefile.in.udev-rules	2015-07-13 09:50:17.000000000 +0200
+++ hplip-3.15.7/Makefile.in	2015-07-29 16:56:45.707539958 +0200
@@ -2463,6 +2463,7 @@ am__dist_ppd_DATA_DIST = prnt/ps/hp-lase
 am__dist_prnt_DATA_DIST = prnt/cups.py prnt/__init__.py prnt/ldl.py \
 	prnt/pcl.py prnt/colorcal.py
 am__dist_rules_DATA_DIST = data/rules/56-hpmud.rules \
+	data/rules/56-hp_conf_usb.rules \
 	data/rules/56-hpmud_sysfs.rules
 am__dist_rulessystem_DATA_DIST = data/rules/hplip-printer@.service
 am__dist_scan_DATA_DIST = scan/__init__.py scan/sane.py
@@ -4607,8 +4608,8 @@ cups_drv = prnt/drv/hpcups.drv
 @HPLIP_BUILD_TRUE@dist_rulessystem_DATA = data/rules/hplip-printer@.service
 
 # hpmud.rules
-@HPLIP_BUILD_TRUE@rulesdir = /etc/udev/rules.d
-@HPLIP_BUILD_TRUE@@UDEV_SYSFS_RULES_FALSE@dist_rules_DATA = data/rules/56-hpmud.rules
+@HPLIP_BUILD_TRUE@rulesdir = /usr/lib/udev/rules.d
+@HPLIP_BUILD_TRUE@@UDEV_SYSFS_RULES_FALSE@dist_rules_DATA = data/rules/56-hpmud.rules data/rules/56-hp_conf_usb.rules
 @HPLIP_BUILD_TRUE@@UDEV_SYSFS_RULES_TRUE@dist_rules_DATA = data/rules/56-hpmud_sysfs.rules
 @HPLIP_BUILD_TRUE@halpredir = /usr/share/hal/fdi/preprobe/10osvendor
 @HPLIP_BUILD_TRUE@dist_halpre_DATA = data/rules/20-hplip-devices.fdi
