diff -up hplip-3.13.11/scan.py.scan-tmp hplip-3.13.11/scan.py
--- hplip-3.13.11/scan.py.scan-tmp	2014-04-04 17:17:50.529576108 +0100
+++ hplip-3.13.11/scan.py	2014-04-04 17:18:26.807727330 +0100
@@ -1072,7 +1072,7 @@ try:
         if ('editor' in dest or 'viewer' in dest or 'email' in dest or 'print' in dest) \
             and not file_saved:
 
-            output_fd, output = utils.make_temp_file(suffix='.png')
+            output_fd, output = utils.make_temp_file(dir='/var/tmp', suffix='.png')
             try:
                 im.save(output)
             except IOError, e:
