diff -up hplip-3.13.2/io/hpmud/musb.c.hplip-IEEE-1284-4 hplip-3.13.2/io/hpmud/musb.c
--- hplip-3.13.2/io/hpmud/musb.c.hplip-IEEE-1284-4	2013-02-13 17:38:44.000000000 +0100
+++ hplip-3.13.2/io/hpmud/musb.c	2013-02-14 12:44:50.594152799 +0100
@@ -713,7 +713,8 @@ static libusb_device *get_libusb_device(
 				{
 					if ((altptr->bInterfaceClass == LIBUSB_CLASS_PRINTER ) && /* Printer */
 							(altptr->bInterfaceSubClass == 1) &&
-							(altptr->bInterfaceProtocol == 1 || altptr->bInterfaceProtocol == 2)) /* Unidirectional or Bidirectional*/
+							(altptr->bInterfaceProtocol == 1 || altptr->bInterfaceProtocol == 2
+							|| altptr->bInterfaceProtocol == 3)) /* Unidirectional, Bidirectional or 1284.4 */
 					{
 						if (is_uri(dev, uri))
 						{
