diff -up hplip-3.14.4/fax/backend/hpfax.py.log hplip-3.14.4/fax/backend/hpfax.py
--- hplip-3.14.4/fax/backend/hpfax.py.log	2014-04-04 12:42:15.000000000 +0200
+++ hplip-3.14.4/fax/backend/hpfax.py	2014-05-22 11:39:06.917683733 +0200
@@ -52,7 +52,10 @@ home_dir = ''
 
 def bug(msg):
     syslog.syslog("hpfax[%d]: error: %s\n" % (pid, msg))
-    log.stderr("ERROR: %s\n" % msg)
+    try:
+        log.stderr("ERROR: %s" % msg)
+    except NameError:
+        sys.stderr.write("ERROR: %s\n" % msg)
 
 
 if os.path.exists(config_file):
diff -up hplip-3.14.4/prnt/filters/hpps.log hplip-3.14.4/prnt/filters/hpps
--- hplip-3.14.4/prnt/filters/hpps.log	2014-04-04 12:39:47.000000000 +0200
+++ hplip-3.14.4/prnt/filters/hpps	2014-05-22 11:38:30.620187178 +0200
@@ -49,10 +49,16 @@ home_dir = ''
 
 
 def bug(m):
-    log.stderr("ERROR: %s" % m)
+    try:
+        log.stderr("ERROR: %s" % m)
+    except NameError:
+        sys.stderr.write("ERROR: %s\n" % m)
 
 def msg(m):
-    log.stderr("INFO: %s" % m)
+    try:
+        log.stderr("INFO: %s" % msg)
+    except NameError:
+        sys.stderr.write("INFO: %s\n" % msg)
 
 if os.path.exists(config_file):
     config = ConfigParser.ConfigParser()
