diff -up hplip-3.10.9/prnt/hpcups/HPCupsFilter.cpp.newline hplip-3.10.9/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.10.9/prnt/hpcups/HPCupsFilter.cpp.newline	2010-12-14 15:10:25.730407116 +0000
+++ hplip-3.10.9/prnt/hpcups/HPCupsFilter.cpp	2010-12-14 15:11:27.343440547 +0000
@@ -608,7 +608,7 @@ int HPCupsFilter::processRasterData(cups
             WriteBMPHeader (kfp, cups_header.cupsWidth, cups_header.cupsHeight, BLACK_RASTER);
         }
 
-        fprintf(stderr, "PAGE: %d %s", current_page_number, m_argv[4]);
+        fprintf(stderr, "PAGE: %d %s\n", current_page_number, m_argv[4]);
         // Iterating through the raster per page
         for (int y = 0; y < (int) cups_header.cupsHeight; y++) {
             cupsRasterReadPixels (cups_raster, m_pPrinterBuffer, cups_header.cupsBytesPerLine);
