diff -up hplip-3.12.9/prnt/cupsext/cupsext.c.addprinter hplip-3.12.9/prnt/cupsext/cupsext.c
--- hplip-3.12.9/prnt/cupsext/cupsext.c.addprinter	2012-09-04 14:31:10.000000000 +0200
+++ hplip-3.12.9/prnt/cupsext/cupsext.c	2012-09-07 14:20:24.626193617 +0200
@@ -511,6 +511,7 @@ PyObject * addPrinter( PyObject * self,
         strcpy( info, name );
 
     /* Connect to the HTTP server */
+    cupsSetUser ("root");
     if ( ( http = httpConnectEncrypt( cupsServer(), ippPort(), cupsEncryption() ) ) == NULL )
     {
         r = 0;
