diff -up hplip-3.14.4/scan.py.scan-tmp hplip-3.14.4/scan.py
--- hplip-3.14.4/scan.py.scan-tmp	2014-04-04 11:42:20.000000000 +0100
+++ hplip-3.14.4/scan.py	2014-04-29 17:20:05.653622531 +0100
@@ -1072,7 +1072,7 @@ try:
         if ('editor' in dest or 'viewer' in dest or 'email' in dest or 'print' in dest) \
             and not file_saved:
 
-            output_fd, output = utils.make_temp_file(suffix='.png')
+            output_fd, output = utils.make_temp_file(dir='/var/tmp', suffix='.png')
             try:
                 im.save(output)
             except IOError, e:
diff -up hplip-3.14.4/scan/sane.py.scan-tmp hplip-3.14.4/scan/sane.py
--- hplip-3.14.4/scan/sane.py.scan-tmp	2014-04-04 11:41:13.000000000 +0100
+++ hplip-3.14.4/scan/sane.py	2014-04-29 17:21:34.663061497 +0100
@@ -424,7 +424,8 @@ class ScanThread(threading.Thread):
         self.dev = device
         self.update_queue = update_queue
         self.event_queue = event_queue
-        self.buffer_fd, self.buffer_path = utils.make_temp_file(prefix='hpscan')
+        self.buffer_fd, self.buffer_path = utils.make_temp_file(dir='/var/tmp',
+                                                                prefix='hpscan')
         self.buffer = os.fdopen(self.buffer_fd, "w+b")
         self.format = -1
         self.format_name = ''
