diff -up hplip-3.12.9/ui4/systemtray.py.notification-exception hplip-3.12.9/ui4/systemtray.py
--- hplip-3.12.9/ui4/systemtray.py.notification-exception	2012-09-24 14:11:44.764151285 +0200
+++ hplip-3.12.9/ui4/systemtray.py	2012-09-24 14:11:45.574138763 +0200
@@ -761,7 +761,10 @@ class SystemTrayApp(QApplication):
                                     else:
                                         n.set_timeout(TRAY_MESSAGE_DELAY)
 
-                                    n.show()
+                                    try:
+                                        n.show()
+                                    except:
+                                        log.error("Failed to show notification!")
 
                                 else: # Use "standard" message bubbles
                                     icon = ERROR_STATE_TO_ICON.get(error_state, QSystemTrayIcon.Information)
