diff -up hplip-3.12.11/ui4/systemtray.py.systray-dbus-exception hplip-3.12.11/ui4/systemtray.py
--- hplip-3.12.11/ui4/systemtray.py.systray-dbus-exception	2012-11-20 10:51:40.000000000 +0100
+++ hplip-3.12.11/ui4/systemtray.py	2012-11-27 10:08:51.333805871 +0100
@@ -802,7 +802,13 @@ def run(read_pipe):
     log.set_module("hp-systray(qt4)")
     log.debug("PID=%d" % os.getpid())
 
-    app = SystemTrayApp(sys.argv, read_pipe)
+    try:
+        app = SystemTrayApp(sys.argv, read_pipe)
+    except DBusException, e:
+        # No session bus
+        log.debug("Caught exception: %s" % e)
+        sys.exit(1)
+
     app.setQuitOnLastWindowClosed(False) # If not set, settings dlg closes app
 
     i = 0
