diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_2500-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_2500-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_2500-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_2500-ps.ppd	2013-04-09 12:24:19.815228054 +0200
@@ -42,6 +42,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 2500;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 2500)"
 *Product: "(Hewlett-Packard hp color LaserJet 2500)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_2500_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_2500_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_2500_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_2500_series-ps.ppd	2013-04-09 12:24:19.816228040 +0200
@@ -42,6 +42,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 2500;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 2500)"
 *Product: "(Hewlett-Packard hp color LaserJet 2500)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_2550_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_2550_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_2550_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_2550_series-ps.ppd	2013-04-09 12:24:19.817228026 +0200
@@ -42,6 +42,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 2550 series;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 2550 series)"
 *Product: "(Hewlett-Packard hp color LaserJet 2550 series)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_2605dn-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_2605dn-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_2605dn-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_2605dn-ps.ppd	2013-04-09 12:24:19.818228012 +0200
@@ -46,6 +46,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 2605dn;DES:Hewlett-Packard Color LaserJet 2605dn;"
 *Product: "(HP Color LaserJet 2605)"
 *Product: "(Hewlett-Packard HP Color LaserJet 2605)"
 *Product: "(HP Color LaserJet 2605dn)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_2840-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_2840-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_2840-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_2840-ps.ppd	2013-04-09 12:24:19.819227998 +0200
@@ -43,6 +43,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID:  "MFG:Hewlett-Packard;CMD:PJL,PML,MLC,PCL,POSTSCRIPT,PCLXL;MDL:HP Color LaserJet 2840;DES:Hewlett-Packard Color LaserJet 2840;"
 *Product: "(HP Color LaserJet 2820)"
 *Product: "(Hewlett-Packard HP Color LaserJet 2820)"
 *Product: "(HP Color LaserJet 2830)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_3700n-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_3700n-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_3700n-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_3700n-ps.ppd	2013-04-09 12:24:19.820227984 +0200
@@ -41,6 +41,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 3700;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 3700)"
 *Product: "(Hewlett-Packard hp color LaserJet 3700)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_3700-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_3700-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_3700-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_3700-ps.ppd	2013-04-09 12:24:19.822227956 +0200
@@ -41,6 +41,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 3700;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 3700)"
 *Product: "(Hewlett-Packard hp color LaserJet 3700)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_3800-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_3800-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_3800-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_3800-ps.ppd	2013-04-09 12:24:19.823227942 +0200
@@ -47,6 +47,7 @@
 
 *% Product Name and Version Information
 *Manufacturer:	"HP"
+*1284DeviceID:  "MFG:Hewlett-Packard;MDL:HP Color LaserJet 3800;CMD:PJL,MLC,BIDI-ECP,PCLXL,PCL,PDF,PJL,POSTSCRIPT;"
 *Product: 	"(HP Color LaserJet 3800)"
 *Product: "(Hewlett-Packard HP Color LaserJet 3800)"
 *ModelName:	"HP Color LaserJet 3800"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_4550-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_4550-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_4550-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_4550-ps.ppd	2013-04-09 12:24:19.824227928 +0200
@@ -40,6 +40,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 4550 ;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet 4550 )"
 *Product: "(Hewlett-Packard HP Color LaserJet 4550 )"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_4600-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_4600-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_4600-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_4600-ps.ppd	2013-04-09 12:24:19.826227900 +0200
@@ -41,6 +41,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 4600;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 4600)"
 *Product: "(Hewlett-Packard hp color LaserJet 4600)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_4600_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_4600_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_4600_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_4600_series-ps.ppd	2013-04-09 12:24:19.827227886 +0200
@@ -41,6 +41,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 4600;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 4600)"
 *Product: "(Hewlett-Packard hp color LaserJet 4600)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_4650-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_4650-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_4650-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:24.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_4650-ps.ppd	2013-04-09 12:24:19.828227872 +0200
@@ -43,6 +43,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 4650;"
 *Manufacturer:	"HP"
 *Product: 	"(hp color LaserJet 4650)"
 *Product: "(Hewlett-Packard hp color LaserJet 4650)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_4700-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_4700-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_4700-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:25.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_4700-ps.ppd	2013-04-09 12:24:19.830227844 +0200
@@ -45,6 +45,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet 4700;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet 4700)"
 *Product: "(Hewlett-Packard HP Color LaserJet 4700)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_5550-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_5550-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_5550-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:25.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_5550-ps.ppd	2013-04-09 12:24:19.831227830 +0200
@@ -43,6 +43,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp color LaserJet 5550 ;"
 *Manufacturer:	"HP"
 *Product: "(hp color LaserJet 5550 )"
 *Product: "(Hewlett-Packard hp color LaserJet 5550 )"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:25.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_cm3530_mfp-ps.ppd	2013-04-09 12:24:19.832227816 +0200
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet CM3530 MFP;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet CM3530 MFP)"
 *Product: "(Hewlett-Packard HP Color LaserJet CM3530 MFP)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:25.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_cm4730_mfp-ps.ppd	2013-04-09 12:24:19.834227788 +0200
@@ -43,6 +43,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet CM4730 MFP;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet CM4730 MFP)"
 *Product: "(Hewlett-Packard HP Color LaserJet CM4730 MFP)"
diff -up hplip-3.13.4/prnt/ps/hp-color_laserjet_cp3525-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-color_laserjet_cp3525-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-color_laserjet_cp3525-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:25.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-color_laserjet_cp3525-ps.ppd	2013-04-09 12:24:19.836227760 +0200
@@ -44,6 +44,7 @@
 
 
 *% Product Name and Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Color LaserJet CP3525;"
 *Manufacturer:	"HP"
 *Product: 	"(HP Color LaserJet CP3525)"
 *Product: "(Hewlett-Packard HP Color LaserJet CP3525)"
diff -up hplip-3.13.4/prnt/ps/hp-designjet_t770_postscript-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-designjet_t770_postscript-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-designjet_t770_postscript-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-designjet_t770_postscript-ps.ppd	2013-04-09 12:24:19.836227760 +0200
@@ -38,6 +38,7 @@
 *ShortNickName: "HP Designjet T770 PostScript"
 *NickName: "HP Designjet T770ps 44in Postscript (recommended)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Designjet T770;DES:HP Designjet T770;CMD:POSTSCRIPT;"
 
 
 *% =============== Device Capabilities And Keywords for supporting Alert and Status Mechanism =============== 
diff -up hplip-3.13.4/prnt/ps/hp-designjet_t770ps_24in-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-designjet_t770ps_24in-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-designjet_t770ps_24in-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-designjet_t770ps_24in-ps.ppd	2013-04-09 12:24:19.837227746 +0200
@@ -38,6 +38,7 @@
 *ShortNickName: "HP Designjet T770ps 24in"
 *NickName: "HP Designjet T770ps 24in Postscript (recommended)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP Designjet T770;DES:HP Designjet T770;CMD:POSTSCRIPT;"
 
 
 *% =============== Device Capabilities And Keywords for supporting Alert and Status Mechanism =============== 
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1200n-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1200n-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1200n-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1200n-ps.ppd	2013-04-09 12:24:19.838227732 +0200
@@ -54,6 +54,7 @@
 *Product: "(HP LaserJet 1200)"
 *Product: "(Hewlett-Packard HP LaserJet 1200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 1200;DES:Hewlett-Packard LaserJet 1200;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1200-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1200-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1200-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1200-ps.ppd	2013-04-09 12:24:19.838227732 +0200
@@ -54,6 +54,7 @@
 *Product: "(HP LaserJet 1200)"
 *Product: "(Hewlett-Packard HP LaserJet 1200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 1200;DES:Hewlett-Packard LaserJet 1200;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1300n-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1300n-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1300n-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1300n-ps.ppd	2013-04-09 12:24:19.839227718 +0200
@@ -58,6 +58,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1300n)"
 *Product: "(HP LaserJet 1300n)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1300;DES:Hewlett-Packard LaserJet 1300;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1300-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1300-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1300-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1300-ps.ppd	2013-04-09 12:24:19.840227704 +0200
@@ -58,6 +58,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1300n)"
 *Product: "(HP LaserJet 1300n)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1300;DES:Hewlett-Packard LaserJet 1300;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1300xi-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1300xi-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1300xi-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1300xi-ps.ppd	2013-04-09 12:24:19.841227690 +0200
@@ -58,6 +58,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1300n)"
 *Product: "(HP LaserJet 1300n)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1300;DES:Hewlett-Packard LaserJet 1300;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1320n-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1320n-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1320n-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1320n-ps.ppd	2013-04-09 12:24:19.842227677 +0200
@@ -55,6 +55,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1320nw-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1320nw-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1320nw-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1320nw-ps.ppd	2013-04-09 12:24:19.843227663 +0200
@@ -55,6 +55,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1320-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1320-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1320-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1320-ps.ppd	2013-04-09 12:24:19.844227649 +0200
@@ -55,6 +55,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1320_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1320_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1320_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1320_series-ps.ppd	2013-04-09 12:24:19.845227635 +0200
@@ -55,6 +55,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_1320tn-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_1320tn-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_1320tn-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_1320tn-ps.ppd	2013-04-09 12:24:19.846227621 +0200
@@ -55,6 +55,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 1320 series)"
 *Product: "(HP LaserJet 1320 series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 1320 series;DES:Hewlett-Packard LaserJet 1320 series;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2100-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2100-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2100-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2100-ps.ppd	2013-04-09 12:24:19.847227607 +0200
@@ -54,6 +54,7 @@
 *Product: "(HP LaserJet 2100 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 2100 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 2100 Series;"
 *PSVersion: "(2014.111) 1"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2100_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2100_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2100_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2100_series-ps.ppd	2013-04-09 12:24:19.847227607 +0200
@@ -54,6 +54,7 @@
 *Product: "(HP LaserJet 2100 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 2100 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 2100 Series;"
 *PSVersion: "(2014.111) 1"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2200-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2200-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2200-ps.ppd.deviceIDs-ppd	2013-05-10 17:02:26.193850985 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2200-ps.ppd	2013-05-10 17:01:12.961873869 +0200
@@ -54,6 +54,7 @@
 *Product: "(HP LaserJet 2200)"
 *Product: "(Hewlett-Packard HP LaserJet 2200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 2200;DES:Hewlett-Packard LaserJet 2200;CMD:PJL,MLC,PCL,PCLXL,POSTSCRIPT;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2200_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2200_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2200_series-ps.ppd.deviceIDs-ppd	2013-05-10 17:02:23.190892928 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2200_series-ps.ppd	2013-05-10 17:01:02.231023762 +0200
@@ -54,6 +54,7 @@
 *Product: "(HP LaserJet 2200)"
 *Product: "(Hewlett-Packard HP LaserJet 2200)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 2200;DES:Hewlett-Packard LaserJet 2200;CMD:PJL,MLC,PCL,PCLXL,POSTSCRIPT;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2300-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2300-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2300-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2300-ps.ppd	2013-04-09 12:24:19.848227593 +0200
@@ -59,6 +59,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2300L)"
 *Product: "(HP LaserJet 2300L)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2300 series;DES:Hewlett-Packard LaserJet 2300 series"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2300_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2300_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2300_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2300_series-ps.ppd	2013-04-09 12:24:19.849227579 +0200
@@ -59,6 +59,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2300L)"
 *Product: "(HP LaserJet 2300L)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2300 series;DES:Hewlett-Packard LaserJet 2300 series"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_2420-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_2420-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_2420-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_2420-ps.ppd	2013-04-09 12:24:19.851227551 +0200
@@ -57,6 +57,7 @@
 *Product: "(Hewlett-Packard hp LaserJet 2420)"
 *Product: "(HP LaserJet 2420)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 2420;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_3390-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_3390-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_3390-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_3390-ps.ppd	2013-04-09 12:24:19.852227537 +0200
@@ -58,6 +58,7 @@
 *Product: "(HP LaserJet 3390)"
 *Product: "(Hewlett-Packard HP LaserJet 3390)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 3390;DES:Hewlett-Packard LaserJet 3390;"
 *PSVersion: "(2014.116) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4000_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4000_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4000_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4000_series-ps.ppd	2013-04-09 12:24:19.853227523 +0200
@@ -49,6 +49,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 4000 Series;DES:Hewlett-Packard LaserJet 4000 Series"
 *Product: "(HP LaserJet 4000 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 4000 Series)"
 *ModelName: "HP LaserJet 4000 Series"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4050_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4050_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4050_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4050_series-ps.ppd	2013-04-09 12:24:19.854227509 +0200
@@ -49,6 +49,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 4050 Series ;"
 *Product: "(HP LaserJet 4050 Series )"
 *Product: "(Hewlett-Packard HP LaserJet 4050 Series )"
 *Product: "(HP LaserJet 4050 Series)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4100_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4100_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4100_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4100_series-ps.ppd	2013-04-09 12:24:19.855227495 +0200
@@ -48,6 +48,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 4100 Series ;"
 *Product: "(HP LaserJet 4100 Series )"
 *Product: "(Hewlett-Packard HP LaserJet 4100 Series )"
 *Product: "(HP LaserJet 4100 Series)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4200-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4200-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4200-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4200-ps.ppd	2013-04-09 12:24:19.856227481 +0200
@@ -49,6 +49,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4200;"
 *Product: "(hp LaserJet 4200)"
 *Product: "(Hewlett-Packard hp LaserJet 4200)"
 *Product: "(HP LaserJet 4200)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4250-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4250-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4250-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4250-ps.ppd	2013-04-09 12:24:19.857227467 +0200
@@ -50,6 +50,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4250;DES:Hewlett-Packard LaserJet 4250;"
 *Product: "(hp LaserJet 4250)"
 *Product: "(Hewlett-Packard hp LaserJet 4250)"
 *Product: "(HP LaserJet 4250)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4300-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4300-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4300-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4300-ps.ppd	2013-04-09 12:24:19.859227439 +0200
@@ -49,6 +49,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4300;"
 *Product: "(hp LaserJet 4300)"
 *Product: "(Hewlett-Packard hp LaserJet 4300)"
 *Product: "(HP LaserJet 4300)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_4350-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_4350-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_4350-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:26.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_4350-ps.ppd	2013-04-09 12:24:19.860227425 +0200
@@ -50,6 +50,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:hp LaserJet 4350;"
 *Product: "(hp LaserJet 4350)"
 *Product: "(Hewlett-Packard hp LaserJet 4350)"
 *Product: "(HP LaserJet 4350)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_5100_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_5100_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_5100_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:27.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_5100_series-ps.ppd	2013-04-09 12:24:19.861227411 +0200
@@ -50,6 +50,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 5100 Series;"
 *Product: "(HP LaserJet 5100 Series)"
 *Product: "(Hewlett-Packard HP LaserJet 5100 Series)"
 *ModelName: "HP LaserJet 5100 Series"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_8000-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_8000-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_8000-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:27.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_8000-ps.ppd	2013-04-09 12:24:19.863227383 +0200
@@ -52,6 +52,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 8000 Series;"
 *ModelName: "HP LaserJet 8000 Series"
 *ShortNickName: "HP LaserJet 8000 Series"
 *NickName: "HP LaserJet 8000 Series Postscript (recommended)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_8000_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_8000_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_8000_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:27.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_8000_series-ps.ppd	2013-04-09 12:24:19.864227369 +0200
@@ -52,6 +52,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 8000 Series;"
 *ModelName: "HP LaserJet 8000 Series"
 *ShortNickName: "HP LaserJet 8000 Series"
 *NickName: "HP LaserJet 8000 Series Postscript (recommended)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:27.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_m3027_mfp-ps.ppd	2013-04-09 12:24:19.865227355 +0200
@@ -51,6 +51,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet M3027 MFP;"
 *ModelName: "HP LaserJet M3027 MFP"
 *ShortNickName: "HP LaserJet M3027 MFP"
 *NickName: "HP LaserJet M3027 MFP Postscript (recommended)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:27.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_m3035_mfp-ps.ppd	2013-04-09 12:24:19.866227341 +0200
@@ -52,6 +52,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet M3035 MFP;"
 *ModelName: "HP LaserJet M3035 MFP"
 *ShortNickName: "HP LaserJet M3035 MFP"
 *NickName: "HP LaserJet M3035 MFP Postscript (recommended)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_p2015_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_p2015_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_p2015_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:28.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_p2015_series-ps.ppd	2013-04-09 12:24:19.867227327 +0200
@@ -58,6 +58,7 @@
 *Product: "(HP LaserJet P2015 Series)"
 *Product: "(Hewlett-Packard HP LaserJet P2015 Series)"
 *Manufacturer: "HP"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P2015 Series;DES:Hewlett-Packard LaserJet P2015 series;"
 *PSVersion: "(3010.107) 0"
 
 *%=================================================
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_p3005-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_p3005-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_p3005-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:28.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_p3005-ps.ppd	2013-04-09 12:24:19.868227313 +0200
@@ -50,6 +50,7 @@
 *%=================================================
 *% 		 Product Version Information
 *%=================================================
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P3005;"
 *ModelName: "HP LaserJet P3005"
 *ShortNickName: "HP LaserJet P3005"
 *NickName: "HP LaserJet P3005 Postscript (recommended)"
diff -up hplip-3.13.4/prnt/ps/hp-laserjet_p3010_series-ps.ppd.deviceIDs-ppd hplip-3.13.4/prnt/ps/hp-laserjet_p3010_series-ps.ppd
--- hplip-3.13.4/prnt/ps/hp-laserjet_p3010_series-ps.ppd.deviceIDs-ppd	2013-04-08 11:59:28.000000000 +0200
+++ hplip-3.13.4/prnt/ps/hp-laserjet_p3010_series-ps.ppd	2013-04-09 12:24:19.870227285 +0200
@@ -45,6 +45,7 @@
 
 
 *% Product Version Information
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet P3010 Series;"
 *Product: "(HP LaserJet P3015)"
 *Product: "(Hewlett-Packard HP LaserJet P3015)"
 *Product: "(HP LaserJet P3011)"
