diff -up hplip-3.12.11/data/rules/56-hpmud_support.rules.udev-rules hplip-3.12.11/data/rules/56-hpmud_support.rules
--- hplip-3.12.11/data/rules/56-hpmud_support.rules.udev-rules	2012-11-20 10:53:51.000000000 +0100
+++ hplip-3.12.11/data/rules/56-hpmud_support.rules	2012-11-27 10:07:00.377519016 +0100
@@ -6,6 +6,7 @@ SUBSYSTEM!="usb_device", GOTO="hpmud_rul
 
 LABEL="pid_test"
 
+ENV{ID_USB_INTERFACES}=="", IMPORT{program}="usb_id --export %p"
 # Check for LaserJet products (0x03f0xx17).
 ATTRS{idVendor}=="03f0", ATTRS{idProduct}=="??17", ENV{hp_model}="$attrs{product}", ENV{hp_test}="yes"
 # Check for LaserJet products (0x03f0xx2a).
diff -up hplip-3.12.11/Makefile.in.udev-rules hplip-3.12.11/Makefile.in
--- hplip-3.12.11/Makefile.in.udev-rules	2012-11-20 10:55:50.000000000 +0100
+++ hplip-3.12.11/Makefile.in	2012-11-27 10:07:00.380518971 +0100
@@ -4353,7 +4353,7 @@ cups_drv = prnt/drv/hpcups.drv
 @HPLIP_BUILD_TRUE@dist_cron_DATA = data/rules/hplip_cron
 
 # hpmud.rules
-@HPLIP_BUILD_TRUE@rulesdir = /etc/udev/rules.d
+@HPLIP_BUILD_TRUE@rulesdir = /usr/lib/udev/rules.d
 @HPLIP_BUILD_TRUE@@UDEV_SYSFS_RULES_FALSE@dist_rules_DATA = data/rules/56-hpmud_support.rules \
 @HPLIP_BUILD_TRUE@@UDEV_SYSFS_RULES_FALSE@	data/rules/86-hpmud_plugin.rules \
 @HPLIP_BUILD_TRUE@@UDEV_SYSFS_RULES_FALSE@	data/rules/56-hpmud_add_printer.rules \
