diff -up hplip-3.13.2/Makefile.in.chgrp hplip-3.13.2/Makefile.in
--- hplip-3.13.2/Makefile.in.chgrp	2013-02-14 12:40:50.000000000 +0100
+++ hplip-3.13.2/Makefile.in	2013-02-14 12:43:46.844042178 +0100
@@ -8952,18 +8952,18 @@ uninstall-am: uninstall-binPROGRAMS unin
 
 install-dist_hplip_LogDATA:
 @FULL_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_Logdir)" || mkdir -p $(DESTDIR)$(hplip_Logdir)
-@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
+#@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
 @FULL_BUILD_TRUE@	chmod 775 $(DESTDIR)$(hplip_Logdir)
 @HPLIP_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_Logdir)" || mkdir -p $(DESTDIR)$(hplip_Logdir)
-@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
+#@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
 @HPLIP_BUILD_TRUE@	chmod 775 $(DESTDIR)$(hplip_Logdir)
 
 install-dist_hplip_tmpDATA:
 @FULL_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_tmpdir)" || mkdir -p $(DESTDIR)$(hplip_tmpdir)
-@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
+#@FULL_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
 @FULL_BUILD_TRUE@	chmod 1775 $(DESTDIR)$(hplip_tmpdir)
 @HPLIP_BUILD_TRUE@	test -z "$(DESTDIR)$(hplip_tmpdir)" || mkdir -p $(DESTDIR)$(hplip_tmpdir)
-@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
+#@HPLIP_BUILD_TRUE@	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
 @HPLIP_BUILD_TRUE@	chmod 1775 $(DESTDIR)$(hplip_tmpdir)
 
 install-dist_cronDATA:$(dist_cron_DATA)
