diff --git a/org.gnome.nm-applet.gschema.xml.in b/org.gnome.nm-applet.gschema.xml.in
index 71b6a29..1a41d0e 100644
--- network-manager-applet-0.9.5.96/org.gnome.nm-applet.gschema.xml.in
+++ network-manager-applet-0.9.5.96/org.gnome.nm-applet.gschema.xml.in
@@ -2,12 +2,12 @@
 <schemalist gettext-domain="nm-applet">
   <schema id="org.gnome.nm-applet" path="/org/gnome/nm-applet/">
     <key name="disable-connected-notifications" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Disable connected notifications</_summary>
       <_description>Set this to true to disable notifications when connecting to a network.</_description>
     </key>
     <key name="disable-disconnected-notifications" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Disable disconnected notifications</_summary>
       <_description>Set this to true to disable notifications when disconnecting from a network.</_description>
     </key>
@@ -17,7 +17,7 @@
       <_description>Set this to true to disable notifications when connecting to or disconnecting from a VPN.</_description>
     </key>
     <key name="suppress-wireless-networks-available" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Suppress networks available notifications</_summary>
       <_description>Set this to true to disable notifications when wireless networks are available.</_description>
     </key>
