--- network-manager-applet-0.9.7.0/org.gnome.nm-applet.gschema.xml.in.no-notifications	2012-08-14 08:59:42.000000000 -0400
+++ network-manager-applet-0.9.7.0/org.gnome.nm-applet.gschema.xml.in	2012-08-20 14:14:00.148731379 -0400
@@ -17,7 +17,7 @@
       <_description>Set this to true to disable notifications when connecting to or disconnecting from a VPN.</_description>
     </key>
     <key name="suppress-wireless-networks-available" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Suppress networks available notifications</_summary>
       <_description>Set this to true to disable notifications when Wi-Fi networks are available.</_description>
     </key>
