--- network-manager-applet-0.9.7.0/src/libnm-gtk/nm-wifi-dialog.c.applet-wifi-ui	2012-08-14 08:59:42.000000000 -0400
+++ network-manager-applet-0.9.7.0/src/libnm-gtk/nm-wifi-dialog.c	2012-08-20 14:17:10.059729528 -0400
@@ -1140,24 +1140,28 @@
 
 		tmp = g_strdup_printf (_("Passwords or encryption keys are required to access the Wi-Fi network '%s'."),
 		                       esc_ssid ? esc_ssid : "<unknown>");
-		gtk_window_set_title (GTK_WINDOW (self), _("Wi-Fi Network Authentication Required"));
 		label = g_strdup_printf ("<span size=\"larger\" weight=\"bold\">%s</span>\n\n%s",
 		                         _("Authentication required by Wi-Fi network"),
 		                         tmp);
 		g_free (esc_ssid);
 		g_free (tmp);
 	} else if (priv->adhoc_create) {
-		gtk_window_set_title (GTK_WINDOW (self), _("Create New Wi-Fi Network"));
 		label = g_strdup_printf ("<span size=\"larger\" weight=\"bold\">%s</span>\n\n%s",
 		                         _("New Wi-Fi network"),
 		                         _("Enter a name for the Wi-Fi network you wish to create."));
 	} else {
-		gtk_window_set_title (GTK_WINDOW (self), _("Connect to Hidden Wi-Fi Network"));
 		label = g_strdup_printf ("<span size=\"larger\" weight=\"bold\">%s</span>\n\n%s",
 		                         _("Hidden Wi-Fi network"),
 		                         _("Enter the name and security details of the hidden Wi-Fi network you wish to connect to."));
 	}
 
+	/* The primary text should always tell the story, not the title.  And since
+	 * the dialog is always on top, we can skip the taskbar.
+	 */
+	gtk_window_set_title (GTK_WINDOW (self), "");
+	gtk_window_set_keep_above (GTK_WINDOW (self), TRUE);
+	gtk_window_set_skip_taskbar_hint (GTK_WINDOW (self), TRUE);
+
 	widget = GTK_WIDGET (gtk_builder_get_object (priv->builder, "caption_label"));
 	gtk_label_set_markup (GTK_LABEL (widget), label);
 	g_free (label);
