--- pcsx2/gui/AppMain.cpp.orig	2012-11-02 21:57:37.645285973 -0200
+++ pcsx2/gui/AppMain.cpp	2012-11-02 21:59:37.005930262 -0200
@@ -365,7 +365,7 @@ public:
 				// variable exist but it is empty. So use the default value
 				user_local_dir = (wxDirName)Path::Combine( GetUserConfigDir() , wxDirName( L".config/pcsx2" ));
 			} else {
-				user_local_dir = (wxDirName)Path::Combine( xdg_home_value, pxGetAppName());
+				user_local_dir = (wxDirName)Path::Combine( xdg_home_value, wxDirName( L"pcsx2" ));
 			}
 		} else {
 			// variable do not exist
