diff -up WindowMaker-0.92.0/WPrefs.app/Makefile.am.dso WindowMaker-0.92.0/WPrefs.app/Makefile.am
--- WindowMaker-0.92.0/WPrefs.app/Makefile.am.dso	2010-12-08 22:28:57.000000000 +0100
+++ WindowMaker-0.92.0/WPrefs.app/Makefile.am	2010-12-08 22:30:34.000000000 +0100
@@ -54,6 +54,7 @@ WPrefs_LDADD = \
 	$(top_builddir)/WINGs/libWINGs.la\
 	$(top_builddir)/WINGs/libWUtil.la\
 	$(top_builddir)/wrlib/libwraster.la \
+        -lfontconfig \
 	@XFTLIBS@ \
 	@INTLIBS@
 
