Index: lazarus/Makefile.fpc
===================================================================
--- lazarus/Makefile.fpc	(version 0.9.26)
+++ lazarus/Makefile.fpc	(working copy)
@@ -21,7 +21,7 @@
 
 [install]
 fpcpackage=n
-basedir=share/lazarus
+basedir=lib/lazarus
 
 [dist]
 destdir=$(BASEDIR)/dist
@@ -29,7 +29,7 @@
 [prerules]
 RCPP?=$(strip $(firstword cpp$(SRCEXEEXT)))
 #
-LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/share/lazarus
+LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/$(_LIB)/lazarus
 INSTALL_MAN_DIR=$(INSTALL_PREFIX)/share/man
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 LAZARUS_INSTALL_DIR=C:\lazarus
@@ -115,8 +115,8 @@
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 
 else
-	$(MKDIR) $(INSTALL_PREFIX)/share
-	$(MKDIR) $(INSTALL_PREFIX)/share/lazarus
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)/lazarus
 	$(MKDIR) $(INSTALL_BINDIR)
 	$(MKDIR) $(INSTALL_MAN_DIR)
 	$(MKDIR) $(INSTALL_MAN_DIR)/man1
