diff -up libdmtx-0.7.2/wrapper/php/dmtx_write.c.php54 libdmtx-0.7.2/wrapper/php/dmtx_write.c
--- libdmtx-0.7.2/wrapper/php/dmtx_write.c.php54	2012-02-08 10:39:04.000000000 +0100
+++ libdmtx-0.7.2/wrapper/php/dmtx_write.c	2012-02-08 10:39:11.000000000 +0100
@@ -34,7 +34,7 @@ int le_dmtx_image;
 
 ZEND_DECLARE_MODULE_GLOBALS(dmtx)
 
-static function_entry dmtx_functions[] = {
+static zend_function_entry dmtx_functions[] = {
    PHP_FE(dmtx_write, NULL)
    PHP_FE(dmtx_getRow, NULL)
    PHP_FE(dmtx_getSize, NULL)
