From 8c6ec94d8919479704cc564dcf96d47831693d2b Mon Sep 17 00:00:00 2001
From: Gerhard Rieger <gerhard@dest-unreach.org>
Date: Sat, 7 Feb 2015 20:19:33 +0100
Subject: [PATCH] Corrected declarations of diag_immediate_msg etc.

---
 CHANGES | 6 ++++++
 error.h | 4 ++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/CHANGES b/CHANGES
index e6221c6..502e8bd 100644
--- a/CHANGES
+++ b/CHANGES
@@ -1,4 +1,10 @@
 
+corrections:
+	Type conflict between int and sig_atomic_t between declaration and
+	definition of diag_immediate_type and diag_immediate_exit broke
+	compilation on FreeBSD 10.1 with clang. Thanks to Emanuel Haupt for
+	reporting this bug.
+
 ####################### V 1.7.3.0:
 
 security:
diff --git a/error.h b/error.h
index 0c5f2c4..58c1cf9 100644
--- a/error.h
+++ b/error.h
@@ -229,8 +229,8 @@ struct diag_dgram {
 } ;
 
 extern sig_atomic_t diag_in_handler;
-extern int diag_immediate_msg;
-extern int diag_immediate_exit;
+extern sig_atomic_t diag_immediate_msg;
+extern sig_atomic_t diag_immediate_exit;
 
 extern void diag_set(char what, const char *arg);
 extern void diag_set_int(char what, int arg);
-- 
2.3.3.GIT

