diff -up yelp-2.29.5/src/Makefile.am.libs yelp-2.29.5/src/Makefile.am
--- yelp-2.29.5/src/Makefile.am.libs	2010-03-10 23:50:02.021505526 -0500
+++ yelp-2.29.5/src/Makefile.am	2010-03-10 23:50:16.258506223 -0500
@@ -93,7 +93,8 @@ yelp_LDADD =				\
 	$(X_LIBS)			\
 	$(MOZILLA_COMPONENT_LIBS)	\
 	$(MOZILLA_EXTRA_LIBS)		\
-	$(MOZILLA_GLUE_LIBS)
+	$(MOZILLA_GLUE_LIBS)		\
+	-ldl
 
 yelp_LDFLAGS = $(AM_LDFLAGS)
 
