--- astyle/src/astyle.h.orig	2014-11-20 22:57:25.695335483 +0100
+++ astyle/src/astyle.h	2014-11-20 22:58:20.073565069 +0100
@@ -382,6 +382,10 @@
 		bool getPreprocDefineIndent(void) const;
 		bool getSwitchIndent(void) const;
 
+                // set by Code::Blocks IDE
+		void setBlockIndent(bool state);
+		void setBracketIndent(bool state);
+
 	protected:
 		void deleteBeautifierVectors();
 		const string* findHeader(const string &line, int i,
@@ -390,8 +394,6 @@
 		                           const vector<const string*>* possibleOperators) const;
 		int  getNextProgramCharDistance(const string &line, int i) const;
 		int  indexOf(vector<const string*> &container, const string* element) const;
-		void setBlockIndent(bool state);
-		void setBracketIndent(bool state);
 		void setBracketIndentVtk(bool state);
 		string extractPreprocessorStatement(const string &line) const;
 		string trim(const string &str) const;
