From bf7c4573ab8b566667fc6757c2c5204068405e14 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Thu, 29 May 2014 12:18:34 +0200
Subject: [PATCH 1/2] fix supported format check for libe-book

---
 src/ebook2odt.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ebook2odt.cxx b/src/ebook2odt.cxx
index a005e38..e13e8e9 100644
--- a/src/ebook2odt.cxx
+++ b/src/ebook2odt.cxx
@@ -87,7 +87,7 @@ public:
 	}
 	bool isSupportedFormat(librevenge::RVNGInputStream &input)
 	{
-		return libebook::EBOOKDocument::isSupported(&input);
+		return libebook::EBOOKDocument::CONFIDENCE_EXCELLENT == libebook::EBOOKDocument::isSupported(&input);
 	}
 };
 
-- 
1.9.3

