From 3f340a3bc6cf81ee8f84a1e6e297b6000ae47944 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Tue, 19 May 2015 09:56:57 +0200
Subject: [PATCH] fix copypasta

---
 src/odf/numbers2ods.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/odf/numbers2ods.cxx b/src/odf/numbers2ods.cxx
index 678d31c..107d9ed 100644
--- a/src/odf/numbers2ods.cxx
+++ b/src/odf/numbers2ods.cxx
@@ -45,7 +45,7 @@ public:
 
 	bool convertDocument(librevenge::RVNGInputStream &input, bool isFlat)
 	{
-		OdtGenerator collector;
+		OdsGenerator collector;
 		StringDocumentHandler stylesHandler, contentHandler, manifestHandler, metaHandler;
 		if (isFlat)
 			collector.addDocumentHandler(&contentHandler, ODF_FLAT_XML);
-- 
2.4.0

