From 1586f8212e0f8d632dfacf59ecc1ab115a3945d7 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Thu, 29 May 2014 12:33:01 +0200
Subject: [PATCH 2/2] fix return value

---
 src/ebook2odt.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ebook2odt.cxx b/src/ebook2odt.cxx
index e13e8e9..6a7d0af 100644
--- a/src/ebook2odt.cxx
+++ b/src/ebook2odt.cxx
@@ -55,7 +55,7 @@ public:
 		}
 		try
 		{
-			if (!libebook::EBOOKDocument::parse(&input, &collector))
+			if (libebook::EBOOKDocument::RESULT_OK != libebook::EBOOKDocument::parse(&input, &collector))
 				return false;
 		}
 		catch (...)
-- 
1.9.3

