diff -up zsnes_1_51/src/linux/audio.c~ zsnes_1_51/src/linux/audio.c
--- zsnes_1_51/src/linux/audio.c~	2013-03-25 22:13:18.000000000 +0100
+++ zsnes_1_51/src/linux/audio.c	2013-03-25 23:46:56.831583012 +0100
@@ -162,10 +162,10 @@ static void *SoundThread_ao(void *useles
 
 static int SoundInit_ao()
 {
+  ao_sample_format driver_format = { 0, };
   int driver_id = ao_driver_id(libAoDriver);
   if (driver_id < 0) { driver_id = ao_default_driver_id(); }
 
-  ao_sample_format driver_format;
   driver_format.bits = 16;
   driver_format.channels = StereoSound+1;
   driver_format.rate = freqtab[SoundQuality = ((SoundQuality > 6) ? 1 : SoundQuality)];
