diff -up id3lib-3.8.3/Makefile.am.zlib id3lib-3.8.3/Makefile.am
--- id3lib-3.8.3/Makefile.am.zlib	2003-03-01 21:23:00.000000000 -0300
+++ id3lib-3.8.3/Makefile.am	2010-06-05 18:38:20.481848638 -0300
@@ -21,14 +21,9 @@ EXTRA_DIST =                    \
         reconf			\
 	makefile.win32		\
 	makewin32.bat
-if ID3_NEEDZLIB
-zlib_subdir = zlib
-else
-zlib_subdir = 
-endif
 
-SUBDIRS =  . m4 $(zlib_subdir) doc include id3com src examples
-DIST_SUBDIRS = . m4 zlib doc include id3com src examples prj libprj
+SUBDIRS =  . m4 doc include id3com src examples
+DIST_SUBDIRS = . m4 doc include id3com src examples prj libprj
 
 INCLUDES = @ID3LIB_DEBUG_FLAGS@
 
diff -up id3lib-3.8.3/configure.in.zlib id3lib-3.8.3/configure.in
--- id3lib-3.8.3/configure.in.zlib	2010-06-05 18:38:20.467847841 -0300
+++ id3lib-3.8.3/configure.in	2010-06-05 18:38:20.481848638 -0300
@@ -296,10 +296,6 @@ AC_DEFINE_UNQUOTED(_ID3LIB_BINARY_AGE,
 AC_DEFINE_UNQUOTED(_ID3_COMPILED_WITH_DEBUGGING, "${enable_debug}")
 
 CONDITIONAL_SUBDIRS=
-if test "x$ac_cv_lib_z_uncompress" = "xno"; then
-  CONDITIONAL_SUBDIRS="$CONDITIONAL_SUBDIRS zlib"
-fi
-AC_CONFIG_SUBDIRS(zlib)
 
 CFLAGS="$CFLAGS -Wall"
 
diff -up id3lib-3.8.3/examples/Makefile.am.zlib id3lib-3.8.3/examples/Makefile.am
--- id3lib-3.8.3/examples/Makefile.am.zlib	2003-03-01 21:23:00.000000000 -0300
+++ id3lib-3.8.3/examples/Makefile.am	2010-06-05 18:38:20.482848530 -0300
@@ -14,13 +14,8 @@ else
 ID3_DEBUG_LIBS =
 endif
 
-if ID3_NEEDZLIB
-zlib_lib = $(top_builddir)/zlib/src/libz.la
-zlib_include = -I$(top_srcdir)/zlib/include
-else
 zlib_lib = -lz
 zlib_include =
-endif
 
 if ID3_NEEDGETOPT_LONG
 getopt_lib = getopt.o getopt1.o
diff -up id3lib-3.8.3/src/Makefile.am.zlib id3lib-3.8.3/src/Makefile.am
--- id3lib-3.8.3/src/Makefile.am.zlib	2003-03-01 21:23:00.000000000 -0300
+++ id3lib-3.8.3/src/Makefile.am	2010-06-05 18:38:49.459724550 -0300
@@ -11,11 +11,7 @@
 
 LT_VERSION = $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
-if ID3_NEEDZLIB
-zlib_include = -I$(top_srcdir)/zlib/include
-else
 zlib_include = 
-endif
 
 INCLUDES = \
   @ID3LIB_DEBUG_FLAGS@ -I$(top_srcdir)/include/id3 -I$(top_srcdir)/include $(zlib_include)
@@ -72,11 +68,10 @@ lib_LTLIBRARIES   = libid3.la
 
 libid3_la_SOURCES = $(id3lib_sources)
 
-if ID3_NEEDZLIB
-LDADD        = $(top_builddir)/zlib/src/libz.la
-endif
+libid3_la_LIBADD        = -lz
 
 libid3_la_LDFLAGS = \
   -version-info $(LT_VERSION) \
+  -no-undefined \
   -release $(LT_RELEASE) \
   -export-dynamic
