diff -rupN meterbridge-0.9.2.old/src/jf_meters.c meterbridge-0.9.2/src/jf_meters.c
--- meterbridge-0.9.2.old/src/jf_meters.c	2003-06-05 07:23:07.000000000 -0400
+++ meterbridge-0.9.2/src/jf_meters.c	2009-05-06 01:21:14.000000000 -0400
@@ -12,7 +12,7 @@
 
 static float lp[MAX_METERS];
 static float tau_lp[MAX_METERS/2];
-static SDL_Rect buf_rect[MAX_METERS];
+SDL_Rect buf_rect[MAX_METERS];
 
 float tau(float *x, float *y, unsigned int n);
 
diff -rupN meterbridge-0.9.2.old/src/ppm_meters.c meterbridge-0.9.2/src/ppm_meters.c
--- meterbridge-0.9.2.old/src/ppm_meters.c	2003-06-05 05:40:06.000000000 -0400
+++ meterbridge-0.9.2/src/ppm_meters.c	2009-05-06 01:21:14.000000000 -0400
@@ -9,7 +9,7 @@
 #include "envelopes.h"
 
 static float lp[MAX_METERS];
-static SDL_Rect buf_rect[MAX_METERS];
+SDL_Rect buf_rect[MAX_METERS];
 
 void load_graphics_ppm()
 {
diff -rupN meterbridge-0.9.2.old/src/scope.c meterbridge-0.9.2/src/scope.c
--- meterbridge-0.9.2.old/src/scope.c	2003-06-05 05:21:38.000000000 -0400
+++ meterbridge-0.9.2/src/scope.c	2009-05-06 01:21:14.000000000 -0400
@@ -8,7 +8,7 @@
 #include "find_image.h"
 #include "envelopes.h"
 
-static SDL_Rect buf_rect[MAX_METERS];
+SDL_Rect buf_rect[MAX_METERS];
 
 void load_graphics_scope()
 {
diff -rupN meterbridge-0.9.2.old/src/vu_meters.c meterbridge-0.9.2/src/vu_meters.c
--- meterbridge-0.9.2.old/src/vu_meters.c	2003-06-05 07:08:52.000000000 -0400
+++ meterbridge-0.9.2/src/vu_meters.c	2009-05-06 01:21:14.000000000 -0400
@@ -8,7 +8,7 @@
 #include "find_image.h"
 
 static float lp[MAX_METERS];
-static SDL_Rect buf_rect[MAX_METERS];
+SDL_Rect buf_rect[MAX_METERS];
 
 void load_graphics_vu()
 {
