diff -up gtkmathview-0.8.0/gtkmathview-custom-reader.pc.in.jx gtkmathview-0.8.0/gtkmathview-custom-reader.pc.in
--- gtkmathview-0.8.0/gtkmathview-custom-reader.pc.in.jx	2007-10-15 03:12:23.000000000 -0400
+++ gtkmathview-0.8.0/gtkmathview-custom-reader.pc.in	2014-05-28 11:31:56.651369768 -0400
@@ -10,6 +10,7 @@ Name: GtkMathView
 Description: A GTK+ Widget for Rendering MathML Documents (custom reader frontend)
 Version: @VERSION@
 Requires: glib-2.0 gtk+-2.0 mathview-core mathview-backend-gtk mathview-frontend-custom-reader
-Libs: -L${libdir} -lgtkmathview_custom_reader @T1_LIBS@
+Libs: -L${libdir} -lgtkmathview_custom_reader
+Libs.private: @T1_LIBS@
 Cflags: -I${includedir}/@PACKAGE@ @GMV_ENABLE_BOXML_CFLAGS@ @GMV_HAVE_HASH_MAP_CFLAGS@ @GMV_HAVE_EXT_HASH_MAP_CFLAGS@

diff -up gtkmathview-0.8.0/gtkmathview-gmetadom.pc.in.jx gtkmathview-0.8.0/gtkmathview-gmetadom.pc.in
--- gtkmathview-0.8.0/gtkmathview-gmetadom.pc.in.jx	2007-10-15 03:12:20.000000000 -0400
+++ gtkmathview-0.8.0/gtkmathview-gmetadom.pc.in	2014-05-28 11:31:57.906373249 -0400
@@ -10,6 +10,7 @@ Name: GtkMathView
 Description: A GTK+ Widget for Rendering MathML Documents (GMetaDOM frontend)
 Version: @VERSION@
 Requires: glib-2.0 gtk+-2.0 gdome2-cpp-smart mathview-core mathview-backend-gtk mathview-frontend-gmetadom
-Libs: @DOM_LIBS@ -L${libdir} -lgtkmathview_gmetadom @T1_LIBS@
+Libs: @DOM_LIBS@ -L${libdir} -lgtkmathview_gmetadom
+Libs.private: @T1_LIBS@
 Cflags: @DOM_CFLAGS@ -I${includedir}/@PACKAGE@ @GMV_ENABLE_BOXML_CFLAGS@ @GMV_HAVE_HASH_MAP_CFLAGS@ @GMV_HAVE_EXT_HASH_MAP_CFLAGS@

diff -up gtkmathview-0.8.0/gtkmathview-libxml2.pc.in.jx gtkmathview-0.8.0/gtkmathview-libxml2.pc.in
--- gtkmathview-0.8.0/gtkmathview-libxml2.pc.in.jx	2007-10-15 03:12:29.000000000 -0400
+++ gtkmathview-0.8.0/gtkmathview-libxml2.pc.in	2014-05-28 11:31:57.908373235 -0400
@@ -10,6 +10,7 @@ Name: GtkMathView
 Description: A GTK+ Widget for Rendering MathML Documents (libxml2 frontend)
 Version: @VERSION@
 Requires: glib-2.0 gtk+-2.0 mathview-core mathview-backend-gtk mathview-frontend-libxml2
-Libs: @XML_LIBS@ -L${libdir} -lgtkmathview_libxml2 @T1_LIBS@
+Libs: @XML_LIBS@ -L${libdir} -lgtkmathview_libxml2
+Libs.private: @T1_LIBS@
 Cflags: @XML_CFLAGS@ -I${includedir}/@PACKAGE@ @GMV_ENABLE_BOXML_CFLAGS@ @GMV_HAVE_HASH_MAP_CFLAGS@ @GMV_HAVE_EXT_HASH_MAP_CFLAGS@

diff -up gtkmathview-0.8.0/gtkmathview-libxml2-reader.pc.in.jx gtkmathview-0.8.0/gtkmathview-libxml2-reader.pc.in
--- gtkmathview-0.8.0/gtkmathview-libxml2-reader.pc.in.jx	2007-10-15 03:12:26.000000000 -0400
+++ gtkmathview-0.8.0/gtkmathview-libxml2-reader.pc.in	2014-05-28 11:31:57.909373233 -0400
@@ -10,6 +10,7 @@ Name: GtkMathView
 Description: A GTK+ Widget for Rendering MathML Documents (libxml2 reader frontend)
 Version: @VERSION@
 Requires: glib-2.0 gtk+-2.0 mathview-core mathmlview-backend-gtk mathview-frontend-libxml2-reader
-Libs: @XML_LIBS@ -L${libdir} -lgtkmathview_libxml2_reader @T1_LIBS@
+Libs: @XML_LIBS@ -L${libdir} -lgtkmathview_libxml2_reader
+Libs.private: @T1_LIBS@
 Cflags: @XML_CFLAGS@ -I${includedir}/@PACKAGE@ @GMV_ENABLE_BOXML_CFLAGS@ @GMV_HAVE_HASH_MAP_CFLAGS@ @GMV_HAVE_EXT_HASH_MAP_CFLAGS@

diff -up gtkmathview-0.8.0/mathview-core.pc.in.jx gtkmathview-0.8.0/mathview-core.pc.in
--- gtkmathview-0.8.0/mathview-core.pc.in.jx	2007-10-15 03:12:17.000000000 -0400
+++ gtkmathview-0.8.0/mathview-core.pc.in	2014-05-28 11:31:57.912373215 -0400
@@ -10,6 +10,7 @@ Name: MathView
 Description: MathML rendering engine (core)
 Version: @VERSION@
 Requires: glib-2.0
-Libs: -L${libdir} -lmathview @T1_LIBS@
+Libs: -L${libdir} -lmathview
+Libs.private: @T1_LIBS@
 Cflags: -I${includedir}/@PACKAGE@ @GMV_ENABLE_BOXML_CFLAGS@ @GMV_HAVE_HASH_MAP_CFLAGS@ @GMV_HAVE_EXT_HASH_MAP_CFLAGS@
