From 7d938ac846ebc5947455e6e56260f72607319811 Mon Sep 17 00:00:00 2001
From: Luca Padovani <luca@src.gnome.org>
Date: Tue, 10 Jun 2008 16:17:42 +0000
Subject: + applied patch by Ed Catmur for marshalling functions in recent GTK versions

svn path=/trunk/; revision=1182
---
diff --git a/CONTRIBUTORS b/CONTRIBUTORS
index a15c16b..d51c0ca 100644
--- a/CONTRIBUTORS
+++ b/CONTRIBUTORS
@@ -38,3 +38,5 @@ In no particular order:
 * Nicola Rossi
   PostScript backend and accents
 
+* Ed Catmur
+  patch for marshalling functions in new gtk
diff --git a/configure.ac b/configure.ac
index 96ea726..85d26b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -332,6 +332,9 @@ PKG_CHECK_MODULES(GLIB, [glib-2.0 >= 2.2.1],,[AC_MSG_ERROR(could not find GLIB)]
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
+GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
+AC_SUBST(GLIB_GENMARSHAL)
+
 have_gtk="no"
 have_pango="no"
 if test "$enable_gtk" = "auto" -o "$enable_gtk" = "yes"; then
diff --git a/src/widget/Makefile.am b/src/widget/Makefile.am
index 4135f0c..8222d78 100644
--- a/src/widget/Makefile.am
+++ b/src/widget/Makefile.am
@@ -68,10 +68,14 @@ libgtkmathview_libxml2_reader_la_LDFLAGS = -version-info @MATHVIEW_VERSION_INFO@
 libgtkmathview_libxml2_la_LDFLAGS = -version-info @MATHVIEW_VERSION_INFO@
 libgtkmathview_gmetadom_la_LDFLAGS = -version-info @MATHVIEW_VERSION_INFO@
 
-libgtkmathview_custom_reader_la_SOURCES = gtkmathview_custom_reader.cc
-libgtkmathview_libxml2_reader_la_SOURCES = gtkmathview_libxml2_reader.cc
-libgtkmathview_libxml2_la_SOURCES = gtkmathview_libxml2.cc
-libgtkmathview_gmetadom_la_SOURCES = gtkmathview_gmetadom.cc
+libgtkmathview_custom_reader_la_SOURCES = gtkmathview_custom_reader.cc \
+  gtkmathview_marshal.c
+libgtkmathview_libxml2_reader_la_SOURCES = gtkmathview_libxml2_reader.cc \
+  gtkmathview_marshal.c
+libgtkmathview_libxml2_la_SOURCES = gtkmathview_libxml2.cc \
+  gtkmathview_marshal.c
+libgtkmathview_gmetadom_la_SOURCES = gtkmathview_gmetadom.cc \
+  gtkmathview_marshal.c
 
 libgtkmathview_custom_reader_la_CPPFLAGS = -DGMV_Widget_DLL
 libgtkmathview_libxml2_reader_la_CPPFLAGS = -DGMV_Widget_DLL
@@ -81,6 +85,7 @@ libgtkmathview_gmetadom_la_CPPFLAGS = -DGMV_Widget_DLL
 EXTRA_DIST = \
   gtkmathview_common.cc \
   gtkmathview_decorators_common.cc \
+  marshal.list \
   $(NULL)
 
 gtkdir = $(includedir)/$(PACKAGE)/gtk
@@ -94,6 +99,25 @@ gtk_HEADERS = \
   gtkmathview_libxml2.h \
   gtkmathview_gmetadom.h
 
+noinst_HEADERS = \
+  gtkmathview_marshal.h \
+  $(NULL)
+
+BUILT_SOURCES = \
+  gtkmathview_marshal.c \
+  gtkmathview_marshal.h \
+  $(NULL)
+
+CLEANFILES = \
+  $(BUILT_SOURCES) \
+  $(NULL)
+
+gtkmathview_marshal.h: marshal.list
+	$(GLIB_GENMARSHAL) --prefix=gtkmathview_marshal $< --header > $@
+
+gtkmathview_marshal.c: marshal.list
+	$(GLIB_GENMARSHAL) --prefix=gtkmathview_marshal $< --body > $@
+
 INCLUDES = \
   -I$(top_builddir)/auto \
   -I$(top_srcdir)/auto \
diff --git a/src/widget/gtkmathview_common.cc b/src/widget/gtkmathview_common.cc
index 9e1d935..d8d8495 100644
--- a/src/widget/gtkmathview_common.cc
+++ b/src/widget/gtkmathview_common.cc
@@ -36,6 +36,8 @@
 #include <gtk/gtkmain.h>
 #include <gtk/gtksignal.h>
 
+#include "gtkmathview_marshal.h"
+
 #include "gtkmathview_common.h"
 #if GTKMATHVIEW_USES_CUSTOM_READER
 #define GTK_MATH_VIEW_WIDGET_NAME "GtkMathView_Custom_Reader"
@@ -512,7 +514,7 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_LAST,
 		 G_STRUCT_OFFSET(GtkMathViewClass,set_scroll_adjustments),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER_POINTER,
+		 gtkmathview_marshal_VOID__POINTER_POINTER,
 		 G_TYPE_NONE , 2, GTK_TYPE_ADJUSTMENT, GTK_TYPE_ADJUSTMENT); 
 	
   click_signal = 
@@ -521,8 +523,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, click),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 
   select_begin_signal=
     g_signal_new("select_begin",
@@ -530,8 +532,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, select_begin),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 	
   select_over_signal = 
     g_signal_new("select_over",
@@ -539,8 +541,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, select_over),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 	
   select_end_signal = 
     g_signal_new("select_end",
@@ -548,8 +550,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, select_end),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 	
   select_abort_signal = 
     g_signal_new("select_abort",
@@ -557,7 +559,7 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, select_abort),
 		 NULL, NULL,
-		 gtk_marshal_NONE__NONE,
+		 g_cclosure_marshal_VOID__VOID,
 		 G_TYPE_NONE, 0);
 	
   element_over_signal = 
@@ -566,8 +568,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, element_over),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 
   decorate_under_signal =
     g_signal_new("decorate_under",
@@ -575,8 +577,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, decorate_under),
 		 NULL, NULL,
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 
   decorate_over_signal =
     g_signal_new("decorate_over",
@@ -584,8 +586,8 @@ gtk_math_view_class_init(GtkMathViewClass* math_view_class)
 		 G_SIGNAL_RUN_FIRST,
 		 G_STRUCT_OFFSET(GtkMathViewClass, decorate_over),
 		 NULL, NULL, 
-		 gtk_marshal_NONE__POINTER,
-		 G_TYPE_NONE, 1, GTK_TYPE_POINTER);
+		 g_cclosure_marshal_VOID__POINTER,
+		 G_TYPE_NONE, 1, G_TYPE_POINTER);
 
 #if 0	
   g_object_class_install_property(gobject_class,
diff --git a/src/widget/marshal.list b/src/widget/marshal.list
new file mode 100644
index 0000000..cd48853
--- a/dev/null
+++ b/src/widget/marshal.list
@@ -0,0 +1 @@
+VOID:POINTER,POINTER
