From 2102063ce75c41f8ba13370eb2ddf63a7d3ec86e Mon Sep 17 00:00:00 2001
From: Luca Padovani <padovani@di.unito.it>
Date: Fri, 14 May 2010 09:17:53 +0000
Subject: 	modified:   src/backend/common/ComputerModernShaper.cc

	applied patch by Andrew Stacey
---
diff --git a/src/backend/common/ComputerModernShaper.cc b/src/backend/common/ComputerModernShaper.cc
index e44f36f..0372422 100644
--- a/src/backend/common/ComputerModernShaper.cc
+++ b/src/backend/common/ComputerModernShaper.cc
@@ -975,7 +975,7 @@ static ComputerModernShaper::PlainChar msbmMap[] =
 static ComputerModernShaper::PlainChar32 msbmMap32[] =
   {
     { 0x1D538, 0x41 },  //double A
-    { 0x1D538, 0x42 },  //double B
+    { 0x1D539, 0x42 },  //double B
     { 0x1D53B, 0x44 },  //double D
     { 0x1D53C, 0x45 },  //double E
     { 0x1D53D, 0x46 },  //double F
