--- Makefile.orig	2008-12-09 08:07:06.000000000 -0200
+++ Makefile	2010-05-29 20:22:21.919802786 -0300
@@ -695,7 +695,7 @@
 		exit 1;\
 	fi;  
 	$(Q)if [ "$(HAVE_GTKSOURCEVIEW)" = "1" ]; then \
-		if [[ "$(TEST_SRCVIEW_VER)" < "2.4.1" ]]; then \
+		if shell pkg-config gtksourceview-2.0 --atleast-version=2.4.1 &>/dev/null ; then \
 		echo 'ERROR: gtksourceview package is too old ($(shell pkg-config --modversion gtksourceview-2.0))!';\
 		exit 1;\
 		fi;\
