diff -up e16-1.0.2/src/Makefile.in~ e16-1.0.2/src/Makefile.in
--- e16-1.0.2/src/Makefile.in~	2010-01-24 17:40:44.000000000 +0100
+++ e16-1.0.2/src/Makefile.in	2010-02-14 19:14:29.563238707 +0100
@@ -568,7 +568,7 @@ LDADD = \
 	$(X_EXTRA_LIBS)		\
 	$(E_X_LIBS)		\
 	$(DBUS_LIBS)		\
-	-lX11 -lm
+	-lX11 -lm -ldl
 
 @BUILD_MODULES_TRUE@e16_LDFLAGS = -export-dynamic
 @BUILD_MODULES_TRUE@libe16dir = $(pkglibdir)
