--- data/stellarium.desktop.orig	2013-01-31 09:44:06.000000000 -0200
+++ data/stellarium.desktop	2013-02-04 19:05:50.604217978 -0200
@@ -12,23 +12,3 @@ Categories=Astronomy;Education;Science;
 Comment=Planetarium
 Comment[fr]=Planétarium
 Comment[ru]=Планетарий
-
-Actions=Window;Fullscreen;Safemode;
-
-[Desktop Action Window]
-Name=Open in window
-Name[ru]=Открыть в окне
-Exec=stellarium -f no
-OnlyShowIn=Unity;
-
-[Desktop Action Fullscreen]
-Name=Open in fullscreen
-Name[ru]=Открыть в полноэкранном режиме
-Exec=stellarium -f yes
-OnlyShowIn=Unity;
-
-[Desktop Action Safemode]
-Name=Open in safe mode
-Name[ru]=Открыть в безопасном режиме
-Exec=stellarium --safe-mode
-OnlyShowIn=Unity;
