diff -ur ppl-0.10.1/Watchdog/doc/Makefile.in ppl-0.10.1-patched/Watchdog/doc/Makefile.in
--- ppl-0.10.1/Watchdog/doc/Makefile.in	2000-04-11 10:36:59.000000000 +0100
+++ ppl-0.10.1-patched/Watchdog/doc/Makefile.in	2009-04-11 17:53:10.000000000 +0100
@@ -163,7 +163,7 @@
 debug_flag = @debug_flag@
 
 # All the documentation in docdir.
-docdir = @docdir@
+docdir = @docdir@/pwl
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
