diff -Nur geeqie-1.0-orig/src/exif-common.c geeqie-1.0/src/exif-common.c
--- geeqie-1.0-orig/src/exif-common.c	2010-02-17 22:21:19.000000000 +0100
+++ geeqie-1.0/src/exif-common.c	2011-02-22 12:37:11.233079477 +0100
@@ -637,7 +637,7 @@
 
 void exif_free_fd(FileData *fd, ExifData *exif)
 {
-	if (!fd) return;
+	if (!fd || !exif_cache) return;
 	g_assert(fd->exif == exif);
 	
 	file_cache_put(exif_cache, fd, 1);
