diff -Nur geeqie-1.1-orig/src/secure_save.c geeqie-1.1/src/secure_save.c
--- geeqie-1.1-orig/src/secure_save.c	2012-08-12 22:13:41.000000000 +0200
+++ geeqie-1.1/src/secure_save.c	2013-02-20 11:47:41.408733878 +0100
@@ -9,12 +9,12 @@
  * This software comes with no warranty of any kind, use at your own risk!
  */
 
+#include "main.h"
 #include <glib/gprintf.h>
 #include <glib/gstdio.h>
 #include <errno.h>
 #include <utime.h>
 
-#include "main.h"
 #include "secure_save.h"
 
 
diff -Nur geeqie-1.1-orig/src/ui_fileops.h geeqie-1.1/src/ui_fileops.h
--- geeqie-1.1-orig/src/ui_fileops.h	2012-08-12 22:13:41.000000000 +0200
+++ geeqie-1.1/src/ui_fileops.h	2013-02-20 11:33:03.458028384 +0100
@@ -14,7 +14,7 @@
 #ifndef UI_FILEOPS_H
 #define UI_FILEOPS_H
 
-
+#include "config.h"
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/types.h>
