diff -Nur geeqie-1.0-orig/src/image-load.c geeqie-1.0/src/image-load.c
--- geeqie-1.0-orig/src/image-load.c	2010-02-17 22:21:19.000000000 +0100
+++ geeqie-1.0/src/image-load.c	2011-08-09 12:07:28.965109451 +0200
@@ -632,6 +632,7 @@
 			il->preview = TRUE;
 			DEBUG_1("Usable reduced size (preview) image loaded from file %s", il->fd->path);
 			}
+        if (exif)
 		exif_free_fd(il->fd, exif);
 		}
 
