diff -Nur geeqie-1.1-orig/src/filedata.c geeqie-1.1/src/filedata.c
--- geeqie-1.1-orig/src/filedata.c	2012-08-12 22:13:41.000000000 +0200
+++ geeqie-1.1/src/filedata.c	2013-02-01 18:44:12.921684510 +0100
@@ -2707,10 +2707,21 @@
 
 void file_data_send_notification(FileData *fd, NotifyType type)
 {
+	GList *work = notify_func_list;
+
+	while (work)
+		{
+		NotifyData *nd = (NotifyData *)work->data;
+
+		nd->func(fd, type, nd->data);
+		work = work->next;
+		}
+    /*
 	NotifyIdleData *nid = g_new0(NotifyIdleData, 1);
 	nid->fd = file_data_ref(fd);
 	nid->type = type;
 	g_idle_add_full(G_PRIORITY_HIGH, file_data_send_notification_idle_cb, nid, NULL);
+    */
 }
 
 static GHashTable *file_data_monitor_pool = NULL;
