diff -Nurb --strip-trailing-cr geeqie-1.0-orig/src/ui_utildlg.c geeqie-1.0/src/ui_utildlg.c
--- geeqie-1.0-orig/src/ui_utildlg.c	2010-02-17 22:21:19.000000000 +0100
+++ geeqie-1.0/src/ui_utildlg.c	2012-12-24 14:24:27.115586025 +0100
@@ -86,11 +86,12 @@
 static gboolean generic_dialog_key_press_cb(GtkWidget *widget, GdkEventKey *event, gpointer data)
 {
 	GenericDialog *gd = data;
+	gboolean auto_close = gd->auto_close;
 
 	if (event->keyval == GDK_Escape)
 		{
 		if (gd->cancel_cb) gd->cancel_cb(gd, gd->data);
-		if (gd->auto_close) generic_dialog_click_cb(widget, data);
+		if (auto_close) generic_dialog_click_cb(widget, data);
 		return TRUE;
 		}
 	return FALSE;
