diff -Nur geeqie-1.0-orig/src/Makefile.am geeqie-1.0/src/Makefile.am
--- geeqie-1.0-orig/src/Makefile.am	2010-02-17 22:21:19.000000000 +0100
+++ geeqie-1.0/src/Makefile.am	2011-11-06 00:56:32.090574656 +0100
@@ -1,6 +1,8 @@
 SUBDIRS = icons
 DIST_SUBDIRS = icons
 
+AM_LDFLAGS = -Wl,--as-needed
+
 AM_CFLAGS =				\
 	$(GLIB_CFLAGS) $(GTK_CFLAGS)	\
 	$(LCMS_CFLAGS)			\
diff -Nur geeqie-1.0-orig/src/Makefile.in geeqie-1.0/src/Makefile.in
--- geeqie-1.0-orig/src/Makefile.in	2010-02-17 22:21:45.000000000 +0100
+++ geeqie-1.0/src/Makefile.in	2011-11-06 00:56:33.147517364 +0100
@@ -196,7 +196,7 @@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 LCMS_CFLAGS = @LCMS_CFLAGS@
 LCMS_LIBS = @LCMS_LIBS@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -Wl,--as-needed
 LIBCHAMPLAIN_CFLAGS = @LIBCHAMPLAIN_CFLAGS@
 LIBCHAMPLAIN_GTK_CFLAGS = @LIBCHAMPLAIN_GTK_CFLAGS@
 LIBCHAMPLAIN_GTK_LIBS = @LIBCHAMPLAIN_GTK_LIBS@
