--- Scribus/scribus/ui/modetoolbar.h.orig	2012-08-06 19:23:10.575771593 -0400
+++ Scribus/scribus/ui/modetoolbar.h	2012-08-06 19:23:17.106075153 -0400
@@ -51,13 +51,13 @@
 
 	int SubMode;
 	int ValCount;
-	qreal *ShapeVals;
+	double *ShapeVals;
 	
 	virtual void changeEvent(QEvent *e);
 
 public slots:
 	void GetPolyProps();
-	void SelShape(int s, int c, qreal *vals);
+	void SelShape(int s, int c, double *vals);
 	void languageChange();
 		
 protected:
-void ModeToolBar::SelShape(int s, int c, qreal *vals)
+void ModeToolBar::SelShape(int s, int c, double *vals)
 {
 	m_ScMW->scrActions["toolsInsertShape"]->setIcon(QIcon(Rechteck->getIconPixmap(s,16)));
 //	insertShapeButtonMenu->hide();
--- Scribus/scribus/ui/modetoolbar.cpp.orig	2012-08-06 20:58:27.562827072 -0400
+++ Scribus/scribus/ui/modetoolbar.cpp	2012-08-06 20:59:13.344968068 -0400
@@ -40,7 +40,7 @@
 {
 	SubMode = 0;
 	ValCount = 32;
-	static qreal AutoShapes0[] = {0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 100.0, 0.0, 100.0, 0.0, 100.0, 0.0,
+	static double AutoShapes0[] = {0.0, 0.0, 0.0, 0.0, 100.0, 0.0, 100.0, 0.0, 100.0, 0.0, 100.0, 0.0,
 									100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 100.0, 0.0, 100.0, 0.0, 100.0,
 									0.0, 100.0, 0.0, 100.0, 0.0, 0.0, 0.0, 0.0};
 	ShapeVals = AutoShapes0;
@@ -95,7 +95,7 @@
 	delete dia;
 }
 
-void ModeToolBar::SelShape(int s, int c, qreal *vals)
+void ModeToolBar::SelShape(int s, int c, double *vals)
 {
 	m_ScMW->scrActions["toolsInsertShape"]->setIcon(QIcon(Rechteck->getIconPixmap(s,16)));
 //	insertShapeButtonMenu->hide();
