diff -up scribus-1.3.6/scribus/CMakeLists.txt.linker scribus-1.3.6/scribus/CMakeLists.txt
--- scribus-1.3.6/scribus/CMakeLists.txt.linker	2010-03-31 11:52:41.000000000 +0200
+++ scribus-1.3.6/scribus/CMakeLists.txt	2010-03-31 11:56:16.000000000 +0200
@@ -761,6 +761,11 @@ TARGET_LINK_LIBRARIES(${EXE_NAME}
   ${WIN32_ONLY_LIBS}
 )
 
+# required to resolve symbols from libdl when ld is called with --no-add-needed
+IF(UNIX)
+  TARGET_LINK_LIBRARIES(${EXE_NAME} dl)
+ENDIF(UNIX)
+
 IF(WIN32)
   TARGET_LINK_LIBRARIES(${EXE_NAME} ${QT_QTMAIN_LIBRARY})
 ENDIF(WIN32)
