From a39803f21ffec3c94a0151bf5c17180e3fb54e65 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Tue, 30 Nov 2010 11:53:22 +0100
Subject: [PATCH] fix rpath injection

---
 .../plugins/tools/2geomtools/CMakeLists.txt        |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/Scribus/scribus/plugins/tools/2geomtools/CMakeLists.txt b/Scribus/scribus/plugins/tools/2geomtools/CMakeLists.txt
index a283288..59bc9c3 100644
--- a/Scribus/scribus/plugins/tools/2geomtools/CMakeLists.txt
+++ b/Scribus/scribus/plugins/tools/2geomtools/CMakeLists.txt
@@ -22,8 +22,6 @@ IF(Boost_FOUND)
   MESSAGE(STATUS "Boost: FOUND, building 2geomtools")
   INCLUDE_DIRECTORIES( ${Boost_INCLUDE_DIR} )
   ADD_SUBDIRECTORY (lib2geom)
-  SET(GEOMLIB ${CMAKE_CURRENT_BINARY_DIR}/lib2geom/2geom)
-  LINK_DIRECTORIES(${PLUGIN_LIBRARIES} ${GEOMLIB})
   ADD_SUBDIRECTORY (meshdistortion)
   ADD_SUBDIRECTORY (pathalongpath)
 ELSE(Boost_FOUND)
-- 
1.7.3.2

