diff --git a/Scribus/scribus/cupsoptions.cpp b/Scribus/scribus/cupsoptions.cpp
index 134bead..34edc06 100644
--- a/Scribus/scribus/cupsoptions.cpp
+++ b/Scribus/scribus/cupsoptions.cpp
@@ -42,6 +42,7 @@ for which a new license (GPL+exception) is in place.
 #include "scconfig.h"
 #ifdef HAVE_CUPS
 #include <cups/cups.h>
+#include <cups/ppd.h>
 #endif
 #include "util_icon.h"
 
diff --git a/Scribus/scribus/util_printer.cpp b/Scribus/scribus/util_printer.cpp
index 8d08cba..1b173cd 100644
--- a/Scribus/scribus/util_printer.cpp
+++ b/Scribus/scribus/util_printer.cpp
@@ -9,6 +9,7 @@ for which a new license (GPL+exception) is in place.
 
 #if defined( HAVE_CUPS )
  #include <cups/cups.h>
+ #include <cups/ppd.h>
 #elif defined(_WIN32)
  #include <windows.h>
  #include <winspool.h>
