--- applets/maintained/notification-daemon/daemon.c.orig	2011-02-12 03:11:45.417794090 -0200
+++ applets/maintained/notification-daemon/daemon.c	2011-02-12 03:12:00.987220849 -0200
@@ -1487,7 +1487,7 @@ gboolean send_message(gchar *body)
   if (fork() == 0)
   {
     notify_init("notify-send");
-    notify = notify_notification_new(summary, body, icon_str, NULL);
+    notify = notify_notification_new(summary, body, icon_str);
     notify_notification_set_category(notify, type);
     notify_notification_set_urgency(notify, urgency);
     notify_notification_set_timeout(notify, expire_timeout);
