--- gnome-mime-data-2.2.0/gnome-vfs.keys.in.rpminstall	2003-02-04 15:01:42.000000000 -0500
+++ gnome-mime-data-2.2.0/gnome-vfs.keys.in	2003-02-04 15:02:25.000000000 -0500
@@ -824,9 +824,12 @@
 
 application/x-rpm
 	_description=RPM package
-	default_action_type=component
 	_category=Packages
 	use_category_default=yes
+	default_action_type=application
+	short_list_application_ids_for_novice_user_level=redhat-install-packages
+	short_list_application_ids_for_intermediate_user_level=redhat-install-packages
+	short_list_application_ids_for_advanced_user_level=redhat-install-packages
 
 application/x-shar
 	_description=Shell archive
