diff -up bind-9.8.1rc1/lib/isc/unix/socket.c.rh735103 bind-9.8.1rc1/lib/isc/unix/socket.c
--- bind-9.8.1rc1/lib/isc/unix/socket.c.rh735103	2011-07-29 04:19:20.000000000 +0200
+++ bind-9.8.1rc1/lib/isc/unix/socket.c	2011-09-07 18:49:54.100215897 +0200
@@ -57,6 +57,20 @@
 #include <isc/util.h>
 #include <isc/xml.h>
 
+/* See task.c about the following definition: */
+#ifdef BIND9
+#ifdef ISC_PLATFORM_USETHREADS
+#define USE_WATCHER_THREAD
+#else
+#define USE_SHARED_MANAGER
+#endif	/* ISC_PLATFORM_USETHREADS */
+#else /* BIND9 */
+#undef ISC_PLATFORM_HAVESYSUNH
+#undef ISC_PLATFORM_HAVEKQUEUE
+#undef ISC_PLATFORM_HAVEEPOLL
+#undef ISC_PLATFORM_HAVEDEVPOLL
+#endif	/* BIND9 */
+
 #ifdef ISC_PLATFORM_HAVESYSUNH
 #include <sys/un.h>
 #endif
@@ -76,15 +90,6 @@
 
 #include "errno2result.h"
 
-/* See task.c about the following definition: */
-#ifdef BIND9
-#ifdef ISC_PLATFORM_USETHREADS
-#define USE_WATCHER_THREAD
-#else
-#define USE_SHARED_MANAGER
-#endif	/* ISC_PLATFORM_USETHREADS */
-#endif	/* BIND9 */
-
 #ifndef USE_WATCHER_THREAD
 #include "socket_p.h"
 #include "../task_p.h"
