diff -up bind-9.7.0-P2/bin/dig/dig.docbook.rh811566 bind-9.7.0-P2/bin/dig/dig.docbook
--- bind-9.7.0-P2/bin/dig/dig.docbook.rh811566	2012-06-20 15:50:03.206839118 +0200
+++ bind-9.7.0-P2/bin/dig/dig.docbook	2012-06-20 15:50:28.368558830 +0200
@@ -912,8 +912,8 @@ dig +qr www.isc.org any -x 127.0.0.1 isc
       <command>dig</command> appropriately converts character encoding of
       domain name before sending a request to DNS server or displaying a
       reply from the server.
-      If you'd like to turn off the IDN support for some reason, defines
-      the <envar>IDN_DISABLE</envar> environment variable.
+      If you'd like to turn off the IDN support for some reason, define
+      the <envar>CHARSET=ASCII</envar> environment variable.
       The IDN support is disabled if the variable is set when 
       <command>dig</command> runs.
     </para>
