diff -up bind-9.7.3-P3/bin/nsupdate/nsupdate.c.rh700097 bind-9.7.3-P3/bin/nsupdate/nsupdate.c
--- bind-9.7.3-P3/bin/nsupdate/nsupdate.c.rh700097	2011-08-12 16:24:18.730205763 +0200
+++ bind-9.7.3-P3/bin/nsupdate/nsupdate.c	2011-08-12 16:24:27.833205103 +0200
@@ -2298,6 +2298,7 @@ recvsoa(isc_task_t *task, isc_event_t *e
 		dns_message_destroy(&soaquery);
 		ddebug("Out of recvsoa");
 		done_update();
+		seenerror = ISC_TRUE;
 		return;
 	}
 
