diff -up bind-9.9.2/bin/dig/dig.docbook.rh640538 bind-9.9.2/bin/dig/dig.docbook
--- bind-9.9.2/bin/dig/dig.docbook.rh640538	2012-09-27 02:35:19.000000000 +0200
+++ bind-9.9.2/bin/dig/dig.docbook	2012-11-12 14:47:17.385334972 +0100
@@ -961,6 +961,40 @@ dig +qr www.isc.org any -x 127.0.0.1 isc
   </refsect1>
 
   <refsect1>
+    <title>RETURN CODES</title>
+    <para>
+      <command>Dig</command> return codes are:
+      <variablelist>
+        <varlistentry>
+          <listitem>
+            <para>0: Everything went well, including things like NXDOMAIN</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <listitem>
+            <para>1: Usage error</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <listitem>
+            <para>8: Couldn't open batch file</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <listitem>
+            <para>9: No reply from server</para>
+          </listitem>
+        </varlistentry>
+        <varlistentry>
+          <listitem>
+            <para>10: Internal error</para>
+          </listitem>
+        </varlistentry>
+      </variablelist>
+    </para>
+  </refsect1>
+
+  <refsect1>
     <title>FILES</title>
     <para><filename>/etc/resolv.conf</filename>
     </para>
