diff -up bind-9.8.0rc1/lib/dns/include/dns/Makefile.in.includes bind-9.8.0rc1/lib/dns/include/dns/Makefile.in
--- bind-9.8.0rc1/lib/dns/include/dns/Makefile.in.includes	2011-02-22 12:09:51.502603050 +0100
+++ bind-9.8.0rc1/lib/dns/include/dns/Makefile.in	2011-02-22 12:10:07.692556291 +0100
@@ -29,7 +29,7 @@ HEADERS =	acl.h adb.h byaddr.h cache.h c
 		peer.h portlist.h private.h rbt.h rcode.h \
 		rdata.h rdataclass.h rdatalist.h rdataset.h rdatasetiter.h \
 		rdataslab.h rdatatype.h request.h resolver.h result.h \
-		rootns.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
+		rootns.h rpz.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
 		tcpmsg.h time.h tkey.h tsig.h ttl.h types.h \
 		validator.h version.h view.h xfrin.h zone.h zonekey.h zt.h
 
