diff -up bind-9.10.1b1/lib/dns/Makefile.in.parallel bind-9.10.1b1/lib/dns/Makefile.in
--- bind-9.10.1b1/lib/dns/Makefile.in.parallel	2014-07-29 15:48:34.052378742 +0200
+++ bind-9.10.1b1/lib/dns/Makefile.in	2014-07-29 15:49:34.542474686 +0200
@@ -17,10 +17,6 @@ srcdir =	@srcdir@
 VPATH =		@srcdir@
 top_srcdir =	@top_srcdir@
 
-# Attempt to disable parallel processing.
-.NOTPARALLEL:
-.NO_PARALLEL:
-
 @BIND9_VERSION@
 @BIND9_MAJOR@
 
