diff -up bind-9.9.3/lib/dns/include/dns/Makefile.in.update bind-9.9.3/lib/dns/include/dns/Makefile.in
--- bind-9.9.3/lib/dns/include/dns/Makefile.in.update	2013-06-03 09:29:41.049197873 +0200
+++ bind-9.9.3/lib/dns/include/dns/Makefile.in	2013-06-03 09:30:09.229213170 +0200
@@ -30,7 +30,7 @@ HEADERS =	acl.h adb.h byaddr.h cache.h c
 		rdata.h rdataclass.h rdatalist.h rdataset.h rdatasetiter.h \
 		rdataslab.h rdatatype.h request.h resolver.h result.h \
 		rootns.h rpz.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
-		tcpmsg.h time.h tkey.h tsig.h ttl.h types.h \
+		tcpmsg.h time.h tkey.h tsig.h ttl.h types.h update.h\
 		validator.h version.h view.h xfrin.h zone.h zonekey.h zt.h \
 		forward.h rrl.h
 
