diff --git a/config.h.in b/config.h.in
index 71048b6..7382171 100644
--- a/config.h.in
+++ b/config.h.in
@@ -440,7 +440,7 @@ int sigwait(const unsigned int *set, int *sig);
 #undef PORT_NONBLOCK
 
 /* The size of `void *', as computed by sizeof. */
-#undef SIZEOF_VOID_P
+/* #undef SIZEOF_VOID_P */
 
 /* Define to 1 if you have the ANSI C header files. */
 #undef STDC_HEADERS
diff --git a/configure.in b/configure.in
index 1a34bb6..1c29465 100644
--- a/configure.in
+++ b/configure.in
@@ -2370,7 +2370,9 @@ int getnameinfo(const struct sockaddr *, socklen_t, char *,
 		size_t, char *, size_t, int);],
 [ return (0);],
 	[AC_MSG_RESULT(size_t for buflen; int for flags)
-	 AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, size_t)
+	# Changed to solve multilib conflict on Fedora
+	# AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, size_t)
+	 AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, socklen_t)
 	 AC_DEFINE(IRS_GETNAMEINFO_FLAGS_T, int)],
 [AC_MSG_RESULT(not match any subspecies; assume standard definition)
 AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, socklen_t)
diff --git a/isc-config.sh.in b/isc-config.sh.in
index 10df275..9b30482 100644
--- a/isc-config.sh.in
+++ b/isc-config.sh.in
@@ -21,7 +21,19 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 exec_prefix_set=
 includedir=@includedir@
-libdir=@libdir@
+arch=$(uname -m)
+
+case $arch in
+	x86_64 | amd64 | sparc64 | s390x | ppc64)
+		libdir=/usr/lib64
+		sec_libdir=/usr/lib
+		;;
+	* )
+		libdir=/usr/lib
+		sec_libdir=/usr/lib64
+		;;
+esac
+
 
 usage()
 {
@@ -133,6 +145,16 @@ if test x"$echo_libs" = x"true"; then
 	if test x"${exec_prefix_set}" = x"true"; then
 		includes="-L${exec_prefix}/lib"
 	else
+		if [ ! -x $libdir/libisc.so ] ; then
+			if [ ! -x $sec_libdir/libisc.so ] ; then
+				echo "Error: ISC libs not found in $libdir"
+				if [ -d $sec_libdir ] ; then
+					echo "Error: ISC libs not found in $sec_libdir"
+				fi
+				exit 1
+			fi
+			libdir=$sec_libdir
+		fi
 		libs="-L${libdir}"
 	fi
 	if test x"$liblwres" = x"true" ; then
