diff -up bind-9.5.1b1/contrib/dlz/config.dlz.in.64bit bind-9.5.1b1/contrib/dlz/config.dlz.in
--- bind-9.5.1b1/contrib/dlz/config.dlz.in.64bit	2008-06-17 06:03:31.000000000 +0200
+++ bind-9.5.1b1/contrib/dlz/config.dlz.in	2008-07-11 12:08:43.000000000 +0200
@@ -17,6 +17,13 @@
 #
 dlzdir='${DLZ_DRIVER_DIR}'
 
+AC_MSG_CHECKING([for target libdir])
+AC_RUN_IFELSE([int main(void) {exit((sizeof(void *) == 8) ? 0 : 1);}],
+	[target_lib=lib64],
+	[target_lib=lib],
+)
+AC_MSG_RESULT(["$target_lib"])
+
 #
 # Private autoconf macro to simplify configuring drivers:
 #
@@ -135,9 +142,9 @@ then
 		then
 			use_dlz_mysql=$d
 			mysql_include=$d/include/mysql
-			if test -d $d/lib/mysql
+			if test -d $d/${target_lib}/mysql
 			then
-				mysql_lib=$d/lib/mysql
+				mysql_lib=$d/${target_lib}/mysql
 			else
 				mysql_lib=$d/lib
 			fi
@@ -383,7 +390,7 @@ case "$use_dlz_ldap" in
 	*)
 		DLZ_ADD_DRIVER(LDAP, dlz_ldap_driver,
 				[-I$use_dlz_ldap/include],
-				[-L$use_dlz_ldap/lib -lldap -llber])
+				[-L$use_dlz_ldap/${target_lib} -lldap -llber])
 
 		AC_MSG_RESULT(
 [using LDAP from $use_dlz_ldap/lib and $use_dlz_ldap/include])
@@ -407,7 +414,7 @@ then
 	odbcdirs="/usr /usr/local /usr/pkg"
 	for d in $odbcdirs
 	do
-		if test -f $d/include/sql.h -a -f $d/lib/libodbc.a
+		if test -f $d/include/sql.h -a -f $d/${target_lib}/libodbc.a
 		then
 			use_dlz_odbc=$d
 			break
@@ -427,7 +434,7 @@ case "$use_dlz_odbc" in
 	*)
 		DLZ_ADD_DRIVER(ODBC, dlz_odbc_driver,
 				[-I$use_dlz_odbc/include],
-				[-L$use_dlz_odbc/lib -lodbc])
+				[-L$use_dlz_odbc/${target_lib} -lodbc])
 
 		AC_MSG_RESULT([using ODBC from $use_dlz_odbc])
 		;;
