diff -up bind-9.8.0-P2/lib/dns/masterdump.c.libdns-export bind-9.8.0-P2/lib/dns/masterdump.c
--- bind-9.8.0-P2/lib/dns/masterdump.c.libdns-export	2011-05-27 10:48:10.216765461 +0200
+++ bind-9.8.0-P2/lib/dns/masterdump.c	2011-05-27 10:48:50.428765471 +0200
@@ -700,6 +700,26 @@ dns_master_questiontotext(dns_name_t *ow
 				ISC_FALSE, target));
 }
 
+static const char *trustnames[] = {
+	"none",
+	"pending-additional",
+	"pending-answer",
+	"additional",
+	"glue",
+	"answer",
+	"authauthority",
+	"authanswer",
+	"secure",
+	"local" /* aka ultimate */
+};
+
+const char *
+dns_trust_totext(dns_trust_t trust) {
+	if (trust >= sizeof(trustnames)/sizeof(*trustnames))
+		return ("bad");
+	return (trustnames[trust]);
+}
+
 #ifdef BIND9
 /*
  * Print an rdataset.  'buffer' is a scratch buffer, which must have been
@@ -835,26 +855,6 @@ dump_order_compare(const void *a, const 
 
 #define MAXSORT 64
 
-static const char *trustnames[] = {
-	"none",
-	"pending-additional",
-	"pending-answer",
-	"additional",
-	"glue",
-	"answer",
-	"authauthority",
-	"authanswer",
-	"secure",
-	"local" /* aka ultimate */
-};
-
-const char *
-dns_trust_totext(dns_trust_t trust) {
-	if (trust >= sizeof(trustnames)/sizeof(*trustnames))
-		return ("bad");
-	return (trustnames[trust]);
-}
-
 static isc_result_t
 dump_rdatasets_text(isc_mem_t *mctx, dns_name_t *name,
 		    dns_rdatasetiter_t *rdsiter, dns_totext_ctx_t *ctx,
