diff -up bind-9.9.0b1/bin/tests/system/Makefile.in.buildfix bind-9.9.0b1/bin/tests/system/Makefile.in
--- bind-9.9.0b1/bin/tests/system/Makefile.in.buildfix	2011-11-10 12:52:41.990270700 +0100
+++ bind-9.9.0b1/bin/tests/system/Makefile.in	2011-11-10 12:52:52.596269957 +0100
@@ -21,7 +21,7 @@ top_srcdir =	@top_srcdir@
 
 @BIND9_MAKE_INCLUDES@
 
-SUBDIRS =	dlzexternal filter-aaaa lwresd rpz rsabigexponent tkey tsiggss
+SUBDIRS =	filter-aaaa lwresd rpz rsabigexponent tkey tsiggss
 TARGETS =
 
 @BIND9_MAKE_RULES@
