diff -up bind-9.7.3/lib/isccc/Makefile.in.rh665971 bind-9.7.3/lib/isccc/Makefile.in
--- bind-9.7.3/lib/isccc/Makefile.in.rh665971	2011-02-18 15:01:28.718993909 +0100
+++ bind-9.7.3/lib/isccc/Makefile.in	2011-02-18 15:01:41.718965257 +0100
@@ -71,7 +71,7 @@ libisccc.la: ${OBJS}
 	${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisccc.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
-		${OBJS} ${LIBS} ${ISCLIBS}
+		${OBJS} ${ISCLIBS} ${LIBS}
 
 timestamp: libisccc.@A@
 	touch timestamp
diff -up bind-9.7.3/lib/isccfg/Makefile.in.rh665971 bind-9.7.3/lib/isccfg/Makefile.in
--- bind-9.7.3/lib/isccfg/Makefile.in.rh665971	2011-02-18 15:01:06.809042198 +0100
+++ bind-9.7.3/lib/isccfg/Makefile.in	2011-02-18 15:01:19.549014121 +0100
@@ -68,7 +68,7 @@ libisccfg.la: ${OBJS}
 	 ${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisccfg.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
-		${OBJS} ${LIBS} ${DNSLIBS} ${ISCCCLIBS} ${ISCLIBS}
+		${OBJS} ${DNSLIBS} ${ISCCCLIBS} ${ISCLIBS} ${LIBS}
 
 timestamp: libisccfg.@A@
 	touch timestamp
