diff -up bind-9.9.5rc2/lib/dns/include/dns/Makefile.in.rrl bind-9.9.5rc2/lib/dns/include/dns/Makefile.in
--- bind-9.9.5rc2/lib/dns/include/dns/Makefile.in.rrl	2014-01-24 12:48:16.869796283 +0100
+++ bind-9.9.5rc2/lib/dns/include/dns/Makefile.in	2014-01-24 12:49:15.704773787 +0100
@@ -30,7 +30,7 @@ HEADERS =	acl.h adb.h byaddr.h cache.h c
 		peer.h portlist.h private.h rbt.h rcode.h \
 		rdata.h rdataclass.h rdatalist.h rdataset.h rdatasetiter.h \
 		rdataslab.h rdatatype.h request.h resolver.h result.h \
-		rootns.h rpz.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
+		rootns.h rpz.h rrl.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
 		tcpmsg.h time.h tkey.h tsec.h tsig.h ttl.h types.h \
 		validator.h version.h view.h xfrin.h zone.h zonekey.h zt.h
 
