diff -up bind-9.10.1b1/lib/isc/unix/socket.c.rh735103 bind-9.10.1b1/lib/isc/unix/socket.c
--- bind-9.10.1b1/lib/isc/unix/socket.c.rh735103	2014-06-23 06:47:35.000000000 +0200
+++ bind-9.10.1b1/lib/isc/unix/socket.c	2014-07-29 16:25:27.172818662 +0200
@@ -67,6 +67,20 @@
 #include <isc/util.h>
 #include <isc/xml.h>
 
+/* See task.c about the following definition: */
+#ifdef BIND9
+#ifdef ISC_PLATFORM_USETHREADS
+#define USE_WATCHER_THREAD
+#else
+#define USE_SHARED_MANAGER
+#endif	/* ISC_PLATFORM_USETHREADS */
+#else /* BIND9 */
+#undef ISC_PLATFORM_HAVESYSUNH
+#undef ISC_PLATFORM_HAVEKQUEUE
+#undef ISC_PLATFORM_HAVEEPOLL
+#undef ISC_PLATFORM_HAVEDEVPOLL
+#endif	/* BIND9 */
+
 #ifdef ISC_PLATFORM_HAVESYSUNH
 #include <sys/un.h>
 #endif
@@ -86,13 +100,6 @@
 
 #include "errno2result.h"
 
-/* See task.c about the following definition: */
-#ifdef ISC_PLATFORM_USETHREADS
-#define USE_WATCHER_THREAD
-#else
-#define USE_SHARED_MANAGER
-#endif	/* ISC_PLATFORM_USETHREADS */
-
 #ifndef USE_WATCHER_THREAD
 #include "socket_p.h"
 #include "../task_p.h"
