diff -up bind-9.9.5rc2/lib/dns/include/dns/Makefile.in.update bind-9.9.5rc2/lib/dns/include/dns/Makefile.in
--- bind-9.9.5rc2/lib/dns/include/dns/Makefile.in.update	2014-01-24 12:50:47.811725841 +0100
+++ bind-9.9.5rc2/lib/dns/include/dns/Makefile.in	2014-01-24 12:51:33.866701688 +0100
@@ -31,7 +31,7 @@ HEADERS =	acl.h adb.h byaddr.h cache.h c
 		rdata.h rdataclass.h rdatalist.h rdataset.h rdatasetiter.h \
 		rdataslab.h rdatatype.h request.h resolver.h result.h \
 		rootns.h rpz.h rrl.h sdb.h sdlz.h secalg.h secproto.h soa.h ssu.h \
-		tcpmsg.h time.h tkey.h tsec.h tsig.h ttl.h types.h \
+		tcpmsg.h time.h tkey.h tsec.h tsig.h ttl.h types.h update.h \
 		validator.h version.h view.h xfrin.h zone.h zonekey.h zt.h
 
 GENHEADERS =	enumclass.h enumtype.h rdatastruct.h
