diff -up bind-9.9.3rc2/config.h.in.multlib-conflict bind-9.9.3rc2/config.h.in
--- bind-9.9.3rc2/config.h.in.multlib-conflict	2013-04-30 08:38:46.000000000 +0200
+++ bind-9.9.3rc2/config.h.in	2013-05-13 12:10:22.514870894 +0200
@@ -416,7 +416,7 @@ int sigwait(const unsigned int *set, int
 #undef PORT_NONBLOCK
 
 /* The size of `void *', as computed by sizeof. */
-#undef SIZEOF_VOID_P
+/* #undef SIZEOF_VOID_P */
 
 /* Define to 1 if you have the ANSI C header files. */
 #undef STDC_HEADERS
diff -up bind-9.9.3rc2/configure.in.multlib-conflict bind-9.9.3rc2/configure.in
--- bind-9.9.3rc2/configure.in.multlib-conflict	2013-05-13 12:10:22.481870901 +0200
+++ bind-9.9.3rc2/configure.in	2013-05-13 12:10:22.515870894 +0200
@@ -2251,7 +2251,9 @@ int getnameinfo(const struct sockaddr *,
                 size_t, char *, size_t, int);],
 [ return (0);],
 	[AC_MSG_RESULT(size_t for buflen; int for flags)
-	 AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, size_t)
+	# Changed to solve multilib conflict on Fedora 
+	#AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, size_t)
+	 AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, socklen_t)
 	 AC_DEFINE(IRS_GETNAMEINFO_FLAGS_T, int)],
 [AC_MSG_RESULT(not match any subspecies; assume standard definition)
 AC_DEFINE(IRS_GETNAMEINFO_BUFLEN_T, socklen_t)
diff -up bind-9.9.3rc2/isc-config.sh.in.multlib-conflict bind-9.9.3rc2/isc-config.sh.in
--- bind-9.9.3rc2/isc-config.sh.in.multlib-conflict	2013-04-30 08:38:46.000000000 +0200
+++ bind-9.9.3rc2/isc-config.sh.in	2013-05-13 12:26:40.258698745 +0200
@@ -21,7 +21,18 @@ prefix=@prefix@
 exec_prefix=@exec_prefix@
 exec_prefix_set=
 includedir=@includedir@
-libdir=@libdir@
+arch=$(uname -m)
+
+case $arch in
+	x86_64 | amd64 | sparc64 | s390x | ppc64)
+		libdir=/usr/lib64
+		sec_libdir=/usr/lib
+		;;
+	* )
+		libdir=/usr/lib
+		sec_libdir=/usr/lib64
+		;;
+esac
 
 usage()
 {
@@ -133,6 +144,16 @@ if test x"$echo_libs" = x"true"; then
 	if test x"${exec_prefix_set}" = x"true"; then
 		includes="-L${exec_prefix}/lib"
 	else
+		if [ ! -x $libdir/libisc.so ] ; then
+			if [ ! -x $sec_libdir/libisc.so ] ; then
+				echo "Error: ISC libs not found in $libdir"
+				if [ -d $sec_libdir ] ; then
+					echo "Error: ISC libs not found in $sec_libdir"
+				fi
+				exit 1
+			fi
+			libdir=$sec_libdir
+		fi
 		libs="-L${libdir}"
 	fi
 	if test x"$liblwres" = x"true" ; then
