--- src/scripts/festival_server.sh.orig	2013-03-02 09:54:57.126436666 -0600
+++ src/scripts/festival_server.sh	2013-03-02 09:55:10.529140080 -0600
@@ -119,7 +119,7 @@
 (set! server_port $port)
 (set! server_festival_version "$festival" )
 (set! server_log_file "$server_log" )
-(set! server_startup_file "$serer_startup" )
+(set! server_startup_file "$server_startup" )
 
 ;; Marks end of machine created bit
 ;---
