--- include/difs.h.orig	2012-02-29 18:50:25.009841758 -0300
+++ include/difs.h	2012-02-29 18:50:29.953779989 -0300
@@ -42,7 +42,6 @@ extern ReplySwapFunc ReplySwapVector[NUM
 /* FIXME: this is derived from fontstruct.h; should integrate it */
 
 /* difs/atom.c */
-extern Atom MakeAtom ( const char *string, unsigned len, Bool makeit );
 extern int ValidAtom ( Atom atom );
 extern char * NameForAtom ( Atom atom );
 extern void InitAtoms ( void );
