--- m4/dpkg-arch.m4.orig	2011-09-23 01:16:57.000000000 -0300
+++ m4/dpkg-arch.m4	2011-09-28 00:00:33.063768615 -0300
@@ -8,8 +8,7 @@
 AC_DEFUN([_DPKG_ARCHITECTURE], [
 AC_REQUIRE([DPKG_PROG_PERL])dnl
 AC_REQUIRE([AC_CANONICAL_HOST])dnl
-$2=$(cd $srcdir/scripts; \
-    PERL5LIB=$(pwd) $PERL dpkg-architecture.pl -t$host -q$1 2>/dev/null)
+$2=$1
 ])# _DPKG_ARCHITECURE
 
 # DPKG_CPU_TYPE
@@ -18,7 +17,7 @@ $2=$(cd $srcdir/scripts; \
 # the Debian name for it.  Sets ARCHITECTURE_CPU.
 AC_DEFUN([DPKG_CPU_TYPE],
 [AC_MSG_CHECKING([dpkg cpu type])
-_DPKG_ARCHITECTURE([DEB_HOST_ARCH_CPU], [cpu_type])
+_DPKG_ARCHITECTURE([SLK_HOST_ARCH_CPU], [cpu_type])
 if test "x$cpu_type" = "x"; then
 	cpu_type=$host_cpu
 	AC_MSG_RESULT([$cpu_type])
@@ -36,7 +35,7 @@ AC_DEFINE_UNQUOTED(ARCHITECTURE_CPU, "${
 # special cases to determine what type it is.  Sets ARCHITECTURE_OS.
 AC_DEFUN([DPKG_OS_TYPE],
 [AC_MSG_CHECKING([dpkg operating system type])
-_DPKG_ARCHITECTURE([DEB_HOST_ARCH_OS], [os_type])
+_DPKG_ARCHITECTURE([SLK_HOST_ARCH_OS], [os_type])
 if test "x$os_type" = "x"; then
 	os_type=$host_os
 	AC_MSG_RESULT([$os_type])
@@ -56,7 +55,7 @@ AC_DEFUN([DPKG_ARCHITECTURE],
 [DPKG_CPU_TYPE
 DPKG_OS_TYPE
 AC_MSG_CHECKING([dpkg architecture name])
-_DPKG_ARCHITECTURE([DEB_HOST_ARCH], [dpkg_arch])
+_DPKG_ARCHITECTURE([SLK_HOST_ARCH], [dpkg_arch])
 if test "x$dpkg_arch" = "x"; then
 	AC_MSG_ERROR([cannot determine host dpkg architecture])
 else
