diff -up hivex-1.2.3/perl/Makefile.am.dirs hivex-1.2.3/perl/Makefile.am
--- hivex-1.2.3/perl/Makefile.am.dirs	2010-11-19 12:44:02.177644892 -0500
+++ hivex-1.2.3/perl/Makefile.am	2010-11-19 12:44:17.137516124 -0500
@@ -40,7 +40,7 @@ $(TESTS): src_deps all
 TESTS_ENVIRONMENT = \
 	LD_LIBRARY_PATH=$(top_builddir)/lib/.libs
 
-INSTALLDIRS = site
+INSTALLDIRS = vendor
 
 all: Makefile-pl src_deps
 	$(MAKE) -f Makefile-pl
diff -up hivex-1.2.3/perl/Makefile.in.dirs hivex-1.2.3/perl/Makefile.in
--- hivex-1.2.3/perl/Makefile.in.dirs	2010-11-19 12:44:08.577393593 -0500
+++ hivex-1.2.3/perl/Makefile.in	2010-11-19 12:44:32.717483099 -0500
@@ -740,7 +740,7 @@ EXTRA_DIST = \
 @HAVE_PERL_TRUE@TESTS_ENVIRONMENT = \
 @HAVE_PERL_TRUE@	LD_LIBRARY_PATH=$(top_builddir)/lib/.libs
 
-@HAVE_PERL_TRUE@INSTALLDIRS = site
+@HAVE_PERL_TRUE@INSTALLDIRS = vendor
 all: all-am
 
 .SUFFIXES:
