--- ruby/ext/hivex/_hivex.c.rb19	2011-10-29 16:17:04.540760326 -0200
+++ ruby/ext/hivex/_hivex.c	2011-10-29 16:22:16.269864170 -0200
@@ -61,8 +61,8 @@ get_value (VALUE valv, hive_set_value *v
 
   val->key = StringValueCStr (key);
   val->t = NUM2ULL (type);
-  val->len = RSTRING (value)->len;
-  val->value = RSTRING (value)->ptr;
+  val->len = RSTRING_LEN (value);
+  val->value = RSTRING_PTR (value);
 }
 
 static hive_set_value *
