diff -up libpcap/pcap-config.in.multilib libpcap/pcap-config.in
--- libpcap/pcap-config.in.multilib	2011-11-23 14:28:54.580396515 +0100
+++ libpcap/pcap-config.in	2011-11-23 14:36:47.408268484 +0100
@@ -36,16 +36,6 @@ do
 	esac
 	shift
 done
-if [ "$V_RPATH_OPT" != "" ]
-then
-	#
-	# If libdir isn't /usr/lib, add it to the run-time linker path.
-	#
-	if [ "$libdir" != "/usr/lib" ]
-	then
-		RPATH=$V_RPATH_OPT$libdir
-	fi
-fi
 if [ "$static" = 1 ]
 then
 	#
@@ -54,16 +44,16 @@ then
 	#
 	if [ "$show_cflags" = 1 -a "$show_libs" = 1 ]
 	then
-		echo "-I$includedir -L$libdir -lpcap $LIBS"
+		echo "-lpcap @LIBS@"
 	elif [ "$show_cflags" = 1 -a "$show_additional_libs" = 1 ]
 	then
-		echo "-I$includedir -L$libdir $LIBS"
+		echo "@LIBS@"
 	elif [ "$show_cflags" = 1 ]
 	then
-		echo "-I$includedir"
+		echo ""
 	elif [ "$show_libs" = 1 ]
 	then
-		echo "-L$libdir -lpcap $LIBS"
+		echo "-lpcap @LIBS@"
 	elif [ "$show_additional_libs" = 1 ]
 	then
 		echo "$LIBS"
@@ -75,15 +65,15 @@ else
 	#
 	if [ "$show_cflags" = 1 -a "$show_libs" = 1 ]
 	then
-		echo "-I$includedir -L$libdir $RPATH -lpcap"
+		echo "-lpcap"
 	elif [ "$show_cflags" = 1 -a "$show_additional_libs" = 1 ]
 	then
-		echo "-I$includedir"
+		echo ""
 	elif [ "$show_cflags" = 1 ]
 	then
-		echo "-I$includedir"
+		echo ""
 	elif [ "$show_libs" = 1 ]
 	then
-		echo "-L$libdir $RPATH -lpcap"
+		echo "-lpcap"
 	fi
 fi
