diff -up libpcap-1.3.0/pcap-config.in.multilib libpcap-1.3.0/pcap-config.in
--- libpcap-1.3.0/pcap-config.in.multilib	2012-03-29 14:57:32.000000000 +0200
+++ libpcap-1.3.0/pcap-config.in	2013-03-26 09:56:16.293324264 +0100
@@ -7,7 +7,6 @@
 prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 includedir="@includedir@"
-libdir="@libdir@"
 V_RPATH_OPT="@V_RPATH_OPT@"
 LIBS="@LIBS@"
 
@@ -36,16 +35,6 @@ do
 	esac
 	shift
 done
-if [ "$V_RPATH_OPT" != "" ]
-then
-	#
-	# If libdir isn't /usr/lib, add it to the run-time linker path.
-	#
-	if [ "$libdir" != "/usr/lib" ]
-	then
-		RPATH=$V_RPATH_OPT$libdir
-	fi
-fi
 if [ "$static" = 1 ]
 then
 	#
@@ -54,16 +43,16 @@ then
 	#
 	if [ "$show_cflags" = 1 -a "$show_libs" = 1 ]
 	then
-		echo "-I$includedir -L$libdir -lpcap $LIBS"
+		echo "-lpcap @LIBS@"
 	elif [ "$show_cflags" = 1 -a "$show_additional_libs" = 1 ]
 	then
-		echo "-I$includedir -L$libdir $LIBS"
+		echo "@LIBS@"
 	elif [ "$show_cflags" = 1 ]
 	then
-		echo "-I$includedir"
+		echo ""
 	elif [ "$show_libs" = 1 ]
 	then
-		echo "-L$libdir -lpcap $LIBS"
+		echo "-lpcap @LIBS@"
 	elif [ "$show_additional_libs" = 1 ]
 	then
 		echo "$LIBS"
@@ -75,15 +64,15 @@ else
 	#
 	if [ "$show_cflags" = 1 -a "$show_libs" = 1 ]
 	then
-		echo "-I$includedir -L$libdir $RPATH -lpcap"
+		echo "-lpcap"
 	elif [ "$show_cflags" = 1 -a "$show_additional_libs" = 1 ]
 	then
-		echo "-I$includedir"
+		echo ""
 	elif [ "$show_cflags" = 1 ]
 	then
-		echo "-I$includedir"
+		echo ""
 	elif [ "$show_libs" = 1 ]
 	then
-		echo "-L$libdir $RPATH -lpcap"
+		echo "-lpcap"
 	fi
 fi
