diff -up vsftpd-3.0.2/privops.c.pasv-addr vsftpd-3.0.2/privops.c
--- vsftpd-3.0.2/privops.c.pasv-addr	2013-09-03 15:57:47.796756184 +0200
+++ vsftpd-3.0.2/privops.c	2013-09-03 15:57:58.873752201 +0200
@@ -160,7 +160,19 @@ vsf_privop_pasv_listen(struct vsf_sessio
       p_sess->pasv_listen_fd = vsf_sysutil_get_ipv4_sock();
     }
     vsf_sysutil_activate_reuseaddr(p_sess->pasv_listen_fd);
-    vsf_sysutil_sockaddr_clone(&s_p_sockaddr, p_sess->p_local_addr);
+    if (tunable_pasv_address != 0)
+    {
+      vsf_sysutil_sockaddr_alloc_ipv4(&s_p_sockaddr);
+      /* Use passive address as specified in configuration */
+      if (vsf_sysutil_inet_aton(tunable_pasv_address, s_p_sockaddr) == 0)
+      {
+        die("invalid pasv_address");
+      }
+    }
+    else
+    {
+      vsf_sysutil_sockaddr_clone(&s_p_sockaddr, p_sess->p_local_addr);
+    }
     vsf_sysutil_sockaddr_set_port(s_p_sockaddr, the_port);
     retval = vsf_sysutil_bind(p_sess->pasv_listen_fd, s_p_sockaddr);
     if (!vsf_sysutil_retval_is_error(retval))
