diff -up vsftpd-2.3.5/defs.h.aslim vsftpd-2.3.5/defs.h
--- vsftpd-2.3.5/defs.h.aslim	2012-02-09 07:40:04.916658587 +0100
+++ vsftpd-2.3.5/defs.h	2012-02-09 07:50:18.365278392 +0100
@@ -19,7 +19,7 @@
 /* Must be at least the size of VSFTP_MAX_COMMAND_LINE, VSFTP_DIR_BUFSIZE and
    VSFTP_DATA_BUFSIZE*2 */
 #define VSFTP_PRIVSOCK_MAXSTR   VSFTP_DATA_BUFSIZE * 2
-#define VSFTP_AS_LIMIT          100UL * 1024 * 1024
+#define VSFTP_AS_LIMIT          400UL * 1024 * 1024
 
 #endif /* VSF_DEFS_H */
 
