--- module-init-tools-3.13/modprobe.c.orig	2011-05-31 02:01:33.000000000 -0300
+++ module-init-tools-3.13/modprobe.c	2011-05-31 20:07:22.232177181 -0300
@@ -38,6 +38,7 @@
 #include <asm/unistd.h>
 #include <sys/wait.h>
 #include <syslog.h>
+#include <regex.h>
 
 #include "util.h"
 #include "elfops.h"
@@ -789,6 +790,20 @@ static char *strsep_skipspace(char **str
 	return strsep(string, delim);
 }
 
+/* Let's exclude a few file extensions */
+static int valid_file_name(const char *filename)
+{
+	static regex_t *re = NULL;
+
+	if (!re) {
+		re = NOFAIL(malloc(sizeof(regex_t)));
+		if (regcomp(re, "(^(\\.|\\.\\.)|\\.(new|orig|bak)$)",
+			REG_EXTENDED|REG_NOSUB) != 0)
+				fatal("regcomp failed: %s\n", strerror(errno));
+		}
+		return regexec(re, filename, 0, NULL, 0);
+}
+
 static int parse_config_scan(struct modprobe_conf *conf,
 			     int dump_only,
 			     int removing, ...);
@@ -1071,6 +1086,8 @@ static int parse_config_scan(struct modp
 				size_t len;
 				int cmp = -1;
 
+			if (!valid_file_name(i->d_name))
+				continue;
 				if (i->d_name[0] == '.')
 					continue;
 				if (!config_filter(i->d_name))
