--- Source/Settings.py.orig	2009-04-22 19:38:10.000000000 -0300
+++ Source/Settings.py	2010-07-06 19:59:24.901129461 -0300
@@ -40,6 +40,7 @@
 # Application
 Application				= QtCore.QObject()
 Application.sName		= "Q7Z"
+Application.sConfigName	= "Chris_Giles"
 Application.sVersion	= QtCore.QString( "0.8.0" )
 Application.sDomain		= "k7z.sourceforge.net"
 Application.sCurrent	= "/7Z/" + Application.sName + "/Current.txt"
@@ -72,7 +73,7 @@
 
 # Qt
 QtCore.QCoreApplication.setApplicationName( Application.sName )
-QtCore.QCoreApplication.setOrganizationName( Author.sName )
+QtCore.QCoreApplication.setOrganizationName( Application.sConfigName )
 QtCore.QCoreApplication.setOrganizationDomain( Application.sDomain )
 
 # Operating system
@@ -101,7 +102,7 @@
 Path.App.sPrefix	= ""
 Path.App.sSBin		= ""
 Path.App.sInstall		= "/share/" + Application.sName
-Path.App.sConfig		= Env.sHome + '/.config/' + Author.sName + Path.sSep + Application.sName + Path.sSep
+Path.App.sConfig		= Env.sHome + '/.config/' + Application.sConfigName + Path.sSep + Application.sName + Path.sSep
 Path.KDE			= QtCore.QObject()
 Path.KDE.sAutoStart	= ""
 Path.KDE.sConfig	= ""
