diff -up msilbc-2.1.2/configure.ac.new-ilbc msilbc-2.1.2/configure.ac
--- msilbc-2.1.2/configure.ac.new-ilbc	2015-11-02 12:00:06.000000000 -0200
+++ msilbc-2.1.2/configure.ac	2015-11-05 00:12:30.091857463 -0200
@@ -49,7 +49,7 @@ fi
 # check for libraries that have pkg-config files installed
 PKG_CHECK_MODULES(ORTP, ortp >= 0.16.0)
 PKG_CHECK_MODULES(MEDIASTREAMER, mediastreamer >= 2.0.0)
-PKG_CHECK_MODULES(ILBC, ilbc >= 0.0.0)
+PKG_CHECK_MODULES(ILBC, libilbc >= 0.0.0)
 
 # create the following files from their .in counterparts
 AC_CONFIG_FILES([
diff -up msilbc-2.1.2/ilbc.c.new-ilbc msilbc-2.1.2/ilbc.c
--- msilbc-2.1.2/ilbc.c.new-ilbc	2015-11-05 00:12:30.091857463 -0200
+++ msilbc-2.1.2/ilbc.c	2015-11-05 00:13:30.159220631 -0200
@@ -1,13 +1,7 @@
 /*
 	NO LICENSE
 */
-#ifdef ANDROID
-	#include <iLBC_encode.h>
-	#include <iLBC_decode.h>
-#else
-	#include <ilbc/iLBC_encode.h>
-	#include <ilbc/iLBC_decode.h>
-#endif /*ANDROID*/
+#include <ilbc.h>
 
 #include "mediastreamer2/msfilter.h"
 #include "mediastreamer2/mscodecutils.h"
@@ -19,6 +13,9 @@
 #define ms_bufferizer_fill_current_metas(b,m)
 #endif
 
+#define NO_OF_BYTES_20MS 38
+#define NO_OF_BYTES_30MS 50
+
 typedef struct EncState{
 	int nsamples;
 	int nbytes;
