diff -up msilbc-2.0.3/configure.ac.new-ilbc msilbc-2.0.3/configure.ac
--- msilbc-2.0.3/configure.ac.new-ilbc	2011-02-07 12:26:04.000000000 -0200
+++ msilbc-2.0.3/configure.ac	2013-06-15 19:11:32.906901605 -0300
@@ -48,7 +48,7 @@ fi
 # check for libraries that have pkg-config files installed
 PKG_CHECK_MODULES(ORTP, ortp >= 0.16.0)
 PKG_CHECK_MODULES(MEDIASTREAMER, mediastreamer >= 2.0.0)
-PKG_CHECK_MODULES(ILBC, ilbc >= 0.0.0)
+PKG_CHECK_MODULES(ILBC, libilbc >= 0.0.0)
 
 # create the following files from their .in counterparts
 AC_CONFIG_FILES([
diff -up msilbc-2.0.3/ilbc.c.new-ilbc msilbc-2.0.3/ilbc.c
--- msilbc-2.0.3/ilbc.c.new-ilbc	2011-02-07 12:26:04.000000000 -0200
+++ msilbc-2.0.3/ilbc.c	2013-06-15 19:14:29.515307257 -0300
@@ -2,15 +2,16 @@
 	NO LICENSE
 */
 #ifdef ANDROID
-	#include <iLBC_encode.h>
-	#include <iLBC_decode.h>
+	#include <ilbc.h>
 #else
-	#include <ilbc/iLBC_encode.h>
-	#include <ilbc/iLBC_decode.h>
+	#include <ilbc.h>
 #endif /*ANDROID*/
 
 #include "mediastreamer2/msfilter.h"
 
+#define NO_OF_BYTES_20MS 38
+#define NO_OF_BYTES_30MS 50
+
 typedef struct EncState{
 	int nsamples;
 	int nbytes;
