--- src/fr-window.c.orig	2011-09-27 17:11:44.475182076 -0300
+++ src/fr-window.c	2011-09-27 17:14:53.536817999 -0300
@@ -640,7 +640,6 @@ fr_window_free_private_data (FrWindow *w
 	_g_object_unref (window->priv->settings_ui);
 	_g_object_unref (window->priv->settings_general);
 	_g_object_unref (window->priv->settings_dialogs);
-	_g_object_unref (window->priv->settings_nautilus);
 }
 
 
@@ -4647,21 +4646,6 @@ fr_window_delete_event_cb (GtkWidget *ca
 	fr_window_close (window);
 }
 
-
-static gboolean
-is_single_click_policy (FrWindow *window)
-{
-	char     *value;
-	gboolean  result;
-
-	value = g_settings_get_string (window->priv->settings_nautilus, NAUTILUS_CLICK_POLICY);
-	result = (value != NULL) && (strncmp (value, "single", 6) == 0);
-	g_free (value);
-
-	return result;
-}
-
-
 static void
 filename_cell_data_func (GtkTreeViewColumn *column,
 			 GtkCellRenderer   *renderer,
@@ -4781,8 +4765,6 @@ add_file_list_columns (FrWindow    *wind
 
 	/* name */
 
-	window->priv->single_click = is_single_click_policy (window);
-
 	renderer = gtk_cell_renderer_text_new ();
 
 	g_value_init (&value, PANGO_TYPE_ELLIPSIZE_MODE);
@@ -5031,8 +5013,6 @@ pref_click_policy_changed (GSettings  *s
 	GdkWindow  *win = gtk_tree_view_get_bin_window (GTK_TREE_VIEW (window->priv->list_view));
 	GdkDisplay *display;
 
-	window->priv->single_click = is_single_click_policy (window);
-
 	gdk_window_set_cursor (win, NULL);
 	display = gtk_widget_get_display (GTK_WIDGET (window->priv->list_view));
 	if (display != NULL)
