--- configure.py.orig	2014-10-31 16:47:12.477201327 -0200
+++ configure.py	2014-10-31 16:48:21.005766712 -0200
@@ -62,7 +62,7 @@ def llvm_config(args):
 
 llvm_bindir = llvm_config(['--bindir'])
 llvm_core_libs = llvm_config(['--libs', 'core', 'bitreader', 'bitwriter']) + ' ' + \
-                 llvm_config(['--ldflags'])
+                 llvm_config(['--ldflags']) + ' _SB_EXTRA_LIBS_'
 llvm_cxxflags = llvm_config(['--cxxflags']) + ' -fno-exceptions -fno-rtti'
 
 llvm_clang = os.path.join(llvm_bindir, 'clang')
