--- gkrellm-2.3.5/src/sysdeps/linux.c.orig	2012-06-21 22:53:40.860763444 -0300
+++ gkrellm-2.3.5/src/sysdeps/linux.c	2012-06-21 22:54:31.311042662 -0300
@@ -1439,6 +1439,7 @@ gkrellm_sys_fs_get_mounts_list(void)
 			|| !strcmp(type, "usbdevfs")
 			|| !strcmp(type, "usbfs")
 			|| !strcmp(type, "sysfs")
+			|| !strcmp(type, "autofs")
 		   )
 			continue;
 		/* Strip trailing / from the directory.
