diff -up gkrellm-2.3.5/server/monitor.c.cifs gkrellm-2.3.5/server/monitor.c
--- gkrellm-2.3.5/server/monitor.c.cifs	2010-10-05 13:46:59.000000000 -0300
+++ gkrellm-2.3.5/server/monitor.c	2012-06-21 22:56:49.080770163 -0300
@@ -1329,7 +1329,8 @@ static gboolean	nfs_check,
 static gchar *remote_fs_types[] =
 	{
 	"nfs",
-	"smbfs"
+	"smbfs",
+	"cifs"
 	};
 
 void
diff -up gkrellm-2.3.5/src/fs.c.cifs gkrellm-2.3.5/src/fs.c
--- gkrellm-2.3.5/src/fs.c.cifs	2012-06-21 22:55:50.940122867 -0300
+++ gkrellm-2.3.5/src/fs.c	2012-06-21 22:56:34.923777901 -0300
@@ -163,7 +163,8 @@ static gint		style_id;
 static gchar *remote_fs_types[]	=
 	{
 	"nfs",
-	"smbfs"
+	"smbfs",
+	"cifs"
 	};
 
 
