--- ypbind-mt-1.37.1/configure.in.orig	2014-02-22 00:54:10.588309723 -0300
+++ ypbind-mt-1.37.1/configure.in	2014-02-22 00:56:22.320383790 -0300
@@ -105,7 +105,7 @@ AM_CONDITIONAL(ENABLE_REGENERATE_MAN, te
 
 USE_SD_NOTIFY=0
 AC_SUBST(USE_SD_NOTIFY)
-AC_CHECK_LIB(systemd-daemon,sd_notify,LIBSYSTEMD_DAEMON="-lsystemd-daemon",
+AC_CHECK_LIB(systemd,sd_notify,LIBSYSTEMD_DAEMON="-lsystemd",
        LIBSYSTEMD_DAEMON="")
 if test -n "$LIBSYSTEMD_DAEMON" ; then
        AC_CHECK_HEADERS(systemd/sd-daemon.h)
