--- ypserv-2.31/configure.in.orig	2014-02-22 00:56:41.950221824 -0300
+++ ypserv-2.31/configure.in	2014-02-22 00:57:03.390675986 -0300
@@ -272,7 +272,7 @@ fi
 
 USE_SD_NOTIFY=0
 AC_SUBST(USE_SD_NOTIFY)
-AC_CHECK_LIB(systemd-daemon,sd_notify,LIBSYSTEMD_DAEMON="-lsystemd-daemon",
+AC_CHECK_LIB(systemd,sd_notify,LIBSYSTEMD_DAEMON="-lsystemd",
 	LIBSYSTEMD_DAEMON="")
 if test -n "$LIBSYSTEMD_DAEMON" ; then
 	AC_CHECK_HEADERS(systemd/sd-daemon.h)
