diff -up ypserv-2.25/scripts/ypMakefile.in.portmanfix ypserv-2.25/scripts/ypMakefile.in
--- ypserv-2.25/scripts/ypMakefile.in.portmanfix	2011-06-14 15:03:52.876412140 +0200
+++ ypserv-2.25/scripts/ypMakefile.in	2011-06-14 15:03:52.905412140 +0200
@@ -22,6 +22,14 @@ B=
 # /var/yp/ypservers.
 NOPUSH=true
 
+# yppush copies updated NIS databases (or maps) from the master NIS server 
+# to the slave servers within a NIS domain and listens for their answer. 
+# yppush will ask portmap to assign it a random port number to listen on 
+# by default. To specify a fixed port number or any aditional options, 
+# edit variable YPPUSH_ARGS.
+# e.g. YPPUSH_ARGS = --port 836
+YPPUSH_ARGS = 
+
 # We do not put password entries with lower UIDs (the root and system
 # entries) in the NIS password database, for security. MINUID is the
 # lowest uid that will be included in the password maps. If you
@@ -128,7 +136,7 @@ all:  passwd group hosts rpc services ne
 
 DBLOAD = $(YPBINDIR)/makedbm -c -m `$(YPBINDIR)/yphelper --hostname`
 MKNETID = $(YPBINDIR)/mknetid
-YPPUSH = $(YPSBINDIR)/yppush
+YPPUSH = $(YPSBINDIR)/yppush $(YPPUSH_ARGS)
 MERGER = $(YPBINDIR)/yphelper
 DOMAIN = `basename \`pwd\``
 LOCALDOMAIN = `/bin/domainname`
diff -up ypserv-2.25/yppush/yppush.8.portmanfix ypserv-2.25/yppush/yppush.8
--- ypserv-2.25/yppush/yppush.8.portmanfix	2010-09-24 12:13:12.000000000 +0200
+++ ypserv-2.25/yppush/yppush.8	2011-06-14 15:04:48.361412162 +0200
@@ -192,6 +192,13 @@ first constructs a list of NIS slave ser
 within the
 \fIdomain\&.\fR
 A destination host (or a list of hosts with multiple \-h commands) can also be specified on the command line\&. A "transfer map" request is sent to the NIS serger at each slave, along with the information needed by the transfer agent ypxfr(8) to callback to yppush, which may be printed the result to stderr\&. Messages are also printed when a transfer is not possible; for instance when the request message is undeliverable\&.
+
+To specify a port number or use any other 
+\fByppush\fR
+options you can edit 
+\fI/var/yp/Makefile\fR
+directly.
+
 .SH "OPTIONS"
 .PP
 \fB\-d \fR\fIdomain\fR
