Index: source/gameengine/Expressions/Value.cpp
===================================================================
--- source/gameengine/Expressions/Value.cpp.orig	2010-04-18 12:28:37.000000000 +0200
+++ source/gameengine/Expressions/Value.cpp	2010-10-28 14:56:24.000000000 +0200
@@ -543,8 +543,8 @@ CValue* CValue::ConvertPythonToValue(PyO
 		CListValue* listval = new CListValue();
 		bool error = false;
 
-		int i;
-		int numitems = PyList_Size(pyobj);
+		Py_ssize_t i;
+		Py_ssize_t numitems = PyList_Size(pyobj);
 		for (i=0;i<numitems;i++)
 		{
 			PyObject* listitem = PyList_GetItem(pyobj,i); /* borrowed ref */
