diff -Naur blender-2.58.orig/CMakeLists.txt blender-2.58/CMakeLists.txt
--- blender-2.58.orig/CMakeLists.txt	2011-08-05 14:45:26.776618379 -0500
+++ blender-2.58/CMakeLists.txt	2011-08-05 14:47:21.562887635 -0500
@@ -307,6 +307,7 @@
 		# Use our own instead, since wothout py is such a rare case,
 		# require this package
 		find_package(PythonLibsUnix REQUIRED)
+		set(PYTHON_INCLUDE_DIRS ${PYTHON_INCLUDE_DIR})
 	endif()
 
 
