Index: extern/bullet2/src/LinearMath/btVector3.h
===================================================================
--- extern/bullet2/src/LinearMath/btVector3.h.orig	2009-03-17 23:44:58.000000000 +0200
+++ extern/bullet2/src/LinearMath/btVector3.h	2010-10-28 14:56:38.000000000 +0200
@@ -63,7 +63,8 @@ public:
 	public:
 
   /**@brief No initialization constructor */
-	SIMD_FORCE_INLINE btVector3() {}
+	SIMD_FORCE_INLINE btVector3() {
+	m_floats[0]=m_floats[1]=m_floats[2]=m_floats[3]=btScalar(0.);}
 
  
 	
Index: extern/bullet2/src/BulletCollision/BroadphaseCollision/btQuantizedBvh.h
===================================================================
--- extern/bullet2/src/BulletCollision/BroadphaseCollision/btQuantizedBvh.h.orig	2009-09-14 16:45:32.000000000 +0200
+++ extern/bullet2/src/BulletCollision/BroadphaseCollision/btQuantizedBvh.h	2010-10-28 14:56:38.000000000 +0200
@@ -94,7 +94,8 @@ ATTRIBUTE_ALIGNED16 (struct) btOptimized
 	//for child nodes
 	int	m_subPart;
 	int	m_triangleIndex;
-	int	m_padding[5];//bad, due to alignment
+	int:5*8*sizeof(int);
+//	int	m_padding[5];//bad, due to alignment
 
 
 };
