diff -up blender-2.61/source/blender/blenfont/intern/blf_lang.c.syspath blender-2.61/source/blender/blenfont/intern/blf_lang.c
--- blender-2.61/source/blender/blenfont/intern/blf_lang.c.syspath	2011-12-13 20:51:08.000000000 +0100
+++ blender-2.61/source/blender/blenfont/intern/blf_lang.c	2011-12-14 19:45:05.695753793 +0100
@@ -101,7 +101,7 @@ static const char *locales[] = {
 
 void BLF_lang_init(void)
 {
-	char *messagepath= BLI_get_folder(BLENDER_DATAFILES, "locale");
+        char *messagepath = "/usr/share/locale";
 	
 	BLI_strncpy(global_encoding_name, SYSTEM_ENCODING_DEFAULT, sizeof(global_encoding_name));
 	
diff -up blender-2.61/source/blender/blenlib/intern/path_util.c.syspath blender-2.61/source/blender/blenlib/intern/path_util.c
--- blender-2.61/source/blender/blenlib/intern/path_util.c.syspath	2011-12-13 20:51:07.000000000 +0100
+++ blender-2.61/source/blender/blenlib/intern/path_util.c	2011-12-14 19:48:04.557027766 +0100
@@ -982,35 +982,6 @@ static int get_path_system(char *targetp
 	const char *system_base_path;
 
 
-	/* first allow developer only overrides to the system path
-	 * these are only used when running blender from source */
-	char cwd[FILE_MAX];
-	char relfolder[FILE_MAX];
-
-	if(folder_name) {
-		if (subfolder_name) {
-			BLI_join_dirfile(relfolder, sizeof(relfolder), folder_name, subfolder_name);
-		} else {
-			BLI_strncpy(relfolder, folder_name, sizeof(relfolder));
-		}
-	}
-	else {
-		relfolder[0]= '\0';
-	}
-
-	/* try CWD/release/folder_name */
-	if(BLI_current_working_dir(cwd, sizeof(cwd))) {
-		if(test_path(targetpath, cwd, "release", relfolder)) {
-			return 1;
-		}
-	}
-
-	/* try EXECUTABLE_DIR/release/folder_name */
-	if(test_path(targetpath, bprogdir, "release", relfolder))
-		return 1;
-	/* end developer overrides */
-
-
 
 	system_path[0] = '\0';
 
