diff -up blender-2.73/source/blender/blenfont/intern/blf_translation.c.droid blender-2.73/source/blender/blenfont/intern/blf_translation.c
--- blender-2.73/source/blender/blenfont/intern/blf_translation.c.droid	2015-01-08 16:18:54.819213786 +0100
+++ blender-2.73/source/blender/blenfont/intern/blf_translation.c	2015-01-08 16:31:38.720228573 +0100
@@ -64,7 +64,7 @@ unsigned char *BLF_get_unifont(int *r_un
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_ttf == NULL) {
-		const char * const fontpath = BKE_appdir_folder_id(BLENDER_DATAFILES, "fonts");
+	  const char * const fontpath = "/usr/share/fonts/blender";
 		if (fontpath) {
 			char unifont_path[1024];
 
@@ -99,7 +99,7 @@ unsigned char *BLF_get_unifont_mono(int
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_mono_ttf == NULL) {
-		const char *fontpath = BKE_appdir_folder_id(BLENDER_DATAFILES, "fonts");
+		const char *fontpath = "/usr/share/fonts/blender";
 		if (fontpath) {
 			char unifont_path[1024];
 
