diff -up blender-2.67/source/blender/blenfont/intern/blf_lang.c.syspath blender-2.67/source/blender/blenfont/intern/blf_lang.c
--- blender-2.67/source/blender/blenfont/intern/blf_lang.c.syspath	2013-05-12 13:12:30.880887495 -0300
+++ blender-2.67/source/blender/blenfont/intern/blf_lang.c	2013-05-12 13:14:29.624620392 -0300
@@ -78,7 +78,7 @@ static void free_locales(void)
 
 static void fill_locales(void)
 {
-	const char * const languages_path = BLI_get_folder(BLENDER_DATAFILES, "locale");
+	const char const languages_path[FILE_MAX] = "/usr/share/blender";
 	char languages[FILE_MAX];
 	LinkNode *lines = NULL, *line;
 	char *str;
diff -up blender-2.67/source/blender/blenlib/intern/path_util.c.syspath blender-2.67/source/blender/blenlib/intern/path_util.c
--- blender-2.67/source/blender/blenlib/intern/path_util.c.syspath	2013-05-12 13:15:12.300538731 -0300
+++ blender-2.67/source/blender/blenlib/intern/path_util.c	2013-05-12 13:15:15.951761716 -0300
@@ -1127,24 +1127,6 @@ static bool get_path_system(char *target
 		relfolder[0] = '\0';
 	}
 
-	/* first allow developer only overrides to the system path
-	 * these are only used when running blender from source */
-
-	/* try CWD/release/folder_name */
-	if (BLI_current_working_dir(cwd, sizeof(cwd))) {
-		if (test_path(targetpath, cwd, "release", relfolder)) {
-			return true;
-		}
-	}
-
-	/* try EXECUTABLE_DIR/release/folder_name */
-	if (test_path(targetpath, bprogdir, "release", relfolder))
-		return true;
-
-	/* end developer overrides */
-
-
-
 	system_path[0] = '\0';
 
 	if (test_env_path(system_path, envvar)) {
