diff -up blender-2.58/intern/ghost/SConscript.ext blender-2.58/intern/ghost/SConscript
--- blender-2.58/intern/ghost/SConscript.ext	2011-06-22 12:16:24.000000000 -0300
+++ blender-2.58/intern/ghost/SConscript	2011-06-23 23:15:26.645830577 -0300
@@ -60,7 +60,7 @@ else:
 if env['BF_GHOST_DEBUG']:
     defs.append('BF_GHOST_DEBUG')
 
-incs = '. ../string #extern/glew/include #source/blender/imbuf #source/blender/makesdna ' + env['BF_OPENGL_INC']
+incs = '. ../string  #source/blender/imbuf #source/blender/makesdna ' + env['BF_OPENGL_INC']
 if window_system in ('win32-vc', 'win32-mingw', 'cygwin', 'linuxcross', 'win64-vc'):
     incs = env['BF_WINTAB_INC'] + ' ' + incs
 
diff -up blender-2.58/source/blender/editors/space_console/SConscript.ext blender-2.58/source/blender/editors/space_console/SConscript
--- blender-2.58/source/blender/editors/space_console/SConscript.ext	2011-06-22 12:26:12.000000000 -0300
+++ blender-2.58/source/blender/editors/space_console/SConscript	2011-06-23 23:15:26.645830577 -0300
@@ -6,7 +6,7 @@ defs = []
 
 incs = [
     '../include',
-    '#extern/glew/include',
+    '',
     '#/intern/guardedalloc',
     '../../makesdna',
     '../../makesrna',
diff -up blender-2.58/source/blender/editors/space_node/SConscript.ext blender-2.58/source/blender/editors/space_node/SConscript
--- blender-2.58/source/blender/editors/space_node/SConscript.ext	2011-06-22 12:28:10.000000000 -0300
+++ blender-2.58/source/blender/editors/space_node/SConscript	2011-06-23 23:15:26.646830363 -0300
@@ -5,7 +5,7 @@ sources = env.Glob('*.c')
 
 incs = '../include ../../blenfont ../../blenlib ../../blenkernel ../../makesdna ../../makesrna ../../imbuf'
 incs += ' ../../nodes ../../render/extern/include ../../blenloader'
-incs += ' ../../windowmanager #intern/guardedalloc #extern/glew/include'
+incs += ' ../../windowmanager #intern/guardedalloc '
 defs = []
 cf = []
 if env['OURPLATFORM'] in ('win32-vc', 'win64-vc'):
diff -up blender-2.58/source/blender/python/SConscript.ext blender-2.58/source/blender/python/SConscript
--- blender-2.58/source/blender/python/SConscript.ext	2011-06-22 12:22:25.000000000 -0300
+++ blender-2.58/source/blender/python/SConscript	2011-06-23 23:15:26.646830363 -0300
@@ -6,7 +6,7 @@ Import ('env')
 
 incs = '. ../editors/include ../makesdna ../makesrna ../blenlib ../blenkernel ../nodes'
 incs += ' ../imbuf ../blenloader ../render/extern/include ../windowmanager'
-incs += ' #intern/guardedalloc #intern/memutil #extern/glew/include'
+incs += ' #intern/guardedalloc #intern/memutil '
 incs += ' #intern/audaspace/intern ' + env['BF_PYTHON_INC']
 
 is_debug = (env['OURPLATFORM'] in ('win32-mingw', 'win32-vc','win64-vc') and env['BF_DEBUG'])
diff -up blender-2.58/source/gameengine/BlenderRoutines/SConscript.ext blender-2.58/source/gameengine/BlenderRoutines/SConscript
--- blender-2.58/source/gameengine/BlenderRoutines/SConscript.ext	2011-06-22 12:20:18.000000000 -0300
+++ blender-2.58/source/gameengine/BlenderRoutines/SConscript	2011-06-23 23:15:26.646830363 -0300
@@ -19,7 +19,7 @@ incs += ' #source/gameengine/SceneGraph
 incs += ' #source/gameengine/Physics/Bullet'
 incs += ' #source/gameengine/Network/LoopBackNetwork'
 incs += ' #source/blender/misc #source/blender/blenloader'
-incs += ' #extern/glew/include #source/blender/gpu'
+incs += '  #source/blender/gpu'
 incs += ' #source/blender/windowmanager'
 
 if env['WITH_BF_FFMPEG']:
diff -up blender-2.58/source/gameengine/GamePlayer/common/SConscript.ext blender-2.58/source/gameengine/GamePlayer/common/SConscript
--- blender-2.58/source/gameengine/GamePlayer/common/SConscript.ext	2011-06-22 12:20:37.000000000 -0300
+++ blender-2.58/source/gameengine/GamePlayer/common/SConscript	2011-06-23 23:15:26.647830149 -0300
@@ -43,7 +43,7 @@ incs = ['.',
         '#source/blender/misc',
         '#source/blender/blenloader',
         '#source/blender/gpu',
-        '#extern/glew/include']
+        '']
 
 defs = [ 'GLEW_STATIC' ]
 
diff -up blender-2.58/source/gameengine/GamePlayer/ghost/SConscript.ext blender-2.58/source/gameengine/GamePlayer/ghost/SConscript
--- blender-2.58/source/gameengine/GamePlayer/ghost/SConscript.ext	2011-06-22 12:20:20.000000000 -0300
+++ blender-2.58/source/gameengine/GamePlayer/ghost/SConscript	2011-06-23 23:15:26.647830149 -0300
@@ -38,7 +38,7 @@ incs = ['.',
         '#source/blender/misc',
         '#source/blender/blenloader',
         '#source/blender/gpu',
-        '#extern/glew/include']
+        '']
 
 defs = [ 'GLEW_STATIC' ]
 
diff -up blender-2.58/source/gameengine/Ketsji/SConscript.ext blender-2.58/source/gameengine/Ketsji/SConscript
--- blender-2.58/source/gameengine/Ketsji/SConscript.ext	2011-06-22 12:21:20.000000000 -0300
+++ blender-2.58/source/gameengine/Ketsji/SConscript	2011-06-23 23:15:26.647830149 -0300
@@ -18,7 +18,7 @@ incs += ' #source/blender/makesdna #sour
 incs += ' #source/gameengine/GameLogic #source/gameengine/Expressions #source/gameengine/Network'
 incs += ' #source/gameengine/SceneGraph #source/gameengine/Physics/common'
 incs += ' #source/gameengine/Physics/Dummy'
-incs += ' #source/blender/misc #source/blender/blenloader #extern/glew/include #source/blender/gpu'
+incs += ' #source/blender/misc #source/blender/blenloader  #source/blender/gpu'
 
 incs += ' ' + env['BF_BULLET_INC']
 incs += ' ' + env['BF_OPENGL_INC']
diff -up blender-2.58/source/gameengine/Physics/Bullet/SConscript.ext blender-2.58/source/gameengine/Physics/Bullet/SConscript
--- blender-2.58/source/gameengine/Physics/Bullet/SConscript.ext	2011-06-22 12:20:13.000000000 -0300
+++ blender-2.58/source/gameengine/Physics/Bullet/SConscript	2011-06-23 23:15:26.648829935 -0300
@@ -6,7 +6,7 @@ sources = 'CcdPhysicsEnvironment.cpp Ccd
 incs = '. ../common'
 incs += ' #intern/string'
 incs += ' #intern/moto/include'
-incs += ' #extern/glew/include'
+incs += ' '
 incs += ' #source/gameengine/Rasterizer'
 incs += ' #source/gameengine/Ketsji'
 incs += ' #source/gameengine/Expressions'
diff -up blender-2.58/source/gameengine/Rasterizer/RAS_OpenGLRasterizer/SConscript.ext blender-2.58/source/gameengine/Rasterizer/RAS_OpenGLRasterizer/SConscript
--- blender-2.58/source/gameengine/Rasterizer/RAS_OpenGLRasterizer/SConscript.ext	2011-06-22 12:21:26.000000000 -0300
+++ blender-2.58/source/gameengine/Rasterizer/RAS_OpenGLRasterizer/SConscript	2011-06-23 23:18:49.543360464 -0300
@@ -6,7 +6,7 @@ sources = env.Glob('*.cpp')
 defs = [ 'GLEW_STATIC' ]
 
 incs = '. #intern/string #intern/moto/include #source/gameengine/Rasterizer #source/gameengine/BlenderRoutines '
-incs += ' #intern/container #source/blender/gpu #extern/glew/include ' + env['BF_OPENGL_INC']
+incs += ' #intern/container #source/blender/gpu ' + env['BF_OPENGL_INC']
 incs += ' #source/blender/gameengine/Ketsji #source/gameengine/SceneGraph #source/blender/makesdna #source/blender/blenkernel'
 incs += ' #intern/guardedalloc #source/blender/blenlib #source/gameengine/BlenderRoutines'
 
diff -up blender-2.58/source/gameengine/Rasterizer/SConscript.ext blender-2.58/source/gameengine/Rasterizer/SConscript
--- blender-2.58/source/gameengine/Rasterizer/SConscript.ext	2011-06-22 12:21:28.000000000 -0300
+++ blender-2.58/source/gameengine/Rasterizer/SConscript	2011-06-23 23:17:13.895852066 -0300
@@ -4,7 +4,7 @@ Import ('env')
 sources = env.Glob('*.cpp')
 
 
-incs = '. #intern/guardedalloc #intern/string #intern/moto/include #intern/container #source/gameengine/BlenderRoutines #extern/glew/include #source/gameengine/Expressions #source/gameengine/SceneGraph #source/blender/blenkernel #source/blender/makesdna'
+incs = '. #intern/guardedalloc #intern/string #intern/moto/include #intern/container #source/gameengine/BlenderRoutines #source/gameengine/Expressions #source/gameengine/SceneGraph #source/blender/blenkernel #source/blender/makesdna'
 
 defs = [ 'GLEW_STATIC' ]
 
diff -up blender-2.58/source/gameengine/VideoTexture/SConscript.ext blender-2.58/source/gameengine/VideoTexture/SConscript
--- blender-2.58/source/gameengine/VideoTexture/SConscript.ext	2011-06-22 12:20:57.000000000 -0300
+++ blender-2.58/source/gameengine/VideoTexture/SConscript	2011-06-23 23:16:32.755666276 -0300
@@ -12,7 +12,7 @@ incs += ' #source/gameengine/BlenderRout
 incs += ' #source/blender/editors/include #source/blender/blenlib #source/blender/blenkernel'
 incs += ' #source/blender/makesdna #source/blender/imbuf #source/blender/python #source/blender/python/generic'
 incs += ' #source/blender/gpu #intern/string #intern/moto/include'
-incs += ' #intern/guardedalloc #intern/container #extern/glew/include'
+incs += ' #intern/guardedalloc #intern/container'
 incs += ' #intern/ffmpeg'
 
 defs = [] 
