diff -Naur blender-2.56-beta-source/extern/bullet2/src/BulletSoftBody/btSoftBody.h blender-2.56-beta-source-new/extern/bullet2/src/BulletSoftBody/btSoftBody.h
--- blender-2.56-beta-source/extern/bullet2/src/BulletSoftBody/btSoftBody.h	2009-09-14 11:45:32.000000000 -0300
+++ blender-2.56-beta-source-new/extern/bullet2/src/BulletSoftBody/btSoftBody.h	2011-02-08 22:32:34.300937158 -0200
@@ -177,6 +177,7 @@
 	/* Material		*/ 
 	struct	Material : Element
 	{
+                Material (){}
 		btScalar				m_kLST;			// Linear stiffness coefficient [0,1]
 		btScalar				m_kAST;			// Area/Angular stiffness coefficient [0,1]
 		btScalar				m_kVST;			// Volume stiffness coefficient [0,1]
@@ -191,6 +192,7 @@
 	/* Node			*/ 
 	struct	Node : Feature
 	{
+                Node (){}
 		btVector3				m_x;			// Position
 		btVector3				m_q;			// Previous step position
 		btVector3				m_v;			// Velocity
@@ -204,6 +206,7 @@
 	/* Link			*/ 
 	struct	Link : Feature
 	{
+                Link (){}
 		Node*					m_n[2];			// Node pointers
 		btScalar				m_rl;			// Rest length		
 		int						m_bbending:1;	// Bending link
@@ -215,6 +218,7 @@
 	/* Face			*/ 
 	struct	Face : Feature
 	{
+                Face (){} 
 		Node*					m_n[3];			// Node pointers
 		btVector3				m_normal;		// Normal
 		btScalar				m_ra;			// Rest area
@@ -255,6 +259,7 @@
 	/* Note			*/ 
 	struct	Note : Element
 	{
+                Note (){}
 		const char*				m_text;			// Text
 		btVector3				m_offset;		// Offset
 		int						m_rank;			// Rank
