diff -uNr blender-2.76.orig/source/blender/blenfont/intern/blf_font_i18n.c blender-2.76/source/blender/blenfont/intern/blf_font_i18n.c
--- blender-2.76.orig/source/blender/blenfont/intern/blf_font_i18n.c	2015-10-10 10:20:56.000000000 +0200
+++ blender-2.76/source/blender/blenfont/intern/blf_font_i18n.c	2015-10-12 16:40:42.225473358 +0200
@@ -57,7 +57,7 @@
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_ttf == NULL) {
-		const char * const fontpath = BKE_appdir_folder_id(BLENDER_DATAFILES, "fonts");
+		const char * const fontpath = "/usr/share/fonts/blender";
 		if (fontpath) {
 			char unifont_path[1024];
 
@@ -92,7 +92,7 @@
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_mono_ttf == NULL) {
-		const char *fontpath = BKE_appdir_folder_id(BLENDER_DATAFILES, "fonts");
+		const char *fontpath = "/usr/share/fonts/blender";
 		if (fontpath) {
 			char unifont_path[1024];
 
