diff -Naur blender-2.57b.orig/source/blender/collada/CMakeLists.txt blender-2.57b/source/blender/collada/CMakeLists.txt
--- blender-2.57b.orig/source/blender/collada/CMakeLists.txt	2011-03-26 04:09:00.000000000 -0500
+++ blender-2.57b/source/blender/collada/CMakeLists.txt	2011-05-25 09:19:23.922922083 -0500
@@ -42,7 +42,7 @@
 
 )
 
-if(APPLE)
+if(APPLE OR UNIX)
 	list(APPEND INC_SYS
 		${OPENCOLLADA_INCLUDE_DIR}/COLLADAStreamWriter
 		${OPENCOLLADA_INCLUDE_DIR}/COLLADABaseUtils
