diff -up blender-2.66/source/gameengine/Physics/Bullet/CcdPhysicsController.cpp.dbg blender-2.66/source/gameengine/Physics/Bullet/CcdPhysicsController.cpp
--- blender-2.66/source/gameengine/Physics/Bullet/CcdPhysicsController.cpp.dbg	2013-02-22 16:29:56.783083071 +0100
+++ blender-2.66/source/gameengine/Physics/Bullet/CcdPhysicsController.cpp	2013-02-22 16:32:35.832668869 +0100
@@ -33,7 +33,7 @@ subject to the following restrictions:
 #include "KX_GameObject.h"
 
 #include "BulletSoftBody/btSoftBody.h"
-#include "BulletSoftBody//btSoftBodyInternals.h"
+#include "BulletSoftBody/btSoftBodyInternals.h"
 #include "BulletSoftBody/btSoftBodyHelpers.h"
 #include "LinearMath/btConvexHull.h"
 #include "BulletCollision/Gimpact/btGImpactShape.h"
