diff -up blender-2.67/source/blender/blenfont/intern/blf_translation.c.droid blender-2.67/source/blender/blenfont/intern/blf_translation.c
--- blender-2.67/source/blender/blenfont/intern/blf_translation.c.droid	2013-03-20 15:42:09.000000000 -0300
+++ blender-2.67/source/blender/blenfont/intern/blf_translation.c	2013-05-12 13:41:09.093426130 -0300
@@ -59,18 +59,11 @@ unsigned char *BLF_get_unifont(int *unif
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_ttf == NULL) {
-		const char * const fontpath = BLI_get_folder(BLENDER_DATAFILES, "fonts");
-		if (fontpath) {
-			char unifont_path[1024];
-
-			BLI_snprintf(unifont_path, sizeof(unifont_path), "%s/%s", fontpath, unifont_filename);
-
-			unifont_ttf = (unsigned char *)BLI_file_ungzip_to_mem(unifont_path, &unifont_size);
-		}
-		else {
-			printf("%s: 'fonts' data path not found for international font, continuing\n", __func__);
-		}
-	}
+	  const char * const fontpath = "/usr/share/fonts/google-droid";
+	  char unifont_path[1024];
+  	  BLI_snprintf(unifont_path, sizeof(unifont_path), "%s/%s", fontpath, unifont_filename);
+	  unifont_ttf = (unsigned char *)BLI_file_to_mem(unifont_path, &unifont_size);
+}
 
 	*unifont_size_r = unifont_size;
 
diff -up blender-2.67/source/blender/blenlib/BLI_fileops.h.droid blender-2.67/source/blender/blenlib/BLI_fileops.h
--- blender-2.67/source/blender/blenlib/BLI_fileops.h.droid	2013-03-26 04:29:01.000000000 -0300
+++ blender-2.67/source/blender/blenlib/BLI_fileops.h	2013-05-12 13:40:36.754301873 -0300
@@ -81,6 +81,8 @@ bool   BLI_file_touch(const char *file);
 int    BLI_file_gzip(const char *from, const char *to);
 char  *BLI_file_ungzip_to_mem(const char *from_file, int *size_r);
 
+char *BLI_file_to_mem(const char *from_file, int *size_r);
+
 size_t BLI_file_descriptor_size(int file);
 size_t BLI_file_size(const char *file);
 
diff -up blender-2.67/source/blender/blenlib/intern/fileops.c.droid blender-2.67/source/blender/blenlib/intern/fileops.c
--- blender-2.67/source/blender/blenlib/intern/fileops.c.droid	2013-05-01 14:43:33.000000000 -0300
+++ blender-2.67/source/blender/blenlib/intern/fileops.c	2013-05-12 13:40:36.754301873 -0300
@@ -68,6 +68,31 @@
 
 #include "MEM_sys_types.h" // for intptr_t support
 
+char *BLI_file_to_mem(const char *from_file, int *size_r)
+{
+        int file;
+	int size = 0;
+	char *mem = NULL;
+
+	file = BLI_open(from_file, O_RDONLY, 0);
+
+	size = BLI_file_descriptor_size(file);
+       
+	if (size == 0) {
+	  close (file);
+	  return 0;
+	}
+      
+	mem = MEM_callocN(size, "BLI_ungzip_to_mem");
+		
+	read(file, mem, size);
+	
+	close (file);
+
+	*size_r = size;
+
+	return mem;
+}
 
 /* gzip the file in from and write it to "to". 
  * return -1 if zlib fails, -2 if the originating file does not exist
