diff -up blender-2.61/source/blender/imbuf/intern/jp2.c.stdbool blender-2.61/source/blender/imbuf/intern/jp2.c
--- blender-2.61/source/blender/imbuf/intern/jp2.c.stdbool	2011-12-13 13:50:45.000000000 -0600
+++ blender-2.61/source/blender/imbuf/intern/jp2.c	2012-02-11 22:24:26.078025463 -0600
@@ -36,6 +36,7 @@
 #include "IMB_allocimbuf.h"
 #include "IMB_filetype.h"
 
+#include <stdbool.h>
 #include "openjpeg.h"
 
 #define JP2_FILEHEADER_SIZE 14
