Index: release/plugins/sequence/dnr.c
===================================================================
--- release/plugins/sequence/dnr.c.orig	2008-11-12 23:16:53.000000000 +0200
+++ release/plugins/sequence/dnr.c	2010-10-28 14:56:30.000000000 +0200
@@ -110,9 +110,12 @@ static void doit(unsigned char * src_, u
 	unsigned char * dst = dst_;
 
 	while (count--) {
-		*dst = table[(*src++ << 8) | *dst]; dst++;
-		*dst = table[(*src++ << 8) | *dst]; dst++;
-		*dst = table[(*src++ << 8) | *dst]; dst++;
+		*dst = table[(*src++ << 8) | *dst];
+		dst++;
+		*dst = table[(*src++ << 8) | *dst];
+		dst++;
+		*dst = table[(*src++ << 8) | *dst];
+		dst++;
 		*dst++ = *src++;
 	}
 
