diff -up blender-2.72/intern/guardedalloc/intern/mallocn.c.size_t blender-2.72/intern/guardedalloc/intern/mallocn.c
--- blender-2.72/intern/guardedalloc/intern/mallocn.c.size_t	2014-10-11 10:30:15.630102814 -0400
+++ blender-2.72/intern/guardedalloc/intern/mallocn.c	2014-10-11 11:33:02.370102814 -0400
@@ -53,11 +53,11 @@ void (*MEM_set_error_callback)(void (*fu
 bool (*MEM_check_memory_integrity)(void) = MEM_lockfree_check_memory_integrity;
 void (*MEM_set_lock_callback)(void (*lock)(void), void (*unlock)(void)) = MEM_lockfree_set_lock_callback;
 void (*MEM_set_memory_debug)(void) = MEM_lockfree_set_memory_debug;
-uintptr_t (*MEM_get_memory_in_use)(void) = MEM_lockfree_get_memory_in_use;
-uintptr_t (*MEM_get_mapped_memory_in_use)(void) = MEM_lockfree_get_mapped_memory_in_use;
+size_t (*MEM_get_memory_in_use)(void) = MEM_lockfree_get_memory_in_use;
+size_t (*MEM_get_mapped_memory_in_use)(void) = MEM_lockfree_get_mapped_memory_in_use;
 unsigned int (*MEM_get_memory_blocks_in_use)(void) = MEM_lockfree_get_memory_blocks_in_use;
 void (*MEM_reset_peak_memory)(void) = MEM_lockfree_reset_peak_memory;
-uintptr_t (*MEM_get_peak_memory)(void) = MEM_lockfree_get_peak_memory;
+size_t (*MEM_get_peak_memory)(void) = MEM_lockfree_get_peak_memory;
 
 #ifndef NDEBUG
 const char *(*MEM_name_ptr)(void *vmemh) = MEM_lockfree_name_ptr;
diff -up blender-2.72/intern/guardedalloc/intern/mallocn_guarded_impl.c.size_t blender-2.72/intern/guardedalloc/intern/mallocn_guarded_impl.c
--- blender-2.72/intern/guardedalloc/intern/mallocn_guarded_impl.c.size_t	2014-10-11 11:28:19.470102814 -0400
+++ blender-2.72/intern/guardedalloc/intern/mallocn_guarded_impl.c	2014-10-11 11:30:04.110102814 -0400
@@ -1151,9 +1151,9 @@ void MEM_guarded_reset_peak_memory(void)
 	mem_unlock_thread();
 }
 
-uintptr_t MEM_guarded_get_memory_in_use(void)
+size_t MEM_guarded_get_memory_in_use(void)
 {
-	uintptr_t _mem_in_use;
+	size_t _mem_in_use;
 
 	mem_lock_thread();
 	_mem_in_use = mem_in_use;
@@ -1162,9 +1162,9 @@ uintptr_t MEM_guarded_get_memory_in_use(
 	return _mem_in_use;
 }
 
-uintptr_t MEM_guarded_get_mapped_memory_in_use(void)
+size_t MEM_guarded_get_mapped_memory_in_use(void)
 {
-	uintptr_t _mmap_in_use;
+	size_t _mmap_in_use;
 
 	mem_lock_thread();
 	_mmap_in_use = mmap_in_use;
diff -up blender-2.72/intern/guardedalloc/intern/mallocn_intern.h.size_t blender-2.72/intern/guardedalloc/intern/mallocn_intern.h
--- blender-2.72/intern/guardedalloc/intern/mallocn_intern.h.size_t	2014-10-11 10:30:58.000102814 -0400
+++ blender-2.72/intern/guardedalloc/intern/mallocn_intern.h	2014-10-11 11:31:07.570102814 -0400
@@ -132,11 +132,11 @@ void MEM_lockfree_set_error_callback(voi
 bool MEM_lockfree_check_memory_integrity(void);
 void MEM_lockfree_set_lock_callback(void (*lock)(void), void (*unlock)(void));
 void MEM_lockfree_set_memory_debug(void);
-uintptr_t MEM_lockfree_get_memory_in_use(void);
-uintptr_t MEM_lockfree_get_mapped_memory_in_use(void);
+size_t MEM_lockfree_get_memory_in_use(void);
+size_t MEM_lockfree_get_mapped_memory_in_use(void);
 unsigned int MEM_lockfree_get_memory_blocks_in_use(void);
 void MEM_lockfree_reset_peak_memory(void);
-uintptr_t MEM_lockfree_get_peak_memory(void) ATTR_WARN_UNUSED_RESULT;
+size_t MEM_lockfree_get_peak_memory(void) ATTR_WARN_UNUSED_RESULT;
 #ifndef NDEBUG
 const char *MEM_lockfree_name_ptr(void *vmemh);
 #endif
@@ -159,11 +159,11 @@ void MEM_guarded_set_error_callback(void
 bool MEM_guarded_check_memory_integrity(void);
 void MEM_guarded_set_lock_callback(void (*lock)(void), void (*unlock)(void));
 void MEM_guarded_set_memory_debug(void);
-uintptr_t MEM_guarded_get_memory_in_use(void);
-uintptr_t MEM_guarded_get_mapped_memory_in_use(void);
+size_t MEM_guarded_get_memory_in_use(void);
+size_t MEM_guarded_get_mapped_memory_in_use(void);
 unsigned int MEM_guarded_get_memory_blocks_in_use(void);
 void MEM_guarded_reset_peak_memory(void);
-uintptr_t MEM_guarded_get_peak_memory(void) ATTR_WARN_UNUSED_RESULT;
+size_t MEM_guarded_get_peak_memory(void) ATTR_WARN_UNUSED_RESULT;
 #ifndef NDEBUG
 const char *MEM_guarded_name_ptr(void *vmemh);
 #endif
diff -up blender-2.72/intern/guardedalloc/intern/mallocn_lockfree_impl.c.size_t blender-2.72/intern/guardedalloc/intern/mallocn_lockfree_impl.c
--- blender-2.72/intern/guardedalloc/intern/mallocn_lockfree_impl.c.size_t	2014-10-11 10:31:22.310102814 -0400
+++ blender-2.72/intern/guardedalloc/intern/mallocn_lockfree_impl.c	2014-10-11 11:31:35.850102814 -0400
@@ -442,12 +442,12 @@ void MEM_lockfree_set_memory_debug(void)
 	malloc_debug_memset = true;
 }
 
-uintptr_t MEM_lockfree_get_memory_in_use(void)
+size_t MEM_lockfree_get_memory_in_use(void)
 {
 	return mem_in_use;
 }
 
-uintptr_t MEM_lockfree_get_mapped_memory_in_use(void)
+size_t MEM_lockfree_get_mapped_memory_in_use(void)
 {
 	return mmap_in_use;
 }
@@ -463,7 +463,7 @@ void MEM_lockfree_reset_peak_memory(void
 	peak_mem = 0;
 }
 
-uintptr_t MEM_lockfree_get_peak_memory(void)
+size_t MEM_lockfree_get_peak_memory(void)
 {
 	return peak_mem;
 }
diff -up blender-2.72/intern/guardedalloc/MEM_guardedalloc.h.size_t blender-2.72/intern/guardedalloc/MEM_guardedalloc.h
--- blender-2.72/intern/guardedalloc/MEM_guardedalloc.h.size_t	2014-10-11 11:27:43.290102814 -0400
+++ blender-2.72/intern/guardedalloc/MEM_guardedalloc.h	2014-10-11 11:28:05.440102814 -0400
@@ -165,9 +165,9 @@ extern "C" {
 	 * Memory usage stats
 	 * - MEM_get_memory_in_use is all memory
 	 * - MEM_get_mapped_memory_in_use is a subset of all memory */
-	extern uintptr_t (*MEM_get_memory_in_use)(void);
+	extern size_t (*MEM_get_memory_in_use)(void);
 	/** Get mapped memory usage. */
-	extern uintptr_t (*MEM_get_mapped_memory_in_use)(void);
+	extern size_t (*MEM_get_mapped_memory_in_use)(void);
 	/** Get amount of memory blocks in use. */
 	extern unsigned int (*MEM_get_memory_blocks_in_use)(void);
 
