diff -up blender-2.49b/source/blender/blenlib/intern/storage.c.sf blender-2.49b/source/blender/blenlib/intern/storage.c
--- blender-2.49b/source/blender/blenlib/intern/storage.c.sf	2010-03-28 10:14:15.140027561 +0200
+++ blender-2.49b/source/blender/blenlib/intern/storage.c	2010-03-28 10:19:55.630278553 +0200
@@ -381,7 +381,8 @@ void BLI_adddirstrings()
 			struct passwd *pwuser;
 			pwuser = getpwuid(files[num].s.st_uid);
 			if ( pwuser ) {
-			strcpy(files[num].owner, pwuser->pw_name);
+			  strncpy(files[num].owner, pwuser->pw_name, sizeof(files[num].owner)-1);
+                          files[num].owner[sizeof(files[num].owner)-1] = '\0';
 			} else {
 				sprintf(files[num].owner, "%d", files[num].s.st_uid);
             }
