diff -up blender-2.64/source/blender/blenfont/intern/blf_lang.c.syspath blender-2.64/source/blender/blenfont/intern/blf_lang.c
--- blender-2.64/source/blender/blenfont/intern/blf_lang.c.syspath	2012-10-03 16:43:53.000000000 +0200
+++ blender-2.64/source/blender/blenfont/intern/blf_lang.c	2012-10-03 20:57:24.878000025 +0200
@@ -117,13 +117,7 @@ void BLF_lang_init(void)
 
 	BLI_strncpy(global_encoding_name, SYSTEM_ENCODING_DEFAULT, sizeof(global_encoding_name));
 
-	if (messagepath) {
-		BLI_strncpy(global_messagepath, messagepath, sizeof(global_messagepath));
-	}
-	else {
-		printf("%s: 'locale' data path for translations not found, continuing\n", __func__);
-		global_messagepath[0] = '\0';
-	}
+	*global_messagepath = "/usr/share/locale/";
 }
 
 /* Get LANG/LANGUAGE environment variable. */
diff -up blender-2.64/source/blender/blenlib/intern/path_util.c.syspath blender-2.64/source/blender/blenlib/intern/path_util.c
--- blender-2.64/source/blender/blenlib/intern/path_util.c.syspath	2012-10-03 16:43:53.000000000 +0200
+++ blender-2.64/source/blender/blenlib/intern/path_util.c	2012-10-03 20:51:37.886000008 +0200
@@ -983,38 +983,6 @@ static int get_path_system(char *targetp
 	char system_path[FILE_MAX];
 	const char *system_base_path;
 
-
-	/* first allow developer only overrides to the system path
-	 * these are only used when running blender from source */
-	char cwd[FILE_MAX];
-	char relfolder[FILE_MAX];
-
-	if (folder_name) {
-		if (subfolder_name) {
-			BLI_join_dirfile(relfolder, sizeof(relfolder), folder_name, subfolder_name);
-		}
-		else {
-			BLI_strncpy(relfolder, folder_name, sizeof(relfolder));
-		}
-	}
-	else {
-		relfolder[0] = '\0';
-	}
-
-	/* try CWD/release/folder_name */
-	if (BLI_current_working_dir(cwd, sizeof(cwd))) {
-		if (test_path(targetpath, cwd, "release", relfolder)) {
-			return 1;
-		}
-	}
-
-	/* try EXECUTABLE_DIR/release/folder_name */
-	if (test_path(targetpath, bprogdir, "release", relfolder))
-		return 1;
-	/* end developer overrides */
-
-
-
 	system_path[0] = '\0';
 
 	if (test_env_path(system_path, envvar)) {
