=== modified file 'configure.ac'
--- a/configure.ac	2011-09-20 04:32:43 +0000
+++ b/configure.ac	2011-10-15 08:28:24 +0000
@@ -196,8 +196,7 @@
 AM_PATH_PYTHON([$PYTHON_REQUIRED])
 AM_CHECK_PYTHON_HEADERS(,[AC_MSG_ERROR(could not find Python headers)])
 
-PKG_CHECK_MODULES([INDICATE_PYTHON], [python-$PYTHON_VERSION >= $PYTHON_REQUIRED
-                                      pygtk-2.0 >= $PYGTK_REQUIRED
+PKG_CHECK_MODULES([INDICATE_PYTHON], [pygtk-2.0 >= $PYGTK_REQUIRED
                                       pygobject-2.0 >= $PYGOBJECT_REQUIRED],
                                      [have_python=yes]
 )

