From 8828fffd93c6b19bcfd7626ddc472aa8f055d034 Mon Sep 17 00:00:00 2001
From: Behdad Esfahbod <behdad@behdad.org>
Date: Mon, 14 Jan 2013 14:36:38 -0600
Subject: [PATCH] Copy all values from pattern to font if the font doesn't have
 the element

Bug 59376 - FcFontRenderPrepare enhancement
---
 src/fcmatch.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/fcmatch.c b/src/fcmatch.c
index 6778967..3d15605 100644
--- a/src/fcmatch.c
+++ b/src/fcmatch.c
@@ -549,8 +549,9 @@ FcFontRenderPrepare (FcConfig	    *config,
 	fe = FcPatternObjectFindElt (font, pe->object);
 	if (!fe)
 	{
-	    v = FcValueCanonicalize(&FcPatternEltValues(pe)->value);
-	    FcPatternObjectAdd (new, pe->object, v, FcTrue);
+	    FcPatternObjectListAdd (new, pe->object,
+				    FcValueListDuplicate (FcPatternEltValues(pe)),
+				    FcFalse);
 	}
     }
 
-- 
1.8.1.1

