--- fontconfig-2.10.0/fonts.conf.in.orig	2012-05-31 23:26:10.000000000 -0300
+++ fontconfig-2.10.0/fonts.conf.in	2012-07-17 20:14:27.698088257 -0300
@@ -23,12 +23,23 @@
 
 <!-- Font directory list -->
 
-	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+	<dir>/usr/share/fonts/OTF</dir>
+	<dir>/usr/share/fonts/TTF</dir>
+	<dir>/usr/share/fonts/Type1</dir>
+	<dir>/usr/share/fonts/Speedo</dir>
+	<dir>/usr/share/fonts/cyrillic</dir>
+	<dir>/usr/share/fonts/misc</dir>
 	<dir prefix="xdg">fonts</dir>
 	<!-- the following element will be removed in the future -->
 	<dir>~/.fonts</dir>
 
+<!--    Commented out font directories:
+These are not indexed by default, as fontconfig then seems
+to prefer bitmapped fonts in some cases...
+	<dir>/usr/share/fonts/100dpi</dir>
+	<dir>/usr/share/fonts/75dpi</dir>
+-->
+
 <!--
   Accept deprecated 'mono' alias, replacing it with 'monospace'
 -->
