--- fontconfig-2.9.0/conf.d/45-latin.conf.orig	2012-03-10 03:24:45.000000000 -0300
+++ fontconfig-2.9.0/conf.d/45-latin.conf	2012-04-21 03:56:10.547245190 -0300
@@ -10,15 +10,15 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
+		<family>DejaVu Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Serif</family>
+		<family>Liberation Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>Liberation Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
@@ -49,15 +49,15 @@
   Sans-serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Sans</family>
+		<family>DejaVu Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Sans</family>
+		<family>Liberation Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>Liberation Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
@@ -91,10 +91,6 @@
 <!--
   Monospace faces
  -->
- 	<alias>
-		<family>Bitstream Vera Sans Mono</family>
-		<default><family>monospace</family></default>
-	</alias>
 	<alias>
 		<family>DejaVu Sans Mono</family>
 		<default><family>monospace</family></default>
@@ -103,6 +99,10 @@
 		<family>Liberation Mono</family>
 		<default><family>monospace</family></default>
 	</alias>
+ 	<alias>
+		<family>Bitstream Vera Sans Mono</family>
+		<default><family>monospace</family></default>
+	</alias>
 	<alias>
 		<family>Inconsolata</family>
 		<default><family>monospace</family></default>
