--- fontconfig-2.12.0/conf.d/45-latin.conf.orig	2016-05-25 01:05:54.000000000 -0300
+++ fontconfig-2.12.0/conf.d/45-latin.conf	2016-12-25 12:22:36.246981023 -0200
@@ -10,19 +10,19 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
+		<family>DejaVu Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>Cambria</family>
+		<family>Liberation Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>Constantia</family>
+		<family>Cambria</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Serif</family>
+		<family>Constantia</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
@@ -38,7 +38,7 @@
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
-		<family>Liberation Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<default><family>serif</family></default>
 	</alias>
 	<alias>
@@ -81,6 +81,10 @@
   Sans-serif faces
  -->
 	<alias>
+		<family>DejaVu Sans</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>Albany AMT</family>
 		<default><family>sans-serif</family></default>
 	</alias>
@@ -97,7 +101,7 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>Bitstream Vera Sans</family>
+		<family>Liberation Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
@@ -121,10 +125,6 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Sans</family>
-		<default><family>sans-serif</family></default>
-	</alias>
-	<alias>
 		<family>Helvetica</family>
 		<default><family>sans-serif</family></default>
 	</alias>
@@ -133,7 +133,7 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
-		<family>Liberation Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
@@ -172,11 +172,11 @@
   Monospace faces
  -->
 	<alias>
-		<family>Andale Mono</family>
+		<family>DejaVu Sans Mono</family>
 		<default><family>monospace</family></default>
 	</alias>
- 	<alias>
-		<family>Bitstream Vera Sans Mono</family>
+	<alias>
+		<family>Andale Mono</family>
 		<default><family>monospace</family></default>
 	</alias>
 	<alias>
@@ -200,11 +200,11 @@
 		<default><family>monospace</family></default>
 	</alias>
 	<alias>
-		<family>DejaVu Sans Mono</family>
+		<family>Fixedsys</family>
 		<default><family>monospace</family></default>
 	</alias>
-	<alias>
-		<family>Fixedsys</family>
+ 	<alias>
+		<family>Bitstream Vera Sans Mono</family>
 		<default><family>monospace</family></default>
 	</alias>
 	<alias>
