diff -up fontconfig-2.8.0/fc-cache/fc-cache.c.jx fontconfig-2.8.0/fc-cache/fc-cache.c
--- fontconfig-2.8.0/fc-cache/fc-cache.c.jx	2009-11-16 12:24:52.000000000 -0500
+++ fontconfig-2.8.0/fc-cache/fc-cache.c	2010-06-24 15:10:04.000000000 -0400
@@ -461,7 +461,7 @@ main (int argc, char **argv)
      */
     FcConfigDestroy (config);
     FcFini ();
-    sleep (2);
+    sleep (1);
     if (verbose)
 	printf ("%s: %s\n", argv[0], ret ? "failed" : "succeeded");
     return ret;
