--- source/code/physfs_stream.hpp.orig	2012-10-11 13:19:39.829960679 -0300
+++ source/code/physfs_stream.hpp	2012-10-11 13:20:52.994371857 -0300
@@ -196,7 +196,7 @@ namespace PhysFS
                 }
 
                 char_type* xend = (static_cast<size_t> (objectsRead) == BUF_SIZE) ? &*_buf.end() : &_buf[objectsRead];
-                setg(&*_buf.begin(), &*_buf.begin(), xend);
+                this->setg(&*_buf.begin(), &*_buf.begin(), xend);
 
                 return traits_type::to_int_type(_buf.front());
             }
@@ -212,7 +212,7 @@ namespace PhysFS
                 }
 
                 // the seek invalidated the buffer
-                setg(&*_buf.begin(), &*_buf.begin(), &*_buf.begin());
+                this->setg(&*_buf.begin(), &*_buf.begin(), &*_buf.begin());
                 return pos;
             }
 
@@ -277,7 +277,7 @@ namespace PhysFS
                 }
 
                 char_type* xend = (static_cast<size_t> (res) == BUF_SIZE) ? &*_buf.end() : &_buf[res];
-                setp(&*_buf.begin(), xend);
+                this->setp(&*_buf.begin(), xend);
                 return 0;
             }
 
