Index: glib2.0-2.27.2/gio/gunixmounts.c
===================================================================
--- glib2.0-2.27.2.orig/gio/gunixmounts.c	2010-10-17 05:31:18.000000000 +0200
+++ glib2.0-2.27.2/gio/gunixmounts.c	2010-11-01 10:55:55.817621566 +0100
@@ -227,6 +227,8 @@
     "/home",
     "/lib",
     "/lib64",
+    "/live/cow",
+    "/live/image",
     "/media",
     "/mnt",
     "/opt",
@@ -237,7 +239,12 @@
     "/usr",
     "/usr/local",
     "/var",
+    "/var/crash",
+    "/var/local",
+    "/var/log",
     "/var/log/audit", /* https://bugzilla.redhat.com/show_bug.cgi?id=333041 */
+    "/var/mail",
+    "/var/run",
     "/var/tmp",       /* https://bugzilla.redhat.com/show_bug.cgi?id=335241 */
     "/proc",
     "/sbin",
