diff -up kbd-1.15.2/po/Makefile.in.in.orig kbd-1.15.2/po/Makefile.in.in
--- kbd-1.15.2/po/Makefile.in.in.orig	2010-04-10 23:34:12.000000000 +0200
+++ kbd-1.15.2/po/Makefile.in.in	2011-01-05 12:34:33.182143154 +0100
@@ -23,7 +23,7 @@ VPATH = @srcdir@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
 datadir = @datadir@
-localedir = $(datadir)/locale
+localedir = @localedir@
 gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
