diff -up kbd-1.15.2/configure.orig kbd-1.15.2/configure
--- kbd-1.15.2/configure.orig	2011-01-05 13:30:25.826147638 +0100
+++ kbd-1.15.2/configure	2011-01-05 13:31:15.089526084 +0100
@@ -7369,8 +7369,9 @@ fi
 
 
 case $host_cpu in
-	i?86*) RESIZECONS_PROGS=yes ;;
-	*)     RESIZECONS_PROGS=no ;;
+	i?86*)   RESIZECONS_PROGS=yes ;;
+	x86_64*) RESIZECONS_PROGS=yes ;;
+	*)       RESIZECONS_PROGS=no ;;
 esac
  if test "$RESIZECONS_PROGS" = "yes"; then
   RESIZECONS_PROGS_TRUE=
