diff -up kbd-1.15.3/man/man1/dumpkeys.1.in.orig kbd-1.15.3/man/man1/dumpkeys.1.in
--- kbd-1.15.3/man/man1/dumpkeys.1.in.orig	2011-08-23 13:06:16.905892768 +0200
+++ kbd-1.15.3/man/man1/dumpkeys.1.in	2011-08-23 13:15:18.264853435 +0200
@@ -156,6 +156,14 @@ output of
 .BR dumpkeys .
 .LP
 .TP
+.B \-1 \-\-separate-lines
+This forces
+.B dumpkeys
+to write one line per (modifier,keycode) pair. It prefixes the word
+.I plain
+for plain keycodes.
+.LP
+.TP
 .B \-\-funcs-only
 When this option is given,
 .B dumpkeys
