--- emerald/src/Makefile.am	2011-01-19 01:29:13.232159500 +0000
+++ emerald/src/Makefile.am.new	2011-01-19 01:36:14.964906582 +0000
@@ -1,5 +1,5 @@
 ## Process this file with automake to produce Makefile.in
-emerald_LDADD   = @EMERALD_LIBS@ $(top_builddir)/libengine/libemeraldengine.la
+emerald_LDADD   = @EMERALD_LIBS@ $(top_builddir)/libengine/libemeraldengine.la -lm -ldl
 emerald_SOURCES = main.c \
 			   engine_loader.c
 emerald_program = emerald
