--- emerald-0.8.4.orig/src/Makefile.in	2010-02-18 20:01:14.000000000 +0000
+++ emerald-0.8.4/src/Makefile.in	2010-02-18 20:02:46.000000000 +0000
@@ -208,7 +208,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-emerald_LDADD = @EMERALD_LIBS@ $(top_builddir)/libengine/libemeraldengine.la
+emerald_LDADD = @EMERALD_LIBS@ $(top_builddir)/libengine/libemeraldengine.la -lm -ldl
 emerald_SOURCES = main.c \
 			   engine_loader.c
 
