diff -ur kdevplatform-1.2.3/CMakeLists.txt kdevplatform-1.2.3-no-rpath/CMakeLists.txt
--- kdevplatform-1.2.3/CMakeLists.txt	2011-06-24 21:03:56.000000000 +0200
+++ kdevplatform-1.2.3-no-rpath/CMakeLists.txt	2011-12-24 13:27:02.000000000 +0100
@@ -29,7 +29,6 @@
 if("${_isSystemLibDir}" STREQUAL "-1")
   set(CMAKE_INSTALL_RPATH "${LIB_INSTALL_DIR}")
 endif("${_isSystemLibDir}" STREQUAL "-1")
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 set(KOMPARE_FOUND FALSE)
 #macro_optional_find_package(Kompare)
