diff -ur cfitsio/configure cfitsio-fixed/configure
--- cfitsio/configure	2011-01-10 12:03:51.000000000 -0500
+++ cfitsio-fixed/configure	2011-04-11 11:33:54.384980620 -0400
@@ -3778,7 +3778,8 @@
     SHLIB_SUFFIX=".sl"
     ;;
   lnx)
-    SHLIB_LD=":"
+    SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
+    SHLIB_SUFFIX=".so.0"
     ;;
   osf)
     SHLIB_LD="ld -shared -expect_unresolved '*'"
@@ -3799,7 +3800,7 @@
 # Darwin uses gcc (=cc), but needs different flags (see above)
 # if test "x$GCC" = xyes; then
 if test "x$GCC" = xyes && test "x$EXT" != xdarwin && test "x$EXT" != xcygwin; then
-  SHLIB_LD="$CC -shared"
+  SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
   lhea_shlib_cflags='-fPIC'
 fi
 if test "x$lhea_shlib_cflags" != x; then
