diff -ur cfitsio/longnam.h cfitsio.new/longnam.h
--- cfitsio/longnam.h	2013-04-04 21:45:41.000000000 +0200
+++ cfitsio.new/longnam.h	2013-07-04 11:05:16.423095453 +0200
@@ -19,7 +19,8 @@
    that was used to build the CFITSIO library is compatible with the version
    as included when compiling the application program
 */
-#define fits_open_file(A, B, C, D)  ffopentest( CFITSIO_SONAME, A, B, C, D)
+/*#define fits_open_file(A, B, C, D)  ffopentest( CFITSIO_SONAME, A, B, C, D)*/
+#define fits_open_file  ffopen
 
 #define fits_open_data      ffdopn
 #define fits_open_table     fftopn
