diff -up Linux-PAM-1.1.6/configure.in.links Linux-PAM-1.1.6/configure.in
--- Linux-PAM-1.1.6/configure.in.links	2013-04-24 13:13:36.000000000 +0200
+++ Linux-PAM-1.1.6/configure.in	2013-08-07 14:08:03.818055990 +0200
@@ -548,9 +548,9 @@ JH_CHECK_XML_CATALOG([-//OASIS//DTD DocB
 JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl],
                 [DocBook XSL Stylesheets], [], enable_docu=no)
 
-AC_PATH_PROG([BROWSER], [w3m])
+AC_PATH_PROG([BROWSER], [links])
 if test ! -z "$BROWSER"; then
-     BROWSER="$BROWSER -T text/html -dump"
+     BROWSER="$BROWSER -no-numbering -no-references -dump"
 else
      enable_docu=no
 fi
