diff -up Linux-PAM-1.2.0/configure.ac.links Linux-PAM-1.2.0/configure.ac
--- Linux-PAM-1.2.0/configure.ac.links	2015-05-15 16:00:03.406295266 +0200
+++ Linux-PAM-1.2.0/configure.ac	2015-05-15 16:00:03.425295702 +0200
@@ -564,9 +564,9 @@ JH_CHECK_XML_CATALOG([-//OASIS//DTD DocB
 JH_CHECK_XML_CATALOG([http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl],
                 [DocBook XSL Stylesheets], [], enable_docu=no)
 
-AC_PATH_PROG([BROWSER], [w3m])
+AC_PATH_PROG([BROWSER], [elinks])
 if test ! -z "$BROWSER"; then
-     BROWSER="$BROWSER -T text/html -dump"
+     BROWSER="$BROWSER -no-numbering -no-references -dump"
 else
      enable_docu=no
 fi
