diff -up aqbanking-5.0.0/aqbanking-config.in.in.pkgconfig aqbanking-5.0.0/aqbanking-config.in.in
--- aqbanking-5.0.0/aqbanking-config.in.in.pkgconfig	2010-08-05 17:53:41.000000000 -0300
+++ aqbanking-5.0.0/aqbanking-config.in.in	2010-08-30 13:16:44.837648782 -0300
@@ -1,51 +1,44 @@
 # $Id$
 # Author of this file: Martin Preuss<martin@libchipcard.de>
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-datarootdir=@datarootdir@
-includedir=@includedir@
-datadir=@datadir@
-
 result=""
 
 for d in $*; do
     case $d in
 	--includes)
-	    result="$result @aqbanking_includes@ @gwenhywfar_includes@"
+	    result="$result $(pkg-config --cflags aqbanking)"
 	    ;;
 	--libraries)
-	    result="$result @aqbanking_ldflags@ @aqbanking_libs@"
+	    result="$result $(pkg-config --libs aqbanking)"
 	    ;;
 	--libraries++)
 	    result="$result "
 	    ;;
 	--plugins)
-	    result="$result @aqbanking_plugindir@"
+	    result="$result $(pkg-config --variable=plugindir aqbanking)"
 	    ;;
 	--data)
-	    result="$result @aqbanking_pkgdatadir@"
+	    result="$result $(pkg-config --variable=pkgdatadir aqbanking)"
 	    ;;
 
         --has-qbanking)
             result=@with_qbanking@
             ;;
 	--qbanking-libraries)
-	    result="$result @aqbanking_ldflags@ @qbanking_libs@"
+	    result="$result $(pkg-config --variable=qbanking_libs aqbanking)"
 	    ;;
 	--qbanking-plugins)
-	    result="$result @qbanking_plugindir@"
+	    result="$result $(pkg-config --variable=qbanking_plugindir aqbanking)"
 	    ;;
 	--qbanking-helpdir)
-	    result="$result @qbanking_helpdir@"
+	    result="$result $(pkg-config --variable=qbanking_helpdir aqbanking)"
 	    ;;
 
         --has-aqhbci)
             result=@with_aqhbci@
             ;;
 	--aqhbci-libraries)
-	    result="$result @aqbanking_ldflags@ @aqhbci_libs@"
+	    result="$result $(pkg-config --variable=aqhbci_libs aqbanking)"
 	    ;;
 
 	--vmajor)
diff -up aqbanking-5.0.0/aqbanking.pc.in.pkgconfig aqbanking-5.0.0/aqbanking.pc.in
--- aqbanking-5.0.0/aqbanking.pc.in.pkgconfig	2010-08-05 17:53:41.000000000 -0300
+++ aqbanking-5.0.0/aqbanking.pc.in	2010-08-30 13:17:24.996022251 -0300
@@ -27,6 +27,8 @@ aqhbci_datadir=@aqhbci_data@
 qbanking_includes=@qbanking_includes@
 qbanking_libs=@qbanking_libs@
 qbanking_datadir=@qbanking_data@
+qbanking_plugindir=@qbanking_plugindir@
+qbanking_helpdir=@qbanking_helpdir@
 q4banking_includes=@q4banking_includes@
 q4banking_libs=@q4banking_libs@
 q4banking_datadir=@q4banking_data@
