diff -up aqbanking-5.0.2/aqbanking-config.in.in.foo aqbanking-5.0.2/aqbanking-config.in.in
--- aqbanking-5.0.2/aqbanking-config.in.in.foo	2011-02-11 16:46:23.610084051 -0500
+++ aqbanking-5.0.2/aqbanking-config.in.in	2011-02-11 16:49:01.181084055 -0500
@@ -1,51 +1,31 @@
 # $Id$
 # Author of this file: Martin Preuss<martin@libchipcard.de>
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-datarootdir=@datarootdir@
-includedir=@includedir@
-datadir=@datadir@
-
 result=""
 
 for d in $*; do
     case $d in
 	--includes)
-	    result="$result @aqbanking_includes@ @gwenhywfar_includes@"
+	    result="$result $(pkg-config --cflags aqbanking)"
 	    ;;
 	--libraries)
-	    result="$result @aqbanking_ldflags@ @aqbanking_libs@"
+	    result="$result $(pkg-config --libs aqbanking)"
 	    ;;
 	--libraries++)
 	    result="$result "
 	    ;;
 	--plugins)
-	    result="$result @aqbanking_plugindir@"
+	    result="$result $(pkg-config --variable=plugindir aqbanking)"
 	    ;;
 	--data)
-	    result="$result @aqbanking_pkgdatadir@"
-	    ;;
-
-        --has-qbanking)
-            result=@with_qbanking@
-            ;;
-	--qbanking-libraries)
-	    result="$result @aqbanking_ldflags@ @qbanking_libs@"
-	    ;;
-	--qbanking-plugins)
-	    result="$result @qbanking_plugindir@"
-	    ;;
-	--qbanking-helpdir)
-	    result="$result @qbanking_helpdir@"
+	    result="$result $(pkg-config --variable=pkgdatadir aqbanking)"
 	    ;;
 
         --has-aqhbci)
             result=@with_aqhbci@
             ;;
 	--aqhbci-libraries)
-	    result="$result @aqbanking_ldflags@ @aqhbci_libs@"
+	    result="$result $(pkg-config --variable=aqhbci_libs aqbanking)"
 	    ;;
 
 	--vmajor)
