--- wordlist/scowl/speller/make-aspell-dict	2010-06-06 20:53:16.000000000 +0100
+++ wordlist/scowl/speller/make-aspell-dict	2010-06-06 20:53:51.000000000 +0100
@@ -14,22 +14,22 @@
 
 echo base
 
-cat $FINAL/english-*.{10,20,35,40,50,55,60} $FINAL/special-* \
+cat $FINAL/english-*.{10,20,35,40,50,55,60,70} $FINAL/special-* \
   | sort -u > en.wl0
-cat en.wl0 $FINAL/american-*.{10,20,35,40,50,55,60} \
+cat en.wl0 $FINAL/american-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en_US-w_accents.twl
-cat en.wl0 $FINAL/british-*.{10,20,35,40,50,55,60} \
+cat en.wl0 $FINAL/british-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en_GB-ise-w_accents.twl
-cat en.wl0 $FINAL/british_z-*.{10,20,35,40,50,55,60} \
+cat en.wl0 $FINAL/british_z-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en_GB-ize-w_accents.twl
-cat en.wl0 $FINAL/canadian-*.{10,20,35,40,50,55,60} \
+cat en.wl0 $FINAL/canadian-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en_CA-w_accents.twl
 
-cat $FINAL/variant_0-*.{10,20,35,40,50,55,60} \
+cat $FINAL/variant_0-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en-variant_0.wl0
-cat en-variant_0.wl0 $FINAL/variant_1-*.{10,20,35,40,50,55,60} \
+cat en-variant_0.wl0 $FINAL/variant_1-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en-variant_1.wl0
-cat en-variant_1.wl0 $FINAL/variant_2-*.{10,20,35,40,50,55,60} \
+cat en-variant_1.wl0 $FINAL/variant_2-*.{10,20,35,40,50,55,60,70} \
   | sort -u > en-variant_2.wl0
 
 echo deaccent
