--- wordlist.orig/scowl/r/special/proper-names	2008-02-08 11:53:27.000000000 +0000
+++ wordlist/scowl/r/special/proper-names	2008-02-08 12:06:00.000000000 +0000
@@ -6,6 +6,8 @@
 Biden
 Bluetooth
 Bugzilla
+Camino
+ChatZilla
 Debian
 ECMAScript
 Emacs
@@ -13,16 +15,21 @@
 Facebook
 GNU
 Gentoo
+Haskell
 Hunspell
 ISO
 Ispell
 LyX
+Mandriva
 Mozilla
+MySpell
 Netflix
 PayPal
+SeaMonkey
 Roku
 SUSE
 Slackware
+Sunbird
 Thunderbird
 Twitter
 Ubuntu
--- wordlist.orig/scowl/speller/en.aff	2008-02-08 20:28:24.000000000 +0000
+++ wordlist/scowl/speller/en.aff	2008-02-08 20:28:45.000000000 +0000
@@ -110,13 +110,17 @@
 SFX L Y 1
 SFX L   0     ment       .
 
-REP 88
+SFX i N 1
+SFX i   us    i          us
+
+REP 90
 REP a ei
 REP ei a
 REP a ey
 REP ey a
 REP ai ie
 REP ie ai
+REP alot a_lot
 REP are air
 REP are ear
 REP are eir
@@ -199,3 +203,4 @@
 REP shun tion
 REP shun sion
 REP shun cion
+REP sitted sat
