diff -up kipi-plugins-1.4.0/cmake/modules/FindGdk.cmake.gtk3 kipi-plugins-1.4.0/cmake/modules/FindGdk.cmake
--- kipi-plugins-1.4.0/cmake/modules/FindGdk.cmake.gtk3	2010-08-22 03:04:10.000000000 -0500
+++ kipi-plugins-1.4.0/cmake/modules/FindGdk.cmake	2010-08-26 10:25:59.030277097 -0500
@@ -23,7 +23,7 @@ else (GDK_INCLUDE_DIR AND GDK_LIBRARIES)
     set(GDK_DEFINITIONS ${_GDKCflags})
   endif(NOT WIN32)
 
-  FIND_PATH(GDK_INCLUDE_DIR gdk-pixbuf/gdk-pixbuf.h /usr/include/gtk-2.0
+  FIND_PATH(GDK_INCLUDE_DIR gdk-pixbuf/gdk-pixbuf.h /usr/include/gtk-2.0 /usr/include/gdk-pixbuf-2.0
     ${_GDKIncDir}
   )
   
