--- ./include/freetype/config/ftoption.h.orig	2007-04-21 17:42:39.000000000 +0200
+++ ./include/freetype/config/ftoption.h	2007-04-21 17:43:39.000000000 +0200
@@ -456,7 +456,7 @@
   /*   Do not #undef this macro here, since the build system might         */
   /*   define it for certain configurations only.                          */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
