--- ./include/freetype/config/ftoption.h.orig	2007-01-05 10:18:45.000000000 -0600
+++ ./include/freetype/config/ftoption.h	2007-04-21 15:07:04.000000000 -0500
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
