--- freetype-2.6.1/builds/unix/freetype-config.in.multilib	2015-09-25 02:58:57.000000000 -0300
+++ freetype-2.6.1/builds/unix/freetype-config.in	2015-10-04 10:54:15.220303632 -0300
@@ -12,11 +12,11 @@
 LC_ALL=C
 export LC_ALL
 
-prefix="%prefix%"
-exec_prefix="%exec_prefix%"
+prefix=`pkg-config --variable prefix freetype2`
+exec_prefix=`pkg-config --variable exec_prefix freetype2`
 exec_prefix_set="no"
-includedir="%includedir%"
-libdir="%libdir%"
+includedir=`pkg-config --variable includedir freetype2`
+libdir=`pkg-config --variable libdir freetype2`
 
 usage()
 {
