diff -u -r freetype-2.1.7.orig/include/freetype/freetype.h freetype-2.1.7/include/freetype/freetype.h
--- freetype-2.1.7.orig/include/freetype.h	2003-11-09 00:38:13.000000000 -0800
+++ freetype-2.1.7/include/freetype.h	2004-06-05 13:43:55.000000000 -0700
@@ -16,13 +16,20 @@
 /***************************************************************************/
 
 
-#ifndef FT_FREETYPE_H
-#error "`ft2build.h' hasn't been included yet!"
-#error "Please always use macros to include FreeType header files."
-#error "Example:"
-#error "  #include <ft2build.h>"
-#error "  #include FT_FREETYPE_H"
-#endif
+/***************************************************************************/
+/*                                                                         */
+/* If you think breaking almost 100% of all source that links with         */
+/* freetype (anything using a configure file for sure) is a good idea,     */
+/* then feel free to uncomment this block.                                 */
+/*                                                                         */
+/* #ifndef FT_FREETYPE_H                                                   */
+/* #error "`ft2build.h' hasn't been included yet!"                         */
+/* #error "Please always use macros to include FreeType header files."     */
+/* #error "Example:"                                                       */
+/* #error "  #include <ft2build.h>"                                        */
+/* #error "  #include FT_FREETYPE_H"                                       */
+/* #endif                                                                  */
+/***************************************************************************/
 
 
 #ifndef __FREETYPE_H__
