diff -up ./mozilla/nsprpub/config/nspr.pc.in.extralibs ./mozilla/nsprpub/config/nspr.pc.in
--- ./mozilla/nsprpub/config/nspr.pc.in.extralibs	2012-06-16 11:02:11.846447511 -0700
+++ ./mozilla/nsprpub/config/nspr.pc.in	2012-06-16 11:05:30.578619641 -0700
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: NSPR
 Description: The Netscape Portable Runtime
 Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
-Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@
+Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@ @OS_LIBS@
 Cflags: -I@includedir@
