Index: ./mozilla/nsprpub/config/nspr-config.in
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/config/nspr-config.in,v
retrieving revision 1.8
diff -u -p -r1.8 nspr-config.in
--- ./mozilla/nsprpub/config/nspr-config.in	6 Mar 2012 13:13:39 -0000	1.8
+++ ./mozilla/nsprpub/config/nspr-config.in	22 Jun 2012 02:36:56 -0000
@@ -102,7 +102,7 @@ if test -z "$includedir"; then
     includedir=@includedir@
 fi
 if test -z "$libdir"; then
-    libdir=@libdir@
+    libdir=`pkg-config --variable=libdir nspr`
 fi
 
 if test "$echo_prefix" = "yes"; then
@@ -136,12 +136,12 @@ if test "$echo_libs" = "yes"; then
       if test -n "$lib_nspr"; then
 	libdirs="$libdirs -lnspr${major_version}"
       fi
-      os_ldflags="@LDFLAGS@"
+      os_ldflags=`pkg-config --variable=ldflags nspr`
       for i in $os_ldflags ; do
 	if echo $i | grep \^-L >/dev/null; then
 	  libdirs="$libdirs $i"
         fi
       done
-      echo $libdirs @OS_LIBS@
+      echo $libdirs `pkg-config --variable=os_libs nspr`
 fi      
 
Index: ./mozilla/nsprpub/config/nspr.pc.in
===================================================================
RCS file: /cvsroot/mozilla/nsprpub/config/nspr.pc.in,v
retrieving revision 1.1
diff -u -p -r1.1 nspr.pc.in
--- ./mozilla/nsprpub/config/nspr.pc.in	18 May 2012 22:26:17 -0000	1.1
+++ ./mozilla/nsprpub/config/nspr.pc.in	22 Jun 2012 02:36:56 -0000
@@ -6,5 +8,5 @@ includedir=@includedir@
 Name: NSPR
 Description: The Netscape Portable Runtime
 Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
-Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@
+Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@ @OS_LIBS@
 Cflags: -I@includedir@
