diff -up ./mozilla/nsprpub/config/nspr-config-pc.in.x123 ./mozilla/nsprpub/config/nspr-config-pc.in
--- ./mozilla/nsprpub/config/nspr-config-pc.in.x123	2012-02-29 16:00:54.108691393 -0800
+++ ./mozilla/nsprpub/config/nspr-config-pc.in	2012-02-29 16:00:54.111691416 -0800
@@ -98,7 +98,7 @@ if test -z "$includedir"; then
     includedir=@includedir@
 fi
 if test -z "$libdir"; then
-    libdir=@libdir@
+    libdir=`pkg-config --variable=libdir nspr`
 fi
 
 if test "$echo_prefix" = "yes"; then
@@ -132,12 +132,12 @@ if test "$echo_libs" = "yes"; then
       if test -n "$lib_nspr"; then
 	libdirs="$libdirs -lnspr${major_version}"
       fi
-      os_ldflags="@LDFLAGS@"
+      os_ldflags=`pkg-config --variable=ldflags nspr`
       for i in $os_ldflags ; do
 	if echo $i | grep \^-L >/dev/null; then
 	  libdirs="$libdirs $i"
         fi
       done
-      echo $libdirs @OS_LIBS@
+      echo $libdirs `pkg-config --variable=os_libs nspr`
 fi      
 
diff -up ./mozilla/nsprpub/configure.in.x123 ./mozilla/nsprpub/configure.in
--- ./mozilla/nsprpub/configure.in.x123	2012-02-14 11:06:00.000000000 -0800
+++ ./mozilla/nsprpub/configure.in	2012-02-29 16:00:54.113691431 -0800
@@ -3357,6 +3357,8 @@ MAKEFILES="
     config/nsprincl.mk
     config/nsprincl.sh
     config/nspr-config
+    config/nspr-config-pc
+    config/nspr-config-vars
     lib/Makefile
     lib/ds/Makefile
     lib/libc/Makefile
diff -up ./mozilla/nsprpub/configure.x123 ./mozilla/nsprpub/configure
--- ./mozilla/nsprpub/configure.x123	2012-02-29 16:03:11.006689991 -0800
+++ ./mozilla/nsprpub/configure	2012-02-29 16:05:15.646599889 -0800
@@ -6694,6 +6694,8 @@ MAKEFILES="
     config/nsprincl.mk
     config/nsprincl.sh
     config/nspr-config
+    config/nspr-config-pc
+    config/nspr-config-vars
     lib/Makefile
     lib/ds/Makefile
     lib/libc/Makefile
