From 1901a1e9e5282c1deca2f5f98476e2c7e9243d35 Mon Sep 17 00:00:00 2001
From: Jean-Louis Dupond <jean-louis@dupond.be>
Date: Mon, 10 Sep 2012 14:12:11 +0200
Subject: [PATCH] Closes #63: fix typo when fitting window

---
 remmina/src/remmina_connection_window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/remmina/src/remmina_connection_window.c b/remmina/src/remmina_connection_window.c
index 7ff1246..2ba8193 100644
--- a/remmina/src/remmina_connection_window.c
+++ b/remmina/src/remmina_connection_window.c
@@ -436,7 +436,7 @@ static gboolean remmina_connection_holder_toolbar_autofit_restore(RemminaConnect
 		remmina_connection_holder_get_desktop_size(cnnhld, &width, &height, TRUE);
 		gtk_widget_get_allocation(priv->notebook, &na);
 		gtk_widget_get_allocation(cnnobj->scrolled_container, &ca);
-		gtk_widget_get_allocation(priv->toolbar, &ca);
+		gtk_widget_get_allocation(priv->toolbar, &ta);
 		gtk_window_resize(GTK_WINDOW(cnnhld->cnnwin), MAX(1, width + na.width - ca.width),
 				MAX(1, height + ta.height + na.height - ca.height));
 		gtk_container_check_resize(GTK_CONTAINER(cnnhld->cnnwin));
-- 
1.8.1.6

