Index: remmina-1.0.0/CMakeLists.txt
===================================================================
--- remmina-1.0.0.orig/CMakeLists.txt
+++ remmina-1.0.0/CMakeLists.txt
@@ -80,7 +80,7 @@ include_directories(remmina/include)
 set(REMMINA_VERSION "0.9.99.1")
 set(REMMINA_DATADIR "${CMAKE_INSTALL_PREFIX}/share")
 set(REMMINA_LOCALEDIR "${CMAKE_INSTALL_PREFIX}/share/locale")
-set(REMMINA_PLUGINDIR "${CMAKE_INSTALL_PREFIX}/lib/remmina/plugins")
+set(REMMINA_PLUGINDIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/remmina/plugins")
 
 find_suggested_package(PTHREAD)
 if(PTHREAD_FOUND)
