Index: gdb-7.5.50.20130310/gdb/solib-svr4.c
===================================================================
--- gdb-7.5.50.20130310.orig/gdb/solib-svr4.c	2013-03-10 16:37:49.000000000 +0100
+++ gdb-7.5.50.20130310/gdb/solib-svr4.c	2013-03-10 16:43:45.498585091 +0100
@@ -1225,14 +1225,27 @@ svr4_read_so_list (CORE_ADDR lm, struct
 	}
 
       {
-	struct elf_build_id *build_id;
+	struct elf_build_id *build_id = NULL;
 
 	strncpy (new->so_original_name, buffer, SO_NAME_MAX_PATH_SIZE - 1);
 	new->so_original_name[SO_NAME_MAX_PATH_SIZE - 1] = '\0';
 	/* May get overwritten below.  */
 	strcpy (new->so_name, new->so_original_name);
 
-	build_id = build_id_addr_get (new->lm_info->l_ld);
+	/* In the case the main executable was found according to its build-id
+	   (from a core file) prevent loading a different build of a library
+	   with accidentally the same SO_NAME.
+
+	   It suppresses bogus backtraces (and prints "??" there instead) if
+	   the on-disk files no longer match the running program version.
+
+	   If the main executable was not loaded according to its build-id do
+	   not do any build-id checking of the libraries.  There may be missing
+	   build-ids dumped in the core file and we would map all the libraries
+	   to the only existing file loaded that time - the executable.  */
+	if (symfile_objfile != NULL
+	    && (symfile_objfile->flags & OBJF_BUILD_ID_CORE_LOADED) != 0)
+	  build_id = build_id_addr_get (new->lm_info->l_ld);
 	if (build_id != NULL)
 	  {
 	    char *name, *build_id_filename;
@@ -1247,23 +1260,7 @@ svr4_read_so_list (CORE_ADDR lm, struct
 		xfree (name);
 	      }
 	    else
-	      {
-		debug_print_missing (new->so_name, build_id_filename);
-
-		/* In the case the main executable was found according to
-		   its build-id (from a core file) prevent loading
-		   a different build of a library with accidentally the
-		   same SO_NAME.
-
-		   It suppresses bogus backtraces (and prints "??" there
-		   instead) if the on-disk files no longer match the
-		   running program version.  */
-
-		if (symfile_objfile != NULL
-		    && (symfile_objfile->flags
-			& OBJF_BUILD_ID_CORE_LOADED) != 0)
-		  new->so_name[0] = 0;
-	      }
+	      debug_print_missing (new->so_name, build_id_filename);
 
 	    xfree (build_id_filename);
 	    xfree (build_id);
