Index: gdb-7.3.50.20110722/gdb/infrun.c
===================================================================
--- gdb-7.3.50.20110722.orig/gdb/infrun.c	2011-07-22 19:12:56.000000000 +0200
+++ gdb-7.3.50.20110722/gdb/infrun.c	2011-07-22 19:17:06.000000000 +0200
@@ -1549,7 +1549,7 @@ static const char *scheduler_enums[] = {
   schedlock_step,
   NULL
 };
-static const char *scheduler_mode = schedlock_off;
+static const char *scheduler_mode = schedlock_step;
 static void
 show_scheduler_mode (struct ui_file *file, int from_tty,
 		     struct cmd_list_element *c, const char *value)
Index: gdb-7.3.50.20110722/gdb/testsuite/gdb.mi/mi-console.exp
===================================================================
--- gdb-7.3.50.20110722.orig/gdb/testsuite/gdb.mi/mi-console.exp	2011-01-01 16:33:47.000000000 +0100
+++ gdb-7.3.50.20110722/gdb/testsuite/gdb.mi/mi-console.exp	2011-07-22 19:17:06.000000000 +0200
@@ -47,6 +47,9 @@ if  { [gdb_compile "${srcdir}/${subdir}/
 
 mi_run_to_main
 
+# thread-id=\"all\" vs. thread-id=\"1\" below:
+mi_gdb_test "210-gdb-set scheduler-locking off" "210\\^done" "set scheduler-locking off"
+
 # Next over the hello() call which will produce lots of output
 mi_gdb_test "220-exec-next" \
 	    "220\\^running(\r\n\\*running,thread-id=\"all\")?" \
Index: gdb-7.3.50.20110722/gdb/testsuite/gdb.mi/mi2-console.exp
===================================================================
--- gdb-7.3.50.20110722.orig/gdb/testsuite/gdb.mi/mi2-console.exp	2011-06-23 11:40:50.000000000 +0200
+++ gdb-7.3.50.20110722/gdb/testsuite/gdb.mi/mi2-console.exp	2011-07-22 19:17:27.000000000 +0200
@@ -47,6 +47,9 @@ if  { [gdb_compile "${srcdir}/${subdir}/
 
 mi_run_to_main
 
+# thread-id=\"all\" vs. thread-id=\"1\" below:
+mi_gdb_test "210-gdb-set scheduler-locking off" "210\\^done" "set scheduler-locking off"
+
 # Next over the hello() call which will produce lots of output
 mi_gdb_test "220-exec-next" "220\\^running(\r\n)?(\\*running,thread-id=\"all\")?" \
 	"Started step over hello"
Index: gdb-7.3.50.20110722/gdb/testsuite/gdb.mi/mi-cli.exp
===================================================================
--- gdb-7.3.50.20110722.orig/gdb/testsuite/gdb.mi/mi-cli.exp	2011-04-27 12:17:38.000000000 +0200
+++ gdb-7.3.50.20110722/gdb/testsuite/gdb.mi/mi-cli.exp	2011-07-22 19:17:06.000000000 +0200
@@ -176,7 +176,7 @@ mi_execute_to "exec-continue" "breakpoin
 # Test that the token is output even for CLI commands
 # Also test that *stopped includes frame information.
 mi_gdb_test "34 next" \
-    ".*34\\\^running.*\\*running,thread-id=\"all\"" \
+    ".*34\\\^running.*\\*running,thread-id=\"1\"" \
     "34 next: run"
 
 if {!$async} {
