--- ./gdb/value.c-orig	2015-02-11 15:17:58.413241779 +0100
+++ ./gdb/value.c	2015-02-11 15:19:42.068716949 +0100
@@ -3074,7 +3074,7 @@ value_primitive_field (struct value *arg
 	}
 
       if (!TYPE_DATA_LOCATION (type)
-          || !TYPE_DATA_LOCATION_KIND (type) == PROP_CONST)
+          || TYPE_DATA_LOCATION_KIND (type) != PROP_CONST)
 	v->offset = (value_offset (arg1) + offset
 		     + value_embedded_offset (arg1));
     }
