diff -dup -rup gdb-7.4.50.20120603-orig/gdb/configure gdb-7.4.50.20120603/gdb/configure
--- gdb-7.4.50.20120603-orig/gdb/configure	2012-06-06 19:08:32.824824699 +0200
+++ gdb-7.4.50.20120603/gdb/configure	2012-06-06 19:12:12.346992423 +0200
@@ -2706,7 +2706,7 @@ fi
 # Provide more thorough testing by -lmcheck.
 # Set it to 'true' for development snapshots, 'false' for releases or
 # pre-releases.
-development=true
+development=false
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
diff -dup -rup gdb-7.4.50.20120603-orig/gdb/configure.ac gdb-7.4.50.20120603/gdb/configure.ac
--- gdb-7.4.50.20120603-orig/gdb/configure.ac	2012-06-06 19:08:32.817824693 +0200
+++ gdb-7.4.50.20120603/gdb/configure.ac	2012-06-06 19:12:05.545987227 +0200
@@ -26,7 +26,7 @@ AM_MAINTAINER_MODE
 # Provide more thorough testing by -lmcheck.
 # Set it to 'true' for development snapshots, 'false' for releases or
 # pre-releases.
-development=true
+development=false
 
 AC_PROG_CC
 AC_USE_SYSTEM_EXTENSIONS
