Index: gdb-7.5.50.20130118/gdb/gcore.c
===================================================================
--- gdb-7.5.50.20130118.orig/gdb/gcore.c	2013-01-18 23:50:56.698573186 +0100
+++ gdb-7.5.50.20130118/gdb/gcore.c	2013-01-18 23:52:29.636705040 +0100
@@ -549,8 +549,14 @@ gcore_copy_callback (bfd *obfd, asection
       if (size > total_size)
 	size = total_size;
 
+      /* Warn if read error occurs except if we were trying to read the
+	 first page for ia64.  The first page is marked readable, but it cannot
+	 be read.  */
       if (target_read_memory (bfd_section_vma (obfd, osec) + offset,
-			      memhunk, size) != 0)
+			      memhunk, size) != 0
+	  && (strcmp (gdbarch_bfd_arch_info (target_gdbarch ())->arch_name,
+		      "ia64")
+	      || bfd_section_vma (obfd, osec) != 0))
 	{
 	  warning (_("Memory read failed for corefile "
 		     "section, %s bytes at %s."),
