https://bugzilla.redhat.com/show_bug.cgi?id=1085576

--- gdb-7.8.2/gdb/python/lib/gdb/command/type_printers.py-orig	2015-01-15 11:58:12.000000000 +0100
+++ gdb-7.8.2/gdb/python/lib/gdb/command/type_printers.py	2015-06-26 15:33:43.972460415 +0200
@@ -47,7 +47,7 @@ class InfoTypePrinter(gdb.Command):
         sep = ''
         for objfile in gdb.objfiles():
             if objfile.type_printers:
-                print ("%sType printers for %s:" % (sep, objfile.name))
+                print ("%sType printers for %s:" % (sep, objfile.filename))
                 self.list_type_printers(objfile.type_printers)
                 sep = '\n'
         if gdb.current_progspace().type_printers:
