--- gdb-7.1.90.20100806/gdb/infcall.c.orig	2010-08-10 00:34:10.000000000 +0200
+++ gdb-7.1.90.20100806/gdb/infcall.c	2010-08-10 00:36:03.000000000 +0200
@@ -284,7 +284,7 @@ find_function_addr (struct value *functi
   struct type *ftype = check_typedef (value_type (function));
   struct gdbarch *gdbarch = get_type_arch (ftype);
   struct type *value_type = NULL;
-  CORE_ADDR funaddr;
+  CORE_ADDR funaddr = 0;	/* GCC false -fprofile-use warning.  */
 
   /* If it's a member function, just look at the function
      part of it.  */
