--- fox-1.6.49/adie/Adie.cpp.syspath	2013-04-08 03:51:42.000000000 +0900
+++ fox-1.6.49/adie/Adie.cpp	2013-04-30 08:04:45.000000000 +0900
@@ -122,7 +122,10 @@
   associations->setIconPath(reg().readStringEntry("SETTINGS","iconpath",FXIconDict::defaultIconPath));
 
   // Hunt for the syntax file
-  syntaxfile=FXPath::search(FXSystem::getExecPath(),"Adie.stx");
+  //syntaxfile=FXPath::search(FXSystem::getExecPath(),"Adie.stx");
+  FXString syntaxpath("/usr/share/fox");
+  syntaxpath += FXSystem::getExecPath();
+  syntaxfile=FXPath::search(syntaxpath,"Adie.stx");
 
   // Load syntax file
   if(!syntaxfile.empty()){
