--- fox-1.6.49/chart/Makefile.am.linkage	2013-04-08 03:51:42.000000000 +0900
+++ fox-1.6.49/chart/Makefile.am	2013-05-03 18:28:29.000000000 +0900
@@ -8,7 +8,8 @@
 
 lib_LTLIBRARIES = libCHART-1.6.la
 
-libCHART_1_6_la_LIBADD  = $(X_LIBS) $(X_BASE_LIBS) $(X_EXTRA_LIBS)
+#libCHART_1_6_la_LIBADD  = $(X_LIBS) $(X_BASE_LIBS) $(X_EXTRA_LIBS)
+libCHART_1_6_la_LIBADD  = $(top_builddir)/src/libFOX-1.6.la
 
 libCHART_1_6_la_LDFLAGS = \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)    \
