From 296538ef2754df0c6e704ac172854816d4ff5cd2 Mon Sep 17 00:00:00 2001
From: Caolan McNamara <caolanm@fedoraproject.org>
Date: Mon, 2 Mar 2009 12:21:34 +0000
Subject: [PATCH 1/3] Modify GetOnePixel usage to build against current
 ImageMagick api

---
 input-magick.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/input-magick.c b/input-magick.c
index 51c6409..bc36b8f 100644
--- a/input-magick.c
+++ b/input-magick.c
@@ -83,7 +83,11 @@ at_bitmap_type input_magick_reader(at_string filename,
 
   for(j=0,runcount=0,point=0;j<image->rows;j++)
     for(i=0;i<image->columns;i++) {
+#if (MagickLibVersion < 0x0645) || (MagickLibVersion >= 0x0649)
       p=GetOnePixel(image,i,j);
+#else
+      GetOnePixel(image,i,j,pixel);
+#endif
       AT_BITMAP_BITS(bitmap)[point++]=pixel->red; /* if gray: red=green=blue */
       if(np==3) {
         AT_BITMAP_BITS(bitmap)[point++]=pixel->green;
-- 
1.7.11.2

