--- autotrace-config.in.multilib	2002-11-08 16:18:31.000000000 -0200
+++ autotrace-config.in	2011-11-10 20:46:22.909552628 -0200
@@ -204,8 +204,8 @@ fi
 ##
 #dummy because this should always be selected
 
-af_cflags="$af_cflags -I@includedir@"
-af_libs="-L@libdir@ -lautotrace @LIBPNG_LDFLAGS@ @MAGICK_LDFLAGS@ @LIBSWF_LDFLAGS@ @LIBPSTOEDIT_LIBS@ $af_libs"
+af_cflags="-I@includedir@"
+af_libs="-lautotrace"
 
 
 
