--- configure.ac.orig	2014-04-23 18:33:17.188291464 -0300
+++ configure.ac	2014-04-23 18:33:35.424434897 -0300
@@ -107,7 +107,7 @@ AC_ARG_ENABLE(journal,
         [enable_journal="yes"]
 )
 if test "x$enable_journal" = "xyes"; then
-	PKG_CHECK_MODULES([LIBSYSTEMD_JOURNAL], [libsystemd-journal >= 197])
+	PKG_CHECK_MODULES([LIBSYSTEMD_JOURNAL], [libsystemd >= 209])
 fi
 AM_CONDITIONAL(ENABLE_JOURNAL, test x$enable_journal = xyes)
 if test "$enable_journal" = "yes"; then
