--- SOAPpy/Client.py~	2006-12-08 19:01:59.000000000 -0800
+++ SOAPpy/Client.py	2006-12-08 19:02:23.000000000 -0800
@@ -1,3 +1,5 @@
+from __future__ import nested_scopes
+
 """
 ################################################################################
 #
@@ -43,8 +45,6 @@
 ident = '$Id: Client.py,v 1.20 2004/04/10 04:22:52 irjudson Exp $'
 from version import __version__
 
-from __future__ import nested_scopes
-
 #import xml.sax
 import urllib
 from types import *
--- SOAPpy/GSIServer.py~	2006-12-08 19:13:42.000000000 -0800
+++ SOAPpy/GSIServer.py	2006-12-08 19:13:56.000000000 -0800
@@ -1,3 +1,5 @@
+from __future__ import nested_scopes
+
 """
 GSIServer - Contributed by Ivan R. Judson <judson@mcs.anl.gov>
 
@@ -46,8 +48,6 @@
 ident = '$Id: GSIServer.py,v 1.4 2004/02/04 03:21:02 irjudson Exp $'
 from version import __version__
 
-from __future__ import nested_scopes
-
 #import xml.sax
 import re
 import socket
--- SOAPpy/NS.py~	2006-12-08 19:14:02.000000000 -0800
+++ SOAPpy/NS.py	2006-12-08 19:14:15.000000000 -0800
@@ -1,3 +1,5 @@
+from __future__ import nested_scopes
+
 """
 ################################################################################
 #
@@ -40,8 +42,6 @@
 ################################################################################
 """
 
-from __future__ import nested_scopes
-
 ident = '$Id: NS.py,v 1.3 2004/01/31 04:20:05 warnes Exp $'
 from version import __version__
 
--- SOAPpy/Server.py~	2006-12-08 19:14:20.000000000 -0800
+++ SOAPpy/Server.py	2006-12-08 19:14:33.000000000 -0800
@@ -1,3 +1,5 @@
+from __future__ import nested_scopes
+
 """
 ################################################################################
 #
@@ -43,8 +45,6 @@
 ident = '$Id: Server.py,v 1.20 2004/04/28 21:47:10 warnes Exp $'
 from version import __version__
 
-from __future__ import nested_scopes
-
 #import xml.sax
 import re
 import socket
--- SOAPpy/Types.py~	2006-12-08 19:14:39.000000000 -0800
+++ SOAPpy/Types.py	2006-12-08 19:14:52.000000000 -0800
@@ -1,3 +1,5 @@
+from __future__ import nested_scopes
+
 """
 ################################################################################
 # Copyright (c) 2003, Pfizer
@@ -36,8 +38,6 @@
 ident = '$Id: Types.py,v 1.17 2004/09/11 03:03:33 warnes Exp $'
 from version import __version__
 
-from __future__ import nested_scopes
-
 import UserList
 import base64
 import cgi
