diff -up M2Crypto-0.20/SWIG/_m2crypto.i.gcc_macros M2Crypto-0.20/SWIG/_m2crypto.i
--- M2Crypto-0.20/SWIG/_m2crypto.i.gcc_macros	2009-08-11 05:05:22.000000000 +0200
+++ M2Crypto-0.20/SWIG/_m2crypto.i	2009-08-11 13:24:52.943706097 +0200
@@ -4,6 +4,7 @@
  * Portions created by Open Source Applications Foundation (OSAF) are
  * Copyright (C) 2004-2006 OSAF. All Rights Reserved.
  */
+%import "gcc_macros.h"
 
 %module(threads=1) _m2crypto
 /* We really don't need threadblock (PyGILState_Ensure() etc.) anywhere.
