--- npth-config.in.orig	2013-03-15 16:25:51.000000000 -0300
+++ npth-config.in	2014-11-06 19:11:03.207602739 -0200
@@ -19,7 +19,7 @@ LANG=C
 
 # NPTH's own cflags and libs
 cflags="-I@includedir@ @NPTH_CONFIG_CFLAGS@"
-libs="-L@libdir@ @NPTH_CONFIG_LIBS@"
+libs="@NPTH_CONFIG_LIBS@"
 
 output=""
 
@@ -110,10 +110,7 @@ while test $# -gt 0; do
             for i in $libs ; do
               skip=no
               case $i in
-                  -L/usr/lib|-L/lib)
-                      skip=yes
-                      ;;
-                  -L*|-l*)
+                  -l*)
                       for j in $result ; do
                           if test x"$j" = x"$i" ; then
                               skip=yes
