diff -up cups-1.6.3/config-scripts/cups-compiler.m4.full-relro cups-1.6.3/config-scripts/cups-compiler.m4
--- cups-1.6.3/config-scripts/cups-compiler.m4.full-relro	2013-08-15 12:40:51.965747766 +0100
+++ cups-1.6.3/config-scripts/cups-compiler.m4	2013-08-15 12:42:15.083630332 +0100
@@ -171,7 +171,7 @@ if test -n "$GCC"; then
 			# The -z relro option is provided by the Linux linker command to
 			# make relocatable data read-only.
 			if test x$enable_relro = xyes; then
-				RELROFLAGS="-Wl,-z,relro"
+				RELROFLAGS="-Wl,-z,relro,-z,now"
 			fi
 			;;
 	esac
