diff -up cups-1.7b1/config-scripts/cups-ssl.m4.no-gcry cups-1.7b1/config-scripts/cups-ssl.m4
--- cups-1.7b1/config-scripts/cups-ssl.m4.no-gcry	2013-06-24 12:14:44.514018538 +0100
+++ cups-1.7b1/config-scripts/cups-ssl.m4	2013-06-24 12:15:35.805243614 +0100
@@ -66,7 +66,6 @@ if test x$enable_ssl != xno; then
     dnl Then look for GNU TLS...
     if test $have_ssl = 0 -a "x$enable_gnutls" != "xno" -a "x$PKGCONFIG" != x; then
     	AC_PATH_PROG(LIBGNUTLSCONFIG,libgnutls-config)
-    	AC_PATH_PROG(LIBGCRYPTCONFIG,libgcrypt-config)
 	if $PKGCONFIG --exists gnutls; then
 	    have_ssl=1
 	    SSLLIBS=`$PKGCONFIG --libs gnutls`
@@ -84,14 +83,6 @@ if test x$enable_ssl != xno; then
 	if test $have_ssl = 1; then
 	    CUPS_SERVERCERT="ssl/server.crt"
 	    CUPS_SERVERKEY="ssl/server.key"
-
-            if $PKGCONFIG --exists gcrypt; then
-	        SSLLIBS="$SSLLIBS `$PKGCONFIG --libs gcrypt`"
-	        SSLFLAGS="$SSLFLAGS `$PKGCONFIG --cflags gcrypt`"
-	    elif test "x$LIBGCRYPTCONFIG" != x; then
-	        SSLLIBS="$SSLLIBS `$LIBGCRYPTCONFIG --libs`"
-	        SSLFLAGS="$SSLFLAGS `$LIBGCRYPTCONFIG --cflags`"
-	    fi
 	fi
     fi
 
diff -up cups-1.7b1/cups/http-private.h.no-gcry cups-1.7b1/cups/http-private.h
--- cups-1.7b1/cups/http-private.h.no-gcry	2013-06-24 12:15:57.833339957 +0100
+++ cups-1.7b1/cups/http-private.h	2013-06-24 12:16:06.175376393 +0100
@@ -80,7 +80,6 @@ typedef int socklen_t;
 #  elif defined HAVE_GNUTLS
 #    include <gnutls/gnutls.h>
 #    include <gnutls/x509.h>
-#    include <gcrypt.h>
 #  elif defined(HAVE_CDSASSL)
 #    include <CoreFoundation/CoreFoundation.h>
 #    include <Security/Security.h>
