diff -up cups-1.4.3/systemv/lpstat.c.str3541 cups-1.4.3/systemv/lpstat.c
--- cups-1.4.3/systemv/lpstat.c.str3541	2009-12-15 23:40:37.000000000 +0100
+++ cups-1.4.3/systemv/lpstat.c	2010-04-16 12:56:04.000000000 +0200
@@ -1423,7 +1423,7 @@ show_jobs(const char *dests,		/* I - Des
 
       rank ++;
 
-      if (match_list(dests, dest) || match_list(users, username))
+      if (match_list(dests, dest) && match_list(users, username))
       {
         jobdate = localtime(&jobtime);
         snprintf(temp, sizeof(temp), "%s-%d", dest, jobid);
