diff -up cups-2.0rc1/conf/cups-files.conf.in.synconclose cups-2.0rc1/conf/cups-files.conf.in
--- cups-2.0rc1/conf/cups-files.conf.in.synconclose	2014-09-12 16:48:36.510689633 +0200
+++ cups-2.0rc1/conf/cups-files.conf.in	2014-09-12 16:48:36.517689534 +0200
@@ -7,7 +7,7 @@
 #FatalErrors @CUPS_FATAL_ERRORS@
 
 # Do we call fsync() after writing configuration or status files?
-#SyncOnClose No
+#SyncOnClose Yes
 
 # Default user and group for filters/backends/helper programs; this cannot be
 # any user or group that resolves to ID 0 for security reasons...
diff -up cups-2.0rc1/doc/help/man-cups-files.conf.html.synconclose cups-2.0rc1/doc/help/man-cups-files.conf.html
--- cups-2.0rc1/doc/help/man-cups-files.conf.html.synconclose	2014-09-12 16:48:36.517689534 +0200
+++ cups-2.0rc1/doc/help/man-cups-files.conf.html	2014-09-12 16:49:20.643066238 +0200
@@ -140,7 +140,7 @@ The default is "/etc/cups".
 <dd style="margin-left: 5.0em">Specifies whether the scheduler calls
 <b>fsync</b>(2)
 after writing configuration or state files.
-The default is "No".
+The default is "Yes".
 <dt><b>SystemGroup </b><i>group-name </i>[ ... <i>group-name</i> ]
 <dd style="margin-left: 5.0em">Specifies the group(s) to use for <i>@SYSTEM</i> group authentication.
 The default contains "admin", "lpadmin", "root", "sys", and/or "system".
diff -up cups-2.0rc1/man/cups-files.conf.man.in.synconclose cups-2.0rc1/man/cups-files.conf.man.in
--- cups-2.0rc1/man/cups-files.conf.man.in.synconclose	2014-09-12 16:48:36.518689520 +0200
+++ cups-2.0rc1/man/cups-files.conf.man.in	2014-09-12 16:49:52.793612092 +0200
@@ -182,7 +182,7 @@ The default is "/etc/cups".
 Specifies whether the scheduler calls
 .BR fsync (2)
 after writing configuration or state files.
-The default is "No".
+The default is "Yes".
 .TP 5
 \fBSystemGroup \fIgroup-name \fR[ ... \fIgroup-name\fR ]
 Specifies the group(s) to use for \fI@SYSTEM\fR group authentication.
diff -up cups-2.0rc1/scheduler/conf.c.synconclose cups-2.0rc1/scheduler/conf.c
--- cups-2.0rc1/scheduler/conf.c.synconclose	2014-09-12 16:48:36.519689506 +0200
+++ cups-2.0rc1/scheduler/conf.c	2014-09-12 16:50:24.289167199 +0200
@@ -715,7 +715,7 @@ cupsdReadConfiguration(void)
   RootCertDuration         = 300;
   Sandboxing               = CUPSD_SANDBOXING_STRICT;
   StrictConformance        = FALSE;
-  SyncOnClose              = FALSE;
+  SyncOnClose              = TRUE;
   Timeout                  = DEFAULT_TIMEOUT;
   WebInterface             = CUPS_DEFAULT_WEBIF;
 
