diff -up cups-1.7.5/scheduler/colorman.c.dbus-unref cups-1.7.5/scheduler/colorman.c
--- cups-1.7.5/scheduler/colorman.c.dbus-unref	2014-10-21 12:43:10.401299966 +0100
+++ cups-1.7.5/scheduler/colorman.c	2014-10-21 12:43:12.870312593 +0100
@@ -208,7 +208,9 @@ void
 cupsdStopColor(void)
 {
 #if !defined(__APPLE__) && defined(HAVE_DBUS)
-  dbus_connection_unref(colord_con);
+  if (colord_con)
+    dbus_connection_unref(colord_con);
+
   colord_con = NULL;
 #endif /* !__APPLE__ && HAVE_DBUS */
 }
