--- cups-1.7.0/conf/cups-files.conf.in.synconclose	2013-11-04 11:43:27.989512747 +0000
+++ cups-1.7.0/conf/cups-files.conf.in	2013-11-04 11:43:37.830557404 +0000
@@ -9,7 +9,7 @@
 #FatalErrors @CUPS_FATAL_ERRORS@
 
 # Do we call fsync() after writing configuration or status files?
-#SyncOnClose No
+#SyncOnClose Yes
 
 # Default user and group for filters/backends/helper programs; this cannot be
 # any user or group that resolves to ID 0 for security reasons...
diff -up cups-1.7.0/doc/help/man-cups-files.conf.html.synconclose cups-1.7.0/doc/help/man-cups-files.conf.html
--- cups-1.7.0/doc/help/man-cups-files.conf.html.synconclose	2013-10-31 11:19:11.927824302 +0000
+++ cups-1.7.0/doc/help/man-cups-files.conf.html	2013-10-31 11:19:24.061871113 +0000
@@ -146,7 +146,7 @@ be found.
 <dt>SyncOnClose No
 </dt>
 <dd>Specifies whether the scheduler calls <i>fsync(2)</i> after writing configuration
-or state files. The default is No.
+or state files. The default is Yes.
 </dd>
 <dt>SystemGroup group-name [group-name ...]
 </dt>
diff -up cups-1.7.0/doc/help/ref-cups-files-conf.html.in.synconclose cups-1.7.0/doc/help/ref-cups-files-conf.html.in
--- cups-1.7.0/doc/help/ref-cups-files-conf.html.in.synconclose	2013-10-31 11:19:33.382907073 +0000
+++ cups-1.7.0/doc/help/ref-cups-files-conf.html.in	2013-10-31 11:20:31.567131557 +0000
@@ -442,8 +442,8 @@ SyncOnClose Yes
 
 <P>The <CODE>SyncOnClose</CODE> directive determines whether the scheduler
 flushes changes to configuration and state files to disk. The default is
-<CODE>No</CODE> which relies on the operating system to schedule a suitable
-time to write changes to disk.</P>
+<CODE>Yes</CODE>. The option <CODE>No</CODE> relies on the operating
+system to schedule a suitable time to write changes to disk.</P>
 
 <BLOCKQUOTE><B>Note:</B>
 
diff -up cups-1.7.0/man/cups-files.conf.man.in.synconclose cups-1.7.0/man/cups-files.conf.man.in
--- cups-1.7.0/man/cups-files.conf.man.in.synconclose	2013-10-31 11:18:43.928716290 +0000
+++ cups-1.7.0/man/cups-files.conf.man.in	2013-10-31 11:19:00.870781647 +0000
@@ -126,7 +126,7 @@ SyncOnClose Yes
 .TP 5
 SyncOnClose No
 Specifies whether the scheduler calls \fIfsync(2)\fR after writing configuration
-or state files. The default is No.
+or state files. The default is Yes.
 .TP 5
 SystemGroup group-name [group-name ...]
 .br
diff -up cups-1.7.0/scheduler/conf.c.synconclose cups-1.7.0/scheduler/conf.c
--- cups-1.7.0/scheduler/conf.c.synconclose	2013-10-31 11:20:45.560185549 +0000
+++ cups-1.7.0/scheduler/conf.c	2013-10-31 11:20:56.679228452 +0000
@@ -738,7 +738,7 @@ cupsdReadConfiguration(void)
   ReloadTimeout	           = DEFAULT_KEEPALIVE;
   RootCertDuration         = 300;
   StrictConformance        = FALSE;
-  SyncOnClose              = FALSE;
+  SyncOnClose              = TRUE;
   Timeout                  = DEFAULT_TIMEOUT;
   WebInterface             = CUPS_DEFAULT_WEBIF;
 
