diff -up cups-1.4b2-svn8404/cups/http-addrlist.c.res_init cups-1.4b2-svn8404/cups/http-addrlist.c
--- cups-1.4b2-svn8404/cups/http-addrlist.c.res_init	2009-03-23 17:41:03.000000000 +0000
+++ cups-1.4b2-svn8404/cups/http-addrlist.c	2009-03-23 17:41:26.000000000 +0000
@@ -373,7 +373,7 @@ httpAddrGetList(const char *hostname,	/*
 
       freeaddrinfo(results);
     }
-    else if (error == EAI_FAIL)
+    else if (error == EAI_FAIL || error == EAI_AGAIN || error == EAI_NODATA)
       cg->need_res_init = 1;
 
 #else
