diff -up cups-1.6.4/backend/ipp.c.ipp-multifile cups-1.6.4/backend/ipp.c
--- cups-1.6.4/backend/ipp.c.ipp-multifile	2013-09-25 14:06:32.027732925 +0100
+++ cups-1.6.4/backend/ipp.c	2013-09-25 14:10:22.134679512 +0100
@@ -1685,7 +1685,10 @@ main(int  argc,				/* I - Number of comm
 	ippAddBoolean(request, IPP_TAG_OPERATION, "last-document",
         	      (i + 1) >= num_files);
 
-	if (document_format)
+	if (num_files > 1)
+	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
+		       "document-format", NULL, "application/octet-stream");
+	else if (document_format)
 	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
 		       "document-format", NULL, document_format);
 
