diff -up cups-1.3.7/backend/usb-unix.c.direct-usb cups-1.3.7/backend/usb-unix.c
--- cups-1.3.7/backend/usb-unix.c.direct-usb	2008-03-26 16:02:45.000000000 +0000
+++ cups-1.3.7/backend/usb-unix.c	2008-07-15 13:25:56.000000000 +0100
@@ -94,6 +94,9 @@ print_device(const char *uri,		/* I - De
              strncasecmp(hostname, "Minolta", 7);
 #endif /* __FreeBSD__ || __NetBSD__ || __OpenBSD__ || __DragonFly__ */
 
+    if (use_bc && !strncmp(uri, "usb:/dev/", 9))
+      use_bc = 0;
+
     if ((device_fd = open_device_unix(uri, &use_bc)) == -1)
     {
       if (getenv("CLASS") != NULL)
@@ -320,12 +323,7 @@ open_device(const char *uri,		/* I - Dev
   if (!strncmp(uri, "usb:/dev/", 9))
 #ifdef __linux
   {
-   /*
-    * Do not allow direct devices anymore...
-    */
-
-    errno = ENODEV;
-    return (-1);
+    return (open(uri + 4, O_RDWR | O_EXCL));
   }
   else if (!strncmp(uri, "usb://", 6))
   {
