diff -up cups-1.4.5/scheduler/printers.c.printer-timeout cups-1.4.5/scheduler/printers.c
--- cups-1.4.5/scheduler/printers.c.printer-timeout	2010-12-16 14:42:07.048865052 +0000
+++ cups-1.4.5/scheduler/printers.c	2010-12-16 14:43:05.325684376 +0000
@@ -785,6 +785,7 @@ cupsdDeletePrinter(
   */
 
   cupsdSetPrinterState(p, IPP_PRINTER_STOPPED, update);
+  p->state = IPP_PRINTER_STOPPED;
 
   if (p->job)
     cupsdSetJobState(p->job, IPP_JOB_PENDING, CUPSD_JOB_FORCE,
