diff -up cups-1.4.4/scheduler/subscriptions.c.str3608 cups-1.4.4/scheduler/subscriptions.c
--- cups-1.4.4/scheduler/subscriptions.c.str3608	2010-06-28 12:07:54.611008831 +0100
+++ cups-1.4.4/scheduler/subscriptions.c	2010-06-28 12:09:25.315887577 +0100
@@ -551,12 +551,14 @@ cupsdEventName(
         return ("printer-media-changed");
 
     case CUPSD_EVENT_PRINTER_ADDED :
+    case CUPSD_EVENT_PRINTER_ADDED | CUPSD_EVENT_PRINTER_CONFIG:
         return ("printer-added");
 
     case CUPSD_EVENT_PRINTER_DELETED :
         return ("printer-deleted");
 
     case CUPSD_EVENT_PRINTER_MODIFIED :
+    case CUPSD_EVENT_PRINTER_MODIFIED | CUPSD_EVENT_PRINTER_CONFIG:
         return ("printer-modified");
 
     case CUPSD_EVENT_PRINTER_QUEUE_ORDER_CHANGED :
