diff -up cups-1.4.2/scheduler/main.c.str3399 cups-1.4.2/scheduler/main.c
--- cups-1.4.2/scheduler/main.c.str3399	2010-03-02 11:04:28.968100478 +0000
+++ cups-1.4.2/scheduler/main.c	2010-03-02 11:04:47.442226409 +0000
@@ -1733,7 +1733,7 @@ process_children(void)
 	  job->backend = -pid;
 
 	if (status && status != SIGTERM && status != SIGKILL &&
-	    job->status >= 0)
+	    status != SIGPIPE && job->status >= 0)
 	{
 	 /*
 	  * An error occurred; save the exit status so we know to stop
@@ -1829,6 +1829,12 @@ process_children(void)
                       "PID %d (%s) was terminated normally with signal %d.",
                       pid, name, status);
     }
+    else if (status == SIGPIPE)
+    {
+      cupsdLogMessage(CUPSD_LOG_DEBUG,
+                      "PID %d (%s) did not catch or ignore signal %d.",
+                      pid, name, status);
+    }
     else if (status)
     {
       if (WIFEXITED(status))
