diff -up cups-1.6.2/cups/file.c.gz-crc cups-1.6.2/cups/file.c
--- cups-1.6.2/cups/file.c.gz-crc	2013-07-11 12:06:49.920878639 +0100
+++ cups-1.6.2/cups/file.c	2013-07-11 12:13:41.169628104 +0100
@@ -2429,8 +2429,8 @@ cups_fill(cups_file_t *fp)		/* I - CUPS
 	}
 	else
 	{
-	  tcrc = (((((trailer[3] << 8) | trailer[2]) << 8) | trailer[1]) << 8) |
-        	 trailer[0];
+	  tcrc = (((uLong) ((trailer[3] << 8) | trailer[2])) << 16) |
+		   (uLong) ((trailer[1] << 8) | trailer[0]);
 
 	  if (tcrc != fp->crc)
 	  {
