diff -up cups-1.4.4/scheduler/ipp.c.str3608 cups-1.4.4/scheduler/ipp.c
--- cups-1.4.4/scheduler/ipp.c.str3608	2010-08-20 11:36:58.490089326 +0100
+++ cups-1.4.4/scheduler/ipp.c	2010-08-20 11:37:11.965213877 +0100
@@ -1245,7 +1245,7 @@ add_class(cupsd_client_t  *con,		/* I - 
 
   if (modify)
   {
-    cupsdAddEvent(CUPSD_EVENT_PRINTER_MODIFIED | CUPSD_EVENT_PRINTER_CONFIG,
+    cupsdAddEvent(CUPSD_EVENT_PRINTER_MODIFIED,
 		  pclass, NULL, "Class \"%s\" modified by \"%s\".",
 		  pclass->name, get_username(con));
 
@@ -1256,7 +1256,7 @@ add_class(cupsd_client_t  *con,		/* I - 
   {
     cupsdAddPrinterHistory(pclass);
 
-    cupsdAddEvent(CUPSD_EVENT_PRINTER_ADDED | CUPSD_EVENT_PRINTER_CONFIG,
+    cupsdAddEvent(CUPSD_EVENT_PRINTER_ADDED,
 		  pclass, NULL, "New class \"%s\" added by \"%s\".",
 		  pclass->name, get_username(con));
 
@@ -3034,7 +3034,7 @@ add_printer(cupsd_client_t  *con,	/* I -
 
   if (modify)
   {
-    cupsdAddEvent(CUPSD_EVENT_PRINTER_MODIFIED | CUPSD_EVENT_PRINTER_CONFIG,
+    cupsdAddEvent(CUPSD_EVENT_PRINTER_MODIFIED,
                   printer, NULL, "Printer \"%s\" modified by \"%s\".",
 		  printer->name, get_username(con));
 
@@ -3045,7 +3045,7 @@ add_printer(cupsd_client_t  *con,	/* I -
   {
     cupsdAddPrinterHistory(printer);
 
-    cupsdAddEvent(CUPSD_EVENT_PRINTER_ADDED | CUPSD_EVENT_PRINTER_CONFIG,
+    cupsdAddEvent(CUPSD_EVENT_PRINTER_ADDED,
                   printer, NULL, "New printer \"%s\" added by \"%s\".",
 		  printer->name, get_username(con));
 
