diff -up cups-1.4.2/cups/http-addr.c.EAI_AGAIN cups-1.4.2/cups/http-addr.c
--- cups-1.4.2/cups/http-addr.c.EAI_AGAIN	2010-02-23 10:39:35.038261623 +0000
+++ cups-1.4.2/cups/http-addr.c	2010-02-23 10:41:14.684385991 +0000
@@ -253,7 +253,7 @@ httpAddrLookup(
 
     if (error)
     {
-      if (error == EAI_FAIL)
+      if (error == EAI_FAIL || error == EAI_AGAIN || error == EAI_NODATA)
         cg->need_res_init = 1;
 
       return (httpAddrString(addr, name, namelen));
