diff -up cups-1.4.3/cups/dest.c.str3503 cups-1.4.3/cups/dest.c
--- cups-1.4.3/cups/dest.c.str3503	2010-03-31 13:31:42.000000000 +0200
+++ cups-1.4.3/cups/dest.c	2010-03-31 13:32:52.000000000 +0200
@@ -553,7 +553,7 @@ cupsGetNamedDest(http_t     *http,	/* I 
 
   if (!cups_get_sdests(http, op, name, 0, &dest))
   {
-    if (op == CUPS_GET_DEFAULT || name)
+    if (op == CUPS_GET_DEFAULT || (name && !set_as_default))
       return (NULL);
 
    /*
