diff -up cups-1.4b2-svn8404/config-scripts/cups-ssl.m4.no-export-ssllibs cups-1.4b2-svn8404/config-scripts/cups-ssl.m4
--- cups-1.4b2-svn8404/config-scripts/cups-ssl.m4.no-export-ssllibs	2009-02-17 17:45:27.000000000 +0000
+++ cups-1.4b2-svn8404/config-scripts/cups-ssl.m4	2009-03-05 11:12:59.000000000 +0000
@@ -110,7 +110,7 @@ fi
 AC_SUBST(SSLFLAGS)
 AC_SUBST(SSLLIBS)
 
-EXPORT_SSLLIBS="$SSLLIBS"
+EXPORT_SSLLIBS=""
 AC_SUBST(EXPORT_SSLLIBS)
 
 
