diff -up cups-1.4.4/scheduler/ipp.c.autotype-crash cups-1.4.4/scheduler/ipp.c
--- cups-1.4.4/scheduler/ipp.c.autotype-crash	2010-10-15 15:25:15.093421917 +0100
+++ cups-1.4.4/scheduler/ipp.c	2010-10-15 15:25:49.645296947 +0100
@@ -10481,8 +10481,9 @@ send_document(cupsd_client_t  *con,	/* I
     if (!filetype)
       filetype = mimeType(MimeDatabase, super, type);
 
-    cupsdLogJob(job, CUPSD_LOG_DEBUG, "Request file type is %s/%s.",
-		filetype->super, filetype->type);
+    if (filetype)
+      cupsdLogJob(job, CUPSD_LOG_DEBUG, "Request file type is %s/%s.",
+		  filetype->super, filetype->type);
   }
   else
     filetype = mimeType(MimeDatabase, super, type);
