diff -up cups-1.6.4/backend/org.cups.usb-quirks.usblp-quirks cups-1.6.4/backend/org.cups.usb-quirks
--- cups-1.6.4/backend/org.cups.usb-quirks.usblp-quirks	2013-07-23 14:24:57.000000000 +0100
+++ cups-1.6.4/backend/org.cups.usb-quirks	2013-09-25 13:54:16.910144515 +0100
@@ -75,6 +75,9 @@
 # Canon, Inc. MP510 Printer, https://bugs.launchpad.net/bugs/1050009
 0x04a9 0x1717 unidir
 
+# Canon, Inc. MP540 Printer, https://bugzilla.redhat.com/967873 */
+0x04a9 0x1730 unidir
+
 # Canon, Inc. MP550 Printer, http://www.cups.org/str.php?L4155
 0x04a9 0x173d unidir
 
diff -up cups-1.6.4/backend/usb-libusb.c.usblp-quirks cups-1.6.4/backend/usb-libusb.c
