--- configure.ac.glx	2010-06-22 17:24:56.020730401 -0300
+++ configure.ac	2010-06-22 17:50:42.291104622 -0300
@@ -179,11 +179,6 @@
                           ${VA_MICRO_VERSION:-0} "*" 100     + \
                           ${VA_SDS_VERSION:-0}`
 VA_DRIVER_INIT_FUNC="__vaDriverInit_${VA_MAJOR_VERSION}_${VA_MINOR_VERSION}"
-if test $va_full_version_int -ge 00310005; then
-    VA_DRIVER_INIT_FUNC="${VA_DRIVER_INIT_FUNC}_${VA_MICRO_VERSION}_sds${VA_SDS_VERSION}"
-else
-    VA_DRIVER_INIT_FUNC="${VA_DRIVER_INIT_FUNC}_sds"
-fi
 if test "$ac_cv_libva_sds_extensions" = "yes"; then
     AC_DEFINE_UNQUOTED([VA_DRIVER_INIT_FUNC], [$VA_DRIVER_INIT_FUNC], [Define driver entry-point])
 else
@@ -197,15 +192,7 @@
   CFLAGS="$CFLAGS $LIBVA_DEPS_CFLAGS"
   AC_TRY_COMPILE([
     #include <va/va_version.h>
-    #if VA_MAJOR_VERSION == 0 && VA_MINOR_VERSION == 30
-    # if !defined(VA_SDS_VERSION) || (VA_SDS_VERSION < $LIBVA_GLX_SDS_VERSION_0_30)
-    #  error "VA API version >= 0.30.0-sds$LIBVA_GLX_SDS_VERSION_0_30 is required"
-    # endif
-    #elif VA_MAJOR_VERSION == 0 && VA_MINOR_VERSION == 31
-    # if !defined(VA_SDS_VERSION) || (VA_SDS_VERSION < $LIBVA_GLX_SDS_VERSION_0_31)
-    #  error "VA API version >= 0.31.0-sds$LIBVA_GLX_SDS_VERSION_0_31 is required"
-    # endif
-    #else
+    #if VA_MAJOR_VERSION == 0 && VA_MINOR_VERSION < 31
     # error "This version of VA API is not supported for OpenGL rendering"
     #endif
   ], [],
