diff -up kdesvn-1.5.5/src/svnfrontend/maintreewidget.cpp.gcc47 kdesvn-1.5.5/src/svnfrontend/maintreewidget.cpp
--- kdesvn-1.5.5/src/svnfrontend/maintreewidget.cpp.gcc47	2010-03-25 16:04:55.000000000 -0600
+++ kdesvn-1.5.5/src/svnfrontend/maintreewidget.cpp	2012-01-14 10:25:17.289604638 -0700
@@ -55,6 +55,7 @@
 #include <kio/copyjob.h>
 #include <kfiledialog.h>
 #include <knotification.h>
+#include <unistd.h>
 
 #include <QSortFilterProxyModel>
 #include <QEvent>
