diff -up tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc.pointersync tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc
--- tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc.pointersync	2014-07-14 16:27:49.412251632 +0100
+++ tigervnc-1.3.0/unix/xserver/hw/vnc/Input.cc	2014-07-14 16:29:06.968643301 +0100
@@ -1,5 +1,5 @@
 /* Copyright (C) 2009 TightVNC Team
- * Copyright (C) 2009 Red Hat, Inc.
+ * Copyright (C) 2009, 2014 Red Hat, Inc.
  * Copyright 2013 Pierre Ossman for Cendio AB
  *
  * This is free software; you can redistribute it and/or modify
@@ -202,6 +202,14 @@ void InputDevice::PointerMove(const rfb:
 
 const rfb::Point &InputDevice::getPointerPos(void)
 {
+	if (pointerDev != NULL) {
+		int x, y;
+
+		GetSpritePosition (pointerDev, &x, &y);
+		cursorPos.x = x;
+		cursorPos.y = y;
+	}
+
 	return cursorPos;
 }
 
