diff -up tigervnc-1.4.2/unix/vncserver.manpages tigervnc-1.4.2/unix/vncserver
--- tigervnc-1.4.2/unix/vncserver.manpages	2015-02-12 19:39:32.253788841 +0000
+++ tigervnc-1.4.2/unix/vncserver	2015-02-12 19:39:32.326789294 +0000
@@ -551,6 +551,7 @@ sub Usage
 	"                 [-geometry <width>x<height>]\n".
 	"                 [-pixelformat rgbNNN|bgrNNN]\n".
 	"                 [-fp <font-path>]\n".
+	"                 [-cc <visual>]\n".
 	"                 [-fg]\n".
 	"                 [-autokill]\n".
 	"                 <Xvnc-options>...\n\n".
diff -up tigervnc-1.4.2/vncviewer/vncviewer.cxx.manpages tigervnc-1.4.2/vncviewer/vncviewer.cxx
--- tigervnc-1.4.2/vncviewer/vncviewer.cxx.manpages	2015-01-23 23:37:23.000000000 +0000
+++ tigervnc-1.4.2/vncviewer/vncviewer.cxx	2015-02-12 19:39:32.327789300 +0000
@@ -265,6 +265,11 @@ static void usage(const char *programNam
           "       %s [parameters] -listen [port] [parameters]\n",
           programName, programName);
   fprintf(stderr,"\n"
+	  "Options:\n\n"
+	  "  -display Xdisplay - Specifies the X display for the viewer window\n"
+	  "  -geometry geometry - Standard X position and sizing specification.\n");
+
+  fprintf(stderr,"\n"
           "Parameters can be turned on with -<param> or off with -<param>=0\n"
           "Parameters which take a value can be specified as "
           "-<param> <value>\n"
diff -up tigervnc-1.4.2/vncviewer/vncviewer.man.manpages tigervnc-1.4.2/vncviewer/vncviewer.man
--- tigervnc-1.4.2/vncviewer/vncviewer.man.manpages	2015-02-12 19:39:32.327789300 +0000
+++ tigervnc-1.4.2/vncviewer/vncviewer.man	2015-02-13 09:54:55.042903357 +0000
@@ -219,6 +219,10 @@ This option specifies the preferred enco
 Disable lossy JPEG compression in Tight encoding. Default is off.
 .
 .TP
+.B \-ImprovedHextile
+Try harder to compress data (default).
+.
+.TP
 .B \-QualityLevel \fIlevel\fP
 JPEG quality level. 0 = Low, 9 = High. May be adjusted automatically if
 \fB-AutoSelect\fP is turned on. Default is 8.
@@ -271,6 +275,10 @@ command is executed with the environment
 \fIR\fR, and \fIG\fR taken the values of the local port number, the remote
 host, the port number on the remote host, and the gateway machine
 respectively.
+.
+.TP
+.B \-ZlibLevel
+Zlib compression level.
 
 .SH FILES
 .TP
