--- tigervnc-1.1.0/unix/xserver/hw/vnc/Makefile.am.bettersed	2012-07-04 12:05:24.578878647 -0400
+++ tigervnc-1.1.0/unix/xserver/hw/vnc/Makefile.am	2012-07-05 10:23:59.186553805 -0400
@@ -62,10 +62,10 @@
 BUILT_SOURCES = $(nodist_Xvnc_SOURCES)
 
 fb.h: $(top_srcdir)/fb/fb.h
-	cat $(top_srcdir)/fb/fb.h | sed -e 's,and,c_and,' -e 's,xor,c_xor,' > $(srcdir)/fb.h
+	cat $(top_srcdir)/fb/fb.h | sed -e 's/\([(, ]\)xor\([,);]\)/\1c_xor\2/g' -e 's/\([(,]\)xor\([,) ;]\)/\1c_xor\2/g' -e 's/\([(, ]\)and\([,);]\)/\1c_and\2/g' -e 's/\([(,]\)and\([,) ;]\)/\1c_and\2/g' > $(srcdir)/fb.h
 
 pixman.h: $(includedir)/pixman-1/pixman.h
-	cat $(includedir)/pixman-1/pixman.h | sed 's/xor/c_xor/' > $(srcdir)/pixman.h
+	cat $(includedir)/pixman-1/pixman.h | sed -e 's/\([(, ]\)xor\([,);]\)/\1c_xor\2/g' -e 's/\([(,]\)xor\([,) ;]\)/\1c_xor\2/g' -e 's/\([(, ]\)and\([,);]\)/\1c_and\2/g' -e 's/\([(,]\)and\([,) ;]\)/\1c_and\2/g' > $(srcdir)/pixman.h
 
 fbrop.h: $(top_srcdir)/fb/fbrop.h
-	cat $(top_srcdir)/fb/fbrop.h | sed -e 's,and,c_and,' -e 's,xor,c_xor,' > $(srcdir)/fbrop.h
+	cat $(top_srcdir)/fb/fbrop.h | sed -e 's/\([(, ]\)xor\([,);]\)/\1c_xor\2/g' -e 's/\([(,]\)xor\([,) ;]\)/\1c_xor\2/g' -e 's/\([(, ]\)and\([,);]\)/\1c_and\2/g' -e 's/\([(,]\)and\([,) ;]\)/\1c_and\2/g' > $(srcdir)/fbrop.h
