diff -up tigervnc-1.2.80-20130314svn5065/unix/vncserver.man.manpages tigervnc-1.2.80-20130314svn5065/unix/vncserver.man
--- tigervnc-1.2.80-20130314svn5065/unix/vncserver.man.manpages	2013-07-03 12:44:03.820392690 +0100
+++ tigervnc-1.2.80-20130314svn5065/unix/vncserver.man	2013-07-03 12:44:44.222587028 +0100
@@ -92,6 +92,10 @@ argument.  Thus, you can invoke "vncserv
 end of your xstartup file after a particular application exits.
 
 .TP
+.B \-list
+Lists running VNC servers.
+
+.TP
 .B \-fp \fIfont-path\fP
 If the vncserver script detects that the X Font Server (XFS) is running, it
 will attempt to start Xvnc and configure Xvnc to use XFS for font handling.
diff -up tigervnc-1.2.80-20130314svn5065/unix/vncserver.manpages tigervnc-1.2.80-20130314svn5065/unix/vncserver
--- tigervnc-1.2.80-20130314svn5065/unix/vncserver.manpages	2013-07-03 12:39:08.754980784 +0100
+++ tigervnc-1.2.80-20130314svn5065/unix/vncserver	2013-07-03 12:43:57.932364387 +0100
@@ -551,6 +551,7 @@ sub Usage
 	"                 [-geometry <width>x<height>]\n".
 	"                 [-pixelformat rgbNNN|bgrNNN]\n".
 	"                 [-fp <font-path>]\n".
+	"                 [-cc <visual>]\n".
 	"                 [-fg]\n".
 	"                 [-autokill]\n".
 	"                 <Xvnc-options>...\n\n".
diff -up tigervnc-1.2.80-20130314svn5065/unix/x0vncserver/x0vncserver.cxx.manpages tigervnc-1.2.80-20130314svn5065/unix/x0vncserver/x0vncserver.cxx
diff -up tigervnc-1.2.80-20130314svn5065/unix/x0vncserver/x0vncserver.man.manpages tigervnc-1.2.80-20130314svn5065/unix/x0vncserver/x0vncserver.man
--- tigervnc-1.2.80-20130314svn5065/unix/x0vncserver/x0vncserver.man.manpages	2013-07-03 12:46:17.814038117 +0100
+++ tigervnc-1.2.80-20130314svn5065/unix/x0vncserver/x0vncserver.man	2013-07-03 12:58:32.814621597 +0100
@@ -68,7 +68,13 @@ to accept connections from any IP addres
 Specify which security scheme to use for incoming connections.  Valid values
 are \fBNone\fP and \fBVncAuth\fP.  Default is \fBVncAuth\fP.
 .TP
-.B PasswordFile
+.B pam_server
+Service name for pam password validation (default is "vnc").
+.TP
+.B PlainUsers
+Users permission to access via Plain security type.
+.TP
+.B PasswordFile, rfbauth
 Password file for VNC authentication.  There is no default, you should
 specify the password file explicitly.  Password file should be created with
 the \fBvncpasswd\fP(1) utility.
@@ -113,6 +119,16 @@ Accept key press and release events from
 .B AcceptPointerEvents
 Accept pointer events from clients.  Default is on.
 .TP
+.B AcceptCutText
+Accept clipboard updates from clients.  Default is on.
+.TP
+.B MaxCutText
+Maximum permitted length of an incoming clipboard update (default is
+262144).
+.TP
+.B SendCutText
+Send clipboard changes to clients.  Default is on.
+.TP
 .B RemapKeys
 Comma-separated list of incoming keysyms to remap.  Mappings are expressed as
 two hex values, prefixed by \fB0x\fP, and separated by \fB->\fP (`dash' and
@@ -154,23 +170,6 @@ clients may choose video selection thems
 \fBVideoArea\fP parameter will take effect.  If the argument is empty, no
 video area is set (this is the default).
 .TP
-.B VideoPriority
-Specify the priority of sending video updates.  \fBx0vncserver\fP can be
-instructed to treat certain rectangular part of the screen as a video area
-and handle it in a special way for improved performance (see documentation on
-the \fBVideoArea\fP parameter).  \fBVideoPriority\fP value controls how often
-video area will be sent to clients as compared to the rest of the screen.
-The priority must be an integer between 0 and 8, and the default value is 2.
-
-\fBVideoPriority\fP set to 1 gives the same priority both to video and to
-other pixels.  Higher values give more priority to video.  For example, the
-value 5 specifies that the rate of sending video will be five times higher
-than the rate of updating the rest of the screen.  If \fBVideoPriority\fP is
-set to 0, it gives equal priority to video and other updates (just like the
-value 1) and also disables special encoding for video data.  In other words,
-\fBVideoPriority\fP video area will be sent as a part of other screen
-contents.
-.TP
 .B CompareFB
 Perform pixel comparison on framebuffer to reduce unnecessary updates.
 Default is on.
@@ -185,15 +184,6 @@ This enables system-specific access to c
 screen (the default X visual often provides 256 colors).  Also, in overlay
 mode, \fBx0vncserver\fP can show correct mouse cursor.  Default is on.
 .TP
-.B UseHardwareJPEG
-Use hardware-accelerated JPEG compressor for video if available.
-\fBx0vncserver\fP can be instructed to treat certain rectangular part of the
-screen as a video area and handle it in a special way for improved
-performance.  If the client supports Tight encoding and JPEG compression,
-such video areas will be sent as JPEG-encoded rectangles.  And if this option
-is on, compression will be hardware-accelerated (currently, supported only in
-SGI/IRIX equipped with appropriate hardware).  Default is on.
-.TP
 .B ZlibLevel
 Zlib compression level for ZRLE encoding (it does not affect Tight encoding).
 Acceptable values are between 0 and 9.  Default is to use the standard
@@ -222,6 +212,18 @@ Terminate after \fIN\fP seconds of user
 .B ClientWaitTimeMillis
 The number of milliseconds to wait for a client which is no longer
 responding.  Default is 20000.
+.TP
+.B DeferUpdate
+Time in milliseconds to defer updates (default is 1).
+.TP
+.B AlwaysSetDeferUpdateTimer
+Always reset the defer update timer on every change.
+.TP
+.B x509key
+Path to key of the x509 certificate in PEM format.
+.TP
+.B 509cert
+Path to x509 certificate in PEM format.
 .SH SEE ALSO
 .BR Xvnc (1),
 .BR vncpasswd (1),
diff -up tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.cxx.manpages tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.cxx
--- tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.cxx.manpages	2013-07-03 12:00:58.005856116 +0100
+++ tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.cxx	2013-07-03 12:04:26.083023050 +0100
@@ -261,6 +261,11 @@ static void usage(const char *programNam
           "       %s [parameters] -listen [port] [parameters]\n",
           programName, programName);
   fprintf(stderr,"\n"
+	  "Options:\n\n"
+	  "  -display Xdisplay - Specifies the X display for the viewer window\n"
+	  "  -geometry geometry - Standard X position and sizing specification.\n");
+
+  fprintf(stderr,"\n"
           "Parameters can be turned on with -<param> or off with -<param>=0\n"
           "Parameters which take a value can be specified as "
           "-<param> <value>\n"
diff -up tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.man.manpages tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.man
--- tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.man.manpages	2013-07-03 11:40:48.905078300 +0100
+++ tigervnc-1.2.80-20130314svn5065/vncviewer/vncviewer.man	2013-07-03 12:00:13.116604372 +0100
@@ -113,12 +113,53 @@ Xvnc supports reverse connections with a
 .B vncconfig.
 
 .TP
-.B \-passwd \fIpassword-file\fP
+.B \-passwd \fIpassword-file\fP, \-PasswordFile \fIpassword-file\fP
 If you are on a filesystem which gives you access to the password file used by
 the server, you can specify it here to avoid typing it in.  It will usually be
 "~/.vnc/passwd".
 
 .TP
+.B \-DotWhenNoCursor
+Show the dot cursor when the server sends an invisible cursor.
+
+.TP
+.B \-PointerEventInterval
+Time in milliseconds to rate-limit successive pointer events.
+
+.TP
+.B \-ImprovedHextile
+Try harder to compress data (default).
+
+.TP
+.B \-QualityLevel \fIlevel\fP
+JPEG quality level (default is 8).
+
+.TP
+.B \-NoJPEG
+Disable lossy JPEG compression in Tight encoding.
+
+.TP
+.B CompressLevel \fIlevel\fP
+Use specified compression level (default is 2).
+
+.TP
+.B CustomCompressLevel \fIlevel\fP
+Use custom compression level. Default is CompressLevel is specified.
+
+.TP
+.B \-SendPrimary
+Send the primary selection and cut buffer to the server as well as the
+clipboard selection (default).
+
+.TP
+.B \-SendClipboard
+Send clipboard changes to the server (default).
+
+.TP
+.B \-AcceptClipboard
+Accept clipboard changes from the server (default).
+
+.TP
 .B \-Shared
 When you make a connection to a VNC server, all other existing connections are
 normally closed.  This option requests that they be left open, allowing you to
@@ -140,6 +181,19 @@ Maximize viewer window.
 Start in full-screen mode.
 
 .TP
+.B \-FullScreenAllMonitors
+Enable full screen over all monitors (default).
+
+.TP
+.B \-FullscreenSystemKeys
+Pass special keys directly to the server in full-screen mode.
+
+.TP
+.B \-RemoteResize
+Dynamically resize the remote desktop size as the size of the local
+client window changes (default).
+
+.TP
 .B \-DesktopSize \fIwidth\fPx\fIheight\fP
 Instead of keeping the existing remote screen size, the client will attempt to
 switch to the specified since when connecting. If the server does not support
@@ -214,6 +268,22 @@ command is executed with the environment
 host, the port number on the remote host, and the gateway machine
 respectively.
 
+.TP
+\fB\-ZlibLevel\fR \fIlevel\fR
+Zlib compression level.
+
+.TP
+.B \-x509crl \fIfile\fP
+X509 CRL file
+
+.TP
+.B \-x509ca \fIfile\fP
+X509 CA certificate
+
+.TP
+.B \-SecurityTypes \fItypes\fP
+Specify which security scheme to use.
+
 .SH SEE ALSO
 .BR Xvnc (1),
 .BR vncpasswd (1),
