diff -up tigervnc-1.2.80-20130314svn5065/unix/xserver/hw/vnc/XserverDesktop.cc.jx tigervnc-1.2.80-20130314svn5065/unix/xserver/hw/vnc/XserverDesktop.cc
--- tigervnc-1.2.80-20130314svn5065/unix/xserver/hw/vnc/XserverDesktop.cc.jx	2012-08-27 09:02:47.000000000 -0400
+++ tigervnc-1.2.80-20130314svn5065/unix/xserver/hw/vnc/XserverDesktop.cc	2013-06-17 09:42:14.310725996 -0400
@@ -479,6 +479,9 @@ void XserverDesktop::setCursor(CursorPtr
       }
     } else {
 #endif
+      if (!cmap)
+	return;
+
       xColorItem fg, bg;
       fg.red   = cursor->foreRed;
       fg.green = cursor->foreGreen;
