diff -up libdvdread-4.1.4/Makefile.multilib libdvdread-4.1.4/Makefile
--- libdvdread-4.1.4/Makefile.multilib	2008-12-31 09:43:03.000000000 +0100
+++ libdvdread-4.1.4/Makefile	2009-09-27 00:41:34.000000000 +0200
@@ -131,9 +131,6 @@ distclean: clean
 dvdread-config: $(.OBJDIR)/dvdread-config
 $(.OBJDIR)/dvdread-config: $(BUILDDEPS)
 	@echo '#!/bin/sh' > $(.OBJDIR)/dvdread-config
-	@echo 'prefix='$(PREFIX) >> $(.OBJDIR)/dvdread-config
-	@echo 'libdir='$(shlibdir) >> $(.OBJDIR)/dvdread-config
-	@echo 'version='$(SHLIB_VERSION) >> $(.OBJDIR)/dvdread-config
 	@echo >> $(.OBJDIR)/dvdread-config
 	cat $(SRC_PATH_BARE)/misc/dvdread-config.sh >> $(.OBJDIR)/dvdread-config
 	chmod 0755 $(.OBJDIR)/dvdread-config
diff -up libdvdread-4.1.4/misc/dvdread-config.sh.multilib libdvdread-4.1.4/misc/dvdread-config.sh
--- libdvdread-4.1.4/misc/dvdread-config.sh.multilib	2008-09-23 11:14:45.000000000 +0200
+++ libdvdread-4.1.4/misc/dvdread-config.sh	2009-09-27 00:39:39.000000000 +0200
@@ -1,4 +1,3 @@
-dvdreadlib="-ldvdread"
 
 usage()
 {
@@ -28,7 +27,7 @@ while test $# -gt 0; do
       echo_prefix=yes
       ;;
     --version)
-      echo $version
+      echo `pkg-config --modversion dvdread`
       ;;
     --cflags)
       echo_cflags=yes
@@ -44,13 +43,13 @@ while test $# -gt 0; do
 done
 
 if test "$echo_prefix" = "yes"; then
-        echo $prefix
+        echo `pkg-config --variable=prefix dvdread`
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $extracflags
+      echo `pkg-config --cflags dvdread`
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir $dvdreadlib
+      echo `pkg-config --libs dvdread`
 fi
