diff -up hdf5-1.8.10/configure.LD_LIBRARY_PATH hdf5-1.8.10/configure
--- hdf5-1.8.10/configure.LD_LIBRARY_PATH	2012-10-26 18:14:59.000000000 -0600
+++ hdf5-1.8.10/configure	2012-11-14 08:56:52.695024834 -0700
@@ -27496,9 +27496,6 @@ if ${hdf5_cv_printf_ll+:} false; then :
   $as_echo_n "(cached) " >&6
 else
 
-LD_LIBRARY_PATH="$LD_LIBRARY_PATH`echo $AM_LDFLAGS $LDFLAGS | sed -e 's/-L/:/g' -e 's/ //g'`"
-export LD_LIBRARY_PATH
-
 for hdf5_cv_printf_ll in l ll L q unknown; do
    if test "$cross_compiling" = yes; then :
   continue
diff -up hdf5-1.8.10/src/Makefile.in.LD_LIBRARY_PATH hdf5-1.8.10/src/Makefile.in
--- hdf5-1.8.10/src/Makefile.in.LD_LIBRARY_PATH	2012-10-26 18:14:55.000000000 -0600
+++ hdf5-1.8.10/src/Makefile.in	2012-11-14 08:55:35.197476914 -0700
@@ -1327,8 +1327,6 @@ help:
 # Remove the generated .c file if errors occur unless HDF5_Make_Ignore
 # is set to ignore the error.
 H5Tinit.c: H5detect$(EXEEXT)
-	LD_LIBRARY_PATH="$$LD_LIBRARY_PATH`echo $(LDFLAGS) |                  \
-		sed -e 's/-L/:/g' -e 's/ //g'`"                               \
 	$(RUNSERIAL) ./H5detect$(EXEEXT) > $@  ||                               \
 	    (test $$HDF5_Make_Ignore && echo "*** Error ignored") ||          \
 	    ($(RM) $@ ; exit 1)
@@ -1339,8 +1337,6 @@ H5Tinit.c: H5detect$(EXEEXT)
 # Remove the generated .c file if errors occur unless HDF5_Make_Ignore
 # is set to ignore the error.
 H5lib_settings.c: H5make_libsettings$(EXEEXT) libhdf5.settings
-	LD_LIBRARY_PATH="$$LD_LIBRARY_PATH`echo $(LDFLAGS) |                  \
-		sed -e 's/-L/:/g' -e 's/ //g'`"                               \
 	$(RUNSERIAL) ./H5make_libsettings$(EXEEXT) > $@  ||                               \
 	    (test $$HDF5_Make_Ignore && echo "*** Error ignored") ||          \
 	    ($(RM) $@ ; exit 1)
