diff -up hdf5-1.8.15/test/error_test.c.format hdf5-1.8.15/test/error_test.c
--- hdf5-1.8.15/test/error_test.c.format	2015-05-01 12:36:27.000000000 -0600
+++ hdf5-1.8.15/test/error_test.c	2015-05-13 20:27:31.977056438 -0600
@@ -498,7 +498,7 @@ test_create(void)
     if(err_num != 0) TEST_ERROR
 
     /* Push an error with a long description */
-    if(H5Epush(estack_id, __FILE__, err_func, __LINE__, ERR_CLS, ERR_MAJ_TEST, ERR_MIN_SUBROUTINE, err_msg) < 0) TEST_ERROR;
+    if(H5Epush(estack_id, __FILE__, err_func, __LINE__, ERR_CLS, ERR_MAJ_TEST, ERR_MIN_SUBROUTINE, "%s", err_msg) < 0) TEST_ERROR;
 
     /* Check the number of errors on stack */
     err_num = H5Eget_num(estack_id);
@@ -543,7 +543,7 @@ test_copy(void)
     herr_t      ret;                 /* Generic return value */
 
     /* Push an error with a long description */
-    if(H5Epush(H5E_DEFAULT, __FILE__, err_func, __LINE__, ERR_CLS, ERR_MAJ_TEST, ERR_MIN_SUBROUTINE, err_msg) < 0) TEST_ERROR;
+    if(H5Epush(H5E_DEFAULT, __FILE__, err_func, __LINE__, ERR_CLS, ERR_MAJ_TEST, ERR_MIN_SUBROUTINE, "%s", err_msg) < 0) TEST_ERROR;
 
     /* Check the number of errors on stack */
     err_num = H5Eget_num(H5E_DEFAULT);
