--- xinit-1.3.0/xinit.c.orig	2010-06-11 22:14:10.000000000 -0300
+++ xinit-1.3.0/xinit.c	2010-10-30 22:57:34.533765821 -0200
@@ -565,11 +565,21 @@
         set_environment();
         setWindowPath();
 
+        int fd;
         if (setuid(getuid()) == -1) {
             Error("cannot change uid");
             _exit(EXIT_FAILURE);
         }
-        setpgid(0, getpid());
+        fd = open ("/dev/null", O_RDONLY);
+
+        if (fd < 0) {
+           Error("cannot open /dev/null: %s\n", strerror(errno));
+            _exit(EXIT_FAILURE);
+        }
+        close (STDIN_FILENO);
+        dup2 (fd, STDIN_FILENO);
+        close (fd);
+        setsid();
         Execute(client);
         Error("Unable to run program \"%s\"", client[0]);
 
