--- binutils-2.26.orig/gold/dirsearch.cc	2016-02-04 10:31:45.228796199 +0000
+++ binutils-2.26/gold/dirsearch.cc	2016-02-04 11:33:15.875824355 +0000
@@ -122,6 +122,7 @@ class Dir_caches
   Cache_hash caches_;
 };
 
+#if 0
 Dir_caches::~Dir_caches()
 {
   for (Cache_hash::iterator p = this->caches_.begin();
@@ -129,6 +130,7 @@ Dir_caches::~Dir_caches()
        ++p)
     delete p->second;
 }
+#endif
 
 void
 Dir_caches::add(const char* dirname)
--- binutils-2.26.orig/gold/arm.cc	2016-02-04 10:31:45.227796193 +0000
+++ binutils-2.26/gold/arm.cc	2016-02-04 12:02:43.897996168 +0000
@@ -4493,7 +4493,7 @@ Stub::do_fixed_endian_write(unsigned cha
 // Reloc_stub::Key methods.
 
 // Dump a Key as a string for debugging.
-
+#if 0
 std::string
 Reloc_stub::Key::name() const
 {
@@ -4524,7 +4524,7 @@ Reloc_stub::Key::name() const
       return std::string(buffer);
     }
 }
-
+#endif
 // Reloc_stub methods.
 
 // Determine the type of stub needed, if any, for a relocation of R_TYPE at
--- binutils-2.26.orig/gas/config/tc-arm.c	2016-02-04 10:31:44.893794316 +0000
+++ binutils-2.26/gas/config/tc-arm.c	2016-02-04 12:41:25.534564806 +0000
@@ -155,10 +155,12 @@ static const arm_feature_set *object_arc
 
 /* Constants for known architecture features.  */
 static const arm_feature_set fpu_default = FPU_DEFAULT;
-static const arm_feature_set fpu_arch_vfp_v1 = FPU_ARCH_VFP_V1;
 static const arm_feature_set fpu_arch_vfp_v2 = FPU_ARCH_VFP_V2;
+#if 0
+static const arm_feature_set fpu_arch_vfp_v1 = FPU_ARCH_VFP_V1;
 static const arm_feature_set fpu_arch_vfp_v3 = FPU_ARCH_VFP_V3;
 static const arm_feature_set fpu_arch_neon_v1 = FPU_ARCH_NEON_V1;
+#endif
 static const arm_feature_set fpu_arch_fpa = FPU_ARCH_FPA;
 static const arm_feature_set fpu_any_hard = FPU_ANY_HARD;
 static const arm_feature_set fpu_arch_maverick = FPU_ARCH_MAVERICK;
@@ -210,7 +212,9 @@ static const arm_feature_set arm_ext_vir
 static const arm_feature_set arm_ext_pan = ARM_FEATURE_CORE_HIGH (ARM_EXT2_PAN);
 
 static const arm_feature_set arm_arch_any = ARM_ANY;
+#if 0
 static const arm_feature_set arm_arch_full = ARM_FEATURE (-1, -1, -1);
+#endif
 static const arm_feature_set arm_arch_t2 = ARM_ARCH_THUMB2;
 static const arm_feature_set arm_arch_none = ARM_ARCH_NONE;
 static const arm_feature_set arm_arch_v6m_only = ARM_ARCH_V6M_ONLY;
@@ -259,8 +263,10 @@ static const arm_feature_set fpu_crypto_
   ARM_FEATURE_COPROC (FPU_CRYPTO_EXT_ARMV8);
 static const arm_feature_set crc_ext_armv8 =
   ARM_FEATURE_COPROC (CRC_EXT_ARMV8);
+#if 0
 static const arm_feature_set fpu_neon_ext_v8_1 =
   ARM_FEATURE_COPROC (FPU_NEON_EXT_ARMV8 | FPU_NEON_EXT_RDMA);
+#endif
 
 static int mfloat_abi_opt = -1;
 /* Record user cpu selection for object attributes.  */
--- binutils-2.26.orig/gas/config/tc-arm.c	2016-05-09 17:21:37.239766096 +0100
+++ binutils-2.26/gas/config/tc-arm.c	2016-05-09 17:23:08.410319859 +0100
@@ -1982,6 +1982,11 @@ parse_neon_el_struct_list (char **str, u
   const char *const type_error = _("mismatched element/structure types in list");
   struct neon_typed_alias firsttype;
 
+  firsttype.defined = 0;
+  firsttype.eltype.type = NT_invtype;
+  firsttype.eltype.size = -1;
+  firsttype.index = -1;
+
   if (skip_past_char (&ptr, '{') == SUCCESS)
     leading_brace = 1;
 
