--- a/ld/emultempl/aarch64elf.em	2013-06-05 09:46:17.499278813 +0100
+++ b/ld/emultempl/aarch64elf.em	2013-06-05 09:54:36.981292659 +0100
@@ -264,7 +264,7 @@
 }
 
 static void
-gld${EMULATION_NAME}_finish (void)
+gld${EMULATION_NAME}_local_finish (void)
 {
   if (! link_info.relocatable)
     {
@@ -276,7 +276,7 @@
 	}
     }
 
-  finish_default ();
+  gld${EMULATION_NAME}_finish ();
 }
 
 /* This is a convenient point to tell BFD about target specific flags.
@@ -413,4 +413,4 @@
 LDEMUL_BEFORE_PARSE=gld"${EMULATION_NAME}"_before_parse
 
 # Call the extra arm-elf function
-LDEMUL_FINISH=gld${EMULATION_NAME}_finish
+LDEMUL_FINISH=gld${EMULATION_NAME}_local_finish
