diff -rcp ../binutils-2.20.51.0.7.original/bfd/ChangeLog ./bfd/ChangeLog
*** ../binutils-2.20.51.0.7.original/bfd/ChangeLog	2010-04-08 15:35:35.000000000 +0100
--- ./bfd/ChangeLog	2010-04-08 15:38:09.000000000 +0100
***************
*** 1,3 ****
--- 1,12 ----
+ 2010-04-07  Nick Clifton  <nickc@redhat.com>
+ 
+ 	Import this patch from the mainline:
+ 
+ 	2010-03-26  H.J. Lu  <hongjiu.lu@intel.com>
+ 
+ 	PR ld/11434
+ 	* elf-bfd.h (SYMBOLIC_BIND): Don't bind unique symbols locally.
+ 
  2010-03-18  H.J. Lu  <hongjiu.lu@intel.com>
  
  	PR binutils/11396
diff -rcp ../binutils-2.20.51.0.7.original/bfd/elf-bfd.h ./bfd/elf-bfd.h
*** ../binutils-2.20.51.0.7.original/bfd/elf-bfd.h	2010-04-08 15:35:35.000000000 +0100
--- ./bfd/elf-bfd.h	2010-04-08 15:37:41.000000000 +0100
*************** extern asection _bfd_elf_large_com_secti
*** 2351,2358 ****
    while (0)
  
  /* Will a symbol be bound to the the definition within the shared
!    library, if any.  */
  #define SYMBOLIC_BIND(INFO, H) \
!     ((INFO)->symbolic || ((INFO)->dynamic && !(H)->dynamic))
  
  #endif /* _LIBELF_H_ */
--- 2351,2359 ----
    while (0)
  
  /* Will a symbol be bound to the the definition within the shared
!    library, if any.  A unique symbol can never be bound locally.  */
  #define SYMBOLIC_BIND(INFO, H) \
!   (! (H)->unique_global        \
!    && ((INFO)->symbolic || ((INFO)->dynamic && !(H)->dynamic)))
  
  #endif /* _LIBELF_H_ */
