--- binutils-2.27.orig/ld/emultempl/aarch64elf.em	2016-08-19 11:53:37.647284625 +0100
+++ binutils-2.27/ld/emultempl/aarch64elf.em	2016-08-19 11:54:41.916673347 +0100
@@ -43,6 +43,7 @@ gld${EMULATION_NAME}_before_parse (void)
   input_flags.dynamic = ${DYNAMIC_LINK-TRUE};
   config.has_shared = `if test -n "$GENERATE_SHLIB_SCRIPT" ; then echo TRUE ; else echo FALSE ; fi`;
   config.separate_code = `if test "x${SEPARATE_CODE}" = xyes ; then echo TRUE ; else echo FALSE ; fi`;
+  link_info.relro = DEFAULT_LD_Z_RELRO;
 }
 
 static void
--- binutils-2.27.orig/ld/emultempl/armelf.em	2016-08-19 11:53:37.647284625 +0100
+++ binutils-2.27/ld/emultempl/armelf.em	2016-08-19 11:54:51.035728502 +0100
@@ -52,6 +52,7 @@ gld${EMULATION_NAME}_before_parse (void)
   input_flags.dynamic = ${DYNAMIC_LINK-TRUE};
   config.has_shared = `if test -n "$GENERATE_SHLIB_SCRIPT" ; then echo TRUE ; else echo FALSE ; fi`;
   config.separate_code = `if test "x${SEPARATE_CODE}" = xyes ; then echo TRUE ; else echo FALSE ; fi`;
+  link_info.relro = DEFAULT_LD_Z_RELRO;
 }
 
 static void
