diff -rcp ../binutils-2.20.51.0.7.original/bfd/configure ./bfd/configure
*** ../binutils-2.20.51.0.7.original/bfd/configure	2010-04-08 15:23:58.000000000 +0100
--- ./bfd/configure	2010-04-08 15:24:06.000000000 +0100
*************** if test "x${ac_cv_sizeof_long}" = "x8"; 
*** 12819,12829 ****
    BFD_HOST_64BIT_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long"
! elif test "x${ac_cv_sizeof_long_long}" = "x8"; then
    BFD_HOST_64BIT_LONG_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long long"
!   if test "x${ac_cv_sizeof_void_p}" = "x8"; then
      BFD_HOSTPTR_T="unsigned long long"
    fi
  fi
--- 12819,12831 ----
    BFD_HOST_64BIT_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long"
! fi
! if test "x${ac_cv_sizeof_long_long}" = "x8"; then
    BFD_HOST_64BIT_LONG_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long long"
!   if test "x${ac_cv_sizeof_void_p}" = "x8" \
!           -a "x${ac_cv_sizeof_long}" != "x8"; then
      BFD_HOSTPTR_T="unsigned long long"
    fi
  fi
diff -rcp ../binutils-2.20.51.0.7.original/bfd/configure.in ./bfd/configure.in
*** ../binutils-2.20.51.0.7.original/bfd/configure.in	2010-04-08 15:23:58.000000000 +0100
--- ./bfd/configure.in	2010-04-08 15:24:06.000000000 +0100
*************** if test "x${ac_cv_sizeof_long}" = "x8"; 
*** 153,163 ****
    BFD_HOST_64BIT_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long"
! elif test "x${ac_cv_sizeof_long_long}" = "x8"; then
    BFD_HOST_64BIT_LONG_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long long"
!   if test "x${ac_cv_sizeof_void_p}" = "x8"; then
      BFD_HOSTPTR_T="unsigned long long"
    fi
  fi
--- 153,165 ----
    BFD_HOST_64BIT_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long"
! fi
! if test "x${ac_cv_sizeof_long_long}" = "x8"; then
    BFD_HOST_64BIT_LONG_LONG=1
    test -n "${HOST_64BIT_TYPE}" || HOST_64BIT_TYPE="long long"
    test -n "${HOST_U_64BIT_TYPE}" || HOST_U_64BIT_TYPE="unsigned long long"
!   if test "x${ac_cv_sizeof_void_p}" = "x8" \
!           -a "x${ac_cv_sizeof_long}" != "x8"; then
      BFD_HOSTPTR_T="unsigned long long"
    fi
  fi
