diff -rcp ../binutils-2.20.51.0.7.original/bfd/Makefile.am ./bfd/Makefile.am
*** ../binutils-2.20.51.0.7.original/bfd/Makefile.am	2010-04-08 15:07:55.000000000 +0100
--- ./bfd/Makefile.am	2010-04-08 15:23:14.000000000 +0100
*************** bfdver.h: $(srcdir)/version.h $(srcdir)/
*** 953,964 ****
  	report_bugs_to="\"$(REPORT_BUGS_TO)\"" ;\
  	if test "x$(RELEASE)" = x ; then \
  	  bfd_version_date=`sed -n -e 's/.*DATE //p' < $(srcdir)/version.h` ;\
! 	  bfd_version_string="\"$(VERSION).$${bfd_version_date}\"" ;\
! 	  bfd_soversion="$(VERSION).$${bfd_version_date}" ;\
  	fi ;\
  	sed -e "s,@bfd_version@,$$bfd_version," \
  	    -e "s,@bfd_version_string@,$$bfd_version_string," \
! 	    -e "s,@bfd_version_package@,$$bfd_version_package," \
  	    -e "s,@report_bugs_to@,$$report_bugs_to," \
  	    < $(srcdir)/version.h > $@; \
  	echo "$${bfd_soversion}" > libtool-soversion
--- 953,964 ----
  	report_bugs_to="\"$(REPORT_BUGS_TO)\"" ;\
  	if test "x$(RELEASE)" = x ; then \
  	  bfd_version_date=`sed -n -e 's/.*DATE //p' < $(srcdir)/version.h` ;\
! 	  bfd_version_string="\"$(VERSION)-%{release} $${bfd_version_date}\"" ;\
! 	  bfd_soversion="$(VERSION)-%{release}" ;\
  	fi ;\
  	sed -e "s,@bfd_version@,$$bfd_version," \
  	    -e "s,@bfd_version_string@,$$bfd_version_string," \
! 	    -e "s,@bfd_version_package@,\"version \"," \
  	    -e "s,@report_bugs_to@,$$report_bugs_to," \
  	    < $(srcdir)/version.h > $@; \
  	echo "$${bfd_soversion}" > libtool-soversion
diff -rcp ../binutils-2.20.51.0.7.original/bfd/Makefile.in ./bfd/Makefile.in
*** ../binutils-2.20.51.0.7.original/bfd/Makefile.in	2010-04-08 15:07:55.000000000 +0100
--- ./bfd/Makefile.in	2010-04-08 15:23:14.000000000 +0100
*************** bfdver.h: $(srcdir)/version.h $(srcdir)/
*** 1982,1993 ****
  	report_bugs_to="\"$(REPORT_BUGS_TO)\"" ;\
  	if test "x$(RELEASE)" = x ; then \
  	  bfd_version_date=`sed -n -e 's/.*DATE //p' < $(srcdir)/version.h` ;\
! 	  bfd_version_string="\"$(VERSION).$${bfd_version_date}\"" ;\
! 	  bfd_soversion="$(VERSION).$${bfd_version_date}" ;\
  	fi ;\
  	sed -e "s,@bfd_version@,$$bfd_version," \
  	    -e "s,@bfd_version_string@,$$bfd_version_string," \
! 	    -e "s,@bfd_version_package@,$$bfd_version_package," \
  	    -e "s,@report_bugs_to@,$$report_bugs_to," \
  	    < $(srcdir)/version.h > $@; \
  	echo "$${bfd_soversion}" > libtool-soversion
--- 1982,1993 ----
  	report_bugs_to="\"$(REPORT_BUGS_TO)\"" ;\
  	if test "x$(RELEASE)" = x ; then \
  	  bfd_version_date=`sed -n -e 's/.*DATE //p' < $(srcdir)/version.h` ;\
! 	  bfd_version_string="\"$(VERSION)-%{release} $${bfd_version_date}\"" ;\
! 	  bfd_soversion="$(VERSION)-%{release}" ;\
  	fi ;\
  	sed -e "s,@bfd_version@,$$bfd_version," \
  	    -e "s,@bfd_version_string@,$$bfd_version_string," \
! 	    -e "s,@bfd_version_package@,\"version \"," \
  	    -e "s,@report_bugs_to@,$$report_bugs_to," \
  	    < $(srcdir)/version.h > $@; \
  	echo "$${bfd_soversion}" > libtool-soversion

