--- binutils-2.26.orig/bfd/elf.c	2016-01-25 10:11:33.482288877 +0000
+++ binutils-2.26/bfd/elf.c	2016-01-25 10:16:48.520223863 +0000
@@ -1300,6 +1300,13 @@ _bfd_elf_copy_private_bfd_data (bfd *ibf
 	}
     }
 
+  /* If the input BFD has the OSABI field set and the
+     output BFD does not, then copy the value.  */
+  if (elf_elfheader (ibfd)->e_ident [EI_OSABI] != ELFOSABI_NONE
+      && elf_elfheader (obfd)->e_ident [EI_OSABI] == ELFOSABI_NONE)
+    elf_elfheader (obfd)->e_ident [EI_OSABI] =
+      elf_elfheader (ibfd)->e_ident [EI_OSABI];
+
   return TRUE;
 }
 
