diff -rcp ../binutils-2.20.51.0.7.original/ld/ldmain.c ./ld/ldmain.c
*** ../binutils-2.20.51.0.7.original/ld/ldmain.c	2010-04-08 15:03:56.000000000 +0100
--- ./ld/ldmain.c	2010-04-08 15:05:09.000000000 +0100
*************** main (int argc, char **argv)
*** 258,271 ****
    command_line.check_section_addresses = -1;
    command_line.disable_target_specific_optimizations = -1;
  
-   if (getenv ("LD_SYMBOLIC") != NULL)
-     command_line.symbolic = symbolic;
-   else if (getenv ("LD_SYMBOLIC_FUNCTIONS") != NULL)
-     command_line.symbolic = symbolic_functions;
- 
-   if (getenv ("LD_AS_NEEDED") != NULL)
-     add_DT_NEEDED_for_regular = TRUE;
- 
    /* We initialize DEMANGLING based on the environment variable
       COLLECT_NO_DEMANGLE.  The gcc collect2 program will demangle the
       output of the linker, unless COLLECT_NO_DEMANGLE is set in the
--- 258,263 ----
diff -rcp ../binutils-2.20.51.0.7.original/ld/ld.texinfo ./ld/ld.texinfo
*** ../binutils-2.20.51.0.7.original/ld/ld.texinfo	2010-04-08 15:03:55.000000000 +0100
--- ./ld/ld.texinfo	2010-04-08 15:05:55.000000000 +0100
*************** their names to the @option{--as-needed} 
*** 1152,1161 ****
  options.  They have been replaced by @option{--copy-dt-needed-entries}
  and @option{--no-copy-dt-needed-entries}.
  
- If the environment variable @code{LD_AS_NEEDED} is set, the linker
- will behave as if the @option{--as-needed} option is passed to the
- linker as the first command line option.
- 
  @kindex -assert @var{keyword}
  @item -assert @var{keyword}
  This option is ignored for SunOS compatibility.
--- 1152,1157 ----
*************** When creating a shared library, bind ref
*** 1206,1226 ****
  definition within the shared library, if any.  Normally, it is possible
  for a program linked against a shared library to override the definition
  within the shared library.  This option is only meaningful on ELF
! platforms which support shared libraries.  If @option{-Bsymbolic} is not
! used when linking a shared library, the linker will also turn on this
! option if the environment variable @code{LD_SYMBOLIC} is set.
  
  @kindex -Bsymbolic-functions
  @item -Bsymbolic-functions
  When creating a shared library, bind references to global function
  symbols to the definition within the shared library, if any.
  This option is only meaningful on ELF platforms which support shared
! libraries.  If @option{-Bsymbolic-functions} is not used when linking a
! shared library, the linker will also turn on this option if the
! environment variable @code{LD_SYMBOLIC_FUNCTIONS} is set.  When
! both environment variables @code{LD_SYMBOLIC} and
! @code{LD_SYMBOLIC_FUNCTIONS} are set, @code{LD_SYMBOLIC} will take
! precedent.
  
  @kindex --dynamic-list=@var{dynamic-list-file}
  @item --dynamic-list=@var{dynamic-list-file}
--- 1202,1215 ----
  definition within the shared library, if any.  Normally, it is possible
  for a program linked against a shared library to override the definition
  within the shared library.  This option is only meaningful on ELF
! platforms which support shared libraries.
  
  @kindex -Bsymbolic-functions
  @item -Bsymbolic-functions
  When creating a shared library, bind references to global function
  symbols to the definition within the shared library, if any.
  This option is only meaningful on ELF platforms which support shared
! libraries.
  
  @kindex --dynamic-list=@var{dynamic-list-file}
  @item --dynamic-list=@var{dynamic-list-file}
diff -rcp ../binutils-2.20.51.0.7.original/ld/NEWS ./ld/NEWS
*** ../binutils-2.20.51.0.7.original/ld/NEWS	2010-04-08 15:03:56.000000000 +0100
--- ./ld/NEWS	2010-04-08 15:04:14.000000000 +0100
*************** Changes in 2.20:
*** 73,83 ****
    For the switch --enable-runtime-pseudo-reloc it uses for 32-bit
    runtime pseudo relocation version one, for 64-bit the version two.
  
- * ELF: Support environment variable LD_AS_NEEDED for --as-needed.
- 
- * ELF: Support environment variables, LD_SYMBOLIC for -Bsymbolic and
-   LD_SYMBOLIC_FUNCTIONS for -Bsymbolic-functions.
- 
  Changes in 2.19:
  
  * Linker scripts support a new INSERT command that makes it easier to
--- 73,78 ----
