*** ../binutils-2.22.52.0.1.orig/bfd/Makefile.am	2012-03-06 14:00:33.229957572 +0000
--- bfd/Makefile.am	2012-04-27 16:46:05.410974817 +0100
*************** if INSTALL_LIBBFD
*** 18,24 ****
  bfdlibdir = @bfdlibdir@
  bfdincludedir = @bfdincludedir@
  bfdlib_LTLIBRARIES = libbfd.la
! bfdinclude_HEADERS = $(BFD_H) $(INCDIR)/ansidecl.h $(INCDIR)/symcat.h $(INCDIR)/bfdlink.h
  else !INSTALL_LIBBFD
  # Empty these so that the respective installation directories will not be created.
  bfdlibdir =
--- 18,24 ----
  bfdlibdir = @bfdlibdir@
  bfdincludedir = @bfdincludedir@
  bfdlib_LTLIBRARIES = libbfd.la
! bfdinclude_HEADERS = $(BFD_H) $(INCDIR)/ansidecl.h $(INCDIR)/symcat.h $(INCDIR)/bfdlink.h $(INCDIR)/demangle.h
  else !INSTALL_LIBBFD
  # Empty these so that the respective installation directories will not be created.
  bfdlibdir =
*** ../binutils-2.22.52.0.1.orig/bfd/Makefile.in	2012-03-06 14:00:32.952957600 +0000
--- bfd/Makefile.in	2012-04-27 16:46:19.718975214 +0100
*************** RECURSIVE_TARGETS = all-recursive check-
*** 138,144 ****
  	installcheck-recursive installdirs-recursive pdf-recursive \
  	ps-recursive uninstall-recursive
  am__bfdinclude_HEADERS_DIST = $(INCDIR)/plugin-api.h bfd.h \
! 	$(INCDIR)/ansidecl.h $(INCDIR)/symcat.h $(INCDIR)/bfdlink.h
  HEADERS = $(bfdinclude_HEADERS)
  RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
    distclean-recursive maintainer-clean-recursive
--- 138,145 ----
  	installcheck-recursive installdirs-recursive pdf-recursive \
  	ps-recursive uninstall-recursive
  am__bfdinclude_HEADERS_DIST = $(INCDIR)/plugin-api.h bfd.h \
! 	$(INCDIR)/ansidecl.h $(INCDIR)/symcat.h $(INCDIR)/bfdlink.h \
! 	$(INCDIR)/demangle.h
  HEADERS = $(bfdinclude_HEADERS)
  RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
    distclean-recursive maintainer-clean-recursive
*************** libbfd_la_LDFLAGS = $(am__append_1) -rel
*** 331,337 ****
  @INSTALL_LIBBFD_FALSE@bfdinclude_HEADERS = $(am__append_2)
  @INSTALL_LIBBFD_TRUE@bfdinclude_HEADERS = $(BFD_H) \
  @INSTALL_LIBBFD_TRUE@	$(INCDIR)/ansidecl.h $(INCDIR)/symcat.h \
! @INSTALL_LIBBFD_TRUE@	$(INCDIR)/bfdlink.h $(am__append_2)
  @INSTALL_LIBBFD_FALSE@rpath_bfdlibdir = @bfdlibdir@
  @INSTALL_LIBBFD_FALSE@noinst_LTLIBRARIES = libbfd.la
  AM_CFLAGS = $(WARN_CFLAGS)
--- 332,339 ----
  @INSTALL_LIBBFD_FALSE@bfdinclude_HEADERS = $(am__append_2)
  @INSTALL_LIBBFD_TRUE@bfdinclude_HEADERS = $(BFD_H) \
  @INSTALL_LIBBFD_TRUE@	$(INCDIR)/ansidecl.h $(INCDIR)/symcat.h \
! @INSTALL_LIBBFD_TRUE@	$(INCDIR)/bfdlink.h $(INCDIR)/demangle.h \
! @INSTALL_LIBBFD_TRUE@	$(am__append_2)
  @INSTALL_LIBBFD_FALSE@rpath_bfdlibdir = @bfdlibdir@
  @INSTALL_LIBBFD_FALSE@noinst_LTLIBRARIES = libbfd.la
  AM_CFLAGS = $(WARN_CFLAGS)
