*** ../binutils-2.21.52.0.1.orig/bfd/elf.c	2011-06-16 13:11:38.286862001 +0100
--- bfd/elf.c	2011-06-16 13:13:45.990862002 +0100
*************** elf_find_function (bfd *abfd,
*** 7508,7513 ****
--- 7508,7516 ----
    enum { nothing_seen, symbol_seen, file_after_symbol_seen } state;
    const struct elf_backend_data *bed = get_elf_backend_data (abfd);
  
+   if (symbols == NULL)
+     return FALSE;
+ 
    filename = NULL;
    func = NULL;
    file = NULL;
