*** ../binutils-2.22.52.0.4.orig/bfd/elf32-arm.c	2012-06-29 13:29:04.526090731 +0100
--- bfd/elf32-arm.c	2012-06-29 13:29:35.977091603 +0100
*************** elf32_arm_check_relocs (bfd *abfd, struc
*** 12629,12635 ****
  
  	  /* If the symbol is a function that doesn't bind locally,
  	     this relocation will need a PLT entry.  */
! 	  root_plt->refcount += 1;
  
  	  if (!call_reloc_p)
  	    arm_plt->noncall_refcount++;
--- 12629,12636 ----
  
  	  /* If the symbol is a function that doesn't bind locally,
  	     this relocation will need a PLT entry.  */
! 	  if (root_plt->refcount != -1)
! 	    root_plt->refcount += 1;
  
  	  if (!call_reloc_p)
  	    arm_plt->noncall_refcount++;
