diff -rcp ../binutils-2.20.51.0.7-original/bfd/archive.c ./bfd/archive.c
*** ../binutils-2.20.51.0.7-original/bfd/archive.c	2010-05-04 19:08:20.000000000 +0100
--- ./bfd/archive.c	2010-05-04 19:09:40.000000000 +0100
*************** _bfd_compute_and_write_armap (bfd *arch,
*** 2196,2201 ****
--- 2196,2202 ----
  		  if ((flags & BSF_GLOBAL
  		       || flags & BSF_WEAK
  		       || flags & BSF_INDIRECT
+ 		       || flags & BSF_GNU_UNIQUE
  		       || bfd_is_com_section (sec))
  		      && ! bfd_is_und_section (sec))
  		    {
diff -rcp ../binutils-2.20.51.0.7-original/bfd/ChangeLog ./bfd/ChangeLog
*** ../binutils-2.20.51.0.7-original/bfd/ChangeLog	2010-05-04 19:08:20.000000000 +0100
--- ./bfd/ChangeLog	2010-05-04 19:11:04.000000000 +0100
***************
*** 1,3 ****
--- 1,12 ----
+ 2010-05-04  Nick Clifton  <nickc@redhat.com>
+ 
+ 	Import this patch:
+ 
+ 	2010-03-18  Matt Rice  <ratmice@gmail.com>
+ 
+         * archive.c (_bfd_compute_and_write_armap): Allow symbols flagged
+         as unique in the armap.
+ 
  2010-04-08  Nick Clifton  <nickc@redhat.com>
  
  	Import these patches from the mainline:
