diff -up emacs-24.1/lisp/org/org-exp-blocks.el.locate-library emacs-24.1/lisp/org/org-exp-blocks.el
--- emacs-24.1/lisp/org/org-exp-blocks.el.locate-library	2012-07-11 17:11:42.053056590 +0200
+++ emacs-24.1/lisp/org/org-exp-blocks.el	2012-07-11 17:11:47.716973793 +0200
@@ -240,7 +240,7 @@ which defaults to the value of `org-expo
 			      (file-name-as-directory
 			       (expand-file-name
 				"../contrib"
-				(file-name-directory (find-library-name "org")))))))
+				(file-name-directory (locate-library "org")))))))
   "Path to the ditaa jar executable."
   :group 'org-babel
   :type 'string)
