diff -Naur gpm-1.20.6.orig/src/lib/report-lib.c gpm-1.20.6/src/lib/report-lib.c
--- gpm-1.20.6.orig/src/lib/report-lib.c	2014-02-05 19:11:58.688000000 +0100
+++ gpm-1.20.6/src/lib/report-lib.c	2014-02-05 19:11:23.968000000 +0100
@@ -55,7 +55,7 @@
                            log_level = LOG_CRIT; break;
    }
 #ifdef HAVE_VSYSLOG
-   syslog(log_level, string);
+   syslog(log_level, "%s", string);
    vsyslog(log_level, text, ap);
 #else
    fprintf(stderr,"%s[%s(%d)]:\n",string,file,line);
