--- src/gs-lock-plug.c.orig	2010-09-28 17:36:51.000000000 -0300
+++ src/gs-lock-plug.c	2011-02-12 00:44:29.434257783 -0200
@@ -1090,7 +1090,7 @@ submit_note (GtkButton  *button,
         strftime (summary, 128, "%X", tmp);
 
         notify_init ("gnome-screensaver-dialog");
-        note = notify_notification_new (summary, escaped_text, NULL, NULL);
+        note = notify_notification_new (summary, escaped_text, NULL);
         notify_notification_set_timeout (note, NOTIFY_EXPIRES_NEVER);
         notify_notification_show (note, NULL);
         g_object_unref (note);
