--- net/third_party/nss/ssl/bodge/secitem_array.c.orig	2013-06-08 00:32:30.670819555 -0300
+++ net/third_party/nss/ssl/bodge/secitem_array.c	2013-06-08 00:32:34.998899621 -0300
@@ -13,11 +13,6 @@
 
 typedef struct SECItemArrayStr SECItemArray;
 
-struct SECItemArrayStr {
-    SECItem *items;
-    unsigned int len;
-};
-
 SECItemArray *
 SECITEM_AllocArray(PLArenaPool *arena, SECItemArray *array, unsigned int len)
 {
