--- trunk/src/media/ffmpeg/file_protocol.cc	2010/06/08 02:13:54	49138
+++ trunk/src/media/ffmpeg/file_protocol.cc	2010/06/08 02:28:24	49139
@@ -51,7 +51,11 @@
   return HANDLE_EINTR(read(GetHandle(h), buf, size));
 }
 
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(52, 68, 0)
+int WriteContext(URLContext* h, const unsigned char* buf, int size) {
+#else
 int WriteContext(URLContext* h, unsigned char* buf, int size) {
+#endif
   return HANDLE_EINTR(write(GetHandle(h), buf, size));
 }
 
