--- media/base/yuv_row_posix.cc.orig	2010-08-13 06:42:21.000000000 +0200
+++ media/base/yuv_row_posix.cc	2010-08-13 06:44:49.000000000 +0200
@@ -270,6 +270,7 @@
                               uint8* rgb_buf,
                               int width);
   asm(
+  ".text\n"
   ".global FastConvertYUVToRGB32Row\n"
 "FastConvertYUVToRGB32Row:\n"
   "pusha\n"
@@ -329,6 +330,7 @@
                         int width,
                         int source_dx);
   asm(
+  ".text\n"
   ".global ScaleYUVToRGB32Row\n"
 "ScaleYUVToRGB32Row:\n"
   "pusha\n"
@@ -402,6 +404,7 @@
                               int width,
                               int source_dx);
   asm(
+  ".text\n"
   ".global LinearScaleYUVToRGB32Row\n"
 "LinearScaleYUVToRGB32Row:\n"
   "pusha\n"
@@ -514,7 +517,7 @@
                                     int width,
                                     int16 *kCoefficientsRgbY);
   asm(
-
+  ".text\n"
 #if defined(OS_MACOSX)
 "_PICConvertYUVToRGB32Row:\n"
 #else
@@ -588,6 +591,7 @@
                                int16 *kCoefficientsRgbY);
 
   asm(
+  ".text\n"
 #if defined(OS_MACOSX)
 "_PICScaleYUVToRGB32Row:\n"
 #else
@@ -676,6 +680,7 @@
                                  int source_dx,
                                  int16 *kCoefficientsRgbY);
   asm(
+  ".text\n"
 #if defined(OS_MACOSX)
 "_PICLinearScaleYUVToRGB32Row:\n"
 #else
@@ -833,29 +838,6 @@
                                         (packuswb(a) << 24);
 }
 
-#if TEST_MMX_YUV
-static inline void YuvPixel(uint8 y,
-                            uint8 u,
-                            uint8 v,
-                            uint8* rgb_buf) {
-
-  asm {
-    movzx     eax, u
-    movq      mm0, [kCoefficientsRgbY+2048 + 8 * eax]
-    movzx     eax, v
-    paddsw    mm0, [kCoefficientsRgbY+4096 + 8 * eax]
-    movzx     eax, y
-    movq      mm1, [kCoefficientsRgbY + 8 * eax]
-    paddsw    mm1, mm0
-    psraw     mm1, 6
-    packuswb  mm1, mm1
-    mov       eax, rgb_buf
-    movd      [eax], mm1
-    emms
-  }
-}
-#endif
-
 void FastConvertYUVToRGB32Row(const uint8* y_buf,
                               const uint8* u_buf,
                               const uint8* v_buf,
