diff -urNp fdupes-1.50-PR2.orig/Makefile fdupes-1.50-PR2/Makefile
--- fdupes-1.50-PR2.orig/Makefile	2009-01-31 20:11:49.577968848 +0530
+++ fdupes-1.50-PR2/Makefile	2009-01-31 21:17:01.207220400 +0530
@@ -40,6 +40,8 @@ include Makefile.inc/VERSION
 #
 PROGRAM_NAME=fdupes
 
+DESTDIR =
+
 #
 # BIN_DIR indicates directory where program is to be installed. 
 # Suggested value is "$(PREFIX)/bin"
@@ -97,12 +99,12 @@ fdupes: $(OBJECT_FILES)
 	$(CC) $(CFLAGS) -o fdupes $(OBJECT_FILES)
 
 installdirs:
-	test -d $(BIN_DIR) || -$(MKDIR) $(BIN_DIR)
-	test -d $(MAN_DIR) || -$(MKDIR) $(MAN_DIR)
+	test -d $(DESTDIR)$(BIN_DIR) || $(MKDIR) $(DESTDIR)$(BIN_DIR)
+	test -d $(DESTDIR)$(MAN_DIR) || $(MKDIR) $(DESTDIR)$(MAN_DIR)
 
 install: fdupes installdirs
-	$(INSTALL_PROGRAM)	fdupes   $(BIN_DIR)/$(PROGRAM_NAME)
-	$(INSTALL_DATA)		fdupes.1 $(MAN_DIR)/$(PROGRAM_NAME).$(MAN_EXT)
+	$(INSTALL_PROGRAM)	fdupes   $(DESTDIR)$(BIN_DIR)/$(PROGRAM_NAME)
+	$(INSTALL_DATA)		fdupes.1 $(DESTDIR)$(MAN_DIR)/$(PROGRAM_NAME).$(MAN_EXT)
 
 clean:
 	$(RM) $(OBJECT_FILES)
