diff -up vdr-1.7.22/config.c~ vdr-1.7.22/config.c
--- vdr-1.7.22/config.c~	2011-12-10 00:12:51.710871792 +0200
+++ vdr-1.7.22/config.c	2011-12-10 00:13:14.325025772 +0200
@@ -397,7 +397,7 @@ cSetup::cSetup(void)
   MarginStop = 10;
   AudioLanguages[0] = -1;
   DisplaySubtitles = 0;
-  SupportTeletext = 0;
+  SupportTeletext = 1;
   SubtitleLanguages[0] = -1;
   SubtitleOffset = 0;
   SubtitleFgTransparency = 0;
