diff -up vdr-1.6.0/menu.c~ vdr-1.6.0/menu.c
--- vdr-1.6.0/menu.c~	2008-03-16 13:15:28.000000000 +0200
+++ vdr-1.6.0/menu.c	2011-02-15 00:39:42.267224859 +0200
@@ -399,11 +399,20 @@ int cMenuChannelItem::Compare(const cLis
 void cMenuChannelItem::Set(void)
 {
   cString buffer;
+  const cEvent *Event = NULL;
   if (!channel->GroupSep()) {
+     cSchedulesLock SchedulesLock;
+     const cSchedules *Schedules = cSchedules::Schedules(SchedulesLock);
+     const cSchedule *Schedule = Schedules->GetSchedule(channel->GetChannelID());
+     if (Schedule)
+        Event = Schedule->GetPresentEvent();
+
      if (sortMode == csmProvider)
-        buffer = cString::sprintf("%d\t%s - %s", channel->Number(), channel->Provider(), channel->Name());
+        buffer = cString::sprintf("%d\t%s - %s %c%s%c", channel->Number(), channel->Provider(), channel->Name(),
+                                  Event ? '(' : ' ', Event ? Event->Title() : "", Event ? ')' : ' ');
      else
-        buffer = cString::sprintf("%d\t%s", channel->Number(), channel->Name());
+        buffer = cString::sprintf("%d\t%s %c%s%c", channel->Number(), channel->Name(),
+                                  Event ? '(' : ' ', Event ? Event->Title() : "", Event ? ')' : ' ');
      }
   else
      buffer = cString::sprintf("---\t%s ----------------------------------------------------------------", channel->Name());
