diff -up vdr-1.7.37/Makefile~ vdr-1.7.37/Makefile
--- vdr-1.7.37/Makefile~	2013-02-14 21:57:22.306077727 +0200
+++ vdr-1.7.37/Makefile	2013-02-14 22:05:04.016086224 +0200
@@ -152,6 +152,12 @@ vdr.pc:
 	@echo "cflags=$(CFLAGS) $(CDEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
 	@echo "cxxflags=$(CXXFLAGS) $(CDEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
 	@echo "" >> $@
+	@echo "datadir=\$${resdir}" >> $@
+	@echo "rundir=$(RUNDIR)" >> $@
+	@echo "vardir=$(VARDIR)" >> $@
+	@echo "user=$(VDR_USER)" >> $@
+	@echo "group=$(VDR_GROUP)" >> $@
+	@echo "" >> $@
 	@echo "Name: VDR" >> $@
 	@echo "Description: Video Disk Recorder" >> $@
 	@echo "URL: http://www.tvdr.de/" >> $@
