From 40a91b44f1dfb7f19f2e3156491d721c62a2c9b3 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Tue, 19 Jul 2011 10:56:19 +0200
Subject: [PATCH] usb-redir: Add flow control support

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 hw/usb/redirect.c | 22 ++++++++++++++++++++--
 1 file changed, 20 insertions(+), 2 deletions(-)

diff --git a/hw/usb/redirect.c b/hw/usb/redirect.c
index bb07c62..4d23b66 100644
--- a/hw/usb/redirect.c
+++ b/hw/usb/redirect.c
@@ -257,8 +257,9 @@ static int usbredir_read(void *priv, uint8_t *data, int count)
 static int usbredir_write(void *priv, uint8_t *data, int count)
 {
     USBRedirDevice *dev = priv;
+    int r;
 
-    if (!dev->cs->opened) {
+    if (!dev->cs->opened || dev->cs->write_blocked) {
         return 0;
     }
 
@@ -267,7 +268,16 @@ static int usbredir_write(void *priv, uint8_t *data, int count)
         return 0;
     }
 
-    return qemu_chr_fe_write(dev->cs, data, count);
+    r = qemu_chr_fe_write(dev->cs, data, count);
+
+    if (r < 0) {
+        if (dev->cs->write_blocked) {
+           return 0;
+        }
+        return -1;
+    }
+
+    return r;
 }
 
 /*
@@ -1227,10 +1237,18 @@ static void usbredir_chardev_event(void *opaque, int event)
     }
 }
 
+static void usbredir_chardev_write_unblocked(void *opaque)
+{
+    USBRedirDevice *dev = opaque;
+
+    usbredirparser_do_write(dev->parser);
+}
+
 static const QemuChrHandlers usbredir_chr_handlers = {
     .fd_can_read = usbredir_chardev_can_read,
     .fd_read = usbredir_chardev_read,
     .fd_event = usbredir_chardev_event,
+    .fd_write_unblocked = usbredir_chardev_write_unblocked,
 };
 
 /*
