--- minitube.pro.orig	2013-02-02 21:49:59.000000000 -0200
+++ minitube.pro	2013-02-03 12:57:49.708924637 -0200
@@ -1,4 +1,4 @@
-CONFIG += release
+CONFIG += release qtsingleapplication
 TEMPLATE = app
 VERSION = 2.0
 DEFINES += APP_VERSION="$$VERSION"
@@ -15,7 +15,6 @@ DEFINES += QT_STRICT_ITERATORS
 
 TARGET = minitube
 QT += network xml phonon
-include(src/qtsingleapplication/qtsingleapplication.pri)
 HEADERS += \
     src/video.h \
     src/searchlineedit.h \
