--- minitube.pro.orig	2010-10-12 12:19:38.558066522 -0300
+++ minitube.pro	2010-10-12 12:28:33.068660167 -0300
@@ -1,4 +1,4 @@
-CONFIG += release
+CONFIG += release qtsingleapplication
 TEMPLATE = app
 VERSION = 1.2
 DEFINES += APP_VERSION="$$VERSION"
@@ -10,7 +10,6 @@
 QT += network \
     xml \
     phonon
-include(src/qtsingleapplication/qtsingleapplication.pri)
 HEADERS += src/MainWindow.h \
     src/SearchView.h \
     src/MediaView.h \
