--- minitube.pro.orig	2015-04-08 06:38:01.000000000 -0300
+++ minitube.pro	2015-04-23 18:57:04.224396472 -0300
@@ -1,4 +1,4 @@
-CONFIG += release
+CONFIG += release qtsingleapplication
 TEMPLATE = app
 VERSION = 2.4
 DEFINES += APP_VERSION="$$VERSION"
@@ -29,8 +29,6 @@ qt:greaterThan(QT_MAJOR_VERSION, 4) {
     contains(QT, gui): QT *= widgets
 }
 
-include(src/qtsingleapplication/qtsingleapplication.pri)
-
 HEADERS += src/video.h \
     src/searchlineedit.h \
     src/urllineedit.h \
