diff -up src/Constants.h.config src/Constants.h
--- src/Constants.h.config	2010-05-03 18:08:27.000000000 -0300
+++ src/Constants.h	2010-07-06 20:03:31.484315685 -0300
@@ -7,6 +7,7 @@ namespace Constants {
     static const char *VERSION = "1.0";
     static const char *APP_NAME = "Minitube";
     static const char *ORG_NAME = "Flavio Tordini";
+    static const char *ORG_UNAME = "Flavio_Tordini";
     static const char *ORG_DOMAIN = "flavio.tordini.org";
     static const char *WEBSITE = "http://flavio.tordini.org/minitube";
     static const char *EMAIL = "flavio.tordini@gmail.com";
diff -up src/main.cpp.config src/main.cpp
--- src/main.cpp.config	2010-05-03 18:08:27.000000000 -0300
+++ src/main.cpp	2010-07-06 20:03:23.485254604 -0300
@@ -10,7 +10,7 @@ int main(int argc, char **argv) {
         return 0;
 
     app.setApplicationName(Constants::APP_NAME);
-    app.setOrganizationName(Constants::ORG_NAME);
+    app.setOrganizationName(Constants::ORG_UNAME);
     app.setOrganizationDomain(Constants::ORG_DOMAIN);
 #ifndef Q_WS_MAC
     app.setWheelScrollLines(1);
