--- minitube.pro.orig	2013-12-18 21:11:13.000000000 -0200
+++ minitube.pro	2014-01-02 17:56:35.212674269 -0200
@@ -1,4 +1,4 @@
-CONFIG += release
+CONFIG += release qtsingleapplication
 TEMPLATE = app
 VERSION = 2.1.5
 DEFINES += APP_VERSION="$$VERSION"
@@ -17,8 +17,6 @@ TARGET = $${APP_UNIX_NAME}
 
 QT += network xml phonon sql script
 
-include(src/qtsingleapplication/qtsingleapplication.pri)
-
 HEADERS += \
     src/video.h \
     src/searchlineedit.h \
