diff -up src/constants.cpp.config src/constants.cpp
--- src/constants.cpp.config	2015-04-08 06:38:01.000000000 -0300
+++ src/constants.cpp	2015-04-23 18:55:15.181494745 -0300
@@ -27,6 +27,7 @@ const char *Constants::VERSION = STRINGI
 const char *Constants::NAME = STRINGIFY(APP_NAME);
 const char *Constants::UNIX_NAME = STRINGIFY(APP_UNIX_NAME);
 const char *Constants::ORG_NAME = "Flavio Tordini";
+const char *Constants::ORG_UNAME = "Flavio_Tordini";
 const char *Constants::ORG_DOMAIN = "flavio.tordini.org";
 const char *Constants::WEBSITE = "http://flavio.tordini.org/minitube";
 const char *Constants::EMAIL = "flavio.tordini@gmail.com";
diff -up src/constants.h.config src/constants.h
--- src/constants.h.config	2015-04-08 06:38:01.000000000 -0300
+++ src/constants.h	2015-04-23 18:55:15.181494745 -0300
@@ -31,6 +31,7 @@ public:
     static const char *NAME;
     static const char *UNIX_NAME;
     static const char *ORG_NAME;
+    static const char *ORG_UNAME;
     static const char *ORG_DOMAIN;
     static const char *WEBSITE;
     static const char *EMAIL;
diff -up src/main.cpp.config src/main.cpp
--- src/main.cpp.config	2015-04-23 18:55:15.181494745 -0300
+++ src/main.cpp	2015-04-23 18:55:57.910443583 -0300
@@ -77,7 +77,7 @@ int main(int argc, char **argv) {
         return 0;
 
     app.setApplicationName(Constants::NAME);
-    app.setOrganizationName(Constants::ORG_NAME);
+    app.setOrganizationName(Constants::ORG_UNAME);
     app.setOrganizationDomain(Constants::ORG_DOMAIN);
     app.setApplicationVersion(Constants::VERSION);
     app.setAttribute(Qt::AA_DontShowIconsInMenus);
