diff -up koffice-2.2.0/kspread/CMakeLists.txt.kutils koffice-2.2.0/kspread/CMakeLists.txt
--- koffice-2.2.0/kspread/CMakeLists.txt.kutils	2010-05-23 01:20:50.000000000 -0500
+++ koffice-2.2.0/kspread/CMakeLists.txt	2010-05-25 07:41:58.461302746 -0500
@@ -221,7 +221,7 @@ target_link_libraries(kspreadcommon
     komain
     ${QT_QTSQL_LIBRARY}
     ${KDE4_KNOTIFYCONFIG_LIBS}
-    kutils
+    ${KDE4_KUTILS_LIBS}
 #    ${KDE4_KDE3SUPPORT_LIBS}
 )
 
