diff -up koffice-2.3.3/filters/kword/msword-odf/wv2/src/olestream.h.glib koffice-2.3.3/filters/kword/msword-odf/wv2/src/olestream.h
--- koffice-2.3.3/filters/kword/msword-odf/wv2/src/olestream.h.glib	2011-02-17 06:12:35.000000000 -0600
+++ koffice-2.3.3/filters/kword/msword-odf/wv2/src/olestream.h	2011-11-04 08:33:12.410475134 -0500
@@ -23,7 +23,7 @@
 #include "global.h"  // U8,... typedefs
 #include <stack>
 
-#include <glib/giochannel.h> // GSeekType
+#include <glib.h> // GSeekType
 #include "wv2_export.h"
 
 namespace wvWare {
