diff -up koffice-2.1.91/libs/flake/KoToolBase_p.h.kdebug235345-2 koffice-2.1.91/libs/flake/KoToolBase_p.h
--- koffice-2.1.91/libs/flake/KoToolBase_p.h.kdebug235345-2	2010-04-23 05:12:32.000000000 -0500
+++ koffice-2.1.91/libs/flake/KoToolBase_p.h	2010-05-12 09:44:38.774150294 -0500
@@ -1,5 +1,6 @@
 /* This file is part of the KDE project
  * Copyright (C) 2006-2010 Thomas Zander <zander@kde.org>
+ * Copyright (C) 2010 KO GmbH <boud@kogmbh.com>
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Library General Public
@@ -42,7 +43,14 @@ public:
 
     ~KoToolBasePrivate()
     {
-        qDeleteAll(optionWidgets);
+       foreach(QWidget *optionWidget, optionWidgets) {
+           // Only delete the option widget if it hasn't got a parent; if it has,
+           // the parent will delete it.
+           if (optionWidget->parent() == 0) {
+               delete optionWidget;
+           }
+       }
+       optionWidgets.clear();
     }
 
     QMap<QString, QWidget *> optionWidgets; ///< the optionwidgets associated witth this tool
