diff -up SQUIRREL2/squirrel/sqmem.h.mem SQUIRREL2/squirrel/sqmem.h
--- SQUIRREL2/squirrel/sqmem.h.mem	2012-01-13 16:59:34.000000000 +0100
+++ SQUIRREL2/squirrel/sqmem.h	2012-01-13 17:00:19.000000000 +0100
@@ -0,0 +1,8 @@
+/*	see copyright notice in squirrel.h */
+#ifndef _SQMEM_H_
+#define _SQMEM_H_
+
+void *sq_vm_malloc(SQUnsignedInteger size);
+void *sq_vm_realloc(void *p,SQUnsignedInteger oldsize,SQUnsignedInteger size);
+void sq_vm_free(void *p,SQUnsignedInteger size);
+#endif //_SQMEM_H_
diff -up SQUIRREL2/squirrel/sqstate.h.mem SQUIRREL2/squirrel/sqstate.h
--- SQUIRREL2/squirrel/sqstate.h.mem	2012-01-13 16:59:28.000000000 +0100
+++ SQUIRREL2/squirrel/sqstate.h	2012-01-13 16:59:51.000000000 +0100
@@ -136,8 +136,4 @@ extern SQObjectPtr _one_;
 extern SQObjectPtr _minusone_;
 
 bool CompileTypemask(SQIntVec &res,const SQChar *typemask);
-
-void *sq_vm_malloc(SQUnsignedInteger size);
-void *sq_vm_realloc(void *p,SQUnsignedInteger oldsize,SQUnsignedInteger size);
-void sq_vm_free(void *p,SQUnsignedInteger size);
 #endif //_SQSTATE_H_
diff -up SQUIRREL2/squirrel/squtils.h.mem SQUIRREL2/squirrel/squtils.h
--- SQUIRREL2/squirrel/squtils.h.mem	2012-01-13 16:59:41.000000000 +0100
+++ SQUIRREL2/squirrel/squtils.h	2012-01-13 17:00:01.000000000 +0100
@@ -2,6 +2,8 @@
 #ifndef _SQUTILS_H_
 #define _SQUTILS_H_
 
+#include "sqmem.h"
+
 #define sq_new(__ptr,__type) {__ptr=(__type *)sq_vm_malloc(sizeof(__type));new (__ptr) __type;}
 #define sq_delete(__ptr,__type) {__ptr->~__type();sq_vm_free(__ptr,sizeof(__type));}
 #define SQ_MALLOC(__size) sq_vm_malloc((__size));
