diff -up htop-0.8.3/acinclude.m4.noplpa htop-0.8.3/acinclude.m4
--- htop-0.8.3/acinclude.m4.noplpa	2010-03-04 15:35:47.457908679 -0500
+++ htop-0.8.3/acinclude.m4	2010-03-04 15:35:50.721919708 -0500
@@ -1 +0,0 @@
-m4_include(plpa-1.1/plpa.m4)
diff -up htop-0.8.3/configure.ac.noplpa htop-0.8.3/configure.ac
--- htop-0.8.3/configure.ac.noplpa	2009-06-23 09:53:18.000000000 -0400
+++ htop-0.8.3/configure.ac	2010-03-04 15:34:41.578909559 -0500
@@ -97,12 +97,7 @@ fi
 AC_CHECK_FILE($PROCDIR/stat,,AC_MSG_ERROR(Cannot find /proc/stat. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
 AC_CHECK_FILE($PROCDIR/meminfo,,AC_MSG_ERROR(Cannot find /proc/meminfo. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
 
-PLPA_INCLUDED(plpa-1.1)
-PLPA_INIT(plpa_happy=yes, plpa_happy=no)
-AM_CONDITIONAL([HAVE_PLPA], [test "$plpa_happy" = "yes"])
-if test "$plpa_happy" = "yes"; then
-   AC_DEFINE([HAVE_PLPA], [1], [Have plpa])
-fi
+AC_CHECK_LIB([plpa], [plpa_sched_getaffinity], [], [missing_libraries="$missing_libraries libplpa"])
 
 AC_CONFIG_FILES([Makefile htop.1])
 AC_OUTPUT
diff -up htop-0.8.3/Makefile.am.noplpa htop-0.8.3/Makefile.am
--- htop-0.8.3/Makefile.am.noplpa	2009-06-02 15:17:57.000000000 -0400
+++ htop-0.8.3/Makefile.am	2010-03-04 15:34:41.579909477 -0500
@@ -1,8 +1,4 @@
 
-if HAVE_PLPA
-SUBDIRS = plpa-1.1
-endif
-
 bin_PROGRAMS = htop
 dist_man_MANS = htop.1
 EXTRA_DIST = $(dist_man_MANS) htop.desktop htop.png scripts/MakeHeader.py \
@@ -37,9 +33,6 @@ SUFFIXES = .h
 
 BUILT_SOURCES = $(myhtopheaders)
 htop_SOURCES = $(myhtopheaders) $(myhtopsources) config.h debug.h
-if HAVE_PLPA
-htop_LDADD = $(top_builddir)/plpa-1.1/src/libplpa_included.la
-endif
 
 profile:
 	$(MAKE) all CFLAGS="-pg -O2"
