diff -up htop-0.9/Makefile.am.noplpa htop-0.9/Makefile.am
--- htop-0.9/Makefile.am.noplpa	2010-11-23 13:56:32.000000000 -0200
+++ htop-0.9/Makefile.am	2010-11-27 21:03:59.146757417 -0200
@@ -1,8 +1,4 @@
 
-if HAVE_PLPA
-SUBDIRS = plpa-1.3.2
-endif
-
 ACLOCAL_AMFLAGS=-I m4
 
 bin_PROGRAMS = htop
@@ -39,9 +35,6 @@ SUFFIXES = .h
 
 BUILT_SOURCES = $(myhtopheaders)
 htop_SOURCES = $(myhtopheaders) $(myhtopsources) config.h debug.h
-if HAVE_PLPA
-htop_LDADD = $(top_builddir)/plpa-1.3.2/src/libplpa/libplpa_included.la
-endif
 
 profile:
 	$(MAKE) all CFLAGS="-pg -O2"
diff -up htop-0.9/acinclude.m4.noplpa htop-0.9/acinclude.m4
--- htop-0.9/acinclude.m4.noplpa	2010-03-29 16:01:44.000000000 -0300
+++ htop-0.9/acinclude.m4	2010-11-27 21:03:14.277829733 -0200
@@ -1 +1 @@
-m4_include(plpa-1.3.2/plpa.m4)
+
diff -up htop-0.9/configure.ac.noplpa htop-0.9/configure.ac
--- htop-0.9/configure.ac.noplpa	2010-11-23 13:56:32.000000000 -0200
+++ htop-0.9/configure.ac	2010-11-27 21:03:14.278829487 -0200
@@ -104,14 +104,7 @@ fi
 AC_CHECK_FILE($PROCDIR/stat,,AC_MSG_ERROR(Cannot find /proc/stat. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
 AC_CHECK_FILE($PROCDIR/meminfo,,AC_MSG_ERROR(Cannot find /proc/meminfo. Make sure you have a Linux-compatible /proc filesystem mounted. See the file README for help.))
 
-AC_ARG_ENABLE(plpa, [AC_HELP_STRING([--enable-plpa], [enable PLPA support for CPU affinity])], ,enable_plpa="yes")
-PLPA_INCLUDED
-PLPA_INIT([plpa-1.3.2], [plpa_happy=yes], [plpa_happy=no])
-AM_CONDITIONAL([HAVE_PLPA], [test "$plpa_happy" = "yes" && test "$enable_plpa" = "yes"])
-if test "$plpa_happy" = "yes" && test "$enable_plpa" = "yes"
-then
-   AC_DEFINE([HAVE_PLPA], [1], [Have plpa])
-fi
+AC_CHECK_LIB([plpa], [plpa_sched_getaffinity], [], [missing_libraries="$missing_libraries libplpa"])
 
 AC_CONFIG_FILES([Makefile htop.1])
 AC_OUTPUT
