--- Makefile.orig	2010-05-19 01:44:05.000000000 -0300
+++ Makefile	2010-05-23 01:07:16.751156402 -0300
@@ -37,9 +37,9 @@
 
 ifndef X_LDFLAGS
   ifeq ($(TARGET_OS)-$(TARGET_ARCH),Linux-x86_64)
-    X_LDFLAGS          = -L/usr/X11R6/lib64
+    X_LDFLAGS          = -L/usr/lib64
   else
-    X_LDFLAGS          = -L/usr/X11R6/lib
+    X_LDFLAGS          = -L/usr/lib
   endif
 endif
 
@@ -69,7 +69,7 @@
 ##############################################################################
 
 XNVCTRL_DIR           ?= src/libXNVCtrl
-XNVCTRL_ARCHIVE       ?= $(XNVCTRL_DIR)/libXNVCtrl.a
+XNVCTRL_LIB           ?= -lXNVCtrl
 XCONFIG_PARSER_DIR    ?= src/XF86Config-parser
 
 
@@ -102,7 +102,7 @@
     LDFLAGS += -lXxf86vm
 endif
 
-LDFLAGS += -lX11 -lXext -lm
+LDFLAGS += -lX11 -ldl -lm
 LDFLAGS += $(GTK_LDFLAGS)
 
 MANPAGE_GZIP ?= 1
@@ -155,8 +155,8 @@
 	$(MKDIR) $(mandir)
 	$(INSTALL) $(INSTALL_BIN_ARGS) $< $(mandir)/$(notdir $<)
 
-$(NVIDIA_SETTINGS): $(OBJS) $(XNVCTRL_ARCHIVE)
-	$(call quiet_cmd,LINK) -o $@ $(OBJS) $(XNVCTRL_ARCHIVE) \
+$(NVIDIA_SETTINGS): $(OBJS)
+	$(call quiet_cmd,LINK) -o $@ $(OBJS) $(XNVCTRL_LIB) \
 		$(CFLAGS) $(LDFLAGS) $(BIN_LDFLAGS)
 	$(call quiet_cmd,STRIP_CMD) $@
 
