--- Mowitz/MwXFormat.c.orig	2013-03-31 02:41:47.565194271 -0300
+++ Mowitz/MwXFormat.c	2013-03-31 02:42:02.978912275 -0300
@@ -1158,7 +1158,7 @@ int MwT1Init(Display *dpy)
                 return -1;
         }
 
-        t1_nfont = T1_Get_no_fonts();
+        t1_nfont = T1_GetNoFonts();
 
         T1_SetX11Params(t1_dpy, DefaultVisual(t1_dpy, t1_scrn),
                 DefaultDepth(t1_dpy, t1_scrn),
