--- Mowitz/MwFilesel.c~	2007-08-26 09:59:58.000000000 +0200
+++ Mowitz/MwFilesel.c	2007-08-28 00:43:07.000000000 +0200
@@ -299,17 +299,15 @@
 
 static void change_dir(Widget w, XtPointer client_data, XtPointer call_data)
 {
-	char path[1024];
-	char newpath[1024];
+	char newpath[PATH_MAX];
 
-	strcpy(path, (char *)client_data);
-
-	if (!realpath(path, newpath))
+	if (!realpath((char *)client_data, newpath)) {
-		fprintf(stderr, "Couldn't realpath %s\n", path);
+		fprintf(stderr, "Couldn't realpath %s\n", (char *)client_data);
+	} else {
+		MwLabelSet(fsel_dirbutton, newpath);
 
-	MwLabelSet(fsel_dirbutton, newpath);
-
-	fsel_scan();
+		fsel_scan();
+	}
 }
 
 static void make_dirmenu(char *dir)
