From 27f609075de02f767611d455ba3f9e29c2986001 Mon Sep 17 00:00:00 2001
From: Kevin Funk <krf@electrostorm.net>
Date: Fri, 28 May 2010 12:36:19 +0200
Subject: [PATCH 1/4] Move some code in TrayIcon class.

---
 src/TrayIcon.cpp |   29 ++++++++++-------------------
 src/TrayIcon.h   |    1 +
 2 files changed, 11 insertions(+), 19 deletions(-)

diff --git a/src/TrayIcon.cpp b/src/TrayIcon.cpp
index 061bc08..fb97483 100644
--- a/src/TrayIcon.cpp
+++ b/src/TrayIcon.cpp
@@ -64,13 +64,13 @@ Amarok::TrayIcon::TrayIcon( QObject *parent )
 
     PERF_LOG( "Before adding actions" );
 
-    #ifdef Q_WS_MAC
+#ifdef Q_WS_MAC
     // Add these functions to the dock icon menu in OS X
     extern void qt_mac_set_dock_menu(QMenu *);
     qt_mac_set_dock_menu( contextMenu() );
     contextMenu()->addAction( ac->action( "playlist_playmedia" ) );
     contextMenu()->addSeparator();
-    #endif
+#endif
 
     contextMenu()->addAction( ac->action( "prev"       ) );
     contextMenu()->addAction( ac->action( "play_pause" ) );
@@ -79,8 +79,15 @@ Amarok::TrayIcon::TrayIcon( QObject *parent )
     contextMenu()->setObjectName( "TrayIconContextMenu" );
 
     PERF_LOG( "Adding system tray icon" );
-    paintIcon();
 
+    m_baseIcon = KIconLoader::global()->loadIcon( "amarok", KIconLoader::Panel );
+    setIconByPixmap( m_baseIcon ); // show icon
+    setOverlayIconByName( QString() );
+
+    m_grayedIcon = m_baseIcon; // copies object
+    KIconEffect::semiTransparent( m_grayedIcon );
+
+    paintIcon();
     setupToolTip();
 
     connect( this, SIGNAL( scrollRequested( int, Qt::Orientation ) ), SLOT( slotScrollRequested(int, Qt::Orientation) ) );
@@ -288,22 +295,6 @@ Amarok::TrayIcon::paintIcon( qint64 trackPosition )
 {
     static qint64 oldMergePos = -1;
 
-    // start up
-    // TODO: Move these two blocks to ctor (warning: might get some regressions)
-    if( m_baseIcon.isNull() )
-    {
-        m_baseIcon = KIconLoader::global()->loadIcon( "amarok", KIconLoader::Panel );
-        setIconByPixmap( m_baseIcon ); // show icon
-        setOverlayIconByName( QString() );
-        return; // HACK: return because m_baseIcon is still null after first startup (why?)
-    }
-
-    if( m_grayedIcon.isNull() )
-    {
-        m_grayedIcon = m_baseIcon; // copies object
-        KIconEffect::semiTransparent( m_grayedIcon );
-    }
-
     // trackPosition < 0 means reset
     if( trackPosition < 0 )
     {
diff --git a/src/TrayIcon.h b/src/TrayIcon.h
index 46a265b..c2602df 100644
--- a/src/TrayIcon.h
+++ b/src/TrayIcon.h
@@ -40,6 +40,7 @@ class TrayIcon : public KStatusNotifierItem, public Engine::EngineObserver, publ
 
 public:
     TrayIcon( QObject *parent );
+
     friend class ::App;
     
     void setVisible( bool visible );
-- 
1.7.0.4

