From 50bf5a7e55b08426fd60f4928e65ac066349f83d Mon Sep 17 00:00:00 2001
From: Bartosz Brachaczek <b.brachaczek@gmail.com>
Date: Tue, 23 Mar 2010 22:53:13 +0100
Subject: [PATCH] Fix logic error in EngineController::supportedMimeTypes causing mp3 support misdetection

Signed-off-by: Bartosz Brachaczek <b.brachaczek@gmail.com>
---
 src/EngineController.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/EngineController.cpp b/src/EngineController.cpp
index 79acb4f..87745ea 100644
--- a/src/EngineController.cpp
+++ b/src/EngineController.cpp
@@ -239,7 +239,7 @@ EngineController::supportedMimeTypes()
     mimeTable << "audio/x-m4b"; // MP4 Audio Books have a different extension that KFileItem/Phonon don't grok
 
     // We special case this, as otherwise the users would hate us
-    if( ( !mimeTable.contains( "audio/mp3" ) || !mimeTable.contains( "audio/x-mp3" ) ) && !installDistroCodec() )
+    if( ( !mimeTable.contains( "audio/mp3" ) && !mimeTable.contains( "audio/x-mp3" ) ) && !installDistroCodec() )
     {
         The::statusBar()->longMessage(
                 i18n( "<p>Phonon claims it <b>cannot</b> play MP3 files. You may want to examine "
-- 
1.6.1

