diff -up amarok-2.3.1/src/core-impl/collections/ipodcollection/handler/IpodHandler.cpp.signals_fix amarok-2.3.1/src/core-impl/collections/ipodcollection/handler/IpodHandler.cpp
--- amarok-2.3.1/src/core-impl/collections/ipodcollection/handler/IpodHandler.cpp.signals_fix	2010-05-27 13:37:21.000000000 -0500
+++ amarok-2.3.1/src/core-impl/collections/ipodcollection/handler/IpodHandler.cpp	2010-06-10 13:57:50.291880276 -0500
@@ -28,6 +28,10 @@
 extern "C" {
 #include <glib-object.h> // g_type_init
 #ifdef GDK_FOUND
+// work around compile issue with glib >= 2.25
+#ifdef signals
+  #undef signals
+#endif
 #include <gdk-pixbuf/gdk-pixbuf.h>
 #endif
 }
