diff -up gview_v4l2core/jpeg_decoder.c.ffmpeg gview_v4l2core/jpeg_decoder.c
--- gview_v4l2core/jpeg_decoder.c.ffmpeg	2015-09-20 12:39:09.498191641 -0300
+++ gview_v4l2core/jpeg_decoder.c	2015-09-20 12:39:20.342898813 -0300
@@ -1436,7 +1436,7 @@ int jpeg_init_decoder(int width, int hei
 		exit(-1);
 	}
 
-	codec_data->context->pix_fmt = PIX_FMT_YUV422P;
+	codec_data->context->pix_fmt = AV_PIX_FMT_YUV422P;
 	codec_data->context->width = width;
 	codec_data->context->height = height;
 	//jpeg_ctx->context->dsp_mask = (FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE);
diff -up gview_v4l2core/uvc_h264.c.ffmpeg gview_v4l2core/uvc_h264.c
--- gview_v4l2core/uvc_h264.c.ffmpeg	2015-09-20 12:38:28.201922648 -0300
+++ gview_v4l2core/uvc_h264.c	2015-09-20 12:38:41.992007098 -0300
@@ -970,7 +970,7 @@ int h264_init_decoder(int width, int hei
 	}
 	
 	h264_ctx->context->flags2 |= CODEC_FLAG2_FAST;
-	h264_ctx->context->pix_fmt = PIX_FMT_YUV420P;
+	h264_ctx->context->pix_fmt = AV_PIX_FMT_YUV420P;
 	h264_ctx->context->width = width;
 	h264_ctx->context->height = height;
 	//h264_ctx->context->dsp_mask = (FF_MM_MMX | FF_MM_MMXEXT | FF_MM_SSE);
