diff -up extremetuxracer-0.4/src/event_race_select.cpp.random-keypress-crash extremetuxracer-0.4/src/event_race_select.cpp
--- extremetuxracer-0.4/src/event_race_select.cpp.random-keypress-crash	2007-09-19 18:04:42.000000000 +0200
+++ extremetuxracer-0.4/src/event_race_select.cpp	2010-11-15 14:10:24.176037375 +0100
@@ -451,13 +451,19 @@ EventRaceSelect::keyPressEvent(SDLKey ke
 			UIMgr.setDirty();
 			return true;
 		case 'c': 
-	    	mp_conditionsSSBtn->simulateMouseClick();
+			if ( mp_conditionsSSBtn != 0 ) {
+				mp_conditionsSSBtn->simulateMouseClick();
+			}
 	    	return true;
 		case 'w': 
-	    	mp_windSSBtn->simulateMouseClick();
+			if ( mp_windSSBtn != 0 ) {
+				mp_windSSBtn->simulateMouseClick();
+			}
 	    	return true;
 		case 'm':
-	    	mp_mirrorSSBtn->simulateMouseClick();
+			if ( mp_mirrorSSBtn != 0 ) {
+				mp_mirrorSSBtn->simulateMouseClick();
+			}
 	    	return true;
 		default:
 			return false;
