diff -up extremetuxracer-0.4/src/highscore.cpp.memset extremetuxracer-0.4/src/highscore.cpp
--- extremetuxracer-0.4/src/highscore.cpp.memset	2007-09-20 21:47:02.000000000 +0200
+++ extremetuxracer-0.4/src/highscore.cpp	2011-11-09 10:59:39.720032708 +0100
@@ -213,19 +213,19 @@ highscore::loadData()
 		bool end=false;
 		bool loop=false;
 		while(!end) {
-			memset(buff,256,'\0');
+			memset(buff,'\0',256);
 			sfile >> buff;
 			if(std::string("<level>").compare(buff)==0) {
 				int n;
 				redim();
 				n = level_hs_length-1;
-				memset(buff,256,'\0');
+				memset(buff,'\0',256);
 				sfile >> buff;
 				level_hs[n].level.assign(fromOutputFormat(buff));
 				loop=true;
 				for(_posts=0;loop&&_posts<10;_posts++) {
 					char nick[256];
-					memset(nick,256,'\0');
+					memset(nick,'\0',256);
 					int score;
 					sfile >> nick >> score;
 					if(std::string("</level>").compare(nick)!=0) {
