--- configure.orig	2002-07-28 00:50:42.000000000 -0300
+++ configure	2012-09-20 00:22:11.212361810 -0300
@@ -2751,9 +2751,7 @@ if test x"$GCC" = x"yes"; then
 
 
 
-    OPT_CFLAGS=`echo "$CFLAGS"|sed "s/-O[0-9]*//g"`
-
-    OPT_CFLAGS="$OPT_CFLAGS -O3"
+    OPT_CFLAGS="$CFLAGS"
     echo "$as_me:$LINENO: checking if $CC supports $OPT_CFLAGS flags" >&5
 echo $ECHO_N "checking if $CC supports $OPT_CFLAGS flags... $ECHO_C" >&6
     SAVE_CFLAGS="$CFLAGS"
@@ -2804,7 +2802,7 @@ echo "${ECHO_T}$ac_cv_try_cflags_ok" >&6
 	:
     fi
 
-        OPT_CFLAGS="$CFLAGS -fomit-frame-pointer"
+        OPT_CFLAGS="$CFLAGS"
     echo "$as_me:$LINENO: checking if $CC supports $OPT_CFLAGS flags" >&5
 echo $ECHO_N "checking if $CC supports $OPT_CFLAGS flags... $ECHO_C" >&6
     SAVE_CFLAGS="$CFLAGS"
@@ -2857,13 +2855,13 @@ echo "${ECHO_T}$ac_cv_try_cflags_ok" >&6
 
         case "$host" in
     i?86-* | k?-*)
-	case "$host" in
-	i386-*) OPT_CFLAGS="$CFLAGS -mcpu=i386";;
-	i486-*) OPT_CFLAGS="$CFLAGS -mcpu=i486";;
-	i586-*) OPT_CFLAGS="$CFLAGS -mcpu=pentium";;
-	i686-*) OPT_CFLAGS="$CFLAGS -mcpu=pentiumpro";;
-	k6-*)   OPT_CFLAGS="$CFLAGS -mcpu=k6";;
-	esac
+#	case "$host" in
+#	i386-*) OPT_CFLAGS="$CFLAGS -mcpu=i386";;
+#	i486-*) OPT_CFLAGS="$CFLAGS -mcpu=i486";;
+#	i586-*) OPT_CFLAGS="$CFLAGS -mcpu=pentium";;
+#	i686-*) OPT_CFLAGS="$CFLAGS -mcpu=pentiumpro";;
+#	k6-*)   OPT_CFLAGS="$CFLAGS -mcpu=k6";;
+#	esac
 	echo "$as_me:$LINENO: checking if $CC supports $OPT_CFLAGS flags" >&5
 echo $ECHO_N "checking if $CC supports $OPT_CFLAGS flags... $ECHO_C" >&6
     SAVE_CFLAGS="$CFLAGS"
