diff -up libvpx-0.9.0/examples.mk.BAD libvpx-0.9.0/examples.mk
--- libvpx-0.9.0/examples.mk.BAD	2010-05-19 17:58:42.611058552 -0400
+++ libvpx-0.9.0/examples.mk	2010-05-19 17:59:11.934181029 -0400
@@ -129,7 +129,6 @@ BINS-$(NOT_MSVS)           += $(addprefi
 # Instantiate linker template for all examples.
 CODEC_LIB=$(if $(CONFIG_DEBUG_LIBS),vpx_g,vpx)
 $(foreach bin,$(BINS-yes),\
-    $(if $(BUILD_OBJS),$(eval $(bin): $(LIB_PATH)/lib$(CODEC_LIB).a))\
     $(if $(BUILD_OBJS),$(eval $(call linker_template,$(bin),\
         $(call objs,$($(notdir $(bin)).SRCS)) \
         -l$(CODEC_LIB) $(addprefix -l,$(CODEC_EXTRA_LIBS))\
