Index: plugins/libnotify/ario-notifier-libnotify.c
===================================================================
--- plugins/libnotify/ario-notifier-libnotify.c	(révision 734)
+++ plugins/libnotify/ario-notifier-libnotify.c	(copie de travail)
@@ -122,9 +122,14 @@
         ARIO_LOG_FUNCTION_START;
         notifier_libnotify->priv = ARIO_NOTIFIER_LIBNOTIFY_GET_PRIVATE (notifier_libnotify);
 
-        notifier_libnotify->priv->notification = notify_notification_new ("Ario",  NULL, NULL, NULL);
+        notifier_libnotify->priv->notification = notify_notification_new ("Ario",  NULL, NULL
+#if HAVE_LIBNOTIFY_07
+           );
+#else
+           , NULL);
         notify_notification_attach_to_status_icon (notifier_libnotify->priv->notification,
                                                    GTK_STATUS_ICON (ario_tray_icon_get_instance ()));
+#endif
 }
 
 static void
Index: configure.ac
===================================================================
--- configure.ac	(révision 734)
+++ configure.ac	(copie de travail)
@@ -229,9 +229,15 @@
 if test x"$enable_notify" != xno; then
    PKG_CHECK_MODULES(NOTIFY, libnotify)
    AC_DEFINE(ENABLE_LIBNOTIFY, 1, [Add support of libnotify])
+   # Check for libnotify >= 0.7
+   PKG_CHECK_MODULES(LIBNOTIFY_07, [libnotify >= 0.7], [have_libnotify_07=yes],[have_libnotify_07=no])
+   if test x"$have_libnotify_07" = "xyes"; then
+      AC_DEFINE(HAVE_LIBNOTIFY_07, 1, [Define if you have libnotify 0.7 or later])
+   fi
 fi
 AM_CONDITIONAL(USE_NOTIFY, test x"$enable_notify" != xno)
 
+
 dnl ================================================================
 dnl AUDIOSCROBBLER
 dnl ================================================================
