--- wscript.orig	2015-03-05 00:20:03.181254505 -0300
+++ wscript	2015-03-05 00:20:12.396302569 -0300
@@ -220,8 +220,7 @@ def configure(conf):
         conf.CHECK_HEADERS('systemd/sd-daemon.h', lib='systemd-daemon')
         conf.CHECK_LIB('systemd-daemon', shlib=True)
 
-    if (conf.CONFIG_SET('HAVE_SYSTEMD_SD_DAEMON_H') and
-        conf.CONFIG_SET('HAVE_LIBSYSTEMD_DAEMON')):
+    if conf.CONFIG_SET('HAVE_SYSTEMD_SD_DAEMON_H'):
         conf.DEFINE('HAVE_SYSTEMD', '1')
         conf.env['ENABLE_SYSTEMD'] = True
     else:
