diff -up libao-1.0.0/src/audio_out.c~ libao-1.0.0/src/audio_out.c
--- libao-1.0.0/src/audio_out.c~	2010-03-24 05:11:42.000000000 +0100
+++ libao-1.0.0/src/audio_out.c	2010-11-05 10:38:29.098337997 +0100
@@ -189,7 +189,9 @@ static driver_list *_get_plugin(char *pl
 
 
 	} else {
-          aerror("Failed to load plugin %s => dlopen() failed\n",plugin_file);
+	  /* HDG: Silence plugin load errors when the needed libs are not
+	     installed */
+          /* aerror("Failed to load plugin %s => dlopen() failed\n",plugin_file); */
           return NULL;
 	}
 
