From 1122d20acd0e1049cf952ffedbcaccc74e3adf12 Mon Sep 17 00:00:00 2001
From: Jeong YunWon <jeong@youknowone.org>
Date: Mon, 18 Jun 2012 14:24:11 +0900
Subject: [PATCH] Autoload from ~/.xchat2/plugins also as like perl plugin

---
 plugins/python/python.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/plugins/python/python.c b/plugins/python/python.c
index fcbe992..58eb826 100644
--- a/plugins/python/python.c
+++ b/plugins/python/python.c
@@ -379,11 +379,21 @@ Util_Autoload_from (const char *dir_name)
 static void
 Util_Autoload()
 {
+	const char *xdir;
+	char *sub_dir;
 	/* we need local filesystem encoding for chdir, opendir etc */
 
 	/* auto-load from ~/.xchat2/ or %APPDATA%\X-Chat 2\ */
+	xdir = xchat_get_info(ph, "xchatdirfs");
 	Util_Autoload_from(xchat_get_info(ph, "xchatdirfs"));
 
+	/* auto-load from subdirectory plugins */
+	sub_dir = malloc (strlen (xdir) + 9);
+	strcpy (sub_dir, xdir);
+	strcat (sub_dir, "/plugins");
+	Util_Autoload_from(sub_dir);
+	free (sub_dir);
+
 #ifdef WIN32	/* also auto-load C:\Program Files\XChat\Plugins\*.py */
 	Util_Autoload_from(XCHATLIBDIR"/plugins");
 #endif
-- 
1.7.7.5 (Apple Git-26)

