diff -urNp xchat-2.8.8.orig/xchat.desktop xchat-2.8.8/xchat.desktop
--- xchat-2.8.8.orig/xchat.desktop	2013-06-10 14:01:07.408432973 +0200
+++ xchat-2.8.8/xchat.desktop	2013-06-10 14:12:56.044209785 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=XChat IRC
+Name=XChat
 Name[zh_TW]=網路清談
+GenericName=IRC Client
 Comment[de]=IRC-Client
 Comment[es]=Aplicación de IRC
 Comment[fi]=IRC-sovellus
