--- src/common/text.c.orig	2012-02-25 11:05:23.890274334 -0600
+++ src/common/text.c	2012-02-25 11:05:39.006195489 -0600
@@ -28,7 +28,7 @@
 #include <sys/mman.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
+#include <glib.h>
 #include "cfgfiles.h"
 #include "chanopt.h"
 #include "plugin.h"
--- src/common/servlist.c.orig	2012-02-25 11:04:30.601501605 -0600
+++ src/common/servlist.c	2012-02-25 11:04:42.964072823 -0600
@@ -24,7 +24,7 @@
 #include <unistd.h>
 
 #include "xchat.h"
-#include <glib/ghash.h>
+#include <glib.h>
 
 #include "cfgfiles.h"
 #include "fe.h"
--- src/common/dbus/dbus-client.c.orig	2012-02-25 10:59:23.948527589 -0600
+++ src/common/dbus/dbus-client.c	2012-02-25 11:00:40.082203787 -0600
@@ -53,11 +53,6 @@
 	GError *error = NULL;
 	char *command = NULL;
 
-	/* GnomeVFS >=2.15 uses D-Bus and threads, so threads should be
-	 * initialised before opening for the first time a D-Bus connection */
-	if (!g_thread_supported ()) {
-		g_thread_init (NULL);
-	}
 	dbus_g_thread_init ();
 
 	/* if there is nothing to do, return now. */
--- src/common/xchat.h.orig	2012-02-25 11:29:33.400499630 -0600
+++ src/common/xchat.h	2012-02-25 11:29:51.774834912 -0600
@@ -1,10 +1,6 @@
 #include "../../config.h"
 
-#include <glib/gslist.h>
-#include <glib/glist.h>
-#include <glib/gutils.h>
-#include <glib/giochannel.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <time.h>			/* need time_t */
 
 #ifndef XCHAT_H
--- src/common/util.c.orig	2012-02-25 11:38:29.404622998 -0600
+++ src/common/util.c	2012-02-25 11:38:51.904482615 -0600
@@ -39,7 +39,7 @@
 #include <errno.h>
 #include "xchat.h"
 #include "xchatc.h"
-#include <glib/gmarkup.h>
+#include <glib.h>
 #include <ctype.h>
 #include "util.h"
 #include "../../config.h"
