--- electricsheep.c.ffmpeg	2011-06-05 11:53:24.093097051 -0300
+++ electricsheep.c	2011-06-05 11:53:27.609353008 -0300
@@ -670,7 +670,7 @@ void copy_out_file(char *fname) {
     input_stream_index = -1;
     for (j = 0; j < ictx->nb_streams; j++) {
 	AVCodecContext *enc = ictx->streams[j]->codec;
-	if (CODEC_TYPE_VIDEO == enc->codec_type) {
+	if (AVMEDIA_TYPE_VIDEO == enc->codec_type) {
 	    input_stream_index = j;
 	    break;
 	}
@@ -696,7 +696,7 @@ void copy_out_file(char *fname) {
 
 	output_ctx = avformat_alloc_context();
 
-	ofmt = guess_format(NULL, fname, NULL);
+	ofmt = av_guess_format(NULL, fname, NULL);
 	if (!ofmt) {
 	    fprintf(logout, "could not determine format from %s.\n", fname);
 	    exit(1);
@@ -738,7 +738,7 @@ void copy_out_file(char *fname) {
 	av_init_packet(&opkt);
 	if (av_parser_change(ictx->streams[input_stream_index]->parser, output_ctx->streams[0]->codec,
 			     &opkt.data, &opkt.size, ipkt.data, ipkt.size,
-			     ipkt.flags & PKT_FLAG_KEY))
+			     ipkt.flags & AV_PKT_FLAG_KEY))
 	    opkt.destruct= av_destruct_packet;
 		
 	if (-1 == av_interleaved_write_frame(output_ctx, &opkt)) {
