diff -up electricsheep-20151107/Client/Player.cpp.ffmpeg electricsheep-20151107/Client/Player.cpp
--- electricsheep-20151107/Client/Player.cpp.ffmpeg	2015-11-07 11:02:18.669020045 -0200
+++ electricsheep-20151107/Client/Player.cpp	2015-11-07 11:02:51.782037623 -0200
@@ -374,9 +374,9 @@ ContentDecoder::CContentDecoder *CPlayer
 
 #else
 
-	PixelFormat pf = PIX_FMT_BGR32;
+	AVPixelFormat pf = AV_PIX_FMT_BGR32;
 #if defined(__BIG_ENDIAN__)
-	pf = PIX_FMT_RGB32_1;
+	pf = AV_PIX_FMT_RGB32_1;
 #endif
 
 #endif
diff -up electricsheep-20151107/ContentDecoder/ContentDecoder.cpp.ffmpeg electricsheep-20151107/ContentDecoder/ContentDecoder.cpp
--- electricsheep-20151107/ContentDecoder/ContentDecoder.cpp.ffmpeg	2015-11-07 11:03:59.423764311 -0200
+++ electricsheep-20151107/ContentDecoder/ContentDecoder.cpp	2015-11-07 11:04:13.405813911 -0200
@@ -38,7 +38,7 @@ namespace ContentDecoder
 	CContentDecoder.
 
 */
-CContentDecoder::CContentDecoder( spCPlaylist _spPlaylist, bool _bStartByRandom, bool _bCalculateTransitions, const uint32 _queueLenght, PixelFormat _wantedFormat )
+CContentDecoder::CContentDecoder( spCPlaylist _spPlaylist, bool _bStartByRandom, bool _bCalculateTransitions, const uint32 _queueLenght, AVPixelFormat _wantedFormat )
 {
 	g_Log->Info( "CContentDecoder()" );
 	m_FadeCount = static_cast<uint32>(g_Settings()->Get("settings.player.fadecount", 30));
diff -up electricsheep-20151107/ContentDecoder/ContentDecoder.h.ffmpeg electricsheep-20151107/ContentDecoder/ContentDecoder.h
--- electricsheep-20151107/ContentDecoder/ContentDecoder.h.ffmpeg	2015-11-07 11:01:16.594104085 -0200
+++ electricsheep-20151107/ContentDecoder/ContentDecoder.h	2015-11-07 11:01:55.778844795 -0200
@@ -196,7 +196,7 @@ class CContentDecoder
 	
 	sOpenVideoInfo		*m_SecondVideoInfo;
 	
-	PixelFormat		m_WantedPixelFormat;
+	AVPixelFormat		m_WantedPixelFormat;
 
 
 	spCPlaylist		m_spPlaylist;
@@ -230,7 +230,7 @@ class CContentDecoder
 	static int DumpError( int _err );
 
 	public:
-			CContentDecoder( spCPlaylist _spPlaylist, bool _bStartByRandom, bool _bAllowTransitions, const uint32 _queueLenght, PixelFormat _wantedPixelFormat = PIX_FMT_RGB24 );
+			CContentDecoder( spCPlaylist _spPlaylist, bool _bStartByRandom, bool _bAllowTransitions, const uint32 _queueLenght, AVPixelFormat _wantedPixelFormat = AV_PIX_FMT_RGB24 );
 			virtual ~CContentDecoder();
 
 			bool	Initialized() { return m_Initialized; }
diff -up electricsheep-20151107/ContentDecoder/Frame.h.ffmpeg electricsheep-20151107/ContentDecoder/Frame.h
--- electricsheep-20151107/ContentDecoder/Frame.h.ffmpeg	2015-11-07 11:00:20.973827628 -0200
+++ electricsheep-20151107/ContentDecoder/Frame.h	2015-11-07 11:00:52.032281182 -0200
@@ -75,7 +75,7 @@ class CVideoFrame
 		AVFrame		*m_pFrame;
 
 	public:
-		CVideoFrame( AVCodecContext *_pCodecContext, PixelFormat _format, std::string _filename ) : m_pFrame(NULL)
+		CVideoFrame( AVCodecContext *_pCodecContext, AVPixelFormat _format, std::string _filename ) : m_pFrame(NULL)
 			{
 				assert( _pCodecContext );
 				if ( _pCodecContext == NULL)
