--- ./etc/identd.conf.orig	1999-01-20 15:59:26.000000000 -0800
+++ ./etc/identd.conf	2004-06-13 12:59:27.000000000 -0700
@@ -1,4 +1,4 @@
-# /usr/local/etc/identd.conf - an example configuration file
+# /etc/identd.conf - an example configuration file
 
 
 #-- The syslog facility for error messages
@@ -18,24 +18,20 @@
 # server:backlog = 256
 
 #-- Where to write the file containing our process id
-# server:pid-file = "/etc/identd.pid"
+server:pid-file = "/var/run/identd.pid"
 
 #-- Maximum number of concurrent requests allowed (0 = unlimited)
 # server:max-requests = 0
 
-
-
 #-- Enable some protocol extensions like "VERSION" or "QUIT"
-protocol:extensions = enabled
+# protocol:extensions = enabled
 
 #-- Allow multiple queries per connection. This slightly breaks RFC1413
-protocol:multiquery = enabled
+# protocol:multiquery = enabled
 
 #-- Timeout in seconds since connection or last query. Zero = disable
 # protocol:timeout = 120
 
-
-
 #-- Maximum number of threads doing kernel lookups
 # kernel:threads = 8
 
@@ -45,10 +41,8 @@
 #-- Maximum number of time to retry a kernel lookup in case of failure
 # kernel:attempts = 5
 
-
-
 #-- Disable username lookups (only return uid numbers)
-# result:uid-only = no
+result:uid-only = yes
 
 #-- Enable the ".noident" file
 # result:noident = enabled
@@ -60,15 +54,14 @@
 # result:opsys = "UNIX"
 
 #-- Log all request replies to syslog (none == don't)
-# result:syslog-level = none
-
+# Valid choices are: none, debug, info, notice, warning, err, crit, emerg
+result:syslog-level = notice
 
 #-- Enable encryption (only available if linked with a DES library)
 # result:encrypt = no
 
 #-- Path to the DES key file (only available if linked with a DES library)
-# encrypt:key-file = "/usr/local/etc/identd.key"
-
+# encrypt:key-file = "/etc/identd.key"
 
 #-- Include a machine local configuration file
-# include = /etc/identd.conf
+# include = /usr/local/etc/identd.conf
