diff -up techne-0.2.3/configure.ac.plplot techne-0.2.3/configure.ac
--- techne-0.2.3/configure.ac.plplot	2015-04-22 21:30:29.414929027 -0600
+++ techne-0.2.3/configure.ac	2015-04-22 21:33:48.771603503 -0600
@@ -50,7 +50,9 @@ fi
 
 PKG_CHECK_MODULES([OPENAL], [openal])
 PKG_CHECK_MODULES([MAGICKWAND], [MagickWand])
-PKG_CHECK_MODULES([PLPLOT], [plplotd])
+PKG_CHECK_MODULES([PLPLOT], [plplot],, [
+   PKG_CHECK_MODULES([PLPLOT], [plplotd])
+])
 PKG_CHECK_MODULES([GDKGLEXT], [gdkglext-1.0])
 PKG_CHECK_MODULES([WEBKIT], [webkit-1.0])
 PKG_CHECK_MODULES([MHD], [libmicrohttpd])
