Index: comix-4.0.4/install.py
===================================================================
--- comix-4.0.4.orig/install.py
+++ comix-4.0.4/install.py
@@ -240,7 +240,7 @@ def check_dependencies():
         print '    !!! PyGTK .................... Not found'
         required_found = False
     try:
-        import Image
+        from PIL import Image
         assert Image.VERSION >= '1.1.5'
         print '    Python Imaging Library ....... OK'
     except ImportError:
Index: comix-4.0.4/mime/comicthumb
===================================================================
--- comix-4.0.4.orig/mime/comicthumb
+++ comix-4.0.4/mime/comicthumb
@@ -22,7 +22,7 @@ import tarfile
 import subprocess
 
 try:
-    import Image
+    from PIL import Image
 except ImportError:
     print '! Could not import the Image module (PIL).'
     print __doc__
Index: comix-4.0.4/src/comix.py
===================================================================
--- comix-4.0.4.orig/src/comix.py
+++ comix-4.0.4/src/comix.py
@@ -49,7 +49,7 @@ except ImportError:
     sys.exit(1)
 
 try:
-    import Image
+    from PIL import Image
     assert Image.VERSION >= '1.1.5'
 except AssertionError:
     print "You don't have the required version of the Python Imaging",
Index: comix-4.0.4/src/histogram.py
===================================================================
--- comix-4.0.4.orig/src/histogram.py
+++ comix-4.0.4/src/histogram.py
@@ -1,9 +1,9 @@
 """histogram.py - Draw histograms (RGB) from pixbufs."""
 
 import gtk
-import Image
-import ImageDraw
-import ImageOps
+from PIL import Image
+from PIL import ImageDraw
+from PIL import ImageOps
 
 import image
 
Index: comix-4.0.4/src/image.py
===================================================================
--- comix-4.0.4.orig/src/image.py
+++ comix-4.0.4/src/image.py
@@ -1,10 +1,10 @@
 """image.py - Various image manipulations."""
 
 import gtk
-import Image
-import ImageEnhance
-import ImageOps
-import ImageStat
+from PIL import Image
+from PIL import ImageEnhance
+from PIL import ImageOps
+from PIL import ImageStat
 
 from preferences import prefs
 
Index: comix-4.0.4/src/library.py
===================================================================
--- comix-4.0.4.orig/src/library.py
+++ comix-4.0.4/src/library.py
@@ -8,8 +8,8 @@ from xml.sax.saxutils import escape as x
 import gtk
 import gobject
 import pango
-import Image
-import ImageDraw
+from PIL import Image
+from PIL import ImageDraw
 
 import archive
 import encoding
Index: comix-4.0.4/src/thumbbar.py
===================================================================
--- comix-4.0.4.orig/src/thumbbar.py
+++ comix-4.0.4/src/thumbbar.py
@@ -4,8 +4,8 @@ import urllib
 
 import gtk
 import gobject
-import Image
-import ImageDraw
+from PIL import Image
+from PIL import ImageDraw
 
 import image
 from preferences import prefs
Index: comix-4.0.4/src/thumbnail.py
===================================================================
--- comix-4.0.4.orig/src/thumbnail.py
+++ comix-4.0.4/src/thumbnail.py
@@ -15,7 +15,7 @@ import shutil
 import tempfile
 
 import gtk
-import Image
+from PIL import Image
 
 import archive
 import constants
Index: comix-4.0.4/src/thumbremover.py
===================================================================
--- comix-4.0.4.orig/src/thumbremover.py
+++ comix-4.0.4/src/thumbremover.py
@@ -7,7 +7,7 @@ import urllib
 
 import gtk
 import pango
-import Image
+from PIL import Image
 
 import encoding
 import labels
