--- comix-4.0.4/src/thumbnail.py.p2url	2008-12-13 01:04:28.000000000 +0900
+++ comix-4.0.4/src/thumbnail.py	2012-11-28 18:19:07.687627250 +0900
@@ -154,6 +154,9 @@
 
 
 def _path_to_thumbpath(path, dst_dir):
+    # https://bugzilla.redhat.com/show_bug.cgi?id=877604
+    if isinstance(path, unicode):
+        path = path.encode("utf-8")
     uri = 'file://' + pathname2url(os.path.normpath(path))
     return _uri_to_thumbpath(uri, dst_dir)
 
