diff -up easytag-2.1.7/src/ogg_tag.c.vorbis-del easytag-2.1.7/src/ogg_tag.c
--- easytag-2.1.7/src/ogg_tag.c.vorbis-del	2012-01-03 18:27:33.000000000 -0200
+++ easytag-2.1.7/src/ogg_tag.c	2012-03-04 11:10:02.431520025 -0300
@@ -110,8 +110,6 @@
  **************/
 gboolean Ogg_Tag_Write_File (FILE *file_in, gchar *filename_in, vcedit_state *state);
 
-static gboolean Ogg_Write_Delimetered_Tag (vorbis_comment *vc, const gchar *tag_name, gchar *values);
-
 
 /*************
  * Functions *
@@ -589,28 +587,6 @@ gboolean Ogg_Tag_Read_File_Tag (gchar *f
 }
 
 
-/*
- * Save field value in separated tags if it contains multifields
- */
-static gboolean Ogg_Write_Delimetered_Tag (vorbis_comment *vc, const gchar *tag_name, gchar *values)
-{
-    gchar **strings = g_strsplit(values,MULTIFIELD_SEPARATOR,255);
-    unsigned int i=0;
-    
-    for (i=0;i<g_strv_length(strings);i++)
-    {
-        if (strlen(strings[i])>0)
-        {
-            char *string = g_strconcat(tag_name,strings[i],NULL);
-            
-            vorbis_comment_add(vc,string);
-            g_free(string);
-        }
-    }
-    g_strfreev(strings);
-    return TRUE;
-}
-
 
 gboolean Ogg_Tag_Write_File_Tag (ET_File *ETFile)
 {
@@ -693,7 +669,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      *********/
     if ( FileTag->title )
     {
-        Ogg_Write_Delimetered_Tag(vc,"TITLE=",FileTag->title);
+        string = g_strconcat("TITLE=",FileTag->title,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /**********
@@ -701,7 +679,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      **********/
     if ( FileTag->artist )
     {
-        Ogg_Write_Delimetered_Tag(vc,"ARTIST=",FileTag->artist);
+        string = g_strconcat("ARTIST=",FileTag->artist,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /****************
@@ -709,7 +689,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      ****************/
     if ( FileTag->album_artist )
     {
-        Ogg_Write_Delimetered_Tag(vc,"ALBUMARTIST=",FileTag->album_artist);
+        string = g_strconcat("ALBUMARTIST=",FileTag->album,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /*********
@@ -717,7 +699,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      *********/
     if ( FileTag->album )
     {
-        Ogg_Write_Delimetered_Tag(vc,"ALBUM=",FileTag->album);
+        string = g_strconcat("ALBUM=",FileTag->album,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /***************
@@ -761,7 +745,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      *********/
     if ( FileTag->genre )
     {
-        Ogg_Write_Delimetered_Tag(vc,"GENRE=",FileTag->genre);
+        string = g_strconcat("GENRE=",FileTag->genre,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /***********
@@ -771,15 +757,21 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
     if ( FileTag->comment )
     {
         // Format of new specification
-        Ogg_Write_Delimetered_Tag(vc,"DESCRIPTION=",FileTag->comment);
+        string = g_strconcat("DESCRIPTION=",FileTag->comment,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
 
         // Format used in winamp plugin
-        Ogg_Write_Delimetered_Tag(vc,"COMMENT=",FileTag->comment);
+        string = g_strconcat("COMMENT=",FileTag->comment,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
 
         if (OGG_TAG_WRITE_XMMS_COMMENT)
         {
             // Format used into xmms-1.2.5
-            Ogg_Write_Delimetered_Tag(vc,"=",FileTag->comment);
+            string = g_strconcat("=",FileTag->comment,NULL);
+            vorbis_comment_add(vc,string);
+            g_free(string);
         }
     }
 
@@ -788,7 +780,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      ************/
     if ( FileTag->composer )
     {
-        Ogg_Write_Delimetered_Tag(vc,"COMPOSER=",FileTag->composer);
+        string = g_strconcat("COMPOSER=",FileTag->composer,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /*******************
@@ -796,7 +790,9 @@ gboolean Ogg_Tag_Write_File_Tag (ET_File
      *******************/
     if ( FileTag->orig_artist )
     {
-        Ogg_Write_Delimetered_Tag(vc,"PERFORMER=",FileTag->orig_artist);
+        string = g_strconcat("PERFORMER=",FileTag->orig_artist,NULL);
+        vorbis_comment_add(vc,string);
+        g_free(string);
     }
 
     /*************
