diff -up easytag-2.1.7/src/flac_tag.c.flac-del easytag-2.1.7/src/flac_tag.c
--- easytag-2.1.7/src/flac_tag.c.flac-del	2012-01-03 18:27:33.000000000 -0200
+++ easytag-2.1.7/src/flac_tag.c	2012-03-04 11:07:50.659493952 -0300
@@ -91,8 +91,6 @@
  **************/
 gboolean Flac_Tag_Write_File (FILE *file_in, gchar *filename_in, vcedit_state *state);
 
-static gboolean Flac_Write_Delimetered_Tag (FLAC__StreamMetadata *vc_block, const gchar *tag_name, gchar *values);
-
 
 /*************
  * Functions *
@@ -773,33 +771,6 @@ gboolean Flac_Tag_Read_File_Tag (gchar *
 
 
 /*
- * Save field value in separated tags if it contains multifields
- */
-static gboolean Flac_Write_Delimetered_Tag (FLAC__StreamMetadata *vc_block, const gchar *tag_name, gchar *values)
-{
-    gchar **strings = g_strsplit(values,MULTIFIELD_SEPARATOR,255);
-    unsigned int i=0;
-    
-    for (i=0;i<g_strv_length(strings);i++)
-    {
-        if (strlen(strings[i])>0)
-        {
-            FLAC__StreamMetadata_VorbisComment_Entry field;
-            char *string = g_strconcat(tag_name,strings[i],NULL);
-            
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string); // Warning : g_utf8_strlen doesn't count the multibyte characters. Here we need the allocated size.
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
-        }
-    }
-    g_strfreev(strings);
-    return TRUE;
-}
-
-
-
-/*
  * Write Flac tag, using the level 2 flac interface
  */
 gboolean Flac_Tag_Write_File_Tag (ET_File *ETFile)
@@ -930,7 +901,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          *********/
         if ( FileTag->title )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"TITLE=",FileTag->title);
+            string = g_strconcat("TITLE=",FileTag->title,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string); // Warning : g_utf8_strlen doesn't count the multibyte characters. Here we need the allocated size.
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /**********
@@ -938,7 +913,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          **********/
         if ( FileTag->artist )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"ARTIST=",FileTag->artist);
+            string = g_strconcat("ARTIST=",FileTag->artist,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /****************
@@ -946,7 +925,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          ****************/
         if ( FileTag->album_artist )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"ALBUMARTIST=",FileTag->album_artist);
+            string = g_strconcat("ALBUMARTIST=",FileTag->album,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /*********
@@ -954,7 +937,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          *********/
         if ( FileTag->album )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"ALBUM=",FileTag->album);
+            string = g_strconcat("ALBUM=",FileTag->album,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /***************
@@ -1006,7 +993,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          *********/
         if ( FileTag->genre )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"GENRE=",FileTag->genre);
+            string = g_strconcat("GENRE=",FileTag->genre,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /***********
@@ -1015,9 +1006,17 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
         // We write the comment using the "both" format
         if ( FileTag->comment )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"DESCRIPTION=",FileTag->comment);
+            string = g_strconcat("DESCRIPTION=",FileTag->comment,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
 
-            Flac_Write_Delimetered_Tag(vc_block,"COMMENT=",FileTag->comment);
+            string = g_strconcat("COMMENT=",FileTag->comment,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /************
@@ -1025,7 +1024,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          ************/
         if ( FileTag->composer )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"COMPOSER=",FileTag->composer);
+            string = g_strconcat("COMPOSER=",FileTag->composer,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /*******************
@@ -1033,7 +1036,11 @@ gboolean Flac_Tag_Write_File_Tag (ET_Fil
          *******************/
         if ( FileTag->orig_artist )
         {
-            Flac_Write_Delimetered_Tag(vc_block,"PERFORMER=",FileTag->orig_artist);
+            string = g_strconcat("PERFORMER=",FileTag->orig_artist,NULL);
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string);
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
         }
 
         /*************
