From 7b487c7571bc6b65759cd0673de4c223e00b6297 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Fri, 03 Jan 2014 10:44:29 +0000
Subject: Clarify licensing of dlm.c and dlm.h, bug 721385

Signed-off-by: Santtu Lakkala <inz@inz.fi>
---
diff --git a/src/dlm.c b/src/dlm.c
index 5e1d10f..77cb19b 100644
--- a/src/dlm.c
+++ b/src/dlm.c
@@ -1,4 +1,20 @@
-/* dlm.c - 2004/07/04 - Santtu Lakkala */
+/* EasyTAG - tag editor for audio files
+ * Copyright (C) 2004  Santtu Lakkala <inzane@ircing.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
 
 #include <glib.h>
 #include <string.h>
diff --git a/src/dlm.h b/src/dlm.h
index 1ef23fd..f405979 100644
--- a/src/dlm.h
+++ b/src/dlm.h
@@ -1,4 +1,20 @@
-/* dlm.h - 2004/07/04 */
+/* EasyTAG - tag editor for audio files
+ * Copyright (C) 2004  Santtu Lakkala <inzane@ircing.org>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
 
 #ifndef __DLM_H__
 #define __DLM_H__
--
cgit v0.9.2
