From 17afda799f95c97263e52ab41fa8c9676aba3021 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Fri, 07 Mar 2014 15:34:08 +0000
Subject: Avoid crash when declining deleting a file

https://retrace.fedoraproject.org/faf/problems/1586226/
---
diff --git a/src/easytag.c b/src/easytag.c
index 40dbeec..52a1619 100644
--- a/src/easytag.c
+++ b/src/easytag.c
@@ -2637,9 +2637,14 @@ Delete_Selected_Files_With_Answer (void)
                 ET_Remove_File_From_File_List(ETFile);
                 break;
             case 0:
-                Log_Print (LOG_ERROR, _("Cannot delete file (%s)"),
-                           error->message);
-                g_error_free (error);
+                /* Distinguish between the file being skipped, and there being
+                 * an error during deletion. */
+                if (error)
+                {
+                    Log_Print (LOG_ERROR, _("Cannot delete file (%s)"),
+                               error->message);
+                    g_clear_error (&error);
+                }
                 break;
             case -1:
                 // Stop deleting files + reinit progress bar
--
cgit v0.9.2
