--- easytag-2.1.5/src/flac_tag.c	2008-01-15 19:43:44.000000000 -0200
+++ easytag-2.1.6/src/flac_tag.c	2008-05-24 19:18:56.000000000 -0300
@@ -90,6 +90,8 @@
  **************/
 gboolean Flac_Tag_Write_File (FILE *file_in, gchar *filename_in, vcedit_state *state);
 
+static gboolean Flac_Write_Delimetered_Tag (FLAC__StreamMetadata *vc_block, const gchar *tag_name, gchar *values);
+
 
 /*************
  * Functions *
@@ -739,6 +741,33 @@
 
 
 /*
+ * Save field value in separated tags if it contains multifields
+ */
+static gboolean Flac_Write_Delimetered_Tag (FLAC__StreamMetadata *vc_block, const gchar *tag_name, gchar *values)
+{
+    gchar **strings = g_strsplit(values,MULTIFIELD_SEPARATOR,255);
+    unsigned int i=0;
+    
+    for (i=0;i<g_strv_length(strings);i++)
+    {
+        if (strlen(strings[i])>0)
+        {
+            FLAC__StreamMetadata_VorbisComment_Entry field;
+            char *string = g_strconcat(tag_name,strings[i],NULL);
+            
+            field.entry = (FLAC__byte *)string;
+            field.length = strlen(string); // Warning : g_utf8_strlen doesn't count the multibyte characters. Here we need the allocated size.
+            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
+            g_free(string);
+        }
+    }
+    g_strfreev(strings);
+    return TRUE;
+}
+
+
+
+/*
  * Write Flac tag, using the level 2 flac interface
  */
 gboolean Flac_Tag_Write_File_Tag (ET_File *ETFile)
@@ -867,11 +896,7 @@
          *********/
         if ( FileTag->title )
         {
-            string = g_strconcat("TITLE=",FileTag->title,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string); // Warning : g_utf8_strlen doesn't count the multibyte characters. Here we need the allocated size.
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"TITLE=",FileTag->title);
         }
 
         /**********
@@ -879,11 +904,7 @@
          **********/
         if ( FileTag->artist )
         {
-            string = g_strconcat("ARTIST=",FileTag->artist,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"ARTIST=",FileTag->artist);
         }
 
         /*********
@@ -891,11 +912,7 @@
          *********/
         if ( FileTag->album )
         {
-            string = g_strconcat("ALBUM=",FileTag->album,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"ALBUM=",FileTag->album);
         }
 
         /***************
@@ -947,11 +964,7 @@
          *********/
         if ( FileTag->genre )
         {
-            string = g_strconcat("GENRE=",FileTag->genre,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"GENRE=",FileTag->genre);
         }
 
         /***********
@@ -960,17 +973,9 @@
         // We write the comment using the "both" format
         if ( FileTag->comment )
         {
-            string = g_strconcat("DESCRIPTION=",FileTag->comment,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"DESCRIPTION=",FileTag->comment);
 
-            string = g_strconcat("COMMENT=",FileTag->comment,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"COMMENT=",FileTag->comment);
         }
 
         /************
@@ -978,11 +983,7 @@
          ************/
         if ( FileTag->composer )
         {
-            string = g_strconcat("COMPOSER=",FileTag->composer,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"COMPOSER=",FileTag->composer);
         }
 
         /*******************
@@ -990,11 +991,7 @@
          *******************/
         if ( FileTag->orig_artist )
         {
-            string = g_strconcat("PERFORMER=",FileTag->orig_artist,NULL);
-            field.entry = (FLAC__byte *)string;
-            field.length = strlen(string);
-            FLAC__metadata_object_vorbiscomment_insert_comment(vc_block,vc_block->data.vorbis_comment.num_comments,field,true);
-            g_free(string);
+            Flac_Write_Delimetered_Tag(vc_block,"PERFORMER=",FileTag->orig_artist);
         }
 
         /*************
