commit 1d0a255ca85d964141945a29f6e92d2ba0d89714
Author: David King <amigadave@amigadave.com>
Date:   Mon Feb 3 22:30:28 2014 +0000

    Fix memory leak in Parse_Date()

diff --git a/src/misc.c b/src/misc.c
index a9a922c..48a1199 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -441,6 +441,7 @@ gboolean Parse_Date (void)
     {
         dt = g_date_time_new_now_local ();
         current_year = g_date_time_format (dt, "%Y");
+        g_date_time_unref (dt);
 
         tmp = &current_year[4-strlen(year)];
         if ( atoi(year) <= atoi(tmp) )
