From 6563a0989a6d35d618a950262bd81d2ac50ed0ed Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Fri, 07 Mar 2014 22:41:43 +0000
Subject: Avoid crash when saving unknown image types

https://retrace.fedoraproject.org/faf/problems/1586225/
---
diff --git a/src/picture.c b/src/picture.c
index b6b8f26..8e74b93 100644
--- a/src/picture.c
+++ b/src/picture.c
@@ -988,7 +988,8 @@ const gchar *Picture_Mime_Type_String (Picture_Format format)
         case PICTURE_FORMAT_PNG:
             return "image/png";
         default:
-            return NULL;
+            g_debug ("%s", "Unrecognised image MIME type");
+            return "application/octet-stream";
     }
 }
 
--
cgit v0.9.2
