From afad898b0394b6eafeaf6f89cf411ac5c0e96ab0 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Tue, 21 Jan 2014 19:42:09 +0000
Subject: Fix memory leak in Log_Format_Date()

---
diff --git a/src/log.c b/src/log.c
index 577fb63..1ab65f0 100644
--- a/src/log.c
+++ b/src/log.c
@@ -231,6 +231,8 @@ Log_Format_Date (void)
     /* Time without date in current locale. */
     time = g_date_time_format (dt, "%X");
 
+    g_date_time_unref (dt);
+
     return time;
 }
 
--
cgit v0.9.2
