--- icu/source/tools/toolutil/udbgutil.cpp.orig	2013-10-20 13:15:40.809787680 -0200
+++ icu/source/tools/toolutil/udbgutil.cpp	2013-10-20 13:15:43.967142060 -0200
@@ -602,7 +602,7 @@ U_CAPI char *udbg_knownIssueURLFrom(cons
 }
 
 
-#if !U_HAVE_STD_STRING
+#if 1
 const char *warning = "WARNING: Don't have std::string (STL) - known issue logs will be deficient.";
 
 U_CAPI void *udbg_knownIssue_openU(void *ptr, const char *ticket, char *where, const UChar *msg, UBool *firstForTicket,
