--- icu/source/test/intltest/numfmtst.cpp	2010-10-04 15:16:58.000000000 +0100
+++ icu/source/test/intltest/numfmtst.cpp	2010-10-04 15:22:37.000000000 +0100
@@ -6144,7 +6144,7 @@
        ) 
     { 
         errln("ERROR: parse failed - expected 123.0, 3  - returned %d, %i", 
-               result.getDouble(), parsePos); 
+               result.getDouble(), parsePos.getErrorIndex()); 
     } 
 } 
 
