--- icu/source/Makefile.in	2010-04-28 17:28:54.000000000 +0200
+++ icu/source/Makefile.in	2010-07-28 22:50:45.056353070 +0200
@@ -18,7 +18,7 @@
 docsubdir = $(PACKAGE)$(ICULIBDASHSUFFIX)/html
 docsubsrchdir = $(docsubdir)/search
 docfilesdir = doc/html
-docfiles = $(docfilesdir)/*.gif $(docfilesdir)/*.png $(docfilesdir)/*.html $(docfilesdir)/*.css $(docfilesdir)/*.tag $(docfilesdir)/installdox
+docfiles = $(docfilesdir)/*.png $(docfilesdir)/*.html $(docfilesdir)/*.css $(docfilesdir)/*.tag $(docfilesdir)/installdox
 docsrchdir = $(docfilesdir)/search
 docsrchfiles = $(docsrchdir)/*
 
