diff -ru icu.orig/source/test/cintltst/cnumtst.c icu/source/test/cintltst/cnumtst.c
--- icu.orig/source/test/cintltst/cnumtst.c	2010-09-09 08:52:26.043634583 +0100
+++ icu/source/test/cintltst/cnumtst.c	2010-09-09 09:03:28.602635460 +0100
@@ -358,7 +358,9 @@
         log_err("parse failed. The error is  : %s\n", myErrorName(status));
     }
 
-    if(d1!=462.12345)
+    a1 = 462.12345;
+
+    if(d1!=a1)
         log_err("Fail: Error in parsing\n");
     else
         log_verbose("Pass: parsing successful\n");
