--- VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp.0011~	2011-08-16 19:20:23.000000000 +0200
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp	2011-08-16 19:22:32.000000000 +0200
@@ -1098,7 +1098,7 @@
             /* Get the USB controller object: */
             CUSBController controller = m_machine.GetUSBController();
             /* Show the machine error message if any: */
-            if (!m_machine.isReallyOk() && !controller.isNull() && controller.GetEnabled())
+            if (!m_machine.isReallyOk() && !controller.isNull() && controller.GetEnabled() && (controller.GetDeviceFilters().size() != 0))
                 msgCenter().cannotAccessUSB(m_machine);
             /* Check if USB is implemented: */
             if (controller.isNull() || !controller.GetProxyAvailable())
