--- ./configure.gcc	2014-05-02 19:18:06.929223329 +0100
+++ ./configure	2014-05-02 19:21:11.010309549 +0100
@@ -420,9 +420,9 @@ check_gcc()
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 8 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
              -o $cc_maj -gt 4 ]; then
-        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<8"
+        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10"
         fail really
       else
         log_success "found version $cc_ver"
