--- VirtualBox-4.3.0/configure.dso	2013-10-19 19:52:54.329622560 -0300
+++ VirtualBox-4.3.0/configure	2013-10-19 19:54:30.009393977 -0300
@@ -2046,7 +2046,7 @@ check_gsoap()
         else
           cnf_append "VBOX_GSOAP_CXX_SOURCES" ""
         fi
-        cnf_append "VBOX_GSOAP_CXX_LIBS"    "`strip_l "$GSOAP_CXX_LIBS"`"
+        cnf_append "VBOX_GSOAP_CXX_LIBS"    "`strip_l "$GSOAP_CXX_LIBS $LIBZ"`"
         gsoap_version=`pkg-config gsoapssl++ --modversion`
         log_success "found version $gsoap_version"
         return
