--- VirtualBox-4.0.0_OSE/src/libs/xpcom18a4/Makefile.kmk
+++ VirtualBox-4.0.0_OSE/src/libs/xpcom18a4/Makefile.kmk
@@ -469,9 +469,11 @@
  libIDL_config_cflags := $(shell $(VBOX_LIBIDL_CONFIG) --cflags)
  libIDL_config_libs   := $(shell $(VBOX_LIBIDL_CONFIG) --libs)
  xpidl_CFLAGS = \
-	$(libIDL_config_cflags)
+	$(shell pkg-config --cflags libIDL-2.0)
  xpidl_LDFLAGS = \
-	$(libIDL_config_libs)
+	$(shell pkg-config --libs-only-L libIDL-2.0)
+xpidl_LIBS.linux = \
+	$(shell pkg-config --libs-only-l libIDL-2.0 | sed -e 's/-l//g')
  xpidl_LDFLAGS.linux = \
 	$(VBOX_LD_as_needed)
 endif
