--- VirtualBox-5.0.0/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup	2015-07-10 19:18:12.040460966 -0300
+++ VirtualBox-5.0.0/src/VBox/Frontends/VirtualBox/Makefile.kmk	2015-07-10 19:18:30.493540304 -0300
@@ -137,10 +137,6 @@ ifdef VBOX_WITH_VIDEOHWACCEL
 else ifdef VBOX_GUI_USE_QGL
  VirtualBox_DEFS        += VBOX_GUI_USE_QGL
 endif
-ifdef VBOX_WITH_LIBCURL
- VirtualBox_DEFS        += VBOX_GUI_WITH_NETWORK_MANAGER
- VBOX_GUI_WITH_NETWORK_MANAGER := 1
-endif
 ifdef VBOX_BLEEDING_EDGE
 VirtualBox_src/selector/UISelectorWindow.cpp_DEFS += \
     VBOX_BLEEDING_EDGE=\"$(VBOX_BLEEDING_EDGE)\"
--- VirtualBox-5.0.0/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp.noup	2015-07-09 06:54:54.000000000 -0300
+++ VirtualBox-5.0.0/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp	2015-07-10 20:36:19.279366959 -0300
@@ -43,9 +43,9 @@
 # include "UIModalWindowManager.h"
 # include "UIExtraDataManager.h"
 # include "UIMedium.h"
-# ifdef VBOX_OSE
+# ifdef VBOX_GUI_WITH_NETWORK_MANAGER
 #  include "UIDownloaderUserManual.h"
-# endif /* VBOX_OSE */
+# endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 # include "UIMachine.h"
 # include "VBoxAboutDlg.h"
 # include "UIHostComboEditor.h"
@@ -2573,7 +2573,7 @@ void UIMessageCenter::sltShowHelpAboutDi
 
 void UIMessageCenter::sltShowHelpHelpDialog()
 {
-#ifndef VBOX_OSE
+#ifndef VBOX_GUI_WITH_NETWORK_MANAGER
     /* For non-OSE version we just open it: */
     sltShowUserManual(vboxGlobal().helpFile());
 #else /* #ifndef VBOX_OSE */
@@ -2602,7 +2602,7 @@ void UIMessageCenter::sltShowHelpHelpDia
         /* Start downloading: */
         pDl->start();
     }
-#endif /* #ifdef VBOX_OSE */
+#endif /* #ifdef VBOX_GUI_WITH_NETWORK_MANAGER */
 }
 
 void UIMessageCenter::sltResetSuppressedMessages()
