--- VirtualBox-5.0.0/Config.kmk.fix	2015-07-09 06:50:42.000000000 -0300
+++ VirtualBox-5.0.0/Config.kmk	2015-07-09 23:53:54.232136132 -0300
@@ -1222,7 +1222,7 @@ ifdef VBOX_OSE
 # of VirtualBox binaries.  We request though that you always use something that
 # clearly identifies your build and makes it clear that it is not one done by
 # the VirtualBox team.
- VBOX_BUILD_PUBLISHER = _OSE
+ VBOX_BUILD_PUBLISHER = __SLKDIST_
 endif
 
 # Keep in sync with G_virtualBoxPackage in glue-jaxws.xsl and glue-java.xsl
