diff -Npru VirtualBox-4.0.4_OSE.orig/configure VirtualBox-4.0.4_OSE/configure
--- VirtualBox-4.0.4_OSE.orig/configure	2011-02-18 00:30:48.000000000 +0800
+++ VirtualBox-4.0.4_OSE/configure	2011-04-10 13:37:42.513743178 +0800
@@ -409,9 +409,9 @@ check_gcc()
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 5 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 6 \) \
              -o $cc_maj -gt 4 ]; then
-        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<5"
+        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<=6"
         fail really
       else
         log_success "found version $cc_ver"
