diff -up VirtualBox-4.3.8/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp.defsys VirtualBox-4.3.8/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp
--- VirtualBox-4.3.8/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp.defsys	2014-02-25 14:08:58.000000000 -0300
+++ VirtualBox-4.3.8/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp	2014-02-25 21:02:43.598770157 -0300
@@ -104,6 +104,9 @@ UINameAndSystemEditor::UINameAndSystemEd
         m_pFamilyCombo->setItemData(i, families[i].GetFamilyId(), TypeID);
     }
     m_pFamilyCombo->setCurrentIndex(0);
+    int iLinuxIndex = m_pFamilyCombo->findData ("Linux", TypeID);
+    if (iLinuxIndex != -1)
+        m_pFamilyCombo->setCurrentIndex (iLinuxIndex);
     sltFamilyChanged(m_pFamilyCombo->currentIndex());
 
     /* Setup connections: */
@@ -206,7 +209,7 @@ void UINameAndSystemEditor::sltFamilyCha
     /* Or select Ubuntu item for Linux family as default: */
     else if (strFamilyId == "Linux")
     {
-        QString strDefaultID = "Ubuntu";
+        QString strDefaultID = "Linux26";
         if (ARCH_BITS == 64 && m_fSupportsHWVirtEx && m_fSupportsLongMode)
             strDefaultID += "_64";
         int iIndexUbuntu = m_pTypeCombo->findData(strDefaultID, TypeID);
