--- VirtualBox-4.1.4_OSE/Config.kmk
+++ VirtualBox-4.1.4_OSE/Config.kmk
@@ -4263,7 +4263,7 @@
 TEMPLATE_VBOXBLDPROG_LDFLAGS.amd64       = -m64
 TEMPLATE_VBOXBLDPROG_LDFLAGS.sparc64     = -m64
  ifeq ($(KBUILD_HOST),linux)
-TEMPLATE_VBOXBLDPROG_LIBS                = pthread m rt dl
+TEMPLATE_VBOXBLDPROG_LIBS                = pthread m rt dl crypt
  else ifeq ($(KBUILD_HOST),os2)
 TEMPLATE_VBOXBLDPROG_TOOL                = GXX3OMF
 TEMPLATE_VBOXBLDPROG_LIBS                = socket iconv
