--- VirtualBox-3.1.6_OSE/src/VBox/Frontends/VirtualBox/src/widgets/VBoxOSTypeSelectorWidget.cpp.old	2010-04-28 23:04:14.509963426 +0300
+++ VirtualBox-3.1.6_OSE/src/VBox/Frontends/VirtualBox/src/widgets/VBoxOSTypeSelectorWidget.cpp	2010-04-28 23:08:01.052462659 +0300
@@ -74,6 +74,9 @@ VBoxOSTypeSelectorWidget::VBoxOSTypeSele
         }
     }
     mCbFamily->setCurrentIndex (0);
+    int liIndex = mCbFamily->findData ("Linux", RoleTypeID);
+    if (liIndex != -1)
+        mCbFamily->setCurrentIndex (liIndex);
     onFamilyChanged (mCbFamily->currentIndex());
 
     /* Set the minimum size for OS Type & Family selectors. */
@@ -198,7 +201,7 @@ void VBoxOSTypeSelectorWidget::onFamilyC
     /* Or select Ubuntu item for Linux family as default */
     else if (familyId == "Linux")
     {
-        int ubIndex = mCbType->findData ("Ubuntu", RoleTypeID);
+        int ubIndex = mCbType->findData ("Linux26", RoleTypeID);
         if (ubIndex != -1)
             mCbType->setCurrentIndex (ubIndex);
     }
