Index: trunk/configure
===================================================================
--- a/trunk/configure
+++ b/trunk/configure
@@ -423,6 +423,7 @@
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
-             -o $cc_maj -gt 4 ]; then
-        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10"
+             -o \( $cc_maj -eq 5 -a $cc_min -gt 1 \) \
+             -o $cc_maj -gt 5 ]; then
+        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10 or gcc 5.1"
         fail really
       else
         log_success "found version $cc_ver"
