diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/Makefile.kmk
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup	2013-10-15 08:12:37.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/Makefile.kmk	2013-10-19 13:26:35.717805248 -0300
@@ -24,6 +24,8 @@ include $(KBUILD_PATH)/subheader.kmk
 VBOX_VIRTUALBOX4_SRC := $(PATH_SUB_CURRENT)
 VBOX_VIRTUALBOX4_OUT_DIR := $(PATH_TARGET)/VirtualBox/misc
 
+# VBOX_WITH_UPDATE := 1
+
 # Show the update notifier dialog during startup. If this setting is disabled
 # the user can still update using the menu if desired.
 ifndef VBOX_OSE
@@ -91,6 +93,9 @@ ifdef VBOX_WITH_ICHAT_THEATER
 endif
 VirtualBox_DEFS          += \
 	$(if $(VBOX_WITH_ADDITIONS_AUTOUPDATE_UI),VBOX_WITH_ADDITIONS_AUTOUPDATE_UI) \
+	$(if $(VBOX_WITH_REGISTRATION),VBOX_WITH_REGISTRATION) \
+	$(if $(VBOX_WITH_REGISTRATION_REQUEST),VBOX_WITH_REGISTRATION_REQUEST) \
+	$(if $(VBOX_WITH_UPDATE),VBOX_WITH_UPDATE) \
 	$(if $(VBOX_WITH_UPDATE_REQUEST),VBOX_WITH_UPDATE_REQUEST) \
 	$(if $(VBOX_WITH_ALSA),VBOX_WITH_ALSA) \
 	$(if $(VBOX_WITH_PULSE),VBOX_WITH_PULSE) \
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp.noup	2013-10-15 08:12:45.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp	2013-10-19 13:26:35.718805036 -0300
@@ -457,6 +457,7 @@ protected:
     }
 };
 
+#ifdef VBOX_WITH_UPDATE
 class UIActionSimpleCheckForUpdates : public UIActionSimple
 {
     Q_OBJECT;
@@ -483,6 +484,7 @@ protected:
         setStatusTip(QApplication::translate("UIActionPool", "Check for a new VirtualBox version"));
     }
 };
+#endif /* VBOX_WITH_UPDATE */
 
 class UIActionSimpleAbout : public UIActionSimple
 {
@@ -650,7 +652,9 @@ void UIActionPool::createActions()
     m_pool[UIActionIndex_Simple_WebSite] = new UIActionSimpleWebSite(this);
     m_pool[UIActionIndex_Simple_ResetWarnings] = new UIActionSimpleResetWarnings(this);
     m_pool[UIActionIndex_Simple_NetworkAccessManager] = new UIActionSimpleNetworkAccessManager(this);
+#ifdef VBOX_WITH_UPDATE
     m_pool[UIActionIndex_Simple_CheckForUpdates] = new UIActionSimpleCheckForUpdates(this);
+#endif /* VBOX_WITH_UPDATE */
     m_pool[UIActionIndex_Simple_About] = new UIActionSimpleAbout(this);
 }
 
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h.noup	2013-10-15 08:12:45.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h	2013-10-19 13:26:35.718805036 -0300
@@ -58,7 +58,9 @@ enum UIActionIndex
     UIActionIndex_Simple_WebSite,
     UIActionIndex_Simple_ResetWarnings,
     UIActionIndex_Simple_NetworkAccessManager,
+#ifdef VBOX_WITH_UPDATE
     UIActionIndex_Simple_CheckForUpdates,
+#endif /* VBOX_WITH_UPDATE */
     UIActionIndex_Simple_About,
 
     /* Maximum index: */
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp.noup	2013-10-15 08:12:45.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp	2013-10-19 13:26:35.718805036 -0300
@@ -31,7 +31,9 @@ const char* UIDefs::GUI_RecentFolderFD =
 const char* UIDefs::GUI_RecentListHD = "GUI/RecentListHD";
 const char* UIDefs::GUI_RecentListCD = "GUI/RecentListCD";
 const char* UIDefs::GUI_RecentListFD = "GUI/RecentListFD";
+#ifdef VBOX_WITH_UPDATE
 const char* UIDefs::GUI_PreventApplicationUpdate = "GUI/PreventApplicationUpdate";
+#endif /* VBOX_WITH_UPDATE */
 
 /* Selector-window definitions: */
 const char* UIDefs::GUI_Input_SelectorShortcuts = "GUI/Input/SelectorShortcuts";
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h.noup	2013-10-15 08:12:45.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h	2013-10-19 13:26:35.719804825 -0300
@@ -102,7 +102,9 @@ namespace UIDefs
     extern const char* GUI_RecentListHD;
     extern const char* GUI_RecentListCD;
     extern const char* GUI_RecentListFD;
+#ifdef VBOX_WITH_UPDATE
     extern const char* GUI_PreventApplicationUpdate;
+#endif /* VBOX_WITH_UPDATE */
 
     /* Selector-window declarations: */
     extern const char* GUI_Input_SelectorShortcuts;
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp.noup	2013-10-15 08:12:45.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp	2013-10-19 13:26:35.720804613 -0300
@@ -3365,12 +3365,14 @@ bool VBoxGlobal::isApprovedByExtraData(C
            || strExtraDataValue == "1";
 }
 
+#ifdef VBOX_WITH_UPDATE
 /* static */
 bool VBoxGlobal::shouldWeAllowApplicationUpdate(CVirtualBox &vbox)
 {
     /* 'true' if disabling is not approved by the extra-data: */
     return !isApprovedByExtraData(vbox, GUI_PreventApplicationUpdate);
 }
+#endif /* VBOX_WITH_UPDATE */
 
 /* static */
 bool VBoxGlobal::shouldWeShowMachine(CMachine &machine)
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h.noup	2013-10-15 08:12:46.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h	2013-10-19 13:26:35.720804613 -0300
@@ -343,7 +343,9 @@ public:
     /* Extra-data settings stuff: */
     static bool isApprovedByExtraData(CVirtualBox &vbox, const QString &strExtraDataKey);
     static bool isApprovedByExtraData(CMachine &machine, const QString &strExtraDataKey);
+#ifdef VBOX_WITH_UPDATE
     static bool shouldWeAllowApplicationUpdate(CVirtualBox &vbox);
+#endif /* VBOX_WITH_UPDATE */
     static bool shouldWeShowMachine(CMachine &machine);
     static bool shouldWeAllowMachineReconfiguration(CMachine &machine,
                                                     bool fIncludingMachineGeneralCheck = false,
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp.noup	2013-10-19 13:26:56.829336135 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp	2013-10-19 13:31:27.003179056 -0300
@@ -1324,12 +1324,14 @@ void UISelectorWindow::prepareMenuHelp(Q
     pMenu->addSeparator();
     m_pNetworkAccessManager = gActionPool->action(UIActionIndex_Simple_NetworkAccessManager);
     pMenu->addAction(m_pNetworkAccessManager);
+#ifdef VBOX_WITH_UPDATE
     m_pUpdateAction = gActionPool->action(UIActionIndex_Simple_CheckForUpdates);
     CVirtualBox vbox = vboxGlobal().virtualBox();
     if (VBoxGlobal::shouldWeAllowApplicationUpdate(vbox))
         pMenu->addAction(m_pUpdateAction);
     else
         m_pUpdateAction->setEnabled(false);
+#endif /* VBOX_WITH_UPDATE */
 #ifndef Q_WS_MAC
     pMenu->addSeparator();
 #endif /* !Q_WS_MAC */
@@ -1462,7 +1464,9 @@ void UISelectorWindow::prepareConnection
     connect(m_pWebAction, SIGNAL(triggered()), &msgCenter(), SLOT(sltShowHelpWebDialog()));
     connect(m_pResetWarningsAction, SIGNAL(triggered()), &msgCenter(), SLOT(sltResetSuppressedMessages()));
     connect(m_pNetworkAccessManager, SIGNAL(triggered()), gNetworkManager, SLOT(show()));
+#ifdef VBOX_WITH_UPDATE
     connect(m_pUpdateAction, SIGNAL(triggered()), gUpdateManager, SLOT(sltForceCheck()));
+#endif /* VBOX_WITH_UPDATE */
     connect(m_pAboutAction, SIGNAL(triggered()), &msgCenter(), SLOT(sltShowHelpAboutDialog()));
 
     /* Status-bar connections: */
diff -up VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp.noup VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
--- VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp.noup	2013-10-15 08:12:49.000000000 -0300
+++ VirtualBox-4.3.0/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp	2013-10-19 13:26:56.830335923 -0300
@@ -35,7 +35,9 @@
 
 #include "UIGlobalSettingsGeneral.h"
 #include "UIGlobalSettingsInput.h"
+#ifdef VBOX_WITH_UPDATE
 #include "UIGlobalSettingsUpdate.h"
+#endif /* VBOX_WITH_UPDATE */
 #include "UIGlobalSettingsLanguage.h"
 #include "UIGlobalSettingsDisplay.h"
 #include "UIGlobalSettingsNetwork.h"
@@ -354,6 +356,7 @@ UISettingsDialogGlobal::UISettingsDialog
                             iPageIndex, "#input", pSettingsPage);
                     break;
                 }
+#ifdef VBOX_WITH_UPDATE
                 /* Update page: */
                 case GlobalSettingsPageType_Update:
                 {
@@ -363,6 +366,7 @@ UISettingsDialogGlobal::UISettingsDialog
                             iPageIndex, "#update", pSettingsPage);
                     break;
                 }
+#endif /* VBOX_WITH_UPDATE */
                 /* Language page: */
                 case GlobalSettingsPageType_Language:
                 {
@@ -489,8 +493,10 @@ void UISettingsDialogGlobal::retranslate
     /* Input page: */
     m_pSelector->setItemText(GlobalSettingsPageType_Input, tr("Input"));
 
+#ifdef VBOX_WITH_UPDATE
     /* Update page: */
     m_pSelector->setItemText(GlobalSettingsPageType_Update, tr("Update"));
+#endif /* VBOX_WITH_UPDATE */
 
     /* Language page: */
     m_pSelector->setItemText(GlobalSettingsPageType_Language, tr("Language"));
