diff -up VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/Makefile.kmk
--- VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup	2016-07-12 17:38:40.688057905 -0300
+++ VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/Makefile.kmk	2016-07-12 17:38:59.638932154 -0300
@@ -156,10 +156,6 @@ ifdef VBOX_WITH_VIDEOHWACCEL
 else ifdef VBOX_GUI_USE_QGL
  VirtualBox_DEFS        += VBOX_GUI_USE_QGL
 endif
-ifdef VBOX_WITH_LIBCURL
- VirtualBox_DEFS        += VBOX_GUI_WITH_NETWORK_MANAGER
- VBOX_GUI_WITH_NETWORK_MANAGER := 1
-endif
 ifneq ($(KBUILD_TYPE),release)
  VirtualBox_DEFS        += VBOX_GUI_WITH_EXTRADATA_MANAGER_UI
 endif
diff -up VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp.noup VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp
--- VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp.noup	2016-07-12 17:39:48.181610043 -0300
+++ VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp	2016-07-12 17:40:28.040345555 -0300
@@ -42,9 +42,9 @@
 # include "UIModalWindowManager.h"
 # include "UIExtraDataManager.h"
 # include "UIMedium.h"
-# ifdef VBOX_OSE
+# ifdef VBOX_GUI_WITH_NETWORK_MANAGER
 #  include "UIDownloaderUserManual.h"
-# endif /* VBOX_OSE */
+# endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 # include "UIMachine.h"
 # include "VBoxAboutDlg.h"
 # include "UIHostComboEditor.h"
@@ -2657,7 +2657,7 @@ void UIMessageCenter::sltShowHelpAboutDi
 
 void UIMessageCenter::sltShowHelpHelpDialog()
 {
-#ifndef VBOX_OSE
+#ifndef VBOX_GUI_WITH_NETWORK_MANAGER
     /* For non-OSE version we just open it: */
     sltShowUserManual(vboxGlobal().helpFile());
 #else /* #ifndef VBOX_OSE */
@@ -2686,7 +2686,7 @@ void UIMessageCenter::sltShowHelpHelpDia
         /* Start downloading: */
         pDl->start();
     }
-#endif /* #ifdef VBOX_OSE */
+#endif /* #ifdef VBOX_GUI_WITH_NETWORK_MANAGER */
 }
 
 void UIMessageCenter::sltResetSuppressedMessages()
