diff -up VirtualBox-4.1.0_OSE/Config.kmk.libcxx VirtualBox-4.1.0_OSE/Config.kmk
--- VirtualBox-4.1.0_OSE/Config.kmk.libcxx	2011-07-19 08:40:48.000000000 -0300
+++ VirtualBox-4.1.0_OSE/Config.kmk	2011-07-19 22:00:54.363452764 -0300
@@ -3131,7 +3131,7 @@ TEMPLATE_VBOXR3EXE_LIBS                =
 TEMPLATE_VBOXR3EXE_LIBS                = \
 	$(L4_LIBDIR)/libvboxserver.s.so \
 	$(L4_LIBDIR)/libuc++.0.s.so \
-	supc++ gcc_eh gcc \
+	stdc++ \
 	$(L4_LIBDIR)/libdl.s.so \
 	$(L4_LIBDIR)/libuc.0.s.so
  else ifeq ($(KBUILD_TARGET),os2)
diff -up VirtualBox-4.1.0_OSE/configure.libcxx VirtualBox-4.1.0_OSE/configure
--- VirtualBox-4.1.0_OSE/configure.libcxx	2011-06-15 15:53:06.000000000 -0300
+++ VirtualBox-4.1.0_OSE/configure	2011-07-19 22:00:54.364452553 -0300
@@ -2582,7 +2582,7 @@ fi
 if [ "$OS" = "linux" ]; then
   # don't check for the static libstdc++ in the PUEL version as we build the
   # additions at a dedicated box
-  [ $OSE -ge 1 ] && check_staticlibstdcxx
+  #[ $OSE -ge 1 ] && check_staticlibstdcxx
   if [ $WITH_KMODS -eq 1 ]; then
     check_linux
   else
diff -up VirtualBox-4.1.0_OSE/src/VBox/Additions/x11/VBoxClient/Makefile.kmk.libcxx VirtualBox-4.1.0_OSE/src/VBox/Additions/x11/VBoxClient/Makefile.kmk
--- VirtualBox-4.1.0_OSE/src/VBox/Additions/x11/VBoxClient/Makefile.kmk.libcxx	2011-07-14 14:03:38.000000000 -0300
+++ VirtualBox-4.1.0_OSE/src/VBox/Additions/x11/VBoxClient/Makefile.kmk	2011-07-19 22:00:54.361453186 -0300
@@ -50,10 +50,7 @@ VBoxClient_LIBS = \
 	X11 \
 	Xrandr \
 	Xt
-# These are static replacements for gcc-specific parts of libstdc++
-VBoxClient_LIBS += \
-	supc++ \
-	gcc_eh
+VBoxClient_LIBS += stdc++
 ifdef VBOX_X11_SEAMLESS_GUEST
  VBoxClient_DEFS += SEAMLESS_GUEST DYNAMIC_RESIZE
  VBoxClient_SOURCES += \
diff -up VirtualBox-4.1.0_OSE/src/VBox/Storage/testcase/Makefile.kmk.libcxx VirtualBox-4.1.0_OSE/src/VBox/Storage/testcase/Makefile.kmk
--- VirtualBox-4.1.0_OSE/src/VBox/Storage/testcase/Makefile.kmk.libcxx	2011-06-15 15:52:53.000000000 -0300
+++ VirtualBox-4.1.0_OSE/src/VBox/Storage/testcase/Makefile.kmk	2011-07-19 22:04:13.736198227 -0300
@@ -68,7 +68,7 @@ ifdef VBOX_WITH_TESTCASES
  #
  # vbox-img - static because it migth be used as at standalone tool.
  #
- vbox-img_TEMPLATE = VBOXR3STATIC
+ vbox-img_TEMPLATE = VBOXR3EXE
  vbox-img_DEFS += IN_VBOXDDU IN_VBOXDDU_STATIC VBOX_HDD_NO_DYNAMIC_BACKENDS IN_RT_R3
  vbox-img_SOURCES = \
 	vbox-img.cpp \
