diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/Makefile.kmk
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/Makefile.kmk.noup	2013-11-29 17:23:50.000000000 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/Makefile.kmk	2013-11-30 02:43:13.346695389 -0200
@@ -24,6 +24,8 @@ include $(KBUILD_PATH)/subheader.kmk
 VBOX_VIRTUALBOX4_SRC := $(PATH_SUB_CURRENT)
 VBOX_VIRTUALBOX4_OUT_DIR := $(PATH_TARGET)/VirtualBox/misc
 
+# VBOX_WITH_UPDATE := 1
+
 # Show the update notifier dialog during startup. If this setting is disabled
 # the user can still update using the menu if desired.
 ifndef VBOX_OSE
@@ -91,6 +93,9 @@ ifdef VBOX_WITH_ICHAT_THEATER
 endif
 VirtualBox_DEFS          += \
 	$(if $(VBOX_WITH_ADDITIONS_AUTOUPDATE_UI),VBOX_WITH_ADDITIONS_AUTOUPDATE_UI) \
+	$(if $(VBOX_WITH_REGISTRATION),VBOX_WITH_REGISTRATION) \
+	$(if $(VBOX_WITH_REGISTRATION_REQUEST),VBOX_WITH_REGISTRATION_REQUEST) \
+	$(if $(VBOX_WITH_UPDATE),VBOX_WITH_UPDATE) \
 	$(if $(VBOX_WITH_UPDATE_REQUEST),VBOX_WITH_UPDATE_REQUEST) \
 	$(if $(VBOX_WITH_ALSA),VBOX_WITH_ALSA) \
 	$(if $(VBOX_WITH_PULSE),VBOX_WITH_PULSE) \
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp.noup	2013-11-29 17:23:54.000000000 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.cpp	2013-11-30 02:49:24.764012721 -0200
@@ -458,6 +458,7 @@ protected:
     }
 };
 
+#ifdef VBOX_WITH_UPDATE
 class UIActionSimpleCheckForUpdates : public UIActionSimple
 {
     Q_OBJECT;
@@ -512,6 +513,7 @@ protected:
         setStatusTip(QApplication::translate("UIActionPool", "Show a window with product information"));
     }
 };
+#endif /* VBOX_WITH_UPDATE */
 
 
 /* UIActionPool stuff: */
@@ -653,7 +655,9 @@ void UIActionPool::createActions()
     m_pool[UIActionIndex_Simple_ResetWarnings] = new UIActionSimpleResetWarnings(this);
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
     m_pool[UIActionIndex_Simple_NetworkAccessManager] = new UIActionSimpleNetworkAccessManager(this);
+#ifdef VBOX_WITH_UPDATE
     m_pool[UIActionIndex_Simple_CheckForUpdates] = new UIActionSimpleCheckForUpdates(this);
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
     m_pool[UIActionIndex_Simple_About] = new UIActionSimpleAbout(this);
 }
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h.noup	2013-11-30 02:43:13.347695188 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIActionPool.h	2013-11-30 02:48:48.931226366 -0200
@@ -59,7 +59,9 @@ enum UIActionIndex
     UIActionIndex_Simple_ResetWarnings,
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
     UIActionIndex_Simple_NetworkAccessManager,
+#ifdef VBOX_WITH_UPDATE
     UIActionIndex_Simple_CheckForUpdates,
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
     UIActionIndex_Simple_About,
 
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp.noup	2013-11-30 02:43:13.348694987 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.cpp	2013-11-30 02:48:13.618333727 -0200
@@ -97,7 +97,9 @@ const char* UIDefs::GUI_InvertMessageOpt
 
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
 /* Update manager definitions: */
+#ifdef VBOX_WITH_UPDATE
 const char* UIDefs::GUI_PreventApplicationUpdate = "GUI/PreventApplicationUpdate";
+#endif /* VBOX_WITH_UPDATE */
 const char* UIDefs::GUI_UpdateDate = "GUI/UpdateDate";
 const char* UIDefs::GUI_UpdateCheckCount = "GUI/UpdateCheckCount";
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h.noup	2013-11-30 02:43:13.348694987 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/UIDefs.h	2013-11-30 02:47:31.450818522 -0200
@@ -166,7 +166,9 @@ namespace UIDefs
 
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
     /* Update manager declarations: */
+#ifdef VBOX_WITH_UPDATE
     extern const char* GUI_PreventApplicationUpdate;
+#endif /* VBOX_WITH_UPDATE */
     extern const char* GUI_UpdateDate;
     extern const char* GUI_UpdateCheckCount;
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp.noup	2013-11-30 02:43:13.352694185 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp	2013-11-30 02:46:49.988159081 -0200
@@ -3415,12 +3415,14 @@ bool VBoxGlobal::isApprovedByExtraData(C
 }
 
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
+#ifdef VBOX_WITH_UPDATE
 /* static */
 bool VBoxGlobal::shouldWeAllowApplicationUpdate(CVirtualBox &vbox)
 {
     /* 'true' if disabling is not approved by the extra-data: */
     return !isApprovedByExtraData(vbox, GUI_PreventApplicationUpdate);
 }
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 
 /* static */
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h.noup	2013-11-30 02:43:13.353693983 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.h	2013-11-30 02:46:16.033987395 -0200
@@ -351,7 +351,9 @@ public:
     static bool isApprovedByExtraData(CVirtualBox &vbox, const QString &strExtraDataKey);
     static bool isApprovedByExtraData(CMachine &machine, const QString &strExtraDataKey);
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
-    static bool shouldWeAllowApplicationUpdate(CVirtualBox &vbox);
+#ifdef VBOX_WITH_UPDATE
+     static bool shouldWeAllowApplicationUpdate(CVirtualBox &vbox);
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
     static bool shouldWeShowMachine(CMachine &machine);
     static bool shouldWeAllowMachineReconfiguration(CMachine &machine,
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp.noup	2013-11-30 02:43:13.355693582 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/selector/UISelectorWindow.cpp	2013-11-30 02:45:26.315982744 -0200
@@ -1349,12 +1349,14 @@ void UISelectorWindow::prepareMenuHelp(Q
     pMenu->addSeparator();
     m_pNetworkAccessManager = gActionPool->action(UIActionIndex_Simple_NetworkAccessManager);
     pMenu->addAction(m_pNetworkAccessManager);
+#ifdef VBOX_WITH_UPDATE
     m_pUpdateAction = gActionPool->action(UIActionIndex_Simple_CheckForUpdates);
     CVirtualBox vbox = vboxGlobal().virtualBox();
     if (VBoxGlobal::shouldWeAllowApplicationUpdate(vbox))
         pMenu->addAction(m_pUpdateAction);
     else
         m_pUpdateAction->setEnabled(false);
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 #ifndef Q_WS_MAC
     pMenu->addSeparator();
@@ -1491,7 +1493,9 @@ void UISelectorWindow::prepareConnection
     connect(m_pResetWarningsAction, SIGNAL(triggered()), &msgCenter(), SLOT(sltResetSuppressedMessages()));
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
     connect(m_pNetworkAccessManager, SIGNAL(triggered()), gNetworkManager, SLOT(show()));
+#ifdef VBOX_WITH_UPDATE
     connect(m_pUpdateAction, SIGNAL(triggered()), gUpdateManager, SLOT(sltForceCheck()));
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
     connect(m_pAboutAction, SIGNAL(triggered()), &msgCenter(), SLOT(sltShowHelpAboutDialog()));
 
diff -up VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp.noup VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
--- VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp.noup	2013-11-30 02:43:13.356693382 -0200
+++ VirtualBox-4.3.4/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp	2013-11-30 02:50:58.641106504 -0200
@@ -36,7 +36,9 @@
 #include "UIGlobalSettingsGeneral.h"
 #include "UIGlobalSettingsInput.h"
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
+#ifdef VBOX_WITH_UPDATE
 # include "UIGlobalSettingsUpdate.h"
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 #include "UIGlobalSettingsLanguage.h"
 #include "UIGlobalSettingsDisplay.h"
@@ -359,6 +361,7 @@ UISettingsDialogGlobal::UISettingsDialog
                     break;
                 }
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
+#ifdef VBOX_WITH_UPDATE
                 /* Update page: */
                 case GlobalSettingsPageType_Update:
                 {
@@ -368,6 +371,7 @@ UISettingsDialogGlobal::UISettingsDialog
                             iPageIndex, "#update", pSettingsPage);
                     break;
                 }
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
                 /* Language page: */
                 case GlobalSettingsPageType_Language:
@@ -498,8 +502,10 @@ void UISettingsDialogGlobal::retranslate
     m_pSelector->setItemText(GlobalSettingsPageType_Input, tr("Input"));
 
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
+#ifdef VBOX_WITH_UPDATE
     /* Update page: */
     m_pSelector->setItemText(GlobalSettingsPageType_Update, tr("Update"));
+#endif /* VBOX_WITH_UPDATE */
 #endif /* VBOX_GUI_WITH_NETWORK_MANAGER */
 
     /* Language page: */
