diff -up VirtualBox-4.2.0/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp.defsys VirtualBox-4.2.0/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp
--- VirtualBox-4.2.0/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp.defsys	2012-09-13 14:18:17.459957901 -0300
+++ VirtualBox-4.2.0/src/VBox/Frontends/VirtualBox/src/widgets/UINameAndSystemEditor.cpp	2012-09-13 14:22:15.346645645 -0300
@@ -104,6 +104,9 @@ UINameAndSystemEditor::UINameAndSystemEd
         m_pFamilyCombo->setItemData(i, families[i].GetFamilyId(), TypeID);
     }
     m_pFamilyCombo->setCurrentIndex(0);
+    int iLinuxIndex = m_pFamilyCombo->findData ("Linux", TypeID);
+    if (iLinuxIndex != -1)
+        m_pFamilyCombo->setCurrentIndex (iLinuxIndex);
     sltFamilyChanged(m_pFamilyCombo->currentIndex());
 
     /* Setup connections: */
@@ -203,7 +206,7 @@ void UINameAndSystemEditor::sltFamilyCha
     /* Or select Ubuntu item for Linux family as default: */
     else if (strFamilyId == "Linux")
     {
-        int iIndexUbuntu = m_pTypeCombo->findData("Ubuntu", TypeID);
+        int iIndexUbuntu = m_pTypeCombo->findData("Linux26", TypeID);
         if (iIndexUbuntu != -1)
             m_pTypeCombo->setCurrentIndex(iIndexUbuntu);
     }
