diff -up VirtualBox-5.1.0/src/VBox/Additions/darwin/Installer/VBoxGuestAdditions_mpkg/Localizable.strings VirtualBox-5.1.0/src/VBox/Additions/darwin/Installer/VBoxGuestAdditions_mpkg/Localizable
diff -up VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/main.cpp.strings VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/main.cpp
--- VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/main.cpp.istrings	2016-07-12 11:54:03.000000000 -0300
+++ VirtualBox-5.1.0/src/VBox/Frontends/VirtualBox/src/main.cpp	2016-07-12 18:03:06.261386064 -0300
@@ -101,11 +101,11 @@ QString g_QStrHintLinuxNoMemory = QAppli
     );
 
 QString g_QStrHintLinuxNoDriver = QApplication::tr(
-    "The VirtualBox Linux kernel driver (vboxdrv) is either not loaded or "
-    "there is a permission problem with /dev/vboxdrv. Please reinstall the kernel "
-    "module by executing<br/><br/>"
-    "  <font color=blue>'/sbin/vboxconfig'</font><br/><br/>"
-    "as root."
+    "The VirtualBox Linux kernel driver (vboxdrv) is either not loaded
+    "or installed."
+    "You may attempt to load the driver by issuing the following"
+    "command with system administrator (root) privileges:<br/><br/>"
+    "  <font color=blue>'/etc/rc.d/rc.vboxdrv restart'</font><br/><br/>"
     );
 
 QString g_QStrHintOtherWrongDriverVersion = QApplication::tr(
@@ -119,7 +119,7 @@ QString g_QStrHintLinuxWrongDriverVersio
     "The VirtualBox kernel modules do not match this version of "
     "VirtualBox. The installation of VirtualBox was apparently not "
     "successful. Executing<br/><br/>"
-    "  <font color=blue>'/sbin/vboxconfig'</font><br/><br/>"
+    "  <font color=blue>'/etc/rc.d/rc.vboxdrv restart'</font><br/><br/>"
     "may correct this. Make sure that you do not mix the "
     "OSE version and the PUEL version of VirtualBox."
     );
diff -up VirtualBox-5.1.0/src/VBox/Installer/darwin/VirtualBox_mpkg/Localizable.strings VirtualBox-5.1.0/src/VBox/Installer/darwin/VirtualBox_mpkg/Localizable
diff -up VirtualBox-5.1.0/src/VBox/VMM/VMMR3/VM.cpp.strings VirtualBox-5.1.0/src/VBox/VMM/VMMR3/VM.cpp
--- VirtualBox-5.1.0/src/VBox/VMM/VMMR3/VM.cpp.istrings	2016-07-12 11:55:10.000000000 -0300
+++ VirtualBox-5.1.0/src/VBox/VMM/VMMR3/VM.cpp	2016-07-12 18:03:06.262386057 -0300
@@ -384,8 +384,8 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
 #ifdef RT_OS_LINUX
                     pszError = N_("VirtualBox kernel driver not loaded. The vboxdrv kernel module "
                                   "was either not loaded or /dev/vboxdrv is not set up properly. "
-                                  "Re-setup the kernel module by executing "
-                                  "'/sbin/vboxconfig' as root");
+                                  "Reboot the computer, or insert the kernel module by executing "
+                                  "'/etc/rc.d/rc.vboxdrv restart' as root");
 #else
                     pszError = N_("VirtualBox kernel driver not loaded");
 #endif
@@ -426,8 +426,8 @@ VMMR3DECL(int)   VMR3Create(uint32_t cCp
 #ifdef RT_OS_LINUX
                     pszError = N_("VirtualBox kernel driver not installed. The vboxdrv kernel module "
                                   "was either not loaded or /dev/vboxdrv was not created for some "
-                                  "reason. Re-setup the kernel module by executing "
-                                  "'/sbin/vboxconfig' as root");
+                                  "reason. Insert the kernel module by executing "
+                                  "'/etc/rc.d/rc.vboxdrv restart' as root");
 #else
                     pszError = N_("VirtualBox kernel driver not installed");
 #endif
