--- configure.gcc	2015-07-25 21:55:07.876086828 -0300
+++ configure	2015-07-25 21:58:10.901182043 -0300
@@ -422,9 +422,9 @@ check_gcc()
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
-             -o \( $cc_maj -eq 5 -a $cc_min -gt 1 \) \
+             -o \( $cc_maj -eq 5 -a $cc_min -gt 2 \) \
              -o $cc_maj -gt 5 ]; then
-        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10 or gcc 5.1"
+        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10 or gcc <= 5.2"
         fail really
       else
         log_success "found version $cc_ver"
