diff -up VirtualBox-4.2.0/src/VBox/HostDrivers/Makefile.kmk.nodkms VirtualBox-4.2.0/src/VBox/HostDrivers/Makefile.kmk
--- VirtualBox-4.2.0/src/VBox/HostDrivers/Makefile.kmk.nodkms	2012-09-13 19:26:42.095689836 -0300
+++ VirtualBox-4.2.0/src/VBox/HostDrivers/Makefile.kmk	2012-09-13 19:26:50.912827596 -0300
@@ -41,7 +41,7 @@ if  !defined(VBOX_ONLY_DOCS) \
   else
   endif
  endif
- if1of ($(KBUILD_TARGET), darwin freebsd linux)
+ if1of ($(KBUILD_TARGET), darwin freebsd)
   include $(PATH_SUB_CURRENT)/$(KBUILD_TARGET)/Makefile.kmk
  endif
 
