--- VirtualBox-4.0.6_OSE/src/VBox/HostDrivers/Support/linux/Makefile.orig	2011-04-21 07:31:13.000000000 -0300
+++ VirtualBox-4.0.6_OSE/src/VBox/HostDrivers/Support/linux/Makefile	2011-04-21 18:35:18.469518900 -0300
@@ -162,7 +162,7 @@ ifeq ($(KERNELRELEASE),)
  # kernel base directory
  ifndef KERN_DIR
   # build for the current kernel, version check
-  KERN_DIR := /lib/modules/$(shell uname -r)/build
+  KERN_DIR := /lib/modules/$(KERNELRELEASE)/build
   ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
    KERN_DIR := /usr/src/linux
    ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
@@ -191,7 +191,7 @@ ifeq ($(KERNELRELEASE),)
  # module install dir, only for current kernel
  ifneq ($(filter install install_rpm,$(MAKECMDGOALS)),)
   ifndef MODULE_DIR
-   MODULE_DIR_TST := /lib/modules/$(shell uname -r)
+   MODULE_DIR_TST := /lib/modules/$(KERNELRELEASE)
    ifeq ($(shell if test -d $(MODULE_DIR_TST); then echo yes; fi),yes)
     MODULE_DIR := $(MODULE_DIR_TST)/misc
    else
@@ -282,7 +282,7 @@ install_rpm: $(MODULE)
 else # eq ($(MAKECMDGOALS),clean)
 
  ifndef KERN_DIR
-  KERN_DIR := /lib/modules/$(shell uname -r)/build
+  KERN_DIR := /lib/modules/$(KERNELRELEASE)/build
   ifeq ($(wildcard $(KERN_DIR)/Makefile),)
    KERN_DIR := /usr/src/linux
   endif
