--- VirtualBox-4.0.0_OSE/doc/manual/docbook2latex.xsl.orig	2010-12-26 13:08:19.134841850 +0200
+++ VirtualBox-4.0.0_OSE/doc/manual/docbook2latex.xsl	2010-12-26 13:08:35.025675882 +0200
@@ -129,9 +129,9 @@
 
 % use Bera Mono (a variant of Bitstream Vera Mono) as typewriter font
 % (requires texlive-fontsextra)
-\usepackage[scaled]{beramono}
+% \usepackage[scaled]{beramono}
 % previously: use Courier as typewriter font:
-% \usepackage{courier}
+\usepackage{courier}
 
 \definecolor{colNote}{rgb}{0,0,0}
 \definecolor{colWarning}{rgb}{0,0,0}
