--- VirtualBox-4.1.2_OSE/src/VBox/HostDrivers/VBoxPci/Makefile.kmk.orig	2011-08-15 09:28:45.000000000 -0300
+++ VirtualBox-4.1.2_OSE/src/VBox/HostDrivers/VBoxPci/Makefile.kmk	2011-08-15 21:55:29.420685104 -0300
@@ -23,6 +23,7 @@ if1of ($(KBUILD_TARGET), linux)
  # The driver.
  # Note! For Linux, this is just a compile test. Don't use the binary!
  #
+ ifdef VBOX_WITH_VBOXDRV
  SYSMODS += VBoxPci
  VBoxPci_TEMPLATE       = VBOXR0DRV
  VBoxPci_INST           = $(INST_VBOXPCI)$(if $(eq $(KBUILD_TARGET),darwin),Contents/MacOS/)
@@ -41,6 +42,7 @@ if1of ($(KBUILD_TARGET), linux)
  VBoxPci_LIBS          += \
  	$(PATH_STAGE_LIB)/SUPR0IdcClient$(VBOX_SUFF_LIB)
 
+ endif # VBOX_WITH_VBOXDRV
 
  #
  # Install the sources.
