--- VirtualBox-5.0.0/configure.orig	2015-07-09 19:42:57.678234368 -0300
+++ VirtualBox-5.0.0/configure	2015-07-09 19:44:55.931186427 -0300
@@ -2688,7 +2688,7 @@ if [ $ONLY_ADDITIONS -eq 0 ]; then
   # don't check for yasm for the time beeing as 0.40 and 0.50 both have known bugs
   # [ "$OS" != "darwin" ] && check_yasm
   [ "$OS" != "darwin"  ] && check_xsltproc
-  [ "$OS" != "darwin"  ] && check_mkisofs
+  #[ "$OS" != "darwin"  ] && check_mkisofs
 fi
 
 # the libraries
@@ -2756,7 +2756,7 @@ if [ "$OS" = "linux" ]; then
   check_compiler_h
   [ $ONLY_ADDITIONS -eq 0 -a "$BUILD_MACHINE" = "amd64" -a $WITH_VMMRAW -eq 1 ] && check_32bit
   # tools/common/makeself*
-  [ $OSE -ge 1 ] && check_makeself
+ # [ $OSE -ge 1 ] && check_makeself
 fi
 
 [ -n "$SETUP_WINE" ] && setup_wine
