--- libquicktime-1.2.4/plugins/ffmpeg/audio.c.0000	2012-03-30 03:44:28.000000000 +0800
+++ libquicktime-1.2.4/plugins/ffmpeg/audio.c	2013-07-15 06:06:24.087006398 +0800
@@ -515,7 +515,7 @@
   if(!chunk_packets)
     return 0;
 
-  new_samples = num_samples + AVCODEC_MAX_AUDIO_FRAME_SIZE / (2 * track_map->channels);
+  new_samples = num_samples + 192000 / (2 * track_map->channels);
   
   if(codec->sample_buffer_alloc <
      codec->sample_buffer_end - codec->sample_buffer_start + new_samples)
@@ -671,7 +671,7 @@
    */
 
   num_samples += 8192;
-  new_samples = num_samples + AVCODEC_MAX_AUDIO_FRAME_SIZE / (2 * track_map->channels);
+  new_samples = num_samples + 192000 / (2 * track_map->channels);
   
   /* Reallocate sample buffer */
   
--- libquicktime-1.2.4/plugins/ffmpeg/lqt_ffmpeg.c.0000	2013-07-15 06:10:56.524786247 +0800
+++ libquicktime-1.2.4/plugins/ffmpeg/lqt_ffmpeg.c	2013-07-15 06:14:24.780733842 +0800
@@ -89,7 +89,6 @@
     PARAM_QSCALE, \
     PARAM_QCOMPRESS, \
     PARAM_QBLUR, \
-    PARAM_QUANTIZER_NOISE_SHAPING, \
     PARAM_TRELLIS
 
 #define ENCODE_PARAM_VIDEO_QUANTIZER_IP \
@@ -109,10 +108,8 @@
     .type =      LQT_PARAMETER_SECTION, \
   }, \
   PARAM_GOP_SIZE, \
-  PARAM_SCENE_CHANGE_THRESHOLD, \
   PARAM_SCENECHANGE_FACTOR, \
-  PARAM_FLAG_CLOSED_GOP, \
-  PARAM_FLAG2_STRICT_GOP
+  PARAM_FLAG_CLOSED_GOP
 
 #define ENCODE_PARAM_VIDEO_FRAMETYPES_IPB \
   ENCODE_PARAM_VIDEO_FRAMETYPES_IP, \
@@ -208,12 +205,10 @@
   PARAM_FLAG_AC_PRED_MPEG4,
   ENCODE_PARAM_VIDEO_RATECONTROL,
   ENCODE_PARAM_VIDEO_QUANTIZER_IPB,
-  PARAM_FLAG_CBP_RD,
   ENCODE_PARAM_VIDEO_ME,
   PARAM_FLAG_GMC,
   PARAM_FLAG_4MV,
   PARAM_FLAG_MV0,
-  PARAM_FLAG_QP_RD,
   ENCODE_PARAM_VIDEO_ME_PRE,
   ENCODE_PARAM_VIDEO_QPEL,
   ENCODE_PARAM_VIDEO_MASKING,
--- libquicktime-1.2.4/plugins/ffmpeg/params.c.0000	2012-03-07 22:10:41.000000000 +0800
+++ libquicktime-1.2.4/plugins/ffmpeg/params.c	2013-07-15 06:13:08.368551514 +0800
@@ -202,8 +202,6 @@
   PARAM_INT("ff_max_b_frames",max_b_frames);
   PARAM_FLOAT("ff_b_quant_factor",b_quant_factor);
   PARAM_INT("ff_b_frame_strategy",b_frame_strategy);
-  PARAM_INT("ff_luma_elim_threshold",luma_elim_threshold);
-  PARAM_INT("ff_chroma_elim_threshold",chroma_elim_threshold);
   PARAM_INT("ff_strict_std_compliance",strict_std_compliance);
   PARAM_QP2LAMBDA("ff_b_quant_offset",b_quant_offset);
   PARAM_INT("ff_rc_min_rate",rc_min_rate);
@@ -241,8 +239,6 @@
   PARAM_QP2LAMBDA("ff_lmax", lmax);
   PARAM_INT("ff_noise_reduction",noise_reduction);
   PARAM_INT_SCALE("ff_rc_initial_buffer_occupancy",rc_initial_buffer_occupancy,1000);
-  PARAM_INT("ff_inter_threshold",inter_threshold);
-  PARAM_INT("ff_quantizer_noise_shaping",quantizer_noise_shaping);
   PARAM_INT("ff_thread_count",thread_count);
   PARAM_INT("ff_me_threshold",me_threshold);
   PARAM_INT("ff_mb_threshold",mb_threshold);
@@ -272,8 +268,6 @@
   PARAM_FLAG("ff_flag_bitexact",CODEC_FLAG_BITEXACT);
   PARAM_FLAG("ff_flag_ac_pred",CODEC_FLAG_AC_PRED);
   //  PARAM_FLAG("ff_flag_h263p_umv",CODEC_FLAG_H263P_UMV); // Unused
-  PARAM_FLAG("ff_flag_cbp_rd",CODEC_FLAG_CBP_RD);
-  PARAM_FLAG("ff_flag_qp_rd",CODEC_FLAG_QP_RD);
 
 #if LIBAVCODEC_VERSION_MAJOR >= 54
   PARAM_DICT_FLAG("ff_flag_h263p_aiv", "aiv");
@@ -288,7 +282,6 @@
   PARAM_FLAG("ff_flag_loop_filter",CODEC_FLAG_LOOP_FILTER);
   PARAM_FLAG("ff_flag_closed_gop",CODEC_FLAG_CLOSED_GOP);
   PARAM_FLAG2("ff_flag2_fast",CODEC_FLAG2_FAST);
-  PARAM_FLAG2("ff_flag2_strict_gop",CODEC_FLAG2_STRICT_GOP);
   PARAM_ENUM("ff_coder_type",coder_type,coder_type);
   
   }
--- libquicktime-1.2.4/plugins/ffmpeg/params.h.0000	2011-05-11 23:13:39.000000000 +0800
+++ libquicktime-1.2.4/plugins/ffmpeg/params.h	2013-07-15 06:14:01.440067233 +0800
@@ -177,32 +177,6 @@
     .val_max =     { .val_int = 1 }, \
   }
 
-#define PARAM_LUMA_ELIM_THRESHOLD \
-  { \
-    .name =        "ff_luma_elim_threshold", \
-    .real_name =   TRS("Luma elimination threshold"),   \
-    .type =        LQT_PARAMETER_INT, \
-    .val_default = { .val_int =  0 }, \
-    .val_min =     { .val_int = -99 }, \
-    .val_max =     { .val_int =  99 }, \
-    .help_string = TRS("Single coefficient elimination threshold for luminance. \
-Negative values also consider dc coefficient. -4 is JVT \
-recommendation") \
-  }
-    
-#define PARAM_CHROMA_ELIM_THRESHOLD \
-  { \
-    .name =        "ff_chroma_elim_threshold", \
-    .real_name =   TRS("Chroma elimination threshold"), \
-    .type =        LQT_PARAMETER_INT, \
-    .val_default = { .val_int =  0 }, \
-    .val_min =     { .val_int = -99 }, \
-    .val_max =     { .val_int =  99 }, \
-    .help_string = TRS("Single coefficient elimination threshold for \
-chrominamce. Negative values also consider dc \
-coefficient. 7 is JVT recommendation") \
-  }
-
 #define PARAM_STRICT_STANDARD_COMPLIANCE \
   { \
     .name =        "ff_strict_std_compliance", \
@@ -739,26 +713,6 @@
 
 /* Does nothing */
 /** Frame types */
-#define PARAM_INTER_THRESHOLD \
-  { \
-    .name =        "ff_inter_threshold", \
-    .real_name =   TRS("Inter threshold"),       \
-    .type =        LQT_PARAMETER_INT, \
-    .val_default = { .val_int = 0 }, \
-  }
-
-/** Quantizer */
-#define PARAM_QUANTIZER_NOISE_SHAPING \
-  { \
-    .name =        "ff_quantizer_noise_shaping", \
-    .real_name =   TRS("Quantizer noise shaping"),      \
-    .type =        LQT_PARAMETER_INT, \
-    .val_default = { .val_int = 0 }, \
-    .val_min =     { .val_int = 0 }, \
-    .val_max =     { .val_int = 3 }, \
-    .help_string = TRS("Choose quantization such that noise will be masked by " \
-                       "similar-frequency content in the image")        \
-  }
 
 /** Motion estimation */
 #define PARAM_ME_THRESHOLD \
@@ -1085,28 +1039,6 @@
     .val_max =     { .val_int = 1 }, \
   }
 
-#define PARAM_FLAG_CBP_RD \
-  { \
-    .name =        "ff_flag_cbp_rd", \
-    .real_name =   TRS("CBP RD"),     \
-    .type =        LQT_PARAMETER_INT, \
-    .val_default = { .val_int = 0 }, \
-    .val_min =     { .val_int = 0 }, \
-    .val_max =     { .val_int = 1 }, \
-       .help_string = TRS("Use rate distortion optimization for cbp. This can only be used together with trellis quantization."),  \
-  }
-
-#define PARAM_FLAG_QP_RD \
-  { \
-    .name =        "ff_flag_qp_rd", \
-    .real_name =   TRS("QP RD"),      \
-    .type =        LQT_PARAMETER_INT, \
-    .val_default = { .val_int = 0 }, \
-    .val_min =     { .val_int = 0 }, \
-    .val_max =     { .val_int = 1 }, \
-    .help_string = TRS("Use rate distortion optimization for qp selection. Can only be used if MB decision mode is \"Rate distoration\""), \
-  }
-
 #define PARAM_FLAG_H263P_AIV \
   { \
     .name =        "ff_flag_h263p_aiv", \
