--- libtorrent-rasterbar-0.14.10/include/libtorrent/udp_socket.hpp.gcc45	2010-02-18 15:51:17.000000000 +0900
+++ libtorrent-rasterbar-0.14.10/include/libtorrent/udp_socket.hpp	2010-07-29 03:25:43.000000000 +0900
@@ -37,6 +37,7 @@
 #include "libtorrent/buffer.hpp"
 #include "libtorrent/session_settings.hpp"
 
+#include <list>
 #include <vector>
 #include <boost/function.hpp>
 #include <boost/thread/mutex.hpp>
