diff -up konq-plugins-4.4.0/adblock/adblock.cpp.kde234624 konq-plugins-4.4.0/adblock/adblock.cpp
--- konq-plugins-4.4.0/adblock/adblock.cpp.kde234624	2010-09-07 15:51:47.901818132 -0500
+++ konq-plugins-4.4.0/adblock/adblock.cpp	2010-09-07 16:07:46.873058854 -0500
@@ -105,10 +105,8 @@ AdBlock::~AdBlock()
     KParts::StatusBarExtension *statusBarEx = KParts::StatusBarExtension::childObject(m_part);
 
     if (statusBarEx)
-    {
         statusBarEx->removeStatusBarItem(m_label);
-        delete m_label;
-    }
+    delete m_label;
     m_label = 0;
     delete m_menu;
     m_menu = 0;
