diff -up konq-plugins-4.4.0/adblock/adblock.cpp.qt47 konq-plugins-4.4.0/adblock/adblock.cpp
--- konq-plugins-4.4.0/adblock/adblock.cpp.qt47	2010-09-07 15:51:03.776797042 -0500
+++ konq-plugins-4.4.0/adblock/adblock.cpp	2010-09-07 15:51:47.901818132 -0500
@@ -300,7 +300,7 @@ void AdBlock::updateFilters()
 // ----------------------------------------------------------------------------
 
 AdElement::AdElement() :
-  m_url(0), m_category(0), m_type(0), m_blocked(false) {}
+  m_blocked(false) {}
 
 AdElement::AdElement(const QString &url, const QString &category,
 		     const QString &type, bool blocked, const DOM::Node&node) :
