diff -Nur blueman-1.21.orig/blueman/Constants.py.in blueman-1.21/blueman/Constants.py.in
--- blueman-1.21.orig/blueman/Constants.py.in	2009-09-20 05:01:26.000000000 -0500
+++ blueman-1.21/blueman/Constants.py.in	2009-12-13 16:26:29.643850538 -0600
@@ -10,7 +10,7 @@
 POLKIT = "@polkit_val@" == "yes"
 HAL_ENABLED = "@hal_en@" == "yes"
 
-DEF_BROWSE_COMMAND = "nautilus --browser obex://[%d]"
+DEF_BROWSE_COMMAND = "/usr/libexec/blueman-open %d"
 
 import os
 import gettext
