--- a/apps/blueman-report	2015-05-24 09:01:34.000000000 +0100
+++ b/apps/blueman-report	2015-05-27 12:43:44.916949194 +0100
@@ -77,7 +77,7 @@ print("")
 print("=====================================")
 print("")
 print("blueman: " + VERSION)
-v = subprocess.Popen(['/usr/sbin/bluetoothd', '-v'], stdout=subprocess.PIPE).stdout.read().decode('UTF-8')[:-1]
+v = subprocess.Popen(['/usr/libexec/bluetooth/bluetoothd', '-v'], stdout=subprocess.PIPE).stdout.read().decode('UTF-8')[:-1]
 print("BlueZ: " + v)
 print("Distribution: ")
 print("Desktop: " + os.environ.get('XDG_CURRENT_DESKTOP'))
