diff -up claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure.ac.orig claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure.ac
--- claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure.ac.orig	2011-07-05 10:27:14.000000000 +0200
+++ claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure.ac	2011-07-05 10:28:44.000000000 +0200
@@ -151,7 +151,7 @@ PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 2.10)
 
 dnl Check for libnotify support
 LIBNOTIFY_REQUIRED=0.4.3
-LIBNOTIFY_TOO_HIGH=0.7.0
+LIBNOTIFY_TOO_HIGH=0.8.0
 AC_ARG_ENABLE(libnotify,
 	[  --disable-libnotify     do not include libnotify support],
  	[enable_libnotify=$enableval], [enable_libnotify=auto])
diff -up claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure.orig claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure
--- claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure.orig	2011-07-05 10:27:03.000000000 +0200
+++ claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/configure	2011-07-05 10:28:37.000000000 +0200
@@ -14934,7 +14934,7 @@ $as_echo "yes" >&6; }
 fi
 
 LIBNOTIFY_REQUIRED=0.4.3
-LIBNOTIFY_TOO_HIGH=0.7.0
+LIBNOTIFY_TOO_HIGH=0.8.0
 # Check whether --enable-libnotify was given.
 if test "${enable_libnotify+set}" = set; then :
   enableval=$enable_libnotify; enable_libnotify=$enableval
diff -up claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_popup.c.orig claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_popup.c
--- claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_popup.c.orig	2010-08-31 08:36:59.000000000 +0200
+++ claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_popup.c	2011-07-05 10:26:16.000000000 +0200
@@ -339,7 +339,7 @@ static gboolean notification_libnotify_c
     break;
   }
 
-  ppopup->notification = notify_notification_new(summary, utf8_str, NULL, NULL);
+  ppopup->notification = notify_notification_new(summary, utf8_str, NULL);
   g_free(utf8_str);
   if(ppopup->notification == NULL) {
     debug_print("Notification Plugin: Failed to create a new "
diff -up claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_trayicon.c.orig claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_trayicon.c
--- claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_trayicon.c.orig	2011-03-04 17:31:08.000000000 +0100
+++ claws-mail-extra-plugins-3.7.9/notification_plugin-0.27/src/notification_trayicon.c	2011-07-05 10:26:16.000000000 +0200
@@ -602,8 +602,7 @@ static gboolean notification_trayicon_po
   summary  = notification_trayicon_popup_assemble_summary();
   utf8_str = notification_trayicon_popup_assemble_body(msginfo);
 
-  popup.notification = notify_notification_new(summary, utf8_str, NULL, NULL);
-  notify_notification_attach_to_status_icon(popup.notification, trayicon);
+  popup.notification = notify_notification_new(summary, utf8_str, NULL);
 
   g_free(summary);
   g_free(utf8_str);
