diff -up clamd-3.5.1/libclamd/clamd-plugin.c.orig clamd-3.5.1/libclamd/clamd-plugin.c
--- clamd-3.5.1/libclamd/clamd-plugin.c.orig	2011-05-20 19:41:51.000000000 +0200
+++ clamd-3.5.1/libclamd/clamd-plugin.c	2011-05-20 19:42:05.000000000 +0200
@@ -210,7 +210,7 @@ void clamd_create_config_automatic(const
 		}
 	}
 	fclose(conf);
-	if (! Socket && (Socket->socket.port || Socket->socket.path)) {
+	if (! (Socket && (Socket->socket.port || Socket->socket.path))) {
 		/*g_error("%s: Not able to find required information", path);*/
 		alertpanel_error(_("%s: Not able to find required information\nclamd will be disabled"), path);
 	}
