diff -up DirectFB-1.4.11/configure.in.cflags DirectFB-1.4.11/configure.in
--- DirectFB-1.4.11/configure.in.cflags	2010-11-27 19:28:34.866252742 +0100
+++ DirectFB-1.4.11/configure.in	2010-11-27 19:38:36.834159439 +0100
@@ -145,13 +145,11 @@ AM_CONDITIONAL(X11_CORE, test "$enable_x
 AC_CHECK_HEADERS(linux/compiler.h linux/unistd.h asm/page.h signal.h linux/videodev.h linux/videodev2.h)
 
 
-dnl Clear default CFLAGS
-if test x"$CFLAGS" = x"-g -O2"; then
-  CFLAGS=
+dnl Set default CFLAGS if unset
+if test x"$CFLAGS" = x""; then
+  CFLAGS="-O3 -ffast-math -pipe"
 fi
 
-CFLAGS="-O3 -ffast-math -pipe $CFLAGS"
-
 DFB_INTERNAL_CFLAGS="-D_GNU_SOURCE $DFB_INTERNAL_CFLAGS"
 
 AC_ARG_ENABLE(extra-warnings,
