diff -up kdepim-runtime-4.7.2/agents/nepomukfeeder/nepomukfeederagentbase.cpp.nepomuk_notification kdepim-runtime-4.7.2/agents/nepomukfeeder/nepomukfeederagentbase.cpp
--- kdepim-runtime-4.7.2/agents/nepomukfeeder/nepomukfeederagentbase.cpp.nepomuk_notification	2011-10-02 08:24:50.000000000 -0500
+++ kdepim-runtime-4.7.2/agents/nepomukfeeder/nepomukfeederagentbase.cpp	2011-10-13 08:45:05.662502890 -0500
@@ -417,7 +417,9 @@ void NepomukFeederAgentBase::selfTest()
   emit status( Broken, i18n( "Nepomuk not operational" ) );
   if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.kde.pim.nepomukfeeder.selftestreport" ) ) )
     return;
-  KNotification::event( KNotification::Warning, i18n( "Nepomuk Indexing Disabled" ), message );
+  // Disable knotification spam 
+  // FIXME/TODO: make this configurable via systemsettings->app&system notifications
+  // KNotification::event( KNotification::Warning, i18n( "Nepomuk Indexing Disabled" ), message );
   QDBusConnection::sessionBus().unregisterService( QLatin1String( "org.kde.pim.nepomukfeeder.selftestreport" ) );
 }
 
