diff -up kdepim-runtime-4.8.90/kcm/serverconfigmodule.cpp.akonadi_kcm_sqlite kdepim-runtime-4.8.90/kcm/serverconfigmodule.cpp
--- kdepim-runtime-4.8.90/kcm/serverconfigmodule.cpp.akonadi_kcm_sqlite	2012-06-06 17:47:53.000000000 -0500
+++ kdepim-runtime-4.8.90/kcm/serverconfigmodule.cpp	2012-06-10 08:30:13.660890777 -0500
@@ -57,6 +57,7 @@ ServerConfigModule::ServerConfigModule(
   // supported drivers
   ui_driver.driverBox->addItem( "Mysql", QVariant( "QMYSQL" ) );
   ui_driver.driverBox->addItem( "PostgreSQL", QVariant( "QPSQL" ) );
+  ui_driver.driverBox->addItem( "SQLite", QVariant( "QSQLITE3" ) );
 
   setButtons( KCModule::Default | KCModule::Apply );
 
@@ -69,6 +70,12 @@ ServerConfigModule::ServerConfigModule(
   m_psqlWidget = new QWidget( this );
   m_stackWidget->addWidget( m_psqlWidget );
   ui_psql.setupUi( m_psqlWidget );
+
+  // SQLite
+  m_sqliteWidget = new QWidget( this );
+  m_stackWidget->addWidget( m_sqliteWidget );
+  // ui_sqlite.setupUi( m_sqliteWidget );
+
   m_stackWidget->setCurrentWidget( m_mysqlWidget );
 
   QWidget *widget = new QWidget( this );
@@ -167,6 +174,12 @@ void ServerConfigModule::save()
   settings.setValue( "Port", ui_psql.port->text() );
   settings.endGroup();
 
+  // sqlite group
+  settings.beginGroup( "SQLITE" );
+  // TODO: make it configurable
+  settings.setValue( "Name", "akonadi" );
+  settings.endGroup();
+
   // selected driver
   settings.beginGroup( "GENERAL" );
   settings.setValue( "Driver", ui_driver.driverBox->itemData( ui_driver.driverBox->currentIndex() ).toString() );
@@ -220,8 +233,10 @@ void ServerConfigModule::driverChanged(
 {
   if ( ui_driver.driverBox->itemData( index ).toString() == "QMYSQL" ) {
     m_stackWidget->setCurrentWidget( m_mysqlWidget );
-  } else {
+  } else if( ui_driver.driverBox->itemData( index ).toString() == "QPSQL" ) {
     m_stackWidget->setCurrentWidget( m_psqlWidget );
+  } else {
+    m_stackWidget->setCurrentWidget( m_sqliteWidget );
   }
 }
 
diff -up kdepim-runtime-4.8.90/kcm/serverconfigmodule.h.akonadi_kcm_sqlite kdepim-runtime-4.8.90/kcm/serverconfigmodule.h
--- kdepim-runtime-4.8.90/kcm/serverconfigmodule.h.akonadi_kcm_sqlite	2012-05-22 19:24:00.000000000 -0500
+++ kdepim-runtime-4.8.90/kcm/serverconfigmodule.h	2012-06-10 08:28:49.748939822 -0500
@@ -56,6 +56,7 @@ class ServerConfigModule : public KCModu
     QStackedWidget *m_stackWidget;
     QWidget *m_mysqlWidget;
     QWidget *m_psqlWidget;
+    QWidget *m_sqliteWidget;
 
     QComboBox *m_driverBox;
 };
