diff -up kdepim-runtime-4.5-beta1/kcm/kcm_akonadi.desktop.show kdepim-runtime-4.5-beta1/kcm/kcm_akonadi.desktop
--- kdepim-runtime-4.5-beta1/kcm/kcm_akonadi.desktop.show	2010-08-11 13:50:43.551908182 -0300
+++ kdepim-runtime-4.5-beta1/kcm/kcm_akonadi.desktop	2010-08-11 13:51:41.562990042 -0300
@@ -8,9 +8,10 @@ X-KDE-ModuleType=Library
 X-KDE-Library=kcm_akonadi
 X-KDE-FactoryName=kcm_akonadi
 X-KDE-HasReadOnlyMode=false
-X-KDE-ParentApp=
+X-KDE-ParentApp=kcontrol
 X-KDE-ParentComponents=
 X-KDE-CfgDlgHierarchy=
+X-KDE-System-Settings-Parent-Category=personal-information
 
 Name=Akonadi Configuration
 Name[ar]=ضبط اكوندا
diff -up kdepim-runtime-4.5-beta1/kcm/kcm_akonadi_resources.desktop.show kdepim-runtime-4.5-beta1/kcm/kcm_akonadi_resources.desktop
--- kdepim-runtime-4.5-beta1/kcm/kcm_akonadi_resources.desktop.show	2010-08-11 13:50:07.427587674 -0300
+++ kdepim-runtime-4.5-beta1/kcm/kcm_akonadi_resources.desktop	2010-08-11 13:51:07.987042701 -0300
@@ -8,10 +8,9 @@ X-KDE-ModuleType=Library
 X-KDE-Library=kcm_akonadi_resources
 X-KDE-FactoryName=kcm_akonadi_resources
 X-KDE-HasReadOnlyMode=false
-X-KDE-ParentApp=kcontrol
+X-KDE-ParentApp=
 X-KDE-ParentComponents=
 X-KDE-CfgDlgHierarchy=
-X-KDE-System-Settings-Parent-Category=personal-information
 
 Name=Akonadi Resources Configuration
 Name[ar]=ضبط موارد اكوندا
