diff -up kdepim-runtime-4.7.1/kcm/serverconfigmodule.cpp.sqlite-support kdepim-runtime-4.7.1/kcm/serverconfigmodule.cpp
--- kdepim-runtime-4.7.1/kcm/serverconfigmodule.cpp.sqlite-support	2011-05-21 01:11:05.000000000 +0200
+++ kdepim-runtime-4.7.1/kcm/serverconfigmodule.cpp	2011-09-20 17:43:26.908150816 +0200
@@ -57,6 +57,7 @@ ServerConfigModule::ServerConfigModule( 
   // supported drivers
   ui_driver.driverBox->addItem( "Mysql", QVariant( "QMYSQL" ) );
   ui_driver.driverBox->addItem( "PostgreSQL", QVariant( "QPSQL" ) );
+  ui_driver.driverBox->addItem( "SQLite", QVariant( "QSQLITE" ) );
 
   setButtons( KCModule::Default | KCModule::Apply );
 
@@ -69,6 +70,12 @@ ServerConfigModule::ServerConfigModule( 
   m_psqlWidget = new QWidget( this );
   m_stackWidget->addWidget( m_psqlWidget );
   ui_psql.setupUi( m_psqlWidget );
+
+  // SQLite
+  m_sqliteWidget = new QWidget( this );
+  m_stackWidget->addWidget( m_sqliteWidget );
+  // ui_sqlite.setupUi( m_sqliteWidget );
+
   m_stackWidget->setCurrentWidget( m_mysqlWidget );
 
   QWidget *widget = new QWidget( this );
@@ -156,6 +163,12 @@ void ServerConfigModule::save()
   settings.setValue( "Port", ui_psql.port->text() );
   settings.endGroup();
 
+  // sqlite group
+  settings.beginGroup( "SQLITE" );
+  // TODO: make it configurable
+  settings.setValue( "Name", "akonadi" );
+  settings.endGroup();
+
   // selected driver
   settings.beginGroup( "GENERAL" );
   settings.setValue( "Driver", ui_driver.driverBox->itemData( ui_driver.driverBox->currentIndex() ).toString() );
@@ -210,9 +223,12 @@ void ServerConfigModule::driverChanged( 
   if( ui_driver.driverBox->itemData( index ).toString() == "QMYSQL" ) {
     m_stackWidget->setCurrentWidget( m_mysqlWidget );
   }
-  else {
+  else if( ui_driver.driverBox->itemData( index ).toString() == "QPSQL" ) {
     m_stackWidget->setCurrentWidget( m_psqlWidget );
   }
+  else {
+    m_stackWidget->setCurrentWidget( m_sqliteWidget );
+  }
 }
 
 #include "serverconfigmodule.moc"
diff -up kdepim-runtime-4.7.1/kcm/serverconfigmodule.h.sqlite-support kdepim-runtime-4.7.1/kcm/serverconfigmodule.h
--- kdepim-runtime-4.7.1/kcm/serverconfigmodule.h.sqlite-support	2011-05-21 01:11:05.000000000 +0200
+++ kdepim-runtime-4.7.1/kcm/serverconfigmodule.h	2011-09-20 17:43:26.908150816 +0200
@@ -56,6 +56,7 @@ class ServerConfigModule : public KCModu
     QStackedWidget *m_stackWidget;
     QWidget *m_mysqlWidget;
     QWidget *m_psqlWidget;
+    QWidget *m_sqliteWidget;
 
     QComboBox *m_driverBox;
 };
