diff -up libQtGTL-0.9.1/QtGTL/CMakeLists.txt.dso libQtGTL-0.9.1/QtGTL/CMakeLists.txt
--- libQtGTL-0.9.1/QtGTL/CMakeLists.txt.dso	2010-02-13 04:23:11.000000000 -0600
+++ libQtGTL-0.9.1/QtGTL/CMakeLists.txt	2010-04-08 10:30:56.984474364 -0500
@@ -13,7 +13,7 @@ set(libQtGTL_SRCS
 qt4_automoc(${libQtShiva_SRCS})
 
 add_library(QtGTL SHARED ${libQtGTL_SRCS})
-target_link_libraries(QtGTL ${QT_QTGUI_LIBRARY} ${OPENSHIVA_LIBRARIES} )
+target_link_libraries(QtGTL ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${OPENSHIVA_LIBRARIES} )
 
 install(TARGETS QtGTL DESTINATION ${LIB_INSTALL_DIR} )
 install( FILES QImageBuffer.h QImageImage.h DESTINATION ${INCLUDE_INSTALL_DIR}/QtGTL )
