diff -up OpenCV-2.2.0/modules/core/include/opencv2/core/core.hpp.gcc46 OpenCV-2.2.0/modules/core/include/opencv2/core/core.hpp
--- OpenCV-2.2.0/modules/core/include/opencv2/core/core.hpp.gcc46	2010-12-05 04:35:24.000000000 +0100
+++ OpenCV-2.2.0/modules/core/include/opencv2/core/core.hpp	2011-02-02 16:44:02.177243993 +0100
@@ -52,6 +52,7 @@
 #ifdef __cplusplus
 
 #ifndef SKIP_INCLUDES
+#include <cstddef>
 #include <limits.h>
 #include <algorithm>
 #include <cmath>
