--- opencv-2.4.11/samples/gpu/CMakeLists.txt.orig	2015-02-25 13:10:31.000000000 +0100
+++ opencv-2.4.11/samples/gpu/CMakeLists.txt	2015-05-19 18:35:11.803554049 +0200
@@ -4,6 +4,15 @@
                                      opencv_superres)
 ocv_check_dependencies(${OPENCV_GPU_SAMPLES_REQUIRED_DEPS})
 
+# remove all matching elements from the list
+MACRO(list_filterout lst regex)
+  foreach(item ${${lst}})
+    if(item MATCHES "${regex}")
+      list(REMOVE_ITEM ${lst} "${item}")
+    endif()
+  endforeach()
+ENDMACRO()
+
 if(BUILD_EXAMPLES AND OCV_DEPENDENCIES_FOUND)
   set(project "gpu")
   string(TOUPPER "${project}" project_upper)
@@ -69,15 +78,6 @@
     endif()
   ENDMACRO()
 
-  # remove all matching elements from the list
-  MACRO(list_filterout lst regex)
-    foreach(item ${${lst}})
-      if(item MATCHES "${regex}")
-        list(REMOVE_ITEM ${lst} "${item}")
-      endif()
-    endforeach()
-  ENDMACRO()
-
   file(GLOB all_samples RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.cpp)
 
   if(NOT HAVE_OPENGL)
