diff -up OpenCV-2.1.0/CMakeLists.txt.lapack OpenCV-2.1.0/CMakeLists.txt
--- OpenCV-2.1.0/CMakeLists.txt.lapack	2010-06-25 15:59:47.000000000 +0200
+++ OpenCV-2.1.0/CMakeLists.txt	2010-06-25 16:01:04.000000000 +0200
@@ -839,7 +839,7 @@ IF(NOT BUILD_SHARED_LIBS)
         set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} stdc++)
     endif()
 
-    set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} flann zlib opencv_lapack)
+    set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} flann z lapack)
 endif()
 
 
diff -up OpenCV-2.1.0/src/cxcore/CMakeLists.txt.lapack OpenCV-2.1.0/src/cxcore/CMakeLists.txt
--- OpenCV-2.1.0/src/cxcore/CMakeLists.txt.lapack	2010-06-25 15:59:47.000000000 +0200
+++ OpenCV-2.1.0/src/cxcore/CMakeLists.txt	2010-06-25 16:00:35.000000000 +0200
@@ -53,7 +53,7 @@ set_target_properties(${the_target} PROP
     )
 
 # Add the required libraries for linking:
-target_link_libraries(${the_target} ${OPENCV_LINKER_LIBS} ${IPP_LIBS} opencv_lapack zlib flann)
+target_link_libraries(${the_target} ${OPENCV_LINKER_LIBS} ${IPP_LIBS} lapack z flann)
 
 # Linker flag needed for Windows Mobile 5 and 6 SDKs
 if(MSVC)
@@ -67,7 +67,7 @@ if(MSVC)
         )
 endif()
 
-add_dependencies(${the_target} opencv_lapack zlib)
+add_dependencies(${the_target} lapack z)
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
