diff -up OpenCV-2.4.4/modules/contrib/doc/facerec/src/crop_face.py.pillow OpenCV-2.4.4/modules/contrib/doc/facerec/src/crop_face.py
--- OpenCV-2.4.4/modules/contrib/doc/facerec/src/crop_face.py.pillow	2013-01-28 00:00:47.000000000 +0100
+++ OpenCV-2.4.4/modules/contrib/doc/facerec/src/crop_face.py	2013-02-18 19:56:11.979405809 +0100
@@ -31,7 +31,8 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
-import sys, math, Image
+import sys, math
+from PIL import Image
 
 def Distance(p1,p2):
   dx = p2[0] - p1[0]
diff -up OpenCV-2.4.4/modules/python/test/ticket_6.py.pillow OpenCV-2.4.4/modules/python/test/ticket_6.py
--- OpenCV-2.4.4/modules/python/test/ticket_6.py.pillow	2013-01-28 00:00:48.000000000 +0100
+++ OpenCV-2.4.4/modules/python/test/ticket_6.py	2013-02-18 19:58:14.288877399 +0100
@@ -2,7 +2,7 @@
 
 import urllib
 import cv2.cv as cv
-import Image
+from PIL import Image
 import unittest
 
 class TestLoadImage(unittest.TestCase):
