diff -up opencv-2.4.7/modules/ts/CMakeLists.txt.ts_static opencv-2.4.7/modules/ts/CMakeLists.txt
--- opencv-2.4.7/modules/ts/CMakeLists.txt.ts_static	2013-11-06 15:26:39.000000000 +0100
+++ opencv-2.4.7/modules/ts/CMakeLists.txt	2013-11-14 21:28:28.289051531 +0100
@@ -4,7 +4,7 @@ if(IOS)
   ocv_module_disable(ts)
 endif()
 
-set(OPENCV_MODULE_TYPE STATIC)
+#set(OPENCV_MODULE_TYPE STATIC)
 set(OPENCV_MODULE_IS_PART_OF_WORLD FALSE)
 
 if(HAVE_CUDA)
