diff -Naur OpenCV-2.0.0.orig/3rdparty/flann/CMakeLists.txt OpenCV-2.0.0/3rdparty/flann/CMakeLists.txt
--- OpenCV-2.0.0.orig/3rdparty/flann/CMakeLists.txt	2009-10-01 01:21:00.000000000 +0100
+++ OpenCV-2.0.0/3rdparty/flann/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -86,8 +86,8 @@
 INSTALL (
     TARGETS flann flann_s
     RUNTIME DESTINATION bin
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX}
 )
 
 # INSTALL (
diff -Naur OpenCV-2.0.0.orig/CMakeLists.txt OpenCV-2.0.0/CMakeLists.txt
--- OpenCV-2.0.0.orig/CMakeLists.txt	2009-10-01 01:21:05.000000000 +0100
+++ OpenCV-2.0.0/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -405,8 +405,8 @@
 
     string(REGEX MATCH "[0-9].[0-9]" PYTHON_VERSION_MAJOR_MINOR "${PYTHON_VERSION_FULL}")
     if(UNIX)
-        set(PYTHON_PLUGIN_INSTALL_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
-        set(PYTHON_PACKAGES_PATH lib/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
+        set(PYTHON_PLUGIN_INSTALL_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages/opencv)
+        set(PYTHON_PACKAGES_PATH lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR_MINOR}/site-packages)
     endif()
     if(WIN32)
         get_filename_component(PYTHON_PATH "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\${PYTHON_VERSION_MAJOR_MINOR}\\InstallPath]" ABSOLUTE CACHE)
@@ -731,7 +731,7 @@
 # Set CMAKE_INCLUDE_DIRS_CONFIGCMAKE to the list of include directories:
 
 set(CMAKE_INCLUDE_DIRS_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/include/opencv")
-set(CMAKE_LIB_DIRS_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_LIB_DIRS_CONFIGCMAKE "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
 
 #exec_program(${CMAKE_COMMAND} ARGS "-E make_directory \"${CMAKE_BINARY_DIR}/unix-install/\"")
 exec_program(mkdir ARGS "-p \"${CMAKE_BINARY_DIR}/unix-install/\"")
@@ -764,19 +764,19 @@
 #  Part 2/2: ${BIN_DIR}/unix-install/opencv.pc -> For use *with* "make install"
 # prefix=/usr
 # exec_prefix=${prefix}
-# libdir=${exec_prefix}/lib
+# libdir=${exec_prefix}/lib${LIB_SUFFIX}
 # includedir=${prefix}/include/opencv
 # -------------------------------------------------------------------------------------------
 set(prefix 		${CMAKE_INSTALL_PREFIX})
 set(exec_prefix		"\${prefix}")
-set(libdir		"\${exec_prefix}/lib")
+set(libdir		"\${exec_prefix}/lib${LIB_SUFFIX}")
 set(includedir		"\${prefix}/include")
 set(VERSION             ${OPENCV_VERSION})
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/opencv.pc.cmake.in" "${CMAKE_BINARY_DIR}/unix-install/opencv.pc" @ONLY IMMEDIATE)
 
 if(UNIX)
-    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib/pkgconfig)
+    install(FILES ${CMAKE_BINARY_DIR}/unix-install/opencv.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 endif()
 
 
diff -Naur OpenCV-2.0.0.orig/interfaces/ffopencv/CMakeLists.txt OpenCV-2.0.0/interfaces/ffopencv/CMakeLists.txt
--- OpenCV-2.0.0.orig/interfaces/ffopencv/CMakeLists.txt	2009-10-01 01:20:57.000000000 +0100
+++ OpenCV-2.0.0/interfaces/ffopencv/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -33,5 +33,5 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
     )
diff -Naur OpenCV-2.0.0.orig/src/cv/CMakeLists.txt OpenCV-2.0.0/src/cv/CMakeLists.txt
--- OpenCV-2.0.0.orig/src/cv/CMakeLists.txt	2009-10-01 01:20:58.000000000 +0100
+++ OpenCV-2.0.0/src/cv/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -66,8 +66,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
     DESTINATION include/opencv
diff -Naur OpenCV-2.0.0.orig/src/cvaux/CMakeLists.txt OpenCV-2.0.0/src/cvaux/CMakeLists.txt
--- OpenCV-2.0.0.orig/src/cvaux/CMakeLists.txt	2009-10-01 01:20:59.000000000 +0100
+++ OpenCV-2.0.0/src/cvaux/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -76,8 +76,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${aux_hdrs_ext}
         DESTINATION include/opencv
diff -Naur OpenCV-2.0.0.orig/src/cxcore/CMakeLists.txt OpenCV-2.0.0/src/cxcore/CMakeLists.txt
--- OpenCV-2.0.0.orig/src/cxcore/CMakeLists.txt	2009-10-01 01:20:59.000000000 +0100
+++ OpenCV-2.0.0/src/cxcore/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -66,8 +66,8 @@
 
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 # install each module headers separately
 install(FILES ${lib_hdrs}
diff -Naur OpenCV-2.0.0.orig/src/highgui/CMakeLists.txt OpenCV-2.0.0/src/highgui/CMakeLists.txt
--- OpenCV-2.0.0.orig/src/highgui/CMakeLists.txt	2009-10-01 01:20:58.000000000 +0100
+++ OpenCV-2.0.0/src/highgui/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -191,8 +191,8 @@
 if(WIN32)
 	target_link_libraries(${the_target} comctl32 gdi32 ole32)
 	
-	if (MSVC)
-		target_link_libraries(${the_target} vfw32)
+	if (MSVC)
+		target_link_libraries(${the_target} vfw32)
 	endif()
 
 	if(MINGW)
@@ -206,8 +206,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${highgui_ext_hdrs}
         DESTINATION include/opencv
diff -Naur OpenCV-2.0.0.orig/src/ml/CMakeLists.txt OpenCV-2.0.0/src/ml/CMakeLists.txt
--- OpenCV-2.0.0.orig/src/ml/CMakeLists.txt	2009-10-01 01:20:59.000000000 +0100
+++ OpenCV-2.0.0/src/ml/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -61,8 +61,8 @@
 
 install(TARGETS ${the_target}
 	RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 
 install(FILES ${lib_hdrs}
         DESTINATION include/opencv
diff -Naur OpenCV-2.0.0.orig/tests/cxts/CMakeLists.txt OpenCV-2.0.0/tests/cxts/CMakeLists.txt
--- OpenCV-2.0.0.orig/tests/cxts/CMakeLists.txt	2009-10-01 01:20:56.000000000 +0100
+++ OpenCV-2.0.0/tests/cxts/CMakeLists.txt	2010-02-25 23:10:44.000000000 +0100
@@ -63,6 +63,6 @@
 if(WIN32)
 install(TARGETS ${the_target}
     RUNTIME DESTINATION bin COMPONENT main
-	LIBRARY DESTINATION lib COMPONENT main
-	ARCHIVE DESTINATION lib COMPONENT main)
+	LIBRARY DESTINATION lib${LIB_SUFFIX} COMPONENT main
+	ARCHIVE DESTINATION lib${LIB_SUFFIX} COMPONENT main)
 endif()
