diff -up opencv-2.4.7/CMakeLists.txt.cmake_paths opencv-2.4.7/CMakeLists.txt
--- opencv-2.4.7/CMakeLists.txt.cmake_paths	2013-11-06 08:26:39.000000000 -0600
+++ opencv-2.4.7/CMakeLists.txt	2013-11-18 14:38:58.763060851 -0600
@@ -286,12 +286,7 @@ else()
   endif()
   set(OPENCV_INCLUDE_INSTALL_PATH "include")
 
-  math(EXPR SIZEOF_VOID_P_BITS "8 * ${CMAKE_SIZEOF_VOID_P}")
-  if(LIB_SUFFIX AND NOT SIZEOF_VOID_P_BITS EQUAL LIB_SUFFIX)
-    set(OPENCV_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/cmake/opencv)
-  else()
-    set(OPENCV_CONFIG_INSTALL_PATH share/OpenCV)
-  endif()
+  set(OPENCV_CONFIG_INSTALL_PATH lib${LIB_SUFFIX}/OpenCV)
 endif()
 
 set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${OPENCV_LIB_INSTALL_PATH}")
