diff -up OpenCV-2.1.0/3rdparty/CMakeLists.txt.nointernal OpenCV-2.1.0/3rdparty/CMakeLists.txt
--- OpenCV-2.1.0/3rdparty/CMakeLists.txt.nointernal	2010-04-06 03:24:34.000000000 +0200
+++ OpenCV-2.1.0/3rdparty/CMakeLists.txt	2010-06-25 15:48:06.000000000 +0200
@@ -1,6 +1,6 @@
 add_subdirectory(flann)
-add_subdirectory(lapack)
-add_subdirectory(zlib)
+#add_subdirectory(lapack)
+#add_subdirectory(zlib)
 if(WITH_JASPER AND NOT JASPER_FOUND)
 	add_subdirectory(libjasper)
 endif()
