diff -up OpenCV-2.2.0/3rdparty/CMakeLists.txt.nointernal OpenCV-2.2.0/3rdparty/CMakeLists.txt
--- OpenCV-2.2.0/3rdparty/CMakeLists.txt.nointernal	2010-12-05 04:35:23.000000000 +0100
+++ OpenCV-2.2.0/3rdparty/CMakeLists.txt	2010-12-07 16:00:53.510070005 +0100
@@ -2,8 +2,12 @@ if(ANDROID)
  configure_file("${CMAKE_SOURCE_DIR}/Android.mk.modules.in" "${CMAKE_CURRENT_BINARY_DIR}/Android.mk")
 endif()
 
-add_subdirectory(lapack)
-add_subdirectory(zlib)
+#if(WITH_LAPACK AND NOT LAPACK_FOUND)
+#	add_subdirectory(lapack)
+#endif()
+#if(WITH_ZLIB AND NOT ZLIB_FOUND)
+#	add_subdirectory(zlib)
+#endif()
 if(WITH_JASPER AND NOT JASPER_FOUND)
 	add_subdirectory(libjasper)
 endif()
diff -up OpenCV-2.2.0/CMakeLists.txt.nointernal OpenCV-2.2.0/CMakeLists.txt
--- OpenCV-2.2.0/CMakeLists.txt.nointernal	2010-12-05 04:35:23.000000000 +0100
+++ OpenCV-2.2.0/CMakeLists.txt	2010-12-07 16:00:53.532070005 +0100
@@ -280,6 +280,8 @@ set(WITH_JPEG ON CACHE BOOL "Include JPE
 set(WITH_JASPER ON CACHE BOOL "Include JPEG2K support")
 set(WITH_TIFF ON CACHE BOOL "Include TIFF support")
 set(WITH_OPENEXR ON CACHE BOOL "Include ILM support via OpenEXR")
+set(WITH_LAPACK ON CACHE BOOL "Include LAPACK support")
+set(WITH_ZLIB ON CACHE BOOL "Include ZLIB support")
 
 if(UNIX)
     set(WITH_FFMPEG ON CACHE BOOL "Include FFMPEG support")
diff -up OpenCV-2.2.0/modules/core/CMakeLists.txt.nointernal OpenCV-2.2.0/modules/core/CMakeLists.txt
--- OpenCV-2.2.0/modules/core/CMakeLists.txt.nointernal	2010-12-05 04:35:24.000000000 +0100
+++ OpenCV-2.2.0/modules/core/CMakeLists.txt	2010-12-07 16:47:14.767070005 +0100
@@ -1,3 +1,11 @@
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../../3rdparty/include")
-set(deps opencv_lapack zlib)
+if(WITH_LAPACK AND NOT LAPACK_FOUND)
+	if(WITH_ZLIB AND NOT ZLIB_FOUND)
+		set(deps opencv_lapack zlib)
+	else()
+		set(deps lapack z)
+	endif()
 define_opencv_module(core ${deps})
+else()
+define_opencv_module(core lapack z)
+endif()
diff -up OpenCV-2.2.0/modules/highgui/CMakeLists.txt.nointernal OpenCV-2.2.0/modules/highgui/CMakeLists.txt
--- OpenCV-2.2.0/modules/highgui/CMakeLists.txt.nointernal	2010-12-05 04:35:24.000000000 +0100
+++ OpenCV-2.2.0/modules/highgui/CMakeLists.txt	2010-12-07 17:01:11.963070004 +0100
@@ -285,7 +285,7 @@ if(WITH_TIFF AND NOT TIFF_FOUND)
 	#target_link_libraries(${the_target} libtiff)
 endif()
 
-target_link_libraries(${the_target} ${OPENCV_LINKER_LIBS} opencv_core opencv_imgproc zlib ${JPEG_LIBRARIES} ${PNG_LIBRARIES} ${TIFF_LIBRARIES} ${JASPER_LIBRARIES} ${HIGHGUI_LIBRARIES} ${OPENEXR_LIBRARIES})
+target_link_libraries(${the_target} ${OPENCV_LINKER_LIBS} opencv_core opencv_imgproc z ${JPEG_LIBRARIES} ${PNG_LIBRARIES} ${TIFF_LIBRARIES} ${JASPER_LIBRARIES} ${HIGHGUI_LIBRARIES} ${OPENEXR_LIBRARIES})
 
 if(APPLE)
 	target_link_libraries(${the_target} "-lbz2 -framework Cocoa -framework QuartzCore")
--- OpenCV-2.2.0/OpenCVFindOpenEXR.cmake.nointernal	2010-12-07 17:12:16.592070004 +0100
+++ OpenCV-2.2.0/OpenCVFindOpenEXR.cmake	2010-12-07 17:12:26.539070004 +0100
@@ -44,7 +44,7 @@ FIND_LIBRARY(OPENEXR_ILMIMF_LIBRARY
 IF (OPENEXR_INCLUDE_PATH AND OPENEXR_IMATH_LIBRARY AND OPENEXR_ILMIMF_LIBRARY AND OPENEXR_IEX_LIBRARY AND OPENEXR_HALF_LIBRARY)
     SET(OPENEXR_FOUND TRUE)
     SET(OPENEXR_INCLUDE_PATHS ${OPENEXR_INCLUDE_PATH} CACHE STRING "The include paths needed to use OpenEXR")
-    SET(OPENEXR_LIBRARIES ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_HALF_LIBRARY} zlib CACHE STRING "The libraries needed to use OpenEXR")
+    SET(OPENEXR_LIBRARIES ${OPENEXR_IMATH_LIBRARY} ${OPENEXR_ILMIMF_LIBRARY} ${OPENEXR_IEX_LIBRARY} ${OPENEXR_HALF_LIBRARY} CACHE STRING "The libraries needed to use OpenEXR")
 ENDIF ()
 
 IF(OPENEXR_FOUND)
