diff -up OpenCV-2.4.2/cmake/templates/OpenCVConfig.cmake.in.pkgcmake2 OpenCV-2.4.2/cmake/templates/OpenCVConfig.cmake.in
--- OpenCV-2.4.2/cmake/templates/OpenCVConfig.cmake.in.pkgcmake2	2012-07-09 18:21:58.374388493 +0200
+++ OpenCV-2.4.2/cmake/templates/OpenCVConfig.cmake.in	2012-07-09 18:22:32.465014020 +0200
@@ -53,11 +53,7 @@ set(OpenCV_USE_MANGLED_PATHS @OpenCV_USE
 get_filename_component(OpenCV_CONFIG_PATH "${CMAKE_CURRENT_LIST_FILE}" PATH CACHE)
 
 if(NOT WIN32 OR OpenCV_ANDROID_NATIVE_API_LEVEL GREATER 0)
-  if(OpenCV_ANDROID_NATIVE_API_LEVEL GREATER 0)
-    set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../../..")
-  else()
-    set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../..")
-  endif()
+  set(OpenCV_INSTALL_PATH "${OpenCV_CONFIG_PATH}/../../..")
   # Get the absolute path with no ../.. relative marks, to eliminate implicit linker warnings
   if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_LESS 2.8)
     get_filename_component(OpenCV_INSTALL_PATH "${OpenCV_INSTALL_PATH}" ABSOLUTE)
