diff -up gsl-1.15/Makefile.in.pom gsl-1.15/Makefile.in
--- gsl-1.15/Makefile.in.pom	2011-05-04 23:10:41.000000000 +0200
+++ gsl-1.15/Makefile.in	2011-05-09 08:51:48.000000000 +0200
@@ -313,7 +313,8 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 SUBDIRS = gsl utils sys test err const complex cheb block vector matrix permutation combination multiset sort ieee-utils cblas blas linalg eigen specfunc dht qrng rng randist fft poly fit multifit statistics siman sum integration interpolation histogram ode-initval ode-initval2 roots multiroots min multimin monte ntuple diff deriv cdf wavelet bspline doc
 SUBLIBS = block/libgslblock.la blas/libgslblas.la \
-	bspline/libgslbspline.la complex/libgslcomplex.la \
+	bspline/libgslbspline.la cblas/libgslcblas.la \
+	complex/libgslcomplex.la \
 	cheb/libgslcheb.la dht/libgsldht.la diff/libgsldiff.la \
 	deriv/libgslderiv.la eigen/libgsleigen.la err/libgslerr.la \
 	fft/libgslfft.la fit/libgslfit.la histogram/libgslhistogram.la \
