commit 02ae6621f28d90b717c679b61cd1d1800d5c69e7
Author: Miroslav Lichvar <mlichvar@redhat.com>
Date:   Mon Dec 13 16:44:16 2010 +0100

    Install shared objects with executable permissions

diff --git a/autoconf/aclocal.m4 b/autoconf/aclocal.m4
index d5eb3f2..b5525cf 100644
--- a/autoconf/aclocal.m4
+++ b/autoconf/aclocal.m4
@@ -578,7 +578,7 @@ ELFLIB_MAJOR_MINOR_MICRO="\$(ELFLIB_MAJOR_MINOR).\$(ELF_MICRO_VERSION)"
 dnl# This specifies the target to use in the makefile to install the shared library
 INSTALL_ELFLIB_TARGET="install-elf-and-links"
 ELFLIB_BUILD_NAME="\$(ELFLIB_MAJOR_MINOR_MICRO)"
-INSTALL_MODULE="\$(INSTALL_DATA)"
+INSTALL_MODULE="\$(INSTALL)"
 SLANG_DLL_CFLAGS=""
 M_LIB="-lm"
 
diff --git a/modules/Makefile.in b/modules/Makefile.in
index 0384359..06b71ec 100644
--- a/modules/Makefile.in
+++ b/modules/Makefile.in
@@ -61,7 +61,7 @@ SOCKET_LIBS = @X_EXTRA_LIBS@
 MODULE_INSTALL_DIR = @libdir@/slang/v@slang_major_version@/modules
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
-INSTALL_MODULE	= @INSTALL_MODULE@
+INSTALL_MODULE	= @INSTALL@
 MKINSDIR	= $(CONFIG_DIR)/autoconf/mkinsdir.sh
 SLSH_DIR	= $(prefix)/share/slsh
 CMAPS_DIR	= $(SLSH_DIR)/cmaps
diff --git a/src/Makefile.in b/src/Makefile.in
index 03fa056..d2b9f1b 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -195,7 +195,7 @@ install-elf-and-links:
 	-$(RM) $(DEST_LIBDIR)/$(ELFLIB)
 	-$(RM) $(DEST_LIBDIR)/$(ELFLIB_MAJOR)
 	@echo installing $(ELFLIB_BUILD_NAME) in $(DEST_LIBDIR)
-	$(INSTALL_DATA) $(ELFDIR_ELF_LIB) $(DEST_LIBDIR)/
+	$(INSTALL) $(ELFDIR_ELF_LIB) $(DEST_LIBDIR)/
 	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_BUILD_NAME) $(ELFLIB_MAJOR)
 	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_MAJOR) $(ELFLIB)
 	@echo ""
