diff -up slang-2.2.3/modules/Makefile.in.makefile slang-2.2.3/modules/Makefile.in
--- slang-2.2.3/modules/Makefile.in.makefile	2010-12-13 08:18:53.000000000 +0100
+++ slang-2.2.3/modules/Makefile.in	2010-12-13 15:35:24.794109086 +0100
@@ -60,7 +60,7 @@ SOCKET_LIBS = @X_EXTRA_LIBS@
 MODULE_INSTALL_DIR = @libdir@/slang/v@slang_major_version@/modules
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
-INSTALL_MODULE	= @INSTALL_MODULE@
+INSTALL_MODULE	= @INSTALL@
 MKINSDIR	= $(CONFIG_DIR)/autoconf/mkinsdir.sh
 SLSH_DIR	= $(prefix)/share/slsh
 CMAPS_DIR	= $(SLSH_DIR)/cmaps
@@ -74,7 +74,7 @@ DEST_SLSH_DIR	= $(DESTDIR)$(SLSH_DIR)
 DEST_CMAPSDIR	= $(DESTDIR)$(CMAPS_DIR)
 DEST_HELPDIR	= $(DESTDIR)$(HELP_DIR)
 #---------------------------------------------------------------------------
-RPATH = @RPATH@
+RPATH =
 
 LIBS = $(LDFLAGS) $(RPATH) $(SLANG_LIB) $(DL_LIB) @M_LIB@
 INCS = $(SLANG_SRCINC)
diff -up slang-2.2.3/slsh/Makefile.in.makefile slang-2.2.3/slsh/Makefile.in
--- slang-2.2.3/slsh/Makefile.in.makefile	2010-12-13 08:18:53.000000000 +0100
+++ slang-2.2.3/slsh/Makefile.in	2010-12-13 15:35:24.794109086 +0100
@@ -19,7 +19,7 @@ SLANG_SRCINC	= -I@SRCDIR@
 SLANG_SRCLIB	= -L@ELFDIR@
 #---------------------------------------------------------------------------
 OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ @M_LIB@
-RPATH		= @RPATH@
+RPATH		=
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
diff -up slang-2.2.3/src/Makefile.in.makefile slang-2.2.3/src/Makefile.in
--- slang-2.2.3/src/Makefile.in.makefile	2010-12-13 08:18:53.000000000 +0100
+++ slang-2.2.3/src/Makefile.in	2010-12-13 15:35:24.801109332 +0100
@@ -195,9 +195,9 @@ install-elf-and-links:
 	-$(RM) $(DEST_LIBDIR)/$(ELFLIB)
 	-$(RM) $(DEST_LIBDIR)/$(ELFLIB_MAJOR)
 	@echo installing $(ELFLIB_BUILD_NAME) in $(DEST_LIBDIR)
-	$(INSTALL_DATA) $(ELFDIR_ELF_LIB) $(DEST_LIBDIR)/
+	$(INSTALL) $(ELFDIR_ELF_LIB) $(DEST_LIBDIR)/
 	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_BUILD_NAME) $(ELFLIB_MAJOR)
-	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_BUILD_NAME) $(ELFLIB)
+	-cd $(DEST_LIBDIR) && $(LN) $(ELFLIB_MAJOR) $(ELFLIB)
 	@echo ""
 	@echo $(ELFLIB_BUILD_NAME) created in $(DEST_LIBDIR).
 	@echo The links $(ELFLIB) and $(ELFLIB_MAJOR) to $(ELFLIB_BUILD_NAME) were also created.
