Don't free an uninitialized pointer. SF#2907102

diff -up cracklib-2.8.15/python/_cracklibmodule.c cracklib-2.8.15/python/_cracklibmodule.c
--- cracklib-2.8.15/python/_cracklibmodule.c	2009-12-01 15:39:57.000000000 -0500
+++ cracklib-2.8.15/python/_cracklibmodule.c	2009-12-01 15:39:59.000000000 -0500
@@ -106,6 +106,7 @@ _cracklib_FascistCheck(PyObject *self, P
             return NULL;
         }
         free(dictfile);
+        defaultdict = NULL;
     } else
     {
         defaultdict = strdup(GetDefaultCracklibDict());
