Do not depend on config.h in public header.
diff -up cracklib-2.9.1/lib/packer.h.inttypes cracklib-2.9.1/lib/packer.h
--- cracklib-2.9.1/lib/packer.h.inttypes	2013-12-03 15:00:15.000000000 +0100
+++ cracklib-2.9.1/lib/packer.h	2013-12-09 09:07:38.306394809 +0100
@@ -30,17 +30,7 @@
 #define _(String) (String)
 #endif
 
-#if defined(HAVE_INTTYPES_H)
-#include <inttypes.h>
-#else
-#if defined(HAVE_STDINT_H)
 #include <stdint.h>
-#else
-typedef unsigned int uint32_t;
-typedef unsigned short uint16_t;
-#endif
-#endif
-
 
 struct pi_header
 {
