--- acinclude.m4.orig	2013-05-04 18:03:53.000000000 -0300
+++ acinclude.m4	2013-05-04 18:04:36.994738539 -0300
@@ -73,28 +73,9 @@ AC_DEFUN([CHECK_QT],
 AC_REQUIRE([AC_PROG_CXX])
 AC_REQUIRE([AC_PATH_X])
 
-AC_MSG_CHECKING([QTDIR])
-AC_ARG_WITH([qtdir], [  --with-qtdir=DIR        Qt installation directory [default=$QTDIR]], QTDIR=$withval)
 AC_ARG_WITH([qt-includes], [  --with-qt-includes=DIR	Directory containing the qt include files ], QT_INC_DIR=$withval)
 AC_ARG_WITH([qt-libs], [  --with-qt-libraries=DIR	Directory containing the qt libraries ], QT_LIB_DIR=$withval)
 
-# Check that QTDIR is defined or that --with-qtdir given
-if test x"$QTDIR" = x; then
-    QT_SEARCH="/usr/lib/qt31 /usr/local/qt31 /usr/lib/qt3 /usr/local/qt3 /usr/lib/qt2 /usr/local/qt2 /usr/lib/qt /usr/local/qt"
-
-    for i in $QT_SEARCH; do
-        if test -f $i/include/qglobal.h -a x$QTDIR = x; then 
-            QTDIR=$i;
-	    QT_INC_DIR=$QTDIR/include
-	    QT_LIB_DIR=$QTDIR/lib
-	    break
-	fi
-    done
-else
-    QT_INC_DIR=$QTDIR/include
-    QT_LIB_DIR=$QTDIR/lib
-fi
-
 #Some distributions don't use QTDIR and use "standard" places like /usr/include/qt3 for headers and /usr/lib for libs
 #We try to detect qt in those standard places when we haven't found anything in QTDIR
 if test x$QT_INC_DIR = x; then 
