--- src/common/Configuration.h.orig	2014-11-20 13:47:24.000000000 -0200
+++ src/common/Configuration.h	2015-08-07 19:02:50.067954111 -0300
@@ -46,7 +46,7 @@ namespace SDDM {
         //      Name   Entries (but it's a regular class again)
         Section(Theme,
             Entry(ThemeDir,            QString,     _S(DATA_INSTALL_DIR "/themes"),             _S("Theme directory path"));
-            Entry(Current,             QString,     _S("maui"),                                 _S("Current theme name"));
+            Entry(Current,             QString,     _S("circles"),                                 _S("Current theme name"));
             Entry(FacesDir,            QString,     _S(DATA_INSTALL_DIR "/faces"),              _S("Face icon directory\n"
                                                                                                    "The files should be in username.face.icon format"));
             Entry(CursorTheme,         QString,     QString(),                                  _S("Cursor theme"));
