--- sddm-7a008602f5f0a4ed8586ce24012983458a687d4e/src/daemon/Display.cpp.savelast	2013-11-25 19:41:15.824359536 +0100
+++ sddm-7a008602f5f0a4ed8586ce24012983458a687d4e/src/daemon/Display.cpp	2013-11-25 19:41:58.793247508 +0100
@@ -77,6 +77,7 @@ namespace SDDM {
         // connect login result signals
         connect(this, SIGNAL(loginFailed(QLocalSocket*)), m_socketServer, SLOT(loginFailed(QLocalSocket*)));
         connect(this, SIGNAL(loginSucceeded(QLocalSocket*)), m_socketServer, SLOT(loginSucceeded(QLocalSocket*)));
+	connect(this, SIGNAL(loginSucceeded(QLocalSocket*)), this, SLOT(slotLoginSucceeded(QLocalSocket*)));
 
         // get auth dir
         QString authDir = daemonApp->configuration()->authDir();
