--- sddm-7a008602f5f0a4ed8586ce24012983458a687d4e/src/daemon/Authenticator.cpp.pam_end	2013-12-16 15:26:35.749298583 +0100
+++ sddm-7a008602f5f0a4ed8586ce24012983458a687d4e/src/daemon/Authenticator.cpp	2013-12-16 15:26:45.273274332 +0100
@@ -418,8 +418,9 @@ namespace SDDM {
         if (m_pam) {
             m_pam->result = pam_close_session(m_pam->handle, 0);
             m_pam->result = pam_setcred(m_pam->handle, PAM_DELETE_CRED);
             delete m_pam;
             m_pam = nullptr;
+            daemonApp->exit(0); // until PAM is fixed, we have to restart the whole service with audit
         }
 #endif
 
