diff --git a/src/greeter/SessionModel.cpp b/src/greeter/SessionModel.cpp
index 8c232ba..8269c4c 100644
--- a/src/greeter/SessionModel.cpp
+++ b/src/greeter/SessionModel.cpp
@@ -56,9 +56,6 @@ namespace SDDM {
         // set role names
         setRoleNames(roleNames);
 #endif
-        // add custom and failsafe session
-        d->sessions << SessionPtr { new Session {"custom", "Custom", "custom", "Custom Session"} };
-        d->sessions << SessionPtr { new Session {"failsafe", "Failsafe", "failsafe", "Failsafe Session"} };
         // read session files
         QDir dir(Configuration::instance()->sessionsDir());
         dir.setNameFilters(QStringList() << "*.desktop");
@@ -84,6 +81,10 @@ namespace SDDM {
             // close file
             inputFile.close();
         }
+        // add custom and failsafe session
+//         hidden for now --mbriza
+//         d->sessions << SessionPtr { new Session {"custom", "Custom", "custom", "Custom Session"} };
+        d->sessions << SessionPtr { new Session {"failsafe", "Failsafe", "failsafe", "Failsafe Session"} };
         // find out index of the last session
         for (int i = 0; i < d->sessions.size(); ++i) {
             if (d->sessions.at(i)->file == Configuration::instance()->lastSession())
