diff -ur dirac.old/libdirac_encoder/quant_chooser.cpp dirac-1.0.2/libdirac_encoder/quant_chooser.cpp
--- dirac.old/libdirac_encoder/quant_chooser.cpp	2009-01-21 05:22:05.000000000 +0000
+++ dirac-1.0.2/libdirac_encoder/quant_chooser.cpp	2010-07-03 17:40:12.000000000 +0000
@@ -340,7 +340,7 @@
     {
         for (int i=cblock.Xstart(); i<cblock.Xend(); ++i )
         {
-            if ( (std::abs(m_coeff_data[j][i])<<2) >= u_threshold )
+            if ( (std::abs(int(m_coeff_data[j][i]))<<2) >= u_threshold )
                 can_skip = false;
         }   
     }
@@ -355,7 +355,7 @@
     {
         for (int i=node.Xp() ; i<node.Xp()+node.Xl(); ++i)
         {    
-            val = std::max( val , std::abs(m_coeff_data[j][i]) );
+            val = std::max( val , std::abs(int(m_coeff_data[j][i])) );
         }// i
     }// j
 
