--- exo-0.5.4/python/Makefile.in.orig	2010-09-05 07:26:36.000000000 -0300
+++ exo-0.5.4/python/Makefile.in	2010-11-14 16:49:43.323791249 -0200
@@ -373,7 +373,7 @@
 
 defsdir = $(datadir)/pygtk/2.0/defs/exo-$(LIBEXO_VERSION_MAJOR).$(LIBEXO_VERSION_MINOR)
 defs_DATA = exo.defs
-pyexodir = $(pythondir)
+pyexodir = $(pyexecdir)
 pyexo_PYTHON = pyexo.py
 exomoduledir = $(pyexecdir)/exo-$(LIBEXO_VERSION_MAJOR).$(LIBEXO_VERSION_MINOR)/exo
 exomodule_PYTHON = __init__.py
