diff -up libqalculate-0.9.7/libqalculate/Calculator.cc.euroref-daily libqalculate-0.9.7/libqalculate/Calculator.cc
--- libqalculate-0.9.7/libqalculate/Calculator.cc.euroref-daily	2010-01-05 09:15:27.000000000 -0600
+++ libqalculate-0.9.7/libqalculate/Calculator.cc	2015-06-17 13:41:27.689297656 -0500
@@ -8144,6 +8144,9 @@ bool Calculator::loadExchangeRates() {
 	string homedir = getLocalDir();
 	filename = homedir;
 	filename += "eurofxref-daily.xml";
+        if(!g_file_test(filename.c_str(),G_FILE_TEST_EXISTS)) {
+                return false;
+        }
 	doc = xmlParseFile(filename.c_str());
 	if(doc == NULL) {
 		//fetchExchangeRates();
