diff -urNp autogen-5.9.4.orig/autoopts/mk-autoopts-pc.in autogen-5.9.4/autoopts/mk-autoopts-pc.in
--- autogen-5.9.4.orig/autoopts/mk-autoopts-pc.in	2007-12-24 02:33:16.000000000 +0530
+++ autogen-5.9.4/autoopts/mk-autoopts-pc.in	2008-02-24 21:21:13.000000000 +0530
@@ -18,7 +18,6 @@
  exec_prefix="@exec_prefix@"
       bindir="@bindir@"
       libdir="@libdir@"
-      ldopts="@AG_LDFLAGS@"
       exeext="@EXEEXT@"
      version="@AO_CURRENT@:@AO_REVISION@:@AO_AGE@"
       dotver="@AO_CURRENT@.@AO_REVISION@.@AO_AGE@"
