=== modified file 'agen5/tpProcess.c'
--- agen5/tpProcess.c	2010-04-21 18:55:14 +0000
+++ agen5/tpProcess.c	2010-04-21 18:55:48 +0000
@@ -318,7 +318,7 @@
     char const * pzOutFile = NULL;
 
     if (strcmp( pOutSpec->zSuffix, "null" ) == 0) {
-        static int const flags = FPF_NOUNLINK;
+        static int const flags = FPF_NOUNLINK | FPF_NOCHMOD;
     null_open:
         open_output_file(zDevNull, sizeof(zDevNull)-1, write_mode, flags);
         return;

