--- autoopts/autoopts-config.in.orig	2011-06-26 10:39:46.000000000 -0400
+++ autoopts/autoopts-config.in	2011-11-25 11:03:15.408677185 -0500
@@ -17,17 +17,17 @@
   includedir="@includedir@"
  exec_prefix="@exec_prefix@"
       bindir="@bindir@"
-      libdir="@libdir@"
-      ldopts="@AG_LDFLAGS@"
+      libdir=""
+      ldopts=""
       exeext="@EXEEXT@"
      version="@AO_CURRENT@:@AO_REVISION@:@AO_AGE@"
       dotver="@AO_CURRENT@.@AO_REVISION@.@AO_AGE@"
   pkgdatadir="${datadir}/${package}"
      autogen="${bindir}/autogen${exeext}"
-     ldflags="-L${libdir} -lopts"
+     ldflags="-lopts"
         libs="${ldflags}"
       libsrc="${pkgdatadir}/libopts-${dotver}.tar.gz"
- static_libs="${libdir}/libopts.a"
+ static_libs=""
       cflags="-I${includedir}"
 case "${libdir}" in
 /lib | /lib64 | /usr/lib | /usr/lib64 )
