diff -urNp xmlto-0.0.24-orig/configure.in xmlto-0.0.24/configure.in
--- xmlto-0.0.24-orig/configure.in	2011-07-14 12:50:47.000000000 +0200
+++ xmlto-0.0.24/configure.in	2011-11-29 12:53:50.614481739 +0100
@@ -109,6 +109,9 @@ AC_PATH_PROG([W3M], [w3m], [w3m])
 AC_ARG_VAR([TAIL], [Name and path of a tail binary that supports -n.])
 AC_PATH_PROG([TAIL], [tail], [tail])
 
+AC_ARG_VAR([GREP], [Name and path of a GNU grep binary.])
+AC_PATH_PROG([GREP], [grep], [grep])
+
 AC_ARG_VAR([GCP], [Name and path of a GNU cp binary (need at least -P)])
 AC_PATH_PROG([GCP], [cp], [cp])
 
Binary files xmlto-0.0.24-orig/.configure.in.swp and xmlto-0.0.24/.configure.in.swp differ
diff -urNp xmlto-0.0.24-orig/xmlto.in xmlto-0.0.24/xmlto.in
--- xmlto-0.0.24-orig/xmlto.in	2011-07-14 12:41:25.000000000 +0200
+++ xmlto-0.0.24/xmlto.in	2011-11-29 12:53:02.105195055 +0100
@@ -22,6 +22,7 @@ MKTEMP=@MKTEMP@ # See http://www.mktemp.
 BASH=@BASH@     # GNU bash, for running the format scripts
 GETOPT=@GETOPT@ # a getopt that supports --longoptions
 TAIL=@TAIL@     # a tail that supports -n (posix)
+GREP=@GREP@     # GNU grep, for searching patterns
 
 version () {
   echo "@PACKAGE@ version @VERSION@"
@@ -236,6 +237,14 @@ then
   echo >&2 "@PACKAGE@: Warning: fop not found or not executable."
   echo >&2 "@PACKAGE@: Using default backend..."
   USE_BACKEND=DEFAULT
+else
+  # we should enable fop.extensions for fop 0_17,0_18 and 0_20*,
+  # fop1.extensions for the rest
+  if [ x`$FOP_PATH -v 2>/dev/null | $GREP 0_[12]` = x ]; then
+    FOPVERSION="fop1"
+  else
+    FOPVERSION="fop"
+  fi
 fi
 if [ x"$USE_BACKEND" = xDBLATEX ] && \
    [ -z "`type -t $DBLATEX_PATH`" ]
@@ -385,7 +394,7 @@ if [ "$BACKEND_EXTENSIONS" -eq 1 ]
 then
   case "$USE_BACKEND" in
   FOP)
-      XSLTOPTS="$XSLTOPTS --param fop.extensions '1' --param fop1.extensions '1'" ;;
+      XSLTOPTS="$XSLTOPTS --param $FOPVERSION.extensions '1'" ;;
   DBLATEX)
       ;;
   DEFAULT)
