diff -urNp xmlto-0.0.25-orig/xmlto.in xmlto-0.0.25/xmlto.in
--- xmlto-0.0.25-orig/xmlto.in	2011-11-29 12:49:57.000000000 +0100
+++ xmlto-0.0.25/xmlto.in	2012-06-08 20:30:11.492008539 +0200
@@ -262,7 +262,7 @@ ZIP_PATH=@ZIP@
 
 # Process any options
 ARGS=$(${GETOPT} \
-	--longoptions=help,version,extensions,searchpath:,skip-validation,stringparam:,noclean,noautosize,with-fop,with-dblatex \
+	--longoptions=help,version,extensions,searchpath:,skip-validation,stringparam:,noclean,noautosize,noextensions,with-fop,with-dblatex \
 	-n xmlto -- x:m:o:p:v "$@")
 [ $? != 0 ] && { usage; exit 1; }
 eval set -- "$ARGS"
