diff -up spandsp-0.0.6/configure.brackets spandsp-0.0.6/configure
--- spandsp-0.0.6/configure.brackets	2011-01-22 01:49:54.000000000 -0600
+++ spandsp-0.0.6/configure	2011-12-27 15:29:07.281452488 -0600
@@ -28574,7 +28574,7 @@ fi
 echo "$as_me:$LINENO: result: `eval echo '${'$as_ac_RealFile'}'`" >&5
 echo "${ECHO_T}`eval echo '${'$as_ac_RealFile'}'`" >&6
 if test `eval echo '${'$as_ac_RealFile'}'` = yes; then
-  libdir='$(exec_prefix)/lib64'
+  libdir='${exec_prefix}/lib64'
 fi
 
         echo "$as_me:$LINENO: checking for /usr/X11R6/lib64" >&5
