--- src/gmameui-zip-utils.c.orig	2012-07-07 16:29:03.697747609 -0300
+++ src/gmameui-zip-utils.c	2012-07-07 16:29:34.652539322 -0300
@@ -91,7 +91,7 @@ read_pixbuf_from_zip_file (gchar *zipfil
 			buffer_data = (gchar *) g_malloc0 (filesize);
 			
 			/* Read the zipped pixbuf into memory */
-			archive_read_data_into_buffer (zipfile, buffer_data, filesize);
+			archive_read_data (zipfile, buffer_data, filesize);
 			
 			pixbuf = load_pixbuf_data (buffer_data, filesize);
 									 
@@ -107,4 +107,4 @@ read_pixbuf_from_zip_file (gchar *zipfil
 	
 	return pixbuf;
 	
-}
\ Falta o caracter nova linha no final do arquivo
+}
