--- configure.py.dso	2010-12-23 08:25:22.000000000 -0200
+++ configure.py	2011-01-01 21:25:52.649362720 -0200
@@ -356,14 +356,15 @@ class ConfigurePyQt4:
             ass_libs = ["ssleay32", "libeay32"]
         else:
             ass_lib_dirs = None
-            ass_libs = None
+            ass_libs = ["QtCore"]
 
         # Note that the order in which we check is important for the
         # consolidated module - a module's dependencies must be checked first.
         pyqt_modules.append("QtCore")
 
         check_module("QtGui", "qwidget.h", "new QWidget()")
-        check_module("QtHelp", "qhelpengine.h", "new QHelpEngine(\"foo\")")
+        check_module("QtHelp", "qhelpengine.h", "new QHelpEngine(\"foo\")",
+                extra_libs=["QtCore"])
         check_module("QtMultimedia", "QAudioDeviceInfo",
                 "new QAudioDeviceInfo()")
         check_module("QtNetwork", "qhostaddress.h", "new QHostAddress()")
