diff -up PyQt-x11-gpl-4.7.2/configure.py.fix-implicit-linking PyQt-x11-gpl-4.7.2/configure.py
--- PyQt-x11-gpl-4.7.2/configure.py.fix-implicit-linking	2010-03-18 07:08:13.000000000 -0500
+++ PyQt-x11-gpl-4.7.2/configure.py	2010-03-18 07:09:46.157734414 -0500
@@ -1173,7 +1173,7 @@ def compile_qt_program(name, mname, extr
     opengl = (mname == "QtOpenGL")
 
     qt = [mname]
-    if mname in ("QtOpenGL", "QtWebKit"):
+    if mname in ("QtOpenGL", "QtWebKit", "QtAssistant", "QtHelp"):
         qt.append("QtCore")
 
     makefile = sipconfig.ProgramMakefile(sipcfg, console=1, qt=qt, warnings=0,
