diff -up PyQt-x11-gpl-4.6.2/pyrcc/rcc.cpp.orig PyQt-x11-gpl-4.6.2/pyrcc/rcc.cpp
--- PyQt-x11-gpl-4.6.2/pyrcc/rcc.cpp.orig	2010-02-23 12:33:44.000000000 +0100
+++ PyQt-x11-gpl-4.6.2/pyrcc/rcc.cpp	2010-02-23 12:45:03.000000000 +0100
@@ -399,8 +399,7 @@ RCCResourceLibrary::writeHeader(FILE *ou
     fprintf(out, "# -*- coding: utf-8 -*-\n\n");
     fprintf(out, "# Resource object code\n");
     fprintf(out, "#\n");
-    fprintf(out, "# Created: %s\n", QDateTime::currentDateTime().toString().toLatin1().constData());
-    fprintf(out, "#      by: The Resource Compiler for PyQt (Qt v%s)\n", QT_VERSION_STR);
+    fprintf(out, "# Created by: The Resource Compiler for PyQt (Qt v%s)\n", QT_VERSION_STR);
     fprintf(out, "#\n");
     fprintf(out, "# WARNING! All changes made in this file will be lost!\n");
     fprintf(out, "\n");
