diff -up PyQt-x11-gpl-4.8.3/configure.py.python_libdir PyQt-x11-gpl-4.8.3/configure.py
--- PyQt-x11-gpl-4.8.3/configure.py.python_libdir	2011-01-23 04:08:20.000000000 -0600
+++ PyQt-x11-gpl-4.8.3/configure.py	2011-01-24 10:10:59.803217498 -0600
@@ -940,13 +940,15 @@ include(%s)
                     # We need to work out how to specify the right framework
                     # version.
                     link = "-framework Python"
-                elif ("--enable-shared" in ducfg.get("CONFIG_ARGS", "") and
-                      glob.glob("%s/lib/libpython%d.%d*" % (ducfg["exec_prefix"], py_major, py_minor))):
-                    lib_dir_flag = quote("-L%s/lib" % ducfg["exec_prefix"])
+                elif ("--enable-shared" in ducfg.get("CONFIG_ARGS", "")):
+                    if (glob.glob("%s/lib/libpython%d.%d*" % (ducfg["exec_prefix"], py_major, py_minor))):
+                        lib_dir_flag = quote("-L%s/lib" % ducfg["exec_prefix"])
+                    elif (glob.glob("%s/libpython%d.%d*" % (ducfg["LIBDIR"], py_major, py_minor))):
+                        lib_dir_flag = quote("-L%s" % ducfg["LIBDIR"])
+                    else:
+                        sipconfig.inform("Qt Designer plugin disabled because Python library is static")
+                        opts.designer_plugin = False
                     link = "%s -lpython%d.%d%s" % (lib_dir_flag, py_major, py_minor, abi)
-                else:
-                    sipconfig.inform("Qt Designer plugin disabled because Python library is static")
-                    opts.designer_plugin = False
 
                 pysh_lib = ducfg["LDLIBRARY"]
 
