--- PyQt-x11-gpl-4.6.2/configure.py.orig	2010-01-08 23:39:46.000000000 +0100
+++ PyQt-x11-gpl-4.6.2/configure.py	2010-01-08 23:45:18.000000000 +0100
@@ -451,7 +451,7 @@
             generate_code("QtXmlPatterns")
 
         if "phonon" in pyqt_modules:
-            generate_code("phonon")
+            generate_code("phonon", extra_include_dirs=["/usr/include/phonon"])
 
         if "QtAssistant" in pyqt_modules:
             generate_code("QtAssistant")
