--- gtkrcfile.cpp.mine	2010-08-11 14:12:48.937435422 -0300
+++ gtkrcfile.cpp	2010-08-11 14:12:54.202187891 -0300
@@ -145,6 +145,8 @@
 	stream << "gtk-theme-name=\"" << m_themeName << "\"\n";
 	stream << "gtk-font-name=\"" << fontName << "\"\n";
 	stream << "gtk-cursor-theme-name = \"" << m_cursorName << "\"\n";
+	stream << "\n";
+	stream << "include \"" << QDir::homePath() << "/.gtkrc-2.0.mine\"\n";
 	
 	QByteArray gtkrc = getenv("GTK2_RC_FILES");
 	QStringList list = QFile::decodeName(gtkrc).split( ':');
