diff -up dansguardian-2.12.0.0/src/ConnectionHandler.cpp.gcc46 dansguardian-2.12.0.0/src/ConnectionHandler.cpp
--- dansguardian-2.12.0.0/src/ConnectionHandler.cpp.gcc46	2011-08-23 09:12:47.000000000 -0300
+++ dansguardian-2.12.0.0/src/ConnectionHandler.cpp	2011-09-11 18:45:30.262738740 -0300
@@ -38,6 +38,7 @@
 
 #ifdef ENABLE_ORIG_IP
 #include <linux/types.h>
+#include <limits.h>
 #include <linux/netfilter_ipv4.h>
 #endif
 
diff -up dansguardian-2.12.0.0/src/authplugins/ntlm.cpp.gcc46 dansguardian-2.12.0.0/src/authplugins/ntlm.cpp
--- dansguardian-2.12.0.0/src/authplugins/ntlm.cpp.gcc46	2011-09-11 18:46:38.513221674 -0300
+++ dansguardian-2.12.0.0/src/authplugins/ntlm.cpp	2011-09-11 18:50:15.275121405 -0300
@@ -14,6 +14,7 @@
 #include "../FDTunnel.hpp"
 #include "../OptionContainer.hpp"
 
+#include <cstddef>
 #include <syslog.h>
 
 #include <iconv.h>
diff -up dansguardian-2.12.0.0/src/contentscanners/commandlinescan.cpp.gcc46 dansguardian-2.12.0.0/src/contentscanners/commandlinescan.cpp
--- dansguardian-2.12.0.0/src/contentscanners/commandlinescan.cpp.gcc46	2011-08-23 09:12:47.000000000 -0300
+++ dansguardian-2.12.0.0/src/contentscanners/commandlinescan.cpp	2011-09-11 18:45:30.263738527 -0300
@@ -17,6 +17,7 @@
 #include "../OptionContainer.hpp"
 #include "../RegExp.hpp"
 
+#include <cstdio>
 #include <syslog.h>
 #include <sys/time.h>
 #include <sys/types.h>
diff -up dansguardian-2.12.0.0/src/contentscanners/icapscan.cpp.gcc46 dansguardian-2.12.0.0/src/contentscanners/icapscan.cpp
--- dansguardian-2.12.0.0/src/contentscanners/icapscan.cpp.gcc46	2011-08-23 09:12:47.000000000 -0300
+++ dansguardian-2.12.0.0/src/contentscanners/icapscan.cpp	2011-09-11 18:45:30.263738527 -0300
@@ -15,6 +15,7 @@
 #include "../ContentScanner.hpp"
 #include "../OptionContainer.hpp"
 
+#include <cstdio>
 #include <syslog.h>
 #include <sys/time.h>
 #include <sys/types.h>
diff -up dansguardian-2.12.0.0/src/downloadmanagers/fancy.cpp.gcc46 dansguardian-2.12.0.0/src/downloadmanagers/fancy.cpp
--- dansguardian-2.12.0.0/src/downloadmanagers/fancy.cpp.gcc46	2011-08-23 09:12:46.000000000 -0300
+++ dansguardian-2.12.0.0/src/downloadmanagers/fancy.cpp	2011-09-11 18:45:30.263738527 -0300
@@ -13,6 +13,7 @@
 #include "../HTMLTemplate.hpp"
 #include "../ConnectionHandler.hpp"
 
+#include <cstdio>
 #include <syslog.h>
 #include <sys/time.h>
 #include <sys/types.h>
