--- configure.in~	2010-03-19 10:14:15.000000000 -0500
+++ configure.in	2010-03-19 10:16:53.808478856 -0500
@@ -16,7 +16,7 @@
 dnl check SDL version                                                           
 AM_PATH_SDL(1.2.0,, AC_MSG_ERROR(lib SDL is needed))                            
 CFLAGS="$CFLAGS $SDL_CFLAGS"                                                    
-LIBS="$LIBS $SDL_LIBS"
+LIBS="$LIBS $SDL_LIBS -lm"
 
 dnl check sound
 audio_flag="-DAUDIO_ENABLED"
