--- src/main.c~	2005-02-14 12:50:17.000000000 -0600
+++ src/main.c	2010-03-19 16:54:25.253605960 -0500
@@ -264,7 +264,7 @@
 	SDL_WriteText( ft_help, dest, x,target_y+50, 
 			"If a jeep makes it safe through the screen you'll" );
 	SDL_WriteText( ft_help, dest, x,target_y+70, 
-			"loose 10 pts. For a tank you'll loose 25 pts. For" );
+			"lose 10 pts. For a tank you'll lose 25 pts. For" );
 	SDL_WriteText( ft_help, dest, x,target_y+90,
 			"soldiers there is no such penalty." );
 
