diff --git a/src/unicapgtk_property_dialog.c b/src/unicapgtk_property_dialog.c
index 0adf8a1..83288ef 100644
--- a/src/unicapgtk_property_dialog.c
+++ b/src/unicapgtk_property_dialog.c
@@ -308,7 +308,7 @@ static gboolean property_update_timeout_cb( UnicapgtkPropertyDialog *ugtk )
 
 static void update_lock_destroy_cb( gpointer _id )
 {
-   guint id = ( guint ) id;
+   guint id = ( guint ) _id;
    
    if( id )
    {
