--- qt-everywhere-opensource-src-4.7.1/src/gui/kernel/qapplication.cpp.orig	2010-11-05 23:55:17.000000000 -0200
+++ qt-everywhere-opensource-src-4.7.1/src/gui/kernel/qapplication.cpp	2010-11-09 20:11:53.656684350 -0200
@@ -801,9 +801,9 @@
     qt_is_gui_used = (qt_appType != QApplication::Tty);
     process_cmdline();
     // the environment variable has the lowest precedence of runtime graphicssystem switches
-    if (graphics_system_name.isEmpty())
+    if (graphics_system_name.isEmpty()) {
         graphics_system_name = QString::fromLocal8Bit(qgetenv("QT_GRAPHICSSYSTEM"));
-
+    }
 #if defined(Q_WS_X11) && !defined(QT_NO_EGL)
     if (graphics_system_name.isEmpty()) {
         bool linksWithMeeGoTouch = dl_iterate_phdr(qt_matchLibraryName, const_cast<char *>("libmeegotouchcore"));
