diff -up qt-everywhere-opensource-src-4.7.1/tools/assistant/tools/assistant/assistant.pro.webkit qt-everywhere-opensource-src-4.7.1/tools/assistant/tools/assistant/assistant.pro
--- qt-everywhere-opensource-src-4.7.1/tools/assistant/tools/assistant/assistant.pro.webkit	2010-11-06 02:55:11.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.1/tools/assistant/tools/assistant/assistant.pro	2010-12-06 18:39:53.667993686 +0100
@@ -2,7 +2,7 @@ include(../../../shared/fontpanel/fontpa
 TEMPLATE = app
 LANGUAGE = C++
 TARGET = assistant
-contains(QT_CONFIG, webkit):QT += webkit
+DEFINES += QT_NO_WEBKIT
 CONFIG += qt \
     warn_on \
     help
@@ -40,12 +40,9 @@ HEADERS += aboutdialog.h \
     topicchooser.h \
     tracer.h \
     xbelsupport.h \
-    ../shared/collectionconfiguration.h
-contains(QT_CONFIG, webkit) {
-    HEADERS += helpviewer_qwv.h
-} else {
-   HEADERS += helpviewer_qtb.h
- }
+    ../shared/collectionconfiguration.h \
+    helpviewer_qtb.h
+
 win32:HEADERS += remotecontrol_win.h
 
 SOURCES += aboutdialog.cpp \
@@ -72,12 +69,8 @@ SOURCES += aboutdialog.cpp \
     searchwidget.cpp \
     topicchooser.cpp \
     xbelsupport.cpp \
-    ../shared/collectionconfiguration.cpp
- contains(QT_CONFIG, webkit) {
-    SOURCES += helpviewer_qwv.cpp
-} else {
-    SOURCES += helpviewer_qtb.cpp
-}
+    ../shared/collectionconfiguration.cpp \
+    helpviewer_qtb.cpp
 
 FORMS += bookmarkdialog.ui \
     bookmarkmanagerwidget.ui \
