diff -up qt-everywhere-opensource-src-4.8.3/src/dbus/qdbusconnection.cpp.qdbusconnection_nodebug qt-everywhere-opensource-src-4.8.3/src/dbus/qdbusconnection.cpp
--- qt-everywhere-opensource-src-4.8.3/src/dbus/qdbusconnection.cpp.qdbusconnection_nodebug	2012-09-10 22:36:52.000000000 -0300
+++ qt-everywhere-opensource-src-4.8.3/src/dbus/qdbusconnection.cpp	2012-09-13 11:19:29.243460667 -0300
@@ -1109,8 +1109,10 @@ public:
         // make sure this connection is running on the main thread
         QCoreApplication *instance = QCoreApplication::instance();
         if (!instance) {
+#ifndef QT_NO_DEBUG
             qWarning("QDBusConnection: %s D-Bus connection created before QCoreApplication. Application may misbehave.",
                      type == SessionBus ? "session" : type == SystemBus ? "system" : "generic");
+#endif
         } else if (QDBusConnectionPrivate::d(*this)) {
             QDBusConnectionPrivate::d(*this)->moveToThread(instance->thread());
         }
