diff -up qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.cpp.me qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.cpp
--- qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.cpp.me	2012-05-16 14:45:30.542709580 +0200
+++ qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.cpp	2012-05-16 14:49:02.266017265 +0200
@@ -61,8 +61,8 @@ const char *QElfParser::parseSectionHead
          + sizeof(qelfaddr_t);  // sh_addr
     sh->offset = read<qelfoff_t>(data);
     data += sizeof(qelfoff_t);  // sh_offset
-    sh->size = read<qelfword_t>(data);
-    data += sizeof(qelfword_t); // sh_size
+    sh->size = read<qelfoff_t>(data);
+    data += sizeof(qelfoff_t); // sh_size
     return data;
 }
 
diff -up qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.h.me qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.h
--- qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.h.me	2012-05-16 14:45:41.006724793 +0200
+++ qt-everywhere-opensource-src-4.8.1/src/corelib/plugin/qelfparser_p.h	2012-05-16 14:47:49.017910811 +0200
@@ -80,7 +80,7 @@ public:
         qelfword_t name;
         qelfword_t type;
         qelfoff_t  offset;
-        qelfword_t size;
+        qelfoff_t  size;
     };
 
     int m_endian;
