diff -up qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/embed.cpp.uic_multilib qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/embed.cpp
--- qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/embed.cpp.uic_multilib	2010-05-03 19:43:25.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/embed.cpp	2010-06-25 14:11:21.815474310 -0500
@@ -152,7 +152,7 @@ void Ui3Reader::embed(const char *projec
     for ( it = images.begin(); it != images.end(); ++it )
         out << "**      " << *it << "\n";
     out << "**\n";
-    out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
+    out << "** Created: " << "\n";
     out << "**      by: The User Interface Compiler for Qt version " << QT_VERSION_STR << "\n";
     out << "**\n";
     out << "** WARNING! All changes made in this file will be lost!\n";
diff -up qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/uic.cpp.uic_multilib qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/uic.cpp
--- qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/uic.cpp.uic_multilib	2010-05-03 19:43:25.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic3/uic.cpp	2010-06-25 14:11:32.910460623 -0500
@@ -146,7 +146,7 @@ void Uic::writeCopyrightHeader(DomUI *ui
         out << "/********************************************************************************\n";
         out << "** Form generated from reading UI file '" << QFileInfo(opt.inputFile).fileName() << "'\n";
         out << "**\n";
-        out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
+        out << "** Created: " << "\n";
         out << "**      " << QString::fromLatin1("by: Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
         out << "**\n";
         out << "** WARNING! All changes made in this file will be lost when recompiling UI file!\n";
diff -up qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic/uic.cpp.uic_multilib qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic/uic.cpp
--- qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic/uic.cpp.uic_multilib	2010-05-03 19:43:25.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta1/src/tools/uic/uic.cpp	2010-06-25 14:10:45.360711498 -0500
@@ -139,7 +139,7 @@ void Uic::writeCopyrightHeader(DomUI *ui
         out << "/********************************************************************************\n";
         out << "** Form generated from reading UI file '" << QFileInfo(opt.inputFile).fileName() << "'\n";
         out << "**\n";
-        out << "** Created: " << QDateTime::currentDateTime().toString() << "\n";
+        out << "** Created: " << "\n";
         out << "**      " << QString::fromLatin1("by: Qt User Interface Compiler version %1\n").arg(QLatin1String(QT_VERSION_STR));
         out << "**\n";
         out << "** WARNING! All changes made in this file will be lost when recompiling UI file!\n";
