diff --git a/src/gui/itemviews/qtreeview.cpp b/src/gui/itemviews/qtreeview.cpp
index 406e1f6..4691b17 100644
--- a/src/gui/itemviews/qtreeview.cpp
+++ b/src/gui/itemviews/qtreeview.cpp
@@ -3223,7 +3223,7 @@ int QTreeViewPrivate::itemHeight(int item) const
         return defaultItemHeight;
     if (viewItems.isEmpty())
         return 0;
-    const QModelIndex &index = viewItems.at(item).index;
+    QModelIndex index = viewItems.at(item).index;
     int height = viewItems.at(item).height;
     if (height <= 0 && index.isValid()) {
         height = q_func()->indexRowSizeHint(index);
