diff -up qt-everywhere-opensource-src-4.8.3/mkspecs/features/device_config.prf.qmake_fix qt-everywhere-opensource-src-4.8.3/mkspecs/features/device_config.prf
--- qt-everywhere-opensource-src-4.8.3/mkspecs/features/device_config.prf.qmake_fix	2012-09-10 20:36:35.000000000 -0500
+++ qt-everywhere-opensource-src-4.8.3/mkspecs/features/device_config.prf	2012-09-28 09:06:48.732819678 -0500
@@ -1,6 +1,6 @@
 # Load generated qdevice.pri
 isEmpty(QT_BUILD_TREE): QT_BUILD_TREE = $$fromfile($$_QMAKE_CACHE_, QT_BUILD_TREE)
-isEmpty(QT_BUILD_TREE): QT_BUILD_TREE = . # where to get?
+isEmpty(QT_BUILD_TREE): QT_BUILD_TREE = $$[QT_INSTALL_DATA]
 
 DEVICE_PRI = $$QT_BUILD_TREE/mkspecs/qdevice.pri
 
