diff -up qt-everywhere-opensource-src-4.7.1/src/3rdparty/harfbuzz/src/harfbuzz-indic.cpp.ml_IN-528303 qt-everywhere-opensource-src-4.7.1/src/3rdparty/harfbuzz/src/harfbuzz-indic.cpp
--- qt-everywhere-opensource-src-4.7.1/src/3rdparty/harfbuzz/src/harfbuzz-indic.cpp.ml_IN-528303	2010-11-22 15:14:41.000000000 +0100
+++ qt-everywhere-opensource-src-4.7.1/src/3rdparty/harfbuzz/src/harfbuzz-indic.cpp	2010-11-22 15:13:03.000000000 +0100
@@ -833,7 +833,7 @@ static const unsigned char indicPosition
     None, None, None, None,
     None, None, None, Post,
 
-    Post, None, Below, None,
+    Pre, None, Below, None,
     None, Post, None, None,
     None, None, None, None,
     None, None, Post, Post,
