diff -up qt-everywhere-opensource-src-4.7.0-beta2/src/3rdparty/phonon/gstreamer/gstreamer.desktop.servicesfile qt-everywhere-opensource-src-4.7.0-beta2/src/3rdparty/phonon/gstreamer/gstreamer.desktop
--- qt-everywhere-opensource-src-4.7.0-beta2/src/3rdparty/phonon/gstreamer/gstreamer.desktop.servicesfile	2010-06-29 20:53:12.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta2/src/3rdparty/phonon/gstreamer/gstreamer.desktop	2010-07-08 08:25:50.419855085 -0500
@@ -2,7 +2,7 @@
 Type=Service
 X-KDE-ServiceTypes=PhononBackend
 MimeType=application/x-annodex;video/quicktime;video/x-quicktime;audio/x-m4a;application/x-quicktimeplayer;video/mkv;video/msvideo;video/x-msvideo;video/x-flic;audio/x-aiff;audio/aiff;audio/x-pn-aiff;audio/x-realaudio;audio/basic;audio/x-basic;audio/x-pn-au;audio/x-8svx;audio/8svx;audio/x-16sv;audio/168sv;image/x-ilbm;image/ilbm;video/x-anim;video/anim;image/png;image/x-png;video/mng;video/x-mng;audio/x-ogg;audio/x-speex+ogg;application/ogg;application/ogg;audio/vnd.rn-realaudio;audio/x-pn-realaudio-plugin;audio/x-real-audio;application/vnd.rn-realmedia;video/mpeg;video/x-mpeg;audio/x-wav;audio/wav;audio/x-pn-wav;audio/x-pn-windows-acm;audio/mpeg2;audio/x-mpeg2;audio/mpeg3;audio/x-mpeg3;audio/mpeg;audio/x-mpeg;x-mpegurl;audio/x-mpegurl;audio/mp3;audio/mpeg;
-X-KDE-Library=phonon_gstreamer
+X-KDE-Library=libphonon_gstreamer
 X-KDE-PhononBackendInfo-InterfaceVersion=1
 X-KDE-PhononBackendInfo-Version=0.1
 X-KDE-PhononBackendInfo-Website=http://gstreamer.freedesktop.org/
diff -up qt-everywhere-opensource-src-4.7.0-beta2/src/phonon/phonon.pro.servicesfile qt-everywhere-opensource-src-4.7.0-beta2/src/phonon/phonon.pro
--- qt-everywhere-opensource-src-4.7.0-beta2/src/phonon/phonon.pro.servicesfile	2010-06-29 20:53:13.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta2/src/phonon/phonon.pro	2010-07-08 08:26:58.307612257 -0500
@@ -122,6 +122,11 @@ contains(QT_CONFIG, dbus) {
        DEFINES += QT_NO_DBUS
 }
 
+# install dbus service file
+sources.files = $$PHONON_DIR/org.kde.Phonon.AudioOutput.xml
+sources.path = /usr/share/dbus-1/interfaces
+INSTALLS += sources
+
 contains(QT_CONFIG, reduce_exports): CONFIG += hide_symbols
 
 unix:!isEmpty(QT_CFLAGS_PULSEAUDIO) {
diff -up qt-everywhere-opensource-src-4.7.0-beta2/src/plugins/phonon/gstreamer/gstreamer.pro.servicesfile qt-everywhere-opensource-src-4.7.0-beta2/src/plugins/phonon/gstreamer/gstreamer.pro
--- qt-everywhere-opensource-src-4.7.0-beta2/src/plugins/phonon/gstreamer/gstreamer.pro.servicesfile	2010-06-29 20:53:14.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.0-beta2/src/plugins/phonon/gstreamer/gstreamer.pro	2010-07-08 08:25:50.387854526 -0500
@@ -64,6 +64,8 @@ SOURCES += $$PHONON_GSTREAMER_DIR/abstra
 }
 
 target.path = $$[QT_INSTALL_PLUGINS]/phonon_backend
-INSTALLS += target
+services.files = $$PHONON_GSTREAMER_DIR/gstreamer.desktop
+services.path = /usr/share/kde4/services/phononbackends
+INSTALLS += target services
 
 include(../../qpluginbase.pri)
