diff -up qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp.glib_eventloop_nullcheck qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp
--- qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp.glib_eventloop_nullcheck	2010-06-01 21:03:15.000000000 -0500
+++ qt-everywhere-opensource-src-4.6.3/src/gui/kernel/qguieventdispatcher_glib.cpp	2010-06-29 14:58:12.299073784 -0500
@@ -76,7 +76,7 @@ static gboolean x11EventSourcePrepare(GS
     GX11EventSource *source = reinterpret_cast<GX11EventSource *>(s);
     return (XEventsQueued(X11->display, QueuedAfterFlush)
             || (!(source->flags & QEventLoop::ExcludeUserInputEvents)
-                && !source->d->queuedUserInputEvents.isEmpty()));
+                && source->d && !source->d->queuedUserInputEvents.isEmpty()));
 }
 
 static gboolean x11EventSourceCheck(GSource *s)
@@ -84,7 +84,7 @@ static gboolean x11EventSourceCheck(GSou
     GX11EventSource *source = reinterpret_cast<GX11EventSource *>(s);
     return (XEventsQueued(X11->display, QueuedAfterFlush)
             || (!(source->flags & QEventLoop::ExcludeUserInputEvents)
-                && !source->d->queuedUserInputEvents.isEmpty()));
+                && source->d && !source->d->queuedUserInputEvents.isEmpty()));
 }
 
 static gboolean x11EventSourceDispatch(GSource *s, GSourceFunc callback, gpointer user_data)
