diff -up qt-everywhere-opensource-src-4.7.4/src/declarative/qml/qml.pri.qml_no_strict_aliasing qt-everywhere-opensource-src-4.7.4/src/declarative/qml/qml.pri
--- qt-everywhere-opensource-src-4.7.4/src/declarative/qml/qml.pri.qml_no_strict_aliasing	2011-09-12 01:49:29.000000000 -0500
+++ qt-everywhere-opensource-src-4.7.4/src/declarative/qml/qml.pri	2011-11-03 13:04:50.377884962 -0500
@@ -1,3 +1,6 @@
+
+*-g++*: QMAKE_CXXFLAGS += -fno-strict-aliasing
+
 INCLUDEPATH += $$PWD
 SOURCES += \
     $$PWD/qdeclarativeparser.cpp \
