diff -ur qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h qt-everywhere-opensource-src-4.8.0-qtwebkit-glib231/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h
--- qt-everywhere-opensource-src-4.8.0/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h	2011-10-04 05:44:31.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.0-qtwebkit-glib231/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GTypedefs.h	2011-10-28 02:12:16.000000000 +0200
@@ -52,7 +52,7 @@
 typedef struct _GHashTable GHashTable;
 typedef struct _GInputStream GInputStream;
 typedef struct _GList GList;
-typedef struct _GMutex GMutex;
+typedef union _GMutex GMutex;
 typedef struct _GPatternSpec GPatternSpec;
 typedef struct _GPollableOutputStream GPollableOutputStream;
 typedef struct _GSocketClient GSocketClient;
