diff -up qt-everywhere-opensource-src-4.8.6/configure.qt_build_key qt-everywhere-opensource-src-4.8.6/configure
--- qt-everywhere-opensource-src-4.8.6/configure.qt_build_key	2015-02-16 08:00:01.698531648 -0600
+++ qt-everywhere-opensource-src-4.8.6/configure	2015-02-16 08:06:43.198794608 -0600
@@ -8044,7 +8044,7 @@ g++*)
     3.*)
         COMPILER_VERSION="3.*"
         ;;
-    4.*)
+    5.*|4.*)
         COMPILER_VERSION="4"
         ;;
     *)
