diff --git a/src/create.c b/src/create.c
index 43b5a4c..f98cbb5 100644
--- a/src/create.c
+++ b/src/create.c
@@ -1798,7 +1798,8 @@ dump_file0 (struct tar_stat_info *st, char const *name, char const *p)
 	    }
 	  else if (atime_preserve_option == replace_atime_preserve
 		   && fd && (is_dir || original_size != 0)
-		   && set_file_atime (fd, parentfd, name, st->atime) != 0)
+		   && set_file_atime (fd, parentfd, name, st->atime) != 0
+		   && errno != EROFS )
 	    utime_error (p);
 	}
 
