diff --git a/src/incremen.c b/src/incremen.c
index dc880cd..3da1349 100644
--- a/src/incremen.c
+++ b/src/incremen.c
@@ -894,8 +894,11 @@ append_incremental_renames (struct directory *dir)
   if (obstack_object_size (&stk) != size)
     {
       obstack_1grow (&stk, 0);
-      dumpdir_free (dir->dump);
-      dir->dump = dumpdir_create (obstack_finish (&stk));
+      if (dir)
+	{
+	  dumpdir_free (dir->dump);
+	  dir->dump = dumpdir_create (obstack_finish (&stk));
+	}
     }
   obstack_free (&stk, NULL);
 }
