--- xscreensaver-5.17/hacks/glx/Makefile.in.newxpm	2012-06-23 13:23:12.000000000 +0900
+++ xscreensaver-5.17/hacks/glx/Makefile.in	2012-06-23 13:35:52.000000000 +0900
@@ -1832,7 +1832,9 @@
 lament.o: ../../config.h
 lament.o: $(HACK_SRC)/fps.h
 lament.o: $(srcdir)/gltrackball.h
-lament.o: $(HACK_SRC)/images/lament.xpm
+#lament.o: $(HACK_SRC)/images/lament.xpm
+#lament.o: $(HACK_SRC)/images/lament128.xpm
+lament.o: $(HACK_SRC)/images/lament512.xpm
 lament.o: $(srcdir)/jwzgles.h
 lament.o: $(srcdir)/normals.h
 lament.o: $(srcdir)/rotator.h
