--- xscreensaver-5.11/hacks/screenhack.c.bz587537	2010-04-12 13:48:58.000000000 +0900
+++ xscreensaver-5.11/hacks/screenhack.c	2010-05-01 01:08:46.000000000 +0900
@@ -453,7 +453,8 @@
                (! (event.xany.window == window
                    ? ft->event_cb (dpy, window, closure, &event)
 #ifdef DEBUG_PAIR
-                   : event.xany.window == window2
+                   : (window2 && closure2 && 
+                        (event.xany.window == window2))
                    ? ft->event_cb (dpy, window2, closure2, &event)
 #endif
                    : 0)))
