--- plugins/ntfy/ntfy.c.orig	2010-10-18 13:18:25.000000000 -0200
+++ plugins/ntfy/ntfy.c	2011-02-11 23:26:42.878637354 -0200
@@ -163,7 +163,7 @@ ntfy_enable(EinaNtfy *self, GError **err
 		notify_init("Eina");
 
 	self->enabled = TRUE;
-	self->ntfy = notify_notification_new(N_("Now playing"), NULL, NULL, NULL);
+	self->ntfy = notify_notification_new(N_("Now playing"), NULL, NULL);
 
 	g_signal_connect_swapped(EINA_OBJ_GET_LOMO(self), "play",     (GCallback) ntfy_sync, self);
 	g_signal_connect_swapped(EINA_OBJ_GET_LOMO(self), "change",   (GCallback) ntfy_reset, self);
@@ -266,7 +266,6 @@ vogon_enable(EinaNtfy *self)
 
 	// Set state
 	gtk_toggle_action_set_active(GTK_TOGGLE_ACTION(gtk_ui_manager_get_action(ui_mng, ACTION_PATH)), self->enabled);
-	notify_notification_attach_to_status_icon(self->ntfy, self->status_icon);
 
 	return TRUE;
 }
@@ -277,8 +276,6 @@ vogon_disable(EinaNtfy *self)
 	EinaVogon *vogon = EINA_OBJ_GET_VOGON(self);
 	g_return_val_if_fail(vogon != NULL, FALSE);
 
-	notify_notification_attach_to_status_icon(self->ntfy, NULL);
-
 	GtkUIManager *ui_mng = eina_vogon_get_ui_manager(vogon);
 	g_return_val_if_fail(ui_mng != NULL, FALSE);
 
