--- src/mkvtoolnix-gui/main_window/main_window.cpp.orig	2015-07-22 19:09:57.386698148 -0300
+++ src/mkvtoolnix-gui/main_window/main_window.cpp	2015-07-22 19:10:01.289869184 -0300
@@ -118,7 +118,7 @@ MainWindow::setupMenu() {
 #if defined(HAVE_CURL_EASY_H)
   connect(ui->actionHelpCheckForUpdates,       &QAction::triggered,             this, &MainWindow::checkForUpdates);
 #else
-  ui->actionGUICheckForUpdates->setVisible(false);
+  ui->actionHelpCheckForUpdates->setVisible(false);
 #endif  // HAVE_CURL_EASY_H
 }
 
