--- gawk-3.1.7/awkgram.y.orig	2010-06-25 10:08:11.000000000 +0200
+++ gawk-3.1.7/awkgram.y	2010-06-25 10:10:34.000000000 +0200
@@ -164,7 +164,7 @@
 %nonassoc LEX_IN
 %left FUNC_CALL LEX_BUILTIN LEX_LENGTH
 %nonassoc ','
-%nonassoc MATCHOP
+%left MATCHOP
 %nonassoc RELOP '<' '>' IO_IN IO_OUT
 %left CONCAT_OP
 %left YSTRING YNUMBER
@@ -702,8 +702,8 @@
 	 */
 print_expression_list
 	: opt_expression_list
-	| '(' exp comma expression_list r_paren
-		{ $$ = node($2, Node_expression_list, $4); }
+	| '(' expression_list r_paren
+		{ $$ = $2; }
 	;
 
 output_redir
