diff -r 2a38df26e009 Lib/distutils/command/bdist_rpm.py
--- a/Lib/distutils/command/bdist_rpm.py	Tue Sep 03 11:39:06 2013 -0500
+++ b/Lib/distutils/command/bdist_rpm.py	Wed Sep 04 19:49:37 2013 +0200
@@ -14,6 +14,7 @@
 from distutils.file_util import write_file
 from distutils.errors import (DistutilsOptionError, DistutilsPlatformError,
                               DistutilsFileError, DistutilsExecError)
+from distutils.sysconfig import get_python_version
 from distutils import log
 
 class bdist_rpm (Command):
