diff -up Python-2.7.3/Modules/_testcapimodule.c.fix-test_structmember-on-64bit-bigendian Python-2.7.3/Modules/_testcapimodule.c
--- Python-2.7.3/Modules/_testcapimodule.c.fix-test_structmember-on-64bit-bigendian	2012-04-09 19:07:33.000000000 -0400
+++ Python-2.7.3/Modules/_testcapimodule.c	2012-04-12 17:42:55.725766488 -0400
@@ -1813,7 +1813,7 @@ test_structmembers_new(PyTypeObject *typ
         ;
     test_structmembers *ob;
     const char *s = NULL;
-    Py_ssize_t string_len = 0;
+    int string_len = 0;
     ob = PyObject_New(test_structmembers, type);
     if (ob == NULL)
         return NULL;
