Index: Misc/NEWS
===================================================================
--- Misc/NEWS	(revision 79309)
+++ Misc/NEWS	(revision 79310)
@@ -29,6 +29,9 @@
 Library
 -------
 
+- Issue #8205: Remove the "Modules" directory from sys.path when Python is
+  running from the build directory (POSIX only).
+
 - Issue #7667: Fix doctest failures with non-ASCII paths.
 
 - Issue #7512: shutil.copystat() could raise an OSError when the filesystem
Index: Lib/site.py
===================================================================
--- Lib/site.py	(revision 79309)
+++ Lib/site.py	(revision 79310)
@@ -118,7 +118,7 @@
     s = "build/lib.%s-%.3s" % (get_platform(), sys.version)
     if hasattr(sys, 'gettotalrefcount'):
         s += '-pydebug'
-    s = os.path.join(os.path.dirname(sys.path[-1]), s)
+    s = os.path.join(os.path.dirname(sys.path.pop()), s)
     sys.path.append(s)
 
 
