From df77ccbc6b326409a03919d846f9c74f2f9979d4 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 19 Jul 2010 10:12:31 +0100
Subject: [PATCH 2/2] Fix GSeal compilation

---
 src/test-entry.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/test-entry.c b/src/test-entry.c
index 6dc1380..09c361e 100644
--- a/src/test-entry.c
+++ b/src/test-entry.c
@@ -56,7 +56,7 @@ int main (int argc, char **argv)
 	g_signal_connect (G_OBJECT (entry), "phone_changed",
 			G_CALLBACK (phone_changed_cb), NULL);
 
-	gtk_container_add (GTK_CONTAINER (GTK_DIALOG (window)->vbox), entry);
+	gtk_container_add (GTK_CONTAINER (gtk_dialog_get_content_area (GTK_DIALOG (window))), entry);
 
 	gtk_widget_show_all (window);
 
-- 
1.7.0.1

