--- ares_build.h.in
+++ ares_build.h.in
@@ -96,7 +96,14 @@
 #endif
 
 /* The size of `long', as computed by sizeof. */
-#undef CARES_SIZEOF_LONG
+#include <bits/wordsize.h>
+#if __WORDSIZE == 32
+#define CARES_SIZEOF_LONG 4
+#elif __WORDSIZE == 64
+#define CARES_SIZEOF_LONG 8
+#else
+#error "Unknown word size"
+#endif
 
 /* Integral data type used for ares_socklen_t. */
 #undef CARES_TYPEOF_ARES_SOCKLEN_T
--- configure.ac
+++ configure.ac
@@ -468,7 +468,6 @@ AC_CHECK_SIZEOF(size_t)
 AC_CHECK_SIZEOF(long)
 AC_CHECK_SIZEOF(int)
 AC_CHECK_SIZEOF(short)
-CARES_CONFIGURE_LONG
 AC_CHECK_SIZEOF(time_t)
 
 AC_CHECK_TYPE(long long,
