diff -up ./ares_build.h.in.multilib ./ares_build.h.in
--- ./ares_build.h.in.multilib	2010-03-23 13:41:44.000000000 +0100
+++ ./ares_build.h.in	2010-06-15 10:38:15.804618311 +0200
@@ -96,9 +96,6 @@
 #  include <sys/socket.h>
 #endif
 
-/* The size of `long', as computed by sizeof. */
-#undef CARES_SIZEOF_LONG
-
 /* Integral data type used for ares_socklen_t. */
 #undef CARES_TYPEOF_ARES_SOCKLEN_T
 
diff -up ./ares_rules.h.multilib ./ares_rules.h
--- ./ares_rules.h.multilib	2010-06-15 10:39:22.895368907 +0200
+++ ./ares_rules.h	2010-06-15 10:40:19.271619152 +0200
@@ -69,10 +69,12 @@
  * Verify that some macros are actually defined.
  */
 
+#if 0
 #ifndef CARES_SIZEOF_LONG
 #  error "CARES_SIZEOF_LONG definition is missing!"
    Error Compilation_aborted_CARES_SIZEOF_LONG_is_missing
 #endif
+#endif
 
 #ifndef CARES_TYPEOF_ARES_SOCKLEN_T
 #  error "CARES_TYPEOF_ARES_SOCKLEN_T definition is missing!"
@@ -97,9 +99,11 @@
  * is the same as the one reported by sizeof() at compile time.
  */
 
+#if 0
 typedef char
   __cares_rule_01__
     [CareschkszEQ(long, CARES_SIZEOF_LONG)];
+#endif
 
 /*
  * Verify that the size previously defined and expected for
