diff -up ./configure.optflags ./configure
--- ./configure.optflags	2011-08-17 08:17:31.000000000 +0200
+++ ./configure	2011-08-17 08:24:33.000000000 +0200
@@ -14925,7 +14925,7 @@ $as_echo "$as_me: WARNING: compiler opti
     ac_var_stripped=""
   for word1 in $tmp_CFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_dbg_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -14941,7 +14941,7 @@ $as_echo "$as_me: WARNING: compiler opti
     ac_var_stripped=""
   for word1 in $tmp_CPPFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_dbg_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -14957,12 +14957,12 @@ $as_echo "$as_me: WARNING: compiler opti
     if test "$want_debug" = "yes"; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking if compiler accepts debug enabling options" >&5
 $as_echo_n "checking if compiler accepts debug enabling options... " >&6; }
-      tmp_options="$flags_dbg_yes"
+      tmp_options=""
     fi
     if test "$want_debug" = "no"; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking if compiler accepts debug disabling options" >&5
 $as_echo_n "checking if compiler accepts debug disabling options... " >&6; }
-      tmp_options="$flags_dbg_off"
+      tmp_options=""
     fi
     #
     if test "$flags_prefer_cppflags" = "yes"; then
@@ -15123,7 +15123,7 @@ $as_echo_n "checking if compiler optimiz
 
   ac_var_match_word="no"
   for word1 in $tmp_CFLAGS; do
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_match_word="yes"
       fi
@@ -15141,7 +15141,7 @@ $as_echo_n "checking if compiler optimiz
 
   ac_var_match_word="no"
   for word1 in $tmp_CPPFLAGS; do
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_match_word="yes"
       fi
@@ -15172,7 +15172,7 @@ $as_echo "$honor_optimize_option" >&6; }
     ac_var_stripped=""
   for word1 in $tmp_CFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -15188,7 +15188,7 @@ $as_echo "$honor_optimize_option" >&6; }
     ac_var_stripped=""
   for word1 in $tmp_CPPFLAGS; do
     ac_var_strip_word="no"
-    for word2 in $flags_opt_all; do
+    for word2 in ""; do
       if test "$word1" = "$word2"; then
         ac_var_strip_word="yes"
       fi
@@ -15203,12 +15203,12 @@ $as_echo "$honor_optimize_option" >&6; }
       if test "$want_optimize" = "yes"; then
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking if compiler accepts optimizer enabling options" >&5
 $as_echo_n "checking if compiler accepts optimizer enabling options... " >&6; }
-        tmp_options="$flags_opt_yes"
+        tmp_options=""
       fi
       if test "$want_optimize" = "no"; then
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking if compiler accepts optimizer disabling options" >&5
 $as_echo_n "checking if compiler accepts optimizer disabling options... " >&6; }
-        tmp_options="$flags_opt_off"
+        tmp_options=""
       fi
       if test "$flags_prefer_cppflags" = "yes"; then
         CPPFLAGS="$tmp_CPPFLAGS $tmp_options"
