--- setup/Linux/oss/build/osscore.c	2012-05-22 00:23:41.000000000 +0200
+++ setup/Linux/oss/build/osscore.c	2013-07-20 21:47:18.294007175 +0200
@@ -953,7 +984,9 @@
   fop->llseek = oss_no_llseek;
   fop->read = tmp_read;
   fop->write = tmp_write;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,10,7)
   fop->readdir = NULL;		/* tmp_readdir; */
+#endif
   fop->poll = tmp_poll;
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,35)
   fop->ioctl = tmp_ioctl;
