diff -ur oss-v4.2-build2008-src-gpl.orig/setup/Linux/oss/build/osscore.c oss-v4.2-build2008-src-gpl/setup/Linux/oss/build/osscore.c
--- oss-v4.2-build2008-src-gpl.orig/setup/Linux/oss/build/osscore.c	2013-11-06 22:22:05.760399613 +0400
+++ oss-v4.2-build2008-src-gpl/setup/Linux/oss/build/osscore.c	2013-11-06 22:34:41.908792755 +0400
@@ -436,11 +436,15 @@
 int
 oss_get_uid (void)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,12,0)
+  return __kuid_val(current->cred->uid);
+#else
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,29)
   return current->cred->uid;
 #else
   return current->uid;
 #endif
+#endif
 }
 
 typedef struct tmout_desc
