--- setup/Linux/oss/build/osscore.c.orig	2015-04-22 22:16:47.406935916 -0300
+++ setup/Linux/oss/build/osscore.c	2015-04-22 22:20:28.650172614 -0300
@@ -1950,8 +1950,13 @@ oss_fp_save (short *envbuf, unsigned int
     }
   else
     {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
+      flags[1] = __read_cr4 ();
+      __write_cr4 (flags[1] | 0x600);	/* Set OSFXSR & OSXMMEXCEPT */
+#else
       flags[1] = read_cr4 ();
       write_cr4 (flags[1] | 0x600);	/* Set OSFXSR & OSXMMEXCEPT */
+#endif
       FX_SAVE (envbuf);
       asm ("fninit");
       asm ("fwait");
@@ -1971,7 +1976,11 @@ oss_fp_restore (short *envbuf, unsigned
   else
     {
       FX_RESTORE (envbuf);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
+      __write_cr4 (flags[1]);	/* Restore cr4 */
+#else
       write_cr4 (flags[1]);	/* Restore cr4 */
+#endif
     }
   write_cr0 (flags[0]);		/* Restore cr0 */
 }
