diff -Nur rkhunter-1.4.0.orig/files/rkhunter.conf rkhunter-1.4.0/files/rkhunter.conf
--- rkhunter-1.4.0.orig/files/rkhunter.conf	2011-12-17 14:36:02.000000000 -0700
+++ rkhunter-1.4.0/files/rkhunter.conf	2012-10-06 13:43:28.030872769 -0600
@@ -100,16 +100,19 @@
 # sure that the directory permissions are tight.
 #
 #TMPDIR=/var/lib/rkhunter/tmp
+TMPDIR=/var/lib/rkhunter
 
 #
 # Specify the database directory to use.
 #
 #DBDIR=/var/lib/rkhunter/db
+DBDIR=/var/lib/rkhunter/db
 
 #
 # Specify the script directory to use.
 #
 #SCRIPTDIR=/usr/local/lib/rkhunter/scripts
+SCRIPTDIR=/usr/share/rkhunter/scripts
 
 #
 # This option can be used to modify the command directory list used
@@ -156,13 +159,13 @@
 #
 # NOTE: This option should be present in the configuration file.
 #
-LOGFILE=/var/log/rkhunter.log
+LOGFILE=/var/log/rkhunter/rkhunter.log
 
 #
 # Set the following option to 1 if the log file is to be appended to
 # whenever rkhunter is run.
 #
-APPEND_LOG=0
+APPEND_LOG=1
 
 #
 # Set the following option to 1 if the log file is to be copied when
@@ -184,7 +187,7 @@
 # Setting the value to 'none', or just leaving the option commented out,
 # disables the use of syslog.
 #
-#USE_SYSLOG=authpriv.notice
+USE_SYSLOG=authpriv.notice
 
 #
 # Set the following option to 1 if the second colour set is to be used.
@@ -214,7 +217,7 @@
 # file, then a value here of 'unset' can be used to avoid warning messages.
 # This option has a default value of 'no'.
 #
-ALLOW_SSH_ROOT_USER=no
+ALLOW_SSH_ROOT_USER=unset
 
 #
 # Set this option to '1' to allow the use of the SSH-1 protocol, but note
@@ -225,7 +228,7 @@
 # configuration file, then a value of '2' may be set here in order to
 # suppress a warning message. This option has a default value of '0'.
 #
-ALLOW_SSH_PROT_V1=0
+ALLOW_SSH_PROT_V1=2
 
 #
 # This setting tells rkhunter the directory containing the SSH configuration
@@ -256,7 +259,7 @@
 # tests, the test names, and how rkhunter behaves when these options are used.
 #
 ENABLE_TESTS="all"
-DISABLE_TESTS="suspscan hidden_ports hidden_procs deleted_files packet_cap_apps"
+DISABLE_TESTS="suspscan hidden_procs deleted_files packet_cap_apps apps"
 
 #
 # The HASH_FUNC option can be used to specify the command to use
@@ -433,6 +436,9 @@
 # for each new file, and rkhunter will report this.
 #
 #EXISTWHITELIST=""
+EXISTWHITELIST=/bin/ad
+# FreeIPA Certificate Authority
+EXISTWHITELIST=/var/log/pki-ca/system
 
 #
 # Whitelist various attributes of the specified files.
@@ -468,6 +474,12 @@
 #
 #SCRIPTWHITELIST="/sbin/ifup /sbin/ifdown"
 #SCRIPTWHITELIST="/usr/bin/groups"
+SCRIPTWHITELIST=/usr/bin/whatis
+SCRIPTWHITELIST=/usr/bin/ldd
+SCRIPTWHITELIST=/usr/bin/groups
+SCRIPTWHITELIST=/usr/bin/GET
+SCRIPTWHITELIST=/sbin/ifup
+SCRIPTWHITELIST=/sbin/ifdown
 
 #
 # Allow the specified commands to have the immutable attribute set.
@@ -491,12 +503,24 @@
 # The option may be specified more than once. The option
 # may use wildcard characters.
 #
-#ALLOWHIDDENDIR="/etc/.java"
+ALLOWHIDDENDIR="/etc/.java"
 #ALLOWHIDDENDIR="/dev/.udev /dev/.udevdb /dev/.udev.tdb"
 #ALLOWHIDDENDIR="/dev/.static"
 #ALLOWHIDDENDIR="/dev/.initramfs"
 #ALLOWHIDDENDIR="/dev/.SRC-unix"
 #ALLOWHIDDENDIR="/dev/.mdadm"
+ALLOWHIDDENDIR=/dev/.udev
+ALLOWHIDDENDIR=/dev/.udevdb
+ALLOWHIDDENDIR=/dev/.udev.tdb
+ALLOWHIDDENDIR=/dev/.static
+ALLOWHIDDENDIR=/dev/.initramfs
+ALLOWHIDDENDIR=/dev/.SRC-unix
+ALLOWHIDDENDIR=/dev/.mdadm
+ALLOWHIDDENDIR=/dev/.systemd
+ALLOWHIDDENDIR=/dev/.mount
+# for etckeeper
+ALLOWHIDDENDIR=/etc/.git
+ALLOWHIDDENDIR=/etc/.bzr
 
 #
 # Allow the specified hidden files to be whitelisted.
@@ -506,7 +530,7 @@
 # characters.
 # 
 #ALLOWHIDDENFILE="/etc/.java"
-#ALLOWHIDDENFILE="/usr/share/man/man1/..1.gz"
+ALLOWHIDDENFILE="/usr/share/man/man1/..1.gz"
 #ALLOWHIDDENFILE="/etc/.pwd.lock"
 #ALLOWHIDDENFILE="/etc/.init.state"
 #ALLOWHIDDENFILE="/lib/.libcrypto.so.0.9.8e.hmac /lib/.libcrypto.so.6.hmac"
@@ -521,6 +545,29 @@
 #ALLOWHIDDENFILE="/usr/lib/hmaccalc/sha384hmac.hmac"
 #ALLOWHIDDENFILE="/usr/lib/hmaccalc/sha512hmac.hmac"
 #ALLOWHIDDENFILE="/usr/sbin/.sshd.hmac"
+ALLOWHIDDENFILE=/lib*/.libcrypto.so.*.hmac
+ALLOWHIDDENFILE=/lib*/.libssl.so.*.hmac
+ALLOWHIDDENFILE=/usr/bin/.fipscheck.hmac
+ALLOWHIDDENFILE=/usr/bin/.ssh.hmac
+ALLOWHIDDENFILE=/usr/bin/.ssh-keygen.hmac
+ALLOWHIDDENFILE=/usr/bin/.ssh-keyscan.hmac
+ALLOWHIDDENFILE=/usr/bin/.ssh-add.hmac
+ALLOWHIDDENFILE=/usr/bin/.ssh-agent.hmac
+ALLOWHIDDENFILE=/usr/lib*/.libfipscheck.so.*.hmac
+ALLOWHIDDENFILE=/usr/lib*/.libgcrypt.so.*.hmac
+ALLOWHIDDENFILE=/usr/lib*/hmaccalc/sha1hmac.hmac
+ALLOWHIDDENFILE=/usr/lib*/hmaccalc/sha256hmac.hmac
+ALLOWHIDDENFILE=/usr/lib*/hmaccalc/sha384hmac.hmac
+ALLOWHIDDENFILE=/usr/lib*/hmaccalc/sha512hmac.hmac
+ALLOWHIDDENFILE=/usr/sbin/.sshd.hmac
+ALLOWHIDDENFILE=/dev/.mdadm.map
+ALLOWHIDDENFILE=/usr/share/man/man5/.k5login.5.gz
+ALLOWHIDDENFILE=/usr/share/man/man5/.k5identity.5.gz
+ALLOWHIDDENFILE=/usr/sbin/.ipsec.hmac
+# etckeeper
+ALLOWHIDDENFILE=/etc/.etckeeper
+ALLOWHIDDENFILE=/etc/.gitignore
+ALLOWHIDDENFILE=/etc/.bzrignore
 
 #
 # Allow the specified processes to use deleted files. The
@@ -586,6 +633,16 @@
 #
 #ALLOWDEVFILE="/dev/shm/pulse-shm-*"
 #ALLOWDEVFILE="/dev/shm/sem.ADBE_*"
+ALLOWDEVFILE=/dev/shm/pulse-shm-*
+ALLOWDEVFILE=/dev/md/md-device-map
+# tomboy creates this one
+ALLOWDEVFILE="/dev/shm/mono.*"
+# created by libv4l
+ALLOWDEVFILE="/dev/shm/libv4l-*"
+# created by spice video
+ALLOWDEVFILE="/dev/shm/spice.*"
+# created by mdadm
+ALLOWDEVFILE="/dev/md/autorebuild.pid"
 
 #
 # This setting tells rkhunter where the inetd configuration
@@ -724,6 +781,7 @@
 # The option may be specified more than once.
 #
 #SUSPSCAN_DIRS="/tmp /var/tmp"
+SUSPSCAN_DIRS="/tmp /var/tmp"
 
 #
 # Directory for temporary files. A memory-based one is better (faster).
@@ -822,7 +880,9 @@
 #
 #RTKT_DIR_WHITELIST=""
 #RTKT_FILE_WHITELIST=""
-
+RTKT_FILE_WHITELIST=/bin/ad
+# FreeIPA Certificate Authority
+RTKT_FILE_WHITELIST=/var/log/pki-ca/system
 #
 # The following option can be used to whitelist shared library files that would
 # normally be flagged with a warning during the preloaded shared library check.
@@ -980,3 +1040,5 @@
 # both programs, then disable the 'hidden_procs' test.
 #
 #DISABLE_UNHIDE=0
+
+INSTALLDIR="/usr"
