From feff3aad50304450248aa66aed59d09439699234 Mon Sep 17 00:00:00 2001
From: Stephen Gallagher <sgallagh@redhat.com>
Date: Wed, 12 Jan 2011 15:01:52 -0500
Subject: [PATCH] Install python bindings in the arch-specific location

---
 buildtools/wafsamba/samba_python.py |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/buildtools/wafsamba/samba_python.py b/buildtools/wafsamba/samba_python.py
index a663b1910d063336c1340142ae8855d7e176ecdb..9b72f3fd112288a022a2de0701594814616f202f 100644
--- a/buildtools/wafsamba/samba_python.py
+++ b/buildtools/wafsamba/samba_python.py
@@ -51,6 +51,9 @@ def SAMBA_PYTHON(bld, name,
 
     link_name = 'python/%s' % realname
 
+    from distutils.sysconfig import get_python_lib
+    arch_install_dir = get_python_lib(1)
+
     bld.SAMBA_LIBRARY(name,
                       source=source,
                       deps=deps,
@@ -63,7 +66,7 @@ def SAMBA_PYTHON(bld, name,
                       link_name=link_name,
                       pyembed=True,
                       target_type='PYTHON',
-                      install_path='${PYTHONDIR}',
+                      install_path=arch_install_dir,
                       enabled=enabled)
 
 Build.BuildContext.SAMBA_PYTHON = SAMBA_PYTHON
-- 
1.7.3.4

