diff -up notification-daemon-engine-nodoka-0.1.0/configure.ac.automake notification-daemon-engine-nodoka-0.1.0/configure.ac
--- notification-daemon-engine-nodoka-0.1.0/configure.ac.automake	2012-07-22 23:27:49.000000000 +0200
+++ notification-daemon-engine-nodoka-0.1.0/configure.ac	2012-07-22 23:29:20.000000000 +0200
@@ -25,6 +25,7 @@ AC_DEFINE_UNQUOTED(ENGINE_VERSION, "$ENG
 
 AM_INIT_AUTOMAKE([-Wall -Werror])
 AM_MAINTAINER_MODE
+AM_PROG_AR
 
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
