diff -up librcc-0.2.9/configure.in.libguess librcc-0.2.9/configure.in
--- librcc-0.2.9/configure.in.libguess	2009-10-09 19:08:47.000000000 -0300
+++ librcc-0.2.9/configure.in	2010-07-18 05:00:45.177101502 -0300
@@ -192,7 +192,7 @@ ENCA_LIBS=""
 ENCA_INCLUDES=""
 HAVE_ENCA=no
 LIBGUESS_LIBS=""
-LIBGUESS_INCLUDES=""
+LIBGUESS_CFLAGS=""
 HAVE_LIBGUESS=no
 
 AC_ARG_ENABLE( force-dynamic-engines,
@@ -214,12 +214,7 @@ if test "x$enable_force_dynamic_engines"
     ])])
 fi
 
-AC_CHECK_HEADER(libguess.h, [AC_CHECK_LIB(guess, guess_jp, [ 
-	AC_DEFINE(HAVE_LIBGUESS,1,[Defines if libguess is available])
-	LIBGUESS_LIBS="-lguess"
-	LIBGUESS_INCLUDES=""
-	HAVE_LIBGUESS=yes
-])])
+PKG_CHECK_MODULES(LIBGUESS, libguess >= 1.0, HAVE_LIBGUESS=yes, HAVE_LIBGUESS=no)
 
 AM_CONDITIONAL(HAVE_RCD, [ test "x$HAVE_RCD" = "xyes" ])
 AM_CONDITIONAL(HAVE_ENCA, [ test "x$HAVE_ENCA" = "xyes" ])
@@ -230,7 +225,7 @@ AC_SUBST(RCD_INCLUDES)
 AC_SUBST(ENCA_LIBS)
 AC_SUBST(ENCA_INCLUDES)
 AC_SUBST(LIBGUESS_LIBS)
-AC_SUBST(LIBGUESS_INCLUDES)
+AC_SUBST(LIBGUESS_CFLAGS)
 
 USE_DLOPEN=no
 if test "x$HAVE_DLOPEN" = "xyes"; then
diff -up librcc-0.2.9/engines/Makefile.am.libguess librcc-0.2.9/engines/Makefile.am
--- librcc-0.2.9/engines/Makefile.am.libguess	2009-10-09 12:07:36.000000000 -0300
+++ librcc-0.2.9/engines/Makefile.am	2010-07-18 05:00:45.175097263 -0300
@@ -19,4 +19,4 @@ endif
 western_engine_la_SOURCES = western.c
 western_engine_la_LDFLAGS = -module -avoid-version -export-symbols-regex "rccGetInfo"
 
-AM_CPPFLAGS = -I../src @RCD_INCLUDES@ @LIBGUESS_INCLUDES@
+AM_CPPFLAGS = -I../src @RCD_INCLUDES@ @LIBGUESS_CFLAGS@
diff -up librcc-0.2.9/engines/libguess.c.libguess librcc-0.2.9/engines/libguess.c
--- librcc-0.2.9/engines/libguess.c.libguess	2010-07-18 05:01:38.169481857 -0300
+++ librcc-0.2.9/engines/libguess.c	2010-07-18 05:05:43.253097729 -0300
@@ -64,7 +64,7 @@ struct rcc_guess_engine_t guessJPEngine
     &guessDetect,
     {"UTF-8", "UTF-16", "ISO-2022-JP", "EUC-JP", "SJIS", NULL}
  },
- &guess_jp
+ NULL
 };
 
 struct rcc_guess_engine_t guessZHEngine = {
@@ -110,7 +110,7 @@ struct rcc_guess_engine_t guessKREngine
     &guessDetect,
     {"UTF-8", "UTF-16", "ISO-2022-KR", "EUC-KR", "JOHAB", NULL}
  },
- &guess_kr
+ NULL
 };
         
     
