diff -up system-config-printer-1.4.0/system-config-printer.py.quit system-config-printer-1.4.0/system-config-printer.py
--- system-config-printer-1.4.0/system-config-printer.py.quit	2013-04-12 12:05:58.441494614 +0100
+++ system-config-printer-1.4.0/system-config-printer.py	2013-04-12 12:06:05.500559576 +0100
@@ -1230,11 +1230,11 @@ class GUI(GtkGUI):
         while len (self.jobviewers) > 0:
             # this will call on_jobviewer_exit
             self.jobviewers[0].on_delete_event ()
-        del self.mainlist
-        del self.printers
         self.propertiesDlg.destroy ()
         self.newPrinterGUI.destroy ()
         Gtk.main_quit()
+        del self.mainlist
+        del self.printers
 
     # Rename
     def is_rename_possible (self, name):
