diff -up system-config-printer-1.2.5/installdriver.py.InstallPrinterDrivers-debug system-config-printer-1.2.5/installdriver.py
--- system-config-printer-1.2.5/installdriver.py.InstallPrinterDrivers-debug	2010-10-12 17:03:22.000000000 +0100
+++ system-config-printer-1.2.5/installdriver.py	2010-10-14 14:09:49.205869035 +0100
@@ -22,6 +22,7 @@
 import dbus
 import dbus.glib
 import dbus.service
+from debug import debugprint
 
 class PrinterDriversInstaller(dbus.service.Object):
     DBUS_PATH  = "/com/redhat/PrinterDriversInstaller"
@@ -43,8 +44,12 @@ class PrinterDriversInstaller(dbus.servi
         obj = bus.get_object ("org.freedesktop.PackageKit",
                               "/org/freedesktop/PackageKit")
         proxy = dbus.Interface (obj, "org.freedesktop.PackageKit.Modify")
-        proxy.InstallPrinterDrivers (0, ["MFG:%s;MDL:%s;" % (mfg, mdl)],
-                                     "hide-finished",
+        xid = 0
+        resources = ["MFG:%s;MDL:%s;" % (mfg, mdl)]
+        interaction = "hide-finished"
+        debugprint ("Calling InstallPrinterDrivers (%s, %s, %s)" %
+                    (repr (xid), repr (resources), repr (interaction)))
+        proxy.InstallPrinterDrivers (xid, resources, interaction,
                                      reply_handler=reply_handler,
                                      error_handler=error_handler,
                                      timeout=3600)
diff -up system-config-printer-1.2.5/ppdsloader.py.InstallPrinterDrivers-debug system-config-printer-1.2.5/ppdsloader.py
--- system-config-printer-1.2.5/ppdsloader.py.InstallPrinterDrivers-debug	2010-10-13 13:11:16.000000000 +0100
+++ system-config-printer-1.2.5/ppdsloader.py	2010-10-14 14:09:49.207994422 +0100
@@ -163,8 +163,11 @@ class PPDsLoader:
             obj = self._bus.get_object ("org.freedesktop.PackageKit",
                                         "/org/freedesktop/PackageKit")
             proxy = dbus.Interface (obj, "org.freedesktop.PackageKit.Modify")
-            proxy.InstallPrinterDrivers (xid, [self._device_id],
-                                         "hide-finished",
+            resources = [self._device_id]
+            interaction = "hide-finished"
+            debugprint ("Calling InstallPrinterDrivers (%s, %s, %s)" %
+                        (repr (xid), repr (resources), repr (interaction)))
+            proxy.InstallPrinterDrivers (xid, resources, interaction,
                                          reply_handler=self._packagekit_reply,
                                          error_handler=self._packagekit_error,
                                          timeout=3600)
