diff -up system-config-printer-1.3.5/serversettings.py.bug726682 system-config-printer-1.3.5/serversettings.py
--- system-config-printer-1.3.5/serversettings.py.bug726682	2011-07-28 17:04:13.000000000 +0100
+++ system-config-printer-1.3.5/serversettings.py	2011-07-29 15:57:01.515904062 +0100
@@ -23,6 +23,7 @@
 import config
 from gettext import gettext as _
 import cups
+import dbus
 import gobject
 import gtk
 import os
