diff -up system-config-printer-1.2.5/system-config-printer.py.compiz-rename system-config-printer-1.2.5/system-config-printer.py
--- system-config-printer-1.2.5/system-config-printer.py.compiz-rename	2010-12-02 16:15:00.656238304 +0000
+++ system-config-printer-1.2.5/system-config-printer.py	2010-12-02 16:15:13.265702866 +0000
@@ -3076,6 +3076,7 @@ class GUI(GtkGUI, monitor.Watcher):
         ids.append (cell.connect ('editing-canceled',
                                  self.printer_name_edit_cancel))
         self.rename_sigids = ids
+        self.dests_iconview.grab_focus ()
 
     def printer_name_edited (self, cell, path, newname):
         newname = newname.replace("/", "")
