diff -up system-config-printer-1.3.0/applet.py.no-job-notifications system-config-printer-1.3.0/applet.py
--- system-config-printer-1.3.0/applet.py.no-job-notifications	2011-02-16 15:43:42.215670390 +0000
+++ system-config-printer-1.3.0/applet.py	2011-02-16 15:44:09.041703328 +0000
@@ -441,8 +441,7 @@ if __name__ == '__main__':
             pass
 
     loop = gobject.MainLoop ()
-    runloop = RunLoop (session_bus, system_bus, loop)
     try:
-        runloop.run ()
+        loop.run ()
     except KeyboardInterrupt:
         pass
