Description: port to libxfce4ui
Origin: other, https://bugzilla.xfce.org/attachment.cgi?id=3529
Bug: https://bugzilla.xfce.org/show_bug.cgi?id=7355

diff -ur xfburn-0.4.3/configure.in xfburn-0.4.3-4ui/configure.in
--- xfburn-0.4.3/configure.in	2011-02-26 20:25:11.173013553 +0100
+++ xfburn-0.4.3-4ui/configure.in	2011-02-26 19:16:30.409029549 +0100
@@ -38,7 +38,7 @@
 XDT_CHECK_PACKAGE([LIBISOFS], [libisofs-1], [0.6.2])
 XDT_CHECK_PACKAGE([GTHREAD], [gthread-2.0], [2.6.0])
 XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.10.0])
-XDT_CHECK_PACKAGE([LIBXFCEGUI4], [libxfcegui4-1.0], [4.4.0])
+XDT_CHECK_PACKAGE([LIBXFCE4UI], [libxfce4ui-1], [4.8.0])
 XDT_CHECK_PACKAGE([EXO], [exo-0.3], [0.3.4], [], [XDT_CHECK_PACKAGE([EXO], [exo-1], [0.3.100])])
 XDT_CHECK_PACKAGE([LIBGIO], [gio-2.0], [2.22.0])
 XDT_CHECK_PACKAGE([LIBGIOUNIX], [gio-unix-2.0], [2.16.0])
diff -ur xfburn-0.4.3/xfburn/Makefile.am xfburn-0.4.3-4ui/xfburn/Makefile.am
--- xfburn-0.4.3/xfburn/Makefile.am	2011-02-26 20:25:11.174013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/Makefile.am	2011-02-26 19:17:52.972029553 +0100
@@ -95,7 +95,7 @@
 	$(GTHREAD_CFLAGS)						\
 	$(LIBBURN_CFLAGS)						\
 	$(LIBISOFS_CFLAGS)						\
-	$(LIBXFCEGUI4_CFLAGS)						\
+	$(LIBXFCE4UI_CFLAGS)						\
 	$(EXO_CFLAGS)							\
 	$(GIO_CFLAGS)							\
 	$(GIO_UNIX_CFLAGS)						\
@@ -106,7 +106,7 @@
 	$(GTHREAD_LIBS)							\
 	$(LIBBURN_LIBS)							\
 	$(LIBISOFS_LIBS)						\
-	$(LIBXFCEGUI4_LIBS)						\
+	$(LIBXFCE4UI_LIBS)						\
 	$(EXO_LIBS)							\
 	$(GIO_LIBS)							\
 	$(GIO_UNIX_LIBS)						\
diff -ur xfburn-0.4.3/xfburn/xfburn-audio-composition.c xfburn-0.4.3-4ui/xfburn/xfburn-audio-composition.c
--- xfburn-0.4.3/xfburn/xfburn-audio-composition.c	2011-02-26 20:25:11.175013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-audio-composition.c	2011-02-26 20:12:31.066324063 +0100
@@ -32,7 +32,7 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <gio/gio.h>
 
@@ -577,7 +577,7 @@
     dialog = xfburn_burn_audio_cd_composition_dialog_new (src);
     break;
   case DVD_DISC:
-    xfce_err (_("Cannot burn audio onto a DVD."));
+    xfce_dialog_show_error (NULL, NULL, _("Cannot burn audio onto a DVD."));
     return;
     break;
   }
@@ -1021,7 +1021,7 @@
 {
   XfburnAudioCompositionPrivate *priv = XFBURN_AUDIO_COMPOSITION_GET_PRIVATE (dc);
   
-  xfce_info (xfburn_transcoder_get_description (priv->trans));
+  xfce_dialog_show_info(NULL, NULL, "%s", xfburn_transcoder_get_description (priv->trans));
 }
 
 static void
@@ -1089,14 +1089,14 @@
   g_assert (error != NULL);
 
   if (error->domain != XFBURN_ERROR) {
-    xfce_warn (error->message);
+    xfce_dialog_show_warning(NULL, NULL, "%s", error->message);
     return;
   }
 
   if (g_hash_table_lookup (priv->warned_about, GINT_TO_POINTER (error->code)) == NULL) {
     g_hash_table_insert (priv->warned_about, GINT_TO_POINTER (error->code), did_warn);
 
-    xfce_warn (error->message);
+    xfce_dialog_show_warning(NULL, NULL, "%s", error->message);
   }
 }
 
@@ -1141,7 +1141,7 @@
     
     gdk_threads_enter ();
     if (file_exists_on_same_level (model, tree_path, FALSE, name)) {
-      xfce_err (_("A file with the same name is already present in the composition."));
+      xfce_dialog_show_error (NULL, NULL, _("A file with the same name is already present in the composition."));
 
       gtk_tree_path_free (tree_path);
       gdk_threads_leave ();
@@ -1225,7 +1225,7 @@
         if (g_hash_table_lookup (priv->warned_about, GINT_TO_POINTER (err_code)) == NULL) {
           g_hash_table_insert (priv->warned_about, GINT_TO_POINTER (err_code), did_warn);
           gdk_threads_enter ();
-          xfce_err (_("You can only have a maximum of 99 tracks."));
+          xfce_dialog_show_error (NULL, NULL, _("You can only have a maximum of 99 tracks."));
           gdk_threads_leave ();
         }
 
@@ -1475,7 +1475,7 @@
     
       /*
       if (file_exists_on_same_level (model, path_level, FALSE, name)) {
-        xfce_warn (_("A file named \"%s\" already exists in this directory, the file hasn't been added."), name);
+        xfce_dialog_warning(NULL, NULL, _("A file named \"%s\" already exists in this directory, the file hasn't been added."), name);
         goto cleanup;
       }
       */
diff -ur xfburn-0.4.3/xfburn/xfburn-audio-disc-usage.c xfburn-0.4.3-4ui/xfburn/xfburn-audio-disc-usage.c
--- xfburn-0.4.3/xfburn/xfburn-audio-disc-usage.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-audio-disc-usage.c	2011-02-26 16:21:03.534320584 +0100
@@ -22,7 +22,6 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
 
 #include "xfburn-disc-usage.h"
 #include "xfburn-audio-disc-usage.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-blank-dialog.c xfburn-0.4.3-4ui/xfburn/xfburn-blank-dialog.c
--- xfburn-0.4.3/xfburn/xfburn-blank-dialog.c	2011-02-26 20:25:11.176013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-blank-dialog.c	2011-02-26 20:13:58.518013555 +0100
@@ -20,7 +20,7 @@
 #include <config.h>
 #endif /* !HAVE_CONFIG_H */
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libburn.h>
 
@@ -206,7 +206,7 @@
   g_signal_connect (G_OBJECT (priv->device_box), "volume-changed", G_CALLBACK (cb_volume_changed), obj);
   gtk_widget_show (priv->device_box);
 
-  frame = xfce_create_framebox_with_content (_("Burning device"), priv->device_box);
+  frame = xfce_gtk_frame_box_new_with_content (_("Burning device"), priv->device_box);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -219,7 +219,7 @@
   gtk_cell_layout_set_attributes (GTK_CELL_LAYOUT (priv->combo_type), cell, "text", BLANK_COMBO_NAME_COLUMN, NULL);
   gtk_widget_show (priv->combo_type);
 
-  frame = xfce_create_framebox_with_content (_("Blank mode"), priv->combo_type);
+  frame = xfce_gtk_frame_box_new_with_content (_("Blank mode"), priv->combo_type);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -227,7 +227,7 @@
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox);
 
-  frame = xfce_create_framebox_with_content (_("Options"), vbox);
+  frame = xfce_gtk_frame_box_new_with_content (_("Options"), vbox);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -241,7 +241,7 @@
   gtk_widget_show (button);
   gtk_dialog_add_action_widget (GTK_DIALOG (obj), button, GTK_RESPONSE_CANCEL);
 
-  button = xfce_create_mixed_button ("xfburn-blank-cdrw", _("_Blank"));
+  button = xfce_gtk_button_new_mixed ("xfburn-blank-cdrw", _("_Blank"));
   gtk_widget_show (button);
   gtk_dialog_add_action_widget (GTK_DIALOG (obj), button, GTK_RESPONSE_OK);
   GTK_WIDGET_SET_FLAGS (button, GTK_CAN_DEFAULT);
--- xfburn-0.4.3/xfburn/xfburn-blank-dialog.h~	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3/xfburn/xfburn-blank-dialog.h	2011-02-26 21:17:54.580716604 +0100
@@ -26,6 +26,7 @@
 #include "xfburn-global.h"
 
 #include <gtk/gtk.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 G_BEGIN_DECLS
 
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-audio-cd-composition-dialog.c xfburn-0.4.3-4ui/xfburn/xfburn-burn-audio-cd-composition-dialog.c
--- xfburn-0.4.3/xfburn/xfburn-burn-audio-cd-composition-dialog.c	2009-10-30 02:10:13.000000000 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-audio-cd-composition-dialog.c	2011-02-26 20:14:58.426013553 +0100
@@ -156,7 +156,7 @@
   g_signal_connect (G_OBJECT (priv->device_box), "volume-changed", G_CALLBACK (cb_volume_changed), obj);
   gtk_widget_show (priv->device_box);
 
-  priv->frame_device = xfce_create_framebox_with_content (_("Burning device"), priv->device_box);
+  priv->frame_device = xfce_gtk_frame_box_new_with_content (_("Burning device"), priv->device_box);
   gtk_widget_show (priv->frame_device);
   gtk_box_pack_start (box, priv->frame_device, FALSE, FALSE, BORDER);
 
@@ -168,7 +168,7 @@
     vbox = gtk_vbox_new (FALSE, 0);
     gtk_widget_show (vbox);
 
-    frame = xfce_create_framebox_with_content (_("Composition name"), vbox);
+    frame = xfce_gtk_frame_box_new_with_content (_("Composition name"), vbox);
     gtk_widget_show (frame);
     gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -190,7 +190,7 @@
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox);
 
-  frame = xfce_create_framebox_with_content (_("Options"), vbox);
+  frame = xfce_gtk_frame_box_new_with_content (_("Options"), vbox);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -218,7 +218,7 @@
   gtk_widget_show (button);
   gtk_dialog_add_action_widget (GTK_DIALOG (obj), button, GTK_RESPONSE_CANCEL);
 
-  priv->button_proceed = button = xfce_create_mixed_button ("xfburn-burn-cd", _("_Burn Composition"));
+  priv->button_proceed = button = xfce_gtk_button_new_mixed ("xfburn-burn-cd", _("_Burn Composition"));
 
   gtk_widget_show (button);
   gtk_dialog_add_action_widget (GTK_DIALOG (obj), button, GTK_RESPONSE_OK);
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-audio-cd-composition-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-burn-audio-cd-composition-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-burn-audio-cd-composition-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-audio-cd-composition-dialog.h	2011-02-26 16:21:06.852320585 +0100
@@ -24,7 +24,7 @@
 #endif /* !HAVE_CONFIG_H */
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libisofs.h>
 
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-data-cd-composition-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-cd-composition-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-burn-data-cd-composition-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-cd-composition-dialog.h	2011-02-26 16:21:06.854320585 +0100
@@ -24,7 +24,7 @@
 #endif /* !HAVE_CONFIG_H */
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libisofs.h>
 #include "xfburn-burn-data-composition-base-dialog.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-data-composition-base-dialog.c xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-composition-base-dialog.c
--- xfburn-0.4.3/xfburn/xfburn-burn-data-composition-base-dialog.c	2009-10-30 02:11:13.000000000 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-composition-base-dialog.c	2011-02-26 20:14:13.313013558 +0100
@@ -192,7 +192,7 @@
 
   g_signal_connect (G_OBJECT (priv->device_box), "volume-changed", G_CALLBACK (cb_volume_changed), obj);
 
-  priv->frame_device = xfce_create_framebox_with_content (_("Burning device"), priv->device_box);
+  priv->frame_device = xfce_gtk_frame_box_new_with_content (_("Burning device"), priv->device_box);
   gtk_widget_show (priv->frame_device);
   gtk_box_pack_start (box, priv->frame_device, FALSE, FALSE, BORDER);
 
@@ -203,7 +203,7 @@
     vbox = gtk_vbox_new (FALSE, 0);
     gtk_widget_show (vbox);
 
-    frame = xfce_create_framebox_with_content (_("Composition name"), vbox);
+    frame = xfce_gtk_frame_box_new_with_content (_("Composition name"), vbox);
     gtk_widget_show (frame);
     gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -224,7 +224,7 @@
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox);
 
-  frame = xfce_create_framebox_with_content (_("Options"), vbox);
+  frame = xfce_gtk_frame_box_new_with_content (_("Options"), vbox);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -280,7 +280,7 @@
   gtk_widget_show (button);
   gtk_dialog_add_action_widget (GTK_DIALOG (obj), button, GTK_RESPONSE_CANCEL);
 
-  priv->button_proceed = button = xfce_create_mixed_button ("xfburn-burn-cd", _("_Burn Composition"));
+  priv->button_proceed = button = xfce_gtk_button_new_mixed ("xfburn-burn-cd", _("_Burn Composition"));
   /*
    * Disabled: change button_proceed functionality
   button = create_proceed_button (obj, "xfburn-burn-cd", "");
@@ -650,7 +650,7 @@
 
     if (iso_image_create_burn_source (priv->image, write_opts, &src) < 0) {
       /* could not create source */
-      xfce_err (_("Could not create ISO source structure."));
+      xfce_dialog_show_error (NULL, NULL, _("Could not create ISO source structure."));
       return;
     }
 
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-data-composition-base-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-composition-base-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-burn-data-composition-base-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-composition-base-dialog.h	2011-02-26 16:21:06.856320585 +0100
@@ -24,7 +24,7 @@
 #endif /* !HAVE_CONFIG_H */
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libisofs.h>
 
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-data-dvd-composition-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-dvd-composition-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-burn-data-dvd-composition-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-data-dvd-composition-dialog.h	2011-02-26 16:21:06.857320585 +0100
@@ -24,7 +24,7 @@
 #endif /* !HAVE_CONFIG_H */
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libisofs.h>
 #include "xfburn-burn-data-composition-base-dialog.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-image-dialog.c xfburn-0.4.3-4ui/xfburn/xfburn-burn-image-dialog.c
--- xfburn-0.4.3/xfburn/xfburn-burn-image-dialog.c	2009-10-30 03:47:20.000000000 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-image-dialog.c	2011-02-26 20:14:24.039013553 +0100
@@ -173,7 +173,7 @@
   gtk_file_chooser_add_filter (GTK_FILE_CHOOSER (priv->chooser_image), filter);
   gtk_file_chooser_set_filter (GTK_FILE_CHOOSER (priv->chooser_image), filter);
 
-  frame = xfce_create_framebox_with_content (_("Image to burn"), priv->chooser_image);
+  frame = xfce_gtk_frame_box_new_with_content (_("Image to burn"), priv->chooser_image);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
   
@@ -188,7 +188,7 @@
   priv->device_box = xfburn_device_box_new (SHOW_CD_WRITERS | SHOW_CDRW_WRITERS | SHOW_DVD_WRITERS | SHOW_MODE_SELECTION | SHOW_SPEED_SELECTION);
   gtk_widget_show (priv->device_box);
   
-  frame = xfce_create_framebox_with_content (_("Burning device"), priv->device_box);
+  frame = xfce_gtk_frame_box_new_with_content (_("Burning device"), priv->device_box);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -196,7 +196,7 @@
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox);
 
-  frame = xfce_create_framebox_with_content (_("Options"), vbox);
+  frame = xfce_gtk_frame_box_new_with_content (_("Options"), vbox);
   gtk_widget_show (frame);
   gtk_box_pack_start (box, frame, FALSE, FALSE, BORDER);
 
@@ -219,7 +219,7 @@
   gtk_widget_show (button);
   gtk_dialog_add_action_widget (GTK_DIALOG (obj), button, GTK_RESPONSE_CANCEL);
 
-  priv->burn_button = xfce_create_mixed_button ("xfburn-burn-cd", _("_Burn image"));
+  priv->burn_button = xfce_gtk_button_new_mixed ("xfburn-burn-cd", _("_Burn image"));
   gtk_widget_show (priv->burn_button);
   g_signal_connect (G_OBJECT (priv->burn_button), "clicked", G_CALLBACK (cb_clicked_ok), obj);
   gtk_container_add (GTK_CONTAINER( GTK_DIALOG(obj)->action_area), priv->burn_button);
@@ -400,7 +400,7 @@
 void
 burn_image_dialog_error (XfburnBurnImageDialog * dialog, const gchar * msg_error)
 {
-  xfce_err (msg_error);
+  xfce_dialog_show_error (NULL, NULL, "%s", msg_error); 
 }
 
 static void
diff -ur xfburn-0.4.3/xfburn/xfburn-burn-image-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-burn-image-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-burn-image-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-burn-image-dialog.h	2011-02-26 16:21:06.859320585 +0100
@@ -25,7 +25,7 @@
 
 #include <gtk/gtk.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "xfburn-global.h"
 
diff -ur xfburn-0.4.3/xfburn/xfburn-copy-cd-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-copy-cd-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-copy-cd-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-copy-cd-dialog.h	2011-02-26 16:21:06.866320585 +0100
@@ -25,7 +25,7 @@
 
 #include <gtk/gtk.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 G_BEGIN_DECLS
 
diff -ur xfburn-0.4.3/xfburn/xfburn-copy-dvd-dialog.h xfburn-0.4.3-4ui/xfburn/xfburn-copy-dvd-dialog.h
--- xfburn-0.4.3/xfburn/xfburn-copy-dvd-dialog.h	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-copy-dvd-dialog.h	2011-02-26 16:21:06.869320585 +0100
@@ -25,7 +25,7 @@
 
 #include <gtk/gtk.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 G_BEGIN_DECLS
 
diff -ur xfburn-0.4.3/xfburn/xfburn-data-composition.c xfburn-0.4.3-4ui/xfburn/xfburn-data-composition.c
--- xfburn-0.4.3/xfburn/xfburn-data-composition.c	2011-02-26 20:25:11.177013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-data-composition.c	2011-02-26 20:11:15.445153553 +0100
@@ -35,7 +35,7 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <gio/gio.h>
 
@@ -671,7 +671,7 @@
   GtkTreePath *real_path;
 
   if (strlen (newtext) == 0) {
-    xfce_err (_("You must give a name to the file."));
+    xfce_dialog_show_error (NULL, NULL, _("You must give a name to the file."));
     return;
   }
     
@@ -680,7 +680,7 @@
 
   if (gtk_tree_model_get_iter (model, &iter, real_path)) {
     if (file_exists_on_same_level (model, real_path, TRUE, newtext)) {
-      xfce_err (_("A file with the same name is already present in the composition."));
+      xfce_dialog_show_error (NULL, NULL, _("A file with the same name is already present in the composition."));
     }
     else {
       gtk_tree_store_set (GTK_TREE_STORE (model), &iter, DATA_COMPOSITION_COLUMN_CONTENT, newtext, -1);
@@ -1116,7 +1116,7 @@
     
     gdk_threads_enter ();
     if (file_exists_on_same_level (model, tree_path, FALSE, name)) {
-      xfce_err (_("A file with the same name is already present in the composition."));
+      xfce_dialog_show_error (NULL, NULL, _("A file with the same name is already present in the composition."));
 
       gtk_tree_path_free (tree_path);
       gdk_threads_leave ();
@@ -1195,7 +1195,7 @@
 
       if (s.st_size > MAXIMUM_ISO_FILE_SIZE) {
         gdk_threads_enter ();
-        xfce_err (_("%s cannot be added to the composition, because it exceeds the maximum allowed file size for iso9660."), path);
+        xfce_dialog_show_error (NULL, NULL, _("%s cannot be added to the composition, because it exceeds the maximum allowed file size for iso9660."), path);
         gdk_threads_leave ();
 
         return FALSE;
@@ -1448,7 +1448,7 @@
       }
     
       if (file_exists_on_same_level (model, path_level, FALSE, name)) {
-        xfce_warn (_("A file named \"%s\" already exists in this directory, the file hasn't been added."), name);
+	xfce_dialog_show_warning(NULL, NULL, _("A file named \"%s\" already exists in this directory, the file hasn't been added."), name);
         goto cleanup;
       }
       
@@ -1861,7 +1861,7 @@
 
         if (r == 0) {
           /* The first string is the renamed name, the second one the original name */
-          xfce_warn (_("Duplicate filename '%s' for '%s'"), name, src);
+	  xfce_dialog_show_warning(NULL, NULL, _("Duplicate filename '%s' for '%s'"), name, src);
 
           g_free (basename);
           g_free (name);
diff -ur xfburn-0.4.3/xfburn/xfburn-data-disc-usage.c xfburn-0.4.3-4ui/xfburn/xfburn-data-disc-usage.c
--- xfburn-0.4.3/xfburn/xfburn-data-disc-usage.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-data-disc-usage.c	2011-02-26 16:21:03.628320584 +0100
@@ -22,7 +22,6 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
 
 #include "xfburn-disc-usage.h"
 #include "xfburn-data-disc-usage.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-device-box.c xfburn-0.4.3-4ui/xfburn/xfburn-device-box.c
--- xfburn-0.4.3/xfburn/xfburn-device-box.c	2011-02-26 20:25:11.178013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-device-box.c	2011-02-26 16:21:03.642320584 +0100
@@ -23,7 +23,6 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
 
 #include "xfburn-device-list.h"
 #include "xfburn-device-box.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-directory-browser.c xfburn-0.4.3-4ui/xfburn/xfburn-directory-browser.c
--- xfburn-0.4.3/xfburn/xfburn-directory-browser.c	2011-02-26 20:25:11.182013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-directory-browser.c	2011-02-26 16:21:03.667320584 +0100
@@ -26,7 +26,6 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
 
 #include <gio/gio.h>
 
diff -ur xfburn-0.4.3/xfburn/xfburn-disc-usage.c xfburn-0.4.3-4ui/xfburn/xfburn-disc-usage.c
--- xfburn-0.4.3/xfburn/xfburn-disc-usage.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-disc-usage.c	2011-02-26 20:14:37.314013551 +0100
@@ -30,7 +30,7 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "xfburn-disc-usage.h"
 #include "xfburn-global.h"
@@ -160,7 +160,7 @@
   gtk_box_pack_start (GTK_BOX (disc_usage), disc_usage->combo, FALSE, FALSE, BORDER);
   gtk_widget_show (disc_usage->combo);
 
-  disc_usage->button = xfce_create_mixed_button (XFBURN_STOCK_BURN_CD, _("Proceed to Burn"));
+  disc_usage->button = xfce_gtk_button_new_mixed (XFBURN_STOCK_BURN_CD, _("Proceed to Burn"));
   gtk_box_pack_start (GTK_BOX (disc_usage), disc_usage->button, FALSE, FALSE, BORDER);
   gtk_widget_set_sensitive (disc_usage->button, FALSE);
   gtk_widget_show (disc_usage->button);
@@ -200,7 +200,7 @@
   if (du->size <= class->labels[gtk_combo_box_get_active (GTK_COMBO_BOX (du->combo))].size) {
     g_signal_emit (G_OBJECT (du), signals[BEGIN_BURN], 0);
   } else {
-    xfce_err (_("You are trying to burn more onto the disc than it can hold."));
+    xfce_dialog_show_error (NULL, NULL, _("You are trying to burn more onto the disc than it can hold."));
   }
 }
 
diff -ur xfburn-0.4.3/xfburn/xfburn-file-browser.c xfburn-0.4.3-4ui/xfburn/xfburn-file-browser.c
--- xfburn-0.4.3/xfburn/xfburn-file-browser.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-file-browser.c	2011-02-26 16:21:03.678320584 +0100
@@ -26,7 +26,6 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
 
 #include "xfburn-file-browser.h"
 #include "xfburn-fs-browser.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-fs-browser.c xfburn-0.4.3-4ui/xfburn/xfburn-fs-browser.c
--- xfburn-0.4.3/xfburn/xfburn-fs-browser.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-fs-browser.c	2011-02-26 16:21:03.684320584 +0100
@@ -24,7 +24,6 @@
 #include <string.h>
 #endif
 
-#include <libxfcegui4/libxfcegui4.h>
 #include <libxfce4util/libxfce4util.h>
 
 #include "xfburn-fs-browser.h"
diff -ur xfburn-0.4.3/xfburn/xfburn-main.c xfburn-0.4.3-4ui/xfburn/xfburn-main.c
--- xfburn-0.4.3/xfburn/xfburn-main.c	2011-02-26 20:25:11.186013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-main.c	2011-02-26 20:10:39.291239575 +0100
@@ -29,7 +29,7 @@
 #include <gtk/gtk.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #ifdef HAVE_GST
 #include <gst/gst.h>
@@ -217,7 +217,7 @@
 
   if (!burn_initialize ()) {
     g_critical ("Unable to initialize libburn");
-    xfce_err (_("Unable to initialize the burning backend."));
+    xfce_dialog_show_error (NULL, NULL, _("Unable to initialize the burning backend."));
     gdk_threads_leave ();
     return EXIT_FAILURE;
   }
@@ -274,7 +274,7 @@
 #ifdef HAVE_GUDEV
   error_msg = xfburn_udev_manager_create_global ();
   if (error_msg) {
-    xfce_err (error_msg);
+    xfce_dialog_show_error (NULL, NULL, "%s", error_msg);
     gdk_threads_leave ();
     burn_finish ();
     return EXIT_FAILURE;
@@ -328,7 +328,7 @@
   }
 
   if (!xfburn_transcoder_is_initialized (transcoder, &error)) {
-    xfce_warn (_("Failed to initialize %s transcoder: %s\n\t(falling back to basic implementation)"), xfburn_transcoder_get_name (transcoder), error->message);
+    xfce_dialog_show_warning(NULL, NULL, _("Failed to initialize %s transcoder: %s\n\t(falling back to basic implementation)"), xfburn_transcoder_get_name (transcoder), error->message);
     g_error_free (error);
     g_object_unref (transcoder);
     transcoder = XFBURN_TRANSCODER (xfburn_transcoder_basic_new ());
@@ -361,7 +361,7 @@
       if (g_file_test (image_fullname, G_FILE_TEST_EXISTS))
 	xfburn_burn_image_dialog_set_filechooser_name (dialog, image_fullname);
       else
-	xfce_err ( g_strdup_printf ( _("Image file '%s' does not exist."), image_fullname));
+        xfce_dialog_show_error (NULL, NULL, _("Image file '%s' does not exist."), image_fullname);
     }
 
     gtk_dialog_run (GTK_DIALOG (dialog));
diff -ur xfburn-0.4.3/xfburn/xfburn-main-window.c xfburn-0.4.3-4ui/xfburn/xfburn-main-window.c
--- xfburn-0.4.3/xfburn/xfburn-main-window.c	2009-12-04 08:35:12.000000000 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-main-window.c	2011-02-26 20:22:57.325013554 +0100
@@ -22,7 +22,7 @@
 
 #include <gtk/gtk.h>
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <exo/exo.h>
 
@@ -479,79 +479,69 @@
 static void
 action_about (GtkAction * action, XfburnMainWindow * window)
 {
-  XfceAboutInfo *info;
-  GtkWidget *dialog;
   gint x, y;
-  GdkPixbuf *icon;
-  guint n;
-
-  static const struct
-  {
-    gchar *name, *email, *language;
-  } translators[] = {
-    {"Mohamed Magdy", "mohamed.m.k@gmail.com", "ar",},
-    {"Pau Rul lan Ferragut", "paurullan@bulma.net", "ca",},
-    {"Michal Várady", "miko.vaji@gmail.com", "cs",},
-    {"Enrico Tröger", "enrico.troeger@uvena.de", "de",},
-    {"Fabian Nowak", "timstery@arcor.de", "de",},
-    {"Nico Schümann", "nico@prog.nico22.de", "de",},
-    {"Stavros Giannouris", "stavrosg2002@freemail.gr", "el",},
-    {"Jeff Bailes", "thepizzaking@gmail.com", "en_GB",},
-    {"Diego Rodriguez", "dieymir@yahoo.es", "es",},
-    {"Kristjan Siimson", "kristjan.siimson@gmail.com", "et",},
-    {"Piarres Beobide", "pi@beobide.net", "eu",},
-    {"Jari Rahkonen", "jari.rahkonen@pp1.inet.fi", "fi",},
-    {"Etienne Collet", "xanaxlnx@gmail.com", "fr",},
-    {"Maximilian Schleiss", "maximilian@xfce.org", "fr",},
-    {"Attila Szervác", "sas@321.hu", "hu",},
-    {"Daichi Kawahata", "daichi@xfce.org", "ja",},
-    {"ByungHyun Choi", "byunghyun.choi@debianusers.org", "kr",},
-    {"Mantas", "mantaz@users.sourceforge.net", "lt",},
-    {"Rihards Prieditis", "RPrieditis@inbox.lv", "lv",},
-    {"Terje Uriansrud", "ter@operamail.com", "nb_NO",},
-    {"Stephan Arts", "psybsd@gmail.com", "nl",},
-    {"Szymon Kałasz", "szymon_maestro@gazeta.pl", "pl",},
-    {"Fábio Nogueira", "deb-user-ba@ubuntu.com", "pt_BR",},
-    {"Og Maciel", "omaciel@xfce.org", "pt_BR",},
-    {"Nuno Miguel", "nunis@netcabo.pt", "pt_PT",},
-    {"Sergey Fedoseev", "fedoseev.sergey@gmail.com", "ru",},
-    {"Besnik Bleta", "besnik@programeshqip.org", "sq",},
-    {"Maxim V. Dziumanenko", "mvd@mylinux.com.ua", "uk",},
-    {"Dmitry Nikitin", "", "uk",},
-    {"ﻢﺤﻣﺩ ﻊﻠﻳ ﺎﻠﻤﻜﻳ", "makki.ma@gmail.com", "ur",},
-    {"正龙 赵", "longer.zhao@gmail.com", "zh_CN",},
-    {"Cosmo Chene", "cosmolax@gmail.com", "zh_TW",},
-  };
+  GdkPixbuf *icon = NULL;
+  const gchar *auth[] = { "David Mohr david@mcbf.net Author/Maintainer",
+	  		  "Mario Đanić mario@libburnia-project.org Author/Maintainer",
+			  "Jean-François Wauthy pollux@xfce.org Retired author/maintainer",
+			  NULL };
+  const gchar *translators =
+    "Mohamed Magdy mohamed.m.k@gmail.com ar\n"
+    "Pau Rul lan Ferragut paurullan@bulma.net ca\n"
+    "Michal Várady miko.vaji@gmail.com cs\n"
+    "Enrico Tröger enrico.troeger@uvena.de de\n"
+    "Fabian Nowak timstery@arcor.de de\n"
+    "Nico Schümann nico@prog.nico22.de de\n"
+    "Stavros Giannouris stavrosg2002@freemail.gr el\n"
+    "Jeff Bailes thepizzaking@gmail.com en_GB\n"
+    "Diego Rodriguez dieymir@yahoo.es es\n"
+    "Kristjan Siimson kristjan.siimson@gmail.com et\n"
+    "Piarres Beobide pi@beobide.net eu\n"
+    "Jari Rahkonen jari.rahkonen@pp1.inet.fi fi\n"
+    "Etienne Collet xanaxlnx@gmail.com fr\n"
+    "Maximilian Schleiss maximilian@xfce.org fr\n"
+    "Attila Szervác sas@321.hu hu\n"
+    "Daichi Kawahata daichi@xfce.org ja\n"
+    "ByungHyun Choi byunghyun.choi@debianusers.org kr\n"
+    "Mantas mantaz@users.sourceforge.net lt\n"
+    "Rihards Prieditis RPrieditis@inbox.lv lv\n"
+    "Terje Uriansrud ter@operamail.com nb_NO\n"
+    "Stephan Arts psybsd@gmail.com nl\n"
+    "Szymon Kałasz szymon_maestro@gazeta.pl pl\n"
+    "Fábio Nogueira deb-user-ba@ubuntu.com pt_BR\n"
+    "Og Maciel omaciel@xfce.org pt_BR\n"
+    "Nuno Miguel nunis@netcabo.pt pt_PT\n"
+    "Sergey Fedoseev fedoseev.sergey@gmail.com ru\n"
+    "Besnik Bleta besnik@programeshqip.org sq\n"
+    "Maxim V. Dziumanenko mvd@mylinux.com.ua uk\n"
+    "Dmitry Nikitin  uk\n"
+    "ﻢﺤﻣﺩ ﻊﻠﻳ ﺎﻠﻤﻜﻳ makki.ma@gmail.com ur\n"
+    "正龙 赵 longer.zhao@gmail.com zh_CN\n"
+    "Cosmo Chene cosmolax@gmail.com zh_TW\n";
 
   gtk_icon_size_lookup (GTK_ICON_SIZE_DIALOG, &x, &y);
-  icon = xfce_themed_icon_load ("media-optical", x);
+  icon = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), "media-optical", x, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
   if (!icon)
-    icon = xfce_themed_icon_load ("media-cdrom", x);
+    icon = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), "media-cdrom", x, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
   if (!icon)
-    icon = xfce_themed_icon_load (GTK_STOCK_CDROM, x);
-
-  info = xfce_about_info_new ("Xfburn", VERSION, _("Another cd burning GUI"),
-                              XFCE_COPYRIGHT_TEXT ("2005-2008", "David Mohr, Mario Đanić, Jean-François Wauthy"), XFCE_LICENSE_GPL);
-  xfce_about_info_set_homepage (info, "http://www.xfce.org/projects/xfburn");
-  xfce_about_info_add_credit (info, "David Mohr", "david@mcbf.net", _("Author/Maintainer"));
-  xfce_about_info_add_credit (info, "Mario Đanić", "mario@libburnia-project.org", _("Author/Maintainer"));
-  xfce_about_info_add_credit (info, "Jean-François Wauthy", "pollux@xfce.org", _("Retired author/maintainer"));
-  
-
-  for (n = 0; n < G_N_ELEMENTS (translators); ++n) {
-    gchar *s;
+    icon = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), GTK_STOCK_CDROM, x, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
 
-    s = g_strdup_printf (_("Translator (%s)"), translators[n].language);
-    xfce_about_info_add_credit (info, translators[n].name, translators[n].email, s);
-    g_free (s);
-  }
-
-  dialog = xfce_about_dialog_new_with_values (GTK_WINDOW (window), info, icon);
-  gtk_widget_set_size_request (GTK_WIDGET (dialog), 400, 300);
-  gtk_dialog_run (GTK_DIALOG (dialog));
-  gtk_widget_destroy (dialog);
+#if !GTK_CHECK_VERSION (2, 18, 0)
+  gtk_about_dialog_set_email_hook (exo_gtk_url_about_dialog_hook, NULL, NULL);
+  gtk_about_dialog_set_url_hook (exo_gtk_url_about_dialog_hook, NULL, NULL);
+#endif
+  gtk_show_about_dialog(GTK_WINDOW (window),
+		  "logo", icon,
+		  "program-name", "Xfburn",
+		  "license", xfce_get_license_text (XFCE_LICENSE_TEXT_GPL),
+		  "version", VERSION,
+		  "comments", _("Another cd burning GUI"),
+		  "website", "http://www.xfce.org/projects/xfburn",
+		  "copyright", "2005-2008 David Mohr, Mario Đanić, Jean-François Wauthy",
+		  "authors", auth, 
+		  "translator-credits", translators,
+		  NULL);
 
-  xfce_about_info_free (info);
   if (G_LIKELY (icon != NULL))
     g_object_unref (G_OBJECT (icon));
 }
diff -ur xfburn-0.4.3/xfburn/xfburn-perform-burn.h xfburn-0.4.3-4ui/xfburn/xfburn-perform-burn.h
--- xfburn-0.4.3/xfburn/xfburn-perform-burn.h	2009-10-30 02:07:24.000000000 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-perform-burn.h	2011-02-26 16:21:06.901320585 +0100
@@ -25,7 +25,7 @@
 #endif
 
 #include <gtk/gtk.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libburn.h>
 
diff -ur xfburn-0.4.3/xfburn/xfburn-preferences-dialog.c xfburn-0.4.3-4ui/xfburn/xfburn-preferences-dialog.c
--- xfburn-0.4.3/xfburn/xfburn-preferences-dialog.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-preferences-dialog.c	2011-02-26 20:18:38.177013556 +0100
@@ -20,7 +20,7 @@
 #include <config.h>
 #endif /* !HAVE_CONFIG_H */
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 #include <exo/exo.h>
 
 #include "xfburn-preferences-dialog.h"
@@ -185,7 +185,7 @@
   vbox2 = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox2);
 
-  frame = xfce_create_framebox_with_content (_("Temporary directory"), vbox2);
+  frame = xfce_gtk_frame_box_new_with_content (_("Temporary directory"), vbox2);
   gtk_box_pack_start (GTK_BOX (vbox), frame, FALSE, FALSE, BORDER);
   gtk_widget_show (frame);
 
@@ -200,7 +200,7 @@
   vbox2 = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox2);
 
-  frame = xfce_create_framebox_with_content (_("File browser"), vbox2);
+  frame = xfce_gtk_frame_box_new_with_content (_("File browser"), vbox2);
   gtk_box_pack_start (GTK_BOX (vbox), frame, FALSE, FALSE, BORDER);
   gtk_widget_show (frame);
 
@@ -240,7 +240,7 @@
   vbox2 = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox2);
 
-  frame = xfce_create_framebox_with_content (_("Detected devices"), vbox2);
+  frame = xfce_gtk_frame_box_new_with_content (_("Detected devices"), vbox2);
   gtk_box_pack_start (GTK_BOX (vbox), frame, TRUE, TRUE, BORDER);
   gtk_widget_show (frame);
 
@@ -292,17 +292,17 @@
   gtk_box_pack_start (GTK_BOX (vbox2), hbox, FALSE, FALSE, BORDER);
   gtk_widget_show (hbox);
 
-  priv->button_scan = xfce_create_mixed_button (GTK_STOCK_CDROM, _("Sc_an for devices"));
+  priv->button_scan = xfce_gtk_button_new_mixed (GTK_STOCK_CDROM, _("Sc_an for devices"));
   gtk_box_pack_end (GTK_BOX (hbox), priv->button_scan, FALSE, FALSE, BORDER);
   g_signal_connect (G_OBJECT (priv->button_scan), "clicked", G_CALLBACK (scan_button_clicked_cb), obj);
   gtk_widget_show (priv->button_scan);
 
   gtk_icon_size_lookup (GTK_ICON_SIZE_DIALOG, &x, &y);
-  icon = xfce_themed_icon_load ("media-optical", x);
+  icon = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), "media-optical", x, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
   if (!icon)
-    icon = xfce_themed_icon_load ("media-cdrom", x);
+    icon = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), "media-cdrom", x, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
   if (!icon)
-    icon = xfce_themed_icon_load (GTK_STOCK_CDROM, x);
+    icon = gtk_icon_theme_load_icon ( gtk_icon_theme_get_default(), GTK_STOCK_CDROM, x, GTK_ICON_LOOKUP_GENERIC_FALLBACK, NULL);
 
   
   gtk_list_store_append (icon_store, &iter);
@@ -326,7 +326,7 @@
   vbox3 = gtk_vbox_new (FALSE, 0);
   gtk_widget_show (vbox3);
 
-  frame = xfce_create_framebox_with_content (_("FIFO buffer size (in kb)"), vbox3);
+  frame = xfce_gtk_frame_box_new_with_content (_("FIFO buffer size (in kb)"), vbox3);
   gtk_box_pack_start (GTK_BOX (vbox), frame, TRUE, TRUE, BORDER);
   gtk_widget_show (frame);
 
@@ -460,7 +460,7 @@
 static void 
 cb_show_hidden_clicked (GtkButton * Button, gpointer user_data)
 {
-  xfce_warn (_("Changing this setting only takes full effect after a program restart."));
+  xfce_dialog_show_warning(NULL, NULL, _("Changing this setting only takes full effect after a program restart."));
 }
 
 /* public */
diff -ur xfburn-0.4.3/xfburn/xfburn-progress-dialog.c xfburn-0.4.3-4ui/xfburn/xfburn-progress-dialog.c
--- xfburn-0.4.3/xfburn/xfburn-progress-dialog.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-progress-dialog.c	2011-02-26 20:10:07.506140054 +0100
@@ -30,7 +30,7 @@
 
 #include <gtk/gtk.h>
 
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "xfburn-progress-dialog.h"
 #include "xfburn-global.h"
@@ -603,7 +603,7 @@
   xfburn_progress_dialog_set_status_with_text (dialog, XFBURN_PROGRESS_DIALOG_STATUS_FAILED, _("Failure"));
 
   gdk_threads_enter ();
-  xfce_err (msg_error);
+  xfce_dialog_show_error (NULL, NULL, "%s", msg_error);
   gdk_threads_leave ();
 }
 
diff -ur xfburn-0.4.3/xfburn/xfburn-transcoder-basic.c xfburn-0.4.3-4ui/xfburn/xfburn-transcoder-basic.c
--- xfburn-0.4.3/xfburn/xfburn-transcoder-basic.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-transcoder-basic.c	2011-02-26 20:07:07.312282054 +0100
@@ -33,7 +33,7 @@
 #include <errno.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include <libburn.h>
 
@@ -235,7 +235,7 @@
   fd = open (path, 0);
 
   if (fd == -1) {
-    xfce_warn (_("Could not open %s."), path);
+    xfce_dialog_show_warning(NULL, NULL, _("Could not open %s."), path);
     return FALSE;
   }
 
diff -ur xfburn-0.4.3/xfburn/xfburn-transcoder-gst.c xfburn-0.4.3-4ui/xfburn/xfburn-transcoder-gst.c
--- xfburn-0.4.3/xfburn/xfburn-transcoder-gst.c	2009-10-24 23:18:10.000000000 +0200
+++ xfburn-0.4.3-4ui/xfburn/xfburn-transcoder-gst.c	2011-02-26 16:21:03.773320584 +0100
@@ -36,7 +36,6 @@
 #include <errno.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
 
 #include <libburn.h>
 
diff -ur xfburn-0.4.3/xfburn/xfburn-udev-manager.c xfburn-0.4.3-4ui/xfburn/xfburn-udev-manager.c
--- xfburn-0.4.3/xfburn/xfburn-udev-manager.c	2011-02-26 20:25:11.187013553 +0100
+++ xfburn-0.4.3-4ui/xfburn/xfburn-udev-manager.c	2011-02-26 19:59:29.621029555 +0100
@@ -33,7 +33,7 @@
 #include <errno.h>
 
 #include <libxfce4util/libxfce4util.h>
-#include <libxfcegui4/libxfcegui4.h>
+#include <libxfce4ui/libxfce4ui.h>
 
 #include "xfburn-global.h"
 #include "xfburn-progress-dialog.h"
@@ -514,7 +514,7 @@
   if (unmounted)
     g_message ("Unmounted '%s'", mp);
   else {
-    xfce_err ("Failed to unmount '%s'. Drive cannot be used for burning.", mp);
+    xfce_dialog_show_error (NULL, NULL, "Failed to unmount '%s'. Drive cannot be used for burning.", mp);
     DBG ("Failed to unmount '%s'", mp);
   }
 
