From a02e5b6d422191a3674b13f1becf9c2ec90e83a5 Mon Sep 17 00:00:00 2001
From: gbjbaanb <gbjbaanb@users.sourceforce.net>
Date: Tue, 24 Mar 2015 15:17:28 +0000
Subject: [PATCH 2/6] minor fix to allow full lines to be input to demo queries

---
 src/demo/searchfiles/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/demo/searchfiles/main.cpp b/src/demo/searchfiles/main.cpp
index de4d303..748003c 100644
--- a/src/demo/searchfiles/main.cpp
+++ b/src/demo/searchfiles/main.cpp
@@ -265,7 +265,7 @@ int main(int argc, char* argv[]) {
                 }
             } else {
                 std::wcout << L"Enter query: ";
-                std::wcin >> line;
+				getline(std::wcin, line);
             }
             boost::trim(line);
 
-- 
2.3.6

