--- CMakeLists.txt.orig	2014-02-18 18:18:31.899698305 -0300
+++ CMakeLists.txt	2014-02-18 18:18:37.612494917 -0300
@@ -5,8 +5,8 @@ project(lucene++-base)
 #These versions match the Lucene version
 SET(LUCENE++_VERSION_MAJOR "3")
 SET(LUCENE++_VERSION_MINOR "0")
-SET(LUCENE++_VERSION_REVISION "3")
-SET(LUCENE++_VERSION_PATCH "4")
+SET(LUCENE++_VERSION_REVISION "5")
+SET(LUCENE++_VERSION_PATCH "0")
 
 # SOVERSION information
 #Must be incremented for releases if the api is not backwards compatible
