diff -up curl-7.20.1/tests/data/test513.delay curl-7.20.1/tests/data/test513
--- curl-7.20.1/tests/data/test513.delay	2010-04-22 10:00:08.326108258 +0100
+++ curl-7.20.1/tests/data/test513	2010-04-22 11:09:08.342100834 +0100
@@ -17,7 +17,7 @@ lib513
  <name>
 send HTTP POST using read callback that returns CURL_READFUNC_ABORT
  </name>
- <command>
+ <command delay="1">
 http://%HOSTIP:%HTTPPORT/513
 </command>
 </client>
diff -up curl-7.20.1/tests/data/test514.delay curl-7.20.1/tests/data/test514
--- curl-7.20.1/tests/data/test514.delay	2010-04-22 10:00:08.326108258 +0100
+++ curl-7.20.1/tests/data/test514	2010-04-22 11:09:40.192203636 +0100
@@ -29,7 +29,7 @@ lib514
  <name>
 First set options to POST and then to make HEAD
  </name>
- <command>
+ <command delay="1">
 http://%HOSTIP:%HTTPPORT/514
 </command>
 </client>
