diff -ruNp seabios-0.6.2.orig/Makefile seabios-0.6.2/Makefile
--- seabios-0.6.2.orig/Makefile	2011-07-13 10:52:50.000000000 -0500
+++ seabios-0.6.2/Makefile	2011-07-13 11:14:14.000000000 -0500
@@ -5,7 +5,7 @@
 # This file may be distributed under the terms of the GNU LGPLv3 license.
 
 # Program version
-VERSION=0.6.2-$(shell date +"%Y%m%d_%H%M%S")-$(shell hostname)
+VERSION=0.6.2
 
 # Output directory
 OUT=out/
diff -ruNp seabios-0.6.2.orig/src/ahci.c seabios-0.6.2/src/ahci.c
--- seabios-0.6.2.orig/src/ahci.c	2011-07-13 10:52:50.000000000 -0500
+++ seabios-0.6.2/src/ahci.c	2011-07-13 11:14:14.000000000 -0500
@@ -408,7 +408,6 @@ static void
 ahci_detect(void *data)
 {
     struct ahci_ctrl_s *ctrl = data;
-    struct ahci_port_s *port;
     u32 pnr, max;
     int rc;
 
@@ -422,7 +421,7 @@ ahci_detect(void *data)
         dprintf(1, "AHCI/%d: link %s\n", pnr, rc == 0 ? "up" : "down");
         if (rc != 0)
             continue;
-        port = ahci_port_init(ctrl, pnr);
+        ahci_port_init(ctrl, pnr);
     }
 }
 
diff -ruNp seabios-0.6.2.orig/src/bregs.h seabios-0.6.2/src/bregs.h
--- seabios-0.6.2.orig/src/bregs.h	2011-07-13 10:52:50.000000000 -0500
+++ seabios-0.6.2/src/bregs.h	2011-07-13 11:16:04.000000000 -0500
@@ -37,9 +37,9 @@
 struct bregs {
     u16 ds;
     u16 es;
-    UREG(edi, di, di_hi, di_lo);
-    UREG(esi, si, si_hi, si_lo);
-    UREG(ebp, bp, bp_hi, bp_lo);
+    UREG(edi, di, di8u, di8l);
+    UREG(esi, si, si8u, si8l);
+    UREG(ebp, bp, bp8u, bp8l);
     UREG(ebx, bx, bh, bl);
     UREG(edx, dx, dh, dl);
     UREG(ecx, cx, ch, cl);
Binary files seabios-0.6.2.orig/tools/layoutrom.pyc and seabios-0.6.2/tools/layoutrom.pyc differ
