diff -up exiv2-0.24/src/CMakeLists.txt.mandir exiv2-0.24/src/CMakeLists.txt
--- exiv2-0.24/src/CMakeLists.txt.mandir	2013-03-29 22:32:26.000000000 -0500
+++ exiv2-0.24/src/CMakeLists.txt	2013-12-02 12:14:04.264728159 -0600
@@ -284,7 +284,7 @@ INSTALL( FILES ${LIBEXIV2_HDR} DESTINATI
 
 # ******************************************************************************
 # Man page
-INSTALL( FILES exiv2.1 DESTINATION man/man1 )
+INSTALL( FILES exiv2.1 DESTINATION share/man/man1 )
 
 # That's all Folks!
 ##
