diff -up exiv2-0.20/src/Makefile.deps exiv2-0.20/src/Makefile
--- exiv2-0.20/src/Makefile.deps	2010-05-25 10:25:02.000000000 -0500
+++ exiv2-0.20/src/Makefile	2010-05-30 13:33:38.181957187 -0500
@@ -195,11 +195,19 @@ $(CCOBJ): %.o: %.cpp
 	@$(LIBTOOL) --mode=compile $(COMPILE.cc) -DEXV_BUILDING_LIB=1 -o $@ $<
 	@$(MAKEDEPEND)
 	@$(POSTDEPEND)
+# The dependancies here are wrong.  It generates .lo output,
+# so things are rebuilt *every* time 'make' is called.
+# Let's try a quick-n-dirty hack -- Rex
+	touch $@
 
 $(COBJ): %.o: %.c
 	@$(LIBTOOL) --mode=compile $(COMPILE.c) -DEXV_BUILDING_LIB=1 -o $@ $<
 	@$(MAKEDEPEND)
 	@$(POSTDEPEND)
+# The dependancies here are wrong.  It generates .lo output,
+# so things are rebuilt *every* time 'make' is called.
+# Let's try a quick-n-dirty hack -- Rex
+	touch $@
 
 $(sort $(BINOBJ) $(EXIV2OBJ) $(MCOBJ) path-test.o): %.o: %.cpp
 	$(COMPILE.cc) -o $@ $<
diff -up exiv2-0.20/xmpsdk/src/Makefile.deps exiv2-0.20/xmpsdk/src/Makefile
--- exiv2-0.20/xmpsdk/src/Makefile.deps	2010-04-03 02:53:30.000000000 -0500
+++ exiv2-0.20/xmpsdk/src/Makefile	2010-05-30 13:36:16.004947131 -0500
@@ -108,6 +108,10 @@ $(CCOBJ): %.o: %.cpp
 	@$(LIBTOOL) --mode=compile $(COMPILE.cc) -o $@ $<
 	@$(MAKEDEPEND)
 	@$(POSTDEPEND)
+	# The dependancies here are wrong.  It generates .lo output,
+	# so things are rebuilt *every* time 'make' is called.
+	# Let's try a quick-n-dirty hack -- Rex
+	@touch $@
 
 %.ii: %.cpp
 	set -e; \
