diff -rupN sqlitebrowser-3.6.0/CMakeLists.txt sqlitebrowser-3.6.0-new/CMakeLists.txt
--- sqlitebrowser-3.6.0/CMakeLists.txt	2015-04-27 11:31:47.000000000 +0200
+++ sqlitebrowser-3.6.0-new/CMakeLists.txt	2015-04-30 09:46:33.525874432 +0200
@@ -21,15 +21,9 @@ if(WIN32 AND MSVC)
 	set(CMAKE_PREFIX_PATH "${QT5_PATH};${SQLITE3_PATH}")
 endif()
 
-set(ANTLR_DIR libs/antlr-2.7.7)
-set(QHEXEDIT_DIR libs/qhexedit)
-set(QCUSTOMPLOT_DIR libs/qcustomplot-source)
-set(QSCINTILLA_DIR libs/qscintilla/Qt4Qt5)
-
-add_subdirectory(${ANTLR_DIR})
-add_subdirectory(${QHEXEDIT_DIR})
-add_subdirectory(${QCUSTOMPLOT_DIR})
-add_subdirectory(${QSCINTILLA_DIR})
+find_package(PkgConfig)
+pkg_check_modules(QHEXEDIT2 qhexedit2-qt5)
+pkg_check_modules(QCUSTOMPLOT qcustomplot-qt5)
 
 if(USE_QT5)
 	find_package(Qt5Widgets REQUIRED)
@@ -207,9 +201,8 @@ endif()
 
 include_directories(
 		"${CMAKE_CURRENT_BINARY_DIR}"
-		${ANTLR_DIR}
-		${QHEXEDIT_DIR}
-		${QCUSTOMPLOT_DIR}
+		${QHEXEDIT2_INCLUDE_DIRS}
+		${QCUSTOMPLOT_INCLUDE_DIRS}
 		${QSCINTILLA_DIR}
 		${ADDITIONAL_INCLUDE_PATHS}
 		src)
@@ -227,19 +220,16 @@ if(USE_QT5)
     qt5_use_modules(${PROJECT_NAME} Gui Widgets Network Test PrintSupport)
     set(QT_LIBRARIES "")
 endif()
-add_dependencies(${PROJECT_NAME} antlr qhexedit qcustomplot qscintilla2)
 
 link_directories(
-		"${CMAKE_CURRENT_BINARY_DIR}/${ANTLR_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QHEXEDIT_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QCUSTOMPLOT_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QSCINTILLA_DIR}")
+		${QHEXEDIT2_LIBRARY_DIRS}
+		${QCUSTOMPLOT_LIBRARY_DIRS})
 
 target_link_libraries(${PROJECT_NAME}
 		antlr
-		qhexedit
-		qcustomplot
-		qscintilla2
+		${QHEXEDIT2_LIBRARIES}
+		${QCUSTOMPLOT_LIBRARIES}
+		qscintilla2-qt5
 		${QT_LIBRARIES}
 		${WIN32_STATIC_LINK}
 		${LIBSQLITE}
diff -rupN sqlitebrowser-3.6.0/src/EditDialog.cpp sqlitebrowser-3.6.0-new/src/EditDialog.cpp
--- sqlitebrowser-3.6.0/src/EditDialog.cpp	2015-04-27 11:31:47.000000000 +0200
+++ sqlitebrowser-3.6.0-new/src/EditDialog.cpp	2015-04-29 23:49:39.356537291 +0200
@@ -2,7 +2,7 @@
 #include "ui_EditDialog.h"
 #include "sqlitedb.h"
 #include "PreferencesDialog.h"
-#include "src/qhexedit.h"
+#include <qhexedit.h>
 
 #include <QFileDialog>
 #include <QKeySequence>
diff -rupN sqlitebrowser-3.6.0/tests/CMakeLists.txt sqlitebrowser-3.6.0-new/tests/CMakeLists.txt
--- sqlitebrowser-3.6.0/tests/CMakeLists.txt	2015-04-27 11:31:47.000000000 +0200
+++ sqlitebrowser-3.6.0-new/tests/CMakeLists.txt	2015-04-30 12:47:21.152075973 +0200
@@ -7,12 +7,8 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-set(ANTLR_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../libs/antlr-2.7.7")
-add_subdirectory("${ANTLR_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/antlr")
-
 if(USE_QT5)
     find_package(Qt5Widgets REQUIRED)
-    set(CMAKE_AUTOMOC ON)
     set(CMAKE_INCLUDE_CURRENT_DIR ON)
 else()
     find_package(Qt4 COMPONENTS QtCore QtTest QtGui REQUIRED)
@@ -65,7 +61,7 @@ else()
     QT4_WRAP_UI(SQLB_FORM_HDR ${SQLB_FORMS})
 endif()
 
-include_directories("${CMAKE_CURRENT_BINARY_DIR}" "${ANTLR_DIR}" ../src)
+include_directories("${CMAKE_CURRENT_BINARY_DIR}" ../src)
 
 add_executable(${PROJECT_NAME} ${SQLB_MOC} ${SQLB_HDR} ${SQLB_SRC} ${SQLB_FORM_HDR})
 
@@ -74,7 +70,6 @@ if(USE_QT5)
     set(QT_LIBRARIES "")
 endif()
 
-add_dependencies(${PROJECT_NAME} antlr)
 target_link_libraries(${PROJECT_NAME} antlr ${QT_LIBRARIES} ${LIBSQLITE})
 
 MESSAGE(STATUS "Build type is ${CMAKE_BUILD_TYPE}")
