diff -rupN sqlitebrowser-3.7.0/CMakeLists.txt sqlitebrowser-3.7.0-new/CMakeLists.txt
--- sqlitebrowser-3.7.0/CMakeLists.txt	2015-06-14 12:13:37.000000000 +0200
+++ sqlitebrowser-3.7.0-new/CMakeLists.txt	2015-06-15 00:26:37.372712963 +0200
@@ -26,19 +26,9 @@ if(WIN32 AND MSVC)
 	set(CMAKE_PREFIX_PATH "${QT5_PATH};${SQLITE3_PATH}")
 endif()
 
-find_package(Antlr2)
-
-set(QHEXEDIT_DIR libs/qhexedit)
-set(QCUSTOMPLOT_DIR libs/qcustomplot-source)
-set(QSCINTILLA_DIR libs/qscintilla/Qt4Qt5)
-
-if(NOT ANTLR2_FOUND)
-    set(ANTLR_DIR libs/antlr-2.7.7)
-    add_subdirectory(${ANTLR_DIR})
-endif()
-add_subdirectory(${QHEXEDIT_DIR})
-add_subdirectory(${QCUSTOMPLOT_DIR})
-add_subdirectory(${QSCINTILLA_DIR})
+find_package(PkgConfig)
+pkg_check_modules(QHEXEDIT2 qhexedit2-qt5)
+pkg_check_modules(QCUSTOMPLOT qcustomplot-qt5)
 
 if(USE_QT5)
 	find_package(Qt5Widgets REQUIRED)
@@ -225,16 +215,11 @@ endif()
 
 include_directories(
 		"${CMAKE_CURRENT_BINARY_DIR}"
-		${QHEXEDIT_DIR}
-		${QCUSTOMPLOT_DIR}
+		${QHEXEDIT2_INCLUDE_DIRS}
+		${QCUSTOMPLOT_INCLUDE_DIRS}
 		${QSCINTILLA_DIR}
 		${ADDITIONAL_INCLUDE_PATHS}
 		src)
-if(ANTLR2_FOUND)
-    include_directories(${ANTLR2_INCLUDE_DIRS})
-else()
-    include_directories(${ANTLR_DIR})
-endif()
 
 add_executable(${PROJECT_NAME}
 		${SQLB_HDR}
@@ -249,32 +234,20 @@ if(USE_QT5)
     set(QT_LIBRARIES "")
 endif()
 add_dependencies(${PROJECT_NAME} qhexedit qcustomplot qscintilla2)
-if(NOT ANTLR2_FOUND)
-    add_dependencies(${PROJECT_NAME} antlr)
-endif()
 
 link_directories(
-		"${CMAKE_CURRENT_BINARY_DIR}/${ANTLR_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QHEXEDIT_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QCUSTOMPLOT_DIR}"
-		"${CMAKE_CURRENT_BINARY_DIR}/${QSCINTILLA_DIR}")
-if(NOT ANTLR2_FOUND)
-    link_directories("${CMAKE_CURRENT_BINARY_DIR}/${ANTLR_DIR}")
-endif()
+		${QHEXEDIT2_LIBRARY_DIRS}
+		${QCUSTOMPLOT_LIBRARY_DIRS})
 
 target_link_libraries(${PROJECT_NAME}
-		qhexedit
-		qcustomplot
-		qscintilla2
+		antlr
+		${QHEXEDIT2_LIBRARIES}
+		${QCUSTOMPLOT_LIBRARIES}
+		qscintilla2-qt5
 		${QT_LIBRARIES}
 		${WIN32_STATIC_LINK}
 		${LIBSQLITE}
 		${ADDITIONAL_LIBS})
-if(ANTLR2_FOUND)
-    target_link_libraries(${PROJECT_NAME} ${ANTLR2_LIBRARIES})
-else()
-    target_link_libraries(${PROJECT_NAME} antlr)
-endif()
 
 if(WIN32 AND MSVC)
 	set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS_DEBUG "/SUBSYSTEM:CONSOLE")
diff -rupN sqlitebrowser-3.7.0/src/EditDialog.cpp sqlitebrowser-3.7.0-new/src/EditDialog.cpp
--- sqlitebrowser-3.7.0/src/EditDialog.cpp	2015-06-14 12:13:37.000000000 +0200
+++ sqlitebrowser-3.7.0-new/src/EditDialog.cpp	2015-06-15 00:26:37.373712982 +0200
@@ -2,9 +2,9 @@
 #include "ui_EditDialog.h"
 #include "sqlitedb.h"
 #include "PreferencesDialog.h"
-#include "src/qhexedit.h"
 #include "FileDialog.h"
 
+#include <qhexedit.h>
 #include <QKeySequence>
 #include <QShortcut>
 
diff -rupN sqlitebrowser-3.7.0/src/tests/CMakeLists.txt sqlitebrowser-3.7.0-new/src/tests/CMakeLists.txt
--- sqlitebrowser-3.7.0/src/tests/CMakeLists.txt	2015-06-14 12:13:37.000000000 +0200
+++ sqlitebrowser-3.7.0-new/src/tests/CMakeLists.txt	2015-06-15 09:40:54.347202969 +0200
@@ -58,16 +58,7 @@ if(USE_QT5)
     set(QT_LIBRARIES "")
 endif()
 
-if(ANTLR2_FOUND)
-else()
-    add_dependencies(test-sqlobjects antlr)
-endif()
-target_link_libraries(test-sqlobjects ${QT_LIBRARIES} ${LIBSQLITE})
-if(ANTLR2_FOUND)
-    target_link_libraries(test-sqlobjects ${ANTLR2_LIBRARIES})
-else()
-    target_link_libraries(test-sqlobjects antlr)
-endif()
+target_link_libraries(test-sqlobjects ${QT_LIBRARIES} ${LIBSQLITE} antlr)
 add_test(test-sqlobjects test-sqlobjects)
 
 # test-import
