diff -rupN sqlitebrowser-3.6.0/src/EditDialog.cpp sqlitebrowser-3.6.0-new/src/EditDialog.cpp
--- sqlitebrowser-3.6.0/src/EditDialog.cpp	2015-04-27 11:31:47.000000000 +0200
+++ sqlitebrowser-3.6.0-new/src/EditDialog.cpp	2015-05-20 20:52:51.764063791 +0200
@@ -7,6 +7,7 @@
 #include <QFileDialog>
 #include <QKeySequence>
 #include <QShortcut>
+#include <QImageReader>
 
 EditDialog::EditDialog(QWidget* parent)
     : QDialog(parent),
