diff -rupN sqlitebrowser-3.7.0/src/EditDialog.cpp sqlitebrowser-3.7.0-new/src/EditDialog.cpp
--- sqlitebrowser-3.7.0/src/EditDialog.cpp	2015-06-15 00:02:51.740963614 +0200
+++ sqlitebrowser-3.7.0-new/src/EditDialog.cpp	2015-06-15 00:03:18.535485183 +0200
@@ -7,6 +7,7 @@
 #include <qhexedit.h>
 #include <QKeySequence>
 #include <QShortcut>
+#include <QImageReader>
 
 EditDialog::EditDialog(QWidget* parent)
     : QDialog(parent),
