--- trackedit/Makefile.in.orig	2007-07-15 12:57:29.000000000 +0200
+++ trackedit/Makefile.in	2007-09-01 12:36:43.000000000 +0200
@@ -67,7 +67,7 @@
 am_ustuntstrackedit_OBJECTS = main.$(OBJEXT)
 ustuntstrackedit_OBJECTS = $(am_ustuntstrackedit_OBJECTS)
 ustuntstrackedit_DEPENDENCIES =  \
-	$(top_builddir)/trackedit/libtrackedit.a \
+	libtrackedit.a \
 	$(top_builddir)/graphics/libgraphics.a \
 	$(top_builddir)/simulation/libsimulation.a \
 	$(top_builddir)/shared/libshared.a
@@ -217,7 +217,7 @@
 target_vendor = @target_vendor@
 usdatadir = @usdatadir@
 ustuntstrackedit_SOURCES = main.cpp
-ustuntstrackedit_LDADD = $(top_builddir)/trackedit/libtrackedit.a \
+ustuntstrackedit_LDADD = libtrackedit.a \
 	$(top_builddir)/graphics/libgraphics.a $(top_builddir)/simulation/libsimulation.a \
 	$(top_builddir)/shared/libshared.a @sdllibs@ @graphlibs@ @LIBINTL@
 
--- trackedit/Makefile.am.orig	2007-07-15 12:54:15.000000000 +0200
+++ trackedit/Makefile.am	2007-09-01 12:17:46.000000000 +0200
@@ -1,6 +1,6 @@
 bin_PROGRAMS = ustuntstrackedit
 ustuntstrackedit_SOURCES = main.cpp
-ustuntstrackedit_LDADD = $(top_builddir)/trackedit/libtrackedit.a \
+ustuntstrackedit_LDADD = libtrackedit.a \
 	$(top_builddir)/graphics/libgraphics.a $(top_builddir)/simulation/libsimulation.a \
 	$(top_builddir)/shared/libshared.a @sdllibs@ @graphlibs@ @LIBINTL@
 
