diff -up ark-4.14.3/app/mainwindow.cpp.0004 ark-4.14.3/app/mainwindow.cpp
--- ark-4.14.3/app/mainwindow.cpp.0004	2014-10-25 07:53:24.000000000 -0500
+++ ark-4.14.3/app/mainwindow.cpp	2014-11-24 08:49:01.799233095 -0600
@@ -38,6 +38,7 @@
 #include <KDebug>
 #include <KEditToolBar>
 #include <KShortcutsDialog>
+#include <KXMLGUIFactory>
 
 #include <QDragEnterEvent>
 #include <QDragMoveEvent>
@@ -69,6 +70,8 @@ MainWindow::~MainWindow()
     if (m_recentFilesAction) {
         m_recentFilesAction->saveEntries(KGlobal::config()->group("Recent Files"));
     }
+
+    guiFactory()->removeClient(m_part);
     delete m_part;
     m_part = 0;
 }
diff -up ark-4.14.3/part/part.cpp.0004 ark-4.14.3/part/part.cpp
--- ark-4.14.3/part/part.cpp.0004	2014-11-24 08:49:01.799233095 -0600
+++ ark-4.14.3/part/part.cpp	2014-11-24 08:50:35.638688148 -0600
@@ -138,7 +138,7 @@ Part::Part(QWidget *parentWidget, QObjec
 
 Part::~Part()
 {
-    factory()->removeClient(this);
+    //factory()->removeClient(this);
 
     saveSplitterSizes();
 
