--- sendmail-8.14.4/sendmail/usersmtp.c	2009-06-17 18:26:51.000000000 +0100
+++ sendmail-8.14.4/sendmail/usersmtp.c	2010-06-11 13:13:52.150312505 +0100
@@ -1323,9 +1323,7 @@
 {
 	long sff;
 	int r;
-#if SASL <= 10515
 	size_t len;
-#endif /* SASL <= 10515 */
 	char *p;
 
 	if (file == NULL || *file == '\0')
@@ -1361,9 +1359,16 @@
 #endif /* SASL <= 10515 */
 
 	p = (char *) file;
+	len = strlen(p);
 	if ((r = safefile(p, RunAsUid, RunAsGid, RunAsUserName, sff,
 			  S_IRUSR, NULL)) == 0)
 		return SASL_OK;
+#if SASL > 10515
+	/* Expect /usr/lib/sasl2/Sendmail.conf to be missing - config now in /etc/sasl2 */
+	if (type == SASL_VRFY_CONF && r == ENOENT &&
+            len >= 8 && strncmp(p, "/usr/lib", 8) == 0)
+		return SASL_CONTINUE;
+#endif /* SASL > 10515 */
 	if (LogLevel > (r != ENOENT ? 8 : 10))
 		sm_syslog(LOG_WARNING, NOQID, "error: safesasl(%s) failed: %s",
 			  p, sm_errstring(r));
