--- sendmail-8.14.4/include/sm/bdb.h.old	2003-03-06 17:30:05.000000000 +0100
+++ sendmail-8.14.4/include/sm/bdb.h		2010-04-15 16:02:41.029169004 +0200
@@ -19,7 +19,8 @@ 
 #  define DB_VERSION_MAJOR 1
 # endif /* ! DB_VERSION_MAJOR */
 
-# if DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 1
+# if (DB_VERSION_MAJOR >= 4 && DB_VERSION_MINOR >= 1) || \
+  (DB_VERSION_MAJOR >= 5)
 
 #  define DBTXN	NULL ,
 
