diff --git a/libsm/mbdb.c b/libsm/mbdb.c
index 482ed70..62811a7 100644
--- a/libsm/mbdb.c
+++ b/libsm/mbdb.c
@@ -315,7 +315,7 @@ mbdb_pw_lookup(name, user)
 {
 	struct passwd *pw;
 
-#ifdef HESIOD
+#ifdef disableHESIOD
 	/* DEC Hesiod getpwnam accepts numeric strings -- short circuit it */
 	{
 		char *p;
diff --git a/sendmail/recipient.c b/sendmail/recipient.c
index 84d2f87..6f29f78 100644
--- a/sendmail/recipient.c
+++ b/sendmail/recipient.c
@@ -1174,7 +1174,7 @@ finduser(name, fuzzyp, user)
 
 	*fuzzyp = false;
 
-#if HESIOD
+#if disableHESIOD
 	/* DEC Hesiod getpwnam accepts numeric strings -- short circuit it */
 	for (p = name; *p != '\0'; p++)
 		if (!isascii(*p) || !isdigit(*p))
