--- sendmail-8.14.8/sendmail/helpfile.noversion	2014-01-26 22:18:24.583735645 -0200
+++ sendmail-8.14.8/sendmail/helpfile	2014-01-26 22:18:54.037485628 -0200
@@ -11,9 +11,7 @@ cpyr	By using this file, you agree to th
 cpyr	forth in the LICENSE file which can be found at the top level of
 cpyr	the sendmail distribution.
 cpyr
-cpyr	$$Id: helpfile,v 8.49 2013/11/22 20:51:55 ca Exp $$
-cpyr
-smtp	This is sendmail version $v
+smtp	This is sendmail
 smtp	Topics:
 smtp		HELO	EHLO	MAIL	RCPT	DATA
 smtp		RSET	NOOP	QUIT	HELP	VRFY
