diff -up gnome-session-2.29.6/capplet/.libs gnome-session-2.29.6/capplet/
Only in gnome-session-2.29.6/capplet/: .deps
Only in gnome-session-2.29.6/capplet/: gnome-session-properties
Only in gnome-session-2.29.6/capplet/: gsm-app-dialog.c
Only in gnome-session-2.29.6/capplet/: gsm-app-dialog.h
Only in gnome-session-2.29.6/capplet/: gsm-app-dialog.o
Only in gnome-session-2.29.6/capplet/: gsm-properties-dialog.c
Only in gnome-session-2.29.6/capplet/: gsm-properties-dialog.h
Only in gnome-session-2.29.6/capplet/: gsm-properties-dialog.o
Only in gnome-session-2.29.6/capplet/: gsp-app.c
Only in gnome-session-2.29.6/capplet/: gsp-app.h
Only in gnome-session-2.29.6/capplet/: gsp-app-manager.c
Only in gnome-session-2.29.6/capplet/: gsp-app-manager.h
Only in gnome-session-2.29.6/capplet/: gsp-app-manager.o
Only in gnome-session-2.29.6/capplet/: gsp-app.o
Only in gnome-session-2.29.6/capplet/: gsp-keyfile.c
Only in gnome-session-2.29.6/capplet/: gsp-keyfile.h
Only in gnome-session-2.29.6/capplet/: gsp-keyfile.o
Only in gnome-session-2.29.6/capplet/: .libs
Only in gnome-session-2.29.6/capplet/: main.c
Only in gnome-session-2.29.6/capplet/: main.o
Only in gnome-session-2.29.6/capplet/: Makefile
Only in gnome-session-2.29.6/capplet/: Makefile.am
Only in gnome-session-2.29.6/capplet/: Makefile.in
diff -up gnome-session-2.29.6/compat/.libs gnome-session-2.29.6/compat/
Only in gnome-session-2.29.6/compat/: .deps
Only in gnome-session-2.29.6/compat/: gnome-settings-daemon-helper
Only in gnome-session-2.29.6/compat/: gnome-settings-daemon-helper.c
Only in gnome-session-2.29.6/compat/: gnome-settings-daemon-helper.desktop
Only in gnome-session-2.29.6/compat/: gnome-settings-daemon-helper.desktop.in.in
Only in gnome-session-2.29.6/compat/: gnome-settings-daemon-helper.desktop.in.in.in
Only in gnome-session-2.29.6/compat/: gnome-settings-daemon-helper.o
Only in gnome-session-2.29.6/compat/: .libs
Only in gnome-session-2.29.6/compat/: Makefile
Only in gnome-session-2.29.6/compat/: Makefile.am
Only in gnome-session-2.29.6/compat/: Makefile.in
Only in gnome-session-2.29.6/compat/: README
diff -up gnome-session-2.29.6/configure.in.libs gnome-session-2.29.6/configure.in
--- gnome-session-2.29.6/configure.in.libs	2010-02-11 22:13:48.580010605 -0500
+++ gnome-session-2.29.6/configure.in	2010-02-11 22:13:48.617006448 -0500
@@ -165,6 +165,8 @@ AC_CHECK_LIB(Xau, XauFileName, [X_LIBS="
 *** Can't find the Xauth library. It is needed to compile gnome-session.])],
 	     $X_LIBS)
 
+X_LIBS="$X_LIBS -lXext"
+
 AC_SUBST(X_LIBS)
 
 dnl ====================================================================
diff -up gnome-session-2.29.6/egg/.libs gnome-session-2.29.6/egg/
Only in gnome-session-2.29.6/egg/: .deps
Only in gnome-session-2.29.6/egg/: eggdesktopfile.c
Only in gnome-session-2.29.6/egg/: eggdesktopfile.h
Only in gnome-session-2.29.6/egg/: eggdesktopfile.lo
Binary files gnome-session-2.29.6/egg/.libs/eggdesktopfile.o and gnome-session-2.29.6/egg/eggdesktopfile.o differ
Only in gnome-session-2.29.6/egg/: eggsmclient.c
Only in gnome-session-2.29.6/egg/: eggsmclient.h
Only in gnome-session-2.29.6/egg/: eggsmclient.lo
Binary files gnome-session-2.29.6/egg/.libs/eggsmclient.o and gnome-session-2.29.6/egg/eggsmclient.o differ
Only in gnome-session-2.29.6/egg/: eggsmclient-private.h
Only in gnome-session-2.29.6/egg/: eggsmclient-xsmp.c
Only in gnome-session-2.29.6/egg/: eggsmclient-xsmp.lo
Binary files gnome-session-2.29.6/egg/.libs/eggsmclient-xsmp.o and gnome-session-2.29.6/egg/eggsmclient-xsmp.o differ
Only in gnome-session-2.29.6/egg/.libs: libeggdesktopfile.a
Only in gnome-session-2.29.6/egg/.libs: libeggsmclient.a
Only in gnome-session-2.29.6/egg/.libs: libeggsmclient-gnome.a
Only in gnome-session-2.29.6/egg/: .libs
Only in gnome-session-2.29.6/egg/: Makefile
Only in gnome-session-2.29.6/egg/: Makefile.am
Only in gnome-session-2.29.6/egg/: Makefile.in
diff -up gnome-session-2.29.6/gnome-session/.libs gnome-session-2.29.6/gnome-session/
Only in gnome-session-2.29.6/gnome-session/: .deps
Only in gnome-session-2.29.6/gnome-session/: gdm.c
Only in gnome-session-2.29.6/gnome-session/: gdm.h
Only in gnome-session-2.29.6/gnome-session/: gdm-log.c
Only in gnome-session-2.29.6/gnome-session/: gdm-log.h
Only in gnome-session-2.29.6/gnome-session/: gdm-log.o
Only in gnome-session-2.29.6/gnome-session/: gdm.o
Only in gnome-session-2.29.6/gnome-session/: gdm-signal-handler.c
Only in gnome-session-2.29.6/gnome-session/: gdm-signal-handler.h
Only in gnome-session-2.29.6/gnome-session/: gdm-signal-handler.o
Only in gnome-session-2.29.6/gnome-session/: gnome-session
Only in gnome-session-2.29.6/gnome-session/: gs-idle-monitor.c
Only in gnome-session-2.29.6/gnome-session/: gs-idle-monitor.h
Only in gnome-session-2.29.6/gnome-session/: gs-idle-monitor.o
Only in gnome-session-2.29.6/gnome-session/: gsm-app.c
Only in gnome-session-2.29.6/gnome-session/: gsm-app-glue.h
Only in gnome-session-2.29.6/gnome-session/: gsm-app.h
Only in gnome-session-2.29.6/gnome-session/: gsm-app.o
Only in gnome-session-2.29.6/gnome-session/: gsm-autostart-app.c
Only in gnome-session-2.29.6/gnome-session/: gsm-autostart-app.h
Only in gnome-session-2.29.6/gnome-session/: gsm-autostart-app.o
Only in gnome-session-2.29.6/gnome-session/: gsm-client.c
Only in gnome-session-2.29.6/gnome-session/: gsm-client-glue.h
Only in gnome-session-2.29.6/gnome-session/: gsm-client.h
Only in gnome-session-2.29.6/gnome-session/: gsm-client.o
Only in gnome-session-2.29.6/gnome-session/: gsm-consolekit.c
Only in gnome-session-2.29.6/gnome-session/: gsm-consolekit.h
Only in gnome-session-2.29.6/gnome-session/: gsm-consolekit.o
Only in gnome-session-2.29.6/gnome-session/: gsm-dbus-client.c
Only in gnome-session-2.29.6/gnome-session/: gsm-dbus-client.h
Only in gnome-session-2.29.6/gnome-session/: gsm-dbus-client.o
Only in gnome-session-2.29.6/gnome-session/: gsm-gconf.c
Only in gnome-session-2.29.6/gnome-session/: gsm-gconf.h
Only in gnome-session-2.29.6/gnome-session/: gsm-gconf.o
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibit-dialog.c
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibit-dialog.h
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibit-dialog.o
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibitor.c
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibitor-glue.h
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibitor.h
Only in gnome-session-2.29.6/gnome-session/: gsm-inhibitor.o
Only in gnome-session-2.29.6/gnome-session/: gsm-logout-dialog.c
Only in gnome-session-2.29.6/gnome-session/: gsm-logout-dialog.h
Only in gnome-session-2.29.6/gnome-session/: gsm-logout-dialog.o
Only in gnome-session-2.29.6/gnome-session/: gsm-manager.c
Only in gnome-session-2.29.6/gnome-session/: gsm-manager.c.max-idle
Only in gnome-session-2.29.6/gnome-session/: gsm-manager.c.nag-root-user
Only in gnome-session-2.29.6/gnome-session/: gsm-manager-glue.h
Only in gnome-session-2.29.6/gnome-session/: gsm-manager.h
Only in gnome-session-2.29.6/gnome-session/: gsm-manager.o
Only in gnome-session-2.29.6/gnome-session/: gsm-marshal.c
Only in gnome-session-2.29.6/gnome-session/: gsm-marshal.h
Only in gnome-session-2.29.6/gnome-session/: gsm-marshal.list
Only in gnome-session-2.29.6/gnome-session/: gsm-marshal.o
Only in gnome-session-2.29.6/gnome-session/: gsm-presence.c
Only in gnome-session-2.29.6/gnome-session/: gsm-presence.c.max-idle
Only in gnome-session-2.29.6/gnome-session/: gsm-presence-glue.h
Only in gnome-session-2.29.6/gnome-session/: gsm-presence.h
Only in gnome-session-2.29.6/gnome-session/: gsm-presence.o
Only in gnome-session-2.29.6/gnome-session/: gsm-session-save.c
Only in gnome-session-2.29.6/gnome-session/: gsm-session-save.h
Only in gnome-session-2.29.6/gnome-session/: gsm-session-save.o
Only in gnome-session-2.29.6/gnome-session/: gsm-store.c
Only in gnome-session-2.29.6/gnome-session/: gsm-store.h
Only in gnome-session-2.29.6/gnome-session/: gsm-store.o
Only in gnome-session-2.29.6/gnome-session/: gsm-util.c
Only in gnome-session-2.29.6/gnome-session/: gsm-util.c.nag-root-user
Only in gnome-session-2.29.6/gnome-session/: gsm-util.h
Only in gnome-session-2.29.6/gnome-session/: gsm-util.h.nag-root-user
Only in gnome-session-2.29.6/gnome-session/: gsm-util.lo
Binary files gnome-session-2.29.6/gnome-session/.libs/gsm-util.o and gnome-session-2.29.6/gnome-session/gsm-util.o differ
Only in gnome-session-2.29.6/gnome-session/: gsm-xsmp-client.c
Only in gnome-session-2.29.6/gnome-session/: gsm-xsmp-client.h
Only in gnome-session-2.29.6/gnome-session/: gsm-xsmp-client.o
Only in gnome-session-2.29.6/gnome-session/: gsm-xsmp-server.c
Only in gnome-session-2.29.6/gnome-session/: gsm-xsmp-server.h
Only in gnome-session-2.29.6/gnome-session/: gsm-xsmp-server.o
Only in gnome-session-2.29.6/gnome-session/.libs: libgsmutil.a
Only in gnome-session-2.29.6/gnome-session/: .libs
Only in gnome-session-2.29.6/gnome-session/: main.c
Only in gnome-session-2.29.6/gnome-session/: main.c.max-idle
Only in gnome-session-2.29.6/gnome-session/: main.o
Only in gnome-session-2.29.6/gnome-session/: Makefile
Only in gnome-session-2.29.6/gnome-session/: Makefile.am
Only in gnome-session-2.29.6/gnome-session/: Makefile.am.libs
Only in gnome-session-2.29.6/gnome-session/: Makefile.in
Only in gnome-session-2.29.6/gnome-session/: org.gnome.SessionManager.App.xml
Only in gnome-session-2.29.6/gnome-session/: org.gnome.SessionManager.ClientPrivate.xml
Only in gnome-session-2.29.6/gnome-session/: org.gnome.SessionManager.Client.xml
Only in gnome-session-2.29.6/gnome-session/: org.gnome.SessionManager.Inhibitor.xml
Only in gnome-session-2.29.6/gnome-session/: org.gnome.SessionManager.Presence.xml
Only in gnome-session-2.29.6/gnome-session/: org.gnome.SessionManager.xml
Only in gnome-session-2.29.6/gnome-session/: README
Only in gnome-session-2.29.6/gnome-session/: test-client-dbus
Only in gnome-session-2.29.6/gnome-session/: test-client-dbus.c
Only in gnome-session-2.29.6/gnome-session/: test-client-dbus.o
Only in gnome-session-2.29.6/gnome-session/: test-inhibit
Only in gnome-session-2.29.6/gnome-session/: test-inhibit.c
Only in gnome-session-2.29.6/gnome-session/: test-inhibit.o
diff -up gnome-session-2.29.6/gnome-session/Makefile.am.libs gnome-session-2.29.6/gnome-session/Makefile.am
--- gnome-session-2.29.6/gnome-session/Makefile.am.libs	2010-02-11 22:29:16.556007920 -0500
+++ gnome-session-2.29.6/gnome-session/Makefile.am	2010-02-11 22:29:34.619008005 -0500
@@ -51,6 +51,7 @@ gnome_session_LDADD =				\
 	$(GCONF_LIBS)				\
 	$(XRENDER_LIBS)				\
 	$(XTEST_LIBS)				\
+	$(X_LIBS)				\
 	$(EXECINFO_LIBS)			\
 	$(NULL)
 
diff -up gnome-session-2.29.6/tools/.libs gnome-session-2.29.6/tools/
Only in gnome-session-2.29.6/tools/: .deps
Only in gnome-session-2.29.6/tools/: gnome-session-save
Only in gnome-session-2.29.6/tools/: gnome-session-save.c
Only in gnome-session-2.29.6/tools/: gnome-session-save.o
Only in gnome-session-2.29.6/tools/: .libs
Only in gnome-session-2.29.6/tools/: Makefile
Only in gnome-session-2.29.6/tools/: Makefile.am
Only in gnome-session-2.29.6/tools/: Makefile.in
