--- ./socket_wrapper/src/socket_wrapper.c      2014-09-30 11:23:19.733011302 +0200
+++ ./socket_wrapper/src/socket_wrapper.c      2014-09-30 11:25:11.599141672 +0200
@@ -294,7 +294,7 @@
 	int (*libc_dup)(int fd);
 	int (*libc_dup2)(int oldfd, int newfd);
 #ifdef HAVE_EVENTFD
-	int (*libc_eventfd)(int count, int flags);
+	int (*libc_eventfd)(unsigned int count, int flags);
 #endif
 	int (*libc_getpeername)(int sockfd,
 				struct sockaddr *addr,
@@ -528,7 +528,7 @@
 }
 
 #ifdef HAVE_EVENTFD
-static int libc_eventfd(int count, int flags)
+static int libc_eventfd(unsigned int count, int flags)
 {
 	swrap_load_lib_function(SWRAP_LIBC, eventfd);
 
@@ -4159,7 +4159,7 @@
  ***************************/
 
 #ifdef HAVE_EVENTFD
-static int swrap_eventfd(int count, int flags)
+static int swrap_eventfd(unsigned int count, int flags)
 {
 	int fd;
 
@@ -4171,7 +4171,7 @@
 	return fd;
 }
 
-int eventfd(int count, int flags)
+int eventfd(unsigned int count, int flags)
 {
 	return swrap_eventfd(count, flags);
 }
