--- SConstruct.orig	2011-06-25 03:39:39.000000000 -0500
+++ SConstruct	2011-06-25 03:39:56.000000000 -0500
@@ -129,17 +129,17 @@
 	conf.env.Append(LINKFLAGS = ['/SUBSYSTEM:WINDOWS', '/FORCE'])
 	conf.env.Append(LIBS=['SDLmain'])
 
-if not conf.CheckLibWithHeader('SDL', 'SDL.h', 'c++', "SDL_Init(0);", False):
+if not conf.CheckLibWithHeader('SDL', 'SDL.h', 'c', "SDL_Init(0);", False):
 	Exit(1)
 	
 
 Export('sdl_cflags')
 Export('sdl_libs')
 
-if not conf.CheckLibWithHeader('SDL_image', 'SDL_image.h', 'c++', "IMG_Load(0);", False):
+if not conf.CheckLibWithHeader('SDL_image', 'SDL_image.h', 'c', "IMG_Load(0);", False):
 	Exit(1)
 
-if not conf.CheckLibWithHeader('vorbisfile', 'vorbis/vorbisfile.h', 'c++', "ov_open(0, 0, 0, 0);", False):
+if not conf.CheckLibWithHeader('vorbisfile', 'vorbis/vorbisfile.h', 'c', "ov_open(0, 0, 0, 0);", False):
 	Exit(1)
 
 if sys.platform == 'win32':
