--- SConstruct.orig	2011-06-25 10:38:50.000000000 -0500
+++ SConstruct	2011-06-25 10:41:52.000000000 -0500
@@ -60,7 +60,7 @@
 
 if (sys.platform != "win32" and env['gcc_visibility']): 
 	env.Append(CCFLAGS=['-fvisibility=hidden', '-DGCC_HASCLASSVISIBILITY'])
-	env.Append(CXXFLAGS=['-fvisibility-inlines-hidden', '-fvisibility=hidden', '-DGCC_HASCLASSVISIBILITY'])
+	env.Append(CXXFLAGS=['-fvisibility-inlines-hidden'])
 
 if sys.platform == "win32" and debug:
 	stl_port_debug = True
@@ -98,13 +98,6 @@
 	if debug:
 		env.Append(CCFLAGS=['-ggdb', '-D_FORTIFY_SOURCE=2'])
 		env.Append(CPPFLAGS=['-ggdb', '-D_FORTIFY_SOURCE=2'])
-	else: 
-		env.Append(CCFLAGS=['-O3'])
-		env.Append(CPPFLAGS=['-O3'])
-		
-	env.Append(CPPFLAGS=['-Wall', '-Wno-deprecated', '-pedantic', '-Wno-long-long', '-pipe', '-pthread'])
-	env.Append(CCFLAGS=['-Wall', '-Wno-deprecated', '-pedantic', '-Wno-long-long', '-pipe', '-pthread'])
-
 
 conf_env = env.Clone()
 conf = Configure(conf_env)
