--- editor/SConscript.dso	2011-06-26 13:33:25.000000000 -0500
+++ editor/SConscript	2011-06-26 13:33:42.000000000 -0500
@@ -6,7 +6,7 @@
 
 env = env.Clone()
 
-libs = ['btanks_engine', 'sdlx', 'mrt', 'SDL']
+libs = ['btanks_engine', 'sdlx', 'mrt', 'SDL', 'clunk']
 
 ed_src = [
 	'base_brush.cpp', 
--- engine/SConscript.dso	2011-06-26 13:44:02.000000000 -0500
+++ engine/SConscript	2011-06-26 13:44:34.000000000 -0500
@@ -113,7 +113,7 @@
 	Install('#', bt[0])
 
 bt_main_sources = ['src/main.cpp']
-bt_main_libs =  ['mrt', 'btanks_engine', 'SDL', 'sdlx']
+bt_main_libs =  ['mrt', 'btanks_engine', 'SDL', 'sdlx', 'clunk']
 
 if sys.platform == "win32":
 	bt_rc = env.RES('src/bt.rc')
