--- clunk/SConscript.orig	2011-06-25 02:56:18.000000000 -0500
+++ clunk/SConscript	2011-06-25 02:56:26.000000000 -0500
@@ -16,7 +16,6 @@
 clunk_libs = ['SDL']
 
 if sys.platform != 'win32':
-	env.Append(LINKFLAGS=['-Wl,-rpath,'+ lib_dir])
 	clunk_libs.append('m')
 
 env.MergeFlags(sdl_cflags, sdl_libs)
--- clunk/SConstruct.remove-rpath	2011-06-26 12:25:15.000000000 -0500
+++ clunk/SConstruct	2011-06-26 12:26:08.000000000 -0500
@@ -48,10 +48,4 @@
 	LIBS=['SDL'])
 
 
-if sys.platform != 'win32':
-	env.Append(CFLAGS=['-Wall', '-pedantic'])
-	env.Append(CXXFLAGS=['-Wall', '-pedantic'])
-	env.Append(LINKFLAGS=['-Wl,-rpath,'+ lib_dir])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,.'])
-
 env.Program('clunk_test', ['test.cpp'], LIBS=['clunk'])
--- sdlx/SConscript.remove-rpath	2011-06-26 12:41:22.000000000 -0500
+++ sdlx/SConscript	2011-06-26 12:41:48.000000000 -0500
@@ -22,8 +22,6 @@
 	libs.append('rt')
 	libs.append('X11')
 
-	env.Append(LINKFLAGS=['-Wl,-rpath,'+ lib_dir])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/mrt'])
 else: 
 	c_map = env.SharedObject('c_map.cpp')
 	libs.append('opengl32')
--- objects/SConscript.remove-rpath	2011-06-26 12:40:02.000000000 -0500
+++ objects/SConscript	2011-06-26 12:40:20.000000000 -0500
@@ -14,12 +14,6 @@
 env.Append(CPPPATH=['#/engine', '#/engine/src'])
 env.MergeFlags(sdl_cflags, sdl_libs)
 
-if sys.platform != 'win32':
-	env.Append(LINKFLAGS=['-Wl,-rpath,'+ lib_dir])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/mrt'])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/sdlx'])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/clunk'])
-
 sources = [
 	'bullet.cpp', 'explosion.cpp', 'single_pose.cpp',
 	'tank.cpp', 'shilka.cpp', 'launcher.cpp', 'ai_tank.cpp',
--- engine/SConscript.remove-rpath	2009-12-30 07:50:33.000000000 -0600
+++ engine/SConscript	2011-06-26 12:59:39.000000000 -0500
@@ -104,13 +104,6 @@
 #else: 
 #	bt_libs.append('GL')
 
-if sys.platform != 'win32':
-	env.Append(LINKFLAGS=['-Wl,-rpath,'+ lib_dir])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/mrt'])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/sdlx'])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/clunk'])
-
-
 bt = env.SharedLibrary('btanks_engine', bt_sources, LIBS=bt_libs)
 
 if sys.platform != 'win32' and len(env['prefix']) > 0:
@@ -129,13 +122,7 @@
 	bt_main_libs.append('user32')
 	bt_main_libs.append('sdlx_main')
 
-if sys.platform != 'win32':
-	bt_env.Append(LINKFLAGS=['-Wl,-rpath,'+ lib_dir])
-	bt_env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/mrt'])
-	bt_env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/sdlx'])
-	bt_env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/clunk'])
-
-bt_main = bt_env.Program('btanks', bt_main_sources, LIBS=bt_main_libs, RPATH=[lib_dir])
+bt_main = bt_env.Program('btanks', bt_main_sources, LIBS=bt_main_libs)
 
 if sys.platform == 'win32': 
 	bt_env.AddPostAction(bt_main, 'mt.exe -nologo -manifest ${TARGET}.manifest -outputresource:$TARGET;2')
--- editor/SConscript.remove-rpath	2009-12-30 07:50:33.000000000 -0600
+++ editor/SConscript	2011-06-26 13:01:36.000000000 -0500
@@ -20,19 +20,13 @@
 
 env.Append(CPPPATH=['#/engine', '#/engine/src'])
 
-if sys.platform != 'win32':
-	env.Append(LINKFLAGS=['-Wl,-rpath,' + lib_dir])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode']])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/mrt'])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/sdlx'])
-	env.Append(LINKFLAGS=['-Wl,-rpath-link,build/' + env['mode'] + '/clunk'])
-else: 
+if sys.platform == 'win32':
 	libs.append('sdlx_main')
 	libs.append('user32')
 
 env.Append(LIBPATH=['#/build/' + env['mode'] + '/engine'])
 
-editor = env.Program('bted', ed_src, LIBS=libs, RPATH=[lib_dir])
+editor = env.Program('bted', ed_src, LIBS=libs)
 
 if sys.platform != 'win32' and len(env['prefix']) > 0:
 	Import('install_targets')
