diff -uNr chrpath-0.13.old/killrpath.c chrpath-0.13/killrpath.c
--- chrpath-0.13.old/killrpath.c	2003-06-24 00:46:15.000000000 +0200
+++ chrpath-0.13/killrpath.c	2009-07-19 23:05:11.000000000 +0200
@@ -73,8 +73,11 @@
        if ( ! elf_dynpath_tag(dyns[i].d_tag) )
         dynpos++;
      }
-   for (; dynpos < i; dynpos++)
+   for (; dynpos < i; dynpos++) {
      dyns[dynpos].d_tag = DT_NULL;
+     dyns[dynpos].d_un.d_val = 0x0;
+   }
+
 
    if (lseek(fd, phdr.p_offset, SEEK_SET) == -1
        || write(fd, dyns, phdr.p_filesz) != (int)phdr.p_filesz)
