diff --git a/Source/GUI/Qt/main.cpp b/Source/GUI/Qt/main.cpp
index 922eb7e..6b5ef35 100755
--- a/Source/GUI/Qt/main.cpp
+++ b/Source/GUI/Qt/main.cpp
@@ -28,6 +28,8 @@ int main(int argc, char *argv[])
 {
     QApplication a(argc, argv);
     QStringList args = QCoreApplication::arguments();
+    // # First argument is always app name, not an really argument
+    args.removeFirst();
     bool no_gui=false;
     QStringList filesnames;
     int output=-1;
diff --git a/Source/GUI/Qt/mainwindow.cpp b/Source/GUI/Qt/mainwindow.cpp
index 7afbee8..fb2641f 100755
--- a/Source/GUI/Qt/mainwindow.cpp
+++ b/Source/GUI/Qt/mainwindow.cpp
@@ -103,7 +103,7 @@ MainWindow::MainWindow(QStringList filesnames, int viewasked, QWidget *parent) :
 
     refreshDisplay();
 
-    if(filesnames.count()>1) {
+    if(filesnames.count()>0) {
         openFiles(filesnames);
     }
