diff --git a/Project/QMake/GUI/MediaInfoQt.pro b/Project/QMake/GUI/MediaInfoQt.pro
index 34a35ac..627283a 100755
--- a/Project/QMake/GUI/MediaInfoQt.pro
+++ b/Project/QMake/GUI/MediaInfoQt.pro
@@ -1,17 +1,14 @@
 # -------------------------------------------------
 # Project created by QtCreator 2010-07-23T13:03:11
 # -------------------------------------------------
-TARGET = mediainfo-gui
+TARGET = mediainfo-qt
 TEMPLATE = app
 
 # QT += webkit
 QT += xml
 DEFINES += \
     _UNICODE
-LIBS += -ldl \
-    ../../../../MediaInfoLib/Project/GNU/Library/.libs/libmediainfo.a \
-    ../../../../ZenLib/Project/GNU/Library/.libs/libzen.a \
-    -lz
+LIBS += -ldl -lmediainfo -lzen -lz
 SOURCES += ../../../Source/GUI/Qt/main.cpp \
     ../../../Source/GUI/Qt/mainwindow.cpp \
     ../../../Source/Common/Core.cpp \
diff --git a/Source/GUI/Qt/columneditsheet.cpp b/Source/GUI/Qt/columneditsheet.cpp
index 69858e6..9ebe986 100755
--- a/Source/GUI/Qt/columneditsheet.cpp
+++ b/Source/GUI/Qt/columneditsheet.cpp
@@ -23,7 +23,7 @@ ColumnEditSheet::ColumnEditSheet(column c, int pos, int nb, Core* C, QWidget *pa
     QHBoxLayout(parent),
     pos(pos),
     C(C),
-    col(c),
+    col(c)
 {
     QLineEdit* lineedit = new QLineEdit(c.name);
     this->addWidget(lineedit);
