From 9ac358bf7fe9a6c322edc1fc85a5d288c5f8f92d Mon Sep 17 00:00:00 2001
From: Boris Ranto <branto@redhat.com>
Date: Wed, 25 Mar 2015 21:37:07 +0100
Subject: [PATCH] Hide protected symbols from other modules

---
 src/sys.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sys.h b/src/sys.h
index 25b14bb..f039532 100644
--- a/src/sys.h
+++ b/src/sys.h
@@ -71,8 +71,8 @@
 #endif
 
 #ifndef protected
-# define protected	/* Redefined from elsewhere to "static" */
-			/* When we want to hide everything	*/
+# define protected	__attribute__ ((visibility ("hidden")))
+			/* hide protected symbols from other modules */
 #endif
 
 #ifndef __arraycount
-- 
2.1.0

