diff -rup gpxe-1.0.1.orig/src/config/general.h gpxe-1.0.1.new/src/config/general.h
--- gpxe-1.0.1.orig/src/config/general.h	2010-06-29 20:31:33.000000000 +0100
+++ gpxe-1.0.1.new/src/config/general.h	2012-03-19 11:53:36.035138977 +0000
@@ -31,7 +31,7 @@ FILE_LICENCE ( GPL2_OR_LATER );
  * Timer configuration
  *
  */
-#define BANNER_TIMEOUT	20	/* Tenths of a second for which the shell
+#define BANNER_TIMEOUT	0	/* Tenths of a second for which the shell
 				   banner should appear */
 
 /*
