diff --git a/etc/PackageKit.conf.in b/etc/PackageKit.conf.in
index bc0bea2..67e15f7 100644
--- a/etc/PackageKit.conf.in
+++ b/etc/PackageKit.conf.in
@@ -27,7 +27,7 @@ UseNetworkHeuristic=true
 # enough and will not be set at the very start or end of a transaction.
 #
 # default=true
-UseRemainingTimeEstimation=true
+UseRemainingTimeEstimation=false
 
 # Shut down the daemon after this many seconds idle. 0 means don't shutdown.
 #
