diff -Naur slim-1.3.2/png.c slim-1.3.2-magellan/png.c
--- slim-1.3.2/png.c 2010-07-08 07:04:10.000000000 +0200
+++ slim-1.3.2-magellan/png.c 2011-03-01 20:49:43.000000000 +0100
@@ -57,7 +57,7 @@
     return(0);
     }
   
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
         png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp) NULL);
         fclose(infile);
