diff --exclude=autom4te.cache -urNbw gegl-0.1.6.autoreconf/docs/Makefile.in gegl-0.1.6/docs/Makefile.in
--- gegl-0.1.6.autoreconf/docs/Makefile.in	2011-02-13 19:15:29.000000000 +0100
+++ gegl-0.1.6/docs/Makefile.in	2011-02-22 15:42:51.124566829 +0100
@@ -345,7 +345,7 @@
 # The patterns used for html creation in the gegl build system
 # is very small, and should probably be documented in a tutorial.
 #
-gtkdochtmldir = $(DESTDIR)$(datadir)/gtk-doc/html/gegl
+gtkdochtmldir = $(datadir)/gtk-doc/html/gegl
 HTML_FILES = operations.html class-hierarchy.html $(am__append_1) \
 	$(am__append_3) $(am__append_5)
 GTKDOC_FILES = operations.html gegl.css gegl.devhelp $(am__append_2) \
