diff -up operations/external/ff-load.c.orig operations/external/ff-load.c
--- operations/external/ff-load.c.orig	2011-05-18 22:58:10.302673097 -0300
+++ operations/external/ff-load.c	2011-05-18 23:01:10.734313255 -0300
@@ -69,7 +69,7 @@ print_error (const char *filename, int e
 {
   switch (err)
     {
-    case AVERROR_NUMEXPECTED:
+    case AVERROR(EDOM):
       g_warning ("%s: Incorrect image filename syntax.\n"
                  "Use '%%d' to specify the image number:\n"
                  "  for img1.jpg, img2.jpg, ..., use 'img%%d.jpg';\n"
@@ -79,7 +79,7 @@ print_error (const char *filename, int e
     case AVERROR_INVALIDDATA:
       g_warning ("%s: Error while parsing header\n", filename);
       break;
-    case AVERROR_NOFMT:
+    case AVERROR(EILSEQ):
       g_warning ("%s: Unknown format\n", filename);
       break;
     default:
@@ -278,7 +278,7 @@ prepare (GeglOperation *operation)
       for (i = 0; i< p->ic->nb_streams; i++)
         {
           AVCodecContext *c = p->ic->streams[i]->codec;
-          if (c->codec_type == CODEC_TYPE_VIDEO)
+          if (c->codec_type == AVMEDIA_TYPE_VIDEO)
             {
               p->video_st = p->ic->streams[i];
               p->video_stream = i;
diff -up operations/workshop/external/ff-save.c.orig operations/workshop/external/ff-save.c
--- operations/workshop/external/ff-save.c.orig	2011-05-18 23:01:43.872224730 -0300
+++ operations/workshop/external/ff-save.c	2011-05-18 23:02:39.693285298 -0300
@@ -455,7 +455,7 @@ write_audio_frame (GeglChantO *op, AVFor
                                        p->audio_outbuf_size, p->samples);
 
       pkt.pts = c->coded_frame->pts;
-      pkt.flags |= PKT_FLAG_KEY;
+      pkt.flags |= AV_PKT_FLAG_KEY;
       pkt.stream_index = st->index;
       pkt.data = p->audio_outbuf;
 
@@ -496,7 +496,7 @@ add_video_stream (GeglChantO *op, AVForm
 
   c = st->codec;
   c->codec_id = codec_id;
-  c->codec_type = CODEC_TYPE_VIDEO;
+  c->codec_type = AVMEDIA_TYPE_VIDEO;
 
   /* put sample propeters */
   c->bit_rate = op->bitrate;
@@ -689,7 +689,7 @@ write_video_frame (GeglChantO *op,
       AVPacket  pkt;
       av_init_packet (&pkt);
 
-      pkt.flags |= PKT_FLAG_KEY;
+      pkt.flags |= AV_PKT_FLAG_KEY;
       pkt.stream_index = st->index;
       pkt.data = (uint8_t *) picture_ptr;
       pkt.size = sizeof (AVPicture);
@@ -712,7 +712,7 @@ write_video_frame (GeglChantO *op,
 
           pkt.pts = c->coded_frame->pts;
           if (c->coded_frame->key_frame)
-            pkt.flags |= PKT_FLAG_KEY;
+            pkt.flags |= AV_PKT_FLAG_KEY;
           pkt.stream_index = st->index;
           pkt.data = p->video_outbuf;
           pkt.size = out_size;
