diff -up operations/external/ff-load.c.ffmpeg operations/external/ff-load.c
--- operations/external/ff-load.c.ffmpeg	2012-04-04 22:57:41.738829075 -0300
+++ operations/external/ff-load.c	2012-04-04 22:58:54.140769936 -0300
@@ -271,7 +271,7 @@ prepare (GeglOperation *operation)
       gint err;
 
       ff_cleanup (o);
-      err = av_open_input_file (&p->ic, o->path, NULL, 0, NULL);
+      err = avformat_open_input (&p->ic, o->path, NULL, NULL);
       if (err < 0)
         {
           print_error (o->path, err);
diff -up operations/workshop/external/ff-save.c.ffmpeg operations/workshop/external/ff-save.c
--- operations/workshop/external/ff-save.c.ffmpeg	2012-04-04 22:44:31.726936462 -0300
+++ operations/workshop/external/ff-save.c	2012-04-04 22:55:03.284775923 -0300
@@ -769,12 +769,6 @@ tfile (GeglChantO *self)
      /*XXX: FOO p->audio_st = add_audio_stream (op, p->oc, p->fmt->audio_codec);*/
     }
 
-  if (av_set_parameters (p->oc, NULL) < 0)
-    {
-      fprintf (stderr, "Invalid output format propeters\n%s", "");
-      return -1;
-    }
-
   dump_format (p->oc, 0, self->path, 1);
 
   if (p->video_st)
@@ -782,13 +776,13 @@ tfile (GeglChantO *self)
   if (p->audio_st)
     open_audio (p, p->oc, p->audio_st);
 
-  if (url_fopen (&p->oc->pb, self->path, URL_WRONLY) < 0)
+  if (avio_open (&p->oc->pb, self->path, AVIO_FLAG_WRITE) < 0)
     {
       fprintf (stderr, "couldn't open '%s'\n", self->path);
       return -1;
     }
 
-  av_write_header (p->oc);
+  avformat_write_header (p->oc, NULL);
 
   return 0;
 }
@@ -858,7 +852,7 @@ finalize (GObject *object)
             av_freep (&p->oc->streams[i]);
           }
 
-        url_fclose (&p->oc->pb);
+        avio_close (&p->oc->pb);
         free (p->oc);
       }
       g_free (o->chant_data);
