diff -up gegl-0.2.0/operations/workshop/external/gluas.c.lua-52 gegl-0.2.0/operations/workshop/external/gluas.c
--- gegl-0.2.0/operations/workshop/external/gluas.c.lua-52	2013-05-15 12:29:05.551382036 -0400
+++ gegl-0.2.0/operations/workshop/external/gluas.c	2013-05-15 12:30:05.369382934 -0400
@@ -19,6 +19,9 @@
 #include "config.h"
 #include <glib/gi18n-lib.h>
 
+#ifndef lua_open
+#define lua_open()     luaL_newstate()
+#endif
 
 #ifdef GEGL_CHANT_PROPERTIES
 
@@ -97,7 +100,7 @@ static int l_progress  (lua_State * lua)
 static int l_flush     (lua_State * lua);
 static int l_print     (lua_State * lua);
 
-static const luaL_reg gluas_functions[] =
+static const luaL_Reg gluas_functions[] =
 {
     {"set_rgba",    l_set_rgba},
     {"get_rgba",    l_get_rgba},
@@ -122,7 +125,7 @@ static const luaL_reg gluas_functions[]
 };
 static void
 register_functions (lua_State      *L,
-                    const luaL_reg *l)
+                    const luaL_Reg *l)
 {
   for (;l->name; l++)
     lua_register (L, l->name, l->func);
