From 75a5c4a5042aa3c531977637e5f0b5e62d14978c Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Tue, 19 Oct 2010 16:32:41 +0200
Subject: [PATCH] gegl: make "root" a have file, not global scope

---
 gegl/gegl-instrument.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gegl/gegl-instrument.c b/gegl/gegl-instrument.c
index c43b268..a46b3ca 100644
--- a/gegl/gegl-instrument.c
+++ b/gegl/gegl-instrument.c
@@ -38,7 +38,7 @@ struct _Timing
   Timing *next;
 };
 
-Timing *root = NULL;
+static Timing *root = NULL;
 
 static Timing *iter_next (Timing *iter)
 {
-- 
1.7.3.1

