--- perl-5.10.1/installperl.orig	2009-11-18 15:44:47.000000000 +0100
+++ perl-5.10.1/installperl	2009-11-18 15:50:32.000000000 +0100
@@ -235,7 +235,7 @@
 
 # Do some quick sanity checks.
 
-if (!$opts{notify} && $d_dosuid && $>) { die "You must run as root to install suidperl\n"; }
+# if (!$opts{notify} && $d_dosuid && $>) { die "You must run as root to install suidperl\n"; }
 
    $installbin		|| die "No installbin directory in config.sh\n";
 -d $installbin		|| mkpath($installbin, $opts{verbose}, 0777);
