diff -up perl-5.10.1/lib/Parse/CPAN/Meta.pm.old perl-5.10.1/lib/Parse/CPAN/Meta.pm
--- perl-5.10.1/lib/Parse/CPAN/Meta.pm.old	2009-06-10 18:37:40.000000000 +0200
+++ perl-5.10.1/lib/Parse/CPAN/Meta.pm	2009-12-22 13:08:39.089184165 +0100
@@ -15,7 +15,7 @@ BEGIN {
 	# Class structure
 	require 5.004;
 	require Exporter;
-	$Parse::CPAN::Meta::VERSION   = '1.39';
+	$Parse::CPAN::Meta::VERSION   = '1.40';
 	@Parse::CPAN::Meta::ISA       = qw{ Exporter      };
 	@Parse::CPAN::Meta::EXPORT_OK = qw{ Load LoadFile };
 }
