diff -Nur libmcs-0.7.1-orig/src/tools/mcs-walk-config/mcs_walk_config.c libmcs-0.7.1/src/tools/mcs-walk-config/mcs_walk_config.c
--- libmcs-0.7.1-orig/src/tools/mcs-walk-config/mcs_walk_config.c	2008-04-21 00:05:48.000000000 +0200
+++ libmcs-0.7.1/src/tools/mcs-walk-config/mcs_walk_config.c	2009-12-24 00:10:01.000000000 +0100
@@ -37,7 +37,7 @@
 {
 	mcs_handle_t *h;
 	mowgli_queue_t *groups, *i;
-	int entries = 0, sections = 0;
+	int entries = 0, sections = 0, avgentries = 0;
 
 	if (argc < 2)
 	{
@@ -75,8 +75,9 @@
 	mowgli_queue_destroy(groups);
 	mcs_fini();
 
+    avgentries = sections ? (entries/sections) : 0;
 	printf("\n%d entries across %d groups, average %d entries per group\n",
-		entries, sections, entries / sections);
+		entries, sections, avgentries);
 
 	return 0;
 }
