diff -Nur libmcs-0.7.1-orig/src/backends/default/keyfile.c libmcs-0.7.1/src/backends/default/keyfile.c
--- libmcs-0.7.1-orig/src/backends/default/keyfile.c	2008-04-21 00:05:48.000000000 +0200
+++ libmcs-0.7.1/src/backends/default/keyfile.c	2009-12-29 20:54:48.213678077 +0100
@@ -316,6 +316,8 @@
 
 		if (value != NULL)
 			line->value = strdup(value);
+		else
+			line->value = NULL;
 	}
 	else
 		keyfile_create_line(sec, key, value);
