Bug 965348 - HOST-RESOURCES-MIB::hrFS* not includes btrfs

commit da1fef382591ff45dc92eb3b95a6bfeff9ecfa4f
Author: Jan Safranek <jsafranek@users.sourceforge.net>
Date:   Tue May 21 09:13:41 2013 +0200

    CHANGES: snmpd: Added btrfs support to hrFSTable

diff -up net-snmp-5.7.2/agent/mibgroup/hardware/fsys/fsys_mntent.c.btrfs net-snmp-5.7.2/agent/mibgroup/hardware/fsys/fsys_mntent.c
--- net-snmp-5.7.2/agent/mibgroup/hardware/fsys/fsys_mntent.c.btrfs	2012-10-10 00:28:58.000000000 +0200
+++ net-snmp-5.7.2/agent/mibgroup/hardware/fsys/fsys_mntent.c	2013-05-21 09:22:11.457405066 +0200
@@ -139,6 +139,7 @@ _fsys_type( char *typename )
               !strcmp(typename, MNTTYPE_REISERFS) ||
               !strcmp(typename, MNTTYPE_OCFS2) ||
               !strcmp(typename, MNTTYPE_CVFS) ||
+              !strcmp(typename, MNTTYPE_BTRFS) ||
               !strcmp(typename, MNTTYPE_LOFS))
        return NETSNMP_FS_TYPE_OTHER;
 
diff -up net-snmp-5.7.2/agent/mibgroup/hardware/fsys/mnttypes.h.btrfs net-snmp-5.7.2/agent/mibgroup/hardware/fsys/mnttypes.h
--- net-snmp-5.7.2/agent/mibgroup/hardware/fsys/mnttypes.h.btrfs	2012-10-10 00:28:58.000000000 +0200
+++ net-snmp-5.7.2/agent/mibgroup/hardware/fsys/mnttypes.h	2013-05-21 09:22:33.758303373 +0200
@@ -142,6 +142,9 @@
 #ifndef MNTTYPE_CVFS
 #define MNTTYPE_CVFS     "cvfs"
 #endif
+#ifndef MNTTYPE_BTRFS
+#define MNTTYPE_BTRFS     "btrfs"
+#endif
 
 /*
  *  File systems to skip
