--- layout/lxde-applications.menu.orig	2014-09-08 10:19:08.000000000 -0300
+++ layout/lxde-applications.menu	2014-10-22 23:16:13.691279982 -0200
@@ -134,6 +134,17 @@
 		</Include>
 	</Menu>	 <!-- End System Tools -->
 
+	<!-- Wine-->
+	<Menu>
+		<Name>System</Name>
+		<Directory>wine.directory</Directory>
+		<Include>
+			<And>
+				<Category>X-Wine</Category>
+			</And>
+		</Include>
+	</Menu>	 <!-- End Wine -->
+
 	<!-- Other -->
 	<Menu>
 		<Name>Other</Name>
