--- lxmenu-data-0.1.1.orig/layout/lxde-applications.menu	2009-05-19 18:52:11.000000000 +0200
+++ lxmenu-data-0.1.1/layout/lxde-applications.menu	2009-12-05 23:24:09.000000000 +0100
@@ -20,28 +20,11 @@
 		<Include>
 			<And>
 				<Category>Utility</Category>
-	<!-- Accessibility spec must have either the Utility or Settings
-			 category, and we display an accessibility submenu already for
-			 the ones that do not have Settings, so don't display accessibility
-			 applications here -->
-				<Not><Category>Accessibility</Category></Not>
 				<Not><Category>System</Category></Not>
 			</And>
 		</Include>
 	</Menu> <!-- End Accessories -->
 
-	<!-- Accessibility submenu -->
-	<Menu>
-		<Name>Universal Access</Name>
-		<Directory>lxde-utility-accessibility.directory</Directory>
-		<Include>
-			<And>
-				<Category>Accessibility</Category>
-				<Not><Category>Settings</Category></Not>
-			</And>
-		</Include>
-	</Menu> <!-- End Accessibility -->
-
 	<!-- Development Tools -->
 	<Menu>
 		<Name>Development</Name>
@@ -148,29 +131,43 @@
 		</Include>
 		</Menu> <!-- End Other -->
 
+	<!-- Preferences -->
 	<Menu>
-		<Name>DesktopSettings</Name>
+		<Name>Preferences</Name>
 		<Directory>lxde-settings.directory</Directory>
-		<OnlyUnallocated/>
 		<Include>
-			<Or>
+			<And>
+				<Category>Settings</Category>
+				<Not>
+					<Or>
+						<Category>System</Category>
+						<Category>X-XfceSettingsDialog</Category>
+						<Filename>gnomecc.desktop</Filename>
+						<Filename>gnome-default-applications.desktop</Filename>
+					</Or>
+				</Not>
+			</And>
+		</Include>
+	</Menu>		<!-- End Preferences -->
+
+	<!-- System Settings -->
+	<Menu>
+		<Name>Administration</Name>
+		<Directory>lxde-settings-system.directory</Directory>
+		<Include>
+			<And>
 				<Category>Settings</Category>
-				<Category>PackageManager</Category>
 				<Category>System</Category>
-			</Or>
+			</And>
 		</Include>
-		<Layout>
-			<Merge type="menus"/>
-			<Merge type="files"/>
-		</Layout>
-
-	</Menu> <!-- End Settings -->
+	</Menu>		 <!-- End System Settings -->
 
 	<Layout>
 		<Merge type="files"/>
 		<Merge type="menus"/>
 		<Separator/>
-		<Menuname>DesktopSettings</Menuname>
+		<Menuname>Preferences</Menuname>
+		<Menuname>Administration</Menuname>
 	</Layout>
 
 </Menu> <!-- End Applications -->
