--- decoder_plugins/ffmpeg/ffmpeg.c.ffmpeg	2015-11-07 00:25:23.780895282 -0200
+++ decoder_plugins/ffmpeg/ffmpeg.c	2015-11-07 00:25:33.019939353 -0200
@@ -40,9 +40,6 @@ GCC_DIAG_OFF(deprecated-declarations)
 #include <libavformat/avformat.h>
 GCC_DIAG_ON(deprecated-declarations)
 #include <libavutil/mathematics.h>
-#ifdef HAVE_AV_GET_CHANNEL_LAYOUT_NB_CHANNELS
-#include <libavutil/audioconvert.h>
-#endif
 
 /* FFmpeg also likes common names, without that, our common.h and log.h
  * would not be included. */
