--- Makefile	2007-08-27 12:43:22.000000000 +0200
+++ Makefile.opt	2007-08-29 19:57:24.671354495 +0200
@@ -46,7 +46,7 @@
 export ECHO    = echo
 
 # -DDEBUG and -g only used during development
-CFLAGS += -Wall -pedantic -std=gnu99 -Wundef -Wcast-align -W -Wpointer-arith -Wwrite-strings -Wno-unused-parameter `pkg-config --cflags gtk+-2.0`
+CFLAGS += ${OPTFLAGS} -pedantic -std=gnu99 -Wundef -Wcast-align -W -Wpointer-arith -Wwrite-strings -Wno-unused-parameter `pkg-config --cflags gtk+-2.0`
 ifndef WITHOUT_NLS
 	CFLAGS += -DENABLE_NLS
 endif
--- bk/Makefile	2007-08-27 12:43:22.000000000 +0200
+++ bk/Makefile.opt	2007-08-29 19:58:00.627517349 +0200
@@ -3,7 +3,7 @@
 OBJECTS = bkRead7x.o bkAdd.o bkDelete.o bkExtract.o bkRead.o bkPath.o bkMangle.o bkWrite.o bkWrite7x.o bkTime.o bkSort.o bkError.o bkGet.o bkSet.o bkCache.o bkLink.o bkMisc.o
 
 # -DDEBUG and -g only used during development
-CFLAGS  += -Wall -pedantic -std=gnu99 -Wundef -Wcast-align -W -Wpointer-arith -Wwrite-strings -Wno-unused-parameter
+CFLAGS  += ${OPTFLAGS} -pedantic -std=gnu99 -Wundef -Wcast-align -W -Wpointer-arith -Wwrite-strings -Wno-unused-parameter
 
 # the _FILE_OFFSET_BITS=64 is to enable stat() for large files
 CPPFLAGS = -D_FILE_OFFSET_BITS=64
--- iniparser-2.17/Makefile	2007-07-27 07:27:35.000000000 +0200
+++ iniparser-2.17/Makefile.opt	2007-08-29 20:01:37.364315099 +0200
@@ -4,7 +4,7 @@
 
 # Compiler settings
 CC      = gcc
-CFLAGS  = -O3 -fPIC
+CFLAGS  = ${OPTFLAGS} -fPIC
 
 # Ar settings to build the library
 AR	    = ar
