--- Makefile.in.orig	2015-08-27 07:30:22.000000000 -0300
+++ Makefile.in	2015-09-09 20:27:43.946783216 -0300
@@ -136,12 +136,6 @@ $(BINDIR)/Debug/plugins/%@DLLEXT@: src/p
 $(BINDIR)/Release/plugins/%@DLLEXT@: src/plugins/%/* $(BINDIR)/Release/$(PACKAGE_TARNAME)$(EXEEXT)
 	@make $(PLUGIN_RELEASE_ARGS) -C $(dir $<) Release
 
-$(BINDIR)/Debug/plugins/j3270@DLLEXT@: src/java/* $(BINDIR)/Debug/$(PACKAGE_TARNAME)$(EXEEXT)
-	@make BINDIR="../../$(BINDIR)" PW3270_LIBS="-L../../$(BINDIR)/Debug@DLLDIR@ -l3270 -lpw3270" PW3270_CFLAGS="-I../include" -C $(dir $<) Debug
-
-$(BINDIR)/Release/plugins/j3270@DLLEXT@: src/java/* $(BINDIR)/Release/$(PACKAGE_TARNAME)$(EXEEXT)
-	make BINDIR="../../$(BINDIR)" PW3270_LIBS="-L../../$(BINDIR)/Release@DLLDIR@ -l3270 -lpw3270" PW3270_CFLAGS="-I../include" -C $(dir $<) Release
-
 $(DESTDIR)$(libdir)/$(PACKAGE_NAME)-plugins/%@DLLEXT@: $(BINDIR)/Release/plugins/%@DLLEXT@
 	@$(MKDIR) `dirname $@`
 	@$(INSTALL_PROGRAM) "$^" "$@"
@@ -173,21 +167,6 @@ $(BINDIR)/Release/$(PACKAGE_NAME).oxt:
 	@$(MKDIR) `dirname $@`
 	@cp src/loffice/.bin/Release/$(PACKAGE_NAME).oxt $@
 
-php: $(BINDIR)/Release/tn3270@DLLEXT@
-
-$(BINDIR)/Release/tn3270.so: src/php/*
-	@$(MAKE) -C src/php Release
-	@$(MKDIR) `dirname $@`
-	@cp src/php/.bin/Release/php3270.so $@
-
-java: $(BINDIR)/Release/@DLLPREFIX@jni3270@DLLEXT@
-
-$(BINDIR)/Release/@DLLPREFIX@jni3270@DLLEXT@: src/java/*
-	@$(MKDIR) `dirname $@`
-	@$(MAKE) BINDIR="../../$(BINDIR)" LIB3270_LIBS="$(RLSLIB)" LIB3270_CFLAGS="-I../../src/include" -C src/java Release
-
-rexx: $(BINDIR)/Debug/plugins/rx3270@DLLEXT@
-
 #---[ Debug targets ]----------------------------------------------------------
 
 Debug: $(BINDIR)/Debug/$(PACKAGE_TARNAME)$(EXEEXT) \
@@ -376,15 +355,6 @@ install-sdk:
 install-oxt:
 	@$(MAKE) BINDIR=../../.bin LIB3270_CFLAGS="-I../../src/include" -C src/loffice install
 
-install-php:
-	@$(MAKE) -C src/php install
-
-install-java:
-	@$(MAKE) BINDIR=../../.bin LIB3270_CFLAGS="-I../../src/include" -C src/java install
-
-install-rexx:
-	@$(MAKE) BINDIR=../../../.bin LIB3270_CFLAGS="-I../../src/include" -C src/plugins/rx3270 install
-
 clean:
 	@rm -fr .obj
 	@rm -fr .bin
@@ -394,11 +364,8 @@ clean:
 	@make -C src/pw3270 clean
 	@make -C src/tools clean
 	@make -C src/plugins/hllapi clean
-	@make -C src/plugins/rx3270 clean
 	@make -C src/plugins/dbus3270 clean
 	@make -C src/loffice clean
-	@make -C src/php clean
-	@make -C src/java clean
 	@make -C src/classlib clean
 	@rm -f $(PACKAGE_NAME).png
 	@rm -f $(PACKAGE_NAME)-logo.png
