diff -up telepathy-logger-qt-0.5.80/CMakeLists.txt.version telepathy-logger-qt-0.5.80/CMakeLists.txt
--- telepathy-logger-qt-0.5.80/CMakeLists.txt.version	2013-03-06 18:55:06.000000000 -0600
+++ telepathy-logger-qt-0.5.80/CMakeLists.txt	2013-03-07 07:23:20.414510145 -0600
@@ -40,7 +40,7 @@ endif(POLICY CMP0011)
 #   set nano_version to 1
 set(TP_LOGGER_QT4_MAJOR_VERSION 0)
-set(TP_LOGGER_QT4_MINOR_VERSION 5)
+set(TP_LOGGER_QT4_MINOR_VERSION 6)
-set(TP_LOGGER_QT4_MICRO_VERSION 60)
+set(TP_LOGGER_QT4_MICRO_VERSION 2)
 set(TP_LOGGER_QT4_NANO_VERSION  0)
 
 # This value contains the library's SOVERSION. This value is to be increased everytime an API/ABI break
