diff -up telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4.pc.in.deps telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4.pc.in
--- telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4.pc.in.deps	2012-06-10 05:30:54.000000000 -0500
+++ telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4.pc.in	2012-07-05 14:58:06.276318158 -0500
@@ -6,6 +6,6 @@ includedir=${CMAKE_INSTALL_PREFIX}/${INC
 Name: Telepathy-Logger-Qt4
 Description: Qt4 utility library to access logs in the Telepathy framework
 Version: ${PACKAGE_VERSION}
-Requires.private: QtCore >= 4.5, TelepathyQt4 >= 0.5
+Requires.private: QtCore >= 4.5, QtGLib-2.0, TelepathyQt4 >= 0.5
 Libs: -L${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR} -ltelepathy-logger-qt4
 Cflags: -I${CMAKE_INSTALL_PREFIX}/${INCLUDE_INSTALL_DIR}/telepathy-logger-0.2
diff -up telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4-uninstalled.pc.in.deps telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4-uninstalled.pc.in
--- telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4-uninstalled.pc.in.deps	2012-06-10 05:30:54.000000000 -0500
+++ telepathy-logger-qt-0.4.0/TelepathyLoggerQt4/TelepathyLoggerQt4-uninstalled.pc.in	2012-07-05 14:58:35.530952423 -0500
@@ -6,6 +6,6 @@ abs_top_srcdir=${CMAKE_SOURCE_DIR}
 Name: Telepathy-Logger-Qt4 (uninstalled copy)
 Description: Qt4 utility library to access logs in the Telepathy framework
 Version: ${PACKAGE_VERSION}
-Requires.private: QtCore >= 4.5, TelepathyQt4 >= 0.5
+Requires.private: QtCore >= 4.5, QtGLib-2.0, TelepathyQt4 >= 0.5
 Libs: ${CMAKE_BINARY_DIR}/TelepathyQt/libtelepathy-logger-qt4.a
 Cflags: -I${CMAKE_SOURCE_DIR} -I${CMAKE_BINARY_DIR}
