--- mcomix-0.98/mcomix/i18n.py.debug	2011-09-22 19:53:56.000000000 +0900
+++ mcomix-0.98/mcomix/i18n.py	2012-06-06 10:00:00.000000000 +0900
@@ -84,6 +84,14 @@
             translation = gettext.GNUTranslations(
                     pkg_resources.resource_stream('mcomix.messages', resource))
             break
+        else:
+            root_path = os.path.normpath('/')
+            base_dir = os.path.join(root_path, 'usr')
+            locale_dir = os.path.join (base_dir, 'share', 'locale')
+            if gettext.find(domain=domain, localedir=locale_dir, languages=[lang]):
+                translation = gettext.translation(domain=domain, localedir=locale_dir,
+                    languages=[lang], fallback=False)
+                break
 
     translation.install(unicode=True)
 
