diff -Nur audacious-2.2-orig/man/audacious2.1.in audacious-2.2/man/audacious2.1.in
--- audacious-2.2-orig/man/audacious2.1.in	2009-11-22 23:43:46.000000000 +0100
+++ audacious-2.2/man/audacious2.1.in	2010-04-16 12:41:56.000000000 +0200
@@ -37,12 +37,12 @@
 
 .TP
 .I -e, --enqueue
-Do not clear the current playlist before adding the files on the command line to
-it.
+Add the files on the command line to the current playlist but do not start
+playback.
 
 .TP
 .I -E, --enqueue-to-temp
-Add the files on the command line to a new playlist.
+Add the files on the command line to a temporary playlist and start playback.
 
 .PP
 .B Controlling playback:
diff -Nur audacious-2.2-orig/src/audacious/main.c audacious-2.2/src/audacious/main.c
--- audacious-2.2-orig/src/audacious/main.c	2009-11-22 23:43:47.000000000 +0100
+++ audacious-2.2/src/audacious/main.c	2010-04-16 12:43:53.000000000 +0200
@@ -165,7 +165,7 @@
     {"play-pause", 't', 0, G_OPTION_ARG_NONE, &options.play_pause, N_("Pause if playing, play otherwise"), NULL},
     {"fwd", 'f', 0, G_OPTION_ARG_NONE, &options.fwd, N_("Skip forward in playlist"), NULL},
     {"show-jump-box", 'j', 0, G_OPTION_ARG_NONE, &options.show_jump_box, N_("Display Jump to File dialog"), NULL},
-    {"enqueue", 'e', 0, G_OPTION_ARG_NONE, &options.enqueue, N_("Don't clear the playlist"), NULL},
+    {"enqueue", 'e', 0, G_OPTION_ARG_NONE, &options.enqueue, N_("Add files to the playlist"), NULL},
     {"enqueue-to-temp", 'E', 0, G_OPTION_ARG_NONE, &options.enqueue_to_temp, N_("Add new files to a temporary playlist"), NULL},
     {"show-main-window", 'm', 0, G_OPTION_ARG_NONE, &options.mainwin, N_("Display the main window"), NULL},
     {"activate", 'a', 0, G_OPTION_ARG_NONE, &options.activate, N_("Display all open Audacious windows"), NULL},
