diff -Nur audacious-2.2-orig/src/audacious/ui_fileinfo.c audacious-2.2/src/audacious/ui_fileinfo.c
--- audacious-2.2-orig/src/audacious/ui_fileinfo.c	2009-11-22 23:43:47.000000000 +0100
+++ audacious-2.2/src/audacious/ui_fileinfo.c	2010-03-20 18:57:36.000000000 +0100
@@ -941,6 +941,7 @@
 void ui_fileinfo_show (gint playlist, gint entry)
 {
     const gchar * filename = playlist_entry_get_filename (playlist, entry);
+    char *unescaped = NULL;
     InputPlugin * decoder = playlist_entry_get_decoder (playlist, entry);
     Tuple * tuple;
 
@@ -959,13 +960,15 @@
         g_free (probe);
     }
 
+    unescaped = g_uri_unescape_string(filename,NULL);
     if (decoder == NULL)
     {
         gchar * message = g_strdup_printf ("No decoder found for %s.\n",
-         filename);
+                                           unescaped);
 
         hook_call ("interface show error", message);
         g_free (message);
+        g_free(unescaped);
         return;
     }
 
@@ -985,10 +988,11 @@
     else
     {
         gchar * message = g_strdup_printf ("No info available for "
-         "%s.\n", filename);
+         "%s.\n", unescaped);
 
         hook_call ("interface show error", message);
         g_free (message);
+        g_free(unescaped);
     }
 }
 
