diff -Nur audacious-3.2-alpha1-orig/src/audacious/Makefile audacious-3.2-alpha1/src/audacious/Makefile
--- audacious-3.2-alpha1-orig/src/audacious/Makefile	2011-12-19 21:43:44.000000000 +0100
+++ audacious-3.2-alpha1/src/audacious/Makefile	2011-12-24 09:53:13.356974535 +0100
@@ -121,6 +121,7 @@
 		${LIBINTL}						\
 		${EGGSM_LIBS}					\
 		${GLIB_LIBS}					\
+		`pkg-config --libs gmodule-2.0` \
 		${GTHREAD_LIBS} \
 		${GTK_LIBS}						\
 		${DBUS_LIBS}					\
