diff -Nur audacious-2.4.0-orig/src/libaudcore/vfs_common.c audacious-2.4.0/src/libaudcore/vfs_common.c
--- audacious-2.4.0-orig/src/libaudcore/vfs_common.c	2010-08-26 01:04:25.000000000 +0200
+++ audacious-2.4.0/src/libaudcore/vfs_common.c	2010-11-06 12:00:09.382940382 +0100
@@ -139,6 +139,9 @@
  */
 void vfs_file_get_contents (const gchar * filename, void * * buf, gint64 * size)
 {
+    * buf = NULL;
+    * size = 0;
+
     VFSFile *fd;
     gsize filled_size = 0, buf_size = 4096;
     guchar * ptr;
