Description: Register FLAC file type in .desktop file.
Author: Benjamin Drung <bdrung@debian.org>
Bug-Debian: http://bugs.debian.org/661921
Bug: http://bugzilla.audacityteam.org/show_bug.cgi?id=483

--- a/src/audacity.desktop.in
+++ b/src/audacity.desktop.in
@@ -17,4 +17,4 @@
 Exec=@AUDACITY_NAME@ %F
 StartupNotify=false
 Terminal=false
-MimeType=application/ogg;audio/basic;audio/mpeg;audio/x-aiff;audio/x-mp3;audio/x-wav;application/x-audacity-project;
+MimeType=application/ogg;audio/basic;audio/mpeg;audio/x-aiff;audio/x-mp3;audio/x-wav;application/x-audacity-project;@MIMETYPES@
--- a/configure.in
+++ b/configure.in
@@ -89,6 +89,8 @@
 dnl audacity. Each name in the list should be a _target_ in lib-src/Makefile
 AC_SUBST(LIBSRC_BUILD)
 
+AC_SUBST(MIMETYPES)
+
 dnl allow the user to specify options to configure that change the
 dnl name "audacity" anywhere it appears in a pathname.   This allows
 dnl multiple versions of Audacity to be installed concurrently
@@ -482,6 +484,9 @@
 for lib in $LIBRARIES ; do
    eval LIB_USE_LOCAL=\$${lib}_USE_LOCAL
    eval LIB_USE_SYSTEM=\$${lib}_USE_SYSTEM
+   if test "$LIB_USE_LOCAL" = yes -o "$LIB_USE_SYSTEM" = yes; then
+      eval MIMETYPES=\"${MIMETYPES}\$${lib}_MIMETYPES\"
+   fi
    if test "x$LIB_USE_LOCAL" = "xyes" ; then
       eval LIBSRC_BUILD=\"$LIBSRC_BUILD \$${lib}_LOCAL_BUILD\"
       eval LOCAL_LIBS=\"$LOCAL_LIBS \$${lib}_LOCAL_LIBS\"
--- a/m4/audacity_checklib_libflac.m4
+++ b/m4/audacity_checklib_libflac.m4
@@ -68,5 +68,7 @@
    else
       AC_MSG_NOTICE([FLAC libraries are NOT available in this source tree])
    fi
+
+   LIBFLAC_MIMETYPES="audio/flac;audio/x-flac;"
 ])
 
