--- lib-src/portmixer/configure.ac.nomac	2010-07-31 22:31:56.031028327 -0300
+++ lib-src/portmixer/configure.ac	2010-07-31 22:32:00.013908664 -0300
@@ -64,10 +64,6 @@
 if [[ $have_support = "no" ]] ; then
 		AC_MSG_WARN("Missing support in pa_win_wmme.h");
 fi
-AC_EGREP_HEADER([PaMacCore_GetStreamInputDevice], [pa_mac_core.h], , [have_support=no])
-if [[ $have_support = "no" ]] ; then
-		AC_MSG_WARN("Missing support in pa_mac_core.h");
-fi
 AC_EGREP_HEADER([PaOSS_GetStreamInputDevice], [pa_unix_oss.h], , [have_support=no])
 if [[ $have_support = "no" ]] ; then
 		AC_MSG_WARN("Missing support in pa_unix_oss.h");
