diff -Nur audacity-src-1.3.9-orig/src/effects/ladspa/LoadLadspa.cpp audacity-src-1.3.9/src/effects/ladspa/LoadLadspa.cpp
--- audacity-src-1.3.9-orig/src/effects/ladspa/LoadLadspa.cpp	2009-08-31 12:45:40.000000000 +0200
+++ audacity-src-1.3.9/src/effects/ladspa/LoadLadspa.cpp	2009-09-03 12:01:49.823505468 +0200
@@ -286,7 +286,7 @@
 
    #ifdef __WXGTK__
    wxGetApp().AddUniquePathToPathList(wxT(INSTALL_PREFIX) wxT("/ladspa"), pathList);
-   wxGetApp().AddUniquePathToPathList(wxT("/usr/local/lib/ladspa"), pathList);
+   wxGetApp().AddUniquePathToPathList(wxT("__RPM_LIBDIR__/ladspa"), pathList);
    wxGetApp().AddUniquePathToPathList(wxT(LIBDIR) wxT("/ladspa"), pathList);
    #endif
 
