diff -ur ../audacity-src-1.3.12-beta.orig/configure.in ./configure.in
--- ../audacity-src-1.3.12-beta.orig/configure.in	2010-03-30 12:32:42.000000000 +0100
+++ ./configure.in	2010-04-02 17:03:07.154028262 +0100
@@ -559,6 +559,8 @@
 esac
 
 dnl PortAudio configuration
+AC_ARG_WITH([alsa], AS_HELP_STRING([--without-alsa], [Build without alsa library (default: test)]))
+AC_ARG_WITH([jack], AS_HELP_STRING([--without-jack], [Build without jack library (default: test)]))
 
 LOCAL_LIBS="$LOCAL_LIBS portaudio-v19/lib/libportaudio.a"
 EXTRAOBJS="$EXTRAOBJS ../lib-src/portaudio-v19/lib/libportaudio.a"
@@ -575,13 +577,21 @@
       ;;
    *)
       dnl Unix
-      AC_CHECK_LIB(asound, snd_pcm_open, have_alsa=yes, have_alsa=no)
-      if [[ $have_alsa = "yes" ]] ; then
-         LIBS="$LIBS -lasound"
-      fi
-      PKG_CHECK_MODULES(JACK, jack, have_jack=yes, have_jack=no)
-      if [[ $have_jack = "yes" ]] ; then
-         LIBS="$LIBS $JACK_LIBS"
+      if test "x$with_alsa" != "xno"; then
+         AC_CHECK_LIB(asound, snd_pcm_open, have_alsa=yes, have_alsa=no)
+         if [[ $have_alsa = "yes" ]] ; then
+            LIBS="$LIBS -lasound"
+        else
+            AC_MSG_WARN([Support for alsa not available])
+         fi
+      fi        
+      if test "x$with_jack" != "xno"; then
+         PKG_CHECK_MODULES(JACK, jack, have_jack=yes, have_jack=no)
+         if [[ $have_jack = "yes" ]] ; then
+            LIBS="$LIBS $JACK_LIBS"
+        else
+            AC_MSG_WARN([Support for jack not available])
+         fi
       fi
       AC_CHECK_LIB(hpi, HPI_SubSysCreate, have_asihpi=yes, have_asihpi=no, -lm)
       if [[ $have_asihpi = "yes" ]] ; then
Only in .: configure.in.orig
diff -ur ../audacity-src-1.3.12-beta.orig/lib-src/portmixer/configure.ac ./lib-src/portmixer/configure.ac
--- ../audacity-src-1.3.12-beta.orig/lib-src/portmixer/configure.ac	2010-03-30 12:29:47.000000000 +0100
+++ ./lib-src/portmixer/configure.ac	2010-04-02 17:03:07.154028262 +0100
@@ -97,6 +97,7 @@
 AC_CHECK_HEADER(alsa/asoundlib.h,  have_alsa=yes, have_alsa=no)
 AC_CHECK_HEADER(CoreAudio/CoreAudio.h, have_coreaudio=yes, have_coreaudio=no)
 
+AC_ARG_WITH([alsa], AS_HELP_STRING([--without-alsa], [Build without alsa library (default: test)]))
 #
 # Set up to use the identified ones
 #
@@ -107,7 +108,7 @@
    objects="$objects px_unix_oss.o"
 fi
 
-if [[ $have_alsa = "yes" ]] ; then
+if [[ $have_alsa = "yes" -a "x$with_alsa" != "xno" ]] ; then
    AC_MSG_NOTICE(Including support for ALSA);
    AC_DEFINE(PX_USE_LINUX_ALSA)
    objects="$objects px_linux_alsa.o"
Only in ./lib-src/portmixer: configure.ac.orig
