diff -up src/FFmpeg.cpp.ffmpeg src/FFmpeg.cpp
--- src/FFmpeg.cpp.ffmpeg	2010-03-30 08:28:54.000000000 -0300
+++ src/FFmpeg.cpp	2010-07-31 22:43:31.764028517 -0300
@@ -204,7 +204,7 @@ static int ufile_read(URLContext *h, uns
    return ret;
 }
 
-static int ufile_write(URLContext *h, unsigned char *buf, int size)
+static int ufile_write(URLContext *h, const unsigned char *buf, int size)
 {
    return (int) ((wxFile *) h->priv_data)->Write(buf, size);
 }
@@ -356,7 +356,7 @@ int ufile_fopen_input(AVFormatContext **
          }
          // Otherwize, resort to extension matching if available
          else if (fmt1->extensions) {
-            if (FFmpegLibsInst->match_ext(filename, fmt1->extensions)) {
+            if (FFmpegLibsInst->av_match_ext(filename, fmt1->extensions)) {
                score = 50;
             }
          }
@@ -829,7 +829,7 @@ bool FFmpegLibs::InitLibs(wxString libpa
    INITDYN(avformat,av_open_input_file);
    INITDYN(avformat,av_open_input_stream);
    INITDYN(avformat,get_buffer);
-   INITDYN(avformat,match_ext);
+   INITDYN(avformat,av_match_ext);
 
 #if FFMPEG_STABLE
    INITDYN(avformat,av_init_packet);
diff -up src/FFmpeg.h.ffmpeg src/FFmpeg.h
--- src/FFmpeg.h.ffmpeg	2010-03-30 08:28:54.000000000 -0300
+++ src/FFmpeg.h	2010-07-31 22:44:36.456903659 -0300
@@ -224,7 +224,7 @@ public:
    AVStream*         (*av_new_stream)                 (AVFormatContext *s, int id);
    AVFormatContext*  (*av_alloc_format_context)       (void);
    AVOutputFormat*   (*guess_format)                  (const char *short_name, const char *filename, const char *mime_type);
-   int               (*match_ext)                     (const char *filename, const char *extensions);
+   int               (*av_match_ext)                  (const char *filename, const char *extensions);
    int               (*av_write_trailer)              (AVFormatContext *s);
    int               (*av_interleaved_write_frame)    (AVFormatContext *s, AVPacket *pkt);
    int               (*av_write_frame)                (AVFormatContext *s, AVPacket *pkt);
