--- audacity-minsrc-2.1.0/src/effects/ladspa/LadspaEffect.cpp.orig	2015-03-31 00:37:32.707165333 -0300
+++ audacity-minsrc-2.1.0/src/effects/ladspa/LadspaEffect.cpp	2015-03-31 00:39:45.994083154 -0300
@@ -282,7 +282,7 @@ wxArrayString LadspaEffectsModule::FindP
    
    pathList.Add(wxGetHomeDir() + wxFILE_SEP_PATH + wxT(".ladspa"));
    pathList.Add(wxT("/usr/local/lib/ladspa"));
-   pathList.Add(wxT("/usr/lib/ladspa"));
+   pathList.Add(wxT("__RPM_LIBDIR__/ladspa"));
    pathList.Add(wxT(LIBDIR) wxT("/ladspa"));
 
    // Recursively scan for all shared objects
