--- audacity-src-2.0.6/src/effects/ladspa/LoadLadspa.cpp.orig	2014-09-28 11:27:19.956684235 -0300
+++ audacity-src-2.0.6/src/effects/ladspa/LoadLadspa.cpp	2014-09-28 11:31:01.414373598 -0300
@@ -287,8 +287,7 @@ void LoadLadspaPlugins()
 
    #ifdef __WXGTK__
    wxGetApp().AddUniquePathToPathList(wxT("~/.ladspa"), pathList);
-   wxGetApp().AddUniquePathToPathList(wxT("/usr/local/lib/ladspa"), pathList);
-   wxGetApp().AddUniquePathToPathList(wxT("/usr/lib/ladspa"), pathList);
+   wxGetApp().AddUniquePathToPathList(wxT("__RPM_LIBDIR__/ladspa"), pathList);
    wxGetApp().AddUniquePathToPathList(wxT(LIBDIR) wxT("/ladspa"), pathList);
    #endif
 
