diff -up mutt-1.5.18/doc/Muttrc.head.muttrc mutt-1.5.18/doc/Muttrc.head
--- mutt-1.5.18/doc/Muttrc.head.muttrc	2008-01-30 05:26:50.000000000 +0100
+++ mutt-1.5.18/doc/Muttrc.head	2008-05-19 10:58:21.000000000 +0200
@@ -19,11 +19,15 @@ macro index,pager,attach,compose \cb "\
 
 # Show documentation when pressing F1
 macro generic,pager <F1> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
+# and also F2, as some terminals use F1
+macro generic,pager <F2> "<shell-escape> less @docdir@/manual.txt<Enter>" "show Mutt documentation"
 
 # show the incoming mailboxes list (just like "mutt -y") and back when pressing "y"
 macro index,pager y "<change-folder>?<toggle-mailboxes>" "show incoming mailboxes list"
 bind browser y exit
 
+bind editor <delete> delete-char
+
 # If Mutt is unable to determine your site's domain name correctly, you can
 # set the default here.
 #
diff -up mutt-1.5.18/contrib/sample.muttrc mutt-1.5.18/contrib/sample
