diff -up mutt-1.5.21/crypt-gpgme.c.notation mutt-1.5.21/crypt-gpgme.c
--- mutt-1.5.21/crypt-gpgme.c.notation	2012-04-25 10:26:20.589226791 +0200
+++ mutt-1.5.21/crypt-gpgme.c	2012-04-25 10:28:02.075915855 +0200
@@ -72,7 +72,8 @@
 #define xtoi_2(p)   ((xtoi_1(p) * 16) + xtoi_1((p)+1))
 
 #define PKA_NOTATION_NAME "pka-address@gnupg.org"
-#define is_pka_notation(notation) (! strcmp ((notation)->name, \
+#define is_pka_notation(notation) ((notation)->name && \ 
+                                   ! strcmp ((notation)->name, \ 
 					     PKA_NOTATION_NAME))
 
 /* Values used for comparing addresses. */
