diff -up ./doc/manual.xml.head.add_debug_option ./doc/manual.xml.head
--- ./doc/manual.xml.head.add_debug_option	2014-04-30 17:31:42.062694536 +0200
+++ ./doc/manual.xml.head	2014-04-30 17:53:10.230668282 +0200
@@ -8182,6 +8182,7 @@ mailboxes and to send messages from the
 <row><entry>-a</entry><entry>attach a file to a message</entry></row>
 <row><entry>-b</entry><entry>specify a blind carbon-copy (BCC) address</entry></row>
 <row><entry>-c</entry><entry>specify a carbon-copy (Cc) address</entry></row>
+<row><entry>-d</entry><entry>log debugging output to ~/.muttdebug0 if mutt was complied with +DEBUG; it can range from 1-5 and affects verbosity (a value of 2 is recommended)</entry></row>
 <row><entry>-D</entry><entry>print the value of all Mutt variables to stdout</entry></row>
 <row><entry>-e</entry><entry>specify a config command to be run after initialization files are read</entry></row>
 <row><entry>-f</entry><entry>specify a mailbox to load</entry></row>
