diff -up xmms2-0.8DrO_o/wscript.versionsanity xmms2-0.8DrO_o/wscript
--- xmms2-0.8DrO_o/wscript.versionsanity	2011-10-20 17:26:08.000000000 -0200
+++ xmms2-0.8DrO_o/wscript	2011-10-25 18:10:42.797192152 -0200
@@ -19,7 +19,7 @@ from waflib import Configure, Options, U
 from waftools.compiler_flags import compiler_flags
 from waftools import gittools
 
-BASEVERSION="0.8 DrO_o"
+BASEVERSION="0.8"
 APPNAME='xmms2'
 
 top = '.'
@@ -283,17 +283,7 @@ def configure(conf):
     conf.check_tool('gcc')
     conf.check_tool('g++')
 
-    if conf.options.target_platform:
-        Options.platform = conf.options.target_platform
-
-    nam,changed = gittools.get_info()
-    conf.msg("git commit id", nam)
-    if conf.options.customversion:
-        conf.env.VERSION = "%s (%s + %s)" % (BASEVERSION, nam, conf.options.customversion)
-    else:
-        dirty = changed and "-dirty" or ""
-        conf.msg("uncommited changed", changed and "yes" or "no")
-        conf.env.VERSION = "%s (git commit: %s%s)" % (BASEVERSION, nam, dirty)
+    conf.env["VERSION"] = BASEVERSION
 
     conf.env.append_unique('CFLAGS', ['-g', '-O0'])
     conf.env.append_unique('CXXFLAGS', ['-g', '-O0'])
