--- src/plugins/avcodec/avcodec.c.orig	2011-05-18 20:32:11.963933994 -0300
+++ src/plugins/avcodec/avcodec.c	2011-05-18 20:32:48.641164035 -0300
@@ -160,7 +160,7 @@ xmms_avcodec_init (xmms_xform_t *xform)
 		goto err;
 	}
 
-	if (codec->type != CODEC_TYPE_AUDIO) {
+	if (codec->type != AVMEDIA_TYPE_AUDIO) {
 		XMMS_DBG ("Codec '%s' found but its type is not audio", data->codec_id);
 		goto err;
 	}
