--- src/Map.cpp.orig	2012-09-09 21:52:49.730115001 +0800
+++ src/Map.cpp	2010-09-05 02:38:27.000000000 +0800
@@ -116,7 +116,7 @@
     for (boost::filesystem::directory_iterator dir_iter(path);
          dir_iter != end_iter; dir_iter++) {
       if (dir_iter->path().extension() == ".map") {
+        map_list.push_back(new MapEntry(path, dir_iter->path().filename()));
-        map_list.push_back(new MapEntry(path, dir_iter->filename()));
       }
     }
   }
@@ -126,7 +126,7 @@
     for (boost::filesystem::directory_iterator dir_iter(path);
          dir_iter != end_iter; dir_iter++) {
       if (dir_iter->path().extension() == ".map") {
+        map_list.push_back(new MapEntry(path, dir_iter->path().filename()));
-        map_list.push_back(new MapEntry(path, dir_iter->filename()));
       }
     }
   }
--- src/MapEntry.cpp.orig	2012-09-09 21:58:17.314363412 +0800
+++ src/MapEntry.cpp	2012-09-09 22:03:09.152693552 +0800
@@ -50,7 +50,7 @@
   enabled = true;
 
   // extract map name
-  map_name = filename.stem();
+  map_name = filename.stem().string();
 
   // is the file writable?
   if (access(path / filename, W_OK) == 0) {
