From 3aecef6474094f520b9df954865dc7ce15ed1870 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Sat, 24 May 2014 15:08:44 +0200
Subject: [PATCH 2/2] HTML-based formats are not supported yet

---
 README | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/README b/README
index e1dceeb..7ff1dbb 100644
--- a/README
+++ b/README
@@ -2,19 +2,13 @@ libe-book is a library and a set of tools for reading and converting
 various reflowable e-book formats.
 
 Currently supported are:
-- ePub (v.2)
 - eReader .pdb
 - FictionBook v. 2 (including zipped files)
-- HTML 1(?)-5
-- MS HTML Help
 - PalmDoc Ebook
 - Plucker .pdb
-- Open eBook (OEB)
 - QiOO (mobile format, for java-enabled cellphones)
-- Rocket eBook
 - TCR (simple compressed text format)
 - TealDoc
-- XHTML 1.0
 - zTXT
 - ZVR (simple compressed text format)
 
-- 
1.9.0

