diff -up pcc-090808/pcc-libs/csu/linux/Makefile.orig pcc-090808/pcc-libs/csu/linux/Makefile
--- pcc-090808/pcc-libs/csu/linux/Makefile.orig	2009-01-23 09:10:09.000000000 +0200
+++ pcc-090808/pcc-libs/csu/linux/Makefile	2009-08-08 13:42:43.616850463 +0300
@@ -25,7 +25,7 @@ crtn.o: $(MACH)/crtn.c
 	$(CC) -O $(CFLAGS) -c -o $@ $<
 
 crtbegin.o: crtbegin.c
-	$(CC) -O $(CFLAGS) -c -o $@ $<
+	$(CC) -O $(CFLAGS_NODEBUG) -c -o $@ $<
 
 crtend.o: crtend.c
 	$(CC) -O $(CFLAGS) -c -o $@ $<
