diff -pruN Canna37p3.orig/server/Imakefile Canna37p3/server/Imakefile
--- Canna37p3.orig/server/Imakefile	2004-04-27 06:48:37.000000000 +0900
+++ Canna37p3/server/Imakefile	2011-02-25 17:00:56.000000000 +0900
@@ -3,11 +3,7 @@
 
          CANNAROOT = ..
      INSTUGIDFLAGS = /* -m 6555 $(cannaOwnerGroup) */
-#ifdef __EMX__
          INSTFLAGS = $(INSTUGIDFLAGS)
-#else
-         INSTFLAGS = -s $(INSTUGIDFLAGS)
-#endif
            DEFINES = $(CANNASERVER_DEFINES) $(SIGNAL_DEFINES) ConnectionFlags
           INCLUDES = -I$(CANNAROOT)/include
 
