--- xine-lib-1.2.0/configure.ac.opt	2012-01-01 15:55:17.000000000 -0200
+++ xine-lib-1.2.0/configure.ac	2012-01-02 20:53:41.989751669 -0200
@@ -741,7 +741,7 @@ case "$host_or_hostalias" in
                 CC_CHECK_CFLAGS([-mtune=i386], [sarchopt="-mtune"],
                     [CC_CHECK_CFLAGS([-mcpu=i386], [sarchopt="-mcpu"],
                         [CC_CHECK_CFLAGS([-march=i386], [sarchopt="-march"], [sarchopt="no"])])])
-                if test "$sarchopt" != "no"; then
+                if false ; then
                     case "$host_or_hostalias" in
                         i386-*)         archopt_val="i386" ;;
                         i486-*)         archopt_val="i486" ;;
