diff -Nurp lilv-0.16.0/lilv/lilvmm.hpp lilv-0.16.0.new/lilv/lilvmm.hpp
--- lilv-0.16.0/lilv/lilvmm.hpp	2013-01-14 10:37:52.000000000 +1000
+++ lilv-0.16.0.new/lilv/lilvmm.hpp	2013-02-24 12:22:36.727156297 +1000
@@ -192,7 +192,7 @@ struct Plugin {
 	LILV_WRAP0(Node,        plugin, get_author_homepage);
 	LILV_WRAP0(bool,        plugin, is_replaced);
 	LILV_WRAP0(Nodes,       plugin, get_extension_data);
-	LILV_WRAP0(UIs,         plugin, get_uis);
+	LILV_WRAP0(LilvUIs*,    plugin, get_uis);
 	LILV_WRAP1(Nodes,       plugin, get_related, Node, type);
 
 	inline Port get_port_by_index(unsigned index) {
