diff -up gnome-menus-3.0.0/layout/applications.menu.other-docs gnome-menus-3.0.0/layout/applications.menu
--- gnome-menus-3.0.0/layout/applications.menu.other-docs	2011-03-31 02:48:41.000000000 -0400
+++ gnome-menus-3.0.0/layout/applications.menu	2011-04-04 19:06:56.826452755 -0400
@@ -147,6 +147,7 @@
         <Not><Category>Core</Category></Not>
         <Not><Category>Screensaver</Category></Not>
         <Not><Category>X-GNOME-Settings-Panel</Category></Not>
+        <Not><Category>Documentation</Category></Not>
       </And>
     </Include>
   </Menu> <!-- End Other -->
