diff -up kdelibs-3.5.10/admin/acinclude.m4.in.kde3dirs kdelibs-3.5.10/admin/acinclude.m4.in
--- kdelibs-3.5.10/admin/acinclude.m4.in.kde3dirs	2008-08-19 17:28:39.000000000 -0300
+++ kdelibs-3.5.10/admin/acinclude.m4.in	2012-10-16 21:00:39.076671470 -0300
@@ -428,7 +428,7 @@ AC_DEFUN([KDE_SET_DEFAULT_PATHS],
 if test "$1" = "default"; then
 
   if test -z "$kde_htmldir"; then
-    kde_htmldir='\${datadir}/doc/HTML'
+    kde_htmldir='\${prefix}/doc/HTML'
   fi
   if test -z "$kde_appsdir"; then
     kde_appsdir='\${datadir}/applnk'
@@ -440,7 +440,7 @@ if test "$1" = "default"; then
     kde_sounddir='\${datadir}/sounds'
   fi
   if test -z "$kde_datadir"; then
-    kde_datadir='\${datadir}/apps'
+    kde_datadir='\${datadir}/kde3/apps'
   fi
   if test -z "$kde_locale"; then
     kde_locale='\${datadir}/locale'
@@ -449,16 +449,16 @@ if test "$1" = "default"; then
     kde_cgidir='\${exec_prefix}/cgi-bin'
   fi
   if test -z "$kde_confdir"; then
-    kde_confdir='\${datadir}/config'
+    kde_confdir='\${datadir}/kde3/config'
   fi
   if test -z "$kde_kcfgdir"; then
-    kde_kcfgdir='\${datadir}/config.kcfg'
+    kde_kcfgdir='\${datadir}/kde3/config.kcfg'
   fi
   if test -z "$kde_mimedir"; then
     kde_mimedir='\${datadir}/mimelnk'
   fi
   if test -z "$kde_toolbardir"; then
-    kde_toolbardir='\${datadir}/toolbar'
+    kde_toolbardir='\${datadir}/kde3/toolbar'
   fi
   if test -z "$kde_wallpaperdir"; then
     kde_wallpaperdir='\${datadir}/wallpapers'
@@ -470,10 +470,10 @@ if test "$1" = "default"; then
     kde_bindir='\${exec_prefix}/bin'
   fi
   if test -z "$kde_servicesdir"; then
-    kde_servicesdir='\${datadir}/services'
+    kde_servicesdir='\${datadir}/kde3/services'
   fi
   if test -z "$kde_servicetypesdir"; then
-    kde_servicetypesdir='\${datadir}/servicetypes'
+    kde_servicetypesdir='\${datadir}/kde3/servicetypes'
   fi
   if test -z "$kde_moduledir"; then
     if test "$kde_qtver" = "2"; then
