diff -up FvwmCompositor/main/AWidgets.cxx.gcc45 FvwmCompositor/main/AWidgets.cxx
--- FvwmCompositor/main/AWidgets.cxx.gcc45	2008-06-30 05:11:59.000000000 -0300
+++ FvwmCompositor/main/AWidgets.cxx	2011-03-25 19:09:48.136241018 -0300
@@ -15,6 +15,8 @@
 
 #include "AWidgets.H"
 
+#include <stdio.h>
+
 #include "main/AScreen.H"
 #include "desktop/MetisseWindow.H"
 
diff -up FvwmCompositor/main/ExtendedInputs.cxx.gcc45 FvwmCompositor/main/ExtendedInputs.cxx
--- FvwmCompositor/main/ExtendedInputs.cxx.gcc45	2008-07-02 09:23:32.000000000 -0300
+++ FvwmCompositor/main/ExtendedInputs.cxx	2011-03-25 19:09:51.403553758 -0300
@@ -15,6 +15,8 @@
 
 #include "ExtendedInputs.H"
 
+#include <stdio.h>
+
 #include <nucleo/gl/window/glWindow.H>
 #include <nucleo/gl/glUtils.H>
 
diff -up FvwmCompositor/main/LayerManager.cxx.gcc45 FvwmCompositor/main/LayerManager.cxx
--- FvwmCompositor/main/LayerManager.cxx.gcc45	2008-08-25 10:54:44.000000000 -0300
+++ FvwmCompositor/main/LayerManager.cxx	2011-03-25 19:10:07.619142486 -0300
@@ -16,6 +16,7 @@
 
 #include "LayerManager.H"
 
+#include <stdio.h>
 #include <X11/X.h>
 
 #include "main/AScreen.H"
diff -up FvwmCompositor/renderer/goodies.cxx.gcc45 FvwmCompositor/renderer/goodies.cxx
--- FvwmCompositor/renderer/goodies.cxx.gcc45	2008-06-26 09:32:30.000000000 -0300
+++ FvwmCompositor/renderer/goodies.cxx	2011-03-25 19:07:04.710633240 -0300
@@ -16,6 +16,7 @@
 
 #include "goodies.H"
 
+#include <stdio.h>
 #include <math.h>
 
 #include "fvwmmodule/AFvwm.H"
