diff -up gnote-0.7.3/configure\~ gnote-0.7.3/configure
--- gnote-0.7.3/configure~	2010-11-04 14:20:08.000000000 +0100
+++ gnote-0.7.3/configure	2011-02-04 23:54:01.000000000 +0100
@@ -16394,7 +16394,7 @@ rm -f core conftest.err conftest_ipa8_co
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$boost_cv_lib_system" = xyes; then
-        boost_cv_lib_system_LDFLAGS="-L$boost_ldpath -R$boost_ldpath"
+        boost_cv_lib_system_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
         break 6
       else
         boost_failed_libs="$boost_failed_libs@$boost_lib@"

Diff finished.  Fri Feb  4 23:54:06 2011
