diff -up lib/locale.rb.orig lib/locale.rb
--- lib/locale.rb.orig	2010-07-04 19:36:14.000000000 -0300
+++ lib/locale.rb	2010-07-04 19:48:48.345958661 -0300
@@ -236,9 +236,11 @@ module Locale
     end
 
     tags = []
-    (0...candidate_tags[0].size).each {|i|
-      tags += candidate_tags.collect{|v| v[i]}
-    }
+    unless candidate_tags.empty?
+      (0...candidate_tags[0].size).each {|i|
+        tags += candidate_tags.collect{|v| v[i]}
+      }
+    end
     tags += default_tags
     tags.uniq!
 
diff -up lib/locale/driver/env.rb.orig lib/locale/driver/env.rb
--- lib/locale/driver/env.rb.orig	2010-07-04 19:36:14.000000000 -0300
+++ lib/locale/driver/env.rb	2010-07-04 19:49:43.022679235 -0300
@@ -39,13 +39,16 @@ module Locale
       # Gets the locales from environment variables. (LANGUAGE > LC_ALL > LC_MESSAGES > LANG)
       # * Returns: an Array of the locale as Locale::Tag::Posix or nil.
       def locales
-        if (locales = ENV["LANGUAGE"])
-          Locale::TagList.new(locales.split(/:/).collect{|v| Locale::Tag::Posix.parse(v)})
+        locales = ENV["LANGUAGE"]
+        if (locales != nil and locales.size > 0)
+          locs = locales.split(/:/).collect{|v| Locale::Tag::Posix.parse(v)}.compact
+          if locs.size > 0
+            return Locale::TagList.new(locs)
+          end
         elsif (loc = locale)
-          Locale::TagList.new([loc])
-        else
-          nil
+          return Locale::TagList.new([loc])
         end
+        nil
       end
 
       # Gets the charset from environment variable or return nil.
