diff -up rsyslog-5.8.6/rsyslog.service.in.orig rsyslog-5.8.6/rsyslog.service.in
--- rsyslog-5.8.6/rsyslog.service.in.orig	2011-10-21 07:53:02.000000000 -0200
+++ rsyslog-5.8.6/rsyslog.service.in	2012-01-17 19:29:53.521805154 -0200
@@ -2,10 +2,12 @@
 Description=System Logging Service
 
 [Service]
+EnvironmentFile=-/etc/default/rsyslog
 ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
-ExecStart=@sbindir@/rsyslogd -n -c5
+ExecStart=@sbindir@/rsyslogd -n $SYSLOGD_OPTIONS
 Sockets=syslog.socket
 StandardOutput=null
 
 [Install]
 WantedBy=multi-user.target
+Alias=syslog.service
