diff -up rsyslog-7.4.1/rsyslog.service.in.orig rsyslog-7.4.1/rsyslog.service.in
--- rsyslog-7.4.1/rsyslog.service.in.orig	2013-06-17 15:28:54.430023493 +0200
+++ rsyslog-7.4.1/rsyslog.service.in	2013-06-17 15:30:05.874378084 +0200
@@ -4,7 +4,8 @@ Requires=syslog.socket
 
 [Service]
 Type=notify
-ExecStart=@sbindir@/rsyslogd -n
+EnvironmentFile=-/etc/default/rsyslog
+ExecStart=@sbindir@/rsyslogd -n $SYSLOGD_OPTIONS
 StandardOutput=null
 
 [Install]
