diff -up rsyslog-5.8.11/rsyslog.service.in.orig rsyslog-5.8.11/rsyslog.service.in
--- rsyslog-5.8.11/rsyslog.service.in.orig	2012-06-20 18:28:14.561697697 +0200
+++ rsyslog-5.8.11/rsyslog.service.in	2012-06-20 18:28:45.554308105 +0200
@@ -2,8 +2,8 @@
 Description=System Logging Service
 
 [Service]
-ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
-ExecStart=@sbindir@/rsyslogd -n -c5
+EnvironmentFile=-/etc/default/rsyslog
+ExecStart=@sbindir@/rsyslogd -n $SYSLOGD_OPTIONS
 Sockets=syslog.socket
 StandardOutput=null
 
