diff -up rsyslog-5.8.5/rsyslog.service.in.orig rsyslog-5.8.5/rsyslog.service.in
--- rsyslog-5.8.5/rsyslog.service.in.orig	2011-08-31 12:13:51.000000000 +0200
+++ rsyslog-5.8.5/rsyslog.service.in	2011-09-05 12:49:18.966194902 +0200
@@ -2,8 +2,9 @@
 Description=System Logging Service
 
 [Service]
+EnvironmentFile=-/etc/default/rsyslog
 ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
-ExecStart=@sbindir@/rsyslogd -n -c5
+ExecStart=@sbindir@/rsyslogd -n $SYSLOGD_OPTIONS
 Sockets=syslog.socket
 StandardOutput=null
 
