diff -up rsyslog-5.8.11/runtime/nsd_gtls.c.orig rsyslog-5.8.11/runtime/nsd_gtls.c
--- rsyslog-5.8.11/runtime/nsd_gtls.c.orig	2012-05-23 17:45:55.088757411 +0200
+++ rsyslog-5.8.11/runtime/nsd_gtls.c	2012-05-23 17:46:34.949256350 +0200
@@ -257,8 +257,8 @@ gtlsClientCertCallback(gnutls_session se
 static rsRetVal
 gtlsGetCertInfo(nsd_gtls_t *pThis, cstr_t **ppStr)
 {
-	char dn[128];
-	uchar lnBuf[256];
+	char dn[256];
+	uchar lnBuf[512];
 	size_t size;
 	unsigned int algo, bits;
 	time_t expiration_time, activation_time;
