From 489541d2a901c5fedd9cbbdc8ad0df557fdabddd Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Tue, 11 Jun 2013 16:27:37 +0200
Subject: [PATCH] bugfix: prevent calling tellLostCnt() twice

---
 runtime/ratelimit.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/runtime/ratelimit.c b/runtime/ratelimit.c
index ec24855..8f5bcef 100644
--- a/runtime/ratelimit.c
+++ b/runtime/ratelimit.c
@@ -157,9 +157,9 @@ withinRatelimit(ratelimit_t *ratelimit, time_t tt)
 
 	/* resume if we go out of out time window */
 	if(tt > ratelimit->begin + ratelimit->interval) {
-		tellLostCnt(ratelimit);
 		ratelimit->begin = 0;
 		ratelimit->done = 0;
+		tellLostCnt(ratelimit);
 	}
 
 	/* do actual limit check */
-- 
1.7.10.4

