--- rsyslog-6.4.0/rsyslog.service.in.orig	2012-09-03 22:42:00.144635762 -0300
+++ rsyslog-6.4.0/rsyslog.service.in	2012-09-03 22:43:13.474031505 -0300
@@ -2,8 +2,8 @@
 Description=System Logging Service
 
 [Service]
-ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
-ExecStart=@sbindir@/rsyslogd -n -c6
+EnvironmentFile=-/etc/default/rsyslog
+ExecStart=@sbindir@/rsyslogd -n $SYSLOGD_OPTIONS
 Sockets=syslog.socket
 StandardOutput=null
 
