From ac6c45537391d67beb55b03ccf301bdaa150cd8a Mon Sep 17 00:00:00 2001
From: Tomas Heinrich <theinric@redhat.com>
Date: Mon, 3 Jun 2013 15:09:07 +0200
Subject: [PATCH] imuxsock: don't report an error message for
 $OmitLocalLogging

---
 plugins/imuxsock/imuxsock.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/imuxsock/imuxsock.c b/plugins/imuxsock/imuxsock.c
index c503852..dad09ab 100644
--- a/plugins/imuxsock/imuxsock.c
+++ b/plugins/imuxsock/imuxsock.c
@@ -1284,6 +1284,8 @@ BEGINactivateCnfPrePrivDrop
 	instanceConf_t *inst;
 CODESTARTactivateCnfPrePrivDrop
 	runModConf = pModConf;
+	if(runModConf->bOmitLocalLogging && nfd == 1)
+		ABORT_FINALIZE(RS_RET_OK);
 	for(inst = runModConf->root ; inst != NULL ; inst = inst->next) {
 		addListner(inst);
 	}
@@ -1325,6 +1327,8 @@ BEGINrunInput
 #endif
 
 CODESTARTrunInput
+	if(runModConf->bOmitLocalLogging && nfd == 1)
+		ABORT_FINALIZE(RS_RET_OK);
 	/* this is an endless loop - it is terminated when the thread is
 	 * signalled to do so. This, however, is handled by the framework,
 	 * right into the sleep below.
-- 
1.7.10.4

