diff -up src/applib/cmdex_sync_gui.cpp.orig src/applib/cmdex_sync_gui.cpp
--- src/applib/cmdex_sync_gui.cpp.orig	2013-04-21 21:57:22.729460812 -0300
+++ src/applib/cmdex_sync_gui.cpp	2013-04-21 21:57:37.199375349 -0300
@@ -9,6 +9,7 @@
 /// \weakgroup applib
 /// @{
 
+#include <glibmm.h>
 #include <gtkmm/main.h>  // Gtk::Main
 
 #include "hz/fs_path.h"
diff -up src/gsc_init.cpp.orig src/gsc_init.cpp
--- src/gsc_init.cpp.orig	2013-04-21 21:59:06.092420735 -0300
+++ src/gsc_init.cpp	2013-04-21 21:59:18.225833577 -0300
@@ -15,6 +15,7 @@
 #include <cstdio>  // std::printf
 #include <vector>
 #include <sstream>
+#include <glibmm.h>
 #include <gtkmm/main.h>
 #include <gtkmm/messagedialog.h>
 #include <gtk/gtk.h>  // gtk_window_set_default_icon_name, gtk_icon_theme_*
