diff -urN supertux-0.3.5.old/src/scripting/scripting.cpp supertux-0.3.5/src/scripting/scripting.cpp
--- supertux-0.3.5.old/src/scripting/scripting.cpp	2015-04-11 14:50:03.000000000 +0100
+++ supertux-0.3.5/src/scripting/scripting.cpp	2015-04-13 09:48:48.063689466 +0100
@@ -99,7 +99,7 @@
   sq_pop(global_vm, 1);
 
   // register print function
-  sq_setprintfunc(global_vm, printfunc, printfunc);
+  sq_setprintfunc(global_vm, printfunc);
   // register default error handlers
   sqstd_seterrorhandlers(global_vm);
 
