diff -up firefox-35.0/mozilla-release/browser/app/profile/firefox.js.flash firefox-35.0/mozilla-release/browser/app/profile/firefox.js
--- firefox-35.0/mozilla-release/browser/app/profile/firefox.js.flash	2015-01-22 12:19:27.000000000 +0100
+++ firefox-35.0/mozilla-release/browser/app/profile/firefox.js	2015-01-22 12:21:45.139134814 +0100
@@ -691,7 +691,7 @@ pref("plugin.defaultXpi.state", 2);
 
 // Flash is enabled by default, and Java is click-to-activate by default on
 // all channels.
-pref("plugin.state.flash", 2);
+pref("plugin.state.flash", 1);
 pref("plugin.state.java", 1);
 
 // Whitelist Requests
