--- mozilla-release/browser/base/content/browser.js.orig	2013-06-14 19:02:34.039471674 -0300
+++ mozilla-release/browser/base/content/browser.js	2013-06-14 19:03:31.624301862 -0300
@@ -3559,10 +3559,12 @@ const BrowserSearch = {
    */
   recordSearchInHealthReport: function (engine, source) {
 #ifdef MOZ_SERVICES_HEALTHREPORT
-    let reporter = Cc["@mozilla.org/datareporting/service;1"]
-                     .getService()
-                     .wrappedJSObject
-                     .healthReporter;
+    // let reporter = Cc["@mozilla.org/datareporting/service;1"]
+    //                 .getService()
+    //                 .wrappedJSObject
+    //                 .healthReporter;
+
+    return;
 
     // This can happen if the FHR component of the data reporting service is
     // disabled. This is controlled by a pref that most will never use.
