Index: src/source/compiler/asllookup.c
===================================================================
--- src.orig/source/compiler/asllookup.c
+++ src/source/compiler/asllookup.c
@@ -119,6 +119,7 @@ LkIsObjectUsed (
 {
     ACPI_NAMESPACE_NODE     *Node = ACPI_CAST_PTR (ACPI_NAMESPACE_NODE, ObjHandle);
     ACPI_NAMESPACE_NODE     *Next;
+    ACPI_NAME_UNION         tmp;
     ASL_METHOD_LOCAL        *MethodLocals;
     ASL_METHOD_LOCAL        *MethodArgs;
     UINT32                  i;
@@ -227,8 +228,9 @@ LkIsObjectUsed (
              * Issue a remark even if it is a reserved name (starts
              * with an underscore).
              */
+            ACPI_MOVE_32_TO_32(&tmp.Ascii, Next->Name.Ascii);
             sprintf (MsgBuffer, "Name is within method [%4.4s]",
-                Next->Name.Ascii);
+                tmp.Ascii);
             AslError (ASL_REMARK, ASL_MSG_NOT_REFERENCED,
                 LkGetNameOp (Node->Op), MsgBuffer);
             return (AE_OK);
