diff -Naur acpica-unix2-20150818/source/compiler/asllookup.c acpica-unix2-20150818.fixes/source/compiler/asllookup.c
--- acpica-unix2-20150818/source/compiler/asllookup.c	2015-09-10 16:07:37.573262497 -0600
+++ acpica-unix2-20150818.fixes/source/compiler/asllookup.c	2015-09-10 16:26:26.336912875 -0600
@@ -176,7 +176,8 @@
                  * We ignore the predefined methods since often, not
                  * all arguments are needed or used.
                  */
-                if ((Node->Name.Ascii[0] != '_') &&
+                ACPI_MOVE_32_TO_32(&tmp.Ascii, Node->Name.Ascii);
+                if ((tmp.Ascii[0] != '_') &&
                     (!(MethodArgs[i].Flags & ASL_ARG_REFERENCED)))
                 {
                     sprintf (MsgBuffer, "Arg%u", i);
