--- xfe-1.40/configure.ac.syssn	2014-12-05 23:51:29.000000000 +0900
+++ xfe-1.40/configure.ac	2015-01-13 14:53:37.283949293 +0900
@@ -170,11 +170,16 @@
 AC_MSG_RESULT([$enable_sn])
 AC_SUBST(STARTUPNOTIFY,false)
 if test "x$enable_sn" != "xno"; then
-	CXXFLAGS="${CXXFLAGS} -DSTARTUP_NOTIFICATION"
-	AC_SUBST(STARTUPNOTIFY,true)
-	enable_sn=yes
+	AC_CHECK_PROGS(pkg_config,pkg-config,)
+	if test -n "$pkg_config"; then
+		PKG_CHECK_MODULES(LIBSN, libstartup-notification-1.0)
+		CXXFLAGS="${CXXFLAGS} -DSTARTUP_NOTIFICATION -DSN_API_NOT_YET_FROZEN"
+		AC_SUBST(STARTUPNOTIFY,true)
+		enable_sn=yes
+	fi
 fi
 AM_CONDITIONAL(STARTUPNOTIFY, [test x$enable_sn = xyes])
+AC_SUBST(LIBSN_LIBS)
 
 
 # Building for debugging
--- xfe-1.40/src/Makefile.am.syssn	2014-11-25 19:31:00.000000000 +0900
+++ xfe-1.40/src/Makefile.am	2015-01-13 14:55:30.512706409 +0900
@@ -42,18 +42,7 @@
 			  XFileExplorer.cpp \
               main.cpp
 
-if STARTUPNOTIFY
-xfe_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfe_LDADD = @LIBINTL@ -lutil
+xfe_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 
 xfp_SOURCES = ../xvt/command.c \
@@ -81,18 +70,7 @@
  			  DirHistBox.cpp \
               XFilePackage.cpp
 
-if STARTUPNOTIFY
-xfp_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfp_LDADD = @LIBINTL@ -lutil
+xfp_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 	  
 
 xfw_SOURCES = ../xvt/command.c \
@@ -122,18 +100,7 @@
 			  FontDialog.cpp \
               XFileWrite.cpp
 
-if STARTUPNOTIFY
-xfw_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfw_LDADD = @LIBINTL@ -lutil
+xfw_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 				
 xfi_SOURCES = ../xvt/command.c \
@@ -161,23 +128,12 @@
  			  PathLinker.cpp \
               XFileImage.cpp
 
-if STARTUPNOTIFY
-xfi_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfi_LDADD = @LIBINTL@ -lutil
+xfi_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 
 localedir = $(datadir)/locale
 
-AM_CPPFLAGS = -I. -I$(top_srcdir) -I$(top_srcdir)/intl
+AM_CPPFLAGS = -I. -I$(top_srcdir) -I$(top_srcdir)/intl @LIBSN_CFLAGS@
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 
 EXTRA_DIST =  ../libsn/sn-common.h	\
