--- xfe-1.34/configure.ac.syssn	2013-02-07 17:49:40.000000000 +0900
+++ xfe-1.34/configure.ac	2013-05-04 17:07:43.000000000 +0900
@@ -165,11 +165,16 @@
 AC_MSG_RESULT([$enable_sn])
 AC_SUBST(STARTUPNOTIFY,false)
 if test "x$enable_sn" != "xno"; then
-	CXXFLAGS="${CXXFLAGS} -DSTARTUP_NOTIFICATION"
-	AC_SUBST(STARTUPNOTIFY,true)
-	enable_sn=yes
+	AC_CHECK_PROGS(pkg_config,pkg-config,)
+	if test -n "$pkg_config"; then
+		PKG_CHECK_MODULES(LIBSN, libstartup-notification-1.0)
+		CXXFLAGS="${CXXFLAGS} -DSTARTUP_NOTIFICATION -DSN_API_NOT_YET_FROZEN"
+		AC_SUBST(STARTUPNOTIFY,true)
+		enable_sn=yes
+	fi
 fi
 AM_CONDITIONAL(STARTUPNOTIFY, [test x$enable_sn = xyes])
+AC_SUBST(LIBSN_LIBS)
 
 
 # Building for debugging
--- xfe-1.34/src/Makefile.am.syssn	2013-02-07 17:48:39.000000000 +0900
+++ xfe-1.34/src/Makefile.am	2013-05-04 17:13:07.000000000 +0900
@@ -40,18 +40,7 @@
 			  XFileExplorer.cpp \
               main.cpp
 
-if STARTUPNOTIFY
-xfe_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfe_LDADD = @LIBINTL@ -lutil
+xfe_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 
 xfp_SOURCES = ../xvt/command.c \
@@ -79,18 +68,7 @@
  			  DirHistBox.cpp \
               XFilePackage.cpp
 
-if STARTUPNOTIFY
-xfp_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfp_LDADD = @LIBINTL@ -lutil
+xfp_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 
 xfv_SOURCES = ../xvt/command.c \
@@ -120,18 +98,7 @@
 			  FontDialog.cpp \
               XFileView.cpp
 
-if STARTUPNOTIFY
-xfv_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfv_LDADD = @LIBINTL@ -lutil
+xfv_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 	  
 
 xfw_SOURCES = ../xvt/command.c \
@@ -161,18 +128,7 @@
 			  FontDialog.cpp \
               XFileWrite.cpp
 
-if STARTUPNOTIFY
-xfw_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfw_LDADD = @LIBINTL@ -lutil
+xfw_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 				
 xfi_SOURCES = ../xvt/command.c \
@@ -200,23 +156,12 @@
  			  PathLinker.cpp \
               XFileImage.cpp
 
-if STARTUPNOTIFY
-xfi_SOURCES += ../libsn/sn-common.c  \
-              ../libsn/sn-launchee.c \
-              ../libsn/sn-launcher.c \
-              ../libsn/sn-list.c \
-              ../libsn/sn-monitor.c \
-              ../libsn/sn-util.c \
-              ../libsn/sn-xmessages.c \
-              ../libsn/sn-xutils.c
-endif
-
-xfi_LDADD = @LIBINTL@ -lutil
+xfi_LDADD = @LIBINTL@ -lutil @LIBSN_LIBS@
 
 
 localedir = $(datadir)/locale
 
-AM_CPPFLAGS = -I. -I$(top_srcdir) -I$(top_srcdir)/intl
+AM_CPPFLAGS = -I. -I$(top_srcdir) -I$(top_srcdir)/intl @LIBSN_CFLAGS@
 DEFS = -DLOCALEDIR=\"$(localedir)\" @DEFS@
 
 EXTRA_DIST =  ../libsn/sn-common.h	\
