--- src/players.cpp.orig	2013-05-24 22:32:59.209367303 -0300
+++ src/players.cpp	2013-05-24 22:33:10.199029686 -0300
@@ -50,6 +50,7 @@ QString Player::executable(bool * found)
 Players::Players() {
 	list.push_back( Player("SMPlayer", "smplayer", true) );
 	list.push_back( Player("MPlayer", "mplayer", false, "-title ") );
+	list.push_back( Player("MPlayer2", "mplayer2", false, "-title ") );
 	list.push_back( Player("VLC", "vlc", false, "--meta-title=") );
 	list.push_back( Player("Dragon Player", "dragon", false) );
 	list.push_back( Player("Totem", "totem", false) );
