--- src/players.cpp.orig	2015-09-15 22:31:48.206744743 -0300
+++ src/players.cpp	2015-09-15 22:33:38.007574218 -0300
@@ -117,13 +117,13 @@ Players::Players() {
 	list.push_back( Player("SMPlayer", "smplayer", "%u", true, Player::Video) );
 	list.push_back( Player("SMPlayer (audio)", "smplayer", "%u -media-title %t", true, Player::Audio) );
 	list.push_back( Player("SMPlayer (add to playlist)", "smplayer", "-add-to-playlist %u", true, Player::VideoAudio) );
-	list.push_back( Player("MPlayer", "mplayer", "%u -title %t", false, Player::Video) );
+	list.push_back( Player("MPlayer", "mplayer", "%u -title %t -quiet", false, Player::Video) );
 	list.push_back( Player("VLC", "vlc", "%u --meta-title=%t", false, Player::VideoAudio) );
 	list.push_back( Player("Dragon Player", "dragon", "%u", false, Player::VideoAudio) );
 	list.push_back( Player("Totem", "totem", "%u", false, Player::VideoAudio) );
 	list.push_back( Player("GNOME-MPlayer", "gnome-mplayer", "%u", false, Player::VideoAudio) );
-	list.push_back( Player("mpv", "mpv", "%u --title=%t", false, Player::Video) );
-	list.push_back( Player("mpv + youtube-dl", "mpv", "--ytdl --ytdl-format=best %u", true, Player::Video) );
+	list.push_back( Player("mpv", "mpv", "%u --title=%t --quiet", false, Player::Video) );
+	list.push_back( Player("mpv + youtube-dl", "mpv", "--ytdl --ytdl-format=best %u --quiet", true, Player::Video) );
 	#ifdef D_PLAYERS
 	list.push_back( Player("uget", "uget-gtk", "--quiet --folder=/tmp --filename=%f %u", false, Player::VideoAudio) );
 	list.push_back( Player("yoDownet", "yoDownet", "%u -o %f", false, Player::VideoAudio) );
