--- src/players.cpp.orig	2015-05-21 19:31:38.726733126 -0300
+++ src/players.cpp	2015-05-21 19:35:44.758375176 -0300
@@ -112,13 +112,13 @@ Players::Players() {
 #else
 	list.push_back( Player("SMPlayer", "smplayer", "%u", true) );
 	list.push_back( Player("SMPlayer (add to playlist)", "smplayer", "-add-to-playlist %u", true) );
-	list.push_back( Player("MPlayer", "mplayer", "%u -title %t", false) );
+	list.push_back( Player("MPlayer", "mplayer", "%u -title %t -quiet", false) );
 	list.push_back( Player("VLC", "vlc", "%u --meta-title=%t", false) );
 	list.push_back( Player("Dragon Player", "dragon", "%u", false) );
 	list.push_back( Player("Totem", "totem", "%u", false) );
 	list.push_back( Player("GNOME-MPlayer", "gnome-mplayer", "%u", false) );
-	list.push_back( Player("mpv", "mpv", "%u --title=%t", false) );
-	list.push_back( Player("mpv + youtube-dl", "mpv", "--ytdl --ytdl-format=best %u", true) );
+	list.push_back( Player("mpv", "mpv", "%u --title=%t --quiet", false) );
+	list.push_back( Player("mpv + youtube-dl", "mpv", "--ytdl --ytdl-format=best %u --quiet", true) );
 	/*
 	list.push_back( Player("uget", "uget-gtk", "--quiet --folder=/tmp --filename=%f %u", false) );
 	*/
