diff -up src/players.cpp.orig src/players.cpp
--- src/players.cpp.orig	2013-01-11 22:03:31.000000000 -0200
+++ src/players.cpp	2013-08-05 20:01:18.463841308 -0300
@@ -49,7 +49,9 @@ QString Player::executable(bool * found)
 
 Players::Players() {
 	list.push_back( Player("SMPlayer", "smplayer", true) );
-	list.push_back( Player("MPlayer", "mplayer", false, "-title ") );
+	list.push_back( Player("MPlayer", "mplayer", false, "-title ", "-quiet") );
+	list.push_back( Player("MPlayer2", "mplayer2", false, "-title ", "-quiet") );
+	list.push_back( Player("mpv", "mpv", false, "--title ", "--quiet") );
 	list.push_back( Player("VLC", "vlc", false, "--meta-title=") );
 	list.push_back( Player("Dragon Player", "dragon", false) );
 	list.push_back( Player("Totem", "totem", false) );
diff -up src/players.h.orig src/players.h
--- src/players.h.orig	2013-08-05 19:55:40.663704417 -0300
+++ src/players.h	2013-08-05 19:57:15.763471987 -0300
@@ -26,27 +26,30 @@
 class Player
 {
 public:
-	Player(QString name, QString binary, bool directPlay, QString title_option = QString::null) { 
+	Player(QString name, QString binary, bool directPlay, QString title_option = QString::null, QString quiet_option = QString::null) { 
 		player_name = name;
 		player_bin = binary;
 		direct_play = directPlay;
 		title_opt = title_option;
+		quiet_opt = quiet_option;
 	}
 
 	void setName(QString name) { player_name = name; }
 	void setBinary(QString binary) { player_bin = binary; }
 	void setDirectPlay(bool b) { direct_play = b; }
 	void setTitleOption(QString s) { title_opt = s; }
+	void setQuietOption(QString q) { quiet_opt = q; }
 
 	QString name() { return player_name; }
 	QString binary() { return player_bin; }
 	bool directPlay() { return direct_play; }
 	QString titleOption() { return title_opt; }
+	QString quietOption() { return quiet_opt; }
 
 	QString executable(bool * found = 0);
 
 protected:
-	QString player_name, player_bin, title_opt;
+	QString player_name, player_bin, title_opt, quiet_opt;
 	bool direct_play;
 };
 
diff -up src/ytdialog.cpp.orig src/ytdialog.cpp
--- src/ytdialog.cpp.orig	2013-08-05 19:59:49.834695198 -0300
+++ src/ytdialog.cpp	2013-08-05 19:59:57.798001156 -0300
@@ -742,6 +742,10 @@ void YTDialog::playYTUrl(const QMap<int,
             args << title_opt + title;
         }
     }
+    QString quiet_opt = players.currentPlayer().quietOption();
+    if (!quiet_opt.isEmpty()) {
+        args << quiet_opt;
+    }
     QProcess::startDetached(exec, args);
 }
 
