diff -up gnupg-2.0.16/kbx/keybox-blob.c.gpgsm_realloc gnupg-2.0.16/kbx/keybox-blob.c
--- gnupg-2.0.16/kbx/keybox-blob.c.gpgsm_realloc	2009-09-21 11:53:44.000000000 -0500
+++ gnupg-2.0.16/kbx/keybox-blob.c	2010-07-23 12:29:54.222718366 -0500
@@ -898,6 +898,7 @@ _keybox_create_x509_blob (KEYBOXBLOB *r_
               rc = gpg_error_from_syserror ();
               goto leave;
             }
+          names = tmp;
         }
       names[blob->nuids++] = p;
       if (!i && (p=x509_email_kludge (p)))
