diff -up gnupg-2.0.20/g10/gpg.c.secmem gnupg-2.0.20/g10/gpg.c
--- gnupg-2.0.20/g10/gpg.c.secmem	2013-05-10 09:55:46.000000000 -0300
+++ gnupg-2.0.20/g10/gpg.c	2013-05-10 19:31:54.996003820 -0300
@@ -794,7 +794,7 @@ make_libversion (const char *libname, co
 
   if (maybe_setuid)
     {
-      gcry_control (GCRYCTL_INIT_SECMEM, 0, 0);  /* Drop setuid. */
+      gcry_control (GCRYCTL_INIT_SECMEM, 4096, 0);  /* Drop setuid. */
       maybe_setuid = 0;
     }
   s = getfnc (NULL);
@@ -898,7 +898,7 @@ build_list (const char *text, char lette
   char *string;
 
   if (maybe_setuid)
-    gcry_control (GCRYCTL_INIT_SECMEM, 0, 0);  /* Drop setuid. */
+    gcry_control (GCRYCTL_INIT_SECMEM, 4096, 0);  /* Drop setuid. */
 
   indent = utf8_charcount (text);
   len = 0;
diff -up gnupg-2.0.20/sm/gpgsm.c.secmem gnupg-2.0.20/sm/gpgsm.c
--- gnupg-2.0.20/sm/gpgsm.c.secmem	2013-05-10 09:55:49.000000000 -0300
+++ gnupg-2.0.20/sm/gpgsm.c	2013-05-10 19:30:44.452962639 -0300
@@ -493,7 +493,7 @@ make_libversion (const char *libname, co
 
   if (maybe_setuid)
     {
-      gcry_control (GCRYCTL_INIT_SECMEM, 0, 0);  /* Drop setuid. */
+      gcry_control (GCRYCTL_INIT_SECMEM, 4096, 0);  /* Drop setuid. */
       maybe_setuid = 0;
     }
   s = getfnc (NULL);
