--- xfdesktop-4.12.0/common/xfdesktop-common.h.orig	2015-03-01 01:30:12.438433411 -0300
+++ xfdesktop-4.12.0/common/xfdesktop-common.h	2015-03-01 01:31:28.009405452 -0300
@@ -35,7 +35,8 @@
 #include <stdarg.h>
 
 #define XFDESKTOP_CHANNEL        "xfce4-desktop"
-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-teal.jpg"
+/* Use slack-artwork default background image */
+#define DEFAULT_BACKDROP         DATADIR "/walpapers/slk-default.png"
 #define DEFAULT_ICON_FONT_SIZE   12
 #define DEFAULT_ICON_SIZE        48
 #define ITHEME_FLAGS             (GTK_ICON_LOOKUP_USE_BUILTIN \
