diff -Nur xfdesktop-4.9.2.orig/settings/xfce-backdrop-settings.desktop.in xfdesktop-4.9.2/settings/xfce-backdrop-settings.desktop.in
--- xfdesktop-4.9.2.orig/settings/xfce-backdrop-settings.desktop.in	2012-04-01 03:10:12.000000000 -0600
+++ xfdesktop-4.9.2/settings/xfce-backdrop-settings.desktop.in	2012-04-03 15:10:16.561323417 -0600
@@ -6,7 +6,7 @@
 Icon=preferences-desktop-wallpaper
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 OnlyShowIn=XFCE;
 StartupNotify=true
 X-XfcePluggable=true
