diff -Nur xfdesktop-4.9.3.orig/common/xfdesktop-common.h xfdesktop-4.9.3/common/xfdesktop-common.h
--- xfdesktop-4.9.3.orig/common/xfdesktop-common.h	2012-04-14 02:00:51.000000000 -0600
+++ xfdesktop-4.9.3/common/xfdesktop-common.h	2012-04-14 14:12:58.292166992 -0600
@@ -32,7 +32,8 @@
 #include <X11/Xlib.h>
 
 #define XFDESKTOP_CHANNEL        "xfce4-desktop"
-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-blue.jpg"
+/* Use slack-artwork default background image */
+#define DEFAULT_BACKDROP         DATADIR "/walpapers/slk-default.png"
 #define DEFAULT_BACKDROP_LIST    "xfce4/desktop/backdrop.list"
 #define DEFAULT_ICON_FONT_SIZE   12
 #define DEFAULT_ICON_SIZE        32
