diff -up wget-1.13/NEWS.path wget-1.13/NEWS
--- wget-1.13/NEWS.path	2011-08-06 08:04:16.000000000 -0300
+++ wget-1.13/NEWS	2011-08-09 21:59:01.164484552 -0300
@@ -625,7 +625,7 @@ distributed with Wget.
 
 ** Compiles on pre-ANSI compilers.
 
-** Global wgetrc now goes to /usr/local/etc (i.e. $sysconfdir).
+** Global wgetrc now goes to /etc (i.e. $sysconfdir).
 
 ** Lots of bugfixes.
 
@@ -688,7 +688,7 @@ Emacs, standalone info, or converted to
 ** Fixed a long-standing bug, so that Wget now works over SLIP
 connections.
 
-** You can have a system-wide wgetrc (/usr/local/lib/wgetrc by
+** You can have a system-wide wgetrc (/etc/wgetrc by
 default). Settings in $HOME/.wgetrc override the global ones, of
 course :-)
 
diff -up wget-1.13/README.path wget-1.13/README
--- wget-1.13/README.path	2011-01-01 10:12:33.000000000 -0200
+++ wget-1.13/README	2011-08-09 21:59:01.164484552 -0300
@@ -33,7 +33,7 @@ for socks.
 
 Most of the features are configurable, either through command-line
 options, or via initialization file .wgetrc.  Wget allows you to
-install a global startup file (/usr/local/etc/wgetrc by default) for
+install a global startup file (/etc/wgetrc by default) for
 site settings.
 
 Wget works under almost all Unix variants in use today and, unlike
diff -up wget-1.13/doc/sample.wgetrc.munged_for_texi_inclusion.path wget-1.13/doc/sample.wgetrc.munged_for_texi_inclusion
--- wget-1.13/doc/sample.wgetrc.munged_for_texi_inclusion.path	2011-08-09 10:52:17.000000000 -0300
+++ wget-1.13/doc/sample.wgetrc.munged_for_texi_inclusion	2011-08-09 21:59:01.159485615 -0300
@@ -7,7 +7,7 @@
 ## not contain a comprehensive list of commands -- look at the manual
 ## to find out what you can put into this file.
 ##
-## Wget initialization file can reside in /usr/local/etc/wgetrc
+## Wget initialization file can reside in /etc/wgetrc
 ## (global, for all users) or $HOME/.wgetrc (for a single user).
 ##
 ## To use the settings in this file, you will have to uncomment them,
@@ -16,7 +16,7 @@
 
 
 ##
-## Global settings (useful for setting up in /usr/local/etc/wgetrc).
+## Global settings (useful for setting up in /etc/wgetrc).
 ## Think well before you change them, since they may reduce wget's
 ## functionality, and make it behave contrary to the documentation:
 ##
diff -up wget-1.13/doc/sample.wgetrc.path wget-1.13/doc/sample.wgetrc
--- wget-1.13/doc/sample.wgetrc.path	2011-01-01 10:12:33.000000000 -0200
+++ wget-1.13/doc/sample.wgetrc	2011-08-09 21:59:01.158485827 -0300
@@ -7,7 +7,7 @@
 ## not contain a comprehensive list of commands -- look at the manual
 ## to find out what you can put into this file.
 ##
-## Wget initialization file can reside in /usr/local/etc/wgetrc
+## Wget initialization file can reside in /etc/wgetrc
 ## (global, for all users) or $HOME/.wgetrc (for a single user).
 ##
 ## To use the settings in this file, you will have to uncomment them,
@@ -16,7 +16,7 @@
 
 
 ##
-## Global settings (useful for setting up in /usr/local/etc/wgetrc).
+## Global settings (useful for setting up in /etc/wgetrc).
 ## Think well before you change them, since they may reduce wget's
 ## functionality, and make it behave contrary to the documentation:
 ##
diff -up wget-1.13/doc/wget.info.path wget-1.13/doc/wget.info
--- wget-1.13/doc/wget.info.path	2011-08-09 10:52:17.000000000 -0300
+++ wget-1.13/doc/wget.info	2011-08-09 22:01:38.518013826 -0300
@@ -113,7 +113,7 @@ retrieval through HTTP proxies.
    * Most of the features are fully configurable, either through
      command line options, or via the initialization file `.wgetrc'
      (*note Startup File::).  Wget allows you to define "global"
-     startup files (`/usr/local/etc/wgetrc' by default) for site
+     startup files (`/etc/wgetrc' by default) for site
      settings. You can also specify the location of a startup file with
      the -config option.
 
@@ -2373,8 +2373,8 @@ File: wget.info,  Node: Wgetrc Location,
 ===================
 
 When initializing, Wget will look for a "global" startup file,
-`/usr/local/etc/wgetrc' by default (or some prefix other than
-`/usr/local', if Wget was not installed there) and read commands from
+`/etc/wgetrc' by default (or some prefix other than
+`/etc', if Wget was not installed there) and read commands from
 there, if it exists.
 
    Then it will look for the user's file.  If the environmental variable
@@ -2385,7 +2385,7 @@ further attempts will be made.
 
    The fact that user's settings are loaded after the system-wide ones
 means that in case of collision user's wgetrc _overrides_ the
-system-wide wgetrc (in `/usr/local/etc/wgetrc' by default).  Fascist
+system-wide wgetrc (in `/etc/wgetrc' by default).  Fascist
 admins, away!
 
 
@@ -2910,7 +2910,7 @@ its line.
      ## not contain a comprehensive list of commands -- look at the manual
      ## to find out what you can put into this file.
      ##
-     ## Wget initialization file can reside in /usr/local/etc/wgetrc
+     ## Wget initialization file can reside in /etc/wgetrc
      ## (global, for all users) or $HOME/.wgetrc (for a single user).
      ##
      ## To use the settings in this file, you will have to uncomment them,
@@ -2919,7 +2919,7 @@ its line.
 
 
      ##
-     ## Global settings (useful for setting up in /usr/local/etc/wgetrc).
+     ## Global settings (useful for setting up in /etc/wgetrc).
      ## Think well before you change them, since they may reduce wget's
      ## functionality, and make it behave contrary to the documentation:
      ##
diff -up wget-1.13/doc/wget.texi.path wget-1.13/doc/wget.texi
--- wget-1.13/doc/wget.texi.path	2011-08-06 07:22:58.000000000 -0300
+++ wget-1.13/doc/wget.texi	2011-08-09 22:01:05.519033579 -0300
@@ -190,14 +190,14 @@ gauge can be customized to your preferen
 Most of the features are fully configurable, either through command line
 options, or via the initialization file @file{.wgetrc} (@pxref{Startup
 File}).  Wget allows you to define @dfn{global} startup files
-(@file{/usr/local/etc/wgetrc} by default) for site settings. You can also
+(@file{/etc/wgetrc} by default) for site settings. You can also
 specify the location of a startup file with the --config option.
  
 
 @ignore
 @c man begin FILES
 @table @samp
-@item /usr/local/etc/wgetrc
+@item /etc/wgetrc
 Default location of the @dfn{global} startup file.
 
 @item .wgetrc
@@ -2696,8 +2696,8 @@ commands.
 @cindex location of wgetrc
 
 When initializing, Wget will look for a @dfn{global} startup file,
-@file{/usr/local/etc/wgetrc} by default (or some prefix other than
-@file{/usr/local}, if Wget was not installed there) and read commands
+@file{/etc/wgetrc} by default (or some prefix other than
+@file{/etc}, if Wget was not installed there) and read commands
 from there, if it exists.
 
 Then it will look for the user's file.  If the environmental variable
@@ -2708,7 +2708,7 @@ If @code{WGETRC} is not set, Wget will t
 
 The fact that user's settings are loaded after the system-wide ones
 means that in case of collision user's wgetrc @emph{overrides} the
-system-wide wgetrc (in @file{/usr/local/etc/wgetrc} by default).
+system-wide wgetrc (in @file{/etc/wgetrc} by default).
 Fascist admins, away!
 
 @node Wgetrc Syntax, Wgetrc Commands, Wgetrc Location, Startup File
