diff -Naur avidemux_2.5.4.orig/plugins/ADM_videoFilters/Ass/ADM_vidASS.h avidemux_2.5.4/plugins/ADM_videoFilters/Ass/ADM_vidASS.h
--- avidemux_2.5.4.orig/plugins/ADM_videoFilters/Ass/ADM_vidASS.h	2010-01-09 05:03:17.000000000 -0600
+++ avidemux_2.5.4/plugins/ADM_videoFilters/Ass/ADM_vidASS.h	2011-05-25 14:27:23.828297237 -0500
@@ -14,7 +14,7 @@
 
 extern "C"
 {
-#include "ADM_libAss/ass.h"
+#include <ass/ass.h>
 }
 #include "ADM_vidAss_Params.h" 
 class ADMVideoSubASS : public AVDMGenericVideoStream 
diff -Naur avidemux_2.5.4.orig/plugins/ADM_videoFilters/Ass/CMakeLists.txt avidemux_2.5.4/plugins/ADM_videoFilters/Ass/CMakeLists.txt
--- avidemux_2.5.4.orig/plugins/ADM_videoFilters/Ass/CMakeLists.txt	2010-01-09 05:03:17.000000000 -0600
+++ avidemux_2.5.4/plugins/ADM_videoFilters/Ass/CMakeLists.txt	2011-05-25 15:45:05.219501216 -0500
@@ -2,13 +2,14 @@
 checkFreeType()
 
 IF (USE_FREETYPE)
-	ADD_SUBDIRECTORY(ADM_libAss)
+	#ADD_SUBDIRECTORY(ADM_libAss)
 
 	INCLUDE(vf_plugin)
 	SET(ADM_vf_ssa_SRCS ADM_vidASS.cpp)
 
 	ADD_LIBRARY(ADM_vf_ssa SHARED ${ADM_vf_ssa_SRCS})
-	TARGET_LINK_LIBRARIES(ADM_vf_ssa ADM_libass)
+	FIND_LIBRARY(LIBASS ass)
+	TARGET_LINK_LIBRARIES(ADM_vf_ssa ${LIBASS})
 	ADD_TARGET_CFLAGS(ADM_vf_ssa "-DCONFIG_FREETYPE=1")
 	ADD_TARGET_LDFLAGS(ADM_vf_ssa "${FREETYPE2_LDFLAGS}")
 
