--- avidemux_2.5.3/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/CMakeLists.txt.orig	2010-05-19 18:04:45.640566565 -0400
+++ avidemux_2.5.3/plugins/ADM_videoEncoder/ADM_vidEnc_mpeg2enc/mpeg2enc/CMakeLists.txt	2010-05-19 18:05:01.255568336 -0400
@@ -42,7 +42,7 @@
 	ENDIF (NOT APPLE)
 ENDIF (ADM_CPU_X86)
 
-add_library(${ADM_LIB} ${${ADM_LIB}_SRCS})
+add_library(${ADM_LIB} STATIC ${${ADM_LIB}_SRCS})
 
 if (UNIX)
 	add_target_cflags(mpeg2enc -fPIC)
