diff --git a/avidemux/gtk/ADM_userInterfaces/ui_support.cpp b/avidemux/gtk/ADM_userInterfaces/ui_support.cpp
index f6e6ac9..076c6d6 100644
--- a/avidemux/gtk/ADM_userInterfaces/ui_support.cpp
+++ b/avidemux/gtk/ADM_userInterfaces/ui_support.cpp
@@ -13,6 +13,7 @@ extern "C" int getMainNSWindow(void);
 #include "ADM_inttype.h"
 #include "ADM_files.h"
 #include "DIA_uiTypes.h"
+#include "ADM_default.h"
 
 extern GtkWidget *guiRootWindow;
 
@@ -68,4 +69,4 @@ void getMainWindowHandles(long int *handle,long int *nativeHandle)
 }
 
 void GUI_OpenApplicationLog() { }
-void GUI_OpenApplicationDataFolder() { }
\ No newline at end of file
+void GUI_OpenApplicationDataFolder() { }
