diff -Naur avidemux_2.5.4.orig/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp avidemux_2.5.4/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp
--- avidemux_2.5.4.orig/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp	2010-03-15 13:30:11.000000000 -0500
+++ avidemux_2.5.4/plugins/ADM_audioDecoders/ADM_ad_ac3/ADM_ad_a52.cpp	2011-05-26 13:26:08.620994308 -0500
@@ -18,8 +18,8 @@
 #include "ADM_ad_plugin.h"
 #include "ADM_audioCodecEnum.h"
 extern "C" {
-#include "ADM_liba52/a52.h"
-#include "ADM_liba52/mm_accel.h"
+#include <a52dec/a52.h> 
+#include <a52dec/mm_accel.h>
 };
 
 #define AC3_HANDLE ((a52_state_t *)ac3_handle)
diff -Naur avidemux_2.5.4.orig/plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt avidemux_2.5.4/plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt
--- avidemux_2.5.4.orig/plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt	2008-07-16 12:40:07.000000000 -0500
+++ avidemux_2.5.4/plugins/ADM_audioDecoders/ADM_ad_ac3/CMakeLists.txt	2011-05-26 13:24:40.789865387 -0500
@@ -1,11 +1,12 @@
 INCLUDE(ad_plugin)
 
-ADD_SUBDIRECTORY(ADM_liba52)
+#ADD_SUBDIRECTORY(ADM_liba52)
 
 SET(ADM_ad_a52_SRCS ADM_ad_a52.cpp)
 
 ADD_LIBRARY(ADM_ad_a52 SHARED ${ADM_ad_a52_SRCS})
-TARGET_LINK_LIBRARIES(ADM_ad_a52 ADM_liba52)
+FIND_LIBRARY(LIBA52 a52)
+TARGET_LINK_LIBRARIES(ADM_ad_a52 ${LIBA52})
 
 INIT_AUDIO_PLUGIN(ADM_ad_a52)
 INSTALL_AUDIODECODER(ADM_ad_a52)
