diff -up plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp.orig plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp
--- plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp.orig	2010-09-29 17:20:48.000000000 -0300
+++ plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp	2010-12-16 20:15:06.180386176 -0200
@@ -94,6 +94,13 @@ x264ConfigDialog::x264ConfigDialog(vidEn
 	connect(ui.dct8x8CheckBox, SIGNAL(toggled(bool)), this, SLOT(dct8x8CheckBox_toggled(bool)));
 	connect(ui.p8x8CheckBox, SIGNAL(toggled(bool)), this, SLOT(p8x8CheckBox_toggled(bool)));
 
+#if X264_BUILD < 110
+	ui.weightedPPredictComboBox->clear();
+	ui.weightedPPredictComboBox->addItem(tr("Disabled"));
+	ui.weightedPPredictComboBox->addItem(tr("Blind Offset"));
+	ui.weightedPPredictComboBox->addItem(tr("Smart Analysis"));
+#endif
+
 	// Frame tab
 	connect(ui.loopFilterCheckBox, SIGNAL(toggled(bool)), this, SLOT(loopFilterCheckBox_toggled(bool)));
 
diff -up plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.ui.orig plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.ui
--- plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.ui.orig	2010-09-29 17:04:31.000000000 -0300
+++ plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.ui	2010-12-16 20:14:36.313708033 -0200
@@ -1131,12 +1131,12 @@
                 </item>
                 <item>
                  <property name="text">
-                  <string>Blind Offset</string>
+                  <string>Weighted References</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
-                  <string>Smart Analysis</string>
+                  <string>Weighted References + Duplicates</string>
                  </property>
                 </item>
                </widget>
diff -up plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp.orig plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp
--- plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp.orig	2010-10-10 10:45:42.000000000 -0300
+++ plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2010-12-16 20:13:24.410928058 -0200
@@ -1386,9 +1386,15 @@ void x264Options::addOptionsToXml(xmlNod
 		case X264_WEIGHTP_NONE:
 			strcpy((char*)xmlBuffer, "none");
 			break;
+#if X264_BUILD < 110
 		case X264_WEIGHTP_BLIND:
 			strcpy((char*)xmlBuffer, "blind");
 			break;
+#else
+			case X264_WEIGHTP_SIMPLE:
+			strcpy((char*)xmlBuffer, "simple");
+			break;
+#endif
 		case X264_WEIGHTP_SMART:
 			strcpy((char*)xmlBuffer, "smart");
 			break;
@@ -1895,8 +1901,13 @@ void x264Options::parseAnalyseOptions(xm
 			{
 				int weightedPredPFrames = X264_WEIGHTP_NONE;
 
+#if X264_BUILD < 110
 				if (strcmp(content, "blind") == 0)
 					weightedPredPFrames = X264_WEIGHTP_BLIND;
+#else
+				if (strcmp(content, "simple") == 0)
+					weightedPredPFrames = X264_WEIGHTP_SIMPLE;
+#endif
 				else if (strcmp(content, "smart") == 0)
 					weightedPredPFrames = X264_WEIGHTP_SMART;
 
diff -up plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Param.xsd.orig plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Param.xsd
--- plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Param.xsd.orig	2010-09-29 16:48:29.000000000 -0300
+++ plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Param.xsd	2010-12-16 20:10:07.374728639 -0200
@@ -335,7 +335,9 @@
                       <xs:simpleType>
                         <xs:restriction base="xs:string">
                           <xs:enumeration value="none"/>
+                          <!-- blind deprecated core 110 -->
                           <xs:enumeration value="blind"/>
+                          <xs:enumeration value="simple"/>
                           <xs:enumeration value="smart"/>
                         </xs:restriction>
                       </xs:simpleType>
