diff -Nuar avidemux_2.5.4.orig/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/qt4/xvidConfigDialog.cpp avidemux_2.5.4/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/qt4/xvidConfigDialog.cpp
--- avidemux_2.5.4.orig/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/qt4/xvidConfigDialog.cpp	2010-06-07 01:07:37.000000000 +0200
+++ avidemux_2.5.4/plugins/ADM_videoEncoder/ADM_vidEnc_xvid/qt4/xvidConfigDialog.cpp	2011-04-18 12:12:13.640562284 +0200
@@ -233,7 +233,8 @@
 		delete [] xml;
 
 		fillConfigurationComboBox();
-		selectConfiguration(&QFileInfo(configFileName).completeBaseName(), PLUGIN_CONFIG_USER);
+		QString temporary = QFileInfo(configFileName).completeBaseName();
+		selectConfiguration(&temporary, PLUGIN_CONFIG_USER);
 	}
 
 	delete [] configDirectory;
@@ -373,7 +374,8 @@
 	disableGenericSlots = true;
 	options->getPresetConfiguration(&configurationName, &configurationType);		
 
-	bool foundConfig = selectConfiguration(&QString(configurationName), configurationType);
+	QString temporary2 = QString(configurationName);
+	bool foundConfig = selectConfiguration(&temporary2, configurationType);
 
 	if (!foundConfig)
 		printf("Configuration %s (type %d) could not be found.  Using snapshot.\n", configurationName, configurationType);
