diff -Naur avidemux_2.5.4.orig/plugins/ADM_audioEncoders/twolame/CMakeLists.txt avidemux_2.5.4/plugins/ADM_audioEncoders/twolame/CMakeLists.txt
--- avidemux_2.5.4.orig/plugins/ADM_audioEncoders/twolame/CMakeLists.txt	2010-05-25 15:48:57.000000000 -0500
+++ avidemux_2.5.4/plugins/ADM_audioEncoders/twolame/CMakeLists.txt	2011-05-26 12:39:31.025294147 -0500
@@ -1,12 +1,13 @@
 	INCLUDE(ae_plugin)
-        subdirs(PREORDER ADM_libtwolame)
+        #subdirs(PREORDER ADM_libtwolame)
 
 	SET(ADM_ae_twolame_SRCS audioencoder_twolame.cpp)
 
 	ADD_LIBRARY(ADM_ae_twolame SHARED ${ADM_ae_twolame_SRCS})
 
-	TARGET_LINK_LIBRARIES(ADM_ae_twolame ADM_libtwolame ADM_coreAudio ADM_coreUI)
-        include_directories(ADM_libtwolame)
+	FIND_LIBRARY(LIBTWOLAME twolame)
+	TARGET_LINK_LIBRARIES(ADM_ae_twolame ${LIBTWOLAME} ADM_coreAudio ADM_coreUI)
+        #include_directories(ADM_libtwolame)
 
 
 	INIT_AUDIO_ENCODER(ADM_ae_twolame)
