--- avidemux_2.5.6/avidemux/ADM_coreImage/src/CMakeLists.txt.orig	2011-12-28 07:44:42.000000000 -0200
+++ avidemux_2.5.6/avidemux/ADM_coreImage/src/CMakeLists.txt	2011-12-29 16:24:50.270472593 -0200
@@ -20,6 +20,7 @@ SET(ADM_coreImage_SRCS
 )
 
 ADD_LIBRARY(ADM_coreImage SHARED ${ADM_coreImage_SRCS})
+ADD_DEPENDENCIES(ADM_coreImage ffmpeg)
 ADD_SOURCE_CFLAGS(DIA_flyDialog.cpp " -DADM_UI_TYPE_BUILD=99")
 
 IF (APPLE AND NOT ADM_CPU_X86_64)
