diff -Nuar avidemux_2.5.4.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp avidemux_2.5.4/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp
--- avidemux_2.5.4.orig/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp	2010-09-29 22:20:48.000000000 +0200
+++ avidemux_2.5.4/plugins/ADM_videoEncoder/ADM_vidEnc_x264/qt4/x264ConfigDialog.cpp	2011-04-18 12:26:34.482304155 +0200
@@ -343,7 +343,8 @@
 		delete [] xml;
 
 		fillConfigurationComboBox();
-		selectConfiguration(&QFileInfo(configFileName).completeBaseName(), PLUGIN_CONFIG_USER);
+		QString temporary = QFileInfo(configFileName).completeBaseName();
+		selectConfiguration(&temporary, PLUGIN_CONFIG_USER);
 	}
 
 	delete [] configDirectory;
@@ -580,7 +581,8 @@
 	disableGenericSlots = true;
 	options->getPresetConfiguration(&configurationName, &configurationType);
 
-	bool foundConfig = selectConfiguration(&QString(configurationName), configurationType);
+	QString temporary2 = QString(configurationName);
+	bool foundConfig = selectConfiguration(&temporary2, configurationType);
 
 	if (!foundConfig)
 		printf("Configuration %s (type %d) could not be found.  Using snapshot.\n", configurationName, configurationType);
