diff -Naur avidemux_2.5.6.orig/cmake/admFFmpegBuild.cmake avidemux_2.5.6/cmake/admFFmpegBuild.cmake
--- avidemux_2.5.6.orig/cmake/admFFmpegBuild.cmake	2012-01-27 15:05:15.029793577 -0600
+++ avidemux_2.5.6/cmake/admFFmpegBuild.cmake	2012-01-27 15:06:00.663291081 -0600
@@ -14,7 +14,7 @@
 set(FFMPEG_DECODERS  adpcm_ima_amv  amv  bmp  cinepak  cyuv  dca  dvbsub  dvvideo  ffv1  ffvhuff  flac  flv  fraps  h263  h264  huffyuv
 					 mjpeg  mjpegb  mpeg2video  mpeg4  msmpeg4v2  msmpeg4v3  msvideo1  nellymoser  png  qdm2  rawvideo  snow  svq3
 					 theora  tscc  vc1  vp3  vp6  vp6a  vp6f  wmav2  wmv1  wmv2  wmv3)
-set(FFMPEG_ENCODERS  ac3_fixed  dvvideo  ffv1  ffvhuff  flv  h263  huffyuv  mjpeg  mp2  mpeg1video  mpeg2video  mpeg4  snow)
+set(FFMPEG_ENCODERS  aac ac3_fixed  dvvideo  ffv1  ffvhuff  flv  h263  huffyuv  mjpeg  mp2  mpeg1video  mpeg2video  mpeg4  snow)
 set(FFMPEG_MUXERS  flv  matroska  mpeg1vcd  mpeg2dvd  mpeg2svcd  mpegts  mov  mp4  psp)
 set(FFMPEG_PARSERS  ac3  h263  h264  mpeg4video)
 set(FFMPEG_PROTOCOLS  file)
diff -Naur avidemux_2.5.6.orig/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp avidemux_2.5.6/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp
--- avidemux_2.5.6.orig/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp	2011-12-28 08:51:49.000000000 -0600
+++ avidemux_2.5.6/plugins/ADM_audioEncoders/lavcodec/audioencoder_lavcodec.cpp	2012-01-27 15:05:39.672982290 -0600
@@ -25,13 +25,16 @@
 #include "ADM_lavcodec.h"
 
 #define Join(x,y) x##_##y
-#if defined(ADM_LAV_MP2) && !defined(ADM_LAV_AC3)
+#if defined(ADM_LAV_MP2) && !defined(ADM_LAV_AC3) && !defined(ADM_LAV_AAC)
   #define makeName(x) Join(x,MP2)
   #define AUDMEncoder_Lavcodec AUDMEncoder_Lavcodec_MP2
 #else
- #if !defined(ADM_LAV_MP2) && defined(ADM_LAV_AC3)
+ #if !defined(ADM_LAV_MP2) && defined(ADM_LAV_AC3) && !defined(ADM_LAV_AAC)
   #define makeName(x) Join(x,AC3)
   #define AUDMEncoder_Lavcodec AUDMEncoder_Lavcodec_AC3
+ #elif !defined(ADM_LAV_MP2) && !defined(ADM_LAV_AC3) && defined(ADM_LAV_AAC)
+  #define makeName(x) Join(x,AAC)
+  #define AUDMEncoder_Lavcodec AUDMEncoder_Lavcodec_AAC
  #else
    #error
  #endif
@@ -60,13 +63,19 @@
   2,                    // Max channels
   1,0,0,                // Version
 #else
-  
-
+#ifdef ADM_LAV_AC3
  "LavAC3",            
   "AC3 (lav)",      
   "AC3 LavEncoder encoder plugin Mean 2008",             
   6,                    // Max channels
   1,0,0,                // Version
+#else
+ "LavAAC",
+  "AAC (lav)",
+  "AAC LavEncoder encoder plugin Mean 2008",
+  6,                    // Max channels
+  1,0,0,                // Version
+#endif
 #endif
   makeName(WAV),
 
@@ -132,7 +141,9 @@
 #endif
   _wavheader->byterate=(lavConfig.bitrate*1000)>>3;         
       
-#ifdef ADM_LAV_MP2 
+#if defined(ADM_LAV_AAC)
+    _chunk = 1024*_wavheader->channels;
+#elif defined(ADM_LAV_MP2)
     _chunk = 1152*_wavheader->channels;
 #else
     _chunk = 1536*_wavheader->channels; // AC3
diff -Naur avidemux_2.5.6.orig/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt avidemux_2.5.6/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt
--- avidemux_2.5.6.orig/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt	2010-05-25 15:48:57.000000000 -0500
+++ avidemux_2.5.6/plugins/ADM_audioEncoders/lavcodec/CMakeLists.txt	2012-01-27 15:05:39.673982248 -0600
@@ -27,3 +27,12 @@
 
 	INIT_AUDIO_ENCODER(ADM_ae_lav_ac3)
 	INSTALL_AUDIOENCODER(ADM_ae_lav_ac3)
+
+	SET(ADM_ae_lav_aac_SRCS audioencoder_lavcodec.cpp)
+
+	ADD_LIBRARY(ADM_ae_lav_aac SHARED ${ADM_ae_lav_aac_SRCS})
+	ADD_TARGET_CFLAGS(ADM_ae_lav_aac "-DADM_LAV_AAC")
+	TARGET_LINK_LIBRARIES(ADM_ae_lav_aac ADM_core ADM_coreAudio ADM_coreUI ADM_libavcodec ADM_libavutil)
+
+	INIT_AUDIO_ENCODER(ADM_ae_lav_aac)
+	INSTALL_AUDIOENCODER(ADM_ae_lav_aac)
