--- patches/updates/7.3.202	2011-06-16 21:41:08.000000000 -0300
+++ vim-fdist/7.3.202	2011-06-16 21:46:15.000000000 -0300
@@ -153,16 +153,6 @@ Files:      runtime/doc/indent.txt, src/
   	      statement after it in the same line.
   
 ***************
-*** 256,261 ****
---- 273,279 ----
-  				    break;	      break;
-  				}		  }
-  <
-+ 							*cino-b*
-  	bN    If N != 0 Vim will align a final "break" with the case label,
-  	      so that case..break looks like a sort of block.  (default: 0).
-  	      When using 1, consider adding "0=break" to 'cinkeys'.
-***************
 *** 272,277 ****
 --- 290,296 ----
   			  break;	  break;
@@ -227,16 +217,6 @@ Files:      runtime/doc/indent.txt, src/
   	      initializations, if they start in a new line (otherwise they
   	      are aligned at the right side of the ':').
 ***************
-*** 320,325 ****
---- 357,363 ----
-  		      BaseClass(3)	    BaseClass(3)
-  		  {}			    {}
-  <
-+ 							*cino-+*
-  	+N    Indent a continuation line (a line that spills onto the next)
-                inside a function N additional characters.  (default
-                'shiftwidth').
-***************
 *** 330,335 ****
 --- 368,374 ----
   		  a = b + 9 *		    a = b + 9 *
@@ -368,28 +348,6 @@ Files:      runtime/doc/indent.txt, src/
   	      them with labels.  The value 'N' is currently unused but must be 
   	      non-zero (e.g. 'J1'). >
 ***************
-*** 466,481 ****
---- 514,532 ----
-  		    }
-  		}
-  <
-+ 								*cino-)*
-  	)N    Vim searches for unclosed parentheses at most N lines away.
-  	      This limits the time needed to search for parentheses.  (default
-  	      20 lines).
-  
-+ 								*cino-star*
-  	*N    Vim searches for unclosed comments at most N lines away.  This
-  	      limits the time needed to search for the start of a comment.
-  	      If your /* */ comments stop indenting afer N lines this is the
-  	      value you will want to change.
-  	      (default 70 lines).
-  
-+ 								*cino-#*
-  	#N    When N is non-zero recognize shell/Perl comments, starting with
-  	      '#'.  Default N is zero: don't recognizes '#' comments.  Note
-  	      that lines starting with # will still be seen as preprocessor
-***************
 *** 483,489 ****
   
   
