diff -up vim74/runtime/filetype.vim.syntax1 vim74/runtime/filetype.vim
--- vim74/runtime/filetype.vim.syntax1	2013-08-10 23:07:00.622996076 -0300
+++ vim74/runtime/filetype.vim	2013-08-10 23:10:28.288784754 -0300
@@ -2475,7 +2475,7 @@ au StdinReadPost * if !did_filetype() |
 
 " More Apache config files
 au BufNewFile,BufRead access.conf*,apache.conf*,apache2.conf*,httpd.conf*,srm.conf*	call s:StarSetf('apache')
-au BufNewFile,BufRead */etc/apache2/*.conf*,*/etc/apache2/conf.*/*,*/etc/apache2/mods-*/*,*/etc/apache2/sites-*/*,*/etc/httpd/conf.d/*.conf*		call s:StarSetf('apache')
+au BufNewFile,BufRead /etc/httpd/*.conf*,/etc/httpd/conf.d/*.conf*,/etc/httpd/extra/*.conf*,/etc/httpd/original/*.conf*,auth_mysql.conf*,auth_pgsql.conf*,ssl.conf*,perl.conf*,php.conf*,python.conf*,squirrelmail.conf*		call s:StarSetf('apache')
 
 " Asterisk config file
 au BufNewFile,BufRead *asterisk/*.conf*		call s:StarSetf('asterisk')
