diff -up vim73/runtime/filetype.vim.816848 vim73/runtime/filetype.vim
--- vim73/runtime/filetype.vim.816848	2012-05-21 11:04:08.184266381 +0200
+++ vim73/runtime/filetype.vim	2012-05-21 11:05:13.511269886 +0200
@@ -1098,7 +1098,7 @@ au BufNewFile,BufRead *.mel			setf mel
 au BufNewFile,BufRead *.hgrc,*hgrc		setf cfg
 
 " Messages
-au BufNewFile,BufRead /var/log/messages,/var/log/messages.*[0-9]  setf messages
+au BufNewFile,BufRead */log/{auth,cron,daemon,debug,kern,lpr,mail,messages,news/news,syslog,user}{,.log,.err,.info,.warn,.crit,.notice}{,.[0-9]*,-[0-9]*} setf messages
 
 " Metafont
 au BufNewFile,BufRead *.mf			setf mf
