--- patches/7.3.859.orig	2013-03-29 21:44:43.586773965 -0300
+++ patches/7.3.859	2013-03-29 21:46:07.313032058 -0300
@@ -68,11 +68,11 @@ Files:	    src/main.c, src/option.c, src
       {K_UP,		IF_EB("\033O*A", ESC_STR "O*A")},
       {K_DOWN,		IF_EB("\033O*B", ESC_STR "O*B")},
 ***************
-*** 1221,1226 ****
+*** 1221,1225 ****
 --- 1227,1233 ----
       {(int)KS_CWP,	"[%dCWP%d]"},
   #  endif
-      {(int)KS_CRV,	"[CRV]"},
++     {(int)KS_CRV,	"[CRV]"},
 +     {(int)KS_U7,	"[U7]"},
       {K_UP,		"[KU]"},
       {K_DOWN,		"[KD]"},
