--- runtime/vimrc_example.vim.orig	2010-05-15 08:03:57.000000000 -0300
+++ runtime/vimrc_example.vim	2013-03-29 21:58:42.345058869 -0300
@@ -79,6 +79,11 @@ if has("autocmd")
     \   exe "normal! g`\"" |
     \ endif
 
+  " don't write swapfile on most commonly used directories for NFS mounts or USB sticks
+  autocmd BufNewFile,BufReadPre /media/*,/mnt/* set directory=~/tmp,/var/tmp,/tmp
+  " start with spec file template
+  autocmd BufNewFile *.spec 0r /usr/share/vim/vimfiles/template.spec
+
   augroup END
 
 else
@@ -87,6 +92,9 @@ else
 
 endif " has("autocmd")
 
+" Make vim work with the 'crontab -e' command
+set backupskip+=/var/spool/cron/*
+
 " Convenient command to see the difference between the current buffer and the
 " file it was loaded from, thus the changes you made.
 " Only define it when not defined already.
