--- configure.ac.orig	2014-02-22 11:17:42.029145702 -0300
+++ configure.ac	2014-02-22 11:18:15.132117955 -0300
@@ -146,7 +146,7 @@ AC_SUBST(LIBATASMART_CFLAGS)
 AC_SUBST(LIBATASMART_LIBS)
 
 PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN,
-                  [libsystemd-login >= 44 libsystemd-daemon],
+                  [libsystemd >= 209],
                   have_libsystemd_login=yes,
                   have_libsystemd_login=no)
 AM_CONDITIONAL(HAVE_LIBSYSTEMD_LOGIN, test x$have_libsystemd_login = xyes)
