diff -up kde-baseapps-4.14.3/konqueror/kfmclient_html.desktop.mimetype kde-baseapps-4.14.3/konqueror/kfmclient_html.desktop
--- kde-baseapps-4.14.3/konqueror/kfmclient_html.desktop.mimetype	2014-11-11 05:22:05.000000000 -0600
+++ kde-baseapps-4.14.3/konqueror/kfmclient_html.desktop	2014-11-17 12:13:33.970897926 -0600
@@ -3,7 +3,7 @@ Type=Application
 Exec=kfmclient openURL %u text/html
 Icon=konqueror
 Terminal=false
-MimeType=text/html;application/xhtml+xml;application/xml;
+MimeType=text/html;application/xhtml+xml;application/xml;x-scheme-handler/http;
 InitialPreference=9
 NoDisplay=true
 Name=Konqueror
