diff -up kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.cpp.foo kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.cpp
--- kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.cpp.foo	2011-05-20 15:33:30.000000000 -0500
+++ kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.cpp	2011-10-08 21:20:48.414176661 -0500
@@ -29,7 +29,8 @@
 ItemEditor::ItemEditor(QGraphicsWidget *parent, const QStyleOptionViewItemV4 &option,
                        const QModelIndex &index)
     : QGraphicsProxyWidget(parent),
-      m_index(index)
+      m_index(index),
+      m_uncommitted(true)
 {
     // Create the editor
     m_editor = new KTextEdit();
@@ -66,14 +67,18 @@ ItemEditor::~ItemEditor()
 
 void ItemEditor::commitData()
 {
-    const_cast<QAbstractItemModel*>(m_index.model())->setData(m_index, m_editor->toPlainText(), Qt::EditRole);
+    if (m_uncommitted) {
+        const_cast<QAbstractItemModel*>(m_index.model())->setData(m_index, m_editor->toPlainText(), Qt::EditRole);
+        m_uncommitted = false;
+    }
 }
 
 bool ItemEditor::eventFilter(QObject *watched, QEvent *event)
 {
     KTextEdit *editor = qobject_cast<KTextEdit*>(watched);
-    if (!editor)
+    if (!editor) {
         return false;
+    }
 
     switch (event->type())
     {
@@ -107,8 +112,10 @@ bool ItemEditor::eventFilter(QObject *wa
 
     case QEvent::FocusOut:
     {
-        commitData();
-        emit closeEditor(this, QAbstractItemDelegate::NoHint);
+        if (m_uncommitted) {
+            commitData();
+            emit closeEditor(this, QAbstractItemDelegate::NoHint);
+        }
         return true;
     }
 
diff -up kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.h.foo kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.h
--- kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.h.foo	2011-05-20 15:33:30.000000000 -0500
+++ kde-baseapps-4.7.2/plasma/applets/folderview/itemeditor.h	2011-10-08 21:20:49.359164876 -0500
@@ -49,6 +49,7 @@ protected:
 private:
     KTextEdit *m_editor;
     QModelIndex m_index;
+    bool m_uncommitted;
 };
 
 #endif
