diff -up kde-baseapps-4.12.2/kdepasswd/kcm/chfnprocess.cpp.bz#609039 kde-baseapps-4.12.2/kdepasswd/kcm/chfnprocess.cpp
--- kde-baseapps-4.12.2/kdepasswd/kcm/chfnprocess.cpp.bz#609039	2014-01-30 15:10:31.000000000 -0600
+++ kde-baseapps-4.12.2/kdepasswd/kcm/chfnprocess.cpp	2014-02-01 21:33:15.140447271 -0600
@@ -63,19 +63,16 @@ int ChfnProcess::ConverseChfn(const char
       break;
     }
 
-    if ( line.contains( "Password: " )/*isPrompt( line, "password" )*/ )
+    if ( line.contains( "Changing finger info" ) )
+    {
+      // do nothing
+    }
+    else if ( line.contains( "Password: " )/*isPrompt( line, "password" )*/ )
     {
       WaitSlave();
       write(fd(), pass, strlen(pass));
       write(fd(), "\n", 1);
     }
-
-    line = readLine(); // Let's see what the outcome was
-
-    if ( line.contains( "Changing finger info" ) )
-    {
-      // do nothing
-    }
     else if ( line.contains( "information changed" ) )
     {
       status=0;
