diff -up kde-baseapps-4.9.2/konqueror/konqbrowser.desktop.orig kde-baseapps-4.9.2/konqueror/konqbrowser.desktop
--- kde-baseapps-4.9.2/konqueror/konqbrowser.desktop.orig	2012-10-16 18:11:28.732633574 +0200
+++ kde-baseapps-4.9.2/konqueror/konqbrowser.desktop	2012-10-16 18:12:00.846668435 +0200
@@ -3,6 +3,7 @@ Type=Application
 Exec=kfmclient openProfile webbrowsing
 Icon=konqueror
 X-DocPath=konqueror/index.html
+MimeType=x-scheme-handler/http;
 
 Name=Konqueror
 Name[af]=Konqueror
