--- a/src/utils/wxCas/src/wxcasframe.cpp.orig	2010-09-10 01:52:00.650000228 +0200
+++ b/src/utils/wxCas/src/wxcasframe.cpp	2010-09-10 01:52:16.669000228 +0200
@@ -285,11 +285,11 @@
 #ifdef __WXMSW__
 
 	memdc.
-	SetFont ( wxFont::wxFont ( 6, wxSWISS, wxNORMAL, wxBOLD ) );
+	SetFont ( wxFont ( 6, wxSWISS, wxNORMAL, wxBOLD ) );
 #else
 
 	memdc.
-	SetFont ( wxFont::wxFont ( 8, wxSWISS, wxNORMAL, wxBOLD ) );
+	SetFont ( wxFont ( 8, wxSWISS, wxNORMAL, wxBOLD ) );
 #endif
 
 	memdc.
