From 821f395d8b4926539bb7fbcbdbbbe1639c295680 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Wed, 27 Feb 2013 17:44:52 +0000
Subject: [PATCH] Fix issue with Plugins that add Tab to torrentdetails

---
 deluge/ui/gtkui/torrentdetails.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/deluge/ui/gtkui/torrentdetails.py b/deluge/ui/gtkui/torrentdetails.py
index ea64d7b..589c264 100644
--- a/deluge/ui/gtkui/torrentdetails.py
+++ b/deluge/ui/gtkui/torrentdetails.py
@@ -204,6 +204,8 @@ def add_tab(self, tab, generate_menu=True, visible=None):
             tab.is_visible = False
 
         self.tabs[name] = tab
+        if name not in self.translate_tabs:
+            self.translate_tabs[name] = _(name)
 
         self.regenerate_positions()
         if generate_menu:
-- 
1.8.3.1

