From e7a2a8dceedc693a3c069af35405603c879771f3 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Mon, 26 Nov 2012 21:59:48 +0000
Subject: [PATCH] Fix #2109 : Proxy settings not being cleared by setting None

---
 deluge/core/preferencesmanager.py | 17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

diff --git a/deluge/core/preferencesmanager.py b/deluge/core/preferencesmanager.py
index 2a5cb1b..7a3cc2c 100644
--- a/deluge/core/preferencesmanager.py
+++ b/deluge/core/preferencesmanager.py
@@ -471,15 +471,14 @@ def _on_new_release_check(self, key, value):
 
     def _on_set_proxies(self, key, value):
         for k, v in value.items():
-            if v["type"]:
-                proxy_settings = lt.proxy_settings()
-                proxy_settings.type = lt.proxy_type(v["type"])
-                proxy_settings.username = str(v["username"])
-                proxy_settings.password = str(v["password"])
-                proxy_settings.hostname = str(v["hostname"])
-                proxy_settings.port = v["port"]
-                log.debug("setting %s proxy settings", k)
-                getattr(self.session, "set_%s_proxy" % k)(proxy_settings)
+            proxy_settings = lt.proxy_settings()
+            proxy_settings.type = lt.proxy_type(v["type"])
+            proxy_settings.username = str(v["username"])
+            proxy_settings.password = str(v["password"])
+            proxy_settings.hostname = str(v["hostname"])
+            proxy_settings.port = v["port"]
+            log.debug("setting %s proxy settings", k)
+            getattr(self.session, "set_%s_proxy" % k)(proxy_settings)
 
     def _on_rate_limit_ip_overhead(self, key, value):
         log.debug("%s: %s", key, value)
-- 
1.8.0.2

