From fc14b008686ecb403f8d8e00a9aed621c1f23dcf Mon Sep 17 00:00:00 2001
From: Andrew Resch <andrewresch@gmail.com>
Date: Fri, 14 Sep 2012 15:56:13 -0700
Subject: [PATCH 1/2] Only convert the move path to unicode if on windows

---
 deluge/core/torrent.py | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/deluge/core/torrent.py b/deluge/core/torrent.py
index faf59a4..515bb38 100644
--- a/deluge/core/torrent.py
+++ b/deluge/core/torrent.py
@@ -834,13 +834,14 @@ def connect_peer(self, ip, port):
     def move_storage(self, dest):
         """Move a torrent's storage location"""
 
-        # Attempt to convert utf8 path to unicode
-        # Note: Inconsistent encoding for 'dest', needs future investigation
-        try:
-           dest_u = unicode(dest, "utf-8")
-        except TypeError:
-           # String is already unicode
-           dest_u = dest
+        if deluge.common.is_windows():
+            # Attempt to convert utf8 path to unicode
+            # Note: Inconsistent encoding for 'dest', needs future investigation
+            try:
+               dest_u = unicode(dest, "utf-8")
+            except TypeError:
+               # String is already unicode
+               dest_u = dest
 
         if not os.path.exists(dest_u):
             try:
-- 
1.8.0

