From 9286d43ba8ef60f6c0b8382c5c459fc2d7e10e10 Mon Sep 17 00:00:00 2001
From: Chase Sterling <chase.sterling@gmail.com>
Date: Fri, 7 Dec 2012 23:10:09 -0500
Subject: [PATCH] Increase libtorrent alert queue size. fix #2204

---
 ChangeLog           | 1 +
 deluge/core/core.py | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/deluge/core/core.py b/deluge/core/core.py
index 4ca3d96..42fa0f4 100644
--- a/deluge/core/core.py
+++ b/deluge/core/core.py
@@ -95,6 +95,8 @@ def __init__(self, listen_interface=None):
         # Set the user agent
         self.settings = lt.session_settings()
         self.settings.user_agent = "Deluge %s" % deluge.common.get_version()
+        # Increase the alert queue size so that alerts don't get lost
+        self.settings.alert_queue_size = 10000
 
         # Set session settings
         self.settings.send_redundant_have = True
-- 
1.8.0.2

