From 74d59eb6ec5d1a19bbed526d072238824a9afb14 Mon Sep 17 00:00:00 2001
From: Chase Sterling <chase.sterling@gmail.com>
Date: Mon, 3 Dec 2012 01:08:31 -0500
Subject: [PATCH 1/2] Fix the version check in _libtorrent to work as intended.

---
 deluge/_libtorrent.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/_libtorrent.py b/deluge/_libtorrent.py
index ab782d6..e68e535 100644
--- a/deluge/_libtorrent.py
+++ b/deluge/_libtorrent.py
@@ -47,7 +47,7 @@
 
 REQUIRED_VERSION = "0.14.9.0"
 
-def check_version(LT):
+def check_version(lt):
     from deluge.common import VersionSplit
     if VersionSplit(lt.version) < VersionSplit(REQUIRED_VERSION):
         raise ImportError("This version of Deluge requires libtorrent >=%s!" % REQUIRED_VERSION)
-- 
1.8.0.2

