From fc4ca861dab844dbbcf3d3ea5fc8aad98a00f14b Mon Sep 17 00:00:00 2001
From: Chase Sterling <chase.sterling@gmail.com>
Date: Mon, 3 Dec 2012 01:14:17 -0500
Subject: [PATCH 2/2] Fix libtorrent version check in torrent.py. fix #2223

---
 deluge/core/torrent.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/core/torrent.py b/deluge/core/torrent.py
index dd2fa39..f2403a6 100644
--- a/deluge/core/torrent.py
+++ b/deluge/core/torrent.py
@@ -174,7 +174,7 @@ def __init__(self, handle, options, state=None, filename=None, magnet=None):
             self.trackers = []
             # Create a list of trackers
             for value in self.handle.trackers():
-                if lt.version_minor < 15:
+                if lt.version_major == 0 and lt.version_minor < 15:
                     tracker = {}
                     tracker["url"] = value.url
                     tracker["tier"] = value.tier
-- 
1.8.0.2

