From 3309f59333fcacb688438c2e60bdaf4922938182 Mon Sep 17 00:00:00 2001
From: Andrew Resch <andrewresch@gmail.com>
Date: Tue, 18 Sep 2012 17:59:54 -0700
Subject: [PATCH 2/2] Fix previous commit

---
 deluge/core/torrent.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/deluge/core/torrent.py b/deluge/core/torrent.py
index 515bb38..dff29ca 100644
--- a/deluge/core/torrent.py
+++ b/deluge/core/torrent.py
@@ -834,7 +834,7 @@ def connect_peer(self, ip, port):
     def move_storage(self, dest):
         """Move a torrent's storage location"""
 
-        if deluge.common.is_windows():
+        if deluge.common.windows_check():
             # Attempt to convert utf8 path to unicode
             # Note: Inconsistent encoding for 'dest', needs future investigation
             try:
@@ -842,7 +842,9 @@ def move_storage(self, dest):
             except TypeError:
                # String is already unicode
                dest_u = dest
-
+        else:
+            dest_u = dest
+            
         if not os.path.exists(dest_u):
             try:
                 # Try to make the destination path if it doesn't exist
-- 
1.8.0

