From 17cac01673d479ecb4f4dbbc16a5c82bc19907e9 Mon Sep 17 00:00:00 2001
From: John Garland <johnnybg+deluge@gmail.com>
Date: Sun, 16 Sep 2012 18:36:58 +1000
Subject: [PATCH 2/2] Fix deluge-gtk not working with twisted 12 (fixes #2148)

---
 deluge/ui/gtkui/gtkui.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/deluge/ui/gtkui/gtkui.py b/deluge/ui/gtkui/gtkui.py
index 08bcfd4..dea68fd 100644
--- a/deluge/ui/gtkui/gtkui.py
+++ b/deluge/ui/gtkui/gtkui.py
@@ -373,7 +373,11 @@ def do_connect(try_counter):
 
             if self.config["show_connection_manager_on_start"]:
                 # XXX: We need to call a simulate() here, but this could be a bug in twisted
-                reactor.simulate()
+                try:
+                    reactor._simulate()
+                except AttributeError:
+                    # twisted < 12
+                    reactor.simulate()
                 self.connectionmanager.show()
 
 
-- 
1.8.0

