From a3806b6d7aac040d70b0ff5aa4b49769996ec3af Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Tue, 19 Nov 2013 22:12:58 +0000
Subject: [PATCH 2/2] Fix #2365 : Hiding Progress column generates TypeError

---
 deluge/ui/gtkui/torrentview.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/ui/gtkui/torrentview.py b/deluge/ui/gtkui/torrentview.py
index 7aa2bf5..a6dba6c 100644
--- a/deluge/ui/gtkui/torrentview.py
+++ b/deluge/ui/gtkui/torrentview.py
@@ -158,7 +158,7 @@ def cell_data_progress(column, cell, model, row, data):
 
     textstr = _t(state_str)
     if state_str != "Seeding" and value < 100:
-        textstr = textstr + " %.2f%%" % value
+        textstr = "%s %.2f%%" % (textstr, value)
     if cell.get_property("text") != textstr:
         cell.set_property("text", textstr)
 
-- 
1.8.5.4

