From 23b0f58eb4d8113bf62eb090f0f0c8f22f637dd5 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Wed, 5 Dec 2012 18:35:44 +0000
Subject: [PATCH 4/4] fix snafu in last commit

---
 deluge/ui/gtkui/addtorrentdialog.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deluge/ui/gtkui/addtorrentdialog.py b/deluge/ui/gtkui/addtorrentdialog.py
index b9c8359..54e86ea 100644
--- a/deluge/ui/gtkui/addtorrentdialog.py
+++ b/deluge/ui/gtkui/addtorrentdialog.py
@@ -868,7 +868,7 @@ def _on_filename_edited(self, renderer, path, new_text):
             file_path = os.path.join(self.get_file_path(parent), new_text)
             # Don't rename if filename exists
             for row in self.files_treestore[parent].iterchildren():
-                if new_text == row[1] and row[3] > -1:
+                if new_text == row[1]:
                     return
             if os.path.sep in new_text:
                 # There are folders in this path, so we need to create them
-- 
1.8.0.2

