From 98dcc3f26e97156f50e9931f3d53a6ac4429f629 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sat, 9 Aug 2014 00:17:43 +0100
Subject: [PATCH 14/24] [GTKUI] Fix showing the open_folder menuitem

---
 deluge/ui/gtkui/menubar.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/deluge/ui/gtkui/menubar.py b/deluge/ui/gtkui/menubar.py
index 214c4b7..f390311 100644
--- a/deluge/ui/gtkui/menubar.py
+++ b/deluge/ui/gtkui/menubar.py
@@ -190,6 +190,7 @@ def start(self):
                 self.torrentmenu_glade.get_widget(widget).set_no_show_all(True)
         else:
             for widget in non_remote_items:
+                self.torrentmenu_glade.get_widget(widget).show()
                 self.torrentmenu_glade.get_widget(widget).set_no_show_all(False)
 
         self.window.main_glade.get_widget("separatormenuitem").set_visible(not self.config["classic_mode"])
-- 
2.1.0

