From 223c9319c703e4ab2c276c0bbb5f62048940b3f7 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sat, 16 Aug 2014 22:49:13 +0100
Subject: [PATCH 17/24] Fix firing of Finished event when moving

---
 deluge/core/torrentmanager.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/deluge/core/torrentmanager.py b/deluge/core/torrentmanager.py
index 9ecc414..7eb9c06 100644
--- a/deluge/core/torrentmanager.py
+++ b/deluge/core/torrentmanager.py
@@ -1027,7 +1027,7 @@ def on_alert_storage_moved(self, alert):
         torrent.set_save_path(os.path.normpath(alert.handle.save_path()))
         torrent.set_move_completed(False)
 
-        if torrent in self.waiting_on_finish_moving:
+        if torrent_id in self.waiting_on_finish_moving:
             self.waiting_on_finish_moving.remove(torrent_id)
             torrent.is_finished = True
             component.get("EventManager").emit(TorrentFinishedEvent(torrent_id))
@@ -1041,7 +1041,7 @@ def on_alert_storage_moved_failed(self, alert):
         except (RuntimeError, KeyError):
             return
 
-        if torrent in self.waiting_on_finish_moving:
+        if torrent_id in self.waiting_on_finish_moving:
             self.waiting_on_finish_moving.remove(torrent_id)
             torrent.is_finished = True
             component.get("EventManager").emit(TorrentFinishedEvent(torrent_id))
-- 
2.1.0

