From ecf5af1e163af8e84167e572dabbbadacd93c046 Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sun, 31 Aug 2014 12:17:05 +0100
Subject: [PATCH 21/24] [GTKUI] Fix listview error with new config

---
 deluge/ui/gtkui/listview.py | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/deluge/ui/gtkui/listview.py b/deluge/ui/gtkui/listview.py
index 5bdbd00..8db35a1 100644
--- a/deluge/ui/gtkui/listview.py
+++ b/deluge/ui/gtkui/listview.py
@@ -229,10 +229,11 @@ def create_model_filter(self):
         self.set_model_sort()
 
     def set_model_sort(self):
-        for column_state in self.state:
-            if column_state.sort is not None and column_state.sort > -1:
-                self.treeview.get_model().set_sort_column_id(column_state.sort, column_state.sort_order)
-                break
+        if self.state is not None:
+            for column_state in self.state:
+                if column_state.sort is not None and column_state.sort > -1:
+                    self.treeview.get_model().set_sort_column_id(column_state.sort, column_state.sort_order)
+                    break
 
     def stabilize_sort_func(self, sort_func):
         def stabilized(model, iter1, iter2, data):
-- 
2.1.0

