Index: gtk+-2.24.8/gtk/gtksettings.c
===================================================================
--- gtk+-2.24.8.orig/gtk/gtksettings.c	2011-11-10 18:38:03.000000000 +0100
+++ gtk+-2.24.8/gtk/gtksettings.c	2011-11-12 01:15:24.947735494 +0100
@@ -314,7 +314,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
 								  P_("Fallback Icon Theme Name"),
 								  P_("Name of a icon theme to fall back to"),
-								  NULL,
+								  "gnome",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
