diff -up gtk+-2.22.0/gdk/x11/gdkwindow-x11.c.goi-build-fix gtk+-2.22.0/gdk/x11/gdkwindow-x11.c
--- gtk+-2.22.0/gdk/x11/gdkwindow-x11.c.goi-build-fix	2010-09-27 16:27:50.277110001 -0400
+++ gtk+-2.22.0/gdk/x11/gdkwindow-x11.c	2010-09-27 16:28:05.519110001 -0400
@@ -3707,7 +3707,7 @@ gdk_x11_window_set_user_time (GdkWindow 
 /**
  * gdk_window_set_icon_list:
  * @window: The #GdkWindow toplevel window to set the icon of.
- * @pixbufs: (transfer none) (element-type GdkPixbuf):
+ * @pixbufs::
  *     A list of pixbufs, of different sizes.
  *
  * Sets a list of icons for the window. One of these will be used
