diff -up gtk+-2.24.7/gdk/gdkmarshalers.c.schar gtk+-2.24.7/gdk/gdkmarshalers.c
--- gtk+-2.24.7/gdk/gdkmarshalers.c.schar	2011-10-17 18:05:32.000000000 -0400
+++ gtk+-2.24.7/gdk/gdkmarshalers.c	2011-10-17 20:07:54.859431530 -0400
@@ -5,7 +5,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
diff -up gtk+-2.24.7/gtk/gtkmarshal.c.schar gtk+-2.24.7/gtk/gtkmarshal.c
--- gtk+-2.24.7/gtk/gtkmarshal.c.schar	2011-10-17 20:10:48.682425316 -0400
+++ gtk+-2.24.7/gtk/gtkmarshal.c	2011-10-17 20:10:56.723425027 -0400
@@ -6,7 +6,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
diff -up gtk+-2.24.7/gtk/gtkmarshalers.c.schar gtk+-2.24.7/gtk/gtkmarshalers.c
--- gtk+-2.24.7/gtk/gtkmarshalers.c.schar	2011-10-17 18:05:54.000000000 -0400
+++ gtk+-2.24.7/gtk/gtkmarshalers.c	2011-10-17 20:07:54.860431529 -0400
@@ -6,7 +6,7 @@
 
 #ifdef G_ENABLE_DEBUG
 #define g_marshal_value_peek_boolean(v)  g_value_get_boolean (v)
-#define g_marshal_value_peek_char(v)     g_value_get_schar (v)
+#define g_marshal_value_peek_char(v)     g_value_get_char (v)
 #define g_marshal_value_peek_uchar(v)    g_value_get_uchar (v)
 #define g_marshal_value_peek_int(v)      g_value_get_int (v)
 #define g_marshal_value_peek_uint(v)     g_value_get_uint (v)
