--- cegui/src/RendererModules/Irrlicht/CEGUIIrrlichtGeometryBuffer.cpp.orig	2012-08-05 18:28:02.952111043 -0300
+++ cegui/src/RendererModules/Irrlicht/CEGUIIrrlichtGeometryBuffer.cpp	2012-08-05 18:28:05.658535913 -0300
@@ -56,7 +56,7 @@ IrrlichtGeometryBuffer::IrrlichtGeometry
     d_material.ZWriteEnable = false;
     #if CEGUI_IRR_SDK_VERSION >= 16
         d_material.MaterialType = irr::video::EMT_ONETEXTURE_BLEND;
-        d_material.MaterialTypeParam = irr::video::pack_texureBlendFunc(
+        d_material.MaterialTypeParam = irr::video::pack_textureBlendFunc(
                 irr::video::EBF_SRC_ALPHA,
                 irr::video::EBF_ONE_MINUS_SRC_ALPHA,
                 irr::video::EMFN_MODULATE_1X,
