From 083ded0f35d4404cd378a975e3b6f2af520b4c54 Mon Sep 17 00:00:00 2001
From: Qball Cow <qball@sarine.nl>
Date: Sun, 12 Sep 2010 10:22:00 +0200
Subject: [PATCH] Fix wrong use of screenshot_add_border.

---
 src/plugin.c |   14 +++++++++-----
 1 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/plugin.c b/src/plugin.c
index 6c65a95..76f1fae 100644
--- a/src/plugin.c
+++ b/src/plugin.c
@@ -71,11 +71,15 @@ static void libnotify_update_cover(GmpcMetaWatcher *gmv, mpd_Song *song, MetaDat
 		if(met->content_type == META_DATA_CONTENT_URI)
 		{
 			const gchar *path = meta_data_get_uri(met);
-			GdkPixbuf *pb = gdk_pixbuf_new_from_file_at_scale(path,64,64,TRUE,NULL);
-			if(pb)
-			{
-				screenshot_add_border(&pb);
-			}
+            GError *error = NULL;
+			GdkPixbuf *pb = gdk_pixbuf_new_from_file_at_scale(path,64,64,TRUE,&error);
+            if(error == NULL && pb != NULL)
+            {
+                screenshot_add_border(pb);
+            }else{
+                if(pb) g_object_unref(pb);
+                pb = NULL;
+            }
 			/* Default to gmpc icon when no icon is found */
 			if(!pb) {
 				pb = gtk_icon_theme_load_icon(gtk_icon_theme_get_default(),"gmpc" , 64, 0,NULL);
-- 
1.6.5.GIT

