--- src/plugin.c.org	2010-11-07 14:10:44.370947164 +0100
+++ src/plugin.c	2010-11-07 14:14:58.740995606 +0100
@@ -153,16 +153,13 @@
 		if(not == NULL)
 		{
 			//            notify_notification_close(not, NULL);
-			not = notify_notification_new(summary, buffer,NULL, NULL);
+			not = notify_notification_new(summary, buffer,NULL);
 		}
 		else{
 			notify_notification_update(not, summary, buffer, NULL);
 		}
 		notify_notification_set_urgency(not, NOTIFY_URGENCY_LOW);
 
-		if(cfg_get_single_value_as_int_with_default(config, "libnotify-plugin", "attach-to-tray", TRUE))
-			notify_notification_attach_to_status_icon(not, tray_icon2_gsi);
-
 		g_free(summary);
 		/* Add the song to the widget */
 		g_object_set_data_full(G_OBJECT(not), "mpd-song", mpd_songDup(song),  (GDestroyNotify)mpd_freeSong);
