diff -up farsight2-0.0.21/common/gtk-doc.mak.make farsight2-0.0.21/common/gtk-doc.mak
--- farsight2-0.0.21/common/gtk-doc.mak.make	2010-06-23 19:20:40.000000000 -0300
+++ farsight2-0.0.21/common/gtk-doc.mak	2010-07-29 12:15:35.814081248 -0300
@@ -232,6 +232,6 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
diff -up farsight2-0.0.21/docs/libs/Makefile.am.make farsight2-0.0.21/docs/libs/Makefile.am
diff -up farsight2-0.0.21/python/Makefile.am.make farsight2-0.0.21/python/Makefile.am
--- farsight2-0.0.21/python/Makefile.am.make	2009-05-08 18:34:14.000000000 -0300
+++ farsight2-0.0.21/python/Makefile.am	2010-07-29 12:13:53.223329639 -0300
@@ -1,5 +1,4 @@
 PYDEFS=`pkg-config --variable=defsdir pygobject-2.0`
-[ -z "$PYDEFS" && PYDEFS=`pkg-config --variable=defsdir pygtk-2.0`]
 GSTPYDEFS=`pkg-config --variable=defsdir gst-python-0.10`
 
 AM_CPPFLAGS =						\
