--- SConstruct.orig	2010-04-08 01:03:13.000000000 -0300
+++ SConstruct	2010-04-08 01:04:02.031494660 -0300
@@ -216,7 +216,7 @@
 if sys.platform == 'win32':
     env['CXXFLAGS'] = compileFlags
 else:
-    env['CXXFLAGS'] = compileFlags + [ '-fvisibility-inlines-hidden' ]
+    env['CXXFLAGS'] = compileFlags + [ '-fvisibility-inlines-hidden' ] + [ '_SB_CFLAGS_' ]
 env['CPPDEFINES'] = cppDefines
 
 # pgo - Profile Guided Optimization
