diff -up miniupnpc-1.8/CMakeLists.txt.orig miniupnpc-1.8/CMakeLists.txt
--- miniupnpc-1.8/CMakeLists.txt.orig	2013-08-13 10:04:23.583753440 -0300
+++ miniupnpc-1.8/CMakeLists.txt	2013-08-13 10:05:03.228754958 -0300
@@ -128,25 +128,25 @@ if (UPNPC_BUILD_TESTS)
   target_link_libraries (testminixml ${LDLIBS})
 
   add_executable (minixmlvalid minixmlvalid.c minixml.c)
-  target_link_libraries (minixmlvalid ${LDLIBS})
+  target_link_libraries (minixmlvalid ${LDLIBS} -L. -lminiupnpc)
 
   add_executable (testupnpreplyparse testupnpreplyparse.c
                                      minixml.c upnpreplyparse.c)
-  target_link_libraries (testupnpreplyparse ${LDLIBS})
+  target_link_libraries (testupnpreplyparse ${LDLIBS} -L. -lminiupnpc)
 
   add_executable (testigddescparse testigddescparse.c
                                    igd_desc_parse.c minixml.c miniupnpc.c miniwget.c minissdpc.c
                                    upnpcommands.c upnpreplyparse.c minisoap.c connecthostport.c
                                    portlistingparse.c receivedata.c
   )
-  target_link_libraries (testigddescparse ${LDLIBS})
+  target_link_libraries (testigddescparse ${LDLIBS} -L. -lminiupnpc)
 
   add_executable (testminiwget testminiwget.c
                                miniwget.c miniupnpc.c minisoap.c upnpcommands.c minissdpc.c
                                upnpreplyparse.c minixml.c igd_desc_parse.c connecthostport.c
                                portlistingparse.c receivedata.c
   )
-  target_link_libraries (testminiwget ${LDLIBS})
+  target_link_libraries (testminiwget ${LDLIBS} -L. -lminiupnpc)
 
 # set (UPNPC_INSTALL_TARGETS ${UPNPC_INSTALL_TARGETS} testminixml minixmlvalid testupnpreplyparse testigddescparse testminiwget)
 endif (UPNPC_BUILD_TESTS)
