diff -up miniupnpc-1.8/CMakeLists.txt.orig miniupnpc-1.8/CMakeLists.txt
--- miniupnpc-1.8/CMakeLists.txt.orig	2013-08-13 10:28:07.207807956 -0300
+++ miniupnpc-1.8/CMakeLists.txt	2013-08-13 10:28:22.476808540 -0300
@@ -117,7 +117,6 @@ endif (UPNPC_BUILD_STATIC)
 if (UPNPC_BUILD_SHARED)
   add_library (upnpc-shared SHARED ${MINIUPNPC_SOURCES})
   set_target_properties (upnpc-shared PROPERTIES OUTPUT_NAME "miniupnpc")
-  set_target_properties (upnpc-shared PROPERTIES VERSION ${MINIUPNPC_VERSION})
   set_target_properties (upnpc-shared PROPERTIES SOVERSION ${MINIUPNPC_API_VERSION})
   target_link_libraries (upnpc-shared ${LDLIBS})
   set (UPNPC_INSTALL_TARGETS ${UPNPC_INSTALL_TARGETS} upnpc-shared)
