--- libextractor-1.2/src/plugins/thumbnailffmpeg_extractor.c.orig	2013-08-28 11:30:31.000000000 +0200
+++ libextractor-1.2/src/plugins/thumbnailffmpeg_extractor.c	2013-10-27 07:03:46.045814925 +0100
@@ -327,10 +327,10 @@
 #endif
 }
 
-#if AV_VERSION_INT(54,25,0) > LIBAVUTIL_VERSION_INT
+#if AV_VERSION_INT(54,25,0) > LIBAVCODEC_VERSION_INT
 #define ENUM_CODEC_ID enum CodecID
 #else
-#define ENUM_CODEC_ID enum AvCodecID
+#define ENUM_CODEC_ID enum AVCodecID
 #endif
 
 
