--- src/plugins/gif_extractor.c.orig	2012-10-22 17:57:03.542677323 -0200
+++ src/plugins/gif_extractor.c	2012-10-22 18:05:17.907914333 -0200
@@ -68,7 +68,7 @@ EXTRACTOR_gif_extract_method (struct EXT
   int et;
   char dims[128];
 /* GIFLIB >= 5.0 does not define GIF_LIB_VERSION anymore */
-#if !defined (GIF_LIB_VERSION)
+#if !defined (GIF_LIB_VERSION) && GIFLIB_MAJOR >= 5
   int gif_error;
   
   gif_error = 0;
