--- src/plugins/thumbnailffmpeg_extractor.c.ffmpeg	2013-03-23 18:18:55.532362683 -0300
+++ src/plugins/thumbnailffmpeg_extractor.c	2013-03-23 18:23:16.420980021 -0300
@@ -325,7 +325,7 @@ calculate_thumbnail_dimensions (int src_
  * @param ec extraction context to use
  */
 static void
-extract_image (enum CodecID image_codec_id,
+extract_image (enum AVCodecID image_codec_id,
                struct EXTRACTOR_ExtractContext *ec)
 {
   AVDictionary *opts;
@@ -618,7 +618,7 @@ struct MIMEToDecoderMapping
   /**
    * Corresponding ffmpeg decoder ID.
    */
-  enum CodecID codec_id;
+  enum AVCodecID codec_id;
 };
 
 
