From fc8bd87c9827632828789f50caa196329d35c5a8 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Sun, 6 Feb 2011 14:44:28 +0100
Subject: [PATCH 2/2] Fix build with new libnotify

---
 src/padevchooser.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/padevchooser.c b/src/padevchooser.c
index 3703e34..2f506fd 100644
--- a/src/padevchooser.c
+++ b/src/padevchooser.c
@@ -171,7 +171,7 @@ static void notify_event(const char *title, const char*text) {
 
     if (!notification) {
         s = g_strdup_printf("<i>%s</i>\n%s", title, text);
-        notification = notify_notification_new_with_status_icon(title, s, "audio-card", tray_icon);
+        notification = notify_notification_new(title, s, NULL);
         notify_notification_set_category(notification, "device.added");
         notify_notification_set_urgency(notification, NOTIFY_URGENCY_LOW);
         g_signal_connect_swapped(G_OBJECT(notification), "closed", G_CALLBACK(notification_closed), NULL);
-- 
1.7.1

