diff -up flake-0.11/configure.shared flake-0.11/configure
--- flake-0.11/configure.shared	2007-08-05 02:55:24.000000000 -0300
+++ flake-0.11/configure	2011-10-28 21:13:30.249938075 -0200
@@ -298,7 +298,7 @@ LIBOBJFLAGS=""
 PROJLDFLAGS=-Wl,--warn-common
 LDCONFIG="ldconfig"
 LIBPREF="lib"
-LIBSUF=".a"
+LIBSUF=".so"
 LIB='$(LIBPREF)$(NAME)$(LIBSUF)'
 EXESUF=""
 BUILDSUF=""
@@ -775,6 +775,7 @@ if test "$have_strnlen" = "yes" ; then
 fi
 
 libflake_version=`grep '#define FLAKE_VERSION ' "$source_path/libflake/flake.h" | sed 's/[^0-9\.]//g'`
+soname_version=`echo $libflake_version | cut -d. -f1`
 
 echo "LIB_INSTALL_EXTRA_CMD=${LIB_INSTALL_EXTRA_CMD}" >> config.mak
 echo "EXTRALIBS=$extralibs" >> config.mak
@@ -826,6 +827,8 @@ if test "$source_path_used" = "yes" ; th
 fi
 echo "SRC_PATH=$source_path" >> config.mak
 echo "BUILD_ROOT=$PWD" >> config.mak
+echo "LIB_VER=$libflake_version" >> config.mak
+echo "SONAME=$soname_version" >> config.mak
 
 # Do not overwrite config.h if unchanged to avoid superfluous rebuilds.
 diff $TMPH config.h >/dev/null 2>&1
diff -up flake-0.11/libflake/Makefile.shared flake-0.11/libflake/Makefile
--- flake-0.11/libflake/Makefile.shared	2007-08-05 02:55:24.000000000 -0300
+++ flake-0.11/libflake/Makefile	2011-10-28 21:14:17.564881806 -0200
@@ -3,7 +3,7 @@
 #
 include ../config.mak
 
-CFLAGS=$(OPTFLAGS) -I. -I.. -I$(SRC_PATH)/libflake \
+CFLAGS=$(OPTFLAGS) -fPIC -DPIC -I. -I.. -I$(SRC_PATH)/libflake \
 	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_ISOC9X_SOURCE \
 	-DHAVE_CONFIG_H
 
@@ -23,8 +23,7 @@ all: $(EXTRADEPS) $(LIB)
 
 $(LIB): $(OBJS)
 	rm -f $@
-	$(AR) rc $@ $^ $(EXTRAOBJS)
-	$(RANLIB) $@
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,$@.$(SONAME) -o $@ $^
 
 %.o: %.c
 	$(CC) $(CFLAGS) $(LIBOBJFLAGS) -c -o $@ $<
@@ -44,8 +43,9 @@ install: install-libs install-headers
 
 install-libs: $(LIB)
 	install -d "$(libdir)"
-	install -m 644 $(LIB) "$(libdir)"
-	$(LIB_INSTALL_EXTRA_CMD)
+	install -m 755 $(LIB) "$(libdir)/$(LIB).$(LIB_VER)"
+	ln -s $(LIB).$(LIB_VER) "$(libdir)/$(LIB).$(SONAME)"
+	ln -s $(LIB).$(SONAME) "$(libdir)/$(LIB)"
 
 install-headers:
 	install -d "$(incdir)"
