--- espeak-1.45.03-source/src/Makefile.orig	2011-03-22 07:56:44.000000000 -0300
+++ espeak-1.45.03-source/src/Makefile	2011-04-24 17:50:50.958294259 -0300
@@ -71,19 +71,19 @@ SRCS3 = espeak.cpp
 OBJS3=$(patsubst %.cpp,%.o,$(SRCS3))
 LIBS3=-lstdc++ -L . -lespeak
 
-CXXFLAGS=-O2
+CFLAGS=-O2
 
 
 all: $(BIN_NAME) $(LIB_NAME) $(STATIC_LIB_NAME) $(BIN2_NAME)
 
 .cpp.o:
-	$(CXX) $(CXXFLAGS) $(USE_AUDIO) -D PATH_ESPEAK_DATA=\"$(DATADIR)\" -Wall -pedantic -I. -c -fno-exceptions  $<
+	$(CC) $(CFLAGS) $(USE_AUDIO) -D PATH_ESPEAK_DATA=\"$(DATADIR)\" -Wall -pedantic -I. -c -fno-exceptions  $<
 
 $(BIN_NAME): $(OBJS1)
-	$(CXX) $(LDFLAGS) -o $@ $(OBJS1) $(LIBS1)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS1) $(LIBS1)
 
 $(BIN2_NAME): $(OBJS3) $(LIB_NAME)
-	$(CXX) $(LDFLAGS) -o $@ $(OBJS3) $(LIBS3)
+	$(CC) $(LDFLAGS) -o $@ $(OBJS3) $(LIBS3)
 
 
 
