--- kazehakase-0.5.8-svn3873_trunk/src/kz-notebook.c.gtk222	2009-10-16 23:53:10.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/src/kz-notebook.c	2010-09-08 23:44:28.000000000 +0900
@@ -82,7 +82,11 @@
 			      guint 	   page_num,
 			      gpointer     data);
 static void     switch_page  (GtkNotebook     *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                              gpointer page,
+#else
 			      GtkNotebookPage *page,
+#endif
 			      guint 	   page_num);
 
 static void	cb_page_reordered (GtkNotebook *notebook,
@@ -322,7 +326,13 @@
 }
 
 static void
-switch_page (GtkNotebook *notebook, GtkNotebookPage *page, guint page_num)
+switch_page (GtkNotebook *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+        gpointer page,
+#else
+        GtkNotebookPage *page,
+#endif
+        guint page_num)
 {
 	KzTabLabel *tab;
 	gchar *title;
--- kazehakase-0.5.8-svn3873_trunk/src/kz-window.c.gtk222	2010-05-04 00:38:44.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/src/kz-window.c	2010-09-08 23:40:40.000000000 +0900
@@ -213,11 +213,19 @@
 					       guint            page_num,
 					       KzWindow        *kz);
 static void     cb_notebook_switch_page       (GtkNotebook     *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                                               gpointer page,
+#else
 					       GtkNotebookPage *page,
+#endif
 					       guint            page_num,
 					       KzWindow        *kz);
 static void     cb_notebook_switch_page_after (GtkNotebook     *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                                               gpointer page,
+#else
 					       GtkNotebookPage *page,
+#endif
 					       guint            page_num,
 					       KzWindow        *kz);
 static void     cb_gesture_stack_motion       (KzGesture       *gesture,
@@ -1912,7 +1920,12 @@
 }
 
 static void
-cb_notebook_switch_page (GtkNotebook *notebook, GtkNotebookPage *page,
+cb_notebook_switch_page (GtkNotebook *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                         gpointer page,
+#else
+                         GtkNotebookPage *page,
+#endif
 			 guint page_num, KzWindow *kz)
 {
 	KzWeb *kzweb = KZ_WINDOW_NTH_WEB(kz, page_num);
@@ -1971,7 +1984,12 @@
 }
 
 static void
-cb_notebook_switch_page_after (GtkNotebook *notebook, GtkNotebookPage *page,
+cb_notebook_switch_page_after (GtkNotebook *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                               gpointer page,
+#else
+                               GtkNotebookPage *page,
+#endif
 			       guint page_num, KzWindow *kz)
 {
 	kz_feed_info_change_state(KZ_FEED_INFO(kz->feed_info));
--- kazehakase-0.5.8-svn3873_trunk/src/sidebar/kz-tabtree.c.gtk222	2009-10-16 23:53:09.000000000 +0900
+++ kazehakase-0.5.8-svn3873_trunk/src/sidebar/kz-tabtree.c	2010-09-08 23:36:25.000000000 +0900
@@ -85,7 +85,11 @@
 
 /* signal handlers for Notebook on KzWindow */
 static void     cb_switch_page    (GtkNotebook     *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                                   gpointer page,
+#else
 				   GtkNotebookPage *page,
+#endif
 				   guint            page_num,
 				   KzTabTree       *tabtree);
 
@@ -696,7 +700,12 @@
 
 
 static void
-cb_switch_page (GtkNotebook *notebook, GtkNotebookPage *page,
+cb_switch_page (GtkNotebook *notebook,
+#if GTK_CHECK_VERSION(2, 21, 6)
+                gpointer page,
+#else
+                GtkNotebookPage *page,
+#endif
 		guint page_num, KzTabTree *tabtree)
 {
 	KzWeb *web;
