--- kazehakase-0.5.6-svn3769_trunk/src/dialogs/kz-about-dialog.c.version	2009-01-28 04:25:07.000000000 +0900
+++ kazehakase-0.5.6-svn3769_trunk/src/dialogs/kz-about-dialog.c	2009-04-12 18:01:15.000000000 +0900
@@ -95,7 +95,12 @@
 			GDK_RGB_DITHER_NORMAL,
 			0, 0);
 	/* Draw version and backends under the logo */
-	version = g_strdup_printf (_("Version %s"), VERSION);
+	version = g_strdup_printf (_("Version %s"), 
+#ifdef VERSION_VENDOR
+		VERSION_VENDOR);
+#else
+		VERSION);
+#endif
 	backends = backend_names();
 	if (backends)
 	{
