diff -rc xpaint-2.9.8/image.h xpaint-2.9.8.1/image.h
*** xpaint-2.9.8/image.h	2010-11-07 18:21:29.000000000 +0100
--- xpaint-2.9.8.1/image.h	2010-11-07 22:35:43.923473540 +0100
***************
*** 18,23 ****
--- 18,25 ----
  
  /* $Id: image.h,v 1.17 2005/03/20 20:15:32 demailly Exp $ */
  
+ #include <X11/Xlib.h>
+ 
  typedef struct {
      int refCount;		/* reference count */
  
***************
*** 63,69 ****
      unsigned long sourceMask;
  } Image;
  
- #ifdef _XLIB_H_
  /* Used to transfer information to routines in iprocess.c */
  typedef struct {
      int mergeType;
--- 65,70 ----
***************
*** 110,117 ****
      Image *canvas;
  } imageprocessinfo;
  
- #endif
- 
  #define ImageAlphaPixel(image, x, y) ((image)->alpha+((x)+(y)*(image)->width))
  
  #define ImagePixel(image, x, y)						\
--- 111,116 ----
diff -rc xpaint-2.9.8/Imakefile xpaint-2.9.8.1/Imakefile
*** xpaint-2.9.8/Imakefile	2010-11-07 14:37:19.000000000 +0100
--- xpaint-2.9.8.1/Imakefile	2010-11-07 22:41:08.820473541 +0100
***************
*** 119,125 ****
  
  start::
  	if test `grep -c XAW3DXF Local.xawdefs` = 1 && \
!            test `grep -c 'libXaw3dxft.a' Local.xawdefs` = 0 ; then \
  	cd Xaw3dxft ; xmkmf ; make ; cd .. ; fi
  	@if ! [ -d xaw_incdir ] ; then ./configure ; make ; fi
  
--- 119,125 ----
  
  start::
  	if test `grep -c XAW3DXF Local.xawdefs` = 1 && \
!            test `grep -c libXaw3dxft.a Local.xawdefs` = 1 ; then \
  	cd Xaw3dxft ; xmkmf ; make ; cd .. ; fi
  	@if ! [ -d xaw_incdir ] ; then ./configure ; make ; fi
  
diff -rc xpaint-2.9.8/Local.config xpaint-2.9.8.1/Local.config
*** xpaint-2.9.8/Local.config	2010-08-04 16:33:29.000000000 +0200
--- xpaint-2.9.8.1/Local.config	2010-11-07 22:32:05.343473541 +0100
***************
*** 175,181 ****
  
  XCOMM  You must have the freetype2 library.
  XFT_INCLUDE = -I/usr/include/freetype2
! XFT_LIB = -lXft
  
  XCOMM  You must have the XPM library.
  XCOMM  Make sure the path name to the library is correct.
--- 175,181 ----
  
  XCOMM  You must have the freetype2 library.
  XFT_INCLUDE = -I/usr/include/freetype2
! XFT_LIB = -lXft -lfontconfig
  
  XCOMM  You must have the XPM library.
  XCOMM  Make sure the path name to the library is correct.
diff -rc xpaint-2.9.8/Xaw3dxft/Imakefile xpaint-2.9.8.1/Xaw3dxft/Imakefile
*** xpaint-2.9.8/Xaw3dxft/Imakefile	2010-11-07 14:47:43.000000000 +0100
--- xpaint-2.9.8.1/Xaw3dxft/Imakefile	2010-11-07 22:37:06.900473540 +0100
***************
*** 117,122 ****
--- 117,125 ----
  distclean:: clean
  	rm -f Makefile
  
+ strip::
+ 	strip /usr/lib/libXaw3dxft*
+ 
  laygram.c: laygram.y
  	$(YACC) -d laygram.y
  	sed -e 's/yy/LayYY/g' y.tab.c > laygram.c
