diff -up libQtGTL-0.9.1/examples/QtCTL/CMakeLists.txt.dso libQtGTL-0.9.1/examples/QtCTL/CMakeLists.txt
--- libQtGTL-0.9.1/examples/QtCTL/CMakeLists.txt.dso	2010-02-13 04:23:11.000000000 -0600
+++ libQtGTL-0.9.1/examples/QtCTL/CMakeLists.txt	2012-01-14 06:19:05.807398813 -0600
@@ -21,5 +21,5 @@ qt4_automoc(${Qt_CTL_SRCS})
 
 add_executable(qCTL ${Qt_CTL_SRCS})
 # set_target_properties( QtCTL PROPERTIES COMPILE_FLAGS ${OPENCTL_COMPILE_FLAGS})
-target_link_libraries(qCTL ${QT_LIBRARY_DIR} ${QT_QTGUI_LIBRARY} ${OPENCTL_LIBRARIES} QtGTL)
+target_link_libraries(qCTL ${QT_LIBRARY_DIR} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${OPENCTL_LIBRARIES} QtGTL)
 # install(TARGETS QtCTL DESTINATION ${BIN_INSTALL_DIR} )
diff -up libQtGTL-0.9.1/examples/QtShiva/CMakeLists.txt.dso libQtGTL-0.9.1/examples/QtShiva/CMakeLists.txt
--- libQtGTL-0.9.1/examples/QtShiva/CMakeLists.txt.dso	2010-02-13 04:23:11.000000000 -0600
+++ libQtGTL-0.9.1/examples/QtShiva/CMakeLists.txt	2012-01-14 06:19:05.807398813 -0600
@@ -20,5 +20,5 @@ qt4_wrap_ui(Qt_SHIVA_SRCS MainWindow.ui)
 qt4_automoc(${Qt_SHIVA_SRCS})
 
 add_executable(qShiva ${Qt_SHIVA_SRCS} )
-target_link_libraries(qShiva ${QT_LIBRARY_DIR} ${QT_QTGUI_LIBRARY} ${OPENSHIVA_LIBRARIES} QtGTL QtShiva)
+target_link_libraries(qShiva ${QT_LIBRARY_DIR} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${OPENSHIVA_LIBRARIES} QtGTL QtShiva)
 # install(TARGETS QtShiva DESTINATION ${BIN_INSTALL_DIR} )
diff -up libQtGTL-0.9.1/QtGTL/CMakeLists.txt.dso libQtGTL-0.9.1/QtGTL/CMakeLists.txt
--- libQtGTL-0.9.1/QtGTL/CMakeLists.txt.dso	2010-02-13 04:23:11.000000000 -0600
+++ libQtGTL-0.9.1/QtGTL/CMakeLists.txt	2012-01-14 06:19:05.808398800 -0600
@@ -1,5 +1,6 @@
 
 find_package(GTLCore REQUIRED)
+find_package(OpenShiva REQUIRED)
 
 include_directories(${QT_QTGUI_INCLUDE_DIR} ${QT_QTCORE_INCLUDE_DIR} ${QT_INCLUDE_DIR} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${GTLCORE_INCLUDE_DIR})
 
diff -up libQtGTL-0.9.1/QtGTL/QtGTL.pc.cmake.dso libQtGTL-0.9.1/QtGTL/QtGTL.pc.cmake
--- libQtGTL-0.9.1/QtGTL/QtGTL.pc.cmake.dso	2010-02-13 04:23:11.000000000 -0600
+++ libQtGTL-0.9.1/QtGTL/QtGTL.pc.cmake	2012-01-14 06:24:01.634700459 -0600
@@ -6,6 +6,6 @@ includedir=@INCLUDE_INSTALL_DIR@
 Name: QtGTL
 Description: Qt Wrappers for Graphics Transformation Language
 Version: @QTGTL_VERSION@
-Requires:
-Libs: -L${libdir} -lGTLCore -lQtGTL
+Requires: GTLCore
+Libs: -L${libdir} -lQtGTL
 Cflags: -I${includedir}
diff -up libQtGTL-0.9.1/QtShiva/QtShiva.pc.cmake.dso libQtGTL-0.9.1/QtShiva/QtShiva.pc.cmake
--- libQtGTL-0.9.1/QtShiva/QtShiva.pc.cmake.dso	2010-02-13 04:23:11.000000000 -0600
+++ libQtGTL-0.9.1/QtShiva/QtShiva.pc.cmake	2012-01-14 06:24:54.375041116 -0600
@@ -6,6 +6,6 @@ includedir=@INCLUDE_INSTALL_DIR@
 Name: QtShiva
 Description: Qt Wrapers for the Shiva Kernel Transformation Language
 Version: @QTSHIVA_VERSION@
-Requires:
-Libs: -L${libdir} -lOpenShiva -lQtShiva
+Requires: OpenShiva
+Libs: -L${libdir} -lQtShiva
 Cflags: -I${includedir}
