--- picard/musicdns/avcodec.c.ffmpeg	2011-05-20 22:51:41.302382994 -0300
+++ picard/musicdns/avcodec.c	2011-05-20 22:51:52.219077845 -0300
@@ -251,7 +251,7 @@ decode(PyObject *self, PyObject *args)
     codec_context = NULL;
     for (i = 0; i < format_context->nb_streams; i++) {
         codec_context = (AVCodecContext *)format_context->streams[i]->codec;
-        if (codec_context && codec_context->codec_type == CODEC_TYPE_AUDIO)
+        if (codec_context && codec_context->codec_type == AVMEDIA_TYPE_AUDIO)
             break;
     }
     if (codec_context == NULL) {
