--- xfce-utils-4.7.1/scripts/startxfce4.in.orig	2010-11-07 07:52:14.000000000 -0200
+++ xfce-utils-4.7.1/scripts/startxfce4.in	2010-11-14 18:07:05.008889336 -0200
@@ -29,12 +29,12 @@
 if test "x$XDG_DATA_DIRS" = "x"
 then
   if test "x@_datadir_@" = "x/usr/local/share" -o "x@_datadir_@" = "x/usr/share"; then
-    XDG_DATA_DIRS="/usr/local/share:/usr/share"
+    XDG_DATA_DIRS="/etc/xfce/xdg:/usr/local/share:/usr/share"
   else
-    XDG_DATA_DIRS="@_datadir_@:/usr/local/share:/usr/share"
+    XDG_DATA_DIRS="/etc/xfce/xdg:@_datadir_@:/usr/local/share:/usr/share"
   fi
 else
-  XDG_DATA_DIRS="@_datadir_@:$XDG_DATA_DIRS"
+  XDG_DATA_DIRS="/etc/xfce/xdg:@_datadir_@:$XDG_DATA_DIRS"
 fi
 export XDG_DATA_DIRS
 
