--- xfce-utils-4.6.1/scripts/xinitrc.in.in.orig	2009-04-09 20:21:13.000000000 -0300
+++ xfce-utils-4.6.1/scripts/xinitrc.in.in	2009-12-09 19:58:38.000000000 -0200
@@ -25,6 +25,11 @@
 fi
 [ -d "$XDG_CACHE_HOME" ] || mkdir "$XDG_CACHE_HOME"
 
+if [ -f "$XDG_CONFIG_HOME"/.xfcerc ]; then 
+  source "$XDG_CONFIG_HOME"/.xfcerc
+else
+  source /etc/xfce/xfcerc
+fi
 
 # set up XDG user directores.  see
 # http://freedesktop.org/wiki/Software/xdg-user-dirs
