diff -up sane-backends-1.0.22/backend/Makefile.in.automake sane-backends-1.0.22/backend/Makefile.in
--- sane-backends-1.0.22/backend/Makefile.in.automake	2011-01-30 19:50:55.000000000 +0100
+++ sane-backends-1.0.22/backend/Makefile.in	2011-03-14 15:46:47.946422433 +0100
@@ -4881,7 +4881,8 @@ install-firmware-path:
 	  done
 
 install-lockpath:
-	$(mkinstalldirs) -m 775 -g $(LOCKPATH_GROUP) $(DESTDIR)$(locksanedir)
+	$(mkinstalldirs) -m 775 -g $(LOCKPATH_GROUP) $(DESTDIR)$(locksanedir) || \
+		$(mkinstalldirs) -m 775 $(DESTDIR)$(locksanedir)
 
 uninstall-hook:
 	rm -rf $(DESTDIR)$(libdir)/sane $(DESTDIR)$(configdir) $(DESTDIR)$(locksanedir)
