--- configure.in.orig	2014-02-21 20:06:48.298822402 -0300
+++ configure.in	2014-02-21 20:07:11.375938934 -0300
@@ -349,7 +349,7 @@ $as_echo_n "Checking for systemd..."
 if test -e /usr/include/systemd/sd-daemon.h ; then
     AC_DEFINE(HAVE_SYSTEMD, 1, [Is /usr/include/systemd/sd-daemon.h available?])
     with_systemd="yes"
-    SYSTEMD_LIBS=" -lsystemd-daemon"
+    SYSTEMD_LIBS=" -lsystemd"
     AC_SUBST(SYSTEMD_LIBS)
     $as_echo "yes"
 else
