--- snownews-1.5.12/configure.orig	2009-12-14 14:55:02.059524232 -0500
+++ snownews-1.5.12/configure	2009-12-14 14:55:18.154460825 -0500
@@ -13,7 +13,7 @@ chomp($xmlldflags);
 
 my $prefix = "/usr/local";
 my $cflags = "-Wall -Wno-format-y2k -O2 -DLOCALEPATH=\"\\\"\$(LOCALEPATH)\\\"\" -DOS=\\\"$os\\\" $xmlcflags \$(EXTRA_CFLAGS) ";
-my $ldflags = "-lncurses -lcrypto $xmlldflags \$(EXTRA_LDFLAGS) ";
+my $ldflags = "-lncursesw -lcrypto $xmlldflags \$(EXTRA_LDFLAGS) ";
 
 my $use_nls = 1;
 parse_cmdl_line();
