--- snownews-1.5.10/Makefile.orig	2008-08-02 21:36:22.000000000 -0400
+++ snownews-1.5.10/Makefile	2008-08-02 21:36:46.000000000 -0400
@@ -60,7 +60,7 @@ install-bin: snownews
 	if [ ! -d "$(DESTDIR)$(PREFIX)/bin" ]; then \
 		mkdir -p $(DESTDIR)$(PREFIX)/bin; \
 	fi
-	$(INSTALL) -s snownews $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) snownews $(DESTDIR)$(PREFIX)/bin
 	$(INSTALL) opml2snow $(DESTDIR)$(PREFIX)/bin
 	if [ ! -f "$(DESTDIR)$(PREFIX)/bin/snow2opml" ]; then \
 		(cd $(DESTDIR)$(PREFIX)/bin && \
