--- snownews-1.5.12/doc/man/snownews.1.in.orig	2009-10-05 21:28:01.000000000 -0400
+++ snownews-1.5.12/doc/man/snownews.1.in	2009-10-05 21:29:18.000000000 -0400
@@ -184,19 +184,6 @@ variable "http_proxy". Snownews expects 
 .I http://your_proxy.org:PORT/
 (http://proxy.your_isp.com:8080/).
 .P
-.B Automatic update checking
-.P
-Once a week the program will automatically check for a new version when it exits.
-It will request a file named "version" from my webserver and compare
-the returned value with the local program version. If they differ it
-notifies the user of a new version. It is done on exit to be least
-annoying I hope. This way you can easily ignore it if you don't want
-to upgrade for some reason or can't, because you're using a package
-that came with your distribution, etc. To disable this check, put the value
--1 into the file ~/.snownews/updatecheck. Snownews will
-not transfer anything to my server. It will just download a file
-containing the version number and compare it locally.
-.P
 .B Plugins
 .P
 Snownews has a plugin architecture that allows to load feeds from external
