--- src/mediadec_ffmpeg.cpp.ffmpeg	2015-09-30 22:29:09.673619118 -0300
+++ src/mediadec_ffmpeg.cpp	2015-09-30 22:29:28.406682748 -0300
@@ -271,7 +271,7 @@ wxImage wxFfmpegMediaDecoder::GetNextFra
 			avcodec_decode_video2(m_codecCtx, m_frame, &frameFinished, &packet);
 			if (frameFinished) {
 				SwsContext* imgConvertCtx = sws_getContext(m_codecCtx->width, m_codecCtx->height, m_codecCtx->pix_fmt,
-						m_width, m_height, PIX_FMT_RGB24, SWS_BICUBIC, NULL, NULL, NULL);
+						m_width, m_height, AV_PIX_FMT_RGB24, SWS_BICUBIC, NULL, NULL, NULL);
 				if (imgConvertCtx == NULL) {
 					av_free_packet(&packet);
 					return wxImage();
