diff -up dvd+rw-tools-7.1/growisofs_mmc.cpp.blocks dvd+rw-tools-7.1/growisofs_mmc.cpp
--- dvd+rw-tools-7.1/growisofs_mmc.cpp.blocks	2012-04-06 12:31:49.336132465 +0200
+++ dvd+rw-tools-7.1/growisofs_mmc.cpp	2012-04-06 12:32:15.131645600 +0200
@@ -1588,8 +1588,8 @@ static int minus_r_reserve_track (Scsi_C
   unsigned int blocks;
 
     blocks = (unsigned int)(size/2048);
+    blocks += 15, blocks &= ~15;
     if (is_dao)	dao_blocks = blocks;
-    else	blocks += 15, blocks &= ~15;
 
     fprintf (stderr,"%s: reserving %u blocks",ioctl_device,blocks);
     if (is_dao && blocks<380000)
