diff -Naupr xar-1.5.orig/configure xar-1.5/configure
--- xar-1.5.orig/configure	2007-05-13 18:09:50.000000000 +0200
+++ xar-1.5/configure	2007-05-30 14:40:04.000000000 +0200
@@ -2800,7 +2800,7 @@ case "${host}" in
 	CFLAGS="$CFLAGS"
 	abi="elf"
 		CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE"
-	RPATH="-Wl,-rpath,"
+	RPATH=""
 	;;
   *-*-netbsd*)
 	echo "$as_me:$LINENO: checking ABI" >&5
diff -Naupr xar-1.5.orig/configure.ac xar-1.5/configure.ac
--- xar-1.5.orig/configure.ac	2007-05-11 04:51:25.000000000 +0200
+++ xar-1.5/configure.ac	2007-05-30 14:40:13.000000000 +0200
@@ -109,7 +109,7 @@ case "${host}" in
 	abi="elf"
 	dnl Linux needs this for things like asprintf() and poll() flags.
 	CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE"
-	RPATH="-Wl,-rpath,"
+	RPATH=""
 	;;
   *-*-netbsd*)
 	AC_MSG_CHECKING([ABI])
