diff -up sysvinit-2.88dsf/man/init.8.slk sysvinit-2.88dsf/man/init.8
--- sysvinit-2.88dsf/man/init.8.slk	2009-09-10 05:28:49.000000000 -0300
+++ sysvinit-2.88dsf/man/init.8	2010-11-17 22:16:00.863157562 -0200
@@ -96,7 +96,7 @@ mounted before users can log in.  Then a
 are processed.
 .PP
 When starting a new process, \fBinit\fP first checks whether the file
-\fI/etc/initscript\fP exists. If it does, it uses this script to
+\fI/sbin/initscript\fP exists. If it does, it uses this script to
 start the process.
 .PP
 Each time a child terminates, \fBinit\fP records the fact and the reason
@@ -269,7 +269,7 @@ file in the directory \fI/etc/init.d\fP
 .SH FILES
 .nf
 /etc/inittab
-/etc/initscript
+/sbin/initscript
 /dev/console
 /var/run/utmp
 /var/log/wtmp
diff -up sysvinit-2.88dsf/man/initscript.5.slk sysvinit-2.88dsf/man/initscript.5
--- sysvinit-2.88dsf/man/initscript.5.slk	2009-09-10 05:28:49.000000000 -0300
+++ sysvinit-2.88dsf/man/initscript.5	2010-11-17 22:16:00.864157340 -0200
@@ -19,15 +19,15 @@
 .SH NAME
 initscript \- script that executes inittab commands.
 .SH SYNOPSIS
-/bin/sh /etc/initscript id runlevels action process
+/bin/sh /sbin/initscript id runlevels action process
 .SH DESCRIPTION
-When the shell script \fI/etc/initscript\fP is present, \fBinit\fP
+When the shell script \fI/sbin/initscript\fP is present, \fBinit\fP
 will use it to execute the commands from \fIinittab\fP.
 This script can be used to set things like \fBulimit\fP and
 \fBumask\fP default values for every process.
 .SH EXAMPLES
 This is a sample initscript, which might be installed on your
-system as \fI/etc/initscript.sample\fP.
+system as \fI/sbin/initscript.sample\fP.
 .RS
 .sp
 .nf
@@ -37,7 +37,7 @@ system as \fI/etc/initscript.sample\fP.
 # initscript   Executed by init(8) for every program it
 #              wants to spawn like this:
 #
-#              /bin/sh /etc/initscript <id> <level> <action> <process>
+#              /bin/sh /sbin/initscript <id> <level> <action> <process>
 #
 
   # Set umask to safe level, and enable core dumps.
@@ -65,7 +65,7 @@ with this can make your system un(re)boo
 .RE
 .SH FILES
 /etc/inittab,
-/etc/initscript.
+/sbin/initscript.
 .SH AUTHOR
 Miquel van Smoorenburg ,<miquels@cistron.nl>
 .SH "SEE ALSO"
diff -up sysvinit-2.88dsf/src/Makefile.slk sysvinit-2.88dsf/src/Makefile
--- sysvinit-2.88dsf/src/Makefile.slk	2010-04-11 06:30:27.000000000 -0300
+++ sysvinit-2.88dsf/src/Makefile	2010-11-17 22:18:15.494199915 -0200
@@ -11,7 +11,7 @@
 CPPFLAGS =
 CFLAGS	?= -ansi -O2 -fomit-frame-pointer
 override CFLAGS += -W -Wall -D_GNU_SOURCE
-STATIC	=
+STATIC	= -static
 
 # For some known distributions we do not build all programs, otherwise we do.
 BIN	=
diff -up sysvinit-2.88dsf/src/initscript.sample.slk sysvinit-2.88dsf/src/initscript.sample
--- sysvinit-2.88dsf/src/initscript.sample.slk	2009-09-10 05:28:49.000000000 -0300
+++ sysvinit-2.88dsf/src/initscript.sample	2010-11-17 22:16:00.864157340 -0200
@@ -1,14 +1,14 @@
 #
-# initscript	If this script is intalled as /etc/initscript,
+# initscript	If this script is intalled as /sbin/initscript,
 #		it is executed by init(8) for every program it
 #		wants to spawn like this:
 #
-#		/bin/sh /etc/initscript <id> <level> <action> <process>
+#		/bin/sh /sbin/initscript <id> <level> <action> <process>
 #
 #		It can be used to set the default umask and ulimit
 #		of all processes. By default this script is installed
-#		as /etc/initscript.sample, so to enable it you must
-#		rename this script first to /etc/initscript.
+#		as /sbin/initscript.sample, so to enable it you must
+#		rename this script first to /sbin/initscript.
 #
 # Version:	@(#)initscript 1.10 10-Dec-1995 MvS.
 #
diff -up sysvinit-2.88dsf/src/paths.h.slk sysvinit-2.88dsf/src/paths.h
--- sysvinit-2.88dsf/src/paths.h.slk	2010-04-11 06:30:27.000000000 -0300
+++ sysvinit-2.88dsf/src/paths.h	2010-11-17 22:16:00.864157340 -0200
@@ -29,12 +29,12 @@
 #define INITTAB		"/etc/inittab"		/* Location of inittab */
 #define INIT		"/sbin/init"		/* Location of init itself. */
 #define NOLOGIN		"/etc/nologin"		/* Stop user logging in. */
-#define FASTBOOT	"/fastboot"		/* Enable fast boot. */
-#define FORCEFSCK	"/forcefsck"		/* Force fsck on boot */
+#define FASTBOOT	"/etc/fastboot"		/* Enable fast boot. */
+#define FORCEFSCK	"/etc/forcefsck"		/* Force fsck on boot */
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
 #define SHELL		"/bin/sh"		/* Default shell */
 #define SULOGIN		"/sbin/sulogin"		/* Sulogin */
-#define INITSCRIPT	"/etc/initscript"	/* Initscript. */
+#define INITSCRIPT	"/sbin/initscript"	/* Initscript. */
 #define PWRSTAT_OLD	"/etc/powerstatus"	/* COMPAT: SIGPWR reason (OK/BAD) */
 #define PWRSTAT		"/var/run/powerstatus"	/* COMPAT: SIGPWR reason (OK/BAD) */
 
@@ -42,9 +42,9 @@
 #define INITLVL		"/etc/initrunlvl"	/* COMPAT: New runlevel */
 #define INITLVL2	"/var/log/initrunlvl"	/* COMPAT: New runlevel */
 				/* Note: INITLVL2 definition needs INITLVL */
-#define HALTSCRIPT1	"/etc/init.d/halt"	/* Called by "fast" shutdown */
+#define HALTSCRIPT1	"/etc/init.d/rc.halt"	/* Called by "fast" shutdown */
 #define HALTSCRIPT2	"/etc/rc.d/rc.0"	/* Called by "fast" shutdown */
-#define REBOOTSCRIPT1	"/etc/init.d/reboot"	/* Ditto. */
+#define REBOOTSCRIPT1	"/etc/init.d/rc.reboot"	/* Ditto. */
 #define REBOOTSCRIPT2	"/etc/rc.d/rc.6"	/* Ditto. */
 #endif
 
