diff -up lapack-3.4.0/INSTALL/make.inc.gfortran.orig lapack-3.4.0/INSTALL/make.inc.gfortran
--- lapack-3.4.0/INSTALL/make.inc.gfortran.orig	2011-11-11 16:08:04.000000000 -0500
+++ lapack-3.4.0/INSTALL/make.inc.gfortran	2011-11-28 14:56:08.990672240 -0500
@@ -13,9 +13,9 @@ SHELL = /bin/sh
 #  desired load options for your machine.
 #
 FORTRAN  = gfortran 
-OPTS     = -O2
+#OPTS     = -O2
 DRVOPTS  = $(OPTS)
-NOOPT    = -O0
+NOOPT    = 
 LOADER   = gfortran
 LOADOPTS =
 #
