diff -up lapack-3.4.0/lapacke/Makefile.fedora lapack-3.4.0/lapacke/Makefile
--- lapack-3.4.0/lapacke/Makefile.fedora	2011-11-29 09:13:01.433799657 -0500
+++ lapack-3.4.0/lapacke/Makefile	2011-11-29 09:15:03.035199954 -0500
@@ -48,6 +48,9 @@ lapacke:
 	cd src && $(MAKE)
 	cd utils && $(MAKE)
 
+shlib: lapacke
+	$(CC) $(CFLAGS) -shared -Wl,-soname,$(SHLIB_LAPACKE).@SHORTVER@ -o $(SHLIB_LAPACKE).@LONGVER@ src/*.o utils/*.o $(LIBS) -lgfortran -lc
+
 clean: cleanlib
 
 cleanlib:
diff -up lapack-3.4.0/lapacke/make.inc.fedora lapack-3.4.0/lapacke/make.inc
--- lapack-3.4.0/lapacke/make.inc.fedora	2011-11-29 09:11:35.717927221 -0500
+++ lapack-3.4.0/lapacke/make.inc	2011-11-29 09:17:08.068555023 -0500
@@ -50,14 +50,14 @@
 CC = gcc
 CFLAGS =
 LINKER = gfortran
-LDFLAGS =
+LDFLAGS = $(RPM_LD_FLAGS)
 #
 # The name of the libraries to be created/linked to
 # Ensure that the libraries have the same data model (LP64/ILP64).
 #
-LAPACKE = lapacke.a
-LIBS =  /Users/julie/Documents/Boulot/lapack-dev/lapack/trunk/lapack_LINUX.a \
-        /Users/julie/Documents/Boulot/lapack-dev/lapack/trunk/blas_LINUX.a  -lm
+LAPACKE = liblapacke.a
+SHLIB_LAPACKE = liblapacke.so
+LIBS = -L.. -llapack -lblas -lm
 #  The archiver and the flag(s) to use when building archive (library)
 #  If your system has no ranlib, set RANLIB = echo.
 #
