diff -up lapack-3.4.2/BLAS/SRC/CMakeLists.txt.shared lapack-3.4.2/BLAS/SRC/CMakeLists.txt
--- lapack-3.4.2/BLAS/SRC/CMakeLists.txt.shared	2010-08-10 13:57:20.000000000 -0300
+++ lapack-3.4.2/BLAS/SRC/CMakeLists.txt	2012-11-18 14:28:43.420906797 -0200
@@ -137,6 +137,11 @@ endif()
   
   
 add_library(blas ${ALLOBJ})
+set_target_properties(blas
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
 if(UNIX)
   target_link_libraries(blas m)
 endif()
diff -up lapack-3.4.2/SRC/CMakeLists.txt.shared lapack-3.4.2/SRC/CMakeLists.txt
--- lapack-3.4.2/SRC/CMakeLists.txt.shared	2012-04-02 16:06:36.000000000 -0300
+++ lapack-3.4.2/SRC/CMakeLists.txt	2012-11-18 14:28:43.422906374 -0200
@@ -415,5 +415,10 @@ endif()
 list(REMOVE_DUPLICATES ALLOBJ)
 
 add_library(lapack ${ALLOBJ} ${ALLXOBJ})
+set_target_properties(lapack
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
 target_link_libraries(lapack ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
 lapack_install_library(lapack)
diff -up lapack-3.4.2/TESTING/MATGEN/CMakeLists.txt.shared lapack-3.4.2/TESTING/MATGEN/CMakeLists.txt
--- lapack-3.4.2/TESTING/MATGEN/CMakeLists.txt.shared	2012-11-18 14:30:34.090463461 -0200
+++ lapack-3.4.2/TESTING/MATGEN/CMakeLists.txt	2012-11-18 14:30:44.802194622 -0200
@@ -71,5 +71,10 @@ else()
   list(REMOVE_DUPLICATES ALLOBJ)
 endif()
 add_library(tmglib ${ALLOBJ} )
+set_target_properties(tmglib
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
 target_link_libraries(tmglib ${LAPACK_LIBRARIES})
 lapack_install_library(tmglib)
diff -up lapack-3.4.2/lapacke/CMakeLists.txt.shared lapack-3.4.2/lapacke/CMakeLists.txt
--- lapack-3.4.2/lapacke/CMakeLists.txt.shared	2012-04-26 10:23:14.000000000 -0300
+++ lapack-3.4.2/lapacke/CMakeLists.txt	2012-11-18 14:28:43.422906374 -0200
@@ -45,13 +45,28 @@ append_subdir_files(UTILS_OBJ "utils")
 
 if (USE_XBLAS)
    add_library(lapacke ${SRC_OBJ} ${SRCX_OBJ} ${UTILS_OBJ})
+   set_target_properties(lapacke
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
    target_link_libraries(lapacke ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
 else (USE_XBLAS)
   if (LAPACKE_WITH_TMG)
    add_library(lapacke ${SRC_OBJ} ${MATGEN_OBJ} ${UTILS_OBJ})
+   set_target_properties(lapacke
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
    target_link_libraries(lapacke tmglib ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
   else (LAPACKE_WITH_TMG)
    add_library(lapacke ${SRC_OBJ} ${UTILS_OBJ})
+   set_target_properties(lapacke
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
    target_link_libraries(lapacke ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES})
   endif(LAPACKE_WITH_TMG)
 endif(USE_XBLAS)
