diff -up lapack-3.4.1/BLAS/SRC/CMakeLists.txt.shared lapack-3.4.1/BLAS/SRC/CMakeLists.txt
--- lapack-3.4.1/BLAS/SRC/CMakeLists.txt.shared	2010-08-10 13:57:20.000000000 -0300
+++ lapack-3.4.1/BLAS/SRC/CMakeLists.txt	2012-08-04 17:48:06.056023104 -0300
@@ -137,6 +137,11 @@ endif()
   
   
 add_library(blas ${ALLOBJ})
+set_target_properties(blas
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
 if(UNIX)
   target_link_libraries(blas m)
 endif()
diff -up lapack-3.4.1/SRC/CMakeLists.txt.shared lapack-3.4.1/SRC/CMakeLists.txt
--- lapack-3.4.1/SRC/CMakeLists.txt.shared	2012-04-02 16:06:36.000000000 -0300
+++ lapack-3.4.1/SRC/CMakeLists.txt	2012-08-04 17:53:08.823927854 -0300
@@ -415,5 +415,10 @@ endif()
 list(REMOVE_DUPLICATES ALLOBJ)
 
 add_library(lapack ${ALLOBJ} ${ALLXOBJ})
+set_target_properties(lapack
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
 target_link_libraries(lapack ${BLAS_LIBRARIES} ${XBLAS_LIBRARY})
 lapack_install_library(lapack)
diff -up lapack-3.4.1/TESTING/MATGEN/CMakeLists.txt.shared lapack-3.4.1/TESTING/MATGEN/CMakeLists.txt
--- lapack-3.4.1/TESTING/MATGEN/CMakeLists.txt.shared	2012-08-04 17:57:56.507012193 -0300
+++ lapack-3.4.1/TESTING/MATGEN/CMakeLists.txt	2012-08-04 17:58:17.045662604 -0300
@@ -71,5 +71,10 @@ set(ALLOBJ ${SMATGEN} ${CMATGEN} ${SCATG
   ${DZATGEN})
 endif()
 add_library(tmglib ${ALLOBJ} )
+set_target_properties(tmglib
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
 target_link_libraries(tmglib lapack)
 lapack_install_library(tmglib)
diff -up lapack-3.4.1/lapacke/CMakeLists.txt.shared lapack-3.4.1/lapacke/CMakeLists.txt
--- lapack-3.4.1/lapacke/CMakeLists.txt.shared	2012-02-17 15:23:57.000000000 -0200
+++ lapack-3.4.1/lapacke/CMakeLists.txt	2012-08-04 17:54:44.195735202 -0300
@@ -45,13 +45,28 @@ append_subdir_files(UTILS_OBJ "utils")
 
 if (USE_XBLAS)
    add_library(lapacke ${SRC_OBJ} ${SRCX_OBJ} ${UTILS_OBJ})
+   set_target_properties(lapacke
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
    target_link_libraries(lapacke lapack ${BLAS_LIBRARIES}  ${XBLAS_LIBRARY})
 else (USE_XBLAS)
   if (LAPACKE_WITH_TMG)
    add_library(lapacke ${SRC_OBJ} ${MATGEN_OBJ} ${UTILS_OBJ})
+   set_target_properties(lapacke
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
    target_link_libraries(lapacke tmglib lapack ${BLAS_LIBRARIES} )
   else (LAPACKE_WITH_TMG)
    add_library(lapacke ${SRC_OBJ} ${UTILS_OBJ})
+   set_target_properties(lapacke
+                         PROPERTIES
+                         VERSION @LONGVER@
+                         SOVERSION @SHORTVER@
+                     )
    target_link_libraries(lapacke lapack ${BLAS_LIBRARIES} )
   endif(LAPACKE_WITH_TMG)
 endif(USE_XBLAS)
