From f28e5ecd4af0f47037dc98eee784ff79f0ee5bff Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Tue, 22 Oct 2013 13:25:35 +0200
Subject: [PATCH 13/20] Reskip custom shader tests

---
 LayoutTests/platform/qt-4.8/TestExpectations | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/LayoutTests/platform/qt-4.8/TestExpectations b/LayoutTests/platform/qt-4.8/TestExpectations
index 63aa329..bcdf82f 100644
--- a/LayoutTests/platform/qt-4.8/TestExpectations
+++ b/LayoutTests/platform/qt-4.8/TestExpectations
@@ -625,6 +625,9 @@ webkit.org/b/6033 css3/filters/nested-filters.html [ ImageOnlyFailure ]
 webkit.org/b/6033 css3/filters/regions-expanding.html [ ImageOnlyFailure ]
 webkit.org/b/6033 css3/filters/simple-filter-rendering.html [ ImageOnlyFailure ]
 
+# CSS shaders only work in qt-wk2
+css3/filters/custom
+
 # UndoManager is not yet enabled.
 webkit.org/b/87908 editing/undomanager [ Skip ]
 
-- 
1.8.5.3

