Description: fix build failure with new bison
Origin: upstream, https://qt.gitorious.org/qt/qtwebkit/commit/60ba8bd5b3575d
Last-Update: 2013-12-22

--- a/Source/ThirdParty/ANGLE/src/compiler/glslang.y
+++ b/Source/ThirdParty/ANGLE/src/compiler/glslang.y
@@ -47,6 +47,7 @@
 %expect 1 /* One shift reduce conflict because of if | else */
 %pure-parser
 %parse-param {TParseContext* context}
+%lex-param {YYLEX_PARAM}
 
 %union {
     struct {
