--- qtwebkit-2.3.3/Source/QtWebKit.pro.orig	2014-03-08 23:39:48.000000000 -0300
+++ qtwebkit-2.3.3/Source/QtWebKit.pro	2014-03-08 23:39:59.492811960 -0300
@@ -31,8 +31,4 @@ haveQt(5):contains(DEFINES, HAVE_QTQUICK
         SUBDIRS += tests
     }
 
-    examples.file = WebKit/qt/examples/examples.pro
-    examples.CONFIG += no_default_target
-    examples.makefile = Makefile
-    SUBDIRS += examples
 }
