commit b013884675e8b21546466919a6a73df6a36a3555
Author: Christoph Feck <christoph@maxiom.de>
Date:   Tue May 13 00:07:01 2014 +0200

    Allow quotation marks in passwords
    
    BUG: 334083
    FIXED-IN: 4.13.2
    REVIEW: 118058

diff --git a/kdesu/kdesud/lexer.cpp b/kdesu/kdesud/lexer.cpp
index f2c5db8..fecbb57 100644
--- a/kdesu/kdesud/lexer.cpp
+++ b/kdesu/kdesud/lexer.cpp
@@ -75,7 +75,7 @@ int Lexer::lex()
 		// handle escaped characters
 		if (c == '\\') {
 		    c = m_Input[in++];
-		    if ((c == '"') || iscntrl(c))
+		    if (iscntrl(c))
 		        return Tok_none;
 		    if (c == '^') {
 		        c = m_Input[in++];
