--- a/kioslave/trash/kio_trash.cpp	
+++ a/kioslave/trash/kio_trash.cpp	
@@ -416,7 +416,7 @@ bool TrashProtocol::createUDSEntry( const QString& physicalPath, const QString&
     }
     mode_t type = buff.st_mode & S_IFMT; // extract file type
     mode_t access = buff.st_mode & 07777; // extract permissions
-    access &= 07555; // make it readonly, since it's in the trashcan
+    //access &= 07555; // make it readonly, since it's in the trashcan
     Q_ASSERT(!internalFileName.isEmpty());
     entry.insert( KIO::UDSEntry::UDS_NAME, internalFileName ); // internal filename, like "0-foo"
     entry.insert( KIO::UDSEntry::UDS_DISPLAY_NAME, displayFileName ); // user-visible filename, like "foo"
