diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1967a29..b443b25 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,13 @@ set_package_properties(LibGcrypt PROPERTIES DESCRIPTION "Libgcrypt is a general
                        TYPE REQUIRED
                        PURPOSE "kwalletd needs libgcrypt to perform PBKDF2-SHA512 hashing"
                       )
+
+find_package(GLIB2)
+set_package_properties(GLIB2 PROPERTIES DESCRIPTION "Low-level core library for data structure handling, portability wrappers, etc."
+                       URL "http://www.gtk.org"
+                       TYPE OPTIONAL
+                      )
+
 # Build options
 option(KDERUNTIME_BUILD_NEPOMUK "Build the Nepomuk KCM and kioslaves" FALSE)
 
diff --git a/phonon/kcm/CMakeLists.txt b/phonon/kcm/CMakeLists.txt
index 3b421e0..558a766 100644
--- a/phonon/kcm/CMakeLists.txt
+++ b/phonon/kcm/CMakeLists.txt
@@ -5,12 +5,6 @@ set_package_properties(PulseAudio PROPERTIES DESCRIPTION "PulseAudio Audio Serve
                        PURPOSE "libpulse is needed for audio setup GUI"
                       )
 
-find_package(GLIB2)
-set_package_properties(GLIB2 PROPERTIES DESCRIPTION "Low-level core library for data structure handling, portability wrappers, etc."
-                       URL "http://www.gtk.org"
-                       TYPE OPTIONAL
-                      )
-
 pkg_check_modules(CANBERRA libcanberra)
 add_feature_info ("Audio setup GUI" CANBERRA_FOUND
                   "libcanberra is needed for audio setup GUI
diff --git a/solid-networkstatus/CMakeLists.txt b/solid-networkstatus/CMakeLists.txt
index f3e7527..6c5aaf5 100644
--- a/solid-networkstatus/CMakeLists.txt
+++ b/solid-networkstatus/CMakeLists.txt
@@ -1,2 +1,4 @@
+include_directories(${GLIB2_INCLUDE_DIR})
+
 add_subdirectory(kded)
 add_subdirectory(tests)
