diff -up kde-runtime-4.11.1/plasma/declarativeimports/qtextracomponents/mouseeventlistener.cpp.mouseeventlistener-2 kde-runtime-4.11.1/plasma/declarativeimports/qtextracomponents/mouseeventlistener.cpp
--- kde-runtime-4.11.1/plasma/declarativeimports/qtextracomponents/mouseeventlistener.cpp.mouseeventlistener-2	2013-09-03 12:52:22.673778954 -0500
+++ kde-runtime-4.11.1/plasma/declarativeimports/qtextracomponents/mouseeventlistener.cpp	2013-09-03 12:54:34.166760188 -0500
@@ -164,6 +164,10 @@ bool MouseEventListener::sceneEventFilte
         return false;
     }
 
+    if (m_lastEvent == event) {
+        return QDeclarativeItem::sceneEventFilter(item, event);
+    }
+
     switch (event->type()) {
     case QEvent::GraphicsSceneMousePress: {
         m_lastEvent = event;
