commit e6e6fb53419ec212ccae083162999d66e7d5663b
Author: David Faure <faure@kde.org>
Date:   Thu Dec 8 23:18:50 2011 +0100

    Fix kde-open exit code
    
    FIXED-IN: 4.8
    BUG: 288389

diff --git a/kioclient/kioclient.cpp b/kioclient/kioclient.cpp
index 6f5dd16..48adaa6 100644
--- a/kioclient/kioclient.cpp
+++ b/kioclient/kioclient.cpp
@@ -268,8 +268,7 @@ bool ClientApp::doIt()
         kFatal(101) << "Session bus not found" ;
 
 #ifdef KIOCLIENT_AS_KDEOPEN
-    app.kde_open(args->url(0), QByteArray(), false);
-    return true;
+    return app.kde_open(args->url(0), QByteArray(), false);
 #elif defined(KIOCLIENT_AS_KDECP)
     checkArgumentCount(argc, 2, 0);
     return app.doCopy(0);
@@ -303,9 +302,9 @@ bool ClientApp::doIt()
     else if ( command == "exec" )
     {
         checkArgumentCount(argc, 2, 3);
-        app.kde_open( args->url( 1 ),
-                      argc == 3 ? args->arg( 2 ) : QString(),
-                      true );
+        return app.kde_open( args->url( 1 ),
+                             argc == 3 ? args->arg( 2 ) : QString(),
+                             true );
     }
     else if ( command == "download" )
     {
