diff -up kde-runtime-4.7.2/CMakeLists.txt.no_activitymanager kde-runtime-4.7.2/CMakeLists.txt
--- kde-runtime-4.7.2/CMakeLists.txt.no_activitymanager	2011-09-26 04:40:16.000000000 -0500
+++ kde-runtime-4.7.2/CMakeLists.txt	2011-10-25 18:57:36.367500738 -0500
@@ -69,7 +69,7 @@ if(NOT WINCE)
 add_subdirectory(kuiserver)
 endif(NOT WINCE)
 add_subdirectory(kwalletd)
-add_subdirectory(activitymanager)
+#add_subdirectory(activitymanager)
 if ( UNIX )
    add_subdirectory(soliduiserver)
    add_subdirectory(solidautoeject)
