--- kioslave/nfs/CMakeLists.txt.orig	2011-02-25 18:54:12.000000000 -0300
+++ kioslave/nfs/CMakeLists.txt	2011-06-18 11:22:15.699051814 -0300
@@ -1,10 +1,12 @@
 set(kio_nfs_PART_SRCS kio_nfs.cpp mount_xdr.c nfs_prot_xdr.c )
 
+pkg_check_modules(TIRPC libtirpc)
 
 kde4_add_plugin(kio_nfs ${kio_nfs_PART_SRCS})
 
+include_directories ( ${TIRPC_INCLUDE_DIRS} )
 
-target_link_libraries(kio_nfs   ${KDE4_KIO_LIBS})
+target_link_libraries(kio_nfs ${KDE4_KIO_LIBS} ${TIRPC_LIBRARIES})
 
 install(TARGETS kio_nfs  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
