--- dropbox.in.orig	2010-04-08 16:04:10.000000000 -0300
+++ dropbox.in	2010-06-28 01:21:51.825745137 -0300
@@ -46,7 +46,7 @@
 UNPACKING = u"Unpacking Dropbox... %d%%"
 
 PARENT_DIR = os.path.expanduser("~")
-DROPBOXD_PATH = "%s/.dropbox-dist/dropboxd" % PARENT_DIR
+DROPBOXD_PATH = "_LIBDIR_/dropbox/dropboxd"
 
 enc = locale.getpreferredencoding()
 def console_print(st=u"", f=sys.stdout, linebreak=True):
@@ -524,7 +524,7 @@
     return newmeth
 
 def start_dropbox():
-    db_path = os.path.expanduser(u"~/.dropbox-dist/dropboxd").encode(sys.getfilesystemencoding())
+    db_path = os.path.expanduser(u"_LIBDIR_/dropbox/dropboxd").encode(sys.getfilesystemencoding())
     if os.access(db_path, os.X_OK):
         f = open("/dev/null", "w")
         # we don't reap the child because we're gonna die anyway, let init do it
