--- dropbox.in.orig	2011-11-11 19:13:56.000000000 -0200
+++ dropbox.in	2012-07-01 15:05:21.175546046 -0300
@@ -55,7 +55,7 @@ DOWNLOADING = u"Downloading Dropbox... %
 UNPACKING = u"Unpacking Dropbox... %d%%"
 
 PARENT_DIR = os.path.expanduser("~")
-DROPBOXD_PATH = "%s/.dropbox-dist/dropboxd" % PARENT_DIR
+DROPBOXD_PATH = "_LIBDIR_/dropbox/dropboxd"
 DESKTOP_FILE = u"@DESKTOP_FILE_DIR@/dropbox.desktop"
 
 enc = locale.getpreferredencoding()
@@ -699,7 +699,7 @@ def requires_dropbox_running(meth):
     return newmeth
 
 def start_dropbox():
-    db_path = os.path.expanduser(u"~/.dropbox-dist/dropboxd").encode(sys.getfilesystemencoding())
+    db_path = os.path.expanduser(u"_LIBDIR_/dropbox/dropboxd").encode(sys.getfilesystemencoding())
     if os.access(db_path, os.X_OK):
         f = open("/dev/null", "w")
         # we don't reap the child because we're gonna die anyway, let init do it
