diff -up nautilus-dropbox-1.4.0/Makefile.am.nognome nautilus-dropbox-1.4.0/Makefile.am
--- nautilus-dropbox-1.4.0/Makefile.am.nognome	2011-11-11 19:13:56.000000000 -0200
+++ nautilus-dropbox-1.4.0/Makefile.am	2012-07-01 15:07:57.088320626 -0300
@@ -13,4 +13,4 @@ dropbox.1: dropbox.txt.in dropbox docgen
 	python docgen.py $(PACKAGE_VERSION) < dropbox.txt.in > dropbox.txt
 	$(RST2MAN) dropbox.txt > dropbox.1
 
-SUBDIRS = data src
+SUBDIRS = data
diff -up nautilus-dropbox-1.4.0/configure.in.nognome nautilus-dropbox-1.4.0/configure.in
--- nautilus-dropbox-1.4.0/configure.in.nognome	2012-04-26 00:12:50.000000000 -0300
+++ nautilus-dropbox-1.4.0/configure.in	2012-07-01 15:06:46.881684199 -0300
@@ -26,7 +26,6 @@ if test "x$HAVE_PKGCONFIG" = "xno"; then
 	AC_MSG_ERROR(you need to have pkgconfig installed !)
 fi
 
-PKG_CHECK_MODULES(NAUTILUS, libnautilus-extension >= $NAUTILUS_REQUIRED)
 PKG_CHECK_MODULES(GLIB, glib-2.0 >= $GLIB_REQUIRED)
 
 AC_PATH_PROG([PYTHON], [python])
@@ -79,37 +78,11 @@ AC_ARG_WITH(nautilus-extension-dir,
               [AS_HELP_STRING([--with-nautilus-extension-dir],
                     [specify the nautilus extension directory])])
 
-# Get nautilus extensions directory
-AC_MSG_CHECKING([for nautilus extension directory])
-if test -n "$with_nautilus_extension_dir"; then
-    NAUTILUS_EXTENSION_DIR=$with_nautilus_extension_dir
-else
-    NAUTILUS_EXTENSION_DIR=`$PKG_CONFIG --variable=extensiondir libnautilus-extension`
-fi
-if test -z "$NAUTILUS_EXTENSION_DIR"; then
-    NAUTILUS_EXTENSION_DIR='${exec_prefix}/lib/nautilus/extension-1.0'
-fi
-
-AC_MSG_RESULT([${NAUTILUS_EXTENSION_DIR}])
-AC_SUBST(NAUTILUS_EXTENSION_DIR)
-
-NAUTILUS_VERSION=`$PKG_CONFIG --modversion libnautilus-extension`
-if test -n "$NAUTILUS_VERSION"; then
-    NAUTILUS_VERSION_MAJOR=`echo -n $NAUTILUS_VERSION | cut -d . -f 1`
-    NAUTILUS_VERSION_MINOR=`echo -n $NAUTILUS_VERSION | cut -d . -f 2`
-    NAUTILUS_VERSION_MICRO=`echo -n $NAUTILUS_VERSION | cut -d . -f 3`
-fi
-
-AC_DEFINE_UNQUOTED(NAUTILUS_VERSION_MAJOR, [$NAUTILUS_VERSION_MAJOR], [ nautilus version major])
-AC_DEFINE_UNQUOTED(NAUTILUS_VERSION_MINOR, [$NAUTILUS_VERSION_MINOR], [ nautilus version minor])
-AC_DEFINE_UNQUOTED(NAUTILUS_VERSION_MICRO, [$NAUTILUS_VERSION_MICRO], [ nautilus version micro])
-
 EMBLEM_DIR='${datadir}/nautilus-dropbox/emblems'
 AC_SUBST(EMBLEM_DIR)
 
 AC_CONFIG_FILES([
 	Makefile 
-	src/Makefile
 	data/Makefile
 	data/icons/Makefile
 	data/icons/hicolor/Makefile
