--- ./libltdl/config/ltmain.m4sh.orig	2009-11-16 07:17:22.000000000 -0600
+++ ./libltdl/config/ltmain.m4sh	2010-02-06 18:13:00.000000000 -0600
@@ -4965,7 +4965,7 @@
 	# Find the relevant object directory and library name.
 	if test "X$installed" = Xyes; then
 	  if test ! -f "$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library \`$lib' was moved."
+	    # func_warning "library \`$lib' was moved."
 	    dir="$ladir"
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
@@ -5496,8 +5496,8 @@
 		  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
 		  test -z "$libdir" && \
 		    func_fatal_error "\`$deplib' is not a valid libtool archive"
-		  test "$absdir" != "$libdir" && \
-		    func_warning "\`$deplib' seems to be moved"
+		  # test "$absdir" != "$libdir" && \
+		  #   func_warning "\`$deplib' seems to be moved"
 
 		  path="-L$absdir"
 		fi
--- ./libltdl/config/ltmain.sh.orig	2009-11-16 07:23:18.000000000 -0600
+++ ./libltdl/config/ltmain.sh	2010-02-06 18:13:26.000000000 -0600
@@ -5407,7 +5407,7 @@
 	# Find the relevant object directory and library name.
 	if test "X$installed" = Xyes; then
 	  if test ! -f "$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library \`$lib' was moved."
+	    # func_warning "library \`$lib' was moved."
 	    dir="$ladir"
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
@@ -5938,8 +5938,8 @@
 		  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
 		  test -z "$libdir" && \
 		    func_fatal_error "\`$deplib' is not a valid libtool archive"
-		  test "$absdir" != "$libdir" && \
-		    func_warning "\`$deplib' seems to be moved"
+		  # test "$absdir" != "$libdir" && \
+		  #   func_warning "\`$deplib' seems to be moved"
 
 		  path="-L$absdir"
 		fi
