diff -up libtool-2.4/libltdl/config/ltmain.m4sh.orig libtool-2.4/libltdl/config/ltmain.m4sh
--- libtool-2.4/libltdl/config/ltmain.m4sh.orig	2010-09-22 11:45:18.000000000 -0300
+++ libtool-2.4/libltdl/config/ltmain.m4sh	2010-10-31 23:24:53.762139268 -0200
@@ -5710,7 +5710,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test "X$installed" = Xyes; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library \`$lib' was moved."
+	    # func_warning "library \`$lib' was moved."
 	    dir="$ladir"
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
@@ -6277,8 +6277,8 @@ func_mode_link ()
 		  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
 		  test -z "$libdir" && \
 		    func_fatal_error "\`$deplib' is not a valid libtool archive"
-		  test "$absdir" != "$libdir" && \
-		    func_warning "\`$deplib' seems to be moved"
+		  # test "$absdir" != "$libdir" && \
+		  #   func_warning "\`$deplib' seems to be moved"
 
 		  path="-L$absdir"
 		fi
diff -up libtool-2.4/libltdl/config/ltmain.sh.orig libtool-2.4/libltdl/config/ltmain.sh
--- libtool-2.4/libltdl/config/ltmain.sh.orig	2010-09-22 11:45:43.000000000 -0300
+++ libtool-2.4/libltdl/config/ltmain.sh	2010-10-31 23:25:46.189762734 -0200
@@ -6499,7 +6499,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test "X$installed" = Xyes; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library \`$lib' was moved."
+	    # func_warning "library \`$lib' was moved."
 	    dir="$ladir"
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
@@ -7066,8 +7066,8 @@ func_mode_link ()
 		  eval libdir=`${SED} -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
 		  test -z "$libdir" && \
 		    func_fatal_error "\`$deplib' is not a valid libtool archive"
-		  test "$absdir" != "$libdir" && \
-		    func_warning "\`$deplib' seems to be moved"
+		  # test "$absdir" != "$libdir" && \
+		  #   func_warning "\`$deplib' seems to be moved"
 
 		  path="-L$absdir"
 		fi
