--- libtool-2.4.3/build-aux/ltmain.sh.orig	2014-10-27 22:16:18.532757790 -0200
+++ libtool-2.4.3/build-aux/ltmain.sh	2014-10-27 22:17:03.562169591 -0200
@@ -7890,7 +7890,7 @@ func_mode_link ()
 	# Find the relevant object directory and library name.
 	if test yes = "$installed"; then
 	  if test ! -f "$lt_sysroot$libdir/$linklib" && test -f "$abs_ladir/$linklib"; then
-	    func_warning "library '$lib' was moved."
+	    # func_warning "library '$lib' was moved."
 	    dir=$ladir
 	    absdir=$abs_ladir
 	    libdir=$abs_ladir
@@ -8457,8 +8457,8 @@ func_mode_link ()
 		  eval libdir=`$SED -n -e 's/^libdir=\(.*\)$/\1/p' $deplib`
 		  test -z "$libdir" && \
 		    func_fatal_error "'$deplib' is not a valid libtool archive"
-		  test "$absdir" != "$libdir" && \
-		    func_warning "'$deplib' seems to be moved"
+		  # test "$absdir" != "$libdir" && \
+		    # func_warning "'$deplib' seems to be moved"
 
 		  path=-L$absdir
 		fi
