diff --git a/libtoolize.in b/libtoolize.in
index 2bce172..16b764d 100644
--- a/libtoolize.in
+++ b/libtoolize.in
@@ -69,6 +69,7 @@
 # General help using GNU software: <http://www.gnu.org/gethelp/>.
 
 : ${TAR=tar}
+test "$TAR" = "tar" && : ${TAR_NO_OWNER=--no-same-owner}
 
 PROGRAM=libtoolize
 
@@ -1279,7 +1280,7 @@ func_copy_cb ()
     $RM "$my_destdir/$my_file"
     if $opt_copy; then
       if { ( cd "$my_srcdir" && $TAR chf - "$my_file" ) 2>/dev/null \
-           | ( umask 0 && cd "$my_destdir" && $TAR xf - ) >/dev/null 2>&1; } \
+           | ( umask 0 && cd "$my_destdir" && $TAR xf - $TAR_NO_OWNER) >/dev/null 2>&1; } \
 	 && touch "$my_destdir/$my_file"; then
 	$opt_quiet || func_echo_once "$my_msg_var"
 	$opt_quiet || func_echo "copying $my_copy_msg"
@@ -1288,7 +1289,7 @@ func_copy_cb ()
     else
       if test "$my_file" = "aclocal.m4"; then
 	if { ( cd "$my_srcdir" && $TAR chf - "$my_file" ) 2>/dev/null \
-	     | ( umask 0 && cd "$my_destdir" && $TAR xf - ) >/dev/null 2>&1 ; }
+	     | ( umask 0 && cd "$my_destdir" && $TAR xf - $TAR_NO_OWNER) >/dev/null 2>&1 ; }
 	then
 	  $opt_quiet || func_echo_once "$my_msg_var"
 	  $opt_quiet || func_echo "copying $my_copy_msg"
