--- allegro-4.4.2/misc/allegro-config.in.multilib	2010-03-03 07:22:34.000000000 -0300
+++ allegro-4.4.2/misc/allegro-config.in	2011-09-18 17:44:51.101414670 -0300
@@ -14,12 +14,12 @@
 
 version=4.4.2
 
-prefix=@prefix@
-exec_prefix=$prefix
+prefix=`pkg-config allegro --variable=prefix`
+exec_prefix=`pkg-config allegro --variable=exec_prefix`
 exec_prefix_set=no
-include_prefix=@INCLUDE_PREFIX@
-include_path=${prefix}/include
-lib_path=${exec_prefix}/lib@LIB_SUFFIX@
+include_prefix=`pkg-config allegro --variable=prefix`
+include_path=`pkg-config allegro --variable=includedir`
+lib_path=`pkg-config allegro --variable=libdir`
 bin_path=${exec_prefix}/bin
 
 static_libs=@LINK_WITH_STATIC_LIBS@
@@ -28,7 +28,7 @@ lib_type=@LIB_TO_LINK@
 accepts_frameworks=@ACCEPT_FRAMEWORKS@
 
 allegro_ldflags="@LDFLAGS@"
-allegro_libs="@LIBS@"
+allegro_libs="`pkg-config allegro --libs --static`"
 allegro_frameworks="@FRAMEWORKS@"
 allegro_cflags=""
 allegro_cppflags=""
