diff -up allegro-4.4.2/CMakeLists.txt~ allegro-4.4.2/CMakeLists.txt
--- allegro-4.4.2/CMakeLists.txt~	2011-03-25 00:18:50.000000000 +0100
+++ allegro-4.4.2/CMakeLists.txt	2011-07-12 11:03:07.975383245 +0200
@@ -907,11 +907,7 @@ option(WANT_LOADPNG "Enable loadpng" on)
 option(WANT_LOGG "Enable logg" on)
 option(WANT_JPGALLEG "Enable JPGAlleg" on)
 
-if(WANT_FRAMEWORKS)
-    set(ADDON_LINKAGE SHARED)
-else()
-    set(ADDON_LINKAGE STATIC)
-endif()
+set(ADDON_LINKAGE SHARED)
 
 if(WANT_ALLEGROGL)
     add_subdirectory(addons/allegrogl)
