diff --git a/declarative-plugins/applet/connectionicon.cpp b/declarative-plugins/applet/connectionicon.cpp
index 7de03ab..47df0cc 100644
--- a/declarative-plugins/applet/connectionicon.cpp
+++ b/declarative-plugins/applet/connectionicon.cpp
@@ -375,9 +375,9 @@ void ConnectionIcon::setIcons()
 #endif
                     } else {
                         NMAppletDebug() << "Emit signal connectionIconChanged(bluetooth)";
-                        m_connectionIcon = "bluetooth";
+                        m_connectionIcon = "network-bluetooth-activated";
                         m_connectionTooltipIcon = "preferences-system-bluetooth";
-                        Q_EMIT connectionIconChanged("bluetooth");
+                        Q_EMIT connectionIconChanged("network-bluetooth-activated");
                         Q_EMIT connectionTooltipIconChanged("preferences-system-bluetooth");
                     }
                 }
