diff -up plasma-nm-0.9.3.3/editor/connectioneditor.cpp.team plasma-nm-0.9.3.3/editor/connectioneditor.cpp
--- plasma-nm-0.9.3.3/editor/connectioneditor.cpp.team	2014-07-15 20:50:46.767771712 -0300
+++ plasma-nm-0.9.3.3/editor/connectioneditor.cpp	2014-07-15 20:50:57.629467452 -0300
@@ -117,11 +117,6 @@ ConnectionEditor::ConnectionEditor(QWidg
     action = new QAction(i18n("VLAN"), this);
     action->setData(NetworkManager::ConnectionSettings::Vlan);
     m_menu->addAction(action);
-#if NM_CHECK_VERSION(0, 9, 9)
-    action = new QAction(i18n("Team"), this);
-    action->setData(NetworkManager::ConnectionSettings::Team);
-    m_menu->addAction(action);
-#endif
 
     action = m_menu->addSeparator();
     action->setText(i18n("VPN"));
diff -up plasma-nm-0.9.3.3/lib/editor/CMakeLists.txt.team plasma-nm-0.9.3.3/lib/editor/CMakeLists.txt
--- plasma-nm-0.9.3.3/lib/editor/CMakeLists.txt.team	2014-07-15 20:49:49.490922414 -0300
+++ plasma-nm-0.9.3.3/lib/editor/CMakeLists.txt	2014-07-15 20:52:39.783795011 -0300
@@ -29,12 +29,6 @@ set(plasmanetworkmanagement_libeditor_SR
     connectionwidget.cpp
 )
 
-if (${NETWORKMANAGERQT_VERSION} VERSION_GREATER 0.9.8.9)
-set(plasmanetworkmanagement_libeditor_SRCS
-    ${plasmanetworkmanagement_libeditor_SRCS}
-    teamwidget.cpp)
-endif()
-
 kde4_add_ui_files(plasmanetworkmanagement_libeditor_SRCS
     ui/advancedpermissionswidget.ui
     ui/ipv4.ui
@@ -60,10 +54,6 @@ kde4_add_ui_files(plasmanetworkmanagemen
     ui/connectionwidget.ui
 )
 
-if (${NETWORKMANAGERQT_VERSION} VERSION_GREATER 0.9.8.9)
-    kde4_add_ui_files(plasmanetworkmanagement_libeditor_SRCS ui/team.ui)
-endif()
-
 kde4_add_library(plasmanetworkmanagement-editor SHARED ${plasmanetworkmanagement_libeditor_SRCS})
 target_link_libraries(plasmanetworkmanagement-editor
     ${KDE4_KDEUI_LIBS}
diff -up plasma-nm-0.9.3.3/lib/editor/connectiondetaileditor.cpp.team plasma-nm-0.9.3.3/lib/editor/connectiondetaileditor.cpp
--- plasma-nm-0.9.3.3/lib/editor/connectiondetaileditor.cpp.team	2014-07-15 20:50:05.544516862 -0300
+++ plasma-nm-0.9.3.3/lib/editor/connectiondetaileditor.cpp	2014-07-15 20:51:28.039016108 -0300
@@ -38,9 +38,6 @@
 #include <bridgewidget.h>
 #include <vlanwidget.h>
 #include <wimaxwidget.h>
-#if NM_CHECK_VERSION(0, 9, 9)
-#include <teamwidget.h>
-#endif
 #include "vpnuiplugin.h"
 
 #include <QDebug>
@@ -314,11 +311,6 @@ void ConnectionDetailEditor::initTabs()
     } else if (type == NetworkManager::ConnectionSettings::Vlan) { // Vlan
         VlanWidget * vlanWidget = new VlanWidget(m_connection->setting(NetworkManager::Setting::Vlan), this);
         m_ui->tabWidget->addTab(vlanWidget, i18n("Vlan"));
-#if NM_CHECK_VERSION(0, 9, 9)
-    } else if (type == NetworkManager::ConnectionSettings::Team) { // Team
-        TeamWidget * teamWidget = new TeamWidget(uuid, m_connection->setting(NetworkManager::Setting::Team), this);
-        m_ui->tabWidget->addTab(teamWidget, i18n("Team"));
-#endif
     } else if (type == NetworkManager::ConnectionSettings::Wimax) { // Wimax
         WimaxWidget * wimaxWidget = new WimaxWidget(m_connection->setting(NetworkManager::Setting::Wimax), this);
         m_ui->tabWidget->addTab(wimaxWidget, i18n("Wimax"));
@@ -369,9 +361,6 @@ void ConnectionDetailEditor::initTabs()
          || type == ConnectionSettings::Wireless
          || type == ConnectionSettings::Infiniband
          || type == ConnectionSettings::Wimax
-     #if NM_CHECK_VERSION(0, 9, 9)
-         || type == ConnectionSettings::Team
-     #endif
          || type == ConnectionSettings::Bond
          || type == ConnectionSettings::Bridge
          || type == ConnectionSettings::Vlan) && !isSlave()) {
