--- CMakeLists.txt.nm	2015-09-16 19:48:15.406649688 -0300
+++ CMakeLists.txt	2015-09-16 19:51:47.458961581 -0300
@@ -18,6 +18,12 @@ elseif(${NETWORKMANAGERQT_VERSION} VERSI
     message(FATAL_ERROR "ERROR: NetworkManagerQt required version is 0.9.8.2. http://projects.kde.org/libnm-qt")
 endif()
 
+find_package(GLIB2)
+set_package_properties(GLIB2 PROPERTIES DESCRIPTION "Low-level core library for data structure handling, portability wrappers, etc."
+                       URL "http://www.gtk.org"
+                       TYPE OPTIONAL
+                      )
+
 if (DISABLE_MODEMMANAGER_SUPPORT)
     message(STATUS "Disabling ModemManager support")
     set(WITH_MODEMMANAGER_SUPPORT 0)
@@ -50,7 +56,8 @@ include_directories(${CMAKE_CURRENT_BINA
                     ${CMAKE_CURRENT_SOURCE_DIR}/libs/editor/
                     ${NETWORKMANAGER_INCLUDE_DIRS}
                     ${NETWORKMANAGERQT_INCLUDE_DIRS}
-                    ${KDE4_INCLUDES})
+                    ${KDE4_INCLUDES}
+                    ${GLIB2_INCLUDE_DIR})
 
 if (MODEMMANAGERQT_FOUND)
     include_directories(${MODEMMANAGERQT_INCLUDE_DIRS})
