diff --git a/applet/declarative/contents/ui/ConnectionItem.qml b/applet/declarative/contents/ui/ConnectionItem.qml
index c6a8b10..34dd957 100644
--- a/applet/declarative/contents/ui/ConnectionItem.qml
+++ b/applet/declarative/contents/ui/ConnectionItem.qml
@@ -421,21 +421,9 @@ ListItem {
             var result = itemLastUsed;
             if (itemSecurityType > PlasmaNM.Enums.None)
                 result += ", " + itemSecurityString;
-
-//             if (itemType == PlasmaNM.Enums.Wireless)
-//                 result += ", " + i18n("Strength: %1%", itemSignal);
-
             return result;
         } else if (itemConnectionState == PlasmaNM.Enums.Activated) {
-            if (itemType == PlasmaNM.Enums.Wired) {
-                return i18n("Connected");/* + ", " + itemSpeed;*/
-            } else if (itemType == PlasmaNM.Enums.Wireless) {
-                return i18n("Connected")/* + ", " + itemSpeed + ", " + i18n("Strength: %1%", itemSignal);*/
-            } else if (itemType == PlasmaNM.Enums.Gsm || itemType == PlasmaNM.Enums.Cdma) {
-                // TODO
-            } else {
-                return i18n("Connected");
-            }
+            return i18n("Connected");
         }
     }
 }
