--- iputils-s20121106/Makefile.orig	2012-11-05 19:13:50.000000000 +0100
+++ iputils-s20121106/Makefile	2012-11-06 14:03:03.214464415 +0100
@@ -21,8 +21,7 @@
 # -------------------------------------
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 #CCOPT=-Wstrict-prototypes -Wall -g -Werror
-CCOPT=-Wstrict-prototypes -Wall -g
-CCOPTOPT=-O2
+CCOPT=-Wstrict-prototypes -fno-strict-aliasing -Werror
 GLIBCFIX=-D_GNU_SOURCE
 DEFINES=
 LDLIB=
@@ -42,7 +41,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(RPM_OPT_FLAGS) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 LASTTAG:=`git describe HEAD | sed -e 's/-.*//'`
