--- iputils-s20140519-dist/Makefile	2014-05-19 23:35:28.000000000 +0200
+++ iputils-s20140519-new/Makefile	2014-05-20 10:51:07.388527442 +0200
@@ -35,7 +35,7 @@ WITHOUT_IFADDRS=no
 ARPING_DEFAULT_DEVICE=
 
 # nettle library for ipv6 ping [yes|no|static]
-USE_NETTLE=yes
+USE_NETTLE=no
 # libgcrypt library for ipv6 ping [no|yes|static]
 USE_GCRYPT=no
 # Crypto library for ping6 [shared|static|no]
@@ -49,7 +49,7 @@ ENABLE_RDISC_SERVER=no
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 # CFLAGS+=-fno-strict-aliasing -Wstrict-prototypes -Wall -Werror -g
 CFLAGS?=-O3 -g
-CFLAGS+=-fno-strict-aliasing -Wstrict-prototypes -Wall
+CFLAGS+=$(RPM_OPT_FLAGS) -fno-strict-aliasing -Wstrict-prototypes -Wall -Werror
 CPPFLAGS+=-D_GNU_SOURCE
 LDLIB=
 
@@ -188,6 +188,7 @@ tftpd.o tftpsubs.o: tftp.h
 
 # -------------------------------------
 # ninfod
+ninfod: CFLAGS:=$(filter-out -Werror,$(CFLAGS))
 ninfod:
 	@set -e; \
 		if [ ! -f ninfod/Makefile ]; then \
