--- iputils-s20121121/Makefile.orig	2012-11-21 11:32:18.000000000 +0100
+++ iputils-s20121121/Makefile	2012-11-22 07:29:27.947670530 +0100
@@ -29,7 +29,7 @@
 # -------------------------------------
 # What a pity, all new gccs are buggy and -Werror does not work. Sigh.
 # CCOPT=-fno-strict-aliasing -Wstrict-prototypes -Wall -Werror -g
-CCOPT=-fno-strict-aliasing -Wstrict-prototypes -Wall -g
+CCOPT=-fno-strict-aliasing -Wstrict-prototypes -Wall -g -Werror
 CCOPTOPT=-O3
 GLIBCFIX=-D_GNU_SOURCE
 DEFINES=
@@ -63,7 +63,7 @@
 IPV6_TARGETS=tracepath6 traceroute6 ping6
 TARGETS=$(IPV4_TARGETS) $(IPV6_TARGETS)
 
-CFLAGS=$(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
+CFLAGS+=$(RPM_OPT_FLAGS) $(CCOPTOPT) $(CCOPT) $(GLIBCFIX) $(DEFINES)
 LDLIBS=$(LDLIB) $(ADDLIB)
 
 LASTTAG:=`git describe HEAD | sed -e 's/-.*//'`
