From c6116b3f7eb2d657141a55afa66d573fd1cda7af Mon Sep 17 00:00:00 2001
From: Sandro Santilli <strk@keybit.net>
Date: Sun, 03 Oct 2010 20:13:27 +0000
Subject: FD_ZERO should not be called between curl_multi_fset and select...

(Backported to 0.8.8 by Kevin Kofler: first hunk omitted.)

---
diff --git a/libbase/curl_adapter.cpp b/libbase/curl_adapter.cpp
index 4604bee..b1caf8a 100644
--- a/libbase/curl_adapter.cpp
+++ b/libbase/curl_adapter.cpp
@@ -642,6 +642,11 @@ CurlStreamFile::fillCache(std::streamsize size)
         //log_debug("cached: %d, size: %d", _cached, size);
 #endif
 
+        // Zero these out _before_ calling curl_multi_fdset!
+        FD_ZERO(&readfd);
+        FD_ZERO(&writefd);
+        FD_ZERO(&exceptfd);
+
         mcode = curl_multi_fdset(_mhandle, &readfd, &writefd, 
                 &exceptfd, &maxfd);
 
@@ -663,10 +668,6 @@ CurlStreamFile::fillCache(std::streamsize size)
             break;
         }
 
-        FD_ZERO(&readfd);
-        FD_ZERO(&writefd);
-        FD_ZERO(&exceptfd);
-
         tv.tv_sec = 0;
         tv.tv_usec = maxSleepUsec;
 
--
cgit v0.8.3.2
