diff --git a/libbase/curl_adapter.cpp b/libbase/curl_adapter.cpp
index 4604bee..c18209b 100644
--- a/libbase/curl_adapter.cpp
+++ b/libbase/curl_adapter.cpp
@@ -657,10 +657,21 @@ CurlStreamFile::fillCache(std::streamsize size)
 
         // A value of -1 means no file descriptors were added.
         if (maxfd < 0) {
+            // As of libcurl 7.21.x, the DNS resolving appears to be going
+            // on in the background, so curl_multi_fdset fails to return
+            // anything useful. So we use the user timeout value to
+            // give DNS enough time to resolve the lookup.
+            if (userTimeout && lastProgress.elapsed() > userTimeout) {
+                log_error(_("Timeout (%u milliseconds) while waiting for "
+                       "the socket to become ready."), userTimeout);
+                break;
+            }
+            else {
 #if GNASH_CURL_VERBOSE
-            log_debug("No filedescriptors; breaking");
+                log_debug("No curl filedescriptors, will perform again");
 #endif
-            break;
+                continue;
+            }
         }
 
         tv.tv_sec = 0;
