From c2913487c8053ab649eae6a069e43b97b050712d Mon Sep 17 00:00:00 2001
From: Bastiaan Jacques <bastiaan@bjacques.org>
Date: Wed, 30 Mar 2011 01:38:52 +0000
Subject: Fix off-by-one that causes an invalid opcode to read past the end of the handlers vector. Fixes bug #32950.

---
diff --git a/libcore/vm/ASHandlers.cpp b/libcore/vm/ASHandlers.cpp
index 2137b01..089e947 100644
--- a/libcore/vm/ASHandlers.cpp
+++ b/libcore/vm/ASHandlers.cpp
@@ -242,7 +242,7 @@ ActionHandler::execute(ActionExec& thread) const
 
 SWFHandlers::SWFHandlers()
     :
-    _handlers(255)
+    _handlers(256)
 {
 
     _handlers[ACTION_END] = ActionHandler(ACTION_END, ActionEnd);
--
cgit v0.8.3.4

