Index: gnash-0.8.9/plugin/npapi/mozilla-sdk/np_entry.cpp
===================================================================
--- gnash-0.8.9.orig/plugin/npapi/mozilla-sdk/np_entry.cpp
+++ gnash-0.8.9/plugin/npapi/mozilla-sdk/np_entry.cpp
@@ -238,7 +238,7 @@ NP_Initialize(NPNetscapeFuncs* aNPNFuncs
     return NS_PluginInitialize();
 }
 
-char *
+const char *
 NP_GetMIMEDescription(void)
 {
     return NPP_GetMIMEDescription();
Index: gnash-0.8.9/plugin/npapi/plugin.cpp
===================================================================
--- gnash-0.8.9.orig/plugin/npapi/plugin.cpp
+++ gnash-0.8.9/plugin/npapi/plugin.cpp
@@ -128,7 +128,7 @@ NPBool plugInitialized = FALSE;
 }
 
 /// \brief Return the MIME Type description for this plugin.
-char*
+const char*
 NPP_GetMIMEDescription(void)
 {
     return const_cast<char *>(MIME_TYPES_DESCRIPTION);
