--- gnash-0.8.10/libbase/GnashImageGif.cpp.orig	2012-01-19 17:17:48.000000000 -0200
+++ gnash-0.8.10/libbase/GnashImageGif.cpp	2014-06-14 17:02:18.308638442 -0300
@@ -121,7 +121,7 @@ GifInput::GifInput(boost::shared_ptr<IOC
 GifInput::~GifInput()
 {
     // Clean up allocated data.
-    DGifCloseFile(_gif);
+    DGifCloseFile(_gif, NULL);
 }
 
 size_t
@@ -269,7 +269,7 @@ GifInput::processRecord(GifRecordType re
 void
 GifInput::read()
 {
-    _gif = DGifOpen(_inStream.get(), &readData); 
+    _gif = DGifOpen(_inStream.get(), &readData, NULL); 
 
     GifRecordType record;
 
