diff -up llvm-3.4/tools/llvm-shlib/Makefile.orig llvm-3.4/tools/llvm-shlib/Makefile
--- llvm-3.4/tools/llvm-shlib/Makefile.orig	2013-11-01 00:35:00.000000000 +1000
+++ llvm-3.4/tools/llvm-shlib/Makefile	2014-01-14 10:13:20.069858909 +1000
@@ -75,7 +75,7 @@ endif
 
 ifeq ($(HOST_OS), $(filter $(HOST_OS), Linux GNU GNU/kFreeBSD))
     # Don't allow unresolved symbols.
-    LLVMLibsOptions += -Wl,--no-undefined
+    LLVMLibsOptions += -Wl,--no-undefined -Wl,-Bsymbolic
 endif
 
 ifeq ($(HOST_OS),SunOS)
