diff -up llvm-3.4.1.src/tools/clang/lib/Driver/Tools.cpp.jx llvm-3.4.1.src/tools/clang/lib/Driver/Tools.cpp
--- llvm-3.4.1.src/tools/clang/lib/Driver/Tools.cpp.jx	2013-12-08 21:59:27.000000000 -0500
+++ llvm-3.4.1.src/tools/clang/lib/Driver/Tools.cpp	2014-06-02 11:24:07.628292753 -0400
@@ -763,9 +763,9 @@ static StringRef getARMFloatABI(const Dr
         break;
       }
       default:
-        // Assume "soft", but warn the user we are guessing.
-        FloatABI = "soft";
-        D.Diag(diag::warn_drv_assuming_mfloat_abi_is) << "soft";
+        // Assume "hard", but warn the user we are guessing.
+        FloatABI = "hard";
+        D.Diag(diag::warn_drv_assuming_mfloat_abi_is) << "hard";
         break;
       }
     }
