--- llvm-3.0.src/tools/gold/Makefile.link_llvmgold_to_lto	2011-05-31 22:00:45.000000000 +0200
+++ llvm-3.0.src/tools/gold/Makefile	2012-02-25 09:40:01.889923932 +0100
@@ -26,6 +26,6 @@
 # Because off_t is used in the public API, the largefile parts are required for
 # ABI compatibility.
 CXXFLAGS+=-I$(BINUTILS_INCDIR) -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CXXFLAGS+=$(SharedLibDir)/$(SharedPrefix)LTO$(SHLIBEXT)
+CXXFLAGS+=-L$(SharedLibDir)/$(SharedPrefix) -lLTO
 
 include $(LEVEL)/Makefile.common
