diff --git a/tools/clang/lib/Driver/Tools.cpp b/tools/clang/lib/Driver/Tools.cpp
index 29713ed..0d23694 100644
--- a/tools/clang/lib/Driver/Tools.cpp
+++ b/tools/clang/lib/Driver/Tools.cpp
@@ -747,7 +747,7 @@ static StringRef getARMFloatABI(const Driver &D,
         FloatABI = "hard";
         break;
       case llvm::Triple::GNUEABI:
-        FloatABI = "softfp";
+        FloatABI = Triple.getVendorName() == "hardfloat" ? "hard" : "softfp";
         break;
       case llvm::Triple::EABI:
         // EABI is always AAPCS, and if it was not marked 'hard', it's softfp
