diff -dur xfce4-netload-plugin-1.1.0.orig/panel-plugin/Makefile.am xfce4-netload-plugin-1.1.0/panel-plugin/Makefile.am
--- xfce4-netload-plugin-1.1.0.orig/panel-plugin/Makefile.am	2012-01-07 15:50:00.000000000 +0100
+++ xfce4-netload-plugin-1.1.0/panel-plugin/Makefile.am	2012-03-09 22:05:17.201838947 +0100
@@ -1,4 +1,4 @@
-plugindir = $(libexecdir)/xfce4/panel-plugins
+plugindir = $(libdir)/xfce4/panel/plugins
 plugin_PROGRAMS = xfce4-netload-plugin
 
 LIBS = @LIBS@ @SOLLIBS@
@@ -34,7 +34,7 @@
 desktop_in_in_files = netload.desktop.in.in
 desktop_in_files = $(desktop_in_in_files:.desktop.in.in=.desktop.in)
 	
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(datadir)/xfce4/panel/plugins
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 @INTLTOOL_DESKTOP_RULE@
 
@@ -47,6 +47,6 @@
 	
 # get full path into .desktop file
 %.desktop.in: %.desktop.in.in
-	sed -e "s^@PLUGIN_PATH@^$(libexecdir)/xfce4/panel-plugins^" \
+	sed -e "s^@PLUGIN_PATH@^$(libdir)/xfce4/panel/plugins^" \
 		$< > $@
 
diff -dur xfce4-netload-plugin-1.1.0.orig/panel-plugin/Makefile.in xfce4-netload-plugin-1.1.0/panel-plugin/Makefile.in
--- xfce4-netload-plugin-1.1.0.orig/panel-plugin/Makefile.in	2012-01-13 12:51:08.000000000 +0100
+++ xfce4-netload-plugin-1.1.0/panel-plugin/Makefile.in	2012-03-09 22:07:05.618821744 +0100
@@ -213,7 +213,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-plugindir = $(libexecdir)/xfce4/panel-plugins
+plugindir = $(libdir)/xfce4/panel/plugins
 xfce4_netload_plugin_SOURCES = \
 	netload.c							\
 	utils.c                                     			\
@@ -245,7 +245,7 @@
 # 
 desktop_in_in_files = netload.desktop.in.in
 desktop_in_files = $(desktop_in_in_files:.desktop.in.in=.desktop.in)
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(datadir)/xfce4/panel/plugins
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
 	wormulon commandline.c						\
@@ -619,7 +619,7 @@
 
 # get full path into .desktop file
 %.desktop.in: %.desktop.in.in
-	sed -e "s^@PLUGIN_PATH@^$(libexecdir)/xfce4/panel-plugins^" \
+	sed -e "s^@PLUGIN_PATH@^$(libdir)/xfce4/panel/plugins^" \
 		$< > $@
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
