diff -up bash-completion-2.1/bash_completion~ bash-completion-2.1/bash_completion
--- bash-completion-2.1/bash_completion~	2013-04-05 13:55:51.000000000 +0300
+++ bash-completion-2.1/bash_completion	2013-04-06 11:43:56.292170674 +0300
@@ -59,7 +59,7 @@ complete -d pushd
 # start of section containing compspecs that can be handled within bash
 
 # user commands see only users
-complete -u write chfn groups slay w sux runuser
+complete -u groups slay w sux
 
 # bg completes with stopped jobs
 complete -A stopped -P '"%' -S '"' bg
diff -up bash-completion-2.1/completions/Makefile.am~ bash-completion-2.1/completions/Makefile.am
--- bash-completion-2.1/completions/Makefile.am~	2013-04-05 13:55:51.000000000 +0300
+++ bash-completion-2.1/completions/Makefile.am	2013-04-06 11:49:59.955323723 +0300
@@ -27,7 +27,6 @@ bashcomp_DATA = a2x \
 		brctl \
 		btdownloadheadless.py \
 		bzip2 \
-		cal \
 		cancel \
 		cardctl \
 		ccache \
@@ -43,7 +42,6 @@ bashcomp_DATA = a2x \
 		chpasswd \
 		chronyc \
 		chrpath \
-		chsh \
 		cksfv \
 		cleanarch \
 		clisp \
@@ -65,7 +63,6 @@ bashcomp_DATA = a2x \
 		desktop-file-validate \
 		dhclient \
 		dict \
-		dmesg \
 		dnsspoof \
 		dot \
 		dpkg \
@@ -76,7 +73,6 @@ bashcomp_DATA = a2x \
 		dumpe2fs \
 		e2freefrag \
 		e2label \
-		eject \
 		eog \
 		ether-wake \
 		evince \
@@ -120,12 +116,10 @@ bashcomp_DATA = a2x \
 		gzip \
 		hcitool \
 		hddtemp \
-		hexdump \
 		hid2hci \
 		hping2 \
 		htop \
 		htpasswd \
-		hwclock \
 		iconv \
 		id \
 		idn \
@@ -137,7 +131,6 @@ bashcomp_DATA = a2x \
 		installpkg \
 		interdiff \
 		invoke-rc.d \
-		ionice \
 		ip \
 		iperf \
 		ipmitool \
@@ -177,7 +170,6 @@ bashcomp_DATA = a2x \
 		list_lists \
 		list_members \
 		list_owners \
-		look \
 		lpq \
 		lpr \
 		lrzip \
@@ -225,7 +217,6 @@ bashcomp_DATA = a2x \
 		nc \
 		ncftp \
 		nethogs \
-		newgrp \
 		newlist \
 		newusers \
 		ngrep \
@@ -283,7 +274,6 @@ bashcomp_DATA = a2x \
 		rdesktop \
 		remove_members \
 		removepkg \
-		renice \
 		reportbug \
 		reptyr \
 		resolvconf \
@@ -298,7 +288,6 @@ bashcomp_DATA = a2x \
 		rpmcheck \
 		rrdtool \
 		rsync \
-		rtcwake \
 		sbcl \
 		sbopkg \
 		screen \
@@ -320,7 +309,6 @@ bashcomp_DATA = a2x \
 		sshow \
 		strace \
 		strings \
-		su \
 		sudo \
 		svcadm \
 		svk \
@@ -478,7 +466,6 @@ CLEANFILES = \
 	mplayer2 \
 	msgsnarf \
 	muttng \
-	ncal \
 	pbzip2 \
 	pccardctl \
 	pdlzip \
@@ -603,10 +590,6 @@ symlinks: $(targetdir) $(DATA)
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) bzip2 $(targetdir)/$$file ; \
 	done
-	for file in ncal ; do \
-		rm -f $(targetdir)/$$file && \
-			$(LN_S) cal $(targetdir)/$$file ; \
-	done
 	for file in pccardctl ; do \
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) cardctl $(targetdir)/$$file ; \
@@ -669,10 +652,6 @@ symlinks: $(targetdir) $(DATA)
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) hcitool $(targetdir)/$$file ; \
 	done
-	for file in hd ; do \
-		rm -f $(targetdir)/$$file && \
-			$(LN_S) hexdump $(targetdir)/$$file ; \
-	done
 	for file in hping hping3 ; do \
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) hping2 $(targetdir)/$$file ; \
diff -up bash-completion-2.1/completions/Makefile.in~ bash-completion-2.1/completions/Makefile.in
--- bash-completion-2.1/completions/Makefile.in~	2013-04-05 13:56:55.000000000 +0300
+++ bash-completion-2.1/completions/Makefile.in	2013-04-06 11:51:40.564047943 +0300
@@ -198,7 +198,6 @@ bashcomp_DATA = a2x \
 		brctl \
 		btdownloadheadless.py \
 		bzip2 \
-		cal \
 		cancel \
 		cardctl \
 		ccache \
@@ -214,7 +213,6 @@ bashcomp_DATA = a2x \
 		chpasswd \
 		chronyc \
 		chrpath \
-		chsh \
 		cksfv \
 		cleanarch \
 		clisp \
@@ -236,7 +234,6 @@ bashcomp_DATA = a2x \
 		desktop-file-validate \
 		dhclient \
 		dict \
-		dmesg \
 		dnsspoof \
 		dot \
 		dpkg \
@@ -247,7 +244,6 @@ bashcomp_DATA = a2x \
 		dumpe2fs \
 		e2freefrag \
 		e2label \
-		eject \
 		eog \
 		ether-wake \
 		evince \
@@ -291,12 +287,10 @@ bashcomp_DATA = a2x \
 		gzip \
 		hcitool \
 		hddtemp \
-		hexdump \
 		hid2hci \
 		hping2 \
 		htop \
 		htpasswd \
-		hwclock \
 		iconv \
 		id \
 		idn \
@@ -308,7 +302,6 @@ bashcomp_DATA = a2x \
 		installpkg \
 		interdiff \
 		invoke-rc.d \
-		ionice \
 		ip \
 		iperf \
 		ipmitool \
@@ -348,7 +341,6 @@ bashcomp_DATA = a2x \
 		list_lists \
 		list_members \
 		list_owners \
-		look \
 		lpq \
 		lpr \
 		lrzip \
@@ -396,7 +388,6 @@ bashcomp_DATA = a2x \
 		nc \
 		ncftp \
 		nethogs \
-		newgrp \
 		newlist \
 		newusers \
 		ngrep \
@@ -454,7 +445,6 @@ bashcomp_DATA = a2x \
 		rdesktop \
 		remove_members \
 		removepkg \
-		renice \
 		reportbug \
 		reptyr \
 		resolvconf \
@@ -469,7 +459,6 @@ bashcomp_DATA = a2x \
 		rpmcheck \
 		rrdtool \
 		rsync \
-		rtcwake \
 		sbcl \
 		sbopkg \
 		screen \
@@ -491,7 +480,6 @@ bashcomp_DATA = a2x \
 		sshow \
 		strace \
 		strings \
-		su \
 		sudo \
 		svcadm \
 		svk \
@@ -649,7 +637,6 @@ CLEANFILES = \
 	mplayer2 \
 	msgsnarf \
 	muttng \
-	ncal \
 	pbzip2 \
 	pccardctl \
 	pdlzip \
@@ -984,10 +971,6 @@ symlinks: $(targetdir) $(DATA)
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) bzip2 $(targetdir)/$$file ; \
 	done
-	for file in ncal ; do \
-		rm -f $(targetdir)/$$file && \
-			$(LN_S) cal $(targetdir)/$$file ; \
-	done
 	for file in pccardctl ; do \
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) cardctl $(targetdir)/$$file ; \
@@ -1050,10 +1033,6 @@ symlinks: $(targetdir) $(DATA)
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) hcitool $(targetdir)/$$file ; \
 	done
-	for file in hd ; do \
-		rm -f $(targetdir)/$$file && \
-			$(LN_S) hexdump $(targetdir)/$$file ; \
-	done
 	for file in hping hping3 ; do \
 		rm -f $(targetdir)/$$file && \
 			$(LN_S) hping2 $(targetdir)/$$file ; \
