diff -up virtuoso-opensource-6.1.0/configure.in.external_iodbc virtuoso-opensource-6.1.0/configure.in
--- virtuoso-opensource-6.1.0/configure.in.external_iodbc	2010-02-03 14:49:18.000000000 -0600
+++ virtuoso-opensource-6.1.0/configure.in	2010-02-04 08:16:51.265605024 -0600
@@ -1833,7 +1833,7 @@ AC_DEFINE(NO_UDBC_SDK,1, [Define if usin
 #
 if test "x$virt_do_iodbc" = "xyes"
 then
-    iodbc_CPPFLAGS="-I$virt_iodbc_dir/include"
+    iodbc_CPPFLAGS=`pkg-config --cflags libiodbc`
 
     saved_CPPFLAGS="$CPPFLAGS"
     CPPFLAGS="$iodbc_CPPFLAGS $CPPFLAGS"
@@ -1842,7 +1842,7 @@ then
 
     ccincl="$iodbc_CPPFLAGS $ccincl"
 
-    iodbc_LDFLAGS="-L$virt_iodbc_dir/lib -liodbc -liodbcinst $LIBDL"
+    iodbc_LDFLAGS=`pkg-config --libs libiodbc`
     AC_CHECK_LIB(iodbc, SQLTables,
 		 [
 		  virt_do_iodbc="yes"
