diff -up virtuoso-opensource-6.1.4/configure.in.no_strip virtuoso-opensource-6.1.4/configure.in
--- virtuoso-opensource-6.1.4/configure.in.no_strip	2011-10-28 18:28:08.000000000 -0500
+++ virtuoso-opensource-6.1.4/configure.in	2012-01-18 11:33:50.367289715 -0600
@@ -632,8 +632,8 @@ if test "x$with_debug" = "xyes"; then
     CCOPT="-g"
     AM_LDFLAGS="-g"
     CCDEBUG=""
-else
-    AM_LDFLAGS="-s"
+#else
+#    AM_LDFLAGS="-s"
 fi
 
 
