From 04b2ab6bff36f89b7c15334c8761cd3d2bbdd13a Mon Sep 17 00:00:00 2001
From: Dario Freddi <drf@kde.org>
Date: Thu, 9 Dec 2010 00:05:50 +0100
Subject: [PATCH 1/5] Post-tagging version bump.

---
 CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16cc262..ca245ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,7 +50,7 @@ endif(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION} VERSI
 
 set(POLKITQT-1_VERSION_MAJOR 0 CACHE INT "PolkitQt-1's major version number" FORCE)
 set(POLKITQT-1_VERSION_MINOR 99 CACHE INT "PolkitQt-1's minor version number" FORCE)
-set(POLKITQT-1_VERSION_PATCH 0 CACHE INT "PolkitQt-1's release version number" FORCE)
+set(POLKITQT-1_VERSION_PATCH 1 CACHE INT "PolkitQt-1's release version number" FORCE)
 set(POLKITQT-1_VERSION_STRING
     "${POLKITQT-1_VERSION_MAJOR}.${POLKITQT-1_VERSION_MINOR}.${POLKITQT-1_VERSION_PATCH}")
 
-- 
1.7.7.4

