From ea0b2e4a923cf3d8c73c623bd8072e39c148a8ff Mon Sep 17 00:00:00 2001
From: Andre Woebbeking <Woebbeking@kde.org>
Date: Mon, 7 Feb 2011 18:19:05 +0100
Subject: [PATCH 2/5] -pedantic

---
 core/polkitqt1-actiondescription.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/core/polkitqt1-actiondescription.h b/core/polkitqt1-actiondescription.h
index e01f2ff..deed65d 100644
--- a/core/polkitqt1-actiondescription.h
+++ b/core/polkitqt1-actiondescription.h
@@ -144,6 +144,6 @@ private:
 };
 }
 
-Q_DECLARE_METATYPE(PolkitQt1::ActionDescription::List);
+Q_DECLARE_METATYPE(PolkitQt1::ActionDescription::List)
 
 #endif //POLKIT_QT_ACTION_DESCRIPTION_H
-- 
1.7.7.4

