diff -up ffmpeg2theora-20151107/src/avinfo.c.ffmpeg ffmpeg2theora-20151107/src/avinfo.c
--- ffmpeg2theora-20151107/src/avinfo.c.ffmpeg	2014-12-11 09:17:17.000000000 -0200
+++ ffmpeg2theora-20151107/src/avinfo.c	2015-11-07 11:35:37.334259827 -0200
@@ -179,7 +179,7 @@ void json_codec_info(FILE *output, AVCod
     case AVMEDIA_TYPE_VIDEO:
         codec_name = fix_codec_name(codec_name);
         json_add_key_value(output, "codec", (void *)codec_name, JSON_STRING, 0, indent);
-        if (enc->pix_fmt != PIX_FMT_NONE) {
+        if (enc->pix_fmt != AV_PIX_FMT_NONE) {
             json_add_key_value(output, "pixel_format", (void *)av_get_pix_fmt_name(enc->pix_fmt), JSON_STRING, 0, indent);
         }
         if (enc->width) {
diff -up ffmpeg2theora-20151107/src/ffmpeg2theora.c.ffmpeg ffmpeg2theora-20151107/src/ffmpeg2theora.c
--- ffmpeg2theora-20151107/src/ffmpeg2theora.c.ffmpeg	2014-12-11 09:17:17.000000000 -0200
+++ ffmpeg2theora-20151107/src/ffmpeg2theora.c	2015-11-07 11:36:21.447937854 -0200
@@ -149,7 +149,7 @@ static AVFrame *frame_alloc(int pix_fmt,
     uint8_t *picture_buf;
     int size;
 
-    picture = avcodec_alloc_frame();
+    picture = av_frame_alloc();
     if (!picture)
         return NULL;
     size = avpicture_get_size (pix_fmt, width, height);
@@ -232,7 +232,7 @@ static ff2theora ff2theora_init() {
         this->kate_streams=NULL;
         this->ignore_non_utf8 = 0;
 
-        this->pix_fmt = PIX_FMT_YUV420P;
+        this->pix_fmt = AV_PIX_FMT_YUV420P;
 
         // ffmpeg2theora --nosound -f dv -H 32000 -S 0 -v 8 -x 384 -y 288 -G 1.5 input.dv
         this->video_gamma  = 0.0;
@@ -1467,17 +1467,8 @@ void ff2theora_output(ff2theora this) {
                                                 display_width, display_height);
                                 output_tmp_p=NULL;
                             }
-                            if ((this->deinterlace==0 && frame->interlaced_frame) ||
-                                this->deinterlace==1) {
-                                if (avpicture_deinterlace((AVPicture *)output,(AVPicture *)output_tmp,this->pix_fmt,display_width,display_height)<0) {
-                                        fprintf(stderr, "Deinterlace failed.\n");
-                                        exit(1);
-                                }
-                            }
-                            else{
-                                av_picture_copy((AVPicture *)output, (AVPicture *)output_tmp, this->pix_fmt,
-                                                display_width, display_height);
-                            }
+                            av_picture_copy((AVPicture *)output, (AVPicture *)output_tmp, this->pix_fmt,
+                                            display_width, display_height);
                             // now output
 
                             if (ppMode)
@@ -1566,7 +1557,7 @@ void ff2theora_output(ff2theora this) {
                     int bytes_per_sample = av_get_bytes_per_sample(aenc->sample_fmt);
 
                     if (avpkt.size > 0) {
-                        if (!audio_frame && !(audio_frame = avcodec_alloc_frame())) {
+                        if (!audio_frame && !(audio_frame = av_frame_alloc())) {
                             fprintf(stderr, "Failed to allocate memory\n");
                             exit(1);
                         }
@@ -1616,7 +1607,7 @@ void ff2theora_output(ff2theora this) {
                             }
                         }
                         oggmux_add_audio(&info, audio_p, dst_nb_samples, audio_eos);
-                        avcodec_free_frame(&audio_frame);
+                        av_frame_free(&audio_frame);
                         this->sample_count += dst_nb_samples;
                     }
                     if(audio_eos) {
