diff -up gnome-panel-2.29.5.1/gnome-panel/panel-menu-items.c.bookmarks-submenu gnome-panel-2.29.5.1/gnome-panel/panel-menu-items.c
--- gnome-panel-2.29.5.1/gnome-panel/panel-menu-items.c.bookmarks-submenu	2010-01-17 18:19:36.000000000 -0500
+++ gnome-panel-2.29.5.1/gnome-panel/panel-menu-items.c	2010-01-17 18:21:00.466529924 -0500
@@ -61,7 +61,7 @@
 #define NAMES_DIR               "/apps/nautilus/desktop"
 #define HOME_NAME_KEY           "/apps/nautilus/desktop/home_icon_name"
 #define COMPUTER_NAME_KEY       "/apps/nautilus/desktop/computer_icon_name"
-#define MAX_ITEMS_OR_SUBMENU    5
+#define MAX_ITEMS_OR_SUBMENU    8
 #define MAX_BOOKMARK_ITEMS      100
 
 G_DEFINE_TYPE (PanelPlaceMenuItem, panel_place_menu_item, GTK_TYPE_IMAGE_MENU_ITEM)
