diff -up pyOpenSSL-0.9/setup.py.posixIncludes pyOpenSSL-0.9/setup.py
--- pyOpenSSL-0.9/setup.py.posixIncludes	2009-03-07 17:50:11.000000000 +0100
+++ pyOpenSSL-0.9/setup.py	2009-09-29 17:46:13.449930310 +0200
@@ -61,6 +61,9 @@ else:
 if sys.platform == 'darwin':
     IncludeDirs = ['/sw/include']
     LibraryDirs = ['/sw/lib']
+elif os.name == 'posix':
+    IncludeDirs = ['/usr/kerberos/include']
+    LibraryDirs = ['/usr/kerberos/lib']
 
 # On Windows, make sure the necessary .dll's get added to the egg.
 data_files = []
