diff -Nur Thunar-1.0.1.orig/thunar/Makefile.in Thunar-1.0.1/thunar/Makefile.in
--- Thunar-1.0.1.orig/thunar/Makefile.in	2009-04-18 08:41:06.000000000 -0600
+++ Thunar-1.0.1/thunar/Makefile.in	2010-02-13 18:06:06.000000000 -0700
@@ -372,7 +372,7 @@
 INTLTOOL_XML_NOMERGE_RULE = @INTLTOOL_XML_NOMERGE_RULE@
 INTLTOOL_XML_RULE = @INTLTOOL_XML_RULE@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lX11
 LIBFAM_CFLAGS = @LIBFAM_CFLAGS@
 LIBFAM_LIBS = @LIBFAM_LIBS@
 LIBFAM_REQUIRED_VERSION = @LIBFAM_REQUIRED_VERSION@
diff -Nur Thunar-1.0.1.orig/thunar-vfs/Makefile.in Thunar-1.0.1/thunar-vfs/Makefile.in
--- Thunar-1.0.1.orig/thunar-vfs/Makefile.in	2009-04-18 08:41:03.000000000 -0600
+++ Thunar-1.0.1/thunar-vfs/Makefile.in	2010-02-13 16:51:16.000000000 -0700
@@ -202,7 +202,7 @@
 thunar_vfs_font_thumbnailer_1_LINK = $(LIBTOOL) --tag=CC \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CCLD) \
 	$(thunar_vfs_font_thumbnailer_1_CFLAGS) $(CFLAGS) \
-	$(thunar_vfs_font_thumbnailer_1_LDFLAGS) $(LDFLAGS) -o $@
+	$(thunar_vfs_font_thumbnailer_1_LDFLAGS) $(LDFLAGS) -lm -o $@
 am_thunar_vfs_mime_cleaner_1_OBJECTS =  \
 	thunar_vfs_mime_cleaner_1-thunar-vfs-mime-cleaner.$(OBJEXT)
 thunar_vfs_mime_cleaner_1_OBJECTS =  \
