diff -up nx-3.5.0/nx-X11/config/cf/host.def~ nx-3.5.0/nx-X11/config/cf/host.def
--- nx-3.5.0/nx-X11/config/cf/host.def~	2011-05-09 16:20:22.000000000 +0300
+++ nx-3.5.0/nx-X11/config/cf/host.def	2011-06-29 01:13:59.677059431 +0300
@@ -908,7 +908,6 @@ XCOMM $XFree86: xc/config/cf/xf86site.de
  *
 #define HasFontconfig NO
  */
-#define HasFontconfig NO
 
 /* 
  * To use a Fontconfig library already installed outside the default search
@@ -924,7 +923,6 @@ XCOMM $XFree86: xc/config/cf/xf86site.de
  *
 #define HasExpat NO
  */
-#define HasExpat NO
 
 /* 
  * To use a Expat library already installed outside the default search
