From 924cc346475dea7dc394316cd7c3d5d0414e538e Mon Sep 17 00:00:00 2001
From: Adam Sampson <ats@offog.org>
Date: Thu, 15 Sep 2016 16:54:42 +0100
Subject: [PATCH] Use sizeof rather than hardcoding snprintf's size argument.

---
 src/su.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/su.c b/src/su.c
index 93ffd2f..d605af8 100644
--- a/src/su.c
+++ b/src/su.c
@@ -373,8 +373,8 @@ static void prepare_pam_close_session (void)
 		              stderr);
 		(void) kill (-pid_child, caught);
 
-		snprintf (kill_msg, _(" ...killed.\n"));
-		snprintf (wait_msg, _(" ...waiting for child to terminate.\n"));
+		snprintf (kill_msg, sizeof kill_msg, _(" ...killed.\n"));
+		snprintf (wait_msg, sizeof wait_msg, _(" ...waiting for child to terminate.\n"));
 
 		(void) signal (SIGALRM, kill_child);
 		(void) alarm (2);
