diff -up shadow-4.3.1/src/useradd.c.process-defaults shadow-4.3.1/src/useradd.c
--- shadow-4.3.1/src/useradd.c.process-defaults	2016-08-22 17:30:01.000000000 +0200
+++ shadow-4.3.1/src/useradd.c	2016-08-25 09:49:51.454720951 +0200
@@ -1974,6 +1974,8 @@ int main (int argc, char **argv)
 	is_shadow_grp = sgr_file_present ();
 #endif
 
+	get_defaults ();
+
 	process_flags (argc, argv);
 
 #ifdef ENABLE_SUBIDS
@@ -1983,8 +1985,6 @@ int main (int argc, char **argv)
 	    (!user_id || (user_id <= uid_max && user_id >= uid_min));
 #endif				/* ENABLE_SUBIDS */
 
-	get_defaults ();
-
 #ifdef ACCT_TOOLS_SETUID
 #ifdef USE_PAM
 	{
