diff -up shadow-4.1.4.3/libmisc/chkname.c.goodname shadow-4.1.4.3/libmisc/chkname.c
--- shadow-4.1.4.3/libmisc/chkname.c.goodname	2011-02-13 18:58:11.000000000 +0100
+++ shadow-4.1.4.3/libmisc/chkname.c	2011-05-18 20:38:50.639341048 +0200
@@ -49,20 +49,28 @@
 static bool is_valid_name (const char *name)
 {
 	/*
-	 * User/group names must match [a-z_][a-z0-9_-]*[$]
-	 */
-	if (('\0' == *name) ||
-	    !((('a' <= *name) && ('z' >= *name)) || ('_' == *name))) {
+         * User/group names must match gnu e-regex:
+         *    [a-zA-Z0-9_.][a-zA-Z0-9_.-]{0,30}[a-zA-Z0-9_.$-]?
+         *
+         * as a non-POSIX, extension, allow "$" as the last char for
+         * sake of Samba 3.x "add machine script"
+         */
+	if ( ('\0' == *name) ||
+             !((*name >= 'a' && *name <= 'z') ||
+               (*name >= 'A' && *name <= 'Z') ||
+               (*name >= '0' && *name <= '9') ||
+               (*name == '_') || (*name == '.') 
+	      )) {
 		return false;
 	}
 
 	while ('\0' != *++name) {
-		if (!(( ('a' <= *name) && ('z' >= *name) ) ||
-		      ( ('0' <= *name) && ('9' >= *name) ) ||
-		      ('_' == *name) ||
-		      ('-' == *name) ||
-		      ( ('$' == *name) && ('\0' == *(name + 1)) )
-		     )) {
+                if (!(  (*name >= 'a' && *name <= 'z') ||
+                        (*name >= 'A' && *name <= 'Z') ||
+                        (*name >= '0' && *name <= '9') ||
+                        (*name == '_') || (*name == '.') || (*name == '-') ||
+                        (*name == '$' && *(name + 1) == '\0') 
+                     )) {
 			return false;
 		}
 	}
diff -up shadow-4.1.4.3/man/groupadd.8.goodname shadow-4.1.4.3/man/groupadd.8
--- shadow-4.1.4.3/man/groupadd.8.goodname	2011-02-16 00:12:34.000000000 +0100
+++ shadow-4.1.4.3/man/groupadd.8	2011-05-18 20:42:53.204324534 +0200
@@ -162,9 +162,7 @@ Shadow password suite configuration\&.
 .RE
 .SH "CAVEATS"
 .PP
-Groupnames must start with a lower case letter or an underscore, followed by lower case letters, digits, underscores, or dashes\&. They can end with a dollar sign\&. In regular expression terms: [a\-z_][a\-z0\-9_\-]*[$]?
-.PP
-Groupnames may only be up to characters long\&.
+Groupnames may only be up to 32 characters long\&.
 .PP
 You may not add a NIS or LDAP group\&. This must be performed on the corresponding server\&.
 .PP
diff -up shadow-4.1.4.3/man/useradd.8.goodname shadow-4.1.4.3/man/useradd.8
--- shadow-4.1.4.3/man/useradd.8.goodname	2011-05-18 20:38:50.635376546 +0200
+++ shadow-4.1.4.3/man/useradd.8	2011-05-18 20:38:50.641332694 +0200
@@ -414,8 +414,6 @@ Similarly, if the username already exist
 \fBuseradd\fR
 will deny the user account creation request\&.
 .PP
-Usernames must start with a lower case letter or an underscore, followed by lower case letters, digits, underscores, or dashes\&. They can end with a dollar sign\&. In regular expression terms: [a\-z_][a\-z0\-9_\-]*[$]?
-.PP
 Usernames may only be up to 32 characters long\&.
 .SH "CONFIGURATION"
 .PP
