diff -up shadow-4.3.1/src/useradd.c.defs-chroot shadow-4.3.1/src/useradd.c
--- shadow-4.3.1/src/useradd.c.defs-chroot	2016-08-22 17:13:29.420068883 +0200
+++ shadow-4.3.1/src/useradd.c	2016-08-22 17:15:14.040343275 +0200
@@ -1938,8 +1938,8 @@ int main (int argc, char **argv)
 #endif				/* ACCT_TOOLS_SETUID */
 
 #ifdef ENABLE_SUBIDS
-	uid_t uid_min = (uid_t) getdef_ulong ("UID_MIN", 1000UL);
-	uid_t uid_max = (uid_t) getdef_ulong ("UID_MAX", 60000UL);
+	uid_t uid_min;
+	uid_t uid_max;
 #endif
 
 	/*
@@ -1958,6 +1958,9 @@ int main (int argc, char **argv)
 	audit_help_open ();
 #endif
 
+	uid_min = (uid_t) getdef_ulong ("UID_MIN", 1000UL);
+	uid_max = (uid_t) getdef_ulong ("UID_MAX", 60000UL);
+
 	sys_ngroups = sysconf (_SC_NGROUPS_MAX);
 	user_groups = (char **) xmalloc ((1 + sys_ngroups) * sizeof (char *));
 	/*
