--- SRC/CMakeLists.txt.soname	2011-03-01 14:26:23.000000000 -0300
+++ SRC/CMakeLists.txt	2011-03-01 15:26:57.912070178 -0300
@@ -378,7 +378,10 @@ endif()
 
 add_library(clapack SHARED ${ALLOBJ} ${ALLXOBJ})
 target_link_libraries(clapack blas f2c)
-set_target_properties(clapack PROPERTIES SOVERSION "${CLAPACK_VERSION}")
+set_target_properties(clapack PROPERTIES
+                              VERSION "${CLAPACK_VERSION}"
+                              SOVERSION "${CPACK_PACKAGE_VERSION_MAJOR}"
+                              )
 
 install(
   TARGETS clapack EXPORT clapack 
