--- netatalk-2.2.2/configure.ac.orig	2012-04-19 18:57:34.424065848 -0300
+++ netatalk-2.2.2/configure.ac	2012-04-19 18:58:43.709254880 -0300
@@ -674,6 +674,10 @@ fi
 #		AC_MSG_RESULT([enabling quotactl wrapper])
 #	)
 
+AC_ARG_WITH([xfs],
+       AS_HELP_STRING([--without-xfs], [Build without XFS filesystem quota support]))
+
+if test "x$with_xfs" != "xno"; then
         # For quotas on Linux XFS filesystems
         
         # For linux > 2.5.56
@@ -681,7 +685,7 @@ fi
 		[AC_CHECK_HEADERS(linux/xqm.h linux/xfs_fs.h)
         	AC_CHECK_HEADERS(xfs/libxfs.h xfs/xqm.h xfs/xfs_fs.h)]
 	)
-
+fi
 
 	dnl ----- as far as I can tell, dbtob always does the wrong thing
 	dnl ----- on every single version of linux I've ever played with.
