--- putty-20100504/unix/uxmisc.c.org	2010-05-19 15:39:00.927479235 +0800
+++ putty-20100504/unix/uxmisc.c	2010-05-19 15:39:32.027256092 +0800
@@ -144,7 +144,7 @@
 	assert(mode[0] == 'w');	       /* is_private is meaningless for read,
 					  and tricky for append */
 	fd = open(filename.path, O_WRONLY | O_CREAT | O_TRUNC,
-		      0700);
+		      0600);
 	if (fd < 0)
 	    return NULL;
 	return fdopen(fd, mode);
