Description: Build with -fgnu89-inline for GCC 5.
Author: Matthias Klose <doko@ubuntu.com>
Bug-Debian: https://bugs.debian.org/778001
Origin: http://launchpadlibrarian.net/199536689/memtest86%2B_4.20-1.1ubuntu8_4.20-1.1ubuntu9.diff.gz

--- memtest86+-5.01.orig/Makefile
+++ memtest86+-5.01/Makefile
@@ -13,7 +13,7 @@ AS=as -32
 CC=gcc
 
 CFLAGS= -Wall -march=i486 -m32 -O0 -fomit-frame-pointer -fno-builtin \
-	-ffreestanding -fPIC $(SMP_FL) -fno-stack-protector 
+	-ffreestanding -fPIC $(SMP_FL) -fno-stack-protector -fgnu89-inline
 CPPFLAGS=-m32
 ASFLAGS=-32
 	
