diff -up memtest86+-4.20/test.c.test memtest86+-4.20/test.c
--- memtest86+-4.20/test.c.test	2011-01-23 19:11:04.000000000 +0100
+++ memtest86+-4.20/test.c	2012-03-22 14:21:27.737524043 +0100
@@ -260,7 +260,7 @@ void movinvr()
 	int i, j, done, seed1, seed2;
 	volatile ulong *pe;
 	volatile ulong *start,*end;
-	ulong num;
+	ulong num, bad;
 	uintptr_t seg_start;
 
 	/* Initialize memory with initial sequence of random numbers.  */
@@ -296,12 +296,12 @@ void movinvr()
 				break;
 			}
 /* Original C code replaced with hand tuned assembly code */
-/*
+
 			for (; p < pe; p++) {
 				*p = rand();
 			}
- */
 
+/*
 			asm __volatile__ (
 				"jmp L200\n\t"
 				".p2align 4,,7\n\t"
@@ -315,7 +315,7 @@ void movinvr()
 				: "D" (p), "b" (pe)
 				: "eax", "edx"
 			);
-
+*/
 			do_tick();
 			BAILR
 		} while (!done);
