diff -up crack-attack-1.1.14/src/gtk-gui/callbacks.cxx.broken crack-attack-1.1.14/src/gtk-gui/callbacks.cxx
--- crack-attack-1.1.14/src/gtk-gui/callbacks.cxx.broken	2011-07-17 19:08:23.000000000 -0300
+++ crack-attack-1.1.14/src/gtk-gui/callbacks.cxx	2011-07-17 19:09:24.954161206 -0300
@@ -249,19 +249,19 @@ on_cmbQuality_changed                  (
   gchar *filename = NULL;
   switch (tmp) {
     case 0: {
-              filename = find_pixmap_file("preview_normal.tga");
+              filename = find_pixmap_file("preview_normal.png");
               break;
             }
     case 1: {
-              filename = find_pixmap_file("preview_reduced.tga");
+              filename = find_pixmap_file("preview_reduced.png");
               break;
             }
     case 2: {
-              filename = find_pixmap_file("preview_extremely_reduced.tga");
+              filename = find_pixmap_file("preview_extremely_reduced.png");
               break;
             }
     default: {
-              filename = find_pixmap_file("preview_normal.tga");
+              filename = find_pixmap_file("preview_normal.png");
              }
   }
   gtk_image_set_from_file(imgLogo, filename);
diff -up crack-attack-1.1.14/src/gtk-gui/interface.cxx.broken crack-attack-1.1.14/src/gtk-gui/interface.cxx
--- crack-attack-1.1.14/src/gtk-gui/interface.cxx.broken	2011-07-17 19:08:25.000000000 -0300
+++ crack-attack-1.1.14/src/gtk-gui/interface.cxx	2011-07-17 19:09:21.691860230 -0300
@@ -162,7 +162,7 @@ create_winCrackAttackSplash (void)
   gtk_widget_show (vbox4);
   gtk_container_add (GTK_CONTAINER (alignment1), vbox4);
 
-  imgLogo = create_pixmap (winCrackAttackSplash, "preview_normal.tga");
+  imgLogo = create_pixmap (winCrackAttackSplash, "preview_normal.png");
   gtk_widget_set_name (imgLogo, "imgLogo");
   gtk_widget_show (imgLogo);
   gtk_box_pack_start (GTK_BOX (vbox4), imgLogo, FALSE, FALSE, 0);
