diff -urN orig/crack-attack-1.1.8.1/src/DrawGarbage.cxx crack-attack-1.1.8.1/src/DrawGarbage.cxx
--- orig/crack-attack-1.1.8.1/src/DrawGarbage.cxx	2002-05-28 14:48:15.000000000 -0700
+++ crack-attack-1.1.8.1/src/DrawGarbage.cxx	2003-09-10 21:05:56.000000000 -0700
@@ -25,6 +25,8 @@
  * Handle's the displaying of the garbage.
  */
 
+#define GL_GLEXT_PROTOTYPES
+
 #include <GL/glut.h>
 
 #ifndef _WIN32
diff -urN orig/crack-attack-1.1.8.1/src/obj_garbage.cxx crack-attack-1.1.8.1/src/obj_garbage.cxx
--- orig/crack-attack-1.1.8.1/src/obj_garbage.cxx	2002-05-28 14:48:16.000000000 -0700
+++ crack-attack-1.1.8.1/src/obj_garbage.cxx	2003-09-10 21:06:25.000000000 -0700
@@ -25,6 +25,8 @@
  * Generates the lightmap and flavor textures for the garbage.
  */
 
+#define GL_GLEXT_PROTOTYPES
+
 #include <GL/glut.h>
 
 #ifndef _WIN32
