From f9d0d07970535d58d14b437930174163a6d5df1f Mon Sep 17 00:00:00 2001
From: Pino Toscano <pino@kde.org>
Date: Tue, 27 Mar 2012 13:02:50 +0200
Subject: [PATCH 01/17] set LIB_INSTALL_DIR even with no KDE4 stuff

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6edf2d7..28f256c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,7 @@ if(NOT KDE4_FOUND)
     cmake_policy(SET CMP0002 OLD)
     find_package(Qt4)
     set(DATA_INSTALL_DIR share)
+    set(LIB_INSTALL_DIR lib)
 endif(NOT KDE4_FOUND)
 
 # Use colored output (since cmake 2.4.0)
-- 
1.9.3

