From 5e74eb9335fbfaa84323e173d081afcfdbf39800 Mon Sep 17 00:00:00 2001
From: Milian Wolff <mail@milianw.de>
Date: Mon, 16 Apr 2012 15:01:58 +0200
Subject: [PATCH 05/17] force building of unit tests

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 121ad3d..0ed26eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,6 +47,7 @@ add_subdirectory(examples EXCLUDE_FROM_ALL)
 
 if(KDE4_FOUND)
     enable_testing()
+    set(KDE4_BUILD_TESTS "ON" CACHE "BOOL" "Enable building of tests" FORCE )
     add_subdirectory(tests)
 endif(KDE4_FOUND)
 
-- 
1.9.3

