From f32988dbaefcad34ab7966676058629e73c438fd Mon Sep 17 00:00:00 2001
From: Milian Wolff <mail@milianw.de>
Date: Sun, 4 Aug 2013 21:37:01 +0200
Subject: [PATCH 14/17] Fix compile with clang, also enable exceptions there.

---
 kdev-pg/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kdev-pg/CMakeLists.txt b/kdev-pg/CMakeLists.txt
index 7022121..516f5b3 100644
--- a/kdev-pg/CMakeLists.txt
+++ b/kdev-pg/CMakeLists.txt
@@ -4,9 +4,9 @@ PROJECT(kdevpg)
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 REMOVE_DEFINITIONS(-DQT_NO_STL)
 ADD_DEFINITIONS(-DQT_STL)
-if(CMAKE_COMPILER_IS_GNUCC)
-    add_definitions(-fexceptions)
-endif(CMAKE_COMPILER_IS_GNUCC)
+if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
+    add_definitions(-fexceptions -std=c++11)
+endif(CMAKE_COMPILER_IS_GNUCC OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
 
 QT4_ADD_RESOURCES(kdevpg_rcc_srcs ../unidata.qrc)
 
-- 
1.9.3

