--- libproxy-0.4.7/libproxy/cmake/modules/pacrunner_webkit.cmk.orig	2012-02-28 09:15:55.160816707 +0000
+++ libproxy-0.4.7/libproxy/cmake/modules/pacrunner_webkit.cmk	2012-02-28 09:16:25.314819274 +0000
@@ -13,7 +13,7 @@
   endif()
 else()
   if(WITH_WEBKIT3)
-    px_check_modules(WEBKIT webkitgtk-3.0)
+    px_check_modules(WEBKIT webkitgtk-3.0 javascriptcoregtk-3.0)
   else()
     px_check_modules(WEBKIT webkit-1.0)
   endif()
