--- libproxy-0.4.7/libproxy/url.cpp.orig	2012-02-26 19:04:10.670952501 +0000
+++ libproxy-0.4.7/libproxy/url.cpp	2012-02-26 19:04:54.474963813 +0000
@@ -446,7 +446,7 @@
 		/* Check for correct mime type and content length */
 		for (line = recvline(sock) ; line != "\r" && line != "" ; line = recvline(sock)) {
 			// Check for chunked encoding
-			if (line.find("Content-Transfer-Encoding: chunked") == 0)
+			if (line.find("Content-Transfer-Encoding: chunked") == 0 || line.find("Transfer-Encoding: chunked") == 0)
 				chunked = true;
 
 			// Check for content length
