--- src/mediaenc_ffmpeg.cpp.ffmpeg	2013-03-15 18:46:22.520049789 -0300
+++ src/mediaenc_ffmpeg.cpp	2013-03-15 18:50:24.550991447 -0300
@@ -146,7 +146,11 @@ bool wxFfmpegMediaEncoder::addVideoStrea
 
 	AVCodecContext* c = m_videoStm->codec;
 	c->thread_count = m_threadCount;
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55, 0, 100)
+	c->codec_id = (AVCodecID) codecId;
+#else
 	c->codec_id = (CodecID) codecId;
+#endif
 	c->codec_type = AVMEDIA_TYPE_VIDEO;
 	c->bit_rate = videoBitrate * 1000;
 	wxSize frameSize = GetFrameSize(videoFormat);
@@ -181,7 +185,11 @@ bool wxFfmpegMediaEncoder::addAudioStrea
 	c->thread_count = m_threadCount;
 	c->time_base.den = 25;
 	c->time_base.num = 1;
+#if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(55, 0, 100)
+	c->codec_id = (AVCodecID) codecId;
+#else
 	c->codec_id = (CodecID) codecId;
+#endif
 	c->codec_type = AVMEDIA_TYPE_AUDIO;
 	c->bit_rate = 64000;
 	c->sample_rate = 48000;
