--- subtitleeditor-svn/src/subtitleview.cc.orig	2014-12-31 18:27:41.556444986 +0100
+++ subtitleeditor-svn/src/subtitleview.cc	2014-12-31 18:31:23.079171738 +0100
@@ -38,12 +38,12 @@
 /*
  *
  */
-class TimeCell : public Gtk::TextView, public Gtk::CellEditable
+class TimeCell : public Gtk::CellEditable, public Gtk::TextView
 {
 public:
 
 	TimeCell()
-	:Glib::ObjectBase(typeid(TimeCell)), Gtk::CellEditable()
+	:Gtk::CellEditable(), Glib::ObjectBase(typeid(TimeCell))
 	{
 		se_debug(SE_DEBUG_VIEW);
 	}
