diff -up wicd-1.7.3/wicd/wicd-daemon.py.orig wicd-1.7.3/wicd/wicd-daemon.py
--- wicd-1.7.3/wicd/wicd-daemon.py.orig	2014-12-21 14:40:46.000000000 -0500
+++ wicd-1.7.3/wicd/wicd-daemon.py	2015-04-09 10:27:54.203884159 -0400
@@ -835,7 +835,7 @@ class WicdDaemon(dbus.service.Object, ob
         pass
 
     @dbus.service.signal(dbus_interface='org.wicd.daemon', signature='')
-    def DaemonClosing(self):
+    def DaemonClosing(self, arg1=None, arg2=None):
         """ Emits a signal indicating the daemon will be closing. """
         # By default, disconnect network links on close.
         if not self.keep_connection:
