diff -up wicd-1.7.3/in/man=wicd-curses.8.in.orig wicd-1.7.3/in/man=wicd-curses.8.in
--- wicd-1.7.3/in/man=wicd-curses.8.in.orig	2014-12-21 14:40:46.000000000 -0500
+++ wicd-1.7.3/in/man=wicd-curses.8.in	2015-04-09 10:24:30.923580121 -0400
@@ -60,10 +60,6 @@ Raise the Ad-Hoc network creation dialog
 .BR X
 Show the list of saved wireless networks, to be able to remove them
 
-.SH "FILES"
-.TP
-.I ~/.wicd/WHEREAREMYFILES
-Reminder that your network configuration files are not here ;-)
 .PP
 .SH "SEE ALSO"
 .BR wicd-client (1),
diff -up wicd-1.7.3/in/scripts=wicd-client.in.orig wicd-1.7.3/in/scripts=wicd-client.in
--- wicd-1.7.3/in/scripts=wicd-client.in.orig	2014-12-21 14:40:46.000000000 -0500
+++ wicd-1.7.3/in/scripts=wicd-client.in	2015-04-09 10:24:30.923580121 -0400
@@ -3,10 +3,6 @@
 if [ ! -d "$HOME/.wicd" ]; then
     mkdir -p "$HOME/.wicd"
 fi
-# Make sure the user knows WHEREAREMYFILES ;-)
-if [ -e "%VARLIB%WHEREAREMYFILES" ] && [ ! -L "$HOME/.wicd/WHEREAREMYFILES" ]; then
-    ln -s "%VARLIB%WHEREAREMYFILES" "$HOME/.wicd/WHEREAREMYFILES"
-fi
 if [ "$DISPLAY" = "" ]; then
     BOLD=$(tput bold)
     BLUE=$(tput setaf 4)
diff -up wicd-1.7.3/in/scripts=wicd-curses.in.orig wicd-1.7.3/in/scripts=wicd-curses.in
--- wicd-1.7.3/in/scripts=wicd-curses.in.orig	2014-12-21 14:40:46.000000000 -0500
+++ wicd-1.7.3/in/scripts=wicd-curses.in	2015-04-09 10:24:30.924580127 -0400
@@ -2,8 +2,5 @@
 if [ ! -d "$HOME/.wicd" ]; then
     mkdir -p "$HOME/.wicd"
 fi
-if [ -e "%VARLIB%WHEREAREMYFILES" ] && [ ! -L "$HOME/.wicd/WHEREAREMYFILES" ]; then
-    ln -s "%VARLIB%WHEREAREMYFILES" "$HOME/.wicd/WHEREAREMYFILES"
-fi
 
 exec %PYTHON% -O %SHARE%curses/wicd-curses.py $@
diff -up wicd-1.7.3/in/scripts=wicd-gtk.in.orig wicd-1.7.3/in/scripts=wicd-gtk.in
--- wicd-1.7.3/in/scripts=wicd-gtk.in.orig	2014-12-21 14:40:46.000000000 -0500
+++ wicd-1.7.3/in/scripts=wicd-gtk.in	2015-04-09 10:24:30.924580127 -0400
@@ -4,9 +4,5 @@
 if [ ! -d "$HOME/.wicd" ]; then
     mkdir -p "$HOME/.wicd"
 fi
-# Make sure the user knows WHEREAREMYFILES ;-)
-if [ -e "/var/lib/wicd/WHEREAREMYFILES" ] && [ ! -L "$HOME/.wicd/WHEREAREMYFILES" ]; then
-    ln -s "/var/lib/wicd/WHEREAREMYFILES" "$HOME/.wicd/WHEREAREMYFILES"
-fi
 
 exec %PYTHON% -O %SHARE%gtk/wicd-client.py $@
