diff -up wicd-1.7.3/curses/netentry_curses.py.orig wicd-1.7.3/curses/netentry_curses.py
--- wicd-1.7.3/curses/netentry_curses.py.orig	2014-12-21 14:40:46.000000000 -0500
+++ wicd-1.7.3/curses/netentry_curses.py	2015-04-09 13:18:04.624248481 -0400
@@ -340,8 +340,7 @@ class WiredSettingsDialog(AdvancedSettin
         self.set_values()
 
         self.prof_name = name
-        title = _('Configuring preferences for wired profile "$A"'). \
-            replace('$A', self.prof_name)
+        title = ' '.join([_('Configuring preferences for wired profile'), self.prof_name])
         self._w.header = urwid.Text(('header', title), align='right')
 
         self.set_values()
@@ -502,9 +501,9 @@ class WirelessSettingsDialog(AdvancedSet
         self.encrypt_types = misc.LoadEncryptionMethods()
         self.set_values()
 
-        title = _('Configuring preferences for wireless network "$A" ($B)'). \
-            replace('$A', wireless.GetWirelessProperty(networkID, 'essid')). \
-            replace('$B', wireless.GetWirelessProperty(networkID, 'bssid'))
+        title = ' '.join([_('Configuring preferences for wireless network'),
+                          wireless.GetWirelessProperty(networkID,'essid'),
+                          "(%s)" % wireless.GetWirelessProperty(networkID,'bssid')])
         self._w.header = urwid.Text(('header', title), align='right')
 
     def set_values(self):
