--- core/libs/database/imagehistory/imagehistorygraph_boost.h.orig	2011-12-08 18:04:22.711327707 -0200
+++ core/libs/database/imagehistory/imagehistorygraph_boost.h	2011-12-08 18:04:25.006874378 -0200
@@ -1393,6 +1393,7 @@ protected:
             {
                 return vertexLessThan(boost::target(a.toEdge(), g), boost::target(b.toEdge(), g));
             }
+            return vertexLessThan(boost::target(a.toEdge(), g), boost::target(b.toEdge(), g));
         };
 
         // This is boost's simple, old, recursive DFS algorithm adapted with lessThan
