diff -up digikam-3.5.0/extra/kipi-plugins/photolayoutseditor/extra/qtpropertybrowser/CMakeLists.txt.no_libPropertyBrowser digikam-3.5.0/extra/kipi-plugins/photolayoutseditor/extra/qtpropertybrowser/CMakeLists.txt
--- digikam-3.5.0/extra/kipi-plugins/photolayoutseditor/extra/qtpropertybrowser/CMakeLists.txt.no_libPropertyBrowser	2013-10-01 01:23:11.000000000 -0500
+++ digikam-3.5.0/extra/kipi-plugins/photolayoutseditor/extra/qtpropertybrowser/CMakeLists.txt	2013-10-10 08:13:18.005001339 -0500
@@ -119,4 +119,5 @@ ADD_DEFINITIONS(${QT_DEFINITIONS})
 ADD_DEFINITIONS(-DQT_DLL)
 ADD_DEFINITIONS(-DQT_PLUGIN)
 
-INSTALL( TARGETS PropertyBrowser ARCHIVE DESTINATION lib ${INSTALL_TARGETS_DEFAULT_ARGS} )
+# Do not install (yet), see https://bugs.kde.org/show_bug.cgi?id=319664
+# INSTALL( TARGETS PropertyBrowser ARCHIVE DESTINATION lib ${INSTALL_TARGETS_DEFAULT_ARGS} )
