diff -up ./configure.in.missinglib ./configure.in
--- ./configure.in.missinglib	2011-10-12 05:37:01.783508981 +0200
+++ ./configure.in	2011-10-12 05:37:13.883580016 +0200
@@ -57,7 +57,7 @@ dnl ====================================
 #PKG_CHECK_MODULES(GRISBI, [gtk+-2.0 >= 2.2.0 glib-2.0 >= 2.2 gmodule-2.0 >= 2.2 zlib])
 # Temporary hack until Mandriva ships zlib.pc
 PKG_CHECK_MODULES(GRISBI, [gtk+-2.0 >= 2.12.0 glib-2.0 >= 2.18.0 gmodule-2.0 >= 2.18.0])
-GRISBI_LIBS="$GRISBI_LIBS -lz"
+GRISBI_LIBS="$GRISBI_LIBS -lz -lm"
 AC_SUBST(GRISBI_LIBS)
 # </hack>
 
diff -up ./src/Makefile.missinglib ./src/Makefile
