diff -up freenx-server-0.7.3/init.d/freenx-server~ freenx-server-0.7.3/init.d/freenx-server
--- freenx-server-0.7.3/init.d/freenx-server~	2011-08-17 01:25:51.872295221 +0300
+++ freenx-server-0.7.3/init.d/freenx-server	2011-08-17 11:00:57.996250732 +0300
@@ -10,9 +10,8 @@
 #
 # Modified to be chkconfig compatible by Johnny Hughes <johnny@centos.org>
 #
-# chkconfig: 2345 91 35
-# description: Creates /tmp/.X11-unix/ if required and cleans up dead \
-#              NX sessions.
+# chkconfig: - 91 35
+# description: FreeNX Server
 
 
 # Read the config file
@@ -22,8 +21,16 @@
 . /etc/init.d/functions
 prog="freenx-server"
 
+nxstatus() {
+	if [ -e "/var/lock/subsys/freenx-server" ]; then
+	  echo $"$prog is running"
+	  return 0
+	fi
+	echo $"$prog is stopped"
+	return 3
+}
+
 start() {
-	runuser -s /bin/bash - nx -c "$PATH_BIN/nxserver --cleanup" > /dev/null 2>&1 < /dev/null
 	if [ ! -d "/tmp/.X11-unix" ]; then
 	  mkdir -m1777 /tmp/.X11-unix/
 	  [ ! -x /sbin/restorecon ] || /sbin/restorecon /tmp/.X11-unix
@@ -36,6 +43,8 @@ start() {
 	  ret=0
 	fi
 	if [ $ret -eq 0 ]; then
+	  nxstatus > /dev/null || $PATH_BIN/nxserver --cleanup > /dev/null 2>&1
+	  $PATH_BIN/nxserver --start > /dev/null 2>&1
 	  touch /var/lock/subsys/freenx-server
 	  action $"Starting $prog: " /bin/true
 	else
@@ -46,7 +55,8 @@ start() {
 }
 
 stop() {
-	runuser -s /bin/bash - nx -c "$PATH_BIN/nxserver --cleanup" > /dev/null 2>&1 < /dev/null
+	$PATH_BIN/nxserver --stop > /dev/null 2>&1
+	$PATH_BIN/nxserver --cleanup > /dev/null 2>&1
 	if [ -e "/var/lock/subsys/freenx-server" ]; then
 	  rm -f /var/lock/subsys/freenx-server
 	  ret=$?
@@ -79,11 +89,7 @@ case "$1" in
         restart
         ;;
   status)
-        if [ -e "/var/lock/subsys/freenx-server" ]; then
-          echo $"$prog is running"
-        else
-          echo $"$prog is stopped"
-        fi
+        nxstatus
         ;;
   *)
         echo $"Usage: $prog {start|stop|restart|status}"
