diff -up freenx-server-0.7.3/node.conf.sample~ freenx-server-0.7.3/node.conf.sample
--- freenx-server-0.7.3/node.conf.sample~	2013-01-04 20:21:54.802989341 +0200
+++ freenx-server-0.7.3/node.conf.sample	2013-01-04 20:26:24.468442246 +0200
@@ -548,8 +548,8 @@
 # 'umount.cifs'.
 #COMMAND_SMBUMOUNT_CIFS=/sbin/umount.cifs
 
-# The key that contains the name of the complete path of the 'netcat' command.
-#COMMAND_NETCAT=netcat
+# The key that contains the name of the complete path of the 'ncat' command.
+#COMMAND_NETCAT=ncat
 
 # The key that contains the name of the complete path of the 'ssh' and
 # 'ssh-keygen' command.
diff -up freenx-server-0.7.3/nxloadconfig~ freenx-server-0.7.3/nxloadconfig
--- freenx-server-0.7.3/nxloadconfig~	2013-01-04 20:21:54.800989309 +0200
+++ freenx-server-0.7.3/nxloadconfig	2013-01-04 20:25:53.020925393 +0200
@@ -231,7 +231,7 @@ COMMAND_SMBMOUNT=smbmount
 COMMAND_SMBUMOUNT=smbumount
 COMMAND_SMBMOUNT_CIFS=/sbin/mount.cifs
 COMMAND_SMBUMOUNT_CIFS=/sbin/umount.cifs
-COMMAND_NETCAT=netcat
+COMMAND_NETCAT=ncat
 COMMAND_SSH=ssh
 COMMAND_SSH_KEYGEN=ssh-keygen
 COMMAND_CUPSD=/usr/sbin/cupsd
diff -up freenx-server-0.7.3/nxserver~ freenx-server-0.7.3/nxserver
--- freenx-server-0.7.3/nxserver~	2013-01-04 20:21:54.798989277 +0200
+++ freenx-server-0.7.3/nxserver	2013-01-04 20:23:57.377974918 +0200
@@ -1483,7 +1483,7 @@ server_startrestore_session()
 
 			# Check if there is already an agent running on that display on that host
 			let AGENT_DISPLAY=$SESS_DISPLAY-$DISPLAY_BASE+6000
-			if $COMMAND_NETCAT -z "$SERVER_HOST" $AGENT_DISPLAY 2>/dev/null
+			if $COMMAND_NETCAT --send-only "$SERVER_HOST" $AGENT_DISPLAY </dev/null 2>/dev/null
 			then
 				log 2 "Warning: Stray nxagent without .nX$SESS_DISPLAY-lock found on host:port $SERVER_HOST:$AGENT_DISPLAY."
 				let SESS_DISPLAY=$SESS_DISPLAY+1
@@ -1491,7 +1491,7 @@ server_startrestore_session()
 			fi
 
                         let PROXY_DISPLAY=$SESS_DISPLAY+4000
-                        if $COMMAND_NETCAT -z "$SERVER_HOST" $PROXY_DISPLAY
+                        if $COMMAND_NETCAT --send-only "$SERVER_HOST" $PROXY_DISPLAY </dev/null 2>/dev/null
                         then
                                 log 2 "Warning: nxagent proxy without .nX$SESS_DISPLAY-lock found on host:port $SERVER_HOST:$AGENT_DISPLAY."
                                 let SESS_DISPLAY=$SESS_DISPLAY+1
@@ -1501,7 +1501,7 @@ server_startrestore_session()
 			# Now check for the other enabled services
 			
 			let SAMBA_DISPLAY=$SESS_DISPLAY+3000
-			if [ "$(getparam 'samba')" = 1 ] && $COMMAND_NETCAT -z "$SERVER_HOST" $SAMBA_DISPLAY
+			if [ "$(getparam 'samba')" = 1 ] && $COMMAND_NETCAT --send-only "$SERVER_HOST" $SAMBA_DISPLAY </dev/null 2>/dev/null
 			then
 				log 2 "Warning: Skipping $SERVER_HOST:$AGENT_DISPLAY as samba port is not free."
 				let SESS_DISPLAY=$SESS_DISPLAY+1
@@ -1509,7 +1509,7 @@ server_startrestore_session()
 			fi
 			
 			let MEDIA_DISPLAY=$SESS_DISPLAY+7000
-			if [ "$(getparam 'media')" = 1 ] && $COMMAND_NETCAT -z "$SERVER_HOST" $MEDIA_DISPLAY
+			if [ "$(getparam 'media')" = 1 ] && $COMMAND_NETCAT --send-only "$SERVER_HOST" $MEDIA_DISPLAY </dev/null 2>/dev/null
 			then
 				log 2 "Warning: Skipping $SERVER_HOST:$AGENT_DISPLAY as media port is not free."
 				let SESS_DISPLAY=$SESS_DISPLAY+1
@@ -1518,7 +1518,7 @@ server_startrestore_session()
 
 			
 			let CUPS_DISPLAY=$SESS_DISPLAY+9000
-			if [ "$(getparam 'cups')" = 1 ] && $COMMAND_NETCAT -z "$SERVER_HOST" $CUPS_DISPLAY
+			if [ "$(getparam 'cups')" = 1 ] && $COMMAND_NETCAT --send-only "$SERVER_HOST" $CUPS_DISPLAY </dev/null 2>/dev/null
 			then
 				log 2 "Warning: Skipping $SERVER_HOST:$AGENT_DISPLAY as cups port is not free."
 				let SESS_DISPLAY=$SESS_DISPLAY+1
