diff -up freenx-server-0.7.3/nxredir/Makefile~ freenx-server-0.7.3/nxredir/Makefile
--- freenx-server-0.7.3/nxredir/Makefile~	2011-12-31 11:39:17.288142560 +0200
+++ freenx-server-0.7.3/nxredir/Makefile	2011-12-31 11:39:52.273330858 +0200
@@ -1,12 +1,12 @@
 all: libnxredir.so
 
 CC=gcc
-CFLAGS=-g -O2 -Wall -fPIC
+CFLAGS+=-fPIC
 LIBNAME=libnxredir.so
 VERSION=0
 
 libnxredir.so: nxredir.o
-	$(CC) -fPIC $(CFLAGS) -nostdlib -shared -Wl,-soname,$(LIBNAME).$(VERSION) -o $(LIBNAME).$(VERSION) nxredir.o -ldl -lc
+	$(CC) -fPIC $(LDFLAGS) -nostdlib -shared -Wl,-soname,$(LIBNAME).$(VERSION) -o $(LIBNAME).$(VERSION) nxredir.o -ldl -lc
 
 clean:
 	rm -f $(LIBNAME).$(VERSION)
diff -up freenx-server-0.7.3/nx-session-launcher/Makefile~ freenx-server-0.7.3/nx-session-launcher/Makefile
--- freenx-server-0.7.3/nx-session-launcher/Makefile~	2008-08-05 20:54:32.000000000 +0300
+++ freenx-server-0.7.3/nx-session-launcher/Makefile	2011-12-31 11:39:23.728729509 +0200
@@ -1,7 +1,7 @@
 .PHONY: all install
 
 CC=gcc
-CFLAGS=-g -O2 -Wall -fPIC
+CFLAGS+=-fPIC
 
 SOURCES = nx-session-launcher-suid.c
 PROGRAMS = nx-session-launcher-suid
