diff -up freenx-server-0.7.3/nxkeygen~ freenx-server-0.7.3/nxkeygen
--- freenx-server-0.7.3/nxkeygen~	2012-06-14 21:36:03.271909685 +0300
+++ freenx-server-0.7.3/nxkeygen	2012-06-14 21:37:46.177685991 +0300
@@ -51,7 +51,7 @@ main ()
 		chown nx:root $x
 	done
 	
-	# copy the key to the authorized_keys2 file
+	# copy the key to the authorized_keys file
 	rm -f $NX_KEY_DIR/$SSH_AUTHORIZED_KEYS
 	echo -n "no-port-forwarding,no-agent-forwarding,command=\"$PATH_BIN/nxserver\" " >$NX_KEY_DIR/$SSH_AUTHORIZED_KEYS
 	cat ${NX_SERVER_KEY} >> $NX_KEY_DIR/$SSH_AUTHORIZED_KEYS
diff -up freenx-server-0.7.3/nxloadconfig~ freenx-server-0.7.3/nxloadconfig
--- freenx-server-0.7.3/nxloadconfig~	2012-06-14 21:36:03.307909232 +0300
+++ freenx-server-0.7.3/nxloadconfig	2012-06-14 21:37:50.437634443 +0300
@@ -70,7 +70,7 @@ APPLICATION_LIBRARY_PATH="" #Calculated
 APPLICATION_LIBRARY_PRELOAD="" #Calculated
 
 # the name of the authorized keys file for ssh
-SSH_AUTHORIZED_KEYS="authorized_keys2"
+SSH_AUTHORIZED_KEYS="authorized_keys"
 
 # retest values like xauth, netcat and checks for nxdesktop/nxviewer availability
 # set to 0 if you are sure, you set the right values
