diff -up freenx-server-0.7.3/nxdialog~ freenx-server-0.7.3/nxdialog
--- freenx-server-0.7.3/nxdialog~	2011-06-20 00:02:55.175387849 +0300
+++ freenx-server-0.7.3/nxdialog	2011-06-20 00:20:11.551917170 +0300
@@ -88,7 +88,7 @@ if [ -x /usr/bin/Xdialog ] 
 then
 	dialog_interface="xdialog"
 	DIALOG=/usr/bin/Xdialog # just in case that we have no good path
-elif [ -x /usr/bin/dialog ]
+elif [ -x /usr/bin/dialog ] && which xterm &>/dev/null
 then
 	#JJK: Added 'dialog_interface=dialog' option because Xdialog not standard
 	#JJK: on some distros such as Fedora and xmessage won't handle long
