--- init.d/freenx-server~	2009-11-23 07:06:13.000000000 -0500
+++ init.d/freenx-server	2010-01-14 11:29:35.000000000 -0500
@@ -26,6 +26,7 @@
 	runuser -s /bin/bash - nx -c "$PATH_BIN/nxserver --cleanup" > /dev/null 2>&1 < /dev/null
 	if [ ! -d "/tmp/.X11-unix" ]; then
 	  mkdir -m1777 /tmp/.X11-unix/
+	  [ ! -x /sbin/restorecon ] || /sbin/restorecon /tmp/.X11-unix
 	  ret=$?
 	else
 	  X11_owner=`/bin/ls -ald /tmp/.X11-unix | /bin/gawk {'print $3'}`
