diff -up claws-mail-3.11.1/src/prefs_account.c.orig claws-mail-3.11.1/src/prefs_account.c
--- claws-mail-3.11.1/src/prefs_account.c.orig	2015-01-15 17:19:26.109501284 +0100
+++ claws-mail-3.11.1/src/prefs_account.c	2015-01-15 17:19:48.201429854 +0100
@@ -3847,6 +3847,7 @@ static void auto_configure_cb (GtkWidget
 		g_free(address);
 		gtk_label_set_text(GTK_LABEL(basic_page.auto_configure_lbl),
 			   _("Failed (wrong address)"));
+		return;
 	}
 	domain = strchr(address, '@') + 1;
 
diff -up claws-mail-3.11.1/src/wizard.c.orig claws-mail-3.11.1/src/wizard.c
--- claws-mail-3.11.1/src/wizard.c.orig	2015-01-15 17:19:57.325400354 +0100
+++ claws-mail-3.11.1/src/wizard.c	2015-01-15 17:20:39.580263731 +0100
@@ -1372,6 +1372,7 @@ static void auto_configure_cb (GtkWidget
 		g_free(address);
 		gtk_label_set_text(GTK_LABEL(wizard->auto_configure_lbl),
 			   _("Failed (wrong address)"));
+		return;
 	}
 	domain = strchr(address, '@') + 1;
 
