diff -up claws-mail-3.11.1/configure.ac.orig claws-mail-3.11.1/configure.ac
--- claws-mail-3.11.1/configure.ac.orig	2014-10-27 19:58:19.000000000 +0100
+++ claws-mail-3.11.1/configure.ac	2015-01-16 10:40:37.368721862 +0100
@@ -1824,6 +1824,20 @@ fi
 
 AC_MSG_CHECKING([whether to build vcalendar plugin])
 if test x"$enable_vcalendar_plugin" != xno; then
+
+        dnl Check for libical
+	PKG_CHECK_MODULES(LIBICAL, libical,
+        [
+		echo "Building with external libical"
+        ],
+        [
+                echo "Building without gnutls"
+		enable_vcalendar_plugin=no
+        ])
+        AC_SUBST(LIBICAL_LIBS)
+        AC_SUBST(LIBICAL_CFLAGS)
+
+
 	dependencies_missing=""
 
 	if test x"$HAVE_CURL" = xno; then
@@ -1934,11 +1948,6 @@ src/plugins/spamassassin/Makefile
 src/plugins/spam_report/Makefile
 src/plugins/tnef_parse/Makefile
 src/plugins/vcalendar/Makefile
-src/plugins/vcalendar/libical/Makefile
-src/plugins/vcalendar/libical/libical/icalversion.h
-src/plugins/vcalendar/libical/libical/Makefile
-src/plugins/vcalendar/libical/design-data/Makefile
-src/plugins/vcalendar/libical/scripts/Makefile
 doc/Makefile
 doc/man/Makefile
 tools/Makefile
diff -up claws-mail-3.11.1/Makefile.in.orig claws-mail-3.11.1/Makefile.in
diff -up claws-mail-3.11.1/src/plugins/vcalendar/Makefile.am.orig claws-mail-3.11.1/src/plugins/vcalendar/Makefile.am
--- claws-mail-3.11.1/src/plugins/vcalendar/Makefile.am.orig	2014-10-27 19:58:16.000000000 +0100
+++ claws-mail-3.11.1/src/plugins/vcalendar/Makefile.am	2015-01-16 10:40:37.369721859 +0100
@@ -3,15 +3,13 @@
 # terms of the General Public License version 3 (or later).
 # See COPYING file for license details.
 
-SUBDIRS = libical
 EXTRA_DIST = claws.def plugin.def version.rc
 
 IFLAGS = \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/common \
 	-I$(top_builddir)/src/common \
-	-I$(top_srcdir)/src/gtk \
-	-I$(top_srcdir)/src/plugins/vcalendar/libical/libical
+	-I$(top_srcdir)/src/gtk
 
 if OS_WIN32
 
@@ -86,12 +84,13 @@ vcalendar_la_DEPENDENCIES = $(plugin_dep
 vcalendar_la_LIBADD = $(plugin_ldadd) $(cygwin_export_lib) \
 	$(GTK_LIBS) \
 	$(CURL_LIBS) \
-        $(top_builddir)/src/plugins/vcalendar/libical/libical/libical.la
+	$(LIBICAL_LIBS)
 
 vcalendar_la_CPPFLAGS = \
 	$(IFLAGS) \
 	$(GLIB_CFLAGS) \
 	$(GTK_CFLAGS) \
+	$(LIBICAL_CFLAGS) \
 	$(CURL_CFLAGS)
 
 clean-local:
diff -up claws-mail-3.11.1/src/plugins/vcalendar/vcal_manager.c.orig claws-mail-3.11.1/src/plugins/vcalendar/vcal_manager.c
--- claws-mail-3.11.1/src/plugins/vcalendar/vcal_manager.c.orig	2015-01-16 10:46:27.844609299 +0100
+++ claws-mail-3.11.1/src/plugins/vcalendar/vcal_manager.c	2015-01-16 10:55:33.788875499 +0100
@@ -408,10 +408,14 @@ gchar *vcal_manager_event_dump(VCalEvent
 		icalproperty_new_summary(event->summary));
 	icalcomponent_add_property(ievent,
 		icalproperty_new_sequence(modif && !is_reply ? event->sequence + 1 : event->sequence));
+	enum icalproperty_class ical_prop_class = ICAL_CLASS_PUBLIC;
 	icalcomponent_add_property(ievent,
-		icalproperty_new_class("PUBLIC"));
+		icalproperty_new_class(ical_prop_class));
+
+	enum icalproperty_transp ical_prop_transp = ICAL_TRANSP_OPAQUE;
 	icalcomponent_add_property(ievent,
-		icalproperty_new_transp("OPAQUE"));
+		icalproperty_new_transp(ical_prop_transp));
+
 	if (event->location && *event->location)
 		icalcomponent_add_property(ievent,
 			icalproperty_new_location(event->location));
