diff -up claws-mail-3.7.9/src/mainwindow.c.orig claws-mail-3.7.9/src/mainwindow.c
--- claws-mail-3.7.9/src/mainwindow.c.orig	2011-04-11 14:31:00.000000000 +0200
+++ claws-mail-3.7.9/src/mainwindow.c	2011-04-11 14:31:27.000000000 +0200
@@ -3573,6 +3573,7 @@ void main_window_popup(MainWindow *mainw
 	if (first_start) {
 #ifdef G_OS_UNIX
 		gtk_window_deiconify(GTK_WINDOW(mainwin->window));
+		gtkut_window_popup(mainwin->window);
 #endif
 		first_start = FALSE;
 	} else {
