diff -up claws-mail-3.11.1/src/common/ssl.c.orig claws-mail-3.11.1/src/common/ssl.c
--- claws-mail-3.11.1/src/common/ssl.c.orig	2015-01-16 11:39:17.764332632 +0100
+++ claws-mail-3.11.1/src/common/ssl.c	2015-01-16 11:40:07.480171586 +0100
@@ -323,7 +323,7 @@ gboolean ssl_init_socket(SockInfo *socki
 			    sockinfo->gnutls_priority, r);
 	}
 	else {
-		gnutls_priority_set_direct(session, "NORMAL:-VERS-SSL3.0", NULL);
+		gnutls_set_default_priority(session);
 	}
 	gnutls_record_disable_padding(session);
 
