From 8cd3d8443dfd5ab9cfa0880ac76d3e78de7a0dd4 Mon Sep 17 00:00:00 2001
From: Paul <paul@claws-mail.org>
Date: Mon, 10 Jun 2013 14:54:15 +0100
Subject: [PATCH] fix double-free crasher in Edit Accounts dialogue

---
 src/account.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/src/account.c b/src/account.c
index da4c807..110cf5d 100644
--- a/src/account.c
+++ b/src/account.c
@@ -661,7 +661,6 @@ void account_destroy(PrefsAccount *ac_prefs)
 
 	folder_unref_account_all(ac_prefs);
 
-	prefs_account_free(ac_prefs);
 	account_list = g_list_remove(account_list, ac_prefs);
 
 	if (cur_account == ac_prefs) cur_account = NULL;
-- 
1.7.0.4

