diff -ru glm/glm/gtc/ulp.inl glm-ulp/glm/gtc/ulp.inl
--- glm/glm/gtc/ulp.inl	2014-12-09 21:13:54.000000000 +0200
+++ glm-ulp/glm/gtc/ulp.inl	2015-01-06 16:29:47.497932633 +0200
@@ -316,7 +316,7 @@
 		if(x < y)
 		{
 			T temp = x;
-			while(temp != y)// && ulp < std::numeric_limits<std::size_t>::max())
+			while(temp < y)// && ulp < std::numeric_limits<std::size_t>::max())
 			{
 				++ulp;
 				temp = next_float(temp);
@@ -325,7 +325,7 @@
 		else if(y < x)
 		{
 			T temp = y;
-			while(temp != x)// && ulp < std::numeric_limits<std::size_t>::max())
+			while(temp < x)// && ulp < std::numeric_limits<std::size_t>::max())
 			{
 				++ulp;
 				temp = next_float(temp);
