diff -ru glm/test/core/core_func_matrix.cpp glm_count/test/core/core_func_matrix.cpp
--- glm/test/core/core_func_matrix.cpp	2014-02-08 21:28:22.000000000 +0200
+++ glm_count/test/core/core_func_matrix.cpp	2014-06-10 19:50:43.355854306 +0300
@@ -186,7 +186,7 @@
 	return Failed;
 }
 
-std::size_t const Count(10000000);
+std::size_t const Count(1000000);
 
 template <typename VEC3, typename MAT4>
 int test_inverse_perf(std::size_t Instance, char const * Message)
