diff -ru glm/CMakeLists.txt glm-nom64/CMakeLists.txt
--- glm/CMakeLists.txt	2015-02-14 00:51:30.000000000 +0200
+++ glm-nom64/CMakeLists.txt	2015-04-25 10:14:57.585689646 +0300
@@ -129,7 +129,7 @@
 if(CMAKE_COMPILER_IS_GNUCXX)
 	#add_definitions(-S)
 	#add_definitions(-s)
-	add_definitions(-m64)
+	#add_definitions(-m64)
 	add_definitions(-O2)
 
 	#add_definitions(-fprofile-arcs -ftest-coverage) gcov
