diff -ru sqlite-3.6.22.orig/src/os_unix.c sqlite-3.6.22/src/os_unix.c
--- sqlite-3.6.22.orig/src/os_unix.c	2010-01-15 09:00:06 +0100
+++ sqlite-3.6.22/src/os_unix.c	2010-01-15 09:01:42 +0100
@@ -436,7 +436,9 @@
   case ENODEV:
   case ENXIO:
   case ENOENT:
+#ifdef ESTALE
   case ESTALE:
+#endif
   case ENOSYS:
     /* these should force the client to close the file and reconnect */
     
@@ -1862,7 +1864,7 @@
     pFile->locktype = locktype;
 #if !OS_VXWORKS
     /* Always update the timestamp on the old file */
-    utimes(zLockFile, NULL);
+    utime(zLockFile, NULL);
 #endif
     return SQLITE_OK;
   }
