diff -up sqlite-src-3081002/configure.ac.malloc_usable_size sqlite-src-3081002/configure.ac
--- sqlite-src-3081002/configure.ac.malloc_usable_size	2015-05-29 10:11:21.154578129 +0200
+++ sqlite-src-3081002/configure.ac	2015-05-29 10:11:35.405646670 +0200
@@ -109,7 +109,7 @@ AC_CHECK_HEADERS([sys/types.h stdlib.h s
 #########
 # Figure out whether or not we have these functions
 #
-AC_CHECK_FUNCS([fdatasync gmtime_r isnan localtime_r localtime_s malloc_usable_size strchrnul usleep utime])
+AC_CHECK_FUNCS([fdatasync gmtime_r isnan localtime_r localtime_s strchrnul usleep utime])
 
 #########
 # By default, we use the amalgamation (this may be changed below...)
diff -up sqlite-src-3081002/configure.malloc_usable_size sqlite-src-3081002/configure
--- sqlite-src-3081002/configure.malloc_usable_size	2015-05-29 10:11:55.285742284 +0200
+++ sqlite-src-3081002/configure	2015-05-29 10:12:11.741821430 +0200
@@ -10303,7 +10303,7 @@ done
 #########
 # Figure out whether or not we have these functions
 #
-for ac_func in fdatasync gmtime_r isnan localtime_r localtime_s malloc_usable_size strchrnul usleep utime
+for ac_func in fdatasync gmtime_r isnan localtime_r localtime_s strchrnul usleep utime
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 ac_fn_c_check_func "$LINENO" "$ac_func" "$as_ac_var"
