diff -up sqlite-src-3071500/configure.ac.malloc-usable-size sqlite-src-3071500/configure.ac
--- sqlite-src-3071500/configure.ac.malloc-usable-size	2012-12-13 14:00:32.685869844 +0200
+++ sqlite-src-3071500/configure.ac	2012-12-13 14:00:43.321830496 +0200
@@ -127,7 +127,7 @@ AC_CHECK_HEADERS([sys/types.h stdlib.h s
 #########
 # Figure out whether or not we have these functions
 #
-AC_CHECK_FUNCS([usleep fdatasync localtime_r gmtime_r localtime_s utime malloc_usable_size])
+AC_CHECK_FUNCS([usleep fdatasync localtime_r gmtime_r localtime_s utime])
 
 #########
 # By default, we use the amalgamation (this may be changed below...)
diff -up sqlite-src-3071500/configure.malloc-usable-size sqlite-src-3071500/configure
--- sqlite-src-3071500/configure.malloc-usable-size	2012-12-13 14:00:28.552884276 +0200
+++ sqlite-src-3071500/configure	2012-12-13 14:00:54.720788024 +0200
@@ -12147,7 +12147,7 @@ done
 
 
 
-for ac_func in usleep fdatasync localtime_r gmtime_r localtime_s utime malloc_usable_size
+for ac_func in usleep fdatasync localtime_r gmtime_r localtime_s utime
 do
 as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 { $as_echo "$as_me:$LINENO: checking for $ac_func" >&5
