diff -up sqlite-src-3080800/test/oserror.test.stupid-openfiles-test sqlite-src-3080800/test/oserror.test
--- sqlite-src-3080800/test/oserror.test.stupid-openfiles-test	2015-01-20 09:23:35.085785827 +0100
+++ sqlite-src-3080800/test/oserror.test	2015-01-20 09:24:12.446938604 +0100
@@ -51,20 +51,20 @@ proc do_re_test {tn script expression} {
 # a call to getcwd() may fail if there are no free file descriptors. So
 # an error may be reported for either open() or getcwd() here.
 #
-if {![clang_sanitize_address]} {
-  do_test 1.1.1 {
-    set ::log [list]
-    list [catch {
-      for {set i 0} {$i < 2000} {incr i} { sqlite3 dbh_$i test.db -readonly 1 }
-    } msg] $msg
-  } {1 {unable to open database file}}
-  do_test 1.1.2 {
-    catch { for {set i 0} {$i < 2000} {incr i} { dbh_$i close } }
-  } {1}
-  do_re_test 1.1.3 { 
-    lindex $::log 0 
-  } {^os_unix.c:\d+: \(\d+\) (open|getcwd)\(.*test.db\) - }
-}
+#if {![clang_sanitize_address]} {
+#  do_test 1.1.1 {
+#    set ::log [list]
+#    list [catch {
+#      for {set i 0} {$i < 2000} {incr i} { sqlite3 dbh_$i test.db -readonly 1 }
+#    } msg] $msg
+#  } {1 {unable to open database file}}
+#  do_test 1.1.2 {
+#    catch { for {set i 0} {$i < 2000} {incr i} { dbh_$i close } }
+#  } {1}
+#  do_re_test 1.1.3 { 
+#    lindex $::log 0 
+#  } {^os_unix.c:\d+: \(\d+\) (open|getcwd)\(.*test.db\) - }
+#}
 
 
 # Test a failure in open() due to the path being a directory.
