diff -up sqlite-src-3081001/Makefile.in.old sqlite-src-3081001/Makefile.in
--- sqlite-src-3081001/Makefile.in.old	2015-05-11 09:24:56.310577370 +0200
+++ sqlite-src-3081001/Makefile.in	2015-05-11 09:25:42.618876370 +0200
@@ -995,7 +995,9 @@ smoketest:	testfixture$(TEXE) fuzzershel
 
 sqlite3_analyzer.c: sqlite3.c $(TOP)/src/tclsqlite.c $(TOP)/tool/spaceanal.tcl
 	echo "#define TCLSH 2" > $@
-	echo "#define SQLITE_ENABLE_DBSTAT_VTAB 1" >> $@
+	echo "#ifndef SQLITE_ENABLE_DBSTAT_VTAB" >> $@
+	echo "#define SQLITE_ENABLE_DBSTAT_VTAB 0" >> $@
+	echo "#endif" >> $@
 	cat sqlite3.c $(TOP)/src/tclsqlite.c >> $@
 	echo "static const char *tclsh_main_loop(void){" >> $@
 	echo "static const char *zMainloop = " >> $@
