diff -up sqlite-src-3080800/configure.ac.no-malloc-usable-size sqlite-src-3080800/configure.ac
--- sqlite-src-3080800/configure.ac.no-malloc-usable-size	2015-01-20 09:25:03.791148561 +0100
+++ sqlite-src-3080800/configure.ac	2015-01-20 09:27:36.215771856 +0100
@@ -109,7 +109,7 @@ AC_CHECK_HEADERS([sys/types.h stdlib.h s
 #########
 # Figure out whether or not we have these functions
 #
-AC_CHECK_FUNCS([fdatasync gmtime_r isnan localtime_r localtime_s malloc_usable_size strchrnul usleep utime])
+AC_CHECK_FUNCS([fdatasync gmtime_r isnan localtime_r localtime_s strchrnul usleep utime])
 
 #########
 # By default, we use the amalgamation (this may be changed below...)
diff -up sqlite-src-3080800/configure.no-malloc-usable-size sqlite-src-3080800/configure
--- sqlite-src-3080800/configure.no-malloc-usable-size	2015-01-20 09:25:03.793148569 +0100
+++ sqlite-src-3080800/configure	2015-01-20 09:28:08.109902277 +0100
@@ -12140,7 +12140,7 @@ done
 
 
 
-for ac_func in fdatasync gmtime_r isnan localtime_r localtime_s malloc_usable_size strchrnul usleep utime
+for ac_func in fdatasync gmtime_r isnan localtime_r localtime_s strchrnul usleep utime
 do
 as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 { $as_echo "$as_me:$LINENO: checking for $ac_func" >&5
