diff -up wireshark-1.2.4/gtk/filter_autocomplete.c.bckp wireshark-1.2.4/gtk/filter_autocomplete.c
--- wireshark-1.2.4/gtk/filter_autocomplete.c.bckp	2009-11-16 21:30:43.000000000 +0100
+++ wireshark-1.2.4/gtk/filter_autocomplete.c	2009-12-14 15:51:18.000000000 +0100
@@ -776,7 +776,9 @@ filter_autocomplete_handle_backspace(Gtk
   GtkRequisition requisition;
   size_t prefix_len;
   gboolean protocols_only = FALSE;
-
+  
+  if (prefix == NULL) return;
+  
   /* Delete the last character in the prefix string */
   prefix_len = strlen(prefix)-1;
   prefix[prefix_len] = '\0';
