diff -up wireshark-1.6.8/ltmain.sh.pie wireshark-1.6.8/ltmain.sh
--- wireshark-1.6.8/ltmain.sh.pie	2012-05-23 10:05:07.900326513 +0200
+++ wireshark-1.6.8/ltmain.sh	2012-05-23 10:05:11.786312666 +0200
@@ -3488,7 +3488,6 @@ static const void *lt_preloaded_setup()
 	symtab_cflags=
 	for arg in $LTCFLAGS; do
 	  case $arg in
-	  -pie | -fpie | -fPIE) ;;
 	  *) func_append symtab_cflags " $arg" ;;
 	  esac
 	done
