--- wireshark-1.10.0/epan/wslua/template-init.lua.enable_lua	2013-06-11 20:12:21.599739328 -0300
+++ wireshark-1.10.0/epan/wslua/template-init.lua	2013-06-11 20:13:18.997537996 -0300
@@ -43,7 +43,7 @@ if running_superuser then
     local disabled_lib = {}
     setmetatable(disabled_lib,{ __index = function() error("this package ".. hint) end } );
 
-    dofile = function() error("dofile " .. hint) end
+--    dofile = function() error("dofile " .. hint) end
     loadfile = function() error("loadfile " .. hint) end
     loadlib = function() error("loadlib " .. hint) end
     require = function() error("require " .. hint) end
