diff -up wireshark-1.6.2/epan/dissectors/packet-nfs.c.orig wireshark-1.6.2/epan/dissectors/packet-nfs.c
--- wireshark-1.6.2/epan/dissectors/packet-nfs.c.orig	2011-09-07 21:54:18.000000000 -0400
+++ wireshark-1.6.2/epan/dissectors/packet-nfs.c	2011-10-17 12:43:30.116761420 -0400
@@ -768,7 +768,7 @@ static int nfsv4_operation_tiers[] = {
 		 1 /* 50, NFS4_OP_LAYOUTGET */,
 		 1 /* 51, NFS4_OP_LAYOUTRETURN */,
 		 1 /* 52, NFS4_OP_SECINFO_NO_NAME */,
-		 1 /* 53, NFS4_OP_SEQUENCE */,
+		 4 /* 53, NFS4_OP_SEQUENCE */,
 		 1 /* 54, NFS4_OP_SET_SSV */,
 		 1 /* 55, NFS4_OP_TEST_STATEID */,
 		 1 /* 56, NFS4_OP_WANT_DELEGATION  */,
@@ -8077,7 +8077,6 @@ static value_string_ext names_nfsv4_oper
 static gint *nfsv4_operation_ett[] =
 {
 	 &ett_nfs_access4 ,
-	 &ett_nfs_access_supp4,
 	 &ett_nfs_close4 ,
 	 &ett_nfs_commit4 ,
 	 &ett_nfs_create4 ,
@@ -8970,6 +8969,8 @@ dissect_nfs_argop4(tvbuff_t *tvb, int of
 		ftree = proto_item_add_subtree(fitem, ett_nfs_argop4);
 	}
 
+	proto_item_append_text(tree, ", Ops(%d):", ops);
+
 	for (ops_counter=0; ops_counter<ops; ops_counter++)
 	{
 		op_summary[ops_counter].optext = g_string_new("");
@@ -9003,6 +9004,7 @@ dissect_nfs_argop4(tvbuff_t *tvb, int of
 		g_string_printf (op_summary[ops_counter].optext, "%s",
                                  val_to_str_ext_const(opcode, &names_nfsv4_operation_ext, "Unknown"));
 
+		proto_item_append_text(tree, " %s", opname);
 
 		switch(opcode)
 		{
@@ -9621,6 +9623,8 @@ dissect_nfs_resop4(tvbuff_t *tvb, int of
 		ftree = proto_item_add_subtree(fitem, ett_nfs_resop4);
 	}
 
+	proto_item_append_text(tree, ", Ops(%d):", ops);
+
 	for (ops_counter = 0; ops_counter < ops; ops_counter++)
 	{
 		op_summary[ops_counter].optext = g_string_new("");
@@ -9654,6 +9658,12 @@ dissect_nfs_resop4(tvbuff_t *tvb, int of
 		g_string_append_printf (op_summary[ops_counter].optext, "%s", opname);
 
 		offset = dissect_nfs_nfsstat4(tvb, offset, newftree, &status);
+		if (status != NFS4_OK) {
+			proto_item_append_text(tree, " %s(%s)", opname,
+				val_to_str_ext(status, &names_nfs_stat_ext, "Unknown error:%u"));
+		} else {
+			proto_item_append_text(tree, " %s", opname);
+		}
 
 		/*
 		 * With the exception of NFS4_OP_LOCK, NFS4_OP_LOCKT, and
