Index: kompare/libdiff2/parserbase.cpp
===================================================================
--- kompare/libdiff2/parserbase.cpp	(revision 1193415)
+++ kompare/libdiff2/parserbase.cpp	(revision 1193416)
@@ -636,7 +636,9 @@
 void ParserBase::checkHeader( const QRegExp& header )
 {
 	if ( m_diffIterator != m_diffLines.end()
-	     && !header.exactMatch( *m_diffIterator ) )
+	     && !header.exactMatch( *m_diffIterator )
+	     && !m_diffIterator->startsWith("Index: ") /* SVN diff */
+	     && !m_diffIterator->startsWith("diff ") /* concatenated diff */)
 		m_malformed = true;
 }
 
