--- osmo-0.2.10.orig/src/options_prefs.c	2010-03-31 23:47:34.000000000 +0200
+++ osmo-0.2.10/src/options_prefs.c	2010-10-27 00:22:48.252574820 +0200
@@ -120,7 +120,7 @@
     g_strlcpy (config.spell_lang, g_getenv("LANG"), MAXNAME);
 	g_strlcpy (config.web_browser, "xdg-open %s", MAXHELPERCMD);
 	g_strlcpy (config.email_client, "xdg-email %s", MAXHELPERCMD);
-	g_strlcpy (config.sound_player, "play %s", MAXHELPERCMD);
+	g_strlcpy (config.sound_player, "aplay %s", MAXHELPERCMD);
 
     /* calendar */
     config.fy_window_size_x = 750;
