diff -up geany-plugins-0.20/addons/src/addons.c.orig geany-plugins-0.20/addons/src/addons.c
--- geany-plugins-0.20/addons/src/addons.c.orig	2011-01-11 09:44:15.000000000 -0200
+++ geany-plugins-0.20/addons/src/addons.c	2011-10-07 12:21:44.034821425 -0300
@@ -22,6 +22,10 @@
  */
 
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "geanyplugin.h"
 
 #include "ao_blanklines.h"
diff -up geany-plugins-0.20/geanyextrasel/src/extrasel.c.orig geany-plugins-0.20/geanyextrasel/src/extrasel.c
--- geany-plugins-0.20/geanyextrasel/src/extrasel.c.orig	2010-12-20 06:29:55.000000000 -0200
+++ geany-plugins-0.20/geanyextrasel/src/extrasel.c	2011-10-07 12:21:44.035821214 -0300
@@ -17,6 +17,10 @@
  *  along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <gdk/gdkkeysyms.h>
 
 #include "geanyplugin.h"
diff -up geany-plugins-0.20/geanygendoc/src/ggd-file-type-manager.c.orig geany-plugins-0.20/geanygendoc/src/ggd-file-type-manager.c
--- geany-plugins-0.20/geanygendoc/src/ggd-file-type-manager.c.orig	2011-10-07 12:25:32.549450260 -0300
+++ geany-plugins-0.20/geanygendoc/src/ggd-file-type-manager.c	2011-10-07 12:25:53.071078674 -0300
@@ -18,6 +18,9 @@
  *  
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
 
 #include "ggd-file-type-manager.h"
 
diff -up geany-plugins-0.20/geanygendoc/src/ggd.c.orig geany-plugins-0.20/geanygendoc/src/ggd.c
--- geany-plugins-0.20/geanygendoc/src/ggd.c.orig	2011-10-07 12:23:29.926495337 -0300
+++ geany-plugins-0.20/geanygendoc/src/ggd.c	2011-10-07 12:25:05.727168687 -0300
@@ -18,6 +18,10 @@
  *  
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "ggd.h"
 
 #include <string.h>
diff -up geany-plugins-0.20/geanyinsertnum/src/insertnum.c.orig geany-plugins-0.20/geanyinsertnum/src/insertnum.c
--- geany-plugins-0.20/geanyinsertnum/src/insertnum.c.orig	2011-01-04 11:15:31.000000000 -0200
+++ geany-plugins-0.20/geanyinsertnum/src/insertnum.c	2011-10-07 12:21:44.035821214 -0300
@@ -17,6 +17,10 @@
  *  along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <ctype.h>
 #include <limits.h>
 #include <stdlib.h>
diff -up geany-plugins-0.20/geanylatex/src/geanylatex.c.orig geany-plugins-0.20/geanylatex/src/geanylatex.c
--- geany-plugins-0.20/geanylatex/src/geanylatex.c.orig	2011-01-04 11:15:31.000000000 -0200
+++ geany-plugins-0.20/geanylatex/src/geanylatex.c	2011-10-07 12:21:44.036821003 -0300
@@ -25,6 +25,10 @@
 /* LaTeX plugin */
 /* This plugin improves the work with LaTeX and Geany.*/
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "geanylatex.h"
 #include "ctype.h"
 
diff -up geany-plugins-0.20/geanylipsum/src/geanylipsum.c.orig geany-plugins-0.20/geanylipsum/src/geanylipsum.c
--- geany-plugins-0.20/geanylipsum/src/geanylipsum.c.orig	2011-01-04 11:15:31.000000000 -0200
+++ geany-plugins-0.20/geanylipsum/src/geanylipsum.c	2011-10-07 12:21:44.036821003 -0300
@@ -18,6 +18,10 @@
  *      MA 02110-1301, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "geanyplugin.h"
 
 #include <string.h>
diff -up geany-plugins-0.20/geanylua/geanylua.c.orig geany-plugins-0.20/geanylua/geanylua.c
--- geany-plugins-0.20/geanylua/geanylua.c.orig	2010-06-10 13:46:27.000000000 -0300
+++ geany-plugins-0.20/geanylua/geanylua.c	2011-10-07 12:21:44.036821003 -0300
@@ -29,7 +29,9 @@
  * information, or visit  http://www.lua.org/license.html .
  */
 
-
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
 
 #include "geany.h"
 #include "plugindata.h"
diff -up geany-plugins-0.20/geanylua/glspi_app.c.orig geany-plugins-0.20/geanylua/glspi_app.c
--- geany-plugins-0.20/geanylua/glspi_app.c.orig	2011-01-04 11:15:31.000000000 -0200
+++ geany-plugins-0.20/geanylua/glspi_app.c	2011-10-07 12:21:44.036821003 -0300
@@ -4,6 +4,10 @@
  * See the file "geanylua.c" for copyright information.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <sys/stat.h>
 #include <stdlib.h>
 #include <errno.h>
diff -up geany-plugins-0.20/geanylua/glspi_init.c.orig geany-plugins-0.20/geanylua/glspi_init.c
--- geany-plugins-0.20/geanylua/glspi_init.c.orig	2010-06-10 13:46:27.000000000 -0300
+++ geany-plugins-0.20/geanylua/glspi_init.c	2011-10-07 12:21:44.037820793 -0300
@@ -3,6 +3,10 @@
  * See the file "geanylua.c" for copyright information.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "glspi.h"
 
 
diff -up geany-plugins-0.20/geanyprj/src/geanyprj.c.orig geany-plugins-0.20/geanyprj/src/geanyprj.c
--- geany-plugins-0.20/geanyprj/src/geanyprj.c.orig	2010-06-10 13:46:27.000000000 -0300
+++ geany-plugins-0.20/geanyprj/src/geanyprj.c	2011-10-07 12:21:44.037820793 -0300
@@ -20,6 +20,10 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <sys/time.h>
 #include <string.h>
 
diff -up geany-plugins-0.20/geanysendmail/src/geanysendmail.c.orig geany-plugins-0.20/geanysendmail/src/geanysendmail.c
--- geany-plugins-0.20/geanysendmail/src/geanysendmail.c.orig	2011-01-04 11:15:31.000000000 -0200
+++ geany-plugins-0.20/geanysendmail/src/geanysendmail.c	2011-10-07 12:21:44.037820793 -0300
@@ -23,6 +23,10 @@
 
 /* A little plugin to send a document as attachment using the preferred mail client */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "geanyplugin.h"
 #include "icon.h"
 
diff -up geany-plugins-0.20/shiftcolumn/src/shiftcolumn.c.orig geany-plugins-0.20/shiftcolumn/src/shiftcolumn.c
--- geany-plugins-0.20/shiftcolumn/src/shiftcolumn.c.orig	2010-06-10 13:46:27.000000000 -0300
+++ geany-plugins-0.20/shiftcolumn/src/shiftcolumn.c	2011-10-07 12:21:44.037820793 -0300
@@ -18,6 +18,9 @@
  *  MA 02110-1301, USA.
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
 
 #include "geany.h"
 #include "support.h"
diff -up geany-plugins-0.20/spellcheck/src/scplugin.c.orig geany-plugins-0.20/spellcheck/src/scplugin.c
--- geany-plugins-0.20/spellcheck/src/scplugin.c.orig	2011-01-11 09:44:15.000000000 -0200
+++ geany-plugins-0.20/spellcheck/src/scplugin.c	2011-10-07 12:21:44.037820793 -0300
@@ -22,6 +22,9 @@
  * $Id$
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
 
 #include "geanyplugin.h"
 
diff -up geany-plugins-0.20/treebrowser/src/treebrowser.c.orig geany-plugins-0.20/treebrowser/src/treebrowser.c
--- geany-plugins-0.20/treebrowser/src/treebrowser.c.orig	2010-12-20 06:29:56.000000000 -0200
+++ geany-plugins-0.20/treebrowser/src/treebrowser.c	2011-10-07 12:21:44.038820583 -0300
@@ -4,6 +4,10 @@
  *      Copyright 2010 Adrian Dimitrov <dimitrov.adrian@gmail.com>
  */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <string.h>
diff -up geany-plugins-0.20/updatechecker/src/updatechecker.c.orig geany-plugins-0.20/updatechecker/src/updatechecker.c
--- geany-plugins-0.20/updatechecker/src/updatechecker.c.orig	2011-01-04 11:15:31.000000000 -0200
+++ geany-plugins-0.20/updatechecker/src/updatechecker.c	2011-10-07 12:21:44.038820583 -0300
@@ -20,6 +20,10 @@
 
 /* A little plugin for regular checks for updates of Geany */
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include "geanyplugin.h"
 #include "libsoup/soup.h"
 #include "stdlib.h"
