From dd114e116bf73a616c95a5b9e400199eb3bfa4c7 Mon Sep 17 00:00:00 2001
From: Thomas Hindoe Paaboel Andersen <phomes@gmail.com>
Date: Sat, 23 May 2015 13:02:56 +0200
Subject: [PATCH] bootctl: fix an error check

---
 src/boot/bootctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/boot/bootctl.c b/src/boot/bootctl.c
index ba534b1..1e65597 100644
--- a/src/boot/bootctl.c
+++ b/src/boot/bootctl.c
@@ -805,7 +805,7 @@ static int remove_boot_efi(const char *esp_path) {
                         continue;
 
                 fd = openat(dirfd(d), de->d_name, O_RDONLY|O_CLOEXEC);
-                if (r < 0)
+                if (fd < 0)
                         return log_error_errno(errno, "Failed to open \"%s/%s\" for reading: %m", p, de->d_name);
 
                 r = get_file_version(fd, &v);
-- 
2.4.1

