From 3062c15117ab6eac5e8b3a3ceb5351ec22ea4481 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 23 Nov 2012 21:38:42 +0100
Subject: [PATCH] timedatectl: improve english language DST change texts a bit

---
 src/timedate/timedatectl.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/timedate/timedatectl.c b/src/timedate/timedatectl.c
index d379bcc..6a1c19a 100644
--- a/src/timedate/timedatectl.c
+++ b/src/timedate/timedatectl.c
@@ -173,10 +173,10 @@ static void print_status_info(StatusInfo *i) {
                 zero(tm);
                 assert_se(strftime(b, sizeof(b), "%a %Y-%m-%d %H:%M:%S %Z", localtime_r(&tc, &tm)) > 0);
                 char_array_0(b);
-                printf(" Last DST change: DST became %s\n"
+                printf(" Last DST change: DST %s at\n"
                        "                  %s\n"
                        "                  %s\n",
-                       is_dstc ? "active" : "inactive", a, b);
+                       is_dstc ? "began" : "ended", a, b);
 
                 t = tn - 1;
                 zero(tm);
@@ -186,10 +186,10 @@ static void print_status_info(StatusInfo *i) {
                 zero(tm);
                 assert_se(strftime(b, sizeof(b), "%a %Y-%m-%d %H:%M:%S %Z", localtime_r(&tn, &tm)) > 0);
                 char_array_0(b);
-                printf(" Next DST change: DST will become %s, the clock will jump %s\n"
+                printf(" Next DST change: DST %s (the clock jumps %s) at\n"
                        "                  %s\n"
                        "                  %s\n",
-                       is_dstn ? "active" : "inactive", jump_str(dn, s, sizeof(s)), a, b);
+                       is_dstn ? "begins" : "ends", jump_str(dn, s, sizeof(s)), a, b);
 
                 free(zc);
                 free(zn);
-- 
1.8.0

