From 21bf2ab082b42f03df5b4685df2fddc4d6e0d572 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Tue, 23 Jul 2013 12:25:32 +0200
Subject: [PATCH 1/2] remove left-over initrd time stamp handling

---
 src/core/main.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/core/main.c b/src/core/main.c
index 77cdcfe..ad155e1 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -1280,18 +1280,8 @@ int main(int argc, char *argv[]) {
                 log_set_target(LOG_TARGET_KMSG);
                 log_open();
 
-                if (in_initrd()) {
-                        char *rd_timestamp = NULL;
-
+                if (in_initrd())
                         initrd_timestamp = userspace_timestamp;
-                        asprintf(&rd_timestamp, "%llu %llu",
-                                 (unsigned long long) initrd_timestamp.realtime,
-                                 (unsigned long long) initrd_timestamp.monotonic);
-                        if (rd_timestamp) {
-                                setenv("RD_TIMESTAMP", rd_timestamp, 1);
-                                free(rd_timestamp);
-                        }
-                }
 
                 if (!skip_setup) {
                         mount_setup_early();
-- 
1.8.3.4

