From ddb7f7fc07a31937989afad53edb87b50e2cad72 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Mon, 18 Aug 2014 17:22:26 -0400
Subject: [PATCH] networkctl: do not mix dns and ntp servers

---
 src/network/networkctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/networkctl.c b/src/network/networkctl.c
index 2a7a1da..d6d2e1d 100644
--- a/src/network/networkctl.c
+++ b/src/network/networkctl.c
@@ -484,7 +484,7 @@ static int link_status(char **args, unsigned n) {
                 if (!strv_isempty(domains))
                         dump_list("      Domain: ", domains);
 
-                sd_network_get_dns(&ntp);
+                sd_network_get_ntp(&ntp);
                 if (!strv_isempty(ntp))
                         dump_list("         NTP: ", ntp);
 
-- 
2.1.0

