From b5fb37899d34ec16f2a88dc3a8be3f4f7bf12ef3 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Mon, 20 Aug 2012 14:33:21 +0200
Subject: [PATCH 1/6] load-fragment: initialize bool invert before use

---
 src/core/load-fragment.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/load-fragment.c b/src/core/load-fragment.c
index 9438aa3..c642261 100644
--- a/src/core/load-fragment.c
+++ b/src/core/load-fragment.c
@@ -2031,7 +2031,7 @@ int config_parse_syscall_filter(
 
         ExecContext *c = data;
         Unit *u = userdata;
-        bool invert;
+        bool invert = false;
         char *w;
         size_t l;
         char *state;
-- 
1.7.12

