From 78e39b43b89c6bf9ce401d6030939a004a23c850 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Sun, 5 Jun 2011 17:22:37 +0200
Subject: [PATCH 2/4] systemctl: fix double unref of a dbus message

---
 src/systemctl.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/systemctl.c b/src/systemctl.c
index 99ada38..a82cce4 100644
--- a/src/systemctl.c
+++ b/src/systemctl.c
@@ -1565,6 +1565,7 @@ static int check_unit(DBusConnection *bus, char **args, unsigned n) {
 
                         dbus_error_free(&error);
                         dbus_message_unref(m);
+                        m = NULL;
                         continue;
                 }
 
-- 
1.7.4.4

