From 5a8e21785907df7466fef5e1cb54ce3bf99e5362 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Fri, 19 Apr 2013 13:58:57 +0200
Subject: [PATCH 3/3] crypt-setup-generator: correctly check return of strdup

---
 src/cryptsetup/cryptsetup-generator.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/cryptsetup/cryptsetup-generator.c b/src/cryptsetup/cryptsetup-generator.c
index ac0ed58..b31329d 100644
--- a/src/cryptsetup/cryptsetup-generator.c
+++ b/src/cryptsetup/cryptsetup-generator.c
@@ -302,7 +302,7 @@ static int parse_proc_cmdline(char ***arg_proc_cmdline_disks, char **arg_proc_cm
 
                 } else if (startswith(word, "luks.key=")) {
                         *arg_proc_cmdline_keyfile = strdup(word + 9);
-                        if (! arg_proc_cmdline_keyfile)
+                        if (!*arg_proc_cmdline_keyfile)
                                 return log_oom();
 
                 } else if (startswith(word, "rd.luks.key=")) {
@@ -311,7 +311,7 @@ static int parse_proc_cmdline(char ***arg_proc_cmdline_disks, char **arg_proc_cm
                                 if (*arg_proc_cmdline_keyfile)
                                         free(*arg_proc_cmdline_keyfile);
                                 *arg_proc_cmdline_keyfile = strdup(word + 12);
-                                if (!arg_proc_cmdline_keyfile)
+                                if (!*arg_proc_cmdline_keyfile)
                                         return log_oom();
                         }
 
-- 
1.8.2.1

