From 403a56bf813014b111640ee804f6b1aa463ca15a Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Mon, 20 Aug 2012 15:15:40 +0200
Subject: [PATCH 4/6] readahead-analyze: don't call fclose on null

---
 src/readahead/readahead-analyze.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/readahead/readahead-analyze.c b/src/readahead/readahead-analyze.c
index 11b2b2d..9a929c0 100644
--- a/src/readahead/readahead-analyze.c
+++ b/src/readahead/readahead-analyze.c
@@ -144,6 +144,7 @@ int main_analyze(const char *pack_path) {
         return EXIT_SUCCESS;
 
 fail:
-        fclose(pack);
+        if(pack)
+                fclose(pack);
         return EXIT_FAILURE;
 }
-- 
1.7.12

