From ba9904e9ce0628cce3bbd8106f6dc6914e418eda Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 21 Nov 2012 18:54:46 +0100
Subject: [PATCH 4/4] catalog: correct substitution

---
 catalog/systemd.catalog | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/catalog/systemd.catalog b/catalog/systemd.catalog
index bc8790f..c2f38a3 100644
--- a/catalog/systemd.catalog
+++ b/catalog/systemd.catalog
@@ -167,7 +167,7 @@ Kernel start-up required @KERNEL_USEC@ microseconds.
 
 Initial RAM disk start-up required @INITRD_USEC@ microseconds.
 
-Userspace start-up required @KERNEL_USEC@ microseconds.
+Userspace start-up required @USERSPACE_USEC@ microseconds.
 
 -- 6bbd95ee977941e497c48be27c254128
 Subject: System sleep state @SLEEP@ entered
-- 
1.8.0

