From 7a855149eaf6dbd336d9defab5b4a9c70b75d5e6 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Wed, 20 Aug 2014 14:51:27 +0200
Subject: [PATCH 6/9] journal-remote: remove unreachable code

---
 src/journal-remote/journal-remote.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/journal-remote/journal-remote.c b/src/journal-remote/journal-remote.c
index aa659d1..7f422bf 100644
--- a/src/journal-remote/journal-remote.c
+++ b/src/journal-remote/journal-remote.c
@@ -579,7 +579,6 @@ static int request_handler(
                         log_error("MHD_get_connection_info failed: cannot get remote fd");
                         return mhd_respond(connection, MHD_HTTP_INTERNAL_SERVER_ERROR,
                                            "Cannot check remote address");
-                        return code;
                 }
 
                 fd = ci->connect_fd;
-- 
2.1.0

