From f284860764076e2a1f0b34eb4c681bb2762f7889 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Thu, 23 Aug 2012 12:28:47 +0200
Subject: [PATCH 6/6] sd-journal: return null when mmap_cache_new fails

---
 src/journal/sd-journal.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/journal/sd-journal.c b/src/journal/sd-journal.c
index 725c979..0f7c02c 100644
--- a/src/journal/sd-journal.c
+++ b/src/journal/sd-journal.c
@@ -1445,6 +1445,7 @@ static sd_journal *journal_new(int flags, const char *path) {
                 hashmap_free(j->directories_by_path);
                 free(j->path);
                 free(j);
+                return NULL;
         }
 
         return j;
-- 
1.7.12

