diff -up xfce4-panel-4.7.4/panel/Makefile.in.multilib xfce4-panel-4.7.4/panel/Makefile.in
--- xfce4-panel-4.7.4/panel/Makefile.in.multilib	2010-11-14 17:18:26.363665076 -0200
+++ xfce4-panel-4.7.4/panel/Makefile.in	2010-11-14 17:18:39.280612944 -0200
@@ -329,7 +329,7 @@ top_srcdir = @top_srcdir@
 INCLUDES = \
 	-I$(top_srcdir) \
 	-DG_LOG_DOMAIN=\"xfce4-panel\" \
-	-DDATADIR=\"$(datadir)/xfce4\" \
+	-DDATADIR=\"$(libdir)/xfce4\" \
 	-DLIBDIR=\"$(libdir)/xfce4\" \
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\" \
 	-DDBUS_API_SUBJECT_TO_CHANGE \
diff -up xfce4-panel-4.7.4/plugins/actions/Makefile.in.multilib xfce4-panel-4.7.4/plugins/actions/Makefile.in
--- xfce4-panel-4.7.4/plugins/actions/Makefile.in.multilib	2010-11-14 17:19:46.543719428 -0200
+++ xfce4-panel-4.7.4/plugins/actions/Makefile.in	2010-11-14 17:20:30.770269167 -0200
@@ -385,7 +385,7 @@ libactions_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = actions.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/applicationsmenu/Makefile.in.multilib xfce4-panel-4.7.4/plugins/applicationsmenu/Makefile.in
--- xfce4-panel-4.7.4/plugins/applicationsmenu/Makefile.in.multilib	2010-11-14 17:19:46.550717774 -0200
+++ xfce4-panel-4.7.4/plugins/applicationsmenu/Makefile.in	2010-11-14 17:20:39.386233325 -0200
@@ -399,7 +399,7 @@ bin_SCRIPTS = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = applicationsmenu.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/clock/Makefile.in.multilib xfce4-panel-4.7.4/plugins/clock/Makefile.in
--- xfce4-panel-4.7.4/plugins/clock/Makefile.in.multilib	2010-11-14 17:19:46.557716121 -0200
+++ xfce4-panel-4.7.4/plugins/clock/Makefile.in	2010-11-14 17:20:50.057711784 -0200
@@ -401,7 +401,7 @@ libclock_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = clock.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/directorymenu/Makefile.in.multilib xfce4-panel-4.7.4/plugins/directorymenu/Makefile.in
--- xfce4-panel-4.7.4/plugins/directorymenu/Makefile.in.multilib	2010-11-14 17:19:46.564714467 -0200
+++ xfce4-panel-4.7.4/plugins/directorymenu/Makefile.in	2010-11-14 17:21:00.264300080 -0200
@@ -401,7 +401,7 @@ bin_SCRIPTS = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = directorymenu.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/launcher/Makefile.in.multilib xfce4-panel-4.7.4/plugins/launcher/Makefile.in
--- xfce4-panel-4.7.4/plugins/launcher/Makefile.in.multilib	2010-11-14 17:19:46.571712813 -0200
+++ xfce4-panel-4.7.4/plugins/launcher/Makefile.in	2010-11-14 17:21:08.324395575 -0200
@@ -396,7 +396,7 @@ liblauncher_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = launcher.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/pager/Makefile.in.multilib xfce4-panel-4.7.4/plugins/pager/Makefile.in
--- xfce4-panel-4.7.4/plugins/pager/Makefile.in.multilib	2010-11-14 17:19:46.577711395 -0200
+++ xfce4-panel-4.7.4/plugins/pager/Makefile.in	2010-11-14 17:21:14.060040309 -0200
@@ -391,7 +391,7 @@ libpager_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = pager.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/separator/Makefile.in.multilib xfce4-panel-4.7.4/plugins/separator/Makefile.in
--- xfce4-panel-4.7.4/plugins/separator/Makefile.in.multilib	2010-11-14 17:19:46.584709742 -0200
+++ xfce4-panel-4.7.4/plugins/separator/Makefile.in	2010-11-14 17:21:19.644720708 -0200
@@ -389,7 +389,7 @@ libseparator_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = separator.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/showdesktop/Makefile.in.multilib xfce4-panel-4.7.4/plugins/showdesktop/Makefile.in
--- xfce4-panel-4.7.4/plugins/showdesktop/Makefile.in.multilib	2010-11-14 17:19:46.591708089 -0200
+++ xfce4-panel-4.7.4/plugins/showdesktop/Makefile.in	2010-11-14 17:21:24.895479988 -0200
@@ -373,7 +373,7 @@ libshowdesktop_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = showdesktop.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/systray/Makefile.in.multilib xfce4-panel-4.7.4/plugins/systray/Makefile.in
--- xfce4-panel-4.7.4/plugins/systray/Makefile.in.multilib	2010-11-14 17:19:46.598706435 -0200
+++ xfce4-panel-4.7.4/plugins/systray/Makefile.in	2010-11-14 17:21:30.270210009 -0200
@@ -397,7 +397,7 @@ libsystray_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = systray.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/tasklist/Makefile.in.multilib xfce4-panel-4.7.4/plugins/tasklist/Makefile.in
--- xfce4-panel-4.7.4/plugins/tasklist/Makefile.in.multilib	2010-11-14 17:19:46.605704781 -0200
+++ xfce4-panel-4.7.4/plugins/tasklist/Makefile.in	2010-11-14 17:21:35.559960116 -0200
@@ -393,7 +393,7 @@ libtasklist_la_DEPENDENCIES = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = tasklist.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
diff -up xfce4-panel-4.7.4/plugins/windowmenu/Makefile.in.multilib xfce4-panel-4.7.4/plugins/windowmenu/Makefile.in
--- xfce4-panel-4.7.4/plugins/windowmenu/Makefile.in.multilib	2010-11-14 17:19:46.612703127 -0200
+++ xfce4-panel-4.7.4/plugins/windowmenu/Makefile.in	2010-11-14 17:21:40.780726509 -0200
@@ -400,7 +400,7 @@ bin_SCRIPTS = \
 #
 # .desktop file
 #
-desktopdir = $(datadir)/xfce4/panel-plugins
+desktopdir = $(libdir)/xfce4/panel-plugins
 desktop_in_files = windowmenu.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 EXTRA_DIST = \
