From 70e7228da13b61421fd27c2b18f429bdc4803e71 Mon Sep 17 00:00:00 2001
From: David Malcolm <dmalcolm@redhat.com>
Date: Wed, 29 Sep 2010 12:40:07 -0400
Subject: [PATCH] Fix missing import

---
 heap/glibc.py |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/heap/glibc.py b/heap/glibc.py
index 0c61643..4b340d2 100644
--- a/heap/glibc.py
+++ b/heap/glibc.py
@@ -27,7 +27,8 @@ import re
 
 import gdb
 
-from heap import WrappedPointer, WrappedValue, caching_lookup_type, type_char_ptr
+from heap import WrappedPointer, WrappedValue, caching_lookup_type, \
+    type_char_ptr, check_missing_debuginfo
 
 class MChunkPtr(WrappedPointer):
     '''Wrapper around glibc's mchunkptr
-- 
1.5.5.6

