diff -up libplist-1.8/CMakeLists.txt.cmake_lib_suffix libplist-1.8/CMakeLists.txt
--- libplist-1.8/CMakeLists.txt.cmake_lib_suffix	2012-01-11 08:29:30.000000000 -0600
+++ libplist-1.8/CMakeLists.txt	2012-04-11 08:57:36.208333767 -0500
@@ -11,8 +11,12 @@ SET( CMAKE_MODULE_PATH ${CMAKE_SOURCE_DI
 
 cmake_minimum_required(VERSION 2.6)
 
+IF(${CMAKE_VERSION} VERSION_GREATER 2.8.4)
+  INCLUDE(GNUInstallDirs)
+ENDIF()
+
 IF(NOT DEFINED CMAKE_INSTALL_LIBDIR)
-   SET(CMAKE_INSTALL_LIBDIR lib CACHE PATH "Output directory for libraries")
+   SET(CMAKE_INSTALL_LIBDIR lib${LIB_SUFFIX} CACHE PATH "Output directory for libraries")
 ENDIF(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 
 FIND_PACKAGE( LibXml2 REQUIRED )
