diff -up openssl-1.1.0c/ssl/statem/statem.c.certreq openssl-1.1.0c/ssl/statem/statem.c
--- openssl-1.1.0c/ssl/statem/statem.c.certreq	2016-11-10 15:03:46.000000000 +0100
+++ openssl-1.1.0c/ssl/statem/statem.c	2016-11-30 14:23:14.274758723 +0100
@@ -361,6 +361,8 @@ static int state_machine(SSL *s, int ser
                  */
                 s->ctx->stats.sess_accept_renegotiate++;
             }
+
+            s->s3->tmp.cert_request = 0;
         } else {
             s->ctx->stats.sess_connect++;
 
@@ -368,7 +370,7 @@ static int state_machine(SSL *s, int ser
             memset(s->s3->client_random, 0, sizeof(s->s3->client_random));
             s->hit = 0;
 
-            s->s3->tmp.cert_request = 0;
+            s->s3->tmp.cert_req = 0;
 
             if (SSL_IS_DTLS(s)) {
                 st->use_timer = 1;
