diff -up openssl-1.1.0c/Configurations/unix-Makefile.tmpl.build openssl-1.1.0c/Configurations/unix-Makefile.tmpl
--- openssl-1.1.0c/Configurations/unix-Makefile.tmpl.build	2016-11-10 15:03:43.000000000 +0100
+++ openssl-1.1.0c/Configurations/unix-Makefile.tmpl	2016-11-11 13:26:36.094400833 +0100
@@ -630,7 +630,7 @@ install_man_docs:
 	@\
 	OUTSUFFIX='.$${SEC}$(MANSUFFIX)'; \
 	OUTTOP="$(DESTDIR)$(MANDIR)"; \
-	GENERATE='pod2man --name=$$NAME --section=$$SEC --center=OpenSSL --release=$(VERSION)'; \
+	GENERATE='TZ=UTC pod2man --name=$$NAME --section=$$SEC --center=OpenSSL --release=$(VERSION)'; \
 	$(PROCESS_PODS)
 
 uninstall_man_docs:
diff -up openssl-1.1.0c/Configurations/10-main.conf.build openssl-1.1.0c/Configurations/10-main.conf
--- openssl-1.1.0c/Configurations/10-main.conf.build	2016-11-10 15:03:43.000000000 +0100
+++ openssl-1.1.0c/Configurations/10-main.conf	2016-11-11 13:29:26.502289226 +0100
@@ -656,6 +656,7 @@ sub vms_info {
         cflags           => add("-m64 -DL_ENDIAN"),
         perlasm_scheme   => "linux64le",
         shared_ldflag    => add("-m64"),
+        multilib         => "64",
     },
 
     "linux-armv4" => {
@@ -696,6 +697,7 @@ sub vms_info {
     "linux-aarch64" => {
         inherit_from     => [ "linux-generic64", asm("aarch64_asm") ],
         perlasm_scheme   => "linux64",
+        multilib         => "64",
     },
     "linux-arm64ilp32" => {  # https://wiki.linaro.org/Platform/arm64-ilp32
         inherit_from     => [ "linux-generic32", asm("aarch64_asm") ],
diff -up openssl-1.1.0c/crypto/ec/ec_lib.c.build openssl-1.1.0c/crypto/ec/ec_lib.c
--- openssl-1.1.0c/crypto/ec/ec_lib.c.build	2016-11-10 15:03:44.000000000 +0100
+++ openssl-1.1.0c/crypto/ec/ec_lib.c	2016-11-11 13:26:36.097400901 +0100
@@ -74,9 +74,6 @@ void EC_pre_comp_free(EC_GROUP *group)
         break;
 #endif
 #ifndef OPENSSL_NO_EC_NISTP_64_GCC_128
-    case PCT_nistp224:
-        EC_nistp224_pre_comp_free(group->pre_comp.nistp224);
-        break;
     case PCT_nistp256:
         EC_nistp256_pre_comp_free(group->pre_comp.nistp256);
         break;
@@ -152,9 +149,6 @@ int EC_GROUP_copy(EC_GROUP *dest, const
         break;
 #endif
 #ifndef OPENSSL_NO_EC_NISTP_64_GCC_128
-    case PCT_nistp224:
-        dest->pre_comp.nistp224 = EC_nistp224_pre_comp_dup(src->pre_comp.nistp224);
-        break;
     case PCT_nistp256:
         dest->pre_comp.nistp256 = EC_nistp256_pre_comp_dup(src->pre_comp.nistp256);
         break;
diff -up openssl-1.1.0c/test/ecdhtest_cavs.h.build openssl-1.1.0c/test/ecdhtest_cavs.h
--- openssl-1.1.0c/test/ecdhtest_cavs.h.build	2016-11-11 13:26:36.079400491 +0100
+++ openssl-1.1.0c/test/ecdhtest_cavs.h	2016-11-11 13:26:36.097400901 +0100
@@ -29,6 +29,7 @@ typedef struct {
 
 static const ecdh_cavs_kat_t ecdh_cavs_kats[] = {
     /* curves over prime fields go here */
+#if 0
     { NID_X9_62_prime192v1,
     "42ea6dd9969dd2a61fea1aac7f8e98edcc896c6e55857cc0",
     "dfbe5d7c61fac88b11811bde328e8a0d12bf01a9d204b523",
@@ -379,6 +380,7 @@ static const ecdh_cavs_kat_t ecdh_cavs_k
     "a6b29632db94da2125dc1cf80e03702687b2acc1122022fa2174765a",
     "61723edd73e10daed73775278f1958ba56f1fc9d085ebc2b64c84fe5",
     "71954e2261e8510be1a060733671d2e9d0a2d012eb4e09556d697d2a" },
+#endif
     { NID_X9_62_prime256v1,
     "700c48f77f56584c5cc632ca65640db91b6bacce3a4df6b42ce7cc838833d287",
     "db71e509e3fd9b060ddb20ba5c51dcc5948d46fbf640dfe0441782cab85fa4ac",
diff -up openssl-1.1.0c/test/ecdhtest.c.build openssl-1.1.0c/test/ecdhtest.c
--- openssl-1.1.0c/test/ecdhtest.c.build	2016-11-10 15:03:47.000000000 +0100
+++ openssl-1.1.0c/test/ecdhtest.c	2016-11-11 13:26:36.097400901 +0100
@@ -252,6 +252,7 @@ typedef struct {
 
 static const ecdh_kat_t ecdh_kats[] = {
     /* Keys and shared secrets from RFC 5114 */
+#if 0
     { NID_X9_62_prime192v1,
     "323FA3169D8E9C6593F59476BC142000AB5BE0E249C43426",
     "631F95BB4A67632C9C476EEE9AB695AB240A0499307FCF62",
@@ -260,6 +261,7 @@ static const ecdh_kat_t ecdh_kats[] = {
     "B558EB6C288DA707BBB4F8FBAE2AB9E9CB62E3BC5C7573E22E26D37F",
     "AC3B1ADD3D9770E6F6A708EE9F3B8E0AB3B480E9F27F85C88B5E6D18",
     "52272F50F46F4EDC9151569092F46DF2D96ECC3B6DC1714A4EA949FA" },
+#endif
     { NID_X9_62_prime256v1,
     "814264145F2F56F2E96A8E337A1284993FAF432A5ABCE59E867B7291D507A3AF",
     "2CE1788EC197E096DB95A200CC0AB26A19CE6BCCAD562B8EEE1B593761CF7F41",
@@ -303,6 +305,7 @@ static const ecdh_kat_t ecdh_kats[] = {
     "01144C7D79AE6956BC8EDB8E7C787C4521CB086FA64407F97894E5E6B2D79B04"
     "D1427E73CA4BAA240A34786859810C06B3C715A3A8CC3151F2BEE417996D19F3"
     "DDEA" },
+#if 0
     /* Keys and shared secrets from RFC 7027 */
     { NID_brainpoolP256r1,
     "81DB1EE100150FF2EA338D708271BE38300CB54241D79950F77B063039804F1D",
@@ -322,6 +325,7 @@ static const ecdh_kat_t ecdh_kats[] = {
     "ABBC19963DAB8E2F1EBA00BFFB29E4D72D13F2224562F405CB80503666B25429",
     "A7927098655F1F9976FA50A9D566865DC530331846381C87256BAF3226244B76"
     "D36403C024D7BBF0AA0803EAFF405D3D24F11A9B5C0BEF679FE1454B21C4CD1F" }
+#endif
 };
 
 /* Given private value and NID, create EC_KEY structure */
diff -up openssl-1.1.0c/test/ecdsatest.c.build openssl-1.1.0c/test/ecdsatest.c
--- openssl-1.1.0c/test/ecdsatest.c.build	2016-11-10 15:03:47.000000000 +0100
+++ openssl-1.1.0c/test/ecdsatest.c	2016-11-11 13:26:36.097400901 +0100
@@ -216,6 +216,7 @@ int x9_62_tests(BIO *out)
     if (!change_rand())
         goto x962_err;
 
+#if 0
     if (!x9_62_test_internal(out, NID_X9_62_prime192v1,
                              "3342403536405981729393488334694600415596881826869351677613",
                              "5735822328888155254683894997897571951568553642892029982342"))
@@ -226,6 +227,7 @@ int x9_62_tests(BIO *out)
                              "3238135532097973577080787768312505059318910517550078427819"
                              "78505179448783"))
         goto x962_err;
+#endif
 # ifndef OPENSSL_NO_EC2M
     if (!x9_62_test_internal(out, NID_X9_62_c2tnb191v1,
                              "87194383164871543355722284926904419997237591535066528048",
