diff -up openssl-1.0.0b/crypto/opensslv.h.version openssl-1.0.0b/crypto/opensslv.h
--- openssl-1.0.0b/crypto/opensslv.h.version	2010-11-16 17:31:23.000000000 +0100
+++ openssl-1.0.0b/crypto/opensslv.h	2010-11-16 17:32:59.000000000 +0100
@@ -25,7 +25,8 @@
  * (Prior to 0.9.5a beta1, a different scheme was used: MMNNFFRBB for
  *  major minor fix final patch/beta)
  */
-#define OPENSSL_VERSION_NUMBER	0x1000002f
+/* we have to keep the version number to not break the abi */
+#define OPENSSL_VERSION_NUMBER	0x10000003
 #ifdef OPENSSL_FIPS
 #define OPENSSL_VERSION_TEXT	"OpenSSL 1.0.0b-fips 16 Nov 2010"
 #else
@@ -83,7 +84,7 @@
  * should only keep the versions that are binary compatible with the current.
  */
 #define SHLIB_VERSION_HISTORY ""
-#define SHLIB_VERSION_NUMBER "1.0.0"
+#define SHLIB_VERSION_NUMBER "1.0.0b"
 
 
 #endif /* HEADER_OPENSSLV_H */
