--- ./Makefile.shared.orig	2016-12-23 02:50:29.419169456 -0200
+++ ./Makefile.shared	2016-12-23 02:53:04.187260786 -0200
@@ -164,7 +164,7 @@ LINK_SO_SHLIB_UNPACKED=	\
 DETECT_GNU_LD=($(CC) -Wl,-V /dev/null 2>&1 | grep '^GNU ld' )>/dev/null
 
 DO_GNU_SO_COMMON=\
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
+	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$$SHLIB._SB_SOVER_"
 DO_GNU_DSO=\
 	SHLIB=$(LIBNAME).so; \
 	SHLIB_SOVER=; \
