diff -up openssl-1.1.0b/engines/afalg/e_afalg.c.eventfd2 openssl-1.1.0b/engines/afalg/e_afalg.c
--- openssl-1.1.0b/engines/afalg/e_afalg.c.eventfd2	2016-09-26 11:46:07.000000000 +0200
+++ openssl-1.1.0b/engines/afalg/e_afalg.c	2016-10-11 10:56:00.884185249 +0200
@@ -104,7 +104,7 @@ static ossl_inline int io_setup(unsigned
 
 static ossl_inline int eventfd(int n)
 {
-    return syscall(__NR_eventfd, n);
+    return syscall(__NR_eventfd2, n, 0);
 }
 
 static ossl_inline int io_destroy(aio_context_t ctx)
