--- openssl-1.1.0c/test/build.info.orig	2016-12-23 04:02:24.807268687 -0200
+++ openssl-1.1.0c/test/build.info	2016-12-23 04:02:41.612699829 -0200
@@ -14,7 +14,7 @@ IF[{- !$disabled{tests} -}]
           danetest heartbeat_test p5_crpt2_test bad_dtls_test \
           constant_time_test verify_extra_test clienthellotest \
           packettest asynctest secmemtest srptest memleaktest \
-          dtlsv1listentest ct_test threadstest afalgtest d2i_test \
+          dtlsv1listentest ct_test afalgtest d2i_test \
           ssl_test_ctx_test ssl_test x509aux cipherlist_test asynciotest \
           bioprinttest sslapitest dtlstest sslcorrupttest bio_enc_test
 
@@ -222,10 +222,6 @@ IF[{- !$disabled{tests} -}]
   INCLUDE[ct_test]=../crypto/include ../include
   DEPEND[ct_test]=../libcrypto
 
-  SOURCE[threadstest]=threadstest.c
-  INCLUDE[threadstest]=.. ../include
-  DEPEND[threadstest]=../libcrypto
-
   SOURCE[afalgtest]=afalgtest.c
   INCLUDE[afalgtest]=.. ../include
   DEPEND[afalgtest]=../libcrypto
