diff -up Source/FreeImage/PluginPNG.cpp.png15 Source/FreeImage/PluginPNG.cpp
--- Source/FreeImage/PluginPNG.cpp.png15	2012-02-04 12:57:32.570607886 -0200
+++ Source/FreeImage/PluginPNG.cpp	2012-02-04 12:57:32.575606813 -0200
@@ -37,6 +37,7 @@
 
 // ----------------------------------------------------------
 
+#include <zlib.h>
 #include <png.h>
 
 // ----------------------------------------------------------
@@ -568,7 +569,7 @@ Load(FreeImageIO *io, fi_handle handle,
 
 			if (png_get_valid(png_ptr, info_ptr, PNG_INFO_iCCP)) {
 				png_charp profile_name = NULL;
-				png_charp profile_data = NULL;
+				png_bytep profile_data = NULL;
 				png_uint_32 profile_length = 0;
 				int  compression_type;
 
@@ -840,7 +841,7 @@ Save(FreeImageIO *io, FIBITMAP *dib, fi_
 
 			FIICCPROFILE *iccProfile = FreeImage_GetICCProfile(dib);
 			if (iccProfile->size && iccProfile->data) {
-				png_set_iCCP(png_ptr, info_ptr, "Embedded Profile", 0, (png_charp)iccProfile->data, iccProfile->size);
+				png_set_iCCP(png_ptr, info_ptr, "Embedded Profile", 0, (png_const_bytep)iccProfile->data, iccProfile->size);
 			}
 
 			// write metadata
