--- FreeImage/Wrapper/FreeImagePlus/doc/FreeImagePlus.dox.orig	2012-02-04 13:19:16.669100940 -0200
+++ FreeImage/Wrapper/FreeImagePlus/doc/FreeImagePlus.dox	2012-02-04 13:20:20.390709396 -0200
@@ -51,7 +51,7 @@ PROJECT_LOGO           =
 # If a relative path is entered, it will be relative to the location 
 # where doxygen was started. If left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = D:/Projects/FreeImage/Wrapper/FreeImagePlus/
+OUTPUT_DIRECTORY       = .
 
 # If the CREATE_SUBDIRS tag is set to YES, then doxygen will create 
 # 4096 sub-directories (in 2 levels) under the output directory of each output 
@@ -598,7 +598,7 @@ WARN_FORMAT            = "$file:$line: $
 # and error messages should be written. If left blank the output is written 
 # to stderr.
 
-WARN_LOGFILE           = D:\Projects\FreeImage\Wrapper\FreeImagePlus\doc\doxygen.log
+WARN_LOGFILE           = ./doxygen.log
 
 #---------------------------------------------------------------------------
 # configuration options related to the input files
@@ -609,8 +609,7 @@ WARN_LOGFILE           = D:\Projects\Fre
 # directories like "/usr/src/myproject". Separate the files or directories 
 # with spaces.
 
-INPUT                  = D:/Projects/FreeImage/Source/FreeImage.h \
-                         D:/Projects/FreeImage/Wrapper/FreeImagePlus/FreeImagePlus.h
+INPUT                  = ../../../Source/FreeImage.h ../FreeImagePlus.h
 
 # This tag can be used to specify the character encoding of the source files 
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is 
@@ -839,7 +838,7 @@ HTML_HEADER            =
 # each generated HTML page. If it is left blank doxygen will generate a 
 # standard footer.
 
-HTML_FOOTER            = footer.html
+HTML_FOOTER            = ./footer.html
 
 # The HTML_STYLESHEET tag can be used to specify a user-defined cascading 
 # style sheet that is used by each HTML page. It can be used to 
