--- SConstruct.orig	2013-07-24 21:29:14.244903671 -0300
+++ SConstruct	2013-07-24 21:49:09.430556918 -0300
@@ -209,6 +209,7 @@ if sys.platform != 'win32':
                '-Wdeclaration-after-statement',
                '-Wmissing-prototypes',
              ])
+  env.Append(LINKFLAGS='-Wl,-soname=%s.so.$MAJOR' % (LIBNAME,) )
 
   ### -Wall is not available on Solaris
   if sys.platform != 'sunos5': 
