--- extra/cups-pdf.conf.oldconf	2007-05-04 13:54:49.000000000 +0200
+++ extra/cups-pdf.conf	2007-11-24 16:15:12.000000000 +0100
@@ -36,11 +36,14 @@
 ##  special qualifiers: 
 ##     ${HOME} will be expanded to the user's home directory
 ##     ${USER} will be expanded to the user name
+##  Add for Fedora (~/.config/user-dirs.dirs)
+##     ${DESKTOP} will be expanded to the user's desktop directory
 ##  in case it is an NFS export make sure it is exported without
 ##  root_squash! 
 ### Default: /var/spool/cups-pdf/${USER}
 
 #Out /var/spool/cups-pdf/${USER}
+Out ${DESKTOP}
 
 ### Key: AnonDirName
 ##  ABSOLUTE path for anonymously created PDF files
