diff -up ClanLib-0.6.5/Sources/Network/Generic/network_delivery_impl.h~ ClanLib-0.6.5/Sources/Network/Generic/network_delivery_impl.h
--- ClanLib-0.6.5/Sources/Network/Generic/network_delivery_impl.h~	2001-11-01 21:27:57.000000000 +0100
+++ ClanLib-0.6.5/Sources/Network/Generic/network_delivery_impl.h	2011-02-08 12:14:13.062932999 +0100
@@ -32,7 +32,7 @@ public:
 	void *data;
 	//: The packet data.
 
-	CL_ConnectionPacket() { data = NULL; }
+	CL_ConnectionPacket() { data = (void *)0; }
 	//: Default constructor
 
 	CL_ConnectionPacket(void *data, int size)
