--- Sources/PNG/provider_png.cpp.png15	2011-11-09 19:59:14.220477079 -0200
+++ Sources/PNG/provider_png.cpp	2011-11-09 20:01:17.153277669 -0200
@@ -544,7 +544,7 @@ void CL_PNGProvider::perform_lock()
 					(png_infopp)NULL);
 		cl_assert(false);
 	}   
-	if (setjmp(png_ptr->jmpbuf))
+	if (setjmp(png_jmpbuf(png_ptr)))
 	{
 		png_destroy_read_struct(&png_ptr, &info_ptr, &end_info);
 		cl_assert(false);
