--- ktorrent-4.0.1/ktupnptest/CMakeLists.txt	2010-06-14 19:29:36.000000000 +0300
+++ ktorrent-4.0.1/ktupnptest/CMakeLists.txt	2010-06-15 01:06:57.000000000 +0300
@@ -6,6 +6,6 @@
 kde4_add_ui_files(ktupnptest_SRC upnptestwidget.ui)
 kde4_add_executable(ktupnptest ${ktupnptest_SRC})
 
-target_link_libraries(ktupnptest ${LIBKTORRENT_LIBRARIES} ktupnp ${QT_QTNETWORK_LIBRARY} ${KDE4_KDEUI_LIBS})
+target_link_libraries(ktupnptest ${LIBKTORRENT_LIBRARIES} ktupnp ktcore ${QT_QTNETWORK_LIBRARY} ${KDE4_KDEUI_LIBS})
 install(TARGETS ktupnptest ${INSTALL_TARGETS_DEFAULT_ARGS})
 
