--- soundtouch/source/SoundTouch/Makefile.am.orig	2011-08-21 21:43:54.215212227 -0300
+++ soundtouch/source/SoundTouch/Makefile.am	2011-08-21 21:44:17.139307635 -0300
@@ -36,8 +36,9 @@ libSoundTouch_la_SOURCES=AAFilter.cpp FI
 # Note by authore: '-msse2' might not work in non-X86 compilations. If someone can
 # fix this script to automatically check for CPU architecture, please submit a patch 
 # to me.
-AM_CXXFLAGS=-O3 -msse2 -fcheck-new -I../../include
-
+AM_CXXFLAGS = -fcheck-new -I../../include
+mmx_optimized.lo : AM_CXXFLAGS = -mmmx -fcheck-new -I../../include
+sse_optimized.lo : AM_CXXFLAGS = -msse -fcheck-new -I../../include
 
 # other linking flags to add
 # noinst_LTLIBRARIES = libSoundTouchOpt.la
