diff -up kopete-4.11.95/protocols/jabber/CMakeLists.txt.system_iris kopete-4.11.95/protocols/jabber/CMakeLists.txt
--- kopete-4.11.95/protocols/jabber/CMakeLists.txt.system_iris	2013-11-20 11:41:22.000000000 -0600
+++ kopete-4.11.95/protocols/jabber/CMakeLists.txt	2013-11-23 00:16:47.590399786 -0600
@@ -1,17 +1,29 @@
 add_definitions(-DIRIS_XMPP_JID_DEPRECATED)
 
 add_subdirectory( icons ) 
-add_subdirectory( libiris ) 
 
 #FIXME:glib : necessary ?
 include_directories( 
 ${KOPETE_INCLUDES} 
 ${CMAKE_CURRENT_SOURCE_DIR}/ui/
 ${CMAKE_CURRENT_SOURCE_DIR}/tasks/
+)
+
+if(USE_SYSTEM_IRIS)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(IRIS REQUIRED iris irisnet)
+include_directories( ${IRIS_INCLUDE_DIRS} )
+link_directories( ${IRIS_LIBRARY_DIRS} )
+else(USE_SYSTEM_IRIS)
+add_subdirectory( libiris )
+
+include_directories(
 ${CMAKE_CURRENT_SOURCE_DIR}/libiris/include/iris
 ${CMAKE_CURRENT_SOURCE_DIR}/libiris/src
 ${QCA2_INCLUDE_DIR}
 )
+link_directories( ${CMAKE_CURRENT_BINARY_DIR}/libiris/ )
+endif(USE_SYSTEM_IRIS)
 
 if(BUILD_GOOGLETALK)
   add_subdirectory(googletalk/libjingle)
@@ -24,9 +36,6 @@ IF(BUILD_JINGLE)
   include_directories(${CMAKE_CURRENT_SOURCE_DIR}/jingle)
 ENDIF(BUILD_JINGLE)
 
-#link_directories( ${CMAKE_CURRENT_BINARY_DIR}/libiris/ ${CMAKE_CURRENT_BINARY_DIR}/jingle/ )
-link_directories( ${CMAKE_CURRENT_BINARY_DIR}/libiris/ )
-
 ########### next target ###############
 
 set(kopete_jabber_ui_SRCS 
@@ -160,7 +169,13 @@ endif(BUILD_JINGLE)
 kde4_add_plugin(kopete_jabber ${kopete_jabber_PART_SRCS})
 
 
-target_link_libraries(kopete_jabber ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${QCA2_LIBRARIES} ${KDE4_SOLID_LIBS} kopete iris_kopete)
+target_link_libraries(kopete_jabber ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${QCA2_LIBRARIES} ${KDE4_SOLID_LIBS} kopete)
+if(USE_SYSTEM_IRIS)
+target_link_libraries(kopete_jabber ${IRIS_LIBRARIES})
+else(USE_SYSETM_IRIS)
+target_link_libraries(kopete_jabber iris_kopete)
+endif(USE_SYSTEM_IRIS)
+
 if(BUILD_JINGLE)	
   target_link_libraries(kopete_jabber ortp asound speex)
 endif(BUILD_JINGLE)
