diff -up kopete-4.11.0/protocols/jabber/CMakeLists.txt.kopete_system_iris kopete-4.11.0/protocols/jabber/CMakeLists.txt
--- kopete-4.11.0/protocols/jabber/CMakeLists.txt.kopete_system_iris	2013-07-09 20:45:56.000000000 -0300
+++ kopete-4.11.0/protocols/jabber/CMakeLists.txt	2013-08-15 18:46:46.168124255 -0300
@@ -4,18 +4,31 @@ add_definitions(-DIRIS_XMPP_JID_DEPRECAT
 add_definitions(-DIRIS_FILE_TRANSFER_PREVIEW -DIRIS_SOCKET_ACCESS -DIRIS_XEP_0115_HASH_ATTRIBUTE -DIRIS_XEP_0184_ID_ATTRIBUTE -DIRIS_XEP_0027_XSIGNED)
 
 add_subdirectory( icons ) 
-add_subdirectory( libiris ) 
 
 #FIXME:glib : necessary ?
 include_directories( 
 ${KOPETE_INCLUDES} 
 ${CMAKE_CURRENT_SOURCE_DIR}/ui/
 ${CMAKE_CURRENT_SOURCE_DIR}/tasks/
+)
+
+if(USE_SYSTEM_IRIS)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(IRIS REQUIRED iris irisnet)
+include_directories( ${IRIS_INCLUDE_DIRS} )
+link_directories( ${IRIS_LIBRARY_DIRS} )
+else(USE_SYSTEM_IRIS)
+add_subdirectory( libiris )
+
+include_directories(
 ${CMAKE_CURRENT_SOURCE_DIR}/libiris/include/iris
 ${CMAKE_CURRENT_SOURCE_DIR}/libiris/src
 ${QCA2_INCLUDE_DIR}
 )
 
+link_directories( ${CMAKE_CURRENT_BINARY_DIR}/libiris/ )
+endif(USE_SYSTEM_IRIS)
+
 if(BUILD_GOOGLETALK)
   add_subdirectory(googletalk/libjingle)
   add_definitions(-DGOOGLETALK_SUPPORT)
@@ -27,9 +40,6 @@ IF(BUILD_JINGLE)
   include_directories(${CMAKE_CURRENT_SOURCE_DIR}/jingle)
 ENDIF(BUILD_JINGLE)
 
-#link_directories( ${CMAKE_CURRENT_BINARY_DIR}/libiris/ ${CMAKE_CURRENT_BINARY_DIR}/jingle/ )
-link_directories( ${CMAKE_CURRENT_BINARY_DIR}/libiris/ )
-
 ########### next target ###############
 
 set(kopete_jabber_ui_SRCS 
