diff -up protocols/wlm/wlmchatsession.cpp.giflib protocols/wlm/wlmchatsession.cpp
--- protocols/wlm/wlmchatsession.cpp.giflib	2014-06-14 11:59:18.574282492 -0300
+++ protocols/wlm/wlmchatsession.cpp	2014-06-14 12:00:56.426631498 -0300
@@ -557,7 +557,7 @@ WlmChatSession::convertToGif( const QPix
                 screenColourmap);
 
     if (status != GIF_OK) {
-        EGifCloseFile(GifFile);
+        EGifCloseFile(GifFile, NULL);
         return;
     }
 
@@ -580,11 +580,11 @@ WlmChatSession::convertToGif( const QPix
 
     if (status != GIF_OK) {
         printGifErrorMessage();
-        EGifCloseFile(GifFile);
+        EGifCloseFile(GifFile, NULL);
         return;
     }
 
-    if (EGifCloseFile(GifFile) != GIF_OK) {
+    if (EGifCloseFile(GifFile, NULL) != GIF_OK) {
         printGifErrorMessage();
         return;
     }
