--- gnuplot-5.0.1/src/gplt_x11.c.orig	2014-12-15 02:24:07.000000000 -0200
+++ gnuplot-5.0.1/src/gplt_x11.c	2015-06-14 18:43:37.070263508 -0300
@@ -2208,7 +2208,8 @@ exec_cmd(plot_struct *plot, char *comman
 	    /* Grey out key box */
 	    if (!fill_gc)
 		fill_gc = XCreateGC(dpy, plot->window, 0, 0);
-	    XCopyGC(dpy, *current_gc, ~0, fill_gc);
+	    if (current_gc)
+		    XCopyGC(dpy, *current_gc, ~0, fill_gc);
 	    XSetForeground(dpy, fill_gc, plot->cmap->colors[1]);
 	    XSetStipple(dpy, fill_gc, stipple_dots);
 	    XSetFillStyle(dpy, fill_gc, FillStippled);
@@ -2387,7 +2388,8 @@ exec_cmd(plot_struct *plot, char *comman
 		    /* Load selected pattern or fill into a separate gc */
 		    if (!fill_gc)
 			fill_gc = XCreateGC(dpy,plot->window,0,0);
-		    XCopyGC(dpy, *current_gc, ~0, fill_gc);
+		    if (current_gc)
+		      XCopyGC(dpy, *current_gc, ~0, fill_gc);
 		    XSetFillStyle(dpy, fill_gc, FillSolid);
 		    XSetForeground(dpy, fill_gc, plot->cmap->colors[0]);
 		    XFillRectangle(dpy, plot->pixmap, fill_gc, 
@@ -2497,7 +2499,8 @@ exec_cmd(plot_struct *plot, char *comman
 	    /* Load selected pattern or fill into a separate gc */
 	    if (!fill_gc)
 		fill_gc = XCreateGC(dpy, plot->window, 0, 0);
-	    XCopyGC(dpy, *current_gc, ~0, fill_gc);
+	    if (current_gc)
+		    XCopyGC(dpy, *current_gc, ~0, fill_gc);
 	    x11_setfill(&fill_gc, style);
 
 	    /* gnuplot has origin at bottom left, but X uses top left
@@ -2908,7 +2911,8 @@ exec_cmd(plot_struct *plot, char *comman
 		/* Load selected pattern or fill into a separate gc */
 		if (!fill_gc)
 		    fill_gc = XCreateGC(dpy, plot->window, 0, 0);
-		XCopyGC(dpy, *current_gc, ~0, fill_gc);
+		if (current_gc)
+		    XCopyGC(dpy, *current_gc, ~0, fill_gc);
 
 		x11_setfill(&fill_gc, style);
 
