diff -up gnome-applets-2.28.0/invest-applet/docs/invest-applet.omf.in.seriesid gnome-applets-2.28.0/invest-applet/docs/invest-applet.omf.in
--- gnome-applets-2.28.0/invest-applet/docs/invest-applet.omf.in.seriesid	2009-10-27 00:25:35.351806142 -0400
+++ gnome-applets-2.28.0/invest-applet/docs/invest-applet.omf.in	2009-10-27 00:29:15.578549690 -0400
@@ -3,7 +3,7 @@
   <resource>
     <subject category="GNOME|Applets|Utility"/>
     <type>user's guide</type>
-    <relation seriesid="df109296-40e7-11d6-935f-beaae1cfe4f9"/>
+    <relation seriesid="04cc2ede-586b-2643-7e6f-37a004e0fbd470"/>
     <rights type="GNU FDL" license.version="1.1" holder="Terence Hall"/>
   </resource>        
 </omf>
