diff -up gnome-applets-2.30.0/Makefile.am.stickynotes-null gnome-applets-2.30.0/Makefile.am
--- gnome-applets-2.30.0/Makefile.am.stickynotes-null	2010-03-28 01:29:51.000000000 -0400
+++ gnome-applets-2.30.0/Makefile.am	2010-03-28 23:24:47.360759295 -0400
@@ -42,7 +42,6 @@ always_built_SUBDIRS =  \
 	geyes		\
 	mini-commander	\
 	gkb-new		\
-	stickynotes	\
 	trashapplet	\
 	null_applet
 
diff -up gnome-applets-2.30.0/null_applet/GNOME_StickyNotesApplet.server.in.stickynotes-null gnome-applets-2.30.0/null_applet/GNOME_StickyNotesApplet.server.in
--- gnome-applets-2.30.0/null_applet/GNOME_StickyNotesApplet.server.in.stickynotes-null	2010-03-28 23:24:47.362759364 -0400
+++ gnome-applets-2.30.0/null_applet/GNOME_StickyNotesApplet.server.in	2010-03-28 23:24:47.363762926 -0400
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<oaf_info>
+	<oaf_server iid="OAFIID:GNOME_StickyNotesApplet" type="factory" location="OAFIID:GNOME_NullApplet_Factory">
+	    <oaf_attribute name="repo_ids" type="stringv">
+		<item value="IDL:GNOME/Vertigo/PanelAppletShell:1.0"/>
+		<item value="IDL:Bonobo/Control:1.0"/>
+		<item value="IDL:Bonobo/Unknown:1.0"/>
+	    </oaf_attribute>
+	    <oaf_attribute name="name" type="string" _value="Sticky Notes (Deprecated)"/>
+	    <oaf_attribute name="description" type="string" _value="Create, view, and manage sticky notes on the desktop"/>
+	    <oaf_attribute name="bugzilla:bugzilla" type="string" value="GNOME"/>
+	    <oaf_attribute name="bugzilla:product" type="string" value="gnome-applets"/>
+	    <oaf_attribute name="bugzilla:component" type="string" value="stickynotes"/>
+	    <oaf_attribute name="bugzilla:other_binaries" type="string" value="stickynotes_applet"/>
+	</oaf_server>
+</oaf_info>
+
diff -up gnome-applets-2.30.0/null_applet/Makefile.am.stickynotes-null gnome-applets-2.30.0/null_applet/Makefile.am
--- gnome-applets-2.30.0/null_applet/Makefile.am.stickynotes-null	2010-03-28 23:24:47.000000000 -0400
+++ gnome-applets-2.30.0/null_applet/Makefile.am	2010-03-28 23:26:45.638021343 -0400
@@ -25,6 +25,7 @@ server_in_files =				\
 	GNOME_MailcheckApplet_Factory.server.in	\
 	GNOME_Panel_WirelessApplet.server.in	\
 	GNOME_KeyboardApplet.server.in		\
+	GNOME_StickyNotesApplet.server.in	\
 	$(battstat_applet)			\
 	$(mixer_server_in)
 server_DATA = $(server_in_files:.server.in=.server)
@@ -44,6 +45,7 @@ EXTRA_DIST =					\
 	GNOME_MailcheckApplet_Factory.server.in	\
 	GNOME_Panel_WirelessApplet.server.in	\
 	GNOME_MixerApplet.server.in		\
+	GNOME_StickyNotesApplet.server.in	\
 	GNOME_BattstatApplet.server.in		\
 	GNOME_KeyboardApplet.server.in
 
diff -up gnome-applets-2.30.0/null_applet/null_applet.c.stickynotes-null gnome-applets-2.30.0/null_applet/null_applet.c
--- gnome-applets-2.30.0/null_applet/null_applet.c.stickynotes-null	2010-03-28 23:24:47.000000000 -0400
+++ gnome-applets-2.30.0/null_applet/null_applet.c	2010-03-28 23:27:18.451011564 -0400
@@ -47,7 +47,9 @@ insert_oafiids (GHashTable *hash_table)
 	g_hash_table_insert (hash_table,
 			     "OAFIID:GNOME_KeyboardApplet", _("Keyboard Indicator"));
         g_hash_table_insert (hash_table,
-                            "OAFIID:GNOME_BattstatApplet", "Battery Charge Monitor");
+                            "OAFIID:GNOME_BattstatApplet", _("Battery Charge Monitor"));
+        g_hash_table_insert (hash_table,
+                            "OAFIID:GNOME_StickyNotesApplet", _("Sticky Notes"));
 }
 
 static gboolean already_running;
