diff -up mxcl-liblastfm-3539836/src/fingerprint/fingerprint.pro.qmake mxcl-liblastfm-3539836/src/fingerprint/fingerprint.pro
--- mxcl-liblastfm-3539836/src/fingerprint/fingerprint.pro.qmake	2010-09-11 09:16:54.000000000 -0500
+++ mxcl-liblastfm-3539836/src/fingerprint/fingerprint.pro	2010-09-17 07:34:57.620666383 -0500
@@ -6,7 +6,7 @@ include( _files.qmake )
 DEFINES += LASTFM_OHAI_QMAKE
 
 INSTALLS = target
-target.path = /lib
+target.path = /$$(LIB)
 
 mac:CONFIG( app_bundle ) {
     LIBS += libfftw3f.a libsamplerate.a -L/opt/local/include
@@ -25,4 +25,4 @@ mac:CONFIG( app_bundle ) {
         # precision, and I'm not sure if that matters or not
         PKGCONFIG += fftw3f
     }
-}
\ No newline at end of file
+}
diff -up mxcl-liblastfm-3539836/src/lastfm.pro.qmake mxcl-liblastfm-3539836/src/lastfm.pro
--- mxcl-liblastfm-3539836/src/lastfm.pro.qmake	2010-09-11 09:16:54.000000000 -0500
+++ mxcl-liblastfm-3539836/src/lastfm.pro	2010-09-17 07:34:41.153388690 -0500
@@ -3,7 +3,7 @@ QT = core network xml
 include( _files.qmake )
 
 INSTALLS = target
-target.path = /lib
+target.path = /$$(LIB)
 
 win32{
     DEFINES += LASTFM_OHAI_QMAKE _ATL_DLL 
