--- src/plugin_xmms/Makefile.am.orig	2014-11-26 23:19:47.000000000 -0200
+++ src/plugin_xmms/Makefile.am	2014-11-27 19:34:22.871634365 -0200
@@ -59,8 +59,7 @@ libxmms_flac_la_LIBADD = \
 	$(top_builddir)/src/share/replaygain_analysis/libreplaygain_analysis.la \
 	$(top_builddir)/src/share/replaygain_synthesis/libreplaygain_synthesis.la \
 	$(top_builddir)/src/share/utf8/libutf8.la \
-	$(top_builddir)/src/libFLAC/libFLAC.la \
-	-L$(top_builddir)/src/libFLAC/.libs \
+	-L$(libdir) -lFLAC \
 	@XMMS_LIBS@ \
 	@LIBICONV@
 libxmms_flac_la_LDFLAGS = -module -avoid-version
