diff -NarU5 aumix-2.8.orig/src/common.c aumix-2.8/src/common.c
--- aumix-2.8.orig/src/common.c	2008-09-26 13:45:15.000000000 -0400
+++ aumix-2.8/src/common.c	2008-09-26 13:49:45.000000000 -0400
@@ -256,10 +256,13 @@
 		Gpm_Close();
 #endif				/* HAVE_CURSES */
 		close(mixer_fd);
 #if HAVE_CURSES
 		CloseScreen();
+		
+		/* simple carriage return to reset the cursor */
+		printf( "\n" );
 	}
 #endif				/* HAVE_CURSES */
 	exit(EXIT_SUCCESS);
 }
 
