diff -up os-prober-1.53/linux-boot-probes/mounted/powerpc/40yaboot.parsefix os-prober-1.53/linux-boot-probes/mounted/powerpc/40yaboot
--- os-prober-1.53/linux-boot-probes/mounted/powerpc/40yaboot.parsefix	2012-05-25 15:57:13.477647064 +0430
+++ os-prober-1.53/linux-boot-probes/mounted/powerpc/40yaboot	2012-05-25 16:16:58.844827933 +0430
@@ -74,8 +74,8 @@ parse_yaboot_conf () {
 				kernel="$(dequote "$2")"
 			;;
 			append)
-				shift 1
-				parameters="$(dequote "${line#append=}")"
+				cleanappend=`printf %s "$line" | sed -E 's/append(\s)*=(\s)*//'`
+				parameters="$(dequote "$cleanappend")"
 			;;
 			initrd)
 				initrd="$(dequote "$2")"
