diff -up os-prober-1.53/os-probes/mounted/common/90linux-distro.usermovefix os-prober-1.53/os-probes/mounted/common/90linux-distro
--- os-prober-1.53/os-probes/mounted/common/90linux-distro.usermovefix	2012-06-02 20:07:39.878619263 +0430
+++ os-prober-1.53/os-probes/mounted/common/90linux-distro	2012-06-02 20:14:40.341362757 +0430
@@ -13,7 +13,8 @@ type="$3"
 # better. Make sure this test has a high number so that more accurate tests
 # can come first.
 # TODO: look for ld-linux.so on arches that have it
-if ls "$dir"/lib*/ld*.so* >/dev/null 2>/dev/null; then
+if ls "$dir"/lib*/ld*.so* >/dev/null 2>/dev/null && [ -d "$dir/boot" ] \
+	|| ls "$dir"/usr/lib*/ld*.so* >/dev/null 2>/dev/null; then
 	if [ -e "$dir/etc/debian_version" ]; then
 		short="Debian"
 		long="$(printf "Debian GNU/Linux (%s)\n" "$(cat "$dir/etc/debian_version")")"
