diff -up os-prober-1.57/common.sh.remove-basename os-prober-1.57/common.sh
--- os-prober-1.57/common.sh.remove-basename	2013-01-23 01:06:50.023992389 +0330
+++ os-prober-1.57/common.sh	2013-01-23 02:01:09.435244141 +0330
@@ -57,7 +57,7 @@ progname=
 cache_progname() {
   case $progname in
     '')
-      progname="$(basename "$0")"
+      progname="${0##*/}"
       ;;
   esac
 }
diff -up os-prober-1.57/linux-boot-probes/mounted/powerpc/40yaboot.remove-basename os-prober-1.57/linux-boot-probes/mounted/powerpc/40yaboot
--- os-prober-1.57/linux-boot-probes/mounted/powerpc/40yaboot.remove-basename	2013-01-23 02:04:03.855063593 +0330
+++ os-prober-1.57/linux-boot-probes/mounted/powerpc/40yaboot	2013-01-23 02:02:34.876175982 +0330
@@ -27,7 +27,7 @@ recordstanza () {
 		fi
 
 		if [ -z "$title" ]; then
-			title="$(basename "$kernel")"
+			title="${kernel##*/}"
 		fi
 		if [ "$read_only" ]; then
 			parameters="ro $parameters"
