diff -up os-prober-1.58/os-probes/mounted/x86/05efi.factor-out-logger-efi-fix os-prober-1.58/os-probes/mounted/x86/05efi
--- os-prober-1.58/os-probes/mounted/x86/05efi.factor-out-logger-efi-fix	2013-04-25 04:23:06.000000000 +0430
+++ os-prober-1.58/os-probes/mounted/x86/05efi	2013-06-19 01:42:05.428750558 +0430
@@ -58,7 +58,11 @@ ret=1
 for test in /usr/lib/os-probes/mounted/efi/*; do
 	debug "running subtest $test"
 	if [ -f "$test" ] && [ -x "$test" ]; then
+		# we need results of subtest in stdout
+		orig_fd_res=$fd_result
+		export fd_result=1
 		entry=$("$test" "$mpoint/$efi")
+		export fd_result=$orig_fd_res
 		if [ -n "$entry" ]; then
 			debug "bootloader $entry found by subtest $test"
 			ret=0
