diff -up elektra-0.7.0/scripts/elektraenv.sh.local elektra-0.7.0/scripts/elektraenv.sh
--- elektra-0.7.0/scripts/elektraenv.sh.local	2011-03-17 12:46:52.265269003 +0100
+++ elektra-0.7.0/scripts/elektraenv.sh	2011-03-17 15:22:31.531268999 +0100
@@ -43,9 +43,9 @@ FILE="`mktemp -t elektraenv.XXXXXXXXX`"
 
 
 readEnvTree() {
-	local keysAvailable=0
-	local stage=0
-	local key
+	keysAvailable=0
+	stage=0
+	key=""
 
 	for stage in 1 2 3; do
 		echo "# Stage $stage"
