diff -up gnome-settings-daemon-2.31.91/plugins/xrandr/gsd-xrandr-manager.c.capplet gnome-settings-daemon-2.31.91/plugins/xrandr/gsd-xrandr-manager.c
--- gnome-settings-daemon-2.31.91/plugins/xrandr/gsd-xrandr-manager.c.capplet	2010-09-01 14:25:51.121940001 -0400
+++ gnome-settings-daemon-2.31.91/plugins/xrandr/gsd-xrandr-manager.c	2010-09-01 14:26:06.346940001 -0400
@@ -76,7 +76,7 @@
 #define GSD_XRANDR_ICON_NAME "gsd-xrandr"
 
 /* executable of the control center's display configuration capplet */
-#define GSD_XRANDR_DISPLAY_CAPPLET "gnome-control-center display"
+#define GSD_XRANDR_DISPLAY_CAPPLET "gnome-display-properties"
 
 #define GSD_DBUS_PATH "/org/gnome/SettingsDaemon"
 #define GSD_DBUS_NAME "org.gnome.SettingsDaemon"
