diff -up gnome-settings-daemon-2.29.90/plugins/keyboard/gsd-keyboard-xkb.c.keyboard-icon gnome-settings-daemon-2.29.90/plugins/keyboard/gsd-keyboard-xkb.c
--- gnome-settings-daemon-2.29.90/plugins/keyboard/gsd-keyboard-xkb.c.keyboard-icon	2010-02-17 18:27:01.013904630 -0500
+++ gnome-settings-daemon-2.29.90/plugins/keyboard/gsd-keyboard-xkb.c	2010-02-17 18:31:52.121659134 -0500
@@ -446,6 +446,7 @@ show_hide_icon ()
 		if (icon == NULL) {
 			xkl_debug (150, "Creating new icon\n");
 			icon = gkbd_status_new ();
+                        gtk_status_icon_set_name (icon, "keyboard");
 			g_signal_connect (icon, "popup-menu",
 					  G_CALLBACK
 					  (status_icon_popup_menu_cb),
