diff -up squid-3.2.0.12/compat/Makefile.am.fpic squid-3.2.0.12/compat/Makefile.am
--- squid-3.2.0.12/compat/Makefile.am.fpic	2011-09-28 22:14:21.115126558 -0300
+++ squid-3.2.0.12/compat/Makefile.am	2011-09-28 22:16:07.058607172 -0300
@@ -4,6 +4,7 @@
 
 include $(top_srcdir)/src/Common.am
 
+AM_CFLAGS = $(SQUID_CFLAGS) -fPIC
 
 # Ideally this would be 100% inline functions and macro wrappers.
 
diff -up squid-3.2.0.12/snmplib/Makefile.am.fpic squid-3.2.0.12/snmplib/Makefile.am
--- squid-3.2.0.12/snmplib/Makefile.am.fpic	2011-09-28 22:17:00.102332440 -0300
+++ squid-3.2.0.12/snmplib/Makefile.am	2011-09-28 22:05:19.500248414 -0300
@@ -2,7 +2,7 @@
 ##
 ## Makefile for libsnmp.
 ## 
-AM_CFLAGS = $(SQUID_CFLAGS)
+AM_CFLAGS = $(SQUID_CFLAGS) -fPIC
 AM_CXXFLAGS = $(SQUID_CXXFLAGS)
 noinst_LIBRARIES = libsnmplib.a
 libsnmplib_a_SOURCES  = asn1.c parse.c snmp_vars.c \
