diff -up squid-3.2.0.13/src/ssl/ErrorDetailManager.cc.dbg squid-3.2.0.13/src/ssl/ErrorDetailManager.cc
--- squid-3.2.0.13/src/ssl/ErrorDetailManager.cc.dbg	2011-10-14 03:42:56.000000000 +0200
+++ squid-3.2.0.13/src/ssl/ErrorDetailManager.cc	2011-10-26 08:16:12.968976995 +0200
@@ -209,7 +209,8 @@ Ssl::ErrorDetailFile::parse(const char *
                 return false;
             }
 
-            const char *errorName = parser.getByName("name").termedBuf();
+            String strErrName = parser.getByName("name");
+            const char *errorName = strErrName.termedBuf();
             if (!errorName) {
                 debugs(83, DBG_IMPORTANT, HERE <<
                        "WARNING! invalid or no error detail name on:" << s);
