--- ./src/cf.data.pre	2013-04-27 05:34:48.000000000 +0200
+++ ./src/cf.data.pre	2013-05-03 10:13:16.007067356 +0200
@@ -4178,7 +4178,7 @@
 
 NAME: logfile_rotate
 TYPE: int
-DEFAULT: 10
+DEFAULT: 0
 LOC: Config.Log.rotateNumber
 DOC_START
 	Specifies the number of logfile rotations to make when you
@@ -5782,11 +5782,11 @@
 
 NAME: cache_mgr
 TYPE: string
-DEFAULT: webmaster
+DEFAULT: root
 LOC: Config.adminEmail
 DOC_START
 	Email-address of local cache manager who will receive
-	mail if the cache dies.  The default is "webmaster".
+	mail if the cache dies.  The default is "root".
 DOC_END
 
 NAME: mail_from
