diff -up squid-3.2.0.14/src/ssl/support.cc.gcc47 squid-3.2.0.14/src/ssl/support.cc
--- squid-3.2.0.14/src/ssl/support.cc.gcc47	2012-01-16 09:02:32.097769625 +0100
+++ squid-3.2.0.14/src/ssl/support.cc	2012-01-16 09:43:53.149248015 +0100
@@ -333,7 +333,7 @@ ssl_options[] = {
 #endif
 #if SSL_OP_ALL
     {
-        "ALL", SSL_OP_ALL
+        "ALL", (long)SSL_OP_ALL
     },
 #endif
 #if SSL_OP_SINGLE_DH_USE
