--- libqtxdg-1.1.0/qiconfix/qiconloader.cpp.orig	2015-02-11 14:55:32.449576420 -0200
+++ libqtxdg-1.1.0/qiconfix/qiconloader.cpp	2015-02-11 14:56:37.668729867 -0200
@@ -252,6 +252,7 @@
         // Parent themes provide fallbacks for missing icons
         m_parents = indexReader.value(
                 QLatin1String("Icon Theme/Inherits")).toStringList();
+		m_parents.removeAll(QString());
 
         // Ensure a default platform fallback for all themes
         if (m_parents.isEmpty()) {
