diff -up kdeutils-4.7.1/printer-applet/monitor.py.job-originating-user-name kdeutils-4.7.1/printer-applet/monitor.py
--- kdeutils-4.7.1/printer-applet/monitor.py.job-originating-user-name	2011-09-01 16:22:40.000000000 -0500
+++ kdeutils-4.7.1/printer-applet/monitor.py	2011-09-27 10:01:41.958634749 -0500
@@ -395,7 +395,7 @@ class Monitor:
                         continue
 
                     jobs[jobid] = attrs
-                except AttributeError:
+                except (AttributeError, KeyError):
                     jobs[jobid] = {'job-k-octets': 0}
                 except cups.IPPError, (e, m):
                     self.watcher.cups_ipp_error (self, e, m)
