diff --git a/src/io_functions.c b/src/io_functions.c
index 20917df..b159ed8 100644
--- a/src/io_functions.c
+++ b/src/io_functions.c
@@ -227,7 +227,7 @@ write_vmap_to_layer (LqrVMap * vmap, gpointer data)
         gimp_layer_new (image_ID, name, w, h, GIMP_RGBA_IMAGE, 100,
                         GIMP_NORMAL_MODE);
       gimp_drawable_fill (seam_layer_ID, GIMP_TRANSPARENT_FILL);
-      gimp_image_add_layer (image_ID, seam_layer_ID, -1);
+      gimp_image_insert_layer (image_ID, seam_layer_ID, -1, -1);
       gimp_layer_translate (seam_layer_ID, x_off, y_off);
       if (seam_layer_p)
         {
diff --git a/src/layers_combo.c b/src/layers_combo.c
index 10d810e..e032eed 100644
--- a/src/layers_combo.c
+++ b/src/layers_combo.c
@@ -199,7 +199,7 @@ callback_new_mask_button (GtkWidget * button, gpointer data)
     gimp_layer_new (p_data->image_ID, nl_data->name,
 		    p_data->old_width, p_data->old_height,
 		    image_type, 50, GIMP_NORMAL_MODE);
-  gimp_image_add_layer (p_data->image_ID, layer_ID, -1);
+  gimp_image_insert_layer (p_data->image_ID, layer_ID, -1, -1);
   gimp_drawable_fill (layer_ID, GIMP_TRANSPARENT_FILL);
   gimp_layer_translate (layer_ID, p_data->x_off, p_data->y_off);
   gimp_image_undo_group_end (p_data->image_ID);
diff --git a/src/render.c b/src/render.c
index f50fc8c..b7e8c7e 100644
--- a/src/render.c
+++ b/src/render.c
@@ -172,7 +172,7 @@ render_init_carver (PlugInImageVals * image_vals,
     {
       g_snprintf (new_layer_name, LQR_MAX_NAME_LENGTH, "%s LqR", layer_name);
       layer_ID = gimp_layer_copy (layer_ID);
-      gimp_image_add_layer (image_ID, layer_ID, -1);
+      gimp_image_insert_layer (image_ID, layer_ID, -1, -1);
       gimp_drawable_set_name (layer_ID, new_layer_name);
       gimp_drawable_set_visible (layer_ID, FALSE);
     }
@@ -181,7 +181,7 @@ render_init_carver (PlugInImageVals * image_vals,
       image_ID = gimp_image_new (old_width, old_height, gimp_image_base_type(image_ID));
       gimp_image_undo_group_start(image_ID);
       layer_ID = gimp_layer_new_from_drawable (layer_ID, image_ID);
-      gimp_image_add_layer (image_ID, layer_ID, -1);
+      gimp_image_insert_layer (image_ID, layer_ID, -1, -1);
       gimp_layer_translate(layer_ID, -x_off, -y_off);
       gimp_drawable_set_visible (layer_ID, TRUE);
       if (vals->resize_aux_layers)
@@ -858,7 +858,7 @@ static gboolean copy_aux_layer_to_new_image (gint32 image_ID, gint32 * layer_ID_
   if (old_layer_ID)
     {
       new_layer_ID = gimp_layer_new_from_drawable (old_layer_ID, image_ID);
-      gimp_image_add_layer (image_ID, new_layer_ID, -1);
+      gimp_image_insert_layer (image_ID, new_layer_ID, -1, -1);
       /* gimp_drawable_offsets (old_layer_ID, &aux_x_off, &aux_y_off); */
       gimp_layer_translate (new_layer_ID, -x_off, -y_off);
     }
