--- evince-2.29.91/backend/dvi/mdvi-lib/t1.c	2009-04-24 09:24:16.000000000 +0200
+++ evince-2.29.91/backend/dvi/mdvi-lib/t1.c	2010-03-09 15:37:26.000000000 +0100
@@ -143,7 +143,7 @@ char	*t1_lookup_font(const char *name, U
 
 	/* get the `base' name */
 	if(ext) {
-		newname = mdvi_strdup(name);
+		newname = mdvi_strdup(info.fontfile);
 		newname[ext - info.fontfile - 1] = 0;
 	} else
 		newname = (char *)name; /* we don't modify this */
