diff -ru snes9express-1.43.orig/Makefile.in snes9express-1.43/Makefile.in
--- snes9express-1.43.orig/Makefile.in	2009-12-19 18:08:54.000000000 -0500
+++ snes9express-1.43/Makefile.in	2010-04-12 21:07:42.180375468 -0400
@@ -60,7 +60,7 @@
 	extra.$(OBJEXT) prefs.$(OBJEXT) misc.$(OBJEXT) \
 	s9xskin.$(OBJEXT) skins.$(OBJEXT) interface.$(OBJEXT)
 snes9express_OBJECTS = $(am_snes9express_OBJECTS)
-snes9express_LDADD = $(LDADD)
+snes9express_LDADD = $(GTK_LIBS) @LIBOBJS@
 snes9express_LINK = $(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) \
 	$(snes9express_LDFLAGS) $(LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@
@@ -217,7 +217,7 @@
 snes9express_SOURCES = frend.h frend.cc defines.h controller.h controller.cc profiler.h profiler.cc rom.h rom.cc sound.h sound.cc video.h video.cc cpu.h cpu.cc netplay.h netplay.cc extra.h extra.cc prefs.h prefs.cc misc.h misc.cc s9xskin.h s9xskin.cc skins.h skins.cc interface.h interface.cc 
 snes9express_DEPENDENCIES = frend.h defines.h Makefile config.h
 AM_CXXFLAGS = @GTK_CFLAGS@
-snes9express_LDFLAGS = $(GTK_LIBS) @LIBOBJS@
+snes9express_LDFLAGS =
 EXTRA_DIST = snes.s9xskin snsp.s9xskin snes9express.spec
 pkgdata_DATA = snes.s9xskin snsp.s9xskin
 AM_CPPFLAGS = -DS9XDATADIR=\"$(pkgdatadir)\"
