--- configure.in.aud	2007-09-02 01:50:47.000000000 -0300
+++ configure.in	2010-08-04 02:14:24.953258549 -0300
@@ -10,15 +10,15 @@
 AC_PROG_INSTALL
 
 # Checks for libraries.
-PKG_CHECK_MODULES(AUDACIOUS, [audacious >= 1.4.0], , [
+PKG_CHECK_MODULES(AUDACIOUS, [audclient >= 1.4.0], , [
         AC_MSG_RESULT(no)
         AC_MSG_ERROR([
 
 You must have audacious >= 1.4.0 development headers installed to build.
 ])])
-AUD_CFLAGS=`pkg-config --cflags audacious 2> /dev/null`
-AUD_LIBS=`pkg-config --libs audacious 2> /dev/null`
-AUD_LIB_DIR=`pkg-config --variable=lib_dir audacious 2> /dev/null`
+AUD_CFLAGS=`pkg-config --cflags audclient 2> /dev/null`
+AUD_LIBS=`pkg-config --libs audclient 2> /dev/null`
+AUD_LIB_DIR=`pkg-config --variable=lib_dir audclient 2> /dev/null`
 AC_SUBST(AUD_CFLAGS)
 AC_SUBST(AUD_LIBS)
 AC_SUBST(AUD_LIB_DIR)
