diff -up lyx-1.6.6/src/frontends/qt4/Makefile.am.ImplicitDSOLinking_X11 lyx-1.6.6/src/frontends/qt4/Makefile.am
--- lyx-1.6.6/src/frontends/qt4/Makefile.am.ImplicitDSOLinking_X11	2009-06-05 00:06:17.000000000 -0500
+++ lyx-1.6.6/src/frontends/qt4/Makefile.am	2010-05-21 07:34:22.182603709 -0500
@@ -35,7 +35,7 @@ noinst_LTLIBRARIES = liblyxqt4.la
 
 liblyxqt4_la_DEPENDENCIES = $(MOCEDFILES)
 liblyxqt4_la_LDFLAGS = $(QT4_LDFLAGS)
-liblyxqt4_la_LIBADD = $(QT4_LIB) 
+liblyxqt4_la_LIBADD = $(QT4_LIB) $(X_LIBS)
 
 AM_CPPFLAGS += \
 	$(QT4_CPPFLAGS) \
diff -up lyx-1.6.6/src/frontends/qt4/Makefile.in.ImplicitDSOLinking_X11 lyx-1.6.6/src/frontends/qt4/Makefile.in
--- lyx-1.6.6/src/frontends/qt4/Makefile.in.ImplicitDSOLinking_X11	2010-05-17 13:13:17.000000000 -0500
+++ lyx-1.6.6/src/frontends/qt4/Makefile.in	2010-05-21 07:35:55.500354702 -0500
@@ -428,7 +428,7 @@ MOCEDFILES = $(MOCHEADER:%.h=%_moc.cpp)
 noinst_LTLIBRARIES = liblyxqt4.la
 liblyxqt4_la_DEPENDENCIES = $(MOCEDFILES)
 liblyxqt4_la_LDFLAGS = $(QT4_LDFLAGS)
-liblyxqt4_la_LIBADD = $(QT4_LIB) 
+liblyxqt4_la_LIBADD = $(QT4_LIB) $(X_LIBS)
 SOURCEFILES = \
 	ButtonPolicy.cpp \
 	ButtonPolicy.h \
