--- supertuxkart-0.7.3/src/graphics/material.cpp.orig	2012-09-29 02:55:21.207020615 -0300
+++ supertuxkart-0.7.3/src/graphics/material.cpp	2012-09-29 02:55:46.650652512 -0300
@@ -512,7 +512,7 @@ void  Material::setMaterialProperties(vi
         // customizable EMT_ONETEXTURE_BLEND instead
         m->MaterialType = video::EMT_ONETEXTURE_BLEND ;
         m->MaterialTypeParam = 
-            pack_texureBlendFunc(video::EBF_SRC_ALPHA, 
+            pack_textureBlendFunc(video::EBF_SRC_ALPHA, 
                                  video::EBF_ONE_MINUS_SRC_ALPHA,
                                  video::EMFN_MODULATE_1X, 
                                  video::EAS_TEXTURE | video::EAS_VERTEX_COLOR);
@@ -539,7 +539,7 @@ void  Material::setMaterialProperties(vi
         // account, which messes up fading in/out effects. So we use the 
         // more customizable EMT_ONETEXTURE_BLEND instead
         m->MaterialType = video::EMT_ONETEXTURE_BLEND ;
-        m->MaterialTypeParam = pack_texureBlendFunc(video::EBF_SRC_ALPHA, 
+        m->MaterialTypeParam = pack_textureBlendFunc(video::EBF_SRC_ALPHA, 
                                                     video::EBF_ONE,
                                                     video::EMFN_MODULATE_1X,
                                                     video::EAS_TEXTURE | 
