diff -up at-spi-1.32.0/at-spi.schemas.in.use-orbit-stack at-spi-1.32.0/at-spi.schemas.in
--- at-spi-1.32.0/at-spi.schemas.in.use-orbit-stack	2010-10-04 14:15:43.691509987 -0400
+++ at-spi-1.32.0/at-spi.schemas.in	2010-10-04 14:15:45.963770826 -0400
@@ -18,7 +18,7 @@ support.</long>
       <applyto>/desktop/gnome/interface/at-spi-corba</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Enable CORBA-based AT-SPI infrastructure</short>
         <long>In combination with /desktop/gnome/interface/accessibility, this key is needed to enable the CORBA-based AT-SPI infrastructure if the CORBA-based AT-SPI infrastructure has been relocated.  It is ignored if the CORBA-based AT-SPI infrastructure has not been relocated.</long>
