--- poppler-0.18.4/poppler/CairoOutputDev.cc	2012-02-15 18:06:48.000000000 +0100
+++ poppler-0.18.4/poppler/CairoOutputDev.cc	2012-05-04 13:04:57.000000000 +0200
@@ -2789,6 +2789,7 @@ void CairoOutputDev::drawImage(GfxState
     cairo_rectangle (cairo, 0., 0., 1., 1.);
   if (maskPattern) {
     cairo_clip (cairo);
+    cairo_set_matrix (cairo, &mask_matrix);
     cairo_mask (cairo, maskPattern);
   } else {
     cairo_fill (cairo);
