--- usr/share/applications/steam.desktop.orig	2012-12-04 20:15:50.000000000 -0200
+++ usr/share/applications/steam.desktop	2012-12-20 19:20:57.221783049 -0200
@@ -8,35 +8,3 @@ Type=Application
 Categories=Network;FileTransfer;Game;
 MimeType=x-scheme-handler/steam
 Actions=Store;Community;Library;Servers;Screenshots;News;Settings;Friends
-
-[Desktop Action Store]
-Name=Store
-Exec=steam steam://store
-
-[Desktop Action Community]
-Name=Community
-Exec=steam steam://url/SteamIDControlPage
-
-[Desktop Action Library]
-Name=Library
-Exec=steam steam://open/games
-
-[Desktop Action Servers]
-Name=Servers
-Exec=steam steam://open/servers
-
-[Desktop Action Screenshots]
-Name=Screenshots
-Exec=steam steam://open/screenshots
-
-[Desktop Action News]
-Name=News
-Exec=steam steam://open/news
-
-[Desktop Action Settings]
-Name=Settings
-Exec=steam steam://open/settings
-
-[Desktop Action Friends]
-Name=Friends
-Exec=steam steam://open/friends
