--- steam.orig	2014-02-10 22:25:25.000000000 -0200
+++ steam	2014-06-10 17:27:48.247239443 -0300
@@ -17,6 +17,12 @@ export STEAMSCRIPT_VERSION=100047
 # Set up domain for script localization
 export TEXTDOMAIN=steam
 
+# Close to tray icon
+export STEAM_FRAME_FORCE_CLOSE=1
+
+PATH=/usr/local/sbin:/usr/sbin:/sbin:$PATH
+export PATH
+
 function show_message()
 {
 	style=$1
@@ -109,10 +115,10 @@ function install_bootstrap()
 	test -f "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs" && source "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs"
 	DESKTOP_DIR="${XDG_DESKTOP_DIR:-$HOME/Desktop}"
 
-	if [ -d "$DESKTOP_DIR" ]; then
-		cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
-		chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
-	fi
+#	if [ -d "$DESKTOP_DIR" ]; then
+#		cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
+#		chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
+#	fi
 
 	# Restore the umask
 	umask "$omask"
@@ -186,9 +192,5 @@ fi
 # go to the install directory and run the client
 cp "$LAUNCHSTEAMBOOTSTRAPFILE" "$LAUNCHSTEAMDIR/bootstrap.tar.xz"
 cd "$LAUNCHSTEAMDIR"
-if [ "`command -v tee`" != "" ]; then
-	mkdir -p --mode=777 /tmp/dumps
-	exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@" 2>&1 | tee "/tmp/dumps/${USER}_stdout.txt"
-else
-	exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"
-fi
+mkdir -p "$DEFAULTSTEAMDIR/logs"
+exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@" &>"$DEFAULTSTEAMDIR/logs/stdout.log"
