--- usr/bin/steam.orig	2012-12-14 21:32:02.000000000 -0200
+++ usr/bin/steam	2012-12-20 19:17:35.601008039 -0200
@@ -12,6 +12,9 @@ export STEAMSCRIPT_VERSION=0
 # Set up domain for script localization
 export TEXTDOMAIN=steam
 
+PATH=/usr/local/sbin:/usr/sbin:/sbin:$PATH
+export PATH
+
 function has_command()
 {
 	command -v "$1" >/dev/null
@@ -171,14 +174,14 @@ function install_bootstrap()
 	test -f "${XDG_CONFIG_HOME:-~/.config}/user-dirs.dirs" && source "${XDG_CONFIG_HOME:-~/.config}/user-dirs.dirs"
 	DESKTOP_DIR="${XDG_DESKTOP_DIR:-$HOME/Desktop}"
 
-	cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
-	chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
+	#cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
+	#chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
 
 	# Restore the umask
 	umask "$omask"
 
 	# Try and install any extra packages
-	install_extra_packages jockey-common python-apt
+	#install_extra_packages jockey-common python-apt
 }
 
 function repair_bootstrap()
