--- steam.orig	2013-03-12 15:23:00.000000000 -0300
+++ steam	2013-03-18 18:44:29.780694262 -0300
@@ -17,6 +17,9 @@ export STEAMSCRIPT_VERSION=100036
 # Set up domain for script localization
 export TEXTDOMAIN=steam
 
+PATH=/usr/local/sbin:/usr/sbin:/sbin:$PATH
+export PATH
+
 function show_message()
 {
 	style=$1
@@ -109,10 +112,10 @@ function install_bootstrap()
 	test -f "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs" && source "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs"
 	DESKTOP_DIR="${XDG_DESKTOP_DIR:-$HOME/Desktop}"
 
-	if [ -d "$DESKTOP_DIR" ]; then
-		cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
-		chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
-	fi
+#	if [ -d "$DESKTOP_DIR" ]; then
+#		cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
+#		chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
+#	fi
 
 	# Restore the umask
 	umask "$omask"
@@ -191,9 +194,5 @@ fi
 # go to the install directory and run the client
 cp "$LAUNCHSTEAMBOOTSTRAPFILE" "$LAUNCHSTEAMDIR/bootstrap.tar.xz"
 cd "$LAUNCHSTEAMDIR"
-if [ "`command -v tee`" != "" ]; then
-	mkdir -p --mode=777 /tmp/dumps
-	exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@" 2>&1 | tee "/tmp/dumps/${USER}_stdout.txt"
-else
-	exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"
-fi
+mkdir -p "$DEFAULTSTEAMDIR/logs"
+exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@" &>"$DEFAULTSTEAMDIR/logs/stdout.log"
