--- steam.orig	2013-02-24 22:50:40.000000000 -0300
+++ steam	2013-02-26 18:11:33.337982300 -0300
@@ -17,6 +17,9 @@ export STEAMSCRIPT_VERSION=100034
 # Set up domain for script localization
 export TEXTDOMAIN=steam
 
+PATH=/usr/local/sbin:/usr/sbin:/sbin:$PATH
+export PATH
+
 function show_message()
 {
 	style=$1
@@ -109,10 +112,10 @@ function install_bootstrap()
 	test -f "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs" && source "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs"
 	DESKTOP_DIR="${XDG_DESKTOP_DIR:-$HOME/Desktop}"
 
-	if [ -d "$DESKTOP_DIR" ]; then
-		cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
-		chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
-	fi
+#	if [ -d "$DESKTOP_DIR" ]; then
+#		cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
+#		chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
+#	fi
 
 	# Restore the umask
 	umask "$omask"
