--- steam.orig	2013-01-18 21:24:42.684787165 -0200
+++ steam	2013-01-18 21:26:04.763387986 -0200
@@ -17,6 +17,9 @@ export STEAMSCRIPT_VERSION=100022
 # Set up domain for script localization
 export TEXTDOMAIN=steam
 
+PATH=/usr/local/sbin:/usr/sbin:/sbin:$PATH
+export PATH
+
 function show_message()
 {
 	style=$1
@@ -106,8 +109,8 @@ function install_bootstrap()
 	test -f "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs" && source "${XDG_CONFIG_HOME:-$HOME/.config}/user-dirs.dirs"
 	DESKTOP_DIR="${XDG_DESKTOP_DIR:-$HOME/Desktop}"
 
-	cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
-	chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
+	#cp "/usr/share/applications/$STEAMPACKAGE.desktop" "$DESKTOP_DIR"
+	#chmod +x "$DESKTOP_DIR/$STEAMPACKAGE.desktop"
 
 	# Restore the umask
 	umask "$omask"
