diff -up makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.cpp.noup makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.cpp
--- makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.cpp.noup	2012-01-15 19:25:05.498287828 -0200
+++ makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.cpp	2012-01-15 19:27:09.607937030 -0200
@@ -98,9 +98,6 @@ void CSettingDialog::ReadSettings()
     int show_debug = client->GetSettingInt(apset_app_ShowDebug);
     generalTab->check_DebugLog->setCheckState( (show_debug==0) ? Qt::Unchecked : Qt::Checked );
 
-    int site_access = client->GetSettingInt(apset_app_UpdateEnable);
-    generalTab->check_SiteAccess->setCheckState( (site_access==0) ? Qt::Unchecked : Qt::Checked );
-
     int expert_mode = client->GetSettingInt(apset_app_ExpertMode);
     generalTab->check_ExpertMode->setCheckState( (expert_mode==0) ? Qt::Unchecked : Qt::Checked );
 
@@ -181,7 +178,6 @@ bool CSettingDialog::WriteSettings(bool&
     // general
     client->SetSettingString(apset_app_DataDir,Utf16FromQString(generalTab->dataDir->text()));
     client->SetSettingInt( apset_app_ShowDebug , (generalTab->check_DebugLog->checkState() == Qt::Checked) ? 1 : 0 );
-    client->SetSettingInt( apset_app_UpdateEnable , (generalTab->check_SiteAccess->checkState() == Qt::Checked) ? 1 : 0 );
     client->SetSettingInt( apset_app_ExpertMode , (generalTab->check_ExpertMode->checkState() == Qt::Checked) ? 1 : 0 );
     client->SetSettingInt( apset_app_ShowAVSyncMessages , (generalTab->check_ShowAV->checkState() == Qt::Checked) ? 1 : 0 );
 
@@ -397,9 +393,6 @@ CGeneralTab::CGeneralTab(IIconPath* icon
     check_DebugLog = new QCheckBox();
     m_lay->addWidget(createLabel(UI_QSTRING(APP_IFACE_SETTINGS_LOG_DEBUG_MSG)),0,0,Qt::AlignRight);
     m_lay->addWidget(check_DebugLog,0,1);
-    check_SiteAccess = new QCheckBox();
-    m_lay->addWidget(createLabel(UI_QSTRING(APP_IFACE_SETTINGS_ENABLE_SITE_ACCESS)),1,0,Qt::AlignRight);
-    m_lay->addWidget(check_SiteAccess,1,1);
     check_ExpertMode = new QCheckBox();
     m_lay->addWidget(createLabel(UI_QSTRING(APP_IFACE_SETTINGS_EXPERT_MODE)),2,0,Qt::AlignRight);
     m_lay->addWidget(check_ExpertMode,2,1);
diff -up makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.h.noup makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.h
--- makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.h.noup	2012-01-15 19:26:31.804963430 -0200
+++ makemkv_v1.7.0_oss/makemkvgui/src/settingdlg.h	2012-01-15 19:26:52.818501826 -0200
@@ -29,7 +29,6 @@ public:
     CDirSelectBox*   dataDir;
 
     QCheckBox*      check_DebugLog;
-    QCheckBox*      check_SiteAccess;
     QCheckBox*      check_ExpertMode;
     QCheckBox*      check_ShowAV;
 
