diff -up parted-1.8.1/libparted/fs/ext2/parted_io.c.blkoflow parted-1.8.1/libparted/fs/ext2/parted_io.c
--- parted-1.8.1/libparted/fs/ext2/parted_io.c.blkoflow	2006-12-03 22:27:47.000000000 +0000
+++ parted-1.8.1/libparted/fs/ext2/parted_io.c	2010-04-20 15:55:06.000000000 +0100
@@ -82,7 +82,9 @@ static int do_read(void *cookie, void *p
 {
 	struct my_cookie *monster = cookie;
 
-	return ped_geometry_read(monster->geom, ptr, block << (monster->logsize - 9), num << (monster->logsize - 9));
+	return ped_geometry_read(monster->geom, ptr,
+				 (PedSector)block << (monster->logsize - 9),
+				 (PedSector)num << (monster->logsize - 9));
 }
 
 static int do_set_blocksize(void *cookie, int logsize)
@@ -98,8 +100,8 @@ static int do_write(void *cookie, void *
 	struct my_cookie *monster = cookie;
 
 	return ped_geometry_write(monster->geom, ptr,
-				  block << (monster->logsize - 9),
-				  num << (monster->logsize - 9));
+				  (PedSector)block << (monster->logsize - 9),
+				  (PedSector)num << (monster->logsize - 9));
 }
 
 
