--- stepmake/aclocal.m4~	2012-05-03 11:53:07.000000000 -0500
+++ stepmake/aclocal.m4	2012-05-11 09:43:55.571617264 -0500
@@ -287,12 +287,17 @@
 ])
 
 AC_DEFUN(STEPMAKE_GXXCODEGENBUG, [
-    AC_MSG_CHECKING([options for known g++ tail call bug])
+    AC_MSG_CHECKING([options for known g++ bugs])
     case "$GXX:$CXX_VERSION" in
 	yes:400600?)
-	    AC_MSG_RESULT([-fno-optimize-sibling-calls])
+	    AC_MSG_RESULT([-fno-optimize-sibling-calls (tail call bug)])
 	    CXXFLAGS="$CXXFLAGS -fno-optimize-sibling-calls"
 	    ;;
+        yes:400700?)
+            AC_MSG_RESULT([-fno-tree-vrp (comparison bug)])
+            CXXFLAGS="$CXXFLAGS -fno-tree-vrp"
+            ;;
+	    
 	*) AC_MSG_RESULT([none])
     esac
     AC_SUBST(CXXFLAGS)
