--- pygobject-3.7.92/tests/Makefile.in.ignore-more-pep8-errors	2013-03-18 13:55:04.000000000 +0100
+++ pygobject-3.7.92/tests/Makefile.in	2013-03-20 21:34:25.815686535 +0100
@@ -759,7 +759,7 @@
 	@if type pyflakes >/dev/null 2>&1; then pyflakes $(top_srcdir); else echo "skipped, pyflakes not installed"; fi
 	@if test -z "$$SKIP_PEP8"; then \
 		echo "  CHECK  PEP8"; \
-		if type pep8 >/dev/null 2>&1; then pep8 --ignore=E501,E123,E124 --repeat --show-source $(top_srcdir); else echo "skipped, pep8 not installed"; fi; \
+		if type pep8 >/dev/null 2>&1; then pep8 --ignore=E501,E123,E124,E127 --repeat --show-source $(top_srcdir); else echo "skipped, pep8 not installed"; fi; \
 	fi
 	export `$(DBUS_LAUNCH)` && \
 	$(RUN_TESTS_ENV_VARS) $(EXEC_NAME) $(PYTHON) -Wd $(srcdir)/runtests.py; rc=$$?; \
--- pygobject-3.7.92/tests/Makefile.am.ignore-more-pep8-errors	2013-02-27 21:14:17.000000000 +0100
+++ pygobject-3.7.92/tests/Makefile.am	2013-03-20 21:34:25.815686535 +0100
@@ -136,7 +136,7 @@
 	@if type pyflakes >/dev/null 2>&1; then pyflakes $(top_srcdir); else echo "skipped, pyflakes not installed"; fi
 	@if test -z "$$SKIP_PEP8"; then \
 		echo "  CHECK  PEP8"; \
-		if type pep8 >/dev/null 2>&1; then pep8 --ignore=E501,E123,E124 --repeat --show-source $(top_srcdir); else echo "skipped, pep8 not installed"; fi; \
+		if type pep8 >/dev/null 2>&1; then pep8 --ignore=E501,E123,E124,E127 --repeat --show-source $(top_srcdir); else echo "skipped, pep8 not installed"; fi; \
 	fi
 	export `$(DBUS_LAUNCH)` && \
 	$(RUN_TESTS_ENV_VARS) $(EXEC_NAME) $(PYTHON) -Wd $(srcdir)/runtests.py; rc=$$?; \
