Index: pygobject-3.3.91/tests/test_overrides_gtk.py
===================================================================
--- pygobject-3.3.91.orig/tests/test_overrides_gtk.py
+++ pygobject-3.3.91/tests/test_overrides_gtk.py
@@ -1553,3 +1553,8 @@ class TestTextBuffer(unittest.TestCase):
                                         None)
         self.assertEqual(start.get_offset(), 6)
         self.assertEqual(end.get_offset(), 11)
+
+    def test_rhbz842880(self):
+        # Regression test for https://bugzilla.redhat.com/show_bug.cgi?id=842880
+        store = Gtk.ListStore(int, int)
+        store.append([1, 2])
