--- Makefile.orig	2013-04-16 22:18:47.580203490 -0300
+++ Makefile	2013-04-16 22:19:54.784925790 -0300
@@ -680,6 +680,7 @@ COMMON_LIBS-$(FFMPEG_A)           += $(F
 COMMON_LIBS                       += $(COMMON_LIBS-yes)
 
 OBJS_COMMON                   += $(addsuffix .o, $(basename $(SRCS_COMMON)))
+libmpdemux/demux_mkv.o: CFLAGS += -O2
 OBJS_MENCODER                 += $(addsuffix .o, $(basename $(SRCS_MENCODER)))
 OBJS_MPLAYER                  += $(addsuffix .o, $(basename $(SRCS_MPLAYER)))
 OBJS_MPLAYER-$(PE_EXECUTABLE) += osdep/mplayer-rc.o
