diff -up INCHI-1-API/INCHI_API/gcc_so_makefile/makefile.r INCHI-1-API/INCHI_API/gcc_so_makefile/makefile
--- INCHI-1-API/INCHI_API/gcc_so_makefile/makefile.r	2010-02-02 02:24:54.000000000 +0100
+++ INCHI-1-API/INCHI_API/gcc_so_makefile/makefile	2010-07-08 01:07:51.000000000 +0200
@@ -1,5 +1,5 @@
 # Comment out the next line to create so/dll only
-CREATE_MAIN = 1
+#CREATE_MAIN = 1
 
 ifdef windir
 # Under Win32/MinGW/gcc to add functions compliant to Pascal/stdcall calling
@@ -84,7 +84,7 @@ endif
 
 # === C Compiler Options =======
 ifndef C_OPTIONS
-  C_OPTIONS = -ansi -O3 -c
+  C_OPTIONS = -ansi $(OPTFLAGS) -c
   ifndef windir
   ifdef ISLINUX
     ifndef C_SO_OPTIONS
