From: James Westby <james.westby@linaro.org>
Date: Thu, 28 Oct 2010 09:14:26 -0400
Subject: Fix the race that leads to the password box disappearing, but the dialog remaining.
Bug: http://bugs.freedesktop.org/show_bug.cgi?id=30515
Bug-Ubuntu: https://launchpad.net/bugs/649939
Bug-Ubuntu: https://launchpad.net/bugs/445303

Index: policykit-1/src/polkitagent/polkitagentsession.c
===================================================================
--- policykit-1.orig/src/polkitagent/polkitagentsession.c	2011-03-03 23:21:50.827981036 +0100
+++ policykit-1/src/polkitagent/polkitagentsession.c	2011-03-03 23:30:01.335981036 +0100
@@ -421,6 +421,7 @@
                   gpointer user_data)
 {
   PolkitAgentSession *session = POLKIT_AGENT_SESSION (user_data);
+  GMainContext *context = g_main_context_default();
 
   if (G_UNLIKELY (_show_debug ()))
     {
@@ -432,6 +433,11 @@
 
   /* kill all the watches we have set up, except for the child since it has exited already */
   session->child_pid = 0;
+  /* Allow the stdout of the child to be processed if we haven't finished yet */
+  while (g_main_context_pending(context))
+    {
+      g_main_context_iteration(context, FALSE);
+    }
   complete_session (session, FALSE);
 }
 
@@ -632,15 +638,15 @@
   if (G_UNLIKELY (_show_debug ()))
     g_print ("PolkitAgentSession: spawned helper with pid %d\n", (gint) session->child_pid);
 
-  session->child_watch_source = g_child_watch_source_new (session->child_pid);
-  g_source_set_callback (session->child_watch_source, (GSourceFunc) child_watch_func, session, NULL);
-  g_source_attach (session->child_watch_source, g_main_context_get_thread_default ());
-
   session->child_stdout_channel = g_io_channel_unix_new (session->child_stdout);
   session->child_stdout_watch_source = g_io_create_watch (session->child_stdout_channel, G_IO_IN);
   g_source_set_callback (session->child_stdout_watch_source, (GSourceFunc) io_watch_have_data, session, NULL);
   g_source_attach (session->child_stdout_watch_source, g_main_context_get_thread_default ());
 
+  session->child_watch_source = g_child_watch_source_new (session->child_pid);
+  g_source_set_callback (session->child_watch_source, (GSourceFunc) child_watch_func, session, NULL);
+  g_source_attach (session->child_watch_source, g_main_context_get_thread_default ());
+
 
   session->success = FALSE;
 
