--- configure.ac.orig	2014-02-20 18:34:46.540443777 -0300
+++ configure.ac	2014-02-20 18:37:16.389672105 -0300
@@ -177,7 +177,7 @@ AC_ARG_ENABLE([libsystemd-login],
               [enable_libsystemd_login=auto])
 if test "$enable_libsystemd_login" != "no"; then
   PKG_CHECK_MODULES(LIBSYSTEMD_LOGIN,
-                    [libsystemd-login],
+                    [libsystemd],
                     have_libsystemd_login=yes,
                     have_libsystemd_login=no)
   if test "$have_libsystemd_login" = "yes"; then
