From ca1b0cd633fb14d2e152abd7b965b02ea58f66d0 Mon Sep 17 00:00:00 2001
From: David Zeuthen <zeuthen@gmail.com>
Date: Wed, 14 Nov 2012 14:29:16 -0500
Subject: [PATCH] Include gmodule-2.0 to avoid linker errors

Signed-off-by: David Zeuthen <zeuthen@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ecf56f1..fefde5d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -122,7 +122,7 @@ if test "x$GCC" = "xyes"; then
   changequote([,])dnl
 fi
 
-PKG_CHECK_MODULES(GLIB, [gio-2.0 >= 2.30.0])
+PKG_CHECK_MODULES(GLIB, [gmodule-2.0 gio-2.0 >= 2.30.0])
 AC_SUBST(GLIB_CFLAGS)
 AC_SUBST(GLIB_LIBS)
 
-- 
1.8.0

