From d9fa44186200e1cd17d69a1ae5b0dead202f7a58 Mon Sep 17 00:00:00 2001
From: Olivier Crête <olivier.crete@collabora.com>
Date: Tue, 28 Feb 2012 22:13:43 +0000
Subject: rtpcodecnego: Ignore config while comparing send codecs

---
diff --git a/gst/fsrtpconference/fs-rtp-codec-negotiation.c b/gst/fsrtpconference/fs-rtp-codec-negotiation.c
index 12f498f..a63faa0 100644
--- a/gst/fsrtpconference/fs-rtp-codec-negotiation.c
+++ b/gst/fsrtpconference/fs-rtp-codec-negotiation.c
@@ -1393,17 +1393,24 @@ lookup_codec_association_by_codec_for_sending (GList *codec_associations,
     FsCodec *codec)
 {
   GList *item;
+  CodecAssociation *res = NULL;
+  FsCodec *tmpcodec = codec_copy_filtered (codec, FS_PARAM_TYPE_CONFIG);
 
   for (item = codec_associations; item; item = g_list_next (item))
   {
     CodecAssociation *ca = item->data;
 
     if (codec_association_is_valid_for_sending (ca, FALSE) &&
-        fs_codec_are_equal (ca->codec, codec))
-      return ca;
+        fs_codec_are_equal (ca->send_codec, tmpcodec))
+    {
+      res = ca;
+      break;
+    }
   }
 
-  return NULL;
+  fs_codec_destroy (tmpcodec);
+
+  return res;
 }
 
 FsRtpHeaderExtension *
--
cgit v0.9
