diff -up dconf-0.7.3/editor/dconf-schema.vala.crash dconf-0.7.3/editor/dconf-schema.vala
--- dconf-0.7.3/editor/dconf-schema.vala.crash	2011-04-06 14:57:43.277748785 -0400
+++ dconf-0.7.3/editor/dconf-schema.vala	2011-04-06 14:58:41.160881491 -0400
@@ -52,9 +52,9 @@ public class SchemaKey
                 }
             }
             else if (child->name == "summary")
-                summary = child->children->content;
+                summary = child->children == null ? "" : child->children->content;
             else if (child->name == "description")
-                description = child->children->content;
+                description = child->children == null ? "" : child->children->content;
             else if (child->name == "range")
                 range = new SchemaValueRange.from_xml(type, child);
             else if (child->name == "choices")
