diff -urP smplayer-0.6.9.org/src/findsubtitles/findsubtitles.pro smplayer-0.6.9/src/findsubtitles/findsubtitles.pro
--- smplayer-0.6.9.org/src/findsubtitles/findsubtitles.pro	2009-01-01 15:34:51.000000000 -0700
+++ smplayer-0.6.9/src/findsubtitles/findsubtitles.pro	2010-04-26 03:51:13.821648677 -0600
@@ -28,30 +28,14 @@
 DEFINES += NO_SMPLAYER_SUPPORT DOWNLOAD_SUBS
 
 contains( DEFINES, DOWNLOAD_SUBS ) {
-	INCLUDEPATH += filedownloader quazip
-	DEPENDPATH += filedownloader quazip
+	INCLUDEPATH += filedownloader 
+	DEPENDPATH += filedownloader 
 
 	HEADERS += filedownloader.h subchooserdialog.h
 	SOURCES += filedownloader.cpp subchooserdialog.cpp
 
 	FORMS += subchooserdialog.ui
 
-	HEADERS += crypt.h \
-	           ioapi.h \
-	           quazip.h \
-	           quazipfile.h \
-	           quazipfileinfo.h \
-	           quazipnewinfo.h \
-	           unzip.h \
-	           zip.h
-
-	SOURCES += ioapi.c \
-	           quazip.cpp \
-	           quazipfile.cpp \
-	           quazipnewinfo.cpp \
-	           unzip.c \
-	           zip.c
-
 	LIBS += -lz
 	
 	win32 {
--- smplayer-0.7.1/src/smplayer.pro.orig	2012-02-09 11:07:02.000000000 +0000
+++ smplayer-0.7.1/src/smplayer.pro	2012-03-24 01:57:08.950048887 +0000
@@ -256,31 +256,17 @@ TRANSLATIONS = translations/smplayer_es.
                translations/smplayer_da.ts translations/smplayer_hr.ts
 
 contains( DEFINES, DOWNLOAD_SUBS ) {
-	INCLUDEPATH += findsubtitles/filedownloader findsubtitles/quazip
-	DEPENDPATH += findsubtitles/filedownloader findsubtitles/quazip
+	INCLUDEPATH += findsubtitles/filedownloader
+	DEPENDPATH += findsubtitles/filedownloader
 
 	HEADERS += filedownloader.h subchooserdialog.h
 	SOURCES += filedownloader.cpp subchooserdialog.cpp
 
 	FORMS += subchooserdialog.ui
 
-	HEADERS += crypt.h \
-	           ioapi.h \
-	           quazip.h \
-	           quazipfile.h \
-	           quazipfileinfo.h \
-	           quazipnewinfo.h \
-	           unzip.h \
-	           zip.h
+	INCLUDEPATH += /usr/include/quazip /usr/include/minizip
 
-	SOURCES += ioapi.c \
-	           quazip.cpp \
-	           quazipfile.cpp \
-	           quazipnewinfo.cpp \
-	           unzip.c \
-	           zip.c
-
-	LIBS += -lz
+	LIBS += -lz -lquazip
 	
 	win32 {
 		INCLUDEPATH += ..\\zlib-1.2.6
