--- ./src/smplayer.pro.orig	2015-04-18 03:11:32.000000000 +0100
+++ ./src/smplayer.pro	2015-06-06 19:37:48.769799609 +0100
@@ -3,6 +3,7 @@ LANGUAGE = C++
 
 CONFIG += qt warn_on
 CONFIG += release
+CONFIG += qtsingleapplication
 #CONFIG += debug
 
 QT += network xml
@@ -283,14 +284,6 @@ FORMS = inputdvddirectory.ui logwindowba
         about.ui inputmplayerversion.ui errordialog.ui timedialog.ui stereo3ddialog.ui \
         toolbareditor.ui multilineinputdialog.ui
 
-# qtsingleapplication
-contains( DEFINES, SINGLE_INSTANCE ) {
-	INCLUDEPATH += qtsingleapplication
-	DEPENDPATH += qtsingleapplication
-
-	SOURCES += qtsingleapplication/qtsingleapplication.cpp qtsingleapplication/qtlocalpeer.cpp
-	HEADERS += qtsingleapplication/qtsingleapplication.h qtsingleapplication/qtlocalpeer.h
-}
 
 # Find subtitles dialog
 contains( DEFINES, FIND_SUBTITLES ) {
