diff -up gnome-utils-2.30.0/baobab/src/Makefile.am.linking gnome-utils-2.30.0/baobab/src/Makefile.am
--- gnome-utils-2.30.0/baobab/src/Makefile.am.linking	2010-06-19 22:53:12.119416772 -0400
+++ gnome-utils-2.30.0/baobab/src/Makefile.am	2010-06-19 22:53:32.177166948 -0400
@@ -45,7 +45,7 @@ baobab_CFLAGS = 			\
 	$(LIBGTOP_CFLAGS)		\
 	$(NULL)
 
-baobab_LDFLAGS = -export-dynamic
+baobab_LDFLAGS = -export-dynamic -Wl,-O1,--as-needed
 baobab_LDADD = 				\
 	-lm				\
 	$(GNOME_UTILS_LIBS)		\
diff -up gnome-utils-2.30.0/gnome-dictionary/src/Makefile.am.linking gnome-utils-2.30.0/gnome-dictionary/src/Makefile.am
--- gnome-utils-2.30.0/gnome-dictionary/src/Makefile.am.linking	2010-03-30 10:26:34.000000000 -0400
+++ gnome-utils-2.30.0/gnome-dictionary/src/Makefile.am	2010-06-19 22:46:23.894414468 -0400
@@ -45,6 +45,7 @@ gnome_dictionary_LDADD = \
 	$(top_builddir)/gnome-dictionary/libgdict/libgdict-1.0.la \
 	$(GNOME_UTILS_LIBS) \
 	$(NULL)
+gnome_dictionary_LDFLAGS = -Wl,-O1,--as-needed
 
 if BUILD_GDICT_APPLET
 
@@ -78,5 +79,6 @@ gnome_dictionary_applet_LDADD = \
 	$(GNOME_UTILS_LIBS) \
 	$(APPLET_LIBS) \
 	$(NULL)
+gnome_dictionary_applet_LDFLAGS = -Wl,-O1,--as-needed
 
 endif # BUILD_GDICT_APPLET
diff -up gnome-utils-2.30.0/gnome-screenshot/Makefile.am.linking gnome-utils-2.30.0/gnome-screenshot/Makefile.am
--- gnome-utils-2.30.0/gnome-screenshot/Makefile.am.linking	2010-06-19 22:54:00.422166808 -0400
+++ gnome-utils-2.30.0/gnome-screenshot/Makefile.am	2010-06-19 22:54:27.584419567 -0400
@@ -28,7 +28,7 @@ gnome_screenshot_SOURCES = \
 	screenshot-xfer.h		\
 	$(NULL)
 
-gnome_screenshot_LDFLAGS = -export-dynamic
+gnome_screenshot_LDFLAGS = -export-dynamic -Wl,-O1,--as-needed
 
 gnome_screenshot_LDADD =		\
 	$(XSHAPE_LIBS)			\
diff -up gnome-utils-2.30.0/gsearchtool/Makefile.am.linking gnome-utils-2.30.0/gsearchtool/Makefile.am
--- gnome-utils-2.30.0/gsearchtool/Makefile.am.linking	2010-06-19 22:54:45.207195443 -0400
+++ gnome-utils-2.30.0/gsearchtool/Makefile.am	2010-06-19 22:55:18.428423058 -0400
@@ -38,6 +38,7 @@ gnome_search_tool_LDADD =            \
 	$(GIO_UNIX_LIBS)             \
 	$(libgnomeui_deprecated_LIB) \
 	$(libeggsmclient_LIB)
+gnome_search_tool_LDFLAGS = -Wl,-O1,--as-needed
 
 man_MANS = gnome-search-tool.1
 
