--- caca-config.in.multilib	2010-02-08 10:34:28.000000000 -0200
+++ caca-config.in	2010-06-22 21:33:06.215851182 -0300
@@ -2,11 +2,15 @@
 
 ##  config script for libcaca -- Sam Hocevar <sam@hocevar.net>
 
-prefix=@prefix@
-exec_prefix=@exec_prefix@
+version=`pkg-config caca --modversion`
+prefix=`pkg-config caca --variable prefix`
+exec_prefix=`pkg-config caca --variable exec_prefix`
+
+lib_dir=`pkg-config caca --variable libdir`
+include_dir=`pkg-config caca --variable includedir`
+cflags=`pkg-config caca --cflags`
+libs=`pkg-config caca --libs`
 
-lib_dir=@libdir@
-include_dir=@includedir@
 
 usage()
 {
@@ -54,7 +58,7 @@
       echo_exec_prefix=yes
       ;;
     --version)
-      echo @VERSION@
+      echo $version
       exit 0
       ;;
     --cflags)
@@ -101,7 +105,6 @@
 
 if test "$echo_cflags" = "yes"
 then
-  cflags="-I$include_dir/"
   echo $cflags
 fi
 
@@ -113,6 +116,6 @@
 
 if test "$echo_libs" = "yes"
 then
-  echo -L@libdir@ $libs
+  echo $libs
 fi
 
