--- libjpeg-turbo-1.4.1/Makefile.am.orig	2015-06-16 19:23:23.866620985 +0100
+++ libjpeg-turbo-1.4.1/Makefile.am	2015-06-16 19:26:38.070227353 +0100
@@ -90,9 +90,7 @@
 
 if WITH_TURBOJPEG
 
-bin_PROGRAMS += tjbench
-
-noinst_PROGRAMS += tjunittest
+noinst_PROGRAMS += tjbench tjunittest
 
 tjbench_SOURCES = tjbench.c bmp.h bmp.c tjutil.h tjutil.c rdbmp.c rdppm.c \
 	wrbmp.c wrppm.c
@@ -158,13 +156,6 @@
 DOCS= coderules.txt jconfig.txt change.log rdrle.c wrrle.c BUILDING.txt \
 	ChangeLog.txt
 
-dist_doc_DATA = README README-turbo.txt libjpeg.txt structure.txt usage.txt \
-	wizard.txt
-
-exampledir = $(docdir)
-dist_example_DATA = example.c
-
-
 EXTRA_DIST = win release $(DOCS) testimages CMakeLists.txt \
 	sharedlib/CMakeLists.txt cmakescripts libjpeg.map.in doc doxygen.config \
 	doxygen-extra.css jccolext.c jdcolext.c jdcol565.c jdmrgext.c jdmrg565.c \
