diff --git a/jmorecfg.h b/jmorecfg.h
index 36fe971..1cec1e4 100644
--- a/jmorecfg.h
+++ b/jmorecfg.h
@@ -179,6 +179,22 @@ typedef unsigned int JDIMENSION;
 /* a reference to a GLOBAL function: */
 #define EXTERN(type)            extern type
 
+/* Originally, this macro was used as a way of defining function prototypes
+ * for both modern compilers as well as older compilers that did not support
+ * prototype parameters.  libjpeg-turbo has never supported these older,
+ * non-ANSI compilers, but the macro is still included because there is some
+ * software out there that uses it.
+ */
+
+#define JMETHOD(type,methodname,arglist)  type (*methodname) arglist
+
+
+/* libjpeg-turbo no longer supports platforms that have far symbols (MS-DOS),
+ * but again, some software relies on this macro.
+ */
+
+#undef FAR
+#define FAR
 
 /*
  * On a few systems, type boolean and/or its values FALSE, TRUE may appear
diff --git a/jpeglib.h b/jpeglib.h
index 27045a4..e8b8b45 100644
--- a/jpeglib.h
+++ b/jpeglib.h
@@ -874,6 +874,14 @@ struct jpeg_memory_mgr {
  */
 typedef boolean (*jpeg_marker_parser_method) (j_decompress_ptr cinfo);
 
+/* Originally, this macro was used as a way of defining function prototypes
+ * for both modern compilers as well as older compilers that did not support
+ * prototype parameters.  libjpeg-turbo has never supported these older,
+ * non-ANSI compilers, but the macro is still included because there is some
+ * software out there that uses it.
+ */
+
+#define JPP(arglist)    arglist
 
 /* Default error-management setup */
 EXTERN(struct jpeg_error_mgr *) jpeg_std_error (struct jpeg_error_mgr * err);
