diff -up libjpeg-turbo-1.3.0/Makefile.am.noinst libjpeg-turbo-1.3.0/Makefile.am
--- libjpeg-turbo-1.3.0/Makefile.am.noinst	2013-04-23 19:29:00.000000000 -0300
+++ libjpeg-turbo-1.3.0/Makefile.am	2013-05-26 15:13:01.017526254 -0300
@@ -89,9 +89,7 @@ noinst_PROGRAMS = jcstest
 
 if WITH_TURBOJPEG
 
-bin_PROGRAMS += tjbench
-
-noinst_PROGRAMS += tjunittest
+noinst_PROGRAMS += tjbench tjunittest
 
 tjbench_SOURCES = tjbench.c bmp.h bmp.c tjutil.h tjutil.c rdbmp.c rdppm.c \
 	wrbmp.c wrppm.c
@@ -144,14 +142,6 @@ dist_man1_MANS = cjpeg.1 djpeg.1 jpegtra
 DOCS= coderules.txt jconfig.txt change.log rdrle.c wrrle.c BUILDING.txt \
 	ChangeLog.txt
 
-docdir = $(datadir)/doc
-dist_doc_DATA = README README-turbo.txt libjpeg.txt structure.txt usage.txt \
-	wizard.txt 
-
-exampledir = $(datadir)/doc
-dist_example_DATA = example.c
-
-
 EXTRA_DIST = win release $(DOCS) testimages CMakeLists.txt \
 	sharedlib/CMakeLists.txt cmakescripts libjpeg.map.in doc doxygen.config \
 	jccolext.c jdcolext.c jdmrgext.c
