diff --git a/Makefile.am b/Makefile.am
index d767e4f..584d0c0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES = libjpeg.la
 libjpeg_la_LDFLAGS = -version-info ${LIBTOOL_CURRENT}:${SO_MINOR_VERSION}:${SO_AGE} -no-undefined
-include_HEADERS = jerror.h jmorecfg.h jpeglib.h
+include_HEADERS = jerror.h jmorecfg.h jpegint.h jconfig.h jpeglib.h
 
 if WITH_TURBOJPEG
 lib_LTLIBRARIES += libturbojpeg.la
