diff -up libjpeg-turbo-1.1.0/Makefile.am.noinst_jpgtest libjpeg-turbo-1.1.0/Makefile.am
--- libjpeg-turbo-1.1.0/Makefile.am.noinst_jpgtest	2011-03-01 14:40:04.291829489 +0100
+++ libjpeg-turbo-1.1.0/Makefile.am	2011-03-01 14:40:13.081814233 +0100
@@ -58,8 +58,8 @@ endif
 
 TSTHDRS = rrutil.h rrtimer.h
 
-bin_PROGRAMS = cjpeg djpeg jpegtran rdjpgcom wrjpgcom jpgtest
-noinst_PROGRAMS = jpegut
+bin_PROGRAMS = cjpeg djpeg jpegtran rdjpgcom wrjpgcom
+noinst_PROGRAMS = jpegut jpgtest
 
 jpgtest_SOURCES = $(TSTHDRS) jpgtest.c bmp.h bmp.c
 
