--- emelfm2-0.7.1.orig/Makefile	2009-12-21 03:50:47.000000000 +0100
+++ emelfm2-0.7.1/Makefile	2010-02-25 14:44:58.000000000 +0100
@@ -336,7 +336,7 @@
 
 $(TARGET): $(OBJECTS)
 	@echo "linking binary '$(TARGET)'"
-	@$(CC) $(LDFLAGS) $(OBJECTS) -o $(TARGET) -Wl,--as-needed $(lLIBS)
+	@$(CC) $(LDFLAGS) $(OBJECTS) -o $(TARGET) -lm -ldl -Wl,--as-needed $(lLIBS)
 ifeq ($(STRIP), 1)
 ifneq ($(DEBUG), 1)
 	@echo "stripping binary '$(TARGET)'"
