Index: src/dialogs/e2_view_dialog.c
===================================================================
--- src/dialogs/e2_view_dialog.c	(Revision 2489)
+++ src/dialogs/e2_view_dialog.c	(Revision 2490)
@@ -860,8 +860,12 @@
 	rt->textbuffer = gtk_text_buffer_new (NULL); //always create a buffer
 	if (contents)
 	{
+		GtkTextIter iter;
+		g_free (contents);
 		gtk_text_buffer_set_text (rt->textbuffer, contents, -1);
-		g_free (contents);
+		//cursor normally set to end of buffer
+		gtk_text_buffer_get_start_iter (rt->textbuffer, &iter);
+        gtk_text_buffer_place_cursor (rt->textbuffer, &iter);
 		printd (DEBUG, "Read text file charset is %s", rt->charset);
 	}
 
