diff -ur kdebindings-4.4.92/python/pykde4/sip/akonadi/addressattribute.sip kdebindings-4.4.92-sip4103/python/pykde4/sip/akonadi/addressattribute.sip
--- kdebindings-4.4.92/python/pykde4/sip/akonadi/addressattribute.sip	2010-05-27 19:21:46.000000000 +0200
+++ kdebindings-4.4.92-sip4103/python/pykde4/sip/akonadi/addressattribute.sip	2010-07-14 23:07:51.000000000 +0200
@@ -59,6 +59,11 @@
     else if (dynamic_cast<Akonadi::PersistentSearchAttribute*>(sipCpp))
         sipType = sipType_Akonadi_PersistentSearchAttribute;
 %End
+
+private:
+//force
+    Akonadi::AddressAttribute& operator=(const Akonadi::AddressAttribute&);
+//end
 };
 };
 %ModuleHeaderCode
