diff -up kdebindings-4.5.1/ruby/CMakeLists.txt.RUBY_VERSION kdebindings-4.5.1/ruby/CMakeLists.txt
--- kdebindings-4.5.1/ruby/CMakeLists.txt.RUBY_VERSION	2010-08-27 03:11:01.000000000 -0500
+++ kdebindings-4.5.1/ruby/CMakeLists.txt	2010-08-28 23:30:45.517927327 -0500
@@ -74,9 +74,9 @@ if (RUBY_EXECUTABLE AND RUBY_LIBRARY AND
             #include <ruby.h>
             int main() { ruby_init(); return 0; }" RubyCompiles)
         if(RubyCompiles)
-            if(ENABLE_KROSSRUBY AND RUBY_VERSION_MAJOR = 1 AND RUBY_VERSION_MINOR < 9)
+            if(ENABLE_KROSSRUBY AND RUBY_VERSION_MAJOR EQUAL 1 AND RUBY_VERSION_MINOR LESS 9)
                 set(BUILD_KrossRuby TRUE)
-            endif(ENABLE_KROSSRUBY AND RUBY_VERSION_MAJOR = 1 AND RUBY_VERSION_MINOR < 9)
+            endif(ENABLE_KROSSRUBY AND RUBY_VERSION_MAJOR EQUAL 1 AND RUBY_VERSION_MINOR LESS 9)
         else(RubyCompiles)
             message(STATUS "Failed to compile a C++ program that embeds ruby.")
         endif(RubyCompiles)
