Index: smoke/nepomukquery/nepomukquery_includes.h
===================================================================
--- smoke/nepomukquery/nepomukquery_includes.h
+++ smoke/nepomukquery/nepomukquery_includes.h	2010-05-20 17:35:27.000000000 +0200
@@ -1,6 +1,7 @@
 #include <nepomuk/term.h>
 #include <nepomuk/negationterm.h>
 #include <nepomuk/orterm.h>
+#include <nepomuk/optionalterm.h>
 #include <nepomuk/andterm.h>
 #include <nepomuk/comparisonterm.h>
 #include <nepomuk/simpleterm.h>
@@ -13,3 +14,5 @@
 #include <nepomuk/result.h>
 #include <nepomuk/queryserviceclient.h>
 #include <nepomuk/queryparser.h>
+
+#include <nepomuk/variant.h>
Index: smoke/nepomukquery/smokeconfig.xml
===================================================================
--- smoke/nepomukquery/smokeconfig.xml
+++ smoke/nepomukquery/smokeconfig.xml	2010-05-20 17:36:07.000000000 +0200
@@ -42,6 +42,7 @@
         <class>Nepomuk::Query::GroupTerm</class>
         <class>Nepomuk::Query::LiteralTerm</class>
         <class>Nepomuk::Query::NegationTerm</class>
+        <class>Nepomuk::Query::OptionalTerm</class>
         <class>Nepomuk::Query::OrTerm</class>
         <class>Nepomuk::Query::Query</class>
         <class>Nepomuk::Query::QueryParser</class>
