--- app/cms.c	2010-07-26 08:54:44.024451199 +0200
+++ app/cms.c	2010-07-25 21:57:19.196512212 +0200
@@ -37,15 +37,20 @@
 #endif
 
 #ifdef HAVE_OY
-#include <oyranos/oyranos.h>
 #ifndef OYRANOS_VERSION
 #define OYRANOS_VERSION 0
 #endif
 #if OYRANOS_VERSION < 108
+#include <oyranos/oyranos.h>
 #include <arpa/inet.h>  /* ntohl */
 #include <oyranos/oyranos_monitor.h>
 #else
-#include <oyranos/oyranos_alpha.h>
+#include <oyranos.h>
+#if OYRANOS_VERSION < 110
+#include <oyranos_alpha.h>
+#else
+#include <alpha/oyranos_alpha.h>
+#endif
 #endif
 #endif
 
--- app/main.c	2010-07-26 08:54:44.050451458 +0200
+++ app/main.c	2010-07-25 21:08:48.700385102 +0200
@@ -54,9 +54,15 @@
 #endif   /*  WAIT_ANY  */
 
 #ifdef HAVE_OY
+# if OYRANOS_VERSION > 107
+#include <oyranos.h>
+#include <oyranos_config.h>
+#include <oyranos_version.h>
+# else
 #include <oyranos/oyranos.h>
 #include <oyranos/oyranos_config.h>
 #include <oyranos/oyranos_version.h>
+# endif
 char *oyranos_temp_path = NULL;
 void *myAlloc(size_t n) { return calloc(sizeof(char), n); }
 #endif
