diff -up icc_examin-0.46/makefile.in.dso icc_examin-0.46/makefile.in
--- icc_examin-0.46/makefile.in.dso	2010-06-23 13:56:26.000000000 +0200
+++ icc_examin-0.46/makefile.in	2010-06-23 13:56:38.000000000 +0200
@@ -37,7 +37,7 @@ ALL_INCL = $(INCL) \
 CFLAGS_ = $(CFLAGS) $(DEBUG)
 CXXFLAGS_ = $(CXXFLAGS) $(INCL_) $(DEBUG)
 
-LDLIBS = $(LDFLAGS) -L$(libdir) -L./ $(LIBFL_I18N) $(FLTK_LIBS) \
+LDLIBS = $(LDFLAGS) -lpthread -L$(libdir) -L./ $(LIBFL_I18N) $(FLTK_LIBS) \
 	$(Flmm_OBJECTS) $(X11_LIBS) -L/lib \
 	$(LIBOYRANOS_EXTRA) $(OYRANOS_LIBS) $(LCMS_LIBS) \
 	$(FTGL_LIBS) $(FONTCONFIG_LIBS) $(I18N_LIB) $(DBG_LIBS)
