--- src/vfs/vfs-file-task.c.orig	2013-10-22 17:57:09.462045687 -0200
+++ src/vfs/vfs-file-task.c	2013-10-22 18:00:17.208195447 -0200
@@ -1765,15 +1765,7 @@ static void vfs_file_task_exec( char* sr
             argv[a++] = g_strdup( task->exec_as_user );
         }
 
-        if ( !strcmp( use_su, "/usr/bin/gksu" ) || !strcmp( use_su, "/usr/bin/gksudo" ) )
-        {
-            // gksu*
-            argv[a++] = g_strdup( "-g" );
-            argv[a++] = g_strdup( "-D" );
-            argv[a++] = g_strdup( "SpaceFM Command" );
-            single_arg = TRUE;
-        }
-        else if ( strstr( use_su, "kdesu" ) )
+        if ( strstr( use_su, "kdesu" ) )
         {
             // kdesu kdesudo
             argv[a++] = g_strdup( "-d" );
