--- pango-1.31.2/pango/modules.c.orig	2012-09-24 22:54:58.676027615 -0300
+++ pango-1.31.2/pango/modules.c	2012-09-24 23:01:29.951483235 -0300
@@ -548,7 +548,7 @@ read_modules (void)
     {
       files = g_new (char *, 3);
 
-      files[0] = g_build_filename (pango_get_sysconf_subdirectory (),
+      files[0] = g_build_filename (pango_get_lib_subdirectory (),
                                    "pango.modules",
                                    NULL);
       files[1] = g_build_filename (pango_get_lib_subdirectory (),
@@ -667,7 +667,7 @@ build_map (PangoMapInfo *info)
       static gboolean no_module_warning = FALSE; /* MT-safe */
       if (!no_module_warning)
 	{
-	  gchar *filename = g_build_filename (pango_get_sysconf_subdirectory (),
+	  gchar *filename = g_build_filename (pango_get_lib_subdirectory (),
 					      "pango.modules",
 					      NULL);
 	  g_critical ("No modules found:\n"
