diff -up src/conflict.c.cool src/conflict.c
--- src/conflict.c.cool	2009-10-19 17:06:34.000000000 -0200
+++ src/conflict.c	2013-02-03 16:51:08.056411015 -0200
@@ -457,7 +457,7 @@ prelink_build_conflicts (struct prelink_
   for (i = 1; i < ndeps; ++i)
     {
       ent = info->ent->depends[i - 1];
-      if ((dso = open_dso (ent->filename)) == NULL)
+      if ((dso = open_dso (ent->canon_filename)) == NULL)
 	goto error_out;
       info->dsos[i] = dso;
       /* Now check that the DSO matches what we recorded about it.  */
@@ -623,8 +623,8 @@ prelink_build_conflicts (struct prelink_
 		  && dso->phdr[k].p_vaddr + dso->phdr[k].p_memsz
 		     > info->conflict_rela[j].r_offset)
 		{
-		  error (0, 0, "%s: Cannot prelink against non-PIC shared library %s",
-			 info->dso->filename, dso->filename);
+		  error (0, 0, "%s: shared library %s appears possibly non-PIC and contains conflicts. Symbol offset: %lx",
+			info->dso->filename, dso->filename, (long)info->conflict_rela[j].r_offset);
 		  goto error_out;
 		}
 	}
diff -up src/dso.c.cool src/dso.c
--- src/dso.c.cool	2010-11-22 16:12:56.000000000 -0200
+++ src/dso.c	2013-02-03 16:51:08.057410804 -0200
@@ -214,8 +214,11 @@ check_dso (DSO *dso)
 	      || RELOCATE_SCN (dso->shdr[last].sh_flags)
 	      || RELOCATE_SCN (dso->shdr[i].sh_flags))
 	    {
-	      error (0, 0, "%s: section file offsets not monotonically increasing",
-		     dso->filename);
+	      error (0, 0, "%s: section file offsets not monotonically increasing (%s before %s)",
+                     dso->filename,
+                     strptr (dso, dso->ehdr.e_shstrndx, dso->shdr[i-1].sh_name),
+                     strptr (dso, dso->ehdr.e_shstrndx, dso->shdr[i].sh_name));
+
 	      return 1;
 	    }
 	}
diff -up src/gather.c.cool src/gather.c
--- src/gather.c.cool	2011-08-25 16:02:05.000000000 -0300
+++ src/gather.c	2013-02-03 16:51:08.056411015 -0200
@@ -487,7 +487,7 @@ gather_lib (struct prelink_entry *ent)
   DSO *dso;
 
   ent->type = ET_BAD;
-  dso = open_dso (ent->filename);
+  dso = open_dso (ent->canon_filename);
   if (dso == NULL)
     return 1;
 
diff -up src/layout.c.cool src/layout.c
--- src/layout.c.cool	2011-10-12 12:37:36.000000000 -0300
+++ src/layout.c	2013-02-03 16:51:08.056411015 -0200
@@ -642,7 +642,8 @@ not_found:
 		< ((deps[j - 1]->end + max_page_size - 1)
 		   & ~(max_page_size - 1))
 		&& (deps[j]->type == ET_DYN || deps[j - 1]->type == ET_DYN))
-	      abort ();
+		error(EXIT_FAILURE, 0, "The sorted library entries for %s and %s do overlap", 
+		      deps[j-1]->canon_filename, deps[j]->canon_filename);
 	}
 #endif
     }
diff -up src/space.c.cool src/space.c
--- src/space.c.cool	2007-10-08 12:49:15.000000000 -0300
+++ src/space.c	2013-02-03 16:51:08.056411015 -0200
@@ -351,6 +351,8 @@ find_readonly_space (DSO *dso, GElf_Shdr
       shdr[j] = *add;
       shdr[j].sh_addr = (shdr[j - 1].sh_addr + shdr[j - 1].sh_size
 			 + add->sh_addralign - 1) & ~(add->sh_addralign - 1);
+      if (shdr[j].sh_addr < phdr[i].p_vaddr)
+	shdr[j].sh_addr = phdr[i].p_vaddr;
       shdr[j].sh_offset = (shdr[j].sh_addr - phdr[i].p_vaddr)
 			  + phdr[i].p_offset;
       phdr[i].p_filesz = shdr[j].sh_addr + add->sh_size - phdr[i].p_vaddr;
