--- valgrind-3.9.0/glibc-2.X.supp.in.orig	2013-11-24 20:39:55.165152689 +0100
+++ valgrind-3.9.0/glibc-2.X.supp.in	2013-11-24 20:46:04.919736611 +0100
@@ -236,3 +236,11 @@
    Memcheck:Cond
    fun:_dl_relocate_object
 }
+
+# KDE#327943 - s390x missing index/strchr suppression for ld.so bad backtrace?
+{
+   s390x-ld-so-index-dl_open_worker
+   Memcheck:Addr8
+   fun:index
+   fun:dl_open_worker
+}
