--- valgrind/configure.in.jj	2010-10-21 03:14:02.000000000 +0200
+++ valgrind/configure.in	2010-11-05 17:09:18.674455299 +0100
@@ -649,6 +649,16 @@ _ACEOF
   GLIBC_VERSION="`$CPP conftest.$ac_ext | $SED -n 's/^glibc version is: //p' | $SED 's/ /./g'`"
 fi
 
+AC_EGREP_CPP([GLIBC_213], [
+#include <features.h>
+#ifdef __GNU_LIBRARY__
+ #if (__GLIBC__ == 2 && __GLIBC_MINOR__ == 13)
+  GLIBC_213
+ #endif
+#endif
+],
+GLIBC_VERSION="2.13")
+
 AC_EGREP_CPP([AIX5_LIBC], [
 #include <standards.h>
 #if defined(_AIXVERSION_510) || defined(_AIXVERSION_520) || defined(_AIXVERSION_530)
@@ -749,6 +759,13 @@ case "${GLIBC_VERSION}" in
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.13)
+	AC_MSG_RESULT(2.13 family)
+	AC_DEFINE([GLIBC_2_13], 1, [Define to 1 if you're using glibc 2.13.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      aix5)
 	AC_MSG_RESULT(AIX 5.1 or 5.2 or 5.3)
 	AC_DEFINE([AIX5_LIBC], 1, [Define to 1 if you're using AIX 5.1 or 5.2 or 5.3])
@@ -762,7 +779,7 @@ case "${GLIBC_VERSION}" in
 
      *)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.12])
+	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.13])
 	AC_MSG_ERROR([or AIX 5.1 or 5.2 or 5.3 GLIBC_VERSION])
 	AC_MSG_ERROR([or Darwin libc])
 	;;
--- valgrind/config.h.in.jj	2010-10-20 22:20:49.000000000 +0200
+++ valgrind/config.h.in	2010-11-05 17:08:33.863454886 +0100
@@ -33,6 +33,9 @@
 /* Define to 1 if you're using glibc 2.12.x */
 #undef GLIBC_2_12
 
+/* Define to 1 if you're using glibc 2.13.x */
+#undef GLIBC_2_13
+
 /* Define to 1 if you're using glibc 2.2.x */
 #undef GLIBC_2_2
 
--- valgrind/configure.jj	2010-10-21 03:16:18.000000000 +0200
+++ valgrind/configure	2010-11-05 17:09:39.163455396 +0100
@@ -6220,6 +6220,28 @@ fi
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <features.h>
+#ifdef __GNU_LIBRARY__
+ #if (__GLIBC__ == 2 && __GLIBC_MINOR__ == 13)
+  GLIBC_213
+ #endif
+#endif
+
+_ACEOF
+if (eval "$ac_cpp conftest.$ac_ext") 2>&5 |
+  $EGREP "GLIBC_213" >/dev/null 2>&1; then
+  GLIBC_VERSION="2.13"
+fi
+rm -f conftest*
+
+
+cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+
 #include <standards.h>
 #if defined(_AIXVERSION_510) || defined(_AIXVERSION_520) || defined(_AIXVERSION_530)
   AIX5_LIBC
@@ -6367,6 +6389,18 @@ $as_echo "#define GLIBC_2_12 1" >>confde
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.13)
+	echo "$as_me:$LINENO: result: 2.13 family" >&5
+echo "${ECHO_T}2.13 family" >&6
+
+cat >>confdefs.h <<\_ACEOF
+#define GLIBC_2_13 1
+_ACEOF
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      aix5)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: AIX 5.1 or 5.2 or 5.3" >&5
 $as_echo "AIX 5.1 or 5.2 or 5.3" >&6; }
@@ -6387,7 +6421,7 @@ $as_echo "#define DARWIN_LIBC 1" >>confd
      *)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: unsupported version ${GLIBC_VERSION}" >&5
 $as_echo "unsupported version ${GLIBC_VERSION}" >&6; }
-	as_fn_error "Valgrind requires glibc version 2.2 - 2.12" "$LINENO" 5
+	as_fn_error "Valgrind requires glibc version 2.2 - 2.13" "$LINENO" 5
 	as_fn_error "or AIX 5.1 or 5.2 or 5.3 GLIBC_VERSION" "$LINENO" 5
 	as_fn_error "or Darwin libc" "$LINENO" 5
 	;;
