--- valgrind-3.7.0/coregrind/m_debuginfo/readdwarf3.c	(revision 12408)
+++ valgrind-3.7.0/coregrind/m_debuginfo/readdwarf3.c	(revision 12409)
@@ -2547,6 +2547,7 @@ static void parse_type_DIE ( /*MOD*/XArr
             const members in C++ code which are compile time constants
             that do no exist in the class. They're not of any interest
             to us so we ignore them. */
+         ML_(TyEnt__make_EMPTY)(&fieldE);
       }
    }
 
