Index: valgrind/coregrind/m_aspacemgr/aspacemgr-linux.c
===================================================================
--- valgrind/coregrind/m_aspacemgr/aspacemgr-linux.c	(revision 13043)
+++ valgrind/coregrind/m_aspacemgr/aspacemgr-linux.c	(working copy)
@@ -265,10 +265,10 @@
 /* ------ start of STATE for the address-space manager ------ */
 
 /* Max number of segments we can track. */
-#define VG_N_SEGMENTS 5000
+#define VG_N_SEGMENTS 50000
 
 /* Max number of segment file names we can track. */
-#define VG_N_SEGNAMES 1000
+#define VG_N_SEGNAMES 25000
 
 /* Max length of a segment file name. */
 #define VG_MAX_SEGNAMELEN 1000
