--- valgrind/VEX/priv/guest_amd64_toIR.c.jj	2012-10-16 09:53:54.000000000 +0200
+++ valgrind/VEX/priv/guest_amd64_toIR.c	2012-10-16 11:06:26.007515653 +0200
@@ -20117,6 +20117,14 @@ Long dis_ESC_0F (
       delta = dis_movx_E_G ( vbi, pfx, delta, 2, sz, True );
       return delta;
 
+   case 0xC0: { /* XADD Gb,Eb */ 
+      Bool decode_OK = False;
+      delta = dis_xadd_G_E ( &decode_OK, vbi, pfx, 1, delta );
+      if (!decode_OK)
+         goto decode_failure;
+      return delta;
+   }
+
    case 0xC1: { /* XADD Gv,Ev */ 
       Bool decode_OK = False;
       delta = dis_xadd_G_E ( &decode_OK, vbi, pfx, sz, delta );
@@ -26992,14 +27000,6 @@ DisResult disInstr_AMD64_WRK (
 
       /* =-=-=-=-=-=-=-=-=- XADD -=-=-=-=-=-=-=-=-=-= */
 
-      case 0xC0: { /* XADD Gb,Eb */ 
-         Bool decode_OK = False;
-         delta = dis_xadd_G_E ( &decode_OK, vbi, pfx, 1, delta );
-         if (!decode_OK)
-            goto decode_failure;
-         break;
-      }
-
       /* =-=-=-=-=-=-=-=-=- SGDT and SIDT =-=-=-=-=-=-=-=-=-=-= */
 
       /* =-=-=-=-=-=-=-=-=- unimp2 =-=-=-=-=-=-=-=-=-=-= */
