--- valgrind/VEX/priv/guest_s390_cc.c.jj	2010-11-11 16:55:05.000000000 +0100
+++ valgrind/VEX/priv/guest_s390_cc.c	2010-11-11 17:38:10.786580286 +0100
@@ -39,8 +39,6 @@
 #include "guest_s390_priv.h"         /* irsb */
 #include "guest_s390_cc.h"
 
-#if defined(VGA_s390x)   /* guard the file */
-
 static UInt s390_calculate_cc(ULong cc_op, ULong cc_dep1, ULong cc_dep2,
                               ULong cc_ndep);
 
@@ -418,6 +416,7 @@ s390_calculate_cc(ULong cc_op, ULong cc_
 {
    UInt psw;
 
+#if defined(VGA_s390x)
    switch (cc_op) {
 
    case S390_CC_OP_BITWISE:
@@ -611,6 +610,7 @@ s390_calculate_cc(ULong cc_op, ULong cc_
    default:
       break;
    }
+#endif
    vpanic("s390_calculate_cc");
 }
 
@@ -680,8 +680,6 @@ s390_call_calculate_cond(UInt m)
    return call;
 }
 
-#endif /* VGA_s390x */
-
 /*---------------------------------------------------------------*/
 /*--- end                                     guest_s390_cc.c ---*/
 /*---------------------------------------------------------------*/
