Disable some hanging tests, fix up bug125959-x86.c undefined behavior.

--- valgrind-3.7.0/helgrind/tests/Makefile.am.jj	2011-10-26 23:24:25.000000000 +0200
+++ valgrind-3.7.0/helgrind/tests/Makefile.am	2012-01-27 15:01:58.955624195 +0100
@@ -94,7 +94,6 @@ EXTRA_DIST = \
 # XXX: tc18_semabuse uses operations that are unsupported on Darwin.  It
 # should be conditionally compiled like tc20_verifywrap is.
 check_PROGRAMS = \
-	annotate_hbefore \
 	cond_timedwait_invalid \
 	free_is_write \
 	hg01_all_ok \
--- valgrind-3.7.0/helgrind/tests/Makefile.in.jj	2011-11-05 12:26:51.000000000 +0100
+++ valgrind-3.7.0/helgrind/tests/Makefile.in	2012-01-27 15:02:40.285383038 +0100
@@ -46,8 +46,7 @@ host_triplet = @host@
 DIST_COMMON = $(dist_noinst_SCRIPTS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(top_srcdir)/Makefile.all.am \
 	$(top_srcdir)/Makefile.tool-tests.am
-check_PROGRAMS = annotate_hbefore$(EXEEXT) \
-	cond_timedwait_invalid$(EXEEXT) free_is_write$(EXEEXT) \
+check_PROGRAMS = cond_timedwait_invalid$(EXEEXT) free_is_write$(EXEEXT) \
 	hg01_all_ok$(EXEEXT) hg02_deadlock$(EXEEXT) \
 	hg03_inherit$(EXEEXT) hg04_race$(EXEEXT) hg05_race2$(EXEEXT) \
 	hg06_readshared$(EXEEXT) locked_vs_unlocked1$(EXEEXT) \
--- valgrind-3.7.0/none/tests/x86/bug125959-x86.c.jj	2011-10-26 23:24:38.000000000 +0200
+++ valgrind-3.7.0/none/tests/x86/bug125959-x86.c	2012-01-27 15:00:11.648250429 +0100
@@ -22,7 +22,8 @@ static int lookup ( int i )
 
 int main(void) 
 { 
-  int i, j;
+  int i;
+  unsigned int j;
    for (i = 0; i < 256; i++) 
     buf[i] = (unsigned char) i;
 
--- valgrind-3.7.0/gdbserver_tests/Makefile.am.jj	2011-10-26 23:24:26.000000000 +0200
+++ valgrind-3.7.0/gdbserver_tests/Makefile.am	2012-01-27 14:56:43.369465872 +0100
@@ -102,9 +102,7 @@ check_PROGRAMS = \
 	clean_after_fork \
 	fork_chain \
 	passsigalrm \
-	sleepers \
 	main_pic \
-	t \
 	watchpoints
 
 AM_CFLAGS   += $(AM_FLAG_M3264_PRI)
--- valgrind-3.7.0/gdbserver_tests/Makefile.in.jj	2011-11-05 12:26:51.000000000 +0100
+++ valgrind-3.7.0/gdbserver_tests/Makefile.in	2012-01-27 14:57:30.278191493 +0100
@@ -47,8 +47,7 @@ DIST_COMMON = $(dist_noinst_SCRIPTS) $(s
 	$(srcdir)/Makefile.in $(top_srcdir)/Makefile.all.am \
 	$(top_srcdir)/Makefile.tool-tests.am
 check_PROGRAMS = clean_after_fork$(EXEEXT) fork_chain$(EXEEXT) \
-	passsigalrm$(EXEEXT) sleepers$(EXEEXT) main_pic$(EXEEXT) \
-	t$(EXEEXT) watchpoints$(EXEEXT)
+	passsigalrm$(EXEEXT) main_pic$(EXEEXT) watchpoints$(EXEEXT)
 subdir = gdbserver_tests
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/configure.in
