--- valgrind-3.8.1/glibc-2.X.supp.in.orig	2012-10-15 14:10:24.646984028 +0200
+++ valgrind-3.8.1/glibc-2.X.supp.in	2012-10-15 14:13:58.505147375 +0200
@@ -213,6 +213,16 @@
    obj:/*libc-@GLIBC_VERSION@*.so
 }
 
+## ---------------------------------------------------------------------##
+# Workaround for Bug 308427 - s390 memcheck reports tsearch conditional
+# jump or move depends on uninitialized value.
+{
+   s390x-tsearch-node_t-red
+   Memcheck:Cond
+   fun:tsearch
+   obj:/*libc-@GLIBC_VERSION@*.so
+}
+
 ##----------------------------------------------------------------------##
 # MontaVista Linux 4.0.1 on ppc32
 {
