diff -r -u valgrind-3.7.0.old/configure valgrind-3.7.0/configure
--- valgrind-3.7.0.old/configure	2012-03-05 15:16:23.000000000 -0500
+++ valgrind-3.7.0/configure	2012-03-05 13:49:04.000000000 -0500
@@ -5279,7 +5279,7 @@
         ARCH_MAX="s390x"
         ;;
 
-     armv7*)
+     armv[57]*)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: ok (${host_cpu})" >&5
 $as_echo "ok (${host_cpu})" >&6; }
 	ARCH_MAX="arm"
