--- valgrind/none/tests/amd64/gen_insn_test.pl.jj	2010-10-20 22:19:31.000000000 +0200
+++ valgrind/none/tests/amd64/gen_insn_test.pl	2011-02-22 18:36:34.767401963 +0100
@@ -628,7 +628,7 @@ while (<>)
 
     foreach my $result (@results)
     {
-        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w)$/)
+        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w|r(8|16|32|64)|mm|xmm)$/)
         {
             $result->{argnum} = $argnum++;
         }
@@ -642,14 +642,6 @@ while (<>)
         }
     }
     
-    foreach my $result (@results)
-    {
-        if ($result->{type} =~ /^(r(8|16|32|64)|mm|xmm)$/)
-        {
-            $result->{argnum} = $argnum++;
-        }
-    }
-
     my $stateargnum = $argnum++;
 
     print qq|   char state\[108\];\n|;
@@ -689,8 +681,8 @@ while (<>)
         }
         elsif ($arg->{type} eq "xmm")
         {
-            print qq|         \"movlps 0+%$arg->{argnum}, %%$arg->{register}\\n\"\n|;
-            print qq|         \"movhps 8+%$arg->{argnum}, %%$arg->{register}\\n\"\n|;
+            print qq|         \"movlps %$arg->{argnum}, %%$arg->{register}\\n\"\n|;
+            print qq|         \"movhps %H$arg->{argnum}, %%$arg->{register}\\n\"\n|;
         }
         elsif ($arg->{type} eq "st")
         {
@@ -803,8 +795,8 @@ while (<>)
         }
         elsif ($result->{type} eq "xmm")
         {
-            print qq|         \"movlps %%$result->{register}, 0+%$result->{argnum}\\n\"\n|;
-            print qq|         \"movhps %%$result->{register}, 8+%$result->{argnum}\\n\"\n|;
+            print qq|         \"movlps %%$result->{register}, %$result->{argnum}\\n\"\n|;
+            print qq|         \"movhps %%$result->{register}, %H$result->{argnum}\\n\"\n|;
         }
         elsif ($result->{type} eq "st")
         {
@@ -852,11 +844,16 @@ while (<>)
 
     foreach my $result (@results)
     {
-        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w)$/)
+        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w|r(8|16|32|64)|mm)$/)
         {
             print qq|$prefix\"=m\" \($result->{name}\)|;
             $prefix = ", ";
         }
+        elsif ($result->{type} eq "xmm")
+        {
+            print qq|$prefix\"=o\" \($result->{name}\)|;
+            $prefix = ", ";
+        }
     }
 
     print qq|\n|;
@@ -867,20 +864,18 @@ while (<>)
     {
         if (defined($arg->{name}))
         {
-            print qq|$prefix\"m\" \($arg->{name}\)|;
+	    if ($arg->{type} eq "xmm")
+	    {
+		print qq|$prefix\"o\" \($arg->{name}\)|;
+	    }
+	    else
+	    {
+		print qq|$prefix\"m\" \($arg->{name}\)|;
+	    }
             $prefix = ", ";
         }
     }
     
-    foreach my $result (@results)
-    {
-        if ($result->{type} =~ /^(r(8|16|32|64)|mm|xmm)$/)
-        {
-            print qq|$prefix\"m\" \($result->{name}\)|;
-            $prefix = ", ";
-        }
-    }
-
     print qq|$prefix\"m\" \(state[0]\)\n|;
 
     $prefix = "         : ";
--- valgrind/none/tests/x86/gen_insn_test.pl.jj	2010-10-20 22:19:35.000000000 +0200
+++ valgrind/none/tests/x86/gen_insn_test.pl	2011-02-22 18:30:27.750683267 +0100
@@ -605,7 +605,7 @@ while (<>)
 
     foreach my $result (@results)
     {
-        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w)$/)
+        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w|r(8|16|32)|mm|xmm)$/)
         {
             $result->{argnum} = $argnum++;
         }
@@ -619,14 +619,6 @@ while (<>)
         }
     }
     
-    foreach my $result (@results)
-    {
-        if ($result->{type} =~ /^(r(8|16|32)|mm|xmm)$/)
-        {
-            $result->{argnum} = $argnum++;
-        }
-    }
-
     my $stateargnum = $argnum++;
 
     print qq|   char state\[108\];\n|;
@@ -658,8 +650,8 @@ while (<>)
         }
         elsif ($arg->{type} eq "xmm")
         {
-            print qq|         \"movlps 0%$arg->{argnum}, %%$arg->{register}\\n\"\n|;
-            print qq|         \"movhps 8%$arg->{argnum}, %%$arg->{register}\\n\"\n|;
+            print qq|         \"movlps %$arg->{argnum}, %%$arg->{register}\\n\"\n|;
+            print qq|         \"movhps %H$arg->{argnum}, %%$arg->{register}\\n\"\n|;
         }
         elsif ($arg->{type} eq "st")
         {
@@ -767,8 +759,8 @@ while (<>)
         }
         elsif ($result->{type} eq "xmm")
         {
-            print qq|         \"movlps %%$result->{register}, 0%$result->{argnum}\\n\"\n|;
-            print qq|         \"movhps %%$result->{register}, 8%$result->{argnum}\\n\"\n|;
+            print qq|         \"movlps %%$result->{register}, %$result->{argnum}\\n\"\n|;
+            print qq|         \"movhps %%$result->{register}, %H$result->{argnum}\\n\"\n|;
         }
         elsif ($result->{type} eq "st")
         {
@@ -817,11 +809,16 @@ while (<>)
 
     foreach my $result (@results)
     {
-        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w)$/)
+        if ($result->{type} =~ /^(m(8|16|32|64|128)|st|eflags|fpu[cs]w|r(8|16|32)|mm)$/)
         {
             print qq|$prefix\"=m\" \($result->{name}\)|;
             $prefix = ", ";
         }
+        elsif ($result->{type} eq "xmm")
+        {
+            print qq|$prefix\"=o\" \($result->{name}\)|;
+            $prefix = ", ";
+        }
     }
 
     print qq|\n|;
@@ -832,20 +829,18 @@ while (<>)
     {
         if (defined($arg->{name}))
         {
-            print qq|$prefix\"m\" \($arg->{name}\)|;
+	    if ($arg->{type} eq "xmm")
+	    {
+		print qq|$prefix\"o\" \($arg->{name}\)|;
+            }
+	    else
+	    {
+		print qq|$prefix\"m\" \($arg->{name}\)|;
+            }
             $prefix = ", ";
         }
     }
     
-    foreach my $result (@results)
-    {
-        if ($result->{type} =~ /^(r(8|16|32)|mm|xmm)$/)
-        {
-            print qq|$prefix\"m\" \($result->{name}\)|;
-            $prefix = ", ";
-        }
-    }
-
     print qq|$prefix\"m\" \(state[0]\)\n|;
 
     $prefix = "         : ";
