--- valgrind-3.9.0/VEX/priv/guest_s390_toIR.c.orig	2013-11-07 22:45:58.021660641 +0100
+++ valgrind-3.9.0/VEX/priv/guest_s390_toIR.c	2013-11-07 22:46:12.319772145 +0100
@@ -7606,7 +7606,7 @@
       put_gpr_dw0(r1, binop(Iop_And64, mkexpr(op2), mkU64(mask)));
    }
    assign(result, get_gpr_dw0(r1));
-   s390_cc_thunk_putS(S390_CC_OP_LOAD_AND_TEST, op2);
+   s390_cc_thunk_putS(S390_CC_OP_LOAD_AND_TEST, result);
 
    return "risbg";
 }
