--- valgrind-3.7.0/configure.in.jj	2011-11-05 12:13:30.000000000 +0100
+++ valgrind-3.7.0/configure.in	2011-12-09 17:48:21.409848842 +0100
@@ -775,6 +775,13 @@ case "${GLIBC_VERSION}" in
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.15)
+	AC_MSG_RESULT(2.15 family)
+	AC_DEFINE([GLIBC_2_15], 1, [Define to 1 if you're using glibc 2.15.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	AC_MSG_RESULT(Darwin)
 	AC_DEFINE([DARWIN_LIBC], 1, [Define to 1 if you're using Darwin])
@@ -788,7 +795,7 @@ case "${GLIBC_VERSION}" in
 
      *)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.14])
+	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.15])
 	AC_MSG_ERROR([or Darwin libc])
 	;;
 esac
--- valgrind-3.7.0/config.h.in.jj	2011-10-26 23:25:55.000000000 +0200
+++ valgrind-3.7.0/config.h.in	2011-12-09 17:47:42.385804043 +0100
@@ -39,6 +39,9 @@
 /* Define to 1 if you're using glibc 2.14.x */
 #undef GLIBC_2_14
 
+/* Define to 1 if you're using glibc 2.15.x */
+#undef GLIBC_2_15
+
 /* Define to 1 if you're using glibc 2.2.x */
 #undef GLIBC_2_2
 
--- valgrind-3.7.0/configure.jj	2011-11-05 12:24:39.000000000 +0100
+++ valgrind-3.7.0/configure	2011-12-09 17:48:58.371930777 +0100
@@ -6402,6 +6402,16 @@ $as_echo "#define GLIBC_2_14 1" >>confde
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.15)
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: 2.15 family" >&5
+$as_echo "2.15 family" >&6; }
+
+$as_echo "#define GLIBC_2_15 1" >>confdefs.h
+
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Darwin" >&5
 $as_echo "Darwin" >&6; }
@@ -6422,7 +6432,7 @@ $as_echo "#define BIONIC_LIBC 1" >>confd
      *)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: unsupported version ${GLIBC_VERSION}" >&5
 $as_echo "unsupported version ${GLIBC_VERSION}" >&6; }
-	as_fn_error "Valgrind requires glibc version 2.2 - 2.14" "$LINENO" 5
+	as_fn_error "Valgrind requires glibc version 2.2 - 2.15" "$LINENO" 5
 	as_fn_error "or Darwin libc" "$LINENO" 5
 	;;
 esac
