--- valgrind/coregrind/m_syswrap/syswrap-ppc64-linux.c.jj	2010-10-20 22:19:41.000000000 +0200
+++ valgrind/coregrind/m_syswrap/syswrap-ppc64-linux.c	2011-01-28 13:12:32.698670101 +0100
@@ -1379,7 +1379,7 @@ static SyscallTableEntry syscall_table[]
 // _____(__NR_rt_sigsuspend,     sys_rt_sigsuspend),      // 178
    GENXY(__NR_pread64,           sys_pread64),            // 179
 
-// _____(__NR_pwrite64,          sys_pwrite64),           // 180
+   GENX_(__NR_pwrite64,          sys_pwrite64),           // 180
    GENX_(__NR_chown,             sys_chown),              // 181
    GENXY(__NR_getcwd,            sys_getcwd),             // 182
    LINXY(__NR_capget,            sys_capget),             // 183
