--- valgrind-3.10.1/coregrind/m_debuginfo/debuginfo.c.orig	2015-07-06 00:21:14.252904153 +0200
+++ valgrind-3.10.1/coregrind/m_debuginfo/debuginfo.c	2015-07-06 00:22:30.975591973 +0200
@@ -925,7 +925,7 @@
 
    /* Note the details about the mapping. */
    struct _DebugInfoMapping map;
-   map.avma = a;
+   map.avma = seg->start;
    map.size = seg->end + 1 - seg->start;
    map.foff = seg->offset;
    map.rx   = is_rx_map;
