--- valgrind-3.7.0/coregrind/m_debuginfo/readdwarf3.c	(revision 12489)
+++ valgrind-3.7.0/coregrind/m_debuginfo/readdwarf3.c	(revision 12490)
@@ -2661,7 +2661,7 @@ static void parse_type_DIE ( /*MOD*/XArr
       boundE.cuOff = posn;
       vg_assert(parser->qparentE[parser->sp].Te.TyArray.boundRs);
       VG_(addToXA)( parser->qparentE[parser->sp].Te.TyArray.boundRs,
-                    &boundE );
+                    &boundE.cuOff );
       /* And record the child itself */
       goto acquire_Bound;
    }
