--- valgrind/coregrind/m_dispatch/dispatch-ppc64-linux.S	(revision 11699)
+++ valgrind/coregrind/m_dispatch/dispatch-ppc64-linux.S	(revision 11700)
@@ -310,7 +310,7 @@
 	/* start over */
 	b	.VG_(run_innerloop__dispatch_unprofiled)
 	/*NOTREACHED*/
-        .size VG_(run_innerloop), .-VG_(run_innerloop)
+        .size .VG_(run_innerloop), .-.VG_(run_innerloop)
 
 
 /*----------------------------------------------------*/
@@ -385,7 +385,7 @@
 	/* start over */
 	b	.VG_(run_innerloop__dispatch_profiled)
 	/*NOTREACHED*/
-        .size VG_(run_a_noredir_translation), .-VG_(run_a_noredir_translation)
+        .size .VG_(run_a_noredir_translation), .-.VG_(run_a_noredir_translation)
 
 
 /*----------------------------------------------------*/
