--- ext/dba/config.m4.db51	2010-11-02 17:22:29.888611878 -0200
+++ ext/dba/config.m4	2010-11-02 17:22:56.428259077 -0200
@@ -233,7 +233,7 @@
     AC_MSG_CHECKING([for DB4 minor version and patch level])
     AC_EGREP_CPP(yes,[
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MINOR != 1 || (DB_VERSION_MINOR == 1 && DB_VERSION_PATCH >= 25)
+#if DB_VERSION_MAJOR == 5 || DB_VERSION_MINOR != 1 || (DB_VERSION_MINOR == 1 && DB_VERSION_PATCH >= 25)
       yes
 #endif
     ],[
