--- ./php.ini-development.orig	2012-05-08 00:22:56.000000000 -0500
+++ ./php.ini-development	2012-05-14 22:32:06.017001589 -0500
@@ -1391,7 +1430,7 @@
 ; where MODE is the octal representation of the mode. Note that this
 ; does not overwrite the process's umask.
 ; http://php.net/session.save-path
-;session.save_path = "/tmp"
+session.save_path = "/var/lib/php"
 
 ; Whether to use cookies.
 ; http://php.net/session.use-cookies
