diff -up accounts-qt-1.6/common-project-config.pri.64bitarchs accounts-qt-1.6/common-project-config.pri
--- accounts-qt-1.6/common-project-config.pri.64bitarchs	2013-01-23 07:01:07.000000000 +0100
+++ accounts-qt-1.6/common-project-config.pri	2013-03-06 13:45:14.414158442 +0100
@@ -43,11 +43,9 @@ isEmpty( PREFIX ) {
 }
 
 ARCH = $$system(uname -m)
-contains( ARCH, x86_64 ) {
-    INSTALL_LIBDIR = $${INSTALL_PREFIX}/lib64
-} else {
-    INSTALL_LIBDIR = $${INSTALL_PREFIX}/lib
-}
+64BITARCHS= x86_64 ppc64 s390x aarch64 ppc64le
+INSTALL_LIBDIR = $${INSTALL_PREFIX}/lib
+for(a, 64BITARCHS):contains(ARCH, $${a}):INSTALL_LIBDIR = $${INSTALL_PREFIX}/lib64
 
 # default library directory can be overriden by defining LIBDIR when
 # running qmake
