--- qtcreator.pro.orig	2015-06-04 16:35:17.000000000 -0300
+++ qtcreator.pro	2015-06-04 16:36:44.638297792 -0300
@@ -49,7 +49,7 @@ exists(src/shared/qbs/qbs.pro) {
         QBS_APPS_RPATH_DIR = @loader_path/../Frameworks
     } else {
         QBS_PLUGINS_BUILD_DIR = $${IDE_BUILD_TREE}/$${IDE_LIBRARY_BASENAME}/qtcreator/plugins
-        QBS_APPS_RPATH_DIR = \$\$ORIGIN/../$$IDE_LIBRARY_BASENAME/qtcreator
+        QBS_APPS_RPATH_DIR = $${QTC_PREFIX}/$$IDE_LIBRARY_BASENAME/qtcreator
     }
     cache(QBS_PLUGINS_BUILD_DIR)
     cache(QBS_APPS_RPATH_DIR)
--- src/shared/qbs/src/lib/qtprofilesetup/use_qtprofilesetup.pri.orig	2015-06-04 16:48:08.616505376 -0300
+++ src/shared/qbs/src/lib/qtprofilesetup/use_qtprofilesetup.pri	2015-06-04 16:48:42.890060855 -0300
@@ -12,7 +12,7 @@ unix {
 }
 
 !qbs_disable_rpath {
-    linux-*:QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,\$\$ORIGIN/../$${QBS_LIBRARY_DIRNAME}\'
+    linux-*:QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$${QTC_PREFIX}/$${QBS_LIBRARY_DIRNAME}\'
     macx:QMAKE_LFLAGS += -Wl,-rpath,@loader_path/../$${QBS_LIBRARY_DIRNAME}
 }
 
