diff -Nur chkrootkit-0.47-orig/chklastlog.c chkrootkit-0.47/chklastlog.c
--- chkrootkit-0.47-orig/chklastlog.c	2006-02-11 18:02:48.000000000 +0100
+++ chkrootkit-0.47/chklastlog.c	2006-10-20 14:31:43.000000000 +0200
@@ -164,7 +164,7 @@
             {
                 if (*uid > MAX_ID)
                 {
-                   fprintf(stderr, "MAX_ID is %ld and current uid is %ld, please check\n\r", MAX_ID, *uid );
+                    fprintf(stderr, "MAX_ID is %ld and current uid is %ld, please check\n\r", (long int)MAX_ID, (long int)*uid );
                    exit (1);
 
                 }
diff -Nur chkrootkit-0.47-orig/chkproc.c chkrootkit-0.47/chkproc.c
--- chkrootkit-0.47-orig/chkproc.c	2006-07-25 16:55:21.000000000 +0200
+++ chkrootkit-0.47/chkproc.c	2006-10-20 14:34:12.000000000 +0200
@@ -180,10 +180,8 @@
    if (pv < 1 || pv > PS_MAX)
       pv = 1;
    pscmd = ps_cmds[pv];
-/*  printf("pv = %d\n\r", pv); /* -- DEBUG */
 #endif
 
-/* printf("pscmd = %s\n\r", pscmd); /* -- DEBUG */ 
    if (!(ps = popen(pscmd, "r")))
    {
        perror("ps");
@@ -228,7 +226,6 @@
           p++;
       while (isspace(*p)) /* Skip spaces */
           p++;
-/*  printf(">>%s<<\n", p);  /* -- DEBUG */
       ret = atol(p);
       if ( ret < 0 || ret > MAX_PROCESSES )
       {
@@ -263,7 +260,6 @@
       }
 #endif
 
-/*      printf("%s\n", tmp_d_name); /* -- DEBUG */
       dirproc[atol(tmp_d_name)] = 1;
    }
    closedir(proc);
diff -Nur chkrootkit-0.48-orig/chkdirs.c chkrootkit-0.48/chkdirs.c
--- chkrootkit-0.48-orig/chkdirs.c	2007-08-10 23:22:52.000000000 +0200
+++ chkrootkit-0.48/chkdirs.c	2008-02-12 10:36:40.000000000 +0100
@@ -60,7 +60,7 @@
 
 char *make_pathname (char *path, char *dir, char **buffer)
 {
-  int plen, pathname_len, bufsize, offs;
+  int plen, pathname_len, bufsize = 0, offs;
 
   plen = strlen(path);
   pathname_len = plen + strlen(dir) + 2;
@@ -237,7 +237,7 @@
 int main (int argc, char **argv)
 {
   int norecurse = 0;
-  int i, retval;
+  int i, retval = 1;
   char c;
 
   opterr = 0;
