--- gtk/src/gtk_netplay.cpp.orig	2013-05-26 14:27:59.685360422 -0300
+++ gtk/src/gtk_netplay.cpp	2013-05-26 14:29:21.063085173 -0300
@@ -149,10 +149,9 @@ S9xNetplayStartServer (void)
     NPServer.SyncByReset = gui_config->netplay_sync_reset;
     NPServer.SendROMImageOnConnect = gui_config->netplay_send_rom;
 
-    npthread = g_thread_create (S9xNetplayServerThread,
-                                NULL,
-                                TRUE,
-                                NULL);
+    npthread = g_thread_new ("S9xNetplayServerThread",
+                              S9xNetplayServerThread,
+                              NULL);
 
     /* Sleep to let the server create itself */
     usleep (10000);
