diff -up alacarte-0.13.1/configure.ac.python27 alacarte-0.13.1/configure.ac
--- alacarte-0.13.1/configure.ac.python27	2010-07-27 09:49:12.576256802 -0400
+++ alacarte-0.13.1/configure.ac	2010-07-27 09:49:19.906256638 -0400
@@ -20,7 +20,7 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GE
 AM_GLIB_GNU_GETTEXT
 IT_PROG_INTLTOOL([0.40.0])
 
-AM_PATH_PYTHON_VERSION(2.6, 2.6.0, 2.5, 2.5.0, 2.4, 2.4.0)
+AM_PATH_PYTHON_VERSION(2.7, 2.7.0, 2.6, 2.6.0, 2.5, 2.5.0, 2.4, 2.4.0)
 
 PKG_CHECK_MODULES(ALACARTE, libgnome-menu >= 2.27.92)
 
