diff -ru ../akonadi-1.4.0/qsqlite/src/qsql_sqlite.cpp ./qsqlite/src/qsql_sqlite.cpp
--- ../akonadi-1.4.0/qsqlite/src/qsql_sqlite.cpp	2010-07-31 18:12:24.000000000 +0200
+++ ./qsqlite/src/qsql_sqlite.cpp	2010-09-08 15:23:16.152896580 +0200
@@ -323,7 +323,6 @@
 
     setSelect(false);
 
-    Q_ASSERT(SQLITE_VERSION_NUMBER == 3006023);
 #if (SQLITE_VERSION_NUMBER >= 3003011)
 //    int res = sqlite3_prepare16_v2(d->access, query.constData(), (query.size() + 1) * sizeof(QChar),
 //                                   &d->stmt, 0);
