diff -up akonadi/server/src/storage/mysql-global.conf.mysql_conf akonadi/server/src/storage/mysql-global.conf
--- akonadi/server/src/storage/mysql-global.conf.mysql_conf	2010-05-21 05:17:10.000000000 -0500
+++ akonadi/server/src/storage/mysql-global.conf	2010-05-23 17:14:08.144227600 -0500
@@ -49,7 +49,7 @@ innodb_additional_mem_pool_size=1M
 # SELECT sum(data_length) as bla, sum(index_length) as blub FROM information_schema.tables WHERE table_schema not in ("mysql", "information_schema");
 innodb_buffer_pool_size=80M
 # size of average write burst, keep Innob_log_waits small, keep Innodb_buffer_pool_wait_free small (see show global status like "inno%", show global variables)
-innodb_log_file_size=64M
+innodb_log_file_size=8M
 innodb_flush_log_at_trx_commit=2
 
 # Do not drop the connection to the DB after 8 hours of inactivity
