diff -up LibVNCServer-0.9.8.2/configure.ac.system_minilzo LibVNCServer-0.9.8.2/configure.ac
--- LibVNCServer-0.9.8.2/configure.ac.system_minilzo	2011-11-09 05:53:25.000000000 -0600
+++ LibVNCServer-0.9.8.2/configure.ac	2011-12-13 09:24:15.185258499 -0600
@@ -855,7 +855,7 @@ if test "$build_x11vnc" = "yes"; then
 	#
 	# configure.ac:690: required file `x11vnc/Makefile.in' not found
 	#
-	AC_CONFIG_FILES([x11vnc/Makefile x11vnc/misc/Makefile x11vnc/misc/turbovnc/Makefile])
+	#AC_CONFIG_FILES([x11vnc/Makefile x11vnc/misc/Makefile x11vnc/misc/turbovnc/Makefile])
 
 	if test ! -z "$with_system_libvncserver" -a "x$with_system_libvncserver" != "xno"; then
 		# need to move local tarball rfb headers aside:
diff -up LibVNCServer-0.9.8.2/libvncclient/Makefile.am.system_minilzo LibVNCServer-0.9.8.2/libvncclient/Makefile.am
--- LibVNCServer-0.9.8.2/libvncclient/Makefile.am.system_minilzo	2011-10-12 11:44:49.000000000 -0500
+++ LibVNCServer-0.9.8.2/libvncclient/Makefile.am	2011-12-13 09:25:16.923486669 -0600
@@ -1,8 +1,9 @@
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/common
 
-libvncclient_la_SOURCES=cursor.c listen.c rfbproto.c sockets.c vncviewer.c ../common/minilzo.c tls.c
+libvncclient_la_LIBADD=-lminilzo
+libvncclient_la_SOURCES=cursor.c listen.c rfbproto.c sockets.c vncviewer.c tls.c
 
-noinst_HEADERS=../common/lzodefs.h ../common/lzoconf.h ../common/minilzo.h tls.h
+noinst_HEADERS=tls.h
 
 rfbproto.o: rfbproto.c corre.c hextile.c rre.c tight.c zlib.c zrle.c ultra.c
 
diff -up LibVNCServer-0.9.8.2/libvncclient/rfbproto.c.system_minilzo LibVNCServer-0.9.8.2/libvncclient/rfbproto.c
--- LibVNCServer-0.9.8.2/libvncclient/rfbproto.c.system_minilzo	2011-11-09 05:44:18.000000000 -0600
+++ LibVNCServer-0.9.8.2/libvncclient/rfbproto.c	2011-12-13 09:24:15.186258487 -0600
@@ -59,7 +59,7 @@
 #include <gcrypt.h>
 #endif
 
-#include "minilzo.h"
+#include <lzo/minilzo.h>
 #include "tls.h"
 
 /*
diff -up LibVNCServer-0.9.8.2/libvncserver/Makefile.am.system_minilzo LibVNCServer-0.9.8.2/libvncserver/Makefile.am
--- LibVNCServer-0.9.8.2/libvncserver/Makefile.am.system_minilzo	2011-11-09 05:43:52.000000000 -0600
+++ LibVNCServer-0.9.8.2/libvncserver/Makefile.am	2011-12-13 09:24:15.187258475 -0600
@@ -19,7 +19,7 @@ include_HEADERS=../rfb/rfb.h ../rfb/rfbc
 	../rfb/rfbproto.h ../rfb/keysym.h ../rfb/rfbregion.h ../rfb/rfbclient.h
 
 noinst_HEADERS=../common/d3des.h ../rfb/default8x16.h zrleoutstream.h \
-	zrlepalettehelper.h zrletypes.h private.h ../common/minilzo.h ../common/lzoconf.h ../common/lzodefs.h scale.h \
+	zrlepalettehelper.h zrletypes.h private.h scale.h \
 	$(TIGHTVNCFILETRANSFERHDRS)
 
 EXTRA_DIST=tableinit24.c tableinittctemplate.c tabletranstemplate.c \
@@ -36,10 +36,11 @@ endif
 LIB_SRCS = main.c rfbserver.c rfbregion.c auth.c sockets.c \
 	stats.c corre.c hextile.c rre.c translate.c cutpaste.c \
 	httpd.c cursor.c font.c \
-	draw.c selbox.c ../common/d3des.c ../common/vncauth.c cargs.c ../common/minilzo.c ultra.c scale.c \
+	draw.c selbox.c ../common/d3des.c ../common/vncauth.c cargs.c ultra.c scale.c \
 	$(ZLIBSRCS) $(JPEGSRCS) $(TIGHTVNCFILETRANSFERSRCS)
 
 libvncserver_la_SOURCES=$(LIB_SRCS)
+libvncserver_la_LIBADD=-lminilzo
 
 lib_LTLIBRARIES=libvncserver.la
 
diff -up LibVNCServer-0.9.8.2/libvncserver/ultra.c.system_minilzo LibVNCServer-0.9.8.2/libvncserver/ultra.c
--- LibVNCServer-0.9.8.2/libvncserver/ultra.c.system_minilzo	2011-10-12 11:44:50.000000000 -0500
+++ LibVNCServer-0.9.8.2/libvncserver/ultra.c	2011-12-13 09:24:15.187258475 -0600
@@ -8,7 +8,7 @@
  */
 
 #include <rfb/rfb.h>
-#include "minilzo.h"
+#include <lzo/minilzo.h>
 
 /*
  * cl->beforeEncBuf contains pixel data in the client's format.
diff -up LibVNCServer-0.9.8.2/Makefile.am.system_minilzo LibVNCServer-0.9.8.2/Makefile.am
--- LibVNCServer-0.9.8.2/Makefile.am.system_minilzo	2011-06-09 18:14:00.000000000 -0500
+++ LibVNCServer-0.9.8.2/Makefile.am	2011-12-13 09:24:15.187258475 -0600
@@ -1,6 +1,6 @@
-if WITH_X11VNC
-X11VNC=x11vnc
-endif
+#if WITH_X11VNC
+#X11VNC=x11vnc
+#endif
 
 SUBDIRS=libvncserver examples libvncclient vncterm classes client_examples test $(X11VNC)
 DIST_SUBDIRS=libvncserver examples libvncclient vncterm classes client_examples test
