diff -up LibVNCServer-0.9.8/libvncclient/Makefile.in.system_minilzo LibVNCServer-0.9.8/libvncclient/Makefile.in
--- LibVNCServer-0.9.8/libvncclient/Makefile.in.system_minilzo	2011-03-30 08:34:38.000000000 -0300
+++ LibVNCServer-0.9.8/libvncclient/Makefile.in	2011-03-30 20:07:27.750069382 -0300
@@ -72,7 +72,7 @@ am__installdirs = "$(DESTDIR)$(libdir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
 libvncclient_la_LIBADD =
 am_libvncclient_la_OBJECTS = cursor.lo listen.lo rfbproto.lo \
-	sockets.lo vncviewer.lo minilzo.lo tls.lo
+	sockets.lo vncviewer.lo tls.lo
 libvncclient_la_OBJECTS = $(am_libvncclient_la_OBJECTS)
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -225,8 +225,8 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 with_ffmpeg = @with_ffmpeg@
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/common
-libvncclient_la_SOURCES = cursor.c listen.c rfbproto.c sockets.c vncviewer.c ../common/minilzo.c tls.c
-noinst_HEADERS = ../common/lzodefs.h ../common/lzoconf.h ../common/minilzo.h tls.h
+libvncclient_la_SOURCES = cursor.c listen.c rfbproto.c sockets.c vncviewer.c tls.c
+noinst_HEADERS = ../common/lzodefs.h ../common/lzoconf.h tls.h
 EXTRA_DIST = corre.c hextile.c rre.c tight.c zlib.c zrle.c ultra.c tls.c
 lib_LTLIBRARIES = libvncclient.la
 all: all-am
@@ -305,7 +305,6 @@ distclean-compile:
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cursor.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/listen.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/minilzo.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbproto.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sockets.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tls.Plo@am__quote@
diff -up LibVNCServer-0.9.8/libvncclient/rfbproto.c.system_minilzo LibVNCServer-0.9.8/libvncclient/rfbproto.c
diff -up LibVNCServer-0.9.8/libvncserver/Makefile.in.system_minilzo LibVNCServer-0.9.8/libvncserver/Makefile.in
--- LibVNCServer-0.9.8/libvncserver/Makefile.in.system_minilzo	2011-03-30 08:34:38.000000000 -0300
+++ LibVNCServer-0.9.8/libvncserver/Makefile.in	2011-03-30 20:07:28.700859283 -0300
@@ -75,7 +75,7 @@ am__libvncserver_la_SOURCES_DIST = main.
 	auth.c sockets.c stats.c corre.c hextile.c rre.c translate.c \
 	cutpaste.c httpd.c cursor.c font.c draw.c selbox.c \
 	../common/d3des.c ../common/vncauth.c cargs.c \
-	../common/minilzo.c ultra.c scale.c zlib.c zrle.c \
+	ultra.c scale.c zlib.c zrle.c \
 	zrleoutstream.c zrlepalettehelper.c ../common/zywrletemplate.c \
 	tight.c tightvnc-filetransfer/rfbtightserver.c \
 	tightvnc-filetransfer/handlefiletransferrequest.c \
@@ -91,7 +91,7 @@ am__libvncserver_la_SOURCES_DIST = main.
 am__objects_4 = main.lo rfbserver.lo rfbregion.lo auth.lo sockets.lo \
 	stats.lo corre.lo hextile.lo rre.lo translate.lo cutpaste.lo \
 	httpd.lo cursor.lo font.lo draw.lo selbox.lo d3des.lo \
-	vncauth.lo cargs.lo minilzo.lo ultra.lo scale.lo \
+	vncauth.lo cargs.lo ultra.lo scale.lo \
 	$(am__objects_1) $(am__objects_2) $(am__objects_3)
 am_libvncserver_la_OBJECTS = $(am__objects_4)
 libvncserver_la_OBJECTS = $(am_libvncserver_la_OBJECTS)
@@ -112,7 +112,7 @@ SOURCES = $(libvncserver_la_SOURCES)
 DIST_SOURCES = $(am__libvncserver_la_SOURCES_DIST)
 am__noinst_HEADERS_DIST = ../common/d3des.h ../rfb/default8x16.h \
 	zrleoutstream.h zrlepalettehelper.h zrletypes.h private.h \
-	../common/minilzo.h ../common/lzoconf.h ../common/lzodefs.h \
+	../common/lzoconf.h ../common/lzodefs.h \
 	scale.h tightvnc-filetransfer/filelistinfo.h \
 	tightvnc-filetransfer/filetransfermsg.h \
 	tightvnc-filetransfer/handlefiletransferrequest.h \
@@ -268,7 +268,7 @@ include_HEADERS = ../rfb/rfb.h ../rfb/rf
 	../rfb/rfbproto.h ../rfb/keysym.h ../rfb/rfbregion.h ../rfb/rfbclient.h
 
 noinst_HEADERS = ../common/d3des.h ../rfb/default8x16.h zrleoutstream.h \
-	zrlepalettehelper.h zrletypes.h private.h ../common/minilzo.h ../common/lzoconf.h ../common/lzodefs.h scale.h \
+	zrlepalettehelper.h zrletypes.h private.h ../common/lzoconf.h ../common/lzodefs.h scale.h \
 	$(TIGHTVNCFILETRANSFERHDRS)
 
 EXTRA_DIST = tableinit24.c tableinittctemplate.c tabletranstemplate.c \
@@ -280,7 +280,7 @@ EXTRA_DIST = tableinit24.c tableinittcte
 LIB_SRCS = main.c rfbserver.c rfbregion.c auth.c sockets.c \
 	stats.c corre.c hextile.c rre.c translate.c cutpaste.c \
 	httpd.c cursor.c font.c \
-	draw.c selbox.c ../common/d3des.c ../common/vncauth.c cargs.c ../common/minilzo.c ultra.c scale.c \
+	draw.c selbox.c ../common/d3des.c ../common/vncauth.c cargs.c ultra.c scale.c \
 	$(ZLIBSRCS) $(JPEGSRCS) $(TIGHTVNCFILETRANSFERSRCS)
 
 libvncserver_la_SOURCES = $(LIB_SRCS)
@@ -373,7 +373,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hextile.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/httpd.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/main.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/minilzo.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbregion.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbserver.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbtightserver.Plo@am__quote@
