diff -up LibVNCServer-0.9.7/libvncclient/Makefile.in.orig LibVNCServer-0.9.7/libvncclient/Makefile.in
--- LibVNCServer-0.9.7/libvncclient/Makefile.in.orig	2009-01-11 20:53:13.000000000 -0600
+++ LibVNCServer-0.9.7/libvncclient/Makefile.in	2009-05-04 12:33:14.513826561 -0500
@@ -61,7 +61,7 @@ libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 libvncclient_la_LIBADD =
 am_libvncclient_la_OBJECTS = cursor.lo listen.lo rfbproto.lo \
-	sockets.lo vncviewer.lo minilzo.lo
+	sockets.lo vncviewer.lo
 libvncclient_la_OBJECTS = $(am_libvncclient_la_OBJECTS)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -222,8 +222,8 @@ sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 with_ffmpeg = @with_ffmpeg@
 AM_CFLAGS = -g -I $(top_srcdir) -I. -Wall
-libvncclient_la_SOURCES = cursor.c listen.c rfbproto.c sockets.c vncviewer.c minilzo.c
-noinst_HEADERS = lzoconf.h minilzo.h
+libvncclient_la_SOURCES = cursor.c listen.c rfbproto.c sockets.c vncviewer.c
+noinst_HEADERS = lzoconf.h
 EXTRA_DIST = corre.c hextile.c rre.c tight.c zlib.c zrle.c ultra.c
 lib_LTLIBRARIES = libvncclient.la
 all: all-am
@@ -297,7 +297,6 @@ distclean-compile:
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cursor.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/listen.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/minilzo.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbproto.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/sockets.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/vncviewer.Plo@am__quote@
diff -up LibVNCServer-0.9.7/libvncclient/rfbproto.c.orig LibVNCServer-0.9.7/libvncclient/rfbproto.c
--- LibVNCServer-0.9.7/libvncclient/rfbproto.c.orig	2008-02-04 11:58:11.000000000 -0600
+++ LibVNCServer-0.9.7/libvncclient/rfbproto.c	2009-05-04 12:33:14.514826549 -0500
@@ -50,7 +50,7 @@
 #include <stdarg.h>
 #include <time.h>
 
-#include "minilzo.h"
+#include <minilzo.h>
 
 /*
  * rfbClientLog prints a time-stamped message to the log file (stderr).
diff -up LibVNCServer-0.9.7/libvncserver/Makefile.in.orig LibVNCServer-0.9.7/libvncserver/Makefile.in
--- LibVNCServer-0.9.7/libvncserver/Makefile.in.orig	2009-01-11 20:53:13.000000000 -0600
+++ LibVNCServer-0.9.7/libvncserver/Makefile.in	2009-05-04 12:40:31.533854737 -0500
@@ -63,7 +63,7 @@ libvncserver_la_LIBADD =
 am__libvncserver_la_SOURCES_DIST = main.c rfbserver.c rfbregion.c \
 	auth.c sockets.c stats.c corre.c hextile.c rre.c translate.c \
 	cutpaste.c httpd.c cursor.c font.c draw.c selbox.c d3des.c \
-	vncauth.c cargs.c minilzo.c ultra.c scale.c zlib.c zrle.c \
+	vncauth.c cargs.c ultra.c scale.c zlib.c zrle.c \
 	zrleoutstream.c zrlepalettehelper.c zywrletemplate.c tight.c \
 	tightvnc-filetransfer/rfbtightserver.c \
 	tightvnc-filetransfer/handlefiletransferrequest.c \
@@ -79,7 +79,7 @@ am__libvncserver_la_SOURCES_DIST = main.
 am__objects_4 = main.lo rfbserver.lo rfbregion.lo auth.lo sockets.lo \
 	stats.lo corre.lo hextile.lo rre.lo translate.lo cutpaste.lo \
 	httpd.lo cursor.lo font.lo draw.lo selbox.lo d3des.lo \
-	vncauth.lo cargs.lo minilzo.lo ultra.lo scale.lo \
+	vncauth.lo cargs.lo ultra.lo scale.lo \
 	$(am__objects_1) $(am__objects_2) $(am__objects_3)
 am_libvncserver_la_OBJECTS = $(am__objects_4)
 libvncserver_la_OBJECTS = $(am_libvncserver_la_OBJECTS)
@@ -98,7 +98,7 @@ SOURCES = $(libvncserver_la_SOURCES)
 DIST_SOURCES = $(am__libvncserver_la_SOURCES_DIST)
 includeHEADERS_INSTALL = $(INSTALL_HEADER)
 am__noinst_HEADERS_DIST = d3des.h ../rfb/default8x16.h zrleoutstream.h \
-	zrlepalettehelper.h zrletypes.h private.h minilzo.h lzoconf.h \
+	zrlepalettehelper.h zrletypes.h private.h lzoconf.h \
 	scale.h tightvnc-filetransfer/filelistinfo.h \
 	tightvnc-filetransfer/filetransfermsg.h \
 	tightvnc-filetransfer/handlefiletransferrequest.h \
@@ -264,7 +264,7 @@ include_HEADERS = ../rfb/rfb.h ../rfb/rf
 	../rfb/rfbproto.h ../rfb/keysym.h ../rfb/rfbregion.h ../rfb/rfbclient.h
 
 noinst_HEADERS = d3des.h ../rfb/default8x16.h zrleoutstream.h \
-	zrlepalettehelper.h zrletypes.h private.h minilzo.h lzoconf.h scale.h \
+	zrlepalettehelper.h zrletypes.h private.h lzoconf.h scale.h \
 	$(TIGHTVNCFILETRANSFERHDRS)
 
 EXTRA_DIST = tableinit24.c tableinittctemplate.c tabletranstemplate.c \
@@ -276,7 +276,7 @@ EXTRA_DIST = tableinit24.c tableinittcte
 LIB_SRCS = main.c rfbserver.c rfbregion.c auth.c sockets.c \
 	stats.c corre.c hextile.c rre.c translate.c cutpaste.c \
 	httpd.c cursor.c font.c \
-	draw.c selbox.c d3des.c vncauth.c cargs.c minilzo.c ultra.c scale.c \
+	draw.c selbox.c d3des.c vncauth.c cargs.c ultra.c scale.c \
 	$(ZLIBSRCS) $(JPEGSRCS) $(TIGHTVNCFILETRANSFERSRCS)
 
 libvncserver_la_SOURCES = $(LIB_SRCS)
@@ -364,7 +364,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hextile.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/httpd.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/main.Plo@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/minilzo.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbregion.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbserver.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rfbtightserver.Plo@am__quote@
