diff -up LibVNCServer-0.9.9/Makefile.am.system_minilzo LibVNCServer-0.9.9/Makefile.am
--- LibVNCServer-0.9.9/Makefile.am.system_minilzo	2012-05-04 11:19:00.000000000 -0300
+++ LibVNCServer-0.9.9/Makefile.am	2012-05-06 13:09:27.346741358 -0300
@@ -1,6 +1,6 @@
-if WITH_X11VNC
-X11VNC=x11vnc
-endif
+#if WITH_X11VNC
+#X11VNC=x11vnc
+#endif
 
 SUBDIRS=libvncserver examples libvncclient vncterm webclients client_examples test $(X11VNC)
 DIST_SUBDIRS=libvncserver examples libvncclient vncterm webclients client_examples test
diff -up LibVNCServer-0.9.9/configure.ac.system_minilzo LibVNCServer-0.9.9/configure.ac
--- LibVNCServer-0.9.9/configure.ac.system_minilzo	2012-05-04 13:27:08.000000000 -0300
+++ LibVNCServer-0.9.9/configure.ac	2012-05-06 13:09:27.346741358 -0300
@@ -992,7 +992,7 @@ if test "$build_x11vnc" = "yes"; then
 	#
 	# configure.ac:690: required file `x11vnc/Makefile.in' not found
 	#
-	AC_CONFIG_FILES([x11vnc/Makefile x11vnc/misc/Makefile x11vnc/misc/turbovnc/Makefile])
+	#AC_CONFIG_FILES([x11vnc/Makefile x11vnc/misc/Makefile x11vnc/misc/turbovnc/Makefile])
 
 	if test ! -z "$with_system_libvncserver" -a "x$with_system_libvncserver" != "xno"; then
 		# need to move local tarball rfb headers aside:
diff -up LibVNCServer-0.9.9/libvncclient/Makefile.am.system_minilzo LibVNCServer-0.9.9/libvncclient/Makefile.am
--- LibVNCServer-0.9.9/libvncclient/Makefile.am.system_minilzo	2012-05-04 11:19:00.000000000 -0300
+++ LibVNCServer-0.9.9/libvncclient/Makefile.am	2012-05-06 13:10:35.179051246 -0300
@@ -13,10 +13,10 @@ endif
 endif
 
 
-libvncclient_la_SOURCES=cursor.c listen.c rfbproto.c sockets.c vncviewer.c ../common/minilzo.c $(TLSSRCS)
+libvncclient_la_SOURCES=cursor.c listen.c rfbproto.c sockets.c vncviewer.c $(TLSSRCS)
 libvncclient_la_LIBADD=$(TLSLIBS)
 
-noinst_HEADERS=../common/lzodefs.h ../common/lzoconf.h ../common/minilzo.h tls.h
+noinst_HEADERS=tls.h
 
 rfbproto.o: rfbproto.c corre.c hextile.c rre.c tight.c zlib.c zrle.c ultra.c
 
diff -up LibVNCServer-0.9.9/libvncclient/rfbproto.c.system_minilzo LibVNCServer-0.9.9/libvncclient/rfbproto.c
--- LibVNCServer-0.9.9/libvncclient/rfbproto.c.system_minilzo	2012-05-04 11:19:00.000000000 -0300
+++ LibVNCServer-0.9.9/libvncclient/rfbproto.c	2012-05-06 13:09:27.347741142 -0300
@@ -59,7 +59,7 @@
 #include <gcrypt.h>
 #endif
 
-#include "minilzo.h"
+#include <lzo/minilzo.h>
 #include "tls.h"
 
 /*
diff -up LibVNCServer-0.9.9/libvncserver/Makefile.am.system_minilzo LibVNCServer-0.9.9/libvncserver/Makefile.am
--- LibVNCServer-0.9.9/libvncserver/Makefile.am.system_minilzo	2012-05-04 11:19:00.000000000 -0300
+++ LibVNCServer-0.9.9/libvncserver/Makefile.am	2012-05-06 13:10:17.966778831 -0300
@@ -37,7 +37,7 @@ include_HEADERS=../rfb/rfb.h ../rfb/rfbc
 
 noinst_HEADERS=../common/d3des.h ../rfb/default8x16.h zrleoutstream.h \
 	zrlepalettehelper.h zrletypes.h private.h scale.h rfbssl.h rfbcrypto.h \
-	../common/minilzo.h ../common/lzoconf.h ../common/lzodefs.h ../common/md5.h ../common/sha1.h \
+	../common/md5.h ../common/sha1.h \
 	$(TIGHTVNCFILETRANSFERHDRS)
 
 EXTRA_DIST=tableinit24.c tableinittctemplate.c tabletranstemplate.c \
@@ -54,7 +54,7 @@ endif
 LIB_SRCS = main.c rfbserver.c rfbregion.c auth.c sockets.c $(WEBSOCKETSSRCS) \
 	stats.c corre.c hextile.c rre.c translate.c cutpaste.c \
 	httpd.c cursor.c font.c \
-	draw.c selbox.c ../common/d3des.c ../common/vncauth.c cargs.c ../common/minilzo.c ultra.c scale.c \
+	draw.c selbox.c ../common/d3des.c ../common/vncauth.c cargs.c ultra.c scale.c \
 	$(ZLIBSRCS) $(TIGHTSRCS) $(TIGHTVNCFILETRANSFERSRCS)
 
 libvncserver_la_SOURCES=$(LIB_SRCS)
diff -up LibVNCServer-0.9.9/libvncserver/ultra.c.system_minilzo LibVNCServer-0.9.9/libvncserver/ultra.c
--- LibVNCServer-0.9.9/libvncserver/ultra.c.system_minilzo	2012-05-04 11:19:00.000000000 -0300
+++ LibVNCServer-0.9.9/libvncserver/ultra.c	2012-05-06 13:09:27.347741142 -0300
@@ -8,7 +8,7 @@
  */
 
 #include <rfb/rfb.h>
-#include "minilzo.h"
+#include <lzo/minilzo.h>
 
 /*
  * cl->beforeEncBuf contains pixel data in the client's format.
