diff -ur teeworlds-b177-r50edfd37-source.orig/bam.lua teeworlds-b177-r50edfd37-source/bam.lua
--- teeworlds-b177-r50edfd37-source.orig/bam.lua	2011-07-31 23:17:00.000000000 +0200
+++ teeworlds-b177-r50edfd37-source/bam.lua	2011-08-07 11:41:07.596001939 +0200
@@ -9,6 +9,8 @@
 config:Add(OptCCompiler("compiler"))
 config:Add(OptTestCompileC("stackprotector", "int main(){return 0;}", "-fstack-protector -fstack-protector-all"))
 config:Add(OptLibrary("zlib", "zlib.h", false))
+config:Add(OptLibrary("wavpack", "wavpack/wavpack.h", false))
+config:Add(OptLibrary("pnglite", "sys/types.h", false))
 config:Add(SDL.OptFind("sdl", true))
 config:Add(FreeType.OptFind("freetype", true))
 config:Finalize("config.lua")
@@ -132,6 +134,13 @@
 function build(settings)
 	--settings.objdir = Path("objs")
 	settings.cc.Output = Intermediate_Output
+	settings.cc.flags:Add(os.getenv ('CFLAGS'))
+	settings.cc.flags:Add(os.getenv ('CPPFLAGS'))
+	settings.cc.flags:Add(os.getenv ('LDFLAGS'))
+	settings.link.libs:Add("wavpack")
+	settings.link.libs:Add("z")
+	settings.link.libs:Add("pnglite")
+	--settings.link.flags:Add('-lpng')
 
 	if config.compiler.driver == "cl" then
 		settings.cc.flags:Add("/wd4244")
@@ -164,21 +173,6 @@
 		settings.link.libs:Add("shell32")
 	end
 
-	-- compile zlib if needed
-	if config.zlib.value == 1 then
-		settings.link.libs:Add("z")
-		if config.zlib.include_path then
-			settings.cc.includes:Add(config.zlib.include_path)
-		end
-		zlib = {}
-	else
-		zlib = Compile(settings, Collect("src/engine/external/zlib/*.c"))
-		settings.cc.includes:Add("src/engine/external/zlib")
-	end
-
-	-- build the small libraries
-	wavpack = Compile(settings, Collect("src/engine/external/wavpack/*.c"))
-	pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
 
 	-- build game components
 	engine_settings = settings:Copy()
diff -ur teeworlds-b177-r50edfd37-source.orig/src/engine/client/graphics.cpp teeworlds-b177-r50edfd37-source/src/engine/client/graphics.cpp
--- teeworlds-b177-r50edfd37-source.orig/src/engine/client/graphics.cpp	2011-07-31 23:17:00.000000000 +0200
+++ teeworlds-b177-r50edfd37-source/src/engine/client/graphics.cpp	2011-08-07 10:44:09.524885861 +0200
@@ -20,7 +20,7 @@
 #endif
 
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 #include <engine/shared/config.h>
 #include <engine/graphics.h>
diff -ur teeworlds-b177-r50edfd37-source.orig/src/engine/client/sound.cpp teeworlds-b177-r50edfd37-source/src/engine/client/sound.cpp
--- teeworlds-b177-r50edfd37-source.orig/src/engine/client/sound.cpp	2011-07-31 23:17:00.000000000 +0200
+++ teeworlds-b177-r50edfd37-source/src/engine/client/sound.cpp	2011-08-07 11:28:31.946001646 +0200
@@ -10,7 +10,7 @@
 #include "sound.h"
 
 extern "C" { // wavpack
-	#include <engine/external/wavpack/wavpack.h>
+	#include <wavpack/wavpack.h>
 }
 #include <math.h>
 
@@ -320,6 +320,7 @@
 	CSample *pSample;
 	int SampleID = -1;
 	char aError[100];
+	char completefilename[102400] = "";
 	WavpackContext *pContext;
 	
 	// don't waste memory on sound when we are stress testing
@@ -333,7 +334,10 @@
 	if(!m_pStorage)
 		return -1;
 
-	ms_File = m_pStorage->OpenFile(pFilename, IOFLAG_READ, IStorage::TYPE_ALL);
+        strncat(completefilename, "/usr/share/teeworlds/data/", 50);
+        strncat(completefilename, pFilename, strlen(pFilename));
+        ms_File = m_pStorage->OpenFile(completefilename, IOFLAG_READ, IStorage::TYPE_ALL);
+
 	if(!ms_File)
 	{
 		dbg_msg("sound/wv", "failed to open file. filename='%s'", pFilename);
@@ -345,7 +349,7 @@
 		return -1;
 	pSample = &m_aSamples[SampleID];
 
-	pContext = WavpackOpenFileInput(ReadData, aError);
+	pContext = WavpackOpenFileInput(completefilename, aError, 0, 0);
 	if (pContext)
 	{
 		int m_aSamples = WavpackGetNumSamples(pContext);
diff -ur teeworlds-b177-r50edfd37-source.orig/src/tools/dilate.cpp teeworlds-b177-r50edfd37-source/src/tools/dilate.cpp
--- teeworlds-b177-r50edfd37-source.orig/src/tools/dilate.cpp	2011-07-31 23:17:00.000000000 +0200
+++ teeworlds-b177-r50edfd37-source/src/tools/dilate.cpp	2011-08-07 10:50:25.516018255 +0200
@@ -2,7 +2,10 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
 #include <base/math.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdio.h>
+#include <pnglite.h>
 
 typedef struct
 {
diff -ur teeworlds-b177-r50edfd37-source.orig/src/tools/tileset_borderfix.cpp teeworlds-b177-r50edfd37-source/src/tools/tileset_borderfix.cpp
--- teeworlds-b177-r50edfd37-source.orig/src/tools/tileset_borderfix.cpp	2011-07-31 23:17:00.000000000 +0200
+++ teeworlds-b177-r50edfd37-source/src/tools/tileset_borderfix.cpp	2011-08-07 10:59:37.986019889 +0200
@@ -1,7 +1,10 @@
 /* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>   
+#include <stdio.h> 
+#include <pnglite.h>
 
 typedef struct
 {
