--- teeworlds-0.6.0-source/bam.lua~	2011-04-26 13:22:16.000000000 -0500
+++ teeworlds-0.6.0-source/bam.lua	2011-04-26 13:24:47.412441073 -0500
@@ -134,6 +134,14 @@
 function build(settings)
 	--settings.objdir = Path("objs")
 	settings.cc.Output = Intermediate_Output
+	settings.cc.flags:Add(os.getenv ('CFLAGS'))
+	settings.cc.flags:Add(os.getenv ('CPPFLAGS'))
+	settings.cc.flags:Add(os.getenv ('LDFLAGS'))
+	settings.link.libs:Add("wavpack")
+	settings.link.libs:Add("z")
+	settings.link.libs:Add("pnglite")
+	settings.link.flags:Add('-lpng')
+
 
 	if config.compiler.driver == "cl" then
 		settings.cc.flags:Add("/wd4244")
