diff -U3 teeworlds-0.6.0-source.orig/src/engine/client/graphics.cpp teeworlds-0.6.0-source/src/engine/client/graphics.cpp
--- teeworlds-0.6.0-source.orig/src/engine/client/graphics.cpp	2011-04-09 14:15:25.000000000 -0500
+++ teeworlds-0.6.0-source/src/engine/client/graphics.cpp	2011-04-26 13:07:44.443440997 -0500
@@ -19,7 +19,7 @@
 #endif
 
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 #include <engine/shared/config.h>
 #include <engine/graphics.h>
diff -U3 teeworlds-0.6.0-source.orig/src/engine/client/sound.cpp teeworlds-0.6.0-source/src/engine/client/sound.cpp
--- teeworlds-0.6.0-source.orig/src/engine/client/sound.cpp	2011-04-09 14:15:25.000000000 -0500
+++ teeworlds-0.6.0-source/src/engine/client/sound.cpp	2011-04-26 13:09:36.005441312 -0500
@@ -10,7 +10,7 @@
 #include "sound.h"
 
 extern "C" { // wavpack
-	#include <engine/external/wavpack/wavpack.h>
+	#include <wavpack/wavpack.h>
 }
 #include <math.h>
 
#@@ -395,9 +395,6 @@
# 		dbg_msg("sound/wv", "failed to open %s: %s", pFilename, aError);
# 	}
# 
#-	io_close(ms_File);
#-	ms_File = NULL;
#-
# 	if(g_Config.m_Debug)
# 		dbg_msg("sound/wv", "loaded %s", pFilename);
# 
diff -U3 teeworlds-0.6.0-source.orig/src/tools/dilate.cpp teeworlds-0.6.0-source/src/tools/dilate.cpp
--- teeworlds-0.6.0-source.orig/src/tools/dilate.cpp	2011-04-09 14:15:25.000000000 -0500
+++ teeworlds-0.6.0-source/src/tools/dilate.cpp	2011-04-26 13:10:12.820441195 -0500
@@ -2,7 +2,11 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
 #include <base/math.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdio.h>
+#include <pnglite.h>
+
 
 typedef struct
 {
diff -U3 teeworlds-0.6.0-source.orig/bam.lua teeworlds-0.6.0-source/bam.lua
--- teeworlds-0.6.0-source.orig/bam.lua	2011-04-09 14:15:25.000000000 -0500
+++ teeworlds-0.6.0-source/bam.lua	2011-04-26 13:17:48.483440965 -0500
@@ -9,6 +9,8 @@
 config:Add(OptCCompiler("compiler"))
 config:Add(OptTestCompileC("stackprotector", "int main(){return 0;}", "-fstack-protector -fstack-protector-all"))
 config:Add(OptLibrary("zlib", "zlib.h", false))
+config:Add(OptLibrary("wavpack", "wavpack/wavpack.h", false))
+config:Add(OptLibrary("pnglite", "sys/types.h", false))
 config:Add(SDL.OptFind("sdl", true))
 config:Add(FreeType.OptFind("freetype", true))
 config:Finalize("config.lua")
@@ -177,8 +179,29 @@
 	end
 
 	-- build the small libraries
-	wavpack = Compile(settings, Collect("src/engine/external/wavpack/*.c"))
-	pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
+	-- compile wavpack if needed
+	if config.wavpack.value == 1 then
+		settings.link.libs:Add("wavpack")
+		if config.wavpack.include_path then
+			settings.cc.includes:Add(config.wavpack.include_path)
+		end
+		wavpack = {}
+	else
+		wavpack = Compile(settings, Collect("src/engine/external/wavpack/*.c"))
+		settings.cc.includes:Add("src/engine/external")
+	end
+
+	-- compile pnglite if needed
+	if config.pnglite.value == 1 then
+		settings.link.libs:Add("pnglite")
+		if config.pnglite.include_path then
+			settings.cc.includes:Add(config.pnglite.include_path)
+		end
+		pnglite = {}
+	else
+		pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
+		settings.cc.includes:Add("src/engine/external/pnglite")
+	end
 	
 	-- build game components
 	engine_settings = settings:Copy()
--- teeworlds-0.6.0-source.orig/src/tools/tileset_borderfix.cpp	2011-04-09 14:15:25.000000000 -0500
+++ teeworlds-0.6.0-source/src/tools/tileset_borderfix.cpp	2011-04-26 13:21:35.141440910 -0500
@@ -2,7 +2,10 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <stdlib.h>
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>   
+#include <stdio.h> 
+#include <pnglite.h>
 
 typedef struct
 {
--- teeworlds-0.6.0-source.orig/src/engine/client/sound.cpp	2011-04-26 14:11:17.267441002 -0500
+++ teeworlds-0.6.0-source/src/engine/client/sound.cpp	2011-04-26 14:11:02.897441026 -0500
@@ -315,6 +315,7 @@
 	CSample *pSample;
 	int SampleID = -1;
 	char aError[100];
+	char completefilename[102400];
 	WavpackContext *pContext;
 	
 	// don't waste memory on sound when we are stress testing
@@ -340,7 +341,9 @@
 		return -1;
 	pSample = &m_aSamples[SampleID];
 
-	pContext = WavpackOpenFileInput(ReadData, aError);
+	//GetPath(completefilename, sizeof(completefilename), pFilename, IOFLAG_READ);
+	pContext = WavpackOpenFileInput(completefilename, aError, 0, 0);
+
 	if (pContext)
 	{
 		int m_aSamples = WavpackGetNumSamples(pContext);
--- teeworlds-0.6.0-source.orig/src/engine/client/sound.cpp	2011-04-09 14:15:25.000000000 -0500
+++ teeworlds-0.6.0-source/src/engine/client/sound.cpp	2011-04-27 13:36:48.745660009 -0500
@@ -328,7 +329,11 @@
 	if(!m_pStorage)
 		return -1;
 
-	ms_File = m_pStorage->OpenFile(pFilename, IOFLAG_READ, IStorage::TYPE_ALL);
+	//char longname[50];
+	//strncpy(longname, pFilename, strlen(pFilename));
+	strncat(completefilename, "/usr/share/teeworlds/data/", 50);
+	strncat(completefilename, pFilename, strlen(pFilename));
+	ms_File = m_pStorage->OpenFile(completefilename, IOFLAG_READ, IStorage::TYPE_ALL);
 	if(!ms_File)
 	{
 		dbg_msg("sound/wv", "failed to open file. filename='%s'", pFilename);
