--- teeworlds-0.6.0-source/bam.lua.optflags	2011-04-09 16:15:25.000000000 -0300
+++ teeworlds-0.6.0-source/bam.lua	2011-04-11 20:23:25.608425996 -0300
@@ -132,6 +132,9 @@ end
 function build(settings)
 	--settings.objdir = Path("objs")
 	settings.cc.Output = Intermediate_Output
+	settings.cc.flags:Add(os.getenv ('CFLAGS'))
+	settings.cc.flags:Add(os.getenv ('CPPFLAGS'))
+	settings.cc.flags:Add(os.getenv ('LDFLAGS'))
 
 	if config.compiler.driver == "cl" then
 		settings.cc.flags:Add("/wd4244")
