diff -uNr teeworlds-0.6.2-source.orig/bam.lua teeworlds-0.6.2-source/bam.lua
--- teeworlds-0.6.2-source.orig/bam.lua	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/bam.lua	2013-07-02 18:08:17.477703604 +0400
@@ -11,6 +11,8 @@
 config:Add(OptTestCompileC("minmacosxsdk", "int main(){return 0;}", "-mmacosx-version-min=10.5 -isysroot /Developer/SDKs/MacOSX10.5.sdk"))
 config:Add(OptTestCompileC("macosxppc", "int main(){return 0;}", "-arch ppc"))
 config:Add(OptLibrary("zlib", "zlib.h", false))
+config:Add(OptLibrary("wavpack", "wavpack/wavpack.h", false))
+config:Add(OptLibrary("pnglite", "sys/types.h", false))
 config:Add(SDL.OptFind("sdl", true))
 config:Add(FreeType.OptFind("freetype", true))
 config:Finalize("config.lua")
@@ -142,6 +144,13 @@
 	
 	--settings.objdir = Path("objs")
 	settings.cc.Output = Intermediate_Output
+	settings.cc.flags:Add(os.getenv ('CFLAGS'))
+	settings.cc.flags:Add(os.getenv ('CPPFLAGS'))
+	settings.cc.flags:Add(os.getenv ('LDFLAGS'))
+	settings.link.libs:Add("wavpack")
+	settings.link.libs:Add("z")
+	settings.link.libs:Add("pnglite")
+	--settings.link.flags:Add('-lpng12')
 
 	if config.compiler.driver == "cl" then
 		settings.cc.flags:Add("/wd4244")
@@ -186,22 +195,6 @@
 		settings.link.libs:Add("shell32")
 	end
 
-	-- compile zlib if needed
-	if config.zlib.value == 1 then
-		settings.link.libs:Add("z")
-		if config.zlib.include_path then
-			settings.cc.includes:Add(config.zlib.include_path)
-		end
-		zlib = {}
-	else
-		zlib = Compile(settings, Collect("src/engine/external/zlib/*.c"))
-		settings.cc.includes:Add("src/engine/external/zlib")
-	end
-
-	-- build the small libraries
-	wavpack = Compile(settings, Collect("src/engine/external/wavpack/*.c"))
-	pnglite = Compile(settings, Collect("src/engine/external/pnglite/*.c"))
-
 	-- build game components
 	engine_settings = settings:Copy()
 	server_settings = engine_settings:Copy()
diff -uNr teeworlds-0.6.2-source.orig/src/engine/client/graphics.cpp teeworlds-0.6.2-source/src/engine/client/graphics.cpp
--- teeworlds-0.6.2-source.orig/src/engine/client/graphics.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/engine/client/graphics.cpp	2013-07-02 18:59:48.228212271 +0400
@@ -9,7 +9,7 @@
 #include "SDL_opengl.h"
 
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 #include <engine/shared/config.h>
 #include <engine/graphics.h>
diff -uNr teeworlds-0.6.2-source.orig/src/engine/client/graphics_threaded.cpp teeworlds-0.6.2-source/src/engine/client/graphics_threaded.cpp
--- teeworlds-0.6.2-source.orig/src/engine/client/graphics_threaded.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/engine/client/graphics_threaded.cpp	2013-07-02 18:10:13.657098950 +0400
@@ -6,7 +6,10 @@
 #include <base/tl/threading.h>
 
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <sys/types.h>
+#include <string.h>
+#include <stdio.h>
+#include <pnglite.h>
 
 #include <engine/shared/config.h>
 #include <engine/graphics.h>
diff -uNr teeworlds-0.6.2-source.orig/src/engine/client/sound.cpp teeworlds-0.6.2-source/src/engine/client/sound.cpp
--- teeworlds-0.6.2-source.orig/src/engine/client/sound.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/engine/client/sound.cpp	2013-07-02 18:11:42.810402161 +0400
@@ -13,7 +13,7 @@
 #include "sound.h"
 
 extern "C" { // wavpack
-	#include <engine/external/wavpack/wavpack.h>
+	#include <wavpack/wavpack.h>
 }
 #include <math.h>
 
@@ -338,6 +338,7 @@
 	CSample *pSample;
 	int SampleID = -1;
 	char aError[100];
+	char completefilename[102400] = "";
 	WavpackContext *pContext;
 
 	// don't waste memory on sound when we are stress testing
@@ -351,7 +352,9 @@
 	if(!m_pStorage)
 		return -1;
 
-	ms_File = m_pStorage->OpenFile(pFilename, IOFLAG_READ, IStorage::TYPE_ALL);
+        strncat(completefilename, "/usr/share/teeworlds/data/", 50);
+        strncat(completefilename, pFilename, strlen(pFilename));
+        ms_File = m_pStorage->OpenFile(completefilename, IOFLAG_READ, IStorage::TYPE_ALL);
 	if(!ms_File)
 	{
 		dbg_msg("sound/wv", "failed to open file. filename='%s'", pFilename);
@@ -363,7 +366,7 @@
 		return -1;
 	pSample = &m_aSamples[SampleID];
 
-	pContext = WavpackOpenFileInput(ReadData, aError);
+	pContext = WavpackOpenFileInput(completefilename, aError, 0, 0);
 	if (pContext)
 	{
 		int m_aSamples = WavpackGetNumSamples(pContext);
diff -uNr teeworlds-0.6.2-source.orig/src/tools/dilate.cpp teeworlds-0.6.2-source/src/tools/dilate.cpp
--- teeworlds-0.6.2-source.orig/src/tools/dilate.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/tools/dilate.cpp	2013-07-02 19:22:12.749938964 +0400
@@ -2,7 +2,7 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
 #include <base/math.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 typedef struct
 {
diff -uNr teeworlds-0.6.2-source.orig/src/tools/tileset_borderadd.cpp teeworlds-0.6.2-source/src/tools/tileset_borderadd.cpp
--- teeworlds-0.6.2-source.orig/src/tools/tileset_borderadd.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/tools/tileset_borderadd.cpp	2013-07-02 19:07:14.750771008 +0400
@@ -2,7 +2,7 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/math.h>
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 typedef struct
 {
diff -uNr teeworlds-0.6.2-source.orig/src/tools/tileset_borderfix.cpp teeworlds-0.6.2-source/src/tools/tileset_borderfix.cpp
--- teeworlds-0.6.2-source.orig/src/tools/tileset_borderfix.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/tools/tileset_borderfix.cpp	2013-07-02 19:17:57.887034675 +0400
@@ -1,7 +1,7 @@
 /* (c) Magnus Auvinen. See licence.txt in the root of the distribution for more information. */
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 typedef struct
 {
diff -uNr teeworlds-0.6.2-source.orig/src/tools/tileset_borderrem.cpp teeworlds-0.6.2-source/src/tools/tileset_borderrem.cpp
--- teeworlds-0.6.2-source.orig/src/tools/tileset_borderrem.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/tools/tileset_borderrem.cpp	2013-07-02 18:59:50.619220642 +0400
@@ -2,7 +2,7 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/math.h>
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 typedef struct
 {
diff -uNr teeworlds-0.6.2-source.orig/src/tools/tileset_borderset.cpp teeworlds-0.6.2-source/src/tools/tileset_borderset.cpp
--- teeworlds-0.6.2-source.orig/src/tools/tileset_borderset.cpp	2013-05-01 15:47:39.000000000 +0400
+++ teeworlds-0.6.2-source/src/tools/tileset_borderset.cpp	2013-07-02 19:03:44.194037097 +0400
@@ -2,7 +2,7 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #include <base/math.h>
 #include <base/system.h>
-#include <engine/external/pnglite/pnglite.h>
+#include <pnglite.h>
 
 typedef struct
 {
