diff -up gcc/config/alpha/linux-elf.h.hash-style gcc/config/alpha/linux-elf.h
--- gcc/config/alpha/linux-elf.h.hash-style	2010-12-09 11:27:07.000000000 -0200
+++ gcc/config/alpha/linux-elf.h	2011-03-26 20:33:12.613832144 -0300
@@ -41,7 +41,7 @@ along with GCC; see the file COPYING3.
 
 #define ELF_DYNAMIC_LINKER	LINUX_DYNAMIC_LINKER
 
-#define LINK_SPEC "-m elf64alpha %{G*} %{relax:-relax}		\
+#define LINK_SPEC "-m elf64alpha --hash-style=both %{G*} %{relax:-relax}	\
   %{O*:-O3} %{!O*:-O1}						\
   %{shared:-shared}						\
   %{!shared:							\
diff -up gcc/config/i386/linux.h.hash-style gcc/config/i386/linux.h
--- gcc/config/i386/linux.h.hash-style	2011-01-14 16:45:06.000000000 -0200
+++ gcc/config/i386/linux.h	2011-03-26 20:33:12.614831933 -0300
@@ -104,7 +104,7 @@ along with GCC; see the file COPYING3.
   { "dynamic_linker", LINUX_DYNAMIC_LINKER }
 
 #undef	LINK_SPEC
-#define LINK_SPEC "-m %(link_emulation) %{shared:-shared} \
+#define LINK_SPEC "-m %(link_emulation) --hash-style=both %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
diff -up gcc/config/i386/linux64.h.hash-style gcc/config/i386/linux64.h
--- gcc/config/i386/linux64.h.hash-style	2011-03-02 19:35:36.000000000 -0300
+++ gcc/config/i386/linux64.h	2011-03-26 20:33:12.614831933 -0300
@@ -78,7 +78,7 @@ see the files COPYING3 and COPYING.RUNTI
  %{!mno-sse2avx:%{mavx:-msse2avx}} %{msse2avx:%{!mavx:-msse2avx}}"
 
 #undef	LINK_SPEC
-#define LINK_SPEC "%{" SPEC_64 ":-m elf_x86_64} %{" SPEC_32 ":-m elf_i386} \
+#define LINK_SPEC "%{" SPEC_64 ":-m elf_x86_64} %{" SPEC_32 ":-m elf_i386} --hash-style=both \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
diff -up gcc/config/ia64/linux.h.hash-style gcc/config/ia64/linux.h
--- gcc/config/ia64/linux.h.hash-style	2010-12-09 11:27:07.000000000 -0200
+++ gcc/config/ia64/linux.h	2011-03-26 20:33:12.614831933 -0300
@@ -64,7 +64,7 @@ do {						\
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
-#define LINK_SPEC "\
+#define LINK_SPEC "--hash-style=both \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
diff -up gcc/config/rs6000/linux64.h.hash-style gcc/config/rs6000/linux64.h
--- gcc/config/rs6000/linux64.h.hash-style	2011-03-07 04:50:23.000000000 -0300
+++ gcc/config/rs6000/linux64.h	2011-03-26 20:36:14.682295107 -0300
@@ -389,11 +389,11 @@ extern int dot_symbols;
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER64, UCLIBC_DYNAMIC_LINKER64)
 
 
-#define LINK_OS_LINUX_SPEC32 "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC32 "-m elf32ppclinux --hash-style=both %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " LINUX_DYNAMIC_LINKER32 "}}"
 
-#define LINK_OS_LINUX_SPEC64 "-m elf64ppc %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC64 "-m elf64ppc --hash-style=both %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " LINUX_DYNAMIC_LINKER64 "}}"
 
diff -up gcc/config/rs6000/sysv4.h.hash-style gcc/config/rs6000/sysv4.h
--- gcc/config/rs6000/sysv4.h.hash-style	2011-03-07 04:50:23.000000000 -0300
+++ gcc/config/rs6000/sysv4.h	2011-03-26 20:33:12.615831722 -0300
@@ -830,7 +830,7 @@ extern int fixuplabelno;
 #define LINUX_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 
-#define LINK_OS_LINUX_SPEC "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC "-m elf32ppclinux --hash-style=both %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " LINUX_DYNAMIC_LINKER "}}"
 
diff -up gcc/config/s390/linux.h.hash-style gcc/config/s390/linux.h
--- gcc/config/s390/linux.h.hash-style	2010-12-09 11:27:07.000000000 -0200
+++ gcc/config/s390/linux.h	2011-03-26 20:33:12.613832144 -0300
@@ -77,7 +77,7 @@ along with GCC; see the file COPYING3.
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
-  "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
+  "%{m31:-m elf_s390}%{m64:-m elf64_s390} --hash-style=both \
    %{shared:-shared} \
    %{!shared: \
       %{static:-static} \
diff -up gcc/config/sparc/linux.h.hash-style gcc/config/sparc/linux.h
--- gcc/config/sparc/linux.h.hash-style	2011-01-26 18:30:12.000000000 -0200
+++ gcc/config/sparc/linux.h	2011-03-26 20:33:12.613832144 -0300
@@ -74,7 +74,7 @@ along with GCC; see the file COPYING3.
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc --hash-style=both -Y P,/usr/lib %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!static: \
diff -up gcc/config/sparc/linux64.h.hash-style gcc/config/sparc/linux64.h
--- gcc/config/sparc/linux64.h.hash-style	2011-02-17 11:57:21.000000000 -0200
+++ gcc/config/sparc/linux64.h	2011-03-26 20:33:12.614831933 -0300
@@ -113,7 +113,7 @@ along with GCC; see the file COPYING3.
   { "link_arch_default", LINK_ARCH_DEFAULT_SPEC },	  \
   { "link_arch",	 LINK_ARCH_SPEC },
 
-#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/usr/lib %{shared:-shared} \
+#define LINK_ARCH32_SPEC "-m elf32_sparc --hash-style=both -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -121,7 +121,7 @@ along with GCC; see the file COPYING3.
       %{static:-static}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc --hash-style=both -Y P,%R/usr/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -193,7 +193,7 @@ along with GCC; see the file COPYING3.
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc --hash-style=both -Y P,%R/usr/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
