diff -urNp pax-3.4-orig/lib/fts.c pax-3.4/lib/fts.c
--- pax-3.4-orig/lib/fts.c	2005-07-29 09:55:25.000000000 +0200
+++ pax-3.4/lib/fts.c	2011-06-29 14:50:05.256431237 +0200
@@ -465,12 +465,8 @@ name:		t = sp->fts_path + NAPPEND(p->fts
  */
 /* ARGSUSED */
 int
-fts_set(sp, p, instr)
-	FTS *sp;
-	FTSENT *p;
-	int instr;
+fts_set(FTS *sp __attribute__ ((__unused__)), FTSENT *p, int instr)
 {
-sp = NULL;	/* stop compiler whinging */
 	if (instr != 0 && instr != FTS_AGAIN && instr != FTS_FOLLOW &&
 	    instr != FTS_NOINSTR && instr != FTS_SKIP) {
 		errno = EINVAL;
Binary files pax-3.4-orig/lib/.fts.c.swp and pax-3.4/lib/.fts.c.swp differ
