diff -up maradns-1.4.03/server/Makefile.recursive.ldflags maradns-1.4.03/server/Makefile.recursive
--- maradns-1.4.03/server/Makefile.recursive.ldflags	2007-08-16 10:35:43.000000000 -0300
+++ maradns-1.4.03/server/Makefile.recursive	2010-03-21 16:09:29.192551000 -0300
@@ -52,8 +52,8 @@ udpsuccess.o: udpsuccess.c
 	$(CC) -c $(FLAGS) -o udpsuccess.o udpsuccess.c
 
 maradns: MaraDNS.c $(OBJECTS) MaraDNS_locale.h
-	$(CC) $(FLAGS) -DVERSION=\"$(VERSION)\" -DCOMPILED=\"$(COMPILED)\" -o maradns MaraDNS.c $(OBJECTS) -lpthread
+	$(CC) $(FLAGS) $(LDFLAGS) -DVERSION=\"$(VERSION)\" -DCOMPILED=\"$(COMPILED)\" -o maradns MaraDNS.c $(OBJECTS) -lpthread
 
 simple: simple.c $(XOBJECTS) MaraDNS_locale.h
-	$(CC) $(FLAGS) -DVERSION=\"$(VERSION)\" -DCOMPILED=\"$(COMPILED)\" -o simple $(OBJECTS) simple.c 
+	$(CC) $(FLAGS) $(LDFLAGS) -DVERSION=\"$(VERSION)\" -DCOMPILED=\"$(COMPILED)\" -o simple $(OBJECTS) simple.c 
 
diff -up maradns-1.4.03/tcp/Makefile.recursive.ldflags maradns-1.4.03/tcp/Makefile.recursive
--- maradns-1.4.03/tcp/Makefile.recursive.ldflags	2006-07-18 00:50:44.000000000 -0300
+++ maradns-1.4.03/tcp/Makefile.recursive	2010-03-21 16:09:29.192551000 -0300
@@ -26,11 +26,11 @@ clean:
 	rm -f core $(EXECS) $(TCPOBJS) *.exe
 
 getzone: getzone.c $(OBJECTS) 
-	$(CC) $(FLAGS) -o getzone getzone.c $(OBJECTS)
+	$(CC) $(FLAGS) $(LDFLAGS) -o getzone getzone.c $(OBJECTS)
 
 fetchzone: fetchzone.c $(OBJECTS) 
-	$(CC) $(FLAGS) -o fetchzone fetchzone.c $(OBJECTS) 
+	$(CC) $(FLAGS) $(LDFLAGS) -o fetchzone fetchzone.c $(OBJECTS) 
 
 zoneserver: zoneserver.c zoneserver_en.h $(OBJECTS)
-	$(CC) $(FLAGS) -DVERSION=\"$(VERSION)\" -o zoneserver zoneserver.c $(OBJECTS) $(RNGOBJS)
+	$(CC) $(FLAGS) $(LDFLAGS) -DVERSION=\"$(VERSION)\" -o zoneserver zoneserver.c $(OBJECTS) $(RNGOBJS)
 
diff -up maradns-1.4.03/tools/Makefile.recursive.ldflags maradns-1.4.03/tools/Makefile.recursive
--- maradns-1.4.03/tools/Makefile.recursive.ldflags	2006-01-02 14:59:33.000000000 -0200
+++ maradns-1.4.03/tools/Makefile.recursive	2010-03-21 16:09:29.193550900 -0300
@@ -18,17 +18,17 @@ clean:
 	rm -f core $(EXECS) *.exe
 
 askmara: askmara.c askmara_labels_en.h $(OBJECTS) $(Q) $(OUTOBJ)
-	$(CC) $(FLAGS) -o askmara askmara.c $(OBJECTS) $(Q) $(OUTOBJ)
+	$(CC) $(FLAGS) $(LDFLAGS) -o askmara askmara.c $(OBJECTS) $(Q) $(OUTOBJ)
 
 askmara-tcp: askmara-tcp.c askmara_labels_en.h $(OBJECTS) $(ROBJECTS) $(POBJECTS)
-	$(CC) $(FLAGS) -o askmara-tcp askmara-tcp.c $(OBJECTS) $(ROBJECTS) $(POBJECTS)
+	$(CC) $(FLAGS) $(LDFLAGS) -o askmara-tcp askmara-tcp.c $(OBJECTS) $(ROBJECTS) $(POBJECTS)
 
 benchmark: benchmark.c askmara_labels_en.h $(OBJECTS) $(POBJECTS)
-	$(CC) $(FLAGS) -o benchmark benchmark.c $(OBJECTS) $(POBJECTS)
+	$(CC) $(FLAGS) $(LDFLAGS) -o benchmark benchmark.c $(OBJECTS) $(POBJECTS)
 
 duende: duende.c 
-	$(CC) $(FLAGS) -o duende duende.c
+	$(CC) $(FLAGS) $(LDFLAGS) -o duende duende.c
 
 strip.control: strip.control.c
-	$(CC) $(FLAGS) -o strip.control strip.control.c
+	$(CC) $(FLAGS) $(LDFLAGS) -o strip.control strip.control.c
 
