Index: kcontrol/colors/colorscm.cpp
===================================================================
--- kcontrol/colors/colorscm.cpp
+++ kcontrol/colors/colorscm.cpp	2010-06-20 23:16:17.000000000 +0200
@@ -116,6 +116,9 @@
     schemeList->setCurrentItem(currentitem);
     schemeList->blockSignals(false);
 
+    KConfigGroup generalGroup(m_config, "General");
+    m_schemeName = generalGroup.readEntry("Name", "Default");
+
     // add default entry
     m_config->setReadDefaults(true);
     icon = createSchemePreviewIcon(m_config);
@@ -206,6 +209,9 @@
     KSharedConfigPtr temp = m_config;
     m_config = config;
 
+    KConfigGroup generalGroup(config, "General");
+    m_schemeName = generalGroup.readEntry("Name", "Default");
+
     updateColorSchemes();
     updateEffectsPage(); // intentionally before swapping back m_config
 
@@ -217,6 +223,12 @@
     updatePreviews();
 
     m_loadedSchemeHasUnsavedChanges = false;
+    generalGroup = KConfigGroup(m_config, "General");
+    generalGroup.writeEntry("Name", m_schemeName);
+    
+    KConfigGroup kdeGroup(m_config, "KDE");
+    kdeGroup.writeEntry("colorScheme", m_schemeName);
+
     //m_changed = false;
 }
 
Index: kcontrol/colors/colorscm.h
===================================================================
--- kcontrol/colors/colorscm.h
+++ kcontrol/colors/colorscm.h	2010-06-20 23:19:30.000000000 +0200
@@ -202,6 +202,7 @@
     KSharedConfigPtr m_config;
 
     bool m_disableUpdates;
+    QString m_schemeName;
     bool m_loadedSchemeHasUnsavedChanges;
     // don't (re)load the scheme, only select it in schemeList
     bool m_dontLoadSelectedScheme;
