diff -up kdebase-workspace-4.6.3/startkde.cmake.multilib_qt_plugin_path kdebase-workspace-4.6.3/startkde.cmake
--- kdebase-workspace-4.6.3/startkde.cmake.multilib_qt_plugin_path	2011-05-21 20:32:02.373251624 -0500
+++ kdebase-workspace-4.6.3/startkde.cmake	2011-05-21 21:01:37.340718071 -0500
@@ -213,6 +213,10 @@ if test -n "$QT_PLUGIN_PATH"; then
 else
   QT_PLUGIN_PATH="`kde4-config --path qtplugins`"
 fi
+# hack in multilib support
+if ! echo ${QT_PLUGIN_PATH} | /bin/grep -q /usr/lib/kde4/plugins ; then
+  QT_PLUGIN_PATH=${QT_PLUGIN_PATH}:/usr/lib/kde4/plugins
+fi
 export QT_PLUGIN_PATH
 
 # Set a left cursor instead of the standard X11 "X" cursor, since I've heard
