--- powerdevil/daemon/BackendConfig.cmake	2010-11-25 18:40:32.000000000 +0100
+++ powerdevil/daemon/BackendConfig.cmake.old	2010-12-04 14:09:50.000000000 +0100
@@ -35,17 +35,17 @@
 
 ########################## HAL Backend #####################################
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/backends/hal
-                    ${CMAKE_CURRENT_BINARY_DIR}/backends/hal)
-
-set(powerdevilhalbackend_SRCS
-    backends/hal/halsuspendjob.cpp
-    backends/hal/powerdevilhalbackend.cpp
-)
-
-set(powerdevilhalbackend_LIBS ${KDE4_SOLID_LIBS})
+#include_directories(${CMAKE_CURRENT_SOURCE_DIR}/backends/hal
+#                    ${CMAKE_CURRENT_BINARY_DIR}/backends/hal)
+#
+#set(powerdevilhalbackend_SRCS
+#    backends/hal/halsuspendjob.cpp
+#    backends/hal/powerdevilhalbackend.cpp
+#)
+#
+#set(powerdevilhalbackend_LIBS ${KDE4_SOLID_LIBS})
 
 ########################## Daemon variables ################################
 
-set(POWERDEVIL_BACKEND_SRCS ${powerdevilupowerbackend_SRCS} ${powerdevilhalbackend_SRCS})
-set(POWERDEVIL_BACKEND_LIBS ${powerdevilupowerbackend_LIBS} ${powerdevilhalbackend_LIBS})
+set(POWERDEVIL_BACKEND_SRCS ${powerdevilupowerbackend_SRCS})
+set(POWERDEVIL_BACKEND_LIBS ${powerdevilupowerbackend_LIBS})
