Index: plasma/generic/applets/battery/battery.cpp
===================================================================
--- plasma/generic/applets/battery/battery.cpp
+++ plasma/generic/applets/battery/battery.cpp	2010-06-21 00:25:14.000000000 +0200
@@ -177,7 +177,7 @@
 void Battery::configChanged()
 {
     KConfigGroup cg = config();
-    m_showBatteryLabel = cg.readEntry("showBatteryString", false);
+    m_showBatteryLabel = cg.readEntry("showBatteryString", true);
     m_showRemainingTime = cg.readEntry("showRemainingTime", false);
     m_showMultipleBatteries = cg.readEntry("showMultipleBatteries", false);
 
