diff --git a/kwin/opengltest/opengltest.cpp b/kwin/opengltest/opengltest.cpp
index d2d8f70..2a65c05 100644
--- a/kwin/opengltest/opengltest.cpp
+++ b/kwin/opengltest/opengltest.cpp
@@ -87,7 +87,8 @@ int main(int argc, char *argv[])
         return 0;
 
     // The Intel driver doesn't have DRI2 in the renderer string
-    if (strstr((const char *)renderer, "GEM"))
+    // but if Intel is present DRI2 is in use.
+    if (strstr((const char *)renderer, "Intel"))
         return 0;
 
     if (strstr((const char *)renderer, "Gallium"))
