diff -up kdebase-workspace-4.4.80/plasma/generic/applets/battery/battery.cpp.showremainingtime kdebase-workspace-4.4.80/plasma/generic/applets/battery/battery.cpp
--- kdebase-workspace-4.4.80/plasma/generic/applets/battery/battery.cpp.showremainingtime	2010-05-16 12:04:32.000000000 +0200
+++ kdebase-workspace-4.4.80/plasma/generic/applets/battery/battery.cpp	2010-05-21 12:46:26.437154419 +0200
@@ -688,7 +688,11 @@ void Battery::updateStatus()
             m_acInfoLabel->setText(i18n("Not plugged in"));
         }
 
-        if (showRemainingTime && m_remainingMSecs > 0) {
+        // Always show the remaining time in the popup.
+        // We don't just set m_showRemainingTime to true by default because
+        // that'd try to use the time also for the icon. That won't fit on
+        // small panels.
+        if (m_remainingMSecs > 0) {
             m_remainingTimeLabel->show();
             m_remainingInfoLabel->show();
             // we don't have too much accuracy so only give hours and minutes
