diff -up kdebase-workspace-4.6.2/ksysguard/ksysguardd/modules.h.ksysguard-add-missing-module kdebase-workspace-4.6.2/ksysguard/ksysguardd/modules.h
--- kdebase-workspace-4.6.2/ksysguard/ksysguardd/modules.h.ksysguard-add-missing-module	2011-04-01 12:52:43.000000000 +0200
+++ kdebase-workspace-4.6.2/ksysguard/ksysguardd/modules.h	2011-04-08 17:02:27.553086372 +0200
@@ -23,6 +23,7 @@
 #ifndef KSG_MODULES_H
 #define KSG_MODULES_H
 
+#include "config-ksysguardd.h"
 #include "Command.h"
 #include "conf.h"
 #include "ksysguardd.h"
