diff -up kdebase-workspace-4.4.4/kstartupconfig/kdostartupconfig.cpp.kdebug183143 kdebase-workspace-4.4.4/kstartupconfig/kdostartupconfig.cpp
--- kdebase-workspace-4.4.4/kstartupconfig/kdostartupconfig.cpp.kdebug183143	2009-11-30 18:30:43.000000000 -0600
+++ kdebase-workspace-4.4.4/kstartupconfig/kdostartupconfig.cpp	2010-06-23 09:09:00.753934230 -0500
@@ -107,6 +107,8 @@ int main( int argc, char **argv )
             break;
         QString tmp = line;
         QString file, group, key, def;
+        QString screenN, refresh;
+        QRect rect;
         file = get_entry( &tmp );
         group = get_entry( &tmp );
         key = get_entry( &tmp );
@@ -115,21 +117,52 @@ int main( int argc, char **argv )
             return 6;
         if( group.startsWith( '[' ) && group.endsWith( ']' ) )
             { // whole config group
-            KConfig cfg( file );
-            group = group.mid( 1, group.length() - 2 );
-            KConfigGroup cg(&cfg, group);
-            QMap< QString, QString > entries = cg.entryMap( );
-            startupconfig << "# " << line << "\n";
-            for( QMap< QString, QString >::ConstIterator it = entries.constBegin();
-                 it != entries.constEnd();
-                 ++it )
+            if( group.mid( 1, 6 ) == "Screen" )
+                {
+                screenN = group.mid( 7, 1 );
+                KConfig cfg( file );
+                group = "Screen_" + screenN + "_Output_default";
+                KConfigGroup cg(&cfg, group);
+                startupconfig << "# " << line << "\n";
+                refresh = cg.readEntry("RefreshRate", "");
+                if ( !refresh.isEmpty() )
+                    {
+                    startupconfig << file.replace( ' ', '_' ).toLower()
+                        << "_" << "screen" << screenN
+                        << "_" << "refresh"
+                        << "=\"" << refresh << "\"\n";
+                    }
+                rect =  cg.readEntry("Rect", QRect());
+                if ( !rect.isNull() )
+                    {
+                    startupconfig << file.replace( ' ', '_' ).toLower()
+                        << "_" << "screen" << screenN
+                        << "_" << "width"
+                        << "=\"" << rect.width() << "\"\n";
+                    startupconfig << file.replace( ' ', '_' ).toLower()
+                        << "_" << "screen" << screenN
+                        << "_" << "height"
+                        << "=\"" << rect.height() << "\"\n";
+                    }
+                }
+            else
                 {
-                QString key = it.key();
-                QString value = *it;
-                startupconfig << file.replace( ' ', '_' ).toLower()
-                    << "_" << group.replace( ' ', '_' ).toLower()
-                    << "_" << key.replace( ' ', '_' ).toLower()
-                    << "=\"" << value.replace( "\"", "\\\"" ) << "\"\n";
+                KConfig cfg( file );
+                group = group.mid( 1, group.length() - 2 );
+                KConfigGroup cg(&cfg, group);
+                QMap< QString, QString > entries = cg.entryMap( );
+                startupconfig << "# " << line << "\n";
+                for( QMap< QString, QString >::ConstIterator it = entries.constBegin();
+                     it != entries.constEnd();
+                     ++it )
+                    {
+                    QString key = it.key();
+                    QString value = *it;
+                    startupconfig << file.replace( ' ', '_' ).toLower()
+                        << "_" << group.replace( ' ', '_' ).toLower()
+                        << "_" << key.replace( ' ', '_' ).toLower()
+                        << "=\"" << value.replace( "\"", "\\\"" ) << "\"\n";
+                    }
                 }
             }
         else
diff -up kdebase-workspace-4.4.4/startkde.cmake.kdebug183143 kdebase-workspace-4.4.4/startkde.cmake
--- kdebase-workspace-4.4.4/startkde.cmake.kdebug183143	2010-06-23 09:09:00.700934859 -0500
+++ kdebase-workspace-4.4.4/startkde.cmake	2010-06-23 09:09:49.058684685 -0500
@@ -94,11 +94,11 @@ kcminputrc Mouse cursorTheme 'Oxygen_Bla
 kcminputrc Mouse cursorSize ''
 ksplashrc KSplash Theme Default
 ksplashrc KSplash Engine KSplashX
-kcmrandrrc Display ApplyOnStartup false
-kcmrandrrc [Screen0]
-kcmrandrrc [Screen1]
-kcmrandrrc [Screen2]
-kcmrandrrc [Screen3]
+krandrrc Display ApplyOnStartup true
+krandrrc [Screen0]
+krandrrc [Screen1]
+krandrrc [Screen2]
+krandrrc [Screen3]
 kcmfonts General forceFontDPI 0
 kdeglobals Locale Language '' # trigger requesting languages from KLocale
 kdeglobals Locale Country ''
@@ -136,20 +136,20 @@ if test -n "$kcminputrc_mouse_cursorthem
     fi
 fi
 
-if test "$kcmrandrrc_display_applyonstartup" = "true"; then
+if test "$krandrrc_display_applyonstartup" = "true"; then
     # 4 screens is hopefully enough
     for scrn in 0 1 2 3; do
         args=
-        width="\$kcmrandrrc_screen${scrn}_width" ; eval "width=$width"
-        height="\$kcmrandrrc_screen${scrn}_height" ; eval "height=$height"
+        width="\$krandrrc_screen${scrn}_width" ; eval "width=$width"
+        height="\$krandrrc_screen${scrn}_height" ; eval "height=$height"
         if test -n "${width}" -a -n "${height}"; then
             args="$args -s ${width}x${height}"
         fi
-        refresh="\$kcmrandrrc_screen${scrn}_refresh" ; eval "refresh=$refresh"
+        refresh="\$krandrrc_screen${scrn}_refresh" ; eval "refresh=$refresh"
         if test -n "${refresh}"; then
             args="$args -r ${refresh}"
         fi
-        rotation="\$kcmrandrrc_screen${scrn}_rotation" ; eval "rotation=$rotation"
+        rotation="\$krandrrc_screen${scrn}_rotation" ; eval "rotation=$rotation"
         if test -n "${rotation}"; then
             case "${rotation}" in
                 0)
@@ -166,11 +166,11 @@ if test "$kcmrandrrc_display_applyonstar
                     ;;
             esac
         fi
-        reflectx="\$kcmrandrrc_screen${scrn}_reflectx" ; eval "reflectx=$reflectx"
+        reflectx="\$krandrrc_screen${scrn}_reflectx" ; eval "reflectx=$reflectx"
         if test "${refrectx}" = "true"; then
             args="$args -x"
         fi
-        reflecty="\$kcmrandrrc_screen${scrn}_reflecty" ; eval "reflecty=$reflecty"
+        reflecty="\$krandrrc_screen${scrn}_reflecty" ; eval "reflecty=$reflecty"
         if test "${refrecty}" = "true"; then
             args="$args -y"
         fi
