Index: kdebase/workspace/plasma/netbook/containments/newspaper/plasma-containment-newspaper.desktop
===================================================================
--- kdebase/workspace/plasma/netbook/containments/newspaper/plasma-containment-newspaper.desktop	(révision 1122354)
+++ kdebase/workspace/plasma/netbook/containments/newspaper/plasma-containment-newspaper.desktop	(révision 1122355)
@@ -105,7 +105,7 @@
 Comment[zh_TW]=將元件放到兩個欄位的活動
 
 
-Icon=
+Icon=knewsticker
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet,Plasma/Containment
 X-Plasma-ContainmentCategories=netbook,desktop
