commit 09f7dbda30fea57a83346bf6cfffaffef20e2e45
Author: Lukas Tinkl <lukas@kde.org>
Date:   Thu Apr 7 13:00:51 2011 +0200

    give nv_backlight lower preference
    
    CCBUG: 257948

diff --git a/powerdevil/daemon/backends/upower/backlighthelper.cpp b/powerdevil/daemon/backends/upower/backlighthelper.cpp
index e1db55e..a0cbca8 100644
--- a/powerdevil/daemon/backends/upower/backlighthelper.cpp
+++ b/powerdevil/daemon/backends/upower/backlighthelper.cpp
@@ -34,11 +34,11 @@ void BacklightHelper::init()
 {
     // find the first existing device with backlight support
     QStringList interfaces;
-    interfaces << "nv_backlight" << "asus_laptop" << "toshiba"
+    interfaces << "asus_laptop" << "toshiba"
                << "eeepc" << "thinkpad_screen" << "acpi_video0"
                << "mbp_backlight" << "acpi_video1"
                << "fujitsu-laptop" << "samsung"
-               << "nvidia_backlight" << "dell_backlight" << "sony";
+               << "nv_backlight"  << "nvidia_backlight" << "dell_backlight" << "sony";
 
     QDir dir;
     foreach (const QString & interface, interfaces) {
