diff -up kdebase-workspace-4.5.1/kcontrol/kfontinst/apps/kfontview.desktop.dt_validate kdebase-workspace-4.5.1/kcontrol/kfontinst/apps/kfontview.desktop
--- kdebase-workspace-4.5.1/kcontrol/kfontinst/apps/kfontview.desktop.dt_validate	2010-08-04 14:13:52.000000000 -0500
+++ kdebase-workspace-4.5.1/kcontrol/kfontinst/apps/kfontview.desktop	2010-08-29 15:35:01.644474288 -0500
@@ -90,7 +90,7 @@ Exec=kfontview %U
 Icon=kfontview
 X-KDE-StartupNotify=true
 Type=Application
-MimeType=application/x-font-ttf;application/x-font-type1;application/x-font-otf;application/x-font-pcf;application/x-font-bdf;fonts/package;
+MimeType=application/x-font-ttf;application/x-font-type1;application/x-font-otf;application/x-font-pcf;application/x-font-bdf;
 GenericName=Font Viewer
 GenericName[af]=Skriftipe Besigter
 GenericName[ar]=عارض الخطوط
