--- perl_ext/Exports.c.perl514	2011-07-21 19:01:59.437585957 -0300
+++ perl_ext/Exports.c	2011-07-21 19:02:05.936064031 -0300
@@ -181,16 +181,20 @@ static char **export_tags(char *tag) {
    }
 }
 
+#ifndef GvCV_set
+#define GvCV_set(gv,cv) GvCV((gv)) = (cv)
+#endif
 
 void export_cv(SV *class, SV *caller, char *sub) 
 { 
     GV *gv; 
+    CV *cv;
 #if 0
     fprintf(stderr, "Here is the result: *%s::%s = \\&%s::%s\n", 
             SvPVX(caller), sub, SvPVX(class), sub); 
 #endif 
     gv = gv_fetchpv(form("%s::%s",SvPVX( caller), sub), TRUE, SVt_PVCV); 
-    GvCV(gv) = perl_get_cv(form("%s::%s", SvPVX(class), sub), TRUE); 
+    GvCV_set(gv,perl_get_cv(form("%s::%s", SvPVX(class), sub), TRUE));
     GvIMPORTED_CV_on(gv); 
     GvMULTI_on(gv);
 } 
