--- util/Makefile.am.dso	2011-07-21 19:21:45.028462478 -0300
+++ util/Makefile.am	2011-07-21 19:22:23.374170832 -0300
@@ -114,16 +114,16 @@ makefdb_LDADD = libutil.la $(top_builddi
 makefdb_CFLAGS = -DMAKE_FDB
 
 gif2mask_SOURCES = gif2mask.c
-gif2mask_LDADD = $(top_builddir)/src/libming.la
+gif2mask_LDADD = $(top_builddir)/src/libming.la $(GIFLIB) $(MATHLIB) $(ZLIB)
 
 png2dbl_SOURCES = png2dbl.c
-png2dbl_LDADD = $(top_builddir)/src/libming.la
+png2dbl_LDADD = $(top_builddir)/src/libming.la $(PNGLIB) $(MATHLIB) $(ZLIB)
 
 dbl2png_SOURCES = dbl2png.c
-dbl2png_LDADD = $(top_builddir)/src/libming.la
+dbl2png_LDADD = $(top_builddir)/src/libming.la $(PNGLIB) $(ZLIB)
 
 gif2dbl_SOURCES = gif2dbl.c
-gif2dbl_LDADD = $(top_builddir)/src/libming.la
+gif2dbl_LDADD = $(top_builddir)/src/libming.la $(GIFLIB) $(MATHLIB) $(ZLIB)
 
 makeswf_SOURCES = makeswf.c makeswf_utils.c
 makeswf_LDADD = $(top_builddir)/src/libming.la
