--- fpc.spec	2011-03-28 19:13:32.000000000 -0300
+++ fpc-fdist/fpc.spec	2015-04-22 19:13:25.434652023 -0300
@@ -1,6 +1,6 @@
 Name:           fpc
-Version:        2.4.2
-Release:        2%{?dist}
+Version:        2.6.4
+Release:        1%{?dist}
 Summary:        Free Pascal Compiler
 
 Group:          Development/Languages
@@ -10,10 +10,25 @@ Source0:        ftp://ftp.freepascal.org
 # This is only needed when useprebuiltcompiler is defined.
 # But it's not in an 'if defined' block, since the file has to be included in the srpm
 # Thus you should enable this line when useprebuildcompiler is defined for any target
+#
+# Last used for ARM bootstrap, for the ARM boostrap, the following 2 debs:
+#  ftp://ftp.nluug.nl/pub/os/Linux/distr/debian/pool/main/f/fpc/fp-compiler-2.6.2_2.6.2-5_armhf.deb
+#  ftp://ftp.nluug.nl/pub/os/Linux/distr/debian/pool/main/f/fpc/fp-units-base-2.6.2_2.6.2-5_armhf.deb
+# were extracted, usr renamed to fpc-2.6.2-bin-armv7hf and then tarred
+# Source1:        fpc-2.6.2-bin-armv7hf.tar.gz
 # Source1:        http://www.cnoc.nl/fpc/%{name}-%{version}.compiler.bin.tar.gz
-Patch0:         fpc-2.4.2-samplecfg_32and64bit.patch
-Patch1:         fpc-2.4.2-build-id.patch
-Patch2:         fpc-2.4.2-r15671.patch
+# 
+# Configuration templates:
+Source2:        fpc.cft
+Source3:        fppkg.cfg
+Source4:        default.cft
+Patch0:         fpc-2.6.2-r22920.patch
+# armhf support patches, courtesy of Debian
+Patch1:         armhf.diff
+Patch2:         regenerate_messages.diff
+Patch3:         armhf-linker-path.diff
+# Patch for the fpmake-packages (fcl-web and fastcgi) to be installed in lib64
+Patch4:         fpmake-lib64.diff
 
 Requires:       gpm, ncurses, binutils
 %if ! %{defined useprebuiltcompiler}
@@ -21,33 +36,33 @@ BuildRequires:  fpc
 %endif
 BuildRequires:  tetex, tetex-latex, tetex-fonts
 
-ExcludeArch:    s390 s390x
+ExclusiveArch:  %{arm} %{ix86} x86_64 ppc ppc64
 
 %description
 
-Freepascal is a free 32/64bit Pascal Compiler. It comes with a run-time
+Free Pascal is a free 32/64bit Pascal Compiler. It comes with a run-time
 library and is fully compatible with Turbo Pascal 7.0 and nearly Delphi
 compatible. Some extensions are added to the language, like function
 overloading and generics. Shared libraries can be linked. This package
-contains commandline compiler and utils. Provided units are the runtime
+contains command-line compiler and utils. Provided units are the runtime
 library (RTL), free component library (FCL) and the base and extra packages.
 
 %package doc
-Summary:        Freepascal Compiler - documentation and examples
+Summary:        Free Pascal Compiler - documentation and examples
 Group:          Documentation
 
 %description doc
 
 The fpc-doc package contains the documentation (in pdf format) and examples
-of Freepascal.
+of Free Pascal.
 
 %package src
-Summary:        Freepascal Compiler - sources
+Summary:        Free Pascal Compiler - sources
 Group:          Development/Languages
 
 %description src
 
-The fpc-src package contains the sources of Freepascal, for documentation or
+The fpc-src package contains the sources of Free Pascal, for documentation or
 automatical-code generation purposes.
 
 %define smart _smart 
@@ -60,11 +75,16 @@ automatical-code generation purposes.
 %define ppcname ppcx64
 %else
 %ifarch ppc64
+%define ppcname ppcppc64
+%else
+%ifarch %{arm}
+%define ppcname ppcarm
 %else
 %define ppcname ppc386
 %endif
 %endif
 %endif
+%endif
 
 %prep
 %if %{defined useprebuiltcompiler}
@@ -73,8 +93,10 @@ automatical-code generation purposes.
 %setup -n fpcbuild-%{version} -q
 %endif
 %patch0
-%patch1
-%patch2
+%patch1 -p1
+%patch2 -p1
+%patch3 -p1
+%patch4 -p1
 
 %build
 # The source-files:
@@ -85,7 +107,7 @@ rm -rf fpc_src/packages/extra/amunits
 rm -rf fpc_src/packages/extra/winunits
 
 %if %{defined useprebuiltcompiler}
-STARTPP=`pwd`/startcompiler/%{ppcname}
+STARTPP=`pwd`/fpc-2.6.2-bin-armv7hf/bin/%{ppcname}-%{version}
 %else
 STARTPP=%{ppcname}
 %endif
@@ -104,8 +126,8 @@ cd ..
 # FIXME: -j1 as there is a race - seen on "missing" `rtl.xct'.
 make -j1 -C fpcdocs pdf FPC=${NEWPP} FPDOC=${NEWFPDOC}
 
+
 %install
-rm -rf %{buildroot}
 cd fpcsrc
 FPCMAKE=`pwd`/utils/fpcm/fpcmake
 NEWPP=`pwd`/compiler/%{ppcname}
@@ -114,9 +136,9 @@ INSTALLOPTS="-j1 FPC=${NEWPP} FPCMAKE=${
                 INSTALL_LIBDIR=%{buildroot}%{_libdir} \
                 INSTALL_BASEDIR=%{buildroot}%{_libdir}/%{name}/%{version} \
                 CODPATH=%{buildroot}%{_libdir}/%{name}/lexyacc \
-                INSTALL_DOCDIR=%{buildroot}%{_defaultdocdir}/%{name}-%{version} \
+                INSTALL_DOCDIR=%{buildroot}%{_defaultdocdir}/%{name} \
                 INSTALL_BINDIR=%{buildroot}%{_bindir}
-                INSTALL_EXAMPLEDIR=%{buildroot}%{_defaultdocdir}/%{name}-%{version}/examples"
+                INSTALL_EXAMPLEDIR=%{buildroot}%{_defaultdocdir}/%{name}/examples"
 make compiler_distinstall ${INSTALLOPTS}
 make rtl_distinstall ${INSTALLOPTS}
 make packages_distinstall ${INSTALLOPTS}
@@ -131,13 +153,22 @@ make -C fpcdocs pdfinstall ${INSTALLOPTS
 # create link
 ln -sf ../%{_lib}/%{name}/%{version}/%{ppcname} %{buildroot}%{_bindir}/%{ppcname}
 
-# Create a version independent config
-%{buildroot}/%{_libdir}/%{name}/%{version}/samplecfg %{_exec_prefix} %{buildroot}%{_sysconfdir} %{buildroot}%{_exec_prefix}
+# Create a version independent compiler-configuration file with build-id
+# enabled by default
+# For this purpose some non-default templates are used. So the samplecfg
+# script could not be used and fpcmkcfg is called directly.
+%{buildroot}%{_bindir}/fpcmkcfg -p -t %{SOURCE2} -d "basepath=%{_exec_prefix}" -o %{buildroot}%{_sysconfdir}/fpc.cfg
+# Create the IDE configuration files
+%{buildroot}%{_bindir}/fpcmkcfg -p -1 -d "basepath=%{_libdir}/%{name}/\$fpcversion" -o %{buildroot}%{_libdir}/%{name}/%{version}/ide/text/fp.cfg
+%{buildroot}%{_bindir}/fpcmkcfg -p -2 -o %{buildroot}%{_libdir}/%{name}/%{version}/ide/text/fp.ini
+# Create the fppkg configuration files
+%{buildroot}%{_bindir}/fpcmkcfg -p -t %{SOURCE3} -d CompilerConfigDir=%{_sysconfdir}/fppkg -d arch=%{_arch} -o %{buildroot}%{_sysconfdir}/fppkg.cfg
+%{buildroot}%{_bindir}/fpcmkcfg -p -t %{SOURCE4} -d fpcbin=%{_bindir}/fpc -d GlobalPrefix=%{_exec_prefix} -d lib=%{_lib} -o %{buildroot}%{_sysconfdir}/fppkg/default_%{_arch}
 
 # Include the COPYING-information for the compiler/rtl/fcl in the documentation
-cp -a fpcsrc/compiler/COPYING.txt %{buildroot}%{_defaultdocdir}/%{name}-%{version}/COPYING
-cp -a fpcsrc/rtl/COPYING.txt %{buildroot}%{_defaultdocdir}/%{name}-%{version}/COPYING.rtl
-cp -a fpcsrc/rtl/COPYING.FPC %{buildroot}%{_defaultdocdir}/%{name}-%{version}/COPYING.FPC
+cp -a fpcsrc/compiler/COPYING.txt %{buildroot}%{_defaultdocdir}/%{name}/COPYING
+cp -a fpcsrc/rtl/COPYING.txt %{buildroot}%{_defaultdocdir}/%{name}/COPYING.rtl
+cp -a fpcsrc/rtl/COPYING.FPC %{buildroot}%{_defaultdocdir}/%{name}/COPYING.FPC
 
 # The source-files:
 mkdir -p %{buildroot}%{_datadir}/fpcsrc
@@ -148,33 +179,82 @@ cp -a fpc_src/* %{buildroot}%{_datadir}/
 # delete lexyacc
 rm -rf %{buildroot}%{_libdir}/%{name}/lexyacc
 
-%clean
-rm -rf %{buildroot}
 
 %files
-%defattr(-,root,root,-)
 %{_bindir}/*
 %{_libdir}/%{name}
 %config(noreplace) %{_sysconfdir}/%{name}.cfg
-%dir %{_defaultdocdir}/%{name}-%{version}/
-%doc %{_defaultdocdir}/%{name}-%{version}/NEWS
-%doc %{_defaultdocdir}/%{name}-%{version}/README
-%doc %{_defaultdocdir}/%{name}-%{version}/readme.ide
-%doc %{_defaultdocdir}/%{name}-%{version}/faq*
-%doc %{_defaultdocdir}/%{name}-%{version}/COPYING*
+%config(noreplace) %{_sysconfdir}/fppkg.cfg
+%config(noreplace) %{_sysconfdir}/fppkg/default_%{_arch}
+%dir %{_defaultdocdir}/%{name}/
+%doc %{_defaultdocdir}/%{name}/NEWS
+%doc %{_defaultdocdir}/%{name}/README
+%doc %{_defaultdocdir}/%{name}/readme.ide
+%doc %{_defaultdocdir}/%{name}/faq*
+%doc %{_defaultdocdir}/%{name}/COPYING*
 %{_mandir}/*/*
 
 %files doc
-%defattr(-,root,root,-)
-%dir %{_defaultdocdir}/%{name}-%{version}/
-%doc %{_defaultdocdir}/%{name}-%{version}/*.pdf
-%doc %{_defaultdocdir}/%{name}-%{version}/examples
+%dir %{_defaultdocdir}/%{name}/
+%doc %{_defaultdocdir}/%{name}/*.pdf
+%doc %{_defaultdocdir}/%{name}/examples
 
 %files src
-%defattr(-,root,root,-)
 %{_datadir}/fpcsrc
 
+
 %changelog
+* Sat Mar 7 2015 Joost van der Sluis <joost@cnoc.nl> - 2.6.4-1
+- Upgrade to upstream release 2.6.4.
+
+* Tue Jan 20 2015 Dan Horák <dan[at]danny.cz> - 2.6.2-7
+- switch to ExclusiveArch
+
+* Sat Aug 16 2014 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.6.2-6
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_21_22_Mass_Rebuild
+
+* Sat Jun 07 2014 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.6.2-5
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_21_Mass_Rebuild
+
+* Thu Aug 08 2013 Hans de Goede <hdegoede@redhat.com> - 2.6.2-4
+- Drop fpc binaries added to the src.rpm for bootstrapping on ARM, completing
+  the boostrap procedure (rhbz#992285)
+
+* Thu Aug 08 2013 Hans de Goede <hdegoede@redhat.com> - 2.6.2-3
+- Bootstrap for arm using Debian fpc-2.6.2 armhf binaries (rhbz#992285)
+- Use an unversioned docdir (rhbz#993758)
+
+* Sat Aug 03 2013 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.6.2-2
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_20_Mass_Rebuild
+
+* Wed Apr 24 2013 Joost van der Sluis <joost@cnoc.nl> - 2.6.2-1
+- Upgrade to upstream release 2.6.2.
+
+* Wed Feb 13 2013 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.6.2-0.2.rc1
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_19_Mass_Rebuild
+
+* Sat Nov 24 2012 Bruno Wolff III <bruno@wolff.to> - 2.6.2-0.1.rc1
+- Use standard versioning, so non-rc versions will be higher
+- Fix issue with some things using 'rc1' appended to version name and others not
+
+* Sat Nov 3 2012 Joost van der Sluis <joost@cnoc.nl> - 2.6.2rc1-1
+- Upgrade to upstream release 2.6.2rc1.
+
+* Thu Jul 19 2012 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.6.0-3
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_18_Mass_Rebuild
+
+* Mon May 14 2012 Karsten Hopp <karsten@redhat.com> 2.6.0-2
+- define ppcname on ppc64
+
+* Fri Jan 27 2012 Joost van der Sluis <joost@cnoc.nl> - 2.6.0-1
+- Upgrade to upstream release 2.6.0.
+- Do not use samplecfg for generating the configuration files anymore, but
+  call fpcmkcfg directly.
+- Changed the name of the project from Freepascal to Free Pascal
+
+* Fri Jan 13 2012 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.4.2-3
+- Rebuilt for https://fedoraproject.org/wiki/Fedora_17_Mass_Rebuild
+
 * Wed Feb 09 2011 Fedora Release Engineering <rel-eng@lists.fedoraproject.org> - 2.4.2-2
 - Rebuilt for https://fedoraproject.org/wiki/Fedora_15_Mass_Rebuild
 
@@ -243,34 +323,34 @@ rm -rf %{buildroot}
 * Tue Feb 19 2008 Fedora Release Engineering <rel-eng@fedoraproject.org> - 2.2.0-11
 - Autorebuild for GCC 4.3
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-10
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-10
 - Strip the debuginfo from grab_vcsa and ppudump, since debugedit chokes on it
 - Only strip debugdata on x86_64
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-9
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-9
 - Strip the debuginfo from mkxmlrpc, since debugedit chokes on it
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-8
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-8
 - Strip the debuginfo from h2pas, since debugedit chokes on it
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-7
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-7
 - Include the startcompiler on all targets, for the srpm-building
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-6
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-6
 - Disabled debuginfo for ppc64 only
 - Enabled smart-linking on ppc64
 - Added a patch for building documentation without fpc already installed
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-5
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-5
 - Disabled debuginfo
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-4
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-4
 - Enabled BuildId, added it to fpc.cfg
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-3
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-3
 - Disabled smart-linking on ppc64
 
-* Mon Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-2
+* Tue Oct 16 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-2
 - Buildrequirement fpc is not needed when using a pre-built compiler binary
 
 * Sun Oct 14 2007 Joost van der Sluis <joost@cnoc.nl> 2.2.0-1
@@ -315,7 +395,7 @@ rm -rf %{buildroot}
 - Removed gpm-devel requirement
 - Fixed a type in the -src description
 
-* Tue Jul 28 2005 Joost van der Sluis <joost@cnoc.nl> 2.0.0-1
+* Thu Jul 28 2005 Joost van der Sluis <joost@cnoc.nl> 2.0.0-1
 - Added some requirements
 - Added COPYING-info to %%doc
 
@@ -324,7 +404,7 @@ rm -rf %{buildroot}
 - Silenced post-script
 - disabled the debuginfo-package
 
-* Sat Jun 5 2005 Joost van der Sluis <joost@cnoc.nl> 2.0.0-0.5
+* Sun Jun 5 2005 Joost van der Sluis <joost@cnoc.nl> 2.0.0-0.5
 - Added doc-subpackage
 - Added src-subpackage
 
