diff -up sip-4.12.1/siputils.py.no_strip sip-4.12.1/siputils.py
--- sip-4.12.1/siputils.py.no_strip	2011-01-22 07:45:38.000000000 -0600
+++ sip-4.12.1/siputils.py	2011-02-25 13:48:12.473741314 -0600
@@ -1340,7 +1340,7 @@ class ModuleMakefile(Makefile):
     """
     def __init__(self, configuration, build_file, install_dir=None, static=0,
                  console=0, qt=0, opengl=0, threaded=0, warnings=1, debug=0,
-                 dir=None, makefile="Makefile", installs=None, strip=1,
+                 dir=None, makefile="Makefile", installs=None, strip=0,
                  export_all=0, universal=None, arch=None,
                  deployment_target=None):
         """Initialise an instance of a module Makefile.
@@ -1657,7 +1657,7 @@ class SIPModuleMakefile(ModuleMakefile):
     """
     def __init__(self, configuration, build_file, install_dir=None, static=0,
                  console=0, qt=0, opengl=0, threaded=0, warnings=1, debug=0,
-                 dir=None, makefile="Makefile", installs=None, strip=1,
+                 dir=None, makefile="Makefile", installs=None, strip=0,
                  export_all=0, universal=None, arch=None, prot_is_public=0,
                  deployment_target=None):
         """Initialise an instance of a SIP generated module Makefile.
