--- configure.in.orig	2010-01-27 21:30:40.000000000 -0200
+++ configure.in	2010-05-16 17:22:52.761571186 -0300
@@ -427,24 +427,8 @@
 [  # Look for the directory under a standard set of common directories.
   # Check X11 before X11Rn because it's often a symlink to the current release.
   for ac_dir in         \
-    /lib/X11/app-defaults \
-    /lib/app-defaults \
-    /lib/X11R6.5.1/app-defaults \
-    /lib/X11R6.5.1/X11/app-defaults \
-    /lib/X11R6.4/app-defaults \
-    /lib/X11R6.4/X11/app-defaults \
-    /lib/X11R6.3/app-defaults \
-    /lib/X11R6.3/X11/app-defaults \
-    /lib/X11R6.2/app-defaults \
-    /lib/X11R6.2/X11/app-defaults \
-    /lib/X11R6.1/app-defaults \
-    /lib/X11R6.1/X11/app-defaults \
-    /lib/X11R6/app-defaults \
-    /lib/X11R6/X11/app-defaults \
-    /lib/X11R5/app-defaults \
-    /lib/X11R5/X11/app-defaults \
-    /lib/X11R4/app-defaults \
-    /lib/X11R4/X11/app-defaults \
+    /etc/X11/app-defaults \
+    $libdir/X11/app-defaults \
     ; \
   do
     found=""
@@ -456,99 +440,8 @@
   done
   if test -z "$found"; then
   for ac_dir in         \
-    /usr/X11/lib/app-defaults     \
-    /lib/X11R6.5.1/lib/app-defaults \
-    /lib/X11R6.5.1/lib/X11/app-defaults \
-    /lib/X11R6.4/lib/app-defaults \
-    /lib/X11R6.4/lib/X11/app-defaults \
-    /usr/X11R6.3/lib/app-defaults     \
-    /usr/X11R6.3/lib/X11/app-defaults   \
-    /usr/X11R6.2/lib/app-defaults     \
-    /usr/X11R6.2/lib/X11/app-defaults   \
-    /usr/X11R6.1/lib/app-defaults     \
-    /usr/X11R6.1/lib/X11/app-defaults   \
-    /usr/X11R6/lib/app-defaults     \
-    /usr/X11R6/lib/X11/app-defaults   \
-    /usr/X11R5/lib/app-defaults     \
-    /usr/X11R5/lib/X11/app-defaults   \
-    /usr/X11R4/lib/app-defaults     \
-    /usr/X11R4/lib/X11/app-defaults   \
-            \
-    /usr/lib/X11/app-defaults     \
-    /usr/lib/X11R6.5.1/app-defaults \
-    /usr/lib/X11R6.4/app-defaults \
-    /usr/lib/X11R6.3/app-defaults     \
-    /usr/lib/X11R6.2/app-defaults     \
-    /usr/lib/X11R6.1/app-defaults     \
-    /usr/lib/X11R6/app-defaults     \
-    /usr/lib/X11R5/app-defaults     \
-    /usr/lib/X11R4/app-defaults     \
-            \
-    /usr/local/X11/lib/app-defaults   \
-    /usr/remote/X11/lib/app-defaults   \
-    /usr/local/X11R6.5.1/lib/app-defaults   \
-    /usr/local/X11R6.5.1/lib/X11/app-defaults   \
-    /usr/local/X11R6.4/lib/app-defaults   \
-    /usr/local/X11R6.4/lib/X11/app-defaults   \
-    /usr/local/X11R6.3/lib/app-defaults   \
-    /usr/local/X11R6.3/lib/X11/app-defaults   \
-    /usr/local/X11R6.2/lib/app-defaults   \
-    /usr/local/X11R6.2/lib/X11/app-defaults   \
-    /usr/local/X11R6.1/lib/app-defaults   \
-    /usr/local/X11R6.1/lib/X11/app-defaults   \
-    /usr/local/X11R6/lib/app-defaults   \
-    /usr/local/X11R6/lib/X11/app-defaults   \
-    /usr/local/X11R5/lib/app-defaults   \
-    /usr/local/X11R5/lib/X11/app-defaults   \
-    /usr/local/X11R4/lib/app-defaults   \
-    /usr/local/X11R4/lib/X11/app-defaults   \
-            \
-    /usr/local/lib/X11/app-defaults   \
-    /usr/remote/lib/X11/app-defaults   \
-    /usr/local/lib/X11R6.5.1/app-defaults   \
-    /usr/local/lib/X11R6.5.1/X11/app-defaults \
-    /usr/local/lib/X11R6.4/app-defaults   \
-    /usr/local/lib/X11R6.4/X11/app-defaults \
-    /usr/local/lib/X11R6.3/app-defaults   \
-    /usr/local/lib/X11R6.3/X11/app-defaults \
-    /usr/local/lib/X11R6.2/app-defaults   \
-    /usr/local/lib/X11R6.2/X11/app-defaults \
-    /usr/local/lib/X11R6.1/app-defaults   \
-    /usr/local/lib/X11R6.1/X11/app-defaults \
-    /usr/local/lib/X11R6/app-defaults   \
-    /usr/local/lib/X11R6/X11/app-defaults \
-    /usr/local/lib/X11R5/app-defaults   \
-    /usr/local/lib/X11R5/X11/app-defaults \
-    /usr/local/lib/X11R4/app-defaults   \
-    /usr/local/lib/X11R4/X11/app-defaults \
-            \
-    /usr/X386/lib/X11/app-defaults    \
-    /usr/x386/lib/X11/app-defaults    \
-    /usr/XFree86/lib/X11/app-defaults   \
-    /usr/pkg/lib/app-defaults          \
-            \
-    /usr/lib/X11/app-defaults     \
-    /usr/unsupported/lib/X11/app-defaults \
-    /usr/athena/lib/X11/app-defaults    \
-    /usr/local/x11r5/lib/X11/app-defaults \
-    /usr/lpp/Xamples/lib/X11/app-defaults \
-    /lib/usr/lib/X11/app-defaults   \
-    /usr/local/lib/app-defaults   \
-    /usr/remote/lib/app-defaults   \
-            \
-    /usr/openwin/lib/app-defaults   \
-    /usr/openwin/lib/X11/app-defaults   \
-    /usr/openwin/share/lib/app-defaults   \
-    /usr/openwin/share/lib/X11/app-defaults \
-            \
-    /X11R6.5.1/lib/app-defaults   \
-    /X11R6.4/lib/app-defaults     \
-    /X11R6.3/lib/app-defaults     \
-    /X11R6.2/lib/app-defaults     \
-    /X11R6.1/lib/app-defaults     \
-    /X11R6/lib/app-defaults     \
-    /X11R5/lib/app-defaults     \
-    /X11R4/lib/app-defaults     \
+    /etc/X11/app-defaults \
+    $libdir/X11/app-defaults \
     ; \
   do
     if test -d "$ac_dir"; then
@@ -612,66 +505,7 @@
 test -z "$motif_direct_test_function" && motif_direct_test_function=XmCreatePushButton
 test -z "$motif_direct_test_include" && motif_direct_test_include=Xm/Xm.h
   for ac_dir in               \
-    /usr/include/Motif1.2     \
-    /usr/Motif1.2/include     \
-                              \
-    /usr/motif/include        \
-                              \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-                              \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X11/include          \
-    /usr/include/X11          \
-    /usr/local/X11/include    \
-    /usr/local/include/X11    \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/dt/include           \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/athena/include       \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
     $extra_include            \
     ; \
   do
@@ -687,64 +521,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$motif_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_motif_includes" | sed s/include/lib/` \
-    /usr/lib/Motif1.2     \
-    /usr/Motif1.2/lib     \
-                          \
-    /usr/motif/lib        \
-                          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11R6.5.1    \
-    /usr/lib/X11R6.4      \
-    /usr/lib/X11R6.3      \
-    /usr/lib/X11R6.2      \
-    /usr/lib/X11R6.1      \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11R6.5.1  \
-    /usr/local/lib/X11R6.4  \
-    /usr/local/lib/X11R6.3  \
-    /usr/local/lib/X11R6.2  \
-    /usr/local/lib/X11R6.1  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X11/lib          \
-    /usr/lib/X11          \
-    /usr/local/X11/lib    \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -818,61 +595,7 @@
 test -z "$editres_direct_test_function" && editres_direct_test_function=_XEditResCheckMessages
 test -z "$editres_direct_test_include" && editres_direct_test_include=X11/Xmu/Editres.h
   for ac_dir in               \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include      \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/athena/include       \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
     $extra_include            \
     ; \
   do
@@ -888,64 +611,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$editres_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_editres_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib      \
-    /usr/X11R6.4/lib        \
-    /usr/X11R6.3/lib        \
-    /usr/X11R6.2/lib        \
-    /usr/X11R6.1/lib        \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1      \
-    /usr/lib/X11R6.4        \
-    /usr/lib/X11R6.3        \
-    /usr/lib/X11R6.2        \
-    /usr/lib/X11R6.1        \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/lib/X11R6.5.1  \
-    /usr/local/lib/X11R6.4  \
-    /usr/local/lib/X11R6.3  \
-    /usr/local/lib/X11R6.2  \
-    /usr/local/lib/X11R6.1  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -1016,64 +682,7 @@
 test -z "$xpm_direct_test_function" && xpm_direct_test_function=XpmCreateImageFromData
 test -z "$xpm_direct_test_include" && xpm_direct_test_include=X11/xpm.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/athena/include       \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
-    /usr/openwin/include      \
-                              \
-    /usr/include/Vk           \
     $extra_include            \
     ; \
   do
@@ -1089,59 +698,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$xpm_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_xpm_includes" | sed s%/X11$%% | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1    \
-    /usr/lib/X11R6.4      \
-    /usr/lib/X11R6.3      \
-    /usr/lib/X11R6.2      \
-    /usr/lib/X11R6.1      \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -1217,63 +774,7 @@
 # test -z "$gltt_direct_test_function" && gltt_direct_test_function=glttCreateImageFromData
 test -z "$gltt_direct_test_include" && gltt_direct_test_include=gltt/GLTTFont.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/athena/include       \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
-    /usr/openwin/include      \
-                              \
     $extra_include            \
     ; \
   do
@@ -1289,59 +790,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$gltt_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_gltt_includes" | sed s%/X11$%% | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib      \
-    /usr/X11R6.4/lib        \
-    /usr/X11R6.3/lib        \
-    /usr/X11R6.2/lib        \
-    /usr/X11R6.1/lib        \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1      \
-    /usr/lib/X11R6.4        \
-    /usr/lib/X11R6.3        \
-    /usr/lib/X11R6.2        \
-    /usr/lib/X11R6.1        \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -1414,63 +863,7 @@
 test -z "$ttf_direct_test_function" && ttf_direct_test_function=TT_Init_FreeType
 test -z "$ttf_direct_test_include" && ttf_direct_test_include=gltt/FTFace.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
-    /usr/include/X11          \
-    /usr/unsupported/include  \
-    /usr/athena/include       \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
-    /usr/openwin/include      \
-                              \
+    /usr/include              \
     $extra_include            \
     ; \
   do
@@ -1486,59 +879,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$ttf_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_ttf_includes" | sed s%/X11$%% | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib      \
-    /usr/X11R6.4/lib        \
-    /usr/X11R6.3/lib        \
-    /usr/X11R6.2/lib        \
-    /usr/X11R6.1/lib        \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1      \
-    /usr/lib/X11R6.4        \
-    /usr/lib/X11R6.3        \
-    /usr/lib/X11R6.2        \
-    /usr/lib/X11R6.1        \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -1585,9 +926,9 @@
   ttf=yes
   AC_DEFINE(HAVE_TTF)
   if test "x$prefix" = "xNONE"; then
-    AC_DEFINE_UNQUOTED(DEF_TTFONT, "/usr/lib/X11/xlock/fonts/")
+    AC_DEFINE_UNQUOTED(DEF_TTFONT, "/usr/share/fonts/TTF/")
   else
-    AC_DEFINE_UNQUOTED(DEF_TTFONT, "${prefix}/lib/X11/xlock/fonts/")
+    AC_DEFINE_UNQUOTED(DEF_TTFONT, "${prefix}/share/fonts/TTF/")
   fi
   XLOCKLIBS="${XLOCKLIBS} -lttf"
   test "x$ttf_includes" = xNONE && ttf_includes=$ac_ttf_includes
@@ -1616,63 +957,7 @@
 # test -z "$ftgl_direct_test_function" && ftgl_direct_test_function=ftglCreateImageFromData
 test -z "$ftgl_direct_test_include" && ftgl_direct_test_include=FTGL/FTFont.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/athena/include       \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
-    /usr/openwin/include      \
-                              \
     $extra_include            \
     ; \
   do
@@ -1688,59 +973,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$ftgl_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_ftgl_includes" | sed s%/X11$%% | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib      \
-    /usr/X11R6.4/lib        \
-    /usr/X11R6.3/lib        \
-    /usr/X11R6.2/lib        \
-    /usr/X11R6.1/lib        \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1      \
-    /usr/lib/X11R6.4        \
-    /usr/lib/X11R6.3        \
-    /usr/lib/X11R6.2        \
-    /usr/lib/X11R6.1        \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -1813,64 +1046,7 @@
 test -z "$freetype_direct_test_function" && freetype_direct_test_function=FT_Init_FreeType
 test -z "$freetype_direct_test_include" && freetype_direct_test_include=freetype2/freetype/freetype.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include	    \
-    /usr/X11R6.4/include	      \
-    /usr/X11R6.3/include	      \
-    /usr/X11R6.2/include	      \
-    /usr/X11R6.1/include	      \
-    /usr/X11R6/include	        \
-    /usr/X11R5/include	        \
-    /usr/X11R4/include	        \
-    /usr/X11/include	          \
-                              \
-    /usr/include/X11	          \
-    /usr/include/X11R6.5.1	    \
-    /usr/include/X11R6.4	      \
-    /usr/include/X11R6.3	      \
-    /usr/include/X11R6.2	      \
-    /usr/include/X11R6.1	      \
-    /usr/include/X11R6	        \
-    /usr/include/X11R5	        \
-    /usr/include/X11R4	        \
-                              \
-    /usr/local/X11/include	    \
-    /usr/local/X11R6.5.1/include	  \
-    /usr/local/X11R6.4/include	  \
-    /usr/local/X11R6.3/include	  \
-    /usr/local/X11R6.2/include	  \
-    /usr/local/X11R6.1/include	  \
-    /usr/local/X11R6/include	  \
-    /usr/local/X11R5/include	  \
-    /usr/local/X11R4/include	  \
-                              \
-    /usr/local/include/X11	    \
-    /usr/local/include/X11R6.5.1	  \
-    /usr/local/include/X11R6.4	  \
-    /usr/local/include/X11R6.3	  \
-    /usr/local/include/X11R6.2	  \
-    /usr/local/include/X11R6.1	  \
-    /usr/local/include/X11R6	  \
-    /usr/local/include/X11R5	  \
-    /usr/local/include/X11R4	  \
-                              \
-    /usr/X386/include	         \
-    /usr/x386/include	         \
-    /usr/XFree86/include/X11	  \
-    /usr/pkg/include          \
-                              \
-    /usr/include	        \
-    /usr/local/include	        \
-    /usr/remote/include	       \
-    /usr/include/X11	          \
-    /usr/unsupported/include	  \
-    /usr/athena/include	       \
-    /usr/local/x11r5/include	  \
-    /usr/lpp/Xamples/include	  \
-                              \
-    /usr/openwin/include	      \
-    /usr/openwin/share/include	 \
-    /usr/openwin/include	      \
-                              \
+    /usr/include              \
     $extra_include	            \
     ; \
   do
@@ -1886,59 +1062,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$freetype_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_freetype_includes" | sed s%/X11$%% | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib      \
-    /usr/X11R6.4/lib        \
-    /usr/X11R6.3/lib        \
-    /usr/X11R6.2/lib        \
-    /usr/X11R6.1/lib        \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1      \
-    /usr/lib/X11R6.4        \
-    /usr/lib/X11R6.3        \
-    /usr/lib/X11R6.2        \
-    /usr/lib/X11R6.1        \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -1985,9 +1109,9 @@
   freetype=yes
   AC_DEFINE(HAVE_FREETYPE)
   if test "x$prefix" = "xNONE"; then
-    AC_DEFINE_UNQUOTED(DEF_TTFONT, "/usr/lib/X11/xlock/fonts/")
+    AC_DEFINE_UNQUOTED(DEF_TTFONT, "/usr/share/fonts/TTF/")
   else
-    AC_DEFINE_UNQUOTED(DEF_TTFONT, "${prefix}/lib/X11/xlock/fonts/")
+    AC_DEFINE_UNQUOTED(DEF_TTFONT, "${prefix}/share/fonts/TTF/")
   fi
   XLOCKLIBS="${XLOCKLIBS} -lfreetype"
   test "x$freetype_includes" = xNONE && freetype_includes=$ac_freetype_includes
@@ -2019,56 +1143,6 @@
 test -z "$opengl_direct_test_include" && opengl_direct_test_include=GL/gl.h
   for ac_dir in               \
     /usr/include              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
-                              \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
-    /usr/unsupported/include  \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
     $extra_include            \
     ; \
   do
@@ -2086,61 +1160,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$opengl_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_opengl_includes" | sed s/include/lib/` \
-                          \
-    /usr/lib              \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1    \
-    /usr/lib/X11R6.4      \
-    /usr/lib/X11R6.3      \
-    /usr/lib/X11R6.2      \
-    /usr/lib/X11R6.1      \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/lib/X11R6.5.1  \
-    /usr/local/lib/X11R6.4  \
-    /usr/local/lib/X11R6.3  \
-    /usr/local/lib/X11R6.2  \
-    /usr/local/lib/X11R6.1  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -2227,56 +1247,7 @@
 test -z "$mesa_direct_test_function" && mesa_direct_test_function=glXCreateContext
 test -z "$mesa_direct_test_include" && mesa_direct_test_include=GL/gl.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11/include          \
-                              \
-    /usr/include/X11          \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-                              \
-    /usr/local/X11/include    \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-                              \
-    /usr/local/include/X11    \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
     $extra_include            \
     ; \
   do
@@ -2292,60 +1263,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$mesa_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_mesa_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1    \
-    /usr/lib/X11R6.4      \
-    /usr/lib/X11R6.3      \
-    /usr/lib/X11R6.2      \
-    /usr/lib/X11R6.1      \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/lib/X11R6.5.1  \
-    /usr/local/lib/X11R6.4  \
-    /usr/local/lib/X11R6.3  \
-    /usr/local/lib/X11R6.2  \
-    /usr/local/lib/X11R6.1  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -2562,7 +1480,7 @@
 test -z "$dtsaver_direct_test_function" && dtsaver_direct_test_function=DtSaverGetWindows
 test -z "$dtsaver_direct_test_include" && dtsaver_direct_test_include=Dt/Saver.h
   for ac_dir in            \
-    /usr/dt/include        \
+    /usr/include           \
     $extra_include         \
     ; \
   do
@@ -2578,7 +1496,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$dtsaver_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_dtsaver_includes" | sed s/include/lib/` \
+for ac_dir in $libdir \
     $extra_lib         \
     ; \
 do
@@ -2647,64 +1565,7 @@
 test -z "$ext_direct_test_function" && ext_direct_test_function=DPMSSetTimeouts
 test -z "$dpms_direct_test_include" && ext_direct_test_include=X11/extensions/dpms.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-                              \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X11/include          \
-    /usr/include/X11          \
-    /usr/local/X11/include    \
-    /usr/local/include/X11    \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/dt/include           \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/dpms/include         \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
     $extra_include            \
     ; \
   do
@@ -2720,66 +1581,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$ext_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_ext_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11R6.5.1    \
-    /usr/lib/X11R6.4      \
-    /usr/lib/X11R6.3      \
-    /usr/lib/X11R6.2      \
-    /usr/lib/X11R6.1      \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11R6.5.1  \
-    /usr/local/lib/X11R6.4  \
-    /usr/local/lib/X11R6.3  \
-    /usr/local/lib/X11R6.2  \
-    /usr/local/lib/X11R6.1  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X11/lib          \
-    /usr/lib/X11          \
-    /usr/local/X11/lib    \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/dpms/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
-    /shlib                \
-    /usr/shlib            \
-    /usr/shlib/X11        \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -2853,64 +1655,7 @@
 test -z "$dpms_direct_test_function" && dpms_direct_test_function=DPMSSetTimeouts
 test -z "$dpms_direct_test_include" && dpms_direct_test_include=X11/extensions/dpms.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11R4/include        \
-                              \
-    /usr/include/X11R6.5.1    \
-    /usr/include/X11R6.4      \
-    /usr/include/X11R6.3      \
-    /usr/include/X11R6.2      \
-    /usr/include/X11R6.1      \
-    /usr/include/X11R6        \
-    /usr/include/X11R5        \
-    /usr/include/X11R4        \
-                              \
-    /usr/local/X11R6.5.1/include  \
-    /usr/local/X11R6.4/include  \
-    /usr/local/X11R6.3/include  \
-    /usr/local/X11R6.2/include  \
-    /usr/local/X11R6.1/include  \
-    /usr/local/X11R6/include  \
-    /usr/local/X11R5/include  \
-    /usr/local/X11R4/include  \
-                              \
-    /usr/local/include/X11R6.5.1  \
-    /usr/local/include/X11R6.4  \
-    /usr/local/include/X11R6.3  \
-    /usr/local/include/X11R6.2  \
-    /usr/local/include/X11R6.1  \
-    /usr/local/include/X11R6  \
-    /usr/local/include/X11R5  \
-    /usr/local/include/X11R4  \
-                              \
-    /usr/X11/include          \
-    /usr/include/X11          \
-    /usr/local/X11/include    \
-    /usr/local/include/X11    \
-                              \
-    /usr/X386/include         \
-    /usr/x386/include         \
-    /usr/XFree86/include/X11  \
-    /usr/pkg/include          \
-                              \
-    /usr/dt/include           \
-                              \
-    /usr/local/include        \
-    /usr/remote/include       \
     /usr/include              \
-    /usr/unsupported/include  \
-    /usr/dpms/include         \
-    /usr/local/x11r5/include  \
-    /usr/lpp/Xamples/include  \
-                              \
-    /usr/openwin/include      \
-    /usr/openwin/share/include \
     $extra_include            \
     ; \
   do
@@ -2926,66 +1671,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$dpms_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_dpms_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11R6.5.1    \
-    /usr/lib/X11R6.4      \
-    /usr/lib/X11R6.3      \
-    /usr/lib/X11R6.2      \
-    /usr/lib/X11R6.1      \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11R6.5.1  \
-    /usr/local/lib/X11R6.4  \
-    /usr/local/lib/X11R6.3  \
-    /usr/local/lib/X11R6.2  \
-    /usr/local/lib/X11R6.1  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X11/lib          \
-    /usr/lib/X11          \
-    /usr/local/X11/lib    \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/dpms/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
-    /shlib                \
-    /usr/shlib            \
-    /usr/shlib/X11        \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -3059,64 +1745,7 @@
 test -z "$xinerama_direct_test_function" && xinerama_direct_test_function=XineramaQueryExtension
 test -z "$xinerama_direct_test_include" && xinerama_direct_test_include=X11/extensions/Xinerama.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include	    \
-    /usr/X11R6.4/include	      \
-    /usr/X11R6.3/include	      \
-    /usr/X11R6.2/include	      \
-    /usr/X11R6.1/include	      \
-    /usr/X11R6/include	        \
-    /usr/X11R5/include	        \
-    /usr/X11R4/include	        \
-    /usr/X11/include	          \
-                              \
-    /usr/include/X11	          \
-    /usr/include/X11R6.5.1	    \
-    /usr/include/X11R6.4	      \
-    /usr/include/X11R6.3	      \
-    /usr/include/X11R6.2	      \
-    /usr/include/X11R6.1	      \
-    /usr/include/X11R6	        \
-    /usr/include/X11R5	        \
-    /usr/include/X11R4	        \
-                              \
-    /usr/local/X11/include	    \
-    /usr/local/X11R6.5.1/include	  \
-    /usr/local/X11R6.4/include	  \
-    /usr/local/X11R6.3/include	  \
-    /usr/local/X11R6.2/include	  \
-    /usr/local/X11R6.1/include	  \
-    /usr/local/X11R6/include	  \
-    /usr/local/X11R5/include	  \
-    /usr/local/X11R4/include	  \
-                              \
-    /usr/local/include/X11	    \
-    /usr/local/include/X11R6.5.1	  \
-    /usr/local/include/X11R6.4	  \
-    /usr/local/include/X11R6.3	  \
-    /usr/local/include/X11R6.2	  \
-    /usr/local/include/X11R6.1	  \
-    /usr/local/include/X11R6	  \
-    /usr/local/include/X11R5	  \
-    /usr/local/include/X11R4	  \
-                              \
-    /usr/X386/include	         \
-    /usr/x386/include	         \
-    /usr/XFree86/include/X11	  \
-    /usr/pkg/include          \
-                              \
-    /usr/include	        \
-    /usr/local/include	        \
-    /usr/remote/include	       \
-    /usr/include/X11	          \
-    /usr/unsupported/include	  \
-    /usr/athena/include	       \
-    /usr/local/x11r5/include	  \
-    /usr/lpp/Xamples/include	  \
-                              \
-    /usr/openwin/include	      \
-    /usr/openwin/share/include	 \
-    /usr/openwin/include	      \
-                              \
+    /usr/include              \
     $extra_include	            \
     ; \
   do
@@ -3132,59 +1761,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$xinerama_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_xinerama_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11/lib          \
-    /usr/X11R6.5.1/lib      \
-    /usr/X11R6.4/lib        \
-    /usr/X11R6.3/lib        \
-    /usr/X11R6.2/lib        \
-    /usr/X11R6.1/lib        \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11R4/lib        \
-                          \
-    /usr/lib/X11          \
-    /usr/lib/X11R6.5.1      \
-    /usr/lib/X11R6.4        \
-    /usr/lib/X11R6.3        \
-    /usr/lib/X11R6.2        \
-    /usr/lib/X11R6.1        \
-    /usr/lib/X11R6        \
-    /usr/lib/X11R5        \
-    /usr/lib/X11R4        \
-                          \
-    /usr/local/X11/lib    \
-    /usr/local/X11R6/lib  \
-    /usr/local/X11R5/lib  \
-    /usr/local/X11R4/lib  \
-                          \
-    /usr/local/lib/X11    \
-    /usr/local/X11R6.5.1/lib  \
-    /usr/local/X11R6.4/lib  \
-    /usr/local/X11R6.3/lib  \
-    /usr/local/X11R6.2/lib  \
-    /usr/local/X11R6.1/lib  \
-    /usr/local/lib/X11R6  \
-    /usr/local/lib/X11R5  \
-    /usr/local/lib/X11R4  \
-                          \
-    /usr/X386/lib         \
-    /usr/x386/lib         \
-    /usr/XFree86/lib/X11  \
-    /usr/pkg/lib          \
-                          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/unsupported/lib  \
-    /usr/athena/lib       \
-    /usr/local/x11r5/lib  \
-    /usr/lpp/Xamples/lib  \
-    /lib/usr/lib/X11      \
-                          \
-    /usr/openwin/lib      \
-    /usr/openwin/share/lib \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -3257,17 +1834,6 @@
 test -z "$rplay_direct_test_function" && rplay_direct_test_function=rplay_open_default
 test -z "$rplay_direct_test_include" && rplay_direct_test_include=rplay.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11/include          \
-    /usr/local/include        \
-    /usr/remote/include       \
-    /usr/pkg/include          \
     /usr/include              \
     $extra_include            \
     ; \
@@ -3284,20 +1850,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$rplay_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_rplay_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11/lib          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/pkg/lib          \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -3458,7 +2011,7 @@
       if test "x$prefix" = "xNONE"; then
 	AC_DEFINE_UNQUOTED(DEFAULT_SOUND_DIR, "/usr/lib/X11/xlock/sounds/")
       else
-	AC_DEFINE_UNQUOTED(DEFAULT_SOUND_DIR, "${prefix}/lib/X11/xlock/sounds/")
+	AC_DEFINE_UNQUOTED(DEFAULT_SOUND_DIR, "${libdir}/X11/xlock/sounds/")
       fi
       test "x$esound_config" = xNONE && esound_config=$ac_esound_config
       test "x$esound_prefix" = xNONE && esound_prefix=$ac_esound_prefix
@@ -3491,17 +2044,6 @@
 test -z "$nas_direct_test_function" && nas_direct_test_function=AuOpenServer
 test -z "$nas_direct_test_include" && nas_direct_test_include=audio/audio.h
   for ac_dir in               \
-    /usr/X11R6.5.1/include    \
-    /usr/X11R6.4/include      \
-    /usr/X11R6.3/include      \
-    /usr/X11R6.2/include      \
-    /usr/X11R6.1/include      \
-    /usr/X11R6/include        \
-    /usr/X11R5/include        \
-    /usr/X11/include          \
-    /usr/local/include        \
-    /usr/remote/include       \
-    /usr/pkg/include          \
     /usr/include              \
     $extra_include            \
     ; \
@@ -3518,20 +2060,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$nas_direct_test_library $LIBS"
 # First see if replacing the include by lib works.
-for ac_dir in `echo "$ac_nas_includes" | sed s/include/lib/` \
-                          \
-    /usr/X11R6.5.1/lib    \
-    /usr/X11R6.4/lib      \
-    /usr/X11R6.3/lib      \
-    /usr/X11R6.2/lib      \
-    /usr/X11R6.1/lib      \
-    /usr/X11R6/lib        \
-    /usr/X11R5/lib        \
-    /usr/X11/lib          \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/pkg/lib          \
+for ac_dir in $libdir \
     $extra_lib            \
     ; \
 do
@@ -3620,10 +2149,7 @@
 ac_save_LIBS="$LIBS"
 LIBS="-l$crypt_direct_test_library $LIBS"
 for ac_dir in \
-    /usr/lib              \
-    /usr/local/lib        \
-    /usr/remote/lib       \
-    /usr/pkg/lib          \
+    $libdir               \
     $extra_lib            \
     ; \
 do
