From 8ce44e63258229e2d84a14a3b596750b846f368c Mon Sep 17 00:00:00 2001
From: Marius Cirsta <mcirsta@frugalware.org>
Date: Fri, 14 Feb 2014 02:22:41 +0200
Subject: [PATCH] * fix building with llvm 3.4

---
 CMakeLists.txt        | 3 +++
 src/scripting/abc.cpp | 2 ++
 2 files changed, 5 insertions(+)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -238,6 +238,9 @@
 IF(${LLVM_STRING_VERSION} VERSION_GREATER 3.0)
 	ADD_DEFINITIONS(-DLLVM_31)
 ENDIF(${LLVM_STRING_VERSION} VERSION_GREATER 3.0)
+IF(${LLVM_STRING_VERSION} VERSION_GREATER 3.3)
+	ADD_DEFINITIONS(-DLLVM_34)
+ENDIF(${LLVM_STRING_VERSION} VERSION_GREATER 3.3)
 INCLUDE(FindZLIB REQUIRED)
 INCLUDE(FindFreetype REQUIRED)
 IF(NOT(ENABLE_GLES2))
--- a/src/scripting/abc.cpp
+++ b/src/scripting/abc.cpp
@@ -1462,7 +1462,9 @@
 	{
 #ifdef LLVM_31
 		llvm::TargetOptions Opts;
+#ifndef LLVM_34
 		Opts.JITExceptionHandling = true;
+#endif
 #else
 		llvm::JITExceptionHandling = true;
 #endif
