diff -up src/aica/audio.c.glib231 src/aica/audio.c
--- src/aica/audio.c.glib231	2013-06-30 16:40:34.542696305 -0300
+++ src/aica/audio.c	2013-06-30 16:41:32.443410110 -0300
@@ -19,7 +19,7 @@
 
 #include "aica/aica.h"
 #include "aica/audio.h"
-#include <glib/gmem.h>
+#include <glib.h>
 #include "dream.h"
 #include <assert.h>
 #include <string.h>
diff -up src/aica/audio.h.glib231 src/aica/audio.h
--- src/aica/audio.h.glib231	2013-06-30 16:40:34.632677212 -0300
+++ src/aica/audio.h	2013-06-30 16:41:29.379060363 -0300
@@ -20,7 +20,7 @@
 
 #include <stdint.h>
 #include <stdio.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "gettext.h"
 #include "plugin.h"
 
diff -up src/config.c.glib231 src/config.c
--- src/config.c.glib231	2013-06-30 16:34:55.941532473 -0300
+++ src/config.c	2013-06-30 16:35:06.004397920 -0300
@@ -21,8 +21,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <string.h>
-#include <glib/gmem.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include "dream.h"
diff -up src/config.h.glib231 src/config.h
--- src/config.h.glib231	2013-06-30 16:36:41.766083697 -0300
+++ src/config.h	2013-06-30 16:36:49.582425507 -0300
@@ -19,8 +19,7 @@
 #ifndef lxdream_config_H
 #define lxdream_config_H 1
 
-#include <glib/gtypes.h>
-#include <glib/glist.h>
+#include <glib.h>
 #include "gettext.h"
 
 #ifdef __cplusplus
diff -up src/drivers/input_lirc.c.glib231 src/drivers/input_lirc.c
--- src/drivers/input_lirc.c.glib231	2013-06-30 16:40:34.725657483 -0300
+++ src/drivers/input_lirc.c	2013-06-30 16:41:23.948212780 -0300
@@ -28,7 +28,6 @@
 #include <dirent.h>
 #include <ctype.h>
 
-#include <glib/giochannel.h>
 #include <glib.h>
 
 #include <lirc/lirc_client.h>
diff -up src/drivers/joy_linux.c.glib231 src/drivers/joy_linux.c
--- src/drivers/joy_linux.c.glib231	2013-06-30 16:40:34.817637966 -0300
+++ src/drivers/joy_linux.c	2013-06-30 16:41:17.581563766 -0300
@@ -32,7 +32,6 @@
 #include <ctype.h>
 
 #include <linux/joystick.h>
-#include <glib/giochannel.h>
 #include <glib.h>
 
 #include "lxdream.h"
diff -up src/gdlist.c.glib231 src/gdlist.c
--- src/gdlist.c.glib231	2013-06-30 16:37:58.094890462 -0300
+++ src/gdlist.c	2013-06-30 16:38:42.766412810 -0300
@@ -19,7 +19,7 @@
 
 #include <string.h>
 #include <stdlib.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <libgen.h>
 #include "gettext.h"
 #include "gdrom/gdrom.h"
diff -up src/gdrom/gddriver.h.glib231 src/gdrom/gddriver.h
--- src/gdrom/gddriver.h.glib231	2013-06-30 16:40:34.908618661 -0300
+++ src/gdrom/gddriver.h	2013-06-30 16:41:11.886772179 -0300
@@ -24,7 +24,7 @@
 #include <stdio.h>
 #include "lxdream.h"
 #include "gdrom/gdrom.h"
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 #ifdef __cplusplus
 extern "C" {
diff -up src/gdrom/gdi.c.glib231 src/gdrom/gdi.c
--- src/gdrom/gdi.c.glib231	2013-06-30 16:40:35.005598082 -0300
+++ src/gdrom/gdi.c	2013-06-30 16:41:06.975814262 -0300
@@ -23,7 +23,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/stat.h>
-#include <glib/gutils.h>
+#include <glib.h>
 #include "gdrom/gddriver.h"
 
 
diff -up src/gdrom/gdrom.c.glib231 src/gdrom/gdrom.c
--- src/gdrom/gdrom.c.glib231	2013-06-30 16:39:02.498226321 -0300
+++ src/gdrom/gdrom.c	2013-06-30 16:39:08.552941679 -0300
@@ -20,7 +20,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <ctype.h>
-#include <glib/gutils.h>
+#include <glib.h>
 #include "gdrom/ide.h"
 #include "gdrom/gdrom.h"
 #include "gdrom/gddriver.h"
diff -up src/gdrom/nrg.c.glib231 src/gdrom/nrg.c
--- src/gdrom/nrg.c.glib231	2013-06-30 16:40:35.105576868 -0300
+++ src/gdrom/nrg.c	2013-06-30 16:41:03.032650979 -0300
@@ -20,7 +20,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <errno.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 #include "gdrom/gddriver.h"
 #include "dream.h"
 
diff -up src/gui.h.glib231 src/gui.h
--- src/gui.h.glib231	2013-06-30 16:37:58.182871796 -0300
+++ src/gui.h	2013-06-30 16:38:39.135183238 -0300
@@ -19,7 +19,7 @@
 #ifndef lxdream_gui_H
 #define lxdream_gui_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #ifdef __cplusplus
 extern "C" {
diff -up src/loader.h.glib231 src/loader.h
--- src/loader.h.glib231	2013-06-30 16:37:58.273852494 -0300
+++ src/loader.h	2013-06-30 16:38:36.303783966 -0300
@@ -20,7 +20,7 @@
 #define lxdream_loader_H 1
 
 #include <stdio.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #ifdef __cplusplus
 extern "C" {
diff -up src/lxdream.h.glib231 src/lxdream.h
--- src/lxdream.h.glib231	2013-06-30 16:34:23.036512188 -0300
+++ src/lxdream.h	2013-06-30 16:34:29.828071603 -0300
@@ -20,7 +20,7 @@
 #define lxdream_lxdream_H 1
 
 #include <stdint.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #include "../config.h"
 
diff -up src/lxpaths.c.glib231 src/lxpaths.c
--- src/lxpaths.c.glib231	2013-06-30 16:36:05.950681557 -0300
+++ src/lxpaths.c	2013-06-30 16:36:13.346112716 -0300
@@ -19,8 +19,7 @@
 #include <ctype.h>
 #include <unistd.h>
 #include <wordexp.h>
-#include <glib/gstrfuncs.h>
-#include <glib/gutils.h>
+#include <glib.h>
 
 #include "gui.h"
 #include "config.h"
diff -up src/maple/maple.c.glib231 src/maple/maple.c
--- src/maple/maple.c.glib231	2013-06-30 16:40:35.193558199 -0300
+++ src/maple/maple.c	2013-06-30 16:40:59.522395833 -0300
@@ -18,7 +18,7 @@
 #define MODULE maple_module
 
 #include <assert.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "dream.h"
 #include "mem.h"
 #include "asic.h"
diff -up src/mem.c.glib231 src/mem.c
--- src/mem.c.glib231	2013-06-30 16:35:28.726577971 -0300
+++ src/mem.c	2013-06-30 16:35:39.157365300 -0300
@@ -20,7 +20,7 @@
 #include <sys/types.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <assert.h>
 #include <stdint.h>
 #include <stdlib.h>
diff -up src/paths_unix.c.glib231 src/paths_unix.c
--- src/paths_unix.c.glib231	2013-06-30 16:37:58.374831071 -0300
+++ src/paths_unix.c	2013-06-30 16:38:32.913503268 -0300
@@ -18,7 +18,7 @@
 
 #include <string.h>
 #include <stdlib.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 
 #include "lxdream.h"
 #include "config.h"
@@ -47,4 +47,4 @@ const char *get_user_data_path()
         user_data_path = g_strdup_printf( "%s/.lxdream", home );
     }
     return user_data_path;
-}
\ No newline at end of file
+}
diff -up src/plugin.c.glib231 src/plugin.c
--- src/plugin.c.glib231	2013-06-30 16:37:58.472810283 -0300
+++ src/plugin.c	2013-06-30 16:38:29.401248447 -0300
@@ -20,8 +20,7 @@
 #include <dirent.h>
 #include <dlfcn.h>
 #include <string.h>
-#include <glib/gmem.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "plugin.h"
 #include "lxpaths.h"
 
diff -up src/pvr2/glutil.c.glib231 src/pvr2/glutil.c
--- src/pvr2/glutil.c.glib231	2013-06-30 16:40:35.287538258 -0300
+++ src/pvr2/glutil.c	2013-06-30 16:40:56.674000244 -0300
@@ -17,7 +17,7 @@
  */
 
 #include <string.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "pvr2/glutil.h"
 
 gboolean isGLSecondaryColorSupported()
diff -up src/sh4/sh4core.h.glib231 src/sh4/sh4core.h
--- src/sh4/sh4core.h.glib231	2013-06-30 16:40:35.387517044 -0300
+++ src/sh4/sh4core.h	2013-06-30 16:40:52.091972516 -0300
@@ -19,7 +19,7 @@
 #ifndef lxdream_sh4core_H
 #define lxdream_sh4core_H 1
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <stdint.h>
 #include <stdio.h>
 #include "mem.h"
diff -up src/syscall.h.glib231 src/syscall.h
--- src/syscall.h.glib231	2013-06-30 16:37:58.570789496 -0300
+++ src/syscall.h	2013-06-30 16:38:23.418517769 -0300
@@ -20,7 +20,7 @@
 #define lxdream_syscall_H 1
 
 #include <stdint.h>
-#include <glib/gtypes.h>
+#include <glib.h>
 
 #ifdef __cplusplus
 extern "C" {
diff -up src/tools/actparse.c.glib231 src/tools/actparse.c
--- src/tools/actparse.c.glib231	2013-06-30 16:33:23.681101871 -0300
+++ src/tools/actparse.c	2013-06-30 16:33:30.720608786 -0300
@@ -21,7 +21,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <sys/stat.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "tools/gendec.h"
 
 static int add_action( struct action *actions, struct ruleset *rules, char *operation, const char *file, int line, char *action )
diff -up src/tools/gendec.c.glib231 src/tools/gendec.c
--- src/tools/gendec.c.glib231	2013-06-30 16:32:28.972705178 -0300
+++ src/tools/gendec.c	2013-06-30 16:32:38.267733810 -0300
@@ -23,7 +23,7 @@
 #include <getopt.h>
 #include <errno.h>
 #include <ctype.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <assert.h>
 #include "tools/gendec.h"
 
diff -up src/vmu/vmulist.c.glib231 src/vmu/vmulist.c
--- src/vmu/vmulist.c.glib231	2013-06-30 16:40:35.481497102 -0300
+++ src/vmu/vmulist.c	2013-06-30 16:40:48.355765308 -0300
@@ -18,8 +18,7 @@
 
 #include <string.h>
 #include <stdlib.h>
-#include <glib/glist.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include "vmulist.h"
 #include "config.h"
 
diff -up src/vmu/vmuvol.c.glib231 src/vmu/vmuvol.c
--- src/vmu/vmuvol.c.glib231	2013-06-30 16:40:35.571478008 -0300
+++ src/vmu/vmuvol.c	2013-06-30 16:40:43.220854890 -0300
@@ -16,8 +16,7 @@
  * GNU General Public License for more details.
  */
 
-#include <glib/gmem.h>
-#include <glib/gstrfuncs.h>
+#include <glib.h>
 #include <string.h>
 #include <unistd.h>
 #include <stdio.h>
