--- pwauth-2.3.10/config.h.orig	2011-10-05 07:01:12.000000000 -0600
+++ pwauth-2.3.10/config.h	2013-05-21 12:05:22.079346949 -0600
@@ -123,7 +123,7 @@
 
 /* #define SHADOW_NONE		/**/
 /* #define SHADOW_BSD		/* FreeBSD, NetBSD, OpenBSD, BSDI, OS X */
-#define SHADOW_SUN		/* Linux, Solaris, IRIX */
+/* #define SHADOW_SUN		/* Linux, Solaris, IRIX */
 /* #define SHADOW_JFH		/**/
 /* #define SHADOW_MDW		/**/
 /* #define SHADOW_AIX		/* AIX (see also AUTHENTICATE_AIX) */
@@ -131,7 +131,7 @@
 
    /* HIGH-LEVEL OPTIONS */
 
-/* #define PAM			/* Linux PAM or OpenPAM */
+#define PAM			/* Linux PAM or OpenPAM */
 /* #define PAM_OLD_OS_X		/* PAM on OS X version 10.5 or older */
 /* #define PAM_SOLARIS		/* PAM on Solaris other than 2.6 */
 /* #define PAM_SOLARIS_26	/* PAM on Solaris 2.6 */
@@ -227,8 +227,8 @@
  * right with PAM.
  */
 
-#define NOLOGIN_FILE "/etc/nologin"	/**/
-#define MIN_NOLOGIN_UID 1		/**/
+/* #define NOLOGIN_FILE "/etc/nologin"	/**/
+/* #define MIN_NOLOGIN_UID 1		/**/
 
 
 /* Defining CHECK_LOGIN_EXPIRATION and CHECK_PASSWORD_EXPIRATION causes
@@ -278,7 +278,7 @@
  * to change the uid list.
  */
 
-#define SERVER_UIDS 30		/* user "wwwrun" on the author's system */
+/* #define SERVER_UIDS 30		/* user "wwwrun" on the author's system */
 
 
 /* If MIN_UNIX_UID is defined to an integer, logins with uid numbers less than
--- pwauth-2.3.10/Makefile.orig	2011-10-05 07:45:04.000000000 -0600
+++ pwauth-2.3.10/Makefile	2013-05-21 12:31:35.249588745 -0600
@@ -11,7 +11,7 @@
 LOCALFLAGS= -g 
 
 # For PAM on Redhat Linux
-# LIB=-lpam -ldl
+LIB=-lpam -ldl
 
 # For PAM on Solaris or OS X
 # LIB=-lpam
@@ -28,12 +29,12 @@ TAR= README INSTALL CHANGES FORM_AUTH Ma
 
 pwauth: main.o auth_aix.o auth_bsd.o auth_hpux.o auth_mdw.o auth_openbsd.o \
 	auth_pam.o auth_sun.o fail_log.o lastlog.o nologin.o snooze.o
-	$(CC) -o pwauth $(CFLAGS) main.o auth_aix.o auth_bsd.o auth_hpux.o \
+	$(CC) -o pwauth $(CFLAGS) $(LDFLAGS) main.o auth_aix.o auth_bsd.o auth_hpux.o \
 		auth_mdw.o auth_openbsd.o auth_pam.o auth_sun.o fail_log.o \
 		lastlog.o nologin.o snooze.o $(LIB)
 
 checkfaillog: checkfaillog.o fail_check.o
-	$(CC) -o checkfaillog $(CFLAGS) checkfaillog.o fail_check.o $(LIB)
+	$(CC) -o checkfaillog $(CFLAGS) $(LDFLAGS) checkfaillog.o fail_check.o $(LIB)
 
 main.o: main.c config.h pwauth.h fail_log.h
 auth_aix.o: auth_aix.c config.h pwauth.h
