--- pwauth-2.3.10/snooze.c.orig	2011-10-05 07:18:44.000000000 -0600
+++ pwauth-2.3.10/snooze.c	2013-05-21 14:05:07.066802103 -0600
@@ -42,6 +42,7 @@
  * sleep time, if other pwauth processes are in sleeps.
  */
 
+void
 snooze(int seconds)
 {
     int slfd;
--- pwauth-2.3.10/lastlog.c.orig	2009-04-03 22:45:56.000000000 -0600
+++ pwauth-2.3.10/lastlog.c	2013-05-21 14:08:39.134622151 -0600
@@ -31,6 +31,8 @@
  * =======================================================================
  */
 
+#include <time.h>
+
 #include "pwauth.h"
 
 /* LASTLOG - update the system's lastlog */
--- pwauth-2.3.10/pwauth.h.orig	2009-04-03 22:45:56.000000000 -0600
+++ pwauth-2.3.10/pwauth.h	2013-05-21 14:10:16.590658544 -0600
@@ -127,3 +127,7 @@ extern int haveuid;
 #ifndef BFSZ
 # define BFSZ 1024
 #endif
+
+void snooze(int seconds);
+void lastlog();
+int check_auth(char *login, char *passwd);
--- pwauth-2.3.10/main.c.orig	2013-05-21 12:05:22.068346394 -0600
+++ pwauth-2.3.10/main.c	2013-05-21 14:18:29.873121527 -0600
@@ -44,6 +44,7 @@
 #endif
 
 
+int
 main(int argc, char **argv)
 {
 #ifdef ENV_METHOD
