diff -up ntp-4.2.6p5/ntpd/ntp_proto.c.sourceport ntp-4.2.6p5/ntpd/ntp_proto.c
--- ntp-4.2.6p5/ntpd/ntp_proto.c.sourceport	2015-02-06 16:43:27.857879763 +0100
+++ ntp-4.2.6p5/ntpd/ntp_proto.c	2015-02-06 16:43:56.446879712 +0100
@@ -334,7 +334,7 @@ receive(
 	 * reveals a clogging attack.
 	 */
 	sys_received++;
-	if (SRCPORT(&rbufp->recv_srcadr) < NTP_PORT) {
+	if (SRCPORT(&rbufp->recv_srcadr) == 0) {
 		sys_badlength++;
 		return;				/* bogus port */
 	}
