diff -up ntp-4.2.6p5/util/ntp-keygen.c.rsaexp ntp-4.2.6p5/util/ntp-keygen.c
--- ntp-4.2.6p5/util/ntp-keygen.c.rsaexp	2015-02-09 12:18:06.627127473 +0100
+++ ntp-4.2.6p5/util/ntp-keygen.c	2015-02-09 12:28:15.263780280 +0100
@@ -882,7 +882,7 @@ gen_rsa(
 	FILE	*str;
 
 	fprintf(stderr, "Generating RSA keys (%d bits)...\n", modulus);
-	rsa = RSA_generate_key(modulus, 3, cb, "RSA");
+	rsa = RSA_generate_key(modulus, 65537, cb, "RSA");
 	fprintf(stderr, "\n");
 	if (rsa == NULL) {
 		fprintf(stderr, "RSA generate keys fails\n%s\n",
@@ -1223,7 +1223,7 @@ gen_gqkey(
 	fprintf(stderr,
 	    "Generating GQ parameters (%d bits)...\n",
 	     modulus2);
-	rsa = RSA_generate_key(modulus2, 3, cb, "GQ");
+	rsa = RSA_generate_key(modulus2, 65537, cb, "GQ");
 	fprintf(stderr, "\n");
 	if (rsa == NULL) {
 		fprintf(stderr, "RSA generate keys fails\n%s\n",
