diff -up ntp-4.2.6p5/libntp/decodenetnum.c.noservres ntp-4.2.6p5/libntp/decodenetnum.c
--- ntp-4.2.6p5/libntp/decodenetnum.c.noservres	2011-08-04 06:13:43.000000000 +0200
+++ ntp-4.2.6p5/libntp/decodenetnum.c	2013-03-28 18:38:49.239727130 +0100
@@ -70,7 +70,7 @@ decodenetnum(
 	}
 	ZERO(hints);
 	hints.ai_flags = Z_AI_NUMERICHOST;
-	err = getaddrinfo(cp, "ntp", &hints, &ai);
+	err = getaddrinfo(cp, NULL, &hints, &ai);
 	if (err != 0)
 		return 0;
 	NTP_INSIST(ai->ai_addrlen <= sizeof(*netnum));
