diff -up ntp-4.2.6p5/ntpd/ntp_io.c.updatebclient ntp-4.2.6p5/ntpd/ntp_io.c
--- ntp-4.2.6p5/ntpd/ntp_io.c.updatebclient	2012-11-20 15:54:23.516362641 +0100
+++ ntp-4.2.6p5/ntpd/ntp_io.c	2012-11-20 15:55:05.464364038 +0100
@@ -2044,6 +2044,9 @@ update_interfaces(
 		msyslog(LOG_INFO, "peers refreshed");
 	}
 
+	if (sys_bclient)
+		io_setbclient();
+
 	return new_interface_found;
 }
 
@@ -2678,9 +2681,6 @@ io_setbclient(void)
 	set_pktinfo(1);
 	if (nif > 0)
 		DPRINTF(1, ("io_setbclient: Opened broadcast clients\n"));
-	else if (!nif)
-		msyslog(LOG_ERR,
-			"Unable to listen for broadcasts, no broadcast interfaces available");
 #else
 	msyslog(LOG_ERR,
 		"io_setbclient: Broadcast Client disabled by build");
