diff -up ntp-4.2.6p5/ntpd/ntp_proto.c.delaycalib ntp-4.2.6p5/ntpd/ntp_proto.c
--- ntp-4.2.6p5/ntpd/ntp_proto.c.delaycalib	2012-02-28 15:57:57.000000000 +0100
+++ ntp-4.2.6p5/ntpd/ntp_proto.c	2012-02-28 16:01:30.080135978 +0100
@@ -1514,7 +1514,7 @@ process_packet(
 		 */
 		if (FLAG_BC_VOL & peer->flags) {
 			peer->flags &= ~FLAG_BC_VOL;
-			peer->delay = (peer->offset - p_offset) * 2;
+			peer->delay = fabs(peer->offset - p_offset) * 2;
 		}
 		p_del = peer->delay;
 		p_offset += p_del / 2;
