diff -up ntp-4.2.6p2/ntpd/ntpd-opts.c.multiopts ntp-4.2.6p2/ntpd/ntpd-opts.c
--- ntp-4.2.6p2/ntpd/ntpd-opts.c.multiopts	2010-09-15 17:37:10.000000000 +0200
+++ ntp-4.2.6p2/ntpd/ntpd-opts.c	2010-10-01 13:28:49.000000000 +0200
@@ -755,7 +755,7 @@ static tOptDesc optDesc[ OPTION_CT ] = {
   {  /* entry idx, value */ 18, VALUE_OPT_PIDFILE,
      /* equiv idx, value */ 18, VALUE_OPT_PIDFILE,
      /* equivalenced to  */ NO_EQUIVALENT,
-     /* min, max, act ct */ 0, 1, 0,
+     /* min, max, act ct */ 0, 2, 0,
      /* opt state flags  */ PIDFILE_FLAGS, 0,
      /* last opt argumnt */ { NULL },
      /* arg list/cookie  */ NULL,
@@ -839,7 +839,7 @@ static tOptDesc optDesc[ OPTION_CT ] = {
   {  /* entry idx, value */ 25, VALUE_OPT_USER,
      /* equiv idx, value */ 25, VALUE_OPT_USER,
      /* equivalenced to  */ NO_EQUIVALENT,
-     /* min, max, act ct */ 0, 1, 0,
+     /* min, max, act ct */ 0, 2, 0,
      /* opt state flags  */ USER_FLAGS, 0,
      /* last opt argumnt */ { NULL },
      /* arg list/cookie  */ NULL,
