diff -up ntp-4.2.6p5/libntp/authreadkeys.c.keylen ntp-4.2.6p5/libntp/authreadkeys.c
--- ntp-4.2.6p5/libntp/authreadkeys.c.keylen	2015-02-09 12:33:15.549485698 +0100
+++ ntp-4.2.6p5/libntp/authreadkeys.c	2015-02-09 13:03:44.938842731 +0100
@@ -74,7 +74,7 @@ authreadkeys(
 	keyid_t	keyno;
 	int	keytype;
 	char	buf[512];		/* lots of room for line */
-	u_char	keystr[20];
+	u_char	keystr[32];
 	int	len;
 
 	/*
