diff -up ntp-4.2.6p5/ntpq/ntpq-subs.c.mreadvar ntp-4.2.6p5/ntpq/ntpq-subs.c
--- ntp-4.2.6p5/ntpq/ntpq-subs.c.mreadvar	2011-12-25 00:27:15.000000000 +0100
+++ ntp-4.2.6p5/ntpq/ntpq-subs.c	2015-02-09 12:13:02.215449708 +0100
@@ -857,8 +857,8 @@ mreadvar(
 				&from, &to))
 		return;
 
+	memset(tmplist, 0, sizeof(tmplist));
 	if (pcmd->nargs >= 3) {
-		memset(tmplist, 0, sizeof(tmplist));
 		doaddvlist(tmplist, pcmd->argval[2].string);
 		pvars = tmplist;
 	} else {
