diff -up ntp-4.2.6p3/ntpd/ntp_proto.c.delaycalib ntp-4.2.6p3/ntpd/ntp_proto.c
--- ntp-4.2.6p3/ntpd/ntp_proto.c.delaycalib	2010-11-25 15:21:52.578492466 +0100
+++ ntp-4.2.6p3/ntpd/ntp_proto.c	2010-11-25 15:21:55.987625500 +0100
@@ -1498,7 +1498,7 @@ process_packet(
 		 */
 		if (peer->cast_flags & MDF_BCLNT) {
 			peer->cast_flags &= ~MDF_BCLNT;
-			peer->delay = (peer->offset - p_offset) * 2;
+			peer->delay = fabs(peer->offset - p_offset) * 2;
 		}
 		p_del = peer->delay;
 		p_offset += p_del / 2;
