diff -up ntp-4.2.6p5/ntpd/ntpdbase-opts.def.testmain ntp-4.2.6p5/ntpd/ntpdbase-opts.def
--- ntp-4.2.6p5/ntpd/ntpdbase-opts.def.testmain	2014-08-25 15:11:30.998454198 +0200
+++ ntp-4.2.6p5/ntpd/ntpdbase-opts.def	2014-08-25 15:11:31.063454347 +0200
@@ -8,7 +8,6 @@ include = <<- _EOF_
 	#endif
 	_EOF_;
 
-test-main;
 
 flag = {
     name      = ipv4;
diff -up ntp-4.2.6p5/ntpdc/ntpdc-opts.def.testmain ntp-4.2.6p5/ntpdc/ntpdc-opts.def
--- ntp-4.2.6p5/ntpdc/ntpdc-opts.def.testmain	2009-12-09 08:36:37.000000000 +0100
+++ ntp-4.2.6p5/ntpdc/ntpdc-opts.def	2014-08-25 15:11:31.063454347 +0200
@@ -10,7 +10,6 @@ argument       = '[ host ...]';
 #include homerc.def
 #include autogen-version.def
 
-test-main;
 
 flag = {
     name      = ipv4;
diff -up ntp-4.2.6p5/ntpq/ntpq-opts.def.testmain ntp-4.2.6p5/ntpq/ntpq-opts.def
--- ntp-4.2.6p5/ntpq/ntpq-opts.def.testmain	2009-12-09 08:36:36.000000000 +0100
+++ ntp-4.2.6p5/ntpq/ntpq-opts.def	2014-08-25 15:11:31.063454347 +0200
@@ -10,7 +10,6 @@ prog-name      = "ntpq";
 prog-title     = "standard NTP query program";
 argument       = '[ host ...]';
 
-test-main;
 
 flag = {
     name      = ipv4;
diff -up ntp-4.2.6p5/ntpsnmpd/ntpsnmpd-opts.def.testmain ntp-4.2.6p5/ntpsnmpd/ntpsnmpd-opts.def
--- ntp-4.2.6p5/ntpsnmpd/ntpsnmpd-opts.def.testmain	2010-05-03 12:01:12.000000000 +0200
+++ ntp-4.2.6p5/ntpsnmpd/ntpsnmpd-opts.def	2014-08-25 15:11:31.064454349 +0200
@@ -9,7 +9,6 @@ autogen definitions options;
 prog-name      = "ntpsnmpd";
 prog-title     = "NTP SNMP MIB agent";
 
-test-main;
 
 flag = {
     name      = nofork;
diff -up ntp-4.2.6p5/sntp/sntp-opts.def.testmain ntp-4.2.6p5/sntp/sntp-opts.def
--- ntp-4.2.6p5/sntp/sntp-opts.def.testmain	2011-09-23 05:17:26.000000000 +0200
+++ ntp-4.2.6p5/sntp/sntp-opts.def	2014-08-25 15:11:31.064454349 +0200
@@ -19,7 +19,6 @@ environrc;
 
 #include version.def
 
-test-main;
 
 flag = {
     name      = ipv4;
diff -up ntp-4.2.6p5/util/ntp-keygen-opts.def.testmain ntp-4.2.6p5/util/ntp-keygen-opts.def
--- ntp-4.2.6p5/util/ntp-keygen-opts.def.testmain	2014-08-25 15:11:31.005454214 +0200
+++ ntp-4.2.6p5/util/ntp-keygen-opts.def	2014-08-25 15:11:31.064454349 +0200
@@ -10,7 +10,6 @@ prog-name      = "ntp-keygen";
 prog-title     = "Create a NTP host key";
 package        = ntp;
 
-test-main;
 
 include        = '#include <stdlib.h>';
 #include       version.def
