diff -up libimobiledevice-1.1.6/libimobiledevice-1.0.pc.in.pkgconfig_overlinking libimobiledevice-1.1.6/libimobiledevice-1.0.pc.in
--- libimobiledevice-1.1.6/libimobiledevice-1.0.pc.in.pkgconfig_overlinking	2014-04-14 18:30:31.203585495 -0300
+++ libimobiledevice-1.1.6/libimobiledevice-1.0.pc.in	2014-04-14 18:31:16.869851617 -0300
@@ -6,7 +6,7 @@ includedir=@includedir@
 Name: libimobiledevice
 Description: A library to communicate with services running on Apple iPhone/iPod Touch devices.
 Version: @VERSION@
-Requires: libplist >= @LIBPLIST_VERSION@ libusbmuxd >= @LIBUSBMUXD_VERSION@ @ssl_requires@
+Requires.private: libplist >= @LIBPLIST_VERSION@ libusbmuxd >= @LIBUSBMUXD_VERSION@ @ssl_requires@
 Libs: -L${libdir} -limobiledevice
 Cflags: -I${includedir}
 
