diff -up ncurses-5.9/c++/Makefile.in.libs ncurses-5.9/c++/Makefile.in
--- ncurses-5.9/c++/Makefile.in.libs	2013-03-18 17:43:56.589838616 +0100
+++ ncurses-5.9/c++/Makefile.in	2013-03-18 17:54:15.484320959 +0100
@@ -104,7 +104,7 @@ LOCAL_LIBDIR	= @top_builddir@/lib
 
 LINK		= @LINK_PROGS@ $(LIBTOOL_LINK) @CXXLDFLAGS@
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lform@LIB_SUFFIX@ -lmenu@LIB_SUFFIX@ -lpanel@LIB_SUFFIX@ -lncurses@LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lform@LIB_SUFFIX@ -lmenu@LIB_SUFFIX@ -lpanel@LIB_SUFFIX@ -lncurses@LIB_SUFFIX@ #@SHLIB_LIST@
 
 LIBROOT		= ncurses++
 
@@ -139,8 +139,7 @@ LDFLAGS_SHARED	= $(TEST_LDFLAGS) $(CFLAG
 LDFLAGS_DEFAULT	= $(LINK_@DFT_UPR_MODEL@) $(LDFLAGS_@DFT_UPR_MODEL@)
 
 # flags for library built by this makefile
-LDFLAGS		= $(TEST_ARGS) @LDFLAGS@ \
-	@LD_MODEL@ $(TEST_LIBS) @LIBS@ $(CXXLIBS)
+LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@ $(CXXLIBS)
 
 AUTO_SRC	= \
 		etip.h
diff -up ncurses-5.9/form/Makefile.in.libs ncurses-5.9/form/Makefile.in
--- ncurses-5.9/form/Makefile.in.libs	2012-02-07 18:25:33.000000000 +0100
+++ ncurses-5.9/form/Makefile.in	2012-02-07 18:26:45.915065925 +0100
@@ -100,7 +100,7 @@ LINK		= $(LIBTOOL_LINK)
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@LIB_SUFFIX@ #@SHLIB_LIST@
 
 RPATH_LIST	= @RPATH_LIST@
 MK_SHARED_LIB	= @MK_SHARED_LIB@
diff -up ncurses-5.9/menu/Makefile.in.libs ncurses-5.9/menu/Makefile.in
--- ncurses-5.9/menu/Makefile.in.libs	2012-02-07 18:25:33.000000000 +0100
+++ ncurses-5.9/menu/Makefile.in	2012-02-07 18:26:03.251063857 +0100
@@ -100,7 +100,7 @@ LINK		= $(LIBTOOL_LINK)
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@LIB_SUFFIX@ #@SHLIB_LIST@
 
 RPATH_LIST	= @RPATH_LIST@
 MK_SHARED_LIB	= @MK_SHARED_LIB@
diff -up ncurses-5.9/panel/Makefile.in.libs ncurses-5.9/panel/Makefile.in
--- ncurses-5.9/panel/Makefile.in.libs	2012-02-07 18:25:33.000000000 +0100
+++ ncurses-5.9/panel/Makefile.in	2012-02-07 18:26:28.798065062 +0100
@@ -101,7 +101,7 @@ LINK		= $(LIBTOOL_LINK)
 LDFLAGS		= @LDFLAGS@ @LD_MODEL@ @LIBS@
 
 SHLIB_DIRS	= -L../lib
-SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@LIB_SUFFIX@ @SHLIB_LIST@
+SHLIB_LIST	= $(SHLIB_DIRS) -lncurses@LIB_SUFFIX@ #@SHLIB_LIST@
 
 RPATH_LIST	= @RPATH_LIST@
 MK_SHARED_LIB	= @MK_SHARED_LIB@
