diff -up udev-151/extras/keymap/keys-from-name.gperf udev-151/extras/keymap/keys-from-name
diff -up udev-151/Makefile.am.gperf udev-151/Makefile.am
--- udev-151/Makefile.am.gperf	2010-03-19 14:04:36.000000000 +0100
+++ udev-151/Makefile.am	2010-03-19 14:04:52.000000000 +0100
@@ -550,7 +550,7 @@ extras/keymap/keys-from-name.gperf: extr
 	$(AM_V_GEN)$(AWK) 'BEGIN{ print "struct key { const char* name; unsigned short id; };"; print "%null-strings"; print "%%";} { print $$1 ", " $$1 }' < $< > $@
 
 extras/keymap/keys-from-name.h: extras/keymap/keys-from-name.gperf Makefile
-	$(AM_V_GEN)$(GPERF) -L ANSI-C -t --ignore-case -N lookup_key -H hash_key_name -p -C < $< > $@
+	$(AM_V_GEN)$(GPERF) -L ANSI-C -t --ignore-case -N lookup_key -H hash_key_name -p -C -D < $< > $@
 
 extras/keymap/keys-to-name.h: extras/keymap/keys.txt Makefile
 	$(AM_V_GEN)$(AWK) 'BEGIN{ print "const char* const key_names[KEY_CNT] = { "} { print "[" $$1 "] = \"" $$1 "\"," } END{print "};"}' < $< > $@
