--- extras/keymap/95-keymap.rules.orig	2011-10-31 10:47:10.000000000 -0200
+++ extras/keymap/95-keymap.rules	2011-12-09 23:13:45.042777774 -0200
@@ -30,6 +30,7 @@ ENV{ID_VENDOR}=="Lite-On_Technology_Corp
 ENV{ID_VENDOR_ID}=="04b3", ENV{ID_MODEL_ID}=="301[89]", RUN+="keymap $name ibm-thinkpad-usb-keyboard-trackpoint"
 
 ENV{ID_VENDOR}=="Microsoft", ENV{ID_MODEL_ID}=="00db", RUN+="keymap $name 0xc022d zoomin 0xc022e zoomout"
+ENV{ID_VENDOR_ID}=="045e", ENV{ID_MODEL_ID}=="0730", RUN+="keymap $name microsoft-digital-3000"
 
 GOTO="keyboard_end"
 
