diff --git a/rules/redhat/40-redhat.rules b/rules/redhat/40-redhat.rules
index 47ce9e5..aaba82b 100644
--- a/rules/redhat/40-redhat.rules
+++ b/rules/redhat/40-redhat.rules
@@ -5,11 +5,11 @@ KERNEL=="hd*[0-9]", ATTRS{media}=="floppy", ATTRS{removable}=="1", SYMLINK+="flo
 
 KERNEL=="fw*", PROGRAM="fw_unit_symlinks.sh %k %n", SYMLINK+="$result"
 
-ACTION=="add", KERNEL=="sg[0-9]*", BUS=="scsi", ATTRS{type}=="3|6", SYMLINK+="scanner scanner-%k", MODE="0660"
+ACTION=="add", KERNEL=="sg[0-9]*", SUBSYSTEM=="scsi_device", ATTRS{type}=="3|6", SYMLINK+="scanner scanner-%k", MODE="0660"
 
-ACTION=="add", KERNEL=="sg[0-9]*", BUS=="scsi", ATTRS{type}=="8", SYMLINK+="changer changer-%k", MODE="0660", GROUP="disk"
+ACTION=="add", KERNEL=="sg[0-9]*", SUBSYSTEM=="scsi_device", ATTRS{type}=="8", SYMLINK+="changer changer-%k", MODE="0660", GROUP="disk"
 
-BUS=="usb", KERNEL=="sd*", ATTRS{bInterfaceClass}=="08", ATTRS{bInterfaceSubClass}=="04", GROUP="floppy", MODE="0660", 	SYMLINK+="floppy floppy-%k"
+SUBSYSTEM=="usb", KERNEL=="sd*", ATTRS{bInterfaceClass}=="08", ATTRS{bInterfaceSubClass}=="04", GROUP="floppy", MODE="0660", 	SYMLINK+="floppy floppy-%k"
 
 KERNEL=="fd[0-9]*",		SYMLINK+="floppy floppy-%k"
 
@@ -25,3 +25,5 @@ KERNEL=="vcsa[0-9]*",          OWNER="vcsa", GROUP="tty"
 KERNEL=="vcc/*",               OWNER="vcsa", GROUP="tty"
 
 KERNEL=="event*", ATTRS{idVendor}=="03f0", ATTRS{interface}=="Virtual Mouse", ATTRS{bInterfaceProtocol}=="02", SYMLINK+="input/hp_ilo_mouse"
+
+ACTION=="add",  SUBSYSTEM=="cpu", KERNEL=="cpu[0-9]*", ATTR{online}="1"
