diff --git a/src/LYBookmark.c b/src/LYBookmark.c
index 6d3a9ca..deaab00 100644
--- a/src/LYBookmark.c
+++ b/src/LYBookmark.c
@@ -496,7 +496,7 @@ void remove_bookmark_link(int cur,
      */
     if (stat(filename_buffer, &stat_buf) == 0) {
 	regular = (BOOLEAN) (S_ISREG(stat_buf.st_mode) && stat_buf.st_nlink == 1);
-	mode = ((stat_buf.st_mode & 0777) | 0600);	/* make it writable */
+	mode = ((stat_buf.st_mode & HIDE_CHMOD) | 0600);	/* make it writable */
 	(void) chmod(newfile, mode);
 	if ((nfp = LYReopenTemp(newfile)) == NULL) {
 	    (void) LYCloseInput(fp);
