diff -Nur lynx2-8-7.orig/userdefs.h lynx2-8-7/userdefs.h
--- lynx2-8-7.orig/userdefs.h	2009-06-06 19:30:35.000000000 -0500
+++ lynx2-8-7/userdefs.h	2009-12-03 15:47:54.596486242 -0600
@@ -289,7 +289,7 @@
 #ifdef DOSPATH
 #define LYNX_CFG_FILE "./lynx.cfg"
 #else
-#define LYNX_CFG_FILE "/usr/local/lib/lynx.cfg"
+#define LYNX_CFG_FILE "/etc/lynx.cfg"
 #endif /* DOSPATH */
 #endif /* LYNX_CFG_FILE */
 #endif /* HAVE_CONFIG_H */
@@ -305,7 +305,7 @@
  * Mappings in these global and personal files override any SUFFIX
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_EXTENSION_MAP MIME_LIBDIR "mime.types"
+#define GLOBAL_EXTENSION_MAP "@SLACK_LIBDIR@/lynx/mime.types"
 #define PERSONAL_EXTENSION_MAP "~/.mime.types"
 
 /**************************
@@ -315,7 +315,7 @@
  * Mappings in these global and personal files override any VIEWER
  * definitions in lynx.cfg and built-in defaults from src/HTInit.c.
  */
-#define GLOBAL_MAILCAP MIME_LIBDIR "mailcap"
+#define GLOBAL_MAILCAP MIME_LIBDIR "/etc/mailcap"
 #define PERSONAL_MAILCAP "~/.mailcap"
 
 /**************************
@@ -445,7 +445,7 @@
  * -lss command line switch will override these definitions.
  */
 #ifndef LYNX_LSS_FILE
-#define LYNX_LSS_FILE "/usr/local/lib/lynx.lss"
+#define LYNX_LSS_FILE "/etc/lynx.lss"
 #endif /* LYNX_LSS_FILE */
 
 #endif /* VMS OR UNIX */
