diff -Nur lynx2-8-7.orig/lynx.cfg lynx2-8-7/lynx.cfg
--- lynx2-8-7.orig/lynx.cfg	2009-06-23 18:53:58.000000000 -0500
+++ lynx2-8-7/lynx.cfg	2009-12-03 15:35:18.744461098 -0600
@@ -3025,8 +3025,18 @@
 #COLOR:4:magenta:white
 #COLOR:5:blue:white
 #COLOR:6:red:white
-COLOR:6:brightred:black
+#COLOR:6:brightred:black
 #COLOR:7:magenta:cyan
+COLOR:0:black:white
+COLOR:1:blue:white
+COLOR:2:yellow:blue
+COLOR:3:green:white
+COLOR:4:magenta:white
+COLOR:5:blue:white
+# This makes links a *lot* more visible.  The default settings are like
+# "let's torture the color-blind" :^)
+COLOR:6:brightred:black
+COLOR:7:magenta:cyan
 
 
 .h2 COLOR_STYLE
