diff --git a/makefile.in b/makefile.in
index d84837e..b245f38 100644
--- a/makefile.in
+++ b/makefile.in
@@ -96,8 +96,6 @@ COMPRESS_EXT=@COMPRESS_EXT@
 # Path of scripts directory
 scripts_dir=$(srcdir)/scripts
 
-MAKE_RECUR = $(MAKE) DESTDIR="$(DESTDIR)" CC="$(CC)"
-
 # !!!!!!!!!!! SUN resolv LIBRARY !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 # To include resolv in the LIBS="" list for SUN 3, 4 or Solaris OS,
 # point RESOLVLIB to that library.  You need this if you get the message
@@ -189,13 +187,13 @@ SRC_CFLAGS = \
 
 all lynx$x: cfg_defs.h LYHelp.h
 @MSG_DIR_MAKE@	$(SHELL) $(scripts_dir)/fixtext.sh $(srcdir)/LYMessages_en.h >LYMessages.c
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR)
-	cd $(WWW_DIR) && $(MAKE_RECUR) $(WWW_CFLAGS)
-	cd $(SRC_DIR) && $(MAKE_RECUR) all $(SRC_CFLAGS)
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR)
+	$(MAKE) -C $(WWW_DIR) $(WWW_CFLAGS)
+	$(MAKE) -C $(SRC_DIR) all $(SRC_CFLAGS)
 
 lint:
-	cd $(WWW_DIR) && $(MAKE_RECUR) LINT="$(LINT)" $(WWW_CFLAGS) $@
-	cd $(SRC_DIR) && $(MAKE_RECUR) LINT="$(LINT)" $(SRC_CFLAGS) $@
+	$(MAKE) -C $(WWW_DIR) LINT="$(LINT)" $(WWW_CFLAGS) $@
+	$(MAKE) -C $(SRC_DIR) LINT="$(LINT)" $(SRC_CFLAGS) $@
 
 help:
 	@echo
@@ -212,27 +210,27 @@ help:
 clean:
 	rm -f WWW/Library/*/*.[aoib]
 	rm -f WWW/Library/*/.created
-	cd $(WWW_DIR) && $(MAKE_RECUR) $@
-	cd $(SRC_DIR) && $(MAKE_RECUR) $@
+	$(MAKE) -C $(WWW_DIR) $@
+	$(MAKE) -C $(SRC_DIR) $@
 @MSG_DIR_MAKE@	rm -f LYMessages.c
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) $@
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) $@
 	rm -f *.b $(SRC_DIR)/lynx$x *.leaks cfg_defs.h LYHelp.h lint.*
 @LYNXCFG_MAKE@	-rm -f alphatoc.html body.html cattoc.html tmp007
 	rm -f help_files.sed
 	rm -f core *.core
 
 depend: cfg_defs.h LYHelp.h
-	cd $(WWW_DIR) && $(MAKE_RECUR) $@
-	cd $(SRC_DIR) && $(MAKE_RECUR) $@
+	$(MAKE) -C $(WWW_DIR) $@
+	$(MAKE) -C $(SRC_DIR) $@
 
 distclean: clean
 	-rm -f WWW/Library/*/*~
 	-rm -f WWW/Library/*/*.bak
 	-rm -rf $(SRC_DIR)/obsolete
-	-cd $(WWW_DIR) && $(MAKE_RECUR) $@
-	-cd $(SRC_DIR) && $(MAKE_RECUR) $@
-	-cd $(CHR_DIR) && $(MAKE_RECUR) $@
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) $@
+	$(MAKE) -C $(WWW_DIR) $@
+	$(MAKE) -C $(SRC_DIR) $@
+	$(MAKE) -C $(CHR_DIR) $@
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) $@
 @MSG_DIR_MAKE@	-rmdir $(PO_DIR)
 	-rm -f *~ *.bak *.sav tags TAGS
 	-rm -f $(WWW_DIR)/makefile $(SRC_DIR)/makefile $(CHR_DIR)/makefile
@@ -301,7 +299,7 @@ install-full: install install-help install-doc
 	@echo Full installation complete.
 
 install-bin: $(BINDIR) lynx$x
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) install
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) install
 	@ECHO_CC@$(SHELL) -c \
 	'if test -f $(BINDIR)/lynx$x ; then \
 	      mv -f $(BINDIR)/lynx$x $(BINDIR)/lynx.old; fi'
@@ -361,7 +359,7 @@ install-help : help_files.sed $(HELPDIR)
 	- (cd $(HELPDIR) && WD=`pwd` && HEAD=`echo $$WD|sed -e 's!/lynx_help$$!!'` && test $$WD != $$HEAD && rm -fr *)
 	test -d $(HELPDIR)/keystrokes || mkdir $(HELPDIR)/keystrokes
 @LYNXCFG_MAKE@	@echo 'Making htmlized lynx.cfg'
-@LYNXCFG_MAKE@	cd $(SRC_DIR) && $(MAKE_RECUR) LYReadCFG.i
+@LYNXCFG_MAKE@	$(MAKE) -C $(SRC_DIR) LYReadCFG.i
 @LYNXCFG_MAKE@	@-rm -f alphatoc.html body.html cattoc.html
 @LYNXCFG_MAKE@	sed -n -e '/Config_Type  *Config_Table/,/{0, *0, *0}/ p' $(SRC_DIR)/LYReadCFG.i | \
 @LYNXCFG_MAKE@	sed  -e 's/ *{ *"\([^"]*\)".*/\1/' | \
@@ -426,7 +424,7 @@ install-lss : $(SYSCONFDIR)
 	@$(SHELL) $(scripts_dir)/install-lss.sh "$(INSTALL_DATA)" $(srcdir)/samples/lynx.lss $(SYSCONFDIR)/lynx.lss
 
 uninstall ::
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) uninstall
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) uninstall
 	-rm -f $(BINDIR)/lynx$x
 	-rm -f $(MANDIR)/lynx.1
 	-rm -f $(SYSCONFDIR)/lynx.cfg
@@ -443,7 +441,7 @@ uninstall-doc ::
 
 update-po:
 	rsync -Lrtvz  translationproject.org::tp/latest/lynx/  $(PO_SRCDIR)
-	test -f $(PO_SRCDIR)/makefile && cd $(PO_SRCDIR) && $(MAKE_RECUR) $@
+	test -f $(PO_SRCDIR)/makefile && $(MAKE) -C $(PO_SRCDIR) $@
 
 $(BINDIR) \
 $(MANDIR) \
diff --git a/src/makefile.in b/src/makefile.in
index 4183bf0..d9c1587 100644
--- a/src/makefile.in
+++ b/src/makefile.in
@@ -110,7 +110,7 @@ message:
 	@echo "Compiling Lynx sources"
 
 do_chartrans_stuff:
-	-cd chrtrans && $(MAKE) \
+	$(MAKE) -C chrtrans \
 		SITE_DEFS="$(SITE_DEFS)" \
 		BUILD_CFLAGS="$(BUILD_CFLAGS)" \
 		BUILD_CPPFLAGS="$(BUILD_CPPFLAGS)" \
@@ -123,7 +123,7 @@ lint:
 
 clean:
 	rm -f lynx$x core *.core *.leaks *.i *$o *.bak tags TAGS test_*
-	cd chrtrans && $(MAKE) clean
+	$(MAKE) -C chrtrans clean
 
 tags:
 	ctags *.[ch]
@@ -194,8 +194,8 @@ TABLES= \
  $(CHRTR)utf8_uni.h \
  $(CHRTR)viscii_uni.h
 
-$(TABLES):
-	-cd chrtrans && $(MAKE) tables
+$(TABLES): chrtrans/makeuctb$(BUILD_EXEEXT)
+	$(MAKE) -C chrtrans tables
 
 UCdomap$o :	UCdomap.c \
 		chrtrans/UCkd.h \
@@ -204,7 +204,7 @@ UCdomap$o :	UCdomap.c \
 	UCdomap.h $(CMN)UCMap.h $(TABLES) $(top_srcdir)/userdefs.h
 
 chrtrans/makeuctb$(BUILD_EXEEXT):
-	cd chrtrans; make makeuctb$(BUILD_EXEEXT)
+	$(MAKE) -C chrtrans makeuctb$(BUILD_EXEEXT)
 
 UCAux$o : UCAux.c $(CMN)UCAux.h $(CMN)UCDefs.h
 LYCookie$o : $(top_srcdir)/userdefs.h
