diff -up lynx2-8-8/makefile.in.build lynx2-8-8/makefile.in
--- lynx2-8-8/makefile.in.build	2013-11-28 22:52:56.000000000 -0200
+++ lynx2-8-8/makefile.in	2014-01-05 12:38:39.665409277 -0200
@@ -199,14 +199,14 @@ binary_PROG  = $(actual_PROG)$x
 
 all lynx$x: cfg_defs.h LYHelp.h
 @MSG_DIR_MAKE@	$(SHELL) $(scripts_dir)/fixtext.sh $(srcdir)/LYMessages_en.h >LYMessages.c
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR)
-	cd $(WWW_DIR) && $(MAKE_RECUR) $(WWW_CFLAGS)
-	cd $(SRC_DIR) && $(MAKE_RECUR) all $(SRC_CFLAGS)
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR)
+	$(MAKE) -C $(WWW_DIR) $(WWW_CFLAGS)
+	$(MAKE) -C $(SRC_DIR) all $(SRC_CFLAGS)
 
 lint:
-	cd $(WWW_DIR) && $(MAKE_RECUR) LINT="$(LINT)" $(WWW_CFLAGS) $@
-	cd $(SRC_DIR) && $(MAKE_RECUR) LINT="$(LINT)" $(SRC_CFLAGS) $@
-	cd $(CHR_DIR) && $(MAKE_RECUR) LINT="$(LINT)" $@
+	$(MAKE) -C $(WWW_DIR) LINT="$(LINT)" $(WWW_CFLAGS) $@
+	$(MAKE) -C $(SRC_DIR) LINT="$(LINT)" $(SRC_CFLAGS) $@
+	$(MAKE) -C $(CHR_DIR) LINT="$(LINT)" $@
 
 tags:
 	cd $(WWW_DIR) && $(MAKE_RECUR) $(WWW_CFLAGS) $@
@@ -231,27 +231,24 @@ help:
 clean:
 	rm -f WWW/Library/*/*.[aoib]
 	rm -f WWW/Library/*/.created
-	cd $(WWW_DIR) && $(MAKE_RECUR) $@
-	cd $(SRC_DIR) && $(MAKE_RECUR) $@
+	$(MAKE) -C $(WWW_DIR) $@
+	$(MAKE) -C $(SRC_DIR) $@
 @MSG_DIR_MAKE@	rm -f LYMessages.c
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) $@
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) $@
 	rm -f *.b $(SRC_DIR)/lynx$x *.leaks cfg_defs.h LYHelp.h lint.*
 @LYNXCFG_MAKE@	-rm -f $(CFG2HTML)
 	rm -f help_files.sed
 	rm -f core *.core
 
 depend: cfg_defs.h LYHelp.h
-	cd $(WWW_DIR) && $(MAKE_RECUR) $@
-	cd $(SRC_DIR) && $(MAKE_RECUR) $@
+	$(MAKE) -C $(WWW_DIR) $@
+	$(MAKE) -C $(SRC_DIR) $@
 
 distclean: clean
-	-rm -f WWW/Library/*/*~
-	-rm -f WWW/Library/*/*.bak
-	-rm -rf $(SRC_DIR)/obsolete
-	-cd $(WWW_DIR) && $(MAKE_RECUR) $@
-	-cd $(SRC_DIR) && $(MAKE_RECUR) $@
-	-cd $(CHR_DIR) && $(MAKE_RECUR) $@
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) $@
+	$(MAKE) -C $(WWW_DIR) $@
+	$(MAKE) -C $(SRC_DIR) $@
+	$(MAKE) -C $(CHR_DIR) $@
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) $@
 @MSG_DIR_MAKE@	-rmdir $(PO_DIR)
 	-rm -f *~ *.bak *.sav tags TAGS
 	-rm -f $(WWW_DIR)/makefile $(SRC_DIR)/makefile $(CHR_DIR)/makefile
@@ -320,7 +317,7 @@ install-full: install install-help insta
 	@echo Full installation complete.
 
 install-bin: $(BINDIR) lynx$x
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) install
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) install
 	@ECHO_CC@$(SHELL) -c \
 	'if test -f $(BINDIR)/$(binary_PROG) ; then \
 	      mv -f $(BINDIR)/$(binary_PROG) $(BINDIR)/lynx.old; fi'
@@ -356,7 +353,7 @@ LYNXHELP_URL='$(LYNX_URL)/lynx_help/'
 
 @LYNXCFG_MAKE@$(CFG2HTML) :
 @LYNXCFG_MAKE@	@echo 'Making htmlized lynx.cfg'
-@LYNXCFG_MAKE@	cd $(SRC_DIR) && $(MAKE_RECUR) LYReadCFG.i
+@LYNXCFG_MAKE@	$(MAKE) -C $(SRC_DIR) LYReadCFG.i
 @LYNXCFG_MAKE@	@-rm -f $(CFG2HTML)
 @LYNXCFG_MAKE@	sed -n -e '/Config_Type  *Config_Table/,/{0, *0, *0}/ p' $(SRC_DIR)/LYReadCFG.i | \
 @LYNXCFG_MAKE@	sed  -e 's/ *{ *"\([^"]*\)".*/\1/' | \
@@ -454,7 +451,7 @@ install-lss : $(SYSCONFDIR)
 	@$(SHELL) $(scripts_dir)/install-lss.sh "$(INSTALL_DATA)" $(srcdir)/samples/lynx.lss $(SYSCONFDIR)/lynx.lss
 
 uninstall ::
-@MSG_DIR_MAKE@	cd $(PO_DIR) && $(MAKE_RECUR) uninstall
+@MSG_DIR_MAKE@	$(MAKE) -C $(PO_DIR) uninstall
 	-rm -f $(BINDIR)/$(binary_PROG)
 	-rm -f $(MANDIR)/$(actual_PROG).1
 	-rm -f $(SYSCONFDIR)/lynx.cfg
@@ -487,7 +484,7 @@ uninstall-doc ::
 
 update-po:
 	rsync -Lrtvz  translationproject.org::tp/latest/lynx/  $(PO_SRCDIR)
-	test -f $(PO_SRCDIR)/makefile && cd $(PO_SRCDIR) && $(MAKE_RECUR) $@
+	test -f $(PO_SRCDIR)/makefile && $(MAKE) -C $(PO_SRCDIR) $@
 
 preinstall :
 	@ echo ''
diff -up lynx2-8-8/src/makefile.in.build lynx2-8-8/src/makefile.in
--- lynx2-8-8/src/makefile.in.build	2014-01-05 12:33:47.255469754 -0200
+++ lynx2-8-8/src/makefile.in	2014-01-05 12:41:25.358381620 -0200
@@ -116,7 +116,7 @@ message:
 	@echo "Compiling Lynx sources"
 
 do_chartrans_stuff:
-	-cd chrtrans && $(MAKE_RECUR) \
+	$(MAKE) -C chrtrans \
 		SITE_DEFS="$(SITE_DEFS)" \
 		BUILD_CFLAGS="$(BUILD_CFLAGS)" \
 		BUILD_CPPFLAGS="$(BUILD_CPPFLAGS)" \
@@ -129,7 +129,7 @@ lint:
 
 clean:
 	rm -f lynx$x core *.core *.leaks *.i *$o *.bak tags TAGS test_*
-	cd chrtrans && $(MAKE_RECUR) clean
+	$(MAKE) -C chrtrans clean
 
 tags:
 	$(CTAGS) *.[ch]
@@ -207,7 +207,8 @@ TABLES= \
  $(CHRTR)viscii_uni.h
 
 $(TABLES):
-	-cd chrtrans && $(MAKE_RECUR) tables
+	$(TABLES): chrtrans/makeuctb$(BUILD_EXEEXT)
+	$(MAKE) -C chrtrans tables
 
 UCdomap$o :	UCdomap.c \
 		chrtrans/UCkd.h \
@@ -216,7 +217,7 @@ UCdomap$o :	UCdomap.c \
 	UCdomap.h $(CMN)UCMap.h $(TABLES) $(top_srcdir)/userdefs.h
 
 chrtrans/makeuctb$(BUILD_EXEEXT):
-	cd chrtrans && $(MAKE_RECUR) makeuctb$(BUILD_EXEEXT)
+	$(MAKE) -C chrtrans makeuctb$(BUILD_EXEEXT)
 
 UCAux$o : UCAux.c $(CMN)UCAux.h $(CMN)UCDefs.h
 LYCookie$o : $(top_srcdir)/userdefs.h
