diff -up evolution-2.30.1.2/shell/e-shell-window-actions.c.help-contents evolution-2.30.1.2/shell/e-shell-window-actions.c
--- evolution-2.30.1.2/shell/e-shell-window-actions.c.help-contents	2010-04-26 09:13:18.000000000 -0400
+++ evolution-2.30.1.2/shell/e-shell-window-actions.c	2010-05-07 21:29:06.269059830 -0400
@@ -2001,6 +2001,14 @@ e_shell_window_actions_init (EShellWindo
 	if (path == NULL)
 		gtk_action_set_visible (ACTION (SUBMIT_BUG), FALSE);
 	g_free (path);
+
+	/* Viewing user documentation requires the evolution-help
+	 * Fedora package.  Look for one of the files it installs. */
+	path = g_build_filename (
+		EVOLUTION_DATADIR, "omf", PACKAGE, "evolution-C.omf", NULL);
+	if (!g_file_test (path, G_FILE_TEST_IS_REGULAR))
+		gtk_action_set_visible (ACTION (CONTENTS), FALSE);
+	g_free (path);
 }
 
 static GList *
