diff -up evolution-2.31.1/configure.ldaphack evolution-2.31.1/configure
--- evolution-2.31.1/configure.ldaphack	2010-05-03 12:43:34.000000000 +0200
+++ evolution-2.31.1/configure	2010-05-03 19:22:55.000000000 +0200
@@ -15797,7 +15797,7 @@ if test "${ac_cv_lib_ldap_ldap_open+set}
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lldap $LDAP_LDFLAGS $LDAP_LIBS $LIBS"
+LIBS="-lldap -lresolv $LDAP_LDFLAGS $LDAP_LIBS $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -16129,7 +16129,7 @@ if test "${ac_cv_lib_ldap_ldap_open+set}
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lldap $LDAP_LDFLAGS $LDAP_LIBS $LIBS"
+LIBS="-lldap -lresolv $LDAP_LDFLAGS $LDAP_LIBS $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
