diff -up libjingle-0.6.6/talk/session/phone/devicemanager.cc.alsa libjingle-0.6.6/talk/session/phone/devicemanager.cc
--- libjingle-0.6.6/talk/session/phone/devicemanager.cc.alsa	2011-12-14 10:58:28.000000000 -0500
+++ libjingle-0.6.6/talk/session/phone/devicemanager.cc	2012-01-05 13:31:43.238535696 -0500
@@ -25,6 +25,8 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
+
 #include "talk/session/phone/devicemanager.h"
 
 #include "talk/base/logging.h"
