diff -up gnome-pilot-conduits-2.0.17/configure.in.require-libgnomeui gnome-pilot-conduits-2.0.17/configure.in
--- gnome-pilot-conduits-2.0.17/configure.in.require-libgnomeui	2010-10-06 15:51:29.739276533 -0400
+++ gnome-pilot-conduits-2.0.17/configure.in	2010-10-06 15:51:37.252185243 -0400
@@ -21,7 +21,7 @@ PKG_PROG_PKG_CONFIG
 dnl ******************************
 dnl Library Checks
 dnl ******************************
-LIBGNOME_REQUIRED="libgnome-2.0 >= 2.0.0"
+LIBGNOME_REQUIRED="libgnomeui-2.0 >= 2.0.0"
 GNOME_PILOT_REQUIRED="gnome-pilot-2.0 >= 2.0.13"
 LIBXML_REQUIRED="libxml-2.0"
 
diff -up gnome-pilot-conduits-2.0.17/configure.require-libgnomeui gnome-pilot-conduits-2.0.17/configure
--- gnome-pilot-conduits-2.0.17/configure.require-libgnomeui	2010-10-06 15:52:06.523831153 -0400
+++ gnome-pilot-conduits-2.0.17/configure	2010-10-06 15:52:14.500733635 -0400
@@ -21529,7 +21529,7 @@ echo "${ECHO_T}no" >&6; }
 
 fi
 
-LIBGNOME_REQUIRED="libgnome-2.0 >= 2.0.0"
+LIBGNOME_REQUIRED="libgnomeui-2.0 >= 2.0.0"
 GNOME_PILOT_REQUIRED="gnome-pilot-2.0 >= 2.0.13"
 LIBXML_REQUIRED="libxml-2.0"
 
