From 0aa904debe1189bbf41d2d4240e2759f0fc56184 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Tue, 5 May 2015 09:36:17 +0200
Subject: [PATCH] define GNUTLS_SUPPLEMENTAL_USER_MAPPING_DATA

---
 lib/includes/gnutls/compat.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/includes/gnutls/compat.h b/lib/includes/gnutls/compat.h
index 537aef0..0e77943 100644
--- a/lib/includes/gnutls/compat.h
+++ b/lib/includes/gnutls/compat.h
@@ -240,6 +240,9 @@ gnutls_global_set_mem_functions(gnutls_alloc_function alloc_func,
 				gnutls_realloc_function realloc_func,
 				gnutls_free_function free_func) _GNUTLS_GCC_ATTR_DEPRECATED;
 
+/* defined in old headers - unused nevertheless */
+#define GNUTLS_SUPPLEMENTAL_USER_MAPPING_DATA 0
+
 /* *INDENT-OFF* */
 #ifdef __cplusplus
 }
-- 
2.1.0

