From 1d523a44a5ad360c83bff362a625cc68cbe7f296 Mon Sep 17 00:00:00 2001
From: Jan Chaloupka <jchaloup@redhat.com>
Date: Wed, 15 Oct 2014 09:46:56 +0200
Subject: [PATCH] switch man and root in init/systemd/man-db.conf

---
 init/systemd/man-db.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init/systemd/man-db.conf b/init/systemd/man-db.conf
index 10b27b4..43dd2ad 100644
--- a/init/systemd/man-db.conf
+++ b/init/systemd/man-db.conf
@@ -1 +1 @@
-d /var/cache/man 2755 man root 1w
+d /var/cache/man 2755 root man 1w
-- 
1.9.3

