diff -upr man-db-2.6.3.orig/man/man1/apropos.man1 man-db-2.6.3/man/man1/apropos.man1
--- man-db-2.6.3.orig/man/man1/apropos.man1	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man1/apropos.man1	2013-04-08 14:50:51.655718887 +0200
@@ -14,7 +14,7 @@
 %apropos% \- search the manual page names and descriptions
 .SH SYNOPSIS
 .B %apropos% 
-.RB [\| \-dalhvV  \|] 
+.RB [\| \-dalv?V  \|] 
 .RB [\| \-e \||\| \-w \||\| \-r\c
 \|]
 .RB [\| \-s
@@ -181,9 +181,12 @@ determined locale.
 Use this user configuration file rather than the default of
 .IR ~/.manpath .
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH "EXIT STATUS"
diff -upr man-db-2.6.3.orig/man/man1/lexgrog.man1 man-db-2.6.3/man/man1/lexgrog.man1
--- man-db-2.6.3.orig/man/man1/lexgrog.man1	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man1/lexgrog.man1	2013-04-08 14:53:33.021148065 +0200
@@ -12,7 +12,7 @@ lexgrog \- parse header information in m
 .SH SYNOPSIS
 .B lexgrog
 .RB [\| \-m \||\| -c \|]
-.RB [\| \-fhwV \|]
+.RB [\| \-dfw?V \|]
 .RB [\| \-E
 .IR encoding \|]
 .I file
@@ -44,6 +44,9 @@ input files is \(lq\-\(rq, it will read
 file is compressed, a decompressed version will be read automatically.
 .SH OPTIONS
 .TP
+.if !'po4a'hide' .BR \-d ", " \-\-debug
+Print debugging information.
+.TP
 .if !'po4a'hide' .BR \-m ", " \-\-man
 Parse input as man page source files.
 This is the default if neither
@@ -81,9 +84,12 @@ or
 Override the guessed character set for the page to
 .IR encoding .
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH "EXIT STATUS"
diff -upr man-db-2.6.3.orig/man/man1/man.man1 man-db-2.6.3/man/man1/man.man1
--- man-db-2.6.3.orig/man/man1/man.man1	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man1/man.man1	2013-04-08 14:56:47.286033116 +0200
@@ -146,7 +146,7 @@
 .\" --help and --version
 .br
 .B %man%
-.RB [\| \-hV \|]
+.RB [\| \-?V \|]
 .SH DESCRIPTION
 .B %man%
 is the system's manual pager. Each
@@ -622,7 +622,7 @@ before displaying the error message, it
 option was supplied, using the name as a filename and looking for an
 exact match.
 .TP
-.if !'po4a'hide' .BR \-w ", " \-\-where ", " \-\-location
+.if !'po4a'hide' .BR \-w ", " \-\-where ", " \-\-path ", " \-\-location
 Don't actually display the manual pages, but do print the location(s) of
 the source nroff files that would be formatted.
 .TP
@@ -1148,9 +1148,12 @@ It implies
 .BR \-t .
 .SS "Getting help"
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH "EXIT STATUS"
diff -upr man-db-2.6.3.orig/man/man1/manpath.man1 man-db-2.6.3/man/man1/manpath.man1
--- man-db-2.6.3.orig/man/man1/manpath.man1	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man1/manpath.man1	2013-04-08 15:04:06.597117555 +0200
@@ -14,7 +14,7 @@
 %manpath% \- determine search path for manual pages
 .SH SYNOPSIS
 .B %manpath%  
-.RB [\| \-qgdchV \|] 
+.RB [\| \-qgdc?V \|] 
 .RB [\| \-m
 .IR system \|[\|,.\|.\|.\|]\|]
 .RB [\| \-C
@@ -87,9 +87,12 @@ environment variable.
 Use this user configuration file rather than the default of
 .IR ~/.manpath .
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH ENVIRONMENT
diff -upr man-db-2.6.3.orig/man/man1/whatis.man1 man-db-2.6.3/man/man1/whatis.man1
--- man-db-2.6.3.orig/man/man1/whatis.man1	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man1/whatis.man1	2013-04-08 15:04:51.455417497 +0200
@@ -14,7 +14,7 @@
 %whatis% \- display manual page descriptions
 .SH SYNOPSIS
 .B %whatis% 
-.RB [\| \-dlhvV \|] 
+.RB [\| \-dlv?V \|] 
 .RB [\| \-r \||\| \-w\c 
 \|] 
 .RB [\| \-s
@@ -185,9 +185,12 @@ determined locale.
 Use this user configuration file rather than the default of
 .IR ~/.manpath .
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH "EXIT STATUS"
diff -upr man-db-2.6.3.orig/man/man8/accessdb.man8 man-db-2.6.3/man/man8/accessdb.man8
--- man-db-2.6.3.orig/man/man8/accessdb.man8	2011-06-06 11:29:21.000000000 +0200
+++ man-db-2.6.3/man/man8/accessdb.man8	2013-04-08 15:05:40.593651559 +0200
@@ -15,7 +15,7 @@ accessdb \- dumps the content of a man-d
 format
 .SH SYNOPSIS
 .B /usr/sbin/accessdb 
-.RB [\| \-hV \|]
+.RB [\| \-d?V \|]
 .RI [ <index-file> ]
 .SH DESCRIPTION
 .B accessdb 
@@ -42,9 +42,15 @@ zsoelim -> "1 1 795981543 A - - satisfy
 .fi
 .SH OPTIONS
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-d ", " \-\-debug
+Print debugging information.
+.TP
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH AUTHOR
diff -upr man-db-2.6.3.orig/man/man8/catman.man8 man-db-2.6.3/man/man8/catman.man8
--- man-db-2.6.3.orig/man/man8/catman.man8	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man8/catman.man8	2013-04-08 15:06:08.785212520 +0200
@@ -14,7 +14,7 @@
 %catman% \- create or update the pre-formatted manual pages
 .SH SYNOPSIS
 .B %catman% 
-.RB [\| \-dhV \|] 
+.RB [\| \-d?V \|] 
 .RB [\| \-M 
 .IR path \|] 
 .RB [\| \-C
@@ -66,9 +66,12 @@ in the man-db configuration file.
 Use this user configuration file rather than the default of
 .IR ~/.manpath .
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Print a help message and exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Display version information.
 .SH ENVIRONMENT
diff -upr man-db-2.6.3.orig/man/man8/mandb.man8 man-db-2.6.3/man/man8/mandb.man8
--- man-db-2.6.3.orig/man/man8/mandb.man8	2012-02-05 00:36:40.000000000 +0100
+++ man-db-2.6.3/man/man8/mandb.man8	2013-04-08 15:07:02.984369201 +0200
@@ -15,7 +15,7 @@
 %mandb% \- create or update the manual page index caches
 .SH SYNOPSIS
 .B %mandb%  
-.RB [\| \-dqsucpt \||\| \-h \||\| \-V \|]
+.RB [\| \-dqsucpt?V \|]
 .RB [\| \-C
 .IR file \|]
 .RI [\| manpath \|]
@@ -136,9 +136,12 @@ and
 Use this user configuration file rather than the default of
 .IR ~/.manpath .
 .TP
-.if !'po4a'hide' .BR \-h ", " \-\-help
+.if !'po4a'hide' .BR \-? ", " \-\-help
 Show the usage message, then exit.
 .TP
+.if !'po4a'hide' .BR \-\-usage
+Print a short usage message and exit.
+.TP
 .if !'po4a'hide' .BR \-V ", " \-\-version
 Show the version, then exit.
 .SH "EXIT STATUS"
