--- kdepim/kontact/src/main.cpp~	2010-02-26 19:34:11.000000000 +0100
+++ kdepim/kontact/src/main.cpp	2010-03-24 13:00:23.000000000 +0100
@@ -28,8 +28,6 @@
 
 #include <libkdepim/reminderclient.h>
 
-#include <Akonadi/Control>
-
 #include <KontactInterface/Plugin>
 #include <KontactInterface/UniqueAppHandler>
 #ifdef Q_WS_WIN
@@ -199,10 +197,6 @@
 
   KontactApp app;
 
-  // KDE 4.4: do akonadi startup before creating any window, since creating
-  // the window loads kmail. In 4.5 we'll do this startup async instead.
-  Akonadi::Control::start( 0 );
-
   // Qt doesn't treat the system tray as a window, and therefore Qt would quit
   // the event loop when an error message is clicked away while Kontact is in the
   // tray.
