diff -up kdepim-4.7.2/nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp.nepomuk_notification.patch kdepim-4.7.2/nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp
--- kdepim-4.7.2/nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp.nepomuk_notification.patch	2011-10-02 08:24:39.000000000 -0500
+++ kdepim-4.7.2/nepomuk_email_feeder/shared/nepomukfeederagentbase.cpp	2011-10-13 12:13:39.740055399 -0500
@@ -414,7 +414,7 @@ void NepomukFeederAgentBase::selfTest()
   emit status( Broken, i18n( "Nepomuk not operational" ) );
   if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.kde.pim.nepomukfeeder.selftestreport" ) ) )
     return;
-  KNotification::event( KNotification::Warning, i18n( "Nepomuk Indexing Disabled" ), message );
+  // KNotification::event( KNotification::Warning, i18n( "Nepomuk Indexing Disabled" ), message );
   QDBusConnection::sessionBus().unregisterService( QLatin1String( "org.kde.pim.nepomukfeeder.selftestreport" ) );
 }
 
