diff -up kdepim-4.4.93/kleopatra/libkleopatraclient/CMakeLists.txt.install-headers kdepim-4.4.93/kleopatra/libkleopatraclient/CMakeLists.txt
--- kdepim-4.4.93/kleopatra/libkleopatraclient/CMakeLists.txt.install-headers	2010-01-26 22:22:34.000000000 -0200
+++ kdepim-4.4.93/kleopatra/libkleopatraclient/CMakeLists.txt	2010-12-29 17:13:48.422145792 -0200
@@ -6,3 +6,7 @@ add_subdirectory( gui )
 
 add_subdirectory( tests )
 
+install(
+  FILES kleopatraclient_export.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient
+  )
diff -up kdepim-4.4.93/kleopatra/libkleopatraclient/core/CMakeLists.txt.install-headers kdepim-4.4.93/kleopatra/libkleopatraclient/core/CMakeLists.txt
--- kdepim-4.4.93/kleopatra/libkleopatraclient/core/CMakeLists.txt.install-headers	2010-01-26 22:22:34.000000000 -0200
+++ kdepim-4.4.93/kleopatra/libkleopatraclient/core/CMakeLists.txt	2010-12-29 17:13:48.422145792 -0200
@@ -49,4 +49,12 @@ install(
   DESTINATION ${LIB_INSTALL_DIR}
   )
   
-
+install(
+  FILES
+    initialization.h
+    command.h
+    selectcertificatecommand.h
+    signencryptfilescommand.h
+    decryptverifyfilescommand.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient/core
+  )
diff -up kdepim-4.4.93/kleopatra/libkleopatraclient/gui/CMakeLists.txt.install-headers kdepim-4.4.93/kleopatra/libkleopatraclient/gui/CMakeLists.txt
--- kdepim-4.4.93/kleopatra/libkleopatraclient/gui/CMakeLists.txt.install-headers	2010-01-13 13:20:04.000000000 -0200
+++ kdepim-4.4.93/kleopatra/libkleopatraclient/gui/CMakeLists.txt	2010-12-29 17:13:48.422145792 -0200
@@ -20,3 +20,8 @@ install(
   DESTINATION ${LIB_INSTALL_DIR}
   )
 
+install(
+  FILES
+    certificaterequester.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient/gui
+  )
diff -up kdepim-4.4.93/libkleo/CMakeLists.txt.install-headers kdepim-4.4.93/libkleo/CMakeLists.txt
--- kdepim-4.4.93/libkleo/CMakeLists.txt.install-headers	2010-04-07 12:18:39.000000000 -0300
+++ kdepim-4.4.93/libkleo/CMakeLists.txt	2010-12-29 17:20:59.535107497 -0200
@@ -114,4 +114,61 @@ endif(WIN32)
 set_target_properties(kleo PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS kleo ${INSTALL_TARGETS_DEFAULT_ARGS})
 
+install( FILES
+    kleo/kleo_export.h
+    kleo/enum.h
+    kleo/oidmap.h
+    kleo/cryptobackend.h
+    kleo/cryptobackendfactory.h
+    kleo/cryptoconfig.h
+    kleo/dn.h
+    kleo/job.h
+    kleo/keylistjob.h
+    kleo/keygenerationjob.h
+    kleo/abstractimportjob.h
+    kleo/importjob.h
+    kleo/importfromkeyserverjob.h
+    kleo/exportjob.h
+    kleo/changeexpiryjob.h
+    kleo/changeownertrustjob.h
+    kleo/downloadjob.h
+    kleo/deletejob.h
+    kleo/encryptjob.h
+    kleo/decryptjob.h
+    kleo/signjob.h
+    kleo/specialjob.h
+    kleo/verifydetachedjob.h
+    kleo/verifyopaquejob.h
+    kleo/decryptverifyjob.h
+    kleo/signencryptjob.h
+    kleo/signkeyjob.h
+    kleo/adduseridjob.h
+    kleo/refreshkeysjob.h
+    kleo/multideletejob.h
+    kleo/hierarchicalkeylistjob.h
+    kleo/keyfilter.h
+    kleo/keyfiltermanager.h
+    kleo/changepasswdjob.h
+    kleo/checksumdefinition.h
+    kleo/exception.h
+    kleo/kconfigbasedkeyfilter.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/kleo COMPONENT Devel)
+
+install ( FILES
+    ui/kdhorizontalline.h
+    ui/messagebox.h
+    ui/progressbar.h
+    ui/progressdialog.h
+    ui/keylistview.h
+    ui/keyselectiondialog.h
+    ui/keyrequester.h
+    ui/keyapprovaldialog.h
+    ui/dnattributeorderconfigwidget.h
+    ui/cryptoconfigmodule.h
+    ui/cryptoconfigdialog.h
+    ui/directoryserviceswidget.h
+    ui/filenamerequester.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/kleo/ui COMPONENT Devel)
+
+
 install ( FILES libkleopatrarc.desktop DESTINATION ${CONFIG_INSTALL_DIR} RENAME libkleopatrarc )
diff -up kdepim-4.4.93/libkpgp/CMakeLists.txt.install-headers kdepim-4.4.93/libkpgp/CMakeLists.txt
--- kdepim-4.4.93/libkpgp/CMakeLists.txt.install-headers	2010-02-20 09:32:15.000000000 -0200
+++ kdepim-4.4.93/libkpgp/CMakeLists.txt	2010-12-29 17:13:48.423145563 -0200
@@ -35,3 +35,11 @@ add_subdirectory( tests )
 install(FILES kpgp.upd  DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 install(PROGRAMS kpgp-3.1-upgrade-address-data.pl DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 
+install(FILES
+        kpgp.h
+        kpgpbase.h
+        kpgpblock.h
+        kpgpkey.h
+        kpgpui.h
+        libkpgp_export.h
+        DESTINATION ${INCLUDE_INSTALL_DIR}/kpgp COMPONENT Devel )
