--- kaddressbook/main.cpp~	2009-12-11 00:14:25.000000000 +0100
+++ kaddressbook/main.cpp	2010-03-27 20:12:15.000000000 +0100
@@ -21,7 +21,6 @@
 #include "aboutdata.h"
 #include "mainwindow.h"
 
-#include <akonadi/control.h>
 #include <kcmdlineargs.h>
 #include <kdebug.h>
 #include <klocale.h>
@@ -49,11 +48,5 @@
   MainWindow *window = new MainWindow;
   window->show();
 
-  if ( !Akonadi::Control::start( window ) ) {
-    //TODO: add message box after string freeze
-    kWarning() << "Unable to start Akonadi server, exit application";
-    return 1;
-  }
-
   return app.exec();
 }
