--- kdepim/kmail/main.cpp~	2009-12-11 00:14:23.000000000 +0100
+++ kdepim/kmail/main.cpp	2010-03-24 12:54:03.000000000 +0100
@@ -145,13 +145,6 @@
   app.setEventLoopReached();
   app.delayedInstanceCreation();
 
-  // Start Akonadi
-  if ( !Akonadi::Control::start( kmkernel->getKMMainWidget() ) ) {
-    //TODO: add message box after string freeze
-    kWarning() << "Unable to start Akonadi server, exit application";
-    return 1;
-  }
-
   // Go!
   int ret = qApp->exec();
   // clean up
