diff -up kdepim-4.9.95/kleopatra/libkleopatraclient/CMakeLists.txt.install-headers kdepim-4.9.95/kleopatra/libkleopatraclient/CMakeLists.txt
--- kdepim-4.9.95/kleopatra/libkleopatraclient/CMakeLists.txt.install-headers	2012-12-17 08:59:44.000000000 -0600
+++ kdepim-4.9.95/kleopatra/libkleopatraclient/CMakeLists.txt	2012-12-23 15:39:43.179789520 -0600
@@ -6,3 +6,7 @@ add_subdirectory(gui)
 
 add_subdirectory(tests)
 
+install(
+  FILES kleopatraclient_export.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient
+  )
diff -up kdepim-4.9.95/kleopatra/libkleopatraclient/core/CMakeLists.txt.install-headers kdepim-4.9.95/kleopatra/libkleopatraclient/core/CMakeLists.txt
--- kdepim-4.9.95/kleopatra/libkleopatraclient/core/CMakeLists.txt.install-headers	2012-12-17 08:59:44.000000000 -0600
+++ kdepim-4.9.95/kleopatra/libkleopatraclient/core/CMakeLists.txt	2012-12-23 15:40:21.725318920 -0600
@@ -72,3 +72,13 @@ else(WINCE)
 endif()
 
 install(TARGETS kleopatraclientcore ${INSTALL_TARGETS_DEFAULT_ARGS})
+
+install(
+  FILES
+    initialization.h
+    command.h
+    selectcertificatecommand.h
+    signencryptfilescommand.h
+    decryptverifyfilescommand.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient/core
+  )
diff -up kdepim-4.9.95/kleopatra/libkleopatraclient/gui/CMakeLists.txt.install-headers kdepim-4.9.95/kleopatra/libkleopatraclient/gui/CMakeLists.txt
--- kdepim-4.9.95/kleopatra/libkleopatraclient/gui/CMakeLists.txt.install-headers	2012-12-17 08:59:44.000000000 -0600
+++ kdepim-4.9.95/kleopatra/libkleopatraclient/gui/CMakeLists.txt	2012-12-23 15:40:45.286032483 -0600
@@ -16,3 +16,9 @@ set_target_properties(kleopatraclientgui
 )
 
 install(TARGETS kleopatraclientgui ${INSTALL_TARGETS_DEFAULT_ARGS})
+
+install(
+  FILES
+    certificaterequester.h
+  DESTINATION ${INCLUDE_INSTALL_DIR}/libkleopatraclient/gui
+  )
diff -up kdepim-4.9.95/libkleo/CMakeLists.txt.install-headers kdepim-4.9.95/libkleo/CMakeLists.txt
--- kdepim-4.9.95/libkleo/CMakeLists.txt.install-headers	2012-12-17 08:59:44.000000000 -0600
+++ kdepim-4.9.95/libkleo/CMakeLists.txt	2012-12-23 15:27:21.303870886 -0600
@@ -112,6 +112,64 @@ endif(WIN32)
 set_target_properties(kleo PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS kleo ${INSTALL_TARGETS_DEFAULT_ARGS})
 
+install( FILES
+    kleo/kleo_export.h
+    kleo/enum.h
+    kleo/oidmap.h
+    kleo/cryptobackend.h
+    kleo/cryptobackendfactory.h
+    kleo/cryptoconfig.h
+    kleo/dn.h
+    kleo/job.h
+    kleo/keylistjob.h
+    kleo/keygenerationjob.h
+    kleo/abstractimportjob.h
+    kleo/importjob.h
+    kleo/importfromkeyserverjob.h
+    kleo/exportjob.h
+    kleo/changeexpiryjob.h
+    kleo/changeownertrustjob.h
+    kleo/downloadjob.h
+    kleo/deletejob.h
+    kleo/encryptjob.h
+    kleo/decryptjob.h
+    kleo/signjob.h
+    kleo/specialjob.h
+    kleo/verifydetachedjob.h
+    kleo/verifyopaquejob.h
+    kleo/decryptverifyjob.h
+    kleo/signencryptjob.h
+    kleo/signkeyjob.h
+    kleo/adduseridjob.h
+    kleo/refreshkeysjob.h
+    kleo/multideletejob.h
+    kleo/hierarchicalkeylistjob.h
+    kleo/keyfilter.h
+    kleo/keyfiltermanager.h
+    kleo/changepasswdjob.h
+    kleo/checksumdefinition.h
+    kleo/exception.h
+    kleo/kconfigbasedkeyfilter.h
+    kleo/listallkeysjob.h
+    kleo/stl_util.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/kleo COMPONENT Devel)
+
+install ( FILES
+    ui/kdhorizontalline.h
+    ui/messagebox.h
+    ui/progressbar.h
+    ui/progressdialog.h
+    ui/keylistview.h
+    ui/keyselectiondialog.h
+    ui/keyrequester.h
+    ui/keyapprovaldialog.h
+    ui/dnattributeorderconfigwidget.h
+    ui/cryptoconfigmodule.h
+    ui/cryptoconfigdialog.h
+    ui/directoryserviceswidget.h
+    ui/filenamerequester.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/kleo/ui COMPONENT Devel)
+
 if ( WIN32 )
 install ( FILES libkleopatrarc-win32.desktop DESTINATION ${CONFIG_INSTALL_DIR} RENAME libkleopatrarc )
 else ( WIN32 )
diff -up kdepim-4.9.95/libkpgp/CMakeLists.txt.install-headers kdepim-4.9.95/libkpgp/CMakeLists.txt
--- kdepim-4.9.95/libkpgp/CMakeLists.txt.install-headers	2012-12-17 08:59:44.000000000 -0600
+++ kdepim-4.9.95/libkpgp/CMakeLists.txt	2012-12-23 15:27:21.303870886 -0600
@@ -33,3 +33,11 @@ add_subdirectory( tests )
 install(FILES kpgp.upd  DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 install(PROGRAMS kpgp-3.1-upgrade-address-data.pl DESTINATION ${KCONF_UPDATE_INSTALL_DIR})
 
+install(FILES
+        kpgp.h
+        kpgpbase.h
+        kpgpblock.h
+        kpgpkey.h
+        kpgpui.h
+        libkpgp_export.h
+        DESTINATION ${INCLUDE_INSTALL_DIR}/kpgp COMPONENT Devel )
