diff -up libmygpo-qt.1.0.6/libmygpo-qt.pc.in.pkgconfig libmygpo-qt.1.0.6/libmygpo-qt.pc.in
--- libmygpo-qt.1.0.6/libmygpo-qt.pc.in.pkgconfig	2012-07-20 06:58:51.000000000 -0500
+++ libmygpo-qt.1.0.6/libmygpo-qt.pc.in	2012-10-13 17:56:34.899552335 -0500
@@ -6,8 +6,8 @@ includedir=@INCLUDE_INSTALL_DIR@
 Name: libmygpo-qt
 Description: libmygpo-qt is a C++/Qt Library that wraps the gpodder.net WebAPI
 URL: http://wiki.gpodder.org/wiki/Libmygpo-qt
-Version: @MYGPO_LIB_VERSION@
+Version: @MYGPO_QT_VERSION@
 Requires: QtCore QtNetwork
 Requires.private: QJson
 Libs: -L${libdir} -lmygpo-qt
-Cflags: -I${includedir}
\ No newline at end of file
+Cflags: -I${includedir}
