--- transmission-2.51/qt/transmission-qt.desktop.orig	2012-04-09 18:17:40.830418654 -0300
+++ transmission-2.51/qt/transmission-qt.desktop	2012-04-09 18:17:56.741484858 -0300
@@ -8,11 +8,3 @@ Terminal=false
 Type=Application
 MimeType=application/x-bittorrent;x-scheme-handler/magnet;
 Categories=Network;FileTransfer;P2P;Qt;
-X-AppInstall-Keywords=torrent
-Actions=Minimize;
-
-[Desktop Action Minimize]
-_Name=Start Transmission Minimized
-Exec=transmission-qt --minimized
-OnlyShowIn=Unity;
-
