--- qt/app.cc.orig	2010-12-09 17:53:20.000000000 -0200
+++ qt/app.cc	2010-12-10 19:18:56.490931758 -0200
@@ -96,8 +96,10 @@ MyApp :: MyApp( int& argc, char ** argv
     qtTranslator.load( "qt_" + QLocale::system().name(), QLibraryInfo::location(QLibraryInfo::TranslationsPath));
     installTranslator( &qtTranslator );
 
+    QString dataDir = QLatin1String("_PKGDATADIR_");
+
     // install the transmission translator
-    appTranslator.load( QString(MY_CONFIG_NAME) + "_" + QLocale::system().name(), QCoreApplication::applicationDirPath() + "/translations" );
+    appTranslator.load( QString(MY_CONFIG_NAME) + "_" + QLocale::system().name(), dataDir + QDir::separator() + "translations" );
     installTranslator( &appTranslator );
 
     Formatter::initUnits( );
