--- configure.ac.orig	2014-02-21 23:52:12.134345034 -0300
+++ configure.ac	2014-02-21 23:52:38.896674489 -0300
@@ -186,7 +186,7 @@ AC_ARG_WITH([systemd-daemon], AS_HELP_ST
             [Add support for systemd startup notification (default is autodetected)])
             [USE_SYSTEMD_DAEMON=$withval], [USE_SYSTEMD_DAEMON=auto])
 AS_IF([test "x$USE_SYSTEMD_DAEMON" != "xno"], [
-    PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd-daemon],
+    PKG_CHECK_MODULES([SYSTEMD_DAEMON], [libsystemd],
         [AC_DEFINE(USE_SYSTEMD_DAEMON,1,[Use systemd startup notification])],
         [AS_IF([test "x$USE_SYSTEMD_DAEMON" = "xyes"],
             [AC_MSG_ERROR([systemd startup notification support requested, but libsystemd-daemon not found.])]
