diff -up transmission-2.76/gtk/transmission-gtk.desktop.in.orig transmission-2.76/gtk/transmission-gtk.desktop.in
--- transmission-2.76/gtk/transmission-gtk.desktop.in.orig	2013-01-09 20:49:50.020694342 -0200
+++ transmission-2.76/gtk/transmission-gtk.desktop.in	2013-01-09 20:49:59.124765688 -0200
@@ -13,14 +13,3 @@ MimeType=application/x-bittorrent;x-sche
 Categories=Network;FileTransfer;P2P;GTK;
 X-Ubuntu-Gettext-Domain=transmission
 X-AppInstall-Keywords=torrent 
-Actions=Pause;Minimize;
-
-[Desktop Action Pause]
-_Name=Start Transmission with All Torrents Paused
-Exec=transmission-gtk --paused
-OnlyShowIn=Unity;
-
-[Desktop Action Minimize]
-_Name=Start Transmission Minimized
-Exec=transmission-gtk --minimized
-OnlyShowIn=Unity;
diff -up transmission-2.76/qt/transmission-qt.desktop.orig transmission-2.76/qt/transmission-qt.desktop
--- transmission-2.76/qt/transmission-qt.desktop.orig	2013-01-09 20:47:06.598313989 -0200
+++ transmission-2.76/qt/transmission-qt.desktop	2013-01-09 20:50:02.914962751 -0200
@@ -9,10 +9,3 @@ Type=Application
 MimeType=application/x-bittorrent;x-scheme-handler/magnet;
 Categories=Network;FileTransfer;P2P;Qt;
 X-AppInstall-Keywords=torrent
-Actions=Minimize;
-
-[Desktop Action Minimize]
-Name=Start Transmission Minimized
-Exec=transmission-qt --minimized
-OnlyShowIn=Unity;
-
