diff -rupN redland-1.0.10.old/src/Makefile.in redland-1.0.10/src/Makefile.in
--- redland-1.0.10.old/src/Makefile.in	2009-12-14 01:53:52.000000000 -0500
+++ redland-1.0.10/src/Makefile.in	2010-02-14 16:48:45.000000000 -0500
@@ -1383,7 +1383,7 @@ rdf_concepts_test: rdf_concepts.c librdf
 	$(COMPILE_LINK) -DSTANDALONE $(srcdir)/rdf_concepts.c librdf.la
 
 rdf_query_test: rdf_query.c rdf_query_results.c librdf.la
-	$(COMPILE_LINK) -DSTANDALONE $(srcdir)/rdf_query.c $(srcdir)/rdf_query_results.c librdf.la
+	$(COMPILE_LINK) -lraptor -lrasqal -DSTANDALONE $(srcdir)/rdf_query.c $(srcdir)/rdf_query_results.c librdf.la
 
 rdf_serializer_test: rdf_serializer.c librdf.la
 	$(COMPILE_LINK) -DSTANDALONE $(srcdir)/rdf_serializer.c librdf.la
diff -rupN redland-1.0.10.old/utils/Makefile.in redland-1.0.10/utils/Makefile.in
--- redland-1.0.10.old/utils/Makefile.in	2009-12-14 01:53:52.000000000 -0500
+++ redland-1.0.10/utils/Makefile.in	2010-02-14 16:44:20.000000000 -0500
@@ -68,7 +68,7 @@ am__rdfproc_SOURCES_DIST = rdfproc.c get
 @GETOPT_TRUE@am__objects_1 = getopt.$(OBJEXT)
 am_rdfproc_OBJECTS = rdfproc.$(OBJEXT) $(am__objects_1)
 rdfproc_OBJECTS = $(am_rdfproc_OBJECTS)
-rdfproc_LDADD = $(LDADD)
+rdfproc_LDADD = -lraptor -lrasqal $(LDADD)
 rdfproc_DEPENDENCIES = $(top_builddir)/src/librdf.la
 am_redland_db_upgrade_OBJECTS = db_upgrade.$(OBJEXT)
 redland_db_upgrade_OBJECTS = $(am_redland_db_upgrade_OBJECTS)
@@ -317,7 +317,7 @@ LDADD = $(top_builddir)/src/librdf.la
 AM_LDFLAGS = @LIBRDF_LDFLAGS@
 redland_db_upgrade_SOURCES = db_upgrade.c
 redland_virtuoso_test_SOURCES = redland-virtuoso-test.c
-redland_virtuoso_test_LDADD = @LIBRDF_LDFLAGS@ $(top_builddir)/src/librdf.la
+redland_virtuoso_test_LDADD = -lraptor @LIBRDF_LDFLAGS@ $(top_builddir)/src/librdf.la
 rdfproc_SOURCES = rdfproc.c $(am__append_1)
 all: all-am
 
