diff -up kio-giobridge-20110605/giobridge/giobridge-errormapper.h.orig kio-giobridge-20110605/giobridge/giobridge-errormapper.h
--- kio-giobridge-20110605/giobridge/giobridge-errormapper.h.orig	2011-06-05 01:59:26.728291869 -0300
+++ kio-giobridge-20110605/giobridge/giobridge-errormapper.h	2011-06-05 01:59:39.230631514 -0300
@@ -24,6 +24,10 @@
 #include <kurl.h>
 #include <kio/ioslave_defaults.h>
 #include <glib.h>
+//work around compile issue with glib >= 2.25
+#ifdef signals
+ #undef signals
+#endif
 #include <gio/gio.h>
 
 class GioBridgeErrorMapper
diff -up kio-giobridge-20110605/giobridge/giobridge.h.orig kio-giobridge-20110605/giobridge/giobridge.h
--- kio-giobridge-20110605/giobridge/giobridge.h.orig	2011-06-05 01:59:04.377047938 -0300
+++ kio-giobridge-20110605/giobridge/giobridge.h	2011-06-05 02:00:13.402360158 -0300
@@ -35,6 +35,10 @@
 #include <unistd.h>
 
 #include <glib.h>
+//work around compile issue with glib >= 2.25
+#ifdef signals
+ #undef signals
+#endif
 #include <gio/gio.h>
 
 #include "giobridge-errormapper.h"
diff -up kio-giobridge-20110605/kdedmodule/giobridge-remotedirnotify.h.orig kio-giobridge-20110605/kdedmodule/giobridge-remotedirnotify.h
--- kio-giobridge-20110605/kdedmodule/giobridge-remotedirnotify.h.orig	2011-06-05 02:00:33.964984647 -0300
+++ kio-giobridge-20110605/kdedmodule/giobridge-remotedirnotify.h	2011-06-05 02:00:44.940649135 -0300
@@ -24,6 +24,10 @@
 #include <kurl.h>
 #include <QtCore/QObject>
 #include <glib.h>
+//work around compile issue with glib >= 2.25
+#ifdef signals
+ #undef signals
+#endif
 #include <gio/gio.h>
 
 class GioBridgeRemoteDirNotify : public QObject
