diff -up konsole-4.8.0/src/TerminalDisplay.cpp.kdebug280896 konsole-4.8.0/src/TerminalDisplay.cpp
--- konsole-4.8.0/src/TerminalDisplay.cpp.kdebug280896	2011-12-02 15:31:44.000000000 -0600
+++ konsole-4.8.0/src/TerminalDisplay.cpp	2012-01-31 18:03:12.750555833 -0600
@@ -311,7 +311,7 @@ TerminalDisplay::TerminalDisplay(QWidget
 ,_resizeTimer(0)
 ,_flowControlWarningEnabled(false)
 ,_outputSuspendedLabel(0)
-,_lineSpacing(0)
+,_lineSpacing(1)
 ,_colorsInverted(false)
 ,_blendColor(qRgba(0,0,0,0xff))
 ,_filterChain(new TerminalImageFilterChain())
@@ -723,7 +723,7 @@ void TerminalDisplay::drawCharacters(QPa
         if (_bidiEnabled)
             painter.drawText(rect,0,text);
         else
-            painter.drawText(rect,0,LTR_OVERRIDE_CHAR+text);
+            painter.drawText(rect,Qt::AlignBottom,LTR_OVERRIDE_CHAR+text);
     }
 }
 
