diff -up konsole-4.7.95/src/TerminalDisplay.cpp.kde280896 konsole-4.7.95/src/TerminalDisplay.cpp
--- konsole-4.7.95/src/TerminalDisplay.cpp.kde280896	2011-12-02 15:31:44.000000000 -0600
+++ konsole-4.7.95/src/TerminalDisplay.cpp	2011-12-27 21:11:28.386653332 -0600
@@ -723,7 +723,7 @@ void TerminalDisplay::drawCharacters(QPa
         if (_bidiEnabled)
             painter.drawText(rect,0,text);
         else
-            painter.drawText(rect,0,LTR_OVERRIDE_CHAR+text);
+            painter.drawText(rect,Qt::AlignBottom,LTR_OVERRIDE_CHAR+text);
     }
 }
 
