diff -up konsole-4.14.1/src/History.cpp.tmp konsole-4.14.1/src/History.cpp
--- konsole-4.14.1/src/History.cpp.tmp	2014-09-09 07:47:03.000000000 -0500
+++ konsole-4.14.1/src/History.cpp	2014-09-23 11:42:48.891387252 -0500
@@ -61,7 +61,7 @@ HistoryFile::HistoryFile()
       _fileMap(0),
       _readWriteBalance(0)
 {
-    const QString tmpFormat = KStandardDirs::locateLocal("tmp", QString())
+    const QString tmpFormat = KStandardDirs::locateLocal("cache", QString())
                               + "konsole-XXXXXX.history";
     _tmpFile.setFileTemplate(tmpFormat);
     if (_tmpFile.open()) {
