diff -up lzo-2.04/configure.ac.configure lzo-2.04/configure.ac
--- lzo-2.04/configure.ac.configure	2010-10-31 15:21:28.000000000 -0200
+++ lzo-2.04/configure.ac	2010-11-08 19:43:27.442825840 -0200
@@ -120,7 +120,7 @@ asm_dir=
 asm_msg_amd64=no
 asm_msg_i386=no
 
-case "$host_cpu-$ac_cv_sizeof_void_p" in
+case "$target_cpu-$ac_cv_sizeof_void_p" in
     amd64-8 | x86_64-8) asm_arch="amd64"; asm_dir="asm/amd64/src_gas/elf64" ;;
     i?86-4) asm_arch="i386"; asm_dir="asm/i386/src_gas" ;;
     *) enable_asm=no ;;
diff -up lzo-2.04/configure.configure lzo-2.04/configure
--- lzo-2.04/configure.configure	2010-10-31 15:21:28.000000000 -0200
+++ lzo-2.04/configure	2010-11-08 19:45:54.403047000 -0200
@@ -12598,7 +12598,7 @@ asm_dir=
 asm_msg_amd64=no
 asm_msg_i386=no
 
-case "$host_cpu-$ac_cv_sizeof_void_p" in
+case "$target_cpu-$ac_cv_sizeof_void_p" in
     amd64-8 | x86_64-8) asm_arch="amd64"; asm_dir="asm/amd64/src_gas/elf64" ;;
     i?86-4) asm_arch="i386"; asm_dir="asm/i386/src_gas" ;;
     *) enable_asm=no ;;
