diff -up lzo-2.05/configure.ac.configure lzo-2.05/configure.ac
--- lzo-2.05/configure.ac.configure	2011-04-23 07:55:35.000000000 -0300
+++ lzo-2.05/configure.ac	2011-04-23 16:27:11.174617879 -0300
@@ -122,7 +122,7 @@ asm_dir=
 asm_msg_amd64=no
 asm_msg_i386=no
 
-case "$host_cpu-$ac_cv_sizeof_void_p" in
+case "$target_cpu-$ac_cv_sizeof_void_p" in
     amd64-8 | x86_64-8) asm_arch="amd64"; asm_dir="asm/amd64/src_gas/elf64"; asm_mode="amd64_src_gas_elf64" ;;
     i?86-4) asm_arch="i386"; asm_dir="asm/i386/src_gas"; asm_mode="i386_src_gas" ;;
     *) enable_asm=no ;;
diff -up lzo-2.05/configure.configure lzo-2.05/configure
--- lzo-2.05/configure.configure	2011-04-23 07:55:35.000000000 -0300
+++ lzo-2.05/configure	2011-04-23 16:27:33.677827080 -0300
@@ -12860,7 +12860,7 @@ asm_dir=
 asm_msg_amd64=no
 asm_msg_i386=no
 
-case "$host_cpu-$ac_cv_sizeof_void_p" in
+case "$target_cpu-$ac_cv_sizeof_void_p" in
     amd64-8 | x86_64-8) asm_arch="amd64"; asm_dir="asm/amd64/src_gas/elf64"; asm_mode="amd64_src_gas_elf64" ;;
     i?86-4) asm_arch="i386"; asm_dir="asm/i386/src_gas"; asm_mode="i386_src_gas" ;;
     *) enable_asm=no ;;
