diff -up calligra-2.4.91/plugins/staging/googledocs/CMakeLists.txt.tables_to_sheets calligra-2.4.91/plugins/staging/googledocs/CMakeLists.txt
--- calligra-2.4.91/plugins/staging/googledocs/CMakeLists.txt.tables_to_sheets	2012-06-15 12:11:08.000000000 -0500
+++ calligra-2.4.91/plugins/staging/googledocs/CMakeLists.txt	2012-06-15 21:53:25.945071659 -0500
@@ -33,7 +33,7 @@ install(TARGETS calligragoogledocs  DEST
 ########### install files ###############
 
 install( FILES  googledocs-stage.rc  DESTINATION ${DATA_INSTALL_DIR}/stage/kpartplugins)
-install( FILES  googledocs-sheets.rc  DESTINATION ${DATA_INSTALL_DIR}/tables/kpartplugins)
+install( FILES  googledocs-sheets.rc  DESTINATION ${DATA_INSTALL_DIR}/sheets/kpartplugins)
 install( FILES  googledocs-words.rc  DESTINATION ${DATA_INSTALL_DIR}/words/kpartplugins)
 
 
diff -up calligra-2.4.91/plugins/staging/googledocs/googledocumentservice.cpp.tables_to_sheets calligra-2.4.91/plugins/staging/googledocs/googledocumentservice.cpp
--- calligra-2.4.91/plugins/staging/googledocs/googledocumentservice.cpp.tables_to_sheets	2012-06-15 12:11:08.000000000 -0500
+++ calligra-2.4.91/plugins/staging/googledocs/googledocumentservice.cpp	2012-06-15 21:52:31.298754832 -0500
@@ -88,7 +88,7 @@ void GoogleDocumentService::listDocument
     case OnlineDocument::STAGE:
         url = "https://docs.google.com/feeds/default/private/full/-/presentation";
         break;
-    case OnlineDocument::TABLES:
+    case OnlineDocument::SHEETS:
         url = "https://docs.google.com/feeds/default/private/full/-/spreadsheet";
         break;
     }
diff -up calligra-2.4.91/plugins/staging/googledocs/onlinedocument.cpp.tables_to_sheets calligra-2.4.91/plugins/staging/googledocs/onlinedocument.cpp
--- calligra-2.4.91/plugins/staging/googledocs/onlinedocument.cpp.tables_to_sheets	2012-06-15 12:11:08.000000000 -0500
+++ calligra-2.4.91/plugins/staging/googledocs/onlinedocument.cpp	2012-06-15 21:52:31.298754832 -0500
@@ -56,8 +56,8 @@ OnlineDocument::OnlineDocument(QObject *
         m_type = OnlineDocument::WORDS;
     } else if (name.contains("stage")) {
         m_type = OnlineDocument::STAGE;
-    } else if (name.contains("tables")) {
-        m_type = OnlineDocument::TABLES;
+    } else if (name.contains("sheets")) {
+        m_type = OnlineDocument::SHEETS;
     } else {
         m_type = OnlineDocument::UNKNOWN;
     }
diff -up calligra-2.4.91/plugins/staging/googledocs/onlinedocument.h.tables_to_sheets calligra-2.4.91/plugins/staging/googledocs/onlinedocument.h
--- calligra-2.4.91/plugins/staging/googledocs/onlinedocument.h.tables_to_sheets	2012-06-15 12:11:08.000000000 -0500
+++ calligra-2.4.91/plugins/staging/googledocs/onlinedocument.h	2012-06-15 21:52:31.299754820 -0500
@@ -31,7 +31,7 @@ public:
     enum DocumentType {
         WORDS,
         STAGE,
-        TABLES,
+        SHEETS,
         UNKNOWN
     };
 
