--- filters/libmsooxml/MsooXmlTableStyle.cpp	2014-03-24 11:38:42.000000000 -0300
+++ filters/libmsooxml/MsooXmlTableStyle.cpp	2014-03-02 11:46:07.000000000 -0300
@@ -440,7 +440,7 @@ void TableStyleConverter::reapplyTableLe
         //cell-level border set to "None"
         if ((data.outerPen.widthF() == 0) && (data.style == KoBorder::BorderSolid)) {
             if (setProperties & TableStyleProperties::InsideHBorder) {
-                style->borders()->setBorderData(KoBorder::BottomBorDer, properties->insideH);
+                style->borders()->setBorderData(KoBorder::BottomBorder, properties->insideH);
             }
             if (setLocalProperties & TableStyleProperties::InsideHBorder) {
                 style->borders()->setBorderData(KoBorder::BottomBorder, localProperties->insideH);
