diff -up calligra-2.7.90/CMakeLists.txt.no_mpxj calligra-2.7.90/CMakeLists.txt
--- calligra-2.7.90/CMakeLists.txt.no_mpxj	2013-12-01 12:24:21.000000000 -0600
+++ calligra-2.7.90/CMakeLists.txt	2013-12-14 13:03:47.354220510 -0600
@@ -981,6 +981,7 @@ calligra_drop_products_on_old_flag(words
 # by calling calligra_disable_product with the product id and the reason,
 # e.g.:
 # calligra_disable_product(KEXI_APP "isn't buildable at the moment")
+calligra_disable_product(FILTER_MPXJ_IMPORT "missing runtime deps")
 
 #############################################
 ####     Calculate buildable products    ####
