diff -up calligra-2.9.0/CMakeLists.txt.disable_products calligra-2.9.0/CMakeLists.txt
--- calligra-2.9.0/CMakeLists.txt.disable_products	2015-02-24 07:01:01.000000000 -0600
+++ calligra-2.9.0/CMakeLists.txt	2015-02-26 19:28:15.239118472 -0600
@@ -966,6 +966,11 @@ calligra_drop_products_on_old_flag(words
 # by calling calligra_disable_product with the product id and the reason,
 # e.g.:
 # calligra_disable_product(APP_KEXI "isn't buildable at the moment")
+calligra_disable_product(FILTER_MPXJ_IMPORT "missing runtime deps")
+calligra_disable_product(APP_ACTIVE "not wanted")
+calligra_disable_product(APP_CSTESTER "not wanted")
+calligra_disable_product(APP_DEVTOOLS "not wanted")
+calligra_disable_product(APP_GEMINI "not wanted")
 
 #############################################
 ####     Calculate buildable products    ####
