diff -up calligra-2.3.87/filters/words/hancomword/pole.cpp.gcc47 calligra-2.3.87/filters/words/hancomword/pole.cpp
--- calligra-2.3.87/filters/words/hancomword/pole.cpp.gcc47	2012-01-28 01:04:47.000000000 -0600
+++ calligra-2.3.87/filters/words/hancomword/pole.cpp	2012-01-29 17:04:11.520633241 -0600
@@ -866,7 +866,7 @@ void StorageIO::load()
         unsigned char* buffer2 = new unsigned char[ bbat->blockSize ];
         unsigned k = 109;
         unsigned mblock = header->mbat_start;
-        for (unsigned r = 0; r < header->num_mbat; r++) {
+        for (unsigned rr = 0; rr < header->num_mbat; rr++) {
             unsigned long r = loadBigBlock(mblock, buffer2, bbat->blockSize);
             if (r != bbat->blockSize) {
                 delete[] buffer2;
