diff -up calligra-2.3.92/sheets/plugins/scripting/scripts/CMakeLists.txt.tables_to_sheets calligra-2.3.92/sheets/plugins/scripting/scripts/CMakeLists.txt
--- calligra-2.3.92/sheets/plugins/scripting/scripts/CMakeLists.txt.tables_to_sheets	2012-03-16 13:12:48.000000000 -0500
+++ calligra-2.3.92/sheets/plugins/scripting/scripts/CMakeLists.txt	2012-03-18 15:25:53.478480452 -0500
@@ -1,6 +1,6 @@
 install(FILES
     scripts.rc
-    DESTINATION ${DATA_INSTALL_DIR}/tables/scripts)
+    DESTINATION ${DATA_INSTALL_DIR}/sheets/scripts)
 
 install(FILES
     csvimport.py csvexport.py csvoptions.ui
@@ -11,16 +11,16 @@ install(FILES
     logger.py
     xmlviewer.py
     myorca.py
-    DESTINATION ${DATA_INSTALL_DIR}/tables/scripts/extensions)
+    DESTINATION ${DATA_INSTALL_DIR}/sheets/scripts/extensions)
 
 install(FILES
     functions.py
     rpyfunctions.py
     yfinance.py yweather.py
     pytime.py pyregexp.py
-    DESTINATION ${DATA_INSTALL_DIR}/tables/scripts/functions)
+    DESTINATION ${DATA_INSTALL_DIR}/sheets/scripts/functions)
 
 install(FILES
     consoledocker.py
     snippetsdocker.rb
-    DESTINATION ${DATA_INSTALL_DIR}/tables/scripts/docker)
+    DESTINATION ${DATA_INSTALL_DIR}/sheets/scripts/docker)
