--- configure.in.orig	2012-02-15 19:40:55.205760778 -0200
+++ configure.in	2012-02-15 19:41:16.081298955 -0200
@@ -135,7 +135,7 @@ AC_DEFUN([TEST_LIBAV],
             #include <libavcodec/avcodec.h>
             #include <libswscale/swscale.h>
             ]],[[
-                avcodec_init();
+                avcodec_register_all();
                 swscale_version();
                 #ifdef FFMS_USE_FFMPEG_COMPAT
                 int bogus = CODEC_ID_G2M;
