diff -ru ffms-2.17-src/src/core/matroskaparser.c ffms-2.17-src.orig/src/core/matroskaparser.c
--- ffms-2.17-src/src/core/matroskaparser.c	2011-01-27 21:38:25.000000000 +0100
+++ ffms-2.17-src.orig/src/core/matroskaparser.c	2012-10-02 21:45:42.000000000 +0200
@@ -91,6 +91,7 @@
 #define	MAXU64		      ULL(0xffffffffffffffff)
 #define	ONE		      ULL(1)
 
+
 // compatibility
 static char  *mystrdup(struct InputStream *is,const char *src) {
   size_t  len;
@@ -1366,9 +1367,9 @@
       readLangCC(mf, len, t.Language);
       break;
     case 0x86: // CodecID
-      if (t.CodecID)
+      if (t.AVCodecID)
 	errorjmp(mf,"Duplicate CodecID");
-      STRGETA(mf,t.CodecID,len);
+      STRGETA(mf,t.AVCodecID,len);
       break;
     case 0x63a2: // CodecPrivate
       if (cp)
@@ -1459,7 +1460,7 @@
   ENDFOR(mf);
 
   // validate track info
-  if (!t.CodecID)
+  if (!t.AVCodecID)
     errorjmp(mf,"Track has no Codec ID");
 
   if (t.UID != 0) {
@@ -1525,8 +1526,8 @@
   // copy strings
   if (t.Name)
     cpadd += strlen(t.Name)+1;
-  if (t.CodecID)
-    cpadd += strlen(t.CodecID)+1;
+  if (t.AVCodecID)
+    cpadd += strlen(t.AVCodecID)+1;
 
   tp = mf->cache->memalloc(mf->cache,sizeof(*tp) + cplen + cslen + cpadd);
   if (tp == NULL)
@@ -1546,7 +1547,7 @@
 
   cp = (char*)(tp+1) + cplen + cslen;
   CopyStr(&tp->Name,&cp);
-  CopyStr(&tp->CodecID,&cp);
+  CopyStr(&tp->AVCodecID,&cp);
 
   // set default language
   if (!tp->Language[0])
diff -ru ffms-2.17-src/src/core/matroskaparser.h ffms-2.17-src.orig/src/core/matroskaparser.h
--- ffms-2.17-src/src/core/matroskaparser.h	2010-11-28 03:28:40.000000000 +0100
+++ ffms-2.17-src.orig/src/core/matroskaparser.h	2012-10-02 21:43:16.000000000 +0200
@@ -161,7 +161,7 @@
   /* various strings */
   char			*Name;
   char			Language[4];
-  char			*CodecID;
+  char			*AVCodecID;
 };
 
 typedef struct TrackInfo  TrackInfo;