diff -up alsaconf/alsaconf.8.orig alsaconf/alsaconf.8
--- alsaconf/alsaconf.8.orig	2014-06-13 14:21:05.000000000 -0300
+++ alsaconf/alsaconf.8	2014-06-17 22:09:18.996611660 -0300
@@ -16,8 +16,6 @@ alsaconf \- configuration tool for the A
 This manual page documents briefly the
 .B alsaconf
 command.
-This manual page was written for the Debian distribution because the
-original program does not have a manual page.
 .PP
 \fBAlsaconf\fP is a simple shell script which tries to detect the sound cards
 on your system and writes a suitable configuration file for ALSA. It will try
@@ -74,13 +72,6 @@ Use the specified wav file as a test sou
 Set the uid for the ALSA devices (default = 0).
 This option is obsolete in the recent ALSA version.
 
-.SH DEBIAN SPECIFIC
-In Debian, the default gid of the device files is 29 (corresponding to the
-audio group) and the default device mode is 0660.
-
-For the ALSA base package, see also
-.I /usr/share/doc/alsa\-base/
-
 .SH SEE ALSO
 \fB
 alsamixer(1),
@@ -98,5 +89,4 @@ Takashi Iwai <tiwai@suse.de>,
 Bernd Kaindl <bk@suse.de> and
 Jan Ondrej (SAL) <ondrejj@salstar.sk>
 
-This manual page was written by Jordi Mallach <jordi@debian.org>,
-for the Debian system (but may be used by others).
+This manual page was written by Jordi Mallach <jordi@debian.org>.
diff -up alsaconf/alsaconf.in.orig alsaconf/alsaconf.in
--- alsaconf/alsaconf.in.orig	2014-06-13 14:21:05.000000000 -0300
+++ alsaconf/alsaconf.in	2014-06-17 22:14:34.611454874 -0300
@@ -50,7 +50,9 @@ xmsg() {
 }
 
 # Check for GNU/Linux distributions
-if [ -f /etc/SuSE-release ]; then
+if [ -f /etc/chinforinfula-version -o -f /etc/slackware-version ]; then
+  distribution="slackware"
+elif [ -f /etc/SuSE-release ]; then
   distribution="suse"
   suse_version=$(grep 'VERSION = ' /etc/SuSE-release | sed -e s/'VERSION = '//)
 elif [ -f /etc/UnitedLinux-release ]; then
@@ -65,8 +67,6 @@ elif test -f /etc/redhat-release && grep
   distribution="redhat"
 elif test -f /etc/fedora-release && grep -q "Fedora" /etc/fedora-release; then
   distribution="fedora"
-elif [ -f /etc/slackware-version -o -f /etc/slamd64-version ]; then
-  distribution="slackware"
 else
   distribution="unknown"
 fi
@@ -308,7 +308,7 @@ if [ "$kernel" = "new" ]; then
     if [ ! -f $cfgout ]; then
 	case "$(modprobe -V)" in
 	"module-init-tools version "3.[789]*|\
-	"module-init-tools version "3.1[0-9]*)
+	"module-init-tools version "|3.[1-9][0-9]*)
 	  ;;
 	"kmod version "*)
 	  ;;
@@ -321,6 +321,8 @@ if [ "$kernel" = "new" ]; then
   fi
 elif [ "$distribution" = "debian" ]; then
   cfgfile="/etc/modutils/sound"
+if [ "$distribution" = "slackware" ]; then
+  cfgfile="/etc/modprobe.d/alsa.conf"
 elif [ -e /etc/modules.conf ]; then
   cfgfile="/etc/modules.conf"
 elif [ -e /etc/conf.modules ]; then
