diff -up Python-3.2b2/Lib/test/test_sys.py.fix-test-sys-COUNT_ALLOCS Python-3.2b2/Lib/test/test_sys.py
--- Python-3.2b2/Lib/test/test_sys.py.fix-test-sys-COUNT_ALLOCS	2010-12-28 20:52:43.236310184 -0500
+++ Python-3.2b2/Lib/test/test_sys.py	2010-12-28 20:58:09.164184094 -0500
@@ -817,6 +817,9 @@ class SizeofTest(unittest.TestCase):
         # (PyTypeObject + PyNumberMethods + PyMappingMethods +
         #  PySequenceMethods + PyBufferProcs)
         s = size(vh + 'P2P15Pl4PP9PP11PI') + size('16Pi17P 3P 10P 2P 2P')
+        # COUNT_ALLOCS adds a further 3 Py_ssize_t and 2 pointers:
+        if hasattr(sys, 'getcounts'):
+            s += size('3P2P')
         check(int, s)
         # class
         class newstyleclass(object): pass
