diff -up Python-3.2b2/Lib/test/test_subprocess.py.non-readable-path Python-3.2b2/Lib/test/test_subprocess.py
--- Python-3.2b2/Lib/test/test_subprocess.py.non-readable-path	2010-12-29 16:25:38.498184175 -0500
+++ Python-3.2b2/Lib/test/test_subprocess.py	2010-12-29 16:25:51.094184539 -0500
@@ -578,7 +578,7 @@ class ProcessTestCase(BaseTestCase):
         for i in range(1024):
             # Windows raises IOError.  Others raise OSError.
             with self.assertRaises(EnvironmentError) as c:
-                subprocess.Popen(['nonexisting_i_hope'],
+                subprocess.Popen(['/usr/bin/nonexisting_i_hope'],
                                  stdout=subprocess.PIPE,
                                  stderr=subprocess.PIPE)
             if c.exception.errno != errno.ENOENT:  # ignore "no such file"
