diff -up openbox-3.4.11/Makefile.in.xau openbox-3.4.11/Makefile.in
--- openbox-3.4.11/Makefile.in.xau	2010-02-09 15:10:36.000000000 +0100
+++ openbox-3.4.11/Makefile.in	2010-02-09 15:15:01.000000000 +0100
@@ -763,6 +763,7 @@ tools_gnome_panel_control_gnome_panel_co
 	$(X_CFLAGS)
 
 tools_gnome_panel_control_gnome_panel_control_LDADD = \
+	-lXau \
 	$(X_LIBS)
 
 tools_gnome_panel_control_gnome_panel_control_SOURCES = \
@@ -784,6 +785,7 @@ tools_gdm_control_gdm_control_CPPFLAGS =
 	$(GLIB_CFLAGS)
 
 tools_gdm_control_gdm_control_LDADD = \
+	-lXau \
 	$(X_LIBS) \
 	$(GLIB_LIBS)
 
