--- data/autostart/openbox-autostart.in.orig	2011-08-06 13:41:38.954752785 -0300
+++ data/autostart/openbox-autostart.in	2011-08-06 13:44:22.870800841 -0300
@@ -11,6 +11,42 @@ elif which xsetroot >/dev/null; then
 fi
 test -z $BG || $BG -solid "#303030"
 
+# Search for xdg dirs and set the first as default
+# Try gnome first, then lxde, then kde.
+if [ -z "${XDG_CONFIG_DIRS}" ]; then
+  if [ -d /etc/gnome/xdg/menus ] ;then
+    XDG_CONFIG_DIRS="/etc/gnome/xdg:/etc/xdg"
+    if [ -z "${XDG_MENU_PREFIX}" ] ; then
+      if [ -f /etc/gnome/xdg/menus/gnome-applications.menu ] ;then
+        XDG_MENU_PREFIX="gnome-"
+      fi
+    fi
+  elif [ -d /etc/xdg/menus ] ;then
+    XDG_CONFIG_DIRS="/etc/xdg"
+    if [ -z "${XDG_MENU_PREFIX}" ] ; then
+      if [ -f /etc/xdg/menus/lxde-applications.menu ] ;then
+        XDG_MENU_PREFIX="lxde-"
+      fi
+    fi
+  elif [ -d /etc/kde/xdg/menus ] ;then
+    XDG_CONFIG_DIRS="/etc/kde/xdg:/etc/xdg"
+    if [ -z "${XDG_MENU_PREFIX}" ] ; then
+      if [ -f /etc/kde/xdg/menus/kde3-applications.menu ] ;then
+        XDG_MENU_PREFIX="kde3-"
+      elif [ -f /etc/kde/xdg/menus/kde4-applications.menu ] ;then
+        XDG_MENU_PREFIX="kde4-"
+      fi
+    fi
+  fi
+fi
+
+if [ -n "${XDG_CONFIG_DIRS}" ] ; then
+  export XDG_CONFIG_DIRS
+fi
+if [ -n "${XDG_MENU_PREFIX}" ] ; then
+  export XDG_MENU_PREFIX
+fi
+
 GLOBALAUTOSTART="@rcdir@/autostart"
 AUTOSTART="${XDG_CONFIG_HOME:-"$HOME/.config"}/openbox/autostart"
 
