diff -up kdeedu-4.5.2/marble/src/plugins/positionprovider/gpsd/GpsdPositionProviderPlugin.cpp.GPSD_API kdeedu-4.5.2/marble/src/plugins/positionprovider/gpsd/GpsdPositionProviderPlugin.cpp
--- kdeedu-4.5.2/marble/src/plugins/positionprovider/gpsd/GpsdPositionProviderPlugin.cpp.GPSD_API	2010-09-30 16:21:44.000000000 -0500
+++ kdeedu-4.5.2/marble/src/plugins/positionprovider/gpsd/GpsdPositionProviderPlugin.cpp	2010-10-11 12:36:44.189643125 -0500
@@ -64,7 +64,7 @@ void GpsdPositionProviderPlugin::update(
         m_position.set( data.fix.longitude, data.fix.latitude,
                         data.fix.altitude, GeoDataCoordinates::Degree );
         m_accuracy.level = GeoDataAccuracy::Detailed;
-#if defined( GPSD_API_MAJOR_VERSION ) && ( GPSD_API_MAJOR_VERSION >= 3 )
+#if defined( GPSD_API_MAJOR_VERSION ) && ( GPSD_API_MAJOR_VERSION >= 4 )
          if ( !isnan( data.fix.epx ) && !isnan( data.fix.epy ) ) {
              m_accuracy.horizontal = qMax( data.fix.epx, data.fix.epy );
          }
