diff -up QScintilla-gpl-2.7.1/designer-Qt4Qt5/designer.pro.qt4qt5_designer_incpath QScintilla-gpl-2.7.1/designer-Qt4Qt5/designer.pro
--- QScintilla-gpl-2.7.1/designer-Qt4Qt5/designer.pro.qt4qt5_designer_incpath	2012-12-21 11:30:31.000000000 -0600
+++ QScintilla-gpl-2.7.1/designer-Qt4Qt5/designer.pro	2013-03-03 18:21:11.466370720 -0600
@@ -22,4 +22,6 @@ SOURCES = qscintillaplugin.cpp
 target.path = $$[QT_INSTALL_PLUGINS]/designer
 INSTALLS += target
 
-LIBS += -L$$[QT_INSTALL_LIBS] -lqscintilla2
+LIBS += -L$(DESTDIR)../Qt4Qt5 -L$$[QT_INSTALL_LIBS] -lqscintilla2
+
+INCPATH = $(DESTDIR)../Qt4Qt5
