diff -up QScintilla-gpl-2.8.2/Python/configure.py.api_path QScintilla-gpl-2.8.2/Python/configure.py
--- QScintilla-gpl-2.8.2/Python/configure.py.api_path	2014-05-26 10:42:16.000000000 -0300
+++ QScintilla-gpl-2.8.2/Python/configure.py	2014-05-28 20:10:30.704881003 -0300
@@ -907,7 +907,7 @@ class _TargetConfiguration:
                 self.qmake_spec = 'macx-g++'
 
         self.qt_version_str = getattr(qt_config, 'QT_VERSION', '')
-        self.api_dir = qt_config.QT_INSTALL_DATA
+        self.api_dir = os.path.join(qt_config.QT_INSTALL_DATA, 'qsci')
         self.qt_inc_dir = qt_config.QT_INSTALL_HEADERS
         self.qt_lib_dir = qt_config.QT_INSTALL_LIBS
 
