diff -up QScintilla-gpl-2.8.1/Python/configure.py.api_path QScintilla-gpl-2.8.1/Python/configure.py
--- QScintilla-gpl-2.8.1/Python/configure.py.api_path	2014-03-16 14:03:35.875522817 -0500
+++ QScintilla-gpl-2.8.1/Python/configure.py	2014-03-16 14:04:32.008925903 -0500
@@ -314,7 +314,7 @@ class TargetConfiguration:
                 # Qt v4.
                 self.qmake_spec = 'macx-g++'
 
-        self.api_dir = qt_config.QT_INSTALL_DATA
+        self.api_dir = os.path.join(qt_config.QT_INSTALL_DATA, 'qsci')
         self.qsci_default_inc_dir = qt_config.QT_INSTALL_HEADERS
         self.qsci_inc_dir = None
         self.qsci_default_lib_dir = qt_config.QT_INSTALL_LIBS
