diff -up oyranos-0.1.10/makefile.in.NVCtrl oyranos-0.1.10/makefile.in
--- oyranos-0.1.10/makefile.in.NVCtrl	2010-08-22 17:03:50.000000000 +0200
+++ oyranos-0.1.10/makefile.in	2010-08-22 17:12:18.000000000 +0200
@@ -518,13 +518,13 @@ $(LIB_XNVCTRL).a:
 	test -L $(LIB_XNVCTRL).a || $(LNK) $(LIB_XNVCTRL)/$(LIB_XNVCTRL).a .
 
 # the twinview library
-$(TARGET)-monitor-nvidia:	$(SRCDIR)/$(TARGET)_monitor_nvidia.c $(LIB_XNVCTRL_NAME) $(MONI_OBJECTS)
+$(TARGET)-monitor-nvidia:	$(SRCDIR)/$(TARGET)_monitor_nvidia.c $(MONI_OBJECTS)
 	echo Linking $@ $(LIB_XNVCTRL_)1 $(XNVCtrl)2 $(LIB_XNVCTRL)3 $(LIB_XNVCTRL_NAME)4 ...
 	-$(CC) -I./ $(CFLAGS) $(INCL) $(SRCDIR)/$(TARGET)_monitor_nvidia.c \
-	-I$(SRCDIR)/libXNVCtrl \
+	-I/usr/include/NVCtrl \
 	-o $(TARGET)-monitor-nvidia $(MONI_OBJECTS) -L./ \
 	$(LDLIBS) $(X11_LIBS) $(m) \
-	-L./$(LIB_XNVCTRL) -lXNVCtrl $(LIBSONAMEFULL) \
+	-lXNVCtrl $(LIBSONAMEFULL) \
 	$(LINK_LIB_PATH)
 
 $(TARGET)_config_fltk.o:	$(SRCDIR)/$(TARGET)_config_fltk.cxx
