diff -up oyranos-0.2.0/makefile.in.NVCtrl oyranos-0.2.0/makefile.in
--- oyranos-0.2.0/makefile.in.NVCtrl	2010-12-17 19:39:07.000000000 +0100
+++ oyranos-0.2.0/makefile.in	2011-01-07 11:24:35.477805004 +0100
@@ -662,13 +662,13 @@ $(LIB_XNVCTRL).a:
 	test -L $(LIB_XNVCTRL).a || $(LNK) $(LIB_XNVCTRL)/$(LIB_XNVCTRL).a .
 
 # the twinview library
-$(TARGET)-monitor-nvidia:	$(SRCDIR)/$(TARGET)_monitor_nvidia.c $(LIB_XNVCTRL_NAME) $(MONI_X11_OBJECTS)
+$(TARGET)-monitor-nvidia:	$(SRCDIR)/$(TARGET)_monitor_nvidia.c $(MONI_X11_OBJECTS)
 	echo Linking $@ $(LIB_XNVCTRL_)1 $(XNVCtrl)2 $(LIB_XNVCTRL)3 $(LIB_XNVCTRL_NAME)4 ...
 	-$(CC) -I./ $(CFLAGS) $(INCL) $(SRCDIR)/$(TARGET)_monitor_nvidia.c \
-	-I$(SRCDIR)/libXNVCtrl \
+	-I/usr/include/NVCtrl \
 	-o $(TARGET)-monitor-nvidia $(MONI_X11_OBJECTS) -L./ \
 	$(LDLIBS) $(X11_LIBS) $(m) \
-	-L./$(LIB_XNVCTRL) -lXNVCtrl $(LIBSONAMEFULL) \
+	-lXNVCtrl $(LIBSONAMEFULL) \
 	$(LINK_LIB_PATH)
 
 $(TARGET)_config_fltk.o:	$(SRCDIR)/$(TARGET)_config_fltk.cxx
