diff -up oyranos-0.3.1/makefile.in.NVCtrl oyranos-0.3.1/makefile.in
--- oyranos-0.3.1/makefile.in.NVCtrl	2011-07-26 16:33:31.989774008 +0200
+++ oyranos-0.3.1/makefile.in	2011-07-26 16:34:23.565774007 +0200
@@ -756,13 +756,13 @@ $(LIB_XNVCTRL).a:
 	test -L $(LIB_XNVCTRL).a || $(LNK) $(LIB_XNVCTRL)/$(LIB_XNVCTRL).a .
 
 # the twinview library
-$(TARGET)-monitor-nvidia:	$(SRCDIR)/$(TARGET)_monitor_nvidia.c $(LIB_XNVCTRL_NAME) $(MONI_X11_OBJECTS) $(LIBSONAMEFULL)
+$(TARGET)-monitor-nvidia:	$(SRCDIR)/$(TARGET)_monitor_nvidia.c $(MONI_X11_OBJECTS)
 	echo Linking $@ $(LIB_XNVCTRL_)1 $(XNVCtrl)2 $(LIB_XNVCTRL)3 $(LIB_XNVCTRL_NAME)4 ...
 	-$(CC) -I./ $(CFLAGS) $(INCL) $(SRCDIR)/$(TARGET)_monitor_nvidia.c \
-	-I$(SRCDIR)/libXNVCtrl \
+	-I/usr/include/NVCtrl \
 	-o $(TARGET)-monitor-nvidia $(MONI_X11_OBJECTS) -L./ \
 	$(LDLIBS) \
-	-L./$(LIB_XNVCTRL) -lXNVCtrl $(MODULE_LDLIBS) \
+	-lXNVCtrl $(MODULE_LDLIBS) \
 	$(LINK_LIB_PATH) $(X11_LIBS) $(m)
 
 $(TARGET)_config_fltk.o:	$(SRCDIR)/$(TARGET)_config_fltk.cxx
