--- SConstruct.orig	2015-04-27 20:48:51.368872700 -0300
+++ SConstruct	2015-04-27 20:49:08.349273222 -0300
@@ -517,9 +517,6 @@ env.Alias(target = 'bin-package', source
 #----------------#
 Export(['env', 'version', 'src_dir', 'bin_dir'])
 if 'install' in COMMAND_LINE_TARGETS:
-    if not os.path.isfile('data/SConscript'):
-        raise 'VDrift data not found. Please make sure data is placed in vdrift directory. See README.md and http://wiki.vdrift.net.' 
-    SConscript('data/SConscript')
     # desktop appdata installation
     install_desktop = env.Install(env['destdir'] + env['prefix'] + '/share/applications', 'vdrift.desktop')
     install_appdata = env.Install(env['destdir'] + env['prefix'] + '/share/appdata', 'vdrift.appdata.xml')
