diff -urN w3m-0.5.3.old/etc.c w3m-0.5.3/etc.c
--- w3m-0.5.3.old/etc.c	2011-01-04 14:52:20.000000000 +0530
+++ w3m-0.5.3/etc.c	2013-12-04 14:40:19.251408025 +0530
@@ -634,24 +634,6 @@
 }
 #endif				/* not HAVE_STRERROR */
 
-#ifndef HAVE_SYS_ERRLIST
-char **sys_errlist;
-
-prepare_sys_errlist()
-{
-    int i, n;
-
-    i = 1;
-    while (strerror(i) != NULL)
-	i++;
-    n = i;
-    sys_errlist = New_N(char *, n);
-    sys_errlist[0] = "";
-    for (i = 1; i < n; i++)
-	sys_errlist[i] = strerror(i);
-}
-#endif				/* not HAVE_SYS_ERRLIST */
-
 int
 next_status(char c, int *status)
 {
diff -urN w3m-0.5.3.old/main.c w3m-0.5.3/main.c
--- w3m-0.5.3.old/main.c	2013-12-04 14:39:30.746582875 +0530
+++ w3m-0.5.3/main.c	2013-12-04 14:40:02.597468050 +0530
@@ -406,9 +406,6 @@
     textdomain(PACKAGE);
 #endif
 
-#ifndef HAVE_SYS_ERRLIST
-    prepare_sys_errlist();
-#endif				/* not HAVE_SYS_ERRLIST */
 
     NO_proxy_domains = newTextList();
     fileToDelete = newTextList();
