diff -urN w3m-0.5.3.old/w3mbookmark.c w3m-0.5.3/w3mbookmark.c
--- w3m-0.5.3.old/w3mbookmark.c	2011-01-04 14:52:24.000000000 +0530
+++ w3m-0.5.3/w3mbookmark.c	2013-12-03 21:34:33.907332176 +0530
@@ -99,7 +99,7 @@
 	fprintf(f, "<body>\n<h1>Bookmarks</h1>\n");
 	fprintf(f, "<h2>%s</h2>\n<ul>\n", section);
 	fprintf(f, "<li><a href=\"%s\">%s</a>\n", url, title);
-	fprintf(f, end_section);
+	fprintf(f, "%s", end_section);
 	fprintf(f, "</ul>\n</body>\n</html>\n");
 	fclose(f);
     }
