--- w3m-0.5.3/acinclude.m4.orig	2011-01-04 07:22:18.000000000 -0200
+++ w3m-0.5.3/acinclude.m4	2011-01-15 11:47:08.392625823 -0200
@@ -382,7 +382,7 @@ AC_DEFUN([AC_W3M_EXTLIBS],
 [lib=$1
  AC_MSG_CHECKING(for -l$lib)
  extlib="not found"
- for dir in /lib /usr/lib /usr/local/lib /usr/ucblib /usr/ccslib /usr/ccs/lib /lib64 /usr/lib64
+ for dir in /lib64 /lib /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib /usr/ucblib /usr/ccslib /usr/ccs/lib
  do
    if test -f $dir/lib$lib.a -o -f $dir/lib$lib.so ; then 
     LIBS="$LIBS -l$lib"
