commit a81dd2adf20e1b398f930545b32e3140fa1ac19b
Author: Julian Ospald <hasufell@gentoo.org>
Date:   Tue Aug 28 10:37:22 2012 +0200

    CMake: fix underlinking

diff --git a/simgear/CMakeLists.txt b/simgear/CMakeLists.txt
index 46a22b6..014b79d 100644
--- a/simgear/CMakeLists.txt
+++ b/simgear/CMakeLists.txt
@@ -54,7 +54,7 @@ if(SIMGEAR_SHARED)
     set_property(TARGET SimGearCore PROPERTY VERSION   ${SIMGEAR_VERSION})
     set_property(TARGET SimGearCore PROPERTY SOVERSION ${SIMGEAR_SOVERSION})
 
-    target_link_libraries(SimGearCore ${ZLIB_LIBRARY} ${RT_LIBRARY})
+    target_link_libraries(SimGearCore ${ZLIB_LIBRARY} ${RT_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
     install(TARGETS SimGearCore LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
     if(LIBSVN_FOUND)
