--- src/dotnew.orig	2010-08-21 16:45:39.000000000 -0300
+++ src/dotnew	2012-06-04 17:38:09.641596823 -0300
@@ -7,14 +7,18 @@
 # Licensed under the GPL v3
 
 # Translations only work with utf8 locales
-if [ ! `echo $LANG | grep utf8` ]; then
+if [ ! `echo $LANG | grep -i \\.utf8` ] && [ ! `echo $LANG | grep -i \\.utf-8` ]; then
 	LANG=C
 fi
 
 # For gettext internationalization
 export TEXTDOMAIN="dotnew"
 export TEXTDOMAINDIR="/usr/share/locale"
-. gettext.sh
+if [ -x /usr/bin/gettext.sh ] ;then
+  . /usr/bin/gettext.sh
+else
+  echo "Missing /usr/bin/gettext.sh"
+fi
 
 if [ "$UID" != "0" ]; then
 	echo "`eval_gettext 'You need to be root to run this'`"
