diff -up gst-plugins-bad-0.10.19/configure.ac.wildmidi gst-plugins-bad-0.10.19/configure.ac
--- gst-plugins-bad-0.10.19/configure.ac.wildmidi	2010-07-15 17:18:09.457155838 -0300
+++ gst-plugins-bad-0.10.19/configure.ac	2010-07-15 17:18:06.000148270 -0300
@@ -1250,7 +1250,12 @@ AG_GST_CHECK_FEATURE(TIMIDITY, [timidity
 dnl *** wildmidi ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_WILDMIDI, true)
 AG_GST_CHECK_FEATURE(WILDMIDI, [wildmidi midi soft synth plugin], wildmidi, [
-  AC_CHECK_HEADER([wildmidi_lib.h],HAVE_WILDMIDI=yes, HAVE_WILDMIDI=no)
+  AG_GST_CHECK_LIBHEADER(WILDMIDI, WildMidi,
+                         WildMidi_Init, ,
+                         wildmidi_lib.h,
+                         WILDMIDI_LIBS="-lWildMidi")
+  AC_CHECK_LIB(WildMidi, WildMidi_LoadSamples, AC_DEFINE(HAVE_WILDMIDI_0_2_2, [1], [Have WildMidi 0.2.2 or earlier library]))
+
   if test "x$HAVE_WILDMIDI" = "xyes"; then
 	 WILDMIDI_CFLAGS=
 	 WILDMIDI_LIBS=-lWildMidi
diff -up gst-plugins-bad-0.10.19/ext/timidity/gstwildmidi.c.wildmidi gst-plugins-bad-0.10.19/ext/timidity/gstwildmidi.c
--- gst-plugins-bad-0.10.19/ext/timidity/gstwildmidi.c.wildmidi	2010-05-13 21:48:27.000000000 -0300
+++ gst-plugins-bad-0.10.19/ext/timidity/gstwildmidi.c	2010-07-15 17:18:06.001146937 -0300
@@ -69,12 +69,15 @@ enum
 
 enum
 {
-  ARG_0,
-  ARG_LINEAR_VOLUME,
-  ARG_HIGH_QUALITY,
+  PROP_0,
+  PROP_LINEAR_VOLUME,
+  PROP_HIGH_QUALITY,
   /* FILL ME */
 };
 
+#define DEFAULT_LINEAR_VOLUME    TRUE
+#define DEFAULT_HIGH_QUALITY     TRUE
+
 static void gst_wildmidi_finalize (GObject * object);
 
 static gboolean gst_wildmidi_sink_event (GstPad * pad, GstEvent * event);
@@ -235,13 +238,15 @@ gst_wildmidi_class_init (GstWildmidiClas
   gobject_class->set_property = gst_wildmidi_set_property;
   gobject_class->get_property = gst_wildmidi_get_property;
 
-  g_object_class_install_property (gobject_class, ARG_LINEAR_VOLUME,
+  g_object_class_install_property (gobject_class, PROP_LINEAR_VOLUME,
       g_param_spec_boolean ("linear-volume", "Linear volume",
-          "Linear volume", TRUE, G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
+          "Linear volume", DEFAULT_LINEAR_VOLUME,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 
-  g_object_class_install_property (gobject_class, ARG_HIGH_QUALITY,
+  g_object_class_install_property (gobject_class, PROP_HIGH_QUALITY,
       g_param_spec_boolean ("high-quality", "High Quality",
-          "High Quality", TRUE, G_PARAM_READWRITE | G_PARAM_CONSTRUCT));
+          "High Quality", DEFAULT_HIGH_QUALITY,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 
   gstelement_class->change_state = gst_wildmidi_change_state;
 }
@@ -282,6 +287,9 @@ gst_wildmidi_init (GstWildmidi * filter,
   filter->adapter = gst_adapter_new ();
 
   filter->bytes_per_frame = WILDMIDI_BPS;
+
+  filter->high_quality = DEFAULT_HIGH_QUALITY;
+  filter->linear_volume = DEFAULT_LINEAR_VOLUME;
 }
 
 static void
@@ -489,11 +497,16 @@ gst_wildmidi_do_seek (GstWildmidi * wild
   sample = segment->last_stop;
 
   GST_OBJECT_LOCK (wildmidi);
+#ifdef HAVE_WILDMIDI_0_2_2
   if (accurate) {
     WildMidi_SampledSeek (wildmidi->song, &sample);
   } else {
     WildMidi_FastSeek (wildmidi->song, &sample);
   }
+#else
+  WildMidi_FastSeek (wildmidi->song, &sample);
+#endif
+
   GST_OBJECT_UNLOCK (wildmidi);
 
   segment->start = segment->time = segment->last_stop = sample;
@@ -669,12 +682,21 @@ gst_wildmidi_parse_song (GstWildmidi * w
   if (!wildmidi->song)
     goto open_failed;
 
+#ifdef HAVE_WILDMIDI_0_2_2
   WildMidi_LoadSamples (wildmidi->song);
+#endif
 
+#ifdef HAVE_WILDMIDI_0_2_2
   WildMidi_SetOption (wildmidi->song, WM_MO_LINEAR_VOLUME,
       wildmidi->linear_volume);
   WildMidi_SetOption (wildmidi->song, WM_MO_EXPENSIVE_INTERPOLATION,
       wildmidi->high_quality);
+#else
+  WildMidi_SetOption (wildmidi->song, WM_MO_LOG_VOLUME,
+      !wildmidi->linear_volume);
+  WildMidi_SetOption (wildmidi->song, WM_MO_ENHANCED_RESAMPLING,
+      wildmidi->high_quality);
+#endif
 
   info = WildMidi_GetInfo (wildmidi->song);
   GST_OBJECT_UNLOCK (wildmidi);
@@ -895,20 +917,30 @@ gst_wildmidi_set_property (GObject * obj
   wildmidi = GST_WILDMIDI (object);
 
   switch (prop_id) {
-    case ARG_LINEAR_VOLUME:
+    case PROP_LINEAR_VOLUME:
       GST_OBJECT_LOCK (object);
       wildmidi->linear_volume = g_value_get_boolean (value);
       if (wildmidi->song)
+#ifdef HAVE_WILDMIDI_0_2_2
         WildMidi_SetOption (wildmidi->song, WM_MO_LINEAR_VOLUME,
             wildmidi->linear_volume);
+#else
+        WildMidi_SetOption (wildmidi->song, WM_MO_LOG_VOLUME,
+            !wildmidi->linear_volume);
+#endif
       GST_OBJECT_UNLOCK (object);
       break;
-    case ARG_HIGH_QUALITY:
+    case PROP_HIGH_QUALITY:
       GST_OBJECT_LOCK (object);
       wildmidi->high_quality = g_value_get_boolean (value);
       if (wildmidi->song)
+#ifdef HAVE_WILDMIDI_0_2_2
         WildMidi_SetOption (wildmidi->song, WM_MO_EXPENSIVE_INTERPOLATION,
             wildmidi->high_quality);
+#else
+        WildMidi_SetOption (wildmidi->song, WM_MO_ENHANCED_RESAMPLING,
+            wildmidi->high_quality);
+#endif
       GST_OBJECT_UNLOCK (object);
       break;
     default:
@@ -928,12 +960,12 @@ gst_wildmidi_get_property (GObject * obj
   wildmidi = GST_WILDMIDI (object);
 
   switch (prop_id) {
-    case ARG_LINEAR_VOLUME:
+    case PROP_LINEAR_VOLUME:
       GST_OBJECT_LOCK (object);
       g_value_set_boolean (value, wildmidi->linear_volume);
       GST_OBJECT_UNLOCK (object);
       break;
-    case ARG_HIGH_QUALITY:
+    case PROP_HIGH_QUALITY:
       GST_OBJECT_LOCK (object);
       g_value_set_boolean (value, wildmidi->high_quality);
       GST_OBJECT_UNLOCK (object);
