--- ext/celt/gstceltenc.c.celt	2011-02-07 19:29:14.139389213 -0200
+++ ext/celt/gstceltenc.c	2011-02-07 19:30:07.067714038 -0200
@@ -630,7 +630,7 @@ gst_celt_enc_setup (GstCeltEnc * enc)
     goto mode_initialization_failed;
 
 #ifdef HAVE_CELT_0_7
-  celt_header_init (&enc->header, enc->mode, enc->channels);
+  celt_header_init (&enc->header, enc->mode, enc->frame_size, enc->channels);
 #else
   celt_header_init (&enc->header, enc->mode);
 #endif
