diff -Nur xfwm4-4.9.0.orig/settings-dialogs/xfce-wm-settings.desktop.in xfwm4-4.9.0/settings-dialogs/xfce-wm-settings.desktop.in
--- xfwm4-4.9.0.orig/settings-dialogs/xfce-wm-settings.desktop.in	2012-03-30 12:43:56.000000000 -0600
+++ xfwm4-4.9.0/settings-dialogs/xfce-wm-settings.desktop.in	2012-04-02 20:48:57.876282606 -0600
@@ -6,7 +6,7 @@
 Icon=xfwm4
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfwm4-4.9.0.orig/settings-dialogs/xfce-wmtweaks-settings.desktop.in xfwm4-4.9.0/settings-dialogs/xfce-wmtweaks-settings.desktop.in
--- xfwm4-4.9.0.orig/settings-dialogs/xfce-wmtweaks-settings.desktop.in	2012-03-30 12:43:56.000000000 -0600
+++ xfwm4-4.9.0/settings-dialogs/xfce-wmtweaks-settings.desktop.in	2012-04-02 20:49:03.919310767 -0600
@@ -6,7 +6,7 @@
 Icon=wmtweaks
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
diff -Nur xfwm4-4.9.0.orig/settings-dialogs/xfce-workspaces-settings.desktop.in xfwm4-4.9.0/settings-dialogs/xfce-workspaces-settings.desktop.in
--- xfwm4-4.9.0.orig/settings-dialogs/xfce-workspaces-settings.desktop.in	2012-03-30 12:43:56.000000000 -0600
+++ xfwm4-4.9.0/settings-dialogs/xfce-workspaces-settings.desktop.in	2012-04-02 20:49:11.651346794 -0600
@@ -6,7 +6,7 @@
 Icon=xfce4-workspaces
 Terminal=false
 Type=Application
-Categories=XFCE;GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
+Categories=GTK;Settings;DesktopSettings;X-XFCE-SettingsDialog;X-XFCE-PersonalSettings;
 StartupNotify=true
 OnlyShowIn=XFCE;
 X-XfcePluggable=true
