diff -up jack-rack-1.4.7/configure.ac.ldflags jack-rack-1.4.7/configure.ac
--- jack-rack-1.4.7/configure.ac.ldflags	2010-02-14 16:17:12.619828790 +0100
+++ jack-rack-1.4.7/configure.ac	2010-02-14 16:18:32.838830238 +0100
@@ -7,6 +7,12 @@ AM_MAINTAINER_MODE
 AC_PROG_CC
 #AC_PROG_LIBTOOL
 
+### Check to see if dlopen is in default libraries (like Solaris, which
+### has it in libc), or if libdl is needed to get it.
+AC_CHECK_FUNC([dlopen], [],
+ AC_CHECK_LIB([dl], [dlopen], DLOPEN_LIBS="-ldl"))
+AC_SUBST([DLOPEN_LIBS])
+
 
 ############
 ### JACK ###
diff -up jack-rack-1.4.7/src/Makefile.am.ldflags jack-rack-1.4.7/src/Makefile.am
--- jack-rack-1.4.7/src/Makefile.am.ldflags	2010-02-14 15:59:21.166078833 +0100
+++ jack-rack-1.4.7/src/Makefile.am	2010-02-14 16:19:16.301829733 +0100
@@ -68,7 +68,9 @@ jack_rack_LDFLAGS = \
 	$(LASH_LIBS) \
 	$(LRDF_LIBS) \
 	$(ALSA_LIBS) \
-	$(XML_LIBS)
+	$(XML_LIBS) \
+	$(DLOPEN_LIBS) \
+	-lm
 
 splint-check:
 	splint  +systemdirs /usr/include:/usr/local/lib \
