--- qsynth-0.3.3.old/src/qsynthOptions.cpp	2008-07-01 10:04:41.000000000 -0400
+++ qsynth-0.3.3/src/qsynthOptions.cpp	2009-04-18 01:09:18.000000000 -0400
@@ -61,7 +61,7 @@ qsynthOptions::qsynthOptions (void)
 
 	// Load defaults...
 	m_settings.beginGroup("/Defaults");
-	sSoundFontDir  = m_settings.value("/SoundFontDir").toString();
+	sSoundFontDir  = m_settings.value("/SoundFontDir", "/usr/share/soundfonts").toString();
 	bPresetPreview = m_settings.value("/PresetPreview", false).toBool();
 	m_settings.endGroup();
 
@@ -492,7 +492,7 @@ void qsynthOptions::loadSetup ( qsynthSe
 #else
 	pSetup->sMidiDriver      = m_settings.value("/MidiDriver", "alsa_seq").toString();
     pSetup->sAudioDriver     = m_settings.value("/AudioDriver", "jack").toString();
-    pSetup->iAudioBufSize    = m_settings.value("/AudioBufSize", 64).toInt();
+    pSetup->iAudioBufSize    = m_settings.value("/AudioBufSize", 1024).toInt();
     pSetup->iAudioBufCount   = m_settings.value("/AudioBufCount", 2).toInt();
 #endif
     pSetup->sAlsaName        = m_settings.value("/AlsaName", "pid").toString();
@@ -529,7 +529,9 @@ void qsynthOptions::loadSetup ( qsynthSe
 	m_settings.beginGroup("/SoundFonts");
 	const QString sSoundFontPrefix  = "/SoundFont%1";
 	const QString sBankOffsetPrefix = "/BankOffset%1";
+	const QString defaultSoundfont = "/usr/share/soundfonts/default.sf2";
 	int i = 0;
+	int f = 0;
 	for (;;) {
 		++i;
 		QString sSoundFont  = m_settings.value(sSoundFontPrefix.arg(i)).toString();
@@ -538,6 +540,11 @@ void qsynthOptions::loadSetup ( qsynthSe
 			break;
 		pSetup->soundfonts.append(sSoundFont);
 		pSetup->bankoffsets.append(sBankOffset);
+		f++;
+	}
+	if (f == 0) {
+	  pSetup->soundfonts.append(defaultSoundfont);
+	  pSetup->bankoffsets.append("0");
 	}
 	m_settings.endGroup();
 
