diff -rupN qsynth-0.3.4.old/configure.ac qsynth-0.3.4/configure.ac
--- qsynth-0.3.4.old/configure.ac	2009-05-10 08:01:14.000000000 -0400
+++ qsynth-0.3.4/configure.ac	2010-02-13 12:58:05.000000000 -0500
@@ -150,7 +150,12 @@ AC_SUBST(ac_lrelease)
 
 # Checks for libraries.
 AC_CHECK_LIB(m, main)
-AC_CHECK_LIB(X11, main)
+AC_CHECK_LIB(X11, main, [ac_X11_lib="yes"], [ac_X11_lib="no"])
+if test "x$ac_X11_lib" = "xno"; then
+   AC_MSG_ERROR([libX11 library not found.])
+else
+   ac_libs="$ac_libs -lX11"
+fi
 AC_CHECK_LIB(Xext, main)
 
 # Check for round math function.
