diff -up ManiaDrive-1.2-src/mania_localtracks.php~ ManiaDrive-1.2-src/mania_localtracks.php
--- ManiaDrive-1.2-src/mania_localtracks.php~	2006-08-16 23:15:49.000000000 +0200
+++ ManiaDrive-1.2-src/mania_localtracks.php	2011-01-11 11:56:20.764096983 +0100
@@ -26,8 +26,7 @@ while( ($file = readdir($handle)) !== fa
 closedir($handle);
 }
 
-$data_dir=str_pad('',1000);
-raydium_file_home_path_cpy("data",$data_dir);
+$data_dir=raydium_file_home_path("data");
 
 search_here(".");
 search_here($data_dir);
diff -up ManiaDrive-1.2-src/mania_story.php~ ManiaDrive-1.2-src/mania_story.php
--- ManiaDrive-1.2-src/mania_story.php~	2006-07-06 21:50:47.000000000 +0200
+++ ManiaDrive-1.2-src/mania_story.php	2011-01-11 11:54:35.100096998 +0100
@@ -45,8 +45,7 @@ return $best;
 
 $story=file($story_file);
 
-$state_file=str_pad('',1000);
-raydium_file_home_path_cpy("mania_drive.state",$state_file);
+$state_file=raydium_file_home_path("mania_drive.state");
 if(file_exists($state_file))
     $scores=file($state_file);
 
diff -up ManiaDrive-1.2-src/raydium/network.h~ ManiaDrive-1.2-src/raydium/network.h
--- ManiaDrive-1.2-src/raydium/network.h~	2006-07-23 13:24:08.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/network.h	2011-01-11 11:54:15.149097004 +0100
@@ -29,7 +29,7 @@ signed char raydium_network_server_creat
 void raydium_random_randomize(void);
 signed char raydium_parser_db_get(char *key, char *value, char *def);
 void raydium_register_function(void *addr,char *name);
-void raydium_file_home_path_cpy(char *file, char *dest);
+char *raydium_file_home_path(char *name);
 void raydium_timecall_init(void);
 
 
@@ -43,7 +43,7 @@ exit(0);
 // do the minimal reg_api job (this should not be done like this ! :/)
 // part 1
 PHP_i_sss(raydium_parser_db_get);
-PHP_v_ss(raydium_file_home_path_cpy);
+PHP_s_s(raydium_file_home_path);
 #endif
 
 void raydium_network_only_init(int argc, char **argv)
@@ -59,7 +59,7 @@ raydium_php_init();
 // do the minimal reg_api job (this should not be done like this ! :/)
 // part 2
 raydium_register_function(C2PHP(raydium_parser_db_get),"raydium_parser_db_get");
-raydium_register_function(C2PHP(raydium_file_home_path_cpy),"raydium_file_home_path_cpy");
+raydium_register_function(C2PHP(raydium_file_home_path),"raydium_file_home_path");
         
 #endif
 raydium_network_init();
diff -up ManiaDrive-1.2-src/raydium/php_wrappers.c~ ManiaDrive-1.2-src/raydium/php_wrappers.c
--- ManiaDrive-1.2-src/raydium/php_wrappers.c~	2006-04-29 00:46:55.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/php_wrappers.c	2011-01-11 11:59:41.067097001 +0100
@@ -197,6 +197,17 @@ if (zend_parse_parameters(ZEND_NUM_ARGS(
 fname(a);\
 }
 
+// char* f(char *)
+#define PHP_s_s(fname)\
+ZEND_FUNCTION(fname)\
+{\
+char *a;\
+long s_len;\
+if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC,\
+  "s", &a, &s_len) == FAILURE)  return;\
+RETURN_STRING(fname(a),1);\
+}
+
 // int f(char *, char*, char *)
 #define PHP_i_sss(fname)\
 ZEND_FUNCTION(fname)\
diff -up ManiaDrive-1.2-src/raydium/reg_api.c~ ManiaDrive-1.2-src/raydium/reg_api.c
--- ManiaDrive-1.2-src/raydium/reg_api.c~	2006-08-11 15:26:53.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/reg_api.c	2011-01-11 11:53:25.277097231 +0100
@@ -19,7 +19,7 @@
 // Part 1: create PHP wrappers
 
 // file.c
-PHP_v_ss(raydium_file_home_path_cpy);
+PHP_s_s(raydium_file_home_path);
 
 // light.c
 PHP_v_v(raydium_light_enable)
@@ -191,7 +191,7 @@ if(done)
     }
 
 // file.c
-raydium_register_function(C2PHP(raydium_file_home_path_cpy),"raydium_file_home_path_cpy");
+raydium_register_function(C2PHP(raydium_file_home_path),"raydium_file_home_path");
 
 // light.c
 raydium_register_function(C2PHP(raydium_light_enable),"raydium_light_enable");
diff -up ManiaDrive-1.2-src/rayphp/libfile.php~ ManiaDrive-1.2-src/rayphp/libfile.php
--- ManiaDrive-1.2-src/rayphp/libfile.php~	2006-08-11 15:26:53.000000000 +0200
+++ ManiaDrive-1.2-src/rayphp/libfile.php	2011-01-11 11:55:02.572096869 +0100
@@ -98,8 +98,7 @@ function read_repositories_file($repos)
 global $raydium_php_rayphp_path;
 $repos_list=array();
     
-$tmp=str_pad("",256);
-raydium_file_home_path_cpy($repos,$tmp);
+$tmp=raydium_file_home_path($repos);
 
 if(file_exists($tmp))
     read_repositories_file_internal($tmp,$repos_list);
@@ -144,8 +143,7 @@ function valid_entry($r)
 // unzip a given input
 function gzdecode($in)
 {
-  $tmp=str_pad("",256);
-  raydium_file_home_path_cpy("tmp.tmp.gz",$tmp);
+  $tmp=raydium_file_home_path("tmp.tmp.gz");
   $fp=fopen($tmp,"wb");
   if(!$fp) return false;
   fwrite($fp,$in);
