--- data/menu.orig	2012-08-03 10:46:39.000000000 -0300
+++ data/menu	2012-08-07 18:31:19.356840698 -0300
@@ -6,74 +6,12 @@ INCLUDE = "vars"
 RootMenu = "Pekwm" {
 	Entry = "Terminal" { Actions = "Exec $TERM &" }
 	Entry = "Run.." { Actions = "ShowCmdDialog" }
+	Entry = "Web Browser" { Actions = "Exec htmlview &" }
 
 	Separator {}
 
-	Submenu = "Editors" {
-		Entry = "emacs" { Actions = "Exec emacs" }
-		Entry = "emacs terminal" { Actions = "Exec $TERM -title emacs -e emacs -nw" }
-		Entry = "gedit" { Actions = "Exec gedit" }
-		Entry = "gvim" { Actions = "Exec gvim" }
-		Entry = "kate" { Actions = "Exec kate" }
-		Entry = "vim" { Actions = "Exec $TERM -title vim -e vim" }
-	}
-	Submenu = "eMail" {
-		Entry = "claws-mail" { Actions = "Exec claws-mail" }
-		Entry = "evolution" { Actions = "Exec evolution" }
-		Entry = "kmail" { Actions = "Exec kmail" }
-		Entry = "mutt" { Actions = "Exec $TERM -e mutt" }
-		Entry = "thunderbird" { Actions = "Exec thunderbird" }
-	}
-	Submenu = "Filemanager" {
-		Entry = "dolphin" { Actions = "Exec dolphin" }
-		Entry = "mc" { Actions = "Exec $TERM -e mc" }
-		Entry = "nautilus" { Actions = "Exec nautilus" }
-		Entry = "pcmanfm" { Actions = "Exec pcmanfm" }
-		Entry = "rox" { Actions = "Exec rox" }
-		Entry = "thunar" { Actions = "Exec thunar" }
-	}
-	Submenu = "Graphics" {
-		Entry = "gimp" { Actions = "Exec gimp" }
-		Entry = "digikam" { Actions = "Exec digikam" }
-		Entry = "display" { Actions = "Exec display" }
-		Entry = "eog" { Actions = "Exec eog" }
-		Entry = "gwenview" { Actions = "Exec gwenview" }
-		Entry = "shotwell" { Actions = "Exec shotwell" }
-	}
-	Submenu = "IDE" {
-		Entry = "anjuta" { Actions = "Exec anjuta" }
-		Entry = "eclipse" { Actions = "Exec eclipse" }
-		Entry = "kdevelop" { Actions = "Exec kdevelop" }
-	}
-	Submenu = "IM" {
-		Entry = "empathy" { Actions = "Exec empathy" }
-		Entry = "kopete" { Actions = "Exec kopete" }
-		Entry = "pidgin" { Actions = "Exec pidgin" }
-		Entry = "psi" { Actions = "Exec psi" }
-	}
-	Submenu = "Multimedia" {
-		Entry = "vlc" { Actions = "Exec vlc" }
-		Entry = "amarok" { Actions = "Exec amarok" }
-		Entry = "banshee" { Actions = "Exec banshee" }
-		Entry = "rhythmbox" { Actions = "Exec rhythmbox" }
-		Entry = "smplayer" { Actions = "Exec smplayer" }
-		Entry = "alsamixer" { Actions = "Exec $TERM -title alsamixer -e alsamixer" }
-	}
-	Submenu = "Office" {
-		Entry = "LOffice - Calc" { Actions = "Exec libreoffice --calc" }
-		Entry = "LOffice - Draw" { Actions = "Exec libreoffice --draw" }
-		Entry = "LOffice - Impress" { Actions = "Exec libreoffice --impress" }
-		Entry = "LOffice - Writer" { Actions = "Exec libreoffice --writer" }
-	}
-	Submenu = "VoIP" {
-		Entry = "ekiga" { Actions = "Exec ekiga" }
-		Entry = "jitsi" { Actions = "Exec jitsi" }
-		Entry = "sflphone" { Actions = "Exec sflphone" }
-}
-	Submenu = "WWW" {
-		Entry = "firefox" { Actions = "Exec firefox" }
-		Entry = "chromium" { Actions = "Exec chromium" }
-		Entry = "opera" { Actions = "Exec opera" }
+	Submenu = "_SLK_DIST Menu" {
+		Entry { Actions = "Dynamic $_PEKWM_SCRIPT_PATH/pekwm-xdg-menu" }
 	}
 
 	Separator {}
