diff -dur xfce4-datetime-plugin-0.6.1.orig/panel-plugin/datetime.desktop.in.in xfce4-datetime-plugin-0.6.1/panel-plugin/datetime.desktop.in.in
--- xfce4-datetime-plugin-0.6.1.orig/panel-plugin/datetime.desktop.in.in	2008-04-21 06:21:45.000000000 +0200
+++ xfce4-datetime-plugin-0.6.1/panel-plugin/datetime.desktop.in.in	2010-12-19 00:50:46.951565622 +0100
@@ -1,8 +1,7 @@
 [Xfce Panel]
 Type=X-XFCE-PanelPlugin
-Encoding=UTF-8
 _Name=DateTime
 _Comment=Date and Time plugin with a simple calendar
 Icon=xfce-schedule
 X-XFCE-Module=datetime
-X-XFCE-Module-Path=@libdir@/xfce4/panel-plugins
+X-XFCE-Internal=FALSE
diff -dur xfce4-datetime-plugin-0.6.1.orig/panel-plugin/Makefile.in xfce4-datetime-plugin-0.6.1/panel-plugin/Makefile.in
--- xfce4-datetime-plugin-0.6.1.orig/panel-plugin/Makefile.in	2008-11-19 08:10:36.000000000 +0100
+++ xfce4-datetime-plugin-0.6.1/panel-plugin/Makefile.in	2010-12-19 00:42:14.401423201 +0100
@@ -252,7 +252,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-plugindir = $(libdir)/xfce4/panel-plugins/
+plugindir = $(libdir)/xfce4/panel/plugins/
 plugin_LTLIBRARIES = \
 	libdatetime.la
 
