diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2010-12-08 17:55:46.000000000 +0100
+++ b/CMakeLists.txt	2010-12-12 16:46:50.496000030 +0100
@@ -108,6 +108,7 @@
 SET(COMPILE_LIGHTSPARK TRUE CACHE BOOL "Compile Lightspark?")
 SET(COMPILE_TIGHTSPARK TRUE CACHE BOOL "Compile Tightspark?")
 SET(COMPILE_PLUGIN FALSE CACHE BOOL "Compile the browser plugin?")
+SET(AUDIO_BACKEND "pulse" CACHE STRING "Choose audio backend: none pulse alsa, default is pulse")
 SET(ENABLE_CURL TRUE CACHE BOOL "Enable CURL? (Required for Downloader functionality)")
 SET(ENABLE_LIBAVCODEC TRUE CACHE BOOL "Enable libavcodec and dependent functionality?")
 SET(PLUGIN_DIRECTORY "${LIBDIR}/mozilla/plugins" CACHE STRING "Directory to install Firefox plugin to")
@@ -371,7 +372,9 @@
 GETTEXT_CREATE_TRANSLATIONS(i18n/lightspark.pot ALL i18n/pl.po i18n/fr.po
         i18n/zh_CN.po)
 
-ADD_SUBDIRECTORY(backends/interfaces/audio/pulse)
+IF(${AUDIO_BACKEND} MATCHES "pulse")
+  ADD_SUBDIRECTORY(backends/interfaces/audio/pulse)
+ENDIF(${AUDIO_BACKEND} MATCHES "pulse")
 #ADD_SUBDIRECTORY(backends/interfaces/audio/alsa)
 #ADD_SUBDIRECTORY(backends/interfaces/audio/openal)
 
