--- src/libprojectM/CMakeLists.txt.lib64	2012-08-21 19:38:33.016902336 -0300
+++ src/libprojectM/CMakeLists.txt	2012-08-21 19:39:27.200398583 -0300
@@ -36,8 +36,7 @@ set(PROJECTM_ABI_LEVEL 2)
 ADD_DEFINITIONS(-DCMAKE_INSTALL_PREFIX="\\\"${CMAKE_INSTALL_PREFIX}\\\"")
 
 SET(LIB_INSTALL_DIR "\${exec_prefix}/lib"
-  CACHE PATH "The subdirectory relative to the install prefix where libraries will be installed (default is /lib)"
-  FORCE)
+  CACHE PATH "The subdirectory relative to the install prefix where libraries will be installed (default is /lib)")
 
 if (USE_NATIVE_GLEW)
 ADD_DEFINITIONS(-DUSE_NATIVE_GLEW)
@@ -234,7 +233,7 @@ ENDIF(projectM_FONT_MENU MATCHES "${CMAK
 FILE(GLOB presets "presets/*.milk" "presets/*.prjm" "presets/*.tga")
 INSTALL(FILES ${presets} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM/presets)
 INSTALL(FILES ${Renderer_SOURCE_DIR}/projectM.cg ${Renderer_SOURCE_DIR}/blur.cg DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM/shaders)
-INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libprojectM.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/pkgconfig)
+INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libprojectM.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/config.inp DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM)
 INSTALL(FILES projectM.hpp event.h dlldefs.h fatal.h PCM.hpp Common.hpp DESTINATION include/libprojectM)
-INSTALL(TARGETS projectM DESTINATION lib)
+INSTALL(TARGETS projectM DESTINATION lib${LIB_SUFFIX})
