diff -Nur libprojectM-2.0.1-orig/CMakeLists.txt libprojectM-2.0.1/CMakeLists.txt
--- libprojectM-2.0.1-orig/CMakeLists.txt	2009-12-06 02:42:51.000000000 +0100
+++ libprojectM-2.0.1/CMakeLists.txt	2012-06-01 12:47:47.280267225 +0200
@@ -190,8 +190,6 @@
 
 FILE(GLOB presets "presets/*.milk" "presets/*.prjm" "presets/*.tga")
 INSTALL(FILES ${presets} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM/presets)
-FILE(GLOB fonts "fonts/*.ttf")
-INSTALL(FILES ${fonts} DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM/fonts)
 INSTALL(FILES ${Renderer_SOURCE_DIR}/projectM.cg ${Renderer_SOURCE_DIR}/blur.cg DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM/shaders)
 INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/libprojectM.pc DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig)
 INSTALL(FILES ${CMAKE_CURRENT_SOURCE_DIR}/config.inp DESTINATION ${CMAKE_INSTALL_PREFIX}/share/projectM)
diff -Nur libprojectM-2.0.1-orig/config.inp.in libprojectM-2.0.1/config.inp.in
--- libprojectM-2.0.1-orig/config.inp.in	2009-12-06 02:42:51.000000000 +0100
+++ libprojectM-2.0.1/config.inp.in	2012-06-01 12:47:47.281267174 +0200
@@ -17,6 +17,6 @@
 Aspect Correction = true	# Custom Shape Aspect Correction
 
 Preset Path = @CMAKE_INSTALL_PREFIX@/@RESOURCE_PREFIX@/presets # preset location
-Title Font = @CMAKE_INSTALL_PREFIX@/@RESOURCE_PREFIX@/fonts/Vera.ttf
-Menu Font = @CMAKE_INSTALL_PREFIX@/@RESOURCE_PREFIX@/fonts/VeraMono.ttf
+Title Font = @CMAKE_INSTALL_PREFIX@__FEDORA_TITLE_FONT__
+Menu Font = @CMAKE_INSTALL_PREFIX@__FEDORA_MENU_FONT__
  
diff -Nur libprojectM-2.0.1-orig/projectM.cpp libprojectM-2.0.1/projectM.cpp
--- libprojectM-2.0.1-orig/projectM.cpp	2009-12-06 02:42:51.000000000 +0100
+++ libprojectM-2.0.1/projectM.cpp	2012-06-01 12:48:18.322584982 +0200
@@ -30,6 +30,7 @@
 
 #include "timer.h"
 #include <iostream>
+#include <fstream>
 #ifdef LINUX
 #include "time.h"
 #endif
@@ -163,6 +164,13 @@
 
 
 
+bool _fileExists(const std::string& path) {
+    std::ifstream f(path.c_str());
+    return f;
+}
+
+
+
 void projectM::readConfig (const std::string & configFile )
 {
     std::cout << "[projectM] config file: " << configFile << std::endl;
@@ -199,10 +207,20 @@
     #endif
 
     #ifdef LINUX
+    string titleFont = CMAKE_INSTALL_PREFIX  "__FEDORA_TITLE_FONT__";
+    string menuFont = CMAKE_INSTALL_PREFIX  "__FEDORA_MENU_FONT__";
     _settings.titleFontURL = config.read<string>
-    ( "Title Font", CMAKE_INSTALL_PREFIX  "/share/projectM/fonts/Vera.ttf" );
+    ( "Title Font", titleFont );
     _settings.menuFontURL = config.read<string>
-    ( "Menu Font", CMAKE_INSTALL_PREFIX  "/share/projectM/fonts/VeraMono.ttf" );
+    ( "Menu Font", menuFont );
+    if ( !_fileExists(_settings.titleFontURL) ) {
+        std::cerr << "NOT FOUND: " << _settings.titleFontURL << std::endl;
+        _settings.titleFontURL = titleFont;
+    }
+    if ( !_fileExists(_settings.menuFontURL) ) {
+        std::cerr << "NOT FOUND: " << _settings.menuFontURL << std::endl;
+        _settings.menuFontURL = menuFont;
+    }
     #endif
 
     #ifdef WIN32
@@ -251,6 +269,17 @@
     _settings.menuFontURL =  settings.menuFontURL;
     _settings.shuffleEnabled = settings.shuffleEnabled;
 
+    string titleFont = CMAKE_INSTALL_PREFIX  "__FEDORA_TITLE_FONT__";
+    string menuFont = CMAKE_INSTALL_PREFIX  "__FEDORA_MENU_FONT__";
+    if ( !_fileExists(_settings.titleFontURL) ) {
+        std::cerr << "NOT FOUND: " << _settings.titleFontURL << std::endl;
+        _settings.titleFontURL = titleFont;
+    }
+    if ( !_fileExists(_settings.menuFontURL) ) {
+        std::cerr << "NOT FOUND: " << _settings.menuFontURL << std::endl;
+        _settings.menuFontURL = menuFont;
+    }
+
     _settings.easterEgg = settings.easterEgg;
 
     projectM_init ( _settings.meshX, _settings.meshY, _settings.fps,
