From 349278c5a22201e73eadce08a96d211f0bfaf0bd Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Thu, 13 Jan 2011 00:20:39 +0000
Subject: draw: don't swap normal<->active states when drawing buttons

This is a regression from commit
8d881a1ebd4858f7fd586aa879c16fa2689e023b
---
diff --git a/src/clearlooks_draw.c b/src/clearlooks_draw.c
index 3ee38cb..5c8ea28 100644
--- a/src/clearlooks_draw.c
+++ b/src/clearlooks_draw.c
@@ -344,7 +344,7 @@ clearlooks_draw_button (cairo_t *cr,
 	                                     height-(yoffset*2)-2,
 				    radius, clearlooks_get_corners (engine));
 
-	if (state & GTK_STATE_FLAG_ACTIVE)
+	if ((state & GTK_STATE_FLAG_ACTIVE) == 0)
 	{
 		cairo_pattern_t *pattern;
 		CairoColor top_shade, topmiddle_shade, bottom_shade, middle_shade;
--
cgit v0.8.3.1
