diff -up ManiaDrive-1.2-src/raydium/ode.h.foobar ManiaDrive-1.2-src/raydium/ode.h
--- ManiaDrive-1.2-src/raydium/ode.h.foobar	2006-08-11 12:11:13.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/ode.h	2007-08-12 00:30:29.000000000 +0200
@@ -11,7 +11,7 @@
 // ugly workaround to avoid odemath.h ...
 #define _ODE_ODEMATH_H_
 #endif
-#include "../ode/include/ode/ode.h"
+#include <ode/ode.h>
 
 #define RAYDIUM_ODE_MAX_OBJECTS			64
 #define RAYDIUM_ODE_MAX_ELEMENTS		256
diff -up ManiaDrive-1.2-src/raydium/php.c.foobar ManiaDrive-1.2-src/raydium/php.c
--- ManiaDrive-1.2-src/raydium/php.c.foobar	2006-08-13 03:55:37.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/php.c	2007-08-12 00:33:15.000000000 +0200
@@ -151,6 +151,7 @@ int raydium_php_exec(char *name)  
     raydium_php_init_request(name);
 
     zend_llist_init(&global_vars, sizeof(char *), NULL, 0);
+    zend_alter_ini_entry("safe_mode", 10, "0", 1, PHP_INI_SYSTEM, PHP_INI_STAGE_ACTIVATE);
     zend_alter_ini_entry("html_errors", 12, "0", 1, PHP_INI_SYSTEM, PHP_INI_STAGE_ACTIVATE);
     zend_alter_ini_entry("max_execution_time", 19, "0", 1, PHP_INI_SYSTEM, PHP_INI_STAGE_ACTIVATE);
     zend_alter_ini_entry("default_socket_timeout",23,"10",2, PHP_INI_SYSTEM, PHP_INI_STAGE_ACTIVATE);
@@ -301,7 +302,7 @@ sapi_globals = ts_resource(sapi_globals_
 tsrm_ls = ts_resource(0);
 #endif
 raydium_sapi_module.phpinfo_as_text=1;
-path=malloc(strlen(PHP_INI_PATH));
+path=malloc(strlen(PHP_INI_PATH)+1);
 strcpy(path,PHP_INI_PATH);
 raydium_sapi_module.php_ini_path_override=path;
 sapi_startup(&raydium_sapi_module);
diff -up ManiaDrive-1.2-src/mania_drive.c.foobar ManiaDrive-1.2-src/mania_drive.c
--- ManiaDrive-1.2-src/mania_drive.c.foobar	2006-08-21 22:56:03.000000000 +0200
+++ ManiaDrive-1.2-src/mania_drive.c	2007-08-12 00:30:29.000000000 +0200
@@ -2493,6 +2493,7 @@ int full_sx_i,full_sy_i;
 int mode;
 
 raydium_init_args_name(argc,argv,"mania_drive");
+chdir("/usr/share/maniadrive");
 
 raydium_parser_db_get("ManiaDrive-Windowed",str,"0");
 windowed_mode=atoi(str);
@@ -2559,7 +2560,7 @@ raydium_parser_db_get("ManiaDrive-Camera
 camera_lag=(lagActive[0]=='y'?1:0);
 sscanf(lagSpeed,"%f",&camera_lag_speed);
 
-raydium_parser_db_get("ManiaDrive-ShadowSupport",str,"1");
+raydium_parser_db_get("ManiaDrive-ShadowSupport",str,"0");
 shadow_support=atoi(str);
 
 raydium_parser_db_get("ManiaDrive-MusicVolume",str,"1.0");
diff -up ManiaDrive-1.2-src/mania2.c.foobar ManiaDrive-1.2-src/mania2.c
--- ManiaDrive-1.2-src/mania2.c.foobar	2006-08-17 15:36:32.000000000 +0200
+++ ManiaDrive-1.2-src/mania2.c	2007-08-12 00:30:29.000000000 +0200
@@ -15,7 +15,7 @@ char *title="CQFD Corp. Mania2";
 #ifdef WIN32
 #define MANIA_BINARY "mania_drive.exe"
 #else
-#define MANIA_BINARY "./mania_drive.static"
+#define MANIA_BINARY "/usr/bin/maniadrive"
 #endif
 
 
@@ -1027,6 +1027,7 @@ data_init();
 current_track[0]=0;
 
 raydium_init_args_name(argc,argv,"mania_drive");
+chdir("/usr/share/maniadrive");
 
 raydium_window_create(700,700,RAYDIUM_RENDERING_WINDOW,window);
 
diff -up ManiaDrive-1.2-src/Makefile.foobar ManiaDrive-1.2-src/Makefile
--- ManiaDrive-1.2-src/Makefile.foobar	2006-07-22 12:52:04.000000000 +0200
+++ ManiaDrive-1.2-src/Makefile	2007-08-12 00:30:29.000000000 +0200
@@ -7,8 +7,8 @@ CC = gcc
 CC = gcc
 AR = ar
 RANLIB = ranlib
-SYSTEM_LIBS =  -lGL -lGLU -lXinerama -lm -ljpeg -lopenal -lalut -lvorbis -lvorbisfile -logg -lresolv -lcrypt -lz -lcurl -lxml2 -lGLEW
+SYSTEM_LIBS =  -lGL -lGLU -lXinerama -lm -ljpeg -lopenal -lalut -lvorbisfile -lGLEW
-OTHER_LIBS =  raydium/ode/lib/libode.a raydium/php/libs/libphp5.a
+OTHER_LIBS = -lode -lphp5
 INCLUDE_PATH =  -Iraydium/ode/include/ -Iraydium/php/ -Iraydium/php/include -Iraydium/php/main/ -Iraydium/php/Zend -Iraydium/php/TSRM -I/usr/include/curl
 LIBS_PATH =  -L/usr/X11R6/lib/
 CFLAGS=-Wall
diff -up ManiaDrive-1.2-src/raydium/ode.c~ ManiaDrive-1.2-src/raydium/ode.c
--- ManiaDrive-1.2-src/raydium/ode.c~	2008-09-15 19:41:57.000000000 +0200
+++ ManiaDrive-1.2-src/raydium/ode.c	2008-09-15 19:41:57.000000000 +0200
@@ -149,6 +149,8 @@ if(sizeof(dReal) != sizeof(float))
     exit(34);
     }
 
+dInitODE();
+
 raydium_ode_ExplosionCallback=NULL;
 raydium_ode_CollideCallback=NULL;
 raydium_ode_RayCallback=NULL;
@@ -3364,7 +3364,7 @@ for(i=0;i<RAYDIUM_ODE_MAX_ELEMENTS;i++)
 	        glPolygonMode(GL_FRONT_AND_BACK, GL_FILL);	    
 	        }
 	    else
-		//if(dGeomGetClass(raydium_ode_element[i].geom)==dSphereClass)
+		if(dGeomGetClass(raydium_ode_element[i].geom)==dSphereClass)
 		{
 	        glutWireSphere(dGeomSphereGetRadius(raydium_ode_element[i].geom),10,10);
 	        }
