--- freetds-0.95.0/src/apps/bsqldb.c.orig	2015-06-14 23:51:48.598185092 -0300
+++ freetds-0.95.0/src/apps/bsqldb.c	2015-06-14 23:53:15.714702024 -0300
@@ -570,7 +570,7 @@ print_results(DBPROCESS *dbproc)
 								break;
 							}
 						}
-						fprintf(stdout, metadata[c].format_string, ""); /* col/row separator */
+						fprintf(stdout, "%s", metadata[c].format_string, ""); /* col/row separator */
 						continue;
 					}
 					switch (data[c].status) { /* handle nulls */
