diff -up freetds-0.82/src/apps/Makefile.am.foo freetds-0.82/src/apps/Makefile.am
--- freetds-0.82/src/apps/Makefile.am.foo	2008-06-16 23:25:29.000000000 +0200
+++ freetds-0.82/src/apps/Makefile.am	2008-06-16 23:25:34.000000000 +0200
@@ -19,7 +19,7 @@ dist_bin_SCRIPTS = osql
 freebcp_LDADD	= ../dblib/libsybdb.la $(NETWORK_LIBS)
 freebcp_SOURCES = freebcp.c freebcp.h
 
-tsql_LDADD	= ../tds/libtds.la \
+tsql_LDADD	= ../tds/libtds_objects.la \
 		  ../replacements/libreplacements.la \
 		  $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC) $(READLINE_LIBS)
 
diff -up freetds-0.82/src/ctlib/Makefile.am.foo freetds-0.82/src/ctlib/Makefile.am
--- freetds-0.82/src/ctlib/Makefile.am.foo	2008-06-16 23:25:23.000000000 +0200
+++ freetds-0.82/src/ctlib/Makefile.am	2008-06-16 23:25:27.000000000 +0200
@@ -18,5 +18,5 @@ SYMBOLS		=	-export-symbols-regex '^(cs_|
 endif
 libct_la_LDFLAGS=	-version-info 4:0:0 $(SYMBOLS) $(FREETDS_SYMBOLIC)
 
-libct_la_LIBADD	=	../tds/libtds.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
+libct_la_LIBADD	=	../tds/libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 
diff -up freetds-0.82/src/dblib/Makefile.am.foo freetds-0.82/src/dblib/Makefile.am
--- freetds-0.82/src/dblib/Makefile.am.foo	2008-06-16 23:25:40.000000000 +0200
+++ freetds-0.82/src/dblib/Makefile.am	2008-06-16 23:25:49.000000000 +0200
@@ -20,5 +20,5 @@ else
 SYMBOLS		=	-export-symbols-regex '^(db|bcp_|tdsdump_open|tdsdbopen|.*_xact|close_commit|open_commit|.?asprintf).*'
 endif
 libsybdb_la_LDFLAGS=	-version-info 5:0:0 $(SYMBOLS) $(FREETDS_SYMBOLIC)
-libsybdb_la_LIBADD=	../tds/libtds.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
+libsybdb_la_LIBADD=	../tds/libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 
diff -up freetds-0.82/src/odbc/Makefile.am.foo freetds-0.82/src/odbc/Makefile.am
--- freetds-0.82/src/odbc/Makefile.am.foo	2008-06-16 23:25:14.000000000 +0200
+++ freetds-0.82/src/odbc/Makefile.am	2008-06-16 23:25:20.000000000 +0200
@@ -13,14 +13,14 @@ libtdsodbc_la_SOURCES=	odbc.c connectpar
 	descriptor.c prepare_query.c odbc_util.c \
 	native.c sql2tds.c error.c odbc_checks.c $(MINGW_SOURCES)
 if MINGW32
-libtdsodbc_la_LIBADD=	../../win32/setup.res ../tds/libtds.la ../replacements/libreplacements.la $(ODBCINSTLIB) \
+libtdsodbc_la_LIBADD=	../../win32/setup.res ../tds/libtds_objects.la ../replacements/libreplacements.la $(ODBCINSTLIB) \
 	$(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 libtdsodbc_la_LDFLAGS = -Wl,--enable-stdcall-fixup -Wl,-s -Wl,../../win32/FreeTDS.def -Wl,../../win32/setup.res $(FREETDS_SYMBOLIC)
 
 .rc.res:
 	$(RC) -i $< --input-format=rc -o $@ -O coff
 else
-libtdsodbc_la_LIBADD=	../tds/libtds.la ../replacements/libreplacements.la $(ODBCINSTLIB) $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
+libtdsodbc_la_LIBADD=	../tds/libtds_objects.la ../replacements/libreplacements.la $(ODBCINSTLIB) $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 if MACOSX
 SYMBOLS		=
 else
diff -up freetds-0.82/src/server/Makefile.am.foo freetds-0.82/src/server/Makefile.am
--- freetds-0.82/src/server/Makefile.am.foo	2008-06-16 23:25:06.000000000 +0200
+++ freetds-0.82/src/server/Makefile.am	2008-06-16 23:25:12.000000000 +0200
@@ -2,7 +2,7 @@
 AM_CPPFLAGS	=	-I$(top_srcdir)/include
 noinst_LTLIBRARIES	=	libtdssrv.la
 libtdssrv_la_SOURCES=	query.c server.c login.c
-libtdssrv_la_LIBADD =	../tds/libtds.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
+libtdssrv_la_LIBADD =	../tds/libtds_objects.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
 noinst_PROGRAMS	= tdssrv
 tdssrv_LDADD	= libtdssrv.la $(NETWORK_LIBS)
 tdssrv_SOURCES	= unittest.c
diff -up freetds-0.82/src/tds/Makefile.am.foo freetds-0.82/src/tds/Makefile.am
--- freetds-0.82/src/tds/Makefile.am.foo	2007-10-30 16:51:06.000000000 +0100
+++ freetds-0.82/src/tds/Makefile.am	2008-06-16 23:23:50.000000000 +0200
@@ -3,15 +3,25 @@
 SUBDIRS			=	unittests
 AM_CPPFLAGS		=	-I$(top_srcdir)/include
 
-noinst_LTLIBRARIES	=	libtds.la
-libtds_la_SOURCES=	mem.c token.c util.c login.c read.c \
+
+noinst_LTLIBRARIES	=	libtds_objects.la
+libtds_objects_la_SOURCES =	mem.c token.c util.c login.c read.c \
 	write.c convert.c numeric.c config.c query.c iconv.c \
 	locale.c challenge.c threadsafe.c vstrbuild.c md4.c md5.c \
 	des.c tdsstring.c getmac.c data.c net.c \
 	tds_checks.c tds_checks.h enum_cap.h log.c \
 	gssapi.c
-libtds_la_LDFLAGS=
-libtds_la_LIBADD=
+libtds_objects_la_LDFLAGS=
+libtds_objects_la_LIBADD=
+
+
+lib_LTLIBRARIES   = libtds.la
+libtds_la_SOURCES =
+libtds_la_LDFLAGS = -release @PACKAGE_VERSION@ \
+                    -export-symbols-regex '^(tds_|tdsdump_|tds5_|tds7_).*'
+libtds_la_LIBADD  = libtds_objects.la ../replacements/libreplacements.la \
+		    $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
+
 
 noinst_HEADERS		= tds_willconvert.h encodings.h num_limits.h
 EXTRA_DIST		= tds_willconvert.h encodings.h num_limits.h
diff -up freetds-0.82/src/tds/unittests/Makefile.am.foo freetds-0.82/src/tds/unittests/Makefile.am
--- freetds-0.82/src/tds/unittests/Makefile.am.foo	2008-06-16 23:24:34.000000000 +0200
+++ freetds-0.82/src/tds/unittests/Makefile.am	2008-06-16 23:24:54.000000000 +0200
@@ -35,5 +35,5 @@ AM_LDFLAGS	=	-no-fast-install
 else
 AM_LDFLAGS	=	-no-install
 endif
-LIBS		=	../libtds.la ../../replacements/libreplacements.la @NETWORK_LIBS@
+LIBS		=	../libtds_objects.la ../../replacements/libreplacements.la @NETWORK_LIBS@
 CLEANFILES	=	tdsdump.out
diff -up freetds-0.82/include/Makefile.am~ freetds-0.82/include/Makefile.am
--- freetds-0.82/include/Makefile.am~	2008-06-16 23:35:45.000000000 +0200
+++ freetds-0.82/include/Makefile.am	2008-06-16 23:38:41.000000000 +0200
@@ -3,6 +3,7 @@ AUTOMAKE_OPTIONS	=	1.5.3
 include_HEADERS	=	bkpublic.h cspublic.h cstypes.h ctpublic.h \
 			sqldb.h sqlfront.h \
 			sybdb.h sybfront.h \
+			tds.h tdsver.h tdsconvert.h \
 			syberror.h
 
 nodist_include_HEADERS	=	tds_sysdep_public.h
@@ -19,7 +20,6 @@ noinst_HEADERS	=	tds_configs.h \
 			tdsthread.h \
 			tdsbytes.h \
 			replacements/readpassphrase.h \
-			tds.h tdsver.h tdsconvert.h \
 			tdssrv.h
 
 DISTCLEANFILES	=	tds_sysdep_public.h freetds_sysconfdir.h
