diff -rbu freetds-freetds/src/apps/bsqldb.c freetds-freetds-OK/src/apps/bsqldb.c
--- freetds-freetds/src/apps/bsqldb.c	2013-08-22 00:19:20.000000000 +0400
+++ freetds-freetds-OK/src/apps/bsqldb.c	2013-12-03 19:32:39.916501589 +0400
@@ -545,7 +545,7 @@
 							perror("could not write to output file");
 							exit(EXIT_FAILURE);
 						}
-						fprintf(stdout, metadata[c].format_string); /* col/row separator */
+						fprintf(stdout, "%s", metadata[c].format_string); /* col/row separator */
 						continue;
 					}
 					switch (data[c].status) { /* handle nulls */
