Index: fpcsrc/utils/fpdoc/dwlinear.pp
===================================================================
--- fpcsrc/utils/fpdoc/dwlinear.pp	(revision 15670)
+++ fpcsrc/utils/fpdoc/dwlinear.pp	(revision 15671)
@@ -869,7 +869,9 @@
       WriteSeeAlso(DocNode);
       EndProcedure;
       WriteExample(DocNode);
-      end;
+      end
+     else
+      EndProcedure;
     end;
 end;
 
@@ -970,7 +972,9 @@
       WriteSeeAlso(DocNode);
       EndProperty;
       WriteExample(DocNode);
-      end;
+      end
+     else
+      EndProperty;
     end;
 end;
 
