From d3a858030f1945cad6961157ddac42308675adb9 Mon Sep 17 00:00:00 2001
From: dgod <dgod.osa@gmail.com>
Date: Mon, 27 Feb 2012 19:37:49 +0800
Subject: [PATCH]  remove old support for plymouth

---
 src/lxdm.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/src/lxdm.c b/src/lxdm.c
index 8d98d37..e61d529 100644
--- a/src/lxdm.c
+++ b/src/lxdm.c
@@ -529,11 +529,9 @@ void lxdm_get_tty(void)
 	if(!gotvtarg)
 	{
 		/* support plymouth */
-		nr_tty = g_file_test("/var/spool/gdm/force-display-on-active-vt", G_FILE_TEST_EXISTS);
-		if( nr_tty || g_key_file_get_integer(config, "server", "active_vt", 0) )
+		if(g_key_file_get_integer(config, "server", "active_vt", 0) )
 			/* use the active vt */
 			def_tty = old_tty;
-		if( nr_tty ) unlink("/var/spool/gdm/force-display-on-active-vt");
 		if(plymouth)
 		{
 			nr_tty=1;
-- 
1.7.4.1

