--- lxdm/pam/lxdm.orig	2010-01-06 13:42:29.000000000 +0100
+++ lxdm/pam/lxdm	2010-01-08 16:00:04.000000000 +0100
@@ -1,7 +1,17 @@
 #%PAM-1.0
-auth       include	system-auth
-account    include	system-auth
-session    optional	pam_keyinit.so force revoke
-session    include	system-auth
-session    required	pam_loginuid.so
-session    optional	pam_console.so
+auth     [success=done ignore=ignore default=bad] pam_selinux_permit.so
+auth       required    pam_succeed_if.so user != root quiet
+auth       required    pam_env.so
+auth       substack    system-auth
+auth       optional    pam_gnome_keyring.so
+account    required    pam_nologin.so
+account    include     system-auth
+password   include     system-auth
+session    required    pam_selinux.so close
+session    required    pam_loginuid.so
+session    optional    pam_console.so
+session    required    pam_selinux.so open
+session    optional    pam_keyinit.so force revoke
+session    required    pam_namespace.so
+session    optional    pam_gnome_keyring.so auto_start
+session    include     system-auth
