From 8c71ffc87305a5742d3ed9f98134cee31117ddd3 Mon Sep 17 00:00:00 2001
From: dgod <dgod.osa@gmail.com>
Date: Mon, 27 Feb 2012 20:02:16 +0800
Subject: [PATCH] add option to restart xserver when logout

---
 src/lxdm.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/lxdm.c b/src/lxdm.c
index e61d529..13e1cef 100644
--- a/src/lxdm.c
+++ b/src/lxdm.c
@@ -1232,6 +1232,11 @@ static void on_session_stop(void *data,int pid, int status)
 	{
 		lxsession_free(s);
 	}
+	else if(g_key_file_get_integer(config,"server","reset",NULL)==1)
+	{
+		lxsession_free(s);
+		lxsession_greeter();
+	}
 	gchar *argv[] = { "/etc/lxdm/PostLogout", NULL };
 	g_spawn_async(NULL, argv, s->env, G_SPAWN_SEARCH_PATH, NULL, NULL, NULL, NULL);
 }
-- 
1.7.4.1

