From f11ae65ebc732df2e0e5f69200a396db61657d16 Mon Sep 17 00:00:00 2001
From: dgod <dgod.osa@gmail.com>
Date: Sun, 29 Jan 2012 10:32:13 +0800
Subject: [PATCH] exec dbus-launch as bug3477356 suggested

---
 data/Xsession |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/data/Xsession b/data/Xsession
index 287db71..8fe0c79 100755
--- a/data/Xsession
+++ b/data/Xsession
@@ -34,13 +34,19 @@ elif [ -x /etc/X11/xinit/xinitrc ]; then
 else
 # unknown, user should custom /etc/lxdm/xinitrc self
 	if [ -x /etc/lxdm/xinitrc ]; then
-		. /etc/lxdm/xinitrc
+		. /etc/lxdm/xinitrc "$LXSESSION"
 	fi
 	if ! [ -z "$XDG_SESSION_COOKIE" ]; then
 		CK_XINIT_SESSION=
 	elif [ -x /usr/bin/ck-launch-session ]; then
 		CK_XINIT_SESSION="/usr/bin/ck-launch-session"
 	fi
+
+	if which dbus-launch >/dev/null && test -z "$DBUS_SESSION_BUS_ADDRESS";
+	then
+		eval "$(dbus-launch --sh-syntax --exit-with-session)"
+	fi
+
 	exec -l $SHELL -c "$CK_XINIT_SESSION \"$LXSESSION\""
 fi
 
-- 
1.7.4.1

