From 14b6c10344429227b9a2aa85dd5dcdd569853377 Mon Sep 17 00:00:00 2001
From: Martin Bagge / brother <brother@bsnet.se>
Date: Thu, 24 Nov 2011 15:34:10 +0100
Subject: [PATCH] Corrected spelling mistake; Show languanges menu => Show languages menu.

---
 data/config.ui |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/data/config.ui b/data/config.ui
index 423fe56..08aeb73 100644
--- a/data/config.ui
+++ b/data/config.ui
@@ -176,7 +176,7 @@
         </child>
         <child>
           <object class="GtkCheckButton" id="lxdm-show-lang">
-            <property name="label" translatable="yes">Show languanges menu</property>
+            <property name="label" translatable="yes">Show languages menu</property>
             <property name="visible">True</property>
             <property name="can_focus">True</property>
             <property name="receives_default">False</property>
-- 
1.7.4.1

