diff -up gnucash-2.3.12/src/bin/gnucash-bin.c.orig gnucash-2.3.12/src/bin/gnucash-bin.c
--- gnucash-2.3.12/src/bin/gnucash-bin.c.orig	2010-04-10 20:22:18.000000000 -0400
+++ gnucash-2.3.12/src/bin/gnucash-bin.c	2010-04-30 16:12:54.172388707 -0400
@@ -716,8 +716,6 @@ main(int argc, char ** argv)
             g_error_free(binreloc_error);
         }
     }
-#else
-    g_message("main: binreloc relocation support was disabled at configure time.\n");
 #endif
 
     /* This should be called before gettext is initialized
