--- a/bitlbee.conf
+++ b/bitlbee.conf
@@ -23,14 +23,14 @@
 ## If BitlBee is started by root as a daemon, it can drop root privileges,
 ## and change to the specified user.
 ##
-# User = bitlbee
+User = bitlbee
 
 ## DaemonPort/DaemonInterface:
 ##
 ## For daemon mode, you can specify on what interface and port the daemon
 ## should be listening for connections.
 ##
-# DaemonInterface = 0.0.0.0
+DaemonInterface = 127.0.0.1
 # DaemonPort = 6667
 
 ## ClientInterface:
