diff -up bitlbee-3.0.1/bitlbee.conf.forkdaemon bitlbee-3.0.1/bitlbee.conf
--- bitlbee-3.0.1/bitlbee.conf.forkdaemon	2010-11-24 23:44:00.000000000 +0100
+++ bitlbee-3.0.1/bitlbee.conf	2010-12-27 21:17:22.601505634 +0100
@@ -23,14 +23,14 @@
 ## If BitlBee is started by root as a daemon, it can drop root privileges,
 ## and change to the specified user.
 ##
-# User = bitlbee
+User = bitlbee
 
 ## DaemonPort/DaemonInterface:
 ##
 ## For daemon mode, you can specify on what interface and port the daemon
 ## should be listening for connections.
 ##
-# DaemonInterface = 0.0.0.0
+DaemonInterface = 127.0.0.1
 # DaemonPort = 6667
 
 ## ClientInterface:
