diff -up bitlbee-3.0/configure.installP bitlbee-3.0/configure
--- bitlbee-3.0/configure.installP	2010-11-22 03:06:31.000000000 +0100
+++ bitlbee-3.0/configure	2010-11-22 03:05:59.000000000 +0100
@@ -123,6 +123,8 @@ TARGET=$target
 ARCH=$arch
 CPU=$cpu
 
+INSTALL=install -p
+
 DESTDIR=
 LFLAGS=
 EFLAGS=
diff -up bitlbee-3.0/doc/Makefile.installP bitlbee-3.0/doc/Makefile
--- bitlbee-3.0/doc/Makefile.installP	2010-11-22 03:11:47.000000000 +0100
+++ bitlbee-3.0/doc/Makefile	2010-11-22 03:13:42.000000000 +0100
@@ -9,8 +9,8 @@ all:
 
 install:
 	mkdir -p $(DESTDIR)$(MANDIR)/man8/ $(DESTDIR)$(MANDIR)/man5/
-	install -m 0644 $(SRCDIR)bitlbee.8 $(DESTDIR)$(MANDIR)/man8/
-	install -m 0644 $(SRCDIR)bitlbee.conf.5 $(DESTDIR)$(MANDIR)/man5/
+	$(INSTALL) -m 0644 $(SRCDIR)bitlbee.8 $(DESTDIR)$(MANDIR)/man8/
+	$(INSTALL) -m 0644 $(SRCDIR)bitlbee.conf.5 $(DESTDIR)$(MANDIR)/man5/
 	$(MAKE) -C user-guide $@
 
 uninstall:
diff -up bitlbee-3.0/doc/user-guide/Makefile.installP bitlbee-3.0/doc/user-guide/Makefile
--- bitlbee-3.0/doc/user-guide/Makefile.installP	2010-11-22 03:11:47.000000000 +0100
+++ bitlbee-3.0/doc/user-guide/Makefile	2010-11-22 03:13:56.000000000 +0100
@@ -41,7 +41,7 @@ install:
 	mkdir -p $(DESTDIR)$(DATADIR)
 	chmod 0755 $(DESTDIR)$(DATADIR)
 	rm -f $(DESTDIR)$(DATADIR)/help.txt # Prevent help function from breaking in running sessions
-	install -m 0644 $(SRCDIR)help.txt $(DESTDIR)$(DATADIR)/help.txt
+	$(INSTALL) -m 0644 $(SRCDIR)help.txt $(DESTDIR)$(DATADIR)/help.txt
 
 uninstall:
 	rm -f $(DESTDIR)$(DATADIR)/help.txt
diff -up bitlbee-3.0/lib/Makefile.installP bitlbee-3.0/lib/Makefile
diff -up bitlbee-3.0/Makefile.installP bitlbee-3.0/Makefile
diff -up bitlbee-3.0/protocols/jabber/Makefile.installP bitlbee-3.0/protocols/jabber/Makefile
diff -up bitlbee-3.0/protocols/Makefile.installP bitlbee-3.0/protocols/Makefile
diff -up bitlbee-3.0/protocols/msn/Makefile.installP bitlbee-3.0/protocols/msn/Makefile
diff -up bitlbee-3.0/protocols/oscar/Makefile.installP bitlbee-3.0/protocols/oscar/Makefile
diff -up bitlbee-3.0/protocols/purple/Makefile.installP bitlbee-3.0/protocols/purple/Makefile
diff -up bitlbee-3.0/protocols/twitter/Makefile.installP bitlbee-3.0/protocols/twitter/Makefile
diff -up bitlbee-3.0/protocols/yahoo/Makefile.installP bitlbee-3.0/protocols/yahoo/Makefile
diff -up bitlbee-3.0/tests/Makefile.installP bitlbee-3.0/tests/Makefile
