Index: bitlbee-3.0.3/configure
===================================================================
--- bitlbee-3.0.3.orig/configure
+++ bitlbee-3.0.3/configure
@@ -20,7 +20,7 @@ libevent='/usr/'
 pidfile='/var/run/bitlbee.pid'
 ipcsocket='/var/run/bitlbee.sock'
 pcdir='$prefix/lib/pkgconfig'
-systemlibdirs="/lib /lib64 /usr/lib /usr/lib64 /usr/local/lib /usr/local/lib64"
+systemlibdirs="/lib64 /usr/lib64 /usr/local/lib64 /lib /usr/lib /usr/local/lib"
 
 msn=1
 jabber=1
