diff -up bitlbee-3.0/configure.libresolv bitlbee-3.0/configure
--- bitlbee-3.0/configure.libresolv	2010-10-22 02:51:31.000000000 +0200
+++ bitlbee-3.0/configure	2010-11-22 01:15:01.000000000 +0100
@@ -19,7 +19,7 @@ libevent='/usr/'
 pidfile='/var/run/bitlbee.pid'
 ipcsocket='/var/run/bitlbee.sock'
 pcdir='$prefix/lib/pkgconfig'
-systemlibdirs="/lib /lib64 /usr/lib /usr/lib64 /usr/local/lib /usr/local/lib64"
+systemlibdirs="/lib64 /usr/lib64 /usr/local/lib64 /lib /usr/lib /usr/local/lib"
 
 msn=1
 jabber=1
