From 77ed9b724d1bff2a409642795ae81f974b8c312f Mon Sep 17 00:00:00 2001
From: Wilmer van der Gaast <wilmer@gaast.net>
Date: Wed, 17 Jun 2015 23:50:15 +0100
Subject: [PATCH] Fix NULL pointer dereference in lib/json.c. (#1220)

---
 lib/json.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/json.c b/lib/json.c
index cbd6eaa..24288f9 100644
--- a/lib/json.c
+++ b/lib/json.c
@@ -508,7 +508,7 @@ json_value * json_parse_ex(json_settings * settings,
 
 				case ']':
 
-					if (top->type == json_array) {
+					if (top && top->type == json_array) {
 						flags = (flags & ~(flag_need_comma | flag_seek_value)) | flag_next;
 					} else {              sprintf(error, "%d:%d: Unexpected ]", cur_line, e_off);
 						              goto e_failed; }
