--- crossapi.cpp.orig	2016-07-17 20:32:09.597703995 -0300
+++ crossapi.cpp	2016-07-17 20:35:26.191370177 -0300
@@ -241,7 +241,7 @@ int CrossAPI_MoveFile(char *szNewName,ch
 //Moving failed due to different logical drives of source and destination. Let's copy:
 	id=open(szOldName,O_RDONLY);
 	if(id==-1) return 0;
-	od=open(szNewName,O_WRONLY|O_CREAT|O_TRUNC);
+	od=open(szNewName,O_WRONLY|O_CREAT|O_TRUNC, 0600);
 	if(od==-1) {
 		close(id);
 		return 0;
