--- conftest.sh.orig	2012-11-13 20:51:14.559040126 -0200
+++ conftest.sh	2012-11-13 20:54:25.637433947 -0200
@@ -127,7 +127,7 @@ build_cflags() {
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         MACH_CFLAGS="-I$HEADERS/asm-$ARCH/mach-default"
-        if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
+        if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" -o "$ARCH" = "x86" ]; then
             MACH_CFLAGS="$MACH_CFLAGS -I$HEADERS/asm-x86/mach-default"
             MACH_CFLAGS="$MACH_CFLAGS -I$SOURCES/arch/x86/include/asm/mach-default"
             MACH_CFLAGS="$MACH_CFLAGS -I$HEADERS/arch/x86/include/uapi"
@@ -141,7 +141,7 @@ build_cflags() {
         fi
     else
         MACH_CFLAGS="-I$HEADERS/asm/mach-default"
-        if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
+        if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" -o "$ARCH" = "x86" ]; then
             MACH_CFLAGS="$MACH_CFLAGS -I$HEADERS/asm-x86/mach-default"
             MACH_CFLAGS="$MACH_CFLAGS -I$SOURCES/arch/x86/include/asm/mach-default"
             MACH_CFLAGS="$MACH_CFLAGS -I$HEADERS/arch/x86/include/uapi"
@@ -158,7 +158,7 @@ build_cflags() {
     CFLAGS="$BASE_CFLAGS $MACH_CFLAGS $OUTPUT_CFLAGS $AUTOCONF_CFLAGS"
     CFLAGS="$CFLAGS -I$HEADERS -I$HEADERS/uapi"
 
-    if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
+    if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" -o "$ARCH" = "x86" ]; then
         CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
         CFLAGS="$CFLAGS -I$OUTPUT/arch/x86/include/generated"
         CFLAGS="$CFLAGS -I$OUTPUT/arch/x86/include/generated/uapi"
