--- kernel/nv.c.msi	2010-11-29 06:53:54.000000000 -0200
+++ kernel/nv.c	2010-12-04 11:56:48.819446185 -0200
@@ -1802,6 +1802,7 @@ static void __exit nvidia_exit_module(vo
             }
             release_mem_region(NV_PCI_RESOURCE_START(dev, NV_GPU_BAR_INDEX_REGS),
                                NV_PCI_RESOURCE_SIZE(dev, NV_GPU_BAR_INDEX_REGS));
+                               pci_disable_device(dev);
         }
     }
 
