--- uvm/nvidia_uvm_linux.h.orig	2013-11-20 20:48:09.125312609 -0200
+++ uvm/nvidia_uvm_linux.h	2013-11-20 20:48:56.971250360 -0200
@@ -391,7 +391,7 @@ typedef void irqreturn_t;
 //
 #if defined(NV_TASK_STRUCT_HAS_CRED)
 #define NV_CURRENT_EUID() \
-    (((typeof(*current->cred) __force __kernel *)current->cred)->euid)
+    from_kuid(current_user_ns(), current->cred->euid)
 #else
 #define NV_CURRENT_EUID() (current->euid)
 #endif
