--- a/kernel/nv-linux.h
+++ b/kernel/nv-linux.h
@@ -276,6 +276,9 @@ extern int nv_pat_mode;
 #if !defined(NV_VMWARE) && defined(CONFIG_ACPI)
 #include <acpi/acpi.h>
 #include <acpi/acpi_drivers.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+#include <acpi/acpi_bus.h>
+#endif
 #if defined(NV_ACPI_DEVICE_OPS_HAS_MATCH) || defined(ACPI_VIDEO_HID)
 #define NV_LINUX_ACPI_EVENTS_SUPPORTED 1
 #endif
--- a/kernel/uvm/nvidia_uvm_common.c
+++ b/kernel/uvm/nvidia_uvm_common.c
@@ -272,5 +272,9 @@ static void __exit uvm_exit(void)
 
 module_init(uvm_init);
 module_exit(uvm_exit);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+MODULE_LICENSE("GPL");
+#else
 MODULE_LICENSE("MIT");
+#endif
 MODULE_INFO(supported, "external");
