diff -up metacity-2.28.0/src/metacity.schemas.in.in.workspaces metacity-2.28.0/src/metacity.schemas.in.in
--- metacity-2.28.0/src/metacity.schemas.in.in.workspaces	2009-10-15 13:52:37.983401456 -0400
+++ metacity-2.28.0/src/metacity.schemas.in.in	2009-10-15 13:52:58.017654116 -0400
@@ -295,7 +295,7 @@
       <applyto>/apps/metacity/general/num_workspaces</applyto>      
       <owner>metacity</owner>
       <type>int</type>
-      <default>4</default>
+      <default>2</default>
       <locale name="C">
         <short>Number of workspaces</short>
         <long>
