diff -rupN clementine-0.4.2.old/src/core/simpletreeitem.h clementine-0.4.2/src/core/simpletreeitem.h
--- clementine-0.4.2.old/src/core/simpletreeitem.h	2010-07-01 14:16:36.000000000 -0400
+++ clementine-0.4.2/src/core/simpletreeitem.h	2010-08-05 09:10:18.000000000 -0400
@@ -121,12 +121,14 @@ void SimpleTreeItem<T>::DeleteNotify(int
 
 template <typename T>
 void SimpleTreeItem<T>::ClearNotify() {
-  model->BeginDelete(static_cast<T*>(this), 0, children.count()-1);
-
-  qDeleteAll(children);
-  children.clear();
-
-  model->EndDelete();
+  if (children.count()) {
+    model->BeginDelete(static_cast<T*>(this), 0, children.count()-1);
+    
+    qDeleteAll(children);
+    children.clear();
+    
+    model->EndDelete();
+  }
 }
 
 template <typename T>
