diff -rupN clementine-0.4.2.old/src/visualisations/projectmvisualisation.cpp clementine-0.4.2/src/visualisations/projectmvisualisation.cpp
--- clementine-0.4.2.old/src/visualisations/projectmvisualisation.cpp	2010-07-16 22:37:32.000000000 -0400
+++ clementine-0.4.2/src/visualisations/projectmvisualisation.cpp	2010-07-17 21:58:04.000000000 -0400
@@ -106,6 +106,8 @@ void ProjectMVisualisation::InitProjectM
   s.windowHeight = 512;
   s.smoothPresetDuration = 5;
   s.presetDuration = duration_;
+  s.titleFontURL = "/usr/share/fonts/TTF/DejaVuSans.ttf";
+  s.menuFontURL = "/usr/share/fonts/TTF/DejaVuSans-Bold.ttf";
   s.presetURL = preset_path.toStdString();
   s.shuffleEnabled = true;
   s.easterEgg = 0; // ??
