diff -up clementine-1.1.1/CMakeLists.txt.mygpo clementine-1.1.1/CMakeLists.txt
--- clementine-1.1.1/CMakeLists.txt.mygpo	2012-11-18 01:51:26.000000000 -0700
+++ clementine-1.1.1/CMakeLists.txt	2013-02-24 10:49:15.852174934 -0700
@@ -394,9 +394,9 @@ if(NOT CHROMAPRINT_FOUND)
 endif(NOT CHROMAPRINT_FOUND)
 
 # We have to use our own libmygpo-qt for now
-add_subdirectory(3rdparty/libmygpo-qt)
+#add_subdirectory(3rdparty/libmygpo-qt)
 set(MYGPOQT_LIBRARIES mygpo-qt)
-set(MYGPOQT_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/3rdparty/libmygpo-qt/)
+set(MYGPOQT_INCLUDE_DIRS /usr/include/mygpo-qt/)
 
 # Qocoa
 add_subdirectory(3rdparty/qocoa)
