diff -rupN clementine-0.4.2.old/CMakeLists.txt clementine-0.4.2/CMakeLists.txt
--- clementine-0.4.2.old/CMakeLists.txt	2010-07-16 22:46:52.000000000 -0400
+++ clementine-0.4.2/CMakeLists.txt	2010-07-17 00:47:44.000000000 -0400
@@ -181,9 +181,9 @@ endif(NOT QTIOCOMPRESSOR_INCLUDE_DIRS OR
 
 # Subdirectories
 add_subdirectory(src)
-if (NOT APPLE)
-  add_subdirectory(3rdparty/qxt)
-endif (NOT APPLE)
+#if (NOT APPLE)
+#  add_subdirectory(3rdparty/qxt)
+#endif (NOT APPLE)
 add_subdirectory(3rdparty/universalchardet)
 add_subdirectory(tests)
 add_subdirectory(dist)
diff -rupN clementine-0.4.2.old/src/CMakeLists.txt clementine-0.4.2/src/CMakeLists.txt
--- clementine-0.4.2.old/src/CMakeLists.txt	2010-07-16 22:47:48.000000000 -0400
+++ clementine-0.4.2/src/CMakeLists.txt	2010-07-17 01:07:34.000000000 -0400
@@ -6,7 +6,7 @@ set(CMAKE_CXX_FLAGS "-Woverloaded-virtua
 include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${SPARKLE})
 include_directories(../3rdparty/gmock/gtest/include)
-include_directories(../3rdparty/qxt)
+#include_directories(../3rdparty/qxt)
 
 include_directories(${GLIB_INCLUDE_DIRS})
 include_directories(${LIBXML_INCLUDE_DIRS})
@@ -14,6 +14,9 @@ include_directories(${GOBJECT_INCLUDE_DI
 include_directories(${LIBPROJECTM_INCLUDE_DIRS})
 include_directories(${QTSINGLEAPPLICATION_INCLUDE_DIRS})
 include_directories(${QTIOCOMPRESSOR_INCLUDE_DIRS})
+include_directories("/usr/include/QtSolutions/")
+include_directories("/usr/include/QxtGui/")
+include_directories("/usr/include/QxtCore/")
 
 cmake_policy(SET CMP0011 NEW)
 include(../cmake/AddEngine.cmake)
@@ -475,7 +478,7 @@ if (APPLE)
   )
   include_directories(${SPARKLE}/Headers)
 else (APPLE)
-  target_link_libraries(clementine_lib qxt)
+  target_link_libraries(clementine_lib QxtCore QxtGui)
 endif (APPLE)
 
 # Link against the qsqlite plugin on windows and mac
@@ -484,7 +487,7 @@ if(HAVE_STATIC_SQLITE)
   target_link_libraries(clementine_lib qsqlite)
 endif(HAVE_STATIC_SQLITE)
 
-add_dependencies(clementine_lib qtsingleapplication)
+add_dependencies(clementine_lib qtsingleapplication QxtCore QxtGui)
 
 
 ###############################################################################
