diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1104542..65d765c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -887,7 +887,6 @@ if(HAVE_DBUS)
   # Hack to get it to generate interfaces without namespaces - required
   # because otherwise org::freedesktop::UDisks and
   # org::freedesktop::UDisks::Device conflict.
-  list(APPEND QT_DBUSXML2CPP_EXECUTABLE -N)
 
   # MPRIS DBUS interfaces
   qt4_add_dbus_adaptor(SOURCES
@@ -954,6 +953,8 @@ if(HAVE_DBUS)
   list(APPEND HEADERS ${CMAKE_CURRENT_BINARY_DIR}/dbus/avahientrygroup.h)
   list(APPEND SOURCES ${CMAKE_CURRENT_BINARY_DIR}/dbus/avahientrygroup.cpp)
 
+  set_source_files_properties( dbus/org.freedesktop.UDisks.xml dbus/org.freedesktop.UDisks.Device.xml PROPERTIES NO_NAMESPACE TRUE)
+
   # DeviceKit DBUS interfaces
   if(HAVE_DEVICEKIT)
     qt4_add_dbus_interface(SOURCES
