diff -rupN clementine-0.4.2.old/CMakeLists.txt clementine-0.4.2/CMakeLists.txt
--- clementine-0.4.2.old/CMakeLists.txt	2010-07-16 22:42:41.000000000 -0400
+++ clementine-0.4.2/CMakeLists.txt	2010-07-16 22:46:52.000000000 -0400
@@ -169,8 +169,17 @@ else(USE_SYSTEM_QTSINGLEAPPLICATION)
   set(QTSINGLEAPPLICATION_LIBRARIES qtsingleapplication)
 endif(USE_SYSTEM_QTSINGLEAPPLICATION)
 
+# QtIoCompressor isn't patched, so we can use a system version if it's
+# available
+find_path(QTIOCOMPRESSOR_INCLUDE_DIRS qtiocompressor.h PATH_SUFFIXES QtSolutions)
+find_library(QTIOCOMPRESSOR_LIBRARIES QtSolutions_IOCompressor-2.3)
+if(NOT QTIOCOMPRESSOR_INCLUDE_DIRS OR NOT QTIOCOMPRESSOR_LIBRARIES)
+  add_subdirectory(3rdparty/qtiocompressor)
+  set(QTIOCOMPRESSOR_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/qtiocompressor)
+  set(QTIOCOMPRESSOR_LIBRARIES qtiocompressor)
+endif(NOT QTIOCOMPRESSOR_INCLUDE_DIRS OR NOT QTIOCOMPRESSOR_LIBRARIES)
+
 # Subdirectories
-add_subdirectory(3rdparty/qtiocompressor)
 add_subdirectory(src)
 if (NOT APPLE)
   add_subdirectory(3rdparty/qxt)
diff -rupN clementine-0.4.2.old/src/CMakeLists.txt clementine-0.4.2/src/CMakeLists.txt
--- clementine-0.4.2.old/src/CMakeLists.txt	2010-07-16 22:43:28.000000000 -0400
+++ clementine-0.4.2/src/CMakeLists.txt	2010-07-16 22:47:48.000000000 -0400
@@ -6,7 +6,6 @@ set(CMAKE_CXX_FLAGS "-Woverloaded-virtua
 include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${SPARKLE})
 include_directories(../3rdparty/gmock/gtest/include)
-include_directories(../3rdparty/qtiocompressor)
 include_directories(../3rdparty/qxt)
 
 include_directories(${GLIB_INCLUDE_DIRS})
@@ -14,6 +13,7 @@ include_directories(${LIBXML_INCLUDE_DIR
 include_directories(${GOBJECT_INCLUDE_DIRS})
 include_directories(${LIBPROJECTM_INCLUDE_DIRS})
 include_directories(${QTSINGLEAPPLICATION_INCLUDE_DIRS})
+include_directories(${QTIOCOMPRESSOR_INCLUDE_DIRS})
 
 cmake_policy(SET CMP0011 NEW)
 include(../cmake/AddEngine.cmake)
@@ -449,7 +449,6 @@ add_library(clementine_lib STATIC
 add_dependencies(clementine_lib pot)
 
 target_link_libraries(clementine_lib
-  qtiocompressor
   chardet
   ${GOBJECT_LIBRARIES}
   ${GLIB_LIBRARIES}
@@ -458,6 +457,7 @@ target_link_libraries(clementine_lib
   ${ENGINE_LIBRARIES}
   ${LASTFM_LIBRARIES}
   ${QTSINGLEAPPLICATION_LIBRARIES}
+  ${QTIOCOMPRESSOR_LIBRARIES}
 )
 
 if(ENABLE_VISUALISATIONS)
diff -rupN clementine-0.4.2.old/src/radio/magnatuneservice.cpp clementine-0.4.2/src/radio/magnatuneservice.cpp
--- clementine-0.4.2.old/src/radio/magnatuneservice.cpp	2010-07-01 14:16:36.000000000 -0400
+++ clementine-0.4.2/src/radio/magnatuneservice.cpp	2010-07-16 22:49:14.000000000 -0400
@@ -28,11 +28,12 @@
 #include "ui/iconloader.h"
 #include "ui/settingsdialog.h"
 
+#include "qtiocompressor.h"
+
 #include <QNetworkAccessManager>
 #include <QNetworkRequest>
 #include <QNetworkReply>
 #include <QXmlStreamReader>
-#include <QtIOCompressor>
 #include <QSortFilterProxyModel>
 #include <QMenu>
 #include <QDesktopServices>
