diff -rupN clementine-0.4.2.old/src/CMakeLists.txt clementine-0.4.2/src/CMakeLists.txt
--- clementine-0.4.2.old/src/CMakeLists.txt	2010-08-05 09:17:46.000000000 -0400
+++ clementine-0.4.2/src/CMakeLists.txt	2010-08-06 23:11:03.000000000 -0400
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
-set(CMAKE_C_FLAGS "-Wall")
-set(CMAKE_CXX_FLAGS "-Woverloaded-virtual -Wall")
+set(CMAKE_C_FLAGS "-Wall ${CMAKE_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "-Woverloaded-virtual -Wall ${CMAKE_CXX_FLAGS}")
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${SPARKLE})
