diff -up clementine-1.2.3/src/core/mergedproxymodel.h.boost clementine-1.2.3/src/core/mergedproxymodel.h
--- clementine-1.2.3/src/core/mergedproxymodel.h.boost	2014-11-12 23:36:19.318238446 -0200
+++ clementine-1.2.3/src/core/mergedproxymodel.h	2014-11-12 23:36:26.085805867 -0200
@@ -25,10 +25,12 @@
 using std::placeholders::_1;
 using std::placeholders::_2;
 
+#ifndef Q_MOC_RUN
 #include <boost/multi_index_container.hpp>
 #include <boost/multi_index/member.hpp>
 #include <boost/multi_index/ordered_index.hpp>
 #include <boost/multi_index/hashed_index.hpp>
+#endif
 
 using boost::multi_index::multi_index_container;
 using boost::multi_index::indexed_by;
diff -up clementine-1.2.3/src/core/signalchecker.h.boost clementine-1.2.3/src/core/signalchecker.h
--- clementine-1.2.3/src/core/signalchecker.h.boost	2014-11-12 23:40:01.758156026 -0200
+++ clementine-1.2.3/src/core/signalchecker.h	2014-11-12 23:40:26.210980779 -0200
@@ -20,8 +20,10 @@
 
 #include <glib-object.h>
 
+#ifndef Q_MOC_RUN
 #include <boost/function_types/function_arity.hpp>
 #include <boost/typeof/typeof.hpp>
+#endif
 
 // Do not call this directly, use CHECKED_GCONNECT instead.
 bool CheckedGConnect(
diff -up clementine-1.2.3/src/library/groupbydialog.h.boost clementine-1.2.3/src/library/groupbydialog.h
--- clementine-1.2.3/src/library/groupbydialog.h.boost	2014-11-12 23:40:13.692630176 -0200
+++ clementine-1.2.3/src/library/groupbydialog.h	2014-11-12 23:40:28.977395295 -0200
@@ -25,9 +25,11 @@
 using std::placeholders::_1;
 using std::placeholders::_2;
 
+#ifndef Q_MOC_RUN
 #include <boost/multi_index_container.hpp>
 #include <boost/multi_index/member.hpp>
 #include <boost/multi_index/ordered_index.hpp>
+#endif
 
 #include "librarymodel.h"
 
