--- src/devices/imobiledeviceconnection.cpp.orig	2013-03-20 18:32:09.000709138 -0300
+++ src/devices/imobiledeviceconnection.cpp	2013-03-20 19:50:57.712404241 -0300
@@ -33,6 +33,7 @@ iMobileDeviceConnection::iMobileDeviceCo
   }
 
   lockdownd_client_t lockdown;
+  lockdownd_service_descriptor_t service = NULL;
 
   QByteArray label_ascii = QCoreApplication::applicationName().toAscii();
   const char* label = label_ascii.constData();
@@ -43,14 +44,14 @@ iMobileDeviceConnection::iMobileDeviceCo
     return;
   }
 
-  lockdown_err = lockdownd_start_service(lockdown, "com.apple.afc", &afc_port_);
+  lockdown_err = lockdownd_start_service(lockdown, "com.apple.afc", &service);
   if (lockdown_err != LOCKDOWN_E_SUCCESS) {
     qLog(Warning) << "lockdown error:" << lockdown_err;
     lockdownd_client_free(lockdown);
     return;
   }
 
-  afc_error_t afc_err = afc_client_new(device_, afc_port_, &afc_);
+  afc_error_t afc_err = afc_client_new(device_, service, &afc_);
   if (afc_err != 0) {
     qLog(Warning) << "afc error:" << afc_err;
     lockdownd_client_free(lockdown);
