diff -up ntfs-3g-2010.8.8/libfuse-lite/mount.c.context ntfs-3g-2010.8.8/libfuse-lite/mount.c
--- ntfs-3g-2010.8.8/libfuse-lite/mount.c.context	2010-09-09 15:53:39.470942002 -0400
+++ ntfs-3g-2010.8.8/libfuse-lite/mount.c	2010-09-09 15:54:33.335942001 -0400
@@ -65,6 +65,7 @@ static const struct fuse_opt fuse_mount_
     FUSE_OPT_KEY("large_read",          KEY_KERN_OPT),
     FUSE_OPT_KEY("blksize=",            KEY_KERN_OPT),
     FUSE_OPT_KEY("default_permissions", KEY_KERN_OPT),
+    FUSE_OPT_KEY("context=",            KEY_KERN_OPT),
     FUSE_OPT_KEY("max_read=",           KEY_KERN_OPT),
     FUSE_OPT_KEY("max_read=",           FUSE_OPT_KEY_KEEP),
     FUSE_OPT_KEY("user=",               KEY_MTAB_OPT),
