--- hal-0.5.14/hald/linux/probing/probe-input.c	2009-08-24 08:42:30.000000000 -0400
+++ hacked/hald/linux/probing/probe-input.c	2010-09-07 15:55:52.271475007 -0400
@@ -69,6 +69,8 @@
 	ret = 1;
 	fd = -1;
 
+	dbus_error_init (&error);
+
 	setup_logger ();
 
 	button_type = getenv ("HAL_PROP_BUTTON_TYPE");
@@ -96,7 +98,6 @@
 	if (udi == NULL)
 		goto out;
 
-	dbus_error_init (&error);
 	if ((ctx = libhal_ctx_init_direct (&error)) == NULL)
 		goto out;
 
