diff -pruN less-458.orig/charset.c less-458/charset.c
--- less-458.orig/charset.c	2013-04-05 01:55:05.000000000 +0900
+++ less-458/charset.c	2014-03-10 19:47:09.850026052 +0900
@@ -970,7 +970,7 @@ static struct wchar_range ubin_table[] =
 	{  0x2064,  0x2069} /* Cn */,
 	{  0x206A,  0x206F} /* Cf */,
 	{  0x2072,  0x2073} /* Cn */, {  0x208F,  0x208F} /* Cn */,
-	{  0x2095,  0x209F} /* Cn */, {  0x20B6,  0x20CF} /* Cn */,
+	{  0x2095,  0x209F} /* Cn */, {  0x20BB,  0x20CF} /* Cn */,
 	{  0x20F0,  0x20FF} /* Cn */, {  0x214F,  0x2152} /* Cn */,
 	{  0x2185,  0x218F} /* Cn */, {  0x23E8,  0x23FF} /* Cn */,
 	{  0x2427,  0x243F} /* Cn */, {  0x244B,  0x245F} /* Cn */,
