diff --git a/src/wp/main/gtk/libabiword.cpp b/src/wp/main/gtk/libabiword.cpp
index 7567a54..c4d8e12 100644
--- a/src/wp/main/gtk/libabiword.cpp
+++ b/src/wp/main/gtk/libabiword.cpp
@@ -30,8 +30,8 @@ static AP_UnixApp *_abiword_app = NULL;
 
 /**
  * libabiword_init:
- * @argc: (inout): argument count
- * @argv: (array length=argc) (inout) (allow-none): Commandline arguments
+ * @argc: argument count
+ * @argv: (array length=argc): Commandline arguments
  *
  * Initializes libabiword
  */

