diff -dur fbpanel-5.6.orig/panel/Makefile.in fbpanel-5.6/panel/Makefile.in
--- fbpanel-5.6.orig/panel/Makefile.in	2009-06-27 21:18:12.000000000 +0200
+++ fbpanel-5.6/panel/Makefile.in	2010-02-25 20:07:57.000000000 +0100
@@ -2,7 +2,7 @@
 
 SRCS = panel.c misc.c plugin.c gtkbar.c bg.c gtkbgbox.c ev.c configurator.c 
 BINTARGET = fbpanel
-LDFLAGS += $(GMODULE_LIBS)
+LDFLAGS += $(GMODULE_LIBS) -lX11 -lm
 
 ifeq (enabled,$(STATIC_BUILD))
 OBJS += $(wildcard $(TOPDIR)/plugins/*/*.o)
