--- makemkv_v1.6.16_oss/makefile.linux.orig	2011-10-27 20:34:50.000000000 -0200
+++ makemkv_v1.6.16_oss/makefile.linux	2011-11-27 17:20:08.973377958 -0200
@@ -7,6 +7,8 @@ MOC=$(strip $(shell which moc-qt4))
 ifeq ($(MOC),)
 MOC=moc
 endif
+QTINC=$(shell pkg-config QtGui QtCore QtDBus --cflags)
+QTLIBS=$(shell pkg-config QtGui QtCore QtDBus --libs)
 
 CXXOPTS=-Os -D_GNU_SOURCE -D_linux_
 
@@ -39,7 +41,7 @@ out/libdriveio.so.0.full:
 	mkdir -p out
 	$(GCC) $(CXXOPTS) -D_REENTRANT -shared -Wl,-z,defs -o$@ $(LIBDRIVEIO_INC) $(LIBDRIVEIO_SRC) \
 	-fPIC -Xlinker -dy -Xlinker --version-script=libdriveio/src/libdriveio.vers \
-	-Xlinker -soname=libdriveio.so.0 -lc -lstdc++
+	-Xlinker -soname=libdriveio.so.0 $(LDFLAGS) -lc -lstdc++
 
 out/libmakemkv.so.1.full: tmp/gen_buildinfo.h
 	mkdir -p out
@@ -49,15 +51,15 @@ out/libmakemkv.so.1.full: tmp/gen_buildi
 	$(LIBABI_SRC) $(LIBABI_SRC_LINUX) $(LIBFFCODEC_SRC) \
 	-DHAVE_BUILDINFO_H -Itmp \
 	-fPIC -Xlinker -dy -Xlinker --version-script=libmakemkv/src/libmakemkv.vers \
-	-Xlinker -soname=libmakemkv.so.1 -lc -lstdc++ -lcrypto -lz
+	-Xlinker -soname=libmakemkv.so.1 $(LDFLAGS) -lc -lstdc++ -lcrypto -lz
 
 out/makemkv.full: $(MAKEMKVGUI_GEN) tmp/gen_buildinfo.h
 	mkdir -p out
 	$(GCC) $(CXXOPTS) -o$@ $(MAKEMKVGUI_INC) $(LIBMAKEMKV_INC) $(SSTRING_INC) $(LIBDRIVEIO_INC) \
 	$(MAKEMKVGUI_SRC) $(MAKEMKVGUI_SRC_LINUX) $(MAKEMKVGUI_GEN) $(SSTRING_SRC) $(LIBDRIVEIO_SRC_PUB) \
 	-DHAVE_BUILDINFO_H -Itmp \
-	-I/usr/include/qt4 -I/usr/lib/qt4/include -L/usr/lib/qt4 -L/usr/lib/qt4/lib -lc -lstdc++ \
-	-lQtGui -lQtCore -lQtDBus -lpthread -lz -lrt
+	$(QTINC) $(LDFLAGS) -lc -lstdc++ \
+	$(QTLIBS) -lpthread -lz -lrt
 
 tmp/gen_buildinfo.h:
 	mkdir -p tmp
