diff -up makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.cpp.noup makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.cpp
--- makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.cpp.noup	2011-10-27 20:34:50.000000000 -0200
+++ makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.cpp	2011-11-27 18:06:51.947420051 -0200
@@ -98,9 +98,6 @@ void CSettingDialog::ReadSettings()
     int show_debug = client->GetSettingInt(apset_app_ShowDebug);
     generalTab->check_DebugLog->setCheckState( (show_debug==0) ? Qt::Unchecked : Qt::Checked );
 
-    int site_access = client->GetSettingInt(apset_app_UpdateEnable);
-    generalTab->check_SiteAccess->setCheckState( (site_access==0) ? Qt::Unchecked : Qt::Checked );
-
     // language
     languageTab->setValue(languageTab->comboBoxInterfaceLanguage,client->GetSettingString(apset_app_InterfaceLanguage));
     languageTab->setValue(languageTab->comboBoxPreferredLanguage,client->GetSettingString(apset_app_PreferredLanguage));
@@ -175,7 +172,6 @@ bool CSettingDialog::WriteSettings(bool&
     // general
     client->SetSettingString(apset_app_DataDir,Utf16FromQString(generalTab->dataDir->text()));
     client->SetSettingInt( apset_app_ShowDebug , (generalTab->check_DebugLog->checkState() == Qt::Checked) ? 1 : 0 );
-    client->SetSettingInt( apset_app_UpdateEnable , (generalTab->check_SiteAccess->checkState() == Qt::Checked) ? 1 : 0 );
 
     // language
     client->SetSettingString(apset_app_InterfaceLanguage,languageTab->getValue(languageTab->comboBoxInterfaceLanguage,string));
@@ -389,9 +385,6 @@ CGeneralTab::CGeneralTab(IIconPath* icon
     check_DebugLog = new QCheckBox();
     m_lay->addWidget(createLabel(UI_QSTRING(APP_IFACE_SETTINGS_LOG_DEBUG_MSG)),0,0,Qt::AlignRight);
     m_lay->addWidget(check_DebugLog,0,1);
-    check_SiteAccess = new QCheckBox();
-    m_lay->addWidget(createLabel(UI_QSTRING(APP_IFACE_SETTINGS_ENABLE_SITE_ACCESS)),1,0,Qt::AlignRight);
-    m_lay->addWidget(check_SiteAccess,1,1);
     miscBox->setLayout(m_lay);
 
     QBoxLayout *lay = new QVBoxLayout();
diff -up makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.h.noup makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.h
--- makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.h.noup	2011-10-27 20:34:50.000000000 -0200
+++ makemkv_v1.6.16_oss/makemkvgui/src/settingdlg.h	2011-11-27 18:05:31.762478757 -0200
@@ -29,7 +29,6 @@ public:
     CDirSelectBox*   dataDir;
 
     QCheckBox*      check_DebugLog;
-    QCheckBox*      check_SiteAccess;
 
 public:
     CGeneralTab(IIconPath* iconPath,QWidget *parent = 0);
