--- kmyfirewall/genericinterface/kmfgenericinterfaceprotocol.cpp.orig	2012-10-16 22:35:03.749136118 -0300
+++ kmyfirewall/genericinterface/kmfgenericinterfaceprotocol.cpp	2012-10-16 22:35:28.103984079 -0300
@@ -312,7 +312,7 @@ void KMFGenericInterfaceProtocol::slotPr
 					m_zone,
 					i18n("Add protocol %1 to zone %2.").arg( protUsage->protocol()->name() ).arg( m_zone->guiName() )
 					);
-					KMFProtocolUsage * pro = m_zone->addProtocolUsage( protUsage->protocol()->uuid(), protUsage->protocol()->getDOMTree() );
+					pro = m_zone->addProtocolUsage( protUsage->protocol()->uuid(), protUsage->protocol()->getDOMTree() );
 			
 				if ( pro ) {
 					kdDebug() << "Added Protocol to zone: " << m_zone-> name()  << endl;
@@ -349,7 +349,7 @@ void KMFGenericInterfaceProtocol::slotPr
 					m_host,
 					i18n("Add protocol %1 to host %2.").arg( protUsage->protocol()->name() ).arg( m_host->guiName() )
 					);
-					KMFProtocolUsage * pro = m_host->addProtocolUsage( protUsage->protocol()->uuid(), protUsage->protocol() ->getDOMTree() );
+					pro = m_host->addProtocolUsage( protUsage->protocol()->uuid(), protUsage->protocol() ->getDOMTree() );
 			
 				if ( pro ) {
 					kdDebug() << "Added Protocol to host: " << m_host-> name()  << endl;
